---
title: Build figures
execute:
  eval: true
code-fold: true
---

# Building figures

Plot guidance for figures in cell here: [https://www.cell.com/figure-guidelines](https://www.cell.com/figure-guidelines)

Limit is 7 figures/tables, but more can go in supplement.

Width guidance:

- For 2-column formats (such as research articles and reviews), the sizes are 85 mm (1 column), 114 mm (1.5 columns), and 174 mm (full width of the page)

- TODO - look at the correlation of the MAGMA risk genes per celltype, see if there's unique co-expression. Then try by sample for a given celltype and see if there's variation (presumably by amyloid burden?). Maybe check braak stage.

# Load packages

```{r}
source(here::here("04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"))
library(magick)
library(circlize)
library(Cairo)
library(rstatix)  # For statistical tests
library(ggpubr)   # For adding significance brackets

tar_config_set(store = here::here("_targets"))
```


```{r}
#| eval: false
knitr::purl("04_data_analysis/998_paper_figures/04_figures.qmd",
"temp-dotplot.R")
```

```{r}
source("04_data_analysis/998_paper_figures/00_figures_source.R")
```

```{r}
#| eval: false
# Use the Ensembl BioMart database
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

# Retrieve GO terms
go_terms <- getBM(attributes = c('go_id', 'name_1006', 'namespace_1003'),
                  mart = ensembl)

# Filter GO terms for those containing "amyloid"
amyloid_go_terms <- go_terms[grepl("amyloid", go_terms$name_1006, ignore.case = TRUE), ]

readr::write_tsv(amyloid_go_terms, here("03_data/999_data_for_publication/amyloid_go_terms.tsv"))

tau_go_terms <- go_terms[grepl("tau", go_terms$name_1006, ignore.case = TRUE), ]

amyloid_go_terms <- readxl::read_excel(here("03_data/999_data_for_publication/amyloid_go_terms_ZC.xlsx")) |>
  janitor::clean_names() |>
  dplyr::select(!x4) |>
  dplyr::rename(amyloid_group = x5) |>
  dplyr::mutate(amyloid_group = if_else(amyloid_group == "?", "M", amyloid_group)) |>
  dplyr::filter(amyloid_group != "Not AD")

# Query the org.Hs.eg.db package for genes associated with the GO terms
all_genes <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = amyloid_go_terms$go_id,
  columns = c("ENSEMBL", "SYMBOL"),
  keytype = "GO"
)

amyloid_go_terms |>
  dplyr::left_join(all_genes, by = join_by(go_id == GO)) |>
  janitor::clean_names() |>
  readr::write_tsv("03_data/999_data_for_publication/amyloid_go_terms_genes.tsv")
```

# Figure 1 - method and demographic overview

```{r}
# Get seurat object
tar_load(sce)

# the full cell labels are long for plots, let's make abbreviated versions
celltypes_abbrev <-
  c(
    "Astrocyte-activated" = "Astro-A",
    "Astrocyte-quiescent" = "Astro-Q",
    "Ex-Neuron-ADARB2-ERBB4" = "Ex-Neuro-1",
    "Ex-Neuron-CBLN2-NRGN" = "Ex-Neuro-2",
    "Ex-Neuron-DLC1-HS3ST4" = "Ex-Neuro-3",
    "Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2" = "Ex-Neuro-4",
    "Ex-Neuron-L3-5-RORB-PLCH1" = "Ex-Neuro-5",
    "Ex-Neuron-L4-5-RORB-IL1RAPL2" = "Ex-Neuro-6",
    "Ex-Neuron-L6-THEMIS-NFIA" = "Ex-Neuro-7",
    "In-Neuron-ADARB2-GRM7" = "In-Neuro-1",
    "In-Neuron-ADARB2-IL1RAPL2-CSCL14" = "In-Neuro-2",
    "In-Neuron-ADARB2-LAMP5-NRG1" = "In-Neuro-3",
    "In-Neuron-ADARB2-NRG1-RELN" = "In-Neuro-4",
    "In-Neuron-DPP10-MEF2C" = "In-Neuro-5",
    "In-Neuron-PRKG1-TBC1D4" = "In-Neuro-6",
    "In-Neuron-TRHDE-RALYL" = "In-Neuro-7",
    "Microglia-activated" = "Microglia-A",
    "Microglia-quiescent" = "Microglia-Q",
    "Microglia-vascular" = "Microglia-V",
    "Pericyte" = "Pericyte-1",
    "Perivascular Macrophage" = "Perivascular-M",
    "Perivascular" = "Perivascular-M",
    "Perivascular-FB" = "Perivascular-FB-1",
    "Perivascular-FB-KAZN2" = "Perivascular-FB-2",
    "T-cell-parenchymal-1" = "T-cell-P1",
    "T-cell-parenchymal-2" = "T-cell-P2",
    "T-cell-vascular" = "T-cell-V",
    "T-cell-mixed" = "T-cell-M",
    "Vascular-SMC" = "Vascular-SMC-1",
    "Vascular-SMC-LINC00486" = "Vascular-SMC-2"
  )

# Create a new column with abbreviated Idents
abbreviated_idents <- celltypes_abbrev[as.character(sce$subcelltype_annotations)]
abbreviated_idents <- if_else(is.na(abbreviated_idents), sce$subcelltype_annotations, abbreviated_idents)
names(abbreviated_idents) <- NULL
sce$abbreviated_idents <- abbreviated_idents
unique(sce$abbreviated_idents)
Idents(sce) <- sce$abbreviated_idents

celltypes_abbrev_level1 <-
  c(
    "mystery-cluster" = "EndoMT",
    "Neuron_ex" = "Ex-Neuro",
    "Neuron_in" = "In-Neuro"
  )
abbreviated_idents <- celltypes_abbrev_level1[as.character(sce$highlevel_manual_annotations)]
abbreviated_idents <- if_else(is.na(abbreviated_idents), sce$highlevel_manual_annotations, abbreviated_idents)
names(abbreviated_idents) <- NULL
sce$abbreviated_idents_level1 <- abbreviated_idents
unique(sce$abbreviated_idents_level1)
```

```{r}
# Get percentages of level 1 celltypes
celltype_counts <- table(sce$highlevel_manual_annotations)
total <- sum(celltype_counts)
percentages <- (celltype_counts / total) * 100
print(percentages)

# Get the number of endothelial cells per sample for Yang paper
total_endothelial_cells <- 36800
total_neurons <- 2600
number_of_samples <- 25
total_endothelial_cells / number_of_samples
total_neurons / number_of_samples

# And for us - note there's one more parenchymal sample than vascular
v_samples <- grepl("V", unique(sce$manifest)) |> sum()
p_samples <- grepl("P", unique(sce$manifest)) |> sum()
celltype_counts[names(celltype_counts) == "Endothelial"] / v_samples
celltype_counts[names(celltype_counts) == "Endothelial"] / v_samples +
celltype_counts[names(celltype_counts) == "mystery-cluster"] / v_samples

celltype_counts[names(celltype_counts) == "Neuron_in"] / p_samples +
celltype_counts[names(celltype_counts) == "Neuron_ex"] / p_samples

# Get percentages of level 2 celltypes
sce@meta.data |>
  dplyr::filter(prep == "V") |>
  dplyr::count(abbreviated_idents) |>
  dplyr::mutate(percent = round(n / sum(n) * 100, digits = 2))
celltype_counts <- table(sce$abbreviated_idents)
total <- sum(celltype_counts)
percentages <- (celltype_counts / total) * 100
print(percentages |> round(digits = 2))
```


```{r}
DotPlot(sce, features = c("AGBL1",
                          "ARHGAP15",
                          "DOCK2",
                          "RUNX1"))
```

## Top Celltype markers

```{r}
#| eval: false
# Find the top markers for each celltype
df <- data.frame(gene = rownames(sce@assays$RNA@data),
                 ensembl = rownames(sce@assays$RNA@counts))
# Subset to cases or controls
rownames(sce@assays$RNA@data) <- df$ensembl
plan("multisession", workers = parallel::detectCores() - 1)
#markers <- FindAllMarkers(sce)
markers <- qs::qread("sce_current_markers.qs")

head(markers)

tar_load(translated_id)

markers <- markers |>
  dplyr::left_join(translated_id, by = join_by(gene == ensembl_gene_id))

# filter to first instance of all gene symbols per celltype
markers <- markers |>
  dplyr::filter(hgnc_symbol != "") |>
  group_by(cluster, hgnc_symbol) |>
  dplyr::slice(1) |>
  ungroup()

# Get the top 2-3 markers per celltype
top_markers <- markers |>
  group_by(cluster) |>
  slice_max(abs(avg_log2FC), n = 2, with_ties = FALSE) |>
  ungroup()

  # Calculate specificity score
  # High score = high expression in one group, low in others
top_markers <- markers |>
  dplyr::filter(p_val_adj < 0.01 & hgnc_symbol != "" & abs(avg_log2FC) > 1) |>
  group_by(cluster) |>
  mutate(specificity_score = avg_log2FC * (pct.1 / pct.2)) |>
  arrange(cluster, desc(specificity_score)) |>
  slice_head(n = 2) |>
  ungroup()

top_markers <- sce@meta.data |>
  dplyr::select(subcelltype_annotations, abbreviated_idents) |>
  unique() |>
  dplyr::left_join(x = top_markers, y = _, by = join_by(cluster == subcelltype_annotations))

v_celltype <- sce@meta.data |>
  dplyr::filter(prep == "V") |>
  dplyr::summarise(n = n(), .by = abbreviated_idents) |>
  dplyr::filter(n > 500 & abbreviated_idents != "Oligo-A")

top_markers_v <- top_markers |>
  dplyr::filter(abbreviated_idents %in% v_celltype$abbreviated_idents)

# Create the dotplot
ggplot(top_markers, aes(x = abbreviated_idents, y = hgnc_symbol, color = p_val_adj,
                        size = avg_log2FC)) +
  geom_point() +
  scale_color_viridis_c(direction = -1) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(x = "Celltype", y = "Marker Gene",
       color = "Adjusted p-value", size = "Avg Log2 FC")

sce_v <- subset(sce, abbreviated_idents %in% v_celltype$abbreviated_idents)
rownames(sce@assays$RNA@data) <- df$gene
rownames(sce_v@assays$RNA@data) <- df$gene
DotPlot(sce_v, features = unique(top_markers_v$hgnc_symbol)) +
  coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

### Celltype propotions

It'd be good to compare the number of nuclei we recover and the percentage per level 1 celltype

```{r}
# Add Yang 2022 numbers
nuclei_counts <- tribble(
  ~dataset, ~celltype,          ~n_nuclei,
  "Yang et al., 2022",   "Endothelial",      36825,
  "Mathys et al., 2019", "Endothelial",      150,
  "Yang et al., 2022",   "Pericyte",         29196,
  "Mathys et al., 2019", "Pericyte",         150,
  "Yang et al., 2022",   "SMC",              7312,
  "Mathys et al., 2019", "SMC",              0,
  "Yang et al., 2022",   "P. fibroblast",    2985,
  "Mathys et al., 2019", "P. fibroblast",    0,
  "Yang et al., 2022",   "Astrocyte",        22695,
  "Mathys et al., 2019", "Astrocyte",        3392,
  "Yang et al., 2022",   "Ependymal",        1534,
  "Mathys et al., 2019", "Ependymal",        0,
  "Yang et al., 2022",   "M. fibroblast",    428,
  "Mathys et al., 2019", "M. fibroblast",    0,
  "Yang et al., 2022",   "T-cell",           351,
  "Mathys et al., 2019", "T-cell",           0,
  "Mathys et al., 2019", "EndoMT",           0,
  "Yang et al., 2022",   "EndoMT",           0,
  "Yang et al., 2022",   "Oligo",            34800,
  "Yang et al., 2022",   "OPC",              3800,
  "Yang et al., 2022",   "Neuron",           2900,
  "Yang et al., 2022",   "Microglia",        2200,
)

# fibroblast subtypes in our data
fb_subtypes <- table(sce$subcelltype_annotations) |>
  as.data.frame() |>
  dplyr::mutate(dataset = "Method here", .before = Var1) |>
  dplyr::rename(celltype = Var1) |>
  dplyr::filter(grepl("FB", celltype))

# Merge p-fb subtypes
p_fb <- fb_subtypes |>
  dplyr::filter(!grepl("Meningeal", celltype)) |>
  dplyr::mutate(n_nuclei = sum(Freq)) |>
  dplyr::filter(celltype == "Perivascular-FB") |>
  dplyr::select(!Freq)

fb_subtypes_joined <- fb_subtypes |>
  dplyr::filter(celltype == "Meningeal-FB") |>
  dplyr::rename(n_nuclei = Freq) |>
  rbind(p_fb)


our_counts <- table(sce$highlevel_manual_annotations) |>
  as.data.frame() |>
  dplyr::mutate(dataset = "Method here", .before = Var1) |>
  dplyr::rename(celltype = Var1, n_nuclei = Freq) |>
  dplyr::filter(celltype != "Fibroblast") |>
  rbind(fb_subtypes_joined)

neuron_count <- our_counts |>
  dplyr::filter(grepl("Neuron", celltype)) |>
  dplyr::summarise(
    dataset = dataset[1],
    celltype = "Neuron",
    n_nuclei = sum(n_nuclei)
  )

yang_comparison <- our_counts |>
  dplyr::filter(!grepl("Neuron", celltype)) |>
  rbind(neuron_count) |>
  rbind(nuclei_counts) |>
  dplyr::mutate(celltype = case_when(
    celltype == "Astro" ~ "Astrocyte",
    celltype == "mystery-cluster" ~ "EndoMT",
    celltype == "Perivascular-FB" ~ "P. fibroblast",
    celltype == "Meningeal-FB" ~ "M. fibroblast",

    .default = celltype
  )) |>
  dplyr::mutate(total_nuclei = sum(n_nuclei), .by = dataset) |>
  dplyr::mutate(percent = round(n_nuclei / total_nuclei * 100, 1)) |>
  dplyr::mutate(fraction = if_else(celltype %in% c("Endothelial", "Pericyte", "SMC", "EndoMT", "M. fibroblast", "P. fibroblast", "Ependymal"), "Vascular", "Parenchymal"))

yang_comparison |>
  ggplot(aes(x = n_nuclei, y = celltype, fill = dataset)) +
  geom_bar(stat = "identity", position = position_dodge())


total_nuclei <- yang_comparison |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = n_nuclei, y = reorder(celltype, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity", position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(n_nuclei)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(panel.grid.minor = element_blank(), axis.title.y = element_blank(),
    legend.position = "none") +
  # Add labels
  labs(x = "# of nuclei", fill = "Dataset") +
  coord_cartesian(xlim = c(0, 120000)) +
  facet_grid(rows = vars(fraction), scales = "free")

percent_nuclei <- yang_comparison |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = percent, y = reorder(celltype, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity", position = position_dodge(width = 0.9),
           width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(percent)), position = position_dodge(width = 0.9),
            hjust = -0.2, size = 3) +
  # Customize theme
  theme_minimal() +
  theme(panel.grid.minor = element_blank(), axis.title.y = element_blank(),
    legend.position = "bottom") +
  # Add labels
  labs(x = "% of nuclei", fill = "Dataset") +
  facet_grid(rows = vars(fraction), scales = "free")

total_nuclei + percent_nuclei

ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison.png"
  ),
  height = 8,
  width = 10
)
```

```{r}
# Sum fractions
fraction_level <- yang_comparison |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  dplyr::summarise(n_nuclei = sum(n_nuclei), .by = c(dataset, fraction)) |>
  mutate(percent = round(n_nuclei / sum(n_nuclei) * 100, 1), .by = dataset)
total_nuclei <- fraction_level |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = n_nuclei, y = reorder(fraction, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity",
          position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(n_nuclei)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "none"
  ) +
  # Add labels
  labs(
    x = "# of nuclei",
    fill = "Dataset"
  ) +
  coord_cartesian(xlim = c(0, 250000))

percent_nuclei <- fraction_level |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = percent, y = reorder(fraction, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity",
          position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(percent)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "bottom"
  ) +
  # Add labels
  labs(
    x = "% of nuclei",
    fill = "Dataset"
  ) +
  coord_cartesian(xlim = c(0, 65))

total_nuclei + percent_nuclei

ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison_fraction.png"
  ),
  height = 8,
  width = 10
)
```

#### Vascular fraction comparison only

It might be more reasonable to just compare our vascular fraction to vascular papers

```{r}
meta <- sce@meta.data |>
  dplyr::filter(prep == "V")

our_vas_counts <- meta |>
  dplyr::count(highlevel_manual_annotations) |>
  dplyr::rename(celltype = highlevel_manual_annotations)

fb_counts <- meta |>
  dplyr::count(subcelltype_annotations) |>
  dplyr::rename(celltype = subcelltype_annotations) |>
  dplyr::filter(grepl("FB", celltype))

# Merge p-fb subtypes
p_fb <- fb_counts |>
  dplyr::filter(!grepl("Meningeal", celltype)) |>
  dplyr::mutate(n = sum(n)) |>
  dplyr::filter(celltype == "Perivascular-FB")

fb_subtypes_joined <- fb_counts |>
  dplyr::filter(celltype == "Meningeal-FB") |>
  rbind(p_fb)

our_vas_counts <- our_vas_counts |>
  dplyr::filter(celltype != "Fibroblast") |>
  rbind(fb_subtypes_joined)

neuron_count <- our_vas_counts |>
  dplyr::filter(grepl("Neuron", celltype)) |>
  dplyr::summarise(
    celltype = "Neuron",
    n = sum(n)
  )

yang_comparison_vas <- our_vas_counts |>
  dplyr::filter(!grepl("Neuron", celltype)) |>
  rbind(neuron_count) |>
  dplyr::mutate(dataset = "Method here", .before = celltype) |>
  dplyr::rename(n_nuclei = n) |>
  rbind(nuclei_counts) |>
  dplyr::mutate(celltype = case_when(
    celltype == "Astro" ~ "Astrocyte",
    celltype == "mystery-cluster" ~ "EndoMT",
    celltype == "Perivascular-FB" ~ "P. fibroblast",
    celltype == "Meningeal-FB" ~ "M. fibroblast",
    .default = celltype
  )) |>
  dplyr::mutate(total_nuclei = sum(n_nuclei), .by = dataset) |>
  dplyr::mutate(percent = round(n_nuclei / total_nuclei * 100, 1)) |>
  dplyr::mutate(fraction = if_else(celltype %in% c("Endothelial", "Pericyte", "SMC", "EndoMT", "M. fibroblast", "P. fibroblast", "Ependymal"), "Vascular", "Non-Vascular"))
```

#### Get Tsartsalis et al data

```{r}
# Download supplementary files for Tsartsalis et al 2024
#getGEOSuppFiles("GSM8010381")
# Read data
tsar <- readr::read_rds(here("03_data/991_external_data/vascular_snRNAseq_for_caleb-selected/vasc_ALL_new_celltype_for_heatmap.RDS"))
table(tsar$new_celltype)
tsar_sce <- qs::qread(here("03_data/991_external_data/vascular_snRNAseq_for_caleb-selected/GerritsSmith_seurat.qs"))
table(tsar_sce$cluster_celltype)
sum(tsar$barcode %in% tsar_sce$barcode)

tsar <- tsar@meta.data |>
  dplyr::select(barcode , new_celltype) |>
  dplyr::left_join(tsar_sce@meta.data, y = _, by = join_by(barcode)) |>
  dplyr::mutate(celltype = if_else(cluster_celltype == "Endo", new_celltype, cluster_celltype))
table(tsar$celltype)
```

```{r}
yang_comparison_vas <- tsar |>
  dplyr::count(celltype) |>
  dplyr::mutate(total_nuclei = sum(n)) |>
  dplyr::rename(n_nuclei = n) |>
  dplyr::mutate(percent = round(n_nuclei / total_nuclei * 100, 1)) |>
  dplyr::mutate(fraction = if_else(celltype %in% c("EC", "FB", "PC", "SMC"),
                                   "Vascular", "Non-Vascular"),
                dataset = "Tsartsalis et al., 2024") |>
  rbind(yang_comparison_vas)
```

```{r}
total_nuclei <- yang_comparison_vas |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = n_nuclei, y = reorder(celltype, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity",
          position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(n_nuclei)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "none"
  ) +
  # Add labels
  labs(
    x = "# of nuclei",
    fill = "Dataset"
  ) +
  coord_cartesian(xlim = c(0, 90000)) +
  facet_grid(rows = vars(fraction), scales = "free")

percent_nuclei <- yang_comparison_vas |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = percent, y = reorder(celltype, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity",
          position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(percent)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "none"
  ) +
  # Add labels
  labs(
    x = "% of nuclei",
    fill = "Dataset"
  ) +
  coord_cartesian(xlim = c(0, 55)) +
  facet_grid(rows = vars(fraction), scales = "free")

ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison_vas.png"
  ),
  height = 8,
  width = 10
)
```

```{r}
fraction_level <- yang_comparison_vas |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  #dplyr::filter(!(dataset == "Method here" & fraction == "Parenchymal")) |>
  dplyr::summarise(n_nuclei = sum(n_nuclei), .by = c(dataset, fraction)) |>
  mutate(percent = round(n_nuclei / sum(n_nuclei) * 100, 1), .by = dataset)

total_nuclei_frac_v <- fraction_level |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = reorder(dataset, percent * (fraction == "Non-Vascular")), y = n_nuclei, fill = fraction)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity") +
  # Add text labels at the end of bars
  geom_text(aes(label = n_nuclei),
            color = "black", size = 3, fontface = "bold",
            position = position_stack(vjust = 0.5)) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "bottom",
    axis.text.x = element_text(angle = 45, hjust = 1)
  ) +
  # Add labels
  labs(x = "# of nuclei", fill = "Dataset") +
  scale_y_continuous(labels = comma)
  #coord_cartesian(xlim = c(0, 240000))

#reorder(abbreviated_idents, -percentage * (prep == "V"))

percent_nuclei_frac_v <- fraction_level |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  ggplot(aes(x = reorder(dataset, percent * (fraction == "Non-Vascular")), y = percent, fill = fraction)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity") +
  # Add text labels at the end of bars
  #geom_text(aes(label = comma(percent)), hjust = -0.2, size = 3) +
  geom_text(aes(label = paste0(percent, "%")),
            color = "black", size = 3, fontface = "bold",
            position = position_stack(vjust = 0.5)) +
  # Customize theme
  theme_minimal() +
  theme(panel.grid.minor = element_blank(), axis.title.y = element_blank(),
    legend.position = "bottom",
    axis.text.x = element_text(angle = 45, hjust = 1)) +

  # Add labels
  labs(x = "% of nuclei", fill = "Dataset")
  #coord_cartesian(xlim = c(0, 120))

total_nuclei_frac_v | percent_nuclei_frac_v

ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison_vas_fraction.png"
  ),
  height = 5,
  width = 6
)
```

### Compare penchymal fraction

I'll compare the parenchymal by looking at Mathys et al., 2023

```{r}
# Make a dataframe of the numbers reported in firgure 1 of Mathys 2023
cell_type <- c("Ex. neuron", "In. neuron", "Oligo",
               "OPC", "Astrocyte", "Immune", "Vascular")
cell_counts <- c(1043230, 329699, 645142, 90502, 149558, 83889, 17974)
percentage <- c(44.2, 14.0, 27.5, 3.8, 6.3, 3.6, 0.8)

# Create the dataframe
brain_cells_df <- data.frame(
  dataset = "Mathys et al., 2023",
  celltype = cell_type,
  n_nuclei = cell_counts,
  percentage = percentage
)

p_counts <- sce@meta.data |>
  dplyr::filter(prep == "P") |>
  dplyr::count(highlevel_manual_annotations) |>
  dplyr::rename(celltype = highlevel_manual_annotations) |>
  dplyr::mutate(celltype = case_when(
    celltype %in% c("Endothelial", "Pericyte", "mystery-cluster",
                    "Fibroblast", "SMC") ~ "Vascular",
    celltype %in% c("Microglia", "T-cell") ~ "Immune",
    celltype == "Neuron_ex" ~ "Ex. neuron",
    celltype == "Neuron_in" ~ "In. neuron",
    celltype == "Astro" ~ "Astrocyte",
    .default = celltype)) |>
  dplyr::mutate(n_nuclei = sum(n), .by = celltype) |>
  dplyr::select(!n) |>
  dplyr::distinct() |>
  dplyr::mutate(n_total = sum(n_nuclei)) |>
  dplyr::mutate(percentage = round(n_nuclei / n_total * 100, 1)) |>
  dplyr::mutate(dataset = "Method here", .before = celltype) |>
  dplyr::select(!n_total) |>
  rbind(brain_cells_df)
```

```{r}
total_nuclei_p <- p_counts |>
  ggplot(aes(x = n_nuclei, y = reorder(celltype, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity",
          position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(n_nuclei)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "bottom"
  ) +
  # Add labels
  labs(
    x = "# of nuclei",
    fill = "Dataset"
  ) +
  coord_cartesian(c(0, 1350000)) +
  scale_x_continuous(labels = comma)

percent_nuclei_p <- p_counts |>
  ggplot(aes(x = percentage, y = reorder(celltype, n_nuclei), fill = dataset)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity",
          position = position_dodge(width = 0.9),
          width = 0.8) +
  # Add text labels at the end of bars
  geom_text(aes(label = comma(percentage)),
            position = position_dodge(width = 0.9),
            hjust = -0.2,  # adjust this to move labels further/closer to bars
            size = 3) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "bottom"
  ) +
  # Add labels
  labs(
    x = "% of nuclei",
    fill = "Dataset"
  ) +
  coord_cartesian(c(0, 50))

total_nuclei_p + percent_nuclei_p
ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison_par.png"
  ),
  height = 5,
  width = 7
)
```

```{r}
fraction_level <- p_counts |>
  dplyr::mutate(fraction = if_else(celltype == "Vascular", "Vascular", "Parenchymal")) |>
  dplyr::mutate(fraction = fct_relevel(fraction, "Vascular")) |>
  dplyr::summarise(n_nuclei = sum(n_nuclei), .by = c(dataset, fraction)) |>
  mutate(percent = round(n_nuclei / sum(n_nuclei) * 100, 1), .by = dataset)

# Get the Dark2 colors to make the green shade the parenchymal one
dark2_colors <- brewer.pal(n = 8, name = "Dark2")
reordered_colors <- dark2_colors[c(2, 1, 3, 4, 5, 6, 7, 8)]

total_nuclei_frac_p <- fraction_level |>
  ggplot(aes(x = reorder(dataset, percent * (fraction == "Vascular")),
             y = n_nuclei, fill = fraction)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity") +
  # Add text labels at the end of bars
  geom_text(aes(label = n_nuclei),
            color = "black", size = 3, fontface = "bold",
            position = position_stack(vjust = 0.5)) +
  # Customize theme
  theme_minimal() +
  theme(
    panel.grid.minor = element_blank(),
    axis.title.y = element_blank(),
    legend.position = "bottom",
    axis.text.x = element_text(angle = 45, hjust = 1)
  ) +
  # Add labels
  labs(x = "# of nuclei", fill = "Dataset") +
  scale_y_continuous(labels = comma) +
  scale_fill_manual(values = reordered_colors)
  #coord_cartesian(xlim = c(0, 240000))

#reorder(abbreviated_idents, -percentage * (prep == "V"))

percent_nuclei_frac_p <- fraction_level |>
  ggplot(aes(x = reorder(dataset, percent * (fraction == "Vascular")), y = percent, fill = fraction)) +
  # Add bars with consistent thickness
  geom_bar(stat = "identity") +
  # Add text labels at the end of bars
  #geom_text(aes(label = comma(percent)), hjust = -0.2, size = 3) +
  geom_text(aes(label = paste0(percent, "%")),
            color = "black", size = 2, fontface = "bold",
            position = position_stack(vjust = 0.5)) +
  # Customize theme
  theme_minimal() +
  theme(panel.grid.minor = element_blank(), axis.title.y = element_blank(),
    legend.position = "none",
    axis.text.x = element_text(angle = 90, hjust = 1, size = 7),
    axis.text.y = element_text(size = 7)) +

  # Add labels
  labs(x = "", fill = "Dataset") +
  scale_fill_manual(values = reordered_colors)
  #coord_cartesian(xlim = c(0, 120))

total_nuclei_frac_p | percent_nuclei_frac_p

ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison_par_fraction.png"
  ),
  height = 5,
  width = 6
)
```

### Do stats on level 1 celltypes

Want to show vascular celltypes are significantly more enriched and parenchyma is on par

```{r}
n_samples <- tribble(
  ~n_sample, ~dataset,
  77,        "Tsartsalis et al., 2024",
  25,        "Yang et al., 2022",
  427,       "Mathys et al., 2019",
  36,        "Method here"
)

celltype_sig_prop <- yang_comparison_vas |>
  #dplyr::filter(fraction == "Vascular") |>
  dplyr::mutate(celltype = case_when(
    celltype == "EC" ~ "Endothelial",
    grepl("fibroblast", celltype) ~ "Fibroblast",
    celltype == "FB" ~ "Fibroblast",
    celltype == "PC" ~ "Pericyte",
    celltype == "Astrocyte" ~ "Astro",
    celltype == "Micro" ~ "Microglia",
    .default = celltype
  )) |>
  dplyr::mutate(n_nuclei = sum(n_nuclei), percent = sum(percent),
                .by = c(dataset, celltype)) |>
  unique() |>
  dplyr::left_join(n_samples, by = join_by(dataset))

celltype_sig_prop

# need the per sample counts for our data
sample_counts <- sce@meta.data |>
  dplyr::filter(prep == "V") |>
  dplyr::summarise(n_nuclei = n(), .by = c(manifest, abbreviated_idents_level1))
unique(sample_counts$manifest) |> length()
unique(sample_counts$abbreviated_idents_level1)
sample_counts_p <- sce@meta.data |>
  dplyr::filter(prep == "P") |>
  dplyr::mutate(celltypes_updated = if_else(grepl("In-Neuro|Ex-Neuro",
                                                  abbreviated_idents_level1),
                                            "Neuron",abbreviated_idents_level1)) |>
  dplyr::summarise(n_nuclei = n(), .by = c(manifest, celltypes_updated))
```

```{r}
fraction_level <- yang_comparison_vas |>
  dplyr::filter(dataset != "Mathys et al., 2019") |>
  #dplyr::filter(!(dataset == "Method here" & fraction == "Parenchymal")) |>
  dplyr::summarise(n_nuclei = sum(n_nuclei), .by = c(dataset, fraction)) |>
  mutate(percent = round(n_nuclei / sum(n_nuclei) * 100, 1), .by = dataset) |>
  dplyr::left_join(n_samples, by = "dataset")

vas_counts_per_sample <- sce@meta.data |>
  dplyr::filter(prep == "V") |>
  dplyr::count(highlevel_manual_annotations, manifest) |>
  dplyr::rename(celltype = highlevel_manual_annotations) |>
  dplyr::mutate(fraction = if_else(celltype %in% c("Endothelial", "Pericyte",
                                                   "SMC", "mystery-cluster",
                                                   "Fibroblast", "Ependymal"),
                                   "Vascular", "Non-Vascular")) |>
  dplyr::summarise(n_nuclei = sum(n), .by = c(manifest, fraction))

par_counts_per_sample <- sce@meta.data |>
  dplyr::filter(prep == "P") |>
  dplyr::count(highlevel_manual_annotations, manifest) |>
  dplyr::rename(celltype = highlevel_manual_annotations) |>
  dplyr::mutate(fraction = if_else(celltype %in% c("Endothelial", "Pericyte",
                                                   "SMC", "mystery-cluster",
                                                   "Fibroblast", "Ependymal"),
                                   "Vascular", "Parenchymal")) |>
  dplyr::summarise(n_nuclei = sum(n), .by = c(manifest, fraction))


plot_enrichment_comparison <- function(total_data, sample_data, contrast = "Non-Vascular") {

  total_data <- total_data |>
    dplyr::arrange(dataset)
  # Get your method name
  your_method <- "Method here"

  # Calculate proportions for your samples
  your_proportions <- sample_data %>%
    dplyr::summarise(
      total_nuclei = sum(n_nuclei),
      vascular_nuclei = sum(n_nuclei[fraction == "Vascular"]),
      vascular_prop = vascular_nuclei / total_nuclei * 100,
      .by = manifest
    )

  # Get reference proportions from other studies
  reference_props <- total_data %>%
    dplyr::filter(dataset != your_method,
           fraction == "Vascular") %>%
    dplyr::select(dataset, percent)

  # Perform statistical tests
  stats_results <- reference_props %>%
    rowwise() %>%
    mutate(
      p_value = wilcox.test(your_proportions$vascular_prop,
                           mu = percent)$p.value,
      stars = case_when(
        p_value < 0.001 ~ "***",
        p_value < 0.01 ~ "**",
        p_value < 0.05 ~ "*",
        TRUE ~ "ns"
      )
    ) |>
    # Make sure order is in descending percent
    dplyr::arrange(desc(percent))

  # Calculate y positions for brackets - can adjust these numbers
  if(nrow(reference_props) != 1) {
    stats_results$y_bracket <- seq(from = 103, to = 105 + 5 * (nrow(reference_props) - 1), by = 5)
  } else {
    stats_results$y_bracket <- 103
  }
  stats_results$y_text <- stats_results$y_bracket + 0.5

  # Create base plot
  p <- ggplot(total_data, aes(x = reorder(dataset, percent * (fraction == contrast)), y = percent, fill = fraction)) +
    geom_bar(stat = "identity") +
    geom_text(aes(label = sprintf("%.1f%%", percent)),
            color = "black", size = 2, fontface = "bold",
            position = position_stack(vjust = 0.5)) +
    # scale_fill_manual(values = c("Non-Vascular" = "#E69F00",
    #                             "Vascular" = "#0072B2")) +
    theme_minimal() +
    theme(
      axis.text.x = element_text(angle = 90, hjust = 1, size = 7),
      axis.text.y = element_text(size = 7),
      legend.title = element_blank(),
      legend.text = element_text(size = 7),
      legend.position = "top",
      panel.grid.major.x = element_blank(),
      axis.title = element_text(size = 8),
      plot.title = element_text(hjust = 0.5)
    ) +
    labs(
      x = "",
      y = "Percentage of Nuclei (%)"
    )

  # Add brackets for each comparison
  for(i in 1:nrow(stats_results)) {
    ref_study <- stats_results$dataset[i]
    x1 <- which(unique(total_data$dataset) == your_method)
    x2 <- which(unique(total_data$dataset) == ref_study)
    y_pos <- stats_results$y_bracket[i]
    y_text <- stats_results$y_text[i]

    p <- p +
      # Horizontal line
      annotate("segment", x = x1, xend = x2,
               y = y_pos, yend = y_pos, color = "black") +
      # Vertical lines
      annotate("segment", x = x1, xend = x1,
               y = y_pos - 2, yend = y_pos, color = "black") +
      annotate("segment", x = x2, xend = x2,
               y = y_pos - 2, yend = y_pos, color = "black") +
      # Significance stars
      annotate("text", x = mean(c(x1, x2)), y = y_text,
               label = stats_results$stars[i], size = 4) +
      scale_y_continuous(breaks = seq(0, 100, by = 25))
  }

  return(p)
}

percent_nuclei_frac_v <- plot_enrichment_comparison(fraction_level,
                                                    vas_counts_per_sample)
print(percent_nuclei_frac_v)

total_nuclei_frac_v | percent_nuclei_frac_v

ggsave(
  here(
    "05_figures/990_shared_figures/003_final_figures/nuclei_count_comparison_vas_fraction.png"
  ),
  height = 5,
  width = 6
)

head(sample_counts)
x <- sample_counts |>
  dplyr::pull(n_nuclei, abbreviated_idents_level1)
sample_counts_list <- split(x, names(x))

x <- sample_counts_p |>
  dplyr::pull(n_nuclei, celltypes_updated)
sample_counts_list <- split(x, names(x))
```

```{r}
#| eval: false
# Function to compare nuclei counts using either chi-square or t-test
compare_nuclei_counts <- function(celltype, literature_total, literature_n,
                                  literature,
                                test_type = c("chi-square", "t-test")) {
  test_type <- match.arg(test_type)

  our_samples <- unlist(sample_counts_list[celltype])

  if (test_type == "t-test") {
    # Convert literature total to mean for t-test
    literature_mean <- literature_total / literature_n

    # Perform one-sample t-test
    t_result <- t.test(our_samples, mu = literature_mean)

    # Calculate effect size (Cohen's d)
    cohens_d <- (mean(our_samples) - literature_mean) / sd(our_samples)

    # Calculate percentage difference
    percent_diff <- ((mean(our_samples) - literature_mean) / literature_mean) * 100

    summary_stats <- tibble(
      celltype = celltype,
      dataset = literature,
      your_mean = mean(our_samples),
      your_sd = sd(our_samples),
      your_n = length(our_samples),
      your_total = sum(our_samples),
      literature_mean = literature_mean,
      literature_total = literature_total,
      literature_n = literature_n,
      percent_difference = percent_diff,
      cohens_d = cohens_d,
      test_statistic = t_result$statistic,
      p_value = t_result$p.value,
      confidence_interval = t_result$conf.int,
      test_type = "t-test"
    )

  } else {  # chi-square test
    # Calculate expected counts based on literature proportions
    your_total <- sum(our_samples)
    total_samples <- your_total + literature_total

    observed <- c(your_total, literature_total)
    expected <- c(
      (your_total + literature_total) * (length(our_samples) / (length(our_samples) + literature_n)),
      (your_total + literature_total) * (literature_n / (length(our_samples) + literature_n))
    )

    # Perform chi-square test
    chi_result <- chisq.test(observed, p = expected/sum(expected))

    # Calculate effect size (Cramer's V)
    cramers_v <- sqrt(chi_result$statistic / sum(observed))

    summary_stats <- tibble(
      celltype = celltype,
      dataset = literature,
      your_mean = mean(our_samples),
      your_sd = sd(our_samples),
      your_n = length(our_samples),
      your_total = sum(our_samples),
      literature_total = literature_total,
      literature_n = literature_n,
      test_statistic = chi_result$statistic,
      p_value = chi_result$p.value,
      effect_size = cramers_v,
      test_type = "chi-square"
    )
  }

  return(summary_stats)
}

celltype_sig_prop <- celltype_sig_prop |>
  as_tibble() |>
  dplyr::filter(dataset != "Method here" & celltype != "Ependymal")

celltype_sig_prop_p <- celltype_sig_prop |>
  dplyr::filter(fraction == "Non-Vascular")
celltype_sig_prop <- celltype_sig_prop |>
  dplyr::filter(fraction == "Vascular")

args <- list(celltype = celltype_sig_prop$celltype,
             literature_total = celltype_sig_prop$n_nuclei,
             literature_n = celltype_sig_prop$n_sample,
             literature = celltype_sig_prop$dataset)

stats <- pmap_dfr(args, compare_nuclei_counts) |>
  dplyr::mutate(pval_adj = p.adjust(p_value, method = "bonferroni"))

stats |>
  dplyr::filter(pval_adj > 0.05)

args <- list(celltype = celltype_sig_prop_p$celltype,
             literature_total = celltype_sig_prop_p$n_nuclei,
             literature_n = celltype_sig_prop_p$n_sample,
             literature = celltype_sig_prop_p$dataset)

stats_p <- pmap_dfr(args, compare_nuclei_counts) |>
  dplyr::mutate(pval_adj = p.adjust(p_value, method = "bonferroni"))

stats_p |>
  dplyr::filter(pval_adj > 0.05)
# Example usage:
# your_data <- c(150, 160, 145, 155, 165)  # Your sample counts
# lit_total <- 1400                         # Literature total count
# lit_n <- 10                               # Number of samples in literature
#
# # For t-test
# results_t <- compare_nuclei_counts(your_data, lit_total, lit_n, "t-test")
#
# # For chi-square test
# results_chi <- compare_nuclei_counts(your_data, lit_total, lit_n, "chi-square")
#
# # Print results nicely
# print_nuclei_results <- function(results) {
#   cat("Statistical Comparison Results:\n")
#   cat("Your samples (n=", results$your_n, "):\n", sep="")
#   cat("  Total count: ", results$your_total, "\n", sep="")
#   cat("  Mean ± SD: ", round(results$your_mean, 2), " ± ", round(results$your_sd, 2), "\n", sep="")
#   cat("Literature values (n=", results$literature_n, "):\n", sep="")
#   cat("  Total count: ", results$literature_total, "\n", sep="")
#
#   if (results$test_type == "t-test") {
#     cat("  Mean: ", round(results$literature_mean, 2), "\n", sep="")
#     cat("Percent difference: ", round(results$percent_difference, 2), "%\n", sep="")
#     cat("Effect size (Cohen's d): ", round(results$cohens_d, 2), "\n", sep="")
#     cat("95% CI: [", round(results$confidence_interval[1], 2), ", ",
#         round(results$confidence_interval[2], 2), "]\n", sep="")
#   } else {
#     cat("Effect size (Cramer's V): ", round(results$effect_size, 3), "\n", sep="")
#   }
#
#   cat("Test type: ", results$test_type, "\n", sep="")
#   cat("P-value: ", format.pvalue(results$p_value, digits=3), "\n", sep="")
# }
```

```{r}
library(ggstatsplot)
library(tidyr)
library(dplyr)

# Function to create a statistical comparison plot
plot_nuclei_comparison <- function(celltype, your_samples, literature_data) {
  your_samples <- unlist(your_samples[celltype])

  # Convert your samples to a data frame
  your_data <- data.frame(
    count = your_samples,
    source = "Your Data"
  )

  # Create literature data frame with one row per sample
  # (using mean counts for each study since we only have totals)
  lit_data <- literature_data |>
    dplyr::mutate(mean_count = total_nuclei / n_sample) |>
    #dplyr::group_by(dataset) |>
    dplyr::summarise(
      count = rep(mean_count, n_sample[1]),
      source = dataset,
      .by = dataset
    ) #|>
    #dplyr::ungroup()

  # Combine the data
  plot_data <- bind_rows(your_data, lit_data)

  # Create the plot
  p <- ggbetweenstats(
    data = plot_data,
    x = source,
    y = count,
    type = "nonparametric", # Using nonparametric test due to potential non-normality
    plot.type = "box",
    pairwise.comparisons = TRUE,
    pairwise.display = "significant",
    centrality.plotting = TRUE,
    bf.message = FALSE,
    title = "Comparison of Nuclei Counts Across Studies",
    xlab = "Data Source",
    ylab = "Nuclei Count"
  ) +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1),
      plot.title = element_text(hjust = 0.5)
    )

  return(p)
}

# Alternative version using percentages if you're comparing proportions
plot_nuclei_proportions <- function(your_samples, literature_data, total_nuclei) {
  # Convert counts to percentages
  your_data <- data.frame(
    percentage = (your_samples / total_nuclei) * 100,
    source = "Your Data"
  )

  # Create literature percentage data
  lit_data <- literature_data |>
    mutate(
      percentage = (total_count / n_samples / total_nuclei) * 100,
      source = study_name
    ) |>
    group_by(study_name) |>
    summarize(
      percentage = rep(percentage, n_samples[1]),
      source = source[1]
    ) |>
    ungroup()

  # Combine the data
  plot_data <- bind_rows(your_data, lit_data)

  # Create the plot
  p <- ggbetweenstats(
    data = plot_data,
    x = source,
    y = percentage,
    type = "nonparametric",
    plot.type = "box",
    pairwise.comparisons = TRUE,
    pairwise.display = "significant",
    centrality.plotting = TRUE,
    bf.message = FALSE,
    title = "Comparison of Nuclei Proportions Across Studies",
    xlab = "Data Source",
    ylab = "Percentage of Total Nuclei (%)"
  ) +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1),
      plot.title = element_text(hjust = 0.5)
    )

  return(p)
}

# Example usage:
# your_data <- c(150, 160, 145, 155, 165)
#
# literature_data <- data.frame(
#   study_name = c("Smith et al", "Jones et al", "Zhang et al"),
#   total_count = c(1400, 1600, 1300),
#   n_samples = c(10, 12, 8)
# )
#
# # For raw counts
p1 <- map(names(sample_counts_list), ~ plot_nuclei_comparison(.x, sample_counts_list, celltype_sig_prop))
print(p1)
#
# # For proportions (if comparing percentages)
# total_nuclei <- 1000 # Replace with your total nuclei count
# p2 <- plot_nuclei_proportions(your_data, literature_data, total_nuclei)
# print(p2)
```


### Endo-MT markers from paper

Let's check the markers for Endo-MT reported in [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10073145/)

```{r}
endomt_markers <- c("ACTA2", "FN1", "PECAM1", "LUM", "RAMP2", "IFI27", "TM4SF1", "GNG11", "TIMP3", "IGFBP7", "ID3", "CTGF")
DotPlot(sce, features = endomt_markers)
```

```{r}
#| fig-width: 7
#| fig-height: 12
VlnPlot(sce, features = c("pc_mito"), ncol = 1, group.by = "highlevel_manual_annotations", pt.size = 0)
```

# UMAPs

```{r}
#| label: make umap
Idents(sce) <- str_replace_all(Idents(sce), "_", "-")
sce$highlevel_manual_annotations <- if_else(sce$highlevel_manual_annotations == "mystery-cluster", "EndoMT", sce$highlevel_manual_annotations)
sce$highlevel_manual_annotations <- str_replace_all(sce$highlevel_manual_annotations, "_", "-")
unique(sce$highlevel_manual_annotations)

# Define strong colors for high-level cell types
level1_colors <- c(
  "Pericyte" = "#E41A1C",    # Strong Red
  "SMC" = "#377EB8",         # Strong Blue
  "EndoMT" = "#4DAF4A",      # Strong Green
  "Oligo" = "#984EA3",       # Strong Purple
  "Fibroblast" = "#FF7F00",  # Strong Orange
  "Endothelial" = "#A65628", # Strong Brown
  "T-cell" = "#F781BF",      # Strong Pink
  "Microglia" = "#666666",   # Dark Gray (for better contrast)
  "Astro" = "#FFD700",       # Strong Yellow-Gold
  "OPC" = "#1E90FF",         # Vivid Blue
  "Neuron-ex" = "#1F78B4",   # Deep Blue
  "Neuron-in" = "#33A02C"    # Deep Green
)

level2_colors <- c(
  "M-Pericyte" = "#FF6666",   # Lighter red
  "Vascular-SMC-1" = "#66B2FF",  # Lighter blue
  "EndoMT-2" = "#66FF66",   # Lighter green
  "Oligo-A" = "#C266FF",    # Lighter purple
  "Meningeal-FB" = "#FFA07A", # Soft coral
  "Capillary" = "#FFD700",   # Golden yellow
  "T-Pericyte" = "#FF4C4C",  # Stronger red
  "Arterial" = "#FF4500",    # Deep orange-red
  "Pericyte-1" = "#FF8080",  # Soft pink-red
  "Arteriolar-SMC" = "#4682B4", # Steel blue
  "T-cell-P1" = "#FA8072",   # Salmon
  "Pericyte-FB" = "#FF6347", # Tomato
  "Pericyte-2" = "#FF9999",  # Pinkish red
  "Artirial-FB" = "#FFB6C1", # Light pink
  "Venous" = "#8B0000",      # Deep red
  "Perivascular-M" = "#2F4F4F", # Dark slate gray
  "EndoMT-1" = "#32CD32",    # Lime green
  "Microglia-Q" = "#CCCCFF", # Light lavender
  "Vascular-SMC-2" = "#4682B4", # Blue
  "Perivascular-FB-2" = "#B0C4DE", # Light steel blue
  "Astro-Q" = "#FFD700",     # Gold
  "OPC-B" = "#ADD8E6",       # Light blue
  "Perivascular-FB-1" = "#87CEFA", # Sky blue
  "T-cell-M" = "#DC143C",    # Crimson
  "Oligo-B" = "#DDA0DD",     # Plum
  "Microglia-V" = "#FF69B4", # Hot pink
  "In-Neuro-7" = "#8A2BE2",  # Blue violet
  "Ex-Neuro-4" = "#4169E1",  # Royal blue
  "In-Neuro-1" = "#20B2AA",  # Light sea green
  "T-cell-V" = "#E9967A",    # Dark salmon
  "Astro-A" = "#DAA520",     # Goldenrod
  "In-Neuro-4" = "#556B2F",  # Dark olive green
  "Ex-Neuro-6" = "#6A5ACD",  # Slate blue
  "In-Neuro-6" = "#008080",  # Teal
  "In-Neuro-3" = "#ADFF2F",  # Green yellow
  "Microglia-A" = "#FF1493", # Deep pink
  "In-Neuro-2" = "#32CD32",  # Lime green
  "Ex-Neuro-7" = "#9370DB",  # Medium purple
  "Ex-Neuro-5" = "#BA55D3",  # Medium orchid
  "Ex-Neuro-3" = "#8B008B",  # Dark magenta
  "OPC-A" = "#9400D3",       # Dark violet
  "In-Neuro-5" = "#3CB371",  # Medium sea green
  "Ex-Neuro-1" = "#00FA9A",  # Medium spring green
  "Ex-Neuro-2" = "#FF4500",  # Orange red
  "T-cell-P2" = "#FF8C00"    # Dark orange
)

# Plot Level 1 cell types
umap_level1 <- DimPlot(sce, reduction = "umap",
                     group.by = "highlevel_manual_annotations",
                     label.size = 3, cols = level1_colors, repel = TRUE,
                     label = TRUE, pt.size = 1) +
  #scale_color_manual(values = level1_colors) +
  NoAxes() +
  NoGrid() +
  NoLegend() +
  theme(
    plot.margin = unit(c(1,1,1,1), "cm"),
  ) +
  labs(title = NULL)

# Plot Subtypes
umap <- DimPlot(sce, reduction = "umap", repel = TRUE, label = TRUE,
                cols = level2_colors, raster = TRUE, pt.size = 1, label.size = 3) +
  NoLegend() +
  NoAxes() +
  NoGrid() #+
  #ggrastr::rasterise(dpi = 300) # Rasterizes only points

umap_vascular_parenchymal <- DimPlot(sce, reduction = "umap", group.by = "prep",
                                     raster = TRUE, pt.size = 1) +
  NoAxes() +
  NoGrid() +
  scale_colour_npg() +
  #scale_fill_custom() +
  labs(title = NULL)

ggsave(here("05_figures/990_shared_figures/003_final_figures/overall_umap.png"), plot = umap, width = 10, height = 8, dpi = 300)
ggsave(here("05_figures/990_shared_figures/003_final_figures/umap_fractions.png"), plot = umap_vascular_parenchymal, width = 10, height = 8, dpi = 300)
```
```{r}
#| label: make umap old
#| eval: false

umap <- DimPlot(sce, reduction = "umap", repel = TRUE, label = TRUE,
                raster = FALSE, pt.size = 0.1) +
  NoLegend() +
  NoAxes() +
  NoGrid() +
  scale_color_custom() +
  scale_fill_custom()

sce$highlevel_manual_annotations <- if_else(sce$highlevel_manual_annotations == "mystery-cluster", "EndoMT", sce$highlevel_manual_annotations)

umap_level1 <- DimPlot(sce,
                     reduction = "umap",
                     group.by = "highlevel_manual_annotations",
                     label = TRUE,
                     label.size = 5,
                     repel = TRUE,
                     pt.size = 0.1,
                     raster = FALSE,
                     label.box = FALSE) +    # Add boxes around labels
  NoAxes() +
  NoGrid() +
  NoLegend() +
  scale_color_custom() +
  scale_fill_custom() +
  theme(
    legend.text = element_text(size = 12),
    legend.title = element_blank(),
    plot.margin = unit(c(1,1,1,1), "cm")
  ) +
  labs(title = NULL)

# Save the UMAP plot
ggsave(here("05_figures/990_shared_figures/003_final_figures/overall_umap.png"), plot = umap, width = 10, height = 8, dpi = 300)
ggsave(here("05_figures/990_shared_figures/003_final_figures/umap_fractions.png"), plot = umap_vascular_parenchymal, width = 10, height = 8, dpi = 300)

umap_donor <- DimPlot(sce, reduction = "umap", group.by = "donor",
                      raster = FALSE, pt.size = 0.1, alpha = 0.5) +
  NoAxes() +
  NoGrid() +
  scale_color_custom() +
  scale_fill_custom() +
  #scale_fill_custom() +
  labs(title = NULL)

ggsave(filename = here::here("05_figures/990_shared_figures/003_final_figures",
                             "000_sup_figures/umap_by_donor.png"),
       plot = umap_donor,
       width = 8, height = 8, units = "in", dpi = 300)
```

### Dotplot of top markers

```{r}
sce$abbreviated_idents <- str_replace_all(sce$abbreviated_idents, "_", "-")
assertthat::assert_that(sum(!(
  unique(Idents(sce)) %in% unique(sce$abbreviated_idents)
)) == 0)
file <- here::here("05_figures/990_shared_figures/003_final_figures",
                   "001_data_for_plots/top_abbrev_ident_markers.tsv")

# Find markers for each cell type
if(!file.exists(file)) {

  plan("multisession", workers = 8)
  markers <- FindAllMarkers(sce,
                            only.pos = TRUE,  # Only positive markers
                            min.pct = 0.25,   # Gene detected in at least 25% of cells
                            logfc.threshold = 0.5)  # Minimum log fold change
  readr::write_tsv(markers, file)
} else {
  markers <- readr::read_tsv(file)
}

# Select top markers for each cell type
top_markers <- markers %>%
  group_by(cluster) %>%
  slice_max(n = 10, order_by = avg_log2FC)

# Optional: Create a heatmap of top markers
DoHeatmap(sce,
          features = top_markers$gene,
          group.by = "highlevel_manual_annotations")
```

## Bar plot of cell proportions

```{r}
# Get metadata
meta_data <- sce@meta.data

readr::write_tsv(meta_data, here::here("03_data/999_data_for_publication/nuclei_meta_data.tsv"))

meta_data |>
  dplyr::select(manifest, age, sex, diagnosis) |>
  unique() |>
  dplyr::summarise(people = n(), median_age = median(age), iqr_age = IQR(age),
                   .by = c(diagnosis, sex))

# Calculate the percentage of cells in each fraction per cell type
percentages <- meta_data %>%
  group_by(abbreviated_idents, prep) %>%
  summarise(count = n()) %>%
  mutate(total = sum(count), percentage = (count / total) * 100) %>%
  ungroup()

# View the calculated percentages
head(percentages)

# Create the bar plot
bar_plot <- ggplot(percentages, aes(x = reorder(abbreviated_idents, -percentage * (prep == "V")), y = percentage, fill = prep)) +
  geom_bar(stat = "identity") +
 # geom_bar(stat = "identity", position = "dodge") +
  labs(x = "",
       y = "Percentage",
       fill = "Fraction") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
#  scale_fill_custom()

# Display the plot
print(bar_plot)

# Save the plot
ggsave(here("05_figures/990_shared_figures/003_final_figures/bar_plot_P_vs_V.png"), plot = bar_plot, width = 6, height = 4, dpi = 300)
ggsave(here("05_figures/990_shared_figures/003_final_figures/bar_plot_P_vs_V.svg"), plot = bar_plot, width = 6, height = 4)
```

```{r}
#| eval: false
# Calculate the percentage of cells in each fraction per cell type and higher-level cell type
percentages <- meta_data %>%
  group_by(highlevel_manual_annotations, abbreviated_idents, prep) %>%
  summarise(count = n()) %>%
  mutate(total = sum(count), percentage = (count / total) * 100) %>%
  ungroup()

# View the calculated percentages
head(percentages)

# Calculate the overall percentage of V cells for each higher-level cell type
overall_v_percentages <- percentages %>%
  filter(prep == "V") %>%
  group_by(highlevel_manual_annotations) %>%
  summarise(overall_v_percentage = sum(percentage)) %>%
  arrange(desc(overall_v_percentage))

# Reorder the higher-level cell types based on the overall V percentage
percentages$highlevel_manual_annotations <- factor(percentages$highlevel_manual_annotations, levels = overall_v_percentages$highlevel_manual_annotations)

# View the reordered higher-level cell types
head(overall_v_percentages)

# Create the bar plot, ordering the cell types by the percentage in the V fraction within each higher-level cell type
bar_plot <- ggplot(percentages, aes(x = reorder_within(abbreviated_idents, -percentage * (prep == "V"), highlevel_manual_annotations), y = percentage, fill = prep)) +
  geom_bar(stat = "identity") +
  scale_x_reordered() + # Apply the reordered scale
  labs(title = "Percentage of Cells in P vs V Fraction per Cell Type",
       x = "Cell Type",
       y = "Percentage",
       fill = "Fraction") +
  scale_fill_custom() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(~ highlevel_manual_annotations, scales = "free_x", space = "free_x") + # Subtle facet by higher-level cell type
  theme(panel.spacing = unit(0.2, "lines")) # Adjust the gap between the groups

# Display the plot
print(bar_plot)

```

## Boxplot of number of nuclie per donor

```{r}
# Combined half violin plot, boxplot, and jittered points
num_nuclei <- meta_data |>
  dplyr::group_by(abbreviated_idents, donor, diagnosis) |>
  dplyr::summarise(num_nuclei = n())

num_nuclei |>
  ggplot(aes(x = abbreviated_idents, y = num_nuclei, fill = diagnosis)) +
  geom_boxplot(outlier.shape = NA) +  # Hide outliers to avoid overlap with jittered points
  geom_jitter(position = position_jitter(width = 0.2), alpha = 0.6) +
  #scale_fill_brewer(palette = "Set3") +  # Choose a color palette
  labs(x = NULL,
       y = "Number of Nuclei")

num_nuclei |>
  ggplot(aes(x = abbreviated_idents, y = num_nuclei, fill = diagnosis)) +
  #geom_violin(trim = FALSE, alpha = 0.5, position = position_nudge(x = 0.2)) +
  geom_boxplot(width = 0.1, position = position_nudge(x = 0.2), outlier.shape = NA) +
  geom_jitter(position = position_jitter(width = 0.1), alpha = 0.6, size = 0.3) +
  coord_flip() +  # Flip coordinates to make it a half violin plot
  labs(x = NULL,
       y = "Number of Nuclei")
num_nuclei |>
  ggplot(aes(x = abbreviated_idents, y = num_nuclei, fill = diagnosis)) +
  #geom_violin(trim = FALSE, alpha = 0.5, position = position_nudge(x = 0.2)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.6) +
  geom_jitter(position = position_jitter(width = 0.1), alpha = 0.8, size = 0.3) +
  coord_flip() +  # Flip coordinates to make it a half violin plot
  labs(x = NULL,
       y = "Number of Nuclei")
boxplot <- num_nuclei |>
  ggplot(aes(x = abbreviated_idents, y = num_nuclei, fill = diagnosis)) +
  #geom_violin(trim = FALSE, alpha = 0.5, position = position_nudge(x = 0.2)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.6) +
  geom_jitter(position = position_jitter(width = 0.1), alpha = 0.8, size = 0.3) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = NULL,
       y = "Number of Nuclei")
```

```{r}
#| eval: false
library(ggdist)
# Raincloud plot
raincloud_plot <- num_nuclei |>
  ggplot(aes(x = subcelltype_annotations, y = num_nuclei, fill = diagnosis)) +
  ggdist::stat_halfeye(
    adjust = 0.5,
    width = 0.6,
    .width = 0,
    justification = -0.3,
    point_colour = NA
  ) +
  geom_boxplot(
    width = 0.15,
    outlier.shape = NA,
    position = position_nudge(x = 0.15)
  ) +
  geom_jitter(
    width = 0.1,
    alpha = 0.6,
    #position = position_nudge(x = 0.2)
  ) +
  coord_flip() +
  labs(
    x = "Cell Type",
    y = "Number of Nuclei"
  )

num_nuclei <- meta_data |>
  dplyr::group_by(abbreviated_idents, donor, diagnosis) |>
  dplyr::summarise(num_nuclei = n())
num_nuclei <- meta_data |>
  dplyr::group_by(highlevel_manual_annotations, donor, diagnosis) |>
  dplyr::summarise(num_nuclei = n())

raincloud_plot <- num_nuclei |>
  ggplot(aes(x = highlevel_manual_annotations, y = num_nuclei, fill = diagnosis)) +
  stat_slab(aes(thickness = after_stat(pdf*n)), scale = 0.7, position = position_dodge()) +
  stat_dotsinterval(side = "bottom", scale = 0.7, slab_linewidth = NA, alpha = 0.4, position = position_dodge()) +
  coord_flip()
raincloud_plot <- num_nuclei |>
  ggplot(aes(x = highlevel_manual_annotations, y = num_nuclei, fill = diagnosis)) +
  stat_slab(aes(thickness = after_stat(pdf*n)), scale = 0.7, position = position_dodge()) +
  stat_dotsinterval(side = "bottom", scale = 0.7, slab_linewidth = NA, alpha = 0.4, position = position_dodge())
raincloud_plot <- num_nuclei |>
  ggplot(aes(x = abbreviated_idents, y = num_nuclei, fill = diagnosis)) +
  stat_slab(aes(thickness = after_stat(pdf*n)), scale = 0.7, position = position_dodge()) +
  stat_dotsinterval(side = "bottom", scale = 0.7, slab_linewidth = NA, alpha = 0.4, position = position_dodge())
raincloud_plot <- num_nuclei |>
  ggplot(aes(x = abbreviated_idents, y = num_nuclei, fill = diagnosis)) +
  stat_slab(aes(thickness = after_stat(pdf*n)), scale = 0.7, position = position_dodge()) +
  stat_dotsinterval(side = "bottom", scale = 0.7, slab_linewidth = NA, alpha = 0.4, position = position_dodge()) +
  coord_flip()
# Print the raincloud plot
print(raincloud_plot)
```

### Explanation

- **stat_halfeye**: Creates the half violin plot part of the raincloud plot.
- **geom_boxplot**: Adds a boxplot to show summary statistics like the median and interquartile range.
- **geom_jitter**: Adds individual data points to show the actual data distribution and variability.
- **coord_flip**: Flips the coordinates to make the violin plot horizontal, which is typical for raincloud plots.


```{r}
#| eval: true
# Read external images
img1 <- image_read(here::here("05_figures/990_shared_figures/003_final_figures",
  "detailed_vascular_prep.png"))
# maybe not super high quality image?
img1 <- image_read(here::here("05_figures/990_shared_figures/003_final_figures",
  "vascular_prep_labelled.png"))

flow_diagram <- image_read(here::here("05_figures/990_shared_figures",
  "003_final_figures/cohort_flow_diagram.svg"))

vascular_cell_diff <- readr::read_rds(here::here("05_figures/990_shared_figures",
  "cell_pop_difference_level2_vascular.rds")) +
  ggtitle("") +
  scale_fill_discrete(name = "",
                     labels = c("Case", "Control")) +
  scale_x_discrete(name = "", labels = "EndoMT") +
  custom_theme

# Save version without labels
row_2 <- plot_grid(percent_nuclei_frac_v + theme(legend.position = "none"),
  percent_nuclei_frac_p + theme(legend.position = "none"),
  rel_widths = c(1.4, 1), labels = c("", ""), label_colour = "black",
  label_size = 11)

# extract the legend from one of the plots
legend <- get_legend(
  # create some space to the left of the legend
  percent_nuclei_frac_p + theme(legend.position = "bottom",
                                legend.box.margin = margin(0, 0, 0, 12))
)

row_2_l <- plot_grid(row_2, legend, rel_heights = c(2, .1), ncol = 1,
  labels = c("", ""), label_colour = "black")

ggsave(here::here("05_figures/990_shared_figures/003_final_figures",
                             "method_comparisons.svg"), plot = row_2_l,
         width = 3, height = 6, units = "in")

row_2 <- plot_grid(percent_nuclei_frac_v + theme(legend.position = "none"),
  percent_nuclei_frac_p + theme(legend.position = "none"),
  rel_widths = c(1.4, 1), labels = c("D", "E"), label_colour = "black",
  label_size = 11)

# extract the legend from one of the plots
legend <- get_legend(
  # create some space to the left of the legend
  percent_nuclei_frac_p + theme(legend.position = "bottom",
                                legend.box.margin = margin(0, 0, 0, 12))
)

row_2_l <- plot_grid(row_2, legend, rel_heights = c(2, .1), ncol = 1,
  labels = c("", ""), label_colour = "black")


# Add external images
img1_gg <- ggdraw() + draw_image(img1)
img2_gg <- ggdraw() + draw_image(flow_diagram)

row_2_flow <- plot_grid(img2_gg, row_2_l, nrow = 1, labels = c("D", ""),
  label_colour = "black", label_size = 11)

final_plot <- plot_grid(img1_gg, img2_gg, ncol = 1, labels = c("", "D"),
  label_colour = "black")
final_plot
# Save the final compound figure
cairo_pdf(file = here::here("05_figures/990_shared_figures/003_final_figures",
 "figure1.pdf"), width = 7, height = 9)
print(final_plot)
dev.off()
```

### Cohort visualisation

```{r}
cohort <- sce@meta.data |>
  dplyr::select(donor, diagnosis, sex:age, Ethnicity, diagnosis_details,
                apoe_status) |>
  dplyr::distinct() |>
  as_tibble()

cohort |>
  dplyr::count(sex, diagnosis)
cohort |>
  summarise(median_age = median(age), iqr_age = IQR(age),
  .by = c(diagnosis, sex))

sce@meta.data |>
  dplyr::select(donor, diagnosis, sex, prep) |>
  dplyr::distinct() |>
  as_tibble() |>
  dplyr::summarise(n = n(), .by = c(diagnosis, prep))

age_plot <- ggplot(cohort, aes(x = sex, y = age, fill = diagnosis)) +
  geom_boxplot() +
  theme_minimal() +
  labs(x = "", y = "Age")

age_plot <- ggplot(cohort, aes(x = sex, y = age, fill = diagnosis)) +
  geom_boxplot() +
  #geom_jitter(width = 0.3, alpha = 0.5) +
  theme_minimal() +
  labs(x = "", y = "Age", fill = "")
age_plot
ggsave(filename = here::here("05_figures/990_shared_figures/003_final_figures",
                             "000_sup_figures/cohort_boxplot.png"), plot = age_plot,
       width = 8, height = 8, units = "in", dpi = 300)
```

## Asemble UMAP

```{r}
# Combine ggplot objects
row_1 <- plot_grid(umap_vascular_parenchymal + theme(aspect.ratio = 1), umap_level1 + theme(aspect.ratio = 1), nrow = 1, labels = c("A", "B"), label_colour = "black")
row_1 <- plot_grid(umap_vascular_parenchymal + theme(aspect.ratio = 1), umap_level1 + theme(aspect.ratio = 1), labels = c("A", "B"), label_colour = "black")

row_2 <- plot_grid(umap, row_2_l, labels = c("C", ""), label_colour = "black", rel_widths = c(2, 1))
# Combine everything
final_plot <- plot_grid(row_1, row_2, ncol = 1, labels = c("", ""), label_colour = "black")

ggsave(file = here::here("05_figures/990_shared_figures/003_final_figures",
                             "figure1_umaps.png"), plot = final_plot,
         width = 7, height = 9, units = "in", dpi = 300)
cairo_pdf(file = here::here("05_figures/990_shared_figures/003_final_figures",
                             "figure1_umaps.pdf"),
         width = 7, height = 9)
print(final_plot)
dev.off()
```

## Figure - risk

This figure focuses on the risk scores from magma

### GO pathways

```{r}
go_combined <-
  qs::qread(
    here::here(
      "05_figures/990_shared_figures/003_final_figures/001_data_for_plots/go_data_sig_gene_only.qs"
    )
  )

```

```{r}
#| eval: false
make_combined_go_plots <- function(go_combined, plot_title, top_n = 10) {
  # Select top terms for each ontology, excluding the specific terms already selected
  top_terms <- go_combined %>%
    dplyr::filter(Count > 1 & p.adjust < 0.05) %>%
    dplyr::group_by(Ontology) %>%
    dplyr::slice_min(p.adjust, n = top_n) %>%
    dplyr::ungroup() |>
    dplyr::mutate(Ontology = toupper(Ontology))

  # Calculate Gene Ratio
  top_terms$GeneRatio <- sapply(top_terms$GeneRatio, function(x) {
    parts <- unlist(strsplit(x, "/"))
    as.numeric(parts[1]) / as.numeric(parts[2])
  })

  # Create dot plot
  ggplot(top_terms, aes(x = GeneRatio, y = reorder(Description, GeneRatio))) +
    geom_point(aes(size = Count, color = p.adjust)) +
    scale_color_gradient(low = "red", high = "blue") +
    facet_wrap( ~ Ontology, scales = "free_y") +
    labs(title = plot_title,
         x = "Gene Ratio",
         y = "") +
    theme(
      axis.text.x = element_text(angle = 45, hjust = 1),
      legend.position = "bottom",
      strip.background = element_blank(),
      strip.text = element_text(face = "bold"),
      plot.title = element_text(size = 10)
    )
}

plots <-
  map2(go_combined,
       names(go_combined),
      make_combined_go_plots,
      top_n = 20)
```

```{r}
celltype_abbrev_df <- data.frame(celltypes_abbrev) |> rownames_to_column("celltype")
readr::write_tsv(celltype_abbrev_df, here::here("05_figures/990_shared_figures/003_final_figures/001_data_for_plots/celltype_abbreviations.tsv"))
# celltype_abbrev_df <- readr::read_tsv(here::here("05_figures/990_shared_figures/003_final_figures/001_data_for_plots/celltype_abbreviations.tsv"))
top_terms <- map2_dfr(go_combined, names(go_combined), ~ .x |>
                        dplyr::filter(p.adjust < 0.05) |>
                        # dplyr::group_by(Ontology) |>
                        # dplyr::slice_min(p.adjust, n = 20) |>
                        # dplyr::ungroup() |>
                        dplyr::mutate(Ontology = toupper(Ontology),
                                      celltype = .y,
                                      amyloid_pathways = if_else(grepl("amyloid", Description), "Amyloid-related", "Other pathways"))) |>
  dplyr::left_join(celltype_abbrev_df, by = join_by(celltype)) |>
  dplyr::mutate(celltypes_abbrev = if_else(is.na(celltypes_abbrev), celltype, celltypes_abbrev))

# Calculate Gene Ratio
top_terms$GeneRatio <- sapply(top_terms$GeneRatio, function(x) {
  parts <- unlist(strsplit(x, "/"))
  as.numeric(parts[1]) / as.numeric(parts[2])
})

# Remove terms that only have one gene
top_terms <- top_terms |>
  dplyr::filter(Count > 1)

result_list_ont <- split(top_terms, top_terms$Ontology)
simMatrix <- map2(
  result_list_ont,
  names(result_list_ont),
  ~ calculateSimMatrix(
    unique(.x$ID),
    orgdb = "org.Hs.eg.db",
    ont = .y,
    method = "Rel"
  )
)

# make sure the scores match the simMatrix
x <- map2(result_list_ont, simMatrix, ~ .x |>
      dplyr::distinct(ID, .keep_all = TRUE) |>
        dplyr::filter(ID %in% colnames(.y)))

scores <- map(x, ~ setNames(-log(.x$p.adjust), .x$ID))

# assert that the score go terms match the simmatrix rownames
assertthat::are_equal(sum(map2_dbl(
  scores, simMatrix, ~ sum(names(.x) != rownames(.y))
)), 0)
# Assert that there are no missing values in the scores
assertthat::assert_that(sum(map_dbl(scores, ~ sum(is.na(.x)))) == 0)
reducedTerms <- map2(simMatrix, scores, ~ reduceSimMatrix(.x,
                                .y,
                                threshold=0.9,
                                orgdb="org.Hs.eg.db"))

# Add the rrvgo reduced terms to the initial dataframe
top_terms_reduced <- map2(reducedTerms, names(reducedTerms), ~ .x |>
       dplyr::mutate(Ontology = .y)) |>
  list_rbind() %>%
  dplyr::left_join(top_terms, ., by = join_by(ID == go, Ontology)) |>
  dplyr::mutate(celltypes_abbrev = str_remove(celltypes_abbrev, "-Bellenguez|-EUROPEUKBB")) |>
  dplyr::mutate(
    parent_median_p_adj = median(p.adjust),
    parent_median_gene_ratio = median(GeneRatio),
    parent_median_count = median(Count),
    .by = c(Ontology, celltypes_abbrev, parentTerm, celltype_level)
  ) |>
  dplyr::select(
    Ontology,
    celltypes_abbrev,
    parentTerm,
    parent_median_gene_ratio,
    parent_median_p_adj,
    parent_median_count,
    celltype_level
  ) |>
  dplyr::distinct() |>
  # add a counter for terms that appear in more than 1 celltype
  dplyr::mutate(count_overlap = n(), .by = c(Ontology, parentTerm)) |>
  dplyr::filter(!is.na(parentTerm)) |>
  dplyr::mutate(
    amyloid_pathways = if_else(
      grepl("amyloid", parentTerm),
      "Amyloid-related",
      "Other pathways"
    )
  ) |>
  dplyr::rename(celltype = celltypes_abbrev) |>
  dplyr::left_join(celltype_abbrev_df, by = join_by(celltype)) |>
  dplyr::mutate(celltypes_abbrev = if_else(is.na(celltypes_abbrev), celltype, celltypes_abbrev)) |>
  dplyr::mutate(celltypes_abbrev = fct_relevel(celltypes_abbrev, "Fibroblast", "Microglia", "Pericyte-1", "T-cell", "Microglia-A", "Pericyte-2", "Perivascular-FB-2")) |>
  dplyr::mutate(celltypes_abbrev = fct_recode(celltypes_abbrev, Pericyte = "Pericyte-1"))

table(top_terms_reduced$Ontology)
# Function to wrap text
wrap_text <- function(text, width = 20) {
  str_wrap(text, width = width)
}

# Wrap the GO terms in the data
top_terms_reduced$parentTerm <- wrap_text(top_terms_reduced$parentTerm, 60)


go_plot_all <- top_terms |>
  # filter to just BP for make plot easier to parse
  dplyr::filter(Ontology == "BP") |>
  ggplot(aes(x = -log10(p.adjust), y = reorder(Description, -log10(p.adjust)))) +
  geom_point(aes(size = Count, color = p.adjust)) +
  scale_color_gradient(low = "red", high = "blue") +
  facet_grid(celltype ~ Ontology, scales = "free_y") +
  labs(x = "Gene Ratio",
       y = "") +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(face = "bold"),
    plot.title = element_text(size = 10)
  )

# Create dot plot
go_plot_all <- ggplot(top_terms, aes(x = GeneRatio, y = reorder(Description, GeneRatio))) +
  geom_point(aes(size = Count, color = p.adjust)) +
  scale_color_gradient(low = "red", high = "blue") +
  facet_grid(celltype ~ Ontology, scales = "free_y") +
  labs(x = "Gene Ratio",
       y = "") +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(face = "bold"),
    plot.title = element_text(size = 10)
  )
go_plot_all <- top_terms |>
  # filter to just BP for make plot easier to parse
  dplyr::filter(Ontology == "BP") |>
  ggplot(aes(x = GeneRatio, y = reorder(Description, GeneRatio))) +
  geom_point(aes(size = Count, color = p.adjust)) +
  scale_color_gradient(low = "red", high = "blue") +
  facet_grid(amyloid_pathways ~ celltypes_abbrev, scales = "free_y", space = "free_y") +
  labs(x = "Gene Ratio",
       y = "") +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    #legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(face = "bold", size = 8),
    plot.title = element_text(size = 10)
  )
go_plot_all <- top_terms_reduced |>
  # filter to just BP for make plot easier to parse
  dplyr::filter(Ontology == "BP") |>
  dplyr::filter(celltypes_abbrev %in% c("Microglia-A", "Pericyte-2", "Perivascular-FB-2")) |>
  ggplot(aes(x = parent_median_gene_ratio, y = reorder(parentTerm, parent_median_gene_ratio))) +
  geom_point(aes(size = parent_median_count, color = parent_median_p_adj)) +
  scale_color_gradient(low = "red", high = "blue") +
  scale_size_continuous(range = c(1, 4)) +  # Adjust point size scale (min, max)
  facet_grid(amyloid_pathways ~ celltypes_abbrev, scales = "free_y", space = "free_y") +
  labs(x = "Gene Ratio",
       y = "", size = "Median\ncount", color = "Median\nadjusted\nP-value") +
  theme(
    axis.text.x = element_text(size = 6, angle = 90, hjust = 1, vjust = 0.5),
    #legend.position = "bottom",
    strip.background = element_blank(),
    strip.text.y = element_blank(),
    panel.background = element_blank(), # Remove background
    panel.grid.major = element_line(color = "grey80"), # Major grid lines color
    strip.text = element_text(face = "bold", size = 5),
    plot.title = element_text(size = 8),
    axis.text.y = element_text(size = 6),
    axis.title.x = element_text(size = 7),
    legend.text = element_text(size = 6),
    legend.key.width = unit(2, "mm"),
    legend.title = element_text(size = 6, face = "bold")
  ) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 45))
go_plot_filtered <- top_terms_reduced |>
  # filter to just BP for make plot easier to parse
  dplyr::filter(Ontology == "BP") |>
  dplyr::slice_min(parent_median_p_adj, n = 15, by = c(celltypes_abbrev, amyloid_pathways)) |>
  ggplot(aes(x = parent_median_gene_ratio, y = reorder(parentTerm, parent_median_gene_ratio))) +
  geom_point(aes(size = parent_median_count, color = parent_median_p_adj)) +
  scale_color_gradient(low = "red", high = "blue") +
  facet_grid(amyloid_pathways ~ celltypes_abbrev + celltype_level, scales = "free_y", space = "free_y") +
  labs(x = "Gene Ratio",
       y = "", size = "Median\ncount", color = "Median\nadjusted\npval") +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1),
    #legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(face = "bold", size = 4),
    plot.title = element_text(size = 10)
  )
```

### MAGMA heatmap

```{r}
magma_heatmap_data <- qs::qread(here::here("05_figures/990_shared_figures/003_final_figures/magma_celltype_heatmap_data_endomt_subtypes.qs"))
magma_heatmap_data <- qs::qread(here::here("05_figures/990_shared_figures/003_final_figures/magma_celltype_heatmap_data_endomt_subtypes_nominal.qs"))

#ldscore_heatmap_data <- qs::qread(here::here("05_figures/990_shared_figures/003_final_figures/ldscore_celltype_heatmap_data.qs"))
# Update col labels
colnames(magma_heatmap_data) <- "MAGMA"
#colnames(ldscore_heatmap_data) <- "LDScore"
# Merge data
#heatmap_data <- cbind(magma_heatmap_data, ldscore_heatmap_data)
heatmap_data <- magma_heatmap_data

create_annotation_matrix <- function(pval_matrix) {
  # Get dimensions of input matrix
  n_rows <- nrow(pval_matrix)
  n_cols <- ncol(pval_matrix)

  # Create empty character matrix for annotations
  annot_matrix <- matrix("", nrow = n_rows, ncol = n_cols)

  # Convert log10 p-values back to regular p-values
  regular_pvals <- 10^(-pval_matrix)

  # Calculate Bonferroni adjusted p-values
  # Convert matrix to vector, adjust, then back to matrix
  adj_pvals <- matrix(
    p.adjust(as.vector(regular_pvals), method = "bonferroni"),
    nrow = n_rows,
    ncol = n_cols
  )

  # Fill annotation matrix based on conditions
  annot_matrix[regular_pvals < 0.05] <- "*"
  annot_matrix[adj_pvals < 0.05] <- "**"
  annot_matrix[adj_pvals < 0.01] <- "***"

  # Preserve dimnames if they exist
  dimnames(annot_matrix) <- dimnames(pval_matrix)

  return(annot_matrix)
}

heatmap_data <- t(heatmap_data)
colnames(heatmap_data) <- str_replace(colnames(heatmap_data), "_", "-")
row_anno <- create_annotation_matrix(heatmap_data)

# Create the heatmap
heatmap <- Heatmap(heatmap_data,
        name = "-log10 p-value",
#        row_names_side = "left",
#        right_annotation = row_anno,
    # Adjust the clustering method if necessary
    cluster_rows = FALSE,
    cluster_columns = FALSE,
        show_row_names = TRUE,
        show_column_names = TRUE,
    #column_names_rot = 45,
    #row_names_rot = 90,
    #row_names_centered = TRUE,
    #column_names_centered = TRUE,
    # Use the defined color palette
    cell_fun = function(j, i, x, y, width, height, fill) {
      if (row_anno[i, j] != "") {
        grid.text(row_anno[i, j], x, y, gp = gpar(col = "black", fontsize = 5))
      }
    },
    row_names_gp = gpar(fontsize = 6),  # Set row label text size
        column_names_gp = gpar(fontsize = 6),
     #width = unit(1, "cm"))
     height = unit(3, "mm"),
     heatmap_legend_param = list(direction = "horizontal",
                                 labels_gp = gpar(fontsize = 5),
                                 title_gp = gpar(fontsize = 5,
                                                 fontface = "bold"),
                                 title_position = "leftcenter",
                                 gap = unit(0.1, "mm"),
                                 grid_height = unit(0.5, "mm")))
cairo_pdf(file = here::here("05_figures/990_shared_figures/003_final_figures", "magma_heatmap.pdf"),
         width = 7, height = 2)
draw(heatmap, heatmap_legend_side = "top", ht_gap = unit(100, "mm"),
  heatmap_row_names_gp = gpar(fontsize = 12), heatmap_column_names_gp = gpar(fontsize = 12))
dev.off()
p2 = grid.grabExpr(draw(heatmap, heatmap_legend_side = "top"), ht_gap = unit(100, "mm"))
#row_1 <- plot_grid(p2, go_plot_filtered, labels = c("A", "B"), rel_widths = c(1.5, 2.5), label_colour = "black")
# TODO - make the transposed heatmap fit next to the reduced conditional heatmap.
row_1 <- plot_grid(p2, go_plot_all, labels = c("A", "B"), rel_widths = c(1, 3), label_colour = "black")

row_1
```

```{r}
files <- list.files(here::here("03_data/994_magma_inputs/sig_genelist/results"),
                    pattern = "*.gsa.out", full.names = TRUE)

# Read data
parse_magma_set_data <- function(file) {
  file_lines <- readLines(file)
  # Get the index of the line judt before the data starts
  data_starts_index <- grep("^# CONDITIONED_", file_lines)
  data <- read_table(file, skip = data_starts_index[length(data_starts_index)], show_col_types = FALSE)
  return(data)
}
data <- map(files, parse_magma_set_data)

files_sub <- str_replace(basename(files), "level2_all_controls_top-ten-percent_AD_", "") |>
  str_replace("_conditional", "") %>%
  # Use sub() to remove all characters after the first dot
  sub("\\..*$", "", .)

process_magma_data <-
  function(df,
           filename,
           celltypes_to_include = c("Perivascular-FB-KAZN2", "Pericyte-2", "Microglia-activated")) {
    df <- df |> dplyr::filter(VARIABLE %in% celltypes_to_include) |>
    # Add groups and adjusted pvals to data
    dplyr::mutate(conditioned_on = filename) |>
      dplyr::mutate(FULL_NAME = tolower(FULL_NAME)) |>
      dplyr::filter(FULL_NAME != conditioned_on) |>
      dplyr::mutate(p_adj = p.adjust(P, method = "bonferroni"))
    return(df)
  }
df <- map2(data, files_sub, ~ process_magma_data(.x, .y)) |>
  list_rbind() |>
  as_tibble()
```

#### Conditional heatmap

```{r}
df$abbrev <- celltypes_abbrev[as.character(df$VARIABLE)]
df$abbrev <- if_else(is.na(df$abbrev), df$VARIABLE, df$abbrev)

df$abbrev[grepl("Ex-Neuron-L2-3-C", df$abbrev)] <- "Ex-Neuro-4"
df$abbrev[grepl("In-Neuron-ADARB2-IL1RAPL2", df$abbrev)] <- "In-Neuro-2"
df <- df |>
  dplyr::filter(abbrev %in% c("Perivascular-FB-2", "Pericyte-2", "Microglia-A")) |>
  dplyr::mutate(log_p = -log10(p_adj))

# Nest data on broad groups
df_nest <- df |>
  dplyr::group_by(conditioned_on) |>
  tidyr::nest()

get_heatmap_data <- function(df) {
  # Prepare the data for the heatmap
  heatmap_data <- df |>
    dplyr::select(abbrev, log_p, conditioned_on) |>
    pivot_wider(names_from = abbrev, values_from = log_p) |>
    as.data.frame()

  rownames(heatmap_data) <- heatmap_data$conditioned_on
  # Remove the first column if it contains row names (cell types)
  heatmap_data <- heatmap_data[, -1]
  heatmap <- t(heatmap_data)
  # Replace any NAs with 0
  #heatmap[is.na(heatmap)] <- 0
  return(heatmap)
}
heatmap_data_con <- get_heatmap_data(df)
colnames(heatmap_data_con) <- c("Microglia-A", "Pericyte-2", "Perivascular-FB-2")

row_anno <- create_annotation_matrix(heatmap_data_con)

heatmap_conditional <- Heatmap(heatmap_data_con,
        name = "-log10 \nAdjusted\nP-value",
        row_names_side = "left",
        #right_annotation = row_anno_conditional,
    cell_fun = function(j, i, x, y, width, height, fill) {
      if (row_anno[i, j] != "") {
        grid.text(row_anno[i, j], x, y, gp = gpar(col = "black", fontsize = 10))
      }
    },
    # Adjust the clustering method if necessary
    cluster_rows = FALSE,
    cluster_columns = FALSE,
        show_row_names = TRUE,
        show_column_names = TRUE,
    column_names_rot = 90,
    #column_names_centered = TRUE,
    #row_names_rot = 90,
    #row_names_centered = TRUE,
    column_title = "Conditioned on",
    column_title_gp = gpar(fontsize = 6),
    column_title_side = "bottom",
    row_names_gp = gpar(fontsize = 5),  # Set row label text size
    column_names_gp = gpar(fontsize = 5),
    heatmap_legend_param = list(direction = "horizontal",
                                labels_gp = gpar(fontsize = 5),
                                grid_height = unit(2, "mm"),
                                title_gp = gpar(fontsize = 6,
                                                fontface = "bold"),
                                title_position = "leftcenter")
    )
heatmap_conditional
p3 = grid.grabExpr(draw(heatmap_conditional, heatmap_legend_side = "top"))

row_1 <- plot_grid(p2, go_plot_all, labels = c("A", "B"), rel_widths = c(1, 3), label_colour = "black")

row_1
```

#### Avg risk gene expression

```{r}
avg_gene_expr <- readr::read_tsv(here::here("05_figures/990_shared_figures/003_final_figures/001_data_for_plots/risk_gene_avg_expression.tsv"))

# Remove the gene column to create a matrix
expression_matrix <- as.matrix(avg_gene_expr[, -1])

# Optionally, you can set row names to gene names
rownames(expression_matrix) <- avg_gene_expr$features.plot

colnames(expression_matrix) <- sapply(colnames(expression_matrix), function(name) {
  if (name %in% celltype_abbrev_df$celltype) {
    return(celltype_abbrev_df$celltypes_abbrev[celltype_abbrev_df$celltype == name])
  } else {
    return(name)
  }
})

colnames(expression_matrix)[colnames(expression_matrix) == "Pericyte-1"] <- "Pericyte"
expression_matrix[1:5, 1:3]
expression_matrix <- expression_matrix[,colnames(expression_matrix) != "T-cell"]
expression_matrix_t <- expression_matrix
expression_matrix <- t(expression_matrix)
avg_risk_gene_heatmap <- Heatmap(expression_matrix,
        name = "Expression",
        column_names_rot = 90,
        #column_names_centered = TRUE,
    # Adjust the clustering method if necessary
    cluster_rows = TRUE,
    cluster_columns = TRUE,
    show_row_dend = FALSE,
    show_column_dend = FALSE,
    row_names_gp = gpar(fontsize = 6),
    column_names_gp = gpar(fontsize = 5),
    heatmap_legend_param = list(direction = "horizontal",
                                labels_gp = gpar(fontsize = 5),
                                grid_height = unit(0.5, "mm"),
                                title_gp = gpar(fontsize = 5,
                                                fontface = "bold"),
                                title_position = "leftcenter"),
)
avg_risk_gene_heatmap_t <- Heatmap(expression_matrix_t,
        name = "Expression",
        column_names_rot = 90,
        column_names_centered = TRUE,
    # Adjust the clustering method if necessary
    cluster_rows = TRUE,
    cluster_columns = TRUE,
    show_row_dend = FALSE,
    show_column_dend = FALSE,
    row_names_gp = gpar(fontsize = 6),
    column_names_gp = gpar(fontsize = 8)
)
avg_risk_gene_heatmap
avg_risk_gene_heatmap_t
p4 = grid.grabExpr(draw(avg_risk_gene_heatmap, heatmap_legend_side = "top"))
p4_t = grid.grabExpr(draw(avg_risk_gene_heatmap_t))
```

### Kegg AD pathway annotations

Tried mapping the MAGMA risk genes to the KEGG AD pathway, but very few genes mapped...

```{r}
library(KEGGREST)
amyloid_pathways <- keggFind("pathway", "Alzheimer")
print(amyloid_pathways)
pathway_genes <- keggGet("hsa05010")[[1]]$GENE
pathway_gene_symbols <- pathway_genes[seq(1, length(pathway_genes), by=2)]  # Extract gene symbols
head(pathway_genes)

# library(pathview)
# pathview(gene.data = avg_gene_expr$features.plot,
#          pathway.id = "hsa05010",
#          species = "hsa",
#          kegg.native = TRUE)
```

### Mouse phenotypes

```{r}
mpo <- readr::read_tsv(here::here("05_figures/990_shared_figures/003_final_figures/001_data_for_plots/mpo_data.tsv")) |>
  dplyr::mutate(celltype = str_remove(celltype, "_.*")) |>
  dplyr::left_join(celltype_abbrev_df, by = join_by(celltype)) |>
  dplyr::mutate(celltypes_abbrev = if_else(is.na(celltypes_abbrev), celltype, celltypes_abbrev))


g1 <- ggplot(
  data = mpo,
  mapping = aes(
    y = reorder(MP_name, -log10(p_value_adj)),
    x = -log10(p_value_adj),
    size = GenesInMPterm,
    color = -log10(p_value_adj)
  )
) +
  geom_point() +
  scale_color_gradient2(low = "snow",
                        mid = "blue",
                        high = "red") +
  facet_wrap( ~ celltypes_abbrev) +
  scale_size_continuous(breaks = 1:max(mpo$GenesInMPterm),
                        labels = as.character(1:max(mpo$GenesInMPterm)),
                        range = c(1, 3)) +
  labs(y = "", color = "-log10\nAdjusted\nP-value",
       size = "Genes in\nterm", x = "-log10(Adjusted P-Value)") +
  theme(
    #axis.text.x = element_text(angle = 45, hjust = 1),
    #legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(size = 6, face = "bold"),
    axis.text.y = element_text(size = 6),
    axis.text.x = element_text(size = 6, angle = 90, vjust = 0.5, hjust = 1),
    axis.title.x = element_text(size = 6),
    legend.text = element_text(size = 5),
    legend.key.width = unit(3.5, 'mm'),
    legend.key.height = unit(2, 'mm'),
    legend.title = element_text(size = 6, face = "bold")
  )
p_mpo <- g1 +
  theme(legend.position = "bottom",
        legend.box = "vertical",
        panel.background = element_blank(),
        panel.grid.major = element_line(color = "grey80")) +
  scale_y_discrete(labels = function(x) str_wrap(x, width = 12))
p_mpo
```

### Assemble Figure 2

```{r}
magma_sig <-
  readr::read_tsv(here::here(
    "03_data/994_magma_inputs/sig_celltype_sig_genes.tsv"
  )) |>
  janitor::clean_names() |>
  dplyr::rename(entrezgene_id = gene) |>
  # Exclude the top one percents
  dplyr::filter(!percent == "top-one-percent") |>
  # Filter to level 2
  dplyr::filter(celltype_level == "level2") |>
  dplyr::filter(padj < 0.05) #|>
  #left_join(entrez_ids, by = join_by(entrezgene_id))
```

ven diagram of sig risk genes from magma for each celltype

### PPI network plots

```{r}
ppi_plots <- qs::qread(here::here("05_figures/990_shared_figures/003_final_figures",
                     "001_data_for_plots/ppi_network_plots.qs"))
ppi_plots <- ppi_plots[names(ppi_plots) %in% c("Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2_control",
                                               "Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2_control",
                                               "Perivascular-FB-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2_control")]
```

```{r}
row_1_1 <- plot_grid(p2, p4, labels = c("A", "C"), rel_heights = c(1, 0.9),
                   label_colour = "black", ncol = 1)
row_1 <- plot_grid(row_1_1, p3, labels = c("", "B"), rel_widths = c(2, 0.8),
                   rel_heights = c(1, 2), label_colour = "black", ncol = 2)

row_2 <- plot_grid(go_plot_all, p_mpo, rel_widths = c(3, 1.2),
                   label_colour = "black", labels = c("D", "E"), ncol = 2)

row3 <- plot_grid(plotlist = ppi_plots, labels = c("F", "G", "H"), label_colour = "black", nrow = 1)

final_plot2 <- plot_grid(row_1, row_2, row3, ncol = 1, rel_heights = c(1.4, 2.5, 1.8))

ggsave(filename = here::here("05_figures/990_shared_figures/003_final_figures",
                             "figure3_risk.pdf"), plot = final_plot2,
       width = 7, height = 9, units = "in")
```



```{r}
row_2 <- plot_grid(g1, row2_sub, labels = c("C", ""), rel_widths = c(2, 1), label_colour = "black")
row_2 <- plot_grid(g1, row2_sub, labels = c("C", ""), label_colour = "black")
row_3 <- plot_grid(p3, p4, labels = c("G", "H"), label_colour = "black")
row_3


final_plot2 <- plot_grid(row_1, row_2, row_3, ncol = 1, labels = c("", "", ""), label_colour = "black")
final_plot2

ggsave(here("05_figures/990_shared_figures/003_final_figures/figure2_row1.png"), plot = row_1, width = 12, height = 8, units = "in", dpi = 300)
ggsave(here("05_figures/990_shared_figures/003_final_figures/figure2_row2.png"), plot = row_2, width = 10, height = 6, units = "in", dpi = 300)
ggsave(here("05_figures/990_shared_figures/003_final_figures/figure2_row3.png"), plot = row_3, width = 12, height = 8, units = "in", dpi = 300)

ggsave(filename = here::here("05_figures/990_shared_figures/003_final_figures/figure2.png"), plot = final_plot2,
       width = 10, height = 15, units = "in", dpi = 300)
```

## Figure 3 - DEGs

- need to start with barplot of number of up and downregulated genes.

```{r}
res_sig <- readr::read_csv(here::here("03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_no-apoe.csv")) |>
  na.omit() |>
  dplyr::filter(padj < 0.05 & !celltype %in% c("ambiguous", "low-feature-cells")) |>
  dplyr::left_join(celltype_abbrev_df, by = join_by(celltype)) |>
  dplyr::mutate(celltypes_abbrev = if_else(is.na(celltypes_abbrev), celltype, celltypes_abbrev))
# Add nuclei numbers
x <- table(sce$abbreviated_idents) |> as.data.frame() |>
  dplyr::rename(celltypes_abbrev = Var1, nuclei_n = Freq) |>
  dplyr::left_join(res_sig)
res_sig_filtered_celltypes <- res_sig |>
  dplyr::group_by(celltypes_abbrev) |>
  dplyr::mutate(count = n()) |>
  dplyr::filter(count > 30) |>
  dplyr::ungroup()
```

```{r}
tar_load(translated_id)

mast_de <- qs::qread(here::here("03_data/990_processed_data/001_snrnaseq/13_mast_de/level2/mast_de_results_list.qs")) |>
  map(rownames_to_column, var = "ensembl") |>
  list_rbind() |>
  dplyr::left_join(translated_id, by = join_by(ensembl == ensembl_gene_id)) |>
  dplyr::group_by(celltype) |>
  dplyr::distinct(ensembl, .keep_all = TRUE) |>
  dplyr::ungroup() |>
  dplyr::left_join(celltype_abbrev_df, by = join_by(celltype)) |>
  dplyr::mutate(celltypes_abbrev = if_else(is.na(celltypes_abbrev), celltype, celltypes_abbrev), deg_direction = if_else(avg_log2FC > 0, "Up", "Down")) |>
  dplyr::rename(gene = hgnc_symbol)

res_sig <- mast_de |>
  dplyr::filter(p_val_adj < 0.05 & abs(avg_log2FC) > 1)

table(res_sig$celltypes_abbrev)

res_sig_filtered_celltypes <- res_sig |>
  dplyr::group_by(celltypes_abbrev) |>
  dplyr::mutate(count = n()) |>
  dplyr::filter(count > 10) |>
  dplyr::ungroup()
table(res_sig_filtered_celltypes$celltypes_abbrev)

x <- table(sce$abbreviated_idents) |> as.data.frame() |>
  dplyr::rename(celltypes_abbrev = Var1, nuclei_n = Freq) |>
  dplyr::left_join(res_sig)
```

# Check sig risk genes in DEGs

```{r}
risk_deg_overlap <- res_sig |>
  dplyr::filter(gene %in% avg_gene_expr$features.plot)
table(risk_deg_overlap$celltypes_abbrev, risk_deg_overlap$deg_direction)
#readr::write_tsv(risk_deg_overlap, here("risk_deg_overlap.tsv"))
```

### DEGs count barplot

```{r}
# Assuming you have a data frame 'data' with columns 'CellType', 'Regulation' and 'GeneCount'
# Replace 'CellType', 'Regulation', and 'GeneCount' with the actual column names in your data frame

# Function to wrap text
wrap_text <- function(text, width = 20) {
  str_wrap(text, width = width)
}

ylab_title <- wrap_text("Number of significant differential expressed genes with average log2 foldchange of > 1", 45)
ylab <- expression(paste("Number of significant differential expressed genes ",
                         "with average ", log[2], " foldchange of > 1", sep = " "))
ylab <- paste("Number of significant differential expressed genes\nwith average log2 foldchange of > 1", sep = " ")
deg_count_plot <- res_sig |>
  dplyr::group_by(celltypes_abbrev, deg_direction) |>
  dplyr::summarise(count = n()) |>
  ggplot(aes(x = reorder(celltypes_abbrev, count), fill = deg_direction)) +
  geom_bar(aes(y = count), stat = "identity", position = position_dodge()) +
  theme_minimal() +
  labs(x = "", y = ylab_title, fill = "Regulation") +
  theme(axis.title.y = element_text(size = 7),
        axis.text.x = element_text(size = 6, angle = 90,
                                   hjust = 1, vjust = 0.5),
        legend.text = element_text(size = 5),
        legend.key.width = unit(6, 'mm'),
        legend.key.height = unit(4, 'mm'),
        legend.title = element_text(size = 6),
        legend.margin = margin(t = -8, unit = "mm"),  # Move legend closer (reduce top margin)
        legend.position = "bottom")
deg_count_plot
```

```{r}
# Prepare data for heatmap
heatmap_data <- res_sig %>%
  dplyr::group_by(celltypes_abbrev, deg_direction) %>%
  dplyr::summarise(count = n(), .groups = "drop") %>%
  tidyr::pivot_wider(names_from = deg_direction, values_from = count, values_fill = list(count = 0))

# Convert to matrix
heatmap_matrix <- as.matrix(heatmap_data[, -1])  # Exclude celltype column
rownames(heatmap_matrix) <- heatmap_data$celltypes_abbrev

# Order by total DEG count
celltype_order <- rowSums(heatmap_matrix) |> order(decreasing = TRUE)
heatmap_matrix <- heatmap_matrix[celltype_order, ]

# Rotate if needed
#heatmap_matrix <- t(heatmap_matrix)

# Define heatmap colors
#heatmap_colors <- colorRamp2(c(min(heatmap_matrix), max(heatmap_matrix)), c("white", "red"))
```

Make a heatmap annotation denoting if a celltype if from the vascular or parenchymal fraction.

```{r}
vascular <- c('Arterial', 'Arteriolar-SMC', 'Artirial-FB', 'Capillary',
              'EndoMT-1', 'EndoMT-2', 'M-Pericyte', 'Meningeal-FB', 'Microglia-V',
              'Pericyte-1', 'Pericyte-2', 'Pericyte-FB', 'Perivascular-FB-1',
              'Perivascular-FB-2', 'Perivascular-M', 'T-cell-M', 'T-cell-V',
              'T-Pericyte', 'Vascular-SMC-1', 'Vascular-SMC-2', 'Venous') |>
  sort()
parencyhmal <- c('Astro-A', 'Astro-Q', 'Ex-Neuro-1', 'Ex-Neuro-2',
                 'Ex-Neuro-3', 'Ex-Neuro-4', 'Ex-Neuro-5', 'Ex-Neuro-6',
                 'Ex-Neuro-7', 'In-Neuro-1', 'In-Neuro-2', 'In-Neuro-3',
                 'In-Neuro-4', 'In-Neuro-5', 'In-Neuro-6', 'In-Neuro-7',
                 'Microglia-A', 'Microglia-Q', 'Oligo-A', 'Oligo-B', 'OPC-A',
                 'OPC-B', 'T-cell-P1') |> sort()
names(vascular) <- rep("Vascular", length(vascular))
names(parencyhmal) <- rep("Parenchymal", length(parencyhmal))
cell_groups <- c(vascular, parencyhmal)

```

Make the same heatmap without the logFC filter

```{r}
heatmap_data <- mast_de |>
  dplyr::filter(p_val_adj < 0.05) |>
  dplyr::group_by(celltypes_abbrev, deg_direction) |>
  dplyr::summarise(count = n(), .groups = "drop") |>
  tidyr::pivot_wider(names_from = deg_direction, values_from = count, values_fill = list(count = 0))

# Convert to matrix
heatmap_matrix_all <- as.matrix(heatmap_data[, -1])  # Exclude celltype column
rownames(heatmap_matrix_all) <- heatmap_data$celltypes_abbrev

# Order by total DEG count
celltype_order <- rowSums(heatmap_matrix_all) |> order(decreasing = TRUE)
heatmap_matrix_all <- heatmap_matrix_all[celltype_order, ]

heatmap_matrix_all[!(rownames(heatmap_matrix_all) %in% rownames(heatmap_matrix)),]

# Get the order of indices
order_indices <- match(rownames(heatmap_matrix), cell_groups)

# Reorder the vector
reordered_vector <- cell_groups[order_indices]

# Flip names and values
cell_test <- names(reordered_vector)
names(cell_test) <- reordered_vector

# Create the row annotation object
row_ha <- rowAnnotation(
    Compartment = cell_test,
    col = list(
        Compartment = c(
            "Parenchymal" = "#E64B35",
            "Vascular" = "#4DBBD5"
        )
    ),
    show_annotation_name = FALSE,
    #annotation_name_side = "top",
    annotation_width = unit(0.1, "mm"),
    width = unit(1, "mm"),
    simple_anno_size = unit(1, "mm"),
    annotation_legend_param = list(labels_gp = gpar(fontsize = 6),
                                title_gp = gpar(fontsize = 7),
                                labels = c("P", "V"),
                                direction = "vertical",
                                title_position = "leftcenter-rot",
                                #title_position = "leftcenter",
                                #gap = unit(0.1, "mm"),
                                grid_width = unit(1, "mm"),
                                grid_height = unit(10, "mm"))
)

# Get the order of indices
order_indices <- match(rownames(heatmap_matrix_all), cell_groups)

# Reorder the vector
reordered_vector <- cell_groups[order_indices]

# Flip names and values
cell_test <- names(reordered_vector)
names(cell_test) <- reordered_vector

# Create the row annotation object
row_ha_all <- HeatmapAnnotation(
    Compartment = cell_test,
    which = "row",
    col = list(
        Compartment = c(
            "Parenchymal" = "#E64B35",
            "Vascular" = "#4DBBD5"
        )
    ),
    show_annotation_name = FALSE,
    #annotation_name_side = "top",
    annotation_width = unit(0.1, "mm"),
    width = unit(0.5, "mm"),
    simple_anno_size = unit(1, "mm"),
    annotation_legend_param = list(labels_gp = gpar(fontsize = 5),
                                title_gp = gpar(fontsize = 6),
                                labels = c("P", "V"),
                                direction = "vertical",
                                title_position = "leftcenter-rot",
                                #title_position = "leftcenter",
                                #gap = unit(0.1, "mm"),
                                grid_width= unit(1, "mm"),
                                grid_height = unit(10, "mm"))
)

# Create heatmap
deg_count_hmap <- Heatmap(heatmap_matrix,
        name = "Significant DEGs - average log2 foldchange > 1",
        #col = heatmap_colors,
        cluster_rows = FALSE,
        width= unit(6.5, "mm"),
        cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 7),
        column_names_gp = gpar(fontsize = 7),
        row_names_side = "left",
        column_names_rot = 90,
        column_names_centered = TRUE,
        row_names_centered = FALSE,
        right_annotation = row_ha,
        #column_title = "Cell Types",
        #row_title = "Direction of Differential Expression",
        cell_fun = function(j, i, x, y, width, height, fill) {
            grid.text(sprintf("%.0f", heatmap_matrix[i, j]), x, y,
                     gp = gpar(fontsize = 6))
        },
        heatmap_legend_param = list(labels_gp = gpar(fontsize = 5),
                                    title_gp = gpar(fontsize = 6),
                                    title_position = "leftcenter-rot",
                                    direction = "vertical",
                                    legend_width = unit(2, "mm"),
                                    row_gap = unit(14, "cm"),
                                    title_gap = unit(500, "mm"),
                                    #gap = unit(0.1, "mm"),
                                    grid_width = unit(1, "mm"),
                                    grid_height = unit(1, "mm")))
draw(deg_count_hmap, heatmap_legend_side = "right",
     align_heatmap_legend = "heatmap_center",
     show_annotation_legend = FALSE,
     show_heatmap_legend = FALSE,
     gap = unit(1, "mm"))
     #ht_gap = unit(100, "mm"))
p5 <- grid.grabExpr(draw(deg_count_hmap,
                         heatmap_legend_side = "right",
                         align_heatmap_legend = "heatmap_center",
     show_heatmap_legend = FALSE,
                         gap = unit(0.6, "mm")))
deg_count_hmap_all <- Heatmap(heatmap_matrix_all,
        name = "Significant DEGs",
        #col = heatmap_colors,
        cluster_rows = FALSE,
        width= unit(10, "mm"),
        cluster_columns = FALSE,
        row_names_gp = gpar(fontsize = 7),
        column_names_gp = gpar(fontsize = 7),
        row_names_side = "left",
        column_names_rot = 90,
        column_names_centered = FALSE,
        row_names_centered = FALSE,
        #column_title = "Cell Types",
        #row_title = "Direction of Differential Expression",
        cell_fun = function(j, i, x, y, width, height, fill) {
            grid.text(sprintf("%.0f", heatmap_matrix_all[i, j]), x, y,
                     gp = gpar(fontsize = 6))
        },
        heatmap_legend_param = list(labels_gp = gpar(fontsize = 5),
                                    title_gp = gpar(fontsize = 6),
                                    labels = c("0", "1K", "2K", "3K"),
                                    direction = "vertical",
                                    legend_width = unit(9, "mm"),
                                    title_position = "leftcenter-rot",
                                    title_gap = unit(0.1, "mm"),
                                    gap = unit(0, "mm"),
                                    grid_width = unit(1, "mm"),
                                    legend_height = unit(20, "mm"),
                                    grid_height = unit(1, "mm")))
draw(deg_count_hmap_all + row_ha_all, heatmap_legend_side = "right",
     align_heatmap_legend = "heatmap_center", gap = unit(0.6, "mm"),
#ht_gap = unit(0.5, "mm"),
     legend_border = unit(10, "cm"))
p6 <- grid.grabExpr(draw(deg_count_hmap_all + row_ha_all,
                         heatmap_legend_side = "right",
                         align_heatmap_legend = "heatmap_center",
                         gap = unit(0.6, "mm")))
```


### GO summary

#### Hypergeometric GO

```{r}
file <- here::here("03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/go-hypergeometic_subtypes_no-apoe_deg_direction.qs")

# add change based on logfc filter
filter_logfc <- TRUE
file <- here::here("03_data/990_processed_data/001_snrnaseq/09_pathway_analysis")

if (filter_logfc) {
  file <- here::here(file, "go-hypergeometic_subtypes_deg_direction_mast_level2_logfc_filter.qs")
} else {
  file <- here::here(file, "go-hypergeometic_subtypes_deg_direction_mast_level2.qs")
}
result_list <- qs::qread(file)

result_list <- map2(result_list, names(result_list), ~ tibble::as_tibble(.x) |> dplyr::mutate(celltype = .y)) |>
  list_rbind()

result_list |>
  dplyr::filter(grepl("amyloid", Description))


# Remove terms that only have one gene
result_list <- result_list |>
  dplyr::filter(Count > 1)

if (filter_logfc) {
  file <- here::here("03_data/999_data_for_publication/deg_similarity_matrix_filtered.qs")
  file_score <- here::here("03_data/999_data_for_publication/deg_similarity_matrix_filtered_scores.qs")
} else {
  file <- here::here("03_data/999_data_for_publication/deg_similarity_matrix.qs")
  file_score <- here::here("03_data/999_data_for_publication/deg_similarity_matrix_scores.qs")
}

result_list_ont <- split(result_list, result_list$ONTOLOGY)

if (!file.exists(file)) {
  simMatrix <- map2(
    result_list_ont,
    names(result_list_ont),
    ~ calculateSimMatrix(
      unique(.x$ID),
      orgdb = "org.Hs.eg.db",
      ont = .y,
      method = "Rel"
    )
  )
  # make sure the scores match the simMatrix
  x <- map2(result_list_ont, simMatrix, ~ .x |>
    dplyr::distinct(ID, .keep_all = TRUE) |>
    dplyr::filter(ID %in% colnames(.y)))

  scores <- map(x, ~ setNames(-log(.x$p.adjust), .x$ID))


  # assert that the score go terms match the simmatrix rownames
  assertthat::are_equal(sum(map2_dbl(
    scores, simMatrix, ~ sum(names(.x) != rownames(.y))
  )), 0)
  # Assert that there are no missing values in the scores
  assertthat::assert_that(sum(map_dbl(scores, ~ sum(is.na(.x)))) == 0)
  qs::qsave(simMatrix, file)
  qs::qsave(scores, file_score)
}

# Read the sim matrix
simMatrix <- qs::qread(file)
scores <- qs::qread(file_score)
```

```{r}
reducedTerms <- map2(simMatrix, scores, ~ reduceSimMatrix(.x,
  .y,
  threshold = 0.9,
  orgdb = "org.Hs.eg.db"
))
reducedTerms$BP[1:5, 1:10]
```

```{r}
# Add the rrvgo reduced terms to the initial dataframe
result_list_reduced <- map2(reducedTerms, names(reducedTerms), ~ .x |>
       dplyr::mutate(ONTOLOGY = .y)) |>
  list_rbind() %>%
  dplyr::left_join(result_list, ., by = join_by(ID == go, ONTOLOGY)) |>
  dplyr::mutate(deg_direction = if_else(grepl("_Down", celltype), "Down", "Up"),
                celltype = str_remove(celltype, "_Down|_Up")) |>
  dplyr::left_join(celltype_abbrev_df, by = join_by(celltype)) |>
  dplyr::mutate(celltypes_abbrev = if_else(is.na(celltypes_abbrev), celltype, celltypes_abbrev))

# Get genes from pathways of interest
erythro_genes <- result_list_reduced |>
  dplyr::filter(parentTerm == "positive regulation of erythrocyte differentiation") |>
  dplyr::pull(geneID) |>
  strsplit("/") |>
  unlist() |>
  unique()
protein_refold <- result_list_reduced |>
  dplyr::filter(parentTerm == "protein refolding") |>
  dplyr::pull(geneID) |>
  strsplit("/") |>
  unlist() |>
  unique()
# All genes overlap...
sum(!protein_refold %in% erythro_genes)

go_genes <- data.frame(summarised_pathway = "positive regulation of erythrocyte differentiation",
  genes = erythro_genes, celltypes = "ex-neuro-6_oligo-b_arterial_capillary") |>
    rbind(data.frame(summarised_pathway = "protein refolding", genes = protein_refold,
      celltypes = "ex-neuro-6_oligo-b_arterial_venous"))
#readr::write_csv(go_genes, here("go_pathway_genes.csv"))

result_list_reduced <- result_list_reduced |>
  dplyr::select(ONTOLOGY, celltypes_abbrev, parentTerm, deg_direction) |>
  dplyr::distinct() |>
  # add a counter for terms that appear in more than 1 celltype
  dplyr::mutate(count_overlap = n(), .by = c(ONTOLOGY, deg_direction, parentTerm)) |>
  dplyr::filter(!is.na(parentTerm))

result_list_reduced <- result_list_reduced |>
  dplyr::mutate(group = factor(case_when(
    grepl("endothelial", parentTerm) ~ "Endothelial",
    grepl("immune|cytokine|virus|viral", parentTerm) ~ "Immune",
    grepl("protein", parentTerm) ~ "Protein",
    .default = "Other terms"
  ), levels = c("Endothelial", "Immune", "Protein", "Other terms")), fraction = if_else(celltypes_abbrev %in% c("Arterial", "Arteriolar-SMC", "Capillary", "EndoMT-1",
  "M-Pericyte", "Pericyte-1", "Pericyte-2", "Perivascular-FB-1", "T-Pericyte", "Vascular-SMC-1",
  "Venous", "T-cell-V"), "Vascular", "Parenchymal"))

# Wrap the GO terms in the data
result_list_reduced$parentTerm <- wrap_text(result_list_reduced$parentTerm, 24)

# Save difference versions
file_base <- "03_data/999_data_for_publication"
if(filter_logfc) {
  qs::qsave(result_list_reduced, here::here(file_base, "deg_reduced_terms_filtered.qs"))
} else {
  qs::qsave(result_list_reduced, here::here(file_base, "deg_reduced_terms.qs"))
}

# Read in both versions
result_list_reduced <- qs::qread(here::here(file_base, "deg_reduced_terms.qs"))
result_list_reduced_filtered <- qs::qread(here::here(file_base, "deg_reduced_terms_filtered.qs"))
```

```{r}
hyper_go_plot_all <- result_list_reduced |>
  ggplot(aes(x = celltypes_abbrev, y = reorder(parentTerm, count_overlap, sum), fill = deg_direction)) +
  geom_tile() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 6),
        axis.text.y = element_text(size = 6)) +
  facet_grid(rows = vars(ONTOLOGY)) +
  labs(x = "", y = "")

hyper_go_plot_bp <- result_list_reduced |>
  dplyr::filter(ONTOLOGY == "BP") |>
  ggplot(aes(x = celltypes_abbrev, y = reorder(parentTerm, count_overlap, sum), fill = deg_direction)) +
  geom_tile() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5,
                                   size = 7),
        axis.text.y = element_text(size = 6),
        strip.background = element_blank(),
        #strip.text = element_text(face = "bold", size = 6),
        panel.grid.major = element_line(color = "gray80", linewidth = 0.5),
        plot.title = element_text(size = 10)) +
  labs(x = "", y = "", fill = "Expression") +
  facet_grid(group ~ fraction, scales = "free", space = "free")

result_list_reduced_filtered$parentTerm <- wrap_text(result_list_reduced_filtered$parentTerm, 28)
hyper_go_plot_bp_filtered <- result_list_reduced_filtered |>
  dplyr::filter(ONTOLOGY == "BP") |>
  ggplot(aes(x = celltypes_abbrev, y = reorder(parentTerm, count_overlap, sum), fill = deg_direction)) +
  geom_tile() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5,
                                   size = 7),
        axis.text.y = element_text(size = 6),
        strip.background = element_blank(),
        #strip.text = element_text(face = "bold", size = 6),
        panel.grid.major = element_line(color = "gray80", linewidth = 0.5),
        plot.title = element_text(size = 10)) +
  labs(x = "", y = "", fill = "Expression") +
  facet_grid(cols = vars(fraction), scales = "free", space = "free")

hyper_go_plot_all
hyper_go_plot_bp
hyper_go_plot_bp_filtered

ggsave(here::here("05_figures/990_shared_figures/003_final_figures/03_fig_deg/hyper_go_terms_all.png"), hyper_go_plot_all, height = 45, width = 12)

ggsave(here::here("05_figures/990_shared_figures/003_final_figures/03_fig_deg/hyper_go_terms_bp.png"), hyper_go_plot_bp, height = 8, width = 6)
```

#### GSEA results

```{r}
#| eval: false
file <-
  here::here(
    "03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/gene-set-enrichment_go_subtypes_no-apoe.qs"
  )
result_list <- qs::qread(file)

# Remove any NULL entries
result_list <- result_list[!sapply(result_list, is.null)]
# Assert that there are no nulls in list
assertthat::see_if(sum(map_lgl(result_list, is.null)) == 0)
# remove any with no pathways found
gse <- result_list %>%
  purrr::keep(~ nrow(.) > 0)

gse <- gse[!grepl("ambiguous_|low-feature-cells", names(gse))]

amyloid_hits <- map2(gse, names(gse), ~ .x@result |>
       dplyr::mutate(celltype = .y)) |>
  list_rbind() |>
  dplyr::filter(grepl("amyloid", Description))
table(amyloid_hits$celltype)
# treeplots
gse <- map(gse, pairwise_termsim)

go_data <- lapply(names(gse), function(celltype) {
  res <- gse[[celltype]]
  if (!is.null(res)) {
    data.frame(celltype = celltype,
               ID = res$ID,
               Description = res$Description,
               p.adjust = res$p.adjust,
               enrichmentScore = res$enrichmentScore,
               Count = res$setSize)
  } else {
    NULL
  }
}) |> bind_rows() |>
  dplyr::mutate(ontology = case_when(
    grepl("_BP", celltype) ~ "BP",
    grepl("_CC", celltype) ~ "CC",
    grepl("_MF", celltype) ~ "MF",
    .default = NA
  ))

# Get the GO IDs for revigo
go_data_ids <- go_data |>
  dplyr::group_by(ontology) |>
  tidyr::nest()
go_data_ids <- map(go_data_ids$data, dplyr::pull, ID) |>
  set_names(go_data_ids$ontology)
```

### Heatmap of shared DEGs

```{r}
# Create a wide format dataframe to count overlaps
wide_df <- res_sig |>
  dplyr::select(gene, celltypes_abbrev, deg_direction) |>
  dplyr::mutate(value = 1) |>
  unique() |>
  pivot_wider(names_from = celltypes_abbrev,
              values_from = value, values_fill = 0)

# Convert the wide format dataframe to a matrix
gene_matrix <- as.matrix(wide_df[,-c(1,2)])

# Calculate the overlap matrix using matrix multiplication
overlap_matrix <- t(gene_matrix) %*% gene_matrix

# Set the row and column names of the overlap matrix
rownames(overlap_matrix) <- colnames(gene_matrix)
colnames(overlap_matrix) <- colnames(gene_matrix)

# Plot the heatmap
pheatmap(overlap_matrix,
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         display_numbers = TRUE,
         main = "Overlap of DEGs Across Cell Types",
         color = colorRampPalette(c("white", "blue"))(50))
```

```{r}
# List of all cell types
all_celltypes <- unique(res_sig$celltype)

# Filter the dataframe for upregulated and downregulated genes
up_df <- res_sig %>% dplyr::filter(deg_direction == "Up")
down_df <- res_sig %>% dplyr::filter(deg_direction == "Down")

# Create a dataframe with all combinations of genes and celltypes, setting missing values to 0
all_genes <- unique(c(up_df$gene, down_df$gene))

# Create wide format dataframes for upregulated and downregulated genes
wide_up_df <- up_df %>%
  dplyr::select(gene, celltype) %>%
  dplyr::distinct() %>%
  dplyr::mutate(value = 1) %>%
  complete(gene = all_genes, celltype = all_celltypes, fill = list(value = 0)) %>%
  pivot_wider(names_from = celltype, values_from = value, values_fill = list(value = 0))

wide_down_df <- down_df %>%
  dplyr::select(gene, celltype) %>%
  dplyr::distinct() %>%
  dplyr::mutate(value = 1) %>%
  complete(gene = all_genes, celltype = all_celltypes, fill = list(value = 0)) %>%
  pivot_wider(names_from = celltype, values_from = value, values_fill = list(value = 0))

# Convert the wide format dataframes to matrices
gene_matrix_up <- as.matrix(wide_up_df[,-1])
gene_matrix_down <- as.matrix(wide_down_df[,-1])

# Calculate the overlap matrices using matrix multiplication
overlap_matrix_up <- t(gene_matrix_up) %*% gene_matrix_up
overlap_matrix_down <- t(gene_matrix_down) %*% gene_matrix_down

# Set the row and column names of the overlap matrices
rownames(overlap_matrix_up) <- colnames(gene_matrix_up)
colnames(overlap_matrix_up) <- colnames(gene_matrix_up)

rownames(overlap_matrix_down) <- colnames(gene_matrix_down)
colnames(overlap_matrix_down) <- colnames(gene_matrix_down)

# Apply log transformation to normalize the data
log_transform <- function(matrix) {
  log1p(matrix)  # log1p is log(1 + x), which handles zero values
}

overlap_matrix_up_log <- log_transform(overlap_matrix_up)
overlap_matrix_down_log <- log_transform(overlap_matrix_down)
require(circlize)
# Define custom color functions for the heatmaps
col_fun_up <- colorRamp2(c(0, max(overlap_matrix_up_log)), c("white", "red"))
col_fun_down <- colorRamp2(c(0, max(overlap_matrix_down_log)), c("white", "blue"))

# Create the heatmaps
ht_up <- Heatmap(overlap_matrix_up_log, name = "Upregulated", col = col_fun_up, show_row_names = TRUE, show_column_names = TRUE)
ht_down <- Heatmap(overlap_matrix_down_log, name = "Downregulated", col = col_fun_down, show_row_names = TRUE, show_column_names = TRUE)

# Combine the heatmaps into a single plot
draw(ht_up + ht_down, heatmap_legend_side = "right")
```

#### Jaccard similarity index

```{r}
library(vegan)
compute_jaccard_similarity <-
  function(df,
           deg_filter = NULL,
           col_title = "All DEGs") {
    if (!is.null(deg_filter)) {
      df <- df |>
        dplyr::filter(deg_direction == deg_filter)
    }

    # Create a binary matrix of genes vs cell types
    binary_matrix <- df %>%
      dplyr::select(gene, celltypes_abbrev) %>%
      dplyr::distinct() %>%
      dplyr::mutate(present = 1) %>%
      pivot_wider(
        names_from = celltypes_abbrev,
        values_from = present,
        values_fill = list(present = 0)
      )

    # Transpose the matrix to have cell types as rows
    binary_matrix_t <- as.data.frame(t(binary_matrix[-1]))
    colnames(binary_matrix_t) <- binary_matrix$gene

    # Calculate Jaccard similarity matrix
    jaccard_matrix <- vegdist(binary_matrix_t, method = "jaccard")

    # Perform hierarchical clustering
    hc <- hclust(jaccard_matrix, method = "average")

    # Convert distance matrix to similarity matrix for heatmap
    similarity_matrix <- 1 - as.matrix(jaccard_matrix)

    format_cell_values <- function(x) {
      if (x >= 0.15) {
        return(sprintf("%.2f", x))
      } else {
        return("")
      }
    }
    # Set colour to make it more obvious which cells have a similarity of 0.2 or more
    col_fun <- colorRamp2(c(0, 0.15, 1), c("white", "blue", "red"))

    # Create the heatmap
    jaccard_all_hmap <- Heatmap(
      similarity_matrix,
      name = "Jaccard\nSimilarity",
      col = col_fun,
      column_title = col_title,
      column_title_gp = gpar(fontsize = 8),
      cell_fun = function(j, i, x, y, width, height, fill) {
        grid.rect(x, y, width, height, gp = gpar(col = "grey", fill = NA))
        value <- similarity_matrix[i, j]
        if (value >= 0.40 & value != 1) {
          #grid.text(sprintf("%.2f", value),
          grid.text("*",
                    x,
                    y,
                    gp = gpar(col = "black", fontsize = 10))
        }
      },
      show_row_dend = FALSE,
      show_column_dend = FALSE,
      row_names_gp = gpar(fontsize = 6),
      column_names_gp = gpar(fontsize = 6),
      column_names_centered = TRUE,
      heatmap_legend_param = list(#direction = "horizontal",
                                  labels_gp = gpar(fontsize = 6),
                                  #grid_height = unit(0.5, "mm"),
                                  #title_position = "leftcenter"),
                                  title_gp = gpar(fontsize = 6))
    )

    return(jaccard_all_hmap)
  }

# Count unique genes per celltype
celltype_gene_count <- res_sig |>
  summarise(unique_gene_count = n_distinct(gene), .by = celltypes_abbrev) |>
  dplyr::filter(unique_gene_count >= 30)

# Filter original dataframe to keep only these celltypes
res_sig_sub <- res_sig |>
  dplyr::filter(celltypes_abbrev %in% celltype_gene_count$celltypes_abbrev)
jaccard_all_hmap <- compute_jaccard_similarity(res_sig_sub)
jaccard_all_hmap

jaccard_up_hmap <- compute_jaccard_similarity(res_sig_sub, deg_filter = "Up",
                                              "Up DEGs")
jaccard_down_hmap <- compute_jaccard_similarity(res_sig_sub,
                                                deg_filter = "Down", "Down DEGs")

# Save plots
png(here::here("05_figures/990_shared_figures/003_final_figures",
      "jaccard_heatmap_all_degs.png"), width = 4000, height = 4000, res = 300)
jaccard_all_hmap
dev.off()
png(here::here("05_figures/990_shared_figures/003_final_figures",
      "jaccard_heatmap_up_degs.png"), width = 4000, height = 4000, res = 300)
jaccard_up_hmap
dev.off()
png(here::here("05_figures/990_shared_figures/003_final_figures",
      "jaccard_heatmap_down_degs.png"), width = 4000, height = 4000, res = 300)
jaccard_down_hmap
dev.off()

jaccard_up_hmap <- compute_jaccard_similarity(res_sig_sub, deg_filter = "Up",
                                              "Up DEGs")
jaccard_down_hmap <- compute_jaccard_similarity(res_sig_sub,
                                                deg_filter = "Down",
                                                "Down DEGs")
```

```{r}
# Create a binary matrix of genes vs cell types
binary_matrix <- res_sig %>%
  dplyr::select(gene, celltypes_abbrev) %>%
  dplyr::distinct() %>%
  dplyr::mutate(present = 1) %>%
  pivot_wider(names_from = celltypes_abbrev, values_from = present,
              values_fill = list(present = 0))

# Transpose the matrix to have cell types as rows
binary_matrix_t <- as.data.frame(t(binary_matrix[-1]))
colnames(binary_matrix_t) <- binary_matrix$gene

# Calculate Jaccard similarity matrix
jaccard_matrix <- vegdist(binary_matrix_t, method = "jaccard")

# Perform hierarchical clustering
hc <- hclust(jaccard_matrix, method = "average")

# Convert distance matrix to similarity matrix for heatmap
similarity_matrix <- 1 - as.matrix(jaccard_matrix)

# Set colour to make it more obvious which cells have a similarity of 0.2 or more
col_fun <- colorRamp2(c(0, 0.2, 1), c("white", "blue", "red"))
Heatmap(similarity_matrix, col = col_fun)
```

### Overlapping gene plot

There're some blocks of overlaping genes from the Jaccard plots, I'll try looking at those, maybe do an update plots for the two main blocks of celltypes

```{r}
blocks <- list(
  "block1" = c(
    "EndoMT",
    "T-Pericyte",
    "M-Pericyte",
    "Pericyte-1",
    "Pericyte-2",
    "Arteriolar-SMC"
  ),
  "block2" = c("Oligo-A", "Oligo-B", "Astro-A", "Astro-Q")
)

get_overlapping_genes <- function(celltype_subset, dataframe) {
  df <- dataframe |>
    dplyr::filter(celltypes_abbrev %in% celltype_subset) |>
    dplyr::group_by(deg_direction) |>
    tidyr::nest()

  # Prepare data for UpSet plot
  genes <- map(df$data, ~ .x |>
    mutate(present = 1) |>
      dplyr::select(!c(baseMean:celltype)) |>
    pivot_wider(
      names_from = celltypes_abbrev,
      values_from = present,
      values_fill = 0
    )) |>
    set_names(df$deg_direction)

  return(genes)

  # # Create lists of genes for each cell type
  # gene_celltype_list <- map(df$data, ~ split(.x$gene, .x$celltype)) |>
  #   set_names(df$deg_direction)
  #
  # # Convert lists to binary matrices
  # matrix <- map(gene_celltype_list, UpSetR::fromList)
}

upset_matrix <- map(blocks, get_overlapping_genes, res_sig)

plots <- map(upset_matrix, ~ {
  # Create UpSet plots
  # Upregulated genes UpSet plot
  upset_up <- upset(.x$Up, colnames(.x$Up)[-1], name = "Upregulated Genes")
  upset_down <- upset(.x$Down, colnames(.x$Down)[-1], name = "Downregulated Genes")
  return(list("up" = upset_up, "down" = upset_down))
})

plots <- map(plots, ~ .x$up | .x$down)

plots$block1
ggsave(here("05_figures/990_shared_figures/003_final_figures/overlapping_degs_jaccard_block1.png"), height = 10, width = 22)
plots$block2
ggsave(here("05_figures/990_shared_figures/003_final_figures/overlapping_degs_jaccard_block2.png"), height = 10, width = 22)
```

### GOs of overlaps

```{r}
get_gene_celltype_presence <- function(celltype_subset, dataframe) {
  # Group by gene and summarize cell types into a single column
  gene_celltypes <- dataframe |>
    dplyr::filter(celltypes_abbrev %in% celltype_subset) |>
    summarize(celltypes = paste(unique(celltype), collapse = ", "),
              .by = c(deg_direction, gene))
  return(gene_celltypes)
}
celltype_gene_overlap <- map(blocks, get_gene_celltype_presence, res_sig)

# get the genes for each group
genes <- map(celltype_gene_overlap, ~ {
  df <- .x |>
    dplyr::group_by(deg_direction, celltypes) |>
    tidyr::nest()
  genes <- map(df$data, ~ .x |>
                 dplyr::pull(gene)) |>
    set_names(paste0(df$deg_direction, "_", df$celltypes))
})
genes <- purrr::list_flatten(genes)
```

```{r}
plan("multisession", workers = parallel::detectCores() - 1)
go_results_bp <- future_map(genes, ~ enrichGO(gene = .x, OrgDb = "org.Hs.eg.db",
                                    keyType = "SYMBOL", ont = "BP"),
                            .options = furrr_options(seed = 123))
names(go_results_bp) <- paste0(names(go_results_bp), "_bp")
go_results_cc <- future_map(genes, ~ enrichGO(gene = .x, OrgDb = "org.Hs.eg.db",
                                    keyType = "SYMBOL", ont = "CC"),
                            .options = furrr_options(seed = 123))
names(go_results_cc) <- paste0(names(go_results_cc), "_cc")
go_results_mf <- future_map(genes, ~ enrichGO(gene = .x, OrgDb = "org.Hs.eg.db",
                                    keyType = "SYMBOL", ont = "MF"),
                            .options = furrr_options(seed = 123))
names(go_results_mf) <- paste0(names(go_results_mf), "_mf")
go_results <- c(go_results_bp, go_results_cc, go_results_mf)
rm(go_results_bp, go_results_cc, go_results_mf)
```

```{r}
# exclude cases where no pathways were found
go_results_filtered <- go_results |>
  purrr::keep(~ !is.null(.)) |>
  purrr::keep(~ nrow(.) > 0)

go_results_filtered <- map(go_results_filtered, ~ .x |>
  # remove instance of just one gene or fewer
           dplyr::filter(Count > 1)) |>
  purrr::keep(~ nrow(.) > 0)

# treeplots
go_results_filtered <- map(go_results_filtered, pairwise_termsim)

# Define a safe version of your treeplot function that returns NULL on error
safe_treeplot <- possibly(treeplot, otherwise = NULL)
plan("multisession", workers = parallel::detectCores() - 1)
treeplots <- future_map(go_results_filtered, safe_treeplot)
# Remove any NULL entries
treeplots <- treeplots[!sapply(treeplots, is.null)]
# Assert that there are no nulls in list
assertthat::see_if(sum(map_lgl(treeplots, is.null)) == 0)
# Replace those with no treeplot with dotplots
no_treeplot <- names(go_results_filtered)[!names(go_results_filtered) %in%
                                            names(treeplots)]
dotplots <- map(go_results_filtered[no_treeplot], dotplot)
assertthat::see_if(sum(map_lgl(dotplots, is.null)) == 0)
treeplots <- c(treeplots, dotplots)

treeplots <- map2(treeplots, names(treeplots), ~ .x + ggtitle(.y))

# Define the function to subset a single string up to the last underscore
subset_until_last_underscore <- function(input_string) {
  # Find the position of the last underscore
  last_underscore_position <- str_locate_all(input_string, "_")[[1]][, 1] %>% max()

  # Subset the string up to the last underscore
  if (!is.na(last_underscore_position)) {
    return(str_sub(input_string, 1, last_underscore_position - 1))
  } else {
    return(input_string)
  }
}

# Apply the function to each element of the vector
celltypes <- sapply(names(treeplots), subset_until_last_underscore) |>
  unique()


merge_plots <- function(celltype, dotplot_list) {
  # subset to relevant plots
  plots <- dotplot_list[grepl(celltype, names(dotplot_list))]
  # Merge plots
  #plot <- plots[[1]] + plots[[2]] + plots[[3]]
  plot <- purrr::reduce(plots, `+`)
}

plots <- map(celltypes, merge_plots, treeplots) |> set_names(celltypes)

plots$`block1_Up_Pericyte-2, T-Pericyte`

# Save plots
map2(plots, names(plots), ~ ggsave(here::here("05_figures/990_shared_figures/002_pseudobulk/pathways/jaccard_overlapping", paste0(.y, ".png")), plot = .x, width = 32, height = 10))
```


### Assemble Figure

- NOTES:
  - Make a heatmap of the shared DEGs (rows and columns celltypes)

```{r}
p2 = grid.grabExpr(draw(jaccard_up_hmap))
p3 = grid.grabExpr(draw(jaccard_down_hmap))
row_2 <- plot_grid(p2, p3, labels = c("B", "C"), label_colour = "black")

final_plot3 <- plot_grid(p5,
                         p2, p3, ncol = 1,
                         rel_heights = c(2, 1, 1),
                         rel_widths = c(1, 1, 2),
                         labels = c("A", "B", "C"),
                         label_colour = "black")

deg_heatmaps <- plot_grid(p5, p6, labels = c("A", "B"), label_colour = "black",
                          rel_widths = c(1, 1.4))
final_plot3 <- plot_grid(deg_heatmaps, hyper_go_plot_bp_filtered +
                           theme(legend.position = "none"), ncol = 1,
                         rel_heights = c(1, 1.2),
                         labels = c("", "C"),
                         label_colour = "black")

hyper_go_plot_bp <- hyper_go_plot_bp +
  theme(legend.key.size = unit(0.3, "cm"),  # Smaller legend symbols
        legend.key.spacing = unit(0.3, "cm"),
        legend.text = element_text(size = 8),
        legend.title = element_text(size = 9),
        legend.spacing.y = unit(5.9, 'cm'),
        #legend.justification.top =
        legend.position = "bottom")

final_plot3 <- plot_grid(final_plot3, hyper_go_plot_bp,
                         rel_widths = c(1, 1.3),
                         labels = c("", "D"),
          label_colour = "black")

# TODO: try and change font to fix ggsave/base pdf function
# ggsave(filename = here::here("05_figures/990_shared_figures/003_final_figures",
#                              "figure3.pdf"), plot = final_plot3,
#        width = 7, height = 9, units = "in")
cairo_pdf(file = here::here("05_figures/990_shared_figures/003_final_figures", "figure3.pdf"),
         width = 7, height = 9)
print(final_plot3)
dev.off()
```

## Figure 4 - cellchat

```{r}
# differential number of interactions
# Note that red colours denote increased signalling in the second dataset compared to the first, and blue the inverse.
# In our case cases are the first group and controls the second.
img1 <- image_read_svg(here::here("05_figures/990_shared_figures/cellchat/differential_heatmap.svg"))
img1_gg <- ggdraw() + draw_image(img1)
# relative information flow
img2 <- image_read_svg(here::here("05_figures/990_shared_figures/cellchat/differential_information_flow_stacked.svg"))
img2_gg <- ggdraw() + draw_image(img2)
# Overall signalling
img3 <- image_read_svg(here::here("05_figures/990_shared_figures/cellchat/differential_heatmap_overall.svg"))
img3_gg <- ggdraw() + draw_image(img3)
# example bubble plot
img4 <- image_read(here::here("05_figures/990_shared_figures/cellchat",
                              "003_differential_bubble_plots",
                              "EndoMT-1_differential_signalling.png"))
img4_gg <- ggdraw() + draw_image(img4)

# Combine everything
final_plot <- plot_grid(img1_gg, img4_gg, img3_gg, ncol = 1, labels = c("A", "B", "C"), label_colour = "black")
final_plot

ggsave(filename = here::here("05_figures/990_shared_figures/003_final_figures/figure4.png"), plot = final_plot,
       width = 12, height = 16, units = "in", dpi = 300)
```
