<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.54">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>Pathway analysis - MAGMA genes</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Pathway analysis - MAGMA genes</h6>

            <a href="../../04_data_analysis/007_magma/22_pathway_analysis.qmd" class="btn btn-primary quarto-download-embed" download="22_pathway_analysis.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Pathway analysis - MAGMA genes</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in data</a></li>
  <li><a href="#data-prep" id="toc-data-prep" class="nav-link" data-scroll-target="#data-prep">Data prep</a></li>
  <li><a href="#hypergeometric-enrichment" id="toc-hypergeometric-enrichment" class="nav-link" data-scroll-target="#hypergeometric-enrichment">Hypergeometric enrichment</a>
  <ul class="collapse">
  <li><a href="#save-for-ipa" id="toc-save-for-ipa" class="nav-link" data-scroll-target="#save-for-ipa">Save for IPA</a></li>
  </ul></li>
  <li><a href="#gene-set-enrichment-analysis-gsea" id="toc-gene-set-enrichment-analysis-gsea" class="nav-link" data-scroll-target="#gene-set-enrichment-analysis-gsea">Gene set enrichment analysis (GSEA)</a>
  <ul class="collapse">
  <li><a href="#run-gsea" id="toc-run-gsea" class="nav-link" data-scroll-target="#run-gsea">Run GSEA</a></li>
  <li><a href="#gene-ontology" id="toc-gene-ontology" class="nav-link" data-scroll-target="#gene-ontology">Gene Ontology</a>
  <ul class="collapse">
  <li><a href="#insulin-pathways" id="toc-insulin-pathways" class="nav-link" data-scroll-target="#insulin-pathways">Insulin pathways</a></li>
  <li><a href="#plots" id="toc-plots" class="nav-link" data-scroll-target="#plots">Plots</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/007_magma/22_pathway_analysis.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad</code></pre>
</div>
</div></div>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, smiths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'kableExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    group_rows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'janitor'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    chisq.test, fisher.test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    melt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rngtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  To enable shared memory capabilities, try: install.extras('
NMF
')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'future'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    run</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Some features are not enabled in this build of Arrow. Run `arrow_info()` for more information.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'arrow'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    timestamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'viridis'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    viridis_pal</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 methods overwritten by 'proxy':
  method               from    
  print.registry_field registry
  print.registry_entry registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    nrun</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:webshot2':

    resize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built under R 4.4.0 but the current version is
4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI
for R may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'DESeq2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    fpkm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: limma</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DESeq2':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    plotMDS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'edgeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SingleCellExperiment':

    cpm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocParallel'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    register</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation("pathview") within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationFilter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationFilter'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:arrow':

    field</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:future':

    value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ensembldb'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggrepel</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'PCAtools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scran':

    parallelPCA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    biplot, screeplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Seurat':

    components</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    normalize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:future':

    %-&gt;%, %&lt;-%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:NMF':

    algorithm, compare</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'biomaRt'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scRNAseq':

    listDatasets, searchDatasets</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>clusterProfiler v4.12.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use clusterProfiler in published research, please cite:
T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'clusterProfiler'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>DOSE v3.30.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'enrichplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    color_palette</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pals':

    alphabet</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocIO</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rtracklayer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocIO':

    FileForFormat</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    blocks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationHub':

    hubUrl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh37'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'qusage'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    read.gmt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:SNPlocs.Hsapiens.dbSNP155.GRCh37':

    .loadAlleles, .loadLoc, COMPATIBLE_BSGENOMES</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'GenomicFiles'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
simona version 1.2.0
Bioconductor page: http://bioconductor.org/packages/simona/
Github page: https://github.com/jokergoo/simona
Documentation: https://jokergoo.github.io/simona/

If you use it in published research, please cite:
Gu, Z. simona: a Comprehensive R package for Semantic Similarity 
  Analysis on Bio-Ontologies. bioRxiv 2023.

This message can be suppressed by:
  suppressPackageStartupMessages(library(simona))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'ggnetwork':
  method         from  
  fortify.igraph ggtree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'CellChat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scater':

    runPCA, runUMAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
</div></div>
</section>
<section id="read-in-data" class="level1">
<h1>Read in data</h1>
<p>Here we’re looking at the significant celltypes from MAGMA again AD in all controls. The genes are the genes MAGMA found to be significant after correction for each celltype. Note that each celltype was adjusted separately and via Bonferroni.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="co"># read data of sig magma genes from sig celltypes for AD all controls</span></span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/results/sig_celltype_sig_genes.tsv"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltype =</span> SetName)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 169727 Columns: 24
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (10): SetName, input_gene_list, gwas_background, input_filter, celltype_...
dbl (14): GENE, CHR, START, STOP, NSNPS, NPARAM, N, ZSTAT, P, ZFITTED_BASE, ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- readr::read_tsv(here::here("03_data/994_magma_inputs/sig_genelist", "all_sig_magma_set_genes.tsv")) |&gt;</span></span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  dplyr::rename(celltype = SetName)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="data-prep" class="level1">
<h1>Data prep</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter to sig genes</span></span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a>res_sig <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res, padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> </span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a>                           input_gene_list <span class="sc">==</span> <span class="st">"all_controls"</span> <span class="sc">&amp;</span> </span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a>                           percent <span class="sc">==</span> <span class="st">"top-ten-percent"</span>)</span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-6"><a href="#cb202-6" aria-hidden="true" tabindex="-1"></a><span class="co"># get gene names per celltype</span></span>
<span id="cb202-7"><a href="#cb202-7" aria-hidden="true" tabindex="-1"></a>genes_df <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span></span>
<span id="cb202-8"><a href="#cb202-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype, gwas_background, celltype_level) <span class="sc">%&gt;%</span></span>
<span id="cb202-9"><a href="#cb202-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">list_of_columns =</span> <span class="fu">list</span>(hgnc_symbol))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'celltype', 'gwas_background'. You can
override using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> genes_df <span class="sc">|&gt;</span> </span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(list_of_columns)</span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genes) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(genes_df<span class="sc">$</span>celltype, <span class="st">"-"</span>, genes_df<span class="sc">$</span>gwas_background, <span class="st">"-"</span>, genes_df<span class="sc">$</span>celltype_level)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="hypergeometric-enrichment" class="level1">
<h1>Hypergeometric enrichment</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>go_results_bp <span class="ot">&lt;-</span> <span class="fu">map</span>(genes, <span class="sc">~</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> .x, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, </span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"BP"</span>))</span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_results_bp) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(go_results_bp), <span class="st">"_bp"</span>)</span>
<span id="cb205-4"><a href="#cb205-4" aria-hidden="true" tabindex="-1"></a>go_results_cc <span class="ot">&lt;-</span> <span class="fu">map</span>(genes, <span class="sc">~</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> .x, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, </span>
<span id="cb205-5"><a href="#cb205-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"CC"</span>))</span>
<span id="cb205-6"><a href="#cb205-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_results_cc) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(go_results_cc), <span class="st">"_cc"</span>)</span>
<span id="cb205-7"><a href="#cb205-7" aria-hidden="true" tabindex="-1"></a>go_results_mf <span class="ot">&lt;-</span> <span class="fu">map</span>(genes, <span class="sc">~</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> .x, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, </span>
<span id="cb205-8"><a href="#cb205-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"MF"</span>))</span>
<span id="cb205-9"><a href="#cb205-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(go_results_mf) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(go_results_mf), <span class="st">"_mf"</span>)</span>
<span id="cb205-10"><a href="#cb205-10" aria-hidden="true" tabindex="-1"></a>go_results <span class="ot">&lt;-</span> <span class="fu">c</span>(go_results_bp, go_results_cc, go_results_mf)</span>
<span id="cb205-11"><a href="#cb205-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(go_results_bp, go_results_cc, go_results_mf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="co"># exclude cases where no pathways were found</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a>go_results_filtered <span class="ot">&lt;-</span> go_results <span class="sc">|&gt;</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">keep</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">is.null</span>(.)) <span class="sc">|&gt;</span></span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">keep</span>(<span class="sc">~</span> <span class="fu">nrow</span>(.) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-6"><a href="#cb206-6" aria-hidden="true" tabindex="-1"></a><span class="co"># treeplots</span></span>
<span id="cb206-7"><a href="#cb206-7" aria-hidden="true" tabindex="-1"></a>go_results_filtered <span class="ot">&lt;-</span> <span class="fu">map</span>(go_results_filtered, pairwise_termsim)</span>
<span id="cb206-8"><a href="#cb206-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-9"><a href="#cb206-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a safe version of your treeplot function that returns NULL on error</span></span>
<span id="cb206-10"><a href="#cb206-10" aria-hidden="true" tabindex="-1"></a>safe_treeplot <span class="ot">&lt;-</span> <span class="fu">possibly</span>(treeplot, <span class="at">otherwise =</span> <span class="cn">NULL</span>)</span>
<span id="cb206-11"><a href="#cb206-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-12"><a href="#cb206-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plan</span>(<span class="st">"multisession"</span>, <span class="at">workers =</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>() <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb206-13"><a href="#cb206-13" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> <span class="fu">future_map</span>(go_results_filtered, safe_treeplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="sourceCode cell-code" id="cb237"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb237-1"><a href="#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any NULL entries                                 </span></span>
<span id="cb237-2"><a href="#cb237-2" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> treeplots[<span class="sc">!</span><span class="fu">sapply</span>(treeplots, is.null)]      </span>
<span id="cb237-3"><a href="#cb237-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Assert that there are no nulls in list                </span></span>
<span id="cb237-4"><a href="#cb237-4" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">see_if</span>(<span class="fu">sum</span>(<span class="fu">map_lgl</span>(treeplots, is.null)) <span class="sc">==</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace those with no treeplot with dotplots          </span></span>
<span id="cb239-2"><a href="#cb239-2" aria-hidden="true" tabindex="-1"></a>no_treeplot <span class="ot">&lt;-</span> <span class="fu">names</span>(go_results_filtered)[<span class="sc">!</span><span class="fu">names</span>(go_results_filtered) <span class="sc">%in%</span> </span>
<span id="cb239-3"><a href="#cb239-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">names</span>(treeplots)]</span>
<span id="cb239-4"><a href="#cb239-4" aria-hidden="true" tabindex="-1"></a>dotplots <span class="ot">&lt;-</span> <span class="fu">map</span>(go_results_filtered[no_treeplot], dotplot)                         </span>
<span id="cb239-5"><a href="#cb239-5" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">see_if</span>(<span class="fu">sum</span>(<span class="fu">map_lgl</span>(dotplots, is.null)) <span class="sc">==</span> <span class="dv">0</span>)                  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> <span class="fu">c</span>(treeplots, dotplots)                                      </span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> <span class="fu">map2</span>(treeplots, <span class="fu">names</span>(treeplots), <span class="sc">~</span> .x <span class="sc">+</span> <span class="fu">ggtitle</span>(.y))</span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the function to subset a single string up to the last underscore</span></span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>subset_until_last_underscore <span class="ot">&lt;-</span> <span class="cf">function</span>(input_string) {</span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find the position of the last underscore</span></span>
<span id="cb241-8"><a href="#cb241-8" aria-hidden="true" tabindex="-1"></a>  last_underscore_position <span class="ot">&lt;-</span> <span class="fu">str_locate_all</span>(input_string, <span class="st">"_"</span>)[[<span class="dv">1</span>]][, <span class="dv">1</span>] <span class="sc">%&gt;%</span> <span class="fu">max</span>()</span>
<span id="cb241-9"><a href="#cb241-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb241-10"><a href="#cb241-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the string up to the last underscore</span></span>
<span id="cb241-11"><a href="#cb241-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(last_underscore_position)) {</span>
<span id="cb241-12"><a href="#cb241-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">str_sub</span>(input_string, <span class="dv">1</span>, last_underscore_position <span class="sc">-</span> <span class="dv">1</span>))</span>
<span id="cb241-13"><a href="#cb241-13" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb241-14"><a href="#cb241-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(input_string)</span>
<span id="cb241-15"><a href="#cb241-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb241-16"><a href="#cb241-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb241-17"><a href="#cb241-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-18"><a href="#cb241-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each element of the vector</span></span>
<span id="cb241-19"><a href="#cb241-19" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">names</span>(treeplots), subset_until_last_underscore) <span class="sc">|&gt;</span></span>
<span id="cb241-20"><a href="#cb241-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb241-21"><a href="#cb241-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-22"><a href="#cb241-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-23"><a href="#cb241-23" aria-hidden="true" tabindex="-1"></a>merge_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(celltype, dotplot_list) {</span>
<span id="cb241-24"><a href="#cb241-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subset to relevant plots</span></span>
<span id="cb241-25"><a href="#cb241-25" aria-hidden="true" tabindex="-1"></a>  plots <span class="ot">&lt;-</span> dotplot_list[<span class="fu">grepl</span>(celltype, <span class="fu">names</span>(dotplot_list))]</span>
<span id="cb241-26"><a href="#cb241-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge plots</span></span>
<span id="cb241-27"><a href="#cb241-27" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> plots[[<span class="dv">1</span>]] <span class="sc">+</span> plots[[<span class="dv">2</span>]] <span class="sc">+</span> plots[[<span class="dv">3</span>]]</span>
<span id="cb241-28"><a href="#cb241-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb241-29"><a href="#cb241-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-30"><a href="#cb241-30" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">map</span>(celltypes, merge_plots, treeplots) <span class="sc">|&gt;</span> <span class="fu">set_names</span>(celltypes)</span>
<span id="cb241-31"><a href="#cb241-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-32"><a href="#cb241-32" aria-hidden="true" tabindex="-1"></a>plots</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Astro-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Astro-pd_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Astro-wmh_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-3.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Astrocyte-activated-pd_35k10k-level2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-4.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-4.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Endothelial-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-5.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-5.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Fibroblast-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-6.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-6.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Fibroblast-EUROPEUKBB_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-7.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-7.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-8.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-8.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-EUROPEUKBB_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-9.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-9.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated-Bellenguez_2022_noproxy_35k10k-level2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-10.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-10.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated-EUROPEUKBB_35k10k-level2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-11.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-11.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-pd_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-12.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-12.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-13.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-13.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-EUROPEUKBB_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-14.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-14.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2-EUROPEUKBB_35k10k-level2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-15.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-15.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2-EUROPEUKBB_35k10k-level2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-16.png" class="lightbox" data-gallery="quarto-lightbox-gallery-16"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-16.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`SMC-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-17.png" class="lightbox" data-gallery="quarto-lightbox-gallery-17"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-17.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-18.png" class="lightbox" data-gallery="quarto-lightbox-gallery-18"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-18.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell-EUROPEUKBB_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-19.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-19.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell-pd_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-20.png" class="lightbox" data-gallery="quarto-lightbox-gallery-20"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-20.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Trans-Endo-to-mural-Bellenguez_2022_noproxy_35k10k-level1`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-6-21.png" class="lightbox" data-gallery="quarto-lightbox-gallery-21"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-6-21.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plots</span></span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(plots, <span class="fu">names</span>(plots), <span class="sc">~</span> <span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/001_magma/magma_pathways"</span>, <span class="fu">paste0</span>(.y, <span class="st">"_sig_genes_only"</span>, <span class="st">".png"</span>)), <span class="at">plot =</span> .x, <span class="at">width =</span> <span class="dv">32</span>, <span class="at">height =</span> <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Astro-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astro-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`Astro-pd_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astro-pd_35k10k-level1_sig_genes_only.png"

$`Astro-wmh_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astro-wmh_35k10k-level1_sig_genes_only.png"

$`Astrocyte-activated-pd_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astrocyte-activated-pd_35k10k-level2_sig_genes_only.png"

$`Endothelial-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Endothelial-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`Fibroblast-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Fibroblast-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`Fibroblast-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Fibroblast-EUROPEUKBB_35k10k-level1_sig_genes_only.png"

$`Microglia-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`Microglia-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-EUROPEUKBB_35k10k-level1_sig_genes_only.png"

$`Microglia-activated-Bellenguez_2022_noproxy_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-activated-Bellenguez_2022_noproxy_35k10k-level2_sig_genes_only.png"

$`Microglia-activated-EUROPEUKBB_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-activated-EUROPEUKBB_35k10k-level2_sig_genes_only.png"

$`Oligo-pd_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Oligo-pd_35k10k-level1_sig_genes_only.png"

$`Pericyte-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`Pericyte-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-EUROPEUKBB_35k10k-level1_sig_genes_only.png"

$`Pericyte-2-EUROPEUKBB_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-2-EUROPEUKBB_35k10k-level2_sig_genes_only.png"

$`Perivascular-FB-KAZN2-EUROPEUKBB_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Perivascular-FB-KAZN2-EUROPEUKBB_35k10k-level2_sig_genes_only.png"

$`SMC-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/SMC-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`T-cell-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/T-cell-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"

$`T-cell-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/T-cell-EUROPEUKBB_35k10k-level1_sig_genes_only.png"

$`T-cell-pd_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/T-cell-pd_35k10k-level1_sig_genes_only.png"

$`Trans-Endo-to-mural-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Trans-Endo-to-mural-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only.png"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb265"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb265-1"><a href="#cb265-1" aria-hidden="true" tabindex="-1"></a>dotplots <span class="ot">&lt;-</span></span>
<span id="cb265-2"><a href="#cb265-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb265-3"><a href="#cb265-3" aria-hidden="true" tabindex="-1"></a>    go_results_filtered,</span>
<span id="cb265-4"><a href="#cb265-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(go_results_filtered),</span>
<span id="cb265-5"><a href="#cb265-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">dotplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb265-6"><a href="#cb265-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggtitle</span>(.y)</span>
<span id="cb265-7"><a href="#cb265-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb265-8"><a href="#cb265-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-9"><a href="#cb265-9" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">map</span>(celltypes, merge_plots, dotplots) <span class="sc">|&gt;</span> <span class="fu">set_names</span>(celltypes)</span>
<span id="cb265-10"><a href="#cb265-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-11"><a href="#cb265-11" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(plots, <span class="fu">names</span>(plots), <span class="sc">~</span> <span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/001_magma/magma_pathways"</span>, <span class="fu">paste0</span>(.y, <span class="st">"_sig_genes_only-dotplots"</span>, <span class="st">".png"</span>)), <span class="at">plot =</span> .x, <span class="at">width =</span> <span class="dv">26</span>, <span class="at">height =</span> <span class="dv">12</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Astro-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astro-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`Astro-pd_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astro-pd_35k10k-level1_sig_genes_only-dotplots.png"

$`Astro-wmh_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astro-wmh_35k10k-level1_sig_genes_only-dotplots.png"

$`Astrocyte-activated-pd_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Astrocyte-activated-pd_35k10k-level2_sig_genes_only-dotplots.png"

$`Endothelial-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Endothelial-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`Fibroblast-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Fibroblast-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`Fibroblast-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Fibroblast-EUROPEUKBB_35k10k-level1_sig_genes_only-dotplots.png"

$`Microglia-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`Microglia-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-EUROPEUKBB_35k10k-level1_sig_genes_only-dotplots.png"

$`Microglia-activated-Bellenguez_2022_noproxy_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-activated-Bellenguez_2022_noproxy_35k10k-level2_sig_genes_only-dotplots.png"

$`Microglia-activated-EUROPEUKBB_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-activated-EUROPEUKBB_35k10k-level2_sig_genes_only-dotplots.png"

$`Oligo-pd_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Oligo-pd_35k10k-level1_sig_genes_only-dotplots.png"

$`Pericyte-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`Pericyte-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-EUROPEUKBB_35k10k-level1_sig_genes_only-dotplots.png"

$`Pericyte-2-EUROPEUKBB_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-2-EUROPEUKBB_35k10k-level2_sig_genes_only-dotplots.png"

$`Perivascular-FB-KAZN2-EUROPEUKBB_35k10k-level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Perivascular-FB-KAZN2-EUROPEUKBB_35k10k-level2_sig_genes_only-dotplots.png"

$`SMC-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/SMC-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`T-cell-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/T-cell-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"

$`T-cell-EUROPEUKBB_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/T-cell-EUROPEUKBB_35k10k-level1_sig_genes_only-dotplots.png"

$`T-cell-pd_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/T-cell-pd_35k10k-level1_sig_genes_only-dotplots.png"

$`Trans-Endo-to-mural-Bellenguez_2022_noproxy_35k10k-level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Trans-Endo-to-mural-Bellenguez_2022_noproxy_35k10k-level1_sig_genes_only-dotplots.png"</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb267"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu">names</span>(go_results_filtered)) <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb267-2"><a href="#cb267-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-3"><a href="#cb267-3" aria-hidden="true" tabindex="-1"></a>merged_go_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(celltype, go_results_list) {</span>
<span id="cb267-4"><a href="#cb267-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb267-5"><a href="#cb267-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset to celltype</span></span>
<span id="cb267-6"><a href="#cb267-6" aria-hidden="true" tabindex="-1"></a>  gos <span class="ot">&lt;-</span> go_results_list[<span class="fu">grepl</span>(celltype, <span class="fu">names</span>(go_results_list))]</span>
<span id="cb267-7"><a href="#cb267-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb267-8"><a href="#cb267-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract results and add ontology information</span></span>
<span id="cb267-9"><a href="#cb267-9" aria-hidden="true" tabindex="-1"></a>  combined_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="fu">names</span>(gos), <span class="cf">function</span>(name) {</span>
<span id="cb267-10"><a href="#cb267-10" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(gos[[name]]<span class="sc">@</span>result)</span>
<span id="cb267-11"><a href="#cb267-11" aria-hidden="true" tabindex="-1"></a>    df<span class="sc">$</span>Ontology <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_(.*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, name)  <span class="co"># Extract ontology from name</span></span>
<span id="cb267-12"><a href="#cb267-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb267-13"><a href="#cb267-13" aria-hidden="true" tabindex="-1"></a>  }))</span>
<span id="cb267-14"><a href="#cb267-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(combined_df)</span>
<span id="cb267-15"><a href="#cb267-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb267-16"><a href="#cb267-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-17"><a href="#cb267-17" aria-hidden="true" tabindex="-1"></a>go_combined <span class="ot">&lt;-</span> <span class="fu">map</span>(celltypes, merged_go_plot, go_results_filtered) <span class="sc">|&gt;</span></span>
<span id="cb267-18"><a href="#cb267-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(celltypes)</span>
<span id="cb267-19"><a href="#cb267-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Save</span></span>
<span id="cb267-20"><a href="#cb267-20" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(go_combined, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/go_data_sig_gene_only.qs"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="save-for-ipa" class="level2">
<h2 class="anchored" data-anchor-id="save-for-ipa">Save for IPA</h2>
<p>Caleb mentioned it can be useful to see what IPA pulls out in terms of pathways and such, but I need to save the data in a csv first</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb268"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb268-1"><a href="#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb268-2"><a href="#cb268-2" aria-hidden="true" tabindex="-1"></a><span class="co"># save data for IPA</span></span>
<span id="cb268-3"><a href="#cb268-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a function to save each subset as a CSV file</span></span>
<span id="cb268-4"><a href="#cb268-4" aria-hidden="true" tabindex="-1"></a>save_celltype_df <span class="ot">&lt;-</span> <span class="cf">function</span>(cell_type, df, file_name_root) {</span>
<span id="cb268-5"><a href="#cb268-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the dataframe for the current cell type</span></span>
<span id="cb268-6"><a href="#cb268-6" aria-hidden="true" tabindex="-1"></a>  celltype_df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">==</span> cell_type) <span class="sc">|&gt;</span></span>
<span id="cb268-7"><a href="#cb268-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Select relevant columns</span></span>
<span id="cb268-8"><a href="#cb268-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gene, log2FoldChange, padj) <span class="sc">|&gt;</span></span>
<span id="cb268-9"><a href="#cb268-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Help IPA figure the columns out</span></span>
<span id="cb268-10"><a href="#cb268-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">pvalue_adjust =</span> padj)</span>
<span id="cb268-11"><a href="#cb268-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb268-12"><a href="#cb268-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a filename based on the cell type</span></span>
<span id="cb268-13"><a href="#cb268-13" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">paste0</span>(file_name_root, <span class="st">"_"</span>, cell_type, <span class="st">".csv"</span>)</span>
<span id="cb268-14"><a href="#cb268-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb268-15"><a href="#cb268-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the subset dataframe to a CSV file</span></span>
<span id="cb268-16"><a href="#cb268-16" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_csv</span>(celltype_df, filename)</span>
<span id="cb268-17"><a href="#cb268-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb268-18"><a href="#cb268-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-19"><a href="#cb268-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Use map to iterate over each unique cell type and save a file</span></span>
<span id="cb268-20"><a href="#cb268-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="fu">unique</span>(res<span class="sc">$</span>celltype),</span>
<span id="cb268-21"><a href="#cb268-21" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">save_celltype_df</span>(</span>
<span id="cb268-22"><a href="#cb268-22" aria-hidden="true" tabindex="-1"></a>        .x,</span>
<span id="cb268-23"><a href="#cb268-23" aria-hidden="true" tabindex="-1"></a>        res,</span>
<span id="cb268-24"><a href="#cb268-24" aria-hidden="true" tabindex="-1"></a>        here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb268-25"><a href="#cb268-25" aria-hidden="true" tabindex="-1"></a>          <span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/01_ipa_data/magma_sig_genes"</span></span>
<span id="cb268-26"><a href="#cb268-26" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb268-27"><a href="#cb268-27" aria-hidden="true" tabindex="-1"></a>      ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>So apparently IPA is too stupid to handle more than 20 observations. I love it when extremely expensive closed-source software barely usable garbage!</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb269"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb269-1"><a href="#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb269-2"><a href="#cb269-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Prepare the data</span></span>
<span id="cb269-3"><a href="#cb269-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get a list of all unique cell types</span></span>
<span id="cb269-4"><a href="#cb269-4" aria-hidden="true" tabindex="-1"></a>all_celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(res<span class="sc">$</span>celltype)</span>
<span id="cb269-5"><a href="#cb269-5" aria-hidden="true" tabindex="-1"></a><span class="co"># remove unneeded ones</span></span>
<span id="cb269-6"><a href="#cb269-6" aria-hidden="true" tabindex="-1"></a>all_celltypes <span class="ot">&lt;-</span> all_celltypes[<span class="sc">!</span>all_celltypes <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"low-feature-cells"</span>, <span class="st">"ambiguous"</span>)]</span>
<span id="cb269-7"><a href="#cb269-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-8"><a href="#cb269-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the list of cell types into two groups of 20</span></span>
<span id="cb269-9"><a href="#cb269-9" aria-hidden="true" tabindex="-1"></a>celltypes_group1 <span class="ot">&lt;-</span> all_celltypes[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb269-10"><a href="#cb269-10" aria-hidden="true" tabindex="-1"></a>celltypes_group2 <span class="ot">&lt;-</span> all_celltypes[<span class="dv">21</span><span class="sc">:</span><span class="dv">40</span>]</span>
<span id="cb269-11"><a href="#cb269-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-12"><a href="#cb269-12" aria-hidden="true" tabindex="-1"></a>res <span class="sc">|&gt;</span> </span>
<span id="cb269-13"><a href="#cb269-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">%in%</span> celltypes_group1) <span class="sc">|&gt;</span></span>
<span id="cb269-14"><a href="#cb269-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, log2FoldChange, pvalue, padj, celltype) <span class="sc">|&gt;</span></span>
<span id="cb269-15"><a href="#cb269-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype, <span class="at">values_from =</span> <span class="fu">c</span>(log2FoldChange, pvalue, padj)) <span class="sc">|&gt;</span></span>
<span id="cb269-16"><a href="#cb269-16" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/01_ipa_data/pseudobulk_ad-vs-control_deseq2_subtype-annotated_wide_group1.csv"</span>))</span>
<span id="cb269-17"><a href="#cb269-17" aria-hidden="true" tabindex="-1"></a>res <span class="sc">|&gt;</span> </span>
<span id="cb269-18"><a href="#cb269-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">%in%</span> celltypes_group2) <span class="sc">|&gt;</span></span>
<span id="cb269-19"><a href="#cb269-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, log2FoldChange, pvalue, padj, celltype) <span class="sc">|&gt;</span></span>
<span id="cb269-20"><a href="#cb269-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype, <span class="at">values_from =</span> <span class="fu">c</span>(log2FoldChange, pvalue, padj)) <span class="sc">|&gt;</span></span>
<span id="cb269-21"><a href="#cb269-21" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/01_ipa_data/pseudobulk_ad-vs-control_deseq2_subtype-annotated_wide_group2.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb270"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb270-2"><a href="#cb270-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Copy the gene names straight to the clipboard</span></span>
<span id="cb270-3"><a href="#cb270-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Note this only works on MacOS, Windows/Linux would need adapted versions</span></span>
<span id="cb270-4"><a href="#cb270-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">unique</span>(res_sig<span class="sc">$</span>gene) <span class="sc">|&gt;</span> <span class="fu">paste0</span>(<span class="at">collapse =</span> <span class="st">","</span>), <span class="fu">pipe</span>(<span class="st">"pbcopy"</span>), <span class="at">sep =</span> <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<!-- There are a quite a lot of unique pairwise comparisons when looking at 2 factors (fraction and disease status in this case), and the cross groups can probably be dropped for now to prioritise the most interesting comparisons. -->
<!-- In this case, I'd think the most interesting comparisons are within the same fraction across disease status, and across fractions within the same disease state. -->
</section>
</section>
<section id="gene-set-enrichment-analysis-gsea" class="level1">
<h1>Gene set enrichment analysis (GSEA)</h1>
<ul>
<li>Note that GSEA takes pval and foldchange into account when computing the enrichment, so we don’t filter the gene list</li>
</ul>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb271"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb271-1"><a href="#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="co"># nest data by celltype</span></span>
<span id="cb271-2"><a href="#cb271-2" aria-hidden="true" tabindex="-1"></a>gsea_df <span class="ot">&lt;-</span> res <span class="sc">|&gt;</span></span>
<span id="cb271-3"><a href="#cb271-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># group by celltype</span></span>
<span id="cb271-4"><a href="#cb271-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">|&gt;</span></span>
<span id="cb271-5"><a href="#cb271-5" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb272"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb272-1"><a href="#cb272-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb272-2"><a href="#cb272-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb272-3"><a href="#cb272-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Function: Adjacency matrix to list -------------------------</span></span>
<span id="cb272-4"><a href="#cb272-4" aria-hidden="true" tabindex="-1"></a>matrix_to_list <span class="ot">&lt;-</span> <span class="cf">function</span>(pws){</span>
<span id="cb272-5"><a href="#cb272-5" aria-hidden="true" tabindex="-1"></a>  pws.l <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb272-6"><a href="#cb272-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pw <span class="cf">in</span> <span class="fu">colnames</span>(pws)) {</span>
<span id="cb272-7"><a href="#cb272-7" aria-hidden="true" tabindex="-1"></a>    pws.l[[pw]] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pws)[<span class="fu">as.logical</span>(pws[, pw])]</span>
<span id="cb272-8"><a href="#cb272-8" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb272-9"><a href="#cb272-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pws.l)</span>
<span id="cb272-10"><a href="#cb272-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb272-11"><a href="#cb272-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-12"><a href="#cb272-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Function: prepare_gmt --------------------------------------</span></span>
<span id="cb272-13"><a href="#cb272-13" aria-hidden="true" tabindex="-1"></a>prepare_gmt <span class="ot">&lt;-</span> <span class="cf">function</span>(gmt_file, genes_in_data, <span class="at">savefile =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb272-14"><a href="#cb272-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># for debug</span></span>
<span id="cb272-15"><a href="#cb272-15" aria-hidden="true" tabindex="-1"></a>  <span class="co">#file &lt;- gmt_files[1]</span></span>
<span id="cb272-16"><a href="#cb272-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#genes_in_data &lt;- df$gene_symbol</span></span>
<span id="cb272-17"><a href="#cb272-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-18"><a href="#cb272-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read in gmt file</span></span>
<span id="cb272-19"><a href="#cb272-19" aria-hidden="true" tabindex="-1"></a>  gmt <span class="ot">&lt;-</span> <span class="fu">gmtPathways</span>(gmt_file)</span>
<span id="cb272-20"><a href="#cb272-20" aria-hidden="true" tabindex="-1"></a>  hidden <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">unlist</span>(gmt))</span>
<span id="cb272-21"><a href="#cb272-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-22"><a href="#cb272-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert gmt file to a matrix with the genes as rows and for each go annotation (columns) the values are 0 or 1</span></span>
<span id="cb272-23"><a href="#cb272-23" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">&lt;-</span> <span class="fu">matrix</span>(</span>
<span id="cb272-24"><a href="#cb272-24" aria-hidden="true" tabindex="-1"></a>    <span class="cn">NA</span>,</span>
<span id="cb272-25"><a href="#cb272-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">dimnames =</span> <span class="fu">list</span>(hidden, <span class="fu">names</span>(gmt)),</span>
<span id="cb272-26"><a href="#cb272-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">nrow =</span> <span class="fu">length</span>(hidden),</span>
<span id="cb272-27"><a href="#cb272-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">ncol =</span> <span class="fu">length</span>(gmt)</span>
<span id="cb272-28"><a href="#cb272-28" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb272-29"><a href="#cb272-29" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(mat)[<span class="dv">2</span>]) {</span>
<span id="cb272-30"><a href="#cb272-30" aria-hidden="true" tabindex="-1"></a>    mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(hidden <span class="sc">%in%</span> gmt[[i]])</span>
<span id="cb272-31"><a href="#cb272-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb272-32"><a href="#cb272-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-33"><a href="#cb272-33" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Subset to the genes that are present in our data to avoid bias</span></span>
<span id="cb272-34"><a href="#cb272-34" aria-hidden="true" tabindex="-1"></a>  hidden1 <span class="ot">&lt;-</span> <span class="fu">intersect</span>(genes_in_data, hidden)</span>
<span id="cb272-35"><a href="#cb272-35" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">&lt;-</span></span>
<span id="cb272-36"><a href="#cb272-36" aria-hidden="true" tabindex="-1"></a>    mat[hidden1, <span class="fu">colnames</span>(mat)[<span class="fu">which</span>(<span class="fu">colSums</span>(mat[hidden1, ]) <span class="sc">&gt;</span> <span class="dv">5</span>)]] <span class="co"># filter for gene sets with more than 5 genes annotated</span></span>
<span id="cb272-37"><a href="#cb272-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># And get the list again</span></span>
<span id="cb272-38"><a href="#cb272-38" aria-hidden="true" tabindex="-1"></a>  final_list <span class="ot">&lt;-</span></span>
<span id="cb272-39"><a href="#cb272-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">matrix_to_list</span>(mat) <span class="co"># for this we use the function we previously defined</span></span>
<span id="cb272-40"><a href="#cb272-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-41"><a href="#cb272-41" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (savefile) {</span>
<span id="cb272-42"><a href="#cb272-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(final_list, <span class="at">file =</span> <span class="fu">paste0</span>(</span>
<span id="cb272-43"><a href="#cb272-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">gsub</span>(<span class="st">'.gmt'</span>, <span class="st">''</span>, gmt_file),</span>
<span id="cb272-44"><a href="#cb272-44" aria-hidden="true" tabindex="-1"></a>      <span class="st">'_subset_'</span>,</span>
<span id="cb272-45"><a href="#cb272-45" aria-hidden="true" tabindex="-1"></a>      <span class="fu">format</span>(<span class="fu">Sys.time</span>(), <span class="st">'%d%m'</span>),</span>
<span id="cb272-46"><a href="#cb272-46" aria-hidden="true" tabindex="-1"></a>      <span class="st">'.RData'</span></span>
<span id="cb272-47"><a href="#cb272-47" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb272-48"><a href="#cb272-48" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb272-49"><a href="#cb272-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb272-50"><a href="#cb272-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">'Wohoo! .gmt conversion successfull!:)'</span>)</span>
<span id="cb272-51"><a href="#cb272-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(final_list)</span>
<span id="cb272-52"><a href="#cb272-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb272-53"><a href="#cb272-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-54"><a href="#cb272-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all the genes in dataset and assign them to my_genes </span></span>
<span id="cb272-55"><a href="#cb272-55" aria-hidden="true" tabindex="-1"></a>my_genes <span class="ot">&lt;-</span> res<span class="sc">$</span>hgnc_symbol</span>
<span id="cb272-56"><a href="#cb272-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-57"><a href="#cb272-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Download gene sets .gmt files</span></span>
<span id="cb272-58"><a href="#cb272-58" aria-hidden="true" tabindex="-1"></a><span class="co">#https://www.gsea-msigdb.org/gsea/msigdb/human/collections.jsp</span></span>
<span id="cb272-59"><a href="#cb272-59" aria-hidden="true" tabindex="-1"></a><span class="co"># I've downloaded several background sets to try</span></span>
<span id="cb272-60"><a href="#cb272-60" aria-hidden="true" tabindex="-1"></a>gmt_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/991_external_data"</span>), <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.gmt$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb272-61"><a href="#cb272-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-62"><a href="#cb272-62" aria-hidden="true" tabindex="-1"></a>gmt <span class="ot">&lt;-</span> <span class="fu">map</span>(gmt_files, <span class="sc">~</span> <span class="fu">prepare_gmt</span>(.x, my_genes))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Wohoo! .gmt conversion successfull!:)"
[1] "Wohoo! .gmt conversion successfull!:)"
[1] "Wohoo! .gmt conversion successfull!:)"
[1] "Wohoo! .gmt conversion successfull!:)"</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb274"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb274-1"><a href="#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb274-2"><a href="#cb274-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb274-3"><a href="#cb274-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb274-4"><a href="#cb274-4" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the per celltype version - not using fgsea at the moment so I'll skip</span></span>
<span id="cb274-5"><a href="#cb274-5" aria-hidden="true" tabindex="-1"></a><span class="co"># this for now</span></span>
<span id="cb274-6"><a href="#cb274-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all the genes in your dataset and assign them to my_genes </span></span>
<span id="cb274-7"><a href="#cb274-7" aria-hidden="true" tabindex="-1"></a>my_genes <span class="ot">&lt;-</span> <span class="fu">map</span>(gsea_df<span class="sc">$</span>data, <span class="sc">~</span> .x<span class="sc">$</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb274-8"><a href="#cb274-8" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(gsea_df<span class="sc">$</span>celltype)</span>
<span id="cb274-9"><a href="#cb274-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb274-10"><a href="#cb274-10" aria-hidden="true" tabindex="-1"></a>gmt <span class="ot">&lt;-</span> <span class="fu">map</span>(gmt_files, <span class="cf">function</span>(gmt) {</span>
<span id="cb274-11"><a href="#cb274-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(my_genes, <span class="cf">function</span>(genes) {</span>
<span id="cb274-12"><a href="#cb274-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">prepare_gmt</span>(gmt, genes)</span>
<span id="cb274-13"><a href="#cb274-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb274-14"><a href="#cb274-14" aria-hidden="true" tabindex="-1"></a>)})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Get the gene rankings</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb275"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes need to be ranked for GSEA as named gene list</span></span>
<span id="cb275-2"><a href="#cb275-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Here I'll use use the signed p values from spatial DGE as ranking</span></span>
<span id="cb275-3"><a href="#cb275-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb275-4"><a href="#cb275-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to sort rankings</span></span>
<span id="cb275-5"><a href="#cb275-5" aria-hidden="true" tabindex="-1"></a>sort_rankings <span class="ot">&lt;-</span> <span class="cf">function</span>(log2fc, pvals, gene_names, group, <span class="at">plot =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb275-6"><a href="#cb275-6" aria-hidden="true" tabindex="-1"></a>  rankings <span class="ot">&lt;-</span> <span class="fu">sign</span>(log2fc) <span class="sc">*</span> (<span class="sc">-</span><span class="fu">log10</span>(pvals))</span>
<span id="cb275-7"><a href="#cb275-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># genes as names</span></span>
<span id="cb275-8"><a href="#cb275-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(rankings) <span class="ot">&lt;-</span> gene_names</span>
<span id="cb275-9"><a href="#cb275-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(rankings)</span>
<span id="cb275-10"><a href="#cb275-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb275-11"><a href="#cb275-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># sort genes by ranking</span></span>
<span id="cb275-12"><a href="#cb275-12" aria-hidden="true" tabindex="-1"></a>  rankings <span class="ot">&lt;-</span> <span class="fu">sort</span>(rankings, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb275-13"><a href="#cb275-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (plot) {</span>
<span id="cb275-14"><a href="#cb275-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(rankings, <span class="at">main =</span> group)</span>
<span id="cb275-15"><a href="#cb275-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb275-16"><a href="#cb275-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb275-17"><a href="#cb275-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Some genes have such low p values that the signed pval is +- inf, we need to</span></span>
<span id="cb275-18"><a href="#cb275-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># change it to the maximum * constant to avoid problems with fgsea</span></span>
<span id="cb275-19"><a href="#cb275-19" aria-hidden="true" tabindex="-1"></a>  max_ranking <span class="ot">&lt;-</span> <span class="fu">max</span>(rankings[<span class="fu">is.finite</span>(rankings)])</span>
<span id="cb275-20"><a href="#cb275-20" aria-hidden="true" tabindex="-1"></a>  min_ranking <span class="ot">&lt;-</span> <span class="fu">min</span>(rankings[<span class="fu">is.finite</span>(rankings)])</span>
<span id="cb275-21"><a href="#cb275-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># change any inf values to 10 times the min/max non-inf value</span></span>
<span id="cb275-22"><a href="#cb275-22" aria-hidden="true" tabindex="-1"></a>  rankings <span class="ot">&lt;-</span></span>
<span id="cb275-23"><a href="#cb275-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">replace</span>(rankings, rankings <span class="sc">&gt;</span> max_ranking, max_ranking <span class="sc">*</span> <span class="dv">10</span>)</span>
<span id="cb275-24"><a href="#cb275-24" aria-hidden="true" tabindex="-1"></a>  rankings <span class="ot">&lt;-</span></span>
<span id="cb275-25"><a href="#cb275-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">replace</span>(rankings, rankings <span class="sc">&lt;</span> min_ranking, min_ranking <span class="sc">*</span> <span class="dv">10</span>)</span>
<span id="cb275-26"><a href="#cb275-26" aria-hidden="true" tabindex="-1"></a>  rankings <span class="ot">&lt;-</span></span>
<span id="cb275-27"><a href="#cb275-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sort</span>(rankings, <span class="at">decreasing =</span> <span class="cn">TRUE</span>) <span class="co"># sort genes by ranking</span></span>
<span id="cb275-28"><a href="#cb275-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb275-29"><a href="#cb275-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(rankings)</span>
<span id="cb275-30"><a href="#cb275-30" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb276"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb276-1"><a href="#cb276-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb276-2"><a href="#cb276-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb276-3"><a href="#cb276-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb276-4"><a href="#cb276-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Again for more complex contrasts</span></span>
<span id="cb276-5"><a href="#cb276-5" aria-hidden="true" tabindex="-1"></a>rankings <span class="ot">&lt;-</span></span>
<span id="cb276-6"><a href="#cb276-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(gsea_df<span class="sc">$</span>data, gsea_df<span class="sc">$</span>celltype_contrast, <span class="sc">~</span> (<span class="fu">sort_rankings</span>(.x<span class="sc">$</span>log2FoldChange,</span>
<span id="cb276-7"><a href="#cb276-7" aria-hidden="true" tabindex="-1"></a>                                                        .x<span class="sc">$</span>pvalue,</span>
<span id="cb276-8"><a href="#cb276-8" aria-hidden="true" tabindex="-1"></a>                                                        .x<span class="sc">$</span>gene,</span>
<span id="cb276-9"><a href="#cb276-9" aria-hidden="true" tabindex="-1"></a>                                                        .y))) <span class="sc">%&gt;%</span></span>
<span id="cb276-10"><a href="#cb276-10" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(gsea_df<span class="sc">$</span>celltype_contrast)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb277"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1" aria-hidden="true" tabindex="-1"></a>rankings <span class="ot">&lt;-</span></span>
<span id="cb277-2"><a href="#cb277-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(gsea_df<span class="sc">$</span>data, gsea_df<span class="sc">$</span>celltype, <span class="sc">~</span> (<span class="fu">sort_rankings</span>(.x<span class="sc">$</span>ZSTAT,</span>
<span id="cb277-3"><a href="#cb277-3" aria-hidden="true" tabindex="-1"></a>                                                        .x<span class="sc">$</span>P,</span>
<span id="cb277-4"><a href="#cb277-4" aria-hidden="true" tabindex="-1"></a>                                                        .x<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb277-5"><a href="#cb277-5" aria-hidden="true" tabindex="-1"></a>                                                        .y))) <span class="sc">%&gt;%</span></span>
<span id="cb277-6"><a href="#cb277-6" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(gsea_df<span class="sc">$</span>celltype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb278"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb278-1"><a href="#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">data.frame</span>(</span>
<span id="cb278-2"><a href="#cb278-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene_symbol =</span> <span class="fu">names</span>(rankings<span class="sc">$</span><span class="st">`</span><span class="at">Microglia-activated</span><span class="st">`</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>],</span>
<span id="cb278-3"><a href="#cb278-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ranks =</span> rankings<span class="sc">$</span><span class="st">`</span><span class="at">Microglia-activated</span><span class="st">`</span>[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb278-4"><a href="#cb278-4" aria-hidden="true" tabindex="-1"></a>),</span>
<span id="cb278-5"><a href="#cb278-5" aria-hidden="true" tabindex="-1"></a><span class="fu">aes</span>(gene_symbol, ranks)) <span class="sc">+</span></span>
<span id="cb278-6"><a href="#cb278-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb278-7"><a href="#cb278-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb278-8"><a href="#cb278-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(</span>
<span id="cb278-9"><a href="#cb278-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb278-10"><a href="#cb278-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb278-11"><a href="#cb278-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="dv">1</span></span>
<span id="cb278-12"><a href="#cb278-12" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb278-13"><a href="#cb278-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Microglia-activated top 50 ranked genes"</span>,</span>
<span id="cb278-14"><a href="#cb278-14" aria-hidden="true" tabindex="-1"></a>          <span class="at">subtitle =</span> <span class="st">"MAGMA sig genes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-17-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-22"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<section id="run-gsea" class="level2">
<h2 class="anchored" data-anchor-id="run-gsea">Run GSEA</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb279"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb279-1"><a href="#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb279-2"><a href="#cb279-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb279-3"><a href="#cb279-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Run GSEA ---------------------------------------------------------------</span></span>
<span id="cb279-4"><a href="#cb279-4" aria-hidden="true" tabindex="-1"></a>gsea_res <span class="ot">&lt;-</span> <span class="fu">map</span>(gmt, <span class="cf">function</span>(pathway) {</span>
<span id="cb279-5"><a href="#cb279-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(rankings, <span class="cf">function</span>(rank) {</span>
<span id="cb279-6"><a href="#cb279-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fgsea</span>(</span>
<span id="cb279-7"><a href="#cb279-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">pathways =</span> pathway,</span>
<span id="cb279-8"><a href="#cb279-8" aria-hidden="true" tabindex="-1"></a>      <span class="co"># List of gene sets to check</span></span>
<span id="cb279-9"><a href="#cb279-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">stats =</span> rank,</span>
<span id="cb279-10"><a href="#cb279-10" aria-hidden="true" tabindex="-1"></a>      <span class="co"># in this case we have both pos and neg rankings. if only pos or neg, set</span></span>
<span id="cb279-11"><a href="#cb279-11" aria-hidden="true" tabindex="-1"></a>      <span class="co"># to 'pos', 'neg'</span></span>
<span id="cb279-12"><a href="#cb279-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">scoreType =</span> <span class="st">'std'</span>,</span>
<span id="cb279-13"><a href="#cb279-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">minSize =</span> <span class="dv">10</span>,</span>
<span id="cb279-14"><a href="#cb279-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">maxSize =</span> <span class="dv">500</span>,</span>
<span id="cb279-15"><a href="#cb279-15" aria-hidden="true" tabindex="-1"></a>      <span class="co"># for parallelisation</span></span>
<span id="cb279-16"><a href="#cb279-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">nproc =</span> <span class="dv">8</span></span>
<span id="cb279-17"><a href="#cb279-17" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb279-18"><a href="#cb279-18" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb279-19"><a href="#cb279-19" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb279-20"><a href="#cb279-20" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(<span class="fu">basename</span>(gmt_files))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (50.96% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (44.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (26.26% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.78% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (31.98% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.5% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (3.48% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (30.6% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.4% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.39% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (70.17% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.47% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.41% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.3% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.36% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (28.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.56% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (49.85% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (4.13% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.07% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.69% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.46% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.34% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 4 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.88% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 6 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.82% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.86% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.02% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (50.96% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (44.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (26.26% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.78% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (31.98% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.5% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (3.48% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (30.6% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.4% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.39% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (70.17% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.47% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.41% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.3% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.36% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (28.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.56% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (49.85% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (4.13% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.07% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.69% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.46% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.34% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.88% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 3 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.82% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.86% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 2 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.02% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (50.96% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (44.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (26.26% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.78% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (31.98% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.5% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (3.48% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (30.6% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.4% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.39% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 3 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (70.17% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.47% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 4 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.41% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 3 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.3% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.36% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (28.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.56% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 4 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (49.85% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (4.13% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 2 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.07% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 5 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.69% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 2 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 2 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.46% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.34% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.88% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.82% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.86% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 4 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.02% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (50.96% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (44.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (26.26% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.78% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (31.98% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (36.5% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (3.48% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (30.6% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.4% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.39% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (70.17% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.47% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (48.41% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (34.3% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.36% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (28.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (47.56% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (49.85% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (4.13% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.08% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.07% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.92% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.69% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.15% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.46% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (39.34% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,
gseaParam, : There are duplicate gene names, fgsea may produce unexpected
results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.88% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.82% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.86% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : For some of the pathways the P-values were likely overestimated. For
such pathways log2err is set to NA.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (1.02% of the list).
The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in fgseaMultilevel(pathways = pathways, stats = stats, minSize =
minSize, : There were 1 pathways for which P-values were not calculated
properly due to unbalanced (positive and negative) gene-level statistic values.
For such pathways pval, padj, NES, log2err are set to NA. You can try to
increase the value of the argument nPermSimple (for example set it nPermSimple
= 10000)</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb549"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb549-1"><a href="#cb549-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get log2 FC as a vector with the respective gene names, sorted in decreasing </span></span>
<span id="cb549-2"><a href="#cb549-2" aria-hidden="true" tabindex="-1"></a><span class="co"># order</span></span>
<span id="cb549-3"><a href="#cb549-3" aria-hidden="true" tabindex="-1"></a>named_vectors <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span></span>
<span id="cb549-4"><a href="#cb549-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb549-5"><a href="#cb549-5" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() <span class="sc">%&gt;%</span></span>
<span id="cb549-6"><a href="#cb549-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gene_list =</span> <span class="fu">map</span>(data, <span class="sc">~</span><span class="fu">setNames</span>(.x<span class="sc">$</span>ZSTAT, .x<span class="sc">$</span>hgnc_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb549-7"><a href="#cb549-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">sort</span>(<span class="at">decreasing =</span> <span class="cn">TRUE</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="gene-ontology" class="level2">
<h2 class="anchored" data-anchor-id="gene-ontology">Gene Ontology</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb550"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb550-1"><a href="#cb550-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the file path to save the resulting object</span></span>
<span id="cb550-2"><a href="#cb550-2" aria-hidden="true" tabindex="-1"></a>result_file <span class="ot">&lt;-</span></span>
<span id="cb550-3"><a href="#cb550-3" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb550-4"><a href="#cb550-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/007_magma/001_pathway_analysis/gene-set-enrichment_go_magma.rds"</span></span>
<span id="cb550-5"><a href="#cb550-5" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb550-6"><a href="#cb550-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb550-7"><a href="#cb550-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the file exists</span></span>
<span id="cb550-8"><a href="#cb550-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(result_file)) {</span>
<span id="cb550-9"><a href="#cb550-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plan</span>(multisession, <span class="at">workers =</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>() <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb550-10"><a href="#cb550-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Code for your time-consuming analysis</span></span>
<span id="cb550-11"><a href="#cb550-11" aria-hidden="true" tabindex="-1"></a>  gse_bp <span class="ot">&lt;-</span></span>
<span id="cb550-12"><a href="#cb550-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">future_map</span>(</span>
<span id="cb550-13"><a href="#cb550-13" aria-hidden="true" tabindex="-1"></a>      rankings,</span>
<span id="cb550-14"><a href="#cb550-14" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">gseGO</span>(</span>
<span id="cb550-15"><a href="#cb550-15" aria-hidden="true" tabindex="-1"></a>        .x,</span>
<span id="cb550-16"><a href="#cb550-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb550-17"><a href="#cb550-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb550-18"><a href="#cb550-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">keyType =</span> <span class="st">"SYMBOL"</span></span>
<span id="cb550-19"><a href="#cb550-19" aria-hidden="true" tabindex="-1"></a>      ), <span class="at">.options =</span> <span class="fu">furrr_options</span>(<span class="at">seed =</span> <span class="dv">123</span>)</span>
<span id="cb550-20"><a href="#cb550-20" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb550-21"><a href="#cb550-21" aria-hidden="true" tabindex="-1"></a>  gse_cc <span class="ot">&lt;-</span></span>
<span id="cb550-22"><a href="#cb550-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">future_map</span>(</span>
<span id="cb550-23"><a href="#cb550-23" aria-hidden="true" tabindex="-1"></a>      rankings,</span>
<span id="cb550-24"><a href="#cb550-24" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">gseGO</span>(</span>
<span id="cb550-25"><a href="#cb550-25" aria-hidden="true" tabindex="-1"></a>        .x,</span>
<span id="cb550-26"><a href="#cb550-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb550-27"><a href="#cb550-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">ont =</span> <span class="st">"CC"</span>,</span>
<span id="cb550-28"><a href="#cb550-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">keyType =</span> <span class="st">"SYMBOL"</span></span>
<span id="cb550-29"><a href="#cb550-29" aria-hidden="true" tabindex="-1"></a>      ), <span class="at">.options =</span> <span class="fu">furrr_options</span>(<span class="at">seed =</span> <span class="dv">123</span>)</span>
<span id="cb550-30"><a href="#cb550-30" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb550-31"><a href="#cb550-31" aria-hidden="true" tabindex="-1"></a>  gse_mf <span class="ot">&lt;-</span></span>
<span id="cb550-32"><a href="#cb550-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">future_map</span>(</span>
<span id="cb550-33"><a href="#cb550-33" aria-hidden="true" tabindex="-1"></a>      rankings,</span>
<span id="cb550-34"><a href="#cb550-34" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">gseGO</span>(</span>
<span id="cb550-35"><a href="#cb550-35" aria-hidden="true" tabindex="-1"></a>        .x,</span>
<span id="cb550-36"><a href="#cb550-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb550-37"><a href="#cb550-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">ont =</span> <span class="st">"MF"</span>,</span>
<span id="cb550-38"><a href="#cb550-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">keyType =</span> <span class="st">"SYMBOL"</span></span>
<span id="cb550-39"><a href="#cb550-39" aria-hidden="true" tabindex="-1"></a>      ), <span class="at">.options =</span> <span class="fu">furrr_options</span>(<span class="at">seed =</span> <span class="dv">123</span>)</span>
<span id="cb550-40"><a href="#cb550-40" aria-hidden="true" tabindex="-1"></a>    ) </span>
<span id="cb550-41"><a href="#cb550-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb550-42"><a href="#cb550-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gse_bp) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(gse_bp), <span class="st">"_BP"</span>)</span>
<span id="cb550-43"><a href="#cb550-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gse_mf) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(gse_mf), <span class="st">"_MF"</span>)</span>
<span id="cb550-44"><a href="#cb550-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gse_cc) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">names</span>(gse_cc), <span class="st">"_CC"</span>)</span>
<span id="cb550-45"><a href="#cb550-45" aria-hidden="true" tabindex="-1"></a>  gse <span class="ot">&lt;-</span> <span class="fu">c</span>(gse_bp, gse_mf, gse_cc)</span>
<span id="cb550-46"><a href="#cb550-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(gse_bp, gse_mf, gse_cc)</span>
<span id="cb550-47"><a href="#cb550-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb550-48"><a href="#cb550-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the resulting object</span></span>
<span id="cb550-49"><a href="#cb550-49" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(gse, result_file)</span>
<span id="cb550-50"><a href="#cb550-50" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb550-51"><a href="#cb550-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the existing resulting object</span></span>
<span id="cb550-52"><a href="#cb550-52" aria-hidden="true" tabindex="-1"></a>  gse <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(result_file)</span>
<span id="cb550-53"><a href="#cb550-53" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb551"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb551-1"><a href="#cb551-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any with no pathways found</span></span>
<span id="cb551-2"><a href="#cb551-2" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> gse <span class="sc">%&gt;%</span></span>
<span id="cb551-3"><a href="#cb551-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">keep</span>(<span class="sc">~</span> <span class="fu">nrow</span>(.) <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="insulin-pathways" class="level3">
<h3 class="anchored" data-anchor-id="insulin-pathways">Insulin pathways</h3>
<p>Zam is interested in pathways related to insulin, so I’ll filter to celltypes where such terms are present</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb552"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb552-1"><a href="#cb552-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to check for the presence of 'insulin' in the pathway description</span></span>
<span id="cb552-2"><a href="#cb552-2" aria-hidden="true" tabindex="-1"></a>find_insulin <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb552-3"><a href="#cb552-3" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(df)</span>
<span id="cb552-4"><a href="#cb552-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grepl</span>(<span class="st">"insulin"</span>, df<span class="sc">$</span>Description, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)</span>
<span id="cb552-5"><a href="#cb552-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb552-6"><a href="#cb552-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb552-7"><a href="#cb552-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each dataframe in the list</span></span>
<span id="cb552-8"><a href="#cb552-8" aria-hidden="true" tabindex="-1"></a>insulin_related <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gse, find_insulin)</span>
<span id="cb552-9"><a href="#cb552-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb552-10"><a href="#cb552-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Now, determine which celltypes have any insulin-related pathways</span></span>
<span id="cb552-11"><a href="#cb552-11" aria-hidden="true" tabindex="-1"></a>celltypes_with_insulin <span class="ot">&lt;-</span> <span class="fu">names</span>(insulin_related)[<span class="fu">sapply</span>(insulin_related, any)]</span>
<span id="cb552-12"><a href="#cb552-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb552-13"><a href="#cb552-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the celltypes with insulin-related pathways</span></span>
<span id="cb552-14"><a href="#cb552-14" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(celltypes_with_insulin)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>character(0)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb554"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb554-1"><a href="#cb554-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove ambiguous</span></span>
<span id="cb554-2"><a href="#cb554-2" aria-hidden="true" tabindex="-1"></a>celltypes_with_insulin <span class="ot">&lt;-</span></span>
<span id="cb554-3"><a href="#cb554-3" aria-hidden="true" tabindex="-1"></a>  celltypes_with_insulin[<span class="sc">!</span>celltypes_with_insulin <span class="sc">==</span> <span class="st">"ambiguous"</span>]</span>
<span id="cb554-4"><a href="#cb554-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb554-5"><a href="#cb554-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">"Celltype with 'insulin' in pathway description:"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Celltype with 'insulin' in pathway description:"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb556"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb556-1"><a href="#cb556-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(celltypes_with_insulin, <span class="at">collapse =</span> <span class="st">", "</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] ""</code></pre>
</div>
</div></div>
<p>Get the genes of these celltypes too</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb558"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb558-1"><a href="#cb558-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb558-2"><a href="#cb558-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb558-3"><a href="#cb558-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the dotplots of this subset</span></span>
<span id="cb558-4"><a href="#cb558-4" aria-hidden="true" tabindex="-1"></a>insulin_pathways <span class="ot">&lt;-</span> gse[<span class="fu">names</span>(gse) <span class="sc">%in%</span> celltypes_with_insulin]</span>
<span id="cb558-5"><a href="#cb558-5" aria-hidden="true" tabindex="-1"></a>dotplots <span class="ot">&lt;-</span></span>
<span id="cb558-6"><a href="#cb558-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb558-7"><a href="#cb558-7" aria-hidden="true" tabindex="-1"></a>    insulin_pathways,</span>
<span id="cb558-8"><a href="#cb558-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(insulin_pathways),</span>
<span id="cb558-9"><a href="#cb558-9" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">dotplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">split =</span> <span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(. <span class="sc">~</span> .sign) <span class="sc">+</span></span>
<span id="cb558-10"><a href="#cb558-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggtitle</span>(.y)</span>
<span id="cb558-11"><a href="#cb558-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb558-12"><a href="#cb558-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb558-13"><a href="#cb558-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify the filename for the PDF</span></span>
<span id="cb558-14"><a href="#cb558-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(include_apoe) {</span>
<span id="cb558-15"><a href="#cb558-15" aria-hidden="true" tabindex="-1"></a>pdf_file <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"insulin_celltype_dotplots_apoe.pdf"</span>)</span>
<span id="cb558-16"><a href="#cb558-16" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb558-17"><a href="#cb558-17" aria-hidden="true" tabindex="-1"></a>pdf_file <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"insulin_celltype_dotplots_no-apoe.pdf"</span>)</span>
<span id="cb558-18"><a href="#cb558-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb558-19"><a href="#cb558-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Open the PDF device</span></span>
<span id="cb558-20"><a href="#cb558-20" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(pdf_file)</span>
<span id="cb558-21"><a href="#cb558-21" aria-hidden="true" tabindex="-1"></a>dotplots</span>
<span id="cb558-22"><a href="#cb558-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the PDF device</span></span>
<span id="cb558-23"><a href="#cb558-23" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb558-24"><a href="#cb558-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb558-25"><a href="#cb558-25" aria-hidden="true" tabindex="-1"></a>insulin_genes <span class="ot">&lt;-</span> res_sig <span class="sc">|&gt;</span></span>
<span id="cb558-26"><a href="#cb558-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">%in%</span> celltypes_with_insulin)</span>
<span id="cb558-27"><a href="#cb558-27" aria-hidden="true" tabindex="-1"></a>insulin_genes <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">==</span> <span class="st">"Capillary"</span>) <span class="sc">|&gt;</span> <span class="fu">dim</span>()</span>
<span id="cb558-28"><a href="#cb558-28" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(include_apoe) {</span>
<span id="cb558-29"><a href="#cb558-29" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(insulin_genes, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"insulin_celltypes_sig_genes_apoe.csv"</span>))</span>
<span id="cb558-30"><a href="#cb558-30" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb558-31"><a href="#cb558-31" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(insulin_genes, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"insulin_celltypes_sig_genes_no-apoe.csv"</span>))</span>
<span id="cb558-32"><a href="#cb558-32" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="plots" class="level3">
<h3 class="anchored" data-anchor-id="plots">Plots</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb559"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb559-1"><a href="#cb559-1" aria-hidden="true" tabindex="-1"></a>dotplots <span class="ot">&lt;-</span></span>
<span id="cb559-2"><a href="#cb559-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb559-3"><a href="#cb559-3" aria-hidden="true" tabindex="-1"></a>    gse,</span>
<span id="cb559-4"><a href="#cb559-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gse),</span>
<span id="cb559-5"><a href="#cb559-5" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">dotplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">split =</span> <span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(. <span class="sc">~</span> .sign) <span class="sc">+</span></span>
<span id="cb559-6"><a href="#cb559-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggtitle</span>(.y)</span>
<span id="cb559-7"><a href="#cb559-7" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb559-8"><a href="#cb559-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb559-9"><a href="#cb559-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the sub function to extract the substring before the first underscore</span></span>
<span id="cb559-10"><a href="#cb559-10" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu">names</span>(dotplots)) <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb559-11"><a href="#cb559-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb559-12"><a href="#cb559-12" aria-hidden="true" tabindex="-1"></a>merge_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(celltype, dotplot_list) {</span>
<span id="cb559-13"><a href="#cb559-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subset to relevant plots</span></span>
<span id="cb559-14"><a href="#cb559-14" aria-hidden="true" tabindex="-1"></a>  plots <span class="ot">&lt;-</span> dotplot_list[<span class="fu">grepl</span>(celltype, <span class="fu">names</span>(dotplot_list))]</span>
<span id="cb559-15"><a href="#cb559-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge plots</span></span>
<span id="cb559-16"><a href="#cb559-16" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> plots[[<span class="dv">1</span>]] <span class="sc">+</span> plots[[<span class="dv">2</span>]] <span class="sc">+</span> plots[[<span class="dv">3</span>]]</span>
<span id="cb559-17"><a href="#cb559-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb559-18"><a href="#cb559-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb559-19"><a href="#cb559-19" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">map</span>(celltypes, merge_plots, dotplots) <span class="sc">|&gt;</span> <span class="fu">set_names</span>(celltypes)</span>
<span id="cb559-20"><a href="#cb559-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb559-21"><a href="#cb559-21" aria-hidden="true" tabindex="-1"></a>plots</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Microglia-activated`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-24-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-23"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-24-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-24"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-24-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-24-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-25"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-24-3.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb563"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb563-1"><a href="#cb563-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plots</span></span>
<span id="cb563-2"><a href="#cb563-2" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(plots, <span class="fu">names</span>(plots), <span class="sc">~</span> <span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/001_magma/magma_pathways"</span>, <span class="fu">paste0</span>(.y, <span class="st">".png"</span>)), <span class="at">plot =</span> .x, <span class="at">width =</span> <span class="dv">18</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 18 x 5 in image
Saving 18 x 5 in image
Saving 18 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Microglia-activated`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Microglia-activated.png"

$`Pericyte-2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Pericyte-2.png"

$`Perivascular-FB-KAZN2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/001_magma/magma_pathways/Perivascular-FB-KAZN2.png"</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb566"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb566-1"><a href="#cb566-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb566-2"><a href="#cb566-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb566-3"><a href="#cb566-3" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gse,</span>
<span id="cb566-4"><a href="#cb566-4" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(gse),</span>
<span id="cb566-5"><a href="#cb566-5" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">emapplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb566-6"><a href="#cb566-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">ggtitle</span>(.y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb567"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb567-1"><a href="#cb567-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gse,</span>
<span id="cb567-2"><a href="#cb567-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(gse),</span>
<span id="cb567-3"><a href="#cb567-3" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">ridgeplot</span>(.x) <span class="sc">+</span></span>
<span id="cb567-4"><a href="#cb567-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"enrichment distribution"</span>) <span class="sc">+</span></span>
<span id="cb567-5"><a href="#cb567-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">ggtitle</span>(.y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Microglia-activated_BP`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.42</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-26"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_BP`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.57</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-27"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_BP`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.53</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-28"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-3.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_MF`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.95</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-4.png" class="lightbox" data-gallery="quarto-lightbox-gallery-29"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-4.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_MF`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.67</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-5.png" class="lightbox" data-gallery="quarto-lightbox-gallery-30"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-5.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_MF`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 2.03</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-6.png" class="lightbox" data-gallery="quarto-lightbox-gallery-31"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-6.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_CC`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.5</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-7.png" class="lightbox" data-gallery="quarto-lightbox-gallery-32"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-7.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_CC`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.93</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-8.png" class="lightbox" data-gallery="quarto-lightbox-gallery-33"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-8.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_CC`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 1.18</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-26-9.png" class="lightbox" data-gallery="quarto-lightbox-gallery-34"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-26-9.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<p>Note, one can change which pathway is plotted here by setting the title and <code>geneSetID</code> accordingly</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb586"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb586-1"><a href="#cb586-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gse,</span>
<span id="cb586-2"><a href="#cb586-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(gse),</span>
<span id="cb586-3"><a href="#cb586-3" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">gseaplot</span>(.x, <span class="at">by =</span> <span class="st">"all"</span>, <span class="at">title =</span> .x<span class="sc">$</span>Description[<span class="dv">1</span>], <span class="at">geneSetID =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Microglia-activated_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-35"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-36"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-37"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-3.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-4.png" class="lightbox" data-gallery="quarto-lightbox-gallery-38"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-4.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-5.png" class="lightbox" data-gallery="quarto-lightbox-gallery-39"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-5.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-6.png" class="lightbox" data-gallery="quarto-lightbox-gallery-40"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-6.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-7.png" class="lightbox" data-gallery="quarto-lightbox-gallery-41"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-7.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-8.png" class="lightbox" data-gallery="quarto-lightbox-gallery-42"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-8.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="22_pathway_analysis_files/figure-html/unnamed-chunk-27-9.png" class="lightbox" data-gallery="quarto-lightbox-gallery-43"><img src="22_pathway_analysis_files/figure-html/unnamed-chunk-27-9.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"descPosition":"bottom","loop":false,"selector":".lightbox","openEffect":"zoom","closeEffect":"zoom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>