#!/bin/bash

#SBATCH -p c_highmem_dri1 ## dev, compute, htc, highmem
#SBATCH --job-name=pic_vcf_liftover-set2
#SBATCH --ntasks=1
#SBATCH --ntasks-per-node=1
#SBATCH --array=1-16%16
##### #SBATCH --mem-per-cpu=8000 # memory limit per core
#SBATCH --mem=8GB # memory limit per compute node for the job
#SBATCH --time=1-00:00 # maximum job time in D-HH:MM
#SBATCH --account=scw1329
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --mail-user Bernardo-HarringtonG@cardiff.ac.uk # email on fail
#SBATCH --mail-type END,FAIL

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "- SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "- SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "- SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "- SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "This job in the array has:"
echo "- SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "- SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Run on host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"

module purge
module load picard/2.20.2

## VCF files with variants at NeuroChip positions, based on hg38
VCF_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/01_vcf_specific_position/02_set2/"


## VCFs named after those
SAMPLE_ID_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_set2_rename.txt"

## not suitable, since contigs are named "1" "2" instead of "chr1"
## REF_TARGET_FILE="/scratch/c.mpmfw/Tools/Liftover/Homo_sapiens.GRCh37.dna.primary_assembly.fa.gz"

## contigs named as "chr1 1", "chr22 22"
## from Cell Ranger pre-processing script, available from CR website
REF_TARGET_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/02_cellranger_reference/GRCh37/fasta/genome.fa"

## needed for LiftoverVcf, will be generated by Picard CreateSequenceDictionary if not existent
REF_TARGET_DICT="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/02_cellranger_reference/GRCh37/fasta/genome.dict"

## http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz
CHAIN_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/hg38ToHg19.over.chain"

OUTPUT_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/03_full_liftover_to_hg19/02_set2/"


#-----------------------------------------------------------------------
## Picard LiftoverVcf

mkdir -p $OUTPUT_DIR

if [ ! -f "$REF_TARGET_DICT" ]; then

	java -jar $PICARD CreateSequenceDictionary \
		R=$REF_TARGET_FILE \
		O=$REF_TARGET_DICT
fi

N=${SLURM_ARRAY_TASK_ID}
SAMPLE_ID=$(cat $SAMPLE_ID_FILE | tail -n+${N} | head -1)

VCF_FILE=$VCF_DIR""$SAMPLE_ID"_var_raw.vcf"
OUT_VCF=$OUTPUT_DIR""$SAMPLE_ID"_var_raw.vcf"
OUT_REJ=$OUTPUT_DIR""$SAMPLE_ID"_var_raw_reject.vcf"

echo "Sample: "$SAMPLE_ID
echo "VCF file: "$VCF_FILE
echo "OUT VCF file: "$OUT_VCF

java -jar $PICARD LiftoverVcf \
     I=$VCF_FILE \
     O=$OUT_VCF \
     CHAIN=$CHAIN_FILE \
     REJECT=$OUT_REJ \
     R=$REF_TARGET_FILE

echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
