---
title: Get genotyping comparison
execute: 
  eval: false
---

- compare VCF from 10X transcriptomics vs VCF from NeuroChip 

1. filter and index VCFs from RNA
2. index VCFs from DNA
3. Compare VCFs via gtcheck

```{r}
#!/bin/bash

#SBATCH -p c_highmem_dri1 ## dev, compute, htc, highmem
#SBATCH --job-name=pic_vcf_liftover-set3
#SBATCH --ntasks=1
#SBATCH --ntasks-per-node=1
#SBATCH --array=1-40%20
##### #SBATCH --mem-per-cpu=8000 # memory limit per core
#SBATCH --mem=8GB # memory limit per compute node for the job
#SBATCH --time=1-00:00 # maximum job time in D-HH:MM
#SBATCH --account=scw1329
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --mail-user Bernardo-HarringtonG@cardiff.ac.uk # email on fail
#SBATCH --mail-type END,FAIL

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "- SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "- SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "- SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "- SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "This job in the array has:"
echo "- SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "- SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Run on host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"

module purge
module load picard/2.20.2

## VCF files with variants at NeuroChip positions, based on hg38
VCF_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/01_vcf_specific_position/03_set3/"

## VCFs named after those
SAMPLE_ID_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_set3.txt"

## not suitable, since contigs are named "1" "2" instead of "chr1"
## REF_TARGET_FILE="/scratch/c.mpmfw/Tools/Liftover/Homo_sapiens.GRCh37.dna.primary_assembly.fa.gz"

## contigs named as "chr1 1", "chr22 22"
## from Cell Ranger pre-processing script, available from CR website
REF_TARGET_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/02_cellranger_reference/GRCh37/fasta/genome.fa"

## needed for LiftoverVcf, will be generated by Picard CreateSequenceDictionary if not existent
REF_TARGET_DICT="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/02_cellranger_reference/GRCh37/fasta/genome.dict"

## http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz
CHAIN_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/hg38ToHg19.over.chain"

OUTPUT_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/03_full_liftover_to_hg19/03_set3/"


#-----------------------------------------------------------------------
## Picard LiftoverVcf

mkdir -p $OUTPUT_DIR

if [ ! -f "$REF_TARGET_DICT" ]; then

	java -jar $PICARD CreateSequenceDictionary \
		R=$REF_TARGET_FILE \
		O=$REF_TARGET_DICT
fi

N=${SLURM_ARRAY_TASK_ID}
SAMPLE_ID=$(cat $SAMPLE_ID_FILE | tail -n+${N} | head -1)
# Get the sample ID string after the first "_"
SAMPLE_ID="${SAMPLE_ID#*_}"

VCF_FILE=$VCF_DIR""$SAMPLE_ID"_var_raw.vcf"
OUT_VCF=$OUTPUT_DIR""$SAMPLE_ID"_var_raw.vcf"
OUT_REJ=$OUTPUT_DIR""$SAMPLE_ID"_var_raw_reject.vcf"

echo "Sample: "$SAMPLE_ID
echo "VCF file: "$VCF_FILE
echo "OUT VCF file: "$OUT_VCF

java -jar $PICARD LiftoverVcf \
     I=$VCF_FILE \
     O=$OUT_VCF \
     CHAIN=$CHAIN_FILE \
     REJECT=$OUT_REJ \
     R=$REF_TARGET_FILE

echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
```

```{r}
#!/bin/bash

#SBATCH -p c_compute_dri1 ## dev, compute, htc, highmem
#SBATCH --job-name=get_genotyping_comparison
#SBATCH --ntasks=1
#SBATCH --ntasks-per-node=1
##### #SBATCH --mem-per-cpu=8000 # memory limit per core
#SBATCH --mem=24GB # memory limit per compute node for the job
#SBATCH --time=0-10:00 # maximum job time in D-HH:MM
#SBATCH --account=scw1329
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --mail-user Bernardo-HarringtonG@cardiff.ac.uk # email on fail
#SBATCH --mail-type END,FAIL

## 24 genotyped donors by NeuroChip
## HOW WERE THEY SPLIT? - looking at the VCFs, must be done via bcftools view
## 1 sample: 
##bcftools_viewCommand=view -s 1_NP002_2019 -o /Users/frankwessely/Desktop/Documents/Projects/BBB/Endo_10X_Main/Analysis/Genotyping/NeuroChip_24_results/Split_samples/1_NP002_2019.vcf /Users/frankwessely/Desktop/Documents/Projects/BBB/Endo_10X_Main/Analysis/Genotyping/NeuroChip_24_results/Webber_Neurochip_2021_chr.vcf; Date=Fri Jul 23 18:10:48 2021
INPUT_DIR_DNA="/Users/frankwessely/Desktop/Documents/Projects/BBB/Endo_10X_Main/Analysis/Genotyping/NeuroChip_24_results/Split_samples/"
INPUT_DIR_DNA_BATCH1="/gluster/dri02/rdscw/shared/webber/Endo_10X/Genotyping/NeuroChip_20_donors_set_1/Split_samples/"
INPUT_DIR_DNA_BATCH2="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/PLINK_021123_0956/split_samples/"

## raw VCFs from Cell Ranger BAM files (lifted to hg19)
INPUT_DIR_RNA="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/03_full_liftover_to_hg19/"

## store .gz VCF files and index
OUTPUT_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/992_genotyping_data/04_compare_samples_dna_rna_filtered/"

OUTPUT_DIR_DNA=$OUTPUT_DIR"VCF_DNA/"
OUTPUT_DIR_RNA=$OUTPUT_DIR"VCF_RNA/"

## results from pairwise comparison via gtcheck
## columns:
## sample1 sample2 n_sample1_passed n_sample2_passed n_discordance n_compared
OUT_FILE=$OUTPUT_DIR"gtcheck_40_samples_DP_20_800_qual_20.txt"

#BCFTOOLS="/Users/frankwessely/Desktop/Documents/Tools/bcftools-1.9/bcftools"
module purge
module load bcftools/1.10.2

#-----------------------------------------------------------------------

mkdir -p $OUTPUT_DIR
mkdir -p $OUTPUT_DIR_DNA
mkdir -p $OUTPUT_DIR_RNA

cd $OUTPUT_DIR

find $INPUT_DIR_RNA -maxdepth 2 -name '*_raw.vcf' > filenames_1.tmp

## 1. filter and index VCFs from RNA
cat filenames_1.tmp | while read VCF; do
	
	VCF_NAME=$(echo ${VCF##*/})
	VCF_GZ=$OUTPUT_DIR_RNA""$VCF_NAME".gz"
	
	echo $VCF
	echo $VCF_GZ
	echo " "
	
	## 1. filter and then compress VCF file
	## filter here somewhat arbitrary, could be optimised
	bcftools filter -s LowQual -e '%QUAL<20 || INFO/DP>800 || INFO/DP<20' $VCF | bcftools view -f PASS -O z -o $VCF_GZ
	
	## 2. index VCF file
	bcftools index $VCF_GZ
	
done

wait
rm filenames_1.tmp

#-----------------------------------------------------------------------

find $INPUT_DIR_DNA_BATCH1 -maxdepth 1 -name '*.vcf' > filenames_1.tmp
find $INPUT_DIR_DNA_BATCH2 -maxdepth 1 -name '*.vcf' >> filenames_1.tmp

## 2. index VCFs from DNA
cat filenames_1.tmp | while read VCF; do
	
	VCF_NAME=$(echo ${VCF##*/})
	VCF_GZ=$OUTPUT_DIR_DNA""$VCF_NAME".gz"
	
	echo $VCF
	echo $VCF_GZ
	echo " "
	
	## 1. compress VCF file
	bcftools view $VCF -O z -o $VCF_GZ

	## 2. index VCF file
	bcftools index $VCF_GZ
	
done

wait
rm filenames_1.tmp

#-----------------------------------------------------------------------

## 3. Compare VCFs via gtcheck

find $OUTPUT_DIR_DNA -maxdepth 1 -name '*.vcf.gz' > filenames_1.tmp
find $OUTPUT_DIR_RNA -maxdepth 1 -name '*.vcf.gz' > filenames_2.tmp

## new results file
echo -n > $OUT_FILE

cat filenames_1.tmp | while read VCF1; 
do
	cat filenames_2.tmp | while read VCF2; 
	do
	
		VCF_NAME_1=$(echo ${VCF1##*/} | sed 's/.vcf.gz//')
		VCF_NAME_2=$(echo ${VCF2##*/} | sed 's/.vcf.gz//')
		
		## pair of samples
		echo -e -n $VCF_NAME_1"\t"$VCF_NAME_2"\t" >> $OUT_FILE
		
		## number of records in each VCF (after above filtering)
		echo -e -n $(bcftools index --nrecords $VCF1)"\t"$(bcftools index --nrecords $VCF2)"\t" >> $OUT_FILE
		
		## gtcheck
		bcftools gtcheck -G 1 -g $VCF1 $VCF2 | grep '^CN' | cut -f2,4 >> $OUT_FILE
		
	done
done

wait
rm filenames_1.tmp
rm filenames_2.tmp
```

