---
title: "QC Genotyping"
author: "Frank Wessely"
output:
  html_document:
    code_folding: hide
    number_sections: yes
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
---

```{r setup, include=FALSE}

library(knitr)
library(data.table)

knitr::opts_chunk$set(cache = TRUE, echo = FALSE, message = FALSE, warning = FALSE)
```

# Procedure

* genotyping data from NeuroChip platform
* 24 samples related to 20 initial donors were run in Cardiff
* the array has 24 spaces, and all spaces need to be filled to prevent any samples from drying out
* samples with _R suffix are from the same DNA tube as those without
* NP002, 003, 008 and 011 are each on the array twice
 * results based on hg19 (unfortunately not on GRCh38)
* however, GRCh38 reference used for Cell Ranger count, therefore some conversions are needed

***

* convert .ped and .map files to .vcf file
* .ped and .map files can be converted to .vcf file, containing all samples/people
* using plink recode funtion
* input files needed: 
* Webber_Neurochip_2021.ped 
* Webber_Neurochip_2021.map
* add "chr" prefix

`module load plink/1.9`  
`plink --file Webber_Neurochip_2021 --recode vcf --out Webber_Neurochip_2021_chr --output-chr "chrM"`

***

* split complete .vcf file into one .vcf file per sample
* use bcftools
* generate 24 individual samples

`BCFTOOLS="/Users/frankwessely/Desktop/Documents/Tools/bcftools-1.9/bcftools"`
`VCF_FILE="/Users/frankwessely/Desktop/VCF_tester/Webber_Neurochip_2021_chr.vcf"`
`OUT_DIR="/Users/frankwessely/Desktop/VCF_tester/Split_samples/"`

````{verbatim, echo=T}
for SAMPLE in `$BCFTOOLS view -h $VCF_FILE | grep "^#CHROM" | cut -f10-`; do
    $BCFTOOLS view -s $SAMPLE -o $OUT_DIR""$SAMPLE.vcf $VCF_FILE
done
````


***

* generate BAM files from Cell Ranger count as usual with GRCh38
* optionally, could run Cell Ranger count with older hg19 reference to avoid liftover workarounds


***

* `liftover_neurochip.Rmd`
* NeuroChip data is based on GRCh37 (hg19)
* 24 genotyped samples by NeuroChip, 20 donors in total, some donors were run as duplicates
* used rtracklayer to liftover original hg19 positions (from manifest) to GRCh38 positions
* code to extract NeuroChip positions: 
`cut -d',' -f10,11 Neuro_Consortium_v1-1_20015375_A1.csv | tail -n +9 | sort -t ',' -V | tr ',' '\t' | grep -v '0\t0\|XY\|Controls' | grep '\S' > NeuroChip_v1_grch37_positions.txt`
* stored in `/Users/frankwessely/Desktop/Documents/Projects/BBB/Endo_10X_Main/Analysis/Genotyping/NeuroChip_manifest/`
* create file: `NeuroChip_v1_positions_liftover_GRCh38.txt` with rtracklayer package
* download chain file for liftover:
* http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/hg38ToHg19.over.chain.gz


***

* `get_vcf_specific_pos.sh`
* call SNPs at specific NeuroChip positions and create VCFs from Cell Ranger BAM files
* approx. 500K liftover positions from NeuroChip


***

* `run_liftover_vcf.sh`
* next, the newly created VCF files from the BAM files need to be lifted back to hg19 to compare them with original DNA genotyping VCFs based on hg19
* needs a hg19 reference with contigs correctly named
* use picard tools "LiftoverVcf"

***

* compare VCF from 10X transcriptomics vs VCF from NeuroChip 
* `get_genotyping_comparison_DNA_RNA.sh`

1. filter and index VCFs from RNA
2. index VCFs from DNA
3. Compare VCFs via gtcheck

***

* `genotyping_check.Rmd`
* finally, generate heatmaps and tables in Rmd file to visualise and check results

