<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.54">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>CellChat</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> CellChat</h6>

            <a href="../../04_data_analysis/005_cell-cell_interactions/01_cellchat.qmd" class="btn btn-primary quarto-download-embed" download="01_cellchat.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">CellChat</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data">Read data</a></li>
  <li><a href="#compare-pericyte-subtypes" id="toc-compare-pericyte-subtypes" class="nav-link" data-scroll-target="#compare-pericyte-subtypes">Compare pericyte subtypes</a></li>
  <li><a href="#endomt-markers" id="toc-endomt-markers" class="nav-link" data-scroll-target="#endomt-markers">EndoMT markers</a></li>
  <li><a href="#check-complement-gene-expression" id="toc-check-complement-gene-expression" class="nav-link" data-scroll-target="#check-complement-gene-expression">Check complement gene expression</a></li>
  <li><a href="#make-umap-for-vascular-fraction-alone" id="toc-make-umap-for-vascular-fraction-alone" class="nav-link" data-scroll-target="#make-umap-for-vascular-fraction-alone">Make UMAP for vascular fraction alone</a></li>
  <li><a href="#cellchat" id="toc-cellchat" class="nav-link" data-scroll-target="#cellchat">CellChat</a>
  <ul class="collapse">
  <li><a href="#set-the-ligand-receptor-interaction-database" id="toc-set-the-ligand-receptor-interaction-database" class="nav-link" data-scroll-target="#set-the-ligand-receptor-interaction-database">Set the ligand-receptor interaction database</a></li>
  <li><a href="#part-1" id="toc-part-1" class="nav-link" data-scroll-target="#part-1">Part 1</a>
  <ul class="collapse">
  <li><a href="#read-in-merged-cellchat-object" id="toc-read-in-merged-cellchat-object" class="nav-link" data-scroll-target="#read-in-merged-cellchat-object">Read in merged cellchat object</a></li>
  <li><a href="#identify-altered-interactions-and-cell-populations" id="toc-identify-altered-interactions-and-cell-populations" class="nav-link" data-scroll-target="#identify-altered-interactions-and-cell-populations">Identify altered interactions and cell populations</a></li>
  <li><a href="#heatmap-showing-differential-number-of-interactions-or-interaction-strength-among-different-cell-populations-across-two-datasets" id="toc-heatmap-showing-differential-number-of-interactions-or-interaction-strength-among-different-cell-populations-across-two-datasets" class="nav-link" data-scroll-target="#heatmap-showing-differential-number-of-interactions-or-interaction-strength-among-different-cell-populations-across-two-datasets">Heatmap showing differential number of interactions or interaction strength among different cell populations across two datasets</a></li>
  </ul></li>
  <li><a href="#part-ii---identify-altered-signaling-with-distinct-network-architecture-and-interaction-strength" id="toc-part-ii---identify-altered-signaling-with-distinct-network-architecture-and-interaction-strength" class="nav-link" data-scroll-target="#part-ii---identify-altered-signaling-with-distinct-network-architecture-and-interaction-strength">Part II - Identify altered signaling with distinct network architecture and interaction strength</a>
  <ul class="collapse">
  <li><a href="#identify-signaling-networks-with-larger-or-less-difference-as-well-as-signaling-groups-based-on-their-functionalstructure-similarity" id="toc-identify-signaling-networks-with-larger-or-less-difference-as-well-as-signaling-groups-based-on-their-functionalstructure-similarity" class="nav-link" data-scroll-target="#identify-signaling-networks-with-larger-or-less-difference-as-well-as-signaling-groups-based-on-their-functionalstructure-similarity">Identify signaling networks with larger (or less) difference as well as signaling groups based on their functional/structure similarity</a></li>
  <li><a href="#identify-altered-signaling-with-distinct-interaction-strength" id="toc-identify-altered-signaling-with-distinct-interaction-strength" class="nav-link" data-scroll-target="#identify-altered-signaling-with-distinct-interaction-strength">Identify altered signaling with distinct interaction strength</a></li>
  </ul></li>
  <li><a href="#part-iii---identify-the-up-gulated-and-down-regulated-signaling-ligand-receptor-pairs" id="toc-part-iii---identify-the-up-gulated-and-down-regulated-signaling-ligand-receptor-pairs" class="nav-link" data-scroll-target="#part-iii---identify-the-up-gulated-and-down-regulated-signaling-ligand-receptor-pairs">Part III - Identify the up-gulated and down-regulated signaling ligand-receptor pairs</a>
  <ul class="collapse">
  <li><a href="#identify-dysfunctional-signaling-by-comparing-the-communication-probabities" id="toc-identify-dysfunctional-signaling-by-comparing-the-communication-probabities" class="nav-link" data-scroll-target="#identify-dysfunctional-signaling-by-comparing-the-communication-probabities">Identify dysfunctional signaling by comparing the communication probabities</a></li>
  <li><a href="#identify-dysfunctional-signaling-by-using-differential-expression-analysis" id="toc-identify-dysfunctional-signaling-by-using-differential-expression-analysis" class="nav-link" data-scroll-target="#identify-dysfunctional-signaling-by-using-differential-expression-analysis">Identify dysfunctional signaling by using differential expression analysis</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container code-fold"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/005_cell-cell_interactions/01_cellchat.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad</code></pre>
</div>
</div></div>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, smiths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'kableExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    group_rows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'janitor'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    chisq.test, fisher.test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    melt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rngtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  To enable shared memory capabilities, try: install.extras('
NMF
')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'future'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    run</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Some features are not enabled in this build of Arrow. Run `arrow_info()` for more information.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'arrow'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    timestamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'viridis'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    viridis_pal</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 methods overwritten by 'proxy':
  method               from    
  print.registry_field registry
  print.registry_entry registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    nrun</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:webshot2':

    resize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'DESeq2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    fpkm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: limma</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DESeq2':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    plotMDS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'edgeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SingleCellExperiment':

    cpm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocParallel'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    register</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation("pathview") within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationFilter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationFilter'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:arrow':

    field</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:future':

    value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ensembldb'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggrepel</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'PCAtools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scran':

    parallelPCA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    biplot, screeplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Seurat':

    components</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    normalize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:future':

    %-&gt;%, %&lt;-%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:NMF':

    algorithm, compare</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'biomaRt'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scRNAseq':

    listDatasets, searchDatasets</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>clusterProfiler v4.12.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use clusterProfiler in published research, please cite:
T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'clusterProfiler'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>DOSE v3.30.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'enrichplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    color_palette</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pals':

    alphabet</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocIO</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rtracklayer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocIO':

    FileForFormat</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    blocks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationHub':

    hubUrl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh37'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'qusage'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    read.gmt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:SNPlocs.Hsapiens.dbSNP155.GRCh37':

    .loadAlleles, .loadLoc, COMPATIBLE_BSGENOMES</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'GenomicFiles'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
simona version 1.2.0
Bioconductor page: http://bioconductor.org/packages/simona/
Github page: https://github.com/jokergoo/simona
Documentation: https://jokergoo.github.io/simona/

If you use it in published research, please cite:
Gu, Z. simona: a Comprehensive R package for Semantic Similarity 
  Analysis on Bio-Ontologies. bioRxiv 2023.

This message can be suppressed by:
  suppressPackageStartupMessages(library(simona))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'ggnetwork':
  method         from  
  fortify.igraph ggtree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'CellChat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scater':

    runPCA, runUMAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.setenv</span>(<span class="at">TAR_PROJECT =</span> here<span class="sc">::</span><span class="fu">here</span>())</span>
<span id="cb197-2"><a href="#cb197-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_config_set</span>(<span class="at">store =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_targets"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="read-data" class="level1">
<h1>Read data</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(sce)</span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(gene_ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="compare-pericyte-subtypes" class="level1">
<h1>Compare pericyte subtypes</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> gene_ids<span class="sc">$</span>ensembl</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a><span class="co">#sce &lt;- ScaleData(sce, features = rownames(sce))</span></span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a>pericyte_2_vs_1 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(sce, <span class="at">ident.1 =</span> <span class="st">"Pericyte-2"</span>, <span class="at">ident.2 =</span> <span class="st">"Pericyte"</span>)</span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a>pericyte_2_vs_1 <span class="ot">&lt;-</span> pericyte_2_vs_1 <span class="sc">|&gt;</span></span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ensembl"</span>) <span class="sc">|&gt;</span></span>
<span id="cb199-8"><a href="#cb199-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_ids, <span class="at">by =</span> <span class="st">"ensembl"</span>) <span class="sc">|&gt;</span></span>
<span id="cb199-9"><a href="#cb199-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb199-10"><a href="#cb199-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(gene <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb199-11"><a href="#cb199-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-12"><a href="#cb199-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Save pericytes</span></span>
<span id="cb199-13"><a href="#cb199-13" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(pericyte_2_vs_1, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/pericyte2_vs_pericyte1.csv"</span>))</span>
<span id="cb199-14"><a href="#cb199-14" aria-hidden="true" tabindex="-1"></a>sce_pericytes <span class="ot">&lt;-</span> <span class="fu">subset</span>(sce, <span class="at">subset =</span> subcelltype_annotations <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Pericyte-2"</span>, <span class="st">"Pericyte"</span>))</span>
<span id="cb199-15"><a href="#cb199-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> df<span class="sc">$</span>gene</span>
<span id="cb199-16"><a href="#cb199-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce_pericytes<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> df<span class="sc">$</span>gene</span>
<span id="cb199-17"><a href="#cb199-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-18"><a href="#cb199-18" aria-hidden="true" tabindex="-1"></a>top_markers <span class="ot">&lt;-</span> <span class="fu">head</span>(pericyte_2_vs_1[<span class="fu">order</span>(<span class="sc">-</span>pericyte_2_vs_1<span class="sc">$</span>avg_log2FC), ], <span class="at">n=</span><span class="dv">20</span>)  <span class="co"># Adjust 'n' as needed</span></span>
<span id="cb199-19"><a href="#cb199-19" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(sce_pericytes, <span class="at">features =</span> top_markers<span class="sc">$</span>gene) <span class="sc">+</span></span>
<span id="cb199-20"><a href="#cb199-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb199-21"><a href="#cb199-21" aria-hidden="true" tabindex="-1"></a>plot</span>
<span id="cb199-22"><a href="#cb199-22" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(pericyte_2_vs_1, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/pericyte2_vs_pericyte1.csv"</span>))</span>
<span id="cb199-23"><a href="#cb199-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/pericyte2_vs_pericyte1_top20genes.png"</span>), plot)</span>
<span id="cb199-24"><a href="#cb199-24" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(sce_pericytes, <span class="at">features =</span> top_markers<span class="sc">$</span>gene, <span class="at">slot =</span> <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="endomt-markers" class="level1">
<h1>EndoMT markers</h1>
<p>Get top specific markers for Endo-Mt, and most expressed markers, check which are nuclear localised</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a>endo_mt <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(sce, <span class="at">ident.1 =</span> <span class="st">"Trans-Endo-to-mural"</span>)</span>
<span id="cb200-4"><a href="#cb200-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-5"><a href="#cb200-5" aria-hidden="true" tabindex="-1"></a>endo_mt <span class="ot">&lt;-</span> endo_mt <span class="sc">|&gt;</span></span>
<span id="cb200-6"><a href="#cb200-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ensembl"</span>) <span class="sc">|&gt;</span></span>
<span id="cb200-7"><a href="#cb200-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_ids, <span class="at">by =</span> <span class="st">"ensembl"</span>) <span class="sc">|&gt;</span></span>
<span id="cb200-8"><a href="#cb200-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb200-9"><a href="#cb200-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(avg_log2FC))</span>
<span id="cb200-10"><a href="#cb200-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-11"><a href="#cb200-11" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(endo_mt, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/endo_mt/endo-mt_markers.csv"</span>))</span>
<span id="cb200-12"><a href="#cb200-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-13"><a href="#cb200-13" aria-hidden="true" tabindex="-1"></a>sce_endo_mt <span class="ot">&lt;-</span> <span class="fu">subset</span>(sce, <span class="at">subset =</span> highlevel_manual_annotations <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"mystery-cluster"</span>, <span class="st">"SMC"</span>, <span class="st">"Pericyte"</span>, <span class="st">"Endothelial"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb200-14"><a href="#cb200-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="at">subset =</span> subcelltype_annotations <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Oligo-A"</span>, <span class="st">"Astrocyte-activated"</span>), <span class="at">invert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb200-15"><a href="#cb200-15" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> gene_ids[gene_ids<span class="sc">$</span>ensembl <span class="sc">%in%</span> <span class="fu">rownames</span>(sce_endo_mt<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data), ]</span>
<span id="cb200-16"><a href="#cb200-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce_endo_mt<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> df<span class="sc">$</span>gene</span>
<span id="cb200-17"><a href="#cb200-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-18"><a href="#cb200-18" aria-hidden="true" tabindex="-1"></a><span class="co"># nuclear markers for EC and Mural</span></span>
<span id="cb200-19"><a href="#cb200-19" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(sce_endo_mt, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"LEF1"</span>, <span class="st">"NOTCH3"</span>, <span class="st">"TBX2"</span>)) <span class="sc">+</span></span>
<span id="cb200-20"><a href="#cb200-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb200-21"><a href="#cb200-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RotatedAxis</span>()</span>
<span id="cb200-22"><a href="#cb200-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-23"><a href="#cb200-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/endo_mt/nuclear_markers.png"</span>), plot, <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb200-24"><a href="#cb200-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-25"><a href="#cb200-25" aria-hidden="true" tabindex="-1"></a>endo_mt <span class="sc">|&gt;</span></span>
<span id="cb200-26"><a href="#cb200-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(gene <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"LEF1"</span>, <span class="st">"NOTCH3"</span>, <span class="st">"TBX2"</span>))</span>
<span id="cb200-27"><a href="#cb200-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-28"><a href="#cb200-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-29"><a href="#cb200-29" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> gene_ids<span class="sc">$</span>gene</span>
<span id="cb200-30"><a href="#cb200-30" aria-hidden="true" tabindex="-1"></a>marker_avg_expr <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(sce, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"LEF1"</span>, <span class="st">"NOTCH3"</span>, <span class="st">"TBX2"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb200-31"><a href="#cb200-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb200-32"><a href="#cb200-32" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(marker_avg_expr, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/endo_mt/level2_nuclear_marker_avg_expr.csv"</span>))</span>
<span id="cb200-33"><a href="#cb200-33" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(sce) <span class="ot">&lt;-</span> sce<span class="sc">$</span>highlevel_manual_annotations</span>
<span id="cb200-34"><a href="#cb200-34" aria-hidden="true" tabindex="-1"></a>marker_avg_expr <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(sce, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"LEF1"</span>, <span class="st">"NOTCH3"</span>, <span class="st">"TBX2"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb200-35"><a href="#cb200-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb200-36"><a href="#cb200-36" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(marker_avg_expr, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/endo_mt/level1_nuclear_marker_avg_expr.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hpar)</span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a>subcell <span class="ot">&lt;-</span> <span class="fu">hpaSubcellularLoc</span>()</span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to the nuclear terms</span></span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a>subcell_nuc <span class="ot">&lt;-</span> subcell <span class="sc">|&gt;</span></span>
<span id="cb201-7"><a href="#cb201-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Nucleoli|Nucleoplasm|Nuclear membrane"</span>, Main.location))</span>
<span id="cb201-8"><a href="#cb201-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-9"><a href="#cb201-9" aria-hidden="true" tabindex="-1"></a>endo_mt_nucleo_markers <span class="ot">&lt;-</span> endo_mt <span class="sc">|&gt;</span></span>
<span id="cb201-10"><a href="#cb201-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb201-11"><a href="#cb201-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(subcell_nuc, <span class="at">by =</span> <span class="fu">join_by</span>(ensembl <span class="sc">==</span> Gene)) <span class="sc">|&gt;</span></span>
<span id="cb201-12"><a href="#cb201-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Main.location))</span>
<span id="cb201-13"><a href="#cb201-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-14"><a href="#cb201-14" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(endo_mt_nucleo_markers, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/endo_mt/endo-mt_nucleo_markers.csv"</span>))</span>
<span id="cb201-15"><a href="#cb201-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-16"><a href="#cb201-16" aria-hidden="true" tabindex="-1"></a>endo_mt_markers <span class="ot">&lt;-</span> endo_mt_nucleo_markers <span class="sc">|&gt;</span></span>
<span id="cb201-17"><a href="#cb201-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_max</span>(avg_log2FC, <span class="at">n =</span> <span class="dv">30</span>)</span>
<span id="cb201-18"><a href="#cb201-18" aria-hidden="true" tabindex="-1"></a>endo_mt_markers <span class="ot">&lt;-</span> endo_mt_nucleo_markers <span class="sc">|&gt;</span></span>
<span id="cb201-19"><a href="#cb201-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_min</span>(avg_log2FC, <span class="at">n =</span> <span class="dv">30</span>) <span class="sc">|&gt;</span></span>
<span id="cb201-20"><a href="#cb201-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(endo_mt_markers)</span>
<span id="cb201-21"><a href="#cb201-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-22"><a href="#cb201-22" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(sce, <span class="at">features =</span> endo_mt_markers<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">RotatedAxis</span>()</span>
<span id="cb201-23"><a href="#cb201-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-24"><a href="#cb201-24" aria-hidden="true" tabindex="-1"></a>nuclea_membrane <span class="ot">&lt;-</span> endo_mt_nucleo_markers <span class="sc">|&gt;</span></span>
<span id="cb201-25"><a href="#cb201-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Nuclear membrane"</span>, Main.location), pct<span class="fl">.1</span> <span class="sc">&gt;</span> <span class="fl">0.5</span>)</span>
<span id="cb201-26"><a href="#cb201-26" aria-hidden="true" tabindex="-1"></a>nuclea_membrane <span class="ot">&lt;-</span> endo_mt_nucleo_markers <span class="sc">|&gt;</span></span>
<span id="cb201-27"><a href="#cb201-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pct<span class="fl">.1</span> <span class="sc">&gt;</span> <span class="fl">0.5</span>, avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb201-28"><a href="#cb201-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb201-29"><a href="#cb201-29" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::filter(grepl("Nuclear membrane", Main.location))</span></span>
<span id="cb201-30"><a href="#cb201-30" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(sce) <span class="ot">&lt;-</span> sce<span class="sc">$</span>highlevel_manual_annotations  </span>
<span id="cb201-31"><a href="#cb201-31" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(sce, <span class="at">features =</span> nuclea_membrane<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">RotatedAxis</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="check-complement-gene-expression" class="level1">
<h1>Check complement gene expression</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a>complement_gene_list <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="fu">here</span>(<span class="st">"03_data/991_external_data/complement_gene_list.xlsx"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">|&gt;</span></span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gene_name_cap =</span> <span class="fu">toupper</span>(gene_name))</span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-6"><a href="#cb202-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rownames</span>(sce))</span>
<span id="cb202-7"><a href="#cb202-7" aria-hidden="true" tabindex="-1"></a><span class="co"># need to rotate axis</span></span>
<span id="cb202-8"><a href="#cb202-8" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(sce, complement_gene_list<span class="sc">$</span>gene_name_cap) <span class="sc">+</span> </span>
<span id="cb202-9"><a href="#cb202-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb202-10"><a href="#cb202-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb202-11"><a href="#cb202-11" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(sce, complement_gene_list<span class="sc">$</span>gene_name_cap) <span class="sc">+</span> </span>
<span id="cb202-12"><a href="#cb202-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb202-13"><a href="#cb202-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>), <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">colour =</span> <span class="st">"black"</span>) <span class="co"># Set background to white</span></span>
<span id="cb202-14"><a href="#cb202-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb202-15"><a href="#cb202-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-16"><a href="#cb202-16" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"complement_celltype_expression-dotplot.png"</span>), <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">15</span>, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb202-17"><a href="#cb202-17" aria-hidden="true" tabindex="-1"></a>p</span>
<span id="cb202-18"><a href="#cb202-18" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb202-19"><a href="#cb202-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-20"><a href="#cb202-20" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(sce, <span class="at">features =</span> complement_gene_list<span class="sc">$</span>gene_name_cap, <span class="at">ncol =</span> <span class="dv">4</span>, <span class="at">pt.size =</span> <span class="dv">0</span>)</span>
<span id="cb202-21"><a href="#cb202-21" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"complement_celltype_expression-violin.png"</span>), <span class="at">height =</span> <span class="dv">52</span>, <span class="at">width =</span> <span class="dv">38</span>, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb202-22"><a href="#cb202-22" aria-hidden="true" tabindex="-1"></a>p2</span>
<span id="cb202-23"><a href="#cb202-23" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb202-24"><a href="#cb202-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb202-25"><a href="#cb202-25" aria-hidden="true" tabindex="-1"></a><span class="co"># bbb_marker_plot &lt;- VlnPlot(sdata_endo, features = BBB_targets, ncol = 2, pt.size = 0)</span></span>
<span id="cb202-26"><a href="#cb202-26" aria-hidden="true" tabindex="-1"></a><span class="co"># bbb_marker_plot</span></span>
<span id="cb202-27"><a href="#cb202-27" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave(here::here("05_figures/990_shared_figures/bbb_marker_expression_levelone.png"), bbb_marker_plot, device = "png", height = 18)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="make-umap-for-vascular-fraction-alone" class="level1">
<h1>Make UMAP for vascular fraction alone</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> df<span class="sc">$</span>ensembl</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span> <span class="fu">subset</span>(sce, <span class="at">subset =</span> prep <span class="sc">==</span> <span class="st">"V"</span>)</span>
<span id="cb203-5"><a href="#cb203-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any remaining low features in the new subset</span></span>
<span id="cb203-6"><a href="#cb203-6" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span> <span class="fu">subset</span>(subsets, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">200</span>)</span>
<span id="cb203-7"><a href="#cb203-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Cluster to the subsets</span></span>
<span id="cb203-8"><a href="#cb203-8" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(subsets, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb203-9"><a href="#cb203-9" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(subsets, <span class="at">resolution =</span> <span class="fl">0.4</span>)</span>
<span id="cb203-10"><a href="#cb203-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 4: Run UMAP</span></span>
<span id="cb203-11"><a href="#cb203-11" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(subsets, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span>
<span id="cb203-12"><a href="#cb203-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-13"><a href="#cb203-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Reduce size but keep reductions and graphs</span></span>
<span id="cb203-14"><a href="#cb203-14" aria-hidden="true" tabindex="-1"></a>reductions <span class="ot">&lt;-</span> <span class="fu">names</span>(subsets<span class="sc">@</span>reductions)</span>
<span id="cb203-15"><a href="#cb203-15" aria-hidden="true" tabindex="-1"></a>graphs <span class="ot">&lt;-</span> <span class="fu">names</span>(subsets<span class="sc">@</span>graphs)</span>
<span id="cb203-16"><a href="#cb203-16" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span> <span class="fu">DietSeurat</span>(subsets, <span class="at">dimreducs =</span> reductions, <span class="at">graphs =</span> graphs)</span>
<span id="cb203-17"><a href="#cb203-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-18"><a href="#cb203-18" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T)</span>
<span id="cb203-19"><a href="#cb203-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the new clusters</span></span>
<span id="cb203-20"><a href="#cb203-20" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_clusters <span class="ot">&lt;-</span> <span class="fu">Idents</span>(subsets)</span>
<span id="cb203-21"><a href="#cb203-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Restore the annotations</span></span>
<span id="cb203-22"><a href="#cb203-22" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(subsets) <span class="ot">&lt;-</span> subsets<span class="sc">$</span>subcelltype_annotations</span>
<span id="cb203-23"><a href="#cb203-23" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T, <span class="at">repel =</span> T)</span>
<span id="cb203-24"><a href="#cb203-24" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="st">"seurat_clusters"</span>, <span class="at">label =</span> T, <span class="at">repel =</span> T)</span>
<span id="cb203-25"><a href="#cb203-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-26"><a href="#cb203-26" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>new_subtype_annotations <span class="ot">&lt;-</span> subsets<span class="sc">$</span>subcelltype_annotations</span>
<span id="cb203-27"><a href="#cb203-27" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>new_subtype_annotations[subsets<span class="sc">$</span>seurat_clusters <span class="sc">==</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">"Capillary"</span></span>
<span id="cb203-28"><a href="#cb203-28" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="st">"new_subtype_annotations"</span>, <span class="at">label =</span> T, <span class="at">repel =</span> T)</span>
<span id="cb203-29"><a href="#cb203-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-30"><a href="#cb203-30" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(</span>
<span id="cb203-31"><a href="#cb203-31" aria-hidden="true" tabindex="-1"></a>  subsets,</span>
<span id="cb203-32"><a href="#cb203-32" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb203-33"><a href="#cb203-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../../bbb-project_2023-06-30/data/vascular_fraction_subset_seurat.rds"</span></span>
<span id="cb203-34"><a href="#cb203-34" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb203-35"><a href="#cb203-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb203-36"><a href="#cb203-36" aria-hidden="true" tabindex="-1"></a>subsets_controls <span class="ot">&lt;-</span> <span class="fu">subset</span>(sce, <span class="at">subset =</span> diagnosis <span class="sc">==</span> <span class="st">"Control"</span>)</span>
<span id="cb203-37"><a href="#cb203-37" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(subsets_controls<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data) <span class="ot">&lt;-</span> df<span class="sc">$</span>gene</span>
<span id="cb203-38"><a href="#cb203-38" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(</span>
<span id="cb203-39"><a href="#cb203-39" aria-hidden="true" tabindex="-1"></a>  subsets_controls,</span>
<span id="cb203-40"><a href="#cb203-40" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb203-41"><a href="#cb203-41" aria-hidden="true" tabindex="-1"></a>    <span class="st">"../../../bbb-project_2023-06-30/data/controls_subset_seurat.rds"</span></span>
<span id="cb203-42"><a href="#cb203-42" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb203-43"><a href="#cb203-43" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb203-44"><a href="#cb203-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb203-45"><a href="#cb203-45" aria-hidden="true" tabindex="-1"></a><span class="co"># # Find markers</span></span>
<span id="cb203-46"><a href="#cb203-46" aria-hidden="true" tabindex="-1"></a><span class="co"># cluster_markers &lt;- FindAllMarkers(</span></span>
<span id="cb203-47"><a href="#cb203-47" aria-hidden="true" tabindex="-1"></a><span class="co">#   subsets,</span></span>
<span id="cb203-48"><a href="#cb203-48" aria-hidden="true" tabindex="-1"></a><span class="co">#   only.pos = TRUE,</span></span>
<span id="cb203-49"><a href="#cb203-49" aria-hidden="true" tabindex="-1"></a><span class="co">#   min.pct = 0.2,</span></span>
<span id="cb203-50"><a href="#cb203-50" aria-hidden="true" tabindex="-1"></a><span class="co">#   logfc.threshold = 0.5</span></span>
<span id="cb203-51"><a href="#cb203-51" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(sce, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="cellchat" class="level1">
<h1>CellChat</h1>
<p><code>CellChat</code> is a package for looking at cell-cell interaction networks</p>
<p>It requires a normalised matrix of gene expression data (genes as rows, cells as columns). I assume from <a href="https://htmlpreview.github.io/?https://github.com/jinworks/CellChat/blob/master/tutorial/CellChat-vignette.html#load-the-required-libraries">the vignette</a> that one should look at one group at a time, in this case each fraction per disease state, so 4 groups total.</p>
<p>Since a lot of CellChats processing steps take a fair while to run, Ive switched to using the lovely <code>targets</code> package to precompute and just read in the results of that. Refer to the <code>_targets.R</code> script in the project root for the code used.</p>
<section id="set-the-ligand-receptor-interaction-database" class="level3">
<h3 class="anchored" data-anchor-id="set-the-ligand-receptor-interaction-database">Set the ligand-receptor interaction database</h3>
<p>Theres the option to use a subset of the CellChatDB, for example, just using cell-cell interactions. Ill start with that as they recommend not using all of the database. I assume this is because there are non-protein signalling stuff in there that we wouldnt have relevant data for?</p>
<p>This can be modified in the <code>_targets.R</code> file, but here is more or less the code used to start with.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a>CellChatDB <span class="ot">&lt;-</span> CellChatDB.human <span class="co"># use CellChatDB.mouse if running on mouse data</span></span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a><span class="fu">showDatabaseCategory</span>(CellChatDB)</span>
<span id="cb205-4"><a href="#cb205-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-5"><a href="#cb205-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Show the structure of the database</span></span>
<span id="cb205-6"><a href="#cb205-6" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">glimpse</span>(CellChatDB<span class="sc">$</span>interaction)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a><span class="co"># use a subset of CellChatDB for cell-cell communication analysis</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>CellChatDB.use <span class="ot">&lt;-</span> <span class="fu">subsetDB</span>(CellChatDB, <span class="at">search =</span> <span class="st">"Secreted Signaling"</span>, <span class="at">key =</span> <span class="st">"annotation"</span>) <span class="co"># use Secreted Signaling</span></span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Only uses the Secreted Signaling from CellChatDB v1</span></span>
<span id="cb206-6"><a href="#cb206-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  CellChatDB.use &lt;- subsetDB(CellChatDB, search = list(c("Secreted Signaling"), c("CellChatDB v1")), key = c("annotation", "version"))</span></span>
<span id="cb206-7"><a href="#cb206-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-8"><a href="#cb206-8" aria-hidden="true" tabindex="-1"></a><span class="co"># use all CellChatDB except for "Non-protein Signaling" for cell-cell communication analysis</span></span>
<span id="cb206-9"><a href="#cb206-9" aria-hidden="true" tabindex="-1"></a><span class="co"># CellChatDB.use &lt;- subsetDB(CellChatDB)</span></span>
<span id="cb206-10"><a href="#cb206-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-11"><a href="#cb206-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-12"><a href="#cb206-12" aria-hidden="true" tabindex="-1"></a><span class="co"># use all CellChatDB for cell-cell communication analysis</span></span>
<span id="cb206-13"><a href="#cb206-13" aria-hidden="true" tabindex="-1"></a><span class="co"># CellChatDB.use &lt;- CellChatDB # simply use the default CellChatDB. We do not suggest to use it in this way because CellChatDB v2 includes "Non-protein Signaling" (i.e., metabolic and synaptic signaling). </span></span>
<span id="cb206-14"><a href="#cb206-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-15"><a href="#cb206-15" aria-hidden="true" tabindex="-1"></a><span class="co"># set the used database in the object</span></span>
<span id="cb206-16"><a href="#cb206-16" aria-hidden="true" tabindex="-1"></a>cellchat<span class="sc">@</span>DB <span class="ot">&lt;-</span> CellChatDB.use</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="part-1" class="level2">
<h2 class="anchored" data-anchor-id="part-1">Part 1</h2>
<section id="read-in-merged-cellchat-object" class="level3">
<h3 class="anchored" data-anchor-id="read-in-merged-cellchat-object">Read in merged cellchat object</h3>
<p>In order to contrast the controls and cases one needs to process them separately and then merge the 2 cellchat objects, see <a href="https://htmlpreview.github.io/?https://github.com/jinworks/CellChat/blob/master/tutorial/Comparison_analysis_of_multiple_datasets.html">this vignette</a> for more details. Ive done all that in <code>_targets.R</code> so we just need to load the object here.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(cellchat)</span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a><span class="co"># check group numbers</span></span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cellchat<span class="sc">@</span>idents)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "case"    "control" "joint"  </code></pre>
</div>
</div></div>
</section>
<section id="identify-altered-interactions-and-cell-populations" class="level3">
<h3 class="anchored" data-anchor-id="identify-altered-interactions-and-cell-populations">Identify altered interactions and cell populations</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a>ptm <span class="ot">=</span> <span class="fu">Sys.time</span>()</span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">compareInteractions</span>(cellchat, <span class="at">show.legend =</span> F, <span class="at">group =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb209-3"><a href="#cb209-3" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">compareInteractions</span>(cellchat, <span class="at">show.legend =</span> F, <span class="at">group =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">measure =</span> <span class="st">"weight"</span>)</span>
<span id="cb209-4"><a href="#cb209-4" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-13-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="01_cellchat_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="heatmap-showing-differential-number-of-interactions-or-interaction-strength-among-different-cell-populations-across-two-datasets" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-showing-differential-number-of-interactions-or-interaction-strength-among-different-cell-populations-across-two-datasets">Heatmap showing differential number of interactions or interaction strength among different cell populations across two datasets</h3>
<p>Note that red colours denote increased signalling in the second dataset compared to the first, and blue the inverse. In our case cases are the first group and controls the second.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">netVisual_heatmap</span>(cellchat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Do heatmap based on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">netVisual_heatmap</span>(cellchat, <span class="at">measure =</span> <span class="st">"weight"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Do heatmap based on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: Relative values</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `legend_height` you specified is too small, use the default minimal
height.
Warning: `legend_height` you specified is too small, use the default minimal
height.
Warning: `legend_height` you specified is too small, use the default minimal
height.</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-14-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="01_cellchat_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb217"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/cellchat/differential_heatmap.svg"</span>), <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb217-2"><a href="#cb217-2" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: Relative values
Warning: `legend_height` you specified is too small, use the default minimal
height.
Warning: `legend_height` you specified is too small, use the default minimal
height.
Warning: `legend_height` you specified is too small, use the default minimal
height.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
</section>
</section>
<section id="part-ii---identify-altered-signaling-with-distinct-network-architecture-and-interaction-strength" class="level2">
<h2 class="anchored" data-anchor-id="part-ii---identify-altered-signaling-with-distinct-network-architecture-and-interaction-strength">Part II - Identify altered signaling with distinct network architecture and interaction strength</h2>
<section id="identify-signaling-networks-with-larger-or-less-difference-as-well-as-signaling-groups-based-on-their-functionalstructure-similarity" class="level3">
<h3 class="anchored" data-anchor-id="identify-signaling-networks-with-larger-or-less-difference-as-well-as-signaling-groups-based-on-their-functionalstructure-similarity">Identify signaling networks with larger (or less) difference as well as signaling groups based on their functional/structure similarity</h3>
<section id="identify-signaling-groups-based-on-their-functional-similarity" class="level4">
<h4 class="anchored" data-anchor-id="identify-signaling-groups-based-on-their-functional-similarity">Identify signaling groups based on their functional similarity</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb221"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb221-1"><a href="#cb221-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb221-2"><a href="#cb221-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(cellchat_case)</span>
<span id="cb221-3"><a href="#cb221-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(cellchat_control)</span>
<span id="cb221-4"><a href="#cb221-4" aria-hidden="true" tabindex="-1"></a>object.list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Case =</span> cellchat_case, <span class="at">Control =</span> cellchat_control)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb222-2"><a href="#cb222-2" aria-hidden="true" tabindex="-1"></a>ptm <span class="ot">=</span> <span class="fu">Sys.time</span>()</span>
<span id="cb222-3"><a href="#cb222-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb222-4"><a href="#cb222-4" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">computeNetSimilarityPairwise</span>(cellchat, <span class="at">type =</span> <span class="st">"functional"</span>)</span>
<span id="cb222-5"><a href="#cb222-5" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">netEmbedding</span>(cellchat, <span class="at">type =</span> <span class="st">"functional"</span>)</span>
<span id="cb222-6"><a href="#cb222-6" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">netClustering</span>(cellchat, <span class="at">type =</span> <span class="st">"functional"</span>)</span>
<span id="cb222-7"><a href="#cb222-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization in 2D-space</span></span>
<span id="cb222-8"><a href="#cb222-8" aria-hidden="true" tabindex="-1"></a><span class="fu">netVisual_embeddingPairwise</span>(cellchat, <span class="at">type =</span> <span class="st">"functional"</span>, <span class="at">label.size =</span> <span class="fl">3.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="identify-signaling-groups-based-on-structure-similarity" class="level4">
<h4 class="anchored" data-anchor-id="identify-signaling-groups-based-on-structure-similarity">Identify signaling groups based on structure similarity</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb223"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb223-1"><a href="#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb223-2"><a href="#cb223-2" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">computeNetSimilarityPairwise</span>(cellchat, <span class="at">type =</span> <span class="st">"structural"</span>)</span>
<span id="cb223-3"><a href="#cb223-3" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">netEmbedding</span>(cellchat, <span class="at">type =</span> <span class="st">"structural"</span>)</span>
<span id="cb223-4"><a href="#cb223-4" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">netClustering</span>(cellchat, <span class="at">type =</span> <span class="st">"structural"</span>)</span>
<span id="cb223-5"><a href="#cb223-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization in 2D-space</span></span>
<span id="cb223-6"><a href="#cb223-6" aria-hidden="true" tabindex="-1"></a><span class="fu">netVisual_embeddingPairwise</span>(cellchat, <span class="at">type =</span> <span class="st">"structural"</span>, <span class="at">label.size =</span> <span class="fl">3.5</span>)</span>
<span id="cb223-7"><a href="#cb223-7" aria-hidden="true" tabindex="-1"></a><span class="fu">netVisual_embeddingPairwiseZoomIn</span>(cellchat, <span class="at">type =</span> <span class="st">"structural"</span>, <span class="at">nCol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="identify-altered-signaling-with-distinct-interaction-strength" class="level3">
<h3 class="anchored" data-anchor-id="identify-altered-signaling-with-distinct-interaction-strength">Identify altered signaling with distinct interaction strength</h3>
<section id="compare-the-overall-information-flow-of-each-signaling-pathway-or-ligand-receptor-pair" class="level4">
<h4 class="anchored" data-anchor-id="compare-the-overall-information-flow-of-each-signaling-pathway-or-ligand-receptor-pair">Compare the overall information flow of each signaling pathway or ligand-receptor pair</h4>
<p>CellChat can identify the conserved and context-specific signalling pathways by simply comparing the information flow for each signalling pathway, which is defined by the sum of communication probability among all pairs of cell groups in the inferred network (i.e., the total weights in the network).</p>
<p>Note that terms in the following plot highlighted red are significantly enriched in the first condition and the other colour (cyan?) are enriched in the other.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb224"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">rankNet</span>(cellchat, <span class="at">mode =</span> <span class="st">"comparison"</span>, <span class="at">measure =</span> <span class="st">"weight"</span>, <span class="at">sources.use =</span> <span class="cn">NULL</span>, <span class="at">targets.use =</span> <span class="cn">NULL</span>, <span class="at">stacked =</span> T, <span class="at">do.stat =</span> <span class="cn">TRUE</span>)</span>
<span id="cb224-2"><a href="#cb224-2" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">rankNet</span>(cellchat, <span class="at">mode =</span> <span class="st">"comparison"</span>, <span class="at">measure =</span> <span class="st">"weight"</span>, <span class="at">sources.use =</span> <span class="cn">NULL</span>, <span class="at">targets.use =</span> <span class="cn">NULL</span>, <span class="at">stacked =</span> F, <span class="at">do.stat =</span> <span class="cn">TRUE</span>)</span>
<span id="cb224-3"><a href="#cb224-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-4"><a href="#cb224-4" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-18-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="01_cellchat_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/cellchat/differential_information_flow.svg"</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span>
<span id="cb225-3"><a href="#cb225-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
</section>
<section id="compare-outgoing-or-incoming-signaling-patterns-associated-with-each-cell-population" class="level4">
<h4 class="anchored" data-anchor-id="compare-outgoing-or-incoming-signaling-patterns-associated-with-each-cell-population">Compare outgoing (or incoming) signaling patterns associated with each cell population</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a>object.list <span class="ot">&lt;-</span> <span class="fu">map</span>(object.list, netAnalysis_computeCentrality)</span>
<span id="cb227-2"><a href="#cb227-2" aria-hidden="true" tabindex="-1"></a>i <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb227-3"><a href="#cb227-3" aria-hidden="true" tabindex="-1"></a><span class="co"># combining all the identified signaling pathways from different datasets </span></span>
<span id="cb227-4"><a href="#cb227-4" aria-hidden="true" tabindex="-1"></a>pathway.union <span class="ot">&lt;-</span> <span class="fu">union</span>(object.list[[i]]<span class="sc">@</span>netP<span class="sc">$</span>pathways, object.list[[i<span class="sc">+</span><span class="dv">1</span>]]<span class="sc">@</span>netP<span class="sc">$</span>pathways)</span>
<span id="cb227-5"><a href="#cb227-5" aria-hidden="true" tabindex="-1"></a>ht1 <span class="ot">=</span> <span class="fu">netAnalysis_signalingRole_heatmap</span>(object.list[[i]], <span class="at">pattern =</span> <span class="st">"outgoing"</span>, <span class="at">signaling =</span> pathway.union, <span class="at">title =</span> <span class="fu">names</span>(object.list)[i], <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">12</span>)</span>
<span id="cb227-6"><a href="#cb227-6" aria-hidden="true" tabindex="-1"></a>ht2 <span class="ot">=</span> <span class="fu">netAnalysis_signalingRole_heatmap</span>(object.list[[i<span class="sc">+</span><span class="dv">1</span>]], <span class="at">pattern =</span> <span class="st">"outgoing"</span>, <span class="at">signaling =</span> pathway.union, <span class="at">title =</span> <span class="fu">names</span>(object.list)[i<span class="sc">+</span><span class="dv">1</span>], <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">12</span>)</span>
<span id="cb227-7"><a href="#cb227-7" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht1 <span class="sc">+</span> ht2, <span class="at">ht_gap =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-19-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="01_cellchat_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/cellchat/differential_heatmap_incoming.svg"</span>), <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">10</span>)</span>
<span id="cb228-2"><a href="#cb228-2" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht1 <span class="sc">+</span> ht2, <span class="at">ht_gap =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>))</span>
<span id="cb228-3"><a href="#cb228-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a>ht1 <span class="ot">=</span> <span class="fu">netAnalysis_signalingRole_heatmap</span>(object.list[[i]], <span class="at">pattern =</span> <span class="st">"incoming"</span>, <span class="at">signaling =</span> pathway.union, <span class="at">title =</span> <span class="fu">names</span>(object.list)[i], <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">color.heatmap =</span> <span class="st">"GnBu"</span>)</span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a>ht2 <span class="ot">=</span> <span class="fu">netAnalysis_signalingRole_heatmap</span>(object.list[[i<span class="sc">+</span><span class="dv">1</span>]], <span class="at">pattern =</span> <span class="st">"incoming"</span>, <span class="at">signaling =</span> pathway.union, <span class="at">title =</span> <span class="fu">names</span>(object.list)[i<span class="sc">+</span><span class="dv">1</span>], <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">color.heatmap =</span> <span class="st">"GnBu"</span>)</span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht1 <span class="sc">+</span> ht2, <span class="at">ht_gap =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-20-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="01_cellchat_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb231"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb231-1"><a href="#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/cellchat/differential_heatmap_outgoing.svg"</span>), <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb231-2"><a href="#cb231-2" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht1 <span class="sc">+</span> ht2, <span class="at">ht_gap =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>))</span>
<span id="cb231-3"><a href="#cb231-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a>ht1 <span class="ot">=</span> <span class="fu">netAnalysis_signalingRole_heatmap</span>(object.list[[i]], <span class="at">pattern =</span> <span class="st">"all"</span>, <span class="at">signaling =</span> pathway.union, <span class="at">title =</span> <span class="fu">names</span>(object.list)[i], <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">color.heatmap =</span> <span class="st">"OrRd"</span>)</span>
<span id="cb233-2"><a href="#cb233-2" aria-hidden="true" tabindex="-1"></a>ht2 <span class="ot">=</span> <span class="fu">netAnalysis_signalingRole_heatmap</span>(object.list[[i<span class="sc">+</span><span class="dv">1</span>]], <span class="at">pattern =</span> <span class="st">"all"</span>, <span class="at">signaling =</span> pathway.union, <span class="at">title =</span> <span class="fu">names</span>(object.list)[i<span class="sc">+</span><span class="dv">1</span>], <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">color.heatmap =</span> <span class="st">"OrRd"</span>)</span>
<span id="cb233-3"><a href="#cb233-3" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht1 <span class="sc">+</span> ht2, <span class="at">ht_gap =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-21-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="01_cellchat_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/cellchat/differential_heatmap_overall.svg"</span>), <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht1 <span class="sc">+</span> ht2, <span class="at">ht_gap =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>))</span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
</section>
</section>
</section>
<section id="part-iii---identify-the-up-gulated-and-down-regulated-signaling-ligand-receptor-pairs" class="level2">
<h2 class="anchored" data-anchor-id="part-iii---identify-the-up-gulated-and-down-regulated-signaling-ligand-receptor-pairs">Part III - Identify the up-gulated and down-regulated signaling ligand-receptor pairs</h2>
<section id="identify-dysfunctional-signaling-by-comparing-the-communication-probabities" class="level3">
<h3 class="anchored" data-anchor-id="identify-dysfunctional-signaling-by-comparing-the-communication-probabities">Identify dysfunctional signaling by comparing the communication probabities</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get celltype index</span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(cellchat<span class="sc">@</span>idents<span class="sc">$</span>case)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "T-Pericyte"                          "M-Pericyte"                         
 [3] "Pericyte"                            "Pericyte-2"                         
 [5] "Oligo-A"                             "Astrocyte-activated"                
 [7] "Vascular-SMC"                        "Arteriolar-SMC"                     
 [9] "Vascular-SMC-LINC00486"              "Trans-Endo-to-mural"                
[11] "Oligo-B"                             "Perivascular-FB-KAZN2"              
[13] "Perivascular-FB"                     "Meningeal-FB"                       
[15] "Artirial-FB"                         "Pericyte-FB"                        
[17] "Capillary"                           "Arterial"                           
[19] "Venous"                              "T-cell-mixed"                       
[21] "T-cell-vascular"                     "T-cell-parenchymal-1"               
[23] "T-cell-parenchymal-2"                "Microglia-quiescent"                
[25] "Microglia-activated"                 "Perivascular Macrophage"            
[27] "Microglia-vascular"                  "Astrocyte-quiescent"                
[29] "OPC-A"                               "OPC-B"                              
[31] "Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2" "Ex-Neuron-ADARB2-ERBB4"             
[33] "In-Neuron-PRKG1-TBC1D4"              "In-Neuron-ADARB2-LAMP5-NRG1"        
[35] "Ex-Neuron-L4-5-RORB-IL1RAPL2"        "Ex-Neuron-L6-THEMIS-NFIA"           
[37] "Ex-Neuron-L3-5-RORB-PLCH1"           "Ex-Neuron-DLC1-HS3ST4"              
[39] "Ex-Neuron-CBLN2-NRGN"                "In-Neuron-ADARB2-GRM7"              
[41] "In-Neuron-TRHDE-RALYL"               "In-Neuron-ADARB2-IL1RAPL2-CSCL14"   
[43] "In-Neuron-ADARB2-NRG1-RELN"          "In-Neuron-DPP10-MEF2C"              </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">are_equal</span>(<span class="fu">levels</span>(cellchat<span class="sc">@</span>idents<span class="sc">$</span>case), <span class="fu">levels</span>(cellchat<span class="sc">@</span>idents<span class="sc">$</span>control))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at interactions between pericyte 2, microglia activated and perivascular FB KAZN</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a>celltypes_of_interest_plot <span class="ot">&lt;-</span></span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">netVisual_bubble</span>(</span>
<span id="cb240-4"><a href="#cb240-4" aria-hidden="true" tabindex="-1"></a>    cellchat,</span>
<span id="cb240-5"><a href="#cb240-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">sources.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>),</span>
<span id="cb240-6"><a href="#cb240-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">targets.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>),</span>
<span id="cb240-7"><a href="#cb240-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),</span>
<span id="cb240-8"><a href="#cb240-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle.x =</span> <span class="dv">45</span></span>
<span id="cb240-9"><a href="#cb240-9" aria-hidden="true" tabindex="-1"></a>    <span class="co">#remove.isolate = FALSE</span></span>
<span id="cb240-10"><a href="#cb240-10" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Comparing communications on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(</span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb242-3"><a href="#cb242-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"05_figures/990_shared_figures/cellchat"</span>,</span>
<span id="cb242-4"><a href="#cb242-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"pericyte2_microglia-active_peri-fb_interactions.png"</span></span>
<span id="cb242-5"><a href="#cb242-5" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb242-6"><a href="#cb242-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb242-7"><a href="#cb242-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb242-8"><a href="#cb242-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb242-9"><a href="#cb242-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">res =</span> <span class="dv">300</span></span>
<span id="cb242-10"><a href="#cb242-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb242-11"><a href="#cb242-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(celltypes_of_interest_plot)</span>
<span id="cb242-12"><a href="#cb242-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">netVisual_bubble</span>(cellchat, <span class="at">sources.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>), <span class="at">targets.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>),  <span class="at">comparison =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">max.dataset =</span> <span class="dv">2</span>, <span class="at">title.name =</span> <span class="st">"Increased signaling in Controls"</span>, <span class="at">angle.x =</span> <span class="dv">45</span>, <span class="at">remove.isolate =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Comparing communications on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">netVisual_bubble</span>(cellchat, <span class="at">sources.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>), <span class="at">targets.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>),  <span class="at">comparison =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">max.dataset =</span> <span class="dv">1</span>, <span class="at">title.name =</span> <span class="st">"Decreased signaling in Controls"</span>, <span class="at">angle.x =</span> <span class="dv">45</span>, <span class="at">remove.isolate =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Comparing communications on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-23-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="01_cellchat_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(</span>
<span id="cb249-2"><a href="#cb249-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb249-3"><a href="#cb249-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"05_figures/990_shared_figures/cellchat"</span>,</span>
<span id="cb249-4"><a href="#cb249-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"pericyte2_microglia-active_peri-fb_interactions_differential.png"</span></span>
<span id="cb249-5"><a href="#cb249-5" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb249-6"><a href="#cb249-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb249-7"><a href="#cb249-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb249-8"><a href="#cb249-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb249-9"><a href="#cb249-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">res =</span> <span class="dv">300</span></span>
<span id="cb249-10"><a href="#cb249-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb249-11"><a href="#cb249-11" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span>
<span id="cb249-12"><a href="#cb249-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
</section>
<section id="identify-dysfunctional-signaling-by-using-differential-expression-analysis" class="level3">
<h3 class="anchored" data-anchor-id="identify-dysfunctional-signaling-by-using-differential-expression-analysis">Identify dysfunctional signaling by using differential expression analysis</h3>
<p>The above method for identifying the upgulated and down-regulated signaling is perfomed by comparing the communication probability between two datasets for each L-R pair and each pair of cell groups. Alternative, we can identify the upgulated and down-regulated signaling ligand-receptor pairs based on the differential expression analysis (DEA). Specifically, we perform differential expression analysis between two biological conditions (i.e., NL and LS) for each cell group, and then obtain the upgulated and down-regulated signaling based on the fold change of ligands in the sender cells and receptors in the receiver cells.</p>
<p>Of note, users may observe the same LR pairs appearing in both the up-regulated and down-regulated results due to the fact that DEA between conditions is performed for each cell group. To perform DEA between conditions by ignoring cell group information, users can set group.DE.combined = TRUE in identifyOverExpressedGenes for CellChat v2.1.1.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="co"># define a positive dataset, i.e., the dataset with positive fold change against the other dataset</span></span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>pos.dataset <span class="ot">=</span> <span class="st">"case"</span></span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a><span class="co"># define a char name used for storing the results of differential expression analysis</span></span>
<span id="cb251-4"><a href="#cb251-4" aria-hidden="true" tabindex="-1"></a>features.name <span class="ot">=</span> <span class="fu">paste0</span>(pos.dataset, <span class="st">".merged"</span>)</span>
<span id="cb251-5"><a href="#cb251-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-6"><a href="#cb251-6" aria-hidden="true" tabindex="-1"></a><span class="co"># perform differential expression analysis </span></span>
<span id="cb251-7"><a href="#cb251-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Of note, compared to CellChat version &lt; v2, CellChat v2 now performs an ultra-fast Wilcoxon test using the presto package, which gives smaller values of logFC. Thus we here set a smaller value of thresh.fc compared to the original one (thresh.fc = 0.1). Users can also provide a vector and dataframe of customized DEGs by modifying the cellchat@var.features$LS.merged and cellchat@var.features$LS.merged.info. </span></span>
<span id="cb251-8"><a href="#cb251-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-9"><a href="#cb251-9" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">identifyOverExpressedGenes</span>(cellchat, <span class="at">group.dataset =</span> <span class="st">"datasets"</span>, <span class="at">pos.dataset =</span> pos.dataset, <span class="at">features.name =</span> features.name, <span class="at">only.pos =</span> <span class="cn">FALSE</span>, <span class="at">thresh.pc =</span> <span class="fl">0.1</span>, <span class="at">thresh.fc =</span> <span class="fl">0.05</span>,<span class="at">thresh.p =</span> <span class="fl">0.05</span>, <span class="at">group.DE.combined =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Use the joint cell labels from the merged CellChat object</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="co"># map the results of differential expression analysis onto the inferred cell-cell communications to easily manage/subset the ligand-receptor pairs of interest</span></span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">netMappingDEG</span>(cellchat, <span class="at">features.name =</span> features.name, <span class="at">variable.all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the ligand-receptor pairs with upregulated ligands in LS</span></span>
<span id="cb253-4"><a href="#cb253-4" aria-hidden="true" tabindex="-1"></a>net.up <span class="ot">&lt;-</span> <span class="fu">subsetCommunication</span>(cellchat, <span class="at">net =</span> net, <span class="at">datasets =</span> <span class="st">"case"</span>,<span class="at">ligand.logFC =</span> <span class="fl">0.05</span>, <span class="at">receptor.logFC =</span> <span class="cn">NULL</span>)</span>
<span id="cb253-5"><a href="#cb253-5" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the ligand-receptor pairs with upregulated ligands and upregulated receptors in NL, i.e.,downregulated in LS</span></span>
<span id="cb253-6"><a href="#cb253-6" aria-hidden="true" tabindex="-1"></a>net.down <span class="ot">&lt;-</span> <span class="fu">subsetCommunication</span>(cellchat, <span class="at">net =</span> net, <span class="at">datasets =</span> <span class="st">"control"</span>,<span class="at">ligand.logFC =</span> <span class="sc">-</span><span class="fl">0.05</span>, <span class="at">receptor.logFC =</span> <span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Since the signaling genes in the <code>net.up</code> and <code>net.down</code> might be complex with multi-subunits, we can do further deconvolution to obtain the individual signaling genes.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a>gene.up <span class="ot">&lt;-</span> <span class="fu">extractGeneSubsetFromPair</span>(net.up, cellchat)</span>
<span id="cb254-2"><a href="#cb254-2" aria-hidden="true" tabindex="-1"></a>gene.down <span class="ot">&lt;-</span> <span class="fu">extractGeneSubsetFromPair</span>(net.down, cellchat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Users can also find all the significant outgoing/incoming/both signaling according to the customized features and cell groups of interest</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">findEnrichedSignaling</span>(object.list[[<span class="dv">2</span>]], <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"CCL19"</span>, <span class="st">"CXCL12"</span>), <span class="at">idents =</span> <span class="fu">c</span>(<span class="st">"Pericyte-2"</span>, <span class="st">"Microglia-activated"</span>, <span class="st">"Perivascular-FB-KAZN2"</span>), <span class="at">pattern =</span><span class="st">"outgoing"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="visualize-the-identified-up-regulated-and-down-regulated-signaling-ligand-receptor-pairs" class="level4">
<h4 class="anchored" data-anchor-id="visualize-the-identified-up-regulated-and-down-regulated-signaling-ligand-receptor-pairs">Visualize the identified up-regulated and down-regulated signaling ligand-receptor pairs</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a>pairLR.use.up <span class="ot">=</span> net.up[, <span class="st">"interaction_name"</span>, drop <span class="ot">=</span> F]</span>
<span id="cb256-2"><a href="#cb256-2" aria-hidden="true" tabindex="-1"></a>gg1 <span class="ot">&lt;-</span> <span class="fu">netVisual_bubble</span>(cellchat, <span class="at">pairLR.use =</span> pairLR.use.up, <span class="at">sources.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>), <span class="at">targets.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>), <span class="at">comparison =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),  <span class="at">angle.x =</span> <span class="dv">45</span>, <span class="at">remove.isolate =</span> T,<span class="at">title.name =</span> <span class="fu">paste0</span>(<span class="st">"Up-regulated signaling in "</span>, <span class="fu">names</span>(object.list)[<span class="dv">2</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Comparing communications on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb258"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb258-1"><a href="#cb258-1" aria-hidden="true" tabindex="-1"></a>pairLR.use.down <span class="ot">=</span> net.down[, <span class="st">"interaction_name"</span>, drop <span class="ot">=</span> F]</span>
<span id="cb258-2"><a href="#cb258-2" aria-hidden="true" tabindex="-1"></a>gg2 <span class="ot">&lt;-</span> <span class="fu">netVisual_bubble</span>(cellchat, <span class="at">pairLR.use =</span> pairLR.use.down, <span class="at">sources.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>), <span class="at">targets.use =</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">25</span>, <span class="dv">12</span>), <span class="at">comparison =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>),  <span class="at">angle.x =</span> <span class="dv">45</span>, <span class="at">remove.isolate =</span> T,<span class="at">title.name =</span> <span class="fu">paste0</span>(<span class="st">"Down-regulated signaling in "</span>, <span class="fu">names</span>(object.list)[<span class="dv">2</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Comparing communications on a merged object </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="01_cellchat_files/figure-html/unnamed-chunk-27-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="01_cellchat_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(</span>
<span id="cb261-2"><a href="#cb261-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb261-3"><a href="#cb261-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"05_figures/990_shared_figures/cellchat"</span>,</span>
<span id="cb261-4"><a href="#cb261-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"pericyte2_microglia-active_peri-fb_interactions_differential2.png"</span></span>
<span id="cb261-5"><a href="#cb261-5" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb261-6"><a href="#cb261-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb261-7"><a href="#cb261-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">12</span>,</span>
<span id="cb261-8"><a href="#cb261-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb261-9"><a href="#cb261-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">res =</span> <span class="dv">300</span></span>
<span id="cb261-10"><a href="#cb261-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb261-11"><a href="#cb261-11" aria-hidden="true" tabindex="-1"></a>gg1 <span class="sc">+</span> gg2</span>
<span id="cb261-12"><a href="#cb261-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
</section>
</section>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"loop":false,"openEffect":"zoom","descPosition":"bottom","selector":".lightbox","closeEffect":"zoom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>