<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.6.40">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

    <meta name="author" content="Gabriel Mateus Bernardo Harrington">
    <meta name="author" content="Hannah Sleven">
    <meta name="author" content="Jimena Monzón-Sandoval">
    <meta name="author" content="Gabriel Rocha">
    <meta name="author" content="Lara Robinson">
    <meta name="author" content="Michal Rokicki">
    <meta name="author" content="Joanne Morgan">
    <meta name="author" content="Ngoc-Nga Vinh">
    <meta name="author" content="Quenten Schwarz">
    <meta name="author" content="Caleb Webber">
    <meta name="author" content="Zameel Cader">
    <meta name="dcterms.date" content="2025-05-30">
    <meta name="keywords" content="Alzheimer’s diesease, Blood brain barrier">

    <title>Paired Vascular-Parenchymal Single Nuclei Sequencing Reveals Perturbed Endothelial-to-Mesenchymal Transition in Alzheimer’s Disease</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      /* CSS for citations */
      div.csl-bib-body { }
      div.csl-entry {
        clear: both;
      }
      .hanging-indent div.csl-entry {
        margin-left:2em;
        text-indent:-2em;
      }
      div.csl-left-margin {
        min-width:2em;
        float:left;
      }
      div.csl-right-inline {
        margin-left:2em;
        padding-left:1em;
      }
      div.csl-indent {
        margin-left: 2em;
      }    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-d613744cf535a6fbd5c002787a2fa9e9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  <link href="site_libs/tabwid-1.1.3/tabwid.css" rel="stylesheet">
<script src="site_libs/tabwid-1.1.3/tabwid.js"></script>   <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script> 
      <meta name="citation_title" content="Paired Vascular-Parenchymal Single Nuclei Sequencing Reveals Perturbed Endothelial-to-Mesenchymal Transition in Alzheimer&amp;amp;#039;s Disease">
<meta name="citation_abstract" content="The neurovascular unit (NVU) underpins critical brain functions by integrating vascular cells, neurons, and glia to maintain metabolic support and blood–brain barrier integrity. Despite mounting evidence implicating vascular dysfunction in Alzheimer&amp;amp;#039;s disease (AD), the contribution of vascular cell types remains underexplored, largely owing to technical challenges in isolating these cells from post-mortem tissue. Here, we describe a novel approach that separately enriches for vascular and parenchymal nuclei at high purity from a single sample of human brain tissue. By applying this method to prefrontal cortex tissue from AD patients and controls and performing single-nuclei RNA sequencing, we assembled a high-resolution atlas capturing major vascular and parenchymal cell types.
This unbiased approach reveals distinct associations of AD risk with pericyte-, perivascular fibroblast-, and activated microglia-specific gene expression profiles. Functional analyses show that the genes driving these associations converge on amyloid-related pathways, implicating perivascular compartments and immune modulators in the early stages of AD pathobiology. Furthermore, we identify a new endothelial-mesenchymal transition population, diminished in AD suggesting a disruption in endothelial plasticity that may underlie vascular dysfunction in AD. Key ligand–receptor interactions between vascular and parenchymal cell types, including ANGPT2, were markedly dysregulated, highlighting specific targets that could drive NVU breakdown in AD.
Taken together, these findings establish a framework for comprehensively evaluating NVU cells in human disease, emphasizing novel links between vascular dynamics and AD pathogenesis. Our method provides an essential foundation for deciphering disease mechanisms and offers new therapeutic avenues focused on protecting and restoring the brain's neurovascular interface.
">
<meta name="citation_keywords" content="Alzheimer's diesease,Blood brain barrier">
<meta name="citation_author" content="Gabriel Mateus Bernardo Harrington">
<meta name="citation_author" content="Hannah Sleven">
<meta name="citation_author" content="Jimena Monzón-Sandoval">
<meta name="citation_author" content="Gabriel Rocha">
<meta name="citation_author" content="Lara Robinson">
<meta name="citation_author" content="Michal Rokicki">
<meta name="citation_author" content="Joanne Morgan">
<meta name="citation_author" content="Ngoc-Nga Vinh">
<meta name="citation_author" content="Quenten Schwarz">
<meta name="citation_author" content="Caleb Webber">
<meta name="citation_author" content="Zameel Cader">
<meta name="citation_publication_date" content="2025-05-30">
<meta name="citation_cover_date" content="2025-05-30">
<meta name="citation_year" content="2025">
<meta name="citation_online_date" content="2025-05-30">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=New insights into the genetic etiology of Alzheimer’s disease and related dementias;,citation_author=Céline Bellenguez;,citation_author=Fahri Küçükali;,citation_author=Iris E. Jansen;,citation_author=Luca Kleineidam;,citation_author=Sonia Moreno-Grau;,citation_author=Najaf Amin;,citation_author=Adam C. Naj;,citation_author=Rafael Campos-Martin;,citation_author=Benjamin Grenier-Boley;,citation_author=Victor Andrade;,citation_author=Peter A. Holmans;,citation_author=Anne Boland;,citation_author=Vincent Damotte;,citation_author=Sven J. Lee;,citation_author=Marcos R. Costa;,citation_author=Teemu Kuulasmaa;,citation_author=Qiong Yang;,citation_author=Itziar Rojas;,citation_author=Joshua C. Bis;,citation_author=Amber Yaqub;,citation_author=Ivana Prokic;,citation_author=Julien Chapuis;,citation_author=Shahzad Ahmad;,citation_author=Vilmantas Giedraitis;,citation_author=Dag Aarsland;,citation_author=Pablo Garcia-Gonzalez;,citation_author=Carla Abdelnour;,citation_author=Emilio Alarcón-Martín;,citation_author=Daniel Alcolea;,citation_author=Montserrat Alegret;,citation_author=Ignacio Alvarez;,citation_author=Victoria Álvarez;,citation_author=Nicola J. Armstrong;,citation_author=Anthoula Tsolaki;,citation_author=Carmen Antúnez;,citation_author=Ildebrando Appollonio;,citation_author=Marina Arcaro;,citation_author=Silvana Archetti;,citation_author=Alfonso Arias Pastor;,citation_author=Beatrice Arosio;,citation_author=Lavinia Athanasiu;,citation_author=Henri Bailly;,citation_author=Nerisa Banaj;,citation_author=Miquel Baquero;,citation_author=Sandra Barral;,citation_author=Alexa Beiser;,citation_author=Ana Belén Pastor;,citation_author=Jennifer E. Below;,citation_author=Penelope Benchek;,citation_author=Luisa Benussi;,citation_author=Claudine Berr;,citation_author=Céline Besse;,citation_author=Valentina Bessi;,citation_author=Giuliano Binetti;,citation_author=Alessandra Bizarro;,citation_author=Rafael Blesa;,citation_author=Mercè Boada;,citation_author=Eric Boerwinkle;,citation_author=Barbara Borroni;,citation_author=Silvia Boschi;,citation_author=Paola Bossù;,citation_author=Geir Bråthen;,citation_author=Jan Bressler;,citation_author=Catherine Bresner;,citation_author=Henry Brodaty;,citation_author=Keeley J. Brookes;,citation_author=Luis Ignacio Brusco;,citation_author=Dolores Buiza-Rueda;,citation_author=Katharina Bûrger;,citation_author=Vanessa Burholt;,citation_author=William S. Bush;,citation_author=Miguel Calero;,citation_author=Laura B. Cantwell;,citation_author=Geneviève Chene;,citation_author=Jaeyoon Chung;,citation_author=Michael L. Cuccaro;,citation_author=Ángel Carracedo;,citation_author=Roberta Cecchetti;,citation_author=Laura Cervera-Carles;,citation_author=Camille Charbonnier;,citation_author=Hung-Hsin Chen;,citation_author=Caterina Chillotti;,citation_author=Simona Ciccone;,citation_author=Jurgen A. H. R. Claassen;,citation_author=Christopher Clark;,citation_author=Elisa Conti;,citation_author=Anaïs Corma-Gómez;,citation_author=Emanuele Costantini;,citation_author=Carlo Custodero;,citation_author=Delphine Daian;,citation_author=Maria Carolina Dalmasso;,citation_author=Antonio Daniele;,citation_author=Efthimios Dardiotis;,citation_author=Jean-François Dartigues;,citation_author=Peter Paul Deyn;,citation_author=Katia Paiva Lopes;,citation_author=Lot D. Witte;,citation_author=Stéphanie Debette;,citation_author=Jürgen Deckert;,citation_author=Teodoro Ser;,citation_author=Nicola Denning;,citation_author=Anita DeStefano;,citation_author=Martin Dichgans;,citation_author=Janine Diehl-Schmid;,citation_author=Mónica Diez-Fairen;,citation_author=Paolo Dionigi Rossi;,citation_author=Srdjan Djurovic;,citation_author=Emmanuelle Duron;,citation_author=Emrah Düzel;,citation_author=Carole Dufouil;,citation_author=Gudny Eiriksdottir;,citation_author=Sebastiaan Engelborghs;,citation_author=Valentina Escott-Price;,citation_author=Ana Espinosa;,citation_author=Michael Ewers;,citation_author=Kelley M. Faber;,citation_author=Tagliavini Fabrizio;,citation_author=Sune Fallgaard Nielsen;,citation_author=David W. Fardo;,citation_author=Lucia Farotti;,citation_author=Chiara Fenoglio;,citation_author=Marta Fernández-Fuertes;,citation_author=Raffaele Ferrari;,citation_author=Catarina B. Ferreira;,citation_author=Evelyn Ferri;,citation_author=Bertrand Fin;,citation_author=Peter Fischer;,citation_author=Tormod Fladby;,citation_author=Klaus Fließbach;,citation_author=Bernard Fongang;,citation_author=Myriam Fornage;,citation_author=Juan Fortea;,citation_author=Tatiana M. Foroud;,citation_author=Silvia Fostinelli;,citation_author=Nick C. Fox;,citation_author=Emlio Franco-Macías;,citation_author=María J. Bullido;,citation_author=Ana Frank-García;,citation_author=Lutz Froelich;,citation_author=Brian Fulton-Howard;,citation_author=Daniela Galimberti;,citation_author=Jose Maria García-Alberca;,citation_author=Pablo García-González;,citation_author=Sebastian Garcia-Madrona;,citation_author=Guillermo Garcia-Ribas;,citation_author=Roberta Ghidoni;,citation_author=Ina Giegling;,citation_author=Giaccone Giorgio;,citation_author=Alison M. Goate;,citation_author=Oliver Goldhardt;,citation_author=Duber Gomez-Fonseca;,citation_author=Antonio González-Pérez;,citation_author=Caroline Graff;,citation_author=Giulia Grande;,citation_author=Emma Green;,citation_author=Timo Grimmer;,citation_author=Edna Grünblatt;,citation_author=Michelle Grunin;,citation_author=Vilmundur Gudnason;,citation_author=Tamar Guetta-Baranes;,citation_author=Annakaisa Haapasalo;,citation_author=Georgios Hadjigeorgiou;,citation_author=Jonathan L. Haines;,citation_author=Kara L. Hamilton-Nelson;,citation_author=Harald Hampel;,citation_author=Olivier Hanon;,citation_author=John Hardy;,citation_author=Annette M. Hartmann;,citation_author=Lucrezia Hausner;,citation_author=Janet Harwood;,citation_author=Stefanie Heilmann-Heimbach;,citation_author=Seppo Helisalmi;,citation_author=Michael T. Heneka;,citation_author=Isabel Hernández;,citation_author=Martin J. Herrmann;,citation_author=Per Hoffmann;,citation_author=Clive Holmes;,citation_author=Henne Holstege;,citation_author=Raquel Huerto Vilas;,citation_author=Marc Hulsman;,citation_author=Jack Humphrey;,citation_author=Geert Jan Biessels;,citation_author=Xueqiu Jian;,citation_author=Charlotte Johansson;,citation_author=Gyungah R. Jun;,citation_author=Yuriko Kastumata;,citation_author=John Kauwe;,citation_author=Patrick G. Kehoe;,citation_author=Lena Kilander;,citation_author=Anne Kinhult Ståhlbom;,citation_author=Miia Kivipelto;,citation_author=Anne Koivisto;,citation_author=Johannes Kornhuber;,citation_author=Mary H. Kosmidis;,citation_author=Walter A. Kukull;,citation_author=Pavel P. Kuksa;,citation_author=Brian W. Kunkle;,citation_author=Amanda B. Kuzma;,citation_author=Carmen Lage;,citation_author=Erika J. Laukka;,citation_author=Lenore Launer;,citation_author=Alessandra Lauria;,citation_author=Chien-Yueh Lee;,citation_author=Jenni Lehtisalo;,citation_author=Ondrej Lerch;,citation_author=Alberto Lleó;,citation_author=William Longstreth;,citation_author=Oscar Lopez;,citation_author=Adolfo Lopez Munain;,citation_author=Seth Love;,citation_author=Malin Löwemark;,citation_author=Lauren Luckcuck;,citation_author=Kathryn L. Lunetta;,citation_author=Yiyi Ma;,citation_author=Juan Macías;,citation_author=Catherine A. MacLeod;,citation_author=Wolfgang Maier;,citation_author=Francesca Mangialasche;,citation_author=Marco Spallazzi;,citation_author=Marta Marquié;,citation_author=Rachel Marshall;,citation_author=Eden R. Martin;,citation_author=Angel Martín Montes;,citation_author=Carmen Martínez Rodríguez;,citation_author=Carlo Masullo;,citation_author=Richard Mayeux;,citation_author=Simon Mead;,citation_author=Patrizia Mecocci;,citation_author=Miguel Medina;,citation_author=Alun Meggy;,citation_author=Shima Mehrabian;,citation_author=Silvia Mendoza;,citation_author=Manuel Menéndez-González;,citation_author=Pablo Mir;,citation_author=Susanne Moebus;,citation_author=Merel Mol;,citation_author=Laura Molina-Porcel;,citation_author=Laura Montrreal;,citation_author=Laura Morelli;,citation_author=Fermin Moreno;,citation_author=Kevin Morgan;,citation_author=Thomas Mosley;,citation_author=Markus M. Nöthen;,citation_author=Carolina Muchnik;,citation_author=Shubhabrata Mukherjee;,citation_author=Benedetta Nacmias;,citation_author=Tiia Ngandu;,citation_author=Gael Nicolas;,citation_author=Børge G. Nordestgaard;,citation_author=Robert Olaso;,citation_author=Adelina Orellana;,citation_author=Michela Orsini;,citation_author=Gemma Ortega;,citation_author=Alessandro Padovani;,citation_author=Caffarra Paolo;,citation_author=Goran Papenberg;,citation_author=Lucilla Parnetti;,citation_author=Florence Pasquier;,citation_author=Pau Pastor;,citation_author=Gina Peloso;,citation_author=Alba Pérez-Cordón;,citation_author=Jordi Pérez-Tur;,citation_author=Pierre Pericard;,citation_author=Oliver Peters;,citation_author=Yolande A. L. Pijnenburg;,citation_author=Juan A. Pineda;,citation_author=Gerard Piñol-Ripoll;,citation_author=Claudia Pisanu;,citation_author=Thomas Polak;,citation_author=Julius Popp;,citation_author=Danielle Posthuma;,citation_author=Josef Priller;,citation_author=Raquel Puerta;,citation_author=Olivier Quenez;,citation_author=Inés Quintela;,citation_author=Jesper Qvist Thomassen;,citation_author=Alberto Rábano;,citation_author=Innocenzo Rainero;,citation_author=Farid Rajabli;,citation_author=Inez Ramakers;,citation_author=Luis M. Real;,citation_author=Marcel J. T. Reinders;,citation_author=Christiane Reitz;,citation_author=Dolly Reyes-Dumeyer;,citation_author=Perry Ridge;,citation_author=Steffi Riedel-Heller;,citation_author=Peter Riederer;,citation_author=Natalia Roberto;,citation_author=Eloy Rodriguez-Rodriguez;,citation_author=Arvid Rongve;,citation_author=Irene Rosas Allende;,citation_author=Maitée Rosende-Roca;,citation_author=Jose Luis Royo;,citation_author=Elisa Rubino;,citation_author=Dan Rujescu;,citation_author=María Eugenia Sáez;,citation_author=Paraskevi Sakka;,citation_author=Ingvild Saltvedt;,citation_author=Ángela Sanabria;,citation_author=María Bernal Sánchez-Arjona;,citation_author=Florentino Sanchez-Garcia;,citation_author=Pascual Sánchez Juan;,citation_author=Raquel Sánchez-Valle;,citation_author=Sigrid B. Sando;,citation_author=Chloé Sarnowski;,citation_author=Claudia L. Satizabal;,citation_author=Michela Scamosci;,citation_author=Nikolaos Scarmeas;,citation_author=Elio Scarpini;,citation_author=Philip Scheltens;,citation_author=Norbert Scherbaum;,citation_author=Martin Scherer;,citation_author=Matthias Schmid;,citation_author=Anja Schneider;,citation_author=Jonathan M. Schott;,citation_author=Geir Selbæk;,citation_author=Davide Seripa;,citation_author=Manuel Serrano;,citation_author=Jin Sha;,citation_author=Alexey A. Shadrin;,citation_author=Olivia Skrobot;,citation_author=Susan Slifer;,citation_author=Gijsje J. L. Snijders;,citation_author=Hilkka Soininen;,citation_author=Vincenzo Solfrizzi;,citation_author=Alina Solomon;,citation_author=Yeunjoo Song;,citation_author=Sandro Sorbi;,citation_author=Oscar Sotolongo-Grau;,citation_author=Gianfranco Spalletta;,citation_author=Annika Spottke;,citation_author=Alessio Squassina;,citation_author=Eystein Stordal;,citation_author=Juan Pablo Tartan;,citation_author=Lluís Tárraga;,citation_author=Niccolo Tesí;,citation_author=Anbupalam Thalamuthu;,citation_author=Tegos Thomas;,citation_author=Giuseppe Tosto;,citation_author=Latchezar Traykov;,citation_author=Lucio Tremolizzo;,citation_author=Anne Tybjærg-Hansen;,citation_author=Andre Uitterlinden;,citation_author=Abbe Ullgren;,citation_author=Ingun Ulstein;,citation_author=Sergi Valero;,citation_author=Otto Valladares;,citation_author=Christine Van Broeckhoven;,citation_author=Jeffery Vance;,citation_author=Badri N. Vardarajan;,citation_author=Aad Lugt;,citation_author=Jasper Van Dongen;,citation_author=Jeroen Rooij;,citation_author=John Swieten;,citation_author=Rik Vandenberghe;,citation_author=Frans Verhey;,citation_author=Jean-Sébastien Vidal;,citation_author=Jonathan Vogelgsang;,citation_author=Martin Vyhnalek;,citation_author=Michael Wagner;,citation_author=David Wallon;,citation_author=Li-San Wang;,citation_author=Ruiqi Wang;,citation_author=Leonie Weinhold;,citation_author=Jens Wiltfang;,citation_author=Gill Windle;,citation_author=Bob Woods;,citation_author=Mary Yannakoulia;,citation_author=Habil Zare;,citation_author=Yi Zhao;,citation_author=Xiaoling Zhang;,citation_author=Congcong Zhu;,citation_author=Miren Zulaica;,citation_author=Lindsay A. Farrer;,citation_author=Bruce M. Psaty;,citation_author=Mohsen Ghanbari;,citation_author=Towfique Raj;,citation_author=Perminder Sachdev;,citation_author=Karen Mather;,citation_author=Frank Jessen;,citation_author=M. Arfan Ikram;,citation_author=Alexandre Mendonça;,citation_author=Jakub Hort;,citation_author=Magda Tsolaki;,citation_author=Margaret A. Pericak-Vance;,citation_author=Philippe Amouyel;,citation_author=Julie Williams;,citation_author=Ruth Frikke-Schmidt;,citation_author=Jordi Clarimon;,citation_author=Jean-François Deleuze;,citation_author=Giacomina Rossi;,citation_author=Sudha Seshadri;,citation_author=Ole A. Andreassen;,citation_author=Martin Ingelsson;,citation_author=Mikko Hiltunen;,citation_author=Kristel Sleegers;,citation_author=Gerard D. Schellenberg;,citation_author=Cornelia M. Duijn;,citation_author=Rebecca Sims;,citation_author=Wiesje M. Flier;,citation_author=Agustín Ruiz;,citation_author=Alfredo Ramirez;,citation_author=Jean-Charles Lambert;,citation_publication_date=2022-04;,citation_cover_date=2022-04;,citation_year=2022;,citation_fulltext_html_url=https://www.nature.com/articles/s41588-022-01024-z;,citation_issue=4;,citation_doi=10.1038/s41588-022-01024-z;,citation_volume=54;,citation_language=en;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=Genetic meta-analysis of diagnosed Alzheimer’s disease identifies new risk loci and implicates Aβ, tau, immunity and lipid processing;,citation_author=Brian W. Kunkle;,citation_author=Benjamin Grenier-Boley;,citation_author=Rebecca Sims;,citation_author=Joshua C. Bis;,citation_author=Vincent Damotte;,citation_author=Adam C. Naj;,citation_author=Anne Boland;,citation_author=Maria Vronskaya;,citation_author=Sven J. Lee;,citation_author=Alexandre Amlie-Wolf;,citation_author=Céline Bellenguez;,citation_author=Aura Frizatti;,citation_author=Vincent Chouraki;,citation_author=Eden R. Martin;,citation_author=Kristel Sleegers;,citation_author=Nandini Badarinarayan;,citation_author=Johanna Jakobsdottir;,citation_author=Kara L. Hamilton-Nelson;,citation_author=Sonia Moreno-Grau;,citation_author=Robert Olaso;,citation_author=Rachel Raybould;,citation_author=Yuning Chen;,citation_author=Amanda B. Kuzma;,citation_author=Mikko Hiltunen;,citation_author=Taniesha Morgan;,citation_author=Shahzad Ahmad;,citation_author=Badri N. Vardarajan;,citation_author=Jacques Epelbaum;,citation_author=Per Hoffmann;,citation_author=Merce Boada;,citation_author=Gary W. Beecham;,citation_author=Jean-Guillaume Garnier;,citation_author=Denise Harold;,citation_author=Annette L. Fitzpatrick;,citation_author=Otto Valladares;,citation_author=Marie-Laure Moutet;,citation_author=Amy Gerrish;,citation_author=Albert V. Smith;,citation_author=Liming Qu;,citation_author=Delphine Bacq;,citation_author=Nicola Denning;,citation_author=Xueqiu Jian;,citation_author=Yi Zhao;,citation_author=Maria Del Zompo;,citation_author=Nick C. Fox;,citation_author=Seung-Hoan Choi;,citation_author=Ignacio Mateo;,citation_author=Joseph T. Hughes;,citation_author=Hieab H. Adams;,citation_author=John Malamon;,citation_author=Florentino Sanchez-Garcia;,citation_author=Yogen Patel;,citation_author=Jennifer A. Brody;,citation_author=Beth A. Dombroski;,citation_author=Maria Candida Deniz Naranjo;,citation_author=Makrina Daniilidou;,citation_author=Gudny Eiriksdottir;,citation_author=Shubhabrata Mukherjee;,citation_author=David Wallon;,citation_author=James Uphill;,citation_author=Thor Aspelund;,citation_author=Laura B. Cantwell;,citation_author=Fabienne Garzia;,citation_author=Daniela Galimberti;,citation_author=Edith Hofer;,citation_author=Mariusz Butkiewicz;,citation_author=Bertrand Fin;,citation_author=Elio Scarpini;,citation_author=Chloe Sarnowski;,citation_author=Will S. Bush;,citation_author=Stéphane Meslage;,citation_author=Johannes Kornhuber;,citation_author=Charles C. White;,citation_author=Yuenjoo Song;,citation_author=Robert C. Barber;,citation_author=Sebastiaan Engelborghs;,citation_author=Sabrina Sordon;,citation_author=Dina Voijnovic;,citation_author=Perrie M. Adams;,citation_author=Rik Vandenberghe;,citation_author=Manuel Mayhaus;,citation_author=L. Adrienne Cupples;,citation_author=Marilyn S. Albert;,citation_author=Peter P. De Deyn;,citation_author=Wei Gu;,citation_author=Jayanadra J. Himali;,citation_author=Duane Beekly;,citation_author=Alessio Squassina;,citation_author=Annette M. Hartmann;,citation_author=Adelina Orellana;,citation_author=Deborah Blacker;,citation_author=Eloy Rodriguez-Rodriguez;,citation_author=Simon Lovestone;,citation_author=Melissa E. Garcia;,citation_author=Rachelle S. Doody;,citation_author=Carmen Munoz-Fernadez;,citation_author=Rebecca Sussams;,citation_author=Honghuang Lin;,citation_author=Thomas J. Fairchild;,citation_author=Yolanda A. Benito;,citation_author=Clive Holmes;,citation_author=Hata Karamujić-Čomić;,citation_author=Matthew P. Frosch;,citation_author=Hakan Thonberg;,citation_author=Wolfgang Maier;,citation_author=Gennady Roshchupkin;,citation_author=Bernardino Ghetti;,citation_author=Vilmantas Giedraitis;,citation_author=Amit Kawalia;,citation_author=Shuo Li;,citation_author=Ryan M. Huebinger;,citation_author=Lena Kilander;,citation_author=Susanne Moebus;,citation_author=Isabel Hernández;,citation_author=M. Ilyas Kamboh;,citation_author=RoseMarie Brundin;,citation_author=James Turton;,citation_author=Qiong Yang;,citation_author=Mindy J. Katz;,citation_author=Letizia Concari;,citation_author=Jenny Lord;,citation_author=Alexa S. Beiser;,citation_author=C. Dirk Keene;,citation_author=Seppo Helisalmi;,citation_author=Iwona Kloszewska;,citation_author=Walter A. Kukull;,citation_author=Anne Maria Koivisto;,citation_author=Aoibhinn Lynch;,citation_author=Lluís Tarraga;,citation_author=Eric B. Larson;,citation_author=Annakaisa Haapasalo;,citation_author=Brian Lawlor;,citation_author=Thomas H. Mosley;,citation_author=Richard B. Lipton;,citation_author=Vincenzo Solfrizzi;,citation_author=Michael Gill;,citation_author=W. T. Longstreth;,citation_author=Thomas J. Montine;,citation_author=Vincenza Frisardi;,citation_author=Monica Diez-Fairen;,citation_author=Fernando Rivadeneira;,citation_author=Ronald C. Petersen;,citation_author=Vincent Deramecourt;,citation_author=Ignacio Alvarez;,citation_author=Francesca Salani;,citation_author=Antonio Ciaramella;,citation_author=Eric Boerwinkle;,citation_author=Eric M. Reiman;,citation_author=Nathalie Fievet;,citation_author=Jerome I. Rotter;,citation_author=Joan S. Reisch;,citation_author=Olivier Hanon;,citation_author=Chiara Cupidi;,citation_author=A. G. Andre Uitterlinden;,citation_author=Donald R. Royall;,citation_author=Carole Dufouil;,citation_author=Raffaele Giovanni Maletta;,citation_author=Itziar Rojas;,citation_author=Mary Sano;,citation_author=Alexis Brice;,citation_author=Roberta Cecchetti;,citation_author=Peter St George-Hyslop;,citation_author=Karen Ritchie;,citation_author=Magda Tsolaki;,citation_author=Debby W. Tsuang;,citation_author=Bruno Dubois;,citation_author=David Craig;,citation_author=Chuang-Kuo Wu;,citation_author=Hilkka Soininen;,citation_author=Despoina Avramidou;,citation_author=Roger L. Albin;,citation_author=Laura Fratiglioni;,citation_author=Antonia Germanou;,citation_author=Liana G. Apostolova;,citation_author=Lina Keller;,citation_author=Maria Koutroumani;,citation_author=Steven E. Arnold;,citation_author=Francesco Panza;,citation_author=Olymbia Gkatzima;,citation_author=Sanjay Asthana;,citation_author=Didier Hannequin;,citation_author=Patrice Whitehead;,citation_author=Craig S. Atwood;,citation_author=Paolo Caffarra;,citation_author=Harald Hampel;,citation_author=Inés Quintela;,citation_author=Ángel Carracedo;,citation_author=Lars Lannfelt;,citation_author=David C. Rubinsztein;,citation_author=Lisa L. Barnes;,citation_author=Florence Pasquier;,citation_author=Lutz Frölich;,citation_author=Sandra Barral;,citation_author=Bernadette McGuinness;,citation_author=Thomas G. Beach;,citation_author=Janet A. Johnston;,citation_author=James T. Becker;,citation_author=Peter Passmore;,citation_author=Eileen H. Bigio;,citation_author=Jonathan M. Schott;,citation_author=Thomas D. Bird;,citation_author=Jason D. Warren;,citation_author=Bradley F. Boeve;,citation_author=Michelle K. Lupton;,citation_author=James D. Bowen;,citation_author=Petra Proitsi;,citation_author=Adam Boxer;,citation_author=John F. Powell;,citation_author=James R. Burke;,citation_author=John S. K. Kauwe;,citation_author=Jeffrey M. Burns;,citation_author=Michelangelo Mancuso;,citation_author=Joseph D. Buxbaum;,citation_author=Ubaldo Bonuccelli;,citation_author=Nigel J. Cairns;,citation_author=Andrew McQuillin;,citation_author=Chuanhai Cao;,citation_author=Gill Livingston;,citation_author=Chris S. Carlson;,citation_author=Nicholas J. Bass;,citation_author=Cynthia M. Carlsson;,citation_author=John Hardy;,citation_author=Regina M. Carney;,citation_author=Jose Bras;,citation_author=Minerva M. Carrasquillo;,citation_author=Rita Guerreiro;,citation_author=Mariet Allen;,citation_author=Helena C. Chui;,citation_author=Elizabeth Fisher;,citation_author=Carlo Masullo;,citation_author=Elizabeth A. Crocco;,citation_author=Charles DeCarli;,citation_author=Gina Bisceglio;,citation_author=Malcolm Dick;,citation_author=Li Ma;,citation_author=Ranjan Duara;,citation_author=Neill R. Graff-Radford;,citation_author=Denis A. Evans;,citation_author=Angela Hodges;,citation_author=Kelley M. Faber;,citation_author=Martin Scherer;,citation_author=Kenneth B. Fallon;,citation_author=Matthias Riemenschneider;,citation_author=David W. Fardo;,citation_author=Reinhard Heun;,citation_author=Martin R. Farlow;,citation_author=Heike Kölsch;,citation_author=Steven Ferris;,citation_author=Markus Leber;,citation_author=Tatiana M. Foroud;,citation_author=Isabella Heuser;,citation_author=Douglas R. Galasko;,citation_author=Ina Giegling;,citation_author=Marla Gearing;,citation_author=Michael Hüll;,citation_author=Daniel H. Geschwind;,citation_author=John R. Gilbert;,citation_author=John Morris;,citation_author=Robert C. Green;,citation_author=Kevin Mayo;,citation_author=John H. Growdon;,citation_author=Thomas Feulner;,citation_author=Ronald L. Hamilton;,citation_author=Lindy E. Harrell;,citation_author=Dmitriy Drichel;,citation_author=Lawrence S. Honig;,citation_author=Thomas D. Cushion;,citation_author=Matthew J. Huentelman;,citation_author=Paul Hollingworth;,citation_author=Christine M. Hulette;,citation_author=Bradley T. Hyman;,citation_author=Rachel Marshall;,citation_author=Gail P. Jarvik;,citation_author=Alun Meggy;,citation_author=Erin Abner;,citation_author=Georgina E. Menzies;,citation_author=Lee-Way Jin;,citation_author=Ganna Leonenko;,citation_author=Luis M. Real;,citation_author=Gyungah R. Jun;,citation_author=Clinton T. Baldwin;,citation_author=Detelina Grozeva;,citation_author=Anna Karydas;,citation_author=Giancarlo Russo;,citation_author=Jeffrey A. Kaye;,citation_author=Ronald Kim;,citation_author=Frank Jessen;,citation_author=Neil W. Kowall;,citation_author=Bruno Vellas;,citation_author=Joel H. Kramer;,citation_author=Emma Vardy;,citation_author=Frank M. LaFerla;,citation_author=Karl-Heinz Jöckel;,citation_author=James J. Lah;,citation_author=Martin Dichgans;,citation_author=James B. Leverenz;,citation_author=David Mann;,citation_author=Allan I. Levey;,citation_author=Stuart Pickering-Brown;,citation_author=Andrew P. Lieberman;,citation_author=Norman Klopp;,citation_author=Kathryn L. Lunetta;,citation_author=H.-Erich Wichmann;,citation_author=Constantine G. Lyketsos;,citation_author=Kevin Morgan;,citation_author=Daniel C. Marson;,citation_author=Kristelle Brown;,citation_author=Frank Martiniuk;,citation_author=Christopher Medway;,citation_author=Deborah C. Mash;,citation_author=Markus M. Nöthen;,citation_author=Eliezer Masliah;,citation_author=Nigel M. Hooper;,citation_author=Wayne C. McCormick;,citation_author=Antonio Daniele;,citation_author=Susan M. McCurry;,citation_author=Anthony Bayer;,citation_author=Andrew N. McDavid;,citation_author=John Gallacher;,citation_author=Ann C. McKee;,citation_author=Hendrik Bussche;,citation_author=Marsel Mesulam;,citation_author=Carol Brayne;,citation_author=Bruce L. Miller;,citation_author=Steffi Riedel-Heller;,citation_author=Carol A. Miller;,citation_author=Joshua W. Miller;,citation_author=Ammar Al-Chalabi;,citation_author=John C. Morris;,citation_author=Christopher E. Shaw;,citation_author=Amanda J. Myers;,citation_author=Jens Wiltfang;,citation_author=Sid O’Bryant;,citation_author=John M. Olichney;,citation_author=Victoria Alvarez;,citation_author=Joseph E. Parisi;,citation_author=Andrew B. Singleton;,citation_author=Henry L. Paulson;,citation_author=John Collinge;,citation_author=William R. Perry;,citation_author=Simon Mead;,citation_author=Elaine Peskind;,citation_author=David H. Cribbs;,citation_author=Martin Rossor;,citation_author=Aimee Pierce;,citation_author=Natalie S. Ryan;,citation_author=Wayne W. Poon;,citation_author=Benedetta Nacmias;,citation_author=Huntington Potter;,citation_author=Sandro Sorbi;,citation_author=Joseph F. Quinn;,citation_author=Eleonora Sacchinelli;,citation_author=Ashok Raj;,citation_author=Gianfranco Spalletta;,citation_author=Murray Raskind;,citation_author=Carlo Caltagirone;,citation_author=Paola Bossù;,citation_author=Maria Donata Orfei;,citation_author=Barry Reisberg;,citation_author=Robert Clarke;,citation_author=Christiane Reitz;,citation_author=A. David Smith;,citation_author=John M. Ringman;,citation_author=Donald Warden;,citation_author=Erik D. Roberson;,citation_author=Gordon Wilcock;,citation_author=Ekaterina Rogaeva;,citation_author=Amalia Cecilia Bruni;,citation_author=Howard J. Rosen;,citation_author=Maura Gallo;,citation_author=Roger N. Rosenberg;,citation_author=Yoav Ben-Shlomo;,citation_author=Mark A. Sager;,citation_author=Patrizia Mecocci;,citation_author=Andrew J. Saykin;,citation_author=Pau Pastor;,citation_author=Michael L. Cuccaro;,citation_author=Jeffery M. Vance;,citation_author=Julie A. Schneider;,citation_author=Lori S. Schneider;,citation_author=Susan Slifer;,citation_author=William W. Seeley;,citation_author=Amanda G. Smith;,citation_author=Joshua A. Sonnen;,citation_author=Salvatore Spina;,citation_author=Robert A. Stern;,citation_author=Russell H. Swerdlow;,citation_author=Mitchell Tang;,citation_author=Rudolph E. Tanzi;,citation_author=John Q. Trojanowski;,citation_author=Juan C. Troncoso;,citation_author=Vivianna M. Van Deerlin;,citation_author=Linda J. Van Eldik;,citation_author=Harry V. Vinters;,citation_author=Jean Paul Vonsattel;,citation_author=Sandra Weintraub;,citation_author=Kathleen A. Welsh-Bohmer;,citation_author=Kirk C. Wilhelmsen;,citation_author=Jennifer Williamson;,citation_author=Thomas S. Wingo;,citation_author=Randall L. Woltjer;,citation_author=Clinton B. Wright;,citation_author=Chang-En Yu;,citation_author=Lei Yu;,citation_author=Yasaman Saba;,citation_author=Alberto Pilotto;,citation_author=Maria J. Bullido;,citation_author=Oliver Peters;,citation_author=Paul K. Crane;,citation_author=David Bennett;,citation_author=Paola Bosco;,citation_author=Eliecer Coto;,citation_author=Virginia Boccardi;,citation_author=Phil L. De Jager;,citation_author=Alberto Lleo;,citation_author=Nick Warner;,citation_author=Oscar L. Lopez;,citation_author=Martin Ingelsson;,citation_author=Panagiotis Deloukas;,citation_author=Carlos Cruchaga;,citation_author=Caroline Graff;,citation_author=Rhian Gwilliam;,citation_author=Myriam Fornage;,citation_author=Alison M. Goate;,citation_author=Pascual Sanchez-Juan;,citation_author=Patrick G. Kehoe;,citation_author=Najaf Amin;,citation_author=Nilifur Ertekin-Taner;,citation_author=Claudine Berr;,citation_author=Stéphanie Debette;,citation_author=Seth Love;,citation_author=Lenore J. Launer;,citation_author=Steven G. Younkin;,citation_author=Jean-Francois Dartigues;,citation_author=Chris Corcoran;,citation_author=M. Arfan Ikram;,citation_author=Dennis W. Dickson;,citation_author=Gael Nicolas;,citation_author=Dominique Campion;,citation_author=JoAnn Tschanz;,citation_author=Helena Schmidt;,citation_author=Hakon Hakonarson;,citation_author=Jordi Clarimon;,citation_author=Ron Munger;,citation_author=Reinhold Schmidt;,citation_author=Lindsay A. Farrer;,citation_author=Christine Van Broeckhoven;,citation_author=Michael C. O’Donovan;,citation_author=Anita L. DeStefano;,citation_author=Lesley Jones;,citation_author=Jonathan L. Haines;,citation_author=Jean-Francois Deleuze;,citation_author=Michael J. Owen;,citation_author=Vilmundur Gudnason;,citation_author=Richard Mayeux;,citation_author=Valentina Escott-Price;,citation_author=Bruce M. Psaty;,citation_author=Alfredo Ramirez;,citation_author=Li-San Wang;,citation_author=Agustin Ruiz;,citation_author=Cornelia M. Duijn;,citation_author=Peter A. Holmans;,citation_author=Sudha Seshadri;,citation_author=Julie Williams;,citation_author=Phillippe Amouyel;,citation_author=Gerard D. Schellenberg;,citation_author=Jean-Charles Lambert;,citation_author=Margaret A. Pericak-Vance;,citation_publication_date=2019-03;,citation_cover_date=2019-03;,citation_year=2019;,citation_fulltext_html_url=https://www.nature.com/articles/s41588-019-0358-2;,citation_issue=3;,citation_doi=10.1038/s41588-019-0358-2;,citation_volume=51;,citation_language=en;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=scFlow: A scalable and reproducible analysis pipeline for single-cell RNA sequencing dat;,citation_author=Combiz Khozoie;,citation_author=Nurun Fancy;,citation_author=Mahdi Marjaneh;,citation_author=Alan Murphy;,citation_author=Paul Matthews;,citation_author=Nathan Skene;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://doi.org/10.22541/au.162912533.38489960/v2;,citation_doi=10.22541/au.162912533.38489960/v2;">
<meta name="citation_reference" content="citation_title=Nextflow enables reproducible computational workflows;,citation_author=Paolo Di Tommaso;,citation_author=Maria Chatzou;,citation_author=Evan W. Floden;,citation_author=Pablo Prieto Barja;,citation_author=Emilio Palumbo;,citation_author=Cedric Notredame;,citation_publication_date=2017-04;,citation_cover_date=2017-04;,citation_year=2017;,citation_fulltext_html_url=https://www.nature.com/articles/nbt.3820;,citation_issue=4;,citation_doi=10.1038/nbt.3820;,citation_volume=35;,citation_language=en;,citation_journal_title=Nature Biotechnology;">
<meta name="citation_reference" content="citation_title=Dictionary learning for integrative, multimodal and scalable single-cell analysis;,citation_author=Yuhan Hao;,citation_author=Tim Stuart;,citation_author=Madeline H Kowalski;,citation_author=Saket Choudhary;,citation_author=Paul Hoffman;,citation_author=Austin Hartman;,citation_author=Avi Srivastava;,citation_author=Gesmira Molla;,citation_author=Shaista Madad;,citation_author=Carlos Fernandez-Granda;,citation_author=Rahul Satija;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.1038/s41587-023-01767-y;,citation_doi=10.1038/s41587-023-01767-y;">
<meta name="citation_reference" content="citation_title=R: A language and environment for statistical computing;,citation_author=R Core Team undefined;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://www.R-project.org/;">
<meta name="citation_reference" content="citation_title=A human brain vascular atlas reveals diverse mediators of Alzheimer’s risk;,citation_author=Andrew C. Yang;,citation_author=Ryan T. Vest;,citation_author=Fabian Kern;,citation_author=Davis P. Lee;,citation_author=Maayan Agam;,citation_author=Christina A. Maat;,citation_author=Patricia M. Losada;,citation_author=Michelle B. Chen;,citation_author=Nicholas Schaum;,citation_author=Nathalie Khoury;,citation_author=Angus Toland;,citation_author=Kruti Calcuttawala;,citation_author=Heather Shin;,citation_author=Róbert Pálovics;,citation_author=Andrew Shin;,citation_author=Elizabeth Y. Wang;,citation_author=Jian Luo;,citation_author=David Gate;,citation_author=Walter J. Schulz-Schaeffer;,citation_author=Pauline Chu;,citation_author=Julie A. Siegenthaler;,citation_author=M. Windy McNerney;,citation_author=Andreas Keller;,citation_author=Tony Wyss-Coray;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_issue=7903;,citation_doi=10.1038/s41586-021-04369-3;,citation_volume=603;,citation_language=eng;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=Dictionary learning for integrative, multimodal and scalable single-cell analysis;,citation_author=Yuhan Hao;,citation_author=Tim Stuart;,citation_author=Madeline H Kowalski;,citation_author=Saket Choudhary;,citation_author=Paul Hoffman;,citation_author=Austin Hartman;,citation_author=Avi Srivastava;,citation_author=Gesmira Molla;,citation_author=Shaista Madad;,citation_author=Carlos Fernandez-Granda;,citation_author=Rahul Satija;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.1038/s41587-023-01767-y;,citation_doi=10.1038/s41587-023-01767-y;">
<meta name="citation_reference" content="citation_title=CellChat: Inference and analysis of cell-cell communication from single-cell and spatially resolved transcriptomics data;,citation_author=Suoqin Jin;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://github.com/jinworks/CellChat;">
<meta name="citation_reference" content="citation_title=Propeller: Testing for differences in cell type proportions in single cell data;,citation_author=Belinda Phipson;,citation_author=Choon Boon Sim;,citation_author=Enzo R Porrello;,citation_author=Alex W Hewitt;,citation_author=Joseph Powell;,citation_author=Alicia Oshlack;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://doi.org/10.1093/bioinformatics/btac582;,citation_volume=38;">
<meta name="citation_reference" content="citation_title=Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2;,citation_author=Michael I. Love;,citation_author=Wolfgang Huber;,citation_author=Simon Anders;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_doi=10.1186/s13059-014-0550-8;,citation_volume=15;">
<meta name="citation_reference" content="citation_title=clusterProfiler 4.0: A universal enrichment tool for interpreting omics data;,citation_author=Tianzhi Wu;,citation_author=Erqiang Hu;,citation_author=Shuangbin Xu;,citation_author=Meijun Chen;,citation_author=Pingfan Guo;,citation_author=Zehan Dai;,citation_author=Tingze Feng;,citation_author=Lang Zhou;,citation_author=Wenli Tang;,citation_author=Li Zhan;,citation_author=xiaochong Fu;,citation_author=Shanshan Liu;,citation_author=Xiaochen Bo;,citation_author=Guangchuang Yu;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_doi=10.1016/j.xinn.2021.100141;,citation_volume=2;">
<meta name="citation_reference" content="citation_title=MAGMA: Generalized Gene-Set Analysis of GWAS Data;,citation_author=Christiaan A. Leeuw;,citation_author=Joris M. Mooij;,citation_author=Tom Heskes;,citation_author=Danielle Posthuma;,citation_publication_date=2015-04-17;,citation_cover_date=2015-04-17;,citation_year=2015;,citation_fulltext_html_url=https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004219;,citation_issue=4;,citation_doi=10.1371/journal.pcbi.1004219;,citation_volume=11;,citation_language=en;,citation_journal_title=PLOS Computational Biology;">
<meta name="citation_reference" content="citation_title=LD Score regression distinguishes confounding from polygenicity in genome-wide association studies;,citation_author=Brendan K. Bulik-Sullivan;,citation_author=Po-Ru Loh;,citation_author=Hilary K. Finucane;,citation_author=Stephan Ripke;,citation_author=Jian Yang;,citation_author=Nick Patterson;,citation_author=Mark J. Daly;,citation_author=Alkes L. Price;,citation_author=Benjamin M. Neale;,citation_publication_date=2015-03;,citation_cover_date=2015-03;,citation_year=2015;,citation_fulltext_html_url=https://www.nature.com/articles/ng.3211;,citation_issue=3;,citation_doi=10.1038/ng.3211;,citation_volume=47;,citation_language=en;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=Database resources of the national center for biotechnology information;,citation_author=Eric W. Sayers;,citation_author=Evan E. Bolton;,citation_author=J. Rodney Brister;,citation_author=Kathi Canese;,citation_author=Jessica Chan;,citation_author=Donald C. Comeau;,citation_author=Ryan Connor;,citation_author=Kathryn Funk;,citation_author=Chris Kelly;,citation_author=Sunghwan Kim;,citation_author=Tom Madej;,citation_author=Aron Marchler-Bauer;,citation_author=Christopher Lanczycki;,citation_author=Stacy Lathrop;,citation_author=Zhiyong Lu;,citation_author=Francoise Thibaud-Nissen;,citation_author=Terence Murphy;,citation_author=Lon Phan;,citation_author=Yuri Skripchenko;,citation_author=Tony Tse;,citation_author=Jiyao Wang;,citation_author=Rebecca Williams;,citation_author=Barton W. Trawick;,citation_author=Kim D. Pruitt;,citation_author=Stephen T. Sherry;,citation_publication_date=2022-01-07;,citation_cover_date=2022-01-07;,citation_year=2022;,citation_issue=D1;,citation_doi=10.1093/nar/gkab1112;,citation_volume=50;,citation_language=eng;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=A global reference for human genetic variation;,citation_author=Adam Auton;,citation_author=Gonçalo R. Abecasis;,citation_author=David M. Altshuler;,citation_author=Richard M. Durbin;,citation_author=Gonçalo R. Abecasis;,citation_author=David R. Bentley;,citation_author=Aravinda Chakravarti;,citation_author=Andrew G. Clark;,citation_author=Peter Donnelly;,citation_author=Evan E. Eichler;,citation_author=Paul Flicek;,citation_author=Stacey B. Gabriel;,citation_author=Richard A. Gibbs;,citation_author=Eric D. Green;,citation_author=Matthew E. Hurles;,citation_author=Bartha M. Knoppers;,citation_author=Jan O. Korbel;,citation_author=Eric S. Lander;,citation_author=Charles Lee;,citation_author=Hans Lehrach;,citation_author=Elaine R. Mardis;,citation_author=Gabor T. Marth;,citation_author=Gil A. McVean;,citation_author=Deborah A. Nickerson;,citation_author=Jeanette P. Schmidt;,citation_author=Stephen T. Sherry;,citation_author=Jun Wang;,citation_author=Richard K. Wilson;,citation_author=Richard A. Gibbs;,citation_author=Eric Boerwinkle;,citation_author=Harsha Doddapaneni;,citation_author=Yi Han;,citation_author=Viktoriya Korchina;,citation_author=Christie Kovar;,citation_author=Sandra Lee;,citation_author=Donna Muzny;,citation_author=Jeffrey G. Reid;,citation_author=Yiming Zhu;,citation_author=Jun Wang;,citation_author=Yuqi Chang;,citation_author=Qiang Feng;,citation_author=Xiaodong Fang;,citation_author=Xiaosen Guo;,citation_author=Min Jian;,citation_author=Hui Jiang;,citation_author=Xin Jin;,citation_author=Tianming Lan;,citation_author=Guoqing Li;,citation_author=Jingxiang Li;,citation_author=Yingrui Li;,citation_author=Shengmao Liu;,citation_author=Xiao Liu;,citation_author=Yao Lu;,citation_author=Xuedi Ma;,citation_author=Meifang Tang;,citation_author=Bo Wang;,citation_author=Guangbiao Wang;,citation_author=Honglong Wu;,citation_author=Renhua Wu;,citation_author=Xun Xu;,citation_author=Ye Yin;,citation_author=Dandan Zhang;,citation_author=Wenwei Zhang;,citation_author=Jiao Zhao;,citation_author=Meiru Zhao;,citation_author=Xiaole Zheng;,citation_author=Eric S. Lander;,citation_author=David M. Altshuler;,citation_author=Stacey B. Gabriel;,citation_author=Namrata Gupta;,citation_author=Neda Gharani;,citation_author=Lorraine H. Toji;,citation_author=Norman P. Gerry;,citation_author=Alissa M. Resch;,citation_author=Paul Flicek;,citation_author=Jonathan Barker;,citation_author=Laura Clarke;,citation_author=Laurent Gil;,citation_author=Sarah E. Hunt;,citation_author=Gavin Kelman;,citation_author=Eugene Kulesha;,citation_author=Rasko Leinonen;,citation_author=William M. McLaren;,citation_author=Rajesh Radhakrishnan;,citation_author=Asier Roa;,citation_author=Dmitriy Smirnov;,citation_author=Richard E. Smith;,citation_author=Ian Streeter;,citation_author=Anja Thormann;,citation_author=Iliana Toneva;,citation_author=Brendan Vaughan;,citation_author=Xiangqun Zheng-Bradley;,citation_author=David R. Bentley;,citation_author=Russell Grocock;,citation_author=Sean Humphray;,citation_author=Terena James;,citation_author=Zoya Kingsbury;,citation_author=Hans Lehrach;,citation_author=Ralf Sudbrak;,citation_author=Marcus W. Albrecht;,citation_author=Vyacheslav S. Amstislavskiy;,citation_author=Tatiana A. Borodina;,citation_author=Matthias Lienhard;,citation_author=Florian Mertes;,citation_author=Marc Sultan;,citation_author=Bernd Timmermann;,citation_author=Marie-Laure Yaspo;,citation_author=Elaine R. Mardis;,citation_author=Richard K. Wilson;,citation_author=Lucinda Fulton;,citation_author=Robert Fulton;,citation_author=Stephen T. Sherry;,citation_author=Victor Ananiev;,citation_author=Zinaida Belaia;,citation_author=Dimitriy Beloslyudtsev;,citation_author=Nathan Bouk;,citation_author=Chao Chen;,citation_author=Deanna Church;,citation_author=Robert Cohen;,citation_author=Charles Cook;,citation_author=John Garner;,citation_author=Timothy Hefferon;,citation_author=Mikhail Kimelman;,citation_author=Chunlei Liu;,citation_author=John Lopez;,citation_author=Peter Meric;,citation_author=Chris O’Sullivan;,citation_author=Yuri Ostapchuk;,citation_author=Lon Phan;,citation_author=Sergiy Ponomarov;,citation_author=Valerie Schneider;,citation_author=Eugene Shekhtman;,citation_author=Karl Sirotkin;,citation_author=Douglas Slotta;,citation_author=Hua Zhang;,citation_author=Gil A. McVean;,citation_author=Richard M. Durbin;,citation_author=Senduran Balasubramaniam;,citation_author=John Burton;,citation_author=Petr Danecek;,citation_author=Thomas M. Keane;,citation_author=Anja Kolb-Kokocinski;,citation_author=Shane McCarthy;,citation_author=James Stalker;,citation_author=Michael Quail;,citation_author=Jeanette P. Schmidt;,citation_author=Christopher J. Davies;,citation_author=Jeremy Gollub;,citation_author=Teresa Webster;,citation_author=Brant Wong;,citation_author=Yiping Zhan;,citation_author=Adam Auton;,citation_author=Christopher L. Campbell;,citation_author=Yu Kong;,citation_author=Anthony Marcketta;,citation_author=Richard A. Gibbs;,citation_author=Fuli Yu;,citation_author=Lilian Antunes;,citation_author=Matthew Bainbridge;,citation_author=Donna Muzny;,citation_author=Aniko Sabo;,citation_author=Zhuoyi Huang;,citation_author=Jun Wang;,citation_author=Lachlan J. M. Coin;,citation_author=Lin Fang;,citation_author=Xiaosen Guo;,citation_author=Xin Jin;,citation_author=Guoqing Li;,citation_author=Qibin Li;,citation_author=Yingrui Li;,citation_author=Zhenyu Li;,citation_author=Haoxiang Lin;,citation_author=Binghang Liu;,citation_author=Ruibang Luo;,citation_author=Haojing Shao;,citation_author=Yinlong Xie;,citation_author=Chen Ye;,citation_author=Chang Yu;,citation_author=Fan Zhang;,citation_author=Hancheng Zheng;,citation_author=Hongmei Zhu;,citation_author=Can Alkan;,citation_author=Elif Dal;,citation_author=Fatma Kahveci;,citation_author=Gabor T. Marth;,citation_author=Erik P. Garrison;,citation_author=Deniz Kural;,citation_author=Wan-Ping Lee;,citation_author=Wen Fung Leong;,citation_author=Michael Stromberg;,citation_author=Alistair N. Ward;,citation_author=Jiantao Wu;,citation_author=Mengyao Zhang;,citation_author=Mark J. Daly;,citation_author=Mark A. DePristo;,citation_author=Robert E. Handsaker;,citation_author=David M. Altshuler;,citation_author=Eric Banks;,citation_author=Gaurav Bhatia;,citation_author=Guillermo Angel;,citation_author=Stacey B. Gabriel;,citation_author=Giulio Genovese;,citation_author=Namrata Gupta;,citation_author=Heng Li;,citation_author=Seva Kashin;,citation_author=Eric S. Lander;,citation_author=Steven A. McCarroll;,citation_author=James C. Nemesh;,citation_author=Ryan E. Poplin;,citation_author=Seungtai C. Yoon;,citation_author=Jayon Lihm;,citation_author=Vladimir Makarov;,citation_author=Andrew G. Clark;,citation_author=Srikanth Gottipati;,citation_author=Alon Keinan;,citation_author=Juan L. Rodriguez-Flores;,citation_author=Jan O. Korbel;,citation_author=Tobias Rausch;,citation_author=Markus H. Fritz;,citation_author=Adrian M. Stütz;,citation_author=Paul Flicek;,citation_author=Kathryn Beal;,citation_author=Laura Clarke;,citation_author=Avik Datta;,citation_author=Javier Herrero;,citation_author=William M. McLaren;,citation_author=Graham R. S. Ritchie;,citation_author=Richard E. Smith;,citation_author=Daniel Zerbino;,citation_author=Xiangqun Zheng-Bradley;,citation_author=Pardis C. Sabeti;,citation_author=Ilya Shlyakhter;,citation_author=Stephen F. Schaffner;,citation_author=Joseph Vitti;,citation_author=David N. Cooper;,citation_author=Edward V. Ball;,citation_author=Peter D. Stenson;,citation_author=David R. Bentley;,citation_author=Bret Barnes;,citation_author=Markus Bauer;,citation_author=R. Keira Cheetham;,citation_author=Anthony Cox;,citation_author=Michael Eberle;,citation_author=Sean Humphray;,citation_author=Scott Kahn;,citation_author=Lisa Murray;,citation_author=John Peden;,citation_author=Richard Shaw;,citation_author=Eimear E. Kenny;,citation_author=Mark A. Batzer;,citation_author=Miriam K. Konkel;,citation_author=Jerilyn A. Walker;,citation_author=Daniel G. MacArthur;,citation_author=Monkol Lek;,citation_author=Ralf Sudbrak;,citation_author=Vyacheslav S. Amstislavskiy;,citation_author=Ralf Herwig;,citation_author=Elaine R. Mardis;,citation_author=Li Ding;,citation_author=Daniel C. Koboldt;,citation_author=David Larson;,citation_author=Kai Ye;,citation_author=Simon Gravel;,citation_author=undefined The 1000 Genomes Project Consortium;,citation_author=undefined Corresponding authors;,citation_author=undefined Steering committee;,citation_author=undefined Production group;,citation_author=undefined Baylor College of Medicine;,citation_author=undefined BGI-Shenzhen;,citation_author=Broad Institute MIT;,citation_author=undefined Harvard;,citation_author=undefined Coriell Institute for Medical Research;,citation_author=European Bioinformatics Institute European Molecular Biology Laboratory;,citation_author=undefined Illumina;,citation_author=undefined Max Planck Institute for Molecular Genetics;,citation_author=undefined McDonnell Genome Institute at Washington University;,citation_author=undefined US National Institutes of Health;,citation_author=undefined University of Oxford;,citation_author=undefined Wellcome Trust Sanger Institute;,citation_author=undefined Analysis group;,citation_author=undefined Affymetrix;,citation_author=undefined Albert Einstein College of Medicine;,citation_author=undefined Bilkent University;,citation_author=undefined Boston College;,citation_author=undefined Cold Spring Harbor Laboratory;,citation_author=undefined Cornell University;,citation_author=undefined European Molecular Biology Laboratory;,citation_author=undefined Harvard University;,citation_author=undefined Human Gene Mutation Database;,citation_author=undefined Icahn School of Medicine at Mount Sinai;,citation_author=undefined Louisiana State University;,citation_author=undefined Massachusetts General Hospital;,citation_author=undefined McGill University;,citation_author=NIH National Eye Institute;,citation_publication_date=2015-10;,citation_cover_date=2015-10;,citation_year=2015;,citation_fulltext_html_url=https://www.nature.com/articles/nature15393;,citation_issue=7571;,citation_doi=10.1038/nature15393;,citation_volume=526;,citation_language=en;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=The relationship between amyloid-beta and brain capillary endothelial cells in Alzheimer’s disease;,citation_author=Qiang Su;,citation_author=Jun-Hong Guo;,citation_author=Yan-Li Zhang;,citation_author=Juan Wang;,citation_author=Zhi-Na Zhang;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.4103/1673-5374.335829;,citation_issue=11;,citation_doi=10.4103/1673-5374.335829;,citation_volume=17;,citation_language=en;,citation_journal_title=Neural Regeneration Research;">
<meta name="citation_reference" content="citation_title=The BloodBrain Barrier;,citation_author=Richard Daneman;,citation_author=Alexandre Prat;,citation_publication_date=2015-01;,citation_cover_date=2015-01;,citation_year=2015;,citation_fulltext_html_url=http://dx.doi.org/10.1101/cshperspect.a020412;,citation_issue=1;,citation_doi=10.1101/cshperspect.a020412;,citation_volume=7;,citation_language=en;,citation_journal_title=Cold Spring Harbor Perspectives in Biology;">
<meta name="citation_reference" content="citation_title=Immunomodulation by endothelial cells  partnering up with the immune system?;,citation_author=Jacob Amersfoort;,citation_author=Guy Eelen;,citation_author=Peter Carmeliet;,citation_publication_date=2022-03-14;,citation_cover_date=2022-03-14;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41577-022-00694-4;,citation_issue=9;,citation_doi=10.1038/s41577-022-00694-4;,citation_volume=22;,citation_language=en;,citation_journal_title=Nature Reviews Immunology;">
<meta name="citation_reference" content="citation_title=Pericytes and neurovascular function in the healthy and diseased brain;,citation_author=Lachlan S. Brown;,citation_author=Catherine G. Foster;,citation_author=Jo-Maree Courtney;,citation_author=Natalie E. King;,citation_author=David W. Howells;,citation_author=Brad A. Sutherland;,citation_publication_date=2019-06-28;,citation_cover_date=2019-06-28;,citation_year=2019;,citation_fulltext_html_url=http://dx.doi.org/10.3389/fncel.2019.00282;,citation_doi=10.3389/fncel.2019.00282;,citation_volume=13;,citation_journal_title=Frontiers in Cellular Neuroscience;">
<meta name="citation_reference" content="citation_title=Interplay between Brain Pericytes and Endothelial Cells in Dementia;,citation_author=Tessa V. Procter;,citation_author=Anna Williams;,citation_author=Axel Montagne;,citation_publication_date=2021-11;,citation_cover_date=2021-11;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.ajpath.2021.07.003;,citation_issue=11;,citation_doi=10.1016/j.ajpath.2021.07.003;,citation_volume=191;,citation_language=en;,citation_journal_title=The American Journal of Pathology;">
<meta name="citation_reference" content="citation_title=The blood-brain barrier in alzheimer’s disease;,citation_author=Steffen E. Storck;,citation_author=Anika M. S. Hartz;,citation_author=Claus U. Pietrzik;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1007/164_2020_418;,citation_doi=10.1007/164_2020_418;,citation_inbook_title=undefined;">
<meta name="citation_reference" content="citation_title=Bloodbrain barrier breakdown in Alzheimer disease and other neurodegenerative disorders;,citation_author=Melanie D. Sweeney;,citation_author=Abhay P. Sagare;,citation_author=Berislav V. Zlokovic;,citation_publication_date=2018-01-29;,citation_cover_date=2018-01-29;,citation_year=2018;,citation_fulltext_html_url=http://dx.doi.org/10.1038/nrneurol.2017.188;,citation_issue=3;,citation_doi=10.1038/nrneurol.2017.188;,citation_volume=14;,citation_language=en;,citation_journal_title=Nature Reviews Neurology;">
<meta name="citation_reference" content="citation_title=Blood-brain barrier leakage in Alzheimer’s disease: From discovery to clinical relevance;,citation_author=Geetika Nehra;,citation_author=Bjoern Bauer;,citation_author=Anika M. S. Hartz;,citation_publication_date=2022-06;,citation_cover_date=2022-06;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.pharmthera.2022.108119;,citation_doi=10.1016/j.pharmthera.2022.108119;,citation_volume=234;,citation_language=en;,citation_journal_title=Pharmacology &amp;amp;amp; Therapeutics;">
<meta name="citation_reference" content="citation_title=Cerebral blood flow decrease as an early pathological mechanism in Alzheimer’s disease;,citation_author=Nils Korte;,citation_author=Ross Nortley;,citation_author=David Attwell;,citation_publication_date=2020-08-31;,citation_cover_date=2020-08-31;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1007/s00401-020-02215-w;,citation_issue=6;,citation_doi=10.1007/s00401-020-02215-w;,citation_volume=140;,citation_language=en;,citation_journal_title=Acta Neuropathologica;">
<meta name="citation_reference" content="citation_title=Cell densities in the mouse brain: A systematic review;,citation_author=Daniel Keller;,citation_author=Csaba Erö;,citation_author=Henry Markram;,citation_publication_date=2018-10-23;,citation_cover_date=2018-10-23;,citation_year=2018;,citation_fulltext_html_url=http://dx.doi.org/10.3389/fnana.2018.00083;,citation_doi=10.3389/fnana.2018.00083;,citation_volume=12;,citation_journal_title=Frontiers in Neuroanatomy;">
<meta name="citation_reference" content="citation_title=Single-cell transcriptomic analysis of Alzheimer’s disease;,citation_author=Hansruedi Mathys;,citation_author=Jose Davila-Velderrain;,citation_author=Zhuyu Peng;,citation_author=Fan Gao;,citation_author=Shahin Mohammadi;,citation_author=Jennie Z. Young;,citation_author=Madhvi Menon;,citation_author=Liang He;,citation_author=Fatema Abdurrob;,citation_author=Xueqiao Jiang;,citation_author=Anthony J. Martorell;,citation_author=Richard M. Ransohoff;,citation_author=Brian P. Hafler;,citation_author=David A. Bennett;,citation_author=Manolis Kellis;,citation_author=Li-Huei Tsai;,citation_publication_date=2019-05-01;,citation_cover_date=2019-05-01;,citation_year=2019;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41586-019-1195-2;,citation_issue=7761;,citation_doi=10.1038/s41586-019-1195-2;,citation_volume=570;,citation_language=en;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=A single-cell atlas of entorhinal cortex from individuals with Alzheimer’s disease reveals cell-type-specific gene expression regulation;,citation_author=Alexandra Grubman;,citation_author=Gabriel Chew;,citation_author=John F. Ouyang;,citation_author=Guizhi Sun;,citation_author=Xin Yi Choo;,citation_author=Catriona McLean;,citation_author=Rebecca K. Simmons;,citation_author=Sam Buckberry;,citation_author=Dulce B. Vargas-Landin;,citation_author=Daniel Poppe;,citation_author=Jahnvi Pflueger;,citation_author=Ryan Lister;,citation_author=Owen J. L. Rackham;,citation_author=Enrico Petretto;,citation_author=Jose M. Polo;,citation_publication_date=2019-11-25;,citation_cover_date=2019-11-25;,citation_year=2019;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41593-019-0539-4;,citation_issue=12;,citation_doi=10.1038/s41593-019-0539-4;,citation_volume=22;,citation_language=en;,citation_journal_title=Nature Neuroscience;">
<meta name="citation_reference" content="citation_title=A single nuclear transcriptomic characterisation of mechanisms responsible for impaired angiogenesis and blood-brain barrier function in Alzheimer’s disease;,citation_author=Stergios Tsartsalis;,citation_author=Hannah Sleven;,citation_author=Nurun Fancy;,citation_author=Frank Wessely;,citation_author=Amy M. Smith;,citation_author=Nanet Willumsen;,citation_author=To Ka Dorcas Cheung;,citation_author=Michal J. Rokicki;,citation_author=Vicky Chau;,citation_author=Eseoghene Ifie;,citation_author=Combiz Khozoie;,citation_author=Olaf Ansorge;,citation_author=Xin Yang;,citation_author=Marion H. Jenkyns;,citation_author=Karen Davey;,citation_author=Aisling McGarry;,citation_author=Robert C. J. Muirhead;,citation_author=Stephanie Debette;,citation_author=Johanna S. Jackson;,citation_author=Axel Montagne;,citation_author=David R. Owen;,citation_author=J. Scott Miners;,citation_author=Seth Love;,citation_author=Caleb Webber;,citation_author=M. Zameel Cader;,citation_author=Paul M. Matthews;,citation_publication_date=2024-03-12;,citation_cover_date=2024-03-12;,citation_year=2024;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41467-024-46630-z;,citation_issue=1;,citation_doi=10.1038/s41467-024-46630-z;,citation_volume=15;,citation_language=en;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=The Growth Factor Progranulin Binds to TNF Receptors and Is Therapeutic Against Inflammatory Arthritis in Mice;,citation_author=Wei Tang;,citation_author=Yi Lu;,citation_author=Qing-Yun Tian;,citation_author=Yan Zhang;,citation_author=Feng-Jin Guo;,citation_author=Guang-Yi Liu;,citation_author=Nabeel Muzaffar Syed;,citation_author=Yongjie Lai;,citation_author=Edward Alan Lin;,citation_author=Li Kong;,citation_author=Jeffrey Su;,citation_author=Fangfang Yin;,citation_author=Ai-Hao Ding;,citation_author=Alexandra Zanin-Zhorov;,citation_author=Michael L. Dustin;,citation_author=Jian Tao;,citation_author=Joseph Craft;,citation_author=Zhinan Yin;,citation_author=Jian Q. Feng;,citation_author=Steven B. Abramson;,citation_author=Xiu-Ping Yu;,citation_author=Chuan-ju Liu;,citation_publication_date=2011-04-22;,citation_cover_date=2011-04-22;,citation_year=2011;,citation_fulltext_html_url=http://dx.doi.org/10.1126/science.1199214;,citation_issue=6028;,citation_doi=10.1126/science.1199214;,citation_volume=332;,citation_language=en;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Neutralization of TRAIL death pathway protects human neuronal cell line from β-amyloid toxicity;,citation_author=G Cantarella;,citation_author=D Uberti;,citation_author=T Carsana;,citation_author=G Lombardo;,citation_author=R Bernardini;,citation_author=M Memo;,citation_publication_date=2003-01;,citation_cover_date=2003-01;,citation_year=2003;,citation_fulltext_html_url=http://dx.doi.org/10.1038/sj.cdd.4401143;,citation_issue=1;,citation_doi=10.1038/sj.cdd.4401143;,citation_volume=10;,citation_language=en;,citation_journal_title=Cell Death &amp;amp;amp; Differentiation;">
<meta name="citation_reference" content="citation_title=Neutralization of TNFSF10 ameliorates functional outcome in a murine model of Alzheimer’s disease;,citation_author=Giuseppina Cantarella;,citation_author=Giulia Di Benedetto;,citation_author=Daniela Puzzo;,citation_author=Lucia Privitera;,citation_author=Carla Loreto;,citation_author=Salvatore Saccone;,citation_author=Salvatore Giunta;,citation_author=Agostino Palmeri;,citation_author=Renato Bernardini;,citation_publication_date=2014-12-02;,citation_cover_date=2014-12-02;,citation_year=2014;,citation_fulltext_html_url=http://dx.doi.org/10.1093/brain/awu318;,citation_issue=1;,citation_doi=10.1093/brain/awu318;,citation_volume=138;,citation_language=en;,citation_journal_title=Brain;">
<meta name="citation_reference" content="citation_title=Annexin A1 restores cerebrovascular integrity concomitant with reduced amyloid-β and tau pathology;,citation_author=Miriam Ries;,citation_author=Helena Watts;,citation_author=Bibiana C Mota;,citation_author=Maria Yanez Lopez;,citation_author=Cornelius K Donat;,citation_author=Nicoleta Baxan;,citation_author=James A Pickering;,citation_author=Tsz Wing Chau;,citation_author=Annika Semmler;,citation_author=Brinda Gurung;,citation_author=Robertas Aleksynas;,citation_author=Laura Abelleira-Hervas;,citation_author=Soha Jamshed Iqbal;,citation_author=Carmen Romero-Molina;,citation_author=Gerard Hernandez-Mir;,citation_author=Antonio Amati;,citation_author=Chris Reutelingsperger;,citation_author=Marc H Goldfinger;,citation_author=Steve M Gentleman;,citation_author=Fred Van Leuven;,citation_author=Egle Solito;,citation_author=Magdalena Sastre;,citation_publication_date=2021-05-01;,citation_cover_date=2021-05-01;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1093/brain/awab050;,citation_issue=5;,citation_doi=10.1093/brain/awab050;,citation_volume=144;,citation_language=en;,citation_journal_title=Brain;">
<meta name="citation_reference" content="citation_title=Molecular characterization of selectively vulnerable neurons in Alzheimer’s disease;,citation_author=Kun Leng;,citation_author=Emmy Li;,citation_author=Rana Eser;,citation_author=Antonia Piergies;,citation_author=Rene Sit;,citation_author=Michelle Tan;,citation_author=Norma Neff;,citation_author=Song Hua Li;,citation_author=Roberta Diehl Rodriguez;,citation_author=Claudia Kimie Suemoto;,citation_author=Renata Elaine Paraizo Leite;,citation_author=Alexander J. Ehrenberg;,citation_author=Carlos A. Pasqualucci;,citation_author=William W. Seeley;,citation_author=Salvatore Spina;,citation_author=Helmut Heinsen;,citation_author=Lea T. Grinberg;,citation_author=Martin Kampmann;,citation_publication_date=2021-01-11;,citation_cover_date=2021-01-11;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41593-020-00764-7;,citation_issue=2;,citation_doi=10.1038/s41593-020-00764-7;,citation_volume=24;,citation_language=en;,citation_journal_title=Nature Neuroscience;">
<meta name="citation_reference" content="citation_title=Demonstrated brain insulin resistance in Alzheimer’s disease patients is associated with IGF-1 resistance, IRS-1 dysregulation, and cognitive decline;,citation_author=Konrad Talbot;,citation_author=Hoau-Yan Wang;,citation_author=Hala Kazi;,citation_author=Li-Ying Han;,citation_author=Kalindi P. Bakshi;,citation_author=Andres Stucky;,citation_author=Robert L. Fuino;,citation_author=Krista R. Kawaguchi;,citation_author=Andrew J. Samoyedny;,citation_author=Robert S. Wilson;,citation_author=Zoe Arvanitakis;,citation_author=Julie A. Schneider;,citation_author=Bryan A. Wolf;,citation_author=David A. Bennett;,citation_author=John Q. Trojanowski;,citation_author=Steven E. Arnold;,citation_publication_date=2012-04-02;,citation_cover_date=2012-04-02;,citation_year=2012;,citation_fulltext_html_url=https://www.jci.org/articles/view/59903#sd;,citation_issue=4;,citation_doi=10.1172/JCI59903;,citation_volume=122;,citation_language=en;,citation_journal_title=The Journal of Clinical Investigation;">
<meta name="citation_reference" content="citation_title=A human brain vascular atlas reveals diverse mediators of Alzheimer’s risk;,citation_author=Andrew C. Yang;,citation_author=Ryan T. Vest;,citation_author=Fabian Kern;,citation_author=Davis P. Lee;,citation_author=Maayan Agam;,citation_author=Christina A. Maat;,citation_author=Patricia M. Losada;,citation_author=Michelle B. Chen;,citation_author=Nicholas Schaum;,citation_author=Nathalie Khoury;,citation_author=Angus Toland;,citation_author=Kruti Calcuttawala;,citation_author=Heather Shin;,citation_author=Róbert Pálovics;,citation_author=Andrew Shin;,citation_author=Elizabeth Y. Wang;,citation_author=Jian Luo;,citation_author=David Gate;,citation_author=Walter J. Schulz-Schaeffer;,citation_author=Pauline Chu;,citation_author=Julie A. Siegenthaler;,citation_author=M. Windy McNerney;,citation_author=Andreas Keller;,citation_author=Tony Wyss-Coray;,citation_publication_date=2022-03;,citation_cover_date=2022-03;,citation_year=2022;,citation_issue=7903;,citation_doi=10.1038/s41586-021-04369-3;,citation_volume=603;,citation_language=eng;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=Elevation of inactive cleaved annexin A1 in the neocortex is associated with amyloid, inflammatory and apoptotic markers in neurodegenerative dementias;,citation_author=Xin Ying Chua;,citation_author=Joyce R. Chong;,citation_author=Ai Ling Cheng;,citation_author=Jasinda H. Lee;,citation_author=Clive Ballard;,citation_author=Dag Aarsland;,citation_author=Paul T. Francis;,citation_author=Mitchell K. P. Lai;,citation_publication_date=2022-01;,citation_cover_date=2022-01;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.neuint.2021.105251;,citation_doi=10.1016/j.neuint.2021.105251;,citation_volume=152;,citation_language=en;,citation_journal_title=Neurochemistry International;">
<meta name="citation_reference" content="citation_title=Single-cell atlas reveals correlates of high cognitive function, dementia, and resilience to Alzheimer’s disease pathology;,citation_author=Hansruedi Mathys;,citation_author=Zhuyu Peng;,citation_author=Carles A. Boix;,citation_author=Matheus B. Victor;,citation_author=Noelle Leary;,citation_author=Sudhagar Babu;,citation_author=Ghada Abdelhady;,citation_author=Xueqiao Jiang;,citation_author=Ayesha P. Ng;,citation_author=Kimia Ghafari;,citation_author=Alexander K. Kunisky;,citation_author=Julio Mantero;,citation_author=Kyriaki Galani;,citation_author=Vanshika N. Lohia;,citation_author=Gabrielle E. Fortier;,citation_author=Yasmine Lotfi;,citation_author=Jason Ivey;,citation_author=Hannah P. Brown;,citation_author=Pratham R. Patel;,citation_author=Nehal Chakraborty;,citation_author=Jacob I. Beaudway;,citation_author=Elizabeth J. Imhoff;,citation_author=Cameron F. Keeler;,citation_author=Maren M. McChesney;,citation_author=Haishal H. Patel;,citation_author=Sahil P. Patel;,citation_author=Megan T. Thai;,citation_author=David A. Bennett;,citation_author=Manolis Kellis;,citation_author=Li-Huei Tsai;,citation_publication_date=2023-09;,citation_cover_date=2023-09;,citation_year=2023;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.cell.2023.08.039;,citation_issue=20;,citation_doi=10.1016/j.cell.2023.08.039;,citation_volume=186;,citation_language=en;,citation_journal_title=Cell;">
<meta name="citation_reference" content="citation_title=SLC4A11 depletion impairs NRF2 mediated antioxidant signaling and increases reactive oxygen species in human corneal endothelial cells during oxidative stress;,citation_author=Sanjukta Guha;,citation_author=Sunita Chaurasia;,citation_author=Charanya Ramachandran;,citation_author=Sanhita Roy;,citation_publication_date=2017-06-22;,citation_cover_date=2017-06-22;,citation_year=2017;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41598-017-03654-4;,citation_issue=1;,citation_doi=10.1038/s41598-017-03654-4;,citation_volume=7;,citation_language=en;,citation_journal_title=Scientific Reports;">
<meta name="citation_reference" content="citation_title=The mitochondrial protein TSPO in Alzheimer’s disease: relation to the severity of AD pathology and the neuroinflammatory environment;,citation_author=Emma F. Garland;,citation_author=Oliver Dennett;,citation_author=Laurie C. Lau;,citation_author=David S. Chatelet;,citation_author=Michel Bottlaender;,citation_author=James A. R. Nicoll;,citation_author=Delphine Boche;,citation_publication_date=2023-08-14;,citation_cover_date=2023-08-14;,citation_year=2023;,citation_fulltext_html_url=http://dx.doi.org/10.1186/s12974-023-02869-9;,citation_issue=1;,citation_doi=10.1186/s12974-023-02869-9;,citation_volume=20;,citation_language=en;,citation_journal_title=Journal of Neuroinflammation;">
<meta name="citation_reference" content="citation_title=The mammalian phenotype ontology: enabling robust annotation and comparative analysis;,citation_author=Cynthia L. Smith;,citation_author=Janan T. Eppig;,citation_publication_date=2009-11;,citation_cover_date=2009-11;,citation_year=2009;,citation_fulltext_html_url=http://dx.doi.org/10.1002/wsbm.44;,citation_issue=3;,citation_doi=10.1002/wsbm.44;,citation_volume=1;,citation_language=en;,citation_journal_title=WIREs Systems Biology and Medicine;">
<meta name="citation_reference" content="citation_title=&amp;amp;amp;lt;i&amp;gt;simona:&amp;lt;/i&amp;gt;a comprehensive r package for semantic similarity analysis on bio-ontologies;,citation_author=Zuguang Gu;,citation_publication_date=2023-12-05;,citation_cover_date=2023-12-05;,citation_year=2023;,citation_fulltext_html_url=http://dx.doi.org/10.1101/2023.12.03.569758;">
<meta name="citation_reference" content="citation_title=APID database: redefining proteinprotein interaction experimental evidences and binary interactomes;,citation_author=Diego Alonso-López;,citation_author=Francisco J Campos-Laborie;,citation_author=Miguel A Gutiérrez;,citation_author=Luke Lambourne;,citation_author=Michael A Calderwood;,citation_author=Marc Vidal;,citation_author=Javier De Las Rivas;,citation_publication_date=2019-01-01;,citation_cover_date=2019-01-01;,citation_year=2019;,citation_fulltext_html_url=http://dx.doi.org/10.1093/database/baz005;,citation_doi=10.1093/database/baz005;,citation_volume=2019;,citation_language=en;,citation_journal_title=Database;">
<meta name="citation_reference" content="citation_title=The BioGRID database: A comprehensive biomedical resource of curated protein, genetic, and chemical interactions;,citation_author=Rose Oughtred;,citation_author=Jennifer Rust;,citation_author=Christie Chang;,citation_author=Bobby-Joe Breitkreutz;,citation_author=Chris Stark;,citation_author=Andrew Willems;,citation_author=Lorrie Boucher;,citation_author=Genie Leung;,citation_author=Nadine Kolas;,citation_author=Frederick Zhang;,citation_author=Sonam Dolma;,citation_author=Jasmin Coulombe-Huntington;,citation_author=Andrew Chatr-aryamontri;,citation_author=Kara Dolinski;,citation_author=Mike Tyers;,citation_publication_date=2020-11-23;,citation_cover_date=2020-11-23;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1002/pro.3978;,citation_issue=1;,citation_doi=10.1002/pro.3978;,citation_volume=30;,citation_language=en;,citation_journal_title=Protein Science;">
<meta name="citation_reference" content="citation_title=Dual proteome-scale networks reveal cell-specific remodeling of the human interactome;,citation_author=Edward L. Huttlin;,citation_author=Raphael J. Bruckner;,citation_author=Jose Navarrete-Perea;,citation_author=Joe R. Cannon;,citation_author=Kurt Baltier;,citation_author=Fana Gebreab;,citation_author=Melanie P. Gygi;,citation_author=Alexandra Thornock;,citation_author=Gabriela Zarraga;,citation_author=Stanley Tam;,citation_author=John Szpyt;,citation_author=Brandon M. Gassaway;,citation_author=Alexandra Panov;,citation_author=Hannah Parzen;,citation_author=Sipei Fu;,citation_author=Arvene Golbazi;,citation_author=Eila Maenpaa;,citation_author=Keegan Stricker;,citation_author=Sanjukta Guha Thakurta;,citation_author=Tian Zhang;,citation_author=Ramin Rad;,citation_author=Joshua Pan;,citation_author=David P. Nusinow;,citation_author=Joao A. Paulo;,citation_author=Devin K. Schweppe;,citation_author=Laura Pontano Vaites;,citation_author=J. Wade Harper;,citation_author=Steven P. Gygi;,citation_publication_date=2021-05;,citation_cover_date=2021-05;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.cell.2021.04.011;,citation_issue=11;,citation_doi=10.1016/j.cell.2021.04.011;,citation_volume=184;,citation_language=en;,citation_journal_title=Cell;">
<meta name="citation_reference" content="citation_title=CORUM: the comprehensive resource of mammalian protein complexes2022;,citation_author=George Tsitsiridis;,citation_author=Ralph Steinkamp;,citation_author=Madalina Giurgiu;,citation_author=Barbara Brauner;,citation_author=Gisela Fobo;,citation_author=Goar Frishman;,citation_author=Corinna Montrone;,citation_author=Andreas Ruepp;,citation_publication_date=2022-11-16;,citation_cover_date=2022-11-16;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1093/nar/gkac1015;,citation_issue=D1;,citation_doi=10.1093/nar/gkac1015;,citation_volume=51;,citation_language=en;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=HitPredict version 4: comprehensive reliability scoring of physical proteinprotein interactions from more than 100 species;,citation_author=Yosvany López;,citation_author=Kenta Nakai;,citation_author=Ashwini Patil;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_fulltext_html_url=http://dx.doi.org/10.1093/database/bav117;,citation_doi=10.1093/database/bav117;,citation_volume=2015;,citation_language=en;,citation_journal_title=Database;">
<meta name="citation_reference" content="citation_title=A reference map of the human binary protein interactome;,citation_author=Katja Luck;,citation_author=Dae-Kyum Kim;,citation_author=Luke Lambourne;,citation_author=Kerstin Spirohn;,citation_author=Bridget E. Begg;,citation_author=Wenting Bian;,citation_author=Ruth Brignall;,citation_author=Tiziana Cafarelli;,citation_author=Francisco J. Campos-Laborie;,citation_author=Benoit Charloteaux;,citation_author=Dongsic Choi;,citation_author=Atina G. Coté;,citation_author=Meaghan Daley;,citation_author=Steven Deimling;,citation_author=Alice Desbuleux;,citation_author=Amélie Dricot;,citation_author=Marinella Gebbia;,citation_author=Madeleine F. Hardy;,citation_author=Nishka Kishore;,citation_author=Jennifer J. Knapp;,citation_author=István A. Kovács;,citation_author=Irma Lemmens;,citation_author=Miles W. Mee;,citation_author=Joseph C. Mellor;,citation_author=Carl Pollis;,citation_author=Carles Pons;,citation_author=Aaron D. Richardson;,citation_author=Sadie Schlabach;,citation_author=Bridget Teeking;,citation_author=Anupama Yadav;,citation_author=Mariana Babor;,citation_author=Dawit Balcha;,citation_author=Omer Basha;,citation_author=Christian Bowman-Colin;,citation_author=Suet-Feung Chin;,citation_author=Soon Gang Choi;,citation_author=Claudia Colabella;,citation_author=Georges Coppin;,citation_author=Cassandra D’Amata;,citation_author=David De Ridder;,citation_author=Steffi De Rouck;,citation_author=Miquel Duran-Frigola;,citation_author=Hanane Ennajdaoui;,citation_author=Florian Goebels;,citation_author=Liana Goehring;,citation_author=Anjali Gopal;,citation_author=Ghazal Haddad;,citation_author=Elodie Hatchi;,citation_author=Mohamed Helmy;,citation_author=Yves Jacob;,citation_author=Yoseph Kassa;,citation_author=Serena Landini;,citation_author=Roujia Li;,citation_author=Natascha Lieshout;,citation_author=Andrew MacWilliams;,citation_author=Dylan Markey;,citation_author=Joseph N. Paulson;,citation_author=Sudharshan Rangarajan;,citation_author=John Rasla;,citation_author=Ashyad Rayhan;,citation_author=Thomas Rolland;,citation_author=Adriana San-Miguel;,citation_author=Yun Shen;,citation_author=Dayag Sheykhkarimli;,citation_author=Gloria M. Sheynkman;,citation_author=Eyal Simonovsky;,citation_author=Murat Taşan;,citation_author=Alexander Tejeda;,citation_author=Vincent Tropepe;,citation_author=Jean-Claude Twizere;,citation_author=Yang Wang;,citation_author=Robert J. Weatheritt;,citation_author=Jochen Weile;,citation_author=Yu Xia;,citation_author=Xinping Yang;,citation_author=Esti Yeger-Lotem;,citation_author=Quan Zhong;,citation_author=Patrick Aloy;,citation_author=Gary D. Bader;,citation_author=Javier De Las Rivas;,citation_author=Suzanne Gaudet;,citation_author=Tong Hao;,citation_author=Janusz Rak;,citation_author=Jan Tavernier;,citation_author=David E. Hill;,citation_author=Marc Vidal;,citation_author=Frederick P. Roth;,citation_author=Michael A. Calderwood;,citation_publication_date=2020-04-08;,citation_cover_date=2020-04-08;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41586-020-2188-x;,citation_issue=7803;,citation_doi=10.1038/s41586-020-2188-x;,citation_volume=580;,citation_language=en;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=The MIntAct projectIntAct as a common curation platform for 11 molecular interaction databases;,citation_author=Sandra Orchard;,citation_author=Mais Ammari;,citation_author=Bruno Aranda;,citation_author=Lionel Breuza;,citation_author=Leonardo Briganti;,citation_author=Fiona Broackes-Carter;,citation_author=Nancy H. Campbell;,citation_author=Gayatri Chavali;,citation_author=Carol Chen;,citation_author=Noemi Toro;,citation_author=Margaret Duesbury;,citation_author=Marine Dumousseau;,citation_author=Eugenia Galeota;,citation_author=Ursula Hinz;,citation_author=Marta Iannuccelli;,citation_author=Sruthi Jagannathan;,citation_author=Rafael Jimenez;,citation_author=Jyoti Khadake;,citation_author=Astrid Lagreid;,citation_author=Luana Licata;,citation_author=Ruth C. Lovering;,citation_author=Birgit Meldal;,citation_author=Anna N. Melidoni;,citation_author=Mila Milagros;,citation_author=Daniele Peluso;,citation_author=Livia Perfetto;,citation_author=Pablo Porras;,citation_author=Arathi Raghunath;,citation_author=Sylvie Ricard-Blum;,citation_author=Bernd Roechert;,citation_author=Andre Stutz;,citation_author=Michael Tognolli;,citation_author=Kim Roey;,citation_author=Gianni Cesareni;,citation_author=Henning Hermjakob;,citation_publication_date=2013-11-13;,citation_cover_date=2013-11-13;,citation_year=2013;,citation_fulltext_html_url=http://dx.doi.org/10.1093/nar/gkt1115;,citation_issue=D1;,citation_doi=10.1093/nar/gkt1115;,citation_volume=42;,citation_language=en;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=MINT, the molecular interaction database: 2012 update;,citation_author=Luana Licata;,citation_author=Leonardo Briganti;,citation_author=Daniele Peluso;,citation_author=Livia Perfetto;,citation_author=Marta Iannuccelli;,citation_author=Eugenia Galeota;,citation_author=Francesca Sacco;,citation_author=Anita Palma;,citation_author=Aurelio Pio Nardozza;,citation_author=Elena Santonico;,citation_author=Luisa Castagnoli;,citation_author=Gianni Cesareni;,citation_publication_date=2011-11-16;,citation_cover_date=2011-11-16;,citation_year=2011;,citation_fulltext_html_url=http://dx.doi.org/10.1093/nar/gkr930;,citation_issue=D1;,citation_doi=10.1093/nar/gkr930;,citation_volume=40;,citation_language=en;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=The reactome pathway knowledgebase 2022;,citation_author=Marc Gillespie;,citation_author=Bijay Jassal;,citation_author=Ralf Stephan;,citation_author=Marija Milacic;,citation_author=Karen Rothfels;,citation_author=Andrea Senff-Ribeiro;,citation_author=Johannes Griss;,citation_author=Cristoffer Sevilla;,citation_author=Lisa Matthews;,citation_author=Chuqiao Gong;,citation_author=Chuan Deng;,citation_author=Thawfeek Varusai;,citation_author=Eliot Ragueneau;,citation_author=Yusra Haider;,citation_author=Bruce May;,citation_author=Veronica Shamovsky;,citation_author=Joel Weiser;,citation_author=Timothy Brunson;,citation_author=Nasim Sanati;,citation_author=Liam Beckman;,citation_author=Xiang Shao;,citation_author=Antonio Fabregat;,citation_author=Konstantinos Sidiropoulos;,citation_author=Julieth Murillo;,citation_author=Guilherme Viteri;,citation_author=Justin Cook;,citation_author=Solomon Shorser;,citation_author=Gary Bader;,citation_author=Emek Demir;,citation_author=Chris Sander;,citation_author=Robin Haw;,citation_author=Guanming Wu;,citation_author=Lincoln Stein;,citation_author=Henning Hermjakob;,citation_author=Peter D’Eustachio;,citation_publication_date=2021-11-12;,citation_cover_date=2021-11-12;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1093/nar/gkab1028;,citation_issue=D1;,citation_doi=10.1093/nar/gkab1028;,citation_volume=50;,citation_language=en;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=The STRING database in 2023: proteinprotein association networks and functional enrichment analyses for any sequenced genome of interest;,citation_author=Damian Szklarczyk;,citation_author=Rebecca Kirsch;,citation_author=Mikaela Koutrouli;,citation_author=Katerina Nastou;,citation_author=Farrokh Mehryary;,citation_author=Radja Hachilif;,citation_author=Annika L Gable;,citation_author=Tao Fang;,citation_author=Nadezhda&nbsp;T Doncheva;,citation_author=Sampo Pyysalo;,citation_author=Peer Bork;,citation_author=Lars&nbsp;J Jensen;,citation_author=Christian Mering;,citation_publication_date=2022-11-12;,citation_cover_date=2022-11-12;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1093/nar/gkac1000;,citation_issue=D1;,citation_doi=10.1093/nar/gkac1000;,citation_volume=51;,citation_language=en;,citation_journal_title=Nucleic Acids Research;">
<meta name="citation_reference" content="citation_title=The Neurovascular Unit Coming of Age: A Journey through Neurovascular Coupling in Health and Disease;,citation_author=Costantino Iadecola;,citation_publication_date=2017-09;,citation_cover_date=2017-09;,citation_year=2017;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.neuron.2017.07.030;,citation_issue=1;,citation_doi=10.1016/j.neuron.2017.07.030;,citation_volume=96;,citation_language=en;,citation_journal_title=Neuron;">
<meta name="citation_reference" content="citation_title=Role of blood-brain barrier in alzheimer’s disease;,citation_author=Zhiyou Cai;,citation_author=Pei-Feng Qiao;,citation_author=Cheng-Qun Wan;,citation_author=Min Cai;,citation_author=Nan-Kai Zhou;,citation_author=Qin Li;,citation_publication_date=2018-05-30;,citation_cover_date=2018-05-30;,citation_year=2018;,citation_fulltext_html_url=http://dx.doi.org/10.3233/JAD-180098;,citation_issue=4;,citation_doi=10.3233/jad-180098;,citation_volume=63;,citation_journal_title=Journal of Alzheimer’s Disease;">
<meta name="citation_reference" content="citation_title=Blood-brain barrier leakage in Alzheimer’s disease: From discovery to clinical relevance;,citation_author=Geetika Nehra;,citation_author=Bjoern Bauer;,citation_author=Anika M. S. Hartz;,citation_publication_date=2022-06;,citation_cover_date=2022-06;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.pharmthera.2022.108119;,citation_doi=10.1016/j.pharmthera.2022.108119;,citation_volume=234;,citation_language=en;,citation_journal_title=Pharmacology &amp;amp;amp; Therapeutics;">
<meta name="citation_reference" content="citation_title=Cell-type-specific Alzheimer’s disease polygenic risk scores are associated with distinct disease processes in Alzheimer’s disease;,citation_author=Hyun-Sik Yang;,citation_author=Ling Teng;,citation_author=Daniel Kang;,citation_author=Vilas Menon;,citation_author=Tian Ge;,citation_author=Hilary K. Finucane;,citation_author=Aaron P. Schultz;,citation_author=Michael Properzi;,citation_author=Hans-Ulrich Klein;,citation_author=Lori B. Chibnik;,citation_author=Julie A. Schneider;,citation_author=David A. Bennett;,citation_author=Timothy J. Hohman;,citation_author=Richard P. Mayeux;,citation_author=Keith A. Johnson;,citation_author=Philip L. De Jager;,citation_author=Reisa A. Sperling;,citation_publication_date=2023-11-30;,citation_cover_date=2023-11-30;,citation_year=2023;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41467-023-43132-2;,citation_issue=1;,citation_doi=10.1038/s41467-023-43132-2;,citation_volume=14;,citation_language=en;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Microglia in Alzheimer’s Disease: Exploring How Genetics and Phenotype Influence Risk;,citation_author=Amanda McQuade;,citation_author=Mathew Blurton-Jones;,citation_publication_date=2019-04;,citation_cover_date=2019-04;,citation_year=2019;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.jmb.2019.01.045;,citation_issue=9;,citation_doi=10.1016/j.jmb.2019.01.045;,citation_volume=431;,citation_language=en;,citation_journal_title=Journal of Molecular Biology;">
<meta name="citation_reference" content="citation_title=A single nuclear transcriptomic characterisation of mechanisms responsible for impaired angiogenesis and blood-brain barrier function in Alzheimer’s disease;,citation_author=Stergios Tsartsalis;,citation_author=Hannah Sleven;,citation_author=Nurun Fancy;,citation_author=Frank Wessely;,citation_author=Amy M. Smith;,citation_author=Nanet Willumsen;,citation_author=To Ka Dorcas Cheung;,citation_author=Michal J. Rokicki;,citation_author=Vicky Chau;,citation_author=Eseoghene Ifie;,citation_author=Combiz Khozoie;,citation_author=Olaf Ansorge;,citation_author=Xin Yang;,citation_author=Marion H. Jenkyns;,citation_author=Karen Davey;,citation_author=Aisling McGarry;,citation_author=Robert C. J. Muirhead;,citation_author=Stephanie Debette;,citation_author=Johanna S. Jackson;,citation_author=Axel Montagne;,citation_author=David R. Owen;,citation_author=J. Scott Miners;,citation_author=Seth Love;,citation_author=Caleb Webber;,citation_author=M. Zameel Cader;,citation_author=Paul M. Matthews;,citation_publication_date=2024-03-12;,citation_cover_date=2024-03-12;,citation_year=2024;,citation_fulltext_html_url=https://www.nature.com/articles/s41467-024-46630-z;,citation_issue=1;,citation_doi=10.1038/s41467-024-46630-z;,citation_volume=15;,citation_language=en;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Single-cell atlas reveals correlates of high cognitive function, dementia, and resilience to Alzheimer’s disease pathology;,citation_author=Hansruedi Mathys;,citation_author=Zhuyu Peng;,citation_author=Carles A. Boix;,citation_author=Matheus B. Victor;,citation_author=Noelle Leary;,citation_author=Sudhagar Babu;,citation_author=Ghada Abdelhady;,citation_author=Xueqiao Jiang;,citation_author=Ayesha P. Ng;,citation_author=Kimia Ghafari;,citation_author=Alexander K. Kunisky;,citation_author=Julio Mantero;,citation_author=Kyriaki Galani;,citation_author=Vanshika N. Lohia;,citation_author=Gabrielle E. Fortier;,citation_author=Yasmine Lotfi;,citation_author=Jason Ivey;,citation_author=Hannah P. Brown;,citation_author=Pratham R. Patel;,citation_author=Nehal Chakraborty;,citation_author=Jacob I. Beaudway;,citation_author=Elizabeth J. Imhoff;,citation_author=Cameron F. Keeler;,citation_author=Maren M. McChesney;,citation_author=Haishal H. Patel;,citation_author=Sahil P. Patel;,citation_author=Megan T. Thai;,citation_author=David A. Bennett;,citation_author=Manolis Kellis;,citation_author=Li-Huei Tsai;,citation_publication_date=2023-09-28;,citation_cover_date=2023-09-28;,citation_year=2023;,citation_issue=20;,citation_doi=10.1016/j.cell.2023.08.039;,citation_volume=186;,citation_language=eng;,citation_journal_title=Cell;">
<meta name="citation_reference" content="citation_title=Endothelial Cells Lining Sporadic Cerebral Cavernous Malformation Cavernomas Undergo Endothelial-to-Mesenchymal Transition;,citation_author=Luca Bravi;,citation_author=Matteo Malinverno;,citation_author=Federica Pisati;,citation_author=Noemi Rudini;,citation_author=Roberto Cuttano;,citation_author=Roberto Pallini;,citation_author=Maurizio Martini;,citation_author=Luigi Maria Larocca;,citation_author=Marco Locatelli;,citation_author=Vincenzo Levi;,citation_author=Giulio Andrea Bertani;,citation_author=Elisabetta Dejana;,citation_author=Maria Grazia Lampugnani;,citation_publication_date=2016-03;,citation_cover_date=2016-03;,citation_year=2016;,citation_fulltext_html_url=http://dx.doi.org/10.1161/STROKEAHA.115.011867;,citation_issue=3;,citation_doi=10.1161/strokeaha.115.011867;,citation_volume=47;,citation_language=en;,citation_journal_title=Stroke;">
<meta name="citation_reference" content="citation_title=Evidence for endothelial-to-mesenchymal transition in human brain arteriovenous malformations;,citation_author=Lorelei D. Shoemaker;,citation_author=Aaron K. McCormick;,citation_author=Breanna M. Allen;,citation_author=Steven D. Chang;,citation_publication_date=2020-06;,citation_cover_date=2020-06;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1002/ctm2.99;,citation_issue=2;,citation_doi=10.1002/ctm2.99;,citation_volume=10;,citation_language=en;,citation_journal_title=Clinical and Translational Medicine;">
<meta name="citation_reference" content="citation_title=Trajectory-based differential expression analysis for single-cell sequencing data;,citation_author=Koen Van den Berge;,citation_author=Hector Roux de Bézieux;,citation_author=Kelly Street;,citation_author=Wouter Saelens;,citation_author=Robrecht Cannoodt;,citation_author=Yvan Saeys;,citation_author=Sandrine Dudoit;,citation_author=Lieven Clement;,citation_publication_date=2020-03-05;,citation_cover_date=2020-03-05;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41467-020-14766-3;,citation_issue=1;,citation_doi=10.1038/s41467-020-14766-3;,citation_volume=11;,citation_language=en;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Trajectory-based differential expression analysis for single-cell sequencing data;,citation_author=Koen Van den Berge;,citation_author=Hector Roux de Bézieux;,citation_author=Kelly Street;,citation_author=Wouter Saelens;,citation_author=Robrecht Cannoodt;,citation_author=Yvan Saeys;,citation_author=Sandrine Dudoit;,citation_author=Lieven Clement;,citation_publication_date=2020-03-05;,citation_cover_date=2020-03-05;,citation_year=2020;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41467-020-14766-3;,citation_issue=1;,citation_doi=10.1038/s41467-020-14766-3;,citation_volume=11;,citation_language=en;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data;,citation_author=Greg Finak;,citation_author=Andrew McDavid;,citation_author=Masanao Yajima;,citation_author=Jingyuan Deng;,citation_author=Vivian Gersuk;,citation_author=Alex K. Shalek;,citation_author=Chloe K. Slichter;,citation_author=Hannah W. Miller;,citation_author=M. Juliana McElrath;,citation_author=Martin Prlic;,citation_author=Peter S. Linsley;,citation_author=Raphael Gottardo;,citation_publication_date=2015-12;,citation_cover_date=2015-12;,citation_year=2015;,citation_fulltext_html_url=http://dx.doi.org/10.1186/s13059-015-0844-5;,citation_issue=1;,citation_doi=10.1186/s13059-015-0844-5;,citation_volume=16;,citation_language=en;,citation_journal_title=Genome Biology;">
<meta name="citation_reference" content="citation_title=BACE2: A Promising Neuroprotective Candidate for Alzheimer’s Disease;,citation_author=Yee Jie Yeap;,citation_author=Nagaendran Kandiah;,citation_author=Dean Nizetic;,citation_author=Kah-Leong Lim;,citation_publication_date=2022-11-26;,citation_cover_date=2022-11-26;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.3233/JAD-220867;,citation_issue=s1;,citation_doi=10.3233/jad-220867;,citation_volume=94;,citation_language=en;,citation_journal_title=Journal of Alzheimer’s Disease;">
<meta name="citation_reference" content="citation_title=Apolipoprotein D Upregulation in Alzheimer’s Disease but Not Frontotemporal Dementia;,citation_author=Surabhi Bhatia;,citation_author=Woojin Scott Kim;,citation_author=Claire E. Shepherd;,citation_author=Glenda M. Halliday;,citation_publication_date=2018-11-22;,citation_cover_date=2018-11-22;,citation_year=2018;,citation_fulltext_html_url=http://dx.doi.org/10.1007/s12031-018-1217-9;,citation_issue=1;,citation_doi=10.1007/s12031-018-1217-9;,citation_volume=67;,citation_language=en;,citation_journal_title=Journal of Molecular Neuroscience;">
<meta name="citation_reference" content="citation_title=LSD1 and Aberrant DNA Methylation Mediate Persistence of Enteroendocrine Progenitors That Support &amp;amp;amp;lt;i&amp;gt;BRAF&amp;lt;/i&amp;gt;-Mutant Colorectal Cancer;,citation_author=Samuel A. Miller;,citation_author=Robert A. Policastro;,citation_author=Shruthi Sriramkumar;,citation_author=Tim Lai;,citation_author=Thomas D. Huntington;,citation_author=Christopher A. Ladaika;,citation_author=Daeho Kim;,citation_author=Chunhai Hao;,citation_author=Gabriel E. Zentner;,citation_author=Heather M. O’Hagan;,citation_publication_date=2021-05-25;,citation_cover_date=2021-05-25;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1158/0008-5472.CAN-20-3562;,citation_issue=14;,citation_doi=10.1158/0008-5472.can-20-3562;,citation_volume=81;,citation_language=en;,citation_journal_title=Cancer Research;">
<meta name="citation_reference" content="citation_title=Notch3 directs differentiation of brain mural cells from human pluripotent stem cellderived neural crest;,citation_author=Benjamin D. Gastfriend;,citation_author=Margaret E. Snyder;,citation_author=Hope E. Holt;,citation_author=Richard Daneman;,citation_author=Sean P. Palecek;,citation_author=Eric V. Shusta;,citation_publication_date=2024-02-02;,citation_cover_date=2024-02-02;,citation_year=2024;,citation_fulltext_html_url=http://dx.doi.org/10.1126/sciadv.adi1737;,citation_issue=5;,citation_doi=10.1126/sciadv.adi1737;,citation_volume=10;,citation_language=en;,citation_journal_title=Science Advances;">
<meta name="citation_reference" content="citation_title=Regulation of endothelial homeostasis, vascular development and angiogenesis by the transcription factor ERG;,citation_author=Aarti V. Shah;,citation_author=Graeme M. Birdsey;,citation_author=Anna M. Randi;,citation_publication_date=2016-11;,citation_cover_date=2016-11;,citation_year=2016;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.vph.2016.05.003;,citation_doi=10.1016/j.vph.2016.05.003;,citation_volume=86;,citation_language=en;,citation_journal_title=Vascular Pharmacology;">
<meta name="citation_reference" content="citation_title=Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics;,citation_author=Kelly Street;,citation_author=Davide Risso;,citation_author=Russell B. Fletcher;,citation_author=Diya Das;,citation_author=John Ngai;,citation_author=Nir Yosef;,citation_author=Elizabeth Purdom;,citation_author=Sandrine Dudoit;,citation_publication_date=2018-06-19;,citation_cover_date=2018-06-19;,citation_year=2018;,citation_fulltext_html_url=http://dx.doi.org/10.1186/s12864-018-4772-0;,citation_issue=1;,citation_doi=10.1186/s12864-018-4772-0;,citation_volume=19;,citation_language=en;,citation_journal_title=BMC Genomics;">
<meta name="citation_reference" content="citation_title=Inference and analysis of cell-cell communication using CellChat;,citation_author=Suoqin Jin;,citation_author=Christian F. Guerrero-Juarez;,citation_author=Lihua Zhang;,citation_author=Ivan Chang;,citation_author=Raul Ramos;,citation_author=Chen-Hsiang Kuan;,citation_author=Peggy Myung;,citation_author=Maksim V. Plikus;,citation_author=Qing Nie;,citation_publication_date=2021-02-17;,citation_cover_date=2021-02-17;,citation_year=2021;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41467-021-21246-9;,citation_issue=1;,citation_doi=10.1038/s41467-021-21246-9;,citation_volume=12;,citation_language=en;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Angiopoietin 2 in placentation and tumor biology: The yin and yang of vascular biology;,citation_author=Qiong Wang;,citation_author=Gendie E. Lash;,citation_publication_date=2017-08;,citation_cover_date=2017-08;,citation_year=2017;,citation_fulltext_html_url=http://dx.doi.org/10.1016/j.placenta.2017.03.021;,citation_doi=10.1016/j.placenta.2017.03.021;,citation_volume=56;,citation_language=en;,citation_journal_title=Placenta;">
<meta name="citation_reference" content="citation_title=Colony-stimulating factor 1 receptor inhibition prevents microglial plaque association and improves cognition in 3xTg-AD mice;,citation_author=Nabil N. Dagher;,citation_author=Allison R. Najafi;,citation_author=Kara M. Neely Kayala;,citation_author=Monica R. P. Elmore;,citation_author=Terra E. White;,citation_author=Rodrigo Medeiros;,citation_author=Brian L. West;,citation_author=Kim N. Green;,citation_publication_date=2015-08-01;,citation_cover_date=2015-08-01;,citation_year=2015;,citation_fulltext_html_url=http://dx.doi.org/10.1186/s12974-015-0366-9;,citation_issue=1;,citation_doi=10.1186/s12974-015-0366-9;,citation_volume=12;,citation_language=en;,citation_journal_title=Journal of Neuroinflammation;">
<meta name="citation_reference" content="citation_title=Studies on colony stimulating factor receptor-1 and ligands colony stimulating factor-1 and interleukin-34 in alzheimer’s disease brains and human microglia;,citation_author=Douglas G. Walker;,citation_author=Tiffany M. Tang;,citation_author=Lih-Fen Lue;,citation_publication_date=2017-08-09;,citation_cover_date=2017-08-09;,citation_year=2017;,citation_fulltext_html_url=http://dx.doi.org/10.3389/fnagi.2017.00244;,citation_doi=10.3389/fnagi.2017.00244;,citation_volume=9;,citation_journal_title=Frontiers in Aging Neuroscience;">
<meta name="citation_reference" content="citation_title=Pathological changes within the cerebral vasculature in Alzheimer’s disease: New perspectives;,citation_author=Robert A. Fisher;,citation_author=J. Scott Miners;,citation_author=Seth Love;,citation_publication_date=2022-03-14;,citation_cover_date=2022-03-14;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.1111/bpa.13061;,citation_issue=6;,citation_doi=10.1111/bpa.13061;,citation_volume=32;,citation_language=en;,citation_journal_title=Brain Pathology;">
<meta name="citation_reference" content="citation_title=Endothelial-to-mesenchymal transition contributes to cardiac fibrosis;,citation_author=Elisabeth M Zeisberg;,citation_author=Oleg Tarnavski;,citation_author=Michael Zeisberg;,citation_author=Adam L Dorfman;,citation_author=Julie R McMullen;,citation_author=Erika Gustafsson;,citation_author=Anil Chandraker;,citation_author=Xueli Yuan;,citation_author=William T Pu;,citation_author=Anita B Roberts;,citation_author=Eric G Neilson;,citation_author=Mohamed H Sayegh;,citation_author=Seigo Izumo;,citation_author=Raghu Kalluri;,citation_publication_date=2007-07-29;,citation_cover_date=2007-07-29;,citation_year=2007;,citation_fulltext_html_url=http://dx.doi.org/10.1038/nm1613;,citation_issue=8;,citation_doi=10.1038/nm1613;,citation_volume=13;,citation_language=en;,citation_journal_title=Nature Medicine;">
<meta name="citation_reference" content="citation_title=Transforming growth factor-β2 promotes Snail-mediated endothelialmesenchymal transition through convergence of Smad-dependent and Smad-independent signalling;,citation_author=Damian Medici;,citation_author=Scott Potenta;,citation_author=Raghu Kalluri;,citation_publication_date=2011-07-13;,citation_cover_date=2011-07-13;,citation_year=2011;,citation_fulltext_html_url=http://dx.doi.org/10.1042/BJ20101500;,citation_issue=3;,citation_doi=10.1042/bj20101500;,citation_volume=437;,citation_language=en;,citation_journal_title=Biochemical Journal;">
<meta name="citation_reference" content="citation_title=Conversion of vascular endothelial cells into multipotent stem-like cells;,citation_author=Damian Medici;,citation_author=Eileen M Shore;,citation_author=Vitali Y Lounev;,citation_author=Frederick S Kaplan;,citation_author=Raghu Kalluri;,citation_author=Bjorn R Olsen;,citation_publication_date=2010-11-21;,citation_cover_date=2010-11-21;,citation_year=2010;,citation_fulltext_html_url=http://dx.doi.org/10.1038/nm.2252;,citation_issue=12;,citation_doi=10.1038/nm.2252;,citation_volume=16;,citation_language=en;,citation_journal_title=Nature Medicine;">
<meta name="citation_reference" content="citation_title=Induction of alpha-smooth muscle actin expression in cultured human brain pericytes by transforming growth factor-beta 1.;,citation_author=M. M. Verbeek;,citation_author=I. Otte-Höller;,citation_author=P. Wesseling;,citation_author=D. J. Ruiter;,citation_author=R. M. Waal;,citation_publication_date=1994-02;,citation_cover_date=1994-02;,citation_year=1994;,citation_fulltext_html_url=https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1887139/;,citation_issue=2;,citation_volume=144;,citation_journal_title=The American Journal of Pathology;">
<meta name="citation_reference" content="citation_title=EndMT contributes to the onset and progression of cerebral cavernous malformations;,citation_author=Luigi Maddaluno;,citation_author=Noemi Rudini;,citation_author=Roberto Cuttano;,citation_author=Luca Bravi;,citation_author=Costanza Giampietro;,citation_author=Monica Corada;,citation_author=Luca Ferrarini;,citation_author=Fabrizio Orsenigo;,citation_author=Eleanna Papa;,citation_author=Gwenola Boulday;,citation_author=Elisabeth Tournier-Lasserve;,citation_author=Françoise Chapon;,citation_author=Cristina Richichi;,citation_author=Saverio Francesco Retta;,citation_author=Maria Grazia Lampugnani;,citation_author=Elisabetta Dejana;,citation_publication_date=2013-06;,citation_cover_date=2013-06;,citation_year=2013;,citation_fulltext_html_url=http://dx.doi.org/10.1038/nature12207;,citation_issue=7455;,citation_doi=10.1038/nature12207;,citation_volume=498;,citation_language=en;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=The Pericyte: A Forgotten Cell Type with Important Implications for Alzheimer’s Disease?;,citation_author=Ethan A. Winkler;,citation_author=Abhay P. Sagare;,citation_author=Berislav V. Zlokovic;,citation_publication_date=2014-06-19;,citation_cover_date=2014-06-19;,citation_year=2014;,citation_fulltext_html_url=http://dx.doi.org/10.1111/bpa.12152;,citation_issue=4;,citation_doi=10.1111/bpa.12152;,citation_volume=24;,citation_language=en;,citation_journal_title=Brain Pathology;">
<meta name="citation_reference" content="citation_title=BACE2: A Promising Neuroprotective Candidate for Alzheimer’s Disease;,citation_author=Yee Jie Yeap;,citation_author=Nagaendran Kandiah;,citation_author=Dean Nizetic;,citation_author=Kah-Leong Lim;,citation_publication_date=2022-11-26;,citation_cover_date=2022-11-26;,citation_year=2022;,citation_fulltext_html_url=http://dx.doi.org/10.3233/JAD-220867;,citation_issue=s1;,citation_doi=10.3233/jad-220867;,citation_volume=94;,citation_language=en;,citation_journal_title=Journal of Alzheimer’s Disease;">
<meta name="citation_reference" content="citation_title=Elevated CSF angiopoietin-2 correlates with blood-brain barrier leakiness and markers of neuronal injury in early Alzheimer’s disease;,citation_author=Carol Van Hulle;,citation_author=Selvi Ince;,citation_author=Ozioma C. Okonkwo;,citation_author=Barbara B. Bendlin;,citation_author=Sterling C. Johnson;,citation_author=Cynthia M. Carlsson;,citation_author=Sanjay Asthana;,citation_author=Seth Love;,citation_author=Kaj Blennow;,citation_author=Henrik Zetterberg;,citation_author=J. Scott Miners;,citation_publication_date=2024-01-05;,citation_cover_date=2024-01-05;,citation_year=2024;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41398-023-02706-w;,citation_issue=1;,citation_doi=10.1038/s41398-023-02706-w;,citation_volume=14;,citation_language=en;,citation_journal_title=Translational Psychiatry;">
<meta name="citation_reference" content="citation_title=Elevated plasma angiogenesis factors in alzheimer’s disease;,citation_author=Wei Qin;,citation_author=Xiangfei Jia;,citation_author=Fen Wang;,citation_author=Xiumei Zuo;,citation_author=Liyong Wu;,citation_author=Aihong Zhou;,citation_author=Dan Li;,citation_author=Baoquan Min;,citation_author=Cuibai Wei;,citation_author=Yi Tang;,citation_author=Yi Xing;,citation_author=Xiumin Dong;,citation_author=Qi Wang;,citation_author=Yining Gao;,citation_author=Ying Li;,citation_author=Jianping Jia;,citation_publication_date=2015-03-03;,citation_cover_date=2015-03-03;,citation_year=2015;,citation_fulltext_html_url=http://dx.doi.org/10.3233/JAD-142409;,citation_issue=1;,citation_doi=10.3233/jad-142409;,citation_volume=45;,citation_journal_title=Journal of Alzheimer’s Disease;">
<meta name="citation_reference" content="citation_title=Single-cell atlas of the human brain vasculature across development, adulthood and disease;,citation_author=Thomas Wälchli;,citation_author=Moheb Ghobrial;,citation_author=Marc Schwab;,citation_author=Shigeki Takada;,citation_author=Hang Zhong;,citation_author=Samuel Suntharalingham;,citation_author=Sandra Vetiska;,citation_author=Daymé Rodrigues Gonzalez;,citation_author=Ruilin Wu;,citation_author=Hubert Rehrauer;,citation_author=Anuroopa Dinesh;,citation_author=Kai Yu;,citation_author=Edward L. Y. Chen;,citation_author=Jeroen Bisschop;,citation_author=Fiona Farnhammer;,citation_author=Ann Mansur;,citation_author=Joanna Kalucka;,citation_author=Itay Tirosh;,citation_author=Luca Regli;,citation_author=Karl Schaller;,citation_author=Karl Frei;,citation_author=Troy Ketela;,citation_author=Mark Bernstein;,citation_author=Paul Kongkham;,citation_author=Peter Carmeliet;,citation_author=Taufik Valiante;,citation_author=Peter B. Dirks;,citation_author=Mario L. Suva;,citation_author=Gelareh Zadeh;,citation_author=Viviane Tabar;,citation_author=Ralph Schlapbach;,citation_author=Hartland W. Jackson;,citation_author=Katrien De Bock;,citation_author=Jason E. Fish;,citation_author=Philippe P. Monnier;,citation_author=Gary D. Bader;,citation_author=Ivan Radovanovic;,citation_publication_date=2024-07-10;,citation_cover_date=2024-07-10;,citation_year=2024;,citation_fulltext_html_url=http://dx.doi.org/10.1038/s41586-024-07493-y;,citation_issue=8025;,citation_doi=10.1038/s41586-024-07493-y;,citation_volume=632;,citation_language=en;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=A robust and high-throughput Cre reporting and characterization system for the whole mouse brain;,citation_author=Linda Madisen;,citation_author=Theresa A Zwingman;,citation_author=Susan M Sunkin;,citation_author=Seung Wook Oh;,citation_author=Hatim A Zariwala;,citation_author=Hong Gu;,citation_author=Lydia L Ng;,citation_author=Richard D Palmiter;,citation_author=Michael J Hawrylycz;,citation_author=Allan R Jones;,citation_author=Ed S Lein;,citation_author=Hongkui Zeng;,citation_publication_date=2009-12-20;,citation_cover_date=2009-12-20;,citation_year=2009;,citation_fulltext_html_url=http://dx.doi.org/10.1038/nn.2467;,citation_issue=1;,citation_doi=10.1038/nn.2467;,citation_volume=13;,citation_language=en;,citation_journal_title=Nature Neuroscience;">
<meta name="citation_reference" content="citation_title=Runx1 is required for the endothelial to haematopoietic cell transition but not thereafter;,citation_author=Michael J. Chen;,citation_author=Tomomasa Yokomizo;,citation_author=Brandon M. Zeigler;,citation_author=Elaine Dzierzak;,citation_author=Nancy A. Speck;,citation_publication_date=2009-02-01;,citation_cover_date=2009-02-01;,citation_year=2009;,citation_fulltext_html_url=http://dx.doi.org/10.1038/nature07619;,citation_issue=7231;,citation_doi=10.1038/nature07619;,citation_volume=457;,citation_language=en;,citation_journal_title=Nature;">
</head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Article Notebook</h6>

            <a href="./index.qmd" class="btn btn-primary quarto-download-embed" download="index.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Paired Vascular-Parenchymal Single Nuclei Sequencing Reveals Perturbed Endothelial-to-Mesenchymal Transition in Alzheimer’s Disease</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliations</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Gabriel Mateus Bernardo Harrington, PhD <a href="mailto:bernardo-harringtong@cardiff.ac.uk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-6075-3619" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                    <p class="affiliation">
                        <a href="https://ukdri.ac.uk/">
                        UK Dementia Research Institue
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Hannah Sleven, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="https://www.ox.ac.uk/">
                        Oxford University
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Jimena Monzón-Sandoval, PhD <a href="https://orcid.org/0000-0002-4190-2188" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                    <p class="affiliation">
                        <a href="https://ukdri.ac.uk/">
                        UK Dementia Research Institue
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Gabriel Rocha </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="https://www.ox.ac.uk/">
                        Oxford University
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Lara Robinson </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="https://www.ox.ac.uk/">
                        Oxford University
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Michal Rokicki, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Centre for Neuropsychiatric Genetics and Genomics, Division of Psychological Medicine and Clinical Neurosciences, School of Medicine, Cardiff University, Cardiff, United Kingdom
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Joanne Morgan, PhD <a href="https://orcid.org/0000-0003-3091-0040" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Centre for Neuropsychiatric Genetics and Genomics, Division of Psychological Medicine and Clinical Neurosciences, School of Medicine, Cardiff University, Cardiff, United Kingdom
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Ngoc-Nga Vinh, PhD <a href="https://orcid.org/0000-0002-3233-5230" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Centre for Neuropsychiatric Genetics and Genomics, Division of Psychological Medicine and Clinical Neurosciences, School of Medicine, Cardiff University, Cardiff, United Kingdom
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Quenten Schwarz, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Centre for Neuropsychiatric Genetics and Genomics, Division of Psychological Medicine and Clinical Neurosciences, School of Medicine, Cardiff University, Cardiff, United Kingdom
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Caleb Webber, PhD <a href="mailto:webberc4@cardiff.ac.uk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                    <p class="affiliation">
                        <a href="https://ukdri.ac.uk/">
                        UK Dementia Research Institue
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Zameel Cader, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="https://www.ox.ac.uk/">
                        Oxford University
                        </a>
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">May 30, 2025</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>The neurovascular unit (NVU) underpins critical brain functions by integrating vascular cells, neurons, and glia to maintain metabolic support and blood–brain barrier integrity. Despite mounting evidence implicating vascular dysfunction in Alzheimer’s disease (AD), the contribution of vascular cell types remains underexplored, largely owing to technical challenges in isolating these cells from post-mortem tissue. Here, we describe a novel approach that separately enriches for vascular and parenchymal nuclei at high purity from a single sample of human brain tissue. By applying this method to prefrontal cortex tissue from AD patients and controls and performing single-nuclei RNA sequencing, we assembled a high-resolution atlas capturing major vascular and parenchymal cell types. This unbiased approach reveals distinct associations of AD risk with pericyte-, perivascular fibroblast-, and activated microglia-specific gene expression profiles. Functional analyses show that the genes driving these associations converge on amyloid-related pathways, implicating perivascular compartments and immune modulators in the early stages of AD pathobiology. Furthermore, we identify a new endothelial-mesenchymal transition population, diminished in AD suggesting a disruption in endothelial plasticity that may underlie vascular dysfunction in AD. Key ligand–receptor interactions between vascular and parenchymal cell types, including ANGPT2, were markedly dysregulated, highlighting specific targets that could drive NVU breakdown in AD. Taken together, these findings establish a framework for comprehensively evaluating NVU cells in human disease, emphasizing novel links between vascular dynamics and AD pathogenesis. Our method provides an essential foundation for deciphering disease mechanisms and offers new therapeutic avenues focused on protecting and restoring the brain’s neurovascular interface.</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Alzheimer’s diesease, Blood brain barrier</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">2</span> Results</a>
  <ul class="collapse">
  <li><a href="#isolating-the-neurovascular-unit" id="toc-isolating-the-neurovascular-unit" class="nav-link" data-scroll-target="#isolating-the-neurovascular-unit"><span class="header-section-number">2.1</span> Isolating the neurovascular unit</a></li>
  <li><a href="#a-single-cell-atlas-of-the-neurovascular-unit" id="toc-a-single-cell-atlas-of-the-neurovascular-unit" class="nav-link" data-scroll-target="#a-single-cell-atlas-of-the-neurovascular-unit"><span class="header-section-number">2.2</span> A single cell atlas of the neurovascular unit</a></li>
  <li><a href="#activated-microglia-pericytes-perivascular-fibroblasts-and-t-cells-are-enriched-for-ad-risk-genes-focused-on-amyloid-processing-and-immune-cell-activation" id="toc-activated-microglia-pericytes-perivascular-fibroblasts-and-t-cells-are-enriched-for-ad-risk-genes-focused-on-amyloid-processing-and-immune-cell-activation" class="nav-link" data-scroll-target="#activated-microglia-pericytes-perivascular-fibroblasts-and-t-cells-are-enriched-for-ad-risk-genes-focused-on-amyloid-processing-and-immune-cell-activation"><span class="header-section-number">2.3</span> Activated microglia, pericytes, perivascular fibroblasts and T cells are enriched for AD risk genes focused on amyloid processing and immune cell activation</a></li>
  <li><a href="#a-population-of-endothelial-mesenchymal-transitional-cells-are-detected-in-human-prefrontal-cortex-whose-transition-appears-disrupted-in-alzheimers-disease" id="toc-a-population-of-endothelial-mesenchymal-transitional-cells-are-detected-in-human-prefrontal-cortex-whose-transition-appears-disrupted-in-alzheimers-disease" class="nav-link" data-scroll-target="#a-population-of-endothelial-mesenchymal-transitional-cells-are-detected-in-human-prefrontal-cortex-whose-transition-appears-disrupted-in-alzheimers-disease"><span class="header-section-number">2.4</span> A population of Endothelial Mesenchymal Transitional Cells are detected in human prefrontal cortex whose transition appears disrupted in Alzheimer’s Disease</a></li>
  <li><a href="#blood-vessel-stabilisation-signalling-is-altered-in-alzheimers-disease" id="toc-blood-vessel-stabilisation-signalling-is-altered-in-alzheimers-disease" class="nav-link" data-scroll-target="#blood-vessel-stabilisation-signalling-is-altered-in-alzheimers-disease"><span class="header-section-number">2.5</span> Blood vessel stabilisation signalling is altered in Alzheimer’s Disease</a></li>
  <li><a href="#cell-cell-interactions-between-vascular-and-parenchymal-cells-are-through-specific-signalling-pathways" id="toc-cell-cell-interactions-between-vascular-and-parenchymal-cells-are-through-specific-signalling-pathways" class="nav-link" data-scroll-target="#cell-cell-interactions-between-vascular-and-parenchymal-cells-are-through-specific-signalling-pathways"><span class="header-section-number">2.6</span> Cell-cell interactions between vascular and parenchymal cells are through specific signalling pathways</a></li>
  <li><a href="#alzheimers-disease-affects-genes-involved-in-protein-folding-and-vascular-remodeling" id="toc-alzheimers-disease-affects-genes-involved-in-protein-folding-and-vascular-remodeling" class="nav-link" data-scroll-target="#alzheimers-disease-affects-genes-involved-in-protein-folding-and-vascular-remodeling"><span class="header-section-number">2.7</span> Alzheimer’s Disease affects genes involved in protein folding and vascular remodeling</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="header-section-number">3</span> Discussion</a>
  <ul class="collapse">
  <li><a href="#acknowledgments" id="toc-acknowledgments" class="nav-link" data-scroll-target="#acknowledgments"><span class="header-section-number">3.1</span> Acknowledgments</a></li>
  <li><a href="#author-contributions" id="toc-author-contributions" class="nav-link" data-scroll-target="#author-contributions"><span class="header-section-number">3.2</span> Author contributions</a></li>
  <li><a href="#declaration-of-interests" id="toc-declaration-of-interests" class="nav-link" data-scroll-target="#declaration-of-interests"><span class="header-section-number">3.3</span> Declaration of interests</a></li>
  <li><a href="#figure-titles-and-legends" id="toc-figure-titles-and-legends" class="nav-link" data-scroll-target="#figure-titles-and-legends"><span class="header-section-number">3.4</span> Figure titles and legends</a></li>
  <li><a href="#tables-with-title-and-legends" id="toc-tables-with-title-and-legends" class="nav-link" data-scroll-target="#tables-with-title-and-legends"><span class="header-section-number">3.5</span> Tables with title and legends</a></li>
  </ul></li>
  <li><a href="#star-methods" id="toc-star-methods" class="nav-link" data-scroll-target="#star-methods"><span class="header-section-number">4</span> STAR Methods</a>
  <ul class="collapse">
  <li><a href="#key-resoures-table" id="toc-key-resoures-table" class="nav-link" data-scroll-target="#key-resoures-table"><span class="header-section-number">4.1</span> Key resoures table</a></li>
  <li><a href="#resource-availability" id="toc-resource-availability" class="nav-link" data-scroll-target="#resource-availability"><span class="header-section-number">4.2</span> Resource availability</a>
  <ul class="collapse">
  <li><a href="#lead-contact" id="toc-lead-contact" class="nav-link" data-scroll-target="#lead-contact"><span class="header-section-number">4.2.1</span> Lead contact</a></li>
  <li><a href="#materials-availability" id="toc-materials-availability" class="nav-link" data-scroll-target="#materials-availability"><span class="header-section-number">4.2.2</span> Materials availability</a></li>
  <li><a href="#data-and-code-availability" id="toc-data-and-code-availability" class="nav-link" data-scroll-target="#data-and-code-availability"><span class="header-section-number">4.2.3</span> Data and code availability</a></li>
  </ul></li>
  <li><a href="#experimental-model-and-study-participant-details" id="toc-experimental-model-and-study-participant-details" class="nav-link" data-scroll-target="#experimental-model-and-study-participant-details"><span class="header-section-number">4.3</span> Experimental model and study participant details</a>
  <ul class="collapse">
  <li><a href="#post-mortem-tissue-donors" id="toc-post-mortem-tissue-donors" class="nav-link" data-scroll-target="#post-mortem-tissue-donors"><span class="header-section-number">4.3.1</span> Post-mortem tissue donors</a></li>
  </ul></li>
  <li><a href="#method-details" id="toc-method-details" class="nav-link" data-scroll-target="#method-details"><span class="header-section-number">4.4</span> Method details</a>
  <ul class="collapse">
  <li><a href="#sample-prepneurovascular-unit-isolation" id="toc-sample-prepneurovascular-unit-isolation" class="nav-link" data-scroll-target="#sample-prepneurovascular-unit-isolation"><span class="header-section-number">4.4.1</span> Sample prep/neurovascular unit isolation</a></li>
  </ul></li>
  <li><a href="#endo-mt-facs-analysis" id="toc-endo-mt-facs-analysis" class="nav-link" data-scroll-target="#endo-mt-facs-analysis"><span class="header-section-number">4.5</span> Endo-MT FACS Analysis</a></li>
  <li><a href="#mouse-lineage-tracing" id="toc-mouse-lineage-tracing" class="nav-link" data-scroll-target="#mouse-lineage-tracing"><span class="header-section-number">4.6</span> Mouse lineage tracing</a>
  <ul class="collapse">
  <li><a href="#mouse-model" id="toc-mouse-model" class="nav-link" data-scroll-target="#mouse-model"><span class="header-section-number">4.6.1</span> Mouse model</a></li>
  <li><a href="#immunofluorescence-analysis-of-brain-tissue" id="toc-immunofluorescence-analysis-of-brain-tissue" class="nav-link" data-scroll-target="#immunofluorescence-analysis-of-brain-tissue"><span class="header-section-number">4.6.2</span> Immunofluorescence analysis of brain tissue</a></li>
  </ul></li>
  <li><a href="#quantification-and-statistical-analysis" id="toc-quantification-and-statistical-analysis" class="nav-link" data-scroll-target="#quantification-and-statistical-analysis"><span class="header-section-number">4.7</span> Quantification and statistical analysis</a>
  <ul class="collapse">
  <li><a href="#data-processing" id="toc-data-processing" class="nav-link" data-scroll-target="#data-processing"><span class="header-section-number">4.7.1</span> Data Processing</a></li>
  <li><a href="#sec-seq-qc" id="toc-sec-seq-qc" class="nav-link" data-scroll-target="#sec-seq-qc"><span class="header-section-number">4.7.2</span> Quality Control</a></li>
  <li><a href="#clustering" id="toc-clustering" class="nav-link" data-scroll-target="#clustering"><span class="header-section-number">4.7.3</span> Clustering</a></li>
  <li><a href="#cell-type-annotation" id="toc-cell-type-annotation" class="nav-link" data-scroll-target="#cell-type-annotation"><span class="header-section-number">4.7.4</span> Cell Type Annotation</a></li>
  <li><a href="#differences-in-cell-proportions" id="toc-differences-in-cell-proportions" class="nav-link" data-scroll-target="#differences-in-cell-proportions"><span class="header-section-number">4.7.5</span> Differences in cell proportions</a></li>
  <li><a href="#mast-differential-gene-expression" id="toc-mast-differential-gene-expression" class="nav-link" data-scroll-target="#mast-differential-gene-expression"><span class="header-section-number">4.7.6</span> MAST differential gene expression</a></li>
  <li><a href="#mammalian-phenotype-ontology-enrichment" id="toc-mammalian-phenotype-ontology-enrichment" class="nav-link" data-scroll-target="#mammalian-phenotype-ontology-enrichment"><span class="header-section-number">4.7.7</span> Mammalian Phenotype Ontology enrichment</a></li>
  <li><a href="#cell-cell-communication" id="toc-cell-cell-communication" class="nav-link" data-scroll-target="#cell-cell-communication"><span class="header-section-number">4.7.8</span> Cell-cell communication</a></li>
  <li><a href="#protein-protein-interaction-networks" id="toc-protein-protein-interaction-networks" class="nav-link" data-scroll-target="#protein-protein-interaction-networks"><span class="header-section-number">4.7.9</span> Protein-protein interaction networks</a></li>
  <li><a href="#sec-method-risk" id="toc-sec-method-risk" class="nav-link" data-scroll-target="#sec-method-risk"><span class="header-section-number">4.7.10</span> Disease risk enrichment</a></li>
  <li><a href="#pseudotime-analysis" id="toc-pseudotime-analysis" class="nav-link" data-scroll-target="#pseudotime-analysis"><span class="header-section-number">4.7.11</span> Pseudotime analysis</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#supplemental-information-titles-and-legends" id="toc-supplemental-information-titles-and-legends" class="nav-link" data-scroll-target="#supplemental-information-titles-and-legends"><span class="header-section-number">5</span> Supplemental information titles and legends</a>
  <ul class="collapse">
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">5.1</span> References</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <!-- - make upset plot of degs -->
<!-- - clean up cell labels in figures -->
<!-- - figure out what people mean by enrichment - maybe look at yang weight of sample and number nuclei per celltype - ask Hannah -->
<!-- - try reducing semantic similiarity of go terms in fig 2 B -->
<!-- - order cellchat plots manually on x axis - maybe separate the vascular and parenchymal celltypes and order each alphabetically -->
<!-- - “ ” ’‘-->
<div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># purl setup script</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">purl</span>(</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/01_r-packages-used.qmd"</span>),</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/01_r-packages-used.R"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>

processing file: /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/04_data_analysis/990_code_libraries/01_r-packages-used.qmd</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>1/6                   
2/6 [load libraries]  
3/6                   
4/6 [unnamed-chunk-15]
5/6                   
6/6 [unnamed-chunk-16]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>output file: /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/04_data_analysis/990_code_libraries/01_r-packages-used.R</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/04_data_analysis/990_code_libraries/01_r-packages-used.R"</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, smiths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'kableExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    group_rows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'janitor'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    chisq.test, fisher.test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    melt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rngtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  To enable shared memory capabilities, try: install.extras('
NMF
')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'NMF'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:generics':

    fit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'future'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    run</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'flextable'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    width</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:kableExtra':

    as_image, footnote</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'downlit'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:flextable':

    highlight</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:gt':

    as_gtable</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:flextable':

    border, font, rotate</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'viridis'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    viridis_pal</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
circlize version 0.4.16
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'circlize'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:flextable':

    fontsize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 methods overwritten by 'proxy':
  method               from    
  print.registry_field registry
  print.registry_entry registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    nrun</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'DESeq2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    fpkm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: limma</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DESeq2':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    plotMDS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'edgeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SingleCellExperiment':

    cpm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocParallel'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    register</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation("pathview") within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationFilter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationFilter'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:future':

    value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ensembldb'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggrepel</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'PCAtools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scran':

    parallelPCA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    biplot, screeplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Seurat':

    components</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    normalize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:circlize':

    degree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:flextable':

    compose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:future':

    %-&gt;%, %&lt;-%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:NMF':

    algorithm, compare</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:generics':

    components, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'biomaRt'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scRNAseq':

    listDatasets, searchDatasets</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/

Please cite:

G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,
5(6):100722</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'clusterProfiler'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>DOSE v4.2.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/

Please cite:

Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an
R/Bioconductor package for Disease Ontology Semantic and Enrichment
analysis. Bioinformatics. 2015, 31(4):608-609</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>enrichplot v1.28.2 Learn more at https://yulab-smu.top/contribution-knowledge-mining/

Please cite:

S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,
W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize
multiomics data. Nature Protocols. 2024, 19(11):3292-3320</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'enrichplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    color_palette</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pals':

    alphabet</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocIO</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rtracklayer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    blocks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationHub':

    hubUrl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh37'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'qusage'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    read.gmt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:SNPlocs.Hsapiens.dbSNP155.GRCh37':

    .loadAlleles, .loadLoc, COMPATIBLE_BSGENOMES</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'GenomicFiles'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
simona version 1.6.0
Bioconductor page: http://bioconductor.org/packages/simona/
Github page: https://github.com/jokergoo/simona
Documentation: https://jokergoo.github.io/simona/

If you use it in published research, please cite:
Gu, Z. simona: a Comprehensive R package for Semantic Similarity 
  Analysis on Bio-Ontologies. BMC Genomics, 2024.

This message can be suppressed by:
  suppressPackageStartupMessages(library(simona))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.24.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'ggnetwork':
  method         from  
  fortify.igraph ggtree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'CellChat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scater':

    runPCA, runUMAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'monocle3'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:presto':

    top_markers</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    clusters</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    exprs, fData, fData&lt;-, pData, pData&lt;-</code></pre>
</div>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flextable)</span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(YesSiR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggwordcloud</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: officer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'officer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readxl':

    read_xlsx</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: plotrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'plotrix'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    rescale</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    dispersion</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb224"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1" aria-hidden="true" tabindex="-1"></a>donor_metadata <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb224-2"><a href="#cb224-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/999_data_for_publication/donor_meta_data.csv"</span></span>
<span id="cb224-3"><a href="#cb224-3" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 37 Columns: 37
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (22): orig.ident, barcode, manifest, donor_ID, diagnosis, prep, sex, bat...
dbl (15): nCount_RNA, nFeature_RNA, individual, age, PMI, pH_cerebellum, Bra...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a>donor_summary <span class="ot">&lt;-</span> donor_metadata <span class="sc">|&gt;</span></span>
<span id="cb226-2"><a href="#cb226-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb226-3"><a href="#cb226-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">median_age =</span> <span class="fu">median</span>(age),</span>
<span id="cb226-4"><a href="#cb226-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">iqr_age =</span> <span class="fu">IQR</span>(age),</span>
<span id="cb226-5"><a href="#cb226-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">.by =</span> <span class="fu">c</span>(sex, diagnosis)</span>
<span id="cb226-6"><a href="#cb226-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb227-2"><a href="#cb227-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/999_data_for_publication/nuclei_meta_data.tsv"</span></span>
<span id="cb227-3"><a href="#cb227-3" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 396103 Columns: 40
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (25): orig.ident, barcode, manifest, donor_ID, diagnosis, prep, sex, bat...
dbl (15): nCount_RNA, nFeature_RNA, individual, age, PMI, pH_cerebellum, Bra...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb229"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb229-1"><a href="#cb229-1" aria-hidden="true" tabindex="-1"></a>level1_celltypes_n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(meta_data<span class="sc">$</span>highlevel_manual_annotations))</span>
<span id="cb229-2"><a href="#cb229-2" aria-hidden="true" tabindex="-1"></a>level2_celltypes_n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(meta_data<span class="sc">$</span>abbreviated_idents))</span>
<span id="cb229-3"><a href="#cb229-3" aria-hidden="true" tabindex="-1"></a>median_features_post_qc <span class="ot">&lt;-</span> <span class="fu">median</span>(meta_data<span class="sc">$</span>nFeature_RNA)</span>
<span id="cb229-4"><a href="#cb229-4" aria-hidden="true" tabindex="-1"></a>num_cells_post_qc <span class="ot">&lt;-</span> <span class="fu">nrow</span>(meta_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>Alzheimer’s disease (AD), the most prevalent type of dementia, is pathologically hallmarked by extracellular β-amyloid (Aβ) deposits, intracellular neurofibrillary tangles (NFTs) and neurodegeneration. The brain’s microvasculature is the interface between blood and brain and plays a crucial role in AD pathophysiology. It serves as a barrier (the blood-brain-barrier, BBB)<span class="citation" data-cites="cai2018 nehra2022"><sup><a href="#ref-cai2018" role="doc-biblioref">1</a>,<a href="#ref-nehra2022" role="doc-biblioref">2</a></sup></span> preventing harmful substances, such as toxins and pathogens, from entering the central nervous system. The BBB regulates the transport of essential nutrients and aids in the removal of waste products thereby maintaining the brain’s stable environment, essential for proper neural function. This is achieved through specialized vascular cells, including brain endothelial cells (ECs) and mural cells. The concept of the neurovascular unit (NVU) emphasizes the intimate relationship between brain parenchymal and vascular cells for example ensuring sufficient supply of oxygen and glucose to meet the metabolic demands of neurons and glia<span class="citation" data-cites="Iadecola2017"><sup><a href="#ref-Iadecola2017" role="doc-biblioref">3</a></sup></span>.</p>
<p>ECs contribute to the clearance of Aβ and other toxins, regulate the exclusion of harmful blood proteins, and facilitate immune cell trafficking<span class="citation" data-cites="Zhang2022 Daneman2015 Amersfoort2022"><sup><a href="#ref-Zhang2022" role="doc-biblioref">4</a>–<a href="#ref-Amersfoort2022" role="doc-biblioref">6</a></sup></span>. Both ECs and pericytes (PCs) are essential for maintaining brain perfusion, endothelial permeability, and immune activation<span class="citation" data-cites="Brown2019 Procter2021"><sup><a href="#ref-Brown2019" role="doc-biblioref">7</a>,<a href="#ref-Procter2021" role="doc-biblioref">8</a></sup></span>. Evidence from imaging, neuropathological studies, and preclinical models indicates chronic tissue hypoxia, impaired cerebral blood flow regulation, and BBB integrity loss in early AD stages, often linked to increased Aβ levels<span class="citation" data-cites="Nehra2022 Korte2020"><sup><a href="#ref-Nehra2022" role="doc-biblioref">9</a>,<a href="#ref-Korte2020" role="doc-biblioref">10</a></sup></span>. Knowledge of cell-type specific gene expression has transformed the interpretation of AD genetic risk and our understanding of AD molecular pathology<span class="citation" data-cites="mcquade2019 yang2023"><sup><a href="#ref-mcquade2019" role="doc-biblioref">11</a>,<a href="#ref-yang2023" role="doc-biblioref">12</a></sup></span>. However, despite similar numbers of glia and vascular cells within the brain<span class="citation" data-cites="Keller2018"><sup><a href="#ref-Keller2018" role="doc-biblioref">13</a></sup></span>, the processes of extracting nuclei from human post-mortem tissue have favoured retrieval of parenchymal nuclei over vascular cells, resulting in an incomplete picture of the NVU. The first human brain single nuclei atlases examining AD at scale obtained less than a few hundred endothelial cell nuclei as compared to tens of thousands from parenchymal cells, which precluded analyses of the NVU<span class="citation" data-cites="Mathys2019 Grubman2019"><sup><a href="#ref-Mathys2019" role="doc-biblioref">14</a>,<a href="#ref-Grubman2019" role="doc-biblioref">15</a></sup></span>. However, the development of extraction approaches focused on vascular cells has recently enabled the study of these cell types. Enrichment approaches have focused on specifically isolating the neurovasculature from brain samples via mechanical approaches yielding ~50% enrichment of vascular nuclei<span class="citation" data-cites="yang2022 Tsartsalis2024"><sup><a href="#ref-yang2022" role="doc-biblioref">16</a>,<a href="#ref-Tsartsalis2024" role="doc-biblioref">17</a></sup></span>, revealing AD-associated impaired angiogenesis and inflammation. Furthermore, both studies highlight potential roles for cerebrovascular cell types in mediating AD genetic risk.</p>
<p>Given the paucity of approaches to efficiently extract brain vascular cells whilst preserving retrieval of parenchymal cells, we developed a new method for simultaneous vascular and parenchymal cell type extraction. We applied this to prefrontal cortex (PFC) from 20 Alzheimer’s and 20 control donors and performed single nuclei RNA sequencing. This revealed distinct enrichments of Alzheimer’s genetic risk in pericytes, perivascular fibroblasts and activated microglia – representing the first robust association of AD genetic risk with brain vascular cells types. Furthermore, we find that the genes underlying these associations are centred on amyloid biology in all three cell types, suggesting the involvement of these specific cell types in early AD pathobiology. We also find gene pathways involved in recruitment or activation of T cells, astrocytes and microglia. Our isolation approach also revealed a transitional cell population simultaneously expressing markers of endothelial and mural cells whose existence we validate in the mouse. We find this population is significantly reduced in AD indicating changes in vascular plasticity in the AD brain. Our method provides a means to fully capture vascular and parenchymal cell types in a given brain region and the single nuclei gene expression dataset is a valuable resource to understand brain changes in AD.</p>
</section>
<section id="results" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Results</h1>
<section id="isolating-the-neurovascular-unit" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="isolating-the-neurovascular-unit"><span class="header-section-number">2.1</span> Isolating the neurovascular unit</h2>
<p>Single-cell and single-nuclei RNA sequencing (snRNAseq) has yielded significant insights into health and disease, particularly brain disorders. Traditionally, the brain vasculature has been highly challenging to isolate but recent studies have begun to examine neurovascular biology through vascular enrichment<span class="citation" data-cites="yang2022 tsartsalis2024"><sup><a href="#ref-yang2022" role="doc-biblioref">16</a>,<a href="#ref-tsartsalis2024" role="doc-biblioref">18</a></sup></span>. Whilst a significant advance, the reported vascular nuclei isolation protocols remain relatively inefficient and achieve this enrichment at the expense of parenchymal cells, thereby losing the opportunity to examine the complexity of whole tissue brain biology.<br>
We developed a novel method for the simultaneous isolation of highly pure microvascular nuclei with high-quality parenchymal nuclei from human post-mortem brain issue in a single procedure. We considered critical issues to be addressed included poor removal of meningeal tissue, poor control of dissection and loss of valuable tissue mass when traditional 4⁰C dissection of brain tissue was performed using current protocols. We therefore first developed a simple holding unit (<a href="#fig-method" class="quarto-xref">Figure&nbsp;1</a> A) to enable dissection on dry ice to maintain tissue temperature at approximately -80C and allowing careful removal of meninges and other contaminants. We further created a 3D printed tissue chopper (<a href="#fig-method" class="quarto-xref">Figure&nbsp;1</a> A) to process unthawed brain tissue blocks into a coarse powder optimal for nuclei isolation, which can then be stored as required.</p>
<p>Brain vessels are typically lost in standard isolation protocols through under-or over-homogenisation. To mitigate this, we designed a 3D printed pestle to allow multiple rounds of large-volume pestle-driven homogenisation combined with low speed centrifugation. This enabled efficient release of vessels whilst minimising vessel damage (<a href="#fig-method" class="quarto-xref">Figure&nbsp;1</a> B), yielding vascular pellets and a vessel-free supernatant for parenchymal isolation – which is then processed using standard protocols. Purification of the vascular fraction involves two multi-layer dextran gradient centrifugation steps to trap contaminants and filtering to remove unwanted larger vessels. The basement membrane of microvessels is enzymatically digested to release endothelial nuclei, supported by a 3D printed assembly to grind nuclei out of vessels (<a href="#fig-method" class="quarto-xref">Figure&nbsp;1</a> C).</p>
<div id="fig-method" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-method-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure1.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Simultaneous isolation of vascular and parenchymal fractions from post-mortem human brain tissue. (A) A 3D printed holding unit and tissue chopper allows dissection of brain tissue whilst keeping the temperature at -80C to avoid thawing. (B) Isolation of microvessels from the brain parenchymal fraction involves repeated homogenization using a 3D printed pestle; followed by alternating centrifugation and dounce homogenization. (C) To release vascular nuclei from isolated microvessels, samples are subjected to multi-layer dextran centrifugation, removal of large vessels and debris using filters; Collagenase II enzymatic digestion of the basement membrane followed by extrusion of vascular nuclei using a 3D printed funnel and pestle."><embed src="05_figures/990_shared_figures/003_final_figures/figure1.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-method-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Simultaneous isolation of vascular and parenchymal fractions from post-mortem human brain tissue. (A) A 3D printed holding unit and tissue chopper allows dissection of brain tissue whilst keeping the temperature at -80C to avoid thawing. (B) Isolation of microvessels from the brain parenchymal fraction involves repeated homogenization using a 3D printed pestle; followed by alternating centrifugation and dounce homogenization. (C) To release vascular nuclei from isolated microvessels, samples are subjected to multi-layer dextran centrifugation, removal of large vessels and debris using filters; Collagenase II enzymatic digestion of the basement membrane followed by extrusion of vascular nuclei using a 3D printed funnel and pestle.
</figcaption>
</figure>
</div>
</section>
<section id="a-single-cell-atlas-of-the-neurovascular-unit" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="a-single-cell-atlas-of-the-neurovascular-unit"><span class="header-section-number">2.2</span> A single cell atlas of the neurovascular unit</h2>
<p>We used our new method to examine the role of the neurovascular unit in Alzheimer’s Disease (AD). We performed single nuclei RNA sequencing on prefrontal cortex (PFC) samples from 40 individuals, divided into 20 AD cases and 20 controls (<a href="#fig-umap" class="quarto-xref">Figure&nbsp;2</a> A). For each individual, we analysed two fractions: vascular and parenchymal, both obtained from the same PFC tissue. This resulted in a total of 80 nuclei fractions (40 vascular and 40 parenchymal). These samples were captured on 10X genomics chips, after which libraries were prepared and subsequently subject to sequencing on an Illumina platform. We obtained 474,357 nuclei which was subset to 396103 nuclei after quality control (QC) and three donors were also removed (See <a href="#sec-seq-qc" class="quarto-xref">Section&nbsp;4.7.2</a> for QC details). The median number of genes whose expression was detected per nucleus was 2353 post-QC. Nuclei were annotated into 12 main cell types which were further subdivided into 45 subtypes based on published markers (<a href="#fig-umap" class="quarto-xref">Figure&nbsp;2</a> &amp; <a href="#suppfig-level1-markers" class="quarto-xref">Figure S1</a>)<span class="citation" data-cites="yang2022 mathys2023"><sup><a href="#ref-yang2022" role="doc-biblioref">16</a>,<a href="#ref-mathys2023" role="doc-biblioref">19</a></sup></span>.</p>
<div id="fig-umap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure1_umaps_c_label.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: Single nuclei atlas of the human prefrontal cortex neuro-vasculature. (A) Flow diagram of the cohort details with the median age ± IQR for males and females in cases and controls respectively. (B) A post-QC UMAP colored by fraction isolated showing the vascular nuclei are on the right. (C) The same UMAP colored by higher level cell types. (D) The same UMAP colored by cell subtypes. (E) The percentage of isolated nuclei that are vascular/parenchymal cell types from various vascular enrichment methods contrasted to the current method. (F) The percentage of isolated nuclei that are vascular/parenchymal in a typical sample preparation contrasted to our parenchymal fraction."><embed src="05_figures/990_shared_figures/003_final_figures/figure1_umaps_c_label.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Single nuclei atlas of the human prefrontal cortex neuro-vasculature. (A) Flow diagram of the cohort details with the median age ± IQR for males and females in cases and controls respectively. (B) A post-QC UMAP colored by fraction isolated showing the vascular nuclei are on the right. (C) The same UMAP colored by higher level cell types. (D) The same UMAP colored by cell subtypes. (E) The percentage of isolated nuclei that are vascular/parenchymal cell types from various vascular enrichment methods contrasted to the current method. (F) The percentage of isolated nuclei that are vascular/parenchymal in a typical sample preparation contrasted to our parenchymal fraction.
</figcaption>
</figure>
</div>
<p>Our protocol provides excellent coverage of all four major vascular cell types (endothelial, pericyte, smooth muscle cell (SMC), fibroblasts) and their neighboring parenchymal cells (astrocytes, microglia, excitatory neurons, inhibitory neurons, oligodendrocytes, oligodendrocyte precursor cells (OPC)). As expected, cell types were detected in their appropriate vascular or the parenchymal fractions, with for example &gt;95% of endothelial cells coming from the vascular fraction and &gt;95% of neurons from the parenchymal fraction. Certain cell types were present in both fractions, including microglia and T cells. Thus, the isolation method simultaneously captures both parenchymal and vascular cell types that are present in the same prefrontal cortex tissue enabling a detailed examination of the NVU in AD. The two most abundant nuclei were from oligodendrocytes captured in the parenchymal fraction and endothelial cells captured in the vascular fraction. Oligodendrocytes represented ~28% of the total nuclei and endothelial were ~20%. We contrasted the number of nuclei obtained from the vascular fractions of this study to the recent Yang et al.&nbsp;and Tsartsalis et al.&nbsp;papers via a chi-squared test and find significantly more nuclei for all cell types present in all studies, excluding fibroblasts contrasted with Yang et al.&nbsp;which are on par (Bonferroni adjusted P &lt; 0.05, <a href="#fig-umap" class="quarto-xref">Figure&nbsp;2</a> E). Conversely, we obtained over ten times the mean number of neurons per sample<span class="citation" data-cites="yang2022"><sup><a href="#ref-yang2022" role="doc-biblioref">16</a></sup></span>.</p>
</section>
<section id="activated-microglia-pericytes-perivascular-fibroblasts-and-t-cells-are-enriched-for-ad-risk-genes-focused-on-amyloid-processing-and-immune-cell-activation" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="activated-microglia-pericytes-perivascular-fibroblasts-and-t-cells-are-enriched-for-ad-risk-genes-focused-on-amyloid-processing-and-immune-cell-activation"><span class="header-section-number">2.3</span> Activated microglia, pericytes, perivascular fibroblasts and T cells are enriched for AD risk genes focused on amyloid processing and immune cell activation</h2>
<p>Having a more complete picture of the parenchymal and vascular fractions of the PFC enables a more accurate definition of gene cell-type-specificity, through which we sought to more definitively understand which cell types were associated with the genetic risk of AD. We employed Multi-marker Analysis of GenoMic Annotation (MAGMA) along with the latest Genome-Wide Association Study (GWAS) for AD<span class="citation" data-cites="bellenguez2022"><sup><a href="#ref-bellenguez2022" role="doc-biblioref">20</a></sup></span> testing the top 10% of genes most specifically expressed within each cell type amongst nuclei from control individuals (see <a href="#sec-method-risk" class="quarto-xref">Section&nbsp;4.7.10</a>). Amongst the 12 higher level cell types, our analysis identified microglia, fibroblasts, pericytes, and T-cells associated with AD genetic risk (<a href="#suppfig-risk-level1" class="quarto-xref">Figure S2</a>). While microglia have been implicated previously with AD genetic risk, the associations with pericytes and perivascular fibroblasts are novel. This is the first time brain vascular cell type specific expression profiles have been shown to be enriched with AD genetic risk when all major cell types of the neurovascular unit are considered.</p>
<p>Further analysis at the subtype level, implicated, in particular, activated microglia (Microglia A), pericyte-2 and perivascular fibroblast 2 (FB-2) with AD genetic risk (<a href="#fig-risk" class="quarto-xref">Figure&nbsp;3</a> A). Conditional analysis showed significant enrichment for each of the three cell types even when conditioned on the other two cell types (<a href="#fig-risk" class="quarto-xref">Figure&nbsp;3</a> B), indicating that the observed signals are uniquely associated with a cell type and not confounded through shared gene expression across cell types (<a href="#fig-risk" class="quarto-xref">Figure&nbsp;3</a> C). To further understand the biological implications of the AD genetic risk genetic associations, we performed Gene Ontology (GO) enrichment analysis on those genes identified by MAGMA as contributing to that cell subtype’s AD risk association. The GO analysis revealed several enriched pathways, prominently featuring several amyloid-related processes. The enrichment of genes involved in amyloid precursor protein (APP) processing, amyloid-beta (Aβ) formation, and amyloid plaque clearance was particularly notable (<a href="#fig-risk" class="quarto-xref">Figure&nbsp;3</a> D). This finding aligns with the well-established role of amyloid pathology in AD. Additionally, for activated microglia and pericyte-2 cell types, significant GO terms related to T-cell and lymphocyte regulation were identified, suggesting an immune regulatory component in AD risk, especially given the significant risk association of T-cells in the higher level cell type analysis.</p>
<div id="fig-risk" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-risk-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure3_risk.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: Alzheimer’s disease genetic risk enrichment. (A) A heatmap denoting cell type-specific risk enrichment via MAGMA with “*” denoting nominal significance (P < 0.05), “**” significance after Bonferroni adjustment (P < 0.05) and “***” a higher degree of adjusted significance again (P < 0.01). Microglia-A, Pericyte-2 and perivascular-FB-2 are significantly enriched for AD risk via MAGMA after multiple testing adjustment. (B) Cell type-specific risk for the three significant cell types conditioned on each other to determine signal independence. All remain significant even when conditioned on one another. (C) Average expression of the MAGMA significant genes in the relevant higher level cell types. Each cell type has a clear distinct expression profile for the risk genes. (D) Significantly enriched disease terms in mouse ontology. (E) Dotplot of significantly enriched GO pathways from the significant genes derived from MAGMA in significant cell types. Pathways have been reduced on semantic similarity via rrvgo (threshold = 0.9). (F/G/H) Protein-protein interaction networks for the same significant genes from MAGMA. Single genes unconnected to the main network are excluded."><embed src="05_figures/990_shared_figures/003_final_figures/figure3_risk.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-risk-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Alzheimer’s disease genetic risk enrichment. (A) A heatmap denoting cell type-specific risk enrichment via MAGMA with “*” denoting nominal significance (P &lt; 0.05), “**” significance after Bonferroni adjustment (P &lt; 0.05) and “***” a higher degree of adjusted significance again (P &lt; 0.01). Microglia-A, Pericyte-2 and perivascular-FB-2 are significantly enriched for AD risk via MAGMA after multiple testing adjustment. (B) Cell type-specific risk for the three significant cell types conditioned on each other to determine signal independence. All remain significant even when conditioned on one another. (C) Average expression of the MAGMA significant genes in the relevant higher level cell types. Each cell type has a clear distinct expression profile for the risk genes. (D) Significantly enriched disease terms in mouse ontology. (E) Dotplot of significantly enriched GO pathways from the significant genes derived from MAGMA in significant cell types. Pathways have been reduced on semantic similarity via rrvgo (threshold = 0.9). (F/G/H) Protein-protein interaction networks for the same significant genes from MAGMA. Single genes unconnected to the main network are excluded.
</figcaption>
</figure>
</div>
<p>We found significantly more protein-protein interaction (PPI) amongst the MAGMA genes identified in each cell type than expected by chance demonstrating convergent cell type specific gene AD risk networks. Subsequent Louvain clustering of these PPI networks revealed distinct modules associated with amyloid-related processes (via further GO enrichment analysis) across all three cell types (<a href="#fig-risk" class="quarto-xref">Figure&nbsp;3</a> F, G, H). This suggests a convergent risk association involving amyloid processing among activated microglia, pericytes, and perivascular fibroblasts.</p>
<p>Finally, we used the Mouse Phenotype Ontology (MPO) to investigate the phenotypes resulting from the knock-out of these genes’ 1:1 orthologs in the mouse. For the activated microglia risk genes, significant associations were found with the abnormal cell chemotaxis and hematoma MPO terms while pericyte-2 risk genes showed significant associations with amyloidosis and tau protein deposits (<a href="#fig-risk" class="quarto-xref">Figure&nbsp;3</a> E). However, no significant phenotypic associations were identified amongst the mouse orthologues of the perivascular fibroblast risk genes.</p>
</section>
<section id="a-population-of-endothelial-mesenchymal-transitional-cells-are-detected-in-human-prefrontal-cortex-whose-transition-appears-disrupted-in-alzheimers-disease" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="a-population-of-endothelial-mesenchymal-transitional-cells-are-detected-in-human-prefrontal-cortex-whose-transition-appears-disrupted-in-alzheimers-disease"><span class="header-section-number">2.4</span> A population of Endothelial Mesenchymal Transitional Cells are detected in human prefrontal cortex whose transition appears disrupted in Alzheimer’s Disease</h2>
<p>We identified a substantial vascular cell population (16021 nuclei, 10% of vascular fraction), expressing both endothelial and mural cell identity markers (<a href="#fig-endomt-prop" class="quarto-xref">Figure&nbsp;4</a> B) which has not beeen in observed in other single cell datasets of cortical brain tissue. We hypothesized that this cell population, herein termed EndoMT, may be undergoing endothelial mesenchymal transition and, ultimately, differentiating into mural cells. A recent single cell RNA-seq study of brain cavernomas and arteriovenous malformations<span class="citation" data-cites="bravi2016 shoemaker2020 wälchli2024"><sup><a href="#ref-bravi2016" role="doc-biblioref">21</a>–<a href="#ref-wälchli2024" role="doc-biblioref">23</a></sup></span> also reported a brain tissue EndoMT population. However, we did not identify a significant overlap in their markers with our population (<a href="#suppfig-endomt-paper-markers" class="quarto-xref">Figure S3</a>). Furthermore, subsequent differentiation into brain mural cells has not been previously described suggesting our brain vascular isolation method has revealed a novel brain cell-type or cell-state. Importantly, we find EndoMT in control donor brain indicating that the observed endothelial plasticity represents a normal physiological state.</p>
<p>We first compared cell-type proportions between AD and controls in vascular and parenchymal fractions independently, utilising the scProportionTest package<span class="citation" data-cites="Miller2021"><sup><a href="#ref-Miller2021" role="doc-biblioref">24</a></sup></span>. For the vascular fraction, this identified a significant ~50% reduction in the proportion of endothelial-to-mesenchymal transition (EndoMT) cells in AD, with both EndoMT1 and EndoMT2 subtypes being reduced (<a href="#fig-endomt-prop" class="quarto-xref">Figure&nbsp;4</a> A). We also observed in AD reduced percitye-1 subtypes whereas SMC subtypes were increased (both vascular-SMC-1 and –2) and the most increased cell type was meningeal fibroblasts (<a href="#fig-endomt-prop" class="quarto-xref">Figure&nbsp;4</a> A). Relatedly, when correlating celltype proportions across donors, we find a strong negative correlation between endothelial and pericyte/SMC populations (<a href="#suppfig-vascular-population-correlations" class="quarto-xref">Figure S4</a>). With respect to the parenchymal fraction, as would be expected in a neurodegenerative disorder, there was a significant reduction in excitatory neuron subtypes (Ex-Neuro-1 ~75% reduction; Ex-Neuro-5 ~50% reduction) (<a href="#suppfig-parenchymal-prop-diff" class="quarto-xref">Figure S5</a>).</p>
<p>We sought to further validate the EndoMT population and to assess changes in AD by using fluorescence-activated cell sorting (FACS) to detect cells co-expressing an endothelial cell specific and a mural cell specific nuclear proteins. We identified ERG as selectively expressed by EC but not mural cells and Notch3 which is selectively expressed by mural cells but not EC (<a href="#fig-endomt-prop" class="quarto-xref">Figure&nbsp;4</a> B). Anti-ERG and Anti-Notch3 are widely used as identity markers for endothelial cells and mural cells respectively<span class="citation" data-cites="shah2016 gastfriend2024"><sup><a href="#ref-shah2016" role="doc-biblioref">25</a>,<a href="#ref-gastfriend2024" role="doc-biblioref">26</a></sup></span>. Using FACS in 10 independent samples (5 AD and 5 controls), we found 15% of control and 10% of AD prefrontal cortex co-expressed ERG and NOTCH3, confirming a significant reduction of EndoMT cells in AD (<a href="#fig-endomt-prop" class="quarto-xref">Figure&nbsp;4</a> C).</p>
<p>We complimented this validation by performing lineage-tracing using the endothelial-specific Cre mouse line CDH5-Cre coupled to the R26RtdTOMATO (R26RTOM) reporter line to test our hypothesis that EndoMT are an intermediate state between EC and mural cells. Lineage tracing and staining of endothelial cells with CD31 and pericytes with PDGFRB, both markers used for cell clustering, thereby confirmed that a small proportion of pericytes were arising from endothelial cells (<a href="#fig-endomt-prop" class="quarto-xref">Figure&nbsp;4</a> D/E).</p>
<div id="fig-endomt-prop" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-endomt-prop-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure4_endomt_prop.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: proportional differences in Endo-MT. (A) Cell subtype proportional differences from scProportionTest. Signficant cell types are those with an FDR < 0.05 and an absolute LFC > 0.58. (B) A dotplot of endothelial and SMC markers, showing Endo-MT has a unique expression of both ERG and NOTCH3. (C) A boxplot showing the proportions of Endo-MT in independent AD and control samples via FACS. There is a significant (P = 0.02) difference via Wilcoxon rank-sum test. (D/E) Genetically labelled endothelial cells and pericytes in Cdh5TOM mice (red) express pericyte markers (PDGFRb, green) and endothelial markers (CD31, white). (D) Endothelial derived pericyte in the parenchyma. (E) Endothelial derived pericytes in the meninges. Insets (i) show higher magnification."><embed src="05_figures/990_shared_figures/003_final_figures/figure4_endomt_prop.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-endomt-prop-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: proportional differences in Endo-MT. (A) Cell subtype proportional differences from scProportionTest. Signficant cell types are those with an FDR &lt; 0.05 and an absolute LFC &gt; 0.58. (B) A dotplot of endothelial and SMC markers, showing Endo-MT has a unique expression of both ERG and NOTCH3. (C) A boxplot showing the proportions of Endo-MT in independent AD and control samples via FACS. There is a significant (P = 0.02) difference via Wilcoxon rank-sum test. (D/E) Genetically labelled endothelial cells and pericytes in Cdh5TOM mice (red) express pericyte markers (PDGFRb, green) and endothelial markers (CD31, white). (D) Endothelial derived pericyte in the parenchyma. (E) Endothelial derived pericytes in the meninges. Insets (i) show higher magnification.
</figcaption>
</figure>
</div>
<p>To understand the molecular basis of these transitions, we performed pseudotime analysis using Slingshot<span class="citation" data-cites="Street2018"><sup><a href="#ref-Street2018" role="doc-biblioref">27</a></sup></span>. This uncovered two primary lineages of EndoMT cells, each characterized by unique transitional subtypes (<a href="#fig-endomt" class="quarto-xref">Figure&nbsp;5</a> A). Lineage 1 encompasses EC that progress through EndoMT-2 cells toward smooth muscle cells (SMCs), herein named lineage EndoMT-SMC, while Lineage 2 are EC transition to EndoMT-1 cells and then differentiating into pericytes, herein named EndoMT-PC (<a href="#fig-endomt" class="quarto-xref">Figure&nbsp;5</a> B). The distribution across the trajectory was even across controls and AD, and between males and females (<a href="#suppfig-endomt-pca" class="quarto-xref">Figure S6</a>). This confirms the population captured is a novel and ubiquitous phenomenon. We next used tradeSeq to assess changes in gene expression through the trajectories in each lineage<span class="citation" data-cites="Van2020"><sup><a href="#ref-Van2020" role="doc-biblioref">28</a></sup></span>. This revealed 8 transcription factors that diverge in early pseudo-time and thus may be involved in the fate choices. For example, we found FOXP2 and RORA to be increased with pseudotime in lineage EndoMT-PC but not EndoMT-SMC (<a href="#suppfig-endomt-tfs" class="quarto-xref">Figure S7</a> A-B). Conversely ZBTB7C, PRDM16 and NR4A1 are increased in lineage EndoMT-SMC (<a href="#suppfig-endomt-tfs" class="quarto-xref">Figure S7</a> C-E).</p>
<div id="fig-endomt" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-endomt-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/endomt_fig2.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;5: Characterization of the endothelial-mesenchymal (EndoMT) cell type. (A) PCA plot of nuclei subset to the endothelial, SMC, pericyte and EndoMT cell types. Includes the slingshot trajectories of the two EndoMT lineages, the first of which terminally differentiates to SMC and the second to pericytes. (B) Density plots of nuclei over pseudotime for each lineage. Early pseudotime is made up of endothelial cells, which then move into EndoMT-1/2 and finally to pericytes/SMCs. (C) Heatmap of transcription factor network median AUCs denoting how likely the regulons are active across cell types via Scenic. The MEMCON regulon shows a graduated decline from arterial through capillary to EndoMT. The FOXP2 regulon shows a distinct activity in the pericyte populations and EndoMT-1 (D) Log expression plots of the transcription factors from C for each condition and lineage respectively. FOXP2 and LEF1 show increased expression in the EndoMT-PC lineage in later pseudotime. ARIDB5 by contrast is increased in EndoMT-SMC"><embed src="05_figures/990_shared_figures/003_final_figures/endomt_fig2.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-endomt-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Characterization of the endothelial-mesenchymal (EndoMT) cell type. (A) PCA plot of nuclei subset to the endothelial, SMC, pericyte and EndoMT cell types. Includes the slingshot trajectories of the two EndoMT lineages, the first of which terminally differentiates to SMC and the second to pericytes. (B) Density plots of nuclei over pseudotime for each lineage. Early pseudotime is made up of endothelial cells, which then move into EndoMT-1/2 and finally to pericytes/SMCs. (C) Heatmap of transcription factor network median AUCs denoting how likely the regulons are active across cell types via Scenic. The MEMCON regulon shows a graduated decline from arterial through capillary to EndoMT. The FOXP2 regulon shows a distinct activity in the pericyte populations and EndoMT-1 (D) Log expression plots of the transcription factors from C for each condition and lineage respectively. FOXP2 and LEF1 show increased expression in the EndoMT-PC lineage in later pseudotime. ARIDB5 by contrast is increased in EndoMT-SMC
</figcaption>
</figure>
</div>
<p>We next used tradeSeq to identify altered AD and control donor gene expression across EndoMT-SMC and EndoMT-PC trajectories, focusing only the most significant genes (p &lt; 1 x 10-16)<span class="citation" data-cites="vandenberge2020"><sup><a href="#ref-vandenberge2020" role="doc-biblioref">29</a></sup></span>. This showed 410 unique DEGs, including genes altered early, late or throughout the trajectory (<a href="#supptbl-tradeseq-degs-early" class="quarto-xref">Table S2</a>, <a href="#supptbl-tradeseq-degs-late" class="quarto-xref">Table S3</a>, <a href="#supptbl-tradeseq-degs-lin" class="quarto-xref">Table S4</a> &amp; <a href="#supptbl-tradeseq-degs-cond" class="quarto-xref">Table S5</a>). Notable examples include BACE2, a beta-secretase enzyme implicated in AD, is progressively downregulated in both lineages but more highly expressed in AD compared to controls in early pseudo-time (<a href="#suppfig-adriskpseudotime" class="quarto-xref">Figure S9</a> A). APOD, involved in lipid transport and neuroprotection, shows a substantial increase at the late stage in both trajectories, but is significantly reduced in AD in EndoMT-PC (<a href="#suppfig-adriskpseudotime" class="quarto-xref">Figure S9</a> B). TGF-β1, involved in regulating phagocytic clearance of Aβ by microglia, is also increased in cases for both lineages. 138 of the 410 most significantly differentially expressed genes (including BACE2 and APOD) identified by tradeseq were overlapping with the significant DEGs identified by Model-based Analysis of Single Cell Transcriptomics (MAST) in EndoMT1 and EndoMT2<span class="citation" data-cites="finak2015"><sup><a href="#ref-finak2015" role="doc-biblioref">30</a></sup></span>. Overall, we find our data suggests that AD affects the EndoMT process, with the genes identified potentially representing key molecular drivers of vascular transformation in AD<span class="citation" data-cites="bhatia2018 yeap2022"><sup><a href="#ref-bhatia2018" role="doc-biblioref">31</a>,<a href="#ref-yeap2022" role="doc-biblioref">32</a></sup></span>.</p>
<p>To understand whether endothelial plasticity may be affected in Alzheimer’s Disease (AD) pathophysiology we performed differential gene expression between controls and AD cases focusing on vascular cell types. Using MAST, we found 610 significant differentially expressed genes (DEGs) in the endomt-1 population (pericyte lineage) but only 8 DEGs in the endomt-2 population (SMC lineage) (<a href="#fig-degs" class="quarto-xref">Figure&nbsp;6</a> A/B &amp; <a href="#supptbl-mast-degs" class="quarto-xref">Table S6</a>). Endomt-1 DEGs were enriched for the GO terms negative regulation of endothelial cell migration (GO:0010596), protein folding (GO:0006457) and negative regulation of natural killer cell mediated cytotoxicity/immunity (GO:0045953/0002716) while there were no significantly enriched pathways for endomt-2 biological processes (<a href="#supptbl-mast-go-pathways" class="quarto-xref">Table S7</a> &amp; <a href="#supptbl-mast-go-pathways-filter" class="quarto-xref">Table S8</a>).</p>
<div id="fig-degs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-degs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure_degs.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;6: Differential gene expression via MAST. (A/B) Heatmaps denoting the number of significant DEGs, with an absolute LFC > 1 and with no LFC cut-off respectively, for each cell subtype. (C/D) Plots denoting which revigo summarised GO pathways are significantly enriched for each celltype, with an absolute LFC > 1 and with no LFC cut-off respectively, separated in pathways up- or down-regulated."><embed src="05_figures/990_shared_figures/003_final_figures/figure_degs.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-degs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Differential gene expression via MAST. (A/B) Heatmaps denoting the number of significant DEGs, with an absolute LFC &gt; 1 and with no LFC cut-off respectively, for each cell subtype. (C/D) Plots denoting which revigo summarised GO pathways are significantly enriched for each celltype, with an absolute LFC &gt; 1 and with no LFC cut-off respectively, separated in pathways up- or down-regulated.
</figcaption>
</figure>
</div>
</section>
<section id="blood-vessel-stabilisation-signalling-is-altered-in-alzheimers-disease" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="blood-vessel-stabilisation-signalling-is-altered-in-alzheimers-disease"><span class="header-section-number">2.5</span> Blood vessel stabilisation signalling is altered in Alzheimer’s Disease</h2>
<p>We next examined cell-cell interactions relevant for EndoMT by intersecting tradeSeq differential trajectory genes with significantly altered ligand-receptor (L-R pairs) from CellChat<span class="citation" data-cites="Jin2021"><sup><a href="#ref-Jin2021" role="doc-biblioref">33</a></sup></span>. This identified both ANGPT2-TEK and ANGPT1-TEK ligands that destabilize and stabilize blood vessels respectively, as significantly altered ligand-receptor (LR) pairs (<a href="#suppfig-cellchat" class="quarto-xref">Figure S8</a> B &amp; <a href="#fig-angpt2" class="quarto-xref">Figure&nbsp;7</a> D/E). In AD only, EndoMT1 and EndoMT2 are receivers of the vascular destablising ligand ANGPT2 from OPCs through TEK/TIE2 and ITGA5/B5. The role of ANGPT2 in AD is further supported by the tradeSeq with ANGPT2 significantly upregulated throughout both lineage trajectories and using MAST, significantly upregulated in several cell types, including endomt-1 and endothelial subtypes (<a href="#fig-angpt2" class="quarto-xref">Figure&nbsp;7</a> A/B/C). GATA-1 and ETS-1 have been reported as key regulators of ANGPT2, and both are also significantly upregulated in capillary EC nuclei.</p>
<p>Conversely, we find the stabilizing ligand ANGPT1 has the same profile of source-receiver celltypes as ANGPT2, but this is now present only in controls. For ANGPT1, both EndoMT1 and EndoMT2 receive from astrocytes and inhibitory neurons in AD only, whereas controls only receive from excitatory neurons. ANGPT1 is also significantly down regulated in astrocytes in AD via MAST. These results support significant alterations in endothelial plasticity and signalling in AD, with altered signalling dynamics through key genes including ANGPT2 and ANGPT1. The opposite changes we found for ANGPT2 and ANGPT1 is consistent with their opposing functional effects on vessel stability<span class="citation" data-cites="wang2017"><sup><a href="#ref-wang2017" role="doc-biblioref">34</a></sup></span>.</p>
<div id="fig-angpt2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-angpt2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/angpt2.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;7: ANGPT2 alterations in AD. (A) Log expression of ANGPT2 over pseudotime via tradeSeq. (B) Expression of ANGPT2 in cell types with significant differential expression. (C) Average log fold changes in AD for ANGPT2 in significant cell types (adjusted pvalue < 0.05 &amp; {.absolute log fold change > 1), coloured by -log10 adjusted pvalue via MAST. (D/E) Significant (pvalue < 0.01 &amp; log fold change > 0.5) ANGPT2 ligand-receptor pairs from the vascular to parenchymal celltypes (D), and the parenchymal to vascular celltype (E)."><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/angpt2.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-angpt2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: ANGPT2 alterations in AD. (A) Log expression of ANGPT2 over pseudotime via tradeSeq. (B) Expression of ANGPT2 in cell types with significant differential expression. (C) Average log fold changes in AD for ANGPT2 in significant cell types (adjusted pvalue &lt; 0.05 &amp; {.absolute log fold change &gt; 1), coloured by -log10 adjusted pvalue via MAST. (D/E) Significant (pvalue &lt; 0.01 &amp; log fold change &gt; 0.5) ANGPT2 ligand-receptor pairs from the vascular to parenchymal celltypes (D), and the parenchymal to vascular celltype (E).
</figcaption>
</figure>
</div>
</section>
<section id="cell-cell-interactions-between-vascular-and-parenchymal-cells-are-through-specific-signalling-pathways" class="level2" data-number="2.6">
<h2 data-number="2.6" class="anchored" data-anchor-id="cell-cell-interactions-between-vascular-and-parenchymal-cells-are-through-specific-signalling-pathways"><span class="header-section-number">2.6</span> Cell-cell interactions between vascular and parenchymal cells are through specific signalling pathways</h2>
<p>With respect to cell-cell interactions beyond EndoMT, we observed less interaction between neuronal subtypes in AD, both in number and strength (<a href="#suppfig-cellchat" class="quarto-xref">Figure S8</a> A). Excitatory neurons strongly communicated with other neuronal subtypes in terms of the number of interactions and the strength of interactions in controls, and this was reduced in AD. There is also a decrease in the number of neuronal to vascular interactions in AD although the strength of interactions is not as clearly changed. More strikingly, there is a decrease in communication between vascular and parenchymal cell types in AD, particularly for EndoMT-2, M-pericytes and menengial-FB to Ex-Neuro-2/6 and In-Neuro-5. Conversely, there is more communication between vascular cell types in AD, particularly for pericyte-2 and several perivascular-FB subtypes.</p>
</section>
<section id="alzheimers-disease-affects-genes-involved-in-protein-folding-and-vascular-remodeling" class="level2" data-number="2.7">
<h2 data-number="2.7" class="anchored" data-anchor-id="alzheimers-disease-affects-genes-involved-in-protein-folding-and-vascular-remodeling"><span class="header-section-number">2.7</span> Alzheimer’s Disease affects genes involved in protein folding and vascular remodeling</h2>
<p>We observed significant effects on vascular, beyond just EndoMT-1, and glial cells. Confining to the strongest gene expression changes (average log fold change LFC &gt; 1, adjusted p value &lt; 0.05), capillary and venous EC showed the most changes (<a href="#fig-degs" class="quarto-xref">Figure&nbsp;6</a> A) and these were also among the top four cell types with differentially expressed genes (DEGs) without an LFC cut-off (<a href="#fig-degs" class="quarto-xref">Figure&nbsp;6</a> B). Most cell types exhibit few significant DEGs, but a subset have many, including quiescent (Q) and activated (A) astrocytes and Endo-MT1.</p>
<p>We examined functional enrichment among the DEGs and in the gene set with LFC &gt; 1 (<a href="#fig-degs" class="quarto-xref">Figure&nbsp;6</a> C), response to chemokine, erythrocyte differentiation and cytosolic calcium release were identified across astrocytes, neurons, microglia, endothelial, pericytes and SMCs. Of note, ANGPT2 was one genes contributing to the erythrocyte differentiation pathway summary. Similarly, CSF1 was also a gene in these pathways and has been reported to reduce volume and reduce cognitive decline in mice when knocked out<span class="citation" data-cites="dagher2015"><sup><a href="#ref-dagher2015" role="doc-biblioref">35</a></sup></span>, whereas in humans, increased CSF1 expression has been correlated with plaque burden in post-mortem tissue<span class="citation" data-cites="walker2017"><sup><a href="#ref-walker2017" role="doc-biblioref">36</a></sup></span>. Function enrichment on the broader DEGs (<a href="#fig-degs" class="quarto-xref">Figure&nbsp;6</a> D) showed endothelial migration terms across many of the vascular cell types – proposing changes in vascular remodelling in AD<span class="citation" data-cites="fisher2022"><sup><a href="#ref-fisher2022" role="doc-biblioref">37</a></sup></span>. However, the most consistent changes across cell types both with and without an LFC cut-off, were in genes involved in protein folding, which is expected given the central role of amyloid and tau protein aggregation in AD.</p>
</section>
</section>
<section id="discussion" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Discussion</h1>
<p>In this study, by leveraging custom-designed 3D-printed tools and optimizing the isolation workflow we developed an efficient method for the simultaneous isolation of highly pure vascular nuclei and parenchymal nuclei from human post-mortem brain tissue. Comparisons with previous methods demonstrate markedly higher recovery of both neurons and brain endothelial nuclei, underscoring the utility of this protocol for comprehensive neurovascular unit (NVU) analysis. Our approach thereby offers new opportunities for understanding complex brain disorders such as Alzheimer’s disease (AD), considering the contribution of all major cell types in brain cortical tissue. We thereby found enrichment of AD risk genes in vascular cells such as pericytes and perivascular fibroblasts. The identification of these specific cell types and the enrichment of amyloid-related processes underscore the complex interplay between various cellular components in the brain and AD pathology. Activated microglia, pericytes, and perivascular fibroblasts may contribute to disease mechanisms through their roles in immune response, vascular integrity, and amyloid processing, respectively. The independence of these signals, as shown by conditional analysis, highlights the unique contributions of each cell type.</p>
<p>Our isolation method has also revealed an interesting transitional cell-population, EndoMT. Furthermore, the trajectory analysis suggests that endothelial-to-mesenchymal transition processes may be perturbed in AD. A mesenchymal stem cell-like transition has been described for endothelium, particularly in the heart where they have been proposed to underlie cardiac fibrosis<span class="citation" data-cites="zeisberg2007"><sup><a href="#ref-zeisberg2007" role="doc-biblioref">38</a></sup></span> and may also differentiate into cardiac pericytes and vascular smooth muscle. In this transition, endothelial cells show degradation of their vascular basement membrane, loss of cell–cell contact, and acquire a migratory phenotype. There is also loss of expression of specific endothelial markers including VE-Cadherin and gain expression of mesenchymal markers including alpha smooth muscle actin (α-SMA)<span class="citation" data-cites="medici2010"><sup><a href="#ref-medici2010" role="doc-biblioref">39</a></sup></span>. Transforming growth factor-β1 has been identified as a key inducer of EndoMT and can promote smooth muscle-like differentiation in pericytes<span class="citation" data-cites="verbeek1994"><sup><a href="#ref-verbeek1994" role="doc-biblioref">40</a></sup></span>. We identified several transcription factors differentially expressed across pseudotime that distinctly between the two lineages and also between early and late-stage transitioning. EndoMT has also been reported to drive the formation of cerebral cavernous malformations<span class="citation" data-cites="maddaluno2013"><sup><a href="#ref-maddaluno2013" role="doc-biblioref">41</a></sup></span>. However in our study, the EndoMT populations are abundant in control donor brain and are directed into either vascular smooth muscle or pericytes. Whilst this has been described in the developing heart, our study is the first to show this in adult brain. This may therefore represent an important physiological process to dynamically alter and renew mural cell composition in the brain. Further functional studies will be required to characterize the determinants of brain EndoMT. The reduction of EndoMT populations in AD implicates vascular plasticity as an under-explored disease vulnerability in AD. The loss of pericytes in the blood-brain barrier has previously been described in Alzheimer’s disease<span class="citation" data-cites="winkler2014"><sup><a href="#ref-winkler2014" role="doc-biblioref">42</a></sup></span> and our data suggests this may be a consequence of both perturbed EndoMT differentiation and overall abundance. The significant reduction in the EndoMT population in AD is presumably driven by endothelial cells failing to differentiate into them, though this requires further validation. Evaluating whether restoration of the EndoMT population in AD rescues vascular and BBB deficits would be of particular interest, and potentially relevant to other neurodegenerative conditions. The transcriptional changes observed along these lineages, including dysregulation of key genes such as BACE2 and APOD, reinforce the involvement of vascular transformations in AD-related neurodegeneration. BACE2 can cleave APP within the Aβ domain, potentially preventing Aβ42 generation and exhibiting neuroprotective properties<span class="citation" data-cites="yeap2022a"><sup><a href="#ref-yeap2022a" role="doc-biblioref">43</a></sup></span>.</p>
<p>Our data provides insights into cell-cell communication in the NVU. We observed a decrease in overall cell-cell communication between vascular and parenchymal cells in AD highlighting potential disruptions in NVU integrity that warrant further exploration. Furthermore we identified significant alterations in specific signalling pathways, particularly dysregulated ANGPT2 and ANGPT1, with opposing profiles that align with their established roles in vascular stability. These signalling dynamics provide a mechanistic basis for the observed vascular alterations in AD, suggesting that disruptions in endothelial signalling contribute to NVU dysfunction and subsequent neurodegeneration. Elevated cerebrospinal fluid (CSF) ANGPT2 levels correlate with blood-brain barrier leakiness, tau pathology, and neuronal injury in early AD<span class="citation" data-cites="vanhulle2024"><sup><a href="#ref-vanhulle2024" role="doc-biblioref">44</a></sup></span>. Plasma ANGPT2 levels are also higher in AD patients compared to controls<span class="citation" data-cites="qin2015"><sup><a href="#ref-qin2015" role="doc-biblioref">45</a></sup></span>.</p>
<p>In conclusion, our novel protocol enables the simultaneous, high-quality isolation of vascular and parenchymal nuclei, facilitating a detailed characterization of the NVU in health and disease. This approach has revealed previously unappreciated vascular contributions to AD, particularly through novel cellular and molecular insights into EndoMT and its dysregulation. These findings underscore the importance of vascular plasticity in brain health and highlight the NVU as a promising target for therapeutic strategies in neurodegenerative diseases. Future studies leveraging this method may yield further insights into the complex interplay of cell types within the human brain.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a>celltype_abbrev_df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/celltype_abbreviations.tsv"</span></span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 30 Columns: 2
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): celltype, celltypes_abbrev

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in data, add celltype abbreviations and add fraction label</span></span>
<span id="cb232-2"><a href="#cb232-2" aria-hidden="true" tabindex="-1"></a>cell_pop_diff_par <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb232-3"><a href="#cb232-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/990_processed_data/001_snrnaseq/10_cell_subtypes/speckle_level2_celltype_proportions_parenchymal.csv"</span></span>
<span id="cb232-4"><a href="#cb232-4" aria-hidden="true" tabindex="-1"></a>)) <span class="sc">|&gt;</span></span>
<span id="cb232-5"><a href="#cb232-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltype =</span> BaselineProp.clusters) <span class="sc">|&gt;</span></span>
<span id="cb232-6"><a href="#cb232-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb232-7"><a href="#cb232-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">fraction =</span> <span class="st">"P"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 9 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (1): BaselineProp.clusters
dbl (7): BaselineProp.Freq, PropMean.Case, PropMean.Control, PropRatio, Tsta...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Do the same and rbind with parenchymal fraction</span></span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a>cell_pop_diff <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/990_processed_data/001_snrnaseq/10_cell_subtypes/speckle_level2_celltype_proportions_vascular.csv"</span></span>
<span id="cb234-4"><a href="#cb234-4" aria-hidden="true" tabindex="-1"></a>)) <span class="sc">|&gt;</span></span>
<span id="cb234-5"><a href="#cb234-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltype =</span> BaselineProp.clusters) <span class="sc">|&gt;</span></span>
<span id="cb234-6"><a href="#cb234-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb234-7"><a href="#cb234-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">fraction =</span> <span class="st">"V"</span>) <span class="sc">|&gt;</span></span>
<span id="cb234-8"><a href="#cb234-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(cell_pop_diff_par)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 20 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (1): BaselineProp.clusters
dbl (7): BaselineProp.Freq, PropMean.Case, PropMean.Control, PropRatio, Tsta...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to sig pop differences</span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a>cell_pop_diff_sig <span class="ot">&lt;-</span> cell_pop_diff <span class="sc">|&gt;</span></span>
<span id="cb236-3"><a href="#cb236-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(P.Value <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb236-4"><a href="#cb236-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), \(x) <span class="fu">round</span>(x, <span class="at">digits =</span> <span class="dv">3</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="acknowledgments" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="acknowledgments"><span class="header-section-number">3.1</span> Acknowledgments</h2>
<p>This project has received funding (ZC) from the Innovative Medicine Initiative 2 Joint Undertaking under grant agreement No 807015 (IM2PACT). This Joint Undertaking receives support from the European Union’s Horizon 2020 research and innovation programme EFPIA. The manuscript reflects the authors’ view and that neither IMI nor the European Union, EFPIA, or any Associated Partners are responsible for any use that may be made of the information contained therein. This work is also supported by the UK Dementia Research Institute [award number UK DRI-3005] through UK DRI Ltd, principally funded by the Medical Research Council. Part of this work was performed using the computational facilities of the Advance Research Computing @ Cardiff (ARCCA) Division, Cardiff University. We also thank Robert Hedley and Vasiliki Tsioligka for providing technical assistance in Fluorescence Activated Cell Sorting at the Flow Cytometry Facility, Sir William Dunn School of Pathology, University of Oxford.</p>
</section>
<section id="author-contributions" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="author-contributions"><span class="header-section-number">3.2</span> Author contributions</h2>
<p>Author roles were classified using the Contributor Role Taxonomy (CRediT; https://credit.niso.org/) as follows: Gabriel Mateus Bernardo Harrington: formal analysis, software, writing – original draft, writing – review &amp; editing, visualization, data curation, conceptualization; Hannah Sleven: [Left for HS to fill in]; Jimena Monzón-Sandoval: writing – review &amp; editing, software, visualization, conceptualization, supervision; Gabriel Rocha: method, investigation; Lara Robinson: method, investigation, writing – original draft; Michal Rokicki: investigation; Joanne Morgan: investigation; Ngoc-Nga Vinh: investigation; Quenten Schwarz: investigation; Caleb Webber: writing – original draft, writing – review &amp; editing, project administration, supervision, conceptualization, funding acquisition; Zameel Cader: writing – original draft, writing – review &amp; editing, project administration, supervision, conceptualization, funding acquisition</p>
</section>
<section id="declaration-of-interests" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="declaration-of-interests"><span class="header-section-number">3.3</span> Declaration of interests</h2>
<p>Zameel Cader is founder of Oxford StemTech Litd and has received honoraria, speaker fees and consultancy fees from Pfizer, TEVA and Abbive. He has received research funding in unrelated projects from GSK and Novo Nordisk.</p>
<p>Caleb Webber has received research funding in unrelated projects from GSK and Eli Lilly.</p>
</section>
<section id="figure-titles-and-legends" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="figure-titles-and-legends"><span class="header-section-number">3.4</span> Figure titles and legends</h2>
</section>
<section id="tables-with-title-and-legends" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="tables-with-title-and-legends"><span class="header-section-number">3.5</span> Tables with title and legends</h2>
</section>
</section>
<section id="star-methods" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> STAR Methods</h1>
<p>Link to info on this here: <a href="https://www.cell.com/star-authors-guide" class="uri">https://www.cell.com/star-authors-guide</a></p>
<p>And a pdf guide <a href="https://www.cell.com/pb-assets/journals/research/cell/methods/Methods_Guide_Cell-1678470557763.pdf">here</a></p>
<ul>
<li>Plan to make the code available on GitHub and make a docker image with the R environment for all the downstream processing - The seurat object from scflow could be included in this image - This could go on ADDI as well?</li>
<li>Raw data will go on GEO I guess</li>
</ul>
<section id="key-resoures-table" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="key-resoures-table"><span class="header-section-number">4.1</span> Key resoures table</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">

<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div><div class="sourceCode cell-code" id="cb237"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb237-1"><a href="#cb237-1" aria-hidden="true" tabindex="-1"></a>key_resources <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb237-2"><a href="#cb237-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/999_data_for_publication/key_resources_table.xlsx"</span></span>
<span id="cb237-3"><a href="#cb237-3" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb237-4"><a href="#cb237-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb237-5"><a href="#cb237-5" aria-hidden="true" tabindex="-1"></a><span class="co"># gt::gt(key_resources)</span></span>
<span id="cb237-6"><a href="#cb237-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(key_resources)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div><div id="tbl-key-resources" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-key-resources-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Key resources table
</figcaption>
<div aria-describedby="tbl-key-resources-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="do-not-create-environment cell caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 15%">
<col style="width: 27%">
<col style="width: 27%">
<col style="width: 29%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">group</th>
<th style="text-align: left;">REAGENT or RESOURCE</th>
<th style="text-align: left;">SOURCE</th>
<th style="text-align: left;">IDENTIFIER</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Critical commercial assays</td>
<td style="text-align: left;">10X Genomics Chromium Single Cell 3’ Reagent Kits</td>
<td style="text-align: left;">10X Genomics</td>
<td style="text-align: left;">PN-</td>
</tr>
<tr class="even">
<td style="text-align: left;">Deposited data</td>
<td style="text-align: left;">Raw and analysed data</td>
<td style="text-align: left;">This paper</td>
<td style="text-align: left;">GEO:</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Software and algorithms</td>
<td style="text-align: left;">Cellranger v7.1.0</td>
<td style="text-align: left;">10X Genomics</td>
<td style="text-align: left;">https://www.10xgenomics.com/software</td>
</tr>
<tr class="even">
<td style="text-align: left;">Software and algorithms</td>
<td style="text-align: left;">Nextflow v23.04.1.5866</td>
<td style="text-align: left;">https://doi.org/10.1038/nbt.3820</td>
<td style="text-align: left;">https://www.nextflow.io/</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Software and algorithms</td>
<td style="text-align: left;">scFow pipeline v0.7.2</td>
<td style="text-align: left;">https://doi.org/10.22541/au.162912533.38489960/v1</td>
<td style="text-align: left;">https://github.com/combiz/nf-core-scflow/tree/dev-nf</td>
</tr>
<tr class="even">
<td style="text-align: left;">Software and algorithms</td>
<td style="text-align: left;">R v4.4.1</td>
<td style="text-align: left;">R Foundation for Statistical Computing</td>
<td style="text-align: left;">https://www.R-project.org/</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Software and algorithms</td>
<td style="text-align: left;">Seurat v5.1.0</td>
<td style="text-align: left;">https://doi.org/10.1038/s41587-023-01767-y</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">Software and algorithms</td>
<td style="text-align: left;">Data analysis code</td>
<td style="text-align: left;">This paper</td>
<td style="text-align: left;">doi for code goes here</td>
</tr>
</tbody>
</table>
</div>
</div>
</figure>
</div></div></div>
</div></div>
</section>
<section id="resource-availability" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="resource-availability"><span class="header-section-number">4.2</span> Resource availability</h2>
<section id="lead-contact" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="lead-contact"><span class="header-section-number">4.2.1</span> Lead contact</h3>
<p>Further information and requests for resources and reagents should be directed to and will be fulfilled by the lead contacts, Zameel Cader (zameel.cader@ndcn.ox.ac.uk) &amp; Caleb Webber (webberc4@cardiff.ox.ac.uk).</p>
</section>
<section id="materials-availability" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="materials-availability"><span class="header-section-number">4.2.2</span> Materials availability</h3>
<p>No unique reagents were generated for this study.</p>
</section>
<section id="data-and-code-availability" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="data-and-code-availability"><span class="header-section-number">4.2.3</span> Data and code availability</h3>
<ul>
<li>All raw data are available in the GEO database under the accession number GEO:</li>
<li>The Seurat object generated from the scFlow pipeline is included in the Docker image with the R environment used for analysis here:</li>
<li>All original code has been deposited at Zenodo and is publicly available as of the date of publication. DOIs are listed in the key resources table.</li>
</ul>
</section>
</section>
<section id="experimental-model-and-study-participant-details" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="experimental-model-and-study-participant-details"><span class="header-section-number">4.3</span> Experimental model and study participant details</h2>
<section id="post-mortem-tissue-donors" class="level3" data-number="4.3.1">
<h3 data-number="4.3.1" class="anchored" data-anchor-id="post-mortem-tissue-donors"><span class="header-section-number">4.3.1</span> Post-mortem tissue donors</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a>n_males <span class="ot">&lt;-</span> <span class="fu">sum</span>(donor_metadata<span class="sc">$</span>sex <span class="sc">==</span> <span class="st">"M"</span>)</span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>n_females <span class="ot">&lt;-</span> <span class="fu">sum</span>(donor_metadata<span class="sc">$</span>sex <span class="sc">==</span> <span class="st">"F"</span>)</span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a>median_age <span class="ot">&lt;-</span> <span class="fu">median</span>(donor_metadata<span class="sc">$</span>age)</span>
<span id="cb238-4"><a href="#cb238-4" aria-hidden="true" tabindex="-1"></a>age_range <span class="ot">&lt;-</span> <span class="fu">range</span>(donor_metadata<span class="sc">$</span>age)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Post-mortem pre-frontal cortex from 20 controls and 20 AD brain donors were collected from The Oxford Brain Bank, Department of Neuropathology of the Oxford University Hospitals NHS Trust under the South Central – Oxford C Research Ethics Committee reference 15/SC/0639. The clinicopathological parameters were collected and summarized in Table S1, including gender, age, diagnosis, APOE status, ethnicity and Braak stage. The clinicopathological parameters were collected and summarized in <a href="#supptbl-donor-metadata" class="quarto-xref">Table S1</a>, including gender, age, diagnosis, APOE status, ethnicity and Braak stage. 3 donors were excluded in QC and of the remaining 37 donors, 23 were female and 14 were male with a median age of 87 (range 48-101). Sex and age were included as a covariates in pseudobulk differential gene expression analysis.</p>
</section>
</section>
<section id="method-details" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="method-details"><span class="header-section-number">4.4</span> Method details</h2>
<section id="sample-prepneurovascular-unit-isolation" class="level3" data-number="4.4.1">
<h3 data-number="4.4.1" class="anchored" data-anchor-id="sample-prepneurovascular-unit-isolation"><span class="header-section-number">4.4.1</span> Sample prep/neurovascular unit isolation</h3>
<p>Post-mortem fresh-frozen prefrontal cortex was obtained from the Oxford Brain Bank, and blood and meninges dissected from the surface on dry ice using a scalpel. The 3D printed tissue smashing apparatus are removed from the Ultralow Temperature Freezer and assembled on dry ice, and the tissue block minced until resembling a fine pink powder. The minced tissue is then transferred into a 50ml tube on ice, containing 4ml ice-cold Buffer A (1X PBS with 1.87 mM KCl, 0.72 mM CaCl<span class="math inline">\(_2\)</span>, 0.0125 g Glucose) and defrosted for 15 minutes. The tissue is homogenised for 60 strokes with the 3D printed homogeniser, topped up with 10ml Buffer A and spun in a swinging bucket microcentrifuge at 500 rpm for 1 minute. After removing and retaining 10ml supernatant in a tube labelled ‘500 rpm’, these homogenisation steps are repeated twice more, finally topping up with 20 ml Buffer A and spinning down along with the ‘500 rpm’ tube at 500 rpm for 3 minutes. Supernatants are removed and retained - the ‘500 rpm’ into a ‘1200 rpm’ tube, and homogenisation tube in a ‘dounce’ tube - and the pellets resuspended in 4ml Buffer A. Samples are then further homogenised using a 7ml dounce (Wheaton, 357424) and loose pestle for 50 strokes on ice. The solution can be added to the retained supernatant from the original homogenisation tube and spun at 1200 rpm for three minutes. 3ml supernatant from the ‘1200 rpm’ tube can be put to one side, as this will form the parenchymal fraction.</p>
<p>Purification of the vascular fraction begins with two multi-layer dextran gradient centrifugation steps to trap contaminants whilst allowing vascular transfer. Pellets from the previous 1200rpm centrifugation step are resuspended in 5.4% Dextran (VWR, A1847.0500) and layered on top of a gradient constituted of 20ml 16% and 10ml 12% Dextran solutions, before spinning at 2600 xg for 6 minutes. After centrifugation, myelin and debris remains in the top two thirds of the volume, and a vessel suspension resides at the base of the tube. All of the myelin-contaminated volume is removed to around the 10ml mark on the Falcon tube, tips changed, and pellet resuspended in an equal volume of 0.5% BSA-supplemented buffer (Merck, AM2616), before underlaying 20ml 20% Dextran with a clean pipette. After the remaining samples are processed, all are spun down at 4200xg for 2 minutes, and the supernatant removed. The resulting vessel suspension is then filtered through a 100um nylon filter to remove unwanted larger vessels and pelleted at 800xg for 8 minutes. Subsequent vessel capture on 40um PET filters allows depletion of residual contaminating particles, before inverting the filters and washing through captured vessels, again centrifuging at 800xg for 8 minutes. Purified vessels are then transferred onto 20um PET filters and washed before collagenase II (Stemcell Technologies, 07418) digestion of the basement membrane of the vasculature. After the enzyme and released final debris are removed with a further wash step, the 20um filter is placed into the 3D printed funnel assembly and a hand-held homogeniser is used with a 3D printed flat base pestle to grind the nuclei out of the vessels (<a href="#fig-method" class="quarto-xref">Figure&nbsp;1</a> C). Nuclei after staining with DAPI are isolated using FACS. The parenchymal supernatant extracted previously is processed to nuclei using standard protocols and similarly FACS isolated. Extracted and isolated nuclei can then be used with the single-nuclei RNA-seq platform of choice (here 10x Genomics) for transcriptomic analysis.</p>
</section>
</section>
<section id="endo-mt-facs-analysis" class="level2" data-number="4.5">
<h2 data-number="4.5" class="anchored" data-anchor-id="endo-mt-facs-analysis"><span class="header-section-number">4.5</span> Endo-MT FACS Analysis</h2>
<p>Following isolation of vascular nuclei fractions from the prefrontal cortices of 10 donors (5 AD, 5 control), each sample was stained with DAPI, and blocked with Goat Serum before adding primary antibodies - Anti-ERG (abcam, EPR3864, 10 µg/mL) for endothelial nuclei, and Anti-Notch3 (Invitrogen, PA5-142191, 10 µg/mL) for mesenchymal nuclei - then incubated for 30 minutes on ice. Subsequently, secondary antibodies were added: Donkey anti-Rabbit IgG AlexaFluor 488 (Invitrogen, A-21206, 1:) and Donkey anti-Goat IgG AlexaFluor 594 (Invitrogen, A-11058) respectively.</p>
<p>Fluorescence activated sorting was performed on a BD FACSAria III, by sorting nuclei positive for DAPI and either AlexaFluor488, AlexaFluor594, or both. Following sorting, the .fcs files were imported into Floreada.io for careful adjustment of gating using unstained controls to ascertain the cutoffs for positively stained nuclei. The endothelial, mesenchymal, and Endo-MT populations were identified as proportions of the overall DAPI+ cohort of nuclei, with the Endo-MT population defined as those nuclei double-positive for both ERG and Notch3.</p>
</section>
<section id="mouse-lineage-tracing" class="level2" data-number="4.6">
<h2 data-number="4.6" class="anchored" data-anchor-id="mouse-lineage-tracing"><span class="header-section-number">4.6</span> Mouse lineage tracing</h2>
<section id="mouse-model" class="level3" data-number="4.6.1">
<h3 data-number="4.6.1" class="anchored" data-anchor-id="mouse-model"><span class="header-section-number">4.6.1</span> Mouse model</h3>
<p>All animal experiments were reviewed and approved by the University of South Australia and SJCRH Institutional Animal Care and Use Committees. Mice expressing Cre under the Cdh5 promoter (Cdh5-Cre)<span class="citation" data-cites="chen2009"><sup><a href="#ref-chen2009" role="doc-biblioref">46</a></sup></span> were crossed with Ai14 (Rosa-CAG-LSL-tdTomato-WPRE) mice obtained from The Jackson Laboratory (#007914)<span class="citation" data-cites="madisen2009"><sup><a href="#ref-madisen2009" role="doc-biblioref">47</a></sup></span> to generate Cdh5-Cre; R26-tdTom animals for lineage tracing of endothelial cells. Genotyping was carried out by Transnetyx (Cordova, TN). All mice had consistent access to food and water and were housed at ambient temperature (20–25°C) and humidity (40–60%) with 12-h light/12-h dark cycles.</p>
</section>
<section id="immunofluorescence-analysis-of-brain-tissue" class="level3" data-number="4.6.2">
<h3 data-number="4.6.2" class="anchored" data-anchor-id="immunofluorescence-analysis-of-brain-tissue"><span class="header-section-number">4.6.2</span> Immunofluorescence analysis of brain tissue</h3>
<p>Whole brains were fixed in 4% PFA, bisected through the midline, cryopresereved in 20% sucrose, mounted and cryosectioned in OCT compound (Cellpath, KMA-0100-00A). 14mm cryosections were dried at room temperature in the dark for 30 minutes and washed in PBS containing 0.1% Tween 20 (Sigma Aldrich, P2287). To image endogenous tdTomato fluorescence, sections were incubated with 1 mM DAPI for 10 minutes, washed twice with PBS, and mounted with Prolong Gold (Thermo-Fisher Scientific, P36930). For immunofluorescence analysis of pericytes and endothelial cells, sections were were blocked in 10% DAKO block, 0.1% Tween-20 in PBS, and stained with the indicated primary antibodies. Antibodies used were rat anti-CD31 (Biolegend) 1:150; goat anti-PDGFRb (R&amp;D technologies) 1:100. Slides were mounted in Fluoro-mount G with DAPI (ProSciTech). Confocal images were acquired on a LSM 800 (Zeiss) system.</p>
</section>
</section>
<section id="quantification-and-statistical-analysis" class="level2" data-number="4.7">
<h2 data-number="4.7" class="anchored" data-anchor-id="quantification-and-statistical-analysis"><span class="header-section-number">4.7</span> Quantification and statistical analysis</h2>
<section id="data-processing" class="level3" data-number="4.7.1">
<h3 data-number="4.7.1" class="anchored" data-anchor-id="data-processing"><span class="header-section-number">4.7.1</span> Data Processing</h3>
<p>The raw sequencing data was processed using CellRanger (version 7.1.0, 10X Genomics), which performed initial alignment, filtering, barcode counting, and UMI counting. An updated reference genome (Ensembl version 109 and Gencode version 43 annotations for GRCh38) was generated for use with CellRanger. The resulting gene-barcode matrices were then further processed using the scFlow pipeline implemented in Nextflow.<span class="citation" data-cites="khozoie2021 ditommaso2017"><sup><a href="#ref-khozoie2021" role="doc-biblioref">48</a>,<a href="#ref-ditommaso2017" role="doc-biblioref">49</a></sup></span></p>
</section>
<section id="sec-seq-qc" class="level3" data-number="4.7.2">
<h3 data-number="4.7.2" class="anchored" data-anchor-id="sec-seq-qc"><span class="header-section-number">4.7.2</span> Quality Control</h3>
<p>The processed data were read into R (version 4.4.0) using the Seurat package (version 5.1.0).<span class="citation" data-cites="base Seurat"><sup><a href="#ref-base" role="doc-biblioref">50</a>,<a href="#ref-Seurat" role="doc-biblioref">51</a></sup></span> In addition to the filtering performed by scFlow, cells with low feature/RNA counts were filtered out (between &lt;300 &amp; &lt;7500 for features and &gt;1000 for RNA), and donors with an insufficient number of high-quality cells were excluded from further analysis, resulting in 3 donors (both fractions), and the vascular fraction from 1 addition donor being excluded.</p>
</section>
<section id="clustering" class="level3" data-number="4.7.3">
<h3 data-number="4.7.3" class="anchored" data-anchor-id="clustering"><span class="header-section-number">4.7.3</span> Clustering</h3>
<p>A standard Seurat workflow was followed including normalisation, finding variable features, scaling and PCA. For the UMAP 35 dimensions and a resolution of 0.6 was used.</p>
</section>
<section id="cell-type-annotation" class="level3" data-number="4.7.4">
<h3 data-number="4.7.4" class="anchored" data-anchor-id="cell-type-annotation"><span class="header-section-number">4.7.4</span> Cell Type Annotation</h3>
<p>Cell types were annotated based on canonical marker genes identified from the literature,<span class="citation" data-cites="yang2022"><sup><a href="#ref-yang2022" role="doc-biblioref">16</a></sup></span> in particular. The expression levels of these marker genes were used to classify cells into distinct main cell types. These higher level cell types were then subclustered with 20 dimensions and a resolution of 0.4 to identify cell subtypes. This annotation process was validated by comparing the identified cell types to known cell type distributions in similar datasets.</p>
</section>
<section id="differences-in-cell-proportions" class="level3" data-number="4.7.5">
<h3 data-number="4.7.5" class="anchored" data-anchor-id="differences-in-cell-proportions"><span class="header-section-number">4.7.5</span> Differences in cell proportions</h3>
<p>To analyze differences in cell type proportions between case and control groups, we utilized both the scProportionTest R package (version 0.0.0.9000) and the propeller function from the speckle R package (version 1.4.0) on the vascular and parenchymal fractions separately.<span class="citation" data-cites="speckle Miller2021"><sup><a href="#ref-Miller2021" role="doc-biblioref">24</a>,<a href="#ref-speckle" role="doc-biblioref">52</a></sup></span> For scProportionTest a permutation test is used to calculate a p-value for each cluster, and a confidence interval for the magnitude difference is returned via bootstrapping. The speckle method employs a robust linear modelling framework to test for significant differences in cell type proportions across experimental conditions, while accounting for the compositional nature of the data and potential variability between samples.</p>
</section>
<section id="mast-differential-gene-expression" class="level3" data-number="4.7.6">
<h3 data-number="4.7.6" class="anchored" data-anchor-id="mast-differential-gene-expression"><span class="header-section-number">4.7.6</span> MAST differential gene expression</h3>
<p>The FindMarkers function from Seurat was used to perform differential gene expression via MAST. Donor ID, age and sex were included as latent variables. This was then used to perform hypergeometric GO enrichment analysis by filtering to significantly differentially expressed genes (DEGs) (adjusted p-value &lt; 0.05) and running <code>enrichGO</code> from <code>clusterProfiler</code> (version 4.12.0) with all genes identified in this dataset used a background.<span class="citation" data-cites="clusterProfiler"><sup><a href="#ref-clusterProfiler" role="doc-biblioref">53</a></sup></span></p>
</section>
<section id="mammalian-phenotype-ontology-enrichment" class="level3" data-number="4.7.7">
<h3 data-number="4.7.7" class="anchored" data-anchor-id="mammalian-phenotype-ontology-enrichment"><span class="header-section-number">4.7.7</span> Mammalian Phenotype Ontology enrichment</h3>
<p>We use the Jackson Laboratory Mammalian Phenotype Ontology database<span class="citation" data-cites="Smith2009"><sup><a href="#ref-Smith2009" role="doc-biblioref">54</a></sup></span> (release 2024-02-07 ) to test for enrichment of gene lists in phenotypic terms. Given the directed acyclic graph structure of the ontology, we used Simona<span class="citation" data-cites="gu2023"><sup><a href="#ref-gu2023" role="doc-biblioref">55</a></sup></span> (version 1.0.10) to annotate child (more specific) terms to their ancestors (more general) terms, creating a deeply annotated set. We further subset the database to terms with more than 10 and less than 200 genes and tested for gene enrichment using an hypergeometric test. We adjusted the p-values using the Benjamini-Hochberg method to account for multiple testing.</p>
</section>
<section id="cell-cell-communication" class="level3" data-number="4.7.8">
<h3 data-number="4.7.8" class="anchored" data-anchor-id="cell-cell-communication"><span class="header-section-number">4.7.8</span> Cell-cell communication</h3>
<p>To investigate cell-cell communication we employed <code>CellChat</code> (version 2.1.2).<span class="citation" data-cites="CellChat"><sup><a href="#ref-CellChat" role="doc-biblioref">56</a></sup></span> A <code>CellChat</code> object was created for cases and controls separately using the “Secreted Signalling” ligand-receptor interaction database. A standard <code>CellChat</code> was then followed to identify over expressed genes/ligands and contrast these in cases and controls.</p>
</section>
<section id="protein-protein-interaction-networks" class="level3" data-number="4.7.9">
<h3 data-number="4.7.9" class="anchored" data-anchor-id="protein-protein-interaction-networks"><span class="header-section-number">4.7.9</span> Protein-protein interaction networks</h3>
<p>We created a combined protein-protein interaction network by combining the following resources: APID<span class="citation" data-cites="Alonso-López2019"><sup><a href="#ref-Alonso-López2019" role="doc-biblioref">57</a></sup></span>, BIOGRID<span class="citation" data-cites="Oughtred2021"><sup><a href="#ref-Oughtred2021" role="doc-biblioref">58</a></sup></span>, BIOPLEX<span class="citation" data-cites="Huttlin2021"><sup><a href="#ref-Huttlin2021" role="doc-biblioref">59</a></sup></span>, CORUM<span class="citation" data-cites="Tsitsiridis2023"><sup><a href="#ref-Tsitsiridis2023" role="doc-biblioref">60</a></sup></span>, HITPREDICT<span class="citation" data-cites="López2015"><sup><a href="#ref-López2015" role="doc-biblioref">61</a></sup></span>, HuRi<span class="citation" data-cites="Luck2020"><sup><a href="#ref-Luck2020" role="doc-biblioref">62</a></sup></span>, INTACT<span class="citation" data-cites="Orchard2014"><sup><a href="#ref-Orchard2014" role="doc-biblioref">63</a></sup></span>, MINT<span class="citation" data-cites="Licata2012"><sup><a href="#ref-Licata2012" role="doc-biblioref">64</a></sup></span>, REACTOME<span class="citation" data-cites="Gillespie2022"><sup><a href="#ref-Gillespie2022" role="doc-biblioref">65</a></sup></span> and only protein physical links from STRING<span class="citation" data-cites="Szklarczyk2023"><sup><a href="#ref-Szklarczyk2023" role="doc-biblioref">66</a></sup></span>. All datasets were mapped to Ensembl Gene ID from either Entrez, Uniprot or Ensembl Protein IDs using org.Hs.eg.db. All duplicated and self-interactions were removed. For any gene set of interest we tested if we observed more interactions than expected by chance. A 10,000 randomizations were used to obtain an empiric p value, reflecting the number of times that equally sized random samples of genes (with similar degree and gene lenght) had more interactions than our gene set of interest.</p>
</section>
<section id="sec-method-risk" class="level3" data-number="4.7.10">
<h3 data-number="4.7.10" class="anchored" data-anchor-id="sec-method-risk"><span class="header-section-number">4.7.10</span> Disease risk enrichment</h3>
<p>To investigate enrichment for disease risk we employed both MAGMA (version 1.10) and LD score (version 1.0.1).<span class="citation" data-cites="leeuw2015 bulik-sullivan2015"><sup><a href="#ref-leeuw2015" role="doc-biblioref">67</a>,<a href="#ref-bulik-sullivan2015" role="doc-biblioref">68</a></sup></span> We subset to control samples and identified celltype-specific marker genes using <code>FindAllMarkers</code> from the Seurat package.<span class="citation" data-cites="Seurat-2"><sup><a href="#ref-Seurat-2" role="doc-biblioref">69</a></sup></span> Only genes that are detected in a minimum of 25% of cells in either of the two populations are considered and a minimum log fold change of 0.01 were used. This function was used to apply a Wilcoxon Rank Sum test to compare each cluster against all other clusters identifying differentially expressed gene. From this list of genes, the top ten percent of the total number of genes in the dataset (27459 genes in total) which had the lowest p-values from this test were selected as the celltype specific genes.</p>
<p>The 1000 Genomes Project (Phase 3) was used as a reference in combination with the NCBI37 (GRCh37) genome build as an annotation file.<span class="citation" data-cites="auton2015 sayers2022"><sup><a href="#ref-auton2015" role="doc-biblioref">70</a>,<a href="#ref-sayers2022" role="doc-biblioref">71</a></sup></span> Genes were annotated with a window 35Kb and 10Kb up/downstream respectively. The Bellenguez et al.&nbsp;2022 AD GWAS was used.<span class="citation" data-cites="bellenguez2022"><sup><a href="#ref-bellenguez2022" role="doc-biblioref">20</a></sup></span></p>
</section>
<section id="pseudotime-analysis" class="level3" data-number="4.7.11">
<h3 data-number="4.7.11" class="anchored" data-anchor-id="pseudotime-analysis"><span class="header-section-number">4.7.11</span> Pseudotime analysis</h3>
<p>To investigate pseudotime in EndoMT we subset the data to just the endothelial, mural and EndoMT nuclei. Standard processing as applied to the subset (normalization, feature selection and scaling) via Seurat. PCA and UMAP were computed with 2 dimensions/PCs respectively and clustering was performed with a resolution of 0.1. This clustered the endothelial and mural celltypes well, but not the EndoMT, so that was subset and re-clustered with a resolution of 0.02 with the subsequent labels applied to the 2 EndoMT clusters. The slingshot R package (version 2.12.0) was then applied to the data on the PCA with the endothelial cluster as the starting cluster. Other clustered were tested as starting clusters, but a linear pseudotime starting from the endothelial cluster fit the data best. The tradeSeq R package (version 1.18.0) was then applied to identify pseudotemporally differential genes. As the number of nuclei in the EndoMT clusters was much smaller than the endothelia/mural clusters the data was subsampled to match the size of the smallest EndoMT cluster (1500 nuclei). Age and sex were added as covariates and 6 knots were used to fit the tradeSeq models. Functions associationTest, patternTest, earlyDETest, diffEndTest and conditionTest were then used to identify differential genes. As a large number of genes were found to be significantly differential we choose to focus on the most significant genes by filtering to those with a adjusted P-value of &lt; 1 x 10<span class="math inline">\(^{-16}\)</span>.</p>
</section>
</section>
</section>
<section id="supplemental-information-titles-and-legends" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Supplemental information titles and legends</h1>
<div id="suppfig-level1-markers" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Dotplot of markers for higher level celltypes">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-level1-markers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/level1_dotplot.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure S1: "><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/level1_dotplot.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-level1-markers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S1
</figcaption>
</figure>
</div>
<div id="suppfig-risk-level1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-risk-level1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/supp_fig_1_magma_risk.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure S2: A heatmap denoting cell type-specific risk enrichment via MAGMA with “” denoting nominal significance (P < 0.05), ”” significance after Bonferroni adjustment (P < 0.05) and ”” a higher degree of adjusted significance again (P < 0.01)."><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/supp_fig_1_magma_risk.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig" id="suppfig-risk-level1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S2: A heatmap denoting cell type-specific risk enrichment via MAGMA with “<em>” denoting nominal significance (P &lt; 0.05), ”<strong>” significance after Bonferroni adjustment (P &lt; 0.05) and ”</strong></em>” a higher degree of adjusted significance again (P &lt; 0.01).
</figcaption>
</figure>
</div>
<div id="suppfig-endomt-paper-markers" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Expression of EndoMT markers reported in the literature in the higher level celltypes.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-endomt-paper-markers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/supp_fig_2_endmt_markers.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-10" title="Figure S3: "><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/supp_fig_2_endmt_markers.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-endomt-paper-markers-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S3
</figcaption>
</figure>
</div>
<div id="suppfig-vascular-population-correlations" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Correlations between higher level celltypes per donor amoungst the vascular celltypes.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-vascular-population-correlations-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/donor-celltype-proportion-corrlations-level1.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Figure S4: "><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/donor-celltype-proportion-corrlations-level1.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-vascular-population-correlations-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S4
</figcaption>
</figure>
</div>
<div id="suppfig-parenchymal-prop-diff" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Proportional differences in AD via scProportionTest in the parenchymal fraction.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-parenchymal-prop-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/parenchymal_prop_scprop.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure S5: "><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/parenchymal_prop_scprop.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-parenchymal-prop-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S5
</figcaption>
</figure>
</div>
<div id="suppfig-endomt-pca" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="PCA of endothelial/EndoMT/Pericyte/SMC cells faceted by case and sex, colored by pseudotime.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-endomt-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/004_pseudotime/slingshot_pca_diagnosis-sex.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-13" title="Figure S6: "><embed src="05_figures/990_shared_figures/004_pseudotime/slingshot_pca_diagnosis-sex.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-endomt-pca-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S6
</figcaption>
</figure>
</div>
<div id="suppfig-endomt-tfs" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Log expression plots of transcription factors showing early pseudotime divergence across lineages for each condition and lineage respectively.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-endomt-tfs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/endomt-lineage-transcription-factors.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-14" title="Figure S7: "><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/endomt-lineage-transcription-factors.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-endomt-tfs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S7
</figcaption>
</figure>
</div>
<div id="suppfig-cellchat" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Cell-cell interactions via CellChat. (A) Heatmaps showing the differential number and strength of interactions of cell types between AD and controls. Red colours denote a decrease in AD whereas blue colours denote an increase. (B) A barplot showing the mean probability of significant ligand-receptor pairs across the vascular/parenchymal cell types that show differential interactions in A. Split into L-R pairs where the parenchymal cell types are the source of the ligand with the vasculature receiving and vice versa.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-cellchat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure_cellchat.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-15" title="Figure S8: "><embed src="05_figures/990_shared_figures/003_final_figures/figure_cellchat.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-cellchat-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S8
</figcaption>
</figure>
</div>
<div id="suppfig-adriskpseudotime" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-cap="Log expression plots of BACE2 and APOD for each condition and lineage respectively. APOD has higher expression across later pseudotime for controls in the EndoMT-PC lineage, but no difference for cases, whereas BACE2 has higher expression in cases for both lineages in the earlier pseudotime.">
<figure class="quarto-float quarto-float-suppfig figure">
<div aria-describedby="suppfig-adriskpseudotime-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/000_sup_figures/ad_risk_pseudotime.pdf" class="lightbox" data-gallery="quarto-lightbox-gallery-16" title="Figure S9: "><embed src="05_figures/990_shared_figures/003_final_figures/000_sup_figures/ad_risk_pseudotime.pdf" class="img-fluid"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-suppfig quarto-uncaptioned" id="suppfig-adriskpseudotime-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure S9
</figcaption>
</figure>
</div>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a>tbl_1 <span class="ot">&lt;-</span> donor_metadata <span class="sc">|&gt;</span></span>
<span id="cb239-2"><a href="#cb239-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb239-3"><a href="#cb239-3" aria-hidden="true" tabindex="-1"></a>    donor,</span>
<span id="cb239-4"><a href="#cb239-4" aria-hidden="true" tabindex="-1"></a>    sex,</span>
<span id="cb239-5"><a href="#cb239-5" aria-hidden="true" tabindex="-1"></a>    age,</span>
<span id="cb239-6"><a href="#cb239-6" aria-hidden="true" tabindex="-1"></a>    diagnosis,</span>
<span id="cb239-7"><a href="#cb239-7" aria-hidden="true" tabindex="-1"></a>    Braak_Tangle,</span>
<span id="cb239-8"><a href="#cb239-8" aria-hidden="true" tabindex="-1"></a>    pH_cerebellum,</span>
<span id="cb239-9"><a href="#cb239-9" aria-hidden="true" tabindex="-1"></a>    Ethnicity,</span>
<span id="cb239-10"><a href="#cb239-10" aria-hidden="true" tabindex="-1"></a>    apoe_status</span>
<span id="cb239-11"><a href="#cb239-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb239-12"><a href="#cb239-12" aria-hidden="true" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">|&gt;</span></span>
<span id="cb239-13"><a href="#cb239-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">donor =</span> <span class="fu">str_remove_all</span>(donor, <span class="st">"donor-"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb239-14"><a href="#cb239-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(donor) <span class="sc">|&gt;</span></span>
<span id="cb239-15"><a href="#cb239-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb239-16"><a href="#cb239-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">pH_cerebellum =</span> p_h_cerebellum,</span>
<span id="cb239-17"><a href="#cb239-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> sex,</span>
<span id="cb239-18"><a href="#cb239-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">Age =</span> age,</span>
<span id="cb239-19"><a href="#cb239-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">APOE =</span> apoe_status,</span>
<span id="cb239-20"><a href="#cb239-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">BRAAK =</span> braak_tangle,</span>
<span id="cb239-21"><a href="#cb239-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">Donor =</span> donor</span>
<span id="cb239-22"><a href="#cb239-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb239-23"><a href="#cb239-23" aria-hidden="true" tabindex="-1"></a><span class="co"># knitr::kable()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># eval: !expr knitr::is_html_output() | knitr::is_latex_output()</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a>tbl_1 <span class="sc">|&gt;</span></span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>  gt<span class="sc">::</span><span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div id="supptbl-donor-metadata" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-donor-metadata-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="co"># eval: !expr knitr::opts_knit$get("rmarkdown.pandoc.to") == "docx"</span></span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>tbl_1 <span class="sc">|&gt;</span></span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(BRAAK, pH_cerebellum)) <span class="sc">|&gt;</span></span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(Donor) <span class="sc">|&gt;</span></span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-5f9e68fc{}.cl-5f7cc8d2{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-5f81b356{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5f81b36a{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5f81df3e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5f81df52{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5f81df53{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5f81df5c{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5f81df84{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5f81df8e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-5f9e68fc"><thead><tr style="overflow-wrap:break-word;"><th class="cl-5f81df3e"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Donor</span></p></th><th class="cl-5f81df3e"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Sex</span></p></th><th class="cl-5f81df52"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">Age</span></p></th><th class="cl-5f81df3e"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">diagnosis</span></p></th><th class="cl-5f81df3e"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">ethnicity</span></p></th><th class="cl-5f81df3e"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">APOE</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">1</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">86</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">2/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">10</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">89</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">11</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">87</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">12</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">87</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">13</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">89</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">White (European)</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">14</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">90</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">15</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">89</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">16</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">88</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">4/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">17</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">95</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">18</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">95</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">19</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">94</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">White (European)</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">2/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">2</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">87</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Black Caribbean</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">21</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">71</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">22</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">90</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">23</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">74</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">24</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">89</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">4/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">25</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">48</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">26</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">98</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">27</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">81</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">2/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">28</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">88</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">82</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">30</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">92</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">31</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">70</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">32</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">73</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">33</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">89</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">34</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">71</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">35</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">77</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">36</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">81</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">37</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">85</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">38</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">93</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">4</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">81</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">4/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">40</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">101</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2"></span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">5</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">87</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">6</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">87</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">2/4</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">7</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">85</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">8</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">M</span></p></td><td class="cl-5f81df5c"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">85</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Case</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df53"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5f81df84"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">9</span></p></td><td class="cl-5f81df84"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">F</span></p></td><td class="cl-5f81df8e"><p class="cl-5f81b36a"><span class="cl-5f7cc8d2">89</span></p></td><td class="cl-5f81df84"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">Control</span></p></td><td class="cl-5f81df84"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">unknown</span></p></td><td class="cl-5f81df84"><p class="cl-5f81b356"><span class="cl-5f7cc8d2">3/3</span></p></td></tr></tbody></table></div>
</div>
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="co"># knitr::kable()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-donor-metadata-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S1: Clinical Information. Clinical characteristics of the cohort and samples included in the multi-omics analysis.
</figcaption>
</figure>
</div>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get deg tables</span></span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qload</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb243-3"><a href="#cb243-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"05_figures/990_shared_figures/003_final_figures"</span>,</span>
<span id="cb243-4"><a href="#cb243-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"001_data_for_plots/tradeseq_sig_degs.qs"</span></span>
<span id="cb243-5"><a href="#cb243-5" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div id="supptbl-tradeseq-degs-early" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-tradeseq-degs-early-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>top_early_genes <span class="sc">|&gt;</span></span>
<span id="cb244-2"><a href="#cb244-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, waldStat, df, padj, fcMedian, ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb244-3"><a href="#cb244-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> hgnc_symbol, <span class="st">`</span><span class="at">Ensembl Gene ID</span><span class="st">`</span> <span class="ot">=</span> ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb244-4"><a href="#cb244-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb244-5"><a href="#cb244-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb244-6"><a href="#cb244-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-5fcaf516{}.cl-5fc0f14c{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-5fc5472e{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5fc54742{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5fc5782a{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fc57834{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fc5783e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fc57866{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fc57884{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fc578a2{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-5fcaf516"><thead><tr style="overflow-wrap:break-word;"><th class="cl-5fc5782a"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">Gene</span></p></th><th class="cl-5fc57834"><p class="cl-5fc54742"><span class="cl-5fc0f14c">waldStat</span></p></th><th class="cl-5fc57834"><p class="cl-5fc54742"><span class="cl-5fc0f14c">df</span></p></th><th class="cl-5fc57834"><p class="cl-5fc54742"><span class="cl-5fc0f14c">padj</span></p></th><th class="cl-5fc57834"><p class="cl-5fc54742"><span class="cl-5fc0f14c">fcMedian</span></p></th><th class="cl-5fc5782a"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">Ensembl Gene ID</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">PID1</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">745.0242</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">0</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4.557631</span></p></td><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">ENSG00000153823</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">SPEG</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">369.6445</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">6</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">0</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4.451720</span></p></td><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">ENSG00000072195</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">MYOCD-AS1</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">338.4913</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">6</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">0</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4.421520</span></p></td><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">ENSG00000227274</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">MYH11</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">2,124.1836</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">5</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">0</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4.418305</span></p></td><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">ENSG00000133392</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">COLEC12</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">1,172.5259</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">0</span></p></td><td class="cl-5fc57866"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4.343724</span></p></td><td class="cl-5fc5783e"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">ENSG00000158270</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fc57884"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">CSDC2</span></p></td><td class="cl-5fc578a2"><p class="cl-5fc54742"><span class="cl-5fc0f14c">226.8687</span></p></td><td class="cl-5fc578a2"><p class="cl-5fc54742"><span class="cl-5fc0f14c">6</span></p></td><td class="cl-5fc578a2"><p class="cl-5fc54742"><span class="cl-5fc0f14c">0</span></p></td><td class="cl-5fc578a2"><p class="cl-5fc54742"><span class="cl-5fc0f14c">4.190530</span></p></td><td class="cl-5fc57884"><p class="cl-5fc5472e"><span class="cl-5fc0f14c">ENSG00000172346</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-tradeseq-degs-early-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S2: Significant (P &lt; <span class="math inline">\(1\times10^{-16}\)</span>) differential genes in early pseudotime identified via tradeSeq.
</figcaption>
</figure>
</div>
<div id="supptbl-tradeseq-degs-late" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-tradeseq-degs-late-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb245"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb245-1"><a href="#cb245-1" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>top_end_genes <span class="sc">|&gt;</span></span>
<span id="cb245-2"><a href="#cb245-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, waldStat, df, padj, logFC1_2, ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb245-3"><a href="#cb245-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> hgnc_symbol, <span class="st">`</span><span class="at">Ensembl Gene ID</span><span class="st">`</span> <span class="ot">=</span> ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb245-4"><a href="#cb245-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb245-5"><a href="#cb245-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-5fe6b03a{}.cl-5fdd1ee4{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-5fe12da4{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5fe12dae{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5fe15a04{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fe15a0e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fe15a18{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fe15a19{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fe15a22{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5fe15a23{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-5fe6b03a"><thead><tr style="overflow-wrap:break-word;"><th class="cl-5fe15a04"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">Gene</span></p></th><th class="cl-5fe15a0e"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">waldStat</span></p></th><th class="cl-5fe15a0e"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">df</span></p></th><th class="cl-5fe15a0e"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">padj</span></p></th><th class="cl-5fe15a0e"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">logFC1_2</span></p></th><th class="cl-5fe15a04"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">Ensembl Gene ID</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">MYH11</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">229.52501</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">1</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">0</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">29.21307</span></p></td><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ENSG00000133392</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ITGA8</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">158.45637</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">1</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">0</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">26.87878</span></p></td><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ENSG00000077943</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">MYOCD</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">129.96043</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">1</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">0</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">22.55155</span></p></td><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ENSG00000141052</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">LDB3</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">136.57470</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">1</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">0</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">22.44720</span></p></td><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ENSG00000122367</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">WNK2</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">94.02539</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">1</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">0</span></p></td><td class="cl-5fe15a19"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">21.51601</span></p></td><td class="cl-5fe15a18"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ENSG00000165238</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5fe15a22"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">SLIT3</span></p></td><td class="cl-5fe15a23"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">334.55814</span></p></td><td class="cl-5fe15a23"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">1</span></p></td><td class="cl-5fe15a23"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">0</span></p></td><td class="cl-5fe15a23"><p class="cl-5fe12dae"><span class="cl-5fdd1ee4">17.78159</span></p></td><td class="cl-5fe15a22"><p class="cl-5fe12da4"><span class="cl-5fdd1ee4">ENSG00000184347</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-tradeseq-degs-late-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S3: Significant (P &lt; <span class="math inline">\(1\times10^{-16}\)</span>) differential genes in late pseudotime identified via tradeSeq.
</figcaption>
</figure>
</div>
<div id="supptbl-tradeseq-degs-lin" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-tradeseq-degs-lin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>top_lin_genes <span class="sc">|&gt;</span></span>
<span id="cb246-2"><a href="#cb246-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, waldStat, df, padj, fcMedian, ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb246-3"><a href="#cb246-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> hgnc_symbol, <span class="st">`</span><span class="at">Ensembl Gene ID</span><span class="st">`</span> <span class="ot">=</span> ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb246-4"><a href="#cb246-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb246-5"><a href="#cb246-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-6001dc20{}.cl-5ff848d6{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-5ffc4df0{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5ffc4e22{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-5ffc7cbc{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5ffc7cd0{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5ffc7cda{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5ffc7d02{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5ffc7d0c{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-5ffc7d20{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-6001dc20"><thead><tr style="overflow-wrap:break-word;"><th class="cl-5ffc7cbc"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">Gene</span></p></th><th class="cl-5ffc7cd0"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">waldStat</span></p></th><th class="cl-5ffc7cd0"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">df</span></p></th><th class="cl-5ffc7cd0"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">padj</span></p></th><th class="cl-5ffc7cd0"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">fcMedian</span></p></th><th class="cl-5ffc7cbc"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">Ensembl Gene ID</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">PID1</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">745.0242</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">0</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4.557631</span></p></td><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">ENSG00000153823</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">SPEG</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">369.6445</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">6</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">0</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4.451720</span></p></td><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">ENSG00000072195</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">MYOCD-AS1</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">338.4913</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">6</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">0</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4.421520</span></p></td><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">ENSG00000227274</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">MYH11</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">2,124.1836</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">5</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">0</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4.418305</span></p></td><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">ENSG00000133392</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">COLEC12</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">1,172.5259</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">0</span></p></td><td class="cl-5ffc7d02"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4.343724</span></p></td><td class="cl-5ffc7cda"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">ENSG00000158270</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-5ffc7d0c"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">CSDC2</span></p></td><td class="cl-5ffc7d20"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">226.8687</span></p></td><td class="cl-5ffc7d20"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">6</span></p></td><td class="cl-5ffc7d20"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">0</span></p></td><td class="cl-5ffc7d20"><p class="cl-5ffc4e22"><span class="cl-5ff848d6">4.190530</span></p></td><td class="cl-5ffc7d0c"><p class="cl-5ffc4df0"><span class="cl-5ff848d6">ENSG00000172346</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-tradeseq-degs-lin-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S4: Significant (P &lt; <span class="math inline">\(1\times10^{-16}\)</span>) differential genes between conditions for each lineage and overall pseudotime identified via tradeSeq.
</figcaption>
</figure>
</div>
<div id="supptbl-tradeseq-degs-cond" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-tradeseq-degs-cond-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb247"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb247-1"><a href="#cb247-1" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>top_genes_condition <span class="sc">|&gt;</span></span>
<span id="cb247-2"><a href="#cb247-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(hgnc_symbol, lineage, waldStat, df, pvalue, ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb247-3"><a href="#cb247-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Gene =</span> hgnc_symbol, <span class="st">`</span><span class="at">Ensembl Gene ID</span><span class="st">`</span> <span class="ot">=</span> ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb247-4"><a href="#cb247-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb247-5"><a href="#cb247-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb247-6"><a href="#cb247-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-601e7e16{}.cl-6014c466{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-6018e2b2{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-6018e2d0{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-60190e0e{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-60190e18{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-60190e4a{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-60190e4b{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-60190e68{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-60190e72{width:0.75in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-601e7e16"><thead><tr style="overflow-wrap:break-word;"><th class="cl-60190e0e"><p class="cl-6018e2b2"><span class="cl-6014c466">Gene</span></p></th><th class="cl-60190e0e"><p class="cl-6018e2b2"><span class="cl-6014c466">lineage</span></p></th><th class="cl-60190e18"><p class="cl-6018e2d0"><span class="cl-6014c466">waldStat</span></p></th><th class="cl-60190e18"><p class="cl-6018e2d0"><span class="cl-6014c466">df</span></p></th><th class="cl-60190e18"><p class="cl-6018e2d0"><span class="cl-6014c466">pvalue</span></p></th><th class="cl-60190e0e"><p class="cl-6018e2b2"><span class="cl-6014c466">Ensembl Gene ID</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">TFPI</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">2</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">142.40119</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">6</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">0</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">ENSG00000003436</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">STAU2</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">2</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">170.24254</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">6</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">0</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">ENSG00000040341</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">RIPOR3</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">2</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">193.02834</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">5</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">0</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">ENSG00000042062</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">LTBP1</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">1</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">142.16388</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">6</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">0</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">ENSG00000049323</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">LTBP1</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">2</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">97.36938</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">6</span></p></td><td class="cl-60190e4b"><p class="cl-6018e2d0"><span class="cl-6014c466">0</span></p></td><td class="cl-60190e4a"><p class="cl-6018e2b2"><span class="cl-6014c466">ENSG00000049323</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-60190e68"><p class="cl-6018e2b2"><span class="cl-6014c466">PLEKHA5</span></p></td><td class="cl-60190e68"><p class="cl-6018e2b2"><span class="cl-6014c466">1</span></p></td><td class="cl-60190e72"><p class="cl-6018e2d0"><span class="cl-6014c466">88.86166</span></p></td><td class="cl-60190e72"><p class="cl-6018e2d0"><span class="cl-6014c466">6</span></p></td><td class="cl-60190e72"><p class="cl-6018e2d0"><span class="cl-6014c466">0</span></p></td><td class="cl-60190e68"><p class="cl-6018e2b2"><span class="cl-6014c466">ENSG00000052126</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-tradeseq-degs-cond-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S5: Significant (P &lt; <span class="math inline">\(1\times10^{-16}\)</span>) differential genes between lineage pseudotime identified via tradeSeq.
</figcaption>
</figure>
</div>
<div id="supptbl-mast-degs" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-mast-degs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>targets<span class="sc">::</span><span class="fu">tar_load</span>(translated_id)</span>
<span id="cb248-2"><a href="#cb248-2" aria-hidden="true" tabindex="-1"></a>mast_de <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb248-3"><a href="#cb248-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/990_processed_data/001_snrnaseq/13_mast_de/level2/mast_de_results_list.qs"</span></span>
<span id="cb248-4"><a href="#cb248-4" aria-hidden="true" tabindex="-1"></a>)) <span class="sc">|&gt;</span></span>
<span id="cb248-5"><a href="#cb248-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(rownames_to_column, <span class="at">var =</span> <span class="st">"ensembl"</span>) <span class="sc">|&gt;</span></span>
<span id="cb248-6"><a href="#cb248-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb248-7"><a href="#cb248-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(translated_id, <span class="at">by =</span> <span class="fu">join_by</span>(ensembl <span class="sc">==</span> ensembl_gene_id)) <span class="sc">|&gt;</span></span>
<span id="cb248-8"><a href="#cb248-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">|&gt;</span></span>
<span id="cb248-9"><a href="#cb248-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(ensembl, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb248-10"><a href="#cb248-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb248-11"><a href="#cb248-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb248-12"><a href="#cb248-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb248-13"><a href="#cb248-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltypes_abbrev =</span> <span class="fu">if_else</span>(</span>
<span id="cb248-14"><a href="#cb248-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">is.na</span>(celltypes_abbrev),</span>
<span id="cb248-15"><a href="#cb248-15" aria-hidden="true" tabindex="-1"></a>      celltype,</span>
<span id="cb248-16"><a href="#cb248-16" aria-hidden="true" tabindex="-1"></a>      celltypes_abbrev</span>
<span id="cb248-17"><a href="#cb248-17" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb248-18"><a href="#cb248-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">deg_direction =</span> <span class="fu">if_else</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Up Regulated"</span>, <span class="st">"Down Regulated"</span>)</span>
<span id="cb248-19"><a href="#cb248-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb248-20"><a href="#cb248-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> hgnc_symbol)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dplyr::left_join(list_rbind(map(qs::qread(here::here("03_data/990_processed_data/001_snrnaseq/13_mast_de/level2/mast_de_results_list.qs")), : Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 20 of `x` matches multiple rows in `y`.
ℹ Row 18480 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb250"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb250-1"><a href="#cb250-1" aria-hidden="true" tabindex="-1"></a>mast_de <span class="sc">|&gt;</span></span>
<span id="cb250-2"><a href="#cb250-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb250-3"><a href="#cb250-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb250-4"><a href="#cb250-4" aria-hidden="true" tabindex="-1"></a>    <span class="sc">-</span><span class="fu">c</span>(p_val, pct<span class="fl">.1</span><span class="sc">:</span>pct<span class="fl">.2</span>, celltype, entrezgene_id, deg_direction)</span>
<span id="cb250-5"><a href="#cb250-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb250-6"><a href="#cb250-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltypes_abbrev) <span class="sc">|&gt;</span></span>
<span id="cb250-7"><a href="#cb250-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">avg_log2FC =</span> <span class="fu">round</span>(avg_log2FC, <span class="dv">3</span>)) <span class="sc">|&gt;</span></span>
<span id="cb250-8"><a href="#cb250-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb250-9"><a href="#cb250-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb250-10"><a href="#cb250-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, avg_log2FC, p_val_adj, <span class="fu">everything</span>()) <span class="sc">|&gt;</span></span>
<span id="cb250-11"><a href="#cb250-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb250-12"><a href="#cb250-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb250-13"><a href="#cb250-13" aria-hidden="true" tabindex="-1"></a>  flextable<span class="sc">::</span><span class="fu">set_formatter</span>(<span class="at">p_val_adj =</span> <span class="cf">function</span>(x) {</span>
<span id="cb250-14"><a href="#cb250-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">formatC</span>(x, <span class="at">format =</span> <span class="st">"e"</span>, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb250-15"><a href="#cb250-15" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">|&gt;</span></span>
<span id="cb250-16"><a href="#cb250-16" aria-hidden="true" tabindex="-1"></a>  flextable<span class="sc">::</span><span class="fu">set_header_labels</span>(</span>
<span id="cb250-17"><a href="#cb250-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene =</span> <span class="st">"Gene"</span>,</span>
<span id="cb250-18"><a href="#cb250-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">avg_log2FC =</span> <span class="st">"TEMP_FC"</span>, <span class="co"># Temporary placeholder</span></span>
<span id="cb250-19"><a href="#cb250-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_val_adj =</span> <span class="st">"Adjusted P-value"</span>,</span>
<span id="cb250-20"><a href="#cb250-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltypes_abbrev =</span> <span class="st">"Celltype"</span>,</span>
<span id="cb250-21"><a href="#cb250-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">ensembl =</span> <span class="st">"Ensembl ID"</span></span>
<span id="cb250-22"><a href="#cb250-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb250-23"><a href="#cb250-23" aria-hidden="true" tabindex="-1"></a>  flextable<span class="sc">::</span><span class="fu">compose</span>(</span>
<span id="cb250-24"><a href="#cb250-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">j =</span> <span class="st">"avg_log2FC"</span>,</span>
<span id="cb250-25"><a href="#cb250-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">part =</span> <span class="st">"header"</span>,</span>
<span id="cb250-26"><a href="#cb250-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">value =</span> <span class="fu">as_paragraph</span>(<span class="st">"Average log"</span>, <span class="fu">as_sub</span>(<span class="st">"2"</span>), <span class="st">"(FoldChange)"</span>)</span>
<span id="cb250-27"><a href="#cb250-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb250-28"><a href="#cb250-28" aria-hidden="true" tabindex="-1"></a>  flextable<span class="sc">::</span><span class="fu">autofit</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-63bd6b72{}.cl-63b54dd4{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-63b54de8{font-family:'DejaVu Sans';font-size:6.6pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;position: relative;top:3.3pt;}.cl-63b8c5fe{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-63b8c612{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-63b8f0ce{width:1.122in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f0d8{width:2.276in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f0e2{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f0ec{width:1.781in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f0ed{width:1.092in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f100{width:1.122in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f101{width:2.276in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f10a{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f10b{width:1.781in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f114{width:1.092in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f115{width:1.122in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f11e{width:2.276in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f128{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f132{width:1.781in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-63b8f13c{width:1.092in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-63bd6b72"><thead><tr style="overflow-wrap:break-word;"><th class="cl-63b8f0ce"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">Gene</span></p></th><th class="cl-63b8f0d8"><p class="cl-63b8c612"><span class="cl-63b54dd4">Average log</span><span class="cl-63b54de8">2</span><span class="cl-63b54dd4">(FoldChange)</span></p></th><th class="cl-63b8f0e2"><p class="cl-63b8c612"><span class="cl-63b54dd4">Adjusted P-value</span></p></th><th class="cl-63b8f0ec"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">Ensembl ID</span></p></th><th class="cl-63b8f0ed"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">Celltype</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-63b8f100"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">XIST</span></p></td><td class="cl-63b8f101"><p class="cl-63b8c612"><span class="cl-63b54dd4">-0.326</span></p></td><td class="cl-63b8f10a"><p class="cl-63b8c612"><span class="cl-63b54dd4">8.262e-108</span></p></td><td class="cl-63b8f10b"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ENSG00000229807</span></p></td><td class="cl-63b8f114"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">M-Pericyte</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-63b8f100"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">LINC00486</span></p></td><td class="cl-63b8f101"><p class="cl-63b8c612"><span class="cl-63b54dd4">0.185</span></p></td><td class="cl-63b8f10a"><p class="cl-63b8c612"><span class="cl-63b54dd4">3.212e-100</span></p></td><td class="cl-63b8f10b"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ENSG00000230876</span></p></td><td class="cl-63b8f114"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">M-Pericyte</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-63b8f100"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">FKBP5</span></p></td><td class="cl-63b8f101"><p class="cl-63b8c612"><span class="cl-63b54dd4">-0.182</span></p></td><td class="cl-63b8f10a"><p class="cl-63b8c612"><span class="cl-63b54dd4">5.426e-85</span></p></td><td class="cl-63b8f10b"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ENSG00000096060</span></p></td><td class="cl-63b8f114"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">M-Pericyte</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-63b8f100"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">SLC6A1</span></p></td><td class="cl-63b8f101"><p class="cl-63b8c612"><span class="cl-63b54dd4">-0.757</span></p></td><td class="cl-63b8f10a"><p class="cl-63b8c612"><span class="cl-63b54dd4">2.111e-60</span></p></td><td class="cl-63b8f10b"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ENSG00000157103</span></p></td><td class="cl-63b8f114"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">M-Pericyte</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-63b8f100"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ADAMTS9</span></p></td><td class="cl-63b8f101"><p class="cl-63b8c612"><span class="cl-63b54dd4">1.156</span></p></td><td class="cl-63b8f10a"><p class="cl-63b8c612"><span class="cl-63b54dd4">6.461e-56</span></p></td><td class="cl-63b8f10b"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ENSG00000163638</span></p></td><td class="cl-63b8f114"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">M-Pericyte</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-63b8f115"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">FAM13A</span></p></td><td class="cl-63b8f11e"><p class="cl-63b8c612"><span class="cl-63b54dd4">0.884</span></p></td><td class="cl-63b8f128"><p class="cl-63b8c612"><span class="cl-63b54dd4">2.548e-53</span></p></td><td class="cl-63b8f132"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">ENSG00000138640</span></p></td><td class="cl-63b8f13c"><p class="cl-63b8c5fe"><span class="cl-63b54dd4">M-Pericyte</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-mast-degs-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S6: Significant (Bonferroni adjusted P &lt; 0.05) differential genes between AD and controls per celltype via MAST. Age and sex were included as covariates.
</figcaption>
</figure>
</div>
<div id="supptbl-mast-go-pathways" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-mast-go-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis"</span></span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb251-4"><a href="#cb251-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-5"><a href="#cb251-5" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb251-6"><a href="#cb251-6" aria-hidden="true" tabindex="-1"></a>  file,</span>
<span id="cb251-7"><a href="#cb251-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"go-hypergeometic_subtypes_deg_direction_mast_level2.qs"</span></span>
<span id="cb251-8"><a href="#cb251-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb251-9"><a href="#cb251-9" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(file)</span>
<span id="cb251-10"><a href="#cb251-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-11"><a href="#cb251-11" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> <span class="fu">map2</span>(</span>
<span id="cb251-12"><a href="#cb251-12" aria-hidden="true" tabindex="-1"></a>  result_list,</span>
<span id="cb251-13"><a href="#cb251-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(result_list),</span>
<span id="cb251-14"><a href="#cb251-14" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> tibble<span class="sc">::</span><span class="fu">as_tibble</span>(.x) <span class="sc">|&gt;</span></span>
<span id="cb251-15"><a href="#cb251-15" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> .y)</span>
<span id="cb251-16"><a href="#cb251-16" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb251-17"><a href="#cb251-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span>
<span id="cb251-18"><a href="#cb251-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-19"><a href="#cb251-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove terms that only have one gene</span></span>
<span id="cb251-20"><a href="#cb251-20" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> result_list <span class="sc">|&gt;</span></span>
<span id="cb251-21"><a href="#cb251-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Count <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> ONTOLOGY <span class="sc">==</span> <span class="st">"BP"</span>) <span class="sc">|&gt;</span></span>
<span id="cb251-22"><a href="#cb251-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb251-23"><a href="#cb251-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">deg_direction =</span> <span class="fu">if_else</span>(<span class="fu">grepl</span>(<span class="st">"_Down"</span>, celltype), <span class="st">"Down"</span>, <span class="st">"Up"</span>)</span>
<span id="cb251-24"><a href="#cb251-24" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb251-25"><a href="#cb251-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb251-26"><a href="#cb251-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltype =</span> <span class="fu">str_remove_all</span>(celltype, <span class="st">"_Down"</span>),</span>
<span id="cb251-27"><a href="#cb251-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltype =</span> <span class="fu">str_remove_all</span>(celltype, <span class="st">"_Up"</span>)</span>
<span id="cb251-28"><a href="#cb251-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb251-29"><a href="#cb251-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb251-30"><a href="#cb251-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb251-31"><a href="#cb251-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltypes_abbrev =</span> <span class="fu">if_else</span>(</span>
<span id="cb251-32"><a href="#cb251-32" aria-hidden="true" tabindex="-1"></a>      <span class="fu">is.na</span>(celltypes_abbrev),</span>
<span id="cb251-33"><a href="#cb251-33" aria-hidden="true" tabindex="-1"></a>      celltype,</span>
<span id="cb251-34"><a href="#cb251-34" aria-hidden="true" tabindex="-1"></a>      celltypes_abbrev</span>
<span id="cb251-35"><a href="#cb251-35" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb251-36"><a href="#cb251-36" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb251-37"><a href="#cb251-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-38"><a href="#cb251-38" aria-hidden="true" tabindex="-1"></a>result_list <span class="sc">|&gt;</span></span>
<span id="cb251-39"><a href="#cb251-39" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb251-40"><a href="#cb251-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb251-41"><a href="#cb251-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">zScore =</span> <span class="fu">round</span>(zScore, <span class="dv">3</span>),</span>
<span id="cb251-42"><a href="#cb251-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">qvalue =</span> <span class="fu">round</span>(qvalue, <span class="dv">3</span>),</span>
<span id="cb251-43"><a href="#cb251-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">FoldEnrichment =</span> <span class="fu">round</span>(FoldEnrichment, <span class="dv">3</span>),</span>
<span id="cb251-44"><a href="#cb251-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">RichFactor =</span> <span class="fu">round</span>(RichFactor, <span class="dv">3</span>)</span>
<span id="cb251-45"><a href="#cb251-45" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb251-46"><a href="#cb251-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(pvalue, ONTOLOGY, celltype, geneID)) <span class="sc">|&gt;</span></span>
<span id="cb251-47"><a href="#cb251-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltypes_abbrev, deg_direction) <span class="sc">|&gt;</span></span>
<span id="cb251-48"><a href="#cb251-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># dplyr::mutate(avg_log2FC = round(avg_log2FC, 3)) |&gt;</span></span>
<span id="cb251-49"><a href="#cb251-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># dplyr::slice_sample(n = 10) |&gt;</span></span>
<span id="cb251-50"><a href="#cb251-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ID, Description, p.adjust, <span class="fu">everything</span>()) <span class="sc">|&gt;</span></span>
<span id="cb251-51"><a href="#cb251-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb251-52"><a href="#cb251-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb251-53"><a href="#cb251-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb251-54"><a href="#cb251-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_header_labels</span>(</span>
<span id="cb251-55"><a href="#cb251-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="st">"Adjusted P-value"</span></span>
<span id="cb251-56"><a href="#cb251-56" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb251-57"><a href="#cb251-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">autofit</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-6be7ccc0{}.cl-6bdeada2{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-6be2755e{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-6be27572{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-6be29d40{width:1.256in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d54{width:4.197in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d5e{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d68{width:1.084in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d69{width:1.114in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d7c{width:1.096in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d7d{width:1.485in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d90{width:0.797in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d91{width:0.799in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29d9a{width:0.739in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29da4{width:1.322in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29dae{width:1.58in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29de0{width:1.256in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29dea{width:4.197in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29df4{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29dfe{width:1.084in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29dff{width:1.114in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e08{width:1.096in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e09{width:1.485in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e12{width:0.797in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e1c{width:0.799in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e26{width:0.739in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e30{width:1.322in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e31{width:1.58in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e3a{width:1.256in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e44{width:4.197in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e4e{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e4f{width:1.084in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e58{width:1.114in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e62{width:1.096in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e6c{width:1.485in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e6d{width:0.797in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e80{width:0.799in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e81{width:0.739in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e8a{width:1.322in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6be29e8b{width:1.58in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-6be7ccc0"><thead><tr style="overflow-wrap:break-word;"><th class="cl-6be29d40"><p class="cl-6be2755e"><span class="cl-6bdeada2">ID</span></p></th><th class="cl-6be29d54"><p class="cl-6be2755e"><span class="cl-6bdeada2">Description</span></p></th><th class="cl-6be29d5e"><p class="cl-6be27572"><span class="cl-6bdeada2">Adjusted P-value</span></p></th><th class="cl-6be29d68"><p class="cl-6be2755e"><span class="cl-6bdeada2">GeneRatio</span></p></th><th class="cl-6be29d69"><p class="cl-6be2755e"><span class="cl-6bdeada2">BgRatio</span></p></th><th class="cl-6be29d7c"><p class="cl-6be27572"><span class="cl-6bdeada2">RichFactor</span></p></th><th class="cl-6be29d7d"><p class="cl-6be27572"><span class="cl-6bdeada2">FoldEnrichment</span></p></th><th class="cl-6be29d90"><p class="cl-6be27572"><span class="cl-6bdeada2">zScore</span></p></th><th class="cl-6be29d91"><p class="cl-6be27572"><span class="cl-6bdeada2">qvalue</span></p></th><th class="cl-6be29d9a"><p class="cl-6be27572"><span class="cl-6bdeada2">Count</span></p></th><th class="cl-6be29da4"><p class="cl-6be2755e"><span class="cl-6bdeada2">deg_direction</span></p></th><th class="cl-6be29dae"><p class="cl-6be2755e"><span class="cl-6bdeada2">celltypes_abbrev</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-6be29de0"><p class="cl-6be2755e"><span class="cl-6bdeada2">GO:0042026</span></p></td><td class="cl-6be29dea"><p class="cl-6be2755e"><span class="cl-6bdeada2">protein refolding</span></p></td><td class="cl-6be29df4"><p class="cl-6be27572"><span class="cl-6bdeada2">0.009909798</span></p></td><td class="cl-6be29dfe"><p class="cl-6be2755e"><span class="cl-6bdeada2">6/256</span></p></td><td class="cl-6be29dff"><p class="cl-6be2755e"><span class="cl-6bdeada2">26/15717</span></p></td><td class="cl-6be29e08"><p class="cl-6be27572"><span class="cl-6bdeada2">0.231</span></p></td><td class="cl-6be29e09"><p class="cl-6be27572"><span class="cl-6bdeada2">14.168</span></p></td><td class="cl-6be29e12"><p class="cl-6be27572"><span class="cl-6bdeada2">8.647</span></p></td><td class="cl-6be29e1c"><p class="cl-6be27572"><span class="cl-6bdeada2">0.009</span></p></td><td class="cl-6be29e26"><p class="cl-6be27572"><span class="cl-6bdeada2">6</span></p></td><td class="cl-6be29e30"><p class="cl-6be2755e"><span class="cl-6bdeada2">Down</span></p></td><td class="cl-6be29e31"><p class="cl-6be2755e"><span class="cl-6bdeada2">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6be29de0"><p class="cl-6be2755e"><span class="cl-6bdeada2">GO:1990266</span></p></td><td class="cl-6be29dea"><p class="cl-6be2755e"><span class="cl-6bdeada2">neutrophil migration</span></p></td><td class="cl-6be29df4"><p class="cl-6be27572"><span class="cl-6bdeada2">0.017221729</span></p></td><td class="cl-6be29dfe"><p class="cl-6be2755e"><span class="cl-6bdeada2">10/256</span></p></td><td class="cl-6be29dff"><p class="cl-6be2755e"><span class="cl-6bdeada2">110/15717</span></p></td><td class="cl-6be29e08"><p class="cl-6be27572"><span class="cl-6bdeada2">0.091</span></p></td><td class="cl-6be29e09"><p class="cl-6be27572"><span class="cl-6bdeada2">5.581</span></p></td><td class="cl-6be29e12"><p class="cl-6be27572"><span class="cl-6bdeada2">6.204</span></p></td><td class="cl-6be29e1c"><p class="cl-6be27572"><span class="cl-6bdeada2">0.015</span></p></td><td class="cl-6be29e26"><p class="cl-6be27572"><span class="cl-6bdeada2">10</span></p></td><td class="cl-6be29e30"><p class="cl-6be2755e"><span class="cl-6bdeada2">Down</span></p></td><td class="cl-6be29e31"><p class="cl-6be2755e"><span class="cl-6bdeada2">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6be29de0"><p class="cl-6be2755e"><span class="cl-6bdeada2">GO:0044403</span></p></td><td class="cl-6be29dea"><p class="cl-6be2755e"><span class="cl-6bdeada2">biological process involved in symbiotic interaction</span></p></td><td class="cl-6be29df4"><p class="cl-6be27572"><span class="cl-6bdeada2">0.017221729</span></p></td><td class="cl-6be29dfe"><p class="cl-6be2755e"><span class="cl-6bdeada2">16/256</span></p></td><td class="cl-6be29dff"><p class="cl-6be2755e"><span class="cl-6bdeada2">282/15717</span></p></td><td class="cl-6be29e08"><p class="cl-6be27572"><span class="cl-6bdeada2">0.057</span></p></td><td class="cl-6be29e09"><p class="cl-6be27572"><span class="cl-6bdeada2">3.483</span></p></td><td class="cl-6be29e12"><p class="cl-6be27572"><span class="cl-6bdeada2">5.415</span></p></td><td class="cl-6be29e1c"><p class="cl-6be27572"><span class="cl-6bdeada2">0.015</span></p></td><td class="cl-6be29e26"><p class="cl-6be27572"><span class="cl-6bdeada2">16</span></p></td><td class="cl-6be29e30"><p class="cl-6be2755e"><span class="cl-6bdeada2">Down</span></p></td><td class="cl-6be29e31"><p class="cl-6be2755e"><span class="cl-6bdeada2">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6be29de0"><p class="cl-6be2755e"><span class="cl-6bdeada2">GO:0034612</span></p></td><td class="cl-6be29dea"><p class="cl-6be2755e"><span class="cl-6bdeada2">response to tumor necrosis factor</span></p></td><td class="cl-6be29df4"><p class="cl-6be27572"><span class="cl-6bdeada2">0.019364357</span></p></td><td class="cl-6be29dfe"><p class="cl-6be2755e"><span class="cl-6bdeada2">14/256</span></p></td><td class="cl-6be29dff"><p class="cl-6be2755e"><span class="cl-6bdeada2">229/15717</span></p></td><td class="cl-6be29e08"><p class="cl-6be27572"><span class="cl-6bdeada2">0.061</span></p></td><td class="cl-6be29e09"><p class="cl-6be27572"><span class="cl-6bdeada2">3.753</span></p></td><td class="cl-6be29e12"><p class="cl-6be27572"><span class="cl-6bdeada2">5.401</span></p></td><td class="cl-6be29e1c"><p class="cl-6be27572"><span class="cl-6bdeada2">0.017</span></p></td><td class="cl-6be29e26"><p class="cl-6be27572"><span class="cl-6bdeada2">14</span></p></td><td class="cl-6be29e30"><p class="cl-6be2755e"><span class="cl-6bdeada2">Down</span></p></td><td class="cl-6be29e31"><p class="cl-6be2755e"><span class="cl-6bdeada2">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6be29de0"><p class="cl-6be2755e"><span class="cl-6bdeada2">GO:0071356</span></p></td><td class="cl-6be29dea"><p class="cl-6be2755e"><span class="cl-6bdeada2">cellular response to tumor necrosis factor</span></p></td><td class="cl-6be29df4"><p class="cl-6be27572"><span class="cl-6bdeada2">0.022747991</span></p></td><td class="cl-6be29dfe"><p class="cl-6be2755e"><span class="cl-6bdeada2">13/256</span></p></td><td class="cl-6be29dff"><p class="cl-6be2755e"><span class="cl-6bdeada2">209/15717</span></p></td><td class="cl-6be29e08"><p class="cl-6be27572"><span class="cl-6bdeada2">0.062</span></p></td><td class="cl-6be29e09"><p class="cl-6be27572"><span class="cl-6bdeada2">3.819</span></p></td><td class="cl-6be29e12"><p class="cl-6be27572"><span class="cl-6bdeada2">5.279</span></p></td><td class="cl-6be29e1c"><p class="cl-6be27572"><span class="cl-6bdeada2">0.020</span></p></td><td class="cl-6be29e26"><p class="cl-6be27572"><span class="cl-6bdeada2">13</span></p></td><td class="cl-6be29e30"><p class="cl-6be2755e"><span class="cl-6bdeada2">Down</span></p></td><td class="cl-6be29e31"><p class="cl-6be2755e"><span class="cl-6bdeada2">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6be29e3a"><p class="cl-6be2755e"><span class="cl-6bdeada2">GO:0006458</span></p></td><td class="cl-6be29e44"><p class="cl-6be2755e"><span class="cl-6bdeada2">'de novo' protein folding</span></p></td><td class="cl-6be29e4e"><p class="cl-6be27572"><span class="cl-6bdeada2">0.022747991</span></p></td><td class="cl-6be29e4f"><p class="cl-6be2755e"><span class="cl-6bdeada2">6/256</span></p></td><td class="cl-6be29e58"><p class="cl-6be2755e"><span class="cl-6bdeada2">40/15717</span></p></td><td class="cl-6be29e62"><p class="cl-6be27572"><span class="cl-6bdeada2">0.150</span></p></td><td class="cl-6be29e6c"><p class="cl-6be27572"><span class="cl-6bdeada2">9.209</span></p></td><td class="cl-6be29e6d"><p class="cl-6be27572"><span class="cl-6bdeada2">6.689</span></p></td><td class="cl-6be29e80"><p class="cl-6be27572"><span class="cl-6bdeada2">0.020</span></p></td><td class="cl-6be29e81"><p class="cl-6be27572"><span class="cl-6bdeada2">6</span></p></td><td class="cl-6be29e8a"><p class="cl-6be2755e"><span class="cl-6bdeada2">Down</span></p></td><td class="cl-6be29e8b"><p class="cl-6be2755e"><span class="cl-6bdeada2">Arterial</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-mast-go-pathways-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S7: Significant (adjusted P &lt; 0.05) hypergeomtric GO pathways amongst the significantly differenetially expressed genes from MAST per celltype.
</figcaption>
</figure>
</div>
<div id="supptbl-mast-go-pathways-filter" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-mast-go-pathways-filter-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb252-2"><a href="#cb252-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis"</span></span>
<span id="cb252-3"><a href="#cb252-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb252-4"><a href="#cb252-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-5"><a href="#cb252-5" aria-hidden="true" tabindex="-1"></a><span class="co"># file for logfc GO IDs</span></span>
<span id="cb252-6"><a href="#cb252-6" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb252-7"><a href="#cb252-7" aria-hidden="true" tabindex="-1"></a>  file,</span>
<span id="cb252-8"><a href="#cb252-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"go-hypergeometic_subtypes_deg_direction_mast_level2_logfc_filter.qs"</span></span>
<span id="cb252-9"><a href="#cb252-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb252-10"><a href="#cb252-10" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(file)</span>
<span id="cb252-11"><a href="#cb252-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-12"><a href="#cb252-12" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> <span class="fu">map2</span>(</span>
<span id="cb252-13"><a href="#cb252-13" aria-hidden="true" tabindex="-1"></a>  result_list,</span>
<span id="cb252-14"><a href="#cb252-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(result_list),</span>
<span id="cb252-15"><a href="#cb252-15" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> tibble<span class="sc">::</span><span class="fu">as_tibble</span>(.x) <span class="sc">|&gt;</span></span>
<span id="cb252-16"><a href="#cb252-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> .y)</span>
<span id="cb252-17"><a href="#cb252-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb252-18"><a href="#cb252-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span>
<span id="cb252-19"><a href="#cb252-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-20"><a href="#cb252-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove terms that only have one gene</span></span>
<span id="cb252-21"><a href="#cb252-21" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> result_list <span class="sc">|&gt;</span></span>
<span id="cb252-22"><a href="#cb252-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Count <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> ONTOLOGY <span class="sc">==</span> <span class="st">"BP"</span>) <span class="sc">|&gt;</span></span>
<span id="cb252-23"><a href="#cb252-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb252-24"><a href="#cb252-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">deg_direction =</span> <span class="fu">if_else</span>(<span class="fu">grepl</span>(<span class="st">"_Down"</span>, celltype), <span class="st">"Down"</span>, <span class="st">"Up"</span>)</span>
<span id="cb252-25"><a href="#cb252-25" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb252-26"><a href="#cb252-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb252-27"><a href="#cb252-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltype =</span> <span class="fu">str_remove_all</span>(celltype, <span class="st">"_Down"</span>),</span>
<span id="cb252-28"><a href="#cb252-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltype =</span> <span class="fu">str_remove_all</span>(celltype, <span class="st">"_Up"</span>)</span>
<span id="cb252-29"><a href="#cb252-29" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb252-30"><a href="#cb252-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb252-31"><a href="#cb252-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb252-32"><a href="#cb252-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltypes_abbrev =</span> <span class="fu">if_else</span>(</span>
<span id="cb252-33"><a href="#cb252-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">is.na</span>(celltypes_abbrev),</span>
<span id="cb252-34"><a href="#cb252-34" aria-hidden="true" tabindex="-1"></a>      celltype,</span>
<span id="cb252-35"><a href="#cb252-35" aria-hidden="true" tabindex="-1"></a>      celltypes_abbrev</span>
<span id="cb252-36"><a href="#cb252-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb252-37"><a href="#cb252-37" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb252-38"><a href="#cb252-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-39"><a href="#cb252-39" aria-hidden="true" tabindex="-1"></a>result_list <span class="sc">|&gt;</span></span>
<span id="cb252-40"><a href="#cb252-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb252-41"><a href="#cb252-41" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb252-42"><a href="#cb252-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">zScore =</span> <span class="fu">round</span>(zScore, <span class="dv">3</span>),</span>
<span id="cb252-43"><a href="#cb252-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">qvalue =</span> <span class="fu">round</span>(qvalue, <span class="dv">3</span>),</span>
<span id="cb252-44"><a href="#cb252-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">FoldEnrichment =</span> <span class="fu">round</span>(FoldEnrichment, <span class="dv">3</span>),</span>
<span id="cb252-45"><a href="#cb252-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">RichFactor =</span> <span class="fu">round</span>(RichFactor, <span class="dv">3</span>)</span>
<span id="cb252-46"><a href="#cb252-46" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb252-47"><a href="#cb252-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(pvalue, ONTOLOGY, celltype, geneID)) <span class="sc">|&gt;</span></span>
<span id="cb252-48"><a href="#cb252-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltypes_abbrev, deg_direction) <span class="sc">|&gt;</span></span>
<span id="cb252-49"><a href="#cb252-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># dplyr::mutate(avg_log2FC = round(avg_log2FC, 3)) |&gt;</span></span>
<span id="cb252-50"><a href="#cb252-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># dplyr::slice_sample(n = 10) |&gt;</span></span>
<span id="cb252-51"><a href="#cb252-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ID, Description, p.adjust, <span class="fu">everything</span>()) <span class="sc">|&gt;</span></span>
<span id="cb252-52"><a href="#cb252-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb252-53"><a href="#cb252-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb252-54"><a href="#cb252-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb252-55"><a href="#cb252-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_header_labels</span>(</span>
<span id="cb252-56"><a href="#cb252-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">p.adjust =</span> <span class="st">"Adjusted P-value"</span></span>
<span id="cb252-57"><a href="#cb252-57" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb252-58"><a href="#cb252-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">autofit</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-6d98ab98{}.cl-6d900308{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-6d9397fc{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-6d939806{margin:0;text-align:right;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-6d93bbc4{width:1.256in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bbd8{width:7.877in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bbe2{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bbe3{width:1.084in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bbec{width:1.114in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bbf6{width:1.096in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bbf7{width:1.485in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc00{width:0.82in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc0a{width:0.799in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc14{width:0.739in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc1e{width:1.322in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc1f{width:1.58in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc28{width:1.256in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc32{width:7.877in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc33{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc46{width:1.084in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc47{width:1.114in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc50{width:1.096in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc51{width:1.485in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc5a{width:0.82in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc5b{width:0.799in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc5c{width:0.739in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc64{width:1.322in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc6e{width:1.58in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc6f{width:1.256in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc82{width:7.877in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc83{width:1.57in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc84{width:1.084in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc8c{width:1.114in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bc96{width:1.096in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bca0{width:1.485in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bca1{width:0.82in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bcaa{width:0.799in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bcab{width:0.739in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bcb4{width:1.322in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6d93bcbe{width:1.58in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-6d98ab98"><thead><tr style="overflow-wrap:break-word;"><th class="cl-6d93bbc4"><p class="cl-6d9397fc"><span class="cl-6d900308">ID</span></p></th><th class="cl-6d93bbd8"><p class="cl-6d9397fc"><span class="cl-6d900308">Description</span></p></th><th class="cl-6d93bbe2"><p class="cl-6d939806"><span class="cl-6d900308">Adjusted P-value</span></p></th><th class="cl-6d93bbe3"><p class="cl-6d9397fc"><span class="cl-6d900308">GeneRatio</span></p></th><th class="cl-6d93bbec"><p class="cl-6d9397fc"><span class="cl-6d900308">BgRatio</span></p></th><th class="cl-6d93bbf6"><p class="cl-6d939806"><span class="cl-6d900308">RichFactor</span></p></th><th class="cl-6d93bbf7"><p class="cl-6d939806"><span class="cl-6d900308">FoldEnrichment</span></p></th><th class="cl-6d93bc00"><p class="cl-6d939806"><span class="cl-6d900308">zScore</span></p></th><th class="cl-6d93bc0a"><p class="cl-6d939806"><span class="cl-6d900308">qvalue</span></p></th><th class="cl-6d93bc14"><p class="cl-6d939806"><span class="cl-6d900308">Count</span></p></th><th class="cl-6d93bc1e"><p class="cl-6d9397fc"><span class="cl-6d900308">deg_direction</span></p></th><th class="cl-6d93bc1f"><p class="cl-6d9397fc"><span class="cl-6d900308">celltypes_abbrev</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-6d93bc28"><p class="cl-6d9397fc"><span class="cl-6d900308">GO:0070424</span></p></td><td class="cl-6d93bc32"><p class="cl-6d9397fc"><span class="cl-6d900308">regulation of nucleotide-binding domain, leucine rich repeat containing receptor signaling pathway</span></p></td><td class="cl-6d93bc33"><p class="cl-6d939806"><span class="cl-6d900308">0.0001144341</span></p></td><td class="cl-6d93bc46"><p class="cl-6d9397fc"><span class="cl-6d900308">3/14</span></p></td><td class="cl-6d93bc47"><p class="cl-6d9397fc"><span class="cl-6d900308">14/15717</span></p></td><td class="cl-6d93bc50"><p class="cl-6d939806"><span class="cl-6d900308">0.214</span></p></td><td class="cl-6d93bc51"><p class="cl-6d939806"><span class="cl-6d900308">240.566</span></p></td><td class="cl-6d93bc5a"><p class="cl-6d939806"><span class="cl-6d900308">26.776</span></p></td><td class="cl-6d93bc5b"><p class="cl-6d939806"><span class="cl-6d900308">0</span></p></td><td class="cl-6d93bc5c"><p class="cl-6d939806"><span class="cl-6d900308">3</span></p></td><td class="cl-6d93bc64"><p class="cl-6d9397fc"><span class="cl-6d900308">Down</span></p></td><td class="cl-6d93bc6e"><p class="cl-6d9397fc"><span class="cl-6d900308">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6d93bc28"><p class="cl-6d9397fc"><span class="cl-6d900308">GO:0070431</span></p></td><td class="cl-6d93bc32"><p class="cl-6d9397fc"><span class="cl-6d900308">nucleotide-binding oligomerization domain containing 2 signaling pathway</span></p></td><td class="cl-6d93bc33"><p class="cl-6d939806"><span class="cl-6d900308">0.0001279975</span></p></td><td class="cl-6d93bc46"><p class="cl-6d9397fc"><span class="cl-6d900308">3/14</span></p></td><td class="cl-6d93bc47"><p class="cl-6d9397fc"><span class="cl-6d900308">18/15717</span></p></td><td class="cl-6d93bc50"><p class="cl-6d939806"><span class="cl-6d900308">0.167</span></p></td><td class="cl-6d93bc51"><p class="cl-6d939806"><span class="cl-6d900308">187.107</span></p></td><td class="cl-6d93bc5a"><p class="cl-6d939806"><span class="cl-6d900308">23.589</span></p></td><td class="cl-6d93bc5b"><p class="cl-6d939806"><span class="cl-6d900308">0</span></p></td><td class="cl-6d93bc5c"><p class="cl-6d939806"><span class="cl-6d900308">3</span></p></td><td class="cl-6d93bc64"><p class="cl-6d9397fc"><span class="cl-6d900308">Down</span></p></td><td class="cl-6d93bc6e"><p class="cl-6d9397fc"><span class="cl-6d900308">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6d93bc28"><p class="cl-6d9397fc"><span class="cl-6d900308">GO:0009408</span></p></td><td class="cl-6d93bc32"><p class="cl-6d9397fc"><span class="cl-6d900308">response to heat</span></p></td><td class="cl-6d93bc33"><p class="cl-6d939806"><span class="cl-6d900308">0.0002030625</span></p></td><td class="cl-6d93bc46"><p class="cl-6d9397fc"><span class="cl-6d900308">4/14</span></p></td><td class="cl-6d93bc47"><p class="cl-6d9397fc"><span class="cl-6d900308">96/15717</span></p></td><td class="cl-6d93bc50"><p class="cl-6d939806"><span class="cl-6d900308">0.042</span></p></td><td class="cl-6d93bc51"><p class="cl-6d939806"><span class="cl-6d900308">46.777</span></p></td><td class="cl-6d93bc5a"><p class="cl-6d939806"><span class="cl-6d900308">13.433</span></p></td><td class="cl-6d93bc5b"><p class="cl-6d939806"><span class="cl-6d900308">0</span></p></td><td class="cl-6d93bc5c"><p class="cl-6d939806"><span class="cl-6d900308">4</span></p></td><td class="cl-6d93bc64"><p class="cl-6d9397fc"><span class="cl-6d900308">Down</span></p></td><td class="cl-6d93bc6e"><p class="cl-6d9397fc"><span class="cl-6d900308">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6d93bc28"><p class="cl-6d9397fc"><span class="cl-6d900308">GO:0042026</span></p></td><td class="cl-6d93bc32"><p class="cl-6d9397fc"><span class="cl-6d900308">protein refolding</span></p></td><td class="cl-6d93bc33"><p class="cl-6d939806"><span class="cl-6d900308">0.0002030625</span></p></td><td class="cl-6d93bc46"><p class="cl-6d9397fc"><span class="cl-6d900308">3/14</span></p></td><td class="cl-6d93bc47"><p class="cl-6d9397fc"><span class="cl-6d900308">26/15717</span></p></td><td class="cl-6d93bc50"><p class="cl-6d939806"><span class="cl-6d900308">0.115</span></p></td><td class="cl-6d93bc51"><p class="cl-6d939806"><span class="cl-6d900308">129.536</span></p></td><td class="cl-6d93bc5a"><p class="cl-6d939806"><span class="cl-6d900308">19.585</span></p></td><td class="cl-6d93bc5b"><p class="cl-6d939806"><span class="cl-6d900308">0</span></p></td><td class="cl-6d93bc5c"><p class="cl-6d939806"><span class="cl-6d900308">3</span></p></td><td class="cl-6d93bc64"><p class="cl-6d9397fc"><span class="cl-6d900308">Down</span></p></td><td class="cl-6d93bc6e"><p class="cl-6d9397fc"><span class="cl-6d900308">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6d93bc28"><p class="cl-6d9397fc"><span class="cl-6d900308">GO:0035872</span></p></td><td class="cl-6d93bc32"><p class="cl-6d9397fc"><span class="cl-6d900308">nucleotide-binding domain, leucine rich repeat containing receptor signaling pathway</span></p></td><td class="cl-6d93bc33"><p class="cl-6d939806"><span class="cl-6d900308">0.0002044721</span></p></td><td class="cl-6d93bc46"><p class="cl-6d9397fc"><span class="cl-6d900308">3/14</span></p></td><td class="cl-6d93bc47"><p class="cl-6d9397fc"><span class="cl-6d900308">28/15717</span></p></td><td class="cl-6d93bc50"><p class="cl-6d939806"><span class="cl-6d900308">0.107</span></p></td><td class="cl-6d93bc51"><p class="cl-6d939806"><span class="cl-6d900308">120.283</span></p></td><td class="cl-6d93bc5a"><p class="cl-6d939806"><span class="cl-6d900308">18.863</span></p></td><td class="cl-6d93bc5b"><p class="cl-6d939806"><span class="cl-6d900308">0</span></p></td><td class="cl-6d93bc5c"><p class="cl-6d939806"><span class="cl-6d900308">3</span></p></td><td class="cl-6d93bc64"><p class="cl-6d9397fc"><span class="cl-6d900308">Down</span></p></td><td class="cl-6d93bc6e"><p class="cl-6d9397fc"><span class="cl-6d900308">Arterial</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6d93bc6f"><p class="cl-6d9397fc"><span class="cl-6d900308">GO:0009266</span></p></td><td class="cl-6d93bc82"><p class="cl-6d9397fc"><span class="cl-6d900308">response to temperature stimulus</span></p></td><td class="cl-6d93bc83"><p class="cl-6d939806"><span class="cl-6d900308">0.0009640527</span></p></td><td class="cl-6d93bc84"><p class="cl-6d9397fc"><span class="cl-6d900308">4/14</span></p></td><td class="cl-6d93bc8c"><p class="cl-6d9397fc"><span class="cl-6d900308">163/15717</span></p></td><td class="cl-6d93bc96"><p class="cl-6d939806"><span class="cl-6d900308">0.025</span></p></td><td class="cl-6d93bca0"><p class="cl-6d939806"><span class="cl-6d900308">27.550</span></p></td><td class="cl-6d93bca1"><p class="cl-6d939806"><span class="cl-6d900308">10.174</span></p></td><td class="cl-6d93bcaa"><p class="cl-6d939806"><span class="cl-6d900308">0</span></p></td><td class="cl-6d93bcab"><p class="cl-6d939806"><span class="cl-6d900308">4</span></p></td><td class="cl-6d93bcb4"><p class="cl-6d9397fc"><span class="cl-6d900308">Down</span></p></td><td class="cl-6d93bcbe"><p class="cl-6d9397fc"><span class="cl-6d900308">Arterial</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-mast-go-pathways-filter-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S8: Significant (adjusted P &lt; 0.05) hypergeomtric GO pathways amongst the significant differenetially expressed genes with a log2 foldchage &gt; 1 from MAST per celltype.
</figcaption>
</figure>
</div>
<div id="supptbl-r-packages" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-supptbl figure">
<div aria-describedby="supptbl-r-packages-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get session information</span></span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>session_info <span class="ot">&lt;-</span> sessioninfo<span class="sc">::</span><span class="fu">session_info</span>()</span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-4"><a href="#cb253-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract package information</span></span>
<span id="cb253-5"><a href="#cb253-5" aria-hidden="true" tabindex="-1"></a>package_info <span class="ot">&lt;-</span> session_info<span class="sc">$</span>packages <span class="sc">%&gt;%</span></span>
<span id="cb253-6"><a href="#cb253-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(package, loadedversion, source) <span class="sc">%&gt;%</span></span>
<span id="cb253-7"><a href="#cb253-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(package)</span>
<span id="cb253-8"><a href="#cb253-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-9"><a href="#cb253-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to a data frame</span></span>
<span id="cb253-10"><a href="#cb253-10" aria-hidden="true" tabindex="-1"></a>package_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(package_info)</span>
<span id="cb253-11"><a href="#cb253-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Print table</span></span>
<span id="cb253-12"><a href="#cb253-12" aria-hidden="true" tabindex="-1"></a>package_df <span class="sc">|&gt;</span></span>
<span id="cb253-13"><a href="#cb253-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb253-14"><a href="#cb253-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(source) <span class="sc">|&gt;</span></span>
<span id="cb253-15"><a href="#cb253-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flextable</span>(<span class="at">use_labels =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb253-16"><a href="#cb253-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">autofit</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="tabwid"><style>.cl-6ed5304e{}.cl-6eba82e4{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-6ec25dd4{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-6ec2a7a8{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a7c6{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a7d0{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a7d1{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a7ee{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a816{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a848{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a849{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a866{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a87a{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a884{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a898{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a8a2{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a8b6{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a8b7{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a8de{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a8e8{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a8fc{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a906{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a910{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a924{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a938{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a94c{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a960{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a99c{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a9a6{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a9a7{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a9b0{width:3.612in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a9b1{width:1.403in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-6ec2a9c4{width:7.498in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style><table data-quarto-disable-processing="true" class="cl-6ed5304e"><thead><tr style="overflow-wrap:break-word;"><th class="cl-6ec2a7a8"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">package</span></p></th><th class="cl-6ec2a7c6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">loadedversion</span></p></th><th class="cl-6ec2a7d0"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">source</span></p></th></tr></thead><tbody><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">AnnotationDbi</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.70.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">AnnotationFilter</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.32.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">AnnotationHub</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.16.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BSgenome</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.76.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BSgenome.Hsapiens.1000genomes.hs37d5</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.99.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BSgenome.Hsapiens.NCBI.GRCh38</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3.1000</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Biobase</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.68.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocFileCache</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.16.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocGenerics</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.54.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocIO</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.18.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocManager</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.30.25</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocNeighbors</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocParallel</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.42.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocSingular</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.24.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">BiocVersion</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.21.1</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Biostrings</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.76.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CellChat</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (jinworks/CellChat@346fb615aaefa3c1a2830ec2fe3a933c5f723c83)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ComplexHeatmap</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.24.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DBI</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DESeq2</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.48.1</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DEoptimR</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1-3-1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DOSE</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.2.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DT</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.33</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DelayedArray</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.34.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DelayedMatrixStats</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.30.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DirichletReg</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.7-1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DoubletFinder</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (chris-mcginnis-ucsf/DoubletFinder@1b244d8f0d54b4b1cb4365639931bbb16f01e1cd)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">DropletUtils</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.28.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8de"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">EWCE</span></p></td><td class="cl-6ec2a8e8"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.15.1</span></p></td><td class="cl-6ec2a8fc"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (NathanSkene/EWCE@a86e84aad55971be2c56a1ffa73e53b26d21b6c4)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">EnsDb.Hsapiens.v79</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.99.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ExperimentHub</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.16.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">FNN</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.4.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Formula</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2-5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GEOquery</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.76.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a906"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GO.db</span></p></td><td class="cl-6ec2a910"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.21.0</span></p></td><td class="cl-6ec2a924"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GOSemSim</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.34.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GenomeInfoDb</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.44.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a906"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GenomeInfoDbData</span></p></td><td class="cl-6ec2a910"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.14</span></p></td><td class="cl-6ec2a924"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GenomicAlignments</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.44.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GenomicFeatures</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.60.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GenomicFiles</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.44.1</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GenomicRanges</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.60.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GetoptLong</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.5</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">GlobalOptions</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">HDF5Array</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.36.0</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">HGNChelper</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.8.15</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">IRanges</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.42.0</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">KEGGREST</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.48.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">KEGGgraph</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.68.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">KernSmooth</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.23-26</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.2)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">MASS</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">7.3-65</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.3)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">MAST</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.33.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Matrix</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.7-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.3)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">MatrixGenerics</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.20.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">MungeSumstats</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.16.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">NLP</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3-2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">NMF</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.28</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">PCAtools</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.20.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Polychrome</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ProtGenerics</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.40.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">R.methodsS3</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">R.oo</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.27.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">R.utils</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.13.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">R6</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.6.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RANN</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.6.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RApiSerialize</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RColorBrewer</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RCurl</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.98-1.17</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RNOmni</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ROCR</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0-11</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSQLite</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3.11</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSpectra</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.16-2</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Rcpp</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.14</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RcppAnnoy</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.0.22</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RcppHNSW</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RcppParallel</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">5.1.10</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Rdpack</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.6.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ResidualMatrix</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.18.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Rgraphviz</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.52.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Rhdf5lib</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.30.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Rsamtools</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.24.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Rtsne</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.17</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">S4Arrays</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">S4Vectors</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.46.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SC3</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.36.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SNPlocs.Hsapiens.dbSNP155.GRCh37</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.99.24</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SNPlocs.Hsapiens.dbSNP155.GRCh38</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.99.24</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ScaledMatrix</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.16.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Seurat</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">5.3.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SeuratObject</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">5.1.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SingleCellExperiment</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.30.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SnowballC</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.7.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SparseArray</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SparseM</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.84-2</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">SummarizedExperiment</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.38.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">UCSC.utils</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">UpSetR</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">VariantAnnotation</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.54.1</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">WebGestaltR</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.6</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">WriteXLS</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">6.8.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">XML</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.99-0.18</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">XVector</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.48.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8de"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">YesSiR</span></p></td><td class="cl-6ec2a8e8"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.0.0.9000</span></p></td><td class="cl-6ec2a8fc"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (Sebastien-Le/YesSiR@217eb43c6f3f3aec556a515e17671d1588475dee)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">abind</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4-8</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">alabaster.base</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">alabaster.matrix</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">alabaster.ranges</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">alabaster.sce</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">alabaster.schemas</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">alabaster.se</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">apcluster</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.13</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ape</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">5.8-1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">aplot</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.5</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">askpass</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">assertthat</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">babelgene</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">22.9</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">backports</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">base64enc</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">base64url</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">batchelor</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.24.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">beachmat</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.24.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">beeswarm</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">bib2df</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (ropensci/bib2df@4ff27d739a4b000ab71f6ef1e5a9c983d99ff9ed)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">biomaRt</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.64.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">bit</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.6.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">bit64</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.6.0-1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">bitops</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0-9</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">blob</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">bluster</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.18.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">boot</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3-31</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.2)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">broom</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.8</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">bslib</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.9.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">cachem</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">callr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.7.6</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">car</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">carData</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.0-5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">cellity</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.36.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">cellranger</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">circlize</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.16</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">class</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">7.3-23</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.2)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">cli</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.6.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">clue</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3-66</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">cluster</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1.8.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.3)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">clusterProfiler</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.16.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">coda</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.19-4.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">codetools</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2-19</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.2.2)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">colorspace</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1-1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">corrgram</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.14</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">cowplot</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">crayon</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">crosstalk</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">curl</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">6.2.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">data.table</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.17.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">dbplyr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.5.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">deldir</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0-4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">dichromat</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0-0.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">digest</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.37</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">doParallel</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.17</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">doRNG</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.6.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">dotCall64</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">downlit</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">dplyr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">dqrng</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">e1071</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.7-16</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">edgeR</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.6.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">emmeans</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.11.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">english</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2-6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">enrichR</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">enrichplot</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.28.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ensembldb</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.32.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">estimability</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">europepmc</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">evaluate</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ewceData</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.16.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">farver</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fastDummies</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.7.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fastmap</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fastmatch</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1-6</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fftw</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0-9</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fgsea</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.34.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fields</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">16.3.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">filelock</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fitdistrplus</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2-2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">flextable</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.9.8</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fontBitstreamVera</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fontLiberation</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fontquiver</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">forcats</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">foreach</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">formattable</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">fs</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.6.6</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">furrr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">future</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.49.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">future.apply</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.11.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gage</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.58.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gageData</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.46.0</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gdtools</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">generics</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggalluvial</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.12.5</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggbeeswarm</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.7.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggdendro</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggfun</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.8</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggnetwork</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.5.13</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggplot2</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.5.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggplotify</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggpubr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggrepel</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.9.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggridges</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.5.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggsci</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggsignif</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggtangle</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.0.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggtree</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.16.0</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggvenn</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.10</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ggwordcloud</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">globals</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.18.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">glue</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">goftest</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2-3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gprofiler2</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">graph</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.86.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gridBase</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4-7</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gridExtra</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gridGraphics</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.5-1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gridtext</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.5</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">grr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.9.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gson</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gt</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.0</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gtable</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">gypsum</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">h5mread</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.1</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">haven</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.5.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">here</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">hms</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">homologene</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.68.19.3.27</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">htmltools</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.5.8.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">htmlwidgets</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.6.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">httpuv</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.6.16</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">httr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.7</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">httr2</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">humaniformat</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ica</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ieugwasr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">igraph</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">irlba</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3.5.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">iterators</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.14</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">janeaustenr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">janitor</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">jquerylib</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">jsonlite</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8de"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">kBET</span></p></td><td class="cl-6ec2a8e8"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.99.6</span></p></td><td class="cl-6ec2a8fc"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (theislab/kBET@afc5f431bcbefd73267acc066a0f2e4eaa10a355)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">kableExtra</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">knitr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.50</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">labelled</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.14.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">later</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">lattice</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.22-5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.3.1)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">lazyeval</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">leaflet</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">lifecycle</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">limma</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.64.1</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">listenv</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.9.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">lme4</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1-37</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">lmtest</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.9-40</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">locfit</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5-9.12</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">lubridate</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.9.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">magrittr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">mapproj</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.12</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">maps</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.4.2.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">matrixStats</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">maxLik</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5-2.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">memoise</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">metapod</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.16.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">mime</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.13</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">miniUI</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">minqa</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.8</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">miscTools</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6-28</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">monocle3</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3.7</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (cole-trapnell-lab/monocle3@98402ed0c10cac020524bebbb9300614a799f6d1)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">mvoutlier</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">mvtnorm</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">network</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.19.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">nlme</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-168</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.3)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">nloptr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">officer</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.10</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">openssl</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">org.Hs.eg.db</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.21.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">org.Mm.eg.db</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.21.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">orthogene</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.14.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">paletteer</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.6.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pals</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.10</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">parallelly</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.44.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">patchwork</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pathview</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.48.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pbapply</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.7-2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pcaPP</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0-5</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pheatmap</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.12</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pillar</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.10.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">pkgconfig</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">plotly</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.10.4.9000</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (ropensci/plotly@57fe5c009bb4f22f65474c7334ac072bf60b8775)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">plotrix</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.8-4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">plyr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.9</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">png</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1-8</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">polyclip</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.10-7</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">preprocessCore</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.70.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">presto</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (immunogenomics/presto@7636b3d0465c468c35853f82f1717d3a64b3c8f6)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">prettyunits</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">processx</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.8.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">progress</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">progressr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.15.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">promises</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">proxy</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4-27</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ps</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.9.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">purrr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">qs</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.27.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">qusage</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.42.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">qvalue</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.40.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">ragg</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rappdirs</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rbibutils</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">readr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">readxl</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">reformulas</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">registry</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.5-1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rematch2</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rentrez</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">reshape</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.8.9</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">reshape2</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">restfulr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.0.15</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">reticulate</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.42.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rhdf5</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.52.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rhdf5filters</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.20.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rjson</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.23</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rlang</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.1.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rliger</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rmarkdown</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.29</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rngtools</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">robustbase</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.99-4-1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rprojroot</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.0.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rrcov</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.7-7</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rrvgo</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.20.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rstatix</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.7.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rstudioapi</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.17.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rsvd</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">rtracklayer</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.68.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sandwich</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sass</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.10</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scFlow</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.7.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (neurogenomics/scFlow@065bc13dacee707a0437fb65e006647660fc13c7)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scProportionTest</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.0.0.9000</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (rpolicastro/scProportionTest@37a04900be1c991428da15af7f9aa7b0ad84661e)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a938"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scRNAseq</span></p></td><td class="cl-6ec2a94c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.22.0</span></p></td><td class="cl-6ec2a960"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scales</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scater</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.36.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scattermore</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scatterplot3d</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3-44</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scran</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.36.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sctransform</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">scuttle</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.18.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">secretbase</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sessioninfo</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sgeostat</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0-27</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">shape</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.6.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">shiny</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.10.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">simona</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.6.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">slam</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.1-55</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sna</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.8</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">snakecase</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.11.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sp</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2-0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spam</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.11-1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">sparseMatrixStats</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.20.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.data</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.explore</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.4-3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.geom</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.4-1</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.random</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.4-1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.sparse</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-0</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.univar</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">spatstat.utils</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.1-4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">speckle</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">splitstackshape</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.4.8</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">statmod</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">statnet.common</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">4.11.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">stringfish</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.16.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">stringi</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8.7</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">stringr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">survival</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.8-3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.2)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">svglite</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.2.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">switchde</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.34.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">systemfonts</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">targets</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.11.3</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tensor</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">textshaping</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.0.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">threejs</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.4</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tibble</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.2.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tidyr</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tidyselect</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2.1</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tidytext</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.2</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tidytree</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.6</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">timechange</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tm</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.7-16</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tokenizers</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a848"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">topGO</span></p></td><td class="cl-6ec2a849"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.59.0</span></p></td><td class="cl-6ec2a866"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a7d1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">treeio</span></p></td><td class="cl-6ec2a7ee"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.32.0</span></p></td><td class="cl-6ec2a816"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Bioconductor 3.21 (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a99c"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">treemap</span></p></td><td class="cl-6ec2a9a6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.4-4</span></p></td><td class="cl-6ec2a9a7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">triebeard</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">tzdb</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.5.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">umap</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.10.0</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">urltools</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.7.3</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">uuid</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.2-1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">uwot</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.3.9000</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">Github (jlmelville/uwot@4c9d188e84d84907019cada8c04c299ee607bcf6)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">vctrs</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.4.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">vipor</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.7</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">viridis</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.6.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">viridisLite</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">vroom</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.6.5</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">wesanderson</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.3.7</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">CRAN (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">whisker</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.4.1</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">withr</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">3.0.2</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">wordcloud</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.6</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">xfun</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.52</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">xml2</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.3.8</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a8a2"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">xtable</span></p></td><td class="cl-6ec2a8b6"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8-4</span></p></td><td class="cl-6ec2a8b7"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">yaml</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3.10</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">yulab.utils</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">0.2.0</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a87a"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">zip</span></p></td><td class="cl-6ec2a884"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">2.3.3</span></p></td><td class="cl-6ec2a898"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr><tr style="overflow-wrap:break-word;"><td class="cl-6ec2a9b0"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">zoo</span></p></td><td class="cl-6ec2a9b1"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">1.8-14</span></p></td><td class="cl-6ec2a9c4"><p class="cl-6ec25dd4"><span class="cl-6eba82e4">RSPM (R 4.5.0)</span></p></td></tr></tbody></table></div>
</div>
</div></div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-supptbl" id="supptbl-r-packages-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table S9: R packages used
</figcaption>
</figure>
</div>
<section id="references" class="level2 unnumbered" data-number="5.1">
<h2 class="unnumbered anchored" data-number="5.1" data-anchor-id="references"><span class="header-section-number">5.1</span> References</h2>
<div id="refs" class="references csl-bib-body" role="list">
<div id="ref-cai2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Cai, Z., Qiao, P.-F., Wan, C.-Q., Cai, M., Zhou, N.-K., and Li, Q. (2018). Role of blood-brain barrier in alzheimer<span>’</span>s disease. Journal of Alzheimer’s Disease <em>63</em>, 1223–1234. <a href="https://doi.org/10.3233/jad-180098">https://doi.org/10.3233/jad-180098</a>.</div>
</div>
<div id="ref-nehra2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Nehra, G., Bauer, B., and Hartz, A.M.S. (2022). Blood-brain barrier leakage in Alzheimer<span>’</span>s disease: From discovery to clinical relevance. Pharmacology &amp; Therapeutics <em>234</em>, 108119. <a href="https://doi.org/10.1016/j.pharmthera.2022.108119">https://doi.org/10.1016/j.pharmthera.2022.108119</a>.</div>
</div>
<div id="ref-Iadecola2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Iadecola, C. (2017). The Neurovascular Unit Coming of Age: A Journey through Neurovascular Coupling in Health and Disease. Neuron <em>96</em>, 17–42. <a href="https://doi.org/10.1016/j.neuron.2017.07.030">https://doi.org/10.1016/j.neuron.2017.07.030</a>.</div>
</div>
<div id="ref-Zhang2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Su, Q., Guo, J.-H., Zhang, Y.-L., Wang, J., and Zhang, Z.-N. (2022). The relationship between amyloid-beta and brain capillary endothelial cells in Alzheimer<span>’</span>s disease. Neural Regeneration Research <em>17</em>, 2355. <a href="https://doi.org/10.4103/1673-5374.335829">https://doi.org/10.4103/1673-5374.335829</a>.</div>
</div>
<div id="ref-Daneman2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Daneman, R., and Prat, A. (2015). The Blood<span></span>Brain Barrier. Cold Spring Harbor Perspectives in Biology <em>7</em>, a020412. <a href="https://doi.org/10.1101/cshperspect.a020412">https://doi.org/10.1101/cshperspect.a020412</a>.</div>
</div>
<div id="ref-Amersfoort2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Amersfoort, J., Eelen, G., and Carmeliet, P. (2022). Immunomodulation by endothelial cells <span></span> partnering up with the immune system? Nature Reviews Immunology <em>22</em>, 576–588. <a href="https://doi.org/10.1038/s41577-022-00694-4">https://doi.org/10.1038/s41577-022-00694-4</a>.</div>
</div>
<div id="ref-Brown2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Brown, L.S., Foster, C.G., Courtney, J.-M., King, N.E., Howells, D.W., and Sutherland, B.A. (2019). Pericytes and neurovascular function in the healthy and diseased brain. Frontiers in Cellular Neuroscience <em>13</em>. <a href="https://doi.org/10.3389/fncel.2019.00282">https://doi.org/10.3389/fncel.2019.00282</a>.</div>
</div>
<div id="ref-Procter2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Procter, T.V., Williams, A., and Montagne, A. (2021). Interplay between Brain Pericytes and Endothelial Cells in Dementia. The American Journal of Pathology <em>191</em>, 1917–1931. <a href="https://doi.org/10.1016/j.ajpath.2021.07.003">https://doi.org/10.1016/j.ajpath.2021.07.003</a>.</div>
</div>
<div id="ref-Nehra2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Nehra, G., Bauer, B., and Hartz, A.M.S. (2022). Blood-brain barrier leakage in Alzheimer<span>’</span>s disease: From discovery to clinical relevance. Pharmacology <span>&amp;</span> Therapeutics <em>234</em>, 108119. <a href="https://doi.org/10.1016/j.pharmthera.2022.108119">https://doi.org/10.1016/j.pharmthera.2022.108119</a>.</div>
</div>
<div id="ref-Korte2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Korte, N., Nortley, R., and Attwell, D. (2020). Cerebral blood flow decrease as an early pathological mechanism in Alzheimer’s disease. Acta Neuropathologica <em>140</em>, 793–810. <a href="https://doi.org/10.1007/s00401-020-02215-w">https://doi.org/10.1007/s00401-020-02215-w</a>.</div>
</div>
<div id="ref-mcquade2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">11. </div><div class="csl-right-inline">McQuade, A., and Blurton-Jones, M. (2019). Microglia in Alzheimer’s Disease: Exploring How Genetics and Phenotype Influence Risk. Journal of Molecular Biology <em>431</em>, 1805–1817. <a href="https://doi.org/10.1016/j.jmb.2019.01.045">https://doi.org/10.1016/j.jmb.2019.01.045</a>.</div>
</div>
<div id="ref-yang2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">12. </div><div class="csl-right-inline">Yang, H.-S., Teng, L., Kang, D., Menon, V., Ge, T., Finucane, H.K., Schultz, A.P., Properzi, M., Klein, H.-U., Chibnik, L.B., et al. (2023). Cell-type-specific Alzheimer<span>’</span>s disease polygenic risk scores are associated with distinct disease processes in Alzheimer<span>’</span>s disease. Nature Communications <em>14</em>. <a href="https://doi.org/10.1038/s41467-023-43132-2">https://doi.org/10.1038/s41467-023-43132-2</a>.</div>
</div>
<div id="ref-Keller2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">13. </div><div class="csl-right-inline">Keller, D., Erö, C., and Markram, H. (2018). Cell densities in the mouse brain: A systematic review. Frontiers in Neuroanatomy <em>12</em>. <a href="https://doi.org/10.3389/fnana.2018.00083">https://doi.org/10.3389/fnana.2018.00083</a>.</div>
</div>
<div id="ref-Mathys2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">14. </div><div class="csl-right-inline">Mathys, H., Davila-Velderrain, J., Peng, Z., Gao, F., Mohammadi, S., Young, J.Z., Menon, M., He, L., Abdurrob, F., Jiang, X., et al. (2019). Single-cell transcriptomic analysis of Alzheimer<span>’</span>s disease. Nature <em>570</em>, 332–337. <a href="https://doi.org/10.1038/s41586-019-1195-2">https://doi.org/10.1038/s41586-019-1195-2</a>.</div>
</div>
<div id="ref-Grubman2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">15. </div><div class="csl-right-inline">Grubman, A., Chew, G., Ouyang, J.F., Sun, G., Choo, X.Y., McLean, C., Simmons, R.K., Buckberry, S., Vargas-Landin, D.B., Poppe, D., et al. (2019). A single-cell atlas of entorhinal cortex from individuals with Alzheimer<span>’</span>s disease reveals cell-type-specific gene expression regulation. Nature Neuroscience <em>22</em>, 2087–2097. <a href="https://doi.org/10.1038/s41593-019-0539-4">https://doi.org/10.1038/s41593-019-0539-4</a>.</div>
</div>
<div id="ref-yang2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">16. </div><div class="csl-right-inline">Yang, A.C., Vest, R.T., Kern, F., Lee, D.P., Agam, M., Maat, C.A., Losada, P.M., Chen, M.B., Schaum, N., Khoury, N., et al. (2022). A human brain vascular atlas reveals diverse mediators of Alzheimer’s risk. Nature <em>603</em>, 885–892. <a href="https://doi.org/10.1038/s41586-021-04369-3">https://doi.org/10.1038/s41586-021-04369-3</a>.</div>
</div>
<div id="ref-Tsartsalis2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">17. </div><div class="csl-right-inline">Tsartsalis, S., Sleven, H., Fancy, N., Wessely, F., Smith, A.M., Willumsen, N., Cheung, T.K.D., Rokicki, M.J., Chau, V., Ifie, E., et al. (2024). A single nuclear transcriptomic characterisation of mechanisms responsible for impaired angiogenesis and blood-brain barrier function in Alzheimer<span>’</span>s disease. Nature Communications <em>15</em>. <a href="https://doi.org/10.1038/s41467-024-46630-z">https://doi.org/10.1038/s41467-024-46630-z</a>.</div>
</div>
<div id="ref-tsartsalis2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">18. </div><div class="csl-right-inline">Tsartsalis, S., Sleven, H., Fancy, N., Wessely, F., Smith, A.M., Willumsen, N., Cheung, T.K.D., Rokicki, M.J., Chau, V., Ifie, E., et al. (2024). A single nuclear transcriptomic characterisation of mechanisms responsible for impaired angiogenesis and blood-brain barrier function in Alzheimer<span>’</span>s disease. Nature Communications <em>15</em>, 2243. <a href="https://doi.org/10.1038/s41467-024-46630-z">https://doi.org/10.1038/s41467-024-46630-z</a>.</div>
</div>
<div id="ref-mathys2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">19. </div><div class="csl-right-inline">Mathys, H., Peng, Z., Boix, C.A., Victor, M.B., Leary, N., Babu, S., Abdelhady, G., Jiang, X., Ng, A.P., Ghafari, K., et al. (2023). Single-cell atlas reveals correlates of high cognitive function, dementia, and resilience to Alzheimer’s disease pathology. Cell <em>186</em>, 4365–4385.e27. <a href="https://doi.org/10.1016/j.cell.2023.08.039">https://doi.org/10.1016/j.cell.2023.08.039</a>.</div>
</div>
<div id="ref-bellenguez2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">20. </div><div class="csl-right-inline">Bellenguez, C., Küçükali, F., Jansen, I.E., Kleineidam, L., Moreno-Grau, S., Amin, N., Naj, A.C., Campos-Martin, R., Grenier-Boley, B., Andrade, V., et al. (2022). New insights into the genetic etiology of Alzheimer<span>’</span>s disease and related dementias. Nature Genetics <em>54</em>, 412–436. <a href="https://doi.org/10.1038/s41588-022-01024-z">https://doi.org/10.1038/s41588-022-01024-z</a>.</div>
</div>
<div id="ref-bravi2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">21. </div><div class="csl-right-inline">Bravi, L., Malinverno, M., Pisati, F., Rudini, N., Cuttano, R., Pallini, R., Martini, M., Larocca, L.M., Locatelli, M., Levi, V., et al. (2016). Endothelial Cells Lining Sporadic Cerebral Cavernous Malformation Cavernomas Undergo Endothelial-to-Mesenchymal Transition. Stroke <em>47</em>, 886–890. <a href="https://doi.org/10.1161/strokeaha.115.011867">https://doi.org/10.1161/strokeaha.115.011867</a>.</div>
</div>
<div id="ref-shoemaker2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">22. </div><div class="csl-right-inline">Shoemaker, L.D., McCormick, A.K., Allen, B.M., and Chang, S.D. (2020). Evidence for endothelial<span>-</span>to<span>-</span>mesenchymal transition in human brain arteriovenous malformations. Clinical and Translational Medicine <em>10</em>. <a href="https://doi.org/10.1002/ctm2.99">https://doi.org/10.1002/ctm2.99</a>.</div>
</div>
<div id="ref-wälchli2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">23. </div><div class="csl-right-inline">Wälchli, T., Ghobrial, M., Schwab, M., Takada, S., Zhong, H., Suntharalingham, S., Vetiska, S., Gonzalez, D.R., Wu, R., Rehrauer, H., et al. (2024). Single-cell atlas of the human brain vasculature across development, adulthood and disease. Nature <em>632</em>, 603–613. <a href="https://doi.org/10.1038/s41586-024-07493-y">https://doi.org/10.1038/s41586-024-07493-y</a>.</div>
</div>
<div id="ref-Miller2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">24. </div><div class="csl-right-inline">Miller, S.A., Policastro, R.A., Sriramkumar, S., Lai, T., Huntington, T.D., Ladaika, C.A., Kim, D., Hao, C., Zentner, G.E., and O’Hagan, H.M. (2021). LSD1 and Aberrant DNA Methylation Mediate Persistence of Enteroendocrine Progenitors That Support <span><em>BRAF</em></span>-Mutant Colorectal Cancer. Cancer Research <em>81</em>, 3791–3805. <a href="https://doi.org/10.1158/0008-5472.can-20-3562">https://doi.org/10.1158/0008-5472.can-20-3562</a>.</div>
</div>
<div id="ref-shah2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">25. </div><div class="csl-right-inline">Shah, A.V., Birdsey, G.M., and Randi, A.M. (2016). Regulation of endothelial homeostasis, vascular development and angiogenesis by the transcription factor ERG. Vascular Pharmacology <em>86</em>, 3–13. <a href="https://doi.org/10.1016/j.vph.2016.05.003">https://doi.org/10.1016/j.vph.2016.05.003</a>.</div>
</div>
<div id="ref-gastfriend2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">26. </div><div class="csl-right-inline">Gastfriend, B.D., Snyder, M.E., Holt, H.E., Daneman, R., Palecek, S.P., and Shusta, E.V. (2024). Notch3 directs differentiation of brain mural cells from human pluripotent stem cell<span></span>derived neural crest. Science Advances <em>10</em>. <a href="https://doi.org/10.1126/sciadv.adi1737">https://doi.org/10.1126/sciadv.adi1737</a>.</div>
</div>
<div id="ref-Street2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">27. </div><div class="csl-right-inline">Street, K., Risso, D., Fletcher, R.B., Das, D., Ngai, J., Yosef, N., Purdom, E., and Dudoit, S. (2018). Slingshot: cell lineage and pseudotime inference for single-cell transcriptomics. BMC Genomics <em>19</em>. <a href="https://doi.org/10.1186/s12864-018-4772-0">https://doi.org/10.1186/s12864-018-4772-0</a>.</div>
</div>
<div id="ref-Van2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">28. </div><div class="csl-right-inline">Van den Berge, K., Roux de Bézieux, H., Street, K., Saelens, W., Cannoodt, R., Saeys, Y., Dudoit, S., and Clement, L. (2020). Trajectory-based differential expression analysis for single-cell sequencing data. Nature Communications <em>11</em>. <a href="https://doi.org/10.1038/s41467-020-14766-3">https://doi.org/10.1038/s41467-020-14766-3</a>.</div>
</div>
<div id="ref-vandenberge2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">29. </div><div class="csl-right-inline">Van den Berge, K., Roux de Bézieux, H., Street, K., Saelens, W., Cannoodt, R., Saeys, Y., Dudoit, S., and Clement, L. (2020). Trajectory-based differential expression analysis for single-cell sequencing data. Nature Communications <em>11</em>. <a href="https://doi.org/10.1038/s41467-020-14766-3">https://doi.org/10.1038/s41467-020-14766-3</a>.</div>
</div>
<div id="ref-finak2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">30. </div><div class="csl-right-inline">Finak, G., McDavid, A., Yajima, M., Deng, J., Gersuk, V., Shalek, A.K., Slichter, C.K., Miller, H.W., McElrath, M.J., Prlic, M., et al. (2015). MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data. Genome Biology <em>16</em>. <a href="https://doi.org/10.1186/s13059-015-0844-5">https://doi.org/10.1186/s13059-015-0844-5</a>.</div>
</div>
<div id="ref-bhatia2018" class="csl-entry" role="listitem">
<div class="csl-left-margin">31. </div><div class="csl-right-inline">Bhatia, S., Kim, W.S., Shepherd, C.E., and Halliday, G.M. (2018). Apolipoprotein D Upregulation in Alzheimer<span>’</span>s Disease but Not Frontotemporal Dementia. Journal of Molecular Neuroscience <em>67</em>, 125–132. <a href="https://doi.org/10.1007/s12031-018-1217-9">https://doi.org/10.1007/s12031-018-1217-9</a>.</div>
</div>
<div id="ref-yeap2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">32. </div><div class="csl-right-inline">Yeap, Y.J., Kandiah, N., Nizetic, D., and Lim, K.-L. (2022). BACE2: A Promising Neuroprotective Candidate for Alzheimer<span>’</span>s Disease. Journal of Alzheimer<span>’</span>s Disease <em>94</em>, S159–S171. <a href="https://doi.org/10.3233/jad-220867">https://doi.org/10.3233/jad-220867</a>.</div>
</div>
<div id="ref-Jin2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">33. </div><div class="csl-right-inline">Jin, S., Guerrero-Juarez, C.F., Zhang, L., Chang, I., Ramos, R., Kuan, C.-H., Myung, P., Plikus, M.V., and Nie, Q. (2021). Inference and analysis of cell-cell communication using CellChat. Nature Communications <em>12</em>. <a href="https://doi.org/10.1038/s41467-021-21246-9">https://doi.org/10.1038/s41467-021-21246-9</a>.</div>
</div>
<div id="ref-wang2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">34. </div><div class="csl-right-inline">Wang, Q., and Lash, G.E. (2017). Angiopoietin 2 in placentation and tumor biology: The yin and yang of vascular biology. Placenta <em>56</em>, 73–78. <a href="https://doi.org/10.1016/j.placenta.2017.03.021">https://doi.org/10.1016/j.placenta.2017.03.021</a>.</div>
</div>
<div id="ref-dagher2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">35. </div><div class="csl-right-inline">Dagher, N.N., Najafi, A.R., Kayala, K.M.N., Elmore, M.R.P., White, T.E., Medeiros, R., West, B.L., and Green, K.N. (2015). Colony-stimulating factor 1 receptor inhibition prevents microglial plaque association and improves cognition in 3xTg-AD mice. Journal of Neuroinflammation <em>12</em>. <a href="https://doi.org/10.1186/s12974-015-0366-9">https://doi.org/10.1186/s12974-015-0366-9</a>.</div>
</div>
<div id="ref-walker2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">36. </div><div class="csl-right-inline">Walker, D.G., Tang, T.M., and Lue, L.-F. (2017). Studies on colony stimulating factor receptor-1 and ligands colony stimulating factor-1 and interleukin-34 in alzheimer’s disease brains and human microglia. Frontiers in Aging Neuroscience <em>9</em>. <a href="https://doi.org/10.3389/fnagi.2017.00244">https://doi.org/10.3389/fnagi.2017.00244</a>.</div>
</div>
<div id="ref-fisher2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">37. </div><div class="csl-right-inline">Fisher, R.A., Miners, J.S., and Love, S. (2022). Pathological changes within the cerebral vasculature in Alzheimer<span>’</span>s disease: New perspectives. Brain Pathology <em>32</em>. <a href="https://doi.org/10.1111/bpa.13061">https://doi.org/10.1111/bpa.13061</a>.</div>
</div>
<div id="ref-zeisberg2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">38. </div><div class="csl-right-inline">Zeisberg, E.M., Tarnavski, O., Zeisberg, M., Dorfman, A.L., McMullen, J.R., Gustafsson, E., Chandraker, A., Yuan, X., Pu, W.T., Roberts, A.B., et al. (2007). Endothelial-to-mesenchymal transition contributes to cardiac fibrosis. Nature Medicine <em>13</em>, 952–961. <a href="https://doi.org/10.1038/nm1613">https://doi.org/10.1038/nm1613</a>.</div>
</div>
<div id="ref-medici2010" class="csl-entry" role="listitem">
<div class="csl-left-margin">39. </div><div class="csl-right-inline">Medici, D., Shore, E.M., Lounev, V.Y., Kaplan, F.S., Kalluri, R., and Olsen, B.R. (2010). Conversion of vascular endothelial cells into multipotent stem-like cells. Nature Medicine <em>16</em>, 1400–1406. <a href="https://doi.org/10.1038/nm.2252">https://doi.org/10.1038/nm.2252</a>.</div>
</div>
<div id="ref-verbeek1994" class="csl-entry" role="listitem">
<div class="csl-left-margin">40. </div><div class="csl-right-inline">Verbeek, M.M., Otte-Höller, I., Wesseling, P., Ruiter, D.J., and Waal, R.M. de (1994). <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1887139/">Induction of alpha-smooth muscle actin expression in cultured human brain pericytes by transforming growth factor-beta 1.</a> The American Journal of Pathology <em>144</em>, 372–382.</div>
</div>
<div id="ref-maddaluno2013" class="csl-entry" role="listitem">
<div class="csl-left-margin">41. </div><div class="csl-right-inline">Maddaluno, L., Rudini, N., Cuttano, R., Bravi, L., Giampietro, C., Corada, M., Ferrarini, L., Orsenigo, F., Papa, E., Boulday, G., et al. (2013). EndMT contributes to the onset and progression of cerebral cavernous malformations. Nature <em>498</em>, 492–496. <a href="https://doi.org/10.1038/nature12207">https://doi.org/10.1038/nature12207</a>.</div>
</div>
<div id="ref-winkler2014" class="csl-entry" role="listitem">
<div class="csl-left-margin">42. </div><div class="csl-right-inline">Winkler, E.A., Sagare, A.P., and Zlokovic, B.V. (2014). The Pericyte: A Forgotten Cell Type with Important Implications for Alzheimer’s Disease? Brain Pathology <em>24</em>, 371–386. <a href="https://doi.org/10.1111/bpa.12152">https://doi.org/10.1111/bpa.12152</a>.</div>
</div>
<div id="ref-yeap2022a" class="csl-entry" role="listitem">
<div class="csl-left-margin">43. </div><div class="csl-right-inline">Yeap, Y.J., Kandiah, N., Nizetic, D., and Lim, K.-L. (2022). BACE2: A Promising Neuroprotective Candidate for Alzheimer<span>’</span>s Disease. Journal of Alzheimer<span>’</span>s Disease <em>94</em>, S159–S171. <a href="https://doi.org/10.3233/jad-220867">https://doi.org/10.3233/jad-220867</a>.</div>
</div>
<div id="ref-vanhulle2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">44. </div><div class="csl-right-inline">Van Hulle, C., Ince, S., Okonkwo, O.C., Bendlin, B.B., Johnson, S.C., Carlsson, C.M., Asthana, S., Love, S., Blennow, K., Zetterberg, H., et al. (2024). Elevated CSF angiopoietin-2 correlates with blood-brain barrier leakiness and markers of neuronal injury in early Alzheimer<span>’</span>s disease. Translational Psychiatry <em>14</em>. <a href="https://doi.org/10.1038/s41398-023-02706-w">https://doi.org/10.1038/s41398-023-02706-w</a>.</div>
</div>
<div id="ref-qin2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">45. </div><div class="csl-right-inline">Qin, W., Jia, X., Wang, F., Zuo, X., Wu, L., Zhou, A., Li, D., Min, B., Wei, C., Tang, Y., et al. (2015). Elevated plasma angiogenesis factors in alzheimer’s disease. Journal of Alzheimer’s Disease <em>45</em>, 245–252. <a href="https://doi.org/10.3233/jad-142409">https://doi.org/10.3233/jad-142409</a>.</div>
</div>
<div id="ref-chen2009" class="csl-entry" role="listitem">
<div class="csl-left-margin">46. </div><div class="csl-right-inline">Chen, M.J., Yokomizo, T., Zeigler, B.M., Dzierzak, E., and Speck, N.A. (2009). Runx1 is required for the endothelial to haematopoietic cell transition but not thereafter. Nature <em>457</em>, 887–891. <a href="https://doi.org/10.1038/nature07619">https://doi.org/10.1038/nature07619</a>.</div>
</div>
<div id="ref-madisen2009" class="csl-entry" role="listitem">
<div class="csl-left-margin">47. </div><div class="csl-right-inline">Madisen, L., Zwingman, T.A., Sunkin, S.M., Oh, S.W., Zariwala, H.A., Gu, H., Ng, L.L., Palmiter, R.D., Hawrylycz, M.J., Jones, A.R., et al. (2009). A robust and high-throughput Cre reporting and characterization system for the whole mouse brain. Nature Neuroscience <em>13</em>, 133–140. <a href="https://doi.org/10.1038/nn.2467">https://doi.org/10.1038/nn.2467</a>.</div>
</div>
<div id="ref-khozoie2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">48. </div><div class="csl-right-inline">Khozoie, C., Fancy, N., Marjaneh, M., Murphy, A., Matthews, P., and Skene, N. (2021). scFlow: A scalable and reproducible analysis pipeline for single-cell RNA sequencing dat <a href="https://doi.org/10.22541/au.162912533.38489960/v2">https://doi.org/10.22541/au.162912533.38489960/v2</a>.</div>
</div>
<div id="ref-ditommaso2017" class="csl-entry" role="listitem">
<div class="csl-left-margin">49. </div><div class="csl-right-inline">Di Tommaso, P., Chatzou, M., Floden, E.W., Barja, P.P., Palumbo, E., and Notredame, C. (2017). Nextflow enables reproducible computational workflows. Nature Biotechnology <em>35</em>, 316–319. <a href="https://doi.org/10.1038/nbt.3820">https://doi.org/10.1038/nbt.3820</a>.</div>
</div>
<div id="ref-base" class="csl-entry" role="listitem">
<div class="csl-left-margin">50. </div><div class="csl-right-inline">R Core Team (2024). <a href="https://www.R-project.org/">R: A language and environment for statistical computing</a>.</div>
</div>
<div id="ref-Seurat" class="csl-entry" role="listitem">
<div class="csl-left-margin">51. </div><div class="csl-right-inline">Hao, Y., Stuart, T., Kowalski, M.H., Choudhary, S., Hoffman, P., Hartman, A., Srivastava, A., Molla, G., Madad, S., Fernandez-Granda, C., et al. (2023). Dictionary learning for integrative, multimodal and scalable single-cell analysis. <a href="https://doi.org/10.1038/s41587-023-01767-y">https://doi.org/10.1038/s41587-023-01767-y</a>.</div>
</div>
<div id="ref-speckle" class="csl-entry" role="listitem">
<div class="csl-left-margin">52. </div><div class="csl-right-inline">Phipson, B., Sim, C.B., Porrello, E.R., Hewitt, A.W., Powell, J., and Oshlack, A. (2022). <a href="https://doi.org/10.1093/bioinformatics/btac582">Propeller: Testing for differences in cell type proportions in single cell data</a>. <em>38</em>, –6.</div>
</div>
<div id="ref-clusterProfiler" class="csl-entry" role="listitem">
<div class="csl-left-margin">53. </div><div class="csl-right-inline">Wu, T., Hu, E., Xu, S., Chen, M., Guo, P., Dai, Z., Feng, T., Zhou, L., Tang, W., Zhan, L., et al. (2021). clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. <em>2</em>, 100141. <a href="https://doi.org/10.1016/j.xinn.2021.100141">https://doi.org/10.1016/j.xinn.2021.100141</a>.</div>
</div>
<div id="ref-Smith2009" class="csl-entry" role="listitem">
<div class="csl-left-margin">54. </div><div class="csl-right-inline">Smith, C.L., and Eppig, J.T. (2009). The mammalian phenotype ontology: enabling robust annotation and comparative analysis. WIREs Systems Biology and Medicine <em>1</em>, 390–399. <a href="https://doi.org/10.1002/wsbm.44">https://doi.org/10.1002/wsbm.44</a>.</div>
</div>
<div id="ref-gu2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">55. </div><div class="csl-right-inline">Gu, Z. (2023). <a href="http://dx.doi.org/10.1101/2023.12.03.569758"><span><em>simona:</em></span>a comprehensive r package for semantic similarity analysis on bio-ontologies</a>.</div>
</div>
<div id="ref-CellChat" class="csl-entry" role="listitem">
<div class="csl-left-margin">56. </div><div class="csl-right-inline">Jin, S. (2024). <a href="https://github.com/jinworks/CellChat">CellChat: Inference and analysis of cell-cell communication from single-cell and spatially resolved transcriptomics data</a>.</div>
</div>
<div id="ref-Alonso-López2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">57. </div><div class="csl-right-inline">Alonso-López, D., Campos-Laborie, F.J., Gutiérrez, M.A., Lambourne, L., Calderwood, M.A., Vidal, M., and De Las Rivas, J. (2019). APID database: redefining protein<span></span>protein interaction experimental evidences and binary interactomes. Database <em>2019</em>. <a href="https://doi.org/10.1093/database/baz005">https://doi.org/10.1093/database/baz005</a>.</div>
</div>
<div id="ref-Oughtred2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">58. </div><div class="csl-right-inline">Oughtred, R., Rust, J., Chang, C., Breitkreutz, B.-J., Stark, C., Willems, A., Boucher, L., Leung, G., Kolas, N., Zhang, F., et al. (2020). The BioGRID database: A comprehensive biomedical resource of curated protein, genetic, and chemical interactions. Protein Science <em>30</em>, 187–200. <a href="https://doi.org/10.1002/pro.3978">https://doi.org/10.1002/pro.3978</a>.</div>
</div>
<div id="ref-Huttlin2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">59. </div><div class="csl-right-inline">Huttlin, E.L., Bruckner, R.J., Navarrete-Perea, J., Cannon, J.R., Baltier, K., Gebreab, F., Gygi, M.P., Thornock, A., Zarraga, G., Tam, S., et al. (2021). Dual proteome-scale networks reveal cell-specific remodeling of the human interactome. Cell <em>184</em>, 3022–3040.e28. <a href="https://doi.org/10.1016/j.cell.2021.04.011">https://doi.org/10.1016/j.cell.2021.04.011</a>.</div>
</div>
<div id="ref-Tsitsiridis2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">60. </div><div class="csl-right-inline">Tsitsiridis, G., Steinkamp, R., Giurgiu, M., Brauner, B., Fobo, G., Frishman, G., Montrone, C., and Ruepp, A. (2022). CORUM: the comprehensive resource of mammalian protein complexes<span></span>2022. Nucleic Acids Research <em>51</em>, D539–D545. <a href="https://doi.org/10.1093/nar/gkac1015">https://doi.org/10.1093/nar/gkac1015</a>.</div>
</div>
<div id="ref-López2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">61. </div><div class="csl-right-inline">López, Y., Nakai, K., and Patil, A. (2015). HitPredict version 4: comprehensive reliability scoring of physical protein<span></span>protein interactions from more than 100 species. Database <em>2015</em>, bav117. <a href="https://doi.org/10.1093/database/bav117">https://doi.org/10.1093/database/bav117</a>.</div>
</div>
<div id="ref-Luck2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">62. </div><div class="csl-right-inline">Luck, K., Kim, D.-K., Lambourne, L., Spirohn, K., Begg, B.E., Bian, W., Brignall, R., Cafarelli, T., Campos-Laborie, F.J., Charloteaux, B., et al. (2020). A reference map of the human binary protein interactome. Nature <em>580</em>, 402–408. <a href="https://doi.org/10.1038/s41586-020-2188-x">https://doi.org/10.1038/s41586-020-2188-x</a>.</div>
</div>
<div id="ref-Orchard2014" class="csl-entry" role="listitem">
<div class="csl-left-margin">63. </div><div class="csl-right-inline">Orchard, S., Ammari, M., Aranda, B., Breuza, L., Briganti, L., Broackes-Carter, F., Campbell, N.H., Chavali, G., Chen, C., del-Toro, N., et al. (2013). The MIntAct project<span></span>IntAct as a common curation platform for 11 molecular interaction databases. Nucleic Acids Research <em>42</em>, D358–D363. <a href="https://doi.org/10.1093/nar/gkt1115">https://doi.org/10.1093/nar/gkt1115</a>.</div>
</div>
<div id="ref-Licata2012" class="csl-entry" role="listitem">
<div class="csl-left-margin">64. </div><div class="csl-right-inline">Licata, L., Briganti, L., Peluso, D., Perfetto, L., Iannuccelli, M., Galeota, E., Sacco, F., Palma, A., Nardozza, A.P., Santonico, E., et al. (2011). MINT, the molecular interaction database: 2012 update. Nucleic Acids Research <em>40</em>, D857–D861. <a href="https://doi.org/10.1093/nar/gkr930">https://doi.org/10.1093/nar/gkr930</a>.</div>
</div>
<div id="ref-Gillespie2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">65. </div><div class="csl-right-inline">Gillespie, M., Jassal, B., Stephan, R., Milacic, M., Rothfels, K., Senff-Ribeiro, A., Griss, J., Sevilla, C., Matthews, L., Gong, C., et al. (2021). The reactome pathway knowledgebase 2022. Nucleic Acids Research <em>50</em>, D687–D692. <a href="https://doi.org/10.1093/nar/gkab1028">https://doi.org/10.1093/nar/gkab1028</a>.</div>
</div>
<div id="ref-Szklarczyk2023" class="csl-entry" role="listitem">
<div class="csl-left-margin">66. </div><div class="csl-right-inline">Szklarczyk, D., Kirsch, R., Koutrouli, M., Nastou, K., Mehryary, F., Hachilif, R., Gable, A.L., Fang, T., Doncheva, N., Pyysalo, S., et al. (2022). The STRING database in 2023: protein<span></span>protein association networks and functional enrichment analyses for any sequenced genome of interest. Nucleic Acids Research <em>51</em>, D638–D646. <a href="https://doi.org/10.1093/nar/gkac1000">https://doi.org/10.1093/nar/gkac1000</a>.</div>
</div>
<div id="ref-leeuw2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">67. </div><div class="csl-right-inline">Leeuw, C.A. de, Mooij, J.M., Heskes, T., and Posthuma, D. (2015). MAGMA: Generalized Gene-Set Analysis of GWAS Data. PLOS Computational Biology <em>11</em>, e1004219. <a href="https://doi.org/10.1371/journal.pcbi.1004219">https://doi.org/10.1371/journal.pcbi.1004219</a>.</div>
</div>
<div id="ref-bulik-sullivan2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">68. </div><div class="csl-right-inline">Bulik-Sullivan, B.K., Loh, P.-R., Finucane, H.K., Ripke, S., Yang, J., Patterson, N., Daly, M.J., Price, A.L., and Neale, B.M. (2015). LD Score regression distinguishes confounding from polygenicity in genome-wide association studies. Nature Genetics <em>47</em>, 291–295. <a href="https://doi.org/10.1038/ng.3211">https://doi.org/10.1038/ng.3211</a>.</div>
</div>
<div id="ref-Seurat-2" class="csl-entry" role="listitem">
<div class="csl-left-margin">69. </div><div class="csl-right-inline">Hao, Y., Stuart, T., Kowalski, M.H., Choudhary, S., Hoffman, P., Hartman, A., Srivastava, A., Molla, G., Madad, S., Fernandez-Granda, C., et al. (2023). Dictionary learning for integrative, multimodal and scalable single-cell analysis. <a href="https://doi.org/10.1038/s41587-023-01767-y">https://doi.org/10.1038/s41587-023-01767-y</a>.</div>
</div>
<div id="ref-auton2015" class="csl-entry" role="listitem">
<div class="csl-left-margin">70. </div><div class="csl-right-inline">Auton, A., Abecasis, G.R., Altshuler, D.M., Durbin, R.M., Abecasis, G.R., Bentley, D.R., Chakravarti, A., Clark, A.G., Donnelly, P., Eichler, E.E., et al. (2015). A global reference for human genetic variation. Nature <em>526</em>, 68–74. <a href="https://doi.org/10.1038/nature15393">https://doi.org/10.1038/nature15393</a>.</div>
</div>
<div id="ref-sayers2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">71. </div><div class="csl-right-inline">Sayers, E.W., Bolton, E.E., Brister, J.R., Canese, K., Chan, J., Comeau, D.C., Connor, R., Funk, K., Kelly, C., Kim, S., et al. (2022). Database resources of the national center for biotechnology information. Nucleic Acids Research <em>50</em>, D20–D26. <a href="https://doi.org/10.1093/nar/gkab1112">https://doi.org/10.1093/nar/gkab1112</a>.</div>
</div>
</div>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>