<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.54">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>Build figures</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Build figures</h6>

            <a href="../../04_data_analysis/999_results_summary/04_figures.qmd" class="btn btn-primary quarto-download-embed" download="04_figures.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Build figures</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#building-figures" id="toc-building-figures" class="nav-link active" data-scroll-target="#building-figures">Building figures</a></li>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages">Load packages</a>
  <ul class="collapse">
  <li><a href="#define-ggplot-theme" id="toc-define-ggplot-theme" class="nav-link" data-scroll-target="#define-ggplot-theme">Define ggplot theme</a></li>
  <li><a href="#figure-1---method-and-demographic-overview" id="toc-figure-1---method-and-demographic-overview" class="nav-link" data-scroll-target="#figure-1---method-and-demographic-overview">Figure 1 - method and demographic overview</a>
  <ul class="collapse">
  <li><a href="#endo-mt-markers-from-paper" id="toc-endo-mt-markers-from-paper" class="nav-link" data-scroll-target="#endo-mt-markers-from-paper">Endo-MT markers from paper</a></li>
  </ul></li>
  <li><a href="#umaps" id="toc-umaps" class="nav-link" data-scroll-target="#umaps">UMAPs</a></li>
  <li><a href="#bar-plot-of-cell-proportions" id="toc-bar-plot-of-cell-proportions" class="nav-link" data-scroll-target="#bar-plot-of-cell-proportions">Bar plot of cell proportions</a></li>
  <li><a href="#boxplot-of-number-of-nuclie-per-donor" id="toc-boxplot-of-number-of-nuclie-per-donor" class="nav-link" data-scroll-target="#boxplot-of-number-of-nuclie-per-donor">Boxplot of number of nuclie per donor</a>
  <ul class="collapse">
  <li><a href="#explanation" id="toc-explanation" class="nav-link" data-scroll-target="#explanation">Explanation</a></li>
  </ul></li>
  <li><a href="#figure-2" id="toc-figure-2" class="nav-link" data-scroll-target="#figure-2">Figure 2</a>
  <ul class="collapse">
  <li><a href="#go-pathways" id="toc-go-pathways" class="nav-link" data-scroll-target="#go-pathways">GO pathways</a></li>
  <li><a href="#magma-heatmap" id="toc-magma-heatmap" class="nav-link" data-scroll-target="#magma-heatmap">MAGMA heatmap</a></li>
  <li><a href="#mouse-phenotypes" id="toc-mouse-phenotypes" class="nav-link" data-scroll-target="#mouse-phenotypes">Mouse phenotypes</a></li>
  <li><a href="#assemble-figure-2" id="toc-assemble-figure-2" class="nav-link" data-scroll-target="#assemble-figure-2">Assemble Figure 2</a></li>
  <li><a href="#ppi-network-plots" id="toc-ppi-network-plots" class="nav-link" data-scroll-target="#ppi-network-plots">PPI network plots</a></li>
  </ul></li>
  <li><a href="#figure-3---degs" id="toc-figure-3---degs" class="nav-link" data-scroll-target="#figure-3---degs">Figure 3 - DEGs</a>
  <ul class="collapse">
  <li><a href="#degs-count-barplot" id="toc-degs-count-barplot" class="nav-link" data-scroll-target="#degs-count-barplot">DEGs count barplot</a></li>
  <li><a href="#go-summary" id="toc-go-summary" class="nav-link" data-scroll-target="#go-summary">GO summary</a></li>
  <li><a href="#heatmap-of-shared-degs" id="toc-heatmap-of-shared-degs" class="nav-link" data-scroll-target="#heatmap-of-shared-degs">Heatmap of shared DEGs</a></li>
  <li><a href="#assemble-figure" id="toc-assemble-figure" class="nav-link" data-scroll-target="#assemble-figure">Assemble Figure</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <section id="building-figures" class="level1">
<h1>Building figures</h1>
<p>Plot guidance for figures in cell here: <a href="https://www.cell.com/figure-guidelines">https://www.cell.com/figure-guidelines</a></p>
<p>Limit is 7 figures/tables, but more can go in supplement.</p>
<p>Width guidance:</p>
<ul>
<li>For 2-column formats (such as research articles and reviews), the sizes are 85 mm (1 column), 114 mm (1.5 columns), and 174 mm (full width of the page)</li>
</ul>
</section>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, smiths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'kableExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    group_rows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'janitor'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    chisq.test, fisher.test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    melt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rngtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  To enable shared memory capabilities, try: install.extras('
NMF
')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'future'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    run</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Some features are not enabled in this build of Arrow. Run `arrow_info()` for more information.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'arrow'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    timestamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'viridis'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    viridis_pal</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 methods overwritten by 'proxy':
  method               from    
  print.registry_field registry
  print.registry_entry registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    nrun</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:webshot2':

    resize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built under R 4.4.0 but the current version is
4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI
for R may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'DESeq2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    fpkm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: limma</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DESeq2':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    plotMDS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'edgeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SingleCellExperiment':

    cpm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocParallel'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    register</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation("pathview") within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationFilter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationFilter'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:arrow':

    field</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:future':

    value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ensembldb'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggrepel</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'PCAtools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scran':

    parallelPCA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    biplot, screeplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Seurat':

    components</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    normalize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:future':

    %-&gt;%, %&lt;-%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:NMF':

    algorithm, compare</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'biomaRt'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scRNAseq':

    listDatasets, searchDatasets</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>clusterProfiler v4.12.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use clusterProfiler in published research, please cite:
T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'clusterProfiler'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>DOSE v3.30.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'enrichplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    color_palette</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pals':

    alphabet</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocIO</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rtracklayer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocIO':

    FileForFormat</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    blocks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationHub':

    hubUrl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh37'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'qusage'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    read.gmt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:SNPlocs.Hsapiens.dbSNP155.GRCh37':

    .loadAlleles, .loadLoc, COMPATIBLE_BSGENOMES</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'GenomicFiles'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
simona version 1.2.0
Bioconductor page: http://bioconductor.org/packages/simona/
Github page: https://github.com/jokergoo/simona
Documentation: https://jokergoo.github.io/simona/

If you use it in published research, please cite:
Gu, Z. simona: a Comprehensive R package for Semantic Similarity 
  Analysis on Bio-Ontologies. bioRxiv 2023.

This message can be suppressed by:
  suppressPackageStartupMessages(library(simona))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'ggnetwork':
  method         from  
  fortify.igraph ggtree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'CellChat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scater':

    runPCA, runUMAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_config_set</span>(<span class="at">store =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_targets"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="define-ggplot-theme" class="level2">
<h2 class="anchored" data-anchor-id="define-ggplot-theme">Define ggplot theme</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set default discrete color palette</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a>scale_color_discrete <span class="ot">&lt;-</span> <span class="cf">function</span>(...) <span class="fu">scale_color_viridis_d</span>(...)</span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set default continuous color palette</span></span>
<span id="cb198-5"><a href="#cb198-5" aria-hidden="true" tabindex="-1"></a>scale_color_continuous <span class="ot">&lt;-</span> <span class="cf">function</span>(...) <span class="fu">scale_color_viridis_c</span>(...)</span>
<span id="cb198-6"><a href="#cb198-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-7"><a href="#cb198-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Set default fill palette</span></span>
<span id="cb198-8"><a href="#cb198-8" aria-hidden="true" tabindex="-1"></a>scale_fill_discrete <span class="ot">&lt;-</span> <span class="cf">function</span>(...) <span class="fu">scale_fill_viridis_d</span>(...)</span>
<span id="cb198-9"><a href="#cb198-9" aria-hidden="true" tabindex="-1"></a>scale_fill_continuous <span class="ot">&lt;-</span> <span class="cf">function</span>(...) <span class="fu">scale_fill_viridis_c</span>(...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Polychrome)</span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a colorblind-friendly palette with 44 colors</span></span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a>palette <span class="ot">&lt;-</span> <span class="fu">createPalette</span>(<span class="dv">44</span>, <span class="at">seedcolors =</span> <span class="fu">viridis</span>(<span class="dv">3</span>), <span class="at">M =</span> <span class="dv">5000</span>)</span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to a format that ggplot2 can use</span></span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a>palette <span class="ot">&lt;-</span> <span class="fu">unname</span>(palette)</span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-8"><a href="#cb199-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom function to apply the palette</span></span>
<span id="cb199-9"><a href="#cb199-9" aria-hidden="true" tabindex="-1"></a>scale_color_custom <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb199-10"><a href="#cb199-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> palette, ...)</span>
<span id="cb199-11"><a href="#cb199-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb199-12"><a href="#cb199-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-13"><a href="#cb199-13" aria-hidden="true" tabindex="-1"></a>scale_fill_custom <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb199-14"><a href="#cb199-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> palette, ...)</span>
<span id="cb199-15"><a href="#cb199-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom ggplot theme</span></span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a>custom_theme <span class="ot">&lt;-</span> <span class="fu">theme</span>(</span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">8</span>), <span class="co"># Set the base font size - needs to be 6-8 pt</span></span>
<span id="cb200-4"><a href="#cb200-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">8</span>), <span class="co"># Axis titles font size</span></span>
<span id="cb200-5"><a href="#cb200-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">6</span>), <span class="co"># Axis text font size</span></span>
<span id="cb200-6"><a href="#cb200-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">8</span>), <span class="co"># Legend title font size</span></span>
<span id="cb200-7"><a href="#cb200-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">6</span>), <span class="co"># Legend text font size</span></span>
<span id="cb200-8"><a href="#cb200-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"Arial"</span>, <span class="at">size =</span> <span class="dv">18</span>, <span class="at">face =</span> <span class="st">"bold"</span>), <span class="co"># Plot title font size and style</span></span>
<span id="cb200-9"><a href="#cb200-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.background =</span> <span class="fu">element_blank</span>(), <span class="co"># Remove background</span></span>
<span id="cb200-10"><a href="#cb200-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#panel.grid.major = element_line(color = "grey80"), # Major grid lines color</span></span>
<span id="cb200-11"><a href="#cb200-11" aria-hidden="true" tabindex="-1"></a>  <span class="co">#panel.grid.minor = element_line(color = "grey90"), # Minor grid lines color</span></span>
<span id="cb200-12"><a href="#cb200-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="co"># Major grid lines color</span></span>
<span id="cb200-13"><a href="#cb200-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), <span class="co"># Minor grid lines color</span></span>
<span id="cb200-14"><a href="#cb200-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>) <span class="co"># Axis line color</span></span>
<span id="cb200-15"><a href="#cb200-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb200-16"><a href="#cb200-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-17"><a href="#cb200-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the theme globally</span></span>
<span id="cb200-18"><a href="#cb200-18" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(custom_theme)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="figure-1---method-and-demographic-overview" class="level2">
<h2 class="anchored" data-anchor-id="figure-1---method-and-demographic-overview">Figure 1 - method and demographic overview</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get seurat object</span></span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(sce)</span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a><span class="co"># the full cell labels are long for plots, let's make abbreviated versions</span></span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a>celltypes_abbrev <span class="ot">&lt;-</span></span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(</span>
<span id="cb201-7"><a href="#cb201-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Astrocyte-activated"</span> <span class="ot">=</span> <span class="st">"Astro-A"</span>,</span>
<span id="cb201-8"><a href="#cb201-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Astrocyte-quiescent"</span> <span class="ot">=</span> <span class="st">"Astro-Q"</span>,</span>
<span id="cb201-9"><a href="#cb201-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-ADARB2-ERBB4"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-1"</span>,</span>
<span id="cb201-10"><a href="#cb201-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-CBLN2-NRGN"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-2"</span>,</span>
<span id="cb201-11"><a href="#cb201-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-DLC1-HS3ST4"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-3"</span>,</span>
<span id="cb201-12"><a href="#cb201-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-4"</span>,</span>
<span id="cb201-13"><a href="#cb201-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-L3-5-RORB-PLCH1"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-5"</span>,</span>
<span id="cb201-14"><a href="#cb201-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-L4-5-RORB-IL1RAPL2"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-6"</span>,</span>
<span id="cb201-15"><a href="#cb201-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Ex-Neuron-L6-THEMIS-NFIA"</span> <span class="ot">=</span> <span class="st">"Ex-Neuro-7"</span>,</span>
<span id="cb201-16"><a href="#cb201-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-ADARB2-GRM7"</span> <span class="ot">=</span> <span class="st">"In-Neuro-1"</span>,</span>
<span id="cb201-17"><a href="#cb201-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-ADARB2-IL1RAPL2-CSCL14"</span> <span class="ot">=</span> <span class="st">"In-Neuro-2"</span>,</span>
<span id="cb201-18"><a href="#cb201-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-ADARB2-LAMP5-NRG1"</span> <span class="ot">=</span> <span class="st">"In-Neuro-3"</span>,</span>
<span id="cb201-19"><a href="#cb201-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-ADARB2-NRG1-RELN"</span> <span class="ot">=</span> <span class="st">"In-Neuro-4"</span>,</span>
<span id="cb201-20"><a href="#cb201-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-DPP10-MEF2C"</span> <span class="ot">=</span> <span class="st">"In-Neuro-5"</span>,</span>
<span id="cb201-21"><a href="#cb201-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-PRKG1-TBC1D4"</span> <span class="ot">=</span> <span class="st">"In-Neuro-6"</span>,</span>
<span id="cb201-22"><a href="#cb201-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"In-Neuron-TRHDE-RALYL"</span> <span class="ot">=</span> <span class="st">"In-Neuro-7"</span>,</span>
<span id="cb201-23"><a href="#cb201-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Microglia-activated"</span> <span class="ot">=</span> <span class="st">"Microglia-A"</span>,</span>
<span id="cb201-24"><a href="#cb201-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Microglia-quiescent"</span> <span class="ot">=</span> <span class="st">"Microglia-Q"</span>,</span>
<span id="cb201-25"><a href="#cb201-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Microglia-vascular"</span> <span class="ot">=</span> <span class="st">"Microglia-V"</span>,</span>
<span id="cb201-26"><a href="#cb201-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pericyte"</span> <span class="ot">=</span> <span class="st">"Pericyte-1"</span>,</span>
<span id="cb201-27"><a href="#cb201-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Perivascular Macrophage"</span> <span class="ot">=</span> <span class="st">"Perivascular-M"</span>,</span>
<span id="cb201-28"><a href="#cb201-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Perivascular-FB"</span> <span class="ot">=</span> <span class="st">"Perivascular-FB-1"</span>,</span>
<span id="cb201-29"><a href="#cb201-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Perivascular-FB-KAZN2"</span> <span class="ot">=</span> <span class="st">"Perivascular-FB-2"</span>,</span>
<span id="cb201-30"><a href="#cb201-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Trans-Endo-to-mural"</span> <span class="ot">=</span> <span class="st">"EndoMT"</span>,</span>
<span id="cb201-31"><a href="#cb201-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T-cell-parenchymal-1"</span> <span class="ot">=</span> <span class="st">"T-cell-P1"</span>,</span>
<span id="cb201-32"><a href="#cb201-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T-cell-parenchymal-2"</span> <span class="ot">=</span> <span class="st">"T-cell-P2"</span>,</span>
<span id="cb201-33"><a href="#cb201-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T-cell-vascular"</span> <span class="ot">=</span> <span class="st">"T-cell-V"</span>,</span>
<span id="cb201-34"><a href="#cb201-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T-cell-mixed"</span> <span class="ot">=</span> <span class="st">"T-cell-M"</span>,</span>
<span id="cb201-35"><a href="#cb201-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vascular-SMC"</span> <span class="ot">=</span> <span class="st">"Vascular-SMC-1"</span>,</span>
<span id="cb201-36"><a href="#cb201-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vascular-SMC-LINC00486"</span> <span class="ot">=</span> <span class="st">"Vascular-SMC-2"</span></span>
<span id="cb201-37"><a href="#cb201-37" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb201-38"><a href="#cb201-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-39"><a href="#cb201-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new column with abbreviated Idents</span></span>
<span id="cb201-40"><a href="#cb201-40" aria-hidden="true" tabindex="-1"></a>abbreviated_idents <span class="ot">&lt;-</span> celltypes_abbrev[<span class="fu">as.character</span>(sce<span class="sc">$</span>subcelltype_annotations)]</span>
<span id="cb201-41"><a href="#cb201-41" aria-hidden="true" tabindex="-1"></a>abbreviated_idents <span class="ot">&lt;-</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(abbreviated_idents), sce<span class="sc">$</span>subcelltype_annotations, abbreviated_idents)</span>
<span id="cb201-42"><a href="#cb201-42" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(abbreviated_idents) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb201-43"><a href="#cb201-43" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>abbreviated_idents <span class="ot">&lt;-</span> abbreviated_idents</span>
<span id="cb201-44"><a href="#cb201-44" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sce<span class="sc">$</span>abbreviated_idents)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "M-Pericyte"        "Vascular-SMC-1"    "EndoMT"           
 [4] "Oligo-A"           "Meningeal-FB"      "Capillary"        
 [7] "T-Pericyte"        "Arterial"          "Pericyte-1"       
[10] "Arteriolar-SMC"    "T-cell-P1"         "Pericyte-FB"      
[13] "Pericyte-2"        "Artirial-FB"       "Venous"           
[16] "Perivascular-M"    "Microglia-Q"       "Vascular-SMC-2"   
[19] "Perivascular-FB-2" "Astro-Q"           "OPC-B"            
[22] "Perivascular-FB-1" "T-cell-M"          "Oligo-B"          
[25] "Microglia-V"       "In-Neuro-7"        "Ex-Neuro-4"       
[28] "In-Neuro-1"        "T-cell-V"          "Astro-A"          
[31] "In-Neuro-4"        "Ex-Neuro-6"        "In-Neuro-6"       
[34] "In-Neuro-3"        "Microglia-A"       "In-Neuro-2"       
[37] "Ex-Neuro-7"        "Ex-Neuro-5"        "Ex-Neuro-3"       
[40] "OPC-A"             "In-Neuro-5"        "Ex-Neuro-1"       
[43] "Ex-Neuro-2"        "T-cell-P2"        </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(sce) <span class="ot">&lt;-</span> sce<span class="sc">$</span>abbreviated_idents</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(sce, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"AGBL1"</span>,</span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"ARHGAP15"</span>,</span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"DOCK2"</span>,</span>
<span id="cb204-4"><a href="#cb204-4" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"RUNX1"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-6-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="04_figures_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<section id="endo-mt-markers-from-paper" class="level3">
<h3 class="anchored" data-anchor-id="endo-mt-markers-from-paper">Endo-MT markers from paper</h3>
<p>Lets check the markers for Endo-MT reported in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10073145/">this paper</a></p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>endomt_markers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ACTA2"</span>, <span class="st">"FN1"</span>, <span class="st">"PECAM1"</span>, <span class="st">"LUM"</span>, <span class="st">"RAMP2"</span>, <span class="st">"IFI27"</span>, <span class="st">"TM4SF1"</span>, <span class="st">"GNG11"</span>, <span class="st">"TIMP3"</span>, <span class="st">"IGFBP7"</span>, <span class="st">"ID3"</span>, <span class="st">"CTGF"</span>)</span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DotPlot</span>(sce, <span class="at">features =</span> endomt_markers)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The following requested variables were not found: CTGF</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-7-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="04_figures_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
</section>
<section id="umaps" class="level2">
<h2 class="anchored" data-anchor-id="umaps">UMAPs</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: make umap</span></span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a>umap <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(sce, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">raster =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb207-4"><a href="#cb207-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoLegend</span>() <span class="sc">+</span></span>
<span id="cb207-5"><a href="#cb207-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoAxes</span>() <span class="sc">+</span></span>
<span id="cb207-6"><a href="#cb207-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoGrid</span>() <span class="sc">+</span> </span>
<span id="cb207-7"><a href="#cb207-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_custom</span>() <span class="sc">+</span></span>
<span id="cb207-8"><a href="#cb207-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_custom</span>()</span>
<span id="cb207-9"><a href="#cb207-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-10"><a href="#cb207-10" aria-hidden="true" tabindex="-1"></a>umap_vascular_parenchymal <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(sce, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="st">"prep"</span>,</span>
<span id="cb207-11"><a href="#cb207-11" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">raster =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb207-12"><a href="#cb207-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoAxes</span>() <span class="sc">+</span></span>
<span id="cb207-13"><a href="#cb207-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">NoGrid</span>() <span class="sc">+</span> </span>
<span id="cb207-14"><a href="#cb207-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_discrete</span>() <span class="sc">+</span></span>
<span id="cb207-15"><a href="#cb207-15" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_custom() +</span></span>
<span id="cb207-16"><a href="#cb207-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="cn">NULL</span>)</span>
<span id="cb207-17"><a href="#cb207-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-18"><a href="#cb207-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the UMAP plot</span></span>
<span id="cb207-19"><a href="#cb207-19" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("umap_plot_custom_palette.png", plot = umap_plot, width = 10, height = 8, dpi = 300)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="bar-plot-of-cell-proportions" class="level2">
<h2 class="anchored" data-anchor-id="bar-plot-of-cell-proportions">Bar plot of cell proportions</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb208"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get metadata</span></span>
<span id="cb208-2"><a href="#cb208-2" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> sce<span class="sc">@</span>meta.data</span>
<span id="cb208-3"><a href="#cb208-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-4"><a href="#cb208-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the percentage of cells in each fraction per cell type</span></span>
<span id="cb208-5"><a href="#cb208-5" aria-hidden="true" tabindex="-1"></a>percentages <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span></span>
<span id="cb208-6"><a href="#cb208-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(abbreviated_idents, prep) <span class="sc">%&gt;%</span></span>
<span id="cb208-7"><a href="#cb208-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb208-8"><a href="#cb208-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count), <span class="at">percentage =</span> (count <span class="sc">/</span> total) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb208-9"><a href="#cb208-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'abbreviated_idents'. You can override
using the `.groups` argument.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a><span class="co"># View the calculated percentages</span></span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(percentages)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6  5
  abbreviated_idents prep  count total percentage
  &lt;chr&gt;              &lt;fct&gt; &lt;int&gt; &lt;int&gt;      &lt;dbl&gt;
1 Arterial           P       127 21254      0.598
2 Arterial           V     21127 21254     99.4  
3 Arteriolar-SMC     P        12  7249      0.166
4 Arteriolar-SMC     V      7237  7249     99.8  
5 Artirial-FB        P        10   710      1.41 
6 Artirial-FB        V       700   710     98.6  </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the bar plot</span></span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>bar_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(percentages, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(abbreviated_idents, <span class="sc">-</span>percentage <span class="sc">*</span> (prep <span class="sc">==</span> <span class="st">"V"</span>)), <span class="at">y =</span> percentage, <span class="at">fill =</span> prep)) <span class="sc">+</span></span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a> <span class="co"># geom_bar(stat = "identity", position = "dodge") +</span></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Cell Type"</span>,</span>
<span id="cb212-6"><a href="#cb212-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Percentage"</span>,</span>
<span id="cb212-7"><a href="#cb212-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Fraction"</span>) <span class="sc">+</span></span>
<span id="cb212-8"><a href="#cb212-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb212-9"><a href="#cb212-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_fill_custom()</span></span>
<span id="cb212-10"><a href="#cb212-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-11"><a href="#cb212-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plot</span></span>
<span id="cb212-12"><a href="#cb212-12" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(bar_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-9-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="04_figures_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot</span></span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("bar_plot_P_vs_V.png", plot = bar_plot, width = 12, height = 8, dpi = 300)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb214-2"><a href="#cb214-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the percentage of cells in each fraction per cell type and higher-level cell type</span></span>
<span id="cb214-3"><a href="#cb214-3" aria-hidden="true" tabindex="-1"></a>percentages <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span></span>
<span id="cb214-4"><a href="#cb214-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(highlevel_manual_annotations, abbreviated_idents, prep) <span class="sc">%&gt;%</span></span>
<span id="cb214-5"><a href="#cb214-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb214-6"><a href="#cb214-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(count), <span class="at">percentage =</span> (count <span class="sc">/</span> total) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb214-7"><a href="#cb214-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb214-8"><a href="#cb214-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-9"><a href="#cb214-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View the calculated percentages</span></span>
<span id="cb214-10"><a href="#cb214-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(percentages)</span>
<span id="cb214-11"><a href="#cb214-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-12"><a href="#cb214-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the overall percentage of V cells for each higher-level cell type</span></span>
<span id="cb214-13"><a href="#cb214-13" aria-hidden="true" tabindex="-1"></a>overall_v_percentages <span class="ot">&lt;-</span> percentages <span class="sc">%&gt;%</span></span>
<span id="cb214-14"><a href="#cb214-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(prep <span class="sc">==</span> <span class="st">"V"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb214-15"><a href="#cb214-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(highlevel_manual_annotations) <span class="sc">%&gt;%</span></span>
<span id="cb214-16"><a href="#cb214-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">overall_v_percentage =</span> <span class="fu">sum</span>(percentage)) <span class="sc">%&gt;%</span></span>
<span id="cb214-17"><a href="#cb214-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(overall_v_percentage))</span>
<span id="cb214-18"><a href="#cb214-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-19"><a href="#cb214-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder the higher-level cell types based on the overall V percentage</span></span>
<span id="cb214-20"><a href="#cb214-20" aria-hidden="true" tabindex="-1"></a>percentages<span class="sc">$</span>highlevel_manual_annotations <span class="ot">&lt;-</span> <span class="fu">factor</span>(percentages<span class="sc">$</span>highlevel_manual_annotations, <span class="at">levels =</span> overall_v_percentages<span class="sc">$</span>highlevel_manual_annotations)</span>
<span id="cb214-21"><a href="#cb214-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-22"><a href="#cb214-22" aria-hidden="true" tabindex="-1"></a><span class="co"># View the reordered higher-level cell types</span></span>
<span id="cb214-23"><a href="#cb214-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(overall_v_percentages)</span>
<span id="cb214-24"><a href="#cb214-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-25"><a href="#cb214-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the bar plot, ordering the cell types by the percentage in the V fraction within each higher-level cell type</span></span>
<span id="cb214-26"><a href="#cb214-26" aria-hidden="true" tabindex="-1"></a>bar_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(percentages, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder_within</span>(abbreviated_idents, <span class="sc">-</span>percentage <span class="sc">*</span> (prep <span class="sc">==</span> <span class="st">"V"</span>), highlevel_manual_annotations), <span class="at">y =</span> percentage, <span class="at">fill =</span> prep)) <span class="sc">+</span></span>
<span id="cb214-27"><a href="#cb214-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb214-28"><a href="#cb214-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_reordered</span>() <span class="sc">+</span> <span class="co"># Apply the reordered scale</span></span>
<span id="cb214-29"><a href="#cb214-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Percentage of Cells in P vs V Fraction per Cell Type"</span>,</span>
<span id="cb214-30"><a href="#cb214-30" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Cell Type"</span>,</span>
<span id="cb214-31"><a href="#cb214-31" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Percentage"</span>,</span>
<span id="cb214-32"><a href="#cb214-32" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Fraction"</span>) <span class="sc">+</span></span>
<span id="cb214-33"><a href="#cb214-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_custom</span>() <span class="sc">+</span></span>
<span id="cb214-34"><a href="#cb214-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb214-35"><a href="#cb214-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> highlevel_manual_annotations, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">space =</span> <span class="st">"free_x"</span>) <span class="sc">+</span> <span class="co"># Subtle facet by higher-level cell type</span></span>
<span id="cb214-36"><a href="#cb214-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.2</span>, <span class="st">"lines"</span>)) <span class="co"># Adjust the gap between the groups</span></span>
<span id="cb214-37"><a href="#cb214-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-38"><a href="#cb214-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the plot</span></span>
<span id="cb214-39"><a href="#cb214-39" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(bar_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="boxplot-of-number-of-nuclie-per-donor" class="level2">
<h2 class="anchored" data-anchor-id="boxplot-of-number-of-nuclie-per-donor">Boxplot of number of nuclie per donor</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combined half violin plot, boxplot, and jittered points</span></span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>num_nuclei <span class="ot">&lt;-</span> meta_data <span class="sc">|&gt;</span></span>
<span id="cb215-3"><a href="#cb215-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(abbreviated_idents, donor, diagnosis) <span class="sc">|&gt;</span></span>
<span id="cb215-4"><a href="#cb215-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">num_nuclei =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'abbreviated_idents', 'donor'. You can
override using the `.groups` argument.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb217"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a>num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb217-2"><a href="#cb217-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> abbreviated_idents, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb217-3"><a href="#cb217-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span>  <span class="co"># Hide outliers to avoid overlap with jittered points</span></span>
<span id="cb217-4"><a href="#cb217-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>), <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb217-5"><a href="#cb217-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_brewer(palette = "Set3") +  # Choose a color palette</span></span>
<span id="cb217-6"><a href="#cb217-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb217-7"><a href="#cb217-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of Nuclei"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="04_figures_files/figure-html/unnamed-chunk-11-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="04_figures_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb218"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1" aria-hidden="true" tabindex="-1"></a>num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb218-2"><a href="#cb218-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> abbreviated_idents, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb218-3"><a href="#cb218-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_violin(trim = FALSE, alpha = 0.5, position = position_nudge(x = 0.2)) +</span></span>
<span id="cb218-4"><a href="#cb218-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">x =</span> <span class="fl">0.2</span>), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb218-5"><a href="#cb218-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>), <span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb218-6"><a href="#cb218-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip coordinates to make it a half violin plot</span></span>
<span id="cb218-7"><a href="#cb218-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb218-8"><a href="#cb218-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of Nuclei"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="04_figures_files/figure-html/unnamed-chunk-11-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="04_figures_files/figure-html/unnamed-chunk-11-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a>num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb219-2"><a href="#cb219-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> abbreviated_idents, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb219-3"><a href="#cb219-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_violin(trim = FALSE, alpha = 0.5, position = position_nudge(x = 0.2)) +</span></span>
<span id="cb219-4"><a href="#cb219-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb219-5"><a href="#cb219-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb219-6"><a href="#cb219-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip coordinates to make it a half violin plot</span></span>
<span id="cb219-7"><a href="#cb219-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb219-8"><a href="#cb219-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of Nuclei"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="04_figures_files/figure-html/unnamed-chunk-11-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="04_figures_files/figure-html/unnamed-chunk-11-3.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb220"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb220-1"><a href="#cb220-1" aria-hidden="true" tabindex="-1"></a>boxplot <span class="ot">&lt;-</span> num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb220-2"><a href="#cb220-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> abbreviated_idents, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb220-3"><a href="#cb220-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_violin(trim = FALSE, alpha = 0.5, position = position_nudge(x = 0.2)) +</span></span>
<span id="cb220-4"><a href="#cb220-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb220-5"><a href="#cb220-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">size =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb220-6"><a href="#cb220-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb220-7"><a href="#cb220-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,</span>
<span id="cb220-8"><a href="#cb220-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of Nuclei"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb221"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb221-1"><a href="#cb221-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb221-2"><a href="#cb221-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)</span>
<span id="cb221-3"><a href="#cb221-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Raincloud plot</span></span>
<span id="cb221-4"><a href="#cb221-4" aria-hidden="true" tabindex="-1"></a>raincloud_plot <span class="ot">&lt;-</span> num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb221-5"><a href="#cb221-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> subcelltype_annotations, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb221-6"><a href="#cb221-6" aria-hidden="true" tabindex="-1"></a>  ggdist<span class="sc">::</span><span class="fu">stat_halfeye</span>(</span>
<span id="cb221-7"><a href="#cb221-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjust =</span> <span class="fl">0.5</span>, </span>
<span id="cb221-8"><a href="#cb221-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.6</span>, </span>
<span id="cb221-9"><a href="#cb221-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.width =</span> <span class="dv">0</span>, </span>
<span id="cb221-10"><a href="#cb221-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">justification =</span> <span class="sc">-</span><span class="fl">0.3</span>, </span>
<span id="cb221-11"><a href="#cb221-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">point_colour =</span> <span class="cn">NA</span></span>
<span id="cb221-12"><a href="#cb221-12" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb221-13"><a href="#cb221-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(</span>
<span id="cb221-14"><a href="#cb221-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.15</span>, </span>
<span id="cb221-15"><a href="#cb221-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">outlier.shape =</span> <span class="cn">NA</span>, </span>
<span id="cb221-16"><a href="#cb221-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">x =</span> <span class="fl">0.15</span>)</span>
<span id="cb221-17"><a href="#cb221-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb221-18"><a href="#cb221-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(</span>
<span id="cb221-19"><a href="#cb221-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.1</span>,</span>
<span id="cb221-20"><a href="#cb221-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.6</span>,</span>
<span id="cb221-21"><a href="#cb221-21" aria-hidden="true" tabindex="-1"></a>    <span class="co">#position = position_nudge(x = 0.2)</span></span>
<span id="cb221-22"><a href="#cb221-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb221-23"><a href="#cb221-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb221-24"><a href="#cb221-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb221-25"><a href="#cb221-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Cell Type"</span>,</span>
<span id="cb221-26"><a href="#cb221-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Number of Nuclei"</span></span>
<span id="cb221-27"><a href="#cb221-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb221-28"><a href="#cb221-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb221-29"><a href="#cb221-29" aria-hidden="true" tabindex="-1"></a>num_nuclei <span class="ot">&lt;-</span> meta_data <span class="sc">|&gt;</span></span>
<span id="cb221-30"><a href="#cb221-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(abbreviated_idents, donor, diagnosis) <span class="sc">|&gt;</span></span>
<span id="cb221-31"><a href="#cb221-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">num_nuclei =</span> <span class="fu">n</span>())</span>
<span id="cb221-32"><a href="#cb221-32" aria-hidden="true" tabindex="-1"></a>num_nuclei <span class="ot">&lt;-</span> meta_data <span class="sc">|&gt;</span></span>
<span id="cb221-33"><a href="#cb221-33" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(highlevel_manual_annotations, donor, diagnosis) <span class="sc">|&gt;</span></span>
<span id="cb221-34"><a href="#cb221-34" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">num_nuclei =</span> <span class="fu">n</span>())</span>
<span id="cb221-35"><a href="#cb221-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb221-36"><a href="#cb221-36" aria-hidden="true" tabindex="-1"></a>raincloud_plot <span class="ot">&lt;-</span> num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb221-37"><a href="#cb221-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> highlevel_manual_annotations, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb221-38"><a href="#cb221-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_slab</span>(<span class="fu">aes</span>(<span class="at">thickness =</span> <span class="fu">after_stat</span>(pdf<span class="sc">*</span>n)), <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb221-39"><a href="#cb221-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(<span class="at">side =</span> <span class="st">"bottom"</span>, <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">slab_linewidth =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb221-40"><a href="#cb221-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb221-41"><a href="#cb221-41" aria-hidden="true" tabindex="-1"></a>raincloud_plot <span class="ot">&lt;-</span> num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb221-42"><a href="#cb221-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> highlevel_manual_annotations, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb221-43"><a href="#cb221-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_slab</span>(<span class="fu">aes</span>(<span class="at">thickness =</span> <span class="fu">after_stat</span>(pdf<span class="sc">*</span>n)), <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb221-44"><a href="#cb221-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(<span class="at">side =</span> <span class="st">"bottom"</span>, <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">slab_linewidth =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>())</span>
<span id="cb221-45"><a href="#cb221-45" aria-hidden="true" tabindex="-1"></a>raincloud_plot <span class="ot">&lt;-</span> num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb221-46"><a href="#cb221-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> abbreviated_idents, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb221-47"><a href="#cb221-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_slab</span>(<span class="fu">aes</span>(<span class="at">thickness =</span> <span class="fu">after_stat</span>(pdf<span class="sc">*</span>n)), <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb221-48"><a href="#cb221-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(<span class="at">side =</span> <span class="st">"bottom"</span>, <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">slab_linewidth =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>())</span>
<span id="cb221-49"><a href="#cb221-49" aria-hidden="true" tabindex="-1"></a>raincloud_plot <span class="ot">&lt;-</span> num_nuclei <span class="sc">|&gt;</span></span>
<span id="cb221-50"><a href="#cb221-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> abbreviated_idents, <span class="at">y =</span> num_nuclei, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb221-51"><a href="#cb221-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_slab</span>(<span class="fu">aes</span>(<span class="at">thickness =</span> <span class="fu">after_stat</span>(pdf<span class="sc">*</span>n)), <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb221-52"><a href="#cb221-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(<span class="at">side =</span> <span class="st">"bottom"</span>, <span class="at">scale =</span> <span class="fl">0.7</span>, <span class="at">slab_linewidth =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb221-53"><a href="#cb221-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb221-54"><a href="#cb221-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the raincloud plot</span></span>
<span id="cb221-55"><a href="#cb221-55" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(raincloud_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="explanation" class="level3">
<h3 class="anchored" data-anchor-id="explanation">Explanation</h3>
<ul>
<li><strong>stat_halfeye</strong>: Creates the half violin plot part of the raincloud plot.</li>
<li><strong>geom_boxplot</strong>: Adds a boxplot to show summary statistics like the median and interquartile range.</li>
<li><strong>geom_jitter</strong>: Adds individual data points to show the actual data distribution and variability.</li>
<li><strong>coord_flip</strong>: Flips the coordinates to make the violin plot horizontal, which is typical for raincloud plots.</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb222-2"><a href="#cb222-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magick)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Linking to ImageMagick 6.9.11.60
Enabled features: fontconfig, freetype, fftw, heic, lcms, pango, webp, x11
Disabled features: cairo, ghostscript, raw, rsvg</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using 16 threads</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Read external images</span></span>
<span id="cb225-3"><a href="#cb225-3" aria-hidden="true" tabindex="-1"></a>img1 <span class="ot">&lt;-</span> <span class="fu">image_read</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/detailed_vascular_prep.png"</span>))</span>
<span id="cb225-4"><a href="#cb225-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb225-5"><a href="#cb225-5" aria-hidden="true" tabindex="-1"></a>vascular_cell_diff <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/cell_pop_difference_level2_vascular.rds"</span>)) <span class="sc">+</span></span>
<span id="cb225-6"><a href="#cb225-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb225-7"><a href="#cb225-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_discrete</span>(<span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb225-8"><a href="#cb225-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Case"</span>, <span class="st">"Control"</span>)) <span class="sc">+</span></span>
<span id="cb225-9"><a href="#cb225-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">""</span>, <span class="at">labels =</span> <span class="st">"EndoMT"</span>) <span class="sc">+</span></span>
<span id="cb225-10"><a href="#cb225-10" aria-hidden="true" tabindex="-1"></a>  custom_theme</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for fill is already present.
Adding another scale for fill, which will replace the existing scale.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb227-2"><a href="#cb227-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb227-3"><a href="#cb227-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb227-4"><a href="#cb227-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine ggplot objects</span></span>
<span id="cb227-5"><a href="#cb227-5" aria-hidden="true" tabindex="-1"></a>row_2 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(umap_vascular_parenchymal, umap, <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"B"</span>, <span class="st">"C"</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span>
<span id="cb227-6"><a href="#cb227-6" aria-hidden="true" tabindex="-1"></a>row_4 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(boxplot, vascular_cell_diff, <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"E"</span>, <span class="st">"F"</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database
Warning in grid.Call(C_stringMetric, as.graphicsAnnot(x$label)): font family
'Arial' not found in PostScript font database</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb233-2"><a href="#cb233-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add external images</span></span>
<span id="cb233-3"><a href="#cb233-3" aria-hidden="true" tabindex="-1"></a>img1_gg <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(img1)</span>
<span id="cb233-4"><a href="#cb233-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-5"><a href="#cb233-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine everything</span></span>
<span id="cb233-6"><a href="#cb233-6" aria-hidden="true" tabindex="-1"></a>final_plot <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(img1_gg, row_2, bar_plot, row_4, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">""</span>, <span class="st">"D"</span>, <span class="st">""</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a>final_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 35 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-13-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="04_figures_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the final compound figure</span></span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/figure1.png"</span>), <span class="at">plot =</span> final_plot, </span>
<span id="cb238-4"><a href="#cb238-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">15</span>, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 11 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
</div></div>
</section>
</section>
<section id="figure-2" class="level2">
<h2 class="anchored" data-anchor-id="figure-2">Figure 2</h2>
<p>This figure focuses on the risk scores from magma</p>
<section id="go-pathways" class="level3">
<h3 class="anchored" data-anchor-id="go-pathways">GO pathways</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a>go_combined <span class="ot">&lt;-</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a>  qs<span class="sc">::</span><span class="fu">qread</span>(</span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>    here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb240-4"><a href="#cb240-4" aria-hidden="true" tabindex="-1"></a>      <span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/go_data_sig_gene_only.qs"</span></span>
<span id="cb240-5"><a href="#cb240-5" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb240-6"><a href="#cb240-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a>make_combined_go_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(go_combined, plot_title, <span class="at">top_n =</span> <span class="dv">10</span>) {</span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select top terms for each ontology, excluding the specific terms already selected</span></span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a>  top_terms <span class="ot">&lt;-</span> go_combined <span class="sc">%&gt;%</span></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Count <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(Ontology) <span class="sc">%&gt;%</span></span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_min</span>(p.adjust, <span class="at">n =</span> top_n) <span class="sc">%&gt;%</span></span>
<span id="cb241-8"><a href="#cb241-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb241-9"><a href="#cb241-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Ontology =</span> <span class="fu">toupper</span>(Ontology))</span>
<span id="cb241-10"><a href="#cb241-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb241-11"><a href="#cb241-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate Gene Ratio</span></span>
<span id="cb241-12"><a href="#cb241-12" aria-hidden="true" tabindex="-1"></a>  top_terms<span class="sc">$</span>GeneRatio <span class="ot">&lt;-</span> <span class="fu">sapply</span>(top_terms<span class="sc">$</span>GeneRatio, <span class="cf">function</span>(x) {</span>
<span id="cb241-13"><a href="#cb241-13" aria-hidden="true" tabindex="-1"></a>    parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(x, <span class="st">"/"</span>))</span>
<span id="cb241-14"><a href="#cb241-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.numeric</span>(parts[<span class="dv">1</span>]) <span class="sc">/</span> <span class="fu">as.numeric</span>(parts[<span class="dv">2</span>])</span>
<span id="cb241-15"><a href="#cb241-15" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb241-16"><a href="#cb241-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb241-17"><a href="#cb241-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create dot plot</span></span>
<span id="cb241-18"><a href="#cb241-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(top_terms, <span class="fu">aes</span>(<span class="at">x =</span> GeneRatio, <span class="at">y =</span> <span class="fu">reorder</span>(Description, GeneRatio))) <span class="sc">+</span></span>
<span id="cb241-19"><a href="#cb241-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> Count, <span class="at">color =</span> p.adjust)) <span class="sc">+</span></span>
<span id="cb241-20"><a href="#cb241-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"red"</span>, <span class="at">high =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb241-21"><a href="#cb241-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>( <span class="sc">~</span> Ontology, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb241-22"><a href="#cb241-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> plot_title,</span>
<span id="cb241-23"><a href="#cb241-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Gene Ratio"</span>,</span>
<span id="cb241-24"><a href="#cb241-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb241-25"><a href="#cb241-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb241-26"><a href="#cb241-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb241-27"><a href="#cb241-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb241-28"><a href="#cb241-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb241-29"><a href="#cb241-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb241-30"><a href="#cb241-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb241-31"><a href="#cb241-31" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb241-32"><a href="#cb241-32" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb241-33"><a href="#cb241-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-34"><a href="#cb241-34" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span></span>
<span id="cb241-35"><a href="#cb241-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(go_combined,</span>
<span id="cb241-36"><a href="#cb241-36" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(go_combined),</span>
<span id="cb241-37"><a href="#cb241-37" aria-hidden="true" tabindex="-1"></a>      make_combined_go_plots,</span>
<span id="cb241-38"><a href="#cb241-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">top_n =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a>celltype_abbrev_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(celltypes_abbrev) <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>(<span class="st">"celltype"</span>)</span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(celltype_abbrev_df, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/celltype_abbreviations.tsv"</span>))</span>
<span id="cb242-3"><a href="#cb242-3" aria-hidden="true" tabindex="-1"></a>top_terms <span class="ot">&lt;-</span> <span class="fu">map2_dfr</span>(go_combined, <span class="fu">names</span>(go_combined), <span class="sc">~</span> .x <span class="sc">|&gt;</span></span>
<span id="cb242-4"><a href="#cb242-4" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb242-5"><a href="#cb242-5" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">group_by</span>(Ontology) <span class="sc">|&gt;</span></span>
<span id="cb242-6"><a href="#cb242-6" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">slice_min</span>(p.adjust, <span class="at">n =</span> <span class="dv">20</span>) <span class="sc">|&gt;</span></span>
<span id="cb242-7"><a href="#cb242-7" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb242-8"><a href="#cb242-8" aria-hidden="true" tabindex="-1"></a>                        dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Ontology =</span> <span class="fu">toupper</span>(Ontology), </span>
<span id="cb242-9"><a href="#cb242-9" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">celltype =</span> .y,</span>
<span id="cb242-10"><a href="#cb242-10" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">amyloid_pathways =</span> <span class="fu">if_else</span>(<span class="fu">grepl</span>(<span class="st">"amyloid"</span>, Description), <span class="st">"Amyloid-related"</span>, <span class="st">"Other pathways"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb242-11"><a href="#cb242-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb242-12"><a href="#cb242-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltypes_abbrev =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(celltypes_abbrev), celltype, celltypes_abbrev))</span>
<span id="cb242-13"><a href="#cb242-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb242-14"><a href="#cb242-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate Gene Ratio</span></span>
<span id="cb242-15"><a href="#cb242-15" aria-hidden="true" tabindex="-1"></a>top_terms<span class="sc">$</span>GeneRatio <span class="ot">&lt;-</span> <span class="fu">sapply</span>(top_terms<span class="sc">$</span>GeneRatio, <span class="cf">function</span>(x) {</span>
<span id="cb242-16"><a href="#cb242-16" aria-hidden="true" tabindex="-1"></a>  parts <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(x, <span class="st">"/"</span>))</span>
<span id="cb242-17"><a href="#cb242-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>(parts[<span class="dv">1</span>]) <span class="sc">/</span> <span class="fu">as.numeric</span>(parts[<span class="dv">2</span>])</span>
<span id="cb242-18"><a href="#cb242-18" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb242-19"><a href="#cb242-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb242-20"><a href="#cb242-20" aria-hidden="true" tabindex="-1"></a>go_plot_all <span class="ot">&lt;-</span> top_terms <span class="sc">|&gt;</span></span>
<span id="cb242-21"><a href="#cb242-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter to just BP for make plot easier to parse</span></span>
<span id="cb242-22"><a href="#cb242-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Ontology <span class="sc">==</span> <span class="st">"BP"</span>) <span class="sc">|&gt;</span></span>
<span id="cb242-23"><a href="#cb242-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="fu">log10</span>(p.adjust), <span class="at">y =</span> <span class="fu">reorder</span>(Description, <span class="sc">-</span><span class="fu">log10</span>(p.adjust)))) <span class="sc">+</span></span>
<span id="cb242-24"><a href="#cb242-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> Count, <span class="at">color =</span> p.adjust)) <span class="sc">+</span></span>
<span id="cb242-25"><a href="#cb242-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"red"</span>, <span class="at">high =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb242-26"><a href="#cb242-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(celltype <span class="sc">~</span> Ontology, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb242-27"><a href="#cb242-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Gene Ratio"</span>,</span>
<span id="cb242-28"><a href="#cb242-28" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb242-29"><a href="#cb242-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb242-30"><a href="#cb242-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb242-31"><a href="#cb242-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb242-32"><a href="#cb242-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb242-33"><a href="#cb242-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb242-34"><a href="#cb242-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb242-35"><a href="#cb242-35" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb242-36"><a href="#cb242-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb242-37"><a href="#cb242-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Create dot plot</span></span>
<span id="cb242-38"><a href="#cb242-38" aria-hidden="true" tabindex="-1"></a>go_plot_all <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(top_terms, <span class="fu">aes</span>(<span class="at">x =</span> GeneRatio, <span class="at">y =</span> <span class="fu">reorder</span>(Description, GeneRatio))) <span class="sc">+</span></span>
<span id="cb242-39"><a href="#cb242-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> Count, <span class="at">color =</span> p.adjust)) <span class="sc">+</span></span>
<span id="cb242-40"><a href="#cb242-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"red"</span>, <span class="at">high =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb242-41"><a href="#cb242-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(celltype <span class="sc">~</span> Ontology, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb242-42"><a href="#cb242-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Gene Ratio"</span>,</span>
<span id="cb242-43"><a href="#cb242-43" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb242-44"><a href="#cb242-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb242-45"><a href="#cb242-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb242-46"><a href="#cb242-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb242-47"><a href="#cb242-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb242-48"><a href="#cb242-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb242-49"><a href="#cb242-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb242-50"><a href="#cb242-50" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb242-51"><a href="#cb242-51" aria-hidden="true" tabindex="-1"></a>go_plot_all <span class="ot">&lt;-</span> top_terms <span class="sc">|&gt;</span></span>
<span id="cb242-52"><a href="#cb242-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter to just BP for make plot easier to parse</span></span>
<span id="cb242-53"><a href="#cb242-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Ontology <span class="sc">==</span> <span class="st">"BP"</span>) <span class="sc">|&gt;</span></span>
<span id="cb242-54"><a href="#cb242-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GeneRatio, <span class="at">y =</span> <span class="fu">reorder</span>(Description, GeneRatio))) <span class="sc">+</span></span>
<span id="cb242-55"><a href="#cb242-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> Count, <span class="at">color =</span> p.adjust)) <span class="sc">+</span></span>
<span id="cb242-56"><a href="#cb242-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"red"</span>, <span class="at">high =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb242-57"><a href="#cb242-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(amyloid_pathways <span class="sc">~</span> celltypes_abbrev, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb242-58"><a href="#cb242-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Gene Ratio"</span>,</span>
<span id="cb242-59"><a href="#cb242-59" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb242-60"><a href="#cb242-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb242-61"><a href="#cb242-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb242-62"><a href="#cb242-62" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = "bottom",</span></span>
<span id="cb242-63"><a href="#cb242-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb242-64"><a href="#cb242-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb242-65"><a href="#cb242-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb242-66"><a href="#cb242-66" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="magma-heatmap" class="level3">
<h3 class="anchored" data-anchor-id="magma-heatmap">MAGMA heatmap</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a>magma_heatmap_data <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/magma_celltype_heatmap_data.qs"</span>))</span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-3"><a href="#cb243-3" aria-hidden="true" tabindex="-1"></a>heatmap_annotations <span class="ot">&lt;-</span> <span class="cf">function</span>(heatmap_data) {</span>
<span id="cb243-4"><a href="#cb243-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb243-5"><a href="#cb243-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Define significance thresholds</span></span>
<span id="cb243-6"><a href="#cb243-6" aria-hidden="true" tabindex="-1"></a>  sig_threshold1 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>) <span class="co"># Corresponds to p &lt; 0.05</span></span>
<span id="cb243-7"><a href="#cb243-7" aria-hidden="true" tabindex="-1"></a>  sig_threshold2 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>) <span class="co"># Corresponds to p &lt; 0.01</span></span>
<span id="cb243-8"><a href="#cb243-8" aria-hidden="true" tabindex="-1"></a>  sig_threshold3 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.001</span>) <span class="co"># Corresponds to p &lt; 0.001</span></span>
<span id="cb243-9"><a href="#cb243-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb243-10"><a href="#cb243-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create an annotation matrix with the same dimensions as 'heatmap'</span></span>
<span id="cb243-11"><a href="#cb243-11" aria-hidden="true" tabindex="-1"></a>  annotation_matrix <span class="ot">&lt;-</span></span>
<span id="cb243-12"><a href="#cb243-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">matrix</span>(</span>
<span id="cb243-13"><a href="#cb243-13" aria-hidden="true" tabindex="-1"></a>      <span class="st">""</span>,</span>
<span id="cb243-14"><a href="#cb243-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">nrow =</span> <span class="fu">nrow</span>(heatmap_data),</span>
<span id="cb243-15"><a href="#cb243-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">ncol =</span> <span class="fu">ncol</span>(heatmap_data)</span>
<span id="cb243-16"><a href="#cb243-16" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb243-17"><a href="#cb243-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb243-18"><a href="#cb243-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Populate the annotation matrix with '*', '**', or '***' based on significance levels</span></span>
<span id="cb243-19"><a href="#cb243-19" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold3] <span class="ot">&lt;-</span></span>
<span id="cb243-20"><a href="#cb243-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"***"</span></span>
<span id="cb243-21"><a href="#cb243-21" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold2 <span class="sc">&amp;</span></span>
<span id="cb243-22"><a href="#cb243-22" aria-hidden="true" tabindex="-1"></a>                      heatmap_data <span class="sc">&lt;=</span> sig_threshold3] <span class="ot">&lt;-</span> <span class="st">"**"</span></span>
<span id="cb243-23"><a href="#cb243-23" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold1 <span class="sc">&amp;</span></span>
<span id="cb243-24"><a href="#cb243-24" aria-hidden="true" tabindex="-1"></a>                      heatmap_data <span class="sc">&lt;=</span> sig_threshold2] <span class="ot">&lt;-</span> <span class="st">"*"</span></span>
<span id="cb243-25"><a href="#cb243-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb243-26"><a href="#cb243-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb243-27"><a href="#cb243-27" aria-hidden="true" tabindex="-1"></a><span class="fu">return</span>(annotation_matrix)</span>
<span id="cb243-28"><a href="#cb243-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb243-29"><a href="#cb243-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-30"><a href="#cb243-30" aria-hidden="true" tabindex="-1"></a>row_anno <span class="ot">&lt;-</span> <span class="fu">heatmap_annotations</span>(magma_heatmap_data)</span>
<span id="cb243-31"><a href="#cb243-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-32"><a href="#cb243-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb243-33"><a href="#cb243-33" aria-hidden="true" tabindex="-1"></a>heatmap <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(magma_heatmap_data, </span>
<span id="cb243-34"><a href="#cb243-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"-log10 p-value"</span>, </span>
<span id="cb243-35"><a href="#cb243-35" aria-hidden="true" tabindex="-1"></a><span class="co">#        row_names_side = "left", </span></span>
<span id="cb243-36"><a href="#cb243-36" aria-hidden="true" tabindex="-1"></a><span class="co">#        right_annotation = row_anno,</span></span>
<span id="cb243-37"><a href="#cb243-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the clustering method if necessary</span></span>
<span id="cb243-38"><a href="#cb243-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb243-39"><a href="#cb243-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb243-40"><a href="#cb243-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb243-41"><a href="#cb243-41" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb243-42"><a href="#cb243-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_rot =</span> <span class="dv">45</span>,</span>
<span id="cb243-43"><a href="#cb243-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_centered =</span> <span class="cn">TRUE</span>,</span>
<span id="cb243-44"><a href="#cb243-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use the defined color palette</span></span>
<span id="cb243-45"><a href="#cb243-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb243-46"><a href="#cb243-46" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (row_anno[i, j] <span class="sc">!=</span> <span class="st">""</span>) {</span>
<span id="cb243-47"><a href="#cb243-47" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(row_anno[i, j], x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb243-48"><a href="#cb243-48" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb243-49"><a href="#cb243-49" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb243-50"><a href="#cb243-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">6</span>),  <span class="co"># Set row label text size</span></span>
<span id="cb243-51"><a href="#cb243-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb243-52"><a href="#cb243-52" aria-hidden="true" tabindex="-1"></a>     <span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">"cm"</span>))</span>
<span id="cb243-53"><a href="#cb243-53" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">=</span> <span class="fu">grid.grabExpr</span>(<span class="fu">draw</span>(heatmap))</span>
<span id="cb243-54"><a href="#cb243-54" aria-hidden="true" tabindex="-1"></a>row_1 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(p2, go_plot_all, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>), <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">2.5</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb245"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb245-1"><a href="#cb245-1" aria-hidden="true" tabindex="-1"></a>row_1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-17-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="04_figures_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/sig_genelist/results"</span>), </span>
<span id="cb246-2"><a href="#cb246-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pattern =</span> <span class="st">"*.gsa.out"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb246-3"><a href="#cb246-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-4"><a href="#cb246-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read data</span></span>
<span id="cb246-5"><a href="#cb246-5" aria-hidden="true" tabindex="-1"></a>parse_magma_set_data <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb246-6"><a href="#cb246-6" aria-hidden="true" tabindex="-1"></a>  file_lines <span class="ot">&lt;-</span> <span class="fu">readLines</span>(file)</span>
<span id="cb246-7"><a href="#cb246-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the index of the line judt before the data starts</span></span>
<span id="cb246-8"><a href="#cb246-8" aria-hidden="true" tabindex="-1"></a>  data_starts_index <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"^# CONDITIONED_"</span>, file_lines)</span>
<span id="cb246-9"><a href="#cb246-9" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read_table</span>(file, <span class="at">skip =</span> data_starts_index[<span class="fu">length</span>(data_starts_index)], <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb246-10"><a href="#cb246-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(data) </span>
<span id="cb246-11"><a href="#cb246-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb246-12"><a href="#cb246-12" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">map</span>(files, parse_magma_set_data)</span>
<span id="cb246-13"><a href="#cb246-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-14"><a href="#cb246-14" aria-hidden="true" tabindex="-1"></a>files_sub <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(<span class="fu">basename</span>(files), <span class="st">"level2_all_controls_top-ten-percent_AD_"</span>, <span class="st">""</span>) <span class="sc">|&gt;</span></span>
<span id="cb246-15"><a href="#cb246-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"_conditional"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb246-16"><a href="#cb246-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use sub() to remove all characters after the first dot</span></span>
<span id="cb246-17"><a href="#cb246-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*$"</span>, <span class="st">""</span>, .)</span>
<span id="cb246-18"><a href="#cb246-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-19"><a href="#cb246-19" aria-hidden="true" tabindex="-1"></a>process_magma_data <span class="ot">&lt;-</span></span>
<span id="cb246-20"><a href="#cb246-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(df,</span>
<span id="cb246-21"><a href="#cb246-21" aria-hidden="true" tabindex="-1"></a>           filename,</span>
<span id="cb246-22"><a href="#cb246-22" aria-hidden="true" tabindex="-1"></a>           <span class="at">celltypes_to_include =</span> <span class="fu">c</span>(<span class="st">"Perivascular-FB-KAZN2"</span>, <span class="st">"Pericyte-2"</span>, <span class="st">"Microglia-activated"</span>)) {</span>
<span id="cb246-23"><a href="#cb246-23" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(VARIABLE <span class="sc">%in%</span> celltypes_to_include) <span class="sc">|&gt;</span></span>
<span id="cb246-24"><a href="#cb246-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add groups and adjusted pvals to data</span></span>
<span id="cb246-25"><a href="#cb246-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">conditioned_on =</span> filename) <span class="sc">|&gt;</span></span>
<span id="cb246-26"><a href="#cb246-26" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FULL_NAME =</span> <span class="fu">tolower</span>(FULL_NAME)) <span class="sc">|&gt;</span></span>
<span id="cb246-27"><a href="#cb246-27" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(FULL_NAME <span class="sc">!=</span> conditioned_on) <span class="sc">|&gt;</span></span>
<span id="cb246-28"><a href="#cb246-28" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">p_adj =</span> <span class="fu">p.adjust</span>(P, <span class="at">method =</span> <span class="st">"bonferroni"</span>))</span>
<span id="cb246-29"><a href="#cb246-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb246-30"><a href="#cb246-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb246-31"><a href="#cb246-31" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">map2</span>(data, files_sub, <span class="sc">~</span> <span class="fu">process_magma_data</span>(.x, .y)) <span class="sc">|&gt;</span></span>
<span id="cb246-32"><a href="#cb246-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb246-33"><a href="#cb246-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="conditional-heatmap" class="level4">
<h4 class="anchored" data-anchor-id="conditional-heatmap">Conditional heatmap</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb247"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb247-1"><a href="#cb247-1" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>abbrev <span class="ot">&lt;-</span> celltypes_abbrev[<span class="fu">as.character</span>(df<span class="sc">$</span>VARIABLE)]</span>
<span id="cb247-2"><a href="#cb247-2" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>abbrev <span class="ot">&lt;-</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(df<span class="sc">$</span>abbrev), df<span class="sc">$</span>VARIABLE, df<span class="sc">$</span>abbrev)</span>
<span id="cb247-3"><a href="#cb247-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-4"><a href="#cb247-4" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>abbrev[<span class="fu">grepl</span>(<span class="st">"Ex-Neuron-L2-3-C"</span>, df<span class="sc">$</span>abbrev)] <span class="ot">&lt;-</span> <span class="st">"Ex-Neuro-4"</span></span>
<span id="cb247-5"><a href="#cb247-5" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>abbrev[<span class="fu">grepl</span>(<span class="st">"In-Neuron-ADARB2-IL1RAPL2"</span>, df<span class="sc">$</span>abbrev)] <span class="ot">&lt;-</span> <span class="st">"In-Neuro-2"</span></span>
<span id="cb247-6"><a href="#cb247-6" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb247-7"><a href="#cb247-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(abbrev <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Perivascular-FB-2"</span>, <span class="st">"Pericyte-2"</span>, <span class="st">"Microglia-A"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb247-8"><a href="#cb247-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">log_p =</span> <span class="sc">-</span><span class="fu">log10</span>(p_adj))</span>
<span id="cb247-9"><a href="#cb247-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-10"><a href="#cb247-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Nest data on broad groups</span></span>
<span id="cb247-11"><a href="#cb247-11" aria-hidden="true" tabindex="-1"></a>df_nest <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span> </span>
<span id="cb247-12"><a href="#cb247-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(conditioned_on) <span class="sc">|&gt;</span></span>
<span id="cb247-13"><a href="#cb247-13" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span>
<span id="cb247-14"><a href="#cb247-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-15"><a href="#cb247-15" aria-hidden="true" tabindex="-1"></a>get_heatmap_data <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb247-16"><a href="#cb247-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Prepare the data for the heatmap</span></span>
<span id="cb247-17"><a href="#cb247-17" aria-hidden="true" tabindex="-1"></a>  heatmap_data <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb247-18"><a href="#cb247-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(abbrev, log_p, conditioned_on) <span class="sc">|&gt;</span></span>
<span id="cb247-19"><a href="#cb247-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> abbrev, <span class="at">values_from =</span> log_p) <span class="sc">|&gt;</span></span>
<span id="cb247-20"><a href="#cb247-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb247-21"><a href="#cb247-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb247-22"><a href="#cb247-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(heatmap_data) <span class="ot">&lt;-</span> heatmap_data<span class="sc">$</span>conditioned_on </span>
<span id="cb247-23"><a href="#cb247-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove the first column if it contains row names (cell types)</span></span>
<span id="cb247-24"><a href="#cb247-24" aria-hidden="true" tabindex="-1"></a>  heatmap_data <span class="ot">&lt;-</span> heatmap_data[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb247-25"><a href="#cb247-25" aria-hidden="true" tabindex="-1"></a>  heatmap <span class="ot">&lt;-</span> <span class="fu">t</span>(heatmap_data)</span>
<span id="cb247-26"><a href="#cb247-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace any NAs with 0</span></span>
<span id="cb247-27"><a href="#cb247-27" aria-hidden="true" tabindex="-1"></a>  <span class="co">#heatmap[is.na(heatmap)] &lt;- 0</span></span>
<span id="cb247-28"><a href="#cb247-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(heatmap)</span>
<span id="cb247-29"><a href="#cb247-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb247-30"><a href="#cb247-30" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> <span class="fu">get_heatmap_data</span>(df)</span>
<span id="cb247-31"><a href="#cb247-31" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heatmap_data) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Microglia-A"</span>, <span class="st">"Pericyte-2"</span>, <span class="st">"Perivascular-FB-2"</span>)</span>
<span id="cb247-32"><a href="#cb247-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-33"><a href="#cb247-33" aria-hidden="true" tabindex="-1"></a>row_anno <span class="ot">&lt;-</span> <span class="fu">heatmap_annotations</span>(heatmap_data)</span>
<span id="cb247-34"><a href="#cb247-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-35"><a href="#cb247-35" aria-hidden="true" tabindex="-1"></a>heatmap_conditional <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(heatmap_data, </span>
<span id="cb247-36"><a href="#cb247-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"-log10 </span><span class="sc">\n</span><span class="st">adj p-value"</span>, </span>
<span id="cb247-37"><a href="#cb247-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_names_side =</span> <span class="st">"left"</span>, </span>
<span id="cb247-38"><a href="#cb247-38" aria-hidden="true" tabindex="-1"></a>        <span class="co">#right_annotation = row_anno_conditional,</span></span>
<span id="cb247-39"><a href="#cb247-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb247-40"><a href="#cb247-40" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (row_anno[i, j] <span class="sc">!=</span> <span class="st">""</span>) {</span>
<span id="cb247-41"><a href="#cb247-41" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(row_anno[i, j], x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb247-42"><a href="#cb247-42" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb247-43"><a href="#cb247-43" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb247-44"><a href="#cb247-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the clustering method if necessary</span></span>
<span id="cb247-45"><a href="#cb247-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb247-46"><a href="#cb247-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb247-47"><a href="#cb247-47" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_row_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb247-48"><a href="#cb247-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb247-49"><a href="#cb247-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_rot =</span> <span class="dv">0</span>,</span>
<span id="cb247-50"><a href="#cb247-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_title =</span> <span class="st">"Conditioned on"</span>,</span>
<span id="cb247-51"><a href="#cb247-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_title_side =</span> <span class="st">"bottom"</span>,</span>
<span id="cb247-52"><a href="#cb247-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),  <span class="co"># Set row label text size</span></span>
<span id="cb247-53"><a href="#cb247-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>))</span>
<span id="cb247-54"><a href="#cb247-54" aria-hidden="true" tabindex="-1"></a>heatmap_conditional</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-19-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9"><img src="04_figures_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">=</span> <span class="fu">grid.grabExpr</span>(<span class="fu">draw</span>(heatmap_conditional))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="avg-risk-gene-expression" class="level4">
<h4 class="anchored" data-anchor-id="avg-risk-gene-expression">Avg risk gene expression</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a>avg_gene_expr <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/risk_gene_avg_expression.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 56 Columns: 4
 Column specification 
Delimiter: "\t"
chr (1): features.plot
dbl (3): Pericyte-2, Perivascular-FB-KAZN2, Microglia-activated

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the gene column to create a matrix</span></span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>expression_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(avg_gene_expr[, <span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-4"><a href="#cb251-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Optionally, you can set row names to gene names</span></span>
<span id="cb251-5"><a href="#cb251-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expression_matrix) <span class="ot">&lt;-</span> avg_gene_expr<span class="sc">$</span>features.plot</span>
<span id="cb251-6"><a href="#cb251-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-7"><a href="#cb251-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(expression_matrix) <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">colnames</span>(expression_matrix), <span class="cf">function</span>(name) {</span>
<span id="cb251-8"><a href="#cb251-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (name <span class="sc">%in%</span> celltype_abbrev_df<span class="sc">$</span>celltype) {</span>
<span id="cb251-9"><a href="#cb251-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(celltype_abbrev_df<span class="sc">$</span>celltypes_abbrev[celltype_abbrev_df<span class="sc">$</span>celltype <span class="sc">==</span> name])</span>
<span id="cb251-10"><a href="#cb251-10" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb251-11"><a href="#cb251-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(name)</span>
<span id="cb251-12"><a href="#cb251-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb251-13"><a href="#cb251-13" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb251-14"><a href="#cb251-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-15"><a href="#cb251-15" aria-hidden="true" tabindex="-1"></a>avg_risk_gene_heatmap <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(expression_matrix,</span>
<span id="cb251-16"><a href="#cb251-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Expression"</span>,</span>
<span id="cb251-17"><a href="#cb251-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_rot =</span> <span class="dv">0</span>,</span>
<span id="cb251-18"><a href="#cb251-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_names_centered =</span> T,</span>
<span id="cb251-19"><a href="#cb251-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the clustering method if necessary</span></span>
<span id="cb251-20"><a href="#cb251-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb251-21"><a href="#cb251-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">TRUE</span>,</span>
<span id="cb251-22"><a href="#cb251-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_row_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb251-23"><a href="#cb251-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_column_dend =</span> <span class="cn">FALSE</span></span>
<span id="cb251-24"><a href="#cb251-24" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb251-25"><a href="#cb251-25" aria-hidden="true" tabindex="-1"></a>avg_risk_gene_heatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-20-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10"><img src="04_figures_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">=</span> <span class="fu">grid.grabExpr</span>(<span class="fu">draw</span>(avg_risk_gene_heatmap))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="mouse-phenotypes" class="level3">
<h3 class="anchored" data-anchor-id="mouse-phenotypes">Mouse phenotypes</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a>mpo <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/mpo_data.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 6 Columns: 8
 Column specification 
Delimiter: "\t"
chr (3): MP_ID, MP_name, celltype
dbl (5): SizeGeneList, SizeMPterm, GenesInMPterm, p_value, p_value_adj

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a>g1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb255-2"><a href="#cb255-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> mpo,</span>
<span id="cb255-3"><a href="#cb255-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mapping =</span> <span class="fu">aes</span>(</span>
<span id="cb255-4"><a href="#cb255-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">reorder</span>(MP_name, p_value_adj),</span>
<span id="cb255-5"><a href="#cb255-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value_adj),</span>
<span id="cb255-6"><a href="#cb255-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> GenesInMPterm,</span>
<span id="cb255-7"><a href="#cb255-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="sc">-</span><span class="fu">log10</span>(p_value_adj)</span>
<span id="cb255-8"><a href="#cb255-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb255-9"><a href="#cb255-9" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb255-10"><a href="#cb255-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb255-11"><a href="#cb255-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient2</span>(<span class="at">low =</span> <span class="st">"snow"</span>,</span>
<span id="cb255-12"><a href="#cb255-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mid =</span> <span class="st">"blue"</span>,</span>
<span id="cb255-13"><a href="#cb255-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">high =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb255-14"><a href="#cb255-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> celltype) <span class="sc">+</span></span>
<span id="cb255-15"><a href="#cb255-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb255-16"><a href="#cb255-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb255-17"><a href="#cb255-17" aria-hidden="true" tabindex="-1"></a>    <span class="co">#axis.text.x = element_text(angle = 45, hjust = 1),</span></span>
<span id="cb255-18"><a href="#cb255-18" aria-hidden="true" tabindex="-1"></a>    <span class="co">#legend.position = "bottom",</span></span>
<span id="cb255-19"><a href="#cb255-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb255-20"><a href="#cb255-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb255-21"><a href="#cb255-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb255-22"><a href="#cb255-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb255-23"><a href="#cb255-23" aria-hidden="true" tabindex="-1"></a>g1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-21-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11"><img src="04_figures_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="assemble-figure-2" class="level3">
<h3 class="anchored" data-anchor-id="assemble-figure-2">Assemble Figure 2</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a>magma_sig <span class="ot">&lt;-</span></span>
<span id="cb256-2"><a href="#cb256-2" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb256-3"><a href="#cb256-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/994_magma_inputs/results/sig_celltype_sig_genes.tsv"</span></span>
<span id="cb256-4"><a href="#cb256-4" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span></span>
<span id="cb256-5"><a href="#cb256-5" aria-hidden="true" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">|&gt;</span></span>
<span id="cb256-6"><a href="#cb256-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">entrezgene_id =</span> gene) <span class="sc">|&gt;</span></span>
<span id="cb256-7"><a href="#cb256-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Exclude the top one percents</span></span>
<span id="cb256-8"><a href="#cb256-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>percent <span class="sc">==</span> <span class="st">"top-one-percent"</span>) <span class="sc">|&gt;</span></span>
<span id="cb256-9"><a href="#cb256-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter to level 2</span></span>
<span id="cb256-10"><a href="#cb256-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype_level <span class="sc">==</span> <span class="st">"level2"</span>) <span class="sc">|&gt;</span></span>
<span id="cb256-11"><a href="#cb256-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="co">#|&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 125858 Columns: 22
 Column specification 
Delimiter: "\t"
chr  (8): gene_set, SetName, group, input_gene_list, gwas_background, input_...
dbl (14): GENE, CHR, START, STOP, NSNPS, NPARAM, N, ZSTAT, P, ZFITTED_BASE, ...

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#left_join(entrez_ids, by = join_by(entrezgene_id))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>ven diagram of sig risk genes from magma for each celltype</p>
</section>
<section id="ppi-network-plots" class="level3">
<h3 class="anchored" data-anchor-id="ppi-network-plots">PPI network plots</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a>process_image <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb260-2"><a href="#cb260-2" aria-hidden="true" tabindex="-1"></a>  img <span class="ot">&lt;-</span> <span class="fu">image_read</span>(file)</span>
<span id="cb260-3"><a href="#cb260-3" aria-hidden="true" tabindex="-1"></a>  img <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(img)</span>
<span id="cb260-4"><a href="#cb260-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(img)</span>
<span id="cb260-5"><a href="#cb260-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb260-6"><a href="#cb260-6" aria-hidden="true" tabindex="-1"></a>image_path <span class="ot">&lt;-</span> <span class="fu">c</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/"</span>))</span>
<span id="cb260-7"><a href="#cb260-7" aria-hidden="true" tabindex="-1"></a>images <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent.png"</span>, <span class="st">"Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent.png"</span>, <span class="st">"Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent.png"</span>)</span>
<span id="cb260-8"><a href="#cb260-8" aria-hidden="true" tabindex="-1"></a>images <span class="ot">&lt;-</span> <span class="fu">paste0</span>(image_path, images)</span>
<span id="cb260-9"><a href="#cb260-9" aria-hidden="true" tabindex="-1"></a>ppi_networks <span class="ot">&lt;-</span> <span class="fu">map</span>(images, process_image)</span>
<span id="cb260-10"><a href="#cb260-10" aria-hidden="true" tabindex="-1"></a>row2_sub <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(<span class="at">plotlist =</span> ppi_networks, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"D"</span>, <span class="st">"E"</span>, <span class="st">"F"</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span>
<span id="cb260-11"><a href="#cb260-11" aria-hidden="true" tabindex="-1"></a>row2_sub</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-23-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12"><img src="04_figures_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a>row_2 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(g1, row2_sub, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"C"</span>, <span class="st">""</span>), <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a>row_2 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(g1, row2_sub, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"C"</span>, <span class="st">""</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database
Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, : font
family 'Arial' not found in PostScript font database</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb265"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb265-1"><a href="#cb265-1" aria-hidden="true" tabindex="-1"></a>row_3 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(p3, p4, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"G"</span>, <span class="st">"H"</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span>
<span id="cb265-2"><a href="#cb265-2" aria-hidden="true" tabindex="-1"></a>row_3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="04_figures_files/figure-html/unnamed-chunk-24-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13"><img src="04_figures_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb266"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb266-1"><a href="#cb266-1" aria-hidden="true" tabindex="-1"></a>final_plot2 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(row_1, row_2, row_3, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">""</span>, <span class="st">""</span>, <span class="st">""</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span>
<span id="cb266-2"><a href="#cb266-2" aria-hidden="true" tabindex="-1"></a>final_plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="04_figures_files/figure-html/unnamed-chunk-24-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14"><img src="04_figures_files/figure-html/unnamed-chunk-24-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb267"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/figure2.png"</span>), <span class="at">plot =</span> final_plot2,</span>
<span id="cb267-2"><a href="#cb267-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">15</span>, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="figure-3---degs" class="level2">
<h2 class="anchored" data-anchor-id="figure-3---degs">Figure 3 - DEGs</h2>
<ul>
<li>need to start with barplot of number of up and downregulated genes.</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb268"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb268-1"><a href="#cb268-1" aria-hidden="true" tabindex="-1"></a>res_sig <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_no-apoe.csv"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb268-2"><a href="#cb268-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">|&gt;</span></span>
<span id="cb268-3"><a href="#cb268-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="sc">!</span>celltype <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ambiguous"</span>, <span class="st">"low-feature-cells"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb268-4"><a href="#cb268-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(celltype_abbrev_df, <span class="at">by =</span> <span class="fu">join_by</span>(celltype)) <span class="sc">|&gt;</span></span>
<span id="cb268-5"><a href="#cb268-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltypes_abbrev =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(celltypes_abbrev), celltype, celltypes_abbrev))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 996222 Columns: 9
 Column specification 
Delimiter: ","
chr (3): gene, celltype, deg_direction
dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb270"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add nuclei numbers</span></span>
<span id="cb270-2"><a href="#cb270-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">table</span>(sce<span class="sc">$</span>abbreviated_idents) <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb270-3"><a href="#cb270-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltypes_abbrev =</span> Var1, <span class="at">nuclei_n =</span> Freq) <span class="sc">|&gt;</span></span>
<span id="cb270-4"><a href="#cb270-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(res_sig)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(celltypes_abbrev)`</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb272"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb272-1"><a href="#cb272-1" aria-hidden="true" tabindex="-1"></a>res_sig_filtered_celltypes <span class="ot">&lt;-</span> res_sig <span class="sc">|&gt;</span></span>
<span id="cb272-2"><a href="#cb272-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltypes_abbrev) <span class="sc">|&gt;</span></span>
<span id="cb272-3"><a href="#cb272-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">|&gt;</span></span>
<span id="cb272-4"><a href="#cb272-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">30</span>) <span class="sc">|&gt;</span></span>
<span id="cb272-5"><a href="#cb272-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="degs-count-barplot" class="level3">
<h3 class="anchored" data-anchor-id="degs-count-barplot">DEGs count barplot</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb273"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb273-1"><a href="#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a data frame 'data' with columns 'CellType', 'Regulation' and 'GeneCount'</span></span>
<span id="cb273-2"><a href="#cb273-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace 'CellType', 'Regulation', and 'GeneCount' with the actual column names in your data frame</span></span>
<span id="cb273-3"><a href="#cb273-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-4"><a href="#cb273-4" aria-hidden="true" tabindex="-1"></a>deg_count_plot <span class="ot">&lt;-</span> res_sig <span class="sc">|&gt;</span></span>
<span id="cb273-5"><a href="#cb273-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltypes_abbrev, deg_direction) <span class="sc">|&gt;</span></span>
<span id="cb273-6"><a href="#cb273-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">|&gt;</span></span>
<span id="cb273-7"><a href="#cb273-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(celltypes_abbrev, count), <span class="at">fill =</span> deg_direction)) <span class="sc">+</span></span>
<span id="cb273-8"><a href="#cb273-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> count), <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb273-9"><a href="#cb273-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb273-10"><a href="#cb273-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Number of significant differential expressed genes"</span>, <span class="at">fill =</span> <span class="st">"Regulation"</span>) <span class="sc">+</span></span>
<span id="cb273-11"><a href="#cb273-11" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_manual(values = c("Upregulated" = "blue", "Downregulated" = "red")) +</span></span>
<span id="cb273-12"><a href="#cb273-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'celltypes_abbrev'. You can override using
the `.groups` argument.</code></pre>
</div>
</div></div>
</section>
<section id="go-summary" class="level3">
<h3 class="anchored" data-anchor-id="go-summary">GO summary</h3>
<section id="gsea-results" class="level4">
<h4 class="anchored" data-anchor-id="gsea-results">GSEA results</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb275"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span></span>
<span id="cb275-2"><a href="#cb275-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb275-3"><a href="#cb275-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/gene-set-enrichment_go_subtypes_no-apoe.qs"</span></span>
<span id="cb275-4"><a href="#cb275-4" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb275-5"><a href="#cb275-5" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(file)</span>
<span id="cb275-6"><a href="#cb275-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb275-7"><a href="#cb275-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any NULL entries</span></span>
<span id="cb275-8"><a href="#cb275-8" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> result_list[<span class="sc">!</span><span class="fu">sapply</span>(result_list, is.null)]</span>
<span id="cb275-9"><a href="#cb275-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Assert that there are no nulls in list</span></span>
<span id="cb275-10"><a href="#cb275-10" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">see_if</span>(<span class="fu">sum</span>(<span class="fu">map_lgl</span>(result_list, is.null)) <span class="sc">==</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb277"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any with no pathways found</span></span>
<span id="cb277-2"><a href="#cb277-2" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> result_list <span class="sc">%&gt;%</span></span>
<span id="cb277-3"><a href="#cb277-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">keep</span>(<span class="sc">~</span> <span class="fu">nrow</span>(.) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb277-4"><a href="#cb277-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-5"><a href="#cb277-5" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> gse[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"ambiguous_|low-feature-cells"</span>, <span class="fu">names</span>(gse))]</span>
<span id="cb277-6"><a href="#cb277-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-7"><a href="#cb277-7" aria-hidden="true" tabindex="-1"></a><span class="co"># treeplots</span></span>
<span id="cb277-8"><a href="#cb277-8" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="fu">map</span>(gse, pairwise_termsim)</span>
<span id="cb277-9"><a href="#cb277-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-10"><a href="#cb277-10" aria-hidden="true" tabindex="-1"></a>go_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(gse), <span class="cf">function</span>(celltype) {</span>
<span id="cb277-11"><a href="#cb277-11" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> gse[[celltype]]</span>
<span id="cb277-12"><a href="#cb277-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(res)) {</span>
<span id="cb277-13"><a href="#cb277-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">celltype =</span> celltype, </span>
<span id="cb277-14"><a href="#cb277-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">ID =</span> res<span class="sc">$</span>ID, </span>
<span id="cb277-15"><a href="#cb277-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">Description =</span> res<span class="sc">$</span>Description, </span>
<span id="cb277-16"><a href="#cb277-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">p.adjust =</span> res<span class="sc">$</span>p.adjust, </span>
<span id="cb277-17"><a href="#cb277-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">enrichmentScore =</span> res<span class="sc">$</span>enrichmentScore,</span>
<span id="cb277-18"><a href="#cb277-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">Count =</span> res<span class="sc">$</span>setSize)</span>
<span id="cb277-19"><a href="#cb277-19" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb277-20"><a href="#cb277-20" aria-hidden="true" tabindex="-1"></a>    <span class="cn">NULL</span></span>
<span id="cb277-21"><a href="#cb277-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb277-22"><a href="#cb277-22" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">|&gt;</span> <span class="fu">bind_rows</span>() <span class="sc">|&gt;</span></span>
<span id="cb277-23"><a href="#cb277-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">ontology =</span> <span class="fu">case_when</span>(</span>
<span id="cb277-24"><a href="#cb277-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"_BP"</span>, celltype) <span class="sc">~</span> <span class="st">"BP"</span>,</span>
<span id="cb277-25"><a href="#cb277-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"_CC"</span>, celltype) <span class="sc">~</span> <span class="st">"CC"</span>,</span>
<span id="cb277-26"><a href="#cb277-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"_MF"</span>, celltype) <span class="sc">~</span> <span class="st">"MF"</span>,</span>
<span id="cb277-27"><a href="#cb277-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">.default =</span> <span class="cn">NA</span></span>
<span id="cb277-28"><a href="#cb277-28" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb277-29"><a href="#cb277-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-30"><a href="#cb277-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the GO IDs for revigo</span></span>
<span id="cb277-31"><a href="#cb277-31" aria-hidden="true" tabindex="-1"></a>go_data_ids <span class="ot">&lt;-</span> go_data <span class="sc">|&gt;</span></span>
<span id="cb277-32"><a href="#cb277-32" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(ontology) <span class="sc">|&gt;</span></span>
<span id="cb277-33"><a href="#cb277-33" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span>
<span id="cb277-34"><a href="#cb277-34" aria-hidden="true" tabindex="-1"></a>go_data_ids <span class="ot">&lt;-</span> <span class="fu">map</span>(go_data_ids<span class="sc">$</span>data, dplyr<span class="sc">::</span>pull, ID) <span class="sc">|&gt;</span></span>
<span id="cb277-35"><a href="#cb277-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(go_data_ids<span class="sc">$</span>ontology)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [28]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb278"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb278-1"><a href="#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb278-2"><a href="#cb278-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Save IDs for revigo</span></span>
<span id="cb278-3"><a href="#cb278-3" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(go_data_ids, <span class="fu">names</span>(go_data_ids), <span class="sc">~</span> <span class="fu">unique</span>(.x) <span class="sc">|&gt;</span></span>
<span id="cb278-4"><a href="#cb278-4" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write_lines</span>(<span class="fu">paste0</span>(.y, <span class="st">"_ids_for_revigo.txt"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>This seems to produce a huge number of terms, even after reducing with revigo</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [29]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb279"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb279-1"><a href="#cb279-1" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span></span>
<span id="cb279-2"><a href="#cb279-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb279-3"><a href="#cb279-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/go-hypergeometic_subtypes_no-apoe.qs"</span></span>
<span id="cb279-4"><a href="#cb279-4" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb279-5"><a href="#cb279-5" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(file)</span>
<span id="cb279-6"><a href="#cb279-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb279-7"><a href="#cb279-7" aria-hidden="true" tabindex="-1"></a><span class="co"># most don't have too many terms</span></span>
<span id="cb279-8"><a href="#cb279-8" aria-hidden="true" tabindex="-1"></a><span class="fu">map_dbl</span>(result_list, <span class="sc">~</span> .x<span class="sc">@</span>result <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> <span class="fu">nrow</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                        Artirial-FB_BP                Ex-Neuron-CBLN2-NRGN_BP 
                                    37                                     16 
Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_BP         In-Neuron-ADARB2-LAMP5-NRG1_BP 
                                     1                                      2 
         In-Neuron-ADARB2-NRG1-RELN_BP               In-Neuron-DPP10-MEF2C_BP 
                                    73                                     47 
              In-Neuron-TRHDE-RALYL_BP                          M-Pericyte_BP 
                                    62                                      8 
                Microglia-activated_BP                             Oligo-A_BP 
                                     4                                     10 
                            Oligo-B_BP                         Pericyte-FB_BP 
                                    17                                     20 
                         T-Pericyte_BP                T-cell-parenchymal-2_BP 
                                     3                                     21 
             Vascular-SMC-LINC00486_BP                 Astrocyte-activated_CC 
                                    45                                      3 
                Astrocyte-quiescent_CC                Ex-Neuron-CBLN2-NRGN_CC 
                                     8                                     12 
           Ex-Neuron-L6-THEMIS-NFIA_CC          In-Neuron-ADARB2-NRG1-RELN_CC 
                                     8                                      4 
              In-Neuron-DPP10-MEF2C_CC               In-Neuron-TRHDE-RALYL_CC 
                                    15                                      4 
                         T-Pericyte_CC                T-cell-parenchymal-1_CC 
                                    24                                      2 
             Vascular-SMC-LINC00486_CC                              Venous_CC 
                                     3                                      3 
                           Arterial_MF                         Artirial-FB_MF 
                                     2                                     22 
               Ex-Neuron-CBLN2-NRGN_MF Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_MF 
                                     7                                      2 
          Ex-Neuron-L3-5-RORB-PLCH1_MF               In-Neuron-ADARB2-GRM7_MF 
                                    18                                      2 
         In-Neuron-ADARB2-NRG1-RELN_MF               In-Neuron-DPP10-MEF2C_MF 
                                    19                                     29 
             In-Neuron-PRKG1-TBC1D4_MF               In-Neuron-TRHDE-RALYL_MF 
                                     2                                     15 
                         M-Pericyte_MF                 Microglia-activated_MF 
                                    12                                      2 
                Microglia-quiescent_MF                             Oligo-A_MF 
                                    24                                      3 
                        Pericyte-FB_MF                          T-Pericyte_MF 
                                     9                                      6 
               T-cell-parenchymal-2_MF                        Vascular-SMC_MF 
                                     8                                      3 
             Vascular-SMC-LINC00486_MF 
                                     4 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb281"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb281-1"><a href="#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tryCatch({</span></span>
<span id="cb281-2"><a href="#cb281-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   map(data, ~plot_function(.x))</span></span>
<span id="cb281-3"><a href="#cb281-3" aria-hidden="true" tabindex="-1"></a><span class="co"># }, error = function(e) {</span></span>
<span id="cb281-4"><a href="#cb281-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   map(data, ~alternative_plot_function(.x))</span></span>
<span id="cb281-5"><a href="#cb281-5" aria-hidden="true" tabindex="-1"></a><span class="co"># })</span></span>
<span id="cb281-6"><a href="#cb281-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-7"><a href="#cb281-7" aria-hidden="true" tabindex="-1"></a>go_plots <span class="ot">&lt;-</span> <span class="fu">map2</span>(result_list, <span class="fu">names</span>(result_list), <span class="sc">~</span> <span class="fu">tryCatch</span>(</span>
<span id="cb281-8"><a href="#cb281-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">treeplot</span>(.x) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(.y, <span class="st">'- Hyper-GO'</span>)),</span>
<span id="cb281-9"><a href="#cb281-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb281-10"><a href="#cb281-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dotplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">10</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(.y, <span class="st">'- Hyper-GO'</span>))</span>
<span id="cb281-11"><a href="#cb281-11" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb281-12"><a href="#cb281-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-13"><a href="#cb281-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Use the sub function to extract the substring before the first underscore</span></span>
<span id="cb281-14"><a href="#cb281-14" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu">names</span>(go_plots)) <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb281-15"><a href="#cb281-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-16"><a href="#cb281-16" aria-hidden="true" tabindex="-1"></a>merge_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(celltype, dotplot_list) {</span>
<span id="cb281-17"><a href="#cb281-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subset to relevant plots</span></span>
<span id="cb281-18"><a href="#cb281-18" aria-hidden="true" tabindex="-1"></a>  plots <span class="ot">&lt;-</span> dotplot_list[<span class="fu">grepl</span>(<span class="fu">paste0</span>(<span class="st">"^"</span>, celltype, <span class="st">"_"</span>), <span class="fu">names</span>(dotplot_list))]</span>
<span id="cb281-19"><a href="#cb281-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge plots</span></span>
<span id="cb281-20"><a href="#cb281-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#plot &lt;- plots[[1]] + plots[[2]] + plots[[3]]</span></span>
<span id="cb281-21"><a href="#cb281-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if there are any plots available</span></span>
<span id="cb281-22"><a href="#cb281-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(plots) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb281-23"><a href="#cb281-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use Reduce to combine the plots</span></span>
<span id="cb281-24"><a href="#cb281-24" aria-hidden="true" tabindex="-1"></a>    final_plot <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(<span class="st">`</span><span class="at">+</span><span class="st">`</span>, plots)</span>
<span id="cb281-25"><a href="#cb281-25" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb281-26"><a href="#cb281-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">warning</span>(<span class="st">"No plots available for the specified celltype."</span>)</span>
<span id="cb281-27"><a href="#cb281-27" aria-hidden="true" tabindex="-1"></a>    final_plot <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb281-28"><a href="#cb281-28" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb281-29"><a href="#cb281-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb281-30"><a href="#cb281-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(final_plot)</span>
<span id="cb281-31"><a href="#cb281-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb281-32"><a href="#cb281-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-33"><a href="#cb281-33" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">map</span>(celltypes, merge_plots, go_plots) <span class="sc">|&gt;</span> <span class="fu">set_names</span>(celltypes)</span>
<span id="cb281-34"><a href="#cb281-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-35"><a href="#cb281-35" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span><span class="st">`</span><span class="at">Artirial-FB</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-29-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15"><img src="04_figures_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb282"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb282-1"><a href="#cb282-1" aria-hidden="true" tabindex="-1"></a>save_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(plot, filename) {</span>
<span id="cb282-2"><a href="#cb282-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Default dimensions</span></span>
<span id="cb282-3"><a href="#cb282-3" aria-hidden="true" tabindex="-1"></a>  width <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb282-4"><a href="#cb282-4" aria-hidden="true" tabindex="-1"></a>  height <span class="ot">&lt;-</span> <span class="dv">10</span> </span>
<span id="cb282-5"><a href="#cb282-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb282-6"><a href="#cb282-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if the plot has facets</span></span>
<span id="cb282-7"><a href="#cb282-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="st">"FacetGrid"</span> <span class="sc">%in%</span> <span class="fu">class</span>(plot<span class="sc">$</span>facet) <span class="sc">||</span> <span class="st">"FacetWrap"</span> <span class="sc">%in%</span> <span class="fu">class</span>(plot<span class="sc">$</span>facet)) {</span>
<span id="cb282-8"><a href="#cb282-8" aria-hidden="true" tabindex="-1"></a>    width <span class="ot">&lt;-</span> <span class="dv">20</span>  <span class="co"># Adjust width for faceted plots</span></span>
<span id="cb282-9"><a href="#cb282-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb282-10"><a href="#cb282-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb282-11"><a href="#cb282-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the plot using ggsave</span></span>
<span id="cb282-12"><a href="#cb282-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(filename, <span class="at">plot =</span> plot, <span class="at">width =</span> width, <span class="at">height =</span> height)</span>
<span id="cb282-13"><a href="#cb282-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb282-14"><a href="#cb282-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-15"><a href="#cb282-15" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset"</span>, <span class="fu">paste0</span>(<span class="fu">names</span>(plots), <span class="st">".png"</span>))</span>
<span id="cb282-16"><a href="#cb282-16" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(plots, files, save_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$`Artirial-FB`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Artirial-FB.png"

$`Ex-Neuron-CBLN2-NRGN`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Ex-Neuron-CBLN2-NRGN.png"

$`Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2.png"

$`In-Neuron-ADARB2-LAMP5-NRG1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/In-Neuron-ADARB2-LAMP5-NRG1.png"

$`In-Neuron-ADARB2-NRG1-RELN`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/In-Neuron-ADARB2-NRG1-RELN.png"

$`In-Neuron-DPP10-MEF2C`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/In-Neuron-DPP10-MEF2C.png"

$`In-Neuron-TRHDE-RALYL`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/In-Neuron-TRHDE-RALYL.png"

$`M-Pericyte`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/M-Pericyte.png"

$`Microglia-activated`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Microglia-activated.png"

$`Oligo-A`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Oligo-A.png"

$`Oligo-B`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Oligo-B.png"

$`Pericyte-FB`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Pericyte-FB.png"

$`T-Pericyte`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/T-Pericyte.png"

$`T-cell-parenchymal-2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/T-cell-parenchymal-2.png"

$`Vascular-SMC-LINC00486`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Vascular-SMC-LINC00486.png"

$`Astrocyte-activated`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Astrocyte-activated.png"

$`Astrocyte-quiescent`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Astrocyte-quiescent.png"

$`Ex-Neuron-L6-THEMIS-NFIA`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Ex-Neuron-L6-THEMIS-NFIA.png"

$`T-cell-parenchymal-1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/T-cell-parenchymal-1.png"

$Venous
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Venous.png"

$Arterial
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Arterial.png"

$`Ex-Neuron-L3-5-RORB-PLCH1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Ex-Neuron-L3-5-RORB-PLCH1.png"

$`In-Neuron-ADARB2-GRM7`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/In-Neuron-ADARB2-GRM7.png"

$`In-Neuron-PRKG1-TBC1D4`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/In-Neuron-PRKG1-TBC1D4.png"

$`Microglia-quiescent`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Microglia-quiescent.png"

$`Vascular-SMC`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/Vascular-SMC.png"</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb284"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to celltypes with at least 30 genes</span></span>
<span id="cb284-2"><a href="#cb284-2" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> plots[<span class="fu">names</span>(plots) <span class="sc">%in%</span> <span class="fu">unique</span>(res_sig_filtered_celltypes<span class="sc">$</span>celltype)]</span>
<span id="cb284-3"><a href="#cb284-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-4"><a href="#cb284-4" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes"</span>, <span class="fu">paste0</span>(<span class="fu">names</span>(plots), <span class="st">".png"</span>))</span>
<span id="cb284-5"><a href="#cb284-5" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(plots, files, save_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$`Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2.png"

$`M-Pericyte`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/M-Pericyte.png"

$`Microglia-activated`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Microglia-activated.png"

$`Oligo-A`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Oligo-A.png"

$`Oligo-B`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Oligo-B.png"

$`T-Pericyte`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/T-Pericyte.png"

$`Astrocyte-activated`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Astrocyte-activated.png"

$`Astrocyte-quiescent`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Astrocyte-quiescent.png"

$Venous
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Venous.png"

$Arterial
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/Arterial.png"

$`In-Neuron-ADARB2-GRM7`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset/subset_celltypes/In-Neuron-ADARB2-GRM7.png"</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb286"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb286-1"><a href="#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="co"># map2(go_plots, names(go_plots), ~ ggsave(filename = here::here("05_figures/990_shared_figures/002_pseudobulk/hyper-go_subset", paste0(.y, ".png")), plot = .x,</span></span>
<span id="cb286-2"><a href="#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="co">#        width = 7, height = 10, units = "in", dpi = 300)</span></span>
<span id="cb286-3"><a href="#cb286-3" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb286-4"><a href="#cb286-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-5"><a href="#cb286-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-6"><a href="#cb286-6" aria-hidden="true" tabindex="-1"></a>go_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(result_list), <span class="cf">function</span>(celltype) {</span>
<span id="cb286-7"><a href="#cb286-7" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> result_list[[celltype]]</span>
<span id="cb286-8"><a href="#cb286-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(res)) {</span>
<span id="cb286-9"><a href="#cb286-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">celltype =</span> celltype, </span>
<span id="cb286-10"><a href="#cb286-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">ID =</span> res<span class="sc">@</span>result<span class="sc">$</span>ID, </span>
<span id="cb286-11"><a href="#cb286-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">Description =</span> res<span class="sc">@</span>result<span class="sc">$</span>Description, </span>
<span id="cb286-12"><a href="#cb286-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">p.adjust =</span> res<span class="sc">@</span>result<span class="sc">$</span>p.adjust, </span>
<span id="cb286-13"><a href="#cb286-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">GeneRatio =</span> res<span class="sc">@</span>result<span class="sc">$</span>GeneRatio,</span>
<span id="cb286-14"><a href="#cb286-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">Count =</span> res<span class="sc">@</span>result<span class="sc">$</span>Count<span class="co">#,</span></span>
<span id="cb286-15"><a href="#cb286-15" aria-hidden="true" tabindex="-1"></a>               <span class="co">#ontology = res@result$ONTOLOGY</span></span>
<span id="cb286-16"><a href="#cb286-16" aria-hidden="true" tabindex="-1"></a>               ) <span class="sc">|&gt;</span></span>
<span id="cb286-17"><a href="#cb286-17" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">ontology =</span> <span class="fu">case_when</span>(</span>
<span id="cb286-18"><a href="#cb286-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"_BP"</span>, celltype) <span class="sc">~</span> <span class="st">"BP"</span>,</span>
<span id="cb286-19"><a href="#cb286-19" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"_CC"</span>, celltype) <span class="sc">~</span> <span class="st">"CC"</span>,</span>
<span id="cb286-20"><a href="#cb286-20" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grepl</span>(<span class="st">"_MF"</span>, celltype) <span class="sc">~</span> <span class="st">"MF"</span>,</span>
<span id="cb286-21"><a href="#cb286-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">.default =</span> <span class="cn">NA</span></span>
<span id="cb286-22"><a href="#cb286-22" aria-hidden="true" tabindex="-1"></a>      ))</span>
<span id="cb286-23"><a href="#cb286-23" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb286-24"><a href="#cb286-24" aria-hidden="true" tabindex="-1"></a>    <span class="cn">NULL</span></span>
<span id="cb286-25"><a href="#cb286-25" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb286-26"><a href="#cb286-26" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">|&gt;</span> <span class="fu">bind_rows</span>()</span>
<span id="cb286-27"><a href="#cb286-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-28"><a href="#cb286-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the GO IDs for revigo</span></span>
<span id="cb286-29"><a href="#cb286-29" aria-hidden="true" tabindex="-1"></a>go_data_ids <span class="ot">&lt;-</span> go_data <span class="sc">|&gt;</span></span>
<span id="cb286-30"><a href="#cb286-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(ontology) <span class="sc">|&gt;</span></span>
<span id="cb286-31"><a href="#cb286-31" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span>
<span id="cb286-32"><a href="#cb286-32" aria-hidden="true" tabindex="-1"></a>go_data_ids <span class="ot">&lt;-</span> <span class="fu">map</span>(go_data_ids<span class="sc">$</span>data, dplyr<span class="sc">::</span>pull, ID) <span class="sc">|&gt;</span></span>
<span id="cb286-33"><a href="#cb286-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(go_data_ids<span class="sc">$</span>ontology)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [30]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb287"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb287-1"><a href="#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb287-2"><a href="#cb287-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Save IDs for revigo</span></span>
<span id="cb287-3"><a href="#cb287-3" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(go_data_ids, <span class="fu">names</span>(go_data_ids), <span class="sc">~</span> <span class="fu">unique</span>(.x) <span class="sc">|&gt;</span></span>
<span id="cb287-4"><a href="#cb287-4" aria-hidden="true" tabindex="-1"></a>      <span class="fu">write_lines</span>(<span class="fu">paste0</span>(.y, <span class="st">"_ids_for_revigo.txt"</span>)))</span>
<span id="cb287-5"><a href="#cb287-5" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_lines</span>(go_data_ids, <span class="st">"go_hyper_ids_for_revigo.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [31]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb288"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb288-1"><a href="#cb288-1" aria-hidden="true" tabindex="-1"></a>process_revigo_raw_table <span class="ot">&lt;-</span> <span class="cf">function</span>(file, ontology) {</span>
<span id="cb288-2"><a href="#cb288-2" aria-hidden="true" tabindex="-1"></a>  revi_df <span class="ot">&lt;-</span></span>
<span id="cb288-3"><a href="#cb288-3" aria-hidden="true" tabindex="-1"></a>    readr<span class="sc">::</span><span class="fu">read_tsv</span>(file) <span class="sc">|&gt;</span></span>
<span id="cb288-4"><a href="#cb288-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">ontology =</span> ontology)</span>
<span id="cb288-5"><a href="#cb288-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Function to reconstruct the full GO ID for representatives</span></span>
<span id="cb288-6"><a href="#cb288-6" aria-hidden="true" tabindex="-1"></a>  reconstruct_go_id <span class="ot">&lt;-</span> <span class="cf">function</span>(go_id, rep) {</span>
<span id="cb288-7"><a href="#cb288-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (rep <span class="sc">!=</span> <span class="st">"null"</span>) {</span>
<span id="cb288-8"><a href="#cb288-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">"GO:"</span>, <span class="fu">sprintf</span>(<span class="st">"%07d"</span>, <span class="fu">as.numeric</span>(rep)))</span>
<span id="cb288-9"><a href="#cb288-9" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb288-10"><a href="#cb288-10" aria-hidden="true" tabindex="-1"></a>      go_id</span>
<span id="cb288-11"><a href="#cb288-11" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb288-12"><a href="#cb288-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb288-13"><a href="#cb288-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb288-14"><a href="#cb288-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a new column 'Representative_GO' that will hold the full representative GO term</span></span>
<span id="cb288-15"><a href="#cb288-15" aria-hidden="true" tabindex="-1"></a>  revi_df <span class="ot">&lt;-</span> revi_df <span class="sc">|&gt;</span></span>
<span id="cb288-16"><a href="#cb288-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Representative_GO =</span> <span class="fu">mapply</span>(reconstruct_go_id, TermID, Representative))</span>
<span id="cb288-17"><a href="#cb288-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(revi_df)</span>
<span id="cb288-18"><a href="#cb288-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb288-19"><a href="#cb288-19" aria-hidden="true" tabindex="-1"></a>revigo_files <span class="ot">&lt;-</span></span>
<span id="cb288-20"><a href="#cb288-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(</span>
<span id="cb288-21"><a href="#cb288-21" aria-hidden="true" tabindex="-1"></a>    here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb288-22"><a href="#cb288-22" aria-hidden="true" tabindex="-1"></a>      <span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots"</span></span>
<span id="cb288-23"><a href="#cb288-23" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb288-24"><a href="#cb288-24" aria-hidden="true" tabindex="-1"></a>    <span class="co">#pattern = "Revigo_*",</span></span>
<span id="cb288-25"><a href="#cb288-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">pattern =</span> <span class="st">"*_hyper_go.tsv"</span>,</span>
<span id="cb288-26"><a href="#cb288-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">full.names =</span> <span class="cn">TRUE</span></span>
<span id="cb288-27"><a href="#cb288-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb288-28"><a href="#cb288-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-29"><a href="#cb288-29" aria-hidden="true" tabindex="-1"></a>revi_df <span class="ot">&lt;-</span></span>
<span id="cb288-30"><a href="#cb288-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2_dfr</span>(revigo_files, <span class="fu">c</span>(<span class="st">"BP"</span>, <span class="st">"CC"</span>, <span class="st">"MF"</span>), process_revigo_raw_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 310 Columns: 8
 Column specification 
Delimiter: "\t"
chr (3): TermID, Name, Representative
dbl (5): Value, LogSize, Frequency, Uniqueness, Dispensability

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 57 Columns: 8
 Column specification 
Delimiter: "\t"
chr (3): TermID, Name, Representative
dbl (5): Value, LogSize, Frequency, Uniqueness, Dispensability

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 153 Columns: 8
 Column specification 
Delimiter: "\t"
chr (3): TermID, Name, Representative
dbl (5): Value, LogSize, Frequency, Uniqueness, Dispensability

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb290"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb290-1"><a href="#cb290-1" aria-hidden="true" tabindex="-1"></a>revi_representative <span class="ot">&lt;-</span> revi_df <span class="sc">|&gt;</span></span>
<span id="cb290-2"><a href="#cb290-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the representative term name</span></span>
<span id="cb290-3"><a href="#cb290-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(ontology) <span class="sc">|&gt;</span></span>
<span id="cb290-4"><a href="#cb290-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(Representative_GO, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb290-5"><a href="#cb290-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb290-6"><a href="#cb290-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Representative_GO, Name) <span class="sc">|&gt;</span></span>
<span id="cb290-7"><a href="#cb290-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">representative_term_name =</span> Name)</span>
<span id="cb290-8"><a href="#cb290-8" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb290-9"><a href="#cb290-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(go_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>        celltype         ID                                 Description
1 Artirial-FB_BP GO:0030638                polyketide metabolic process
2 Artirial-FB_BP GO:0030647 aminoglycoside antibiotic metabolic process
3 Artirial-FB_BP GO:0044598               doxorubicin metabolic process
4 Artirial-FB_BP GO:0042448              progesterone metabolic process
5 Artirial-FB_BP GO:0016137                 glycoside metabolic process
6 Artirial-FB_BP GO:1902644          tertiary alcohol metabolic process
      p.adjust GeneRatio Count ontology
1 0.0001291181       2/5     2       BP
2 0.0001291181       2/5     2       BP
3 0.0001291181       2/5     2       BP
4 0.0001677894       2/5     2       BP
5 0.0002722352       2/5     2       BP
6 0.0002722352       2/5     2       BP</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb292"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb292-1"><a href="#cb292-1" aria-hidden="true" tabindex="-1"></a>go_data_revi <span class="ot">&lt;-</span> go_data <span class="sc">|&gt;</span></span>
<span id="cb292-2"><a href="#cb292-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(revi_representative, <span class="at">by =</span> <span class="fu">join_by</span>(ID <span class="sc">==</span> Representative_GO)) <span class="sc">|&gt;</span></span>
<span id="cb292-3"><a href="#cb292-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype, ontology, representative_term_name) <span class="sc">|&gt;</span></span>
<span id="cb292-4"><a href="#cb292-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">min_p =</span> <span class="fu">min</span>(p.adjust)) <span class="sc">|&gt;</span></span>
<span id="cb292-5"><a href="#cb292-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb292-6"><a href="#cb292-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(representative_term_name))</span>
<span id="cb292-7"><a href="#cb292-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-8"><a href="#cb292-8" aria-hidden="true" tabindex="-1"></a>go_data_revi_matrix <span class="ot">&lt;-</span> go_data_revi <span class="sc">|&gt;</span></span>
<span id="cb292-9"><a href="#cb292-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(celltype, ontology, representative_term_name, min_p) <span class="sc">|&gt;</span></span>
<span id="cb292-10"><a href="#cb292-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> <span class="fu">str_remove</span>(celltype, <span class="st">"_.*"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb292-11"><a href="#cb292-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb292-12"><a href="#cb292-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-13"><a href="#cb292-13" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(go_data_revi_matrix<span class="sc">$</span>ontology)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
 BP  CC  MF 
772 133 244 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb294"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb294-1"><a href="#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="co"># #revi_df &lt;- </span></span>
<span id="cb294-2"><a href="#cb294-2" aria-hidden="true" tabindex="-1"></a><span class="co"># revi_df &lt;- revi_representative |&gt;</span></span>
<span id="cb294-3"><a href="#cb294-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::left_join(revi_df, by = join_by(Representative_GO))</span></span>
<span id="cb294-4"><a href="#cb294-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb294-5"><a href="#cb294-5" aria-hidden="true" tabindex="-1"></a><span class="co"># head(go_data)</span></span>
<span id="cb294-6"><a href="#cb294-6" aria-hidden="true" tabindex="-1"></a><span class="co"># go_data_revi &lt;- go_data |&gt;</span></span>
<span id="cb294-7"><a href="#cb294-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::left_join(revi_df, by = join_by(ID == TermID, ontology)) |&gt;</span></span>
<span id="cb294-8"><a href="#cb294-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::group_by(celltype, ontology, Representative_GO) |&gt;</span></span>
<span id="cb294-9"><a href="#cb294-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::mutate(min_p = min(p.adjust)) |&gt;</span></span>
<span id="cb294-10"><a href="#cb294-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::ungroup()</span></span>
<span id="cb294-11"><a href="#cb294-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb294-12"><a href="#cb294-12" aria-hidden="true" tabindex="-1"></a><span class="co"># go_data_revi_matrix &lt;- go_data_revi |&gt;</span></span>
<span id="cb294-13"><a href="#cb294-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::select(celltype, ontology, Name, min_p, Representative_GO) |&gt;</span></span>
<span id="cb294-14"><a href="#cb294-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::mutate(celltype = str_remove(celltype, "_.*")) |&gt;</span></span>
<span id="cb294-15"><a href="#cb294-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::group_by(ontology) |&gt;</span></span>
<span id="cb294-16"><a href="#cb294-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::distinct(Representative_GO, .keep_all = TRUE) |&gt;</span></span>
<span id="cb294-17"><a href="#cb294-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::ungroup() |&gt;</span></span>
<span id="cb294-18"><a href="#cb294-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   unique() |&gt;</span></span>
<span id="cb294-19"><a href="#cb294-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::select(!Representative_GO) |&gt;</span></span>
<span id="cb294-20"><a href="#cb294-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::rename(representative_term_name = Name)</span></span>
<span id="cb294-21"><a href="#cb294-21" aria-hidden="true" tabindex="-1"></a><span class="co"># table(go_data_revi_matrix$ontology)</span></span>
<span id="cb294-22"><a href="#cb294-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb294-23"><a href="#cb294-23" aria-hidden="true" tabindex="-1"></a>go_data_revi_matrix <span class="ot">&lt;-</span> go_data_revi_matrix <span class="sc">|&gt;</span></span>
<span id="cb294-24"><a href="#cb294-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype, <span class="at">values_from =</span> min_p)</span>
<span id="cb294-25"><a href="#cb294-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb294-26"><a href="#cb294-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Split dataframe by ontology</span></span>
<span id="cb294-27"><a href="#cb294-27" aria-hidden="true" tabindex="-1"></a>df_list <span class="ot">&lt;-</span> <span class="fu">split</span>(go_data_revi_matrix, go_data_revi_matrix<span class="sc">$</span>ontology)</span>
<span id="cb294-28"><a href="#cb294-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb294-29"><a href="#cb294-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create heatmap for each ontology</span></span>
<span id="cb294-30"><a href="#cb294-30" aria-hidden="true" tabindex="-1"></a>create_heatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(df_subset) {</span>
<span id="cb294-31"><a href="#cb294-31" aria-hidden="true" tabindex="-1"></a>  df_subset <span class="ot">&lt;-</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>(df_subset)</span>
<span id="cb294-32"><a href="#cb294-32" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(df_subset[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)])  <span class="co"># Exclude GO_Term and Ontology columns</span></span>
<span id="cb294-33"><a href="#cb294-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(df_subset<span class="sc">$</span>representative_term_name, <span class="st">" "</span>, <span class="st">"_"</span>)</span>
<span id="cb294-34"><a href="#cb294-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb294-35"><a href="#cb294-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Heatmap</span>(</span>
<span id="cb294-36"><a href="#cb294-36" aria-hidden="true" tabindex="-1"></a>    mat,</span>
<span id="cb294-37"><a href="#cb294-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">unique</span>(df_subset<span class="sc">$</span>ontology),</span>
<span id="cb294-38"><a href="#cb294-38" aria-hidden="true" tabindex="-1"></a>    <span class="co">#column_title = unique(df_subset$ontology),</span></span>
<span id="cb294-39"><a href="#cb294-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb294-40"><a href="#cb294-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">FALSE</span></span>
<span id="cb294-41"><a href="#cb294-41" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb294-42"><a href="#cb294-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb294-43"><a href="#cb294-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb294-44"><a href="#cb294-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each subset</span></span>
<span id="cb294-45"><a href="#cb294-45" aria-hidden="true" tabindex="-1"></a>heatmaps <span class="ot">&lt;-</span> <span class="fu">map</span>(df_list, create_heatmap)</span>
<span id="cb294-46"><a href="#cb294-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb294-47"><a href="#cb294-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine heatmaps</span></span>
<span id="cb294-48"><a href="#cb294-48" aria-hidden="true" tabindex="-1"></a><span class="co">#draw(do.call(HeatmapAnnotation, heatmaps))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="heatmap-of-shared-degs" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-of-shared-degs">Heatmap of shared DEGs</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [32]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb295"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb295-1"><a href="#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a wide format dataframe to count overlaps</span></span>
<span id="cb295-2"><a href="#cb295-2" aria-hidden="true" tabindex="-1"></a>wide_df <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span></span>
<span id="cb295-3"><a href="#cb295-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, celltype, deg_direction) <span class="sc">%&gt;%</span></span>
<span id="cb295-4"><a href="#cb295-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">value =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb295-5"><a href="#cb295-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype, <span class="at">values_from =</span> value, <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">value =</span> <span class="dv">0</span>))</span>
<span id="cb295-6"><a href="#cb295-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-7"><a href="#cb295-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the wide format dataframe to a matrix</span></span>
<span id="cb295-8"><a href="#cb295-8" aria-hidden="true" tabindex="-1"></a>gene_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(wide_df[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)])</span>
<span id="cb295-9"><a href="#cb295-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-10"><a href="#cb295-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the overlap matrix using matrix multiplication</span></span>
<span id="cb295-11"><a href="#cb295-11" aria-hidden="true" tabindex="-1"></a>overlap_matrix <span class="ot">&lt;-</span> <span class="fu">t</span>(gene_matrix) <span class="sc">%*%</span> gene_matrix</span>
<span id="cb295-12"><a href="#cb295-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-13"><a href="#cb295-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the row and column names of the overlap matrix</span></span>
<span id="cb295-14"><a href="#cb295-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(overlap_matrix) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gene_matrix)</span>
<span id="cb295-15"><a href="#cb295-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(overlap_matrix) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gene_matrix)</span>
<span id="cb295-16"><a href="#cb295-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-17"><a href="#cb295-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the heatmap</span></span>
<span id="cb295-18"><a href="#cb295-18" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(overlap_matrix, </span>
<span id="cb295-19"><a href="#cb295-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb295-20"><a href="#cb295-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>, </span>
<span id="cb295-21"><a href="#cb295-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">display_numbers =</span> <span class="cn">TRUE</span>, </span>
<span id="cb295-22"><a href="#cb295-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"Overlap of DEGs Across Cell Types"</span>,</span>
<span id="cb295-23"><a href="#cb295-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"blue"</span>))(<span class="dv">50</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-32-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-16"><img src="04_figures_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [33]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb296"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb296-1"><a href="#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List of all cell types</span></span>
<span id="cb296-2"><a href="#cb296-2" aria-hidden="true" tabindex="-1"></a>all_celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(res_sig<span class="sc">$</span>celltype)</span>
<span id="cb296-3"><a href="#cb296-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-4"><a href="#cb296-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the dataframe for upregulated and downregulated genes</span></span>
<span id="cb296-5"><a href="#cb296-5" aria-hidden="true" tabindex="-1"></a>up_df <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(deg_direction <span class="sc">==</span> <span class="st">"Up"</span>)</span>
<span id="cb296-6"><a href="#cb296-6" aria-hidden="true" tabindex="-1"></a>down_df <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(deg_direction <span class="sc">==</span> <span class="st">"Down"</span>)</span>
<span id="cb296-7"><a href="#cb296-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-8"><a href="#cb296-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dataframe with all combinations of genes and celltypes, setting missing values to 0</span></span>
<span id="cb296-9"><a href="#cb296-9" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(up_df<span class="sc">$</span>gene, down_df<span class="sc">$</span>gene))</span>
<span id="cb296-10"><a href="#cb296-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-11"><a href="#cb296-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Create wide format dataframes for upregulated and downregulated genes</span></span>
<span id="cb296-12"><a href="#cb296-12" aria-hidden="true" tabindex="-1"></a>wide_up_df <span class="ot">&lt;-</span> up_df <span class="sc">%&gt;%</span></span>
<span id="cb296-13"><a href="#cb296-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, celltype) <span class="sc">%&gt;%</span></span>
<span id="cb296-14"><a href="#cb296-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb296-15"><a href="#cb296-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">value =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb296-16"><a href="#cb296-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">complete</span>(<span class="at">gene =</span> all_genes, <span class="at">celltype =</span> all_celltypes, <span class="at">fill =</span> <span class="fu">list</span>(<span class="at">value =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb296-17"><a href="#cb296-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype, <span class="at">values_from =</span> value, <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">value =</span> <span class="dv">0</span>))</span>
<span id="cb296-18"><a href="#cb296-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-19"><a href="#cb296-19" aria-hidden="true" tabindex="-1"></a>wide_down_df <span class="ot">&lt;-</span> down_df <span class="sc">%&gt;%</span></span>
<span id="cb296-20"><a href="#cb296-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, celltype) <span class="sc">%&gt;%</span></span>
<span id="cb296-21"><a href="#cb296-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb296-22"><a href="#cb296-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">value =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb296-23"><a href="#cb296-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">complete</span>(<span class="at">gene =</span> all_genes, <span class="at">celltype =</span> all_celltypes, <span class="at">fill =</span> <span class="fu">list</span>(<span class="at">value =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb296-24"><a href="#cb296-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype, <span class="at">values_from =</span> value, <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">value =</span> <span class="dv">0</span>))</span>
<span id="cb296-25"><a href="#cb296-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-26"><a href="#cb296-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the wide format dataframes to matrices</span></span>
<span id="cb296-27"><a href="#cb296-27" aria-hidden="true" tabindex="-1"></a>gene_matrix_up <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(wide_up_df[,<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb296-28"><a href="#cb296-28" aria-hidden="true" tabindex="-1"></a>gene_matrix_down <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(wide_down_df[,<span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb296-29"><a href="#cb296-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-30"><a href="#cb296-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the overlap matrices using matrix multiplication</span></span>
<span id="cb296-31"><a href="#cb296-31" aria-hidden="true" tabindex="-1"></a>overlap_matrix_up <span class="ot">&lt;-</span> <span class="fu">t</span>(gene_matrix_up) <span class="sc">%*%</span> gene_matrix_up</span>
<span id="cb296-32"><a href="#cb296-32" aria-hidden="true" tabindex="-1"></a>overlap_matrix_down <span class="ot">&lt;-</span> <span class="fu">t</span>(gene_matrix_down) <span class="sc">%*%</span> gene_matrix_down</span>
<span id="cb296-33"><a href="#cb296-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-34"><a href="#cb296-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the row and column names of the overlap matrices</span></span>
<span id="cb296-35"><a href="#cb296-35" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(overlap_matrix_up) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gene_matrix_up)</span>
<span id="cb296-36"><a href="#cb296-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(overlap_matrix_up) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gene_matrix_up)</span>
<span id="cb296-37"><a href="#cb296-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-38"><a href="#cb296-38" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(overlap_matrix_down) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gene_matrix_down)</span>
<span id="cb296-39"><a href="#cb296-39" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(overlap_matrix_down) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(gene_matrix_down)</span>
<span id="cb296-40"><a href="#cb296-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-41"><a href="#cb296-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply log transformation to normalize the data</span></span>
<span id="cb296-42"><a href="#cb296-42" aria-hidden="true" tabindex="-1"></a>log_transform <span class="ot">&lt;-</span> <span class="cf">function</span>(matrix) {</span>
<span id="cb296-43"><a href="#cb296-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">log1p</span>(matrix)  <span class="co"># log1p is log(1 + x), which handles zero values</span></span>
<span id="cb296-44"><a href="#cb296-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb296-45"><a href="#cb296-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-46"><a href="#cb296-46" aria-hidden="true" tabindex="-1"></a>overlap_matrix_up_log <span class="ot">&lt;-</span> <span class="fu">log_transform</span>(overlap_matrix_up)</span>
<span id="cb296-47"><a href="#cb296-47" aria-hidden="true" tabindex="-1"></a>overlap_matrix_down_log <span class="ot">&lt;-</span> <span class="fu">log_transform</span>(overlap_matrix_down)</span>
<span id="cb296-48"><a href="#cb296-48" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(circlize)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: circlize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
circlize version 0.4.16
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'circlize'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    degree</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb301"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb301-1"><a href="#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define custom color functions for the heatmaps</span></span>
<span id="cb301-2"><a href="#cb301-2" aria-hidden="true" tabindex="-1"></a>col_fun_up <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(overlap_matrix_up_log)), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red"</span>))</span>
<span id="cb301-3"><a href="#cb301-3" aria-hidden="true" tabindex="-1"></a>col_fun_down <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(overlap_matrix_down_log)), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"blue"</span>))</span>
<span id="cb301-4"><a href="#cb301-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-5"><a href="#cb301-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmaps</span></span>
<span id="cb301-6"><a href="#cb301-6" aria-hidden="true" tabindex="-1"></a>ht_up <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(overlap_matrix_up_log, <span class="at">name =</span> <span class="st">"Upregulated"</span>, <span class="at">col =</span> col_fun_up, <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, <span class="at">show_column_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb301-7"><a href="#cb301-7" aria-hidden="true" tabindex="-1"></a>ht_down <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(overlap_matrix_down_log, <span class="at">name =</span> <span class="st">"Downregulated"</span>, <span class="at">col =</span> col_fun_down, <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, <span class="at">show_column_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb301-8"><a href="#cb301-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-9"><a href="#cb301-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the heatmaps into a single plot</span></span>
<span id="cb301-10"><a href="#cb301-10" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht_up <span class="sc">+</span> ht_down, <span class="at">heatmap_legend_side =</span> <span class="st">"right"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-33-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-17"><img src="04_figures_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<section id="jaccard-similarity-index" class="level4">
<h4 class="anchored" data-anchor-id="jaccard-similarity-index">Jaccard similarity index</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [34]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb302"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb302-1"><a href="#cb302-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: permute</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'permute'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    permute</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'lattice'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:enrichplot':

    dotplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    dotplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:corrgram':

    panel.fill</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>This is vegan 2.6-6.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'vegan'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    diversity</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb314"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb314-1"><a href="#cb314-1" aria-hidden="true" tabindex="-1"></a>compute_jaccard_similarity <span class="ot">&lt;-</span></span>
<span id="cb314-2"><a href="#cb314-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(df,</span>
<span id="cb314-3"><a href="#cb314-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">deg_filter =</span> <span class="cn">NULL</span>,</span>
<span id="cb314-4"><a href="#cb314-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">col_title =</span> <span class="st">"All DEGs"</span>) {</span>
<span id="cb314-5"><a href="#cb314-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(deg_filter)) {</span>
<span id="cb314-6"><a href="#cb314-6" aria-hidden="true" tabindex="-1"></a>      df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb314-7"><a href="#cb314-7" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(deg_direction <span class="sc">==</span> deg_filter)</span>
<span id="cb314-8"><a href="#cb314-8" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb314-9"><a href="#cb314-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-10"><a href="#cb314-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a binary matrix of genes vs cell types</span></span>
<span id="cb314-11"><a href="#cb314-11" aria-hidden="true" tabindex="-1"></a>    binary_matrix <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb314-12"><a href="#cb314-12" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(gene, celltypes_abbrev) <span class="sc">%&gt;%</span></span>
<span id="cb314-13"><a href="#cb314-13" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb314-14"><a href="#cb314-14" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">present =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb314-15"><a href="#cb314-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pivot_wider</span>(</span>
<span id="cb314-16"><a href="#cb314-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">names_from =</span> celltypes_abbrev,</span>
<span id="cb314-17"><a href="#cb314-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_from =</span> present,</span>
<span id="cb314-18"><a href="#cb314-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">present =</span> <span class="dv">0</span>)</span>
<span id="cb314-19"><a href="#cb314-19" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb314-20"><a href="#cb314-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-21"><a href="#cb314-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Transpose the matrix to have cell types as rows</span></span>
<span id="cb314-22"><a href="#cb314-22" aria-hidden="true" tabindex="-1"></a>    binary_matrix_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(binary_matrix[<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb314-23"><a href="#cb314-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(binary_matrix_t) <span class="ot">&lt;-</span> binary_matrix<span class="sc">$</span>gene</span>
<span id="cb314-24"><a href="#cb314-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-25"><a href="#cb314-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate Jaccard similarity matrix</span></span>
<span id="cb314-26"><a href="#cb314-26" aria-hidden="true" tabindex="-1"></a>    jaccard_matrix <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(binary_matrix_t, <span class="at">method =</span> <span class="st">"jaccard"</span>)</span>
<span id="cb314-27"><a href="#cb314-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-28"><a href="#cb314-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Perform hierarchical clustering</span></span>
<span id="cb314-29"><a href="#cb314-29" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(jaccard_matrix, <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb314-30"><a href="#cb314-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-31"><a href="#cb314-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Convert distance matrix to similarity matrix for heatmap</span></span>
<span id="cb314-32"><a href="#cb314-32" aria-hidden="true" tabindex="-1"></a>    similarity_matrix <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">as.matrix</span>(jaccard_matrix)</span>
<span id="cb314-33"><a href="#cb314-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-34"><a href="#cb314-34" aria-hidden="true" tabindex="-1"></a>    format_cell_values <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb314-35"><a href="#cb314-35" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (x <span class="sc">&gt;=</span> <span class="fl">0.15</span>) {</span>
<span id="cb314-36"><a href="#cb314-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(<span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, x))</span>
<span id="cb314-37"><a href="#cb314-37" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb314-38"><a href="#cb314-38" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(<span class="st">""</span>)</span>
<span id="cb314-39"><a href="#cb314-39" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb314-40"><a href="#cb314-40" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb314-41"><a href="#cb314-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set colour to make it more obvious which cells have a similarity of 0.2 or more</span></span>
<span id="cb314-42"><a href="#cb314-42" aria-hidden="true" tabindex="-1"></a>    col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.15</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"blue"</span>, <span class="st">"red"</span>))</span>
<span id="cb314-43"><a href="#cb314-43" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-44"><a href="#cb314-44" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create the heatmap</span></span>
<span id="cb314-45"><a href="#cb314-45" aria-hidden="true" tabindex="-1"></a>    jaccard_all_hmap <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb314-46"><a href="#cb314-46" aria-hidden="true" tabindex="-1"></a>      similarity_matrix,</span>
<span id="cb314-47"><a href="#cb314-47" aria-hidden="true" tabindex="-1"></a>      <span class="at">name =</span> <span class="st">"Jaccard</span><span class="sc">\n</span><span class="st">Similarity"</span>,</span>
<span id="cb314-48"><a href="#cb314-48" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> col_fun,</span>
<span id="cb314-49"><a href="#cb314-49" aria-hidden="true" tabindex="-1"></a>      <span class="at">column_title =</span> col_title,</span>
<span id="cb314-50"><a href="#cb314-50" aria-hidden="true" tabindex="-1"></a>      <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb314-51"><a href="#cb314-51" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.rect</span>(x, y, width, height, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"grey"</span>, <span class="at">fill =</span> <span class="cn">NA</span>))</span>
<span id="cb314-52"><a href="#cb314-52" aria-hidden="true" tabindex="-1"></a>        value <span class="ot">&lt;-</span> similarity_matrix[i, j]</span>
<span id="cb314-53"><a href="#cb314-53" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (value <span class="sc">&gt;=</span> <span class="fl">0.15</span>) {</span>
<span id="cb314-54"><a href="#cb314-54" aria-hidden="true" tabindex="-1"></a>          <span class="fu">grid.text</span>(<span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, value),</span>
<span id="cb314-55"><a href="#cb314-55" aria-hidden="true" tabindex="-1"></a>                    x,</span>
<span id="cb314-56"><a href="#cb314-56" aria-hidden="true" tabindex="-1"></a>                    y,</span>
<span id="cb314-57"><a href="#cb314-57" aria-hidden="true" tabindex="-1"></a>                    <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb314-58"><a href="#cb314-58" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb314-59"><a href="#cb314-59" aria-hidden="true" tabindex="-1"></a>      },</span>
<span id="cb314-60"><a href="#cb314-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">show_row_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb314-61"><a href="#cb314-61" aria-hidden="true" tabindex="-1"></a>      <span class="at">show_column_dend =</span> <span class="cn">FALSE</span></span>
<span id="cb314-62"><a href="#cb314-62" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb314-63"><a href="#cb314-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb314-64"><a href="#cb314-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(jaccard_all_hmap)</span>
<span id="cb314-65"><a href="#cb314-65" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb314-66"><a href="#cb314-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-67"><a href="#cb314-67" aria-hidden="true" tabindex="-1"></a>jaccard_all_hmap <span class="ot">&lt;-</span> <span class="fu">compute_jaccard_similarity</span>(res_sig)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb319"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb319-1"><a href="#cb319-1" aria-hidden="true" tabindex="-1"></a>jaccard_all_hmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-34-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-18"><img src="04_figures_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb320"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb320-1"><a href="#cb320-1" aria-hidden="true" tabindex="-1"></a>jaccard_up_hmap <span class="ot">&lt;-</span> <span class="fu">compute_jaccard_similarity</span>(res_sig, <span class="at">deg_filter =</span> <span class="st">"Up"</span>, <span class="st">"Up DEGs"</span>)</span>
<span id="cb320-2"><a href="#cb320-2" aria-hidden="true" tabindex="-1"></a>jaccard_down_hmap <span class="ot">&lt;-</span> <span class="fu">compute_jaccard_similarity</span>(res_sig, <span class="at">deg_filter =</span> <span class="st">"Down"</span>, <span class="st">"Down DEGs"</span>)</span>
<span id="cb320-3"><a href="#cb320-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb320-4"><a href="#cb320-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plots</span></span>
<span id="cb320-5"><a href="#cb320-5" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures"</span>,</span>
<span id="cb320-6"><a href="#cb320-6" aria-hidden="true" tabindex="-1"></a>      <span class="st">"jaccard_heatmap_all_degs.png"</span>), <span class="at">width =</span> <span class="dv">4000</span>, <span class="at">height =</span> <span class="dv">4000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb320-7"><a href="#cb320-7" aria-hidden="true" tabindex="-1"></a>jaccard_all_hmap</span>
<span id="cb320-8"><a href="#cb320-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb322"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb322-1"><a href="#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures"</span>,</span>
<span id="cb322-2"><a href="#cb322-2" aria-hidden="true" tabindex="-1"></a>      <span class="st">"jaccard_heatmap_up_degs.png"</span>), <span class="at">width =</span> <span class="dv">4000</span>, <span class="at">height =</span> <span class="dv">4000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb322-3"><a href="#cb322-3" aria-hidden="true" tabindex="-1"></a>jaccard_up_hmap</span>
<span id="cb322-4"><a href="#cb322-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb324"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb324-1"><a href="#cb324-1" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures"</span>,</span>
<span id="cb324-2"><a href="#cb324-2" aria-hidden="true" tabindex="-1"></a>      <span class="st">"jaccard_heatmap_down_degs.png"</span>), <span class="at">width =</span> <span class="dv">4000</span>, <span class="at">height =</span> <span class="dv">4000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb324-3"><a href="#cb324-3" aria-hidden="true" tabindex="-1"></a>jaccard_down_hmap</span>
<span id="cb324-4"><a href="#cb324-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [35]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb326"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb326-1"><a href="#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a binary matrix of genes vs cell types</span></span>
<span id="cb326-2"><a href="#cb326-2" aria-hidden="true" tabindex="-1"></a>binary_matrix <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span></span>
<span id="cb326-3"><a href="#cb326-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene, celltypes_abbrev) <span class="sc">%&gt;%</span></span>
<span id="cb326-4"><a href="#cb326-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb326-5"><a href="#cb326-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">present =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb326-6"><a href="#cb326-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltypes_abbrev, <span class="at">values_from =</span> present, </span>
<span id="cb326-7"><a href="#cb326-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">present =</span> <span class="dv">0</span>))</span>
<span id="cb326-8"><a href="#cb326-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb326-9"><a href="#cb326-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose the matrix to have cell types as rows</span></span>
<span id="cb326-10"><a href="#cb326-10" aria-hidden="true" tabindex="-1"></a>binary_matrix_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(binary_matrix[<span class="sc">-</span><span class="dv">1</span>]))</span>
<span id="cb326-11"><a href="#cb326-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(binary_matrix_t) <span class="ot">&lt;-</span> binary_matrix<span class="sc">$</span>gene</span>
<span id="cb326-12"><a href="#cb326-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb326-13"><a href="#cb326-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate Jaccard similarity matrix</span></span>
<span id="cb326-14"><a href="#cb326-14" aria-hidden="true" tabindex="-1"></a>jaccard_matrix <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(binary_matrix_t, <span class="at">method =</span> <span class="st">"jaccard"</span>)</span>
<span id="cb326-15"><a href="#cb326-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb326-16"><a href="#cb326-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform hierarchical clustering</span></span>
<span id="cb326-17"><a href="#cb326-17" aria-hidden="true" tabindex="-1"></a>hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(jaccard_matrix, <span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb326-18"><a href="#cb326-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb326-19"><a href="#cb326-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert distance matrix to similarity matrix for heatmap</span></span>
<span id="cb326-20"><a href="#cb326-20" aria-hidden="true" tabindex="-1"></a>similarity_matrix <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">as.matrix</span>(jaccard_matrix)</span>
<span id="cb326-21"><a href="#cb326-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb326-22"><a href="#cb326-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Set colour to make it more obvious which cells have a similarity of 0.2 or more</span></span>
<span id="cb326-23"><a href="#cb326-23" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"blue"</span>, <span class="st">"red"</span>))</span>
<span id="cb326-24"><a href="#cb326-24" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(similarity_matrix, <span class="at">col =</span> col_fun)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-35-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19"><img src="04_figures_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="alternative-to-look-at-overlapping-genes" class="level4">
<h4 class="anchored" data-anchor-id="alternative-to-look-at-overlapping-genes">Alternative to look at overlapping genes</h4>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [36]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb327"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb327-1"><a href="#cb327-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to calculate proportion of overlapping DEGs</span></span>
<span id="cb327-2"><a href="#cb327-2" aria-hidden="true" tabindex="-1"></a>calculate_overlap <span class="ot">&lt;-</span> <span class="cf">function</span>(df, celltype1, celltype2, <span class="at">up_down_degs =</span> <span class="st">"Up"</span>) {</span>
<span id="cb327-3"><a href="#cb327-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter DEGs for each cell type</span></span>
<span id="cb327-4"><a href="#cb327-4" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(deg_direction <span class="sc">==</span> up_down_degs)</span>
<span id="cb327-5"><a href="#cb327-5" aria-hidden="true" tabindex="-1"></a>  degs_1 <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">==</span> celltype1) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">pull</span>(gene)</span>
<span id="cb327-6"><a href="#cb327-6" aria-hidden="true" tabindex="-1"></a>  degs_2 <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">==</span> celltype2) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">pull</span>(gene)</span>
<span id="cb327-7"><a href="#cb327-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb327-8"><a href="#cb327-8" aria-hidden="true" tabindex="-1"></a>  assertthat<span class="sc">::</span><span class="fu">assert_that</span>(<span class="fu">sum</span>(<span class="fu">duplicated</span>(degs_1)) <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb327-9"><a href="#cb327-9" aria-hidden="true" tabindex="-1"></a>  assertthat<span class="sc">::</span><span class="fu">assert_that</span>(<span class="fu">sum</span>(<span class="fu">duplicated</span>(degs_2)) <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb327-10"><a href="#cb327-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find overlapping DEGs</span></span>
<span id="cb327-11"><a href="#cb327-11" aria-hidden="true" tabindex="-1"></a>  overlapping_degs <span class="ot">&lt;-</span> <span class="fu">intersect</span>(degs_1, degs_2)</span>
<span id="cb327-12"><a href="#cb327-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb327-13"><a href="#cb327-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate proportions of overlapping DEGs</span></span>
<span id="cb327-14"><a href="#cb327-14" aria-hidden="true" tabindex="-1"></a>  proportion_overlapping_1 <span class="ot">&lt;-</span> <span class="fu">length</span>(overlapping_degs) <span class="sc">/</span> <span class="fu">length</span>(degs_1)</span>
<span id="cb327-15"><a href="#cb327-15" aria-hidden="true" tabindex="-1"></a>  proportion_overlapping_2 <span class="ot">&lt;-</span> <span class="fu">length</span>(overlapping_degs) <span class="sc">/</span> <span class="fu">length</span>(degs_2)</span>
<span id="cb327-16"><a href="#cb327-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb327-17"><a href="#cb327-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">prop1 =</span> proportion_overlapping_1, <span class="at">prop2 =</span> proportion_overlapping_2))</span>
<span id="cb327-18"><a href="#cb327-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb327-19"><a href="#cb327-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-20"><a href="#cb327-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique cell types</span></span>
<span id="cb327-21"><a href="#cb327-21" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(res_sig<span class="sc">$</span>celltype)</span>
<span id="cb327-22"><a href="#cb327-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-23"><a href="#cb327-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame of all combinations of cell types</span></span>
<span id="cb327-24"><a href="#cb327-24" aria-hidden="true" tabindex="-1"></a>combinations <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(<span class="at">celltype1 =</span> celltypes, <span class="at">celltype2 =</span> celltypes)</span>
<span id="cb327-25"><a href="#cb327-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-26"><a href="#cb327-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the calculate_overlap function to each combination</span></span>
<span id="cb327-27"><a href="#cb327-27" aria-hidden="true" tabindex="-1"></a>overlap_values <span class="ot">&lt;-</span> combinations <span class="sc">%&gt;%</span></span>
<span id="cb327-28"><a href="#cb327-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">overlap =</span> <span class="fu">map2</span>(celltype1, celltype2, <span class="sc">~</span> <span class="fu">calculate_overlap</span>(res_sig, .x, .y)))</span>
<span id="cb327-29"><a href="#cb327-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-30"><a href="#cb327-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate the results into two matrices</span></span>
<span id="cb327-31"><a href="#cb327-31" aria-hidden="true" tabindex="-1"></a>overlap_matrix_1 <span class="ot">&lt;-</span> overlap_values <span class="sc">%&gt;%</span></span>
<span id="cb327-32"><a href="#cb327-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">overlap =</span> <span class="fu">map_dbl</span>(overlap, <span class="st">"prop1"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb327-33"><a href="#cb327-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype2, <span class="at">values_from =</span> overlap) <span class="sc">%&gt;%</span></span>
<span id="cb327-34"><a href="#cb327-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"celltype1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb327-35"><a href="#cb327-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb327-36"><a href="#cb327-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-37"><a href="#cb327-37" aria-hidden="true" tabindex="-1"></a>overlap_values_down <span class="ot">&lt;-</span> combinations <span class="sc">%&gt;%</span></span>
<span id="cb327-38"><a href="#cb327-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">overlap =</span> <span class="fu">map2</span>(celltype1, celltype2, <span class="sc">~</span> <span class="fu">calculate_overlap</span>(res_sig, .x, .y, <span class="st">"Down"</span>)))</span>
<span id="cb327-39"><a href="#cb327-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-40"><a href="#cb327-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the results to a matrix</span></span>
<span id="cb327-41"><a href="#cb327-41" aria-hidden="true" tabindex="-1"></a>overlap_matrix_down <span class="ot">&lt;-</span> overlap_values_down <span class="sc">%&gt;%</span></span>
<span id="cb327-42"><a href="#cb327-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">overlap =</span> <span class="fu">map_dbl</span>(overlap, <span class="st">"prop1"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb327-43"><a href="#cb327-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> celltype2, <span class="at">values_from =</span> overlap) <span class="sc">%&gt;%</span></span>
<span id="cb327-44"><a href="#cb327-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"celltype1"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb327-45"><a href="#cb327-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb327-46"><a href="#cb327-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb327-47"><a href="#cb327-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb327-48"><a href="#cb327-48" aria-hidden="true" tabindex="-1"></a>h1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(overlap_matrix_1,</span>
<span id="cb327-49"><a href="#cb327-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Proportion"</span>,</span>
<span id="cb327-50"><a href="#cb327-50" aria-hidden="true" tabindex="-1"></a>        <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb327-51"><a href="#cb327-51" aria-hidden="true" tabindex="-1"></a>          <span class="fu">grid.text</span>(<span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, overlap_matrix_1[i, j]), x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb327-52"><a href="#cb327-52" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb327-53"><a href="#cb327-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb327-54"><a href="#cb327-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb327-55"><a href="#cb327-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_heatmap_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb327-56"><a href="#cb327-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">"Proportion of Overlapping Up DEGs"</span>)</span>
<span id="cb327-57"><a href="#cb327-57" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb327-58"><a href="#cb327-58" aria-hidden="true" tabindex="-1"></a>h2 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(overlap_matrix_down,</span>
<span id="cb327-59"><a href="#cb327-59" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"Proportion"</span>,</span>
<span id="cb327-60"><a href="#cb327-60" aria-hidden="true" tabindex="-1"></a>        <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb327-61"><a href="#cb327-61" aria-hidden="true" tabindex="-1"></a>          <span class="fu">grid.text</span>(<span class="fu">sprintf</span>(<span class="st">"%.2f"</span>, overlap_matrix_down[i, j]), x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb327-62"><a href="#cb327-62" aria-hidden="true" tabindex="-1"></a>        },</span>
<span id="cb327-63"><a href="#cb327-63" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb327-64"><a href="#cb327-64" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb327-65"><a href="#cb327-65" aria-hidden="true" tabindex="-1"></a>        <span class="at">show_heatmap_legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb327-66"><a href="#cb327-66" aria-hidden="true" tabindex="-1"></a>        <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">"Proportion of Overlapping Down DEGs"</span>)</span>
<span id="cb327-67"><a href="#cb327-67" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb327-68"><a href="#cb327-68" aria-hidden="true" tabindex="-1"></a>h1 <span class="sc">+</span> h2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: Proportion</code></pre>
</div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="04_figures_files/figure-html/unnamed-chunk-36-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-20"><img src="04_figures_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb329"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb329-1"><a href="#cb329-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">=</span> <span class="fu">grid.grabExpr</span>(<span class="fu">draw</span>(h1))</span>
<span id="cb329-2"><a href="#cb329-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">=</span> <span class="fu">grid.grabExpr</span>(<span class="fu">draw</span>(h2))</span>
<span id="cb329-3"><a href="#cb329-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb329-4"><a href="#cb329-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/overlapping_up_degs.png"</span>), p1, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">20</span>)</span>
<span id="cb329-5"><a href="#cb329-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/overlapping_down_degs.png"</span>), p2, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
<section id="assemble-figure" class="level3">
<h3 class="anchored" data-anchor-id="assemble-figure">Assemble Figure</h3>
<ul>
<li>NOTES:
<ul>
<li>Make a heatmap of the shared DEGs (rows and columns celltypes)</li>
</ul></li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [37]:</pre></div><div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb330"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb330-1"><a href="#cb330-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb330-2"><a href="#cb330-2" aria-hidden="true" tabindex="-1"></a>img1 <span class="ot">&lt;-</span> <span class="fu">image_read</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/amyloid_ppi_network_with_legend.png"</span>)</span>
<span id="cb330-3"><a href="#cb330-3" aria-hidden="true" tabindex="-1"></a>img1_gg <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>() <span class="sc">+</span> <span class="fu">draw_image</span>(img1)</span>
<span id="cb330-4"><a href="#cb330-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-5"><a href="#cb330-5" aria-hidden="true" tabindex="-1"></a><span class="co"># row_2 &lt;- plot_grid(g1, img1_gg, labels = c("C", "D"), rel_widths = c(2, 1), label_colour = "black")</span></span>
<span id="cb330-6"><a href="#cb330-6" aria-hidden="true" tabindex="-1"></a><span class="co"># row_2 &lt;- plot_grid(g1, img1_gg, labels = c("C", "D"), label_colour = "black")</span></span>
<span id="cb330-7"><a href="#cb330-7" aria-hidden="true" tabindex="-1"></a><span class="co"># row_3 &lt;- plot_grid(p3, p4, labels = c("E", "F"), label_colour = "black")</span></span>
<span id="cb330-8"><a href="#cb330-8" aria-hidden="true" tabindex="-1"></a><span class="co"># row_3</span></span>
<span id="cb330-9"><a href="#cb330-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-10"><a href="#cb330-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-11"><a href="#cb330-11" aria-hidden="true" tabindex="-1"></a>final_plot3 <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(deg_count_plot, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>), <span class="at">label_colour =</span> <span class="st">"black"</span>)</span>
<span id="cb330-12"><a href="#cb330-12" aria-hidden="true" tabindex="-1"></a>final_plot3</span>
<span id="cb330-13"><a href="#cb330-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-14"><a href="#cb330-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/figure2.png"</span>), <span class="at">plot =</span> final_plot2,</span>
<span id="cb330-15"><a href="#cb330-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">15</span>, <span class="at">units =</span> <span class="st">"in"</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"selector":".lightbox","openEffect":"zoom","closeEffect":"zoom","loop":false,"descPosition":"bottom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>