<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Blood brain barrier in AD - scflow prep</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="styles.css">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Blood brain barrier in AD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../06_dissemination/03_website/01_data-analysis.html" rel="" target="" aria-current="page">
 <span class="menu-text">Data analysis</span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools ms-auto">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../04_data_analysis/001_snrnaseq_analysis/08_scflow_prep.html">scflow prep</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../06_dissemination/03_website/01_data-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis Pipeline</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/01_fastqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run FastQC on sample</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/02_multiqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run MultiQC</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/03_generate_cellranger_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/04_cellranger_count_set1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/05_cellranger_count_set2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 2</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/06_cellranger_count_set3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 3</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/07_cellranger_aggr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Aggregate cellranger count reports</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/08_scflow_prep.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">scflow prep</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/09_scflow_run.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run SCflow</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/10_scflow_results.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Analysis scFlow results</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">04_data_analysis/001_snrnaseq_analysis/11_qseudobulk.qmd</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">04_data_analysis/001_snrnaseq_analysis/12_pathway-analysis.qmd</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#getting-scflow-on-hawk" id="toc-getting-scflow-on-hawk" class="nav-link active" data-scroll-target="#getting-scflow-on-hawk"><span class="header-section-number">1</span> Getting scflow on hawk</a></li>
  <li>
<a href="#input-files" id="toc-input-files" class="nav-link" data-scroll-target="#input-files"><span class="header-section-number">2</span> Input files</a>
  <ul class="collapse">
<li><a href="#manifest-file" id="toc-manifest-file" class="nav-link" data-scroll-target="#manifest-file"><span class="header-section-number">2.1</span> Manifest file</a></li>
  <li><a href="#sample-metadata" id="toc-sample-metadata" class="nav-link" data-scroll-target="#sample-metadata"><span class="header-section-number">2.2</span> Sample metadata</a></li>
  <li><a href="#getting-marker-genes" id="toc-getting-marker-genes" class="nav-link" data-scroll-target="#getting-marker-genes"><span class="header-section-number">2.3</span> Getting marker genes</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">scflow prep</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><section id="getting-scflow-on-hawk" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Getting scflow on hawk</h1>
<p>So someone in Super Computing Wales added a handy nf-core profile config we can use, and this also includes a nice guide to getting set up for using nf-core pipelines: <a href="https://github.com/nf-core/configs/blob/master/docs/scw.md">https://github.com/nf-core/configs/blob/master/docs/scw.md</a></p>
<p>I’ve followed the guide for offline use, which I won’t rehash here, check the above link if curious.</p>
</section><section id="input-files" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Input files</h1>
<p>To run scflow we need a manifest tsv file where the first column is a unique sample ID and the second column is the file path to the directory that has the <code>barcodes.tsv</code>, <code>genes.tsv</code> and <code>matrix.mtx</code> files.</p>
<p>I’m assuming we want to use the filtered gene-barcode matrices in MEX format as opposed to the raw files.</p>
<p>Note that this page has a mix of bash (run on Hawk) and R code (run locally), I’ve just run these bits manually and recorded the code here for now.</p>
<section id="manifest-file" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="manifest-file">
<span class="header-section-number">2.1</span> Manifest file</h2>
<p>First let’s sort the manifest for our samples.</p>
<p>Here’s how I generated it:</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># list the cellranger count output directories</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>find <span class="sc">/</span>scratch<span class="sc">/</span>scw1329<span class="sc">/</span>gmbh<span class="sc">/</span>blood<span class="sc">-</span>brain<span class="sc">-</span>barrier<span class="sc">-</span><span class="cf">in</span><span class="sc">-</span>ad<span class="sc">/</span>03_data<span class="sc">/</span>990_processed_data<span class="sc">/</span>001_snrnaseq<span class="sc">/</span>04_cellranger_count<span class="sc">/</span> <span class="sc">-</span>maxdepth <span class="dv">2</span> <span class="sc">-</span>mindepth <span class="dv">2</span> <span class="sc">-</span>type d <span class="sc">&gt;</span> scflow_manifest.txt</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># use the last directory in the path as the sample id</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>awk <span class="sc">-</span>F<span class="st">'/'</span> <span class="st">'{print $NF "</span><span class="sc">\t</span><span class="st">" $0}'</span> scflow_manifest.txt <span class="sc">&gt;</span> scflow_manifest2.txt</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># append "outs/filtered_gene_bc_matrices" to file paths</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>sed <span class="st">"s|$|/outs/filtered_feature_bc_matrix|"</span> scflow_manifest2.txt <span class="sc">&gt;</span> scflow_manifest3.txt</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the undetermined sample and P_39</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>grep <span class="sc">-</span>v <span class="st">"Undetermined"</span> scflow_manifest3.txt <span class="sc">&gt;</span> scflow_manifest4.txt</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>grep <span class="sc">-</span>v <span class="st">"P_39"</span> scflow_manifest4.txt <span class="sc">&gt;</span> scflow_manifest5.txt</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># add the table headers</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>awk <span class="sc">-</span>v line<span class="ot">=</span><span class="st">"key</span><span class="sc">\t</span><span class="st">filepath"</span> <span class="st">'BEGIN{print line}{print}'</span> scflow_manifest5.txt <span class="sc">&gt;</span> scflow_manifest.tsv</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>rm scflow_manifest.txt scflow_manifest2.txt scflow_manifest3.txt scflow_manifest4.txt scflow_manifest5.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We would also like to include the brain atlas from Yang et al.&nbsp;2022.<span class="citation" data-cites="yang2022">(<a href="#ref-yang2022" role="doc-biblioref"><strong>yang2022?</strong></a>)</span> Their data is on GEO so I’ve downloaded and extracted that to gluster.</p>
</section><section id="sample-metadata" class="level2" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="sample-metadata">
<span class="header-section-number">2.2</span> Sample metadata</h2>
<p>We also need to sort out the sample metadata for the Yang data.</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/">purrr</a></span><span class="op">)</span></span>
<span><span class="va">pkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"readr"</span>, <span class="st">"dplyr"</span>, <span class="st">"here"</span>, <span class="st">"GEOquery"</span>, <span class="st">"magrittr"</span>, <span class="st">"tidyr"</span>, <span class="st">"yaml"</span>, </span>
<span>         <span class="st">"gt"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">pkg</span>, <span class="va">library</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[[1]]
[1] "readr"     "purrr"     "stats"     "graphics"  "grDevices" "datasets" 
[7] "utils"     "methods"   "base"     

[[2]]
 [1] "dplyr"     "readr"     "purrr"     "stats"     "graphics"  "grDevices"
 [7] "datasets"  "utils"     "methods"   "base"     

[[3]]
 [1] "here"      "dplyr"     "readr"     "purrr"     "stats"     "graphics" 
 [7] "grDevices" "datasets"  "utils"     "methods"   "base"     

[[4]]
 [1] "GEOquery"     "Biobase"      "BiocGenerics" "here"         "dplyr"       
 [6] "readr"        "purrr"        "stats"        "graphics"     "grDevices"   
[11] "datasets"     "utils"        "methods"      "base"        

[[5]]
 [1] "magrittr"     "GEOquery"     "Biobase"      "BiocGenerics" "here"        
 [6] "dplyr"        "readr"        "purrr"        "stats"        "graphics"    
[11] "grDevices"    "datasets"     "utils"        "methods"      "base"        

[[6]]
 [1] "tidyr"        "magrittr"     "GEOquery"     "Biobase"      "BiocGenerics"
 [6] "here"         "dplyr"        "readr"        "purrr"        "stats"       
[11] "graphics"     "grDevices"    "datasets"     "utils"        "methods"     
[16] "base"        

[[7]]
 [1] "yaml"         "tidyr"        "magrittr"     "GEOquery"     "Biobase"     
 [6] "BiocGenerics" "here"         "dplyr"        "readr"        "purrr"       
[11] "stats"        "graphics"     "grDevices"    "datasets"     "utils"       
[16] "methods"      "base"        

[[8]]
 [1] "gt"           "yaml"         "tidyr"        "magrittr"     "GEOquery"    
 [6] "Biobase"      "BiocGenerics" "here"         "dplyr"        "readr"       
[11] "purrr"        "stats"        "graphics"     "grDevices"    "datasets"    
[16] "utils"        "methods"      "base"        </code></pre>
</div>
</div>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># current metadata structure for first 2 sample batches</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_meta_merged_final.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="co"># add a column for the technique - don't know much about it so I'll just call</span></span>
<span><span class="co"># it Oxford for now</span></span>
<span><span class="va">metadata</span><span class="op">$</span><span class="va">technique</span> <span class="op">&lt;-</span> <span class="st">"oxford"</span></span>
<span><span class="co"># also don't know brain region</span></span>
<span><span class="va">metadata</span><span class="op">$</span><span class="va">brain_region</span> <span class="op">&lt;-</span> <span class="st">"dunno"</span></span>
<span></span>
<span><span class="co"># get yang data</span></span>
<span><span class="va">gsm</span> <span class="op">&lt;-</span> <span class="fu">getGEO</span><span class="op">(</span><span class="st">"GSE163577"</span><span class="op">)</span></span>
<span><span class="co"># get phenotype data</span></span>
<span><span class="va">phenodata</span> <span class="op">&lt;-</span> <span class="va">gsm</span><span class="op">$</span><span class="va">GSE163577_series_matrix.txt.gz</span><span class="op">@</span><span class="va">phenoData</span><span class="op">@</span><span class="va">data</span></span>
<span><span class="co"># select useful cols</span></span>
<span><span class="va">phenodata</span> <span class="op">&lt;-</span> <span class="va">phenodata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span></span>
<span>    <span class="va">title</span>,</span>
<span>    <span class="va">geo_accession</span>,</span>
<span>    <span class="va">extract_protocol_ch1</span>,</span>
<span>    <span class="va">data_processing</span>,</span>
<span>    <span class="va">platform_id</span>,</span>
<span>    <span class="va">instrument_model</span>,</span>
<span>    <span class="va">supplementary_file_1</span>,</span>
<span>    <span class="va">`brain region:ch1`</span>,</span>
<span>    <span class="va">`disease:ch1`</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># load scflow test sample sheet</span></span>
<span><span class="va">example_samplesheet</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/90_sample_info/scflowSampleSheet_example.tsv"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>There are nice geo accessions that would be handy to use as unique sample IDs for the yang manifest file</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save id and path for manifest file</span></span>
<span><span class="va">yang_manifest</span> <span class="op">&lt;-</span> <span class="va">phenodata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">geo_accession</span>, <span class="va">supplementary_file_1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get the last part of path</span></span>
<span><span class="va">last_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">".*/"</span>, <span class="st">""</span>, <span class="va">yang_manifest</span><span class="op">$</span><span class="va">supplementary_file_1</span><span class="op">)</span></span>
<span><span class="co"># get after first _</span></span>
<span><span class="va">last_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"^[^_]*_"</span>, <span class="st">""</span>, <span class="va">last_path</span><span class="op">)</span></span>
<span><span class="co"># remove file extension</span></span>
<span><span class="va">last_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\.tar\\.gz$"</span>, <span class="st">""</span>, <span class="va">last_path</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># update path to yang data on gluster</span></span>
<span><span class="va">yang_manifest</span><span class="op">$</span><span class="va">supplementary_file_1</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span></span>
<span>    <span class="st">"/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/06_scflow/yang_2022_data/"</span>,</span>
<span>    <span class="va">last_path</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co"># update names to be in line with scflow manifest file</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">yang_manifest</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"key"</span>, <span class="st">"filepath"</span><span class="op">)</span></span>
<span><span class="co"># save to tsv</span></span>
<span><span class="fu">write_tsv</span><span class="op">(</span><span class="va">yang_manifest</span>, <span class="fu">here</span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/06_scflow/yang_manifest.tsv"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then we just need to merge the two manifest files together.</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>(cat scflow_manifest.tsv <span class="sc">&amp;&amp;</span> tail <span class="sc">-</span>n <span class="sc">+</span><span class="dv">2</span> yang_manifest.tsv) <span class="sc">&gt;</span> scflow_manifest_both.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>It turns out that you can’t have delimiters in the key for the mainfest/sample files, so I need to remove the underscores too</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>awk <span class="st">'BEGIN{FS=OFS="</span><span class="sc">\t</span><span class="st">"} {gsub("_", "", $1)} 1'</span>  scflow_manifest_both.tsv <span class="sc">&gt;</span> scflow_manifest_final.tsv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We need to get the phenotype data from the yang GEO data in line with our phenotype data to join them. My understanding from the Yang paper is that they only looked at the vascular/perivascular component, so I’ll label the <code>prep</code> as <code>V</code> for now.</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># sort yang phenotype data and join to ours</span></span>
<span><span class="va">merged_meta</span> <span class="op">&lt;-</span> <span class="va">phenodata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">geo_accession</span>,</span>
<span>                brain_region <span class="op">=</span> <span class="va">`brain region:ch1`</span>,</span>
<span>                type <span class="op">=</span> <span class="va">`disease:ch1`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># add cell subsetting technique and cell population</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>technique <span class="op">=</span> <span class="st">"VINE-seq"</span>, prep <span class="op">=</span> <span class="st">"V"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># select cols of interest</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">brain_region</span>, <span class="va">type</span>, <span class="va">technique</span>, <span class="va">prep</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># merge datasets</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html">bind_rows</a></span><span class="op">(</span><span class="va">metadata</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># make control label consistent</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"CT"</span>, <span class="st">"Control"</span>, <span class="va">type</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># add final group column for differential expression</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">type</span>, <span class="va">prep</span>, <span class="va">technique</span>, <span class="va">brain_region</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>,</span>
<span>                sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">""</span>, <span class="va">sample</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># rename to sample sheet style</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>manifest <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                diagnosis <span class="op">=</span> <span class="va">type</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save metadata</span></span>
<span><span class="fu">write_tsv</span><span class="op">(</span><span class="va">merged_meta</span>, file <span class="op">=</span> <span class="fu">here</span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/90_sample_info/sample_metadata.tsv"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="getting-marker-genes" class="level2" data-number="2.3"><h2 data-number="2.3" class="anchored" data-anchor-id="getting-marker-genes">
<span class="header-section-number">2.3</span> Getting marker genes</h2>
<p>The EBI has the <a href="https://www.ebi.ac.uk/gxa/sc/home">Single cell expression atlas</a> so I’ll use that for our gene annotation markers. There is an R package on Bioconductor for querying it, but it doesn’t seem to have an easy way of querying by tissues/organ, so I’ll just manually download them (there aren’t that many).</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># list gene marker csvs</span></span>
<span><span class="va">gene_markers_files</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span></span>
<span>    <span class="fu">here</span><span class="op">(</span></span>
<span>      <span class="st">"03_data/990_processed_data/001_snrnaseq/06_scflow/2023-05-22_gene-markers"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    full.names <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract the final part of the path after the last "/"</span></span>
<span><span class="va">file_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">gene_markers_files</span><span class="op">)</span></span>
<span><span class="co"># Remove the file extension</span></span>
<span><span class="va">file_names</span> <span class="op">&lt;-</span> <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/fileutils.html">file_path_sans_ext</a></span><span class="op">(</span><span class="va">file_names</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene_markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">gene_markers_files</span>, <span class="va">read_csv</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html">set_names</a></span><span class="op">(</span><span class="va">file_names</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add annotation name</span></span>
<span><span class="va">gene_markers</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">gene_markers</span>, <span class="va">file_names</span>, <span class="op">~</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="va">.x</span>, annotation <span class="op">=</span> <span class="va">.y</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># bind rows</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html">bind_rows</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># the odd column headers are the experiment ID and the values are expression</span></span>
<span><span class="co"># pivot long</span></span>
<span><span class="va">gene_markers</span> <span class="op">&lt;-</span> <span class="va">gene_markers</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">Category</span>, <span class="va">annotation</span><span class="op">)</span>,</span>
<span>               names_to <span class="op">=</span> <span class="st">"experiment_id"</span>,</span>
<span>               values_to <span class="op">=</span> <span class="st">"expression"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># remove cases where expression is 0</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">Category</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter out duplicate genes within each annotation</span></span>
<span><span class="va">gene_markers_clean</span> <span class="op">&lt;-</span> <span class="va">gene_markers</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co"># remove the "top-genes" part of the string</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"top-.*"</span>, <span class="st">""</span>, <span class="va">annotation</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gt</span><span class="op">(</span><span class="va">gene_markers_clean</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div id="gvbsxdriwa" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#gvbsxdriwa table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#gvbsxdriwa thead, #gvbsxdriwa tbody, #gvbsxdriwa tfoot, #gvbsxdriwa tr, #gvbsxdriwa td, #gvbsxdriwa th {
  border-style: none;
}

#gvbsxdriwa p {
  margin: 0;
  padding: 0;
}

#gvbsxdriwa .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gvbsxdriwa .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#gvbsxdriwa .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gvbsxdriwa .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gvbsxdriwa .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gvbsxdriwa .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gvbsxdriwa .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gvbsxdriwa .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gvbsxdriwa .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gvbsxdriwa .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gvbsxdriwa .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gvbsxdriwa .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gvbsxdriwa .gt_spanner_row {
  border-bottom-style: hidden;
}

#gvbsxdriwa .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#gvbsxdriwa .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gvbsxdriwa .gt_from_md > :first-child {
  margin-top: 0;
}

#gvbsxdriwa .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gvbsxdriwa .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gvbsxdriwa .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#gvbsxdriwa .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#gvbsxdriwa .gt_row_group_first td {
  border-top-width: 2px;
}

#gvbsxdriwa .gt_row_group_first th {
  border-top-width: 2px;
}

#gvbsxdriwa .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gvbsxdriwa .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#gvbsxdriwa .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#gvbsxdriwa .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gvbsxdriwa .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gvbsxdriwa .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gvbsxdriwa .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#gvbsxdriwa .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gvbsxdriwa .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gvbsxdriwa .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gvbsxdriwa .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gvbsxdriwa .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gvbsxdriwa .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gvbsxdriwa .gt_left {
  text-align: left;
}

#gvbsxdriwa .gt_center {
  text-align: center;
}

#gvbsxdriwa .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gvbsxdriwa .gt_font_normal {
  font-weight: normal;
}

#gvbsxdriwa .gt_font_bold {
  font-weight: bold;
}

#gvbsxdriwa .gt_font_italic {
  font-style: italic;
}

#gvbsxdriwa .gt_super {
  font-size: 65%;
}

#gvbsxdriwa .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#gvbsxdriwa .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#gvbsxdriwa .gt_indent_1 {
  text-indent: 5px;
}

#gvbsxdriwa .gt_indent_2 {
  text-indent: 10px;
}

#gvbsxdriwa .gt_indent_3 {
  text-indent: 15px;
}

#gvbsxdriwa .gt_indent_4 {
  text-indent: 20px;
}

#gvbsxdriwa .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="gene">gene</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="experiment_id">experiment_id</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="expression">expression</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="astrocyte-of-the-cerebra">astrocyte-of-the-cerebra</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="astrocyte-of-the-cerebra  gene" class="gt_row gt_left">CLU</td>
<td headers="astrocyte-of-the-cerebra  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="astrocyte-of-the-cerebra  expression" class="gt_row gt_right">5733.4755</td>
</tr>
<tr>
<td headers="astrocyte-of-the-cerebra  gene" class="gt_row gt_left">PTGDS</td>
<td headers="astrocyte-of-the-cerebra  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="astrocyte-of-the-cerebra  expression" class="gt_row gt_right">2628.1210</td>
</tr>
<tr>
<td headers="astrocyte-of-the-cerebra  gene" class="gt_row gt_left">SLC1A2</td>
<td headers="astrocyte-of-the-cerebra  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="astrocyte-of-the-cerebra  expression" class="gt_row gt_right">4977.4883</td>
</tr>
<tr>
<td headers="astrocyte-of-the-cerebra  gene" class="gt_row gt_left">SLC1A3</td>
<td headers="astrocyte-of-the-cerebra  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="astrocyte-of-the-cerebra  expression" class="gt_row gt_right">4415.0110</td>
</tr>
<tr>
<td headers="astrocyte-of-the-cerebra  gene" class="gt_row gt_left">MT2A</td>
<td headers="astrocyte-of-the-cerebra  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="astrocyte-of-the-cerebra  expression" class="gt_row gt_right">2587.3220</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="astrocyte">astrocyte</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="astrocyte  gene" class="gt_row gt_left">SPARCL1</td>
<td headers="astrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="astrocyte  expression" class="gt_row gt_right">12453.4555</td>
</tr>
<tr>
<td headers="astrocyte  gene" class="gt_row gt_left">ATP1A2</td>
<td headers="astrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="astrocyte  expression" class="gt_row gt_right">2682.5862</td>
</tr>
<tr>
<td headers="astrocyte  gene" class="gt_row gt_left">CPE</td>
<td headers="astrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="astrocyte  expression" class="gt_row gt_right">6516.6291</td>
</tr>
<tr>
<td headers="astrocyte  gene" class="gt_row gt_left">NTRK2</td>
<td headers="astrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="astrocyte  expression" class="gt_row gt_right">2098.7989</td>
</tr>
<tr>
<td headers="astrocyte  gene" class="gt_row gt_left">SLC1A2</td>
<td headers="astrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="astrocyte  expression" class="gt_row gt_right">6660.5045</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="b-cell">b-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="b-cell  gene" class="gt_row gt_left">RPS19</td>
<td headers="b-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="b-cell  expression" class="gt_row gt_right">5104.1410</td>
</tr>
<tr>
<td headers="b-cell  gene" class="gt_row gt_left">RPL36</td>
<td headers="b-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="b-cell  expression" class="gt_row gt_right">2518.1768</td>
</tr>
<tr>
<td headers="b-cell  gene" class="gt_row gt_left">RPS28</td>
<td headers="b-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="b-cell  expression" class="gt_row gt_right">5210.8823</td>
</tr>
<tr>
<td headers="b-cell  gene" class="gt_row gt_left">RPLP1</td>
<td headers="b-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="b-cell  expression" class="gt_row gt_right">5189.7509</td>
</tr>
<tr>
<td headers="b-cell  gene" class="gt_row gt_left">ENSG00000223532</td>
<td headers="b-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="b-cell  expression" class="gt_row gt_right">1006.6358</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="cerebral-cortex-endothel">cerebral-cortex-endothel</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="cerebral-cortex-endothel  gene" class="gt_row gt_left">CLDN5</td>
<td headers="cerebral-cortex-endothel  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-endothel  expression" class="gt_row gt_right">3000.6105</td>
</tr>
<tr>
<td headers="cerebral-cortex-endothel  gene" class="gt_row gt_left">IFITM3</td>
<td headers="cerebral-cortex-endothel  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-endothel  expression" class="gt_row gt_right">3683.9235</td>
</tr>
<tr>
<td headers="cerebral-cortex-endothel  gene" class="gt_row gt_left">TMSB10</td>
<td headers="cerebral-cortex-endothel  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-endothel  expression" class="gt_row gt_right">8179.4912</td>
</tr>
<tr>
<td headers="cerebral-cortex-endothel  gene" class="gt_row gt_left">MT2A</td>
<td headers="cerebral-cortex-endothel  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-endothel  expression" class="gt_row gt_right">6327.5722</td>
</tr>
<tr>
<td headers="cerebral-cortex-endothel  gene" class="gt_row gt_left">RPS28</td>
<td headers="cerebral-cortex-endothel  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-endothel  expression" class="gt_row gt_right">3475.7807</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="cerebral-cortex-glial-ce">cerebral-cortex-glial-ce</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="cerebral-cortex-glial-ce  gene" class="gt_row gt_left">PLP1</td>
<td headers="cerebral-cortex-glial-ce  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-glial-ce  expression" class="gt_row gt_right">4761.9043</td>
</tr>
<tr>
<td headers="cerebral-cortex-glial-ce  gene" class="gt_row gt_left">MBP</td>
<td headers="cerebral-cortex-glial-ce  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-glial-ce  expression" class="gt_row gt_right">2057.6130</td>
</tr>
<tr>
<td headers="cerebral-cortex-glial-ce  gene" class="gt_row gt_left">DST</td>
<td headers="cerebral-cortex-glial-ce  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-glial-ce  expression" class="gt_row gt_right">2372.8813</td>
</tr>
<tr>
<td headers="cerebral-cortex-glial-ce  gene" class="gt_row gt_left">PTGDS</td>
<td headers="cerebral-cortex-glial-ce  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-glial-ce  expression" class="gt_row gt_right">1721.1704</td>
</tr>
<tr>
<td headers="cerebral-cortex-glial-ce  gene" class="gt_row gt_left">PCDH9</td>
<td headers="cerebral-cortex-glial-ce  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cerebral-cortex-glial-ce  expression" class="gt_row gt_right">2728.5132</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="cortical-interneurontop">cortical-interneurontop</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="cortical-interneurontop  gene" class="gt_row gt_left">TCF4</td>
<td headers="cortical-interneurontop  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-interneurontop  expression" class="gt_row gt_right">1566.5796</td>
</tr>
<tr>
<td headers="cortical-interneurontop  gene" class="gt_row gt_left">GAD1</td>
<td headers="cortical-interneurontop  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-interneurontop  expression" class="gt_row gt_right">735.2941</td>
</tr>
<tr>
<td headers="cortical-interneurontop  gene" class="gt_row gt_left">ATP1B1</td>
<td headers="cortical-interneurontop  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-interneurontop  expression" class="gt_row gt_right">2055.8003</td>
</tr>
<tr>
<td headers="cortical-interneurontop  gene" class="gt_row gt_left">ERBB4</td>
<td headers="cortical-interneurontop  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-interneurontop  expression" class="gt_row gt_right">556.0189</td>
</tr>
<tr>
<td headers="cortical-interneurontop  gene" class="gt_row gt_left">SLC6A1</td>
<td headers="cortical-interneurontop  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-interneurontop  expression" class="gt_row gt_right">641.4368</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="cortical-layer-2-3-excit">cortical-layer-2-3-excit</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="cortical-layer-2-3-excit  gene" class="gt_row gt_left">PPFIA2</td>
<td headers="cortical-layer-2-3-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-2-3-excit  expression" class="gt_row gt_right">441.5986</td>
</tr>
<tr>
<td headers="cortical-layer-2-3-excit  gene" class="gt_row gt_left">ENC1</td>
<td headers="cortical-layer-2-3-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-2-3-excit  expression" class="gt_row gt_right">1096.4912</td>
</tr>
<tr>
<td headers="cortical-layer-2-3-excit  gene" class="gt_row gt_left">R3HDM1</td>
<td headers="cortical-layer-2-3-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-2-3-excit  expression" class="gt_row gt_right">403.3885</td>
</tr>
<tr>
<td headers="cortical-layer-2-3-excit  gene" class="gt_row gt_left">CAMK2A</td>
<td headers="cortical-layer-2-3-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-2-3-excit  expression" class="gt_row gt_right">883.3923</td>
</tr>
<tr>
<td headers="cortical-layer-2-3-excit  gene" class="gt_row gt_left">GRIA2</td>
<td headers="cortical-layer-2-3-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-2-3-excit  expression" class="gt_row gt_right">1666.6666</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="cortical-layer-4-excitat">cortical-layer-4-excitat</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="cortical-layer-4-excitat  gene" class="gt_row gt_left">MEF2C</td>
<td headers="cortical-layer-4-excitat  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-4-excitat  expression" class="gt_row gt_right">1343.1840</td>
</tr>
<tr>
<td headers="cortical-layer-4-excitat  gene" class="gt_row gt_left">RORB</td>
<td headers="cortical-layer-4-excitat  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-4-excitat  expression" class="gt_row gt_right">757.6046</td>
</tr>
<tr>
<td headers="cortical-layer-4-excitat  gene" class="gt_row gt_left">SYT1</td>
<td headers="cortical-layer-4-excitat  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-4-excitat  expression" class="gt_row gt_right">3600.6842</td>
</tr>
<tr>
<td headers="cortical-layer-4-excitat  gene" class="gt_row gt_left">IPCEF1</td>
<td headers="cortical-layer-4-excitat  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-4-excitat  expression" class="gt_row gt_right">252.1731</td>
</tr>
<tr>
<td headers="cortical-layer-4-excitat  gene" class="gt_row gt_left">NRGN</td>
<td headers="cortical-layer-4-excitat  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-4-excitat  expression" class="gt_row gt_right">740.1240</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="cortical-layer-5-6-excit">cortical-layer-5-6-excit</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="cortical-layer-5-6-excit  gene" class="gt_row gt_left">NPTX1</td>
<td headers="cortical-layer-5-6-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-5-6-excit  expression" class="gt_row gt_right">553.7098</td>
</tr>
<tr>
<td headers="cortical-layer-5-6-excit  gene" class="gt_row gt_left">CADPS</td>
<td headers="cortical-layer-5-6-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-5-6-excit  expression" class="gt_row gt_right">447.4273</td>
</tr>
<tr>
<td headers="cortical-layer-5-6-excit  gene" class="gt_row gt_left">CELF2</td>
<td headers="cortical-layer-5-6-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-5-6-excit  expression" class="gt_row gt_right">1155.2681</td>
</tr>
<tr>
<td headers="cortical-layer-5-6-excit  gene" class="gt_row gt_left">SLC17A7</td>
<td headers="cortical-layer-5-6-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-5-6-excit  expression" class="gt_row gt_right">440.8200</td>
</tr>
<tr>
<td headers="cortical-layer-5-6-excit  gene" class="gt_row gt_left">ARPP21</td>
<td headers="cortical-layer-5-6-excit  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="cortical-layer-5-6-excit  expression" class="gt_row gt_right">418.7605</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="excitatory-neuron">excitatory-neuron</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="excitatory-neuron  gene" class="gt_row gt_left">CHN1</td>
<td headers="excitatory-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="excitatory-neuron  expression" class="gt_row gt_right">2145.9229</td>
</tr>
<tr>
<td headers="excitatory-neuron  gene" class="gt_row gt_left">NRGN</td>
<td headers="excitatory-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="excitatory-neuron  expression" class="gt_row gt_right">1000.0001</td>
</tr>
<tr>
<td headers="excitatory-neuron  gene" class="gt_row gt_left">PPP3CA</td>
<td headers="excitatory-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="excitatory-neuron  expression" class="gt_row gt_right">1136.3636</td>
</tr>
<tr>
<td headers="excitatory-neuron  gene" class="gt_row gt_left">SYT1</td>
<td headers="excitatory-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="excitatory-neuron  expression" class="gt_row gt_right">3023.4316</td>
</tr>
<tr>
<td headers="excitatory-neuron  gene" class="gt_row gt_left">MT-ND3</td>
<td headers="excitatory-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="excitatory-neuron  expression" class="gt_row gt_right">2506.2656</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="immune-cell">immune-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="immune-cell  gene" class="gt_row gt_left">CD74</td>
<td headers="immune-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="immune-cell  expression" class="gt_row gt_right">9703.2005</td>
</tr>
<tr>
<td headers="immune-cell  gene" class="gt_row gt_left">TYROBP</td>
<td headers="immune-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="immune-cell  expression" class="gt_row gt_right">585.7133</td>
</tr>
<tr>
<td headers="immune-cell  gene" class="gt_row gt_left">FCER1G</td>
<td headers="immune-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="immune-cell  expression" class="gt_row gt_right">791.5301</td>
</tr>
<tr>
<td headers="immune-cell  gene" class="gt_row gt_left">SRGN</td>
<td headers="immune-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="immune-cell  expression" class="gt_row gt_right">1105.6933</td>
</tr>
<tr>
<td headers="immune-cell  gene" class="gt_row gt_left">LAPTM5</td>
<td headers="immune-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="immune-cell  expression" class="gt_row gt_right">1690.8645</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="microglial-cell">microglial-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="microglial-cell  gene" class="gt_row gt_left">FTL</td>
<td headers="microglial-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="microglial-cell  expression" class="gt_row gt_right">10436.6890</td>
</tr>
<tr>
<td headers="microglial-cell  gene" class="gt_row gt_left">CD74</td>
<td headers="microglial-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="microglial-cell  expression" class="gt_row gt_right">3796.1101</td>
</tr>
<tr>
<td headers="microglial-cell  gene" class="gt_row gt_left">TMSB4X</td>
<td headers="microglial-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="microglial-cell  expression" class="gt_row gt_right">7035.1340</td>
</tr>
<tr>
<td headers="microglial-cell  gene" class="gt_row gt_left">SPP1</td>
<td headers="microglial-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="microglial-cell  expression" class="gt_row gt_right">8000.5120</td>
</tr>
<tr>
<td headers="microglial-cell  gene" class="gt_row gt_left">RPL28</td>
<td headers="microglial-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="microglial-cell  expression" class="gt_row gt_right">2514.1431</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="neoplastic-cell">neoplastic-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="neoplastic-cell  gene" class="gt_row gt_left">EGFR</td>
<td headers="neoplastic-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neoplastic-cell  expression" class="gt_row gt_right">704.7337</td>
</tr>
<tr>
<td headers="neoplastic-cell  gene" class="gt_row gt_left">CLU</td>
<td headers="neoplastic-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neoplastic-cell  expression" class="gt_row gt_right">4150.2519</td>
</tr>
<tr>
<td headers="neoplastic-cell  gene" class="gt_row gt_left">CNN3</td>
<td headers="neoplastic-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neoplastic-cell  expression" class="gt_row gt_right">1001.4115</td>
</tr>
<tr>
<td headers="neoplastic-cell  gene" class="gt_row gt_left">SEC61G</td>
<td headers="neoplastic-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neoplastic-cell  expression" class="gt_row gt_right">162.9971</td>
</tr>
<tr>
<td headers="neoplastic-cell  gene" class="gt_row gt_left">TUBA1A</td>
<td headers="neoplastic-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neoplastic-cell  expression" class="gt_row gt_right">2820.4468</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="neuron">neuron</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="neuron  gene" class="gt_row gt_left">GAD1</td>
<td headers="neuron  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neuron  expression" class="gt_row gt_right">847.4346</td>
</tr>
<tr>
<td headers="neuron  gene" class="gt_row gt_left">TMEM130</td>
<td headers="neuron  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neuron  expression" class="gt_row gt_right">1164.7417</td>
</tr>
<tr>
<td headers="neuron  gene" class="gt_row gt_left">SNAP25</td>
<td headers="neuron  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neuron  expression" class="gt_row gt_right">2454.8066</td>
</tr>
<tr>
<td headers="neuron  gene" class="gt_row gt_left">TUBA4A</td>
<td headers="neuron  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neuron  expression" class="gt_row gt_right">308.4678</td>
</tr>
<tr>
<td headers="neuron  gene" class="gt_row gt_left">GAD2</td>
<td headers="neuron  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="neuron  expression" class="gt_row gt_right">497.7964</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="oligodendrocyte-precurso">oligodendrocyte-precurso</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">SCD5</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">1691.6135</td>
</tr>
<tr>
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">PLLP</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">1079.2020</td>
</tr>
<tr>
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">VCAN</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">1857.0101</td>
</tr>
<tr>
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">PTPRZ1</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">3211.9915</td>
</tr>
<tr>
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">APOD</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">1618.1230</td>
</tr>
<tr>
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">TNR</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">1216.5449</td>
</tr>
<tr>
<td headers="oligodendrocyte-precurso  gene" class="gt_row gt_left">OLIG1</td>
<td headers="oligodendrocyte-precurso  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte-precurso  expression" class="gt_row gt_right">1176.4706</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="oligodendrocyte">oligodendrocyte</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="oligodendrocyte  gene" class="gt_row gt_left">DBNDD2</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">5190.3916</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">QDPR</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">4438.4138</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">CRYAB</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">4762.0579</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">GPM6B</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">3275.1094</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">TMEM144</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">2217.2950</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">MBP</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">4975.1240</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">TF</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">3060.4436</td>
</tr>
<tr>
<td headers="oligodendrocyte  gene" class="gt_row gt_left">PLP1</td>
<td headers="oligodendrocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="oligodendrocyte  expression" class="gt_row gt_right">11936.3400</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="phagocyte">phagocyte</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="phagocyte  gene" class="gt_row gt_left">SPP1</td>
<td headers="phagocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="phagocyte  expression" class="gt_row gt_right">4012.8408</td>
</tr>
<tr>
<td headers="phagocyte  gene" class="gt_row gt_left">PLP1</td>
<td headers="phagocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="phagocyte  expression" class="gt_row gt_right">6361.3230</td>
</tr>
<tr>
<td headers="phagocyte  gene" class="gt_row gt_left">MBP</td>
<td headers="phagocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="phagocyte  expression" class="gt_row gt_right">2223.8696</td>
</tr>
<tr>
<td headers="phagocyte  gene" class="gt_row gt_left">SLC1A3</td>
<td headers="phagocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="phagocyte  expression" class="gt_row gt_right">1540.8320</td>
</tr>
<tr>
<td headers="phagocyte  gene" class="gt_row gt_left">FTL</td>
<td headers="phagocyte  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="phagocyte  expression" class="gt_row gt_right">2322.8804</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="pyramidal-neuron">pyramidal-neuron</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="pyramidal-neuron  gene" class="gt_row gt_left">MAP1B</td>
<td headers="pyramidal-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="pyramidal-neuron  expression" class="gt_row gt_right">4305.0430</td>
</tr>
<tr>
<td headers="pyramidal-neuron  gene" class="gt_row gt_left">CALM2</td>
<td headers="pyramidal-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="pyramidal-neuron  expression" class="gt_row gt_right">3703.7036</td>
</tr>
<tr>
<td headers="pyramidal-neuron  gene" class="gt_row gt_left">TUBA1B</td>
<td headers="pyramidal-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="pyramidal-neuron  expression" class="gt_row gt_right">2306.8050</td>
</tr>
<tr>
<td headers="pyramidal-neuron  gene" class="gt_row gt_left">CALM1</td>
<td headers="pyramidal-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="pyramidal-neuron  expression" class="gt_row gt_right">3956.4788</td>
</tr>
<tr>
<td headers="pyramidal-neuron  gene" class="gt_row gt_left">CHN1</td>
<td headers="pyramidal-neuron  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="pyramidal-neuron  expression" class="gt_row gt_right">4028.1975</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="stromal-cell">stromal-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="stromal-cell  gene" class="gt_row gt_left">IFITM3</td>
<td headers="stromal-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="stromal-cell  expression" class="gt_row gt_right">2531.6458</td>
</tr>
<tr>
<td headers="stromal-cell  gene" class="gt_row gt_left">SPARC</td>
<td headers="stromal-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="stromal-cell  expression" class="gt_row gt_right">1148.1056</td>
</tr>
<tr>
<td headers="stromal-cell  gene" class="gt_row gt_left">PDGFRB</td>
<td headers="stromal-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="stromal-cell  expression" class="gt_row gt_right">985.2216</td>
</tr>
<tr>
<td headers="stromal-cell  gene" class="gt_row gt_left">IFITM1</td>
<td headers="stromal-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="stromal-cell  expression" class="gt_row gt_right">1166.8611</td>
</tr>
<tr>
<td headers="stromal-cell  gene" class="gt_row gt_left">DCN</td>
<td headers="stromal-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="stromal-cell  expression" class="gt_row gt_right">1043.8413</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="t-cell">t-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="t-cell  gene" class="gt_row gt_left">RPS28</td>
<td headers="t-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="t-cell  expression" class="gt_row gt_right">1914.8449</td>
</tr>
<tr>
<td headers="t-cell  gene" class="gt_row gt_left">RPS8</td>
<td headers="t-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="t-cell  expression" class="gt_row gt_right">1225.6691</td>
</tr>
<tr>
<td headers="t-cell  gene" class="gt_row gt_left">RPL41</td>
<td headers="t-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="t-cell  expression" class="gt_row gt_right">2276.2886</td>
</tr>
<tr>
<td headers="t-cell  gene" class="gt_row gt_left">ENSG00000223367</td>
<td headers="t-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="t-cell  expression" class="gt_row gt_right">422.7055</td>
</tr>
<tr>
<td headers="t-cell  gene" class="gt_row gt_left">RPLP2</td>
<td headers="t-cell  experiment_id" class="gt_row gt_left">E-HCAD-35</td>
<td headers="t-cell  expression" class="gt_row gt_right">1254.5843</td>
</tr>
<tr class="gt_group_heading_row">
<th colspan="3" class="gt_group_heading" scope="colgroup" id="vascular-cell">vascular-cell</th>
    </tr>
<tr class="gt_row_group_first">
<td headers="vascular-cell  gene" class="gt_row gt_left">IGFBP7</td>
<td headers="vascular-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="vascular-cell  expression" class="gt_row gt_right">2707.6829</td>
</tr>
<tr>
<td headers="vascular-cell  gene" class="gt_row gt_left">IFITM1</td>
<td headers="vascular-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="vascular-cell  expression" class="gt_row gt_right">951.4264</td>
</tr>
<tr>
<td headers="vascular-cell  gene" class="gt_row gt_left">IFITM3</td>
<td headers="vascular-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="vascular-cell  expression" class="gt_row gt_right">1115.2355</td>
</tr>
<tr>
<td headers="vascular-cell  gene" class="gt_row gt_left">SPARC</td>
<td headers="vascular-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="vascular-cell  expression" class="gt_row gt_right">3253.8726</td>
</tr>
<tr>
<td headers="vascular-cell  gene" class="gt_row gt_left">GNG11</td>
<td headers="vascular-cell  experiment_id" class="gt_row gt_left">E-GEOD-84465</td>
<td headers="vascular-cell  expression" class="gt_row gt_right">235.9478</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Now we can generate the yaml file we need from this dataframe</p>
<div class="cell">
<details open=""><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Group the dataframe by 'annotation' and aggregate genes as a list</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="va">gene_markers_clean</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">summarise</span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert the result to a nested list using map()</span></span>
<span><span class="va">result_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html">set_names</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">genes</span>, <span class="va">result</span><span class="op">$</span><span class="va">annotation</span><span class="op">)</span>, <span class="va">as.list</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Write the YAML to a file</span></span>
<span><span class="va">yaml_output</span> <span class="op">&lt;-</span> <span class="fu">write_yaml</span><span class="op">(</span></span>
<span>  <span class="va">result_list</span>,</span>
<span>  <span class="fu">here</span><span class="op">(</span></span>
<span>    <span class="st">"03_data/990_processed_data/001_snrnaseq/06_scflow/reddim_genes.yml"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb12" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> scflow prep</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span><span class="co"> </span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">  purl: false</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Getting scflow on hawk</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>So someone in Super Computing Wales added a handy nf-core profile config we can use, and this also includes a nice guide to getting set up for using nf-core pipelines: <span class="co">[</span><span class="ot">https://github.com/nf-core/configs/blob/master/docs/scw.md</span><span class="co">](https://github.com/nf-core/configs/blob/master/docs/scw.md)</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>I've followed the guide for offline use, which I won't rehash here, check the above link if curious.</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="fu"># Input files</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>To run scflow we need a manifest tsv file where the first column is a unique sample ID and the second column is the file path to the directory that has the <span class="in">`barcodes.tsv`</span>, <span class="in">`genes.tsv`</span> and <span class="in">`matrix.mtx`</span> files.</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>I'm assuming we want to use the filtered gene-barcode matrices in MEX format as opposed to the raw files.</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>Note that this page has a mix of bash (run on Hawk) and R code (run locally), I've just run these bits manually and recorded the code here for now.</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="fu">## Manifest file</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>First let's sort the manifest for our samples.</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>Here's how I generated it:</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a><span class="co"># list the cellranger count output directories</span></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>find <span class="sc">/</span>scratch<span class="sc">/</span>scw1329<span class="sc">/</span>gmbh<span class="sc">/</span>blood<span class="sc">-</span>brain<span class="sc">-</span>barrier<span class="sc">-</span><span class="cf">in</span><span class="sc">-</span>ad<span class="sc">/</span>03_data<span class="sc">/</span>990_processed_data<span class="sc">/</span>001_snrnaseq<span class="sc">/</span>04_cellranger_count<span class="sc">/</span> <span class="sc">-</span>maxdepth <span class="dv">2</span> <span class="sc">-</span>mindepth <span class="dv">2</span> <span class="sc">-</span>type d <span class="sc">&gt;</span> scflow_manifest.txt</span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a><span class="co"># use the last directory in the path as the sample id</span></span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>awk <span class="sc">-</span>F<span class="st">'/'</span> <span class="st">'{print $NF "</span><span class="sc">\t</span><span class="st">" $0}'</span> scflow_manifest.txt <span class="sc">&gt;</span> scflow_manifest2.txt</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a><span class="co"># append "outs/filtered_gene_bc_matrices" to file paths</span></span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>sed <span class="st">"s|$|/outs/filtered_feature_bc_matrix|"</span> scflow_manifest2.txt <span class="sc">&gt;</span> scflow_manifest3.txt</span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the undetermined sample and P_39</span></span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>grep <span class="sc">-</span>v <span class="st">"Undetermined"</span> scflow_manifest3.txt <span class="sc">&gt;</span> scflow_manifest4.txt</span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a>grep <span class="sc">-</span>v <span class="st">"P_39"</span> scflow_manifest4.txt <span class="sc">&gt;</span> scflow_manifest5.txt</span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a><span class="co"># add the table headers</span></span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a>awk <span class="sc">-</span>v line<span class="ot">=</span><span class="st">"key</span><span class="sc">\t</span><span class="st">filepath"</span> <span class="st">'BEGIN{print line}{print}'</span> scflow_manifest5.txt <span class="sc">&gt;</span> scflow_manifest.tsv</span>
<span id="cb12-47"><a href="#cb12-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-48"><a href="#cb12-48" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up</span></span>
<span id="cb12-49"><a href="#cb12-49" aria-hidden="true" tabindex="-1"></a>rm scflow_manifest.txt scflow_manifest2.txt scflow_manifest3.txt scflow_manifest4.txt scflow_manifest5.txt</span>
<span id="cb12-50"><a href="#cb12-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-51"><a href="#cb12-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-52"><a href="#cb12-52" aria-hidden="true" tabindex="-1"></a>We would also like to include the brain atlas from Yang et al. 2022.<span class="co">[</span><span class="ot">@yang2022</span><span class="co">]</span></span>
<span id="cb12-53"><a href="#cb12-53" aria-hidden="true" tabindex="-1"></a>Their data is on GEO so I've downloaded and extracted that to gluster.</span>
<span id="cb12-54"><a href="#cb12-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-57"><a href="#cb12-57" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-58"><a href="#cb12-58" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb12-59"><a href="#cb12-59" aria-hidden="true" tabindex="-1"></a><span class="co"># having extracted the RAW tar file and moved them to their own dir, I'll get </span></span>
<span id="cb12-60"><a href="#cb12-60" aria-hidden="true" tabindex="-1"></a><span class="co"># the full path to these directories</span></span>
<span id="cb12-61"><a href="#cb12-61" aria-hidden="true" tabindex="-1"></a>find <span class="sc">$</span>(pwd) <span class="sc">-</span>mindepth <span class="dv">1</span> <span class="sc">-</span>type d <span class="sc">&gt;</span> file_paths.txt</span>
<span id="cb12-62"><a href="#cb12-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-63"><a href="#cb12-63" aria-hidden="true" tabindex="-1"></a><span class="co"># use the last directory in the path as the sample id</span></span>
<span id="cb12-64"><a href="#cb12-64" aria-hidden="true" tabindex="-1"></a>awk <span class="sc">-</span>F<span class="st">'/'</span> <span class="st">'{print $NF "</span><span class="sc">\t</span><span class="st">" $0}'</span> file_paths.txt <span class="sc">&gt;</span> yang_manifest.tsv </span>
<span id="cb12-65"><a href="#cb12-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-66"><a href="#cb12-66" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the ugly string on the sample IDs</span></span>
<span id="cb12-67"><a href="#cb12-67" aria-hidden="true" tabindex="-1"></a>awk <span class="sc">-</span>F<span class="st">'</span><span class="sc">\t</span><span class="st">'</span> <span class="sc">-</span>v OFS<span class="ot">=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span> <span class="st">'{sub("_filtered_feature_bc_matrix", "", $1); print}'</span> yang_manifest.tsv <span class="sc">&gt;</span> yang_manifest2.tsv</span>
<span id="cb12-68"><a href="#cb12-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-69"><a href="#cb12-69" aria-hidden="true" tabindex="-1"></a><span class="co"># add yang sample manifest to main samples</span></span>
<span id="cb12-70"><a href="#cb12-70" aria-hidden="true" tabindex="-1"></a>cat yang_manifest2.tsv <span class="sc">&gt;</span><span class="er">&gt;</span> scflow_manifest.tsv</span>
<span id="cb12-71"><a href="#cb12-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-72"><a href="#cb12-72" aria-hidden="true" tabindex="-1"></a><span class="co"># clean up </span></span>
<span id="cb12-73"><a href="#cb12-73" aria-hidden="true" tabindex="-1"></a>rm yang_manifest<span class="sc">*</span></span>
<span id="cb12-74"><a href="#cb12-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-75"><a href="#cb12-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-76"><a href="#cb12-76" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sample metadata</span></span>
<span id="cb12-77"><a href="#cb12-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-78"><a href="#cb12-78" aria-hidden="true" tabindex="-1"></a>We also need to sort out the sample metadata for the Yang data.</span>
<span id="cb12-79"><a href="#cb12-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-82"><a href="#cb12-82" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-83"><a href="#cb12-83" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load packages</span></span>
<span id="cb12-84"><a href="#cb12-84" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb12-85"><a href="#cb12-85" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb12-86"><a href="#cb12-86" aria-hidden="true" tabindex="-1"></a>pkg <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"readr"</span>, <span class="st">"dplyr"</span>, <span class="st">"here"</span>, <span class="st">"GEOquery"</span>, <span class="st">"magrittr"</span>, <span class="st">"tidyr"</span>, <span class="st">"yaml"</span>, </span>
<span id="cb12-87"><a href="#cb12-87" aria-hidden="true" tabindex="-1"></a>         <span class="st">"gt"</span>)</span>
<span id="cb12-88"><a href="#cb12-88" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(pkg, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-89"><a href="#cb12-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-90"><a href="#cb12-90" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-91"><a href="#cb12-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-94"><a href="#cb12-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-95"><a href="#cb12-95" aria-hidden="true" tabindex="-1"></a><span class="co"># current metadata structure for first 2 sample batches</span></span>
<span id="cb12-96"><a href="#cb12-96" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_meta_merged_final.txt"</span>))</span>
<span id="cb12-97"><a href="#cb12-97" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(metadata)</span>
<span id="cb12-98"><a href="#cb12-98" aria-hidden="true" tabindex="-1"></a><span class="co"># add a column for the technique - don't know much about it so I'll just call</span></span>
<span id="cb12-99"><a href="#cb12-99" aria-hidden="true" tabindex="-1"></a><span class="co"># it Oxford for now</span></span>
<span id="cb12-100"><a href="#cb12-100" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>technique <span class="ot">&lt;-</span> <span class="st">"oxford"</span></span>
<span id="cb12-101"><a href="#cb12-101" aria-hidden="true" tabindex="-1"></a><span class="co"># also don't know brain region</span></span>
<span id="cb12-102"><a href="#cb12-102" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>brain_region <span class="ot">&lt;-</span> <span class="st">"dunno"</span></span>
<span id="cb12-103"><a href="#cb12-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-104"><a href="#cb12-104" aria-hidden="true" tabindex="-1"></a><span class="co"># get yang data</span></span>
<span id="cb12-105"><a href="#cb12-105" aria-hidden="true" tabindex="-1"></a>gsm <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE163577"</span>)</span>
<span id="cb12-106"><a href="#cb12-106" aria-hidden="true" tabindex="-1"></a><span class="co"># get phenotype data</span></span>
<span id="cb12-107"><a href="#cb12-107" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> gsm<span class="sc">$</span>GSE163577_series_matrix.txt.gz<span class="sc">@</span>phenoData<span class="sc">@</span>data</span>
<span id="cb12-108"><a href="#cb12-108" aria-hidden="true" tabindex="-1"></a><span class="co"># select useful cols</span></span>
<span id="cb12-109"><a href="#cb12-109" aria-hidden="true" tabindex="-1"></a>phenodata <span class="ot">&lt;-</span> phenodata <span class="sc">%&gt;%</span></span>
<span id="cb12-110"><a href="#cb12-110" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb12-111"><a href="#cb12-111" aria-hidden="true" tabindex="-1"></a>    title,</span>
<span id="cb12-112"><a href="#cb12-112" aria-hidden="true" tabindex="-1"></a>    geo_accession,</span>
<span id="cb12-113"><a href="#cb12-113" aria-hidden="true" tabindex="-1"></a>    extract_protocol_ch1,</span>
<span id="cb12-114"><a href="#cb12-114" aria-hidden="true" tabindex="-1"></a>    data_processing,</span>
<span id="cb12-115"><a href="#cb12-115" aria-hidden="true" tabindex="-1"></a>    platform_id,</span>
<span id="cb12-116"><a href="#cb12-116" aria-hidden="true" tabindex="-1"></a>    instrument_model,</span>
<span id="cb12-117"><a href="#cb12-117" aria-hidden="true" tabindex="-1"></a>    supplementary_file_1,</span>
<span id="cb12-118"><a href="#cb12-118" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">brain region:ch1</span><span class="st">`</span>,</span>
<span id="cb12-119"><a href="#cb12-119" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">disease:ch1</span><span class="st">`</span></span>
<span id="cb12-120"><a href="#cb12-120" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb12-121"><a href="#cb12-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-122"><a href="#cb12-122" aria-hidden="true" tabindex="-1"></a><span class="co"># load scflow test sample sheet</span></span>
<span id="cb12-123"><a href="#cb12-123" aria-hidden="true" tabindex="-1"></a>example_samplesheet <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/90_sample_info/scflowSampleSheet_example.tsv"</span>))</span>
<span id="cb12-124"><a href="#cb12-124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-125"><a href="#cb12-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-126"><a href="#cb12-126" aria-hidden="true" tabindex="-1"></a>There are nice geo accessions that would be handy to use as unique sample IDs for the yang manifest file</span>
<span id="cb12-127"><a href="#cb12-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-130"><a href="#cb12-130" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-131"><a href="#cb12-131" aria-hidden="true" tabindex="-1"></a><span class="co"># save id and path for manifest file</span></span>
<span id="cb12-132"><a href="#cb12-132" aria-hidden="true" tabindex="-1"></a>yang_manifest <span class="ot">&lt;-</span> phenodata <span class="sc">%&gt;%</span></span>
<span id="cb12-133"><a href="#cb12-133" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(geo_accession, supplementary_file_1)</span>
<span id="cb12-134"><a href="#cb12-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-135"><a href="#cb12-135" aria-hidden="true" tabindex="-1"></a><span class="co"># get the last part of path</span></span>
<span id="cb12-136"><a href="#cb12-136" aria-hidden="true" tabindex="-1"></a>last_path <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*/"</span>, <span class="st">""</span>, yang_manifest<span class="sc">$</span>supplementary_file_1)</span>
<span id="cb12-137"><a href="#cb12-137" aria-hidden="true" tabindex="-1"></a><span class="co"># get after first _</span></span>
<span id="cb12-138"><a href="#cb12-138" aria-hidden="true" tabindex="-1"></a>last_path <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^[^_]*_"</span>, <span class="st">""</span>, last_path)</span>
<span id="cb12-139"><a href="#cb12-139" aria-hidden="true" tabindex="-1"></a><span class="co"># remove file extension</span></span>
<span id="cb12-140"><a href="#cb12-140" aria-hidden="true" tabindex="-1"></a>last_path <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.tar</span><span class="sc">\\</span><span class="st">.gz$"</span>, <span class="st">""</span>, last_path)</span>
<span id="cb12-141"><a href="#cb12-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-142"><a href="#cb12-142" aria-hidden="true" tabindex="-1"></a><span class="co"># update path to yang data on gluster</span></span>
<span id="cb12-143"><a href="#cb12-143" aria-hidden="true" tabindex="-1"></a>yang_manifest<span class="sc">$</span>supplementary_file_1 <span class="ot">&lt;-</span></span>
<span id="cb12-144"><a href="#cb12-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste0</span>(</span>
<span id="cb12-145"><a href="#cb12-145" aria-hidden="true" tabindex="-1"></a>    <span class="st">"/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/06_scflow/yang_2022_data/"</span>,</span>
<span id="cb12-146"><a href="#cb12-146" aria-hidden="true" tabindex="-1"></a>    last_path</span>
<span id="cb12-147"><a href="#cb12-147" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb12-148"><a href="#cb12-148" aria-hidden="true" tabindex="-1"></a><span class="co"># update names to be in line with scflow manifest file</span></span>
<span id="cb12-149"><a href="#cb12-149" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(yang_manifest) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"key"</span>, <span class="st">"filepath"</span>)</span>
<span id="cb12-150"><a href="#cb12-150" aria-hidden="true" tabindex="-1"></a><span class="co"># save to tsv</span></span>
<span id="cb12-151"><a href="#cb12-151" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(yang_manifest, <span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/06_scflow/yang_manifest.tsv"</span>))</span>
<span id="cb12-152"><a href="#cb12-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-153"><a href="#cb12-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-154"><a href="#cb12-154" aria-hidden="true" tabindex="-1"></a>Then we just need to merge the two manifest files together.</span>
<span id="cb12-155"><a href="#cb12-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-158"><a href="#cb12-158" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-159"><a href="#cb12-159" aria-hidden="true" tabindex="-1"></a>(cat scflow_manifest.tsv <span class="sc">&amp;&amp;</span> tail <span class="sc">-</span>n <span class="sc">+</span><span class="dv">2</span> yang_manifest.tsv) <span class="sc">&gt;</span> scflow_manifest_both.tsv</span>
<span id="cb12-160"><a href="#cb12-160" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-161"><a href="#cb12-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-162"><a href="#cb12-162" aria-hidden="true" tabindex="-1"></a>It turns out that you can't have delimiters in the key for the mainfest/sample files, so I need to remove the underscores too</span>
<span id="cb12-163"><a href="#cb12-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-166"><a href="#cb12-166" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-167"><a href="#cb12-167" aria-hidden="true" tabindex="-1"></a>awk <span class="st">'BEGIN{FS=OFS="</span><span class="sc">\t</span><span class="st">"} {gsub("_", "", $1)} 1'</span>  scflow_manifest_both.tsv <span class="sc">&gt;</span> scflow_manifest_final.tsv</span>
<span id="cb12-168"><a href="#cb12-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-169"><a href="#cb12-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-170"><a href="#cb12-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-171"><a href="#cb12-171" aria-hidden="true" tabindex="-1"></a>We need to get the phenotype data from the yang GEO data in line with our phenotype data to join them.</span>
<span id="cb12-172"><a href="#cb12-172" aria-hidden="true" tabindex="-1"></a>My understanding from the Yang paper is that they only looked at the vascular/perivascular component, so I'll label the <span class="in">`prep`</span> as <span class="in">`V`</span> for now.</span>
<span id="cb12-173"><a href="#cb12-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-176"><a href="#cb12-176" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-177"><a href="#cb12-177" aria-hidden="true" tabindex="-1"></a><span class="co"># sort yang phenotype data and join to ours</span></span>
<span id="cb12-178"><a href="#cb12-178" aria-hidden="true" tabindex="-1"></a>merged_meta <span class="ot">&lt;-</span> phenodata <span class="sc">%&gt;%</span></span>
<span id="cb12-179"><a href="#cb12-179" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sample =</span> geo_accession,</span>
<span id="cb12-180"><a href="#cb12-180" aria-hidden="true" tabindex="-1"></a>                <span class="at">brain_region =</span> <span class="st">`</span><span class="at">brain region:ch1</span><span class="st">`</span>,</span>
<span id="cb12-181"><a href="#cb12-181" aria-hidden="true" tabindex="-1"></a>                <span class="at">type =</span> <span class="st">`</span><span class="at">disease:ch1</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-182"><a href="#cb12-182" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add cell subsetting technique and cell population</span></span>
<span id="cb12-183"><a href="#cb12-183" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">technique =</span> <span class="st">"VINE-seq"</span>, <span class="at">prep =</span> <span class="st">"V"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-184"><a href="#cb12-184" aria-hidden="true" tabindex="-1"></a>  <span class="co"># select cols of interest</span></span>
<span id="cb12-185"><a href="#cb12-185" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, brain_region, type, technique, prep) <span class="sc">%&gt;%</span></span>
<span id="cb12-186"><a href="#cb12-186" aria-hidden="true" tabindex="-1"></a>  <span class="co"># merge datasets</span></span>
<span id="cb12-187"><a href="#cb12-187" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(metadata, .) <span class="sc">%&gt;%</span></span>
<span id="cb12-188"><a href="#cb12-188" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make control label consistent</span></span>
<span id="cb12-189"><a href="#cb12-189" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">ifelse</span>(type <span class="sc">==</span> <span class="st">"CT"</span>, <span class="st">"Control"</span>, type)) <span class="sc">%&gt;%</span></span>
<span id="cb12-190"><a href="#cb12-190" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add final group column for differential expression</span></span>
<span id="cb12-191"><a href="#cb12-191" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">paste</span>(type, prep, technique, brain_region, <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb12-192"><a href="#cb12-192" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample =</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb12-193"><a href="#cb12-193" aria-hidden="true" tabindex="-1"></a>  <span class="co"># rename to sample sheet style</span></span>
<span id="cb12-194"><a href="#cb12-194" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">manifest =</span> sample,</span>
<span id="cb12-195"><a href="#cb12-195" aria-hidden="true" tabindex="-1"></a>                <span class="at">diagnosis =</span> type)</span>
<span id="cb12-196"><a href="#cb12-196" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-197"><a href="#cb12-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-200"><a href="#cb12-200" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-201"><a href="#cb12-201" aria-hidden="true" tabindex="-1"></a><span class="co"># save metadata</span></span>
<span id="cb12-202"><a href="#cb12-202" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(merged_meta, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/90_sample_info/sample_metadata.tsv"</span>))</span>
<span id="cb12-203"><a href="#cb12-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-204"><a href="#cb12-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-205"><a href="#cb12-205" aria-hidden="true" tabindex="-1"></a><span class="fu">## Getting marker genes</span></span>
<span id="cb12-206"><a href="#cb12-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-207"><a href="#cb12-207" aria-hidden="true" tabindex="-1"></a>The EBI has the <span class="co">[</span><span class="ot">Single cell expression atlas</span><span class="co">](https://www.ebi.ac.uk/gxa/sc/home)</span> so I'll use that for our gene annotation markers.</span>
<span id="cb12-208"><a href="#cb12-208" aria-hidden="true" tabindex="-1"></a>There is an R package on Bioconductor for querying it, but it doesn't seem to have an easy way of querying by tissues/organ, so I'll just manually download them (there aren't that many).</span>
<span id="cb12-209"><a href="#cb12-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-212"><a href="#cb12-212" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-213"><a href="#cb12-213" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb12-214"><a href="#cb12-214" aria-hidden="true" tabindex="-1"></a><span class="co"># list gene marker csvs</span></span>
<span id="cb12-215"><a href="#cb12-215" aria-hidden="true" tabindex="-1"></a>gene_markers_files <span class="ot">&lt;-</span></span>
<span id="cb12-216"><a href="#cb12-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(</span>
<span id="cb12-217"><a href="#cb12-217" aria-hidden="true" tabindex="-1"></a>    <span class="fu">here</span>(</span>
<span id="cb12-218"><a href="#cb12-218" aria-hidden="true" tabindex="-1"></a>      <span class="st">"03_data/990_processed_data/001_snrnaseq/06_scflow/2023-05-22_gene-markers"</span></span>
<span id="cb12-219"><a href="#cb12-219" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb12-220"><a href="#cb12-220" aria-hidden="true" tabindex="-1"></a>    <span class="at">full.names =</span> <span class="cn">TRUE</span></span>
<span id="cb12-221"><a href="#cb12-221" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb12-222"><a href="#cb12-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-223"><a href="#cb12-223" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the final part of the path after the last "/"</span></span>
<span id="cb12-224"><a href="#cb12-224" aria-hidden="true" tabindex="-1"></a>file_names <span class="ot">&lt;-</span> <span class="fu">basename</span>(gene_markers_files)</span>
<span id="cb12-225"><a href="#cb12-225" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the file extension</span></span>
<span id="cb12-226"><a href="#cb12-226" aria-hidden="true" tabindex="-1"></a>file_names <span class="ot">&lt;-</span> tools<span class="sc">::</span><span class="fu">file_path_sans_ext</span>(file_names)</span>
<span id="cb12-227"><a href="#cb12-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-228"><a href="#cb12-228" aria-hidden="true" tabindex="-1"></a>gene_markers <span class="ot">&lt;-</span> <span class="fu">map</span>(gene_markers_files, read_csv) <span class="sc">%&gt;%</span></span>
<span id="cb12-229"><a href="#cb12-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(file_names)</span>
<span id="cb12-230"><a href="#cb12-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-231"><a href="#cb12-231" aria-hidden="true" tabindex="-1"></a><span class="co"># add annotation name</span></span>
<span id="cb12-232"><a href="#cb12-232" aria-hidden="true" tabindex="-1"></a>gene_markers <span class="ot">&lt;-</span></span>
<span id="cb12-233"><a href="#cb12-233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(gene_markers, file_names, <span class="sc">~</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(.x, <span class="at">annotation =</span> .y)) <span class="sc">%&gt;%</span></span>
<span id="cb12-234"><a href="#cb12-234" aria-hidden="true" tabindex="-1"></a>  <span class="co"># bind rows</span></span>
<span id="cb12-235"><a href="#cb12-235" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb12-236"><a href="#cb12-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-237"><a href="#cb12-237" aria-hidden="true" tabindex="-1"></a><span class="co"># the odd column headers are the experiment ID and the values are expression</span></span>
<span id="cb12-238"><a href="#cb12-238" aria-hidden="true" tabindex="-1"></a><span class="co"># pivot long</span></span>
<span id="cb12-239"><a href="#cb12-239" aria-hidden="true" tabindex="-1"></a>gene_markers <span class="ot">&lt;-</span> gene_markers <span class="sc">%&gt;%</span></span>
<span id="cb12-240"><a href="#cb12-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="fu">c</span>(Category, annotation),</span>
<span id="cb12-241"><a href="#cb12-241" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"experiment_id"</span>,</span>
<span id="cb12-242"><a href="#cb12-242" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"expression"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-243"><a href="#cb12-243" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove cases where expression is 0</span></span>
<span id="cb12-244"><a href="#cb12-244" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(expression)) <span class="sc">%&gt;%</span></span>
<span id="cb12-245"><a href="#cb12-245" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> Category)</span>
<span id="cb12-246"><a href="#cb12-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-247"><a href="#cb12-247" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out duplicate genes within each annotation</span></span>
<span id="cb12-248"><a href="#cb12-248" aria-hidden="true" tabindex="-1"></a>gene_markers_clean <span class="ot">&lt;-</span> gene_markers <span class="sc">%&gt;%</span></span>
<span id="cb12-249"><a href="#cb12-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(annotation) <span class="sc">%&gt;%</span></span>
<span id="cb12-250"><a href="#cb12-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(gene)) <span class="sc">%&gt;%</span></span>
<span id="cb12-251"><a href="#cb12-251" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove the "top-genes" part of the string</span></span>
<span id="cb12-252"><a href="#cb12-252" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">annotation =</span> <span class="fu">gsub</span>(<span class="st">"top-.*"</span>, <span class="st">""</span>, annotation))</span>
<span id="cb12-253"><a href="#cb12-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-254"><a href="#cb12-254" aria-hidden="true" tabindex="-1"></a><span class="fu">gt</span>(gene_markers_clean)</span>
<span id="cb12-255"><a href="#cb12-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb12-256"><a href="#cb12-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-257"><a href="#cb12-257" aria-hidden="true" tabindex="-1"></a>Now we can generate the yaml file we need from this dataframe</span>
<span id="cb12-258"><a href="#cb12-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-261"><a href="#cb12-261" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb12-262"><a href="#cb12-262" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb12-263"><a href="#cb12-263" aria-hidden="true" tabindex="-1"></a><span class="co"># Group the dataframe by 'annotation' and aggregate genes as a list</span></span>
<span id="cb12-264"><a href="#cb12-264" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> gene_markers_clean <span class="sc">%&gt;%</span></span>
<span id="cb12-265"><a href="#cb12-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(annotation) <span class="sc">%&gt;%</span></span>
<span id="cb12-266"><a href="#cb12-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">genes =</span> <span class="fu">list</span>(<span class="fu">unique</span>(gene))) <span class="sc">%&gt;%</span></span>
<span id="cb12-267"><a href="#cb12-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb12-268"><a href="#cb12-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-269"><a href="#cb12-269" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the result to a nested list using map()</span></span>
<span id="cb12-270"><a href="#cb12-270" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">set_names</span>(result<span class="sc">$</span>genes, result<span class="sc">$</span>annotation), as.list)</span>
<span id="cb12-271"><a href="#cb12-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-272"><a href="#cb12-272" aria-hidden="true" tabindex="-1"></a><span class="co"># Write the YAML to a file</span></span>
<span id="cb12-273"><a href="#cb12-273" aria-hidden="true" tabindex="-1"></a>yaml_output <span class="ot">&lt;-</span> <span class="fu">write_yaml</span>(</span>
<span id="cb12-274"><a href="#cb12-274" aria-hidden="true" tabindex="-1"></a>  result_list,</span>
<span id="cb12-275"><a href="#cb12-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">here</span>(</span>
<span id="cb12-276"><a href="#cb12-276" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/06_scflow/reddim_genes.yml"</span></span>
<span id="cb12-277"><a href="#cb12-277" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb12-278"><a href="#cb12-278" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-279"><a href="#cb12-279" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>