<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Blood brain barrier in AD - Perform Pseudobulk</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="styles.css">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Blood brain barrier in AD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../06_dissemination/03_website/01_data-analysis.html" rel="" target="" aria-current="page">
 <span class="menu-text">Data analysis</span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools ms-auto">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html">Perform Pseudobulk</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../06_dissemination/03_website/01_data-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis Pipeline</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/01_fastqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run FastQC on sample</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/02_multiqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run MultiQC</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/03_generate_cellranger_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/04_cellranger_count_set1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/05_cellranger_count_set2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 2</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/06_cellranger_count_set3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 3</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/07_cellranger_aggr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Aggregate cellranger count reports</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/08_scflow_prep.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scflow prep</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/09_scflow_run.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run SCflow</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/10_scflow_results.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Analysis scFlow results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Perform Pseudobulk</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform pathway analysis</span></a>
  </div>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages"><span class="header-section-number">1</span> Load packages</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data"><span class="header-section-number">2</span> Read in data</a></li>
  <li><a href="#average-expression" id="toc-average-expression" class="nav-link" data-scroll-target="#average-expression"><span class="header-section-number">3</span> Average expression</a></li>
  <li><a href="#pseudobulk" id="toc-pseudobulk" class="nav-link" data-scroll-target="#pseudobulk"><span class="header-section-number">4</span> Pseudobulk</a></li>
  <li><a href="#code-from-alan" id="toc-code-from-alan" class="nav-link" data-scroll-target="#code-from-alan"><span class="header-section-number">5</span> Code from Alan</a></li>
  <li>
<a href="#write-from-scratch" id="toc-write-from-scratch" class="nav-link" data-scroll-target="#write-from-scratch"><span class="header-section-number">6</span> Write from scratch</a>
  <ul class="collapse">
<li>
<a href="#plots" id="toc-plots" class="nav-link" data-scroll-target="#plots"><span class="header-section-number">6.1</span> Plots</a>
  <ul class="collapse">
<li><a href="#counts-of-degs" id="toc-counts-of-degs" class="nav-link" data-scroll-target="#counts-of-degs"><span class="header-section-number">6.1.1</span> Counts of DEGs</a></li>
  <li><a href="#volcano-plots" id="toc-volcano-plots" class="nav-link" data-scroll-target="#volcano-plots"><span class="header-section-number">6.1.2</span> Volcano plots</a></li>
  </ul>
</li>
  </ul>
</li>
  <li>
<a href="#deseq-2" id="toc-deseq-2" class="nav-link" data-scroll-target="#deseq-2"><span class="header-section-number">7</span> DESeq 2</a>
  <ul class="collapse">
<li>
<a href="#plots-1" id="toc-plots-1" class="nav-link" data-scroll-target="#plots-1"><span class="header-section-number">7.1</span> Plots</a>
  <ul class="collapse">
<li><a href="#volcano-plots-1" id="toc-volcano-plots-1" class="nav-link" data-scroll-target="#volcano-plots-1"><span class="header-section-number">7.1.1</span> Volcano plots</a></li>
  </ul>
</li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Perform Pseudobulk</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><p>Note that I’ll use the higher level annotations to start with. This only involved aggregating the venous/capillary/arterial subtypes to a larger “Endothelial” category.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/i_am.html">i_am</a></span><span class="op">(</span><span class="st">"04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.qmd"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-packages" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Load packages</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"04_data_analysis/990_code_libraries/pseudobulk"</span><span class="op">)</span></span>
<span><span class="co">#source necessary functions from other scripts</span></span>
<span><span class="co">#source(here::here(path, "validate_input_parameters_de.R"))</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"make_pseudobulk.R"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"de_analysis.R"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"plot_de_analysis.R"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="read-in-data" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Read in data</h1>
<p>First we’ll read in the scFlow output to a sce</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-updated_dim-35.rds"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># make count rownames match data rownames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">doner</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"doner-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="average-expression" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Average expression</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="fu">AverageExpression</span><span class="op">(</span><span class="va">sce</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ident"</span>, <span class="st">"doner"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="va">avg_exp</span><span class="op">$</span><span class="va">RNA</span></span>
<span> </span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="op">(</span><span class="va">avg_exp</span>, keep.rownames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">setnames</span><span class="op">(</span><span class="va">avg_exp</span>, <span class="st">"rn"</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># pivot longer such that celltypes are separated out</span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="va">avg_exp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">!</span><span class="va">gene</span>,</span>
<span>    names_to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"celltype"</span>, <span class="st">".value"</span><span class="op">)</span>,</span>
<span>    names_pattern <span class="op">=</span> <span class="st">"(.*)_(.*)"</span></span>
<span>  <span class="op">)</span></span>
<span> </span>
<span><span class="va">file_out</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-celltype-per-doner.csv"</span><span class="op">)</span></span>
<span><span class="co">#file_out &lt;- here::here("03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-celltype-per-doner_35-dims.csv")</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_csv</a></span><span class="op">(</span><span class="va">avg_exp</span>, <span class="va">file_out</span><span class="op">)</span></span>
<span> </span>
<span><span class="co">## split AD and CT samples</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># avg_exp &lt;- AverageExpression(sdata_endo, group.by = c("ident", "type"))</span></span>
<span><span class="co"># avg_exp &lt;- avg_exp$RNA</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># avg_exp &lt;- data.table(avg_exp, keep.rownames = T)</span></span>
<span><span class="co"># setnames(avg_exp, "rn", "gene")</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># file_out &lt;- "/scratch/c.mpmfw/Endo_10X_main/Analysis/Expression_matrix/avg_expr_main_cell_type_40_samples_AD_vs_CT_v3.txt"</span></span>
<span><span class="co"># write.table(avg_exp, file = file_out, sep = "\t", quote = F, row.names = F, col.names = T)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Make column with cluster and annotation combined</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">cluster_annoated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"cluster_"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">seurat_clusters</span>, <span class="st">"_"</span>, <span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="fu">AverageExpression</span><span class="op">(</span><span class="va">sce</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cluster_annoated"</span>, <span class="st">"doner"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="va">avg_exp</span><span class="op">$</span><span class="va">RNA</span></span>
<span> </span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="op">(</span><span class="va">avg_exp</span>, keep.rownames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">setnames</span><span class="op">(</span><span class="va">avg_exp</span>, <span class="st">"rn"</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># pivot longer such that celltypes are separated out</span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="va">avg_exp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">!</span><span class="va">gene</span>,</span>
<span>    names_to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"celltype"</span>, <span class="st">".value"</span><span class="op">)</span>,</span>
<span>    names_pattern <span class="op">=</span> <span class="st">"(.*)_(.*)"</span></span>
<span>  <span class="op">)</span></span>
<span> </span>
<span><span class="va">file_out</span> <span class="op">&lt;-</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-cluster-per-doner.csv"</span><span class="op">)</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_csv</a></span><span class="op">(</span><span class="va">avg_exp</span>, <span class="va">file_out</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="pseudobulk" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Pseudobulk</h1>
<p>Now we can subset to variables we want to aggregate across for the pseudobulk</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># aggregate read counts </span></span>
<span><span class="va">aggr_counts</span> <span class="op">&lt;-</span> <span class="fu">Seurat2PB</span><span class="op">(</span><span class="va">sce</span>, sample<span class="op">=</span><span class="st">"group"</span>, cluster<span class="op">=</span><span class="st">"cluster_celltype"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">aggr_counts</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] 27539    36</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">aggr_counts</span><span class="op">$</span><span class="va">samples</span>, n<span class="op">=</span><span class="fl">10L</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>                                                     group  lib.size
AD_P_oxford_dunno_clusterAstrocytes                      1 160707417
AD_P_oxford_dunno_clusterEN-L3-4                         1 176066153
AD_P_oxford_dunno_clusterEndothelial                     1   8064407
AD_P_oxford_dunno_clusterIN-LAMP5                        1  65623124
AD_P_oxford_dunno_clusterMicroglia                       1  28793915
AD_P_oxford_dunno_clusterOligodendrocytes                1 366359329
AD_P_oxford_dunno_clusterOPCs                            1  62663862
AD_P_oxford_dunno_clusterPericytes                       1   4511423
AD_P_oxford_dunno_clusterVascular-and-leptomeningeal     1   3158224
AD_V_oxford_dunno_clusterAstrocytes                      1  60334951
                                                     norm.factors
AD_P_oxford_dunno_clusterAstrocytes                             1
AD_P_oxford_dunno_clusterEN-L3-4                                1
AD_P_oxford_dunno_clusterEndothelial                            1
AD_P_oxford_dunno_clusterIN-LAMP5                               1
AD_P_oxford_dunno_clusterMicroglia                              1
AD_P_oxford_dunno_clusterOligodendrocytes                       1
AD_P_oxford_dunno_clusterOPCs                                   1
AD_P_oxford_dunno_clusterPericytes                              1
AD_P_oxford_dunno_clusterVascular-and-leptomeningeal            1
AD_V_oxford_dunno_clusterAstrocytes                             1
                                                                sample
AD_P_oxford_dunno_clusterAstrocytes                  AD_P_oxford_dunno
AD_P_oxford_dunno_clusterEN-L3-4                     AD_P_oxford_dunno
AD_P_oxford_dunno_clusterEndothelial                 AD_P_oxford_dunno
AD_P_oxford_dunno_clusterIN-LAMP5                    AD_P_oxford_dunno
AD_P_oxford_dunno_clusterMicroglia                   AD_P_oxford_dunno
AD_P_oxford_dunno_clusterOligodendrocytes            AD_P_oxford_dunno
AD_P_oxford_dunno_clusterOPCs                        AD_P_oxford_dunno
AD_P_oxford_dunno_clusterPericytes                   AD_P_oxford_dunno
AD_P_oxford_dunno_clusterVascular-and-leptomeningeal AD_P_oxford_dunno
AD_V_oxford_dunno_clusterAstrocytes                  AD_V_oxford_dunno
                                                                         cluster
AD_P_oxford_dunno_clusterAstrocytes                                   Astrocytes
AD_P_oxford_dunno_clusterEN-L3-4                                         EN-L3-4
AD_P_oxford_dunno_clusterEndothelial                                 Endothelial
AD_P_oxford_dunno_clusterIN-LAMP5                                       IN-LAMP5
AD_P_oxford_dunno_clusterMicroglia                                     Microglia
AD_P_oxford_dunno_clusterOligodendrocytes                       Oligodendrocytes
AD_P_oxford_dunno_clusterOPCs                                               OPCs
AD_P_oxford_dunno_clusterPericytes                                     Pericytes
AD_P_oxford_dunno_clusterVascular-and-leptomeningeal Vascular-and-leptomeningeal
AD_V_oxford_dunno_clusterAstrocytes                                   Astrocytes</code></pre>
</div>
</div>
<p>Convert back to single cell experiment</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/as.SingleCellExperiment.html">as.SingleCellExperiment</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="code-from-alan" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Code from Alan</h1>
<p>There’s an interest in incorporating pseudobuilk into scflow itself, and a member of Nathan’s lab (Alan) has written some code for performing pseudobulk on scflow outputs.</p>
<p>Here I’ll play with his code and start thinking about how this could be incorporated into scflow</p>
<!-- ## A more manual approach -->
<!-- Try extracting just the parts of `sc_cell_type_de.R` -->
<!-- Get cell type counts -->
<!-- One can set the comparison here via the `control` argument. -->
<!-- Changed palette arg in code to make this work -->
<!-- Save pseudobulk results -->
<!-- # try 2 -->
<!-- Try using `region` arg in `make_pseudobulk` -->
<!-- Changed palette arg in code to make this work -->
<!-- Save pseudobulk results -->
</section><section id="write-from-scratch" class="level1" data-number="6"><h1 data-number="6">
<span class="header-section-number">6</span> Write from scratch</h1>
<p>Here I’ll try writing my own version of <code>de_analysis()</code></p>
<p>I’ll base it off the <code>edgeR</code> doc <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">here</a></p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce</span><span class="op">$</span><span class="va">donor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"donor-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">diagnosis</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AD"</span>, <span class="st">"Control"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># inputs</span></span>
<span><span class="va">SCE</span> <span class="op">&lt;-</span> <span class="va">sce</span></span>
<span><span class="va">celltype_ID</span> <span class="op">&lt;-</span> <span class="st">"ident"</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">diagnosis</span></span>
<span><span class="va">pseudobulk_ID</span> <span class="op">=</span> <span class="st">"manifest"</span></span>
<span><span class="va">y</span> <span class="op">=</span> <span class="st">"diagnosis"</span></span>
<span><span class="va">region</span> <span class="op">=</span> <span class="st">"prep"</span></span>
<span><span class="co">#pb_columns &lt;- c("prep", "diagnosis", "manifest")</span></span>
<span><span class="va">pb_columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="st">"manifest"</span><span class="op">)</span></span>
<span><span class="va">verbose</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">control</span> <span class="op">=</span> <span class="st">"Control"</span></span>
<span><span class="va">pval_adjust_method</span> <span class="op">=</span> <span class="st">"BH"</span></span>
<span><span class="va">adj_pval</span><span class="op">=</span><span class="fl">0.05</span></span>
<span><span class="va">coef</span> <span class="op">=</span> <span class="st">"Case"</span></span>
<span><span class="va">folder</span> <span class="op">=</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/graphs/"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#first format formula</span></span>
<span><span class="va">design_txt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/deparse.html">deparse</a></span><span class="op">(</span><span class="va">design</span>,width.cutoff <span class="op">=</span> <span class="fl">500</span><span class="op">)</span>,collapse<span class="op">=</span><span class="st">','</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#make design formula minus anything before ~</span></span>
<span><span class="va">formula</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/formula.html">as.formula</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">".*~"</span>,<span class="st">"~"</span>,<span class="va">design_txt</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#if y not specified take last value in design matrix</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html">is.null</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">y</span> <span class="op">&lt;-</span> <span class="va">design_txt</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">design_txt</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co">#Check if continuous or categorical variable to be modeled</span></span>
<span><span class="va">y_contin</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">is.numeric</a></span><span class="op">(</span><span class="va">SCE</span><span class="op">[[</span><span class="va">y</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">y_contin</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Note that the pseudobulk compute takes a few minutes to run, so I’ll disable eval on compilation.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#get counts of each cell type</span></span>
<span><span class="co">#counts(SCE) &lt;- as.matrix(counts(SCE))</span></span>
<span><span class="va">counts_celltypes</span> <span class="op">&lt;-</span> <span class="va">SCE</span><span class="op">[[</span><span class="va">celltype_ID</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">counts_celltypes</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/vector.html">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">counts_celltypes</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">counts_celltypes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">SCE</span><span class="op">[[</span><span class="va">celltype_ID</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get pseudobulk values</span></span>
<span><span class="va">celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">SCE</span><span class="op">[[</span><span class="va">celltype_ID</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pb_dat</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">celltypes</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>        <span class="fu">make_pseudobulk</span><span class="op">(</span><span class="va">SCE</span><span class="op">[</span>,<span class="va">SCE</span><span class="op">[[</span><span class="va">celltype_ID</span><span class="op">]</span><span class="op">]</span><span class="op">==</span><span class="va">x</span><span class="op">]</span>,</span>
<span>                        pseudobulk_ID<span class="op">=</span><span class="va">pseudobulk_ID</span>,</span>
<span>                        pb_columns<span class="op">=</span><span class="va">pb_columns</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pb_dat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">celltypes</span></span>
<span></span>
<span><span class="co"># the pseudobulk values take a while to compute, so I'll save the output</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">pb_dat</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_values.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># load values</span></span>
<span><span class="va">pb_dat</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_values.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Hannah is after the pseudobulk values themselves, so I’ll add the celltypes and save that data</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">pb_dat_per_individual</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">celltypes</span>,<span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>        <span class="fu">make_pseudobulk</span><span class="op">(</span><span class="va">SCE</span><span class="op">[</span>,<span class="va">SCE</span><span class="op">[[</span><span class="va">celltype_ID</span><span class="op">]</span><span class="op">]</span><span class="op">==</span><span class="va">x</span><span class="op">]</span>,</span>
<span>                        pseudobulk_ID<span class="op">=</span><span class="st">"individual"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pb_dat_per_individual</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">celltypes</span></span>
<span></span>
<span><span class="va">pseudo_vals</span> <span class="op">&lt;-</span> <span class="fu">map2</span><span class="op">(</span><span class="va">pb_dat_per_individual</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pb_dat_per_individual</span><span class="op">)</span>, <span class="op">~</span> <span class="va">.x</span><span class="op">$</span><span class="va">sumDat</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>celltype <span class="op">=</span> <span class="va">.y</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/list_c.html">list_rbind</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_csv</a></span><span class="op">(</span><span class="va">pseudo_vals</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_values_per_individual.csv"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>First part is prep for LRT, this is done per cell type via <code>purrr</code></p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get DGE list per celltype</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">pb_dat</span>, <span class="op">~</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">sumDat</span>,</span>
<span>                    group <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">annot_pb</span><span class="op">$</span><span class="va">group</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate normalization factors to scale the raw library sizes.</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">y</span>, <span class="op">~</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">.x</span>, method <span class="op">=</span> <span class="st">'TMM'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build design matrix</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">pb_dat</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">group</span>, data <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">annot_pb</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Maximizes the negative binomial likelihood to give the estimate of the</span></span>
<span><span class="co"># common,trended and tagwise dispersions across all tags.</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">design</span>, <span class="op">~</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html">estimateDisp</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># LRT</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">design</span>, <span class="op">~</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html">glmFit</a></span><span class="op">(</span><span class="va">.x</span>, design <span class="op">=</span> <span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now there’s a need to think about what comparisons to run.</p>
<p>For now I think I’ll write a more complex approach to run all possible comparisons from a arbitrarily long list</p>
<p>First I’ll make a contrast matrix of all combinations</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get number group levels</span></span>
<span><span class="va">ncls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nlevels.html">nlevels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">group</span><span class="op">)</span></span>
<span></span>
<span><span class="va">get_contrasts</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">design_matrix</span>, <span class="va">group_vector</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">num_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nlevels.html">nlevels</a></span><span class="op">(</span><span class="va">group_vector</span><span class="op">)</span></span>
<span>  <span class="va">contr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">/</span> <span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">num_groups</span><span class="op">)</span>, <span class="va">num_groups</span>, <span class="va">num_groups</span><span class="op">)</span>,</span>
<span>                 <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">design_matrix</span><span class="op">)</span> <span class="op">-</span> <span class="va">num_groups</span>, <span class="va">num_groups</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/diag.html">diag</a></span><span class="op">(</span><span class="va">contr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span>  <span class="va">contr</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">contr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">design_matrix</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">contr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"cluster"</span>, <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">group_vector</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">contr</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">contr</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">design</span>, <span class="op">~</span> <span class="fu">get_contrasts</span><span class="op">(</span><span class="va">.x</span>, <span class="va">sce</span><span class="op">$</span><span class="va">group</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Then apply the LRT model per celltype</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">apply_lrt_to_groups</span> <span class="op">&lt;-</span></span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">glmfit_object</span>,</span>
<span>           <span class="va">contrast_matrix</span>,</span>
<span>           <span class="va">group_levels</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">lrt</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">contrast_matrix</span><span class="op">)</span>, <span class="op">~</span> <span class="op">{</span></span>
<span></span>
<span>        <span class="co"># Apply LRT per comparison</span></span>
<span>        <span class="va">lrt</span> <span class="op">&lt;-</span></span>
<span>          <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html">glmLRT</a></span><span class="op">(</span><span class="va">glmfit_object</span>, contrast <span class="op">=</span> <span class="va">contrast_matrix</span><span class="op">[</span>, <span class="va">.x</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>        <span class="co"># Set name of comparison</span></span>
<span>        <span class="va">lrt</span><span class="op">$</span><span class="va">comparison</span> <span class="op">&lt;-</span></span>
<span>          <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="va">group_levels</span><span class="op">[</span><span class="va">.x</span><span class="op">]</span>, <span class="st">"_vs_others"</span><span class="op">)</span></span>
<span>        <span class="va">lrt</span></span>
<span>      <span class="op">}</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Set names of comparison to list</span></span>
<span>    <span class="va">lrt</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html">set_names</a></span><span class="op">(</span><span class="va">lrt</span>, <span class="fu">map_chr</span><span class="op">(</span><span class="va">group_levels</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"group"</span>, <span class="va">.x</span>, <span class="st">"_vs_others"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">lrt</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="va">lrt</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contr</span>, <span class="op">~</span> <span class="fu">apply_lrt_to_groups</span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span>, <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">group</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Check results</span></span>
<span><span class="fu">topTags</span><span class="op">(</span><span class="va">lrt</span><span class="op">$</span><span class="va">Endothelial</span><span class="op">$</span><span class="va">groupAD_P_oxford_dunno_vs_others</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Coefficient:  groupAD_P_oxford_dunno_vs_others 
           logFC   logCPM       LR       PValue          FDR
PLP1    4.950544 7.132600 196.0722 1.503203e-44 3.836024e-40
FTH1    2.294012 7.170833 189.1651 4.836853e-43 6.171582e-39
SPP1    5.321196 3.926076 176.8246 2.392111e-40 2.034809e-36
ATP5F1E 2.412922 5.936702 164.1003 1.438309e-37 9.176055e-34
RPS21   2.339190 5.601526 154.0947 2.208126e-35 1.126984e-31
RPLP1   2.444752 7.318473 153.4939 2.987658e-35 1.270701e-31
TMSB4X  2.362848 7.448768 150.3683 1.440318e-34 5.250784e-31
OLIG1   5.257273 1.694539 149.0982 2.729416e-34 8.706495e-31
UBA52   2.293557 5.708163 147.0689 7.579998e-34 2.149266e-30
RPS15   2.402022 6.423740 142.4076 7.920476e-33 1.978390e-29</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get number of DE genes</span></span>
<span><span class="va">dt</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">lrt</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">.x</span>, <span class="va">decideTestsDGE</span><span class="op">)</span>, <span class="va">summary</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                   <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="st">"cbind"</span>, <span class="va">.</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">dt</span><span class="op">$</span><span class="va">Endothelial</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>       groupAD_P_oxford_dunno_vs_others groupAD_V_oxford_dunno_vs_others
Down                               2236                             2068
NotSig                            20462                            20712
Up                                 2821                             2739
       groupControl_P_oxford_dunno_vs_others
Down                                    1413
NotSig                                 21288
Up                                      2818
       groupControl_V_oxford_dunno_vs_others
Down                                    3048
NotSig                                 19937
Up                                      2534</code></pre>
</div>
</div>
<p>Clean and extract results from lrt, including getting adjusted pvals</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">clean_lrt_results</span> <span class="op">&lt;-</span></span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">lrt_results</span>,</span>
<span>           <span class="va">pval_adjust_method</span> <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>           <span class="va">verbose</span> <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># Get results table</span></span>
<span>    <span class="va">pvals</span> <span class="op">&lt;-</span> <span class="va">lrt_results</span><span class="op">$</span><span class="va">table</span></span>
<span>    </span>
<span>    <span class="co"># Add adj p-values</span></span>
<span>    <span class="va">pvals</span><span class="op">$</span><span class="va">adj_pval</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">$</span><span class="va">PValue</span>,</span>
<span>                                      method <span class="op">=</span> <span class="va">pval_adjust_method</span><span class="op">)</span></span>
<span>    </span>
<span>    <span class="co"># Make rownames a column</span></span>
<span>    <span class="va">pvals</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span>    </span>
<span>    <span class="co"># Print number of DEGs found if verbose</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">numDEGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">[</span><span class="va">pvals</span><span class="op">$</span><span class="va">adj_pval</span> <span class="op">&lt;</span> <span class="va">adj_pval</span>, <span class="op">]</span><span class="op">)</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/message.html">message</a></span><span class="op">(</span><span class="va">numDEGs</span>, <span class="st">" DEGs found"</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    </span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">lrt_clean</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">lrt</span>, <span class="op">~</span> <span class="fu">map</span><span class="op">(</span><span class="va">.x</span>, <span class="op">~</span><span class="fu">clean_lrt_results</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Get significant genes and generate plots</p>
<ul>
<li>NOTE: plotting function needs to be updated - not currently working</li>
</ul>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">get_sig_genes_and_plots</span> <span class="op">&lt;-</span></span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">celltype_de</span>,</span>
<span>           <span class="va">pval_adjustment</span> <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>           <span class="va">celltype_counts</span>,</span>
<span>           <span class="va">plot_folder</span> <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           <span class="va">verbose</span> <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">{</span></span>
<span></span>
<span>    <span class="co"># get sig DEGs for each</span></span>
<span>    <span class="va">celltype_DEGs</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">celltype_de</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>        <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">adj_pval</span> <span class="op">&lt;</span> <span class="va">pval_adjustment</span>, <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">unique_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">celltype_DEGs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>      <span class="va">x</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">unique_degs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">unique_genes</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># if no DEGs found break but return the DE analysis scores still</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">unique_degs</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span>
<span>      <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span></span>
<span>          <span class="st">"celltype_DEGs"</span> <span class="op">=</span> <span class="va">celltype_DEGs</span>,</span>
<span>          <span class="st">"celltype_all_genes"</span> <span class="op">=</span> <span class="va">celltype_de</span>,</span>
<span>          <span class="st">"celltype_counts"</span> <span class="op">=</span> <span class="va">counts_celltypes</span></span>
<span>        <span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isTRUE</a></span><span class="op">(</span><span class="va">verbose</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/message.html">message</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">unique_degs</span><span class="op">)</span>, <span class="st">" unique DEGs foundacross all cell types"</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">celltype_all_genes_dt</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/rbindlist.html">rbindlist</a></span><span class="op">(</span><span class="va">celltype_de</span>, idcol <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu">setnames</span><span class="op">(</span><span class="va">celltype_all_genes_dt</span>, <span class="st">".id"</span>, <span class="st">"celltype"</span><span class="op">)</span></span>
<span></span>
<span>    <span class="va">celltype_DEGs_dt</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/rbindlist.html">rbindlist</a></span><span class="op">(</span><span class="va">celltype_DEGs</span>, idcol <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu">setnames</span><span class="op">(</span><span class="va">celltype_DEGs_dt</span>, <span class="st">".id"</span>, <span class="st">"celltype"</span><span class="op">)</span></span>
<span></span>
<span>    <span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isFALSE</a></span><span class="op">(</span><span class="va">plot_folder</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Logic.html">isTRUE</a></span><span class="op">(</span><span class="va">verbose</span><span class="op">)</span><span class="op">)</span></span>
<span>            <span class="fu"><a href="https://rdrr.io/r/base/message.html">message</a></span><span class="op">(</span><span class="st">"Plotting the results of differential expression analysis"</span><span class="op">)</span></span>
<span></span>
<span>        <span class="co">#make plots for DE analysis</span></span>
<span>        <span class="co"># plot_de_analysis(pb_dat,y,celltype_DEGs_dt,celltype_all_genes_dt,</span></span>
<span>        <span class="co">#                  counts_celltypes,plot_folder)</span></span>
<span>    <span class="op">}</span></span>
<span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">celltype_DEGs</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">lrt_sig</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span></span>
<span>  <span class="va">lrt_clean</span>,</span>
<span>  <span class="op">~</span> <span class="fu">get_sig_genes_and_plots</span><span class="op">(</span></span>
<span>    <span class="va">.x</span>,</span>
<span>    celltype_counts <span class="op">=</span> <span class="va">counts_celltypes</span>,</span>
<span>    plot_folder <span class="op">=</span> <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"temp"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The results are in a list of lists. The first level is the celltypes and the next is the comparisons.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># the top level is the celltypes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">lrt_sig</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code> [1] "Endothelial" "Pericyte"    "SMC"         "Oligo"       "Fibroblast" 
 [6] "Neuron_ex"   "Microglia"   "T-cell"      "ambiguous"   "OPC"        
[11] "Astro"       "Neuron-in"   "Neuron_in"   "Neuron-ex"  </code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># the next level is the comparisons</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">lrt_sig</span><span class="op">$</span><span class="va">Endothelial</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "groupAD_P_oxford_dunno_vs_others"     
[2] "groupAD_V_oxford_dunno_vs_others"     
[3] "groupControl_P_oxford_dunno_vs_others"
[4] "groupControl_V_oxford_dunno_vs_others"</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">lrt_sig</span><span class="op">$</span><span class="va">Endothelial</span><span class="op">$</span><span class="va">groupAD_P_oxford_dunno_vs_others</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>        logFC   logCPM        LR       PValue     adj_pval  name
6  -0.4687318 6.607357  9.122385 2.525005e-03 1.671368e-02   FGR
7  -0.6893936 9.154802 12.075713 5.108293e-04 4.567573e-03   CFH
13 -0.4399714 4.354509  7.739166 5.403575e-03 3.107116e-02 LAS1L
14  1.0781034 4.703413 28.019890 1.200749e-07 3.129920e-06 ENPP4
20 -0.4048579 4.802171  7.527897 6.075078e-03 3.399033e-02 RAD52
21  1.1804340 2.826355  9.000979 2.698350e-03 1.763341e-02   BAD</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save results</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">lrt_sig</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_sig-genes_ad-vs-control-and-prep.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="plots" class="level2" data-number="6.1"><h2 data-number="6.1" class="anchored" data-anchor-id="plots">
<span class="header-section-number">6.1</span> Plots</h2>
<p>Let’s have a look at the results with some plots.</p>
<p>First I’ll merge the different contrasts into a single dataframe.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># merge contrasts</span></span>
<span><span class="va">lrt_merged</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">lrt_sig</span>,</span>
<span>                  <span class="co"># add the contrast name as a column</span></span>
<span>                  <span class="op">~</span> <span class="fu">map2</span><span class="op">(</span></span>
<span>                    <span class="va">.x</span>,</span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>,</span>
<span>                    <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span></span>
<span>                      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>                        contrast <span class="op">=</span> <span class="va">.y</span>,</span>
<span>                        <span class="co"># add direction of expression</span></span>
<span>                        deg_direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">logFC</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span><span class="op">)</span></span>
<span>                      <span class="op">)</span></span>
<span>                  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                    <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/list_c.html">list_rbind</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># set colour palette </span></span>
<span><span class="va">pal</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu">wesanderson</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/wesanderson/man/wes_palette.html">wes_palette</a></span><span class="op">(</span><span class="st">"Royal2"</span><span class="op">)</span>,</span>
<span>      <span class="fu">wesanderson</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/wesanderson/man/wes_palette.html">wes_palette</a></span><span class="op">(</span><span class="st">"Moonrise3"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">lrt_merged</span> <span class="op">&lt;-</span> <span class="fu">map2</span><span class="op">(</span><span class="va">lrt_merged</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">lrt_merged</span><span class="op">)</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span></span>
<span>                     <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>celltype <span class="op">=</span> <span class="va">.y</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/list_c.html">list_rbind</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="counts-of-degs" class="level3" data-number="6.1.1"><h3 data-number="6.1.1" class="anchored" data-anchor-id="counts-of-degs">
<span class="header-section-number">6.1.1</span> Counts of DEGs</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">lrt_merged</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">deg_direction</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"count"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Cell type counts of DEGs"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y<span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, x <span class="op">=</span> <span class="st">"DEG direction"</span>, fill<span class="op">=</span><span class="st">"Cell Type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_viridis</span><span class="op">(</span>discrete <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="11_pseudobulk_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span>  <span class="co">#scale_fill_manual(values=pal)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">lrt_merged</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">contrast</span>, y <span class="op">=</span> <span class="va">logFC</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">contrast</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Cell type Log2 Fold Change of DEGs"</span><span class="op">)</span> <span class="op">+</span><span class="co">#,</span></span>
<span>          <span class="co"># subtitle = paste0("Median LFC of Down and Up regulated DEGs: ",</span></span>
<span>          <span class="co">#                   round(degs_median_lfc$V1[[2]]*-1,2),", ",</span></span>
<span>          <span class="co">#                   round(degs_median_lfc$V1[[1]],2)))+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y<span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, x <span class="op">=</span> <span class="st">"DEG direction"</span>, fill<span class="op">=</span><span class="st">"Cell Type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">6</span><span class="op">)</span>,</span>
<span>       axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span>, size <span class="op">=</span> <span class="fl">6</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_viridis</span><span class="op">(</span>discrete <span class="op">=</span> <span class="cn">T</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AD P vs others"</span>, <span class="st">"AD V vs others"</span>, <span class="st">"Control P vs others"</span>, <span class="st">"Control V vs others"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="11_pseudobulk_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="volcano-plots" class="level3" data-number="6.1.2"><h3 data-number="6.1.2" class="anchored" data-anchor-id="volcano-plots">
<span class="header-section-number">6.1.2</span> Volcano plots</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># add colour identifier</span></span>
<span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Up"</span> <span class="op">=</span> <span class="st">"#FDE725FF"</span>, <span class="st">"Down"</span> <span class="op">=</span> <span class="st">"#440154FF"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lrt_merged</span> <span class="op">%&gt;%</span></span>
<span>    <span class="co"># scale FDR by log10</span></span>
<span>    <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">logFC</span>, y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">adj_pval</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>colour <span class="op">=</span> <span class="va">deg_direction</span><span class="op">)</span>, stat <span class="op">=</span> <span class="st">"identity"</span>, size <span class="op">=</span> <span class="fl">.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>        yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">"#990000"</span>,</span>
<span>        linetype <span class="op">=</span> <span class="st">"dashed"</span>, linewidth <span class="op">=</span> <span class="fl">.3</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="co"># geom_vline(xintercept = 0, colour="black", size=.3) +</span></span>
<span>    <span class="fu">labs</span><span class="op">(</span>y <span class="op">=</span> <span class="st">"-log10(FDR)"</span>, x <span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, colour <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>        axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>, axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">scale_colour_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">geom_text_repel</span><span class="op">(</span> <span class="co"># give top three genes per cell type by adjusted p value</span></span>
<span>        data <span class="op">=</span> <span class="va">lrt_merged</span> <span class="op">%&gt;%</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice_min</a></span><span class="op">(</span><span class="va">adj_pval</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,</span>
<span>        <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>        box.padding <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.35</span>, <span class="st">"lines"</span><span class="op">)</span>,</span>
<span>        point.padding <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"lines"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="11_pseudobulk_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section></section></section><section id="deseq-2" class="level1" data-number="7"><h1 data-number="7">
<span class="header-section-number">7</span> DESeq 2</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># make count rownames match data rownames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">doner</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"doner-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Let’s first try doing pseudobulk by disease status</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># setup columns to aggregate by</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">celltype</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co"># remove the "_" in the celltype names and replace with "-"</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">celltype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">"-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">celltype</span><span class="op">)</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">donor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"donor-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">diagnosis</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AD"</span>, <span class="st">"Control"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># counts matrix individual level</span></span>
<span><span class="va">cts</span> <span class="op">&lt;-</span> <span class="fu">AggregateExpression</span><span class="op">(</span><span class="va">sce</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"celltype"</span>, <span class="st">"donor"</span>, <span class="st">"diagnosis"</span><span class="op">)</span>,</span>
<span>                    assays <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                    slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>                    return.seurat <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">cts</span> <span class="op">&lt;-</span> <span class="va">cts</span><span class="op">$</span><span class="va">RNA</span></span>
<span><span class="va">cts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>         ambiguous_donor-1_Control ambiguous_donor-14_AD
TSPAN6                           1                     0
TNMD                             0                     0
DPM1                             1                     0
SCYL3                            4                     0
C1orf112                         6                     1
         ambiguous_donor-15_Control ambiguous_donor-16_AD
TSPAN6                            0                     0
TNMD                              0                     0
DPM1                              0                     0
SCYL3                             0                     1
C1orf112                          0                     0
         ambiguous_donor-17_Control ambiguous_donor-18_AD
TSPAN6                            0                     1
TNMD                              0                     0
DPM1                              2                     5
SCYL3                             1                     0
C1orf112                          4                     4
         ambiguous_donor-19_Control ambiguous_donor-2_AD ambiguous_donor-20_AD
TSPAN6                            0                    1                     4
TNMD                              0                    0                     0
DPM1                              1                    1                    71
SCYL3                             1                    6                    64
C1orf112                          1                    9                   130
         ambiguous_donor-21_Control
TSPAN6                            0
TNMD                              0
DPM1                              0
SCYL3                             0
C1orf112                          0</code></pre>
</div>
</div>
<p>Now we need to do some data manipulation to get the count and column data in the format DESeq2 requires</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># transpose</span></span>
<span><span class="va">cts_t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">cts</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">split_rows</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'_.*'</span>, <span class="st">''</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">cts_t</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cts_split</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split.data.frame</a></span><span class="op">(</span><span class="va">cts_t</span>,</span>
<span>                              f <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">split_rows</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cts_split_m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">cts_split</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Function to modify column names</span></span>
<span><span class="va">modify_column_names</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">df</span>, <span class="va">number_of_groups</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span>  <span class="co"># get the number of "_" seperations in the names</span></span>
<span>  <span class="va">number_of_splits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="co"># get the last n - 1 groups</span></span>
<span>  <span class="va">cols</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">map</span><span class="op">(</span><span class="op">(</span><span class="va">number_of_splits</span> <span class="op">-</span> <span class="op">(</span><span class="va">number_of_groups</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">:</span><span class="va">number_of_splits</span>,</span>
<span>        <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span></span>
<span>          <span class="va">df</span></span>
<span>        <span class="op">)</span><span class="op">)</span>, <span class="st">"_"</span><span class="op">)</span>, <span class="st">"["</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="va">paste</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">cols</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">df</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">cts_split_m</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cts_split_m</span>, <span class="va">modify_column_names</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cts_split_m</span><span class="op">$</span><span class="va">`Neuron-ex`</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>         donor-1_Control donor-10_AD
TSPAN6                20           0
TNMD                   0           0
DPM1                1244          34
SCYL3                690          15
C1orf112             422          13
FGR                    5           0
CFH                   31           0
FUCA2                141           3
GCLC                 847          23
NFYA                 172           5</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># get the col data</span></span>
<span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span></span>
<span>  <span class="va">cts_split_m</span>,</span>
<span>  <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span></span>
<span>    samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>,</span>
<span>    condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>, <span class="st">"Control"</span>, <span class="st">"AD"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">column_to_rownames</span><span class="op">(</span>var <span class="op">=</span> <span class="st">"samples"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># check that count and col data cols/rows match</span></span>
<span><span class="fu">map2_lgl</span><span class="op">(</span><span class="va">cts_split_m</span>, <span class="va">colData</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>  ambiguous       Astro Endothelial  Fibroblast   Microglia   Neuron-ex 
       TRUE        TRUE        TRUE        TRUE        TRUE        TRUE 
  Neuron-in       Oligo         OPC    Pericyte         SMC      T-cell 
       TRUE        TRUE        TRUE        TRUE        TRUE        TRUE </code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb52"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># get DESeq2 object</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">map2</span><span class="op">(</span><span class="va">cts_split_m</span>, <span class="va">colData</span>, <span class="op">~</span> <span class="fu">DESeqDataSetFromMatrix</span><span class="op">(</span>countData <span class="op">=</span> <span class="va">.x</span>,</span>
<span>                                                           colData <span class="op">=</span> <span class="va">.y</span>,</span>
<span>                                                           design <span class="op">=</span> <span class="op">~</span> <span class="va">condition</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># run DESeq2</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="va">DESeq</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can see the result names</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">resultsNames</span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">Astro</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"               "condition_Control_vs_AD"</code></pre>
</div>
</div>
<p>And then extract these results</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb56"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">dds</span>, <span class="op">~</span><span class="fu">results</span><span class="op">(</span><span class="va">.x</span>, name <span class="op">=</span> <span class="st">"condition_Control_vs_AD"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">Astro</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition Control vs AD 
Wald test p-value: condition Control vs AD 
DataFrame with 6 rows and 6 columns
           baseMean log2FoldChange     lfcSE      stat    pvalue      padj
          &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
TSPAN6   117.359710      0.1513936 0.2880161  0.525643 0.5991363  0.813982
TNMD       0.124692     -2.0696063 2.9463214 -0.702437 0.4824064        NA
DPM1     272.642690      0.1491363 0.0817873  1.823464 0.0682331  0.298400
SCYL3    179.262618      0.0363757 0.0867459  0.419337 0.6749701  0.853200
C1orf112 118.631936      0.1781016 0.1338314  1.330791 0.1832577  0.478475
FGR        4.899549     -0.7740027 0.4755142 -1.627717 0.1035848        NA</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb58"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># merge celltype dataframes</span></span>
<span><span class="va">res_merged</span> <span class="op">&lt;-</span> <span class="fu">map2</span><span class="op">(</span><span class="va">res</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                     <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                     <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>celltype <span class="op">=</span> <span class="va">.y</span>, deg_direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">log2FoldChange</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/list_c.html">list_rbind</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save results</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">res_merged</span>, <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="plots-1" class="level2" data-number="7.1"><h2 data-number="7.1" class="anchored" data-anchor-id="plots-1">
<span class="header-section-number">7.1</span> Plots</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb60"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">res_merged</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">deg_direction</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"count"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Cell type counts of DEGs - Control VS AD"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y<span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, x <span class="op">=</span> <span class="st">"DEG direction"</span>, fill<span class="op">=</span><span class="st">"Cell Type"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size<span class="op">=</span><span class="fl">9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_viridis</span><span class="op">(</span>discrete <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="11_pseudobulk_files/figure-html/unnamed-chunk-63-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb61"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span>  <span class="co">#scale_fill_manual(values=pal)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="volcano-plots-1" class="level3" data-number="7.1.1"><h3 data-number="7.1.1" class="anchored" data-anchor-id="volcano-plots-1">
<span class="header-section-number">7.1.1</span> Volcano plots</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb62"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># add colour identifier</span></span>
<span><span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Up"</span> <span class="op">=</span> <span class="st">"#FDE725FF"</span>, <span class="st">"Down"</span> <span class="op">=</span> <span class="st">"#440154FF"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_merged</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># filter to significant hits</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># scale FDR by log10</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">log2FoldChange</span>, y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">padj</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>colour <span class="op">=</span> <span class="va">deg_direction</span><span class="op">)</span>, stat <span class="op">=</span> <span class="st">"identity"</span>, size <span class="op">=</span> <span class="fl">.2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span></span>
<span>    yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>,</span>
<span>    colour <span class="op">=</span> <span class="st">"#990000"</span>,</span>
<span>    linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>    linewidth <span class="op">=</span> <span class="fl">.3</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># geom_vline(xintercept = 0, colour="black", size=.3) +</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>y <span class="op">=</span> <span class="st">"-log10(FDR)"</span>, x <span class="op">=</span> <span class="st">"Log2 Fold Change"</span>, colour <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_cowplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>    axis.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">9</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_colour_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">cols</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_text_repel</span><span class="op">(</span></span>
<span>    <span class="co"># give top three genes per cell type by adjusted p value</span></span>
<span>    data <span class="op">=</span> <span class="va">res_merged</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice_min</a></span><span class="op">(</span><span class="va">padj</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>,</span>
<span>    <span class="fu">aes</span><span class="op">(</span>label <span class="op">=</span> <span class="va">gene</span><span class="op">)</span>,</span>
<span>    size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    box.padding <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.35</span>, <span class="st">"lines"</span><span class="op">)</span>,</span>
<span>    point.padding <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">0.3</span>, <span class="st">"lines"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="11_pseudobulk_files/figure-html/unnamed-chunk-64-1.png" class="img-fluid" width="672"></p>
</div>
</div>


<!-- -->

</section></section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb63" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Perform Pseudobulk</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: true</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="co">#self-contained: true</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> true</span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>Note that I'll use the higher level annotations to start with.</span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>This only involved aggregating the venous/capillary/arterial subtypes to a larger "Endothelial" category.</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.qmd"</span>)</span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a><span class="fu"># Load packages</span></span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-29"><a href="#cb63-29" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-30"><a href="#cb63-30" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/pseudobulk"</span>)</span>
<span id="cb63-31"><a href="#cb63-31" aria-hidden="true" tabindex="-1"></a><span class="co">#source necessary functions from other scripts</span></span>
<span id="cb63-32"><a href="#cb63-32" aria-hidden="true" tabindex="-1"></a><span class="co">#source(here::here(path, "validate_input_parameters_de.R"))</span></span>
<span id="cb63-33"><a href="#cb63-33" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(path, <span class="st">"make_pseudobulk.R"</span>))</span>
<span id="cb63-34"><a href="#cb63-34" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(path, <span class="st">"de_analysis.R"</span>))</span>
<span id="cb63-35"><a href="#cb63-35" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(path, <span class="st">"plot_de_analysis.R"</span>))</span>
<span id="cb63-36"><a href="#cb63-36" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-37"><a href="#cb63-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-38"><a href="#cb63-38" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read in data</span></span>
<span id="cb63-39"><a href="#cb63-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-40"><a href="#cb63-40" aria-hidden="true" tabindex="-1"></a>First we'll read in the scFlow output to a sce</span>
<span id="cb63-41"><a href="#cb63-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-44"><a href="#cb63-44" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-45"><a href="#cb63-45" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb63-46"><a href="#cb63-46" aria-hidden="true" tabindex="-1"></a><span class="co"># read data</span></span>
<span id="cb63-47"><a href="#cb63-47" aria-hidden="true" tabindex="-1"></a><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span id="cb63-48"><a href="#cb63-48" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span>))</span>
<span id="cb63-49"><a href="#cb63-49" aria-hidden="true" tabindex="-1"></a><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-updated_dim-35.rds"))</span></span>
<span id="cb63-50"><a href="#cb63-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-51"><a href="#cb63-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-54"><a href="#cb63-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-55"><a href="#cb63-55" aria-hidden="true" tabindex="-1"></a><span class="co"># make count rownames match data rownames</span></span>
<span id="cb63-56"><a href="#cb63-56" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data)</span>
<span id="cb63-57"><a href="#cb63-57" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>doner <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"doner-"</span>, sce<span class="sc">$</span>individual)</span>
<span id="cb63-58"><a href="#cb63-58" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-59"><a href="#cb63-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-60"><a href="#cb63-60" aria-hidden="true" tabindex="-1"></a><span class="fu"># Average expression</span></span>
<span id="cb63-61"><a href="#cb63-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-64"><a href="#cb63-64" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-65"><a href="#cb63-65" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(sce, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"doner"</span>))</span>
<span id="cb63-66"><a href="#cb63-66" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> avg_exp<span class="sc">$</span>RNA</span>
<span id="cb63-67"><a href="#cb63-67" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb63-68"><a href="#cb63-68" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> <span class="fu">data.table</span>(avg_exp, <span class="at">keep.rownames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb63-69"><a href="#cb63-69" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(avg_exp, <span class="st">"rn"</span>, <span class="st">"gene"</span>)</span>
<span id="cb63-70"><a href="#cb63-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-71"><a href="#cb63-71" aria-hidden="true" tabindex="-1"></a><span class="co"># pivot longer such that celltypes are separated out</span></span>
<span id="cb63-72"><a href="#cb63-72" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> avg_exp <span class="sc">%&gt;%</span></span>
<span id="cb63-73"><a href="#cb63-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb63-74"><a href="#cb63-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">!</span>gene,</span>
<span id="cb63-75"><a href="#cb63-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"celltype"</span>, <span class="st">".value"</span>),</span>
<span id="cb63-76"><a href="#cb63-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">"(.*)_(.*)"</span></span>
<span id="cb63-77"><a href="#cb63-77" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb63-78"><a href="#cb63-78" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb63-79"><a href="#cb63-79" aria-hidden="true" tabindex="-1"></a>file_out <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-celltype-per-doner.csv"</span>)</span>
<span id="cb63-80"><a href="#cb63-80" aria-hidden="true" tabindex="-1"></a><span class="co">#file_out &lt;- here::here("03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-celltype-per-doner_35-dims.csv")</span></span>
<span id="cb63-81"><a href="#cb63-81" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(avg_exp, file_out)</span>
<span id="cb63-82"><a href="#cb63-82" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb63-83"><a href="#cb63-83" aria-hidden="true" tabindex="-1"></a><span class="do">## split AD and CT samples</span></span>
<span id="cb63-84"><a href="#cb63-84" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb63-85"><a href="#cb63-85" aria-hidden="true" tabindex="-1"></a><span class="co"># avg_exp &lt;- AverageExpression(sdata_endo, group.by = c("ident", "type"))</span></span>
<span id="cb63-86"><a href="#cb63-86" aria-hidden="true" tabindex="-1"></a><span class="co"># avg_exp &lt;- avg_exp$RNA</span></span>
<span id="cb63-87"><a href="#cb63-87" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb63-88"><a href="#cb63-88" aria-hidden="true" tabindex="-1"></a><span class="co"># avg_exp &lt;- data.table(avg_exp, keep.rownames = T)</span></span>
<span id="cb63-89"><a href="#cb63-89" aria-hidden="true" tabindex="-1"></a><span class="co"># setnames(avg_exp, "rn", "gene")</span></span>
<span id="cb63-90"><a href="#cb63-90" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb63-91"><a href="#cb63-91" aria-hidden="true" tabindex="-1"></a><span class="co"># file_out &lt;- "/scratch/c.mpmfw/Endo_10X_main/Analysis/Expression_matrix/avg_expr_main_cell_type_40_samples_AD_vs_CT_v3.txt"</span></span>
<span id="cb63-92"><a href="#cb63-92" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(avg_exp, file = file_out, sep = "\t", quote = F, row.names = F, col.names = T)</span></span>
<span id="cb63-93"><a href="#cb63-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-94"><a href="#cb63-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-97"><a href="#cb63-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-98"><a href="#cb63-98" aria-hidden="true" tabindex="-1"></a><span class="co"># Make column with cluster and annotation combined</span></span>
<span id="cb63-99"><a href="#cb63-99" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>cluster_annoated <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster_"</span>, sce<span class="sc">$</span>seurat_clusters, <span class="st">"_"</span>, <span class="fu">Idents</span>(sce))</span>
<span id="cb63-100"><a href="#cb63-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-101"><a href="#cb63-101" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(sce, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"cluster_annoated"</span>, <span class="st">"doner"</span>))</span>
<span id="cb63-102"><a href="#cb63-102" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> avg_exp<span class="sc">$</span>RNA</span>
<span id="cb63-103"><a href="#cb63-103" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb63-104"><a href="#cb63-104" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> <span class="fu">data.table</span>(avg_exp, <span class="at">keep.rownames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb63-105"><a href="#cb63-105" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(avg_exp, <span class="st">"rn"</span>, <span class="st">"gene"</span>)</span>
<span id="cb63-106"><a href="#cb63-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-107"><a href="#cb63-107" aria-hidden="true" tabindex="-1"></a><span class="co"># pivot longer such that celltypes are separated out</span></span>
<span id="cb63-108"><a href="#cb63-108" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> avg_exp <span class="sc">%&gt;%</span></span>
<span id="cb63-109"><a href="#cb63-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb63-110"><a href="#cb63-110" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">!</span>gene,</span>
<span id="cb63-111"><a href="#cb63-111" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"celltype"</span>, <span class="st">".value"</span>),</span>
<span id="cb63-112"><a href="#cb63-112" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">"(.*)_(.*)"</span></span>
<span id="cb63-113"><a href="#cb63-113" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb63-114"><a href="#cb63-114" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb63-115"><a href="#cb63-115" aria-hidden="true" tabindex="-1"></a>file_out <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-cluster-per-doner.csv"</span>)</span>
<span id="cb63-116"><a href="#cb63-116" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(avg_exp, file_out)</span>
<span id="cb63-117"><a href="#cb63-117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-118"><a href="#cb63-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-119"><a href="#cb63-119" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pseudobulk</span></span>
<span id="cb63-120"><a href="#cb63-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-121"><a href="#cb63-121" aria-hidden="true" tabindex="-1"></a>Now we can subset to variables we want to aggregate across for the pseudobulk</span>
<span id="cb63-122"><a href="#cb63-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-125"><a href="#cb63-125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-126"><a href="#cb63-126" aria-hidden="true" tabindex="-1"></a><span class="co"># aggregate read counts </span></span>
<span id="cb63-127"><a href="#cb63-127" aria-hidden="true" tabindex="-1"></a>aggr_counts <span class="ot">&lt;-</span> <span class="fu">Seurat2PB</span>(sce, <span class="at">sample=</span><span class="st">"group"</span>, <span class="at">cluster=</span><span class="st">"cluster_celltype"</span>)</span>
<span id="cb63-128"><a href="#cb63-128" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(aggr_counts)</span>
<span id="cb63-129"><a href="#cb63-129" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(aggr_counts<span class="sc">$</span>samples, <span class="at">n=</span>10L)</span>
<span id="cb63-130"><a href="#cb63-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-131"><a href="#cb63-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-132"><a href="#cb63-132" aria-hidden="true" tabindex="-1"></a>Convert back to single cell experiment</span>
<span id="cb63-133"><a href="#cb63-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-136"><a href="#cb63-136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-137"><a href="#cb63-137" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> Seurat<span class="sc">::</span><span class="fu">as.SingleCellExperiment</span>(sce)</span>
<span id="cb63-138"><a href="#cb63-138" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-139"><a href="#cb63-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-140"><a href="#cb63-140" aria-hidden="true" tabindex="-1"></a><span class="fu"># Code from Alan</span></span>
<span id="cb63-141"><a href="#cb63-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-142"><a href="#cb63-142" aria-hidden="true" tabindex="-1"></a>There's an interest in incorporating pseudobuilk into scflow itself, and a member of Nathan's lab (Alan) has written some code for performing pseudobulk on scflow outputs.</span>
<span id="cb63-143"><a href="#cb63-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-144"><a href="#cb63-144" aria-hidden="true" tabindex="-1"></a>Here I'll play with his code and start thinking about how this could be incorporated into scflow</span>
<span id="cb63-145"><a href="#cb63-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-148"><a href="#cb63-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-149"><a href="#cb63-149" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-150"><a href="#cb63-150" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-151"><a href="#cb63-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-152"><a href="#cb63-152" aria-hidden="true" tabindex="-1"></a><span class="co">#Ran on R 4.3</span></span>
<span id="cb63-153"><a href="#cb63-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-154"><a href="#cb63-154" aria-hidden="true" tabindex="-1"></a><span class="co">#Get package dependencies</span></span>
<span id="cb63-155"><a href="#cb63-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-156"><a href="#cb63-156" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>)) <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb63-157"><a href="#cb63-157" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"RcppParallel"</span>)) <span class="fu">install.packages</span>(<span class="st">"RcppParallel"</span>)</span>
<span id="cb63-158"><a href="#cb63-158" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"pacman"</span>)) <span class="fu">install.packages</span>(<span class="st">"pacman"</span>)</span>
<span id="cb63-159"><a href="#cb63-159" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"qs"</span>)) <span class="fu">install.packages</span>(<span class="st">"qs"</span>)</span>
<span id="cb63-160"><a href="#cb63-160" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"SingleCellExperiment"</span>))</span>
<span id="cb63-161"><a href="#cb63-161" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"SingleCellExperiment"</span>)</span>
<span id="cb63-162"><a href="#cb63-162" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"biomaRt"</span>)) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"biomaRt"</span>)</span>
<span id="cb63-163"><a href="#cb63-163" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"ggplot2"</span>)) <span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb63-164"><a href="#cb63-164" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"cowplot"</span>)) <span class="fu">install.packages</span>(<span class="st">"cowplot"</span>)</span>
<span id="cb63-165"><a href="#cb63-165" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"viridis"</span>)) <span class="fu">install.packages</span>(<span class="st">"viridis"</span>)</span>
<span id="cb63-166"><a href="#cb63-166" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"ggrepel"</span>)) <span class="fu">install.packages</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb63-167"><a href="#cb63-167" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"Hmisc"</span>)) <span class="fu">install.packages</span>(<span class="st">"Hmisc"</span>)</span>
<span id="cb63-168"><a href="#cb63-168" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"edgeR"</span>)) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"edgeR"</span>)</span>
<span id="cb63-169"><a href="#cb63-169" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"stats"</span>)) <span class="fu">install.packages</span>(<span class="st">"stats"</span>)</span>
<span id="cb63-170"><a href="#cb63-170" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"data.table"</span>)) <span class="fu">install.packages</span>(<span class="st">"data.table"</span>)</span>
<span id="cb63-171"><a href="#cb63-171" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"reshape2"</span>)) <span class="fu">install.packages</span>(<span class="st">"reshape2"</span>)</span>
<span id="cb63-172"><a href="#cb63-172" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"wesanderson"</span>)) <span class="fu">install.packages</span>(<span class="st">"wesanderson"</span>)</span>
<span id="cb63-173"><a href="#cb63-173" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"EnsDb.Hsapiens.v79"</span>)){</span>
<span id="cb63-174"><a href="#cb63-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-175"><a href="#cb63-175" aria-hidden="true" tabindex="-1"></a>  <span class="co">#This package uses dependencies which are large so increase time for download</span></span>
<span id="cb63-176"><a href="#cb63-176" aria-hidden="true" tabindex="-1"></a>  <span class="fu">options</span>(<span class="at">timeout=</span><span class="dv">2000</span>)</span>
<span id="cb63-177"><a href="#cb63-177" aria-hidden="true" tabindex="-1"></a>  BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"EnsDb.Hsapiens.v79"</span>)</span>
<span id="cb63-178"><a href="#cb63-178" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb63-179"><a href="#cb63-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-180"><a href="#cb63-180" aria-hidden="true" tabindex="-1"></a><span class="co">#source the necessary function to run pb DE</span></span>
<span id="cb63-181"><a href="#cb63-181" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/pseudobulk/sc_cell_type_de.R"</span>))</span>
<span id="cb63-182"><a href="#cb63-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-183"><a href="#cb63-183" aria-hidden="true" tabindex="-1"></a><span class="co">#library necessary packages</span></span>
<span id="cb63-184"><a href="#cb63-184" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb63-185"><a href="#cb63-185" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb63-186"><a href="#cb63-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-187"><a href="#cb63-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-190"><a href="#cb63-190" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-191"><a href="#cb63-191" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-192"><a href="#cb63-192" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-193"><a href="#cb63-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-194"><a href="#cb63-194" aria-hidden="true" tabindex="-1"></a><span class="co">#load dataset and exclude endothelial cells to match Tsai group analysis</span></span>
<span id="cb63-195"><a href="#cb63-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-196"><a href="#cb63-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Mathys_SCE &lt;-</span></span>
<span id="cb63-197"><a href="#cb63-197" aria-hidden="true" tabindex="-1"></a><span class="co">#   qs::qread("../data/sce.qs")</span></span>
<span id="cb63-198"><a href="#cb63-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-199"><a href="#cb63-199" aria-hidden="true" tabindex="-1"></a><span class="co">#remove endothelial cells to match Mathys 2019's approach</span></span>
<span id="cb63-200"><a href="#cb63-200" aria-hidden="true" tabindex="-1"></a><span class="co">#Mathys_SCE &lt;- Mathys_SCE[,!Mathys_SCE$allan_celltype=="Endo"]</span></span>
<span id="cb63-201"><a href="#cb63-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-202"><a href="#cb63-202" aria-hidden="true" tabindex="-1"></a><span class="co"># load data</span></span>
<span id="cb63-203"><a href="#cb63-203" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce.rds"</span>))</span>
<span id="cb63-204"><a href="#cb63-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-205"><a href="#cb63-205" aria-hidden="true" tabindex="-1"></a><span class="co">#run DE analysis</span></span>
<span id="cb63-206"><a href="#cb63-206" aria-hidden="true" tabindex="-1"></a>sc_cell_type_de_return <span class="ot">&lt;-</span></span>
<span id="cb63-207"><a href="#cb63-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sc_cell_type_de</span>(sce, <span class="at">design =</span> <span class="sc">~</span> group,</span>
<span id="cb63-208"><a href="#cb63-208" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pseudobulk_ID=</span><span class="st">"manifest"</span>, <span class="at">celltype_ID=</span><span class="st">"cluster_celltype"</span>,</span>
<span id="cb63-209"><a href="#cb63-209" aria-hidden="true" tabindex="-1"></a>                  <span class="at">folder =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/graphs/"</span>),</span>
<span id="cb63-210"><a href="#cb63-210" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb63-211"><a href="#cb63-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-212"><a href="#cb63-212" aria-hidden="true" tabindex="-1"></a><span class="co">#TO DO - add gene name to DEGs in results.......</span></span>
<span id="cb63-213"><a href="#cb63-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-214"><a href="#cb63-214" aria-hidden="true" tabindex="-1"></a><span class="co">#save results</span></span>
<span id="cb63-215"><a href="#cb63-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-216"><a href="#cb63-216" aria-hidden="true" tabindex="-1"></a><span class="co">#Cell Counts</span></span>
<span id="cb63-217"><a href="#cb63-217" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">fwrite</span>(<span class="fu">data.table</span>(<span class="at">cell=</span><span class="fu">names</span>(</span>
<span id="cb63-218"><a href="#cb63-218" aria-hidden="true" tabindex="-1"></a>  sc_cell_type_de_return<span class="sc">$</span>celltype_counts),</span>
<span id="cb63-219"><a href="#cb63-219" aria-hidden="true" tabindex="-1"></a>  <span class="at">count=</span>sc_cell_type_de_return<span class="sc">$</span>celltype_counts),</span>
<span id="cb63-220"><a href="#cb63-220" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../results/cell_counts.csv"</span>)</span>
<span id="cb63-221"><a href="#cb63-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-222"><a href="#cb63-222" aria-hidden="true" tabindex="-1"></a><span class="co">#DEGs</span></span>
<span id="cb63-223"><a href="#cb63-223" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">fwrite</span>(</span>
<span id="cb63-224"><a href="#cb63-224" aria-hidden="true" tabindex="-1"></a>  data.table<span class="sc">::</span><span class="fu">rbindlist</span>(sc_cell_type_de_return<span class="sc">$</span>celltype_DEGs,<span class="at">idcol=</span><span class="st">'Cell'</span>),</span>
<span id="cb63-225"><a href="#cb63-225" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../results/DEGs.csv"</span>)</span>
<span id="cb63-226"><a href="#cb63-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-227"><a href="#cb63-227" aria-hidden="true" tabindex="-1"></a><span class="co">#All Genes</span></span>
<span id="cb63-228"><a href="#cb63-228" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">fwrite</span>(</span>
<span id="cb63-229"><a href="#cb63-229" aria-hidden="true" tabindex="-1"></a>  data.table<span class="sc">::</span><span class="fu">rbindlist</span>(sc_cell_type_de_return<span class="sc">$</span>celltype_all_genes,<span class="at">idcol=</span><span class="st">'Cell'</span>),</span>
<span id="cb63-230"><a href="#cb63-230" aria-hidden="true" tabindex="-1"></a>  <span class="st">"../results/all_genes.csv"</span>)</span>
<span id="cb63-231"><a href="#cb63-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-232"><a href="#cb63-232" aria-hidden="true" tabindex="-1"></a><span class="co">#R object</span></span>
<span id="cb63-233"><a href="#cb63-233" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(sc_cell_type_de_return,<span class="at">file=</span><span class="st">"../results/Mathys_pb_res.RData"</span>)</span>
<span id="cb63-234"><a href="#cb63-234" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-235"><a href="#cb63-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-236"><a href="#cb63-236" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- ## A more manual approach --&gt;</span></span>
<span id="cb63-237"><a href="#cb63-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-238"><a href="#cb63-238" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Try extracting just the parts of `sc_cell_type_de.R` --&gt;</span></span>
<span id="cb63-239"><a href="#cb63-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-242"><a href="#cb63-242" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-243"><a href="#cb63-243" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-244"><a href="#cb63-244" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-245"><a href="#cb63-245" aria-hidden="true" tabindex="-1"></a><span class="co"># inputs</span></span>
<span id="cb63-246"><a href="#cb63-246" aria-hidden="true" tabindex="-1"></a>SCE <span class="ot">&lt;-</span> sce</span>
<span id="cb63-247"><a href="#cb63-247" aria-hidden="true" tabindex="-1"></a>celltype_ID <span class="ot">&lt;-</span> <span class="st">"ident"</span></span>
<span id="cb63-248"><a href="#cb63-248" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="er">~</span> group</span>
<span id="cb63-249"><a href="#cb63-249" aria-hidden="true" tabindex="-1"></a>pseudobulk_ID <span class="ot">=</span> <span class="st">"manifest"</span></span>
<span id="cb63-250"><a href="#cb63-250" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb63-251"><a href="#cb63-251" aria-hidden="true" tabindex="-1"></a>pb_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"manifest"</span>)</span>
<span id="cb63-252"><a href="#cb63-252" aria-hidden="true" tabindex="-1"></a>verbose <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb63-253"><a href="#cb63-253" aria-hidden="true" tabindex="-1"></a>control <span class="ot">=</span> <span class="st">"Control_V_oxford_dunno"</span></span>
<span id="cb63-254"><a href="#cb63-254" aria-hidden="true" tabindex="-1"></a>pval_adjust_method <span class="ot">=</span> <span class="st">"BH"</span></span>
<span id="cb63-255"><a href="#cb63-255" aria-hidden="true" tabindex="-1"></a>adj_pval<span class="ot">=</span><span class="fl">0.05</span></span>
<span id="cb63-256"><a href="#cb63-256" aria-hidden="true" tabindex="-1"></a>coef <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb63-257"><a href="#cb63-257" aria-hidden="true" tabindex="-1"></a>folder <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/001_snrnaseq_analysis_figures/01_pseudobulk/control_v_oxford"</span>)</span>
<span id="cb63-258"><a href="#cb63-258" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-259"><a href="#cb63-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-262"><a href="#cb63-262" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-263"><a href="#cb63-263" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-264"><a href="#cb63-264" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-265"><a href="#cb63-265" aria-hidden="true" tabindex="-1"></a><span class="co">#first format formula</span></span>
<span id="cb63-266"><a href="#cb63-266" aria-hidden="true" tabindex="-1"></a>design_txt <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">deparse</span>(design,<span class="at">width.cutoff =</span> <span class="dv">500</span>),<span class="at">collapse=</span><span class="st">','</span>)</span>
<span id="cb63-267"><a href="#cb63-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-268"><a href="#cb63-268" aria-hidden="true" tabindex="-1"></a><span class="co">#make design formula minus anything before ~</span></span>
<span id="cb63-269"><a href="#cb63-269" aria-hidden="true" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">gsub</span>(<span class="st">".*~"</span>,<span class="st">"~"</span>,design_txt))</span>
<span id="cb63-270"><a href="#cb63-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-271"><a href="#cb63-271" aria-hidden="true" tabindex="-1"></a><span class="co">#if y not specified take last value in design matrix</span></span>
<span id="cb63-272"><a href="#cb63-272" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(y))</span>
<span id="cb63-273"><a href="#cb63-273" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> design_txt[[<span class="fu">length</span>(design_txt)]]</span>
<span id="cb63-274"><a href="#cb63-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-275"><a href="#cb63-275" aria-hidden="true" tabindex="-1"></a><span class="co">#Check if continuous or categorical variable to be modeled</span></span>
<span id="cb63-276"><a href="#cb63-276" aria-hidden="true" tabindex="-1"></a>y_contin <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb63-277"><a href="#cb63-277" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.numeric</span>(SCE[[y]]))</span>
<span id="cb63-278"><a href="#cb63-278" aria-hidden="true" tabindex="-1"></a>    y_contin <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb63-279"><a href="#cb63-279" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-280"><a href="#cb63-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-281"><a href="#cb63-281" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Get cell type counts --&gt;</span></span>
<span id="cb63-282"><a href="#cb63-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-285"><a href="#cb63-285" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-286"><a href="#cb63-286" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-287"><a href="#cb63-287" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-288"><a href="#cb63-288" aria-hidden="true" tabindex="-1"></a><span class="co">#get counts of each cell type</span></span>
<span id="cb63-289"><a href="#cb63-289" aria-hidden="true" tabindex="-1"></a><span class="co">#counts(SCE) &lt;- as.matrix(counts(SCE))</span></span>
<span id="cb63-290"><a href="#cb63-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-291"><a href="#cb63-291" aria-hidden="true" tabindex="-1"></a>counts_celltypes <span class="ot">&lt;-</span> SCE[[celltype_ID]]</span>
<span id="cb63-292"><a href="#cb63-292" aria-hidden="true" tabindex="-1"></a>counts_celltypes <span class="ot">&lt;-</span><span class="fu">as.vector</span>(<span class="fu">table</span>(counts_celltypes))</span>
<span id="cb63-293"><a href="#cb63-293" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(counts_celltypes) <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">table</span>(SCE[[celltype_ID]]))</span>
<span id="cb63-294"><a href="#cb63-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-295"><a href="#cb63-295" aria-hidden="true" tabindex="-1"></a><span class="co"># Get pseudobulk values</span></span>
<span id="cb63-296"><a href="#cb63-296" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(SCE[[celltype_ID]])</span>
<span id="cb63-297"><a href="#cb63-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-298"><a href="#cb63-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-301"><a href="#cb63-301" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-302"><a href="#cb63-302" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-303"><a href="#cb63-303" aria-hidden="true" tabindex="-1"></a><span class="co"># counts by sample population</span></span>
<span id="cb63-304"><a href="#cb63-304" aria-hidden="true" tabindex="-1"></a><span class="co"># cluster_count_by_sample &lt;- table(sce$cluster_celltype, sce$prep) %&gt;%</span></span>
<span id="cb63-305"><a href="#cb63-305" aria-hidden="true" tabindex="-1"></a><span class="co">#  as.data.frame() %&gt;%</span></span>
<span id="cb63-306"><a href="#cb63-306" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_wider(names_from = Var2, values_from = Freq) %&gt;%</span></span>
<span id="cb63-307"><a href="#cb63-307" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::rename(cell_type = Var1)</span></span>
<span id="cb63-308"><a href="#cb63-308" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb63-309"><a href="#cb63-309" aria-hidden="true" tabindex="-1"></a><span class="co"># gt::gt(cluster_count_by_sample)</span></span>
<span id="cb63-310"><a href="#cb63-310" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb63-311"><a href="#cb63-311" aria-hidden="true" tabindex="-1"></a><span class="co"># # Save data</span></span>
<span id="cb63-312"><a href="#cb63-312" aria-hidden="true" tabindex="-1"></a><span class="co"># readr::write_csv(cluster_count_by_sample, here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/celltype_counts_by_cellpop.csv"))</span></span>
<span id="cb63-313"><a href="#cb63-313" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-314"><a href="#cb63-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-317"><a href="#cb63-317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-318"><a href="#cb63-318" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-319"><a href="#cb63-319" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-320"><a href="#cb63-320" aria-hidden="true" tabindex="-1"></a>pb_dat <span class="ot">&lt;-</span></span>
<span id="cb63-321"><a href="#cb63-321" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(celltypes,<span class="cf">function</span>(x)</span>
<span id="cb63-322"><a href="#cb63-322" aria-hidden="true" tabindex="-1"></a>        <span class="fu">make_pseudobulk</span>(SCE[,SCE[[celltype_ID]]<span class="sc">==</span>x],</span>
<span id="cb63-323"><a href="#cb63-323" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pseudobulk_ID=</span>pseudobulk_ID,</span>
<span id="cb63-324"><a href="#cb63-324" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pb_columns=</span>pb_columns))</span>
<span id="cb63-325"><a href="#cb63-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-326"><a href="#cb63-326" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pb_dat) <span class="ot">&lt;-</span> celltypes</span>
<span id="cb63-327"><a href="#cb63-327" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-328"><a href="#cb63-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-329"><a href="#cb63-329" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- One can set the comparison here via the `control` argument. --&gt;</span></span>
<span id="cb63-330"><a href="#cb63-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-333"><a href="#cb63-333" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-334"><a href="#cb63-334" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-335"><a href="#cb63-335" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-336"><a href="#cb63-336" aria-hidden="true" tabindex="-1"></a><span class="co">#run edgeR LRT DE analysis</span></span>
<span id="cb63-337"><a href="#cb63-337" aria-hidden="true" tabindex="-1"></a>celltype_de <span class="ot">&lt;-</span></span>
<span id="cb63-338"><a href="#cb63-338" aria-hidden="true" tabindex="-1"></a>    <span class="fu">de_analysis</span>(pb_dat,formula,<span class="at">y_name=</span>y,y_contin,coef,control,</span>
<span id="cb63-339"><a href="#cb63-339" aria-hidden="true" tabindex="-1"></a>                    pval_adjust_method, adj_pval, verbose)</span>
<span id="cb63-340"><a href="#cb63-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-341"><a href="#cb63-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-344"><a href="#cb63-344" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-345"><a href="#cb63-345" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-346"><a href="#cb63-346" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-347"><a href="#cb63-347" aria-hidden="true" tabindex="-1"></a><span class="co">#get sig DEGs for each</span></span>
<span id="cb63-348"><a href="#cb63-348" aria-hidden="true" tabindex="-1"></a>celltype_DEGs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(celltype_de, <span class="cf">function</span>(x) x[x<span class="sc">$</span>adj_pval<span class="sc">&lt;</span>adj_pval,])</span>
<span id="cb63-349"><a href="#cb63-349" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span><span class="fu">lapply</span>(celltype_DEGs, <span class="cf">function</span>(x) x<span class="sc">$</span>name)</span>
<span id="cb63-350"><a href="#cb63-350" aria-hidden="true" tabindex="-1"></a>unique_degs <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">unlist</span>(unique_genes))</span>
<span id="cb63-351"><a href="#cb63-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-352"><a href="#cb63-352" aria-hidden="true" tabindex="-1"></a><span class="co">#if no DEGs found break but return the DE analysis scores still</span></span>
<span id="cb63-353"><a href="#cb63-353" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(unique_degs)<span class="sc">==</span><span class="dv">0</span>)</span>
<span id="cb63-354"><a href="#cb63-354" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="st">"celltype_DEGs"</span><span class="ot">=</span>celltype_DEGs,</span>
<span id="cb63-355"><a href="#cb63-355" aria-hidden="true" tabindex="-1"></a>                <span class="st">"celltype_all_genes"</span><span class="ot">=</span>celltype_de,</span>
<span id="cb63-356"><a href="#cb63-356" aria-hidden="true" tabindex="-1"></a>                <span class="st">"celltype_counts"</span><span class="ot">=</span>counts_celltypes))</span>
<span id="cb63-357"><a href="#cb63-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-358"><a href="#cb63-358" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">isTRUE</span>(verbose))</span>
<span id="cb63-359"><a href="#cb63-359" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">length</span>(unique_degs),<span class="st">" unique DEGs foundacross all cell types"</span>)</span>
<span id="cb63-360"><a href="#cb63-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-361"><a href="#cb63-361" aria-hidden="true" tabindex="-1"></a>celltype_all_genes_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_de,<span class="at">idcol =</span> T)</span>
<span id="cb63-362"><a href="#cb63-362" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(celltype_all_genes_dt,<span class="st">".id"</span>,<span class="st">"celltype"</span>)</span>
<span id="cb63-363"><a href="#cb63-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-364"><a href="#cb63-364" aria-hidden="true" tabindex="-1"></a>celltype_DEGs_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_DEGs,<span class="at">idcol =</span> T)</span>
<span id="cb63-365"><a href="#cb63-365" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(celltype_DEGs_dt,<span class="st">".id"</span>,<span class="st">"celltype"</span>)</span>
<span id="cb63-366"><a href="#cb63-366" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-367"><a href="#cb63-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-368"><a href="#cb63-368" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Changed palette arg in code to make this work --&gt;</span></span>
<span id="cb63-369"><a href="#cb63-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-372"><a href="#cb63-372" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-373"><a href="#cb63-373" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-374"><a href="#cb63-374" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-375"><a href="#cb63-375" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">isFALSE</span>(folder)){</span>
<span id="cb63-376"><a href="#cb63-376" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">isTRUE</span>(verbose))</span>
<span id="cb63-377"><a href="#cb63-377" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">"Plotting the results of differential expression analysis"</span>)</span>
<span id="cb63-378"><a href="#cb63-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-379"><a href="#cb63-379" aria-hidden="true" tabindex="-1"></a>        <span class="co">#make plots for DE analysis</span></span>
<span id="cb63-380"><a href="#cb63-380" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot_de_analysis</span>(pb_dat,y,celltype_DEGs_dt,celltype_all_genes_dt,</span>
<span id="cb63-381"><a href="#cb63-381" aria-hidden="true" tabindex="-1"></a>                         counts_celltypes,folder)</span>
<span id="cb63-382"><a href="#cb63-382" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb63-383"><a href="#cb63-383" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-384"><a href="#cb63-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-387"><a href="#cb63-387" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-388"><a href="#cb63-388" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-389"><a href="#cb63-389" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-390"><a href="#cb63-390" aria-hidden="true" tabindex="-1"></a><span class="co">#get gene names for DEGs</span></span>
<span id="cb63-391"><a href="#cb63-391" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_DEGs)<span class="sc">$</span>name)</span>
<span id="cb63-392"><a href="#cb63-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-393"><a href="#cb63-393" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> ensembldb<span class="sc">::</span><span class="fu">select</span>(EnsDb.Hsapiens.v79, <span class="at">keys=</span> genes,</span>
<span id="cb63-394"><a href="#cb63-394" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"GENEID"</span>,</span>
<span id="cb63-395"><a href="#cb63-395" aria-hidden="true" tabindex="-1"></a>                              <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GENEID"</span>,<span class="st">"SYMBOL"</span>))</span>
<span id="cb63-396"><a href="#cb63-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-397"><a href="#cb63-397" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_IDs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>,<span class="st">"hgnc_symbol"</span>)</span>
<span id="cb63-398"><a href="#cb63-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-399"><a href="#cb63-399" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(gene_IDs)</span>
<span id="cb63-400"><a href="#cb63-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-401"><a href="#cb63-401" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setnames</span>(gene_IDs,<span class="st">"ensembl_gene_id"</span>,<span class="st">"name"</span>)</span>
<span id="cb63-402"><a href="#cb63-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-403"><a href="#cb63-403" aria-hidden="true" tabindex="-1"></a><span class="co">#remove any dups in the reference set - two names for one ENSEMBL ID</span></span>
<span id="cb63-404"><a href="#cb63-404" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene_IDs,<span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb63-405"><a href="#cb63-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-406"><a href="#cb63-406" aria-hidden="true" tabindex="-1"></a>celltype_DEGs <span class="ot">&lt;-</span></span>
<span id="cb63-407"><a href="#cb63-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(celltype_DEGs,</span>
<span id="cb63-408"><a href="#cb63-408" aria-hidden="true" tabindex="-1"></a>         <span class="cf">function</span>(i){<span class="fu">setDT</span>(i);<span class="fu">setkey</span>(i,name);</span>
<span id="cb63-409"><a href="#cb63-409" aria-hidden="true" tabindex="-1"></a>           i[,gene_name<span class="sc">:</span><span class="er">=</span>gene_IDs[i, on<span class="ot">=</span>.(name),x.hgnc_symbol]];</span>
<span id="cb63-410"><a href="#cb63-410" aria-hidden="true" tabindex="-1"></a>           <span class="fu">setnames</span>(i,<span class="st">"name"</span>,<span class="st">"ensembl_id"</span>);<span class="fu">setnames</span>(i,<span class="st">"gene_name"</span>,<span class="st">"HGNC"</span>)})</span>
<span id="cb63-411"><a href="#cb63-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-412"><a href="#cb63-412" aria-hidden="true" tabindex="-1"></a><span class="co">#get gene names for all genes</span></span>
<span id="cb63-413"><a href="#cb63-413" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_de)<span class="sc">$</span>name)</span>
<span id="cb63-414"><a href="#cb63-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-415"><a href="#cb63-415" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> ensembldb<span class="sc">::</span><span class="fu">select</span>(EnsDb.Hsapiens.v79, <span class="at">keys=</span> genes,</span>
<span id="cb63-416"><a href="#cb63-416" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"GENEID"</span>,</span>
<span id="cb63-417"><a href="#cb63-417" aria-hidden="true" tabindex="-1"></a>                              <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GENEID"</span>,<span class="st">"SYMBOL"</span>))</span>
<span id="cb63-418"><a href="#cb63-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-419"><a href="#cb63-419" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_IDs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>,<span class="st">"hgnc_symbol"</span>)</span>
<span id="cb63-420"><a href="#cb63-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-421"><a href="#cb63-421" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(gene_IDs)</span>
<span id="cb63-422"><a href="#cb63-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-423"><a href="#cb63-423" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setnames</span>(gene_IDs,<span class="st">"ensembl_gene_id"</span>,<span class="st">"name"</span>)</span>
<span id="cb63-424"><a href="#cb63-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-425"><a href="#cb63-425" aria-hidden="true" tabindex="-1"></a><span class="co">#remove any dups in the reference set - two names for one ENSEMBL ID</span></span>
<span id="cb63-426"><a href="#cb63-426" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene_IDs,<span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb63-427"><a href="#cb63-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-428"><a href="#cb63-428" aria-hidden="true" tabindex="-1"></a>celltype_de <span class="ot">&lt;-</span></span>
<span id="cb63-429"><a href="#cb63-429" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(celltype_de,</span>
<span id="cb63-430"><a href="#cb63-430" aria-hidden="true" tabindex="-1"></a>         <span class="cf">function</span>(i){<span class="fu">setDT</span>(i);<span class="fu">setkey</span>(i,name);</span>
<span id="cb63-431"><a href="#cb63-431" aria-hidden="true" tabindex="-1"></a>           i[,gene_name<span class="sc">:</span><span class="er">=</span>gene_IDs[i, on<span class="ot">=</span>.(name),x.hgnc_symbol]];</span>
<span id="cb63-432"><a href="#cb63-432" aria-hidden="true" tabindex="-1"></a>           <span class="fu">setnames</span>(i,<span class="st">"name"</span>,<span class="st">"ensembl_id"</span>);<span class="fu">setnames</span>(i,<span class="st">"gene_name"</span>,<span class="st">"HGNC"</span>)})</span>
<span id="cb63-433"><a href="#cb63-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-434"><a href="#cb63-434" aria-hidden="true" tabindex="-1"></a>pseudobulk_results <span class="ot">&lt;-</span> (<span class="fu">list</span>(<span class="st">"celltype_DEGs"</span><span class="ot">=</span>celltype_DEGs,</span>
<span id="cb63-435"><a href="#cb63-435" aria-hidden="true" tabindex="-1"></a>            <span class="st">"celltype_all_genes"</span><span class="ot">=</span>celltype_de,</span>
<span id="cb63-436"><a href="#cb63-436" aria-hidden="true" tabindex="-1"></a>            <span class="st">"celltype_counts"</span><span class="ot">=</span>counts_celltypes))</span>
<span id="cb63-437"><a href="#cb63-437" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-438"><a href="#cb63-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-439"><a href="#cb63-439" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Save pseudobulk results --&gt;</span></span>
<span id="cb63-440"><a href="#cb63-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-443"><a href="#cb63-443" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-444"><a href="#cb63-444" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-445"><a href="#cb63-445" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-446"><a href="#cb63-446" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(pseudobulk_results, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/pseudobulk_results.rds"</span>))</span>
<span id="cb63-447"><a href="#cb63-447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-448"><a href="#cb63-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-449"><a href="#cb63-449" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- # try 2 --&gt;</span></span>
<span id="cb63-450"><a href="#cb63-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-451"><a href="#cb63-451" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Try using `region` arg in `make_pseudobulk` --&gt;</span></span>
<span id="cb63-452"><a href="#cb63-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-453"><a href="#cb63-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-456"><a href="#cb63-456" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-457"><a href="#cb63-457" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-458"><a href="#cb63-458" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-459"><a href="#cb63-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-460"><a href="#cb63-460" aria-hidden="true" tabindex="-1"></a><span class="co"># inputs</span></span>
<span id="cb63-461"><a href="#cb63-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-462"><a href="#cb63-462" aria-hidden="true" tabindex="-1"></a>SCE <span class="ot">&lt;-</span> sce</span>
<span id="cb63-463"><a href="#cb63-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-464"><a href="#cb63-464" aria-hidden="true" tabindex="-1"></a>celltype_ID <span class="ot">&lt;-</span> <span class="st">"ident"</span></span>
<span id="cb63-465"><a href="#cb63-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-466"><a href="#cb63-466" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="er">~</span> diagnosis</span>
<span id="cb63-467"><a href="#cb63-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-468"><a href="#cb63-468" aria-hidden="true" tabindex="-1"></a>pseudobulk_ID <span class="ot">=</span> <span class="st">"manifest"</span></span>
<span id="cb63-469"><a href="#cb63-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-470"><a href="#cb63-470" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="st">"diagnosis"</span></span>
<span id="cb63-471"><a href="#cb63-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-472"><a href="#cb63-472" aria-hidden="true" tabindex="-1"></a>region <span class="ot">=</span> <span class="st">"prep"</span></span>
<span id="cb63-473"><a href="#cb63-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-474"><a href="#cb63-474" aria-hidden="true" tabindex="-1"></a>pb_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"prep"</span>, <span class="st">"diagnosis"</span>, <span class="st">"manifest"</span>)</span>
<span id="cb63-475"><a href="#cb63-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-476"><a href="#cb63-476" aria-hidden="true" tabindex="-1"></a>verbose <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb63-477"><a href="#cb63-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-478"><a href="#cb63-478" aria-hidden="true" tabindex="-1"></a>control <span class="ot">=</span> <span class="st">"Control"</span></span>
<span id="cb63-479"><a href="#cb63-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-480"><a href="#cb63-480" aria-hidden="true" tabindex="-1"></a>pval_adjust_method <span class="ot">=</span> <span class="st">"BH"</span></span>
<span id="cb63-481"><a href="#cb63-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-482"><a href="#cb63-482" aria-hidden="true" tabindex="-1"></a>adj_pval<span class="ot">=</span><span class="fl">0.05</span></span>
<span id="cb63-483"><a href="#cb63-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-484"><a href="#cb63-484" aria-hidden="true" tabindex="-1"></a>coef <span class="ot">=</span> <span class="st">"Case"</span></span>
<span id="cb63-485"><a href="#cb63-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-486"><a href="#cb63-486" aria-hidden="true" tabindex="-1"></a>folder <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/graphs/"</span>)</span>
<span id="cb63-487"><a href="#cb63-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-488"><a href="#cb63-488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-489"><a href="#cb63-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-492"><a href="#cb63-492" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-493"><a href="#cb63-493" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-494"><a href="#cb63-494" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-495"><a href="#cb63-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-496"><a href="#cb63-496" aria-hidden="true" tabindex="-1"></a><span class="co">#first format formula</span></span>
<span id="cb63-497"><a href="#cb63-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-498"><a href="#cb63-498" aria-hidden="true" tabindex="-1"></a>design_txt <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">deparse</span>(design,<span class="at">width.cutoff =</span> <span class="dv">500</span>),<span class="at">collapse=</span><span class="st">','</span>)</span>
<span id="cb63-499"><a href="#cb63-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-500"><a href="#cb63-500" aria-hidden="true" tabindex="-1"></a><span class="co">#make design formula minus anything before ~</span></span>
<span id="cb63-501"><a href="#cb63-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-502"><a href="#cb63-502" aria-hidden="true" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">gsub</span>(<span class="st">".*~"</span>,<span class="st">"~"</span>,design_txt))</span>
<span id="cb63-503"><a href="#cb63-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-504"><a href="#cb63-504" aria-hidden="true" tabindex="-1"></a><span class="co">#if y not specified take last value in design matrix</span></span>
<span id="cb63-505"><a href="#cb63-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-506"><a href="#cb63-506" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(y))</span>
<span id="cb63-507"><a href="#cb63-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-508"><a href="#cb63-508" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> design_txt[[<span class="fu">length</span>(design_txt)]]</span>
<span id="cb63-509"><a href="#cb63-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-510"><a href="#cb63-510" aria-hidden="true" tabindex="-1"></a><span class="co">#Check if continuous or categorical variable to be modeled</span></span>
<span id="cb63-511"><a href="#cb63-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-512"><a href="#cb63-512" aria-hidden="true" tabindex="-1"></a>y_contin <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb63-513"><a href="#cb63-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-514"><a href="#cb63-514" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.numeric</span>(SCE[[y]]))</span>
<span id="cb63-515"><a href="#cb63-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-516"><a href="#cb63-516" aria-hidden="true" tabindex="-1"></a>    y_contin <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb63-517"><a href="#cb63-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-518"><a href="#cb63-518" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-519"><a href="#cb63-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-522"><a href="#cb63-522" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-523"><a href="#cb63-523" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-524"><a href="#cb63-524" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-525"><a href="#cb63-525" aria-hidden="true" tabindex="-1"></a><span class="co">#get counts of each cell type</span></span>
<span id="cb63-526"><a href="#cb63-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-527"><a href="#cb63-527" aria-hidden="true" tabindex="-1"></a><span class="co">#counts(SCE) &lt;- as.matrix(counts(SCE))</span></span>
<span id="cb63-528"><a href="#cb63-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-529"><a href="#cb63-529" aria-hidden="true" tabindex="-1"></a>counts_celltypes <span class="ot">&lt;-</span> SCE[[celltype_ID]]</span>
<span id="cb63-530"><a href="#cb63-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-531"><a href="#cb63-531" aria-hidden="true" tabindex="-1"></a>counts_celltypes <span class="ot">&lt;-</span><span class="fu">as.vector</span>(<span class="fu">table</span>(counts_celltypes))</span>
<span id="cb63-532"><a href="#cb63-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-533"><a href="#cb63-533" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(counts_celltypes) <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">table</span>(SCE[[celltype_ID]]))</span>
<span id="cb63-534"><a href="#cb63-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-535"><a href="#cb63-535" aria-hidden="true" tabindex="-1"></a><span class="co"># Get pseudobulk values</span></span>
<span id="cb63-536"><a href="#cb63-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-537"><a href="#cb63-537" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(SCE[[celltype_ID]])</span>
<span id="cb63-538"><a href="#cb63-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-539"><a href="#cb63-539" aria-hidden="true" tabindex="-1"></a>pb_dat <span class="ot">&lt;-</span></span>
<span id="cb63-540"><a href="#cb63-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-541"><a href="#cb63-541" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(celltypes,<span class="cf">function</span>(x)</span>
<span id="cb63-542"><a href="#cb63-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-543"><a href="#cb63-543" aria-hidden="true" tabindex="-1"></a>        <span class="fu">make_pseudobulk</span>(SCE[,SCE[[celltype_ID]]<span class="sc">==</span>x],</span>
<span id="cb63-544"><a href="#cb63-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-545"><a href="#cb63-545" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pseudobulk_ID=</span>pseudobulk_ID,</span>
<span id="cb63-546"><a href="#cb63-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-547"><a href="#cb63-547" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pb_columns=</span>pb_columns))</span>
<span id="cb63-548"><a href="#cb63-548" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-549"><a href="#cb63-549" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pb_dat) <span class="ot">&lt;-</span> celltypes</span>
<span id="cb63-550"><a href="#cb63-550" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-551"><a href="#cb63-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-554"><a href="#cb63-554" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-555"><a href="#cb63-555" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-556"><a href="#cb63-556" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-557"><a href="#cb63-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-558"><a href="#cb63-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-559"><a href="#cb63-559" aria-hidden="true" tabindex="-1"></a><span class="co">#run edgeR LRT DE analysis</span></span>
<span id="cb63-560"><a href="#cb63-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-561"><a href="#cb63-561" aria-hidden="true" tabindex="-1"></a>celltype_de <span class="ot">&lt;-</span></span>
<span id="cb63-562"><a href="#cb63-562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-563"><a href="#cb63-563" aria-hidden="true" tabindex="-1"></a>    <span class="fu">de_analysis</span>(pb_dat,formula,<span class="at">y_name=</span>y,y_contin,coef,control,</span>
<span id="cb63-564"><a href="#cb63-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-565"><a href="#cb63-565" aria-hidden="true" tabindex="-1"></a>                    pval_adjust_method, adj_pval, verbose)</span>
<span id="cb63-566"><a href="#cb63-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-567"><a href="#cb63-567" aria-hidden="true" tabindex="-1"></a><span class="co">#get sig DEGs for each</span></span>
<span id="cb63-568"><a href="#cb63-568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-569"><a href="#cb63-569" aria-hidden="true" tabindex="-1"></a>celltype_DEGs <span class="ot">&lt;-</span> <span class="fu">lapply</span>(celltype_de, <span class="cf">function</span>(x) x[x<span class="sc">$</span>adj_pval<span class="sc">&lt;</span>adj_pval,])</span>
<span id="cb63-570"><a href="#cb63-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-571"><a href="#cb63-571" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span><span class="fu">lapply</span>(celltype_DEGs, <span class="cf">function</span>(x) x<span class="sc">$</span>name)</span>
<span id="cb63-572"><a href="#cb63-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-573"><a href="#cb63-573" aria-hidden="true" tabindex="-1"></a>unique_degs <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">unlist</span>(unique_genes))</span>
<span id="cb63-574"><a href="#cb63-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-575"><a href="#cb63-575" aria-hidden="true" tabindex="-1"></a><span class="co">#if no DEGs found break but return the DE analysis scores still</span></span>
<span id="cb63-576"><a href="#cb63-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-577"><a href="#cb63-577" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(unique_degs)<span class="sc">==</span><span class="dv">0</span>)</span>
<span id="cb63-578"><a href="#cb63-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-579"><a href="#cb63-579" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="st">"celltype_DEGs"</span><span class="ot">=</span>celltype_DEGs,</span>
<span id="cb63-580"><a href="#cb63-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-581"><a href="#cb63-581" aria-hidden="true" tabindex="-1"></a>                <span class="st">"celltype_all_genes"</span><span class="ot">=</span>celltype_de,</span>
<span id="cb63-582"><a href="#cb63-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-583"><a href="#cb63-583" aria-hidden="true" tabindex="-1"></a>                <span class="st">"celltype_counts"</span><span class="ot">=</span>counts_celltypes))</span>
<span id="cb63-584"><a href="#cb63-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-585"><a href="#cb63-585" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">isTRUE</span>(verbose))</span>
<span id="cb63-586"><a href="#cb63-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-587"><a href="#cb63-587" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">length</span>(unique_degs),<span class="st">" unique DEGs foundacross all cell types"</span>)</span>
<span id="cb63-588"><a href="#cb63-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-589"><a href="#cb63-589" aria-hidden="true" tabindex="-1"></a>celltype_all_genes_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_de,<span class="at">idcol =</span> T)</span>
<span id="cb63-590"><a href="#cb63-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-591"><a href="#cb63-591" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(celltype_all_genes_dt,<span class="st">".id"</span>,<span class="st">"celltype"</span>)</span>
<span id="cb63-592"><a href="#cb63-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-593"><a href="#cb63-593" aria-hidden="true" tabindex="-1"></a>celltype_DEGs_dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_DEGs,<span class="at">idcol =</span> T)</span>
<span id="cb63-594"><a href="#cb63-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-595"><a href="#cb63-595" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(celltype_DEGs_dt,<span class="st">".id"</span>,<span class="st">"celltype"</span>)</span>
<span id="cb63-596"><a href="#cb63-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-597"><a href="#cb63-597" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-598"><a href="#cb63-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-599"><a href="#cb63-599" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Changed palette arg in code to make this work --&gt;</span></span>
<span id="cb63-600"><a href="#cb63-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-603"><a href="#cb63-603" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-604"><a href="#cb63-604" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-605"><a href="#cb63-605" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-606"><a href="#cb63-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-607"><a href="#cb63-607" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">isFALSE</span>(folder)){</span>
<span id="cb63-608"><a href="#cb63-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-609"><a href="#cb63-609" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">isTRUE</span>(verbose))</span>
<span id="cb63-610"><a href="#cb63-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-611"><a href="#cb63-611" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">"Plotting the results of differential expression analysis"</span>)</span>
<span id="cb63-612"><a href="#cb63-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-613"><a href="#cb63-613" aria-hidden="true" tabindex="-1"></a>        <span class="co">#make plots for DE analysis</span></span>
<span id="cb63-614"><a href="#cb63-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-615"><a href="#cb63-615" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot_de_analysis</span>(pb_dat,y,celltype_DEGs_dt,celltype_all_genes_dt,</span>
<span id="cb63-616"><a href="#cb63-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-617"><a href="#cb63-617" aria-hidden="true" tabindex="-1"></a>                         counts_celltypes,folder)</span>
<span id="cb63-618"><a href="#cb63-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-619"><a href="#cb63-619" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb63-620"><a href="#cb63-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-621"><a href="#cb63-621" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-622"><a href="#cb63-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-625"><a href="#cb63-625" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-626"><a href="#cb63-626" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-627"><a href="#cb63-627" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-628"><a href="#cb63-628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-629"><a href="#cb63-629" aria-hidden="true" tabindex="-1"></a><span class="co">#get gene names for DEGs</span></span>
<span id="cb63-630"><a href="#cb63-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-631"><a href="#cb63-631" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_DEGs)<span class="sc">$</span>name)</span>
<span id="cb63-632"><a href="#cb63-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-633"><a href="#cb63-633" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> ensembldb<span class="sc">::</span><span class="fu">select</span>(EnsDb.Hsapiens.v79, <span class="at">keys=</span> genes,</span>
<span id="cb63-634"><a href="#cb63-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-635"><a href="#cb63-635" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"GENEID"</span>,</span>
<span id="cb63-636"><a href="#cb63-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-637"><a href="#cb63-637" aria-hidden="true" tabindex="-1"></a>                              <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GENEID"</span>,<span class="st">"SYMBOL"</span>))</span>
<span id="cb63-638"><a href="#cb63-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-639"><a href="#cb63-639" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_IDs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>,<span class="st">"hgnc_symbol"</span>)</span>
<span id="cb63-640"><a href="#cb63-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-641"><a href="#cb63-641" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(gene_IDs)</span>
<span id="cb63-642"><a href="#cb63-642" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-643"><a href="#cb63-643" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setnames</span>(gene_IDs,<span class="st">"ensembl_gene_id"</span>,<span class="st">"name"</span>)</span>
<span id="cb63-644"><a href="#cb63-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-645"><a href="#cb63-645" aria-hidden="true" tabindex="-1"></a><span class="co">#remove any dups in the reference set - two names for one ENSEMBL ID</span></span>
<span id="cb63-646"><a href="#cb63-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-647"><a href="#cb63-647" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene_IDs,<span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb63-648"><a href="#cb63-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-649"><a href="#cb63-649" aria-hidden="true" tabindex="-1"></a>celltype_DEGs <span class="ot">&lt;-</span></span>
<span id="cb63-650"><a href="#cb63-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-651"><a href="#cb63-651" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(celltype_DEGs,</span>
<span id="cb63-652"><a href="#cb63-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-653"><a href="#cb63-653" aria-hidden="true" tabindex="-1"></a>         <span class="cf">function</span>(i){<span class="fu">setDT</span>(i);<span class="fu">setkey</span>(i,name);</span>
<span id="cb63-654"><a href="#cb63-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-655"><a href="#cb63-655" aria-hidden="true" tabindex="-1"></a>           i[,gene_name<span class="sc">:</span><span class="er">=</span>gene_IDs[i, on<span class="ot">=</span>.(name),x.hgnc_symbol]];</span>
<span id="cb63-656"><a href="#cb63-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-657"><a href="#cb63-657" aria-hidden="true" tabindex="-1"></a>           <span class="fu">setnames</span>(i,<span class="st">"name"</span>,<span class="st">"ensembl_id"</span>);<span class="fu">setnames</span>(i,<span class="st">"gene_name"</span>,<span class="st">"HGNC"</span>)})</span>
<span id="cb63-658"><a href="#cb63-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-659"><a href="#cb63-659" aria-hidden="true" tabindex="-1"></a><span class="co">#get gene names for all genes</span></span>
<span id="cb63-660"><a href="#cb63-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-661"><a href="#cb63-661" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_de)<span class="sc">$</span>name)</span>
<span id="cb63-662"><a href="#cb63-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-663"><a href="#cb63-663" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> ensembldb<span class="sc">::</span><span class="fu">select</span>(EnsDb.Hsapiens.v79, <span class="at">keys=</span> genes,</span>
<span id="cb63-664"><a href="#cb63-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-665"><a href="#cb63-665" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"GENEID"</span>,</span>
<span id="cb63-666"><a href="#cb63-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-667"><a href="#cb63-667" aria-hidden="true" tabindex="-1"></a>                              <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GENEID"</span>,<span class="st">"SYMBOL"</span>))</span>
<span id="cb63-668"><a href="#cb63-668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-669"><a href="#cb63-669" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_IDs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>,<span class="st">"hgnc_symbol"</span>)</span>
<span id="cb63-670"><a href="#cb63-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-671"><a href="#cb63-671" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">as.data.table</span>(gene_IDs)</span>
<span id="cb63-672"><a href="#cb63-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-673"><a href="#cb63-673" aria-hidden="true" tabindex="-1"></a>data.table<span class="sc">::</span><span class="fu">setnames</span>(gene_IDs,<span class="st">"ensembl_gene_id"</span>,<span class="st">"name"</span>)</span>
<span id="cb63-674"><a href="#cb63-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-675"><a href="#cb63-675" aria-hidden="true" tabindex="-1"></a><span class="co">#remove any dups in the reference set - two names for one ENSEMBL ID</span></span>
<span id="cb63-676"><a href="#cb63-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-677"><a href="#cb63-677" aria-hidden="true" tabindex="-1"></a>gene_IDs <span class="ot">&lt;-</span> <span class="fu">unique</span>(gene_IDs,<span class="at">by=</span><span class="st">"name"</span>)</span>
<span id="cb63-678"><a href="#cb63-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-679"><a href="#cb63-679" aria-hidden="true" tabindex="-1"></a>celltype_de <span class="ot">&lt;-</span></span>
<span id="cb63-680"><a href="#cb63-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-681"><a href="#cb63-681" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lapply</span>(celltype_de,</span>
<span id="cb63-682"><a href="#cb63-682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-683"><a href="#cb63-683" aria-hidden="true" tabindex="-1"></a>         <span class="cf">function</span>(i){<span class="fu">setDT</span>(i);<span class="fu">setkey</span>(i,name);</span>
<span id="cb63-684"><a href="#cb63-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-685"><a href="#cb63-685" aria-hidden="true" tabindex="-1"></a>           i[,gene_name<span class="sc">:</span><span class="er">=</span>gene_IDs[i, on<span class="ot">=</span>.(name),x.hgnc_symbol]];</span>
<span id="cb63-686"><a href="#cb63-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-687"><a href="#cb63-687" aria-hidden="true" tabindex="-1"></a>           <span class="fu">setnames</span>(i,<span class="st">"name"</span>,<span class="st">"ensembl_id"</span>);<span class="fu">setnames</span>(i,<span class="st">"gene_name"</span>,<span class="st">"HGNC"</span>)})</span>
<span id="cb63-688"><a href="#cb63-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-689"><a href="#cb63-689" aria-hidden="true" tabindex="-1"></a>pseudobulk_results <span class="ot">&lt;-</span> (<span class="fu">list</span>(<span class="st">"celltype_DEGs"</span><span class="ot">=</span>celltype_DEGs,</span>
<span id="cb63-690"><a href="#cb63-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-691"><a href="#cb63-691" aria-hidden="true" tabindex="-1"></a>            <span class="st">"celltype_all_genes"</span><span class="ot">=</span>celltype_de,</span>
<span id="cb63-692"><a href="#cb63-692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-693"><a href="#cb63-693" aria-hidden="true" tabindex="-1"></a>            <span class="st">"celltype_counts"</span><span class="ot">=</span>counts_celltypes))</span>
<span id="cb63-694"><a href="#cb63-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-695"><a href="#cb63-695" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-696"><a href="#cb63-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-697"><a href="#cb63-697" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Save pseudobulk results --&gt;</span></span>
<span id="cb63-698"><a href="#cb63-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-701"><a href="#cb63-701" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-702"><a href="#cb63-702" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-703"><a href="#cb63-703" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-704"><a href="#cb63-704" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(pseudobulk_results, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/pseudobulk_results.rds"</span>))</span>
<span id="cb63-705"><a href="#cb63-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-706"><a href="#cb63-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-707"><a href="#cb63-707" aria-hidden="true" tabindex="-1"></a><span class="fu"># Write from scratch</span></span>
<span id="cb63-708"><a href="#cb63-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-709"><a href="#cb63-709" aria-hidden="true" tabindex="-1"></a>Here I'll try writing my own version of <span class="in">`de_analysis()`</span></span>
<span id="cb63-710"><a href="#cb63-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-711"><a href="#cb63-711" aria-hidden="true" tabindex="-1"></a>I'll base it off the <span class="in">`edgeR`</span> doc <span class="co">[</span><span class="ot">here</span><span class="co">](https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf)</span></span>
<span id="cb63-712"><a href="#cb63-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-715"><a href="#cb63-715" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-716"><a href="#cb63-716" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>donor <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"donor-"</span>, sce<span class="sc">$</span>individual)</span>
<span id="cb63-717"><a href="#cb63-717" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(sce<span class="sc">$</span>diagnosis) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AD"</span>, <span class="st">"Control"</span>)</span>
<span id="cb63-718"><a href="#cb63-718" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-719"><a href="#cb63-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-722"><a href="#cb63-722" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-723"><a href="#cb63-723" aria-hidden="true" tabindex="-1"></a><span class="co"># inputs</span></span>
<span id="cb63-724"><a href="#cb63-724" aria-hidden="true" tabindex="-1"></a>SCE <span class="ot">&lt;-</span> sce</span>
<span id="cb63-725"><a href="#cb63-725" aria-hidden="true" tabindex="-1"></a>celltype_ID <span class="ot">&lt;-</span> <span class="st">"ident"</span></span>
<span id="cb63-726"><a href="#cb63-726" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="er">~</span> diagnosis</span>
<span id="cb63-727"><a href="#cb63-727" aria-hidden="true" tabindex="-1"></a>pseudobulk_ID <span class="ot">=</span> <span class="st">"manifest"</span></span>
<span id="cb63-728"><a href="#cb63-728" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> <span class="st">"diagnosis"</span></span>
<span id="cb63-729"><a href="#cb63-729" aria-hidden="true" tabindex="-1"></a>region <span class="ot">=</span> <span class="st">"prep"</span></span>
<span id="cb63-730"><a href="#cb63-730" aria-hidden="true" tabindex="-1"></a><span class="co">#pb_columns &lt;- c("prep", "diagnosis", "manifest")</span></span>
<span id="cb63-731"><a href="#cb63-731" aria-hidden="true" tabindex="-1"></a>pb_columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"manifest"</span>)</span>
<span id="cb63-732"><a href="#cb63-732" aria-hidden="true" tabindex="-1"></a>verbose <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb63-733"><a href="#cb63-733" aria-hidden="true" tabindex="-1"></a>control <span class="ot">=</span> <span class="st">"Control"</span></span>
<span id="cb63-734"><a href="#cb63-734" aria-hidden="true" tabindex="-1"></a>pval_adjust_method <span class="ot">=</span> <span class="st">"BH"</span></span>
<span id="cb63-735"><a href="#cb63-735" aria-hidden="true" tabindex="-1"></a>adj_pval<span class="ot">=</span><span class="fl">0.05</span></span>
<span id="cb63-736"><a href="#cb63-736" aria-hidden="true" tabindex="-1"></a>coef <span class="ot">=</span> <span class="st">"Case"</span></span>
<span id="cb63-737"><a href="#cb63-737" aria-hidden="true" tabindex="-1"></a>folder <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/graphs/"</span>)</span>
<span id="cb63-738"><a href="#cb63-738" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-739"><a href="#cb63-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-742"><a href="#cb63-742" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-743"><a href="#cb63-743" aria-hidden="true" tabindex="-1"></a><span class="co">#first format formula</span></span>
<span id="cb63-744"><a href="#cb63-744" aria-hidden="true" tabindex="-1"></a>design_txt <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">deparse</span>(design,<span class="at">width.cutoff =</span> <span class="dv">500</span>),<span class="at">collapse=</span><span class="st">','</span>)</span>
<span id="cb63-745"><a href="#cb63-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-746"><a href="#cb63-746" aria-hidden="true" tabindex="-1"></a><span class="co">#make design formula minus anything before ~</span></span>
<span id="cb63-747"><a href="#cb63-747" aria-hidden="true" tabindex="-1"></a>formula <span class="ot">&lt;-</span> <span class="fu">as.formula</span>(<span class="fu">gsub</span>(<span class="st">".*~"</span>,<span class="st">"~"</span>,design_txt))</span>
<span id="cb63-748"><a href="#cb63-748" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-749"><a href="#cb63-749" aria-hidden="true" tabindex="-1"></a><span class="co">#if y not specified take last value in design matrix</span></span>
<span id="cb63-750"><a href="#cb63-750" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(y))</span>
<span id="cb63-751"><a href="#cb63-751" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> design_txt[[<span class="fu">length</span>(design_txt)]]</span>
<span id="cb63-752"><a href="#cb63-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-753"><a href="#cb63-753" aria-hidden="true" tabindex="-1"></a><span class="co">#Check if continuous or categorical variable to be modeled</span></span>
<span id="cb63-754"><a href="#cb63-754" aria-hidden="true" tabindex="-1"></a>y_contin <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb63-755"><a href="#cb63-755" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.numeric</span>(SCE[[y]]))</span>
<span id="cb63-756"><a href="#cb63-756" aria-hidden="true" tabindex="-1"></a>    y_contin <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb63-757"><a href="#cb63-757" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-758"><a href="#cb63-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-759"><a href="#cb63-759" aria-hidden="true" tabindex="-1"></a>Note that the pseudobulk compute takes a few minutes to run, so I'll disable eval on compilation.</span>
<span id="cb63-760"><a href="#cb63-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-763"><a href="#cb63-763" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-764"><a href="#cb63-764" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-765"><a href="#cb63-765" aria-hidden="true" tabindex="-1"></a><span class="co">#get counts of each cell type</span></span>
<span id="cb63-766"><a href="#cb63-766" aria-hidden="true" tabindex="-1"></a><span class="co">#counts(SCE) &lt;- as.matrix(counts(SCE))</span></span>
<span id="cb63-767"><a href="#cb63-767" aria-hidden="true" tabindex="-1"></a>counts_celltypes <span class="ot">&lt;-</span> SCE[[celltype_ID]]</span>
<span id="cb63-768"><a href="#cb63-768" aria-hidden="true" tabindex="-1"></a>counts_celltypes <span class="ot">&lt;-</span><span class="fu">as.vector</span>(<span class="fu">table</span>(counts_celltypes))</span>
<span id="cb63-769"><a href="#cb63-769" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(counts_celltypes) <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">table</span>(SCE[[celltype_ID]]))</span>
<span id="cb63-770"><a href="#cb63-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-771"><a href="#cb63-771" aria-hidden="true" tabindex="-1"></a><span class="co"># Get pseudobulk values</span></span>
<span id="cb63-772"><a href="#cb63-772" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(SCE[[celltype_ID]])</span>
<span id="cb63-773"><a href="#cb63-773" aria-hidden="true" tabindex="-1"></a>pb_dat <span class="ot">&lt;-</span></span>
<span id="cb63-774"><a href="#cb63-774" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(celltypes,<span class="cf">function</span>(x)</span>
<span id="cb63-775"><a href="#cb63-775" aria-hidden="true" tabindex="-1"></a>        <span class="fu">make_pseudobulk</span>(SCE[,SCE[[celltype_ID]]<span class="sc">==</span>x],</span>
<span id="cb63-776"><a href="#cb63-776" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pseudobulk_ID=</span>pseudobulk_ID,</span>
<span id="cb63-777"><a href="#cb63-777" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pb_columns=</span>pb_columns))</span>
<span id="cb63-778"><a href="#cb63-778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-779"><a href="#cb63-779" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pb_dat) <span class="ot">&lt;-</span> celltypes</span>
<span id="cb63-780"><a href="#cb63-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-781"><a href="#cb63-781" aria-hidden="true" tabindex="-1"></a><span class="co"># the pseudobulk values take a while to compute, so I'll save the output</span></span>
<span id="cb63-782"><a href="#cb63-782" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(pb_dat, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_values.rds"</span>))</span>
<span id="cb63-783"><a href="#cb63-783" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-784"><a href="#cb63-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-787"><a href="#cb63-787" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-788"><a href="#cb63-788" aria-hidden="true" tabindex="-1"></a><span class="co"># load values</span></span>
<span id="cb63-789"><a href="#cb63-789" aria-hidden="true" tabindex="-1"></a>pb_dat <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_values.rds"</span>))</span>
<span id="cb63-790"><a href="#cb63-790" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-791"><a href="#cb63-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-792"><a href="#cb63-792" aria-hidden="true" tabindex="-1"></a>Hannah is after the pseudobulk values themselves, so I'll add the celltypes and save that data</span>
<span id="cb63-793"><a href="#cb63-793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-796"><a href="#cb63-796" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-797"><a href="#cb63-797" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-798"><a href="#cb63-798" aria-hidden="true" tabindex="-1"></a>pb_dat_per_individual <span class="ot">&lt;-</span></span>
<span id="cb63-799"><a href="#cb63-799" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(celltypes,<span class="cf">function</span>(x)</span>
<span id="cb63-800"><a href="#cb63-800" aria-hidden="true" tabindex="-1"></a>        <span class="fu">make_pseudobulk</span>(SCE[,SCE[[celltype_ID]]<span class="sc">==</span>x],</span>
<span id="cb63-801"><a href="#cb63-801" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pseudobulk_ID=</span><span class="st">"individual"</span>))</span>
<span id="cb63-802"><a href="#cb63-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-803"><a href="#cb63-803" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pb_dat_per_individual) <span class="ot">&lt;-</span> celltypes</span>
<span id="cb63-804"><a href="#cb63-804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-805"><a href="#cb63-805" aria-hidden="true" tabindex="-1"></a>pseudo_vals <span class="ot">&lt;-</span> <span class="fu">map2</span>(pb_dat_per_individual, <span class="fu">names</span>(pb_dat_per_individual), <span class="sc">~</span> .x<span class="sc">$</span>sumDat <span class="sc">%&gt;%</span></span>
<span id="cb63-806"><a href="#cb63-806" aria-hidden="true" tabindex="-1"></a>              <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb63-807"><a href="#cb63-807" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> .y) <span class="sc">%&gt;%</span></span>
<span id="cb63-808"><a href="#cb63-808" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb63-809"><a href="#cb63-809" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">list_rbind</span>()</span>
<span id="cb63-810"><a href="#cb63-810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-811"><a href="#cb63-811" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(pseudo_vals, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_values_per_individual.csv"</span>))</span>
<span id="cb63-812"><a href="#cb63-812" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-813"><a href="#cb63-813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-814"><a href="#cb63-814" aria-hidden="true" tabindex="-1"></a>First part is prep for LRT, this is done per cell type via <span class="in">`purrr`</span></span>
<span id="cb63-815"><a href="#cb63-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-818"><a href="#cb63-818" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-819"><a href="#cb63-819" aria-hidden="true" tabindex="-1"></a><span class="co"># Get DGE list per celltype</span></span>
<span id="cb63-820"><a href="#cb63-820" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(pb_dat, <span class="sc">~</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(<span class="at">counts =</span> .x<span class="sc">$</span>sumDat,</span>
<span id="cb63-821"><a href="#cb63-821" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> .x<span class="sc">$</span>annot_pb<span class="sc">$</span>group))</span>
<span id="cb63-822"><a href="#cb63-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-823"><a href="#cb63-823" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate normalization factors to scale the raw library sizes.</span></span>
<span id="cb63-824"><a href="#cb63-824" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(y, <span class="sc">~</span> edgeR<span class="sc">::</span><span class="fu">calcNormFactors</span>(.x, <span class="at">method =</span> <span class="st">'TMM'</span>))</span>
<span id="cb63-825"><a href="#cb63-825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-826"><a href="#cb63-826" aria-hidden="true" tabindex="-1"></a><span class="co"># Build design matrix</span></span>
<span id="cb63-827"><a href="#cb63-827" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(pb_dat, <span class="sc">~</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> group, <span class="at">data =</span> .x<span class="sc">$</span>annot_pb))</span>
<span id="cb63-828"><a href="#cb63-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-829"><a href="#cb63-829" aria-hidden="true" tabindex="-1"></a><span class="co"># Maximizes the negative binomial likelihood to give the estimate of the</span></span>
<span id="cb63-830"><a href="#cb63-830" aria-hidden="true" tabindex="-1"></a><span class="co"># common,trended and tagwise dispersions across all tags.</span></span>
<span id="cb63-831"><a href="#cb63-831" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map2</span>(y, design, <span class="sc">~</span> edgeR<span class="sc">::</span><span class="fu">estimateDisp</span>(.x, .y))</span>
<span id="cb63-832"><a href="#cb63-832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-833"><a href="#cb63-833" aria-hidden="true" tabindex="-1"></a><span class="co"># LRT</span></span>
<span id="cb63-834"><a href="#cb63-834" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map2</span>(y, design, <span class="sc">~</span> edgeR<span class="sc">::</span><span class="fu">glmFit</span>(.x, <span class="at">design =</span> .y))</span>
<span id="cb63-835"><a href="#cb63-835" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-836"><a href="#cb63-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-837"><a href="#cb63-837" aria-hidden="true" tabindex="-1"></a>Now there's a need to think about what comparisons to run.</span>
<span id="cb63-838"><a href="#cb63-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-839"><a href="#cb63-839" aria-hidden="true" tabindex="-1"></a>For now I think I'll write a more complex approach to run all possible comparisons from a arbitrarily long list</span>
<span id="cb63-840"><a href="#cb63-840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-841"><a href="#cb63-841" aria-hidden="true" tabindex="-1"></a>First I'll make a contrast matrix of all combinations</span>
<span id="cb63-842"><a href="#cb63-842" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-845"><a href="#cb63-845" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-846"><a href="#cb63-846" aria-hidden="true" tabindex="-1"></a><span class="co"># Get number group levels</span></span>
<span id="cb63-847"><a href="#cb63-847" aria-hidden="true" tabindex="-1"></a>ncls <span class="ot">&lt;-</span> <span class="fu">nlevels</span>(sce<span class="sc">$</span>group)</span>
<span id="cb63-848"><a href="#cb63-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-849"><a href="#cb63-849" aria-hidden="true" tabindex="-1"></a>get_contrasts <span class="ot">&lt;-</span> <span class="cf">function</span>(design_matrix, group_vector) {</span>
<span id="cb63-850"><a href="#cb63-850" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb63-851"><a href="#cb63-851" aria-hidden="true" tabindex="-1"></a>  num_groups <span class="ot">&lt;-</span> <span class="fu">nlevels</span>(group_vector)</span>
<span id="cb63-852"><a href="#cb63-852" aria-hidden="true" tabindex="-1"></a>  contr <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">matrix</span>(<span class="dv">1</span> <span class="sc">/</span> (<span class="dv">1</span> <span class="sc">-</span> num_groups), num_groups, num_groups),</span>
<span id="cb63-853"><a href="#cb63-853" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="fu">ncol</span>(design_matrix) <span class="sc">-</span> num_groups, num_groups))</span>
<span id="cb63-854"><a href="#cb63-854" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-855"><a href="#cb63-855" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(contr) <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb63-856"><a href="#cb63-856" aria-hidden="true" tabindex="-1"></a>  contr[<span class="dv">1</span>,] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb63-857"><a href="#cb63-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-858"><a href="#cb63-858" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(contr) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(design_matrix)</span>
<span id="cb63-859"><a href="#cb63-859" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(contr) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster"</span>, <span class="fu">levels</span>(group_vector))</span>
<span id="cb63-860"><a href="#cb63-860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-861"><a href="#cb63-861" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(contr)</span>
<span id="cb63-862"><a href="#cb63-862" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb63-863"><a href="#cb63-863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-864"><a href="#cb63-864" aria-hidden="true" tabindex="-1"></a>contr <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(design, <span class="sc">~</span> <span class="fu">get_contrasts</span>(.x, sce<span class="sc">$</span>group))</span>
<span id="cb63-865"><a href="#cb63-865" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-866"><a href="#cb63-866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-867"><a href="#cb63-867" aria-hidden="true" tabindex="-1"></a>Then apply the LRT model per celltype</span>
<span id="cb63-868"><a href="#cb63-868" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-871"><a href="#cb63-871" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-872"><a href="#cb63-872" aria-hidden="true" tabindex="-1"></a>apply_lrt_to_groups <span class="ot">&lt;-</span></span>
<span id="cb63-873"><a href="#cb63-873" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(glmfit_object,</span>
<span id="cb63-874"><a href="#cb63-874" aria-hidden="true" tabindex="-1"></a>           contrast_matrix,</span>
<span id="cb63-875"><a href="#cb63-875" aria-hidden="true" tabindex="-1"></a>           group_levels) {</span>
<span id="cb63-876"><a href="#cb63-876" aria-hidden="true" tabindex="-1"></a>    lrt <span class="ot">&lt;-</span></span>
<span id="cb63-877"><a href="#cb63-877" aria-hidden="true" tabindex="-1"></a>      purrr<span class="sc">::</span><span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(contrast_matrix), <span class="sc">~</span> {</span>
<span id="cb63-878"><a href="#cb63-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-879"><a href="#cb63-879" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Apply LRT per comparison</span></span>
<span id="cb63-880"><a href="#cb63-880" aria-hidden="true" tabindex="-1"></a>        lrt <span class="ot">&lt;-</span></span>
<span id="cb63-881"><a href="#cb63-881" aria-hidden="true" tabindex="-1"></a>          edgeR<span class="sc">::</span><span class="fu">glmLRT</span>(glmfit_object, <span class="at">contrast =</span> contrast_matrix[, .x])</span>
<span id="cb63-882"><a href="#cb63-882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-883"><a href="#cb63-883" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Set name of comparison</span></span>
<span id="cb63-884"><a href="#cb63-884" aria-hidden="true" tabindex="-1"></a>        lrt<span class="sc">$</span>comparison <span class="ot">&lt;-</span></span>
<span id="cb63-885"><a href="#cb63-885" aria-hidden="true" tabindex="-1"></a>          <span class="fu">paste0</span>(<span class="st">"group"</span>, group_levels[.x], <span class="st">"_vs_others"</span>)</span>
<span id="cb63-886"><a href="#cb63-886" aria-hidden="true" tabindex="-1"></a>        lrt</span>
<span id="cb63-887"><a href="#cb63-887" aria-hidden="true" tabindex="-1"></a>      })</span>
<span id="cb63-888"><a href="#cb63-888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-889"><a href="#cb63-889" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set names of comparison to list</span></span>
<span id="cb63-890"><a href="#cb63-890" aria-hidden="true" tabindex="-1"></a>    lrt <span class="ot">&lt;-</span></span>
<span id="cb63-891"><a href="#cb63-891" aria-hidden="true" tabindex="-1"></a>      purrr<span class="sc">::</span><span class="fu">set_names</span>(lrt, <span class="fu">map_chr</span>(group_levels, <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"group"</span>, .x, <span class="st">"_vs_others"</span>)))</span>
<span id="cb63-892"><a href="#cb63-892" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb63-893"><a href="#cb63-893" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(lrt)</span>
<span id="cb63-894"><a href="#cb63-894" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb63-895"><a href="#cb63-895" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-896"><a href="#cb63-896" aria-hidden="true" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map2</span>(fit, contr, <span class="sc">~</span> <span class="fu">apply_lrt_to_groups</span>(.x, .y, <span class="fu">levels</span>(sce<span class="sc">$</span>group)))</span>
<span id="cb63-897"><a href="#cb63-897" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-898"><a href="#cb63-898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-901"><a href="#cb63-901" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-902"><a href="#cb63-902" aria-hidden="true" tabindex="-1"></a><span class="co"># Check results</span></span>
<span id="cb63-903"><a href="#cb63-903" aria-hidden="true" tabindex="-1"></a><span class="fu">topTags</span>(lrt<span class="sc">$</span>Endothelial<span class="sc">$</span>groupAD_P_oxford_dunno_vs_others)</span>
<span id="cb63-904"><a href="#cb63-904" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-905"><a href="#cb63-905" aria-hidden="true" tabindex="-1"></a><span class="co"># Get number of DE genes</span></span>
<span id="cb63-906"><a href="#cb63-906" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(lrt, <span class="sc">~</span> <span class="fu">lapply</span>(<span class="fu">lapply</span>(.x, decideTestsDGE), summary) <span class="sc">%&gt;%</span></span>
<span id="cb63-907"><a href="#cb63-907" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">do.call</span>(<span class="st">"cbind"</span>, .))</span>
<span id="cb63-908"><a href="#cb63-908" aria-hidden="true" tabindex="-1"></a>dt<span class="sc">$</span>Endothelial</span>
<span id="cb63-909"><a href="#cb63-909" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-910"><a href="#cb63-910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-911"><a href="#cb63-911" aria-hidden="true" tabindex="-1"></a>Clean and extract results from lrt, including getting adjusted pvals</span>
<span id="cb63-912"><a href="#cb63-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-915"><a href="#cb63-915" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-916"><a href="#cb63-916" aria-hidden="true" tabindex="-1"></a>clean_lrt_results <span class="ot">&lt;-</span></span>
<span id="cb63-917"><a href="#cb63-917" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(lrt_results,</span>
<span id="cb63-918"><a href="#cb63-918" aria-hidden="true" tabindex="-1"></a>           <span class="at">pval_adjust_method =</span> <span class="st">"BH"</span>,</span>
<span id="cb63-919"><a href="#cb63-919" aria-hidden="true" tabindex="-1"></a>           <span class="at">verbose =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb63-920"><a href="#cb63-920" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get results table</span></span>
<span id="cb63-921"><a href="#cb63-921" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">&lt;-</span> lrt_results<span class="sc">$</span>table</span>
<span id="cb63-922"><a href="#cb63-922" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb63-923"><a href="#cb63-923" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add adj p-values</span></span>
<span id="cb63-924"><a href="#cb63-924" aria-hidden="true" tabindex="-1"></a>    pvals<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">p.adjust</span>(pvals<span class="sc">$</span>PValue,</span>
<span id="cb63-925"><a href="#cb63-925" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">method =</span> pval_adjust_method)</span>
<span id="cb63-926"><a href="#cb63-926" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb63-927"><a href="#cb63-927" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Make rownames a column</span></span>
<span id="cb63-928"><a href="#cb63-928" aria-hidden="true" tabindex="-1"></a>    pvals<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pvals)</span>
<span id="cb63-929"><a href="#cb63-929" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pvals) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb63-930"><a href="#cb63-930" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb63-931"><a href="#cb63-931" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Print number of DEGs found if verbose</span></span>
<span id="cb63-932"><a href="#cb63-932" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (verbose) {</span>
<span id="cb63-933"><a href="#cb63-933" aria-hidden="true" tabindex="-1"></a>      numDEGs <span class="ot">&lt;-</span> <span class="fu">nrow</span>(pvals[pvals<span class="sc">$</span>adj_pval <span class="sc">&lt;</span> adj_pval, ])</span>
<span id="cb63-934"><a href="#cb63-934" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(numDEGs, <span class="st">" DEGs found"</span>)</span>
<span id="cb63-935"><a href="#cb63-935" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb63-936"><a href="#cb63-936" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb63-937"><a href="#cb63-937" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(pvals)</span>
<span id="cb63-938"><a href="#cb63-938" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb63-939"><a href="#cb63-939" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-940"><a href="#cb63-940" aria-hidden="true" tabindex="-1"></a>lrt_clean <span class="ot">&lt;-</span> <span class="fu">map</span>(lrt, <span class="sc">~</span> <span class="fu">map</span>(.x, <span class="sc">~</span><span class="fu">clean_lrt_results</span>(.x)))</span>
<span id="cb63-941"><a href="#cb63-941" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-942"><a href="#cb63-942" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-943"><a href="#cb63-943" aria-hidden="true" tabindex="-1"></a>Get significant genes and generate plots</span>
<span id="cb63-944"><a href="#cb63-944" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-945"><a href="#cb63-945" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NOTE: plotting function needs to be updated - not currently working</span>
<span id="cb63-946"><a href="#cb63-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-949"><a href="#cb63-949" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-950"><a href="#cb63-950" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-951"><a href="#cb63-951" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-952"><a href="#cb63-952" aria-hidden="true" tabindex="-1"></a>celltype_de <span class="ot">=</span> lrt_clean<span class="sc">$</span>Endothelial</span>
<span id="cb63-953"><a href="#cb63-953" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-954"><a href="#cb63-954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-957"><a href="#cb63-957" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-958"><a href="#cb63-958" aria-hidden="true" tabindex="-1"></a>get_sig_genes_and_plots <span class="ot">&lt;-</span></span>
<span id="cb63-959"><a href="#cb63-959" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(celltype_de,</span>
<span id="cb63-960"><a href="#cb63-960" aria-hidden="true" tabindex="-1"></a>           <span class="at">pval_adjustment =</span> <span class="fl">0.05</span>,</span>
<span id="cb63-961"><a href="#cb63-961" aria-hidden="true" tabindex="-1"></a>           celltype_counts,</span>
<span id="cb63-962"><a href="#cb63-962" aria-hidden="true" tabindex="-1"></a>           <span class="at">plot_folder =</span> <span class="cn">FALSE</span>,</span>
<span id="cb63-963"><a href="#cb63-963" aria-hidden="true" tabindex="-1"></a>           <span class="at">verbose =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb63-964"><a href="#cb63-964" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-965"><a href="#cb63-965" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get sig DEGs for each</span></span>
<span id="cb63-966"><a href="#cb63-966" aria-hidden="true" tabindex="-1"></a>    celltype_DEGs <span class="ot">&lt;-</span></span>
<span id="cb63-967"><a href="#cb63-967" aria-hidden="true" tabindex="-1"></a>      <span class="fu">lapply</span>(celltype_de, <span class="cf">function</span>(x)</span>
<span id="cb63-968"><a href="#cb63-968" aria-hidden="true" tabindex="-1"></a>        x[x<span class="sc">$</span>adj_pval <span class="sc">&lt;</span> pval_adjustment, ])</span>
<span id="cb63-969"><a href="#cb63-969" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-970"><a href="#cb63-970" aria-hidden="true" tabindex="-1"></a>    unique_genes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(celltype_DEGs, <span class="cf">function</span>(x)</span>
<span id="cb63-971"><a href="#cb63-971" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>name)</span>
<span id="cb63-972"><a href="#cb63-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-973"><a href="#cb63-973" aria-hidden="true" tabindex="-1"></a>    unique_degs <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">unlist</span>(unique_genes))</span>
<span id="cb63-974"><a href="#cb63-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-975"><a href="#cb63-975" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if no DEGs found break but return the DE analysis scores still</span></span>
<span id="cb63-976"><a href="#cb63-976" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(unique_degs) <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb63-977"><a href="#cb63-977" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(</span>
<span id="cb63-978"><a href="#cb63-978" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(</span>
<span id="cb63-979"><a href="#cb63-979" aria-hidden="true" tabindex="-1"></a>          <span class="st">"celltype_DEGs"</span> <span class="ot">=</span> celltype_DEGs,</span>
<span id="cb63-980"><a href="#cb63-980" aria-hidden="true" tabindex="-1"></a>          <span class="st">"celltype_all_genes"</span> <span class="ot">=</span> celltype_de,</span>
<span id="cb63-981"><a href="#cb63-981" aria-hidden="true" tabindex="-1"></a>          <span class="st">"celltype_counts"</span> <span class="ot">=</span> counts_celltypes</span>
<span id="cb63-982"><a href="#cb63-982" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb63-983"><a href="#cb63-983" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb63-984"><a href="#cb63-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-985"><a href="#cb63-985" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">isTRUE</span>(verbose))</span>
<span id="cb63-986"><a href="#cb63-986" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">length</span>(unique_degs), <span class="st">" unique DEGs foundacross all cell types"</span>)</span>
<span id="cb63-987"><a href="#cb63-987" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-988"><a href="#cb63-988" aria-hidden="true" tabindex="-1"></a>    celltype_all_genes_dt <span class="ot">&lt;-</span></span>
<span id="cb63-989"><a href="#cb63-989" aria-hidden="true" tabindex="-1"></a>      data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_de, <span class="at">idcol =</span> T)</span>
<span id="cb63-990"><a href="#cb63-990" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-991"><a href="#cb63-991" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setnames</span>(celltype_all_genes_dt, <span class="st">".id"</span>, <span class="st">"celltype"</span>)</span>
<span id="cb63-992"><a href="#cb63-992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-993"><a href="#cb63-993" aria-hidden="true" tabindex="-1"></a>    celltype_DEGs_dt <span class="ot">&lt;-</span></span>
<span id="cb63-994"><a href="#cb63-994" aria-hidden="true" tabindex="-1"></a>      data.table<span class="sc">::</span><span class="fu">rbindlist</span>(celltype_DEGs, <span class="at">idcol =</span> T)</span>
<span id="cb63-995"><a href="#cb63-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-996"><a href="#cb63-996" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setnames</span>(celltype_DEGs_dt, <span class="st">".id"</span>, <span class="st">"celltype"</span>)</span>
<span id="cb63-997"><a href="#cb63-997" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-998"><a href="#cb63-998" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">isFALSE</span>(plot_folder)){</span>
<span id="cb63-999"><a href="#cb63-999" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">isTRUE</span>(verbose))</span>
<span id="cb63-1000"><a href="#cb63-1000" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">"Plotting the results of differential expression analysis"</span>)</span>
<span id="cb63-1001"><a href="#cb63-1001" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1002"><a href="#cb63-1002" aria-hidden="true" tabindex="-1"></a>        <span class="co">#make plots for DE analysis</span></span>
<span id="cb63-1003"><a href="#cb63-1003" aria-hidden="true" tabindex="-1"></a>        <span class="co"># plot_de_analysis(pb_dat,y,celltype_DEGs_dt,celltype_all_genes_dt,</span></span>
<span id="cb63-1004"><a href="#cb63-1004" aria-hidden="true" tabindex="-1"></a>        <span class="co">#                  counts_celltypes,plot_folder)</span></span>
<span id="cb63-1005"><a href="#cb63-1005" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb63-1006"><a href="#cb63-1006" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1007"><a href="#cb63-1007" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(celltype_DEGs)</span>
<span id="cb63-1008"><a href="#cb63-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1009"><a href="#cb63-1009" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb63-1010"><a href="#cb63-1010" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1011"><a href="#cb63-1011" aria-hidden="true" tabindex="-1"></a>lrt_sig <span class="ot">&lt;-</span> <span class="fu">map</span>(</span>
<span id="cb63-1012"><a href="#cb63-1012" aria-hidden="true" tabindex="-1"></a>  lrt_clean,</span>
<span id="cb63-1013"><a href="#cb63-1013" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> <span class="fu">get_sig_genes_and_plots</span>(</span>
<span id="cb63-1014"><a href="#cb63-1014" aria-hidden="true" tabindex="-1"></a>    .x,</span>
<span id="cb63-1015"><a href="#cb63-1015" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltype_counts =</span> counts_celltypes,</span>
<span id="cb63-1016"><a href="#cb63-1016" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot_folder =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"temp"</span>)</span>
<span id="cb63-1017"><a href="#cb63-1017" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb63-1018"><a href="#cb63-1018" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb63-1019"><a href="#cb63-1019" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1020"><a href="#cb63-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1021"><a href="#cb63-1021" aria-hidden="true" tabindex="-1"></a>The results are in a list of lists.</span>
<span id="cb63-1022"><a href="#cb63-1022" aria-hidden="true" tabindex="-1"></a>The first level is the celltypes and the next is the comparisons.</span>
<span id="cb63-1023"><a href="#cb63-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1026"><a href="#cb63-1026" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1027"><a href="#cb63-1027" aria-hidden="true" tabindex="-1"></a><span class="co"># the top level is the celltypes</span></span>
<span id="cb63-1028"><a href="#cb63-1028" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(lrt_sig)</span>
<span id="cb63-1029"><a href="#cb63-1029" aria-hidden="true" tabindex="-1"></a><span class="co"># the next level is the comparisons</span></span>
<span id="cb63-1030"><a href="#cb63-1030" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(lrt_sig<span class="sc">$</span>Endothelial)</span>
<span id="cb63-1031"><a href="#cb63-1031" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(lrt_sig<span class="sc">$</span>Endothelial<span class="sc">$</span>groupAD_P_oxford_dunno_vs_others)</span>
<span id="cb63-1032"><a href="#cb63-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1033"><a href="#cb63-1033" aria-hidden="true" tabindex="-1"></a><span class="co"># save results</span></span>
<span id="cb63-1034"><a href="#cb63-1034" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(lrt_sig, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_sig-genes_ad-vs-control-and-prep.rds"</span>))</span>
<span id="cb63-1035"><a href="#cb63-1035" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1036"><a href="#cb63-1036" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1037"><a href="#cb63-1037" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plots</span></span>
<span id="cb63-1038"><a href="#cb63-1038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1039"><a href="#cb63-1039" aria-hidden="true" tabindex="-1"></a>Let's have a look at the results with some plots.</span>
<span id="cb63-1040"><a href="#cb63-1040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1041"><a href="#cb63-1041" aria-hidden="true" tabindex="-1"></a>First I'll merge the different contrasts into a single dataframe.</span>
<span id="cb63-1042"><a href="#cb63-1042" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1045"><a href="#cb63-1045" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1046"><a href="#cb63-1046" aria-hidden="true" tabindex="-1"></a><span class="co"># merge contrasts</span></span>
<span id="cb63-1047"><a href="#cb63-1047" aria-hidden="true" tabindex="-1"></a>lrt_merged <span class="ot">&lt;-</span> <span class="fu">map</span>(lrt_sig,</span>
<span id="cb63-1048"><a href="#cb63-1048" aria-hidden="true" tabindex="-1"></a>                  <span class="co"># add the contrast name as a column</span></span>
<span id="cb63-1049"><a href="#cb63-1049" aria-hidden="true" tabindex="-1"></a>                  <span class="sc">~</span> <span class="fu">map2</span>(</span>
<span id="cb63-1050"><a href="#cb63-1050" aria-hidden="true" tabindex="-1"></a>                    .x,</span>
<span id="cb63-1051"><a href="#cb63-1051" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">names</span>(.x),</span>
<span id="cb63-1052"><a href="#cb63-1052" aria-hidden="true" tabindex="-1"></a>                    <span class="sc">~</span> .x <span class="sc">%&gt;%</span></span>
<span id="cb63-1053"><a href="#cb63-1053" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb63-1054"><a href="#cb63-1054" aria-hidden="true" tabindex="-1"></a>                        <span class="at">contrast =</span> .y,</span>
<span id="cb63-1055"><a href="#cb63-1055" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># add direction of expression</span></span>
<span id="cb63-1056"><a href="#cb63-1056" aria-hidden="true" tabindex="-1"></a>                        <span class="at">deg_direction =</span> <span class="fu">ifelse</span>(logFC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>)</span>
<span id="cb63-1057"><a href="#cb63-1057" aria-hidden="true" tabindex="-1"></a>                      )</span>
<span id="cb63-1058"><a href="#cb63-1058" aria-hidden="true" tabindex="-1"></a>                  ) <span class="sc">%&gt;%</span></span>
<span id="cb63-1059"><a href="#cb63-1059" aria-hidden="true" tabindex="-1"></a>                    purrr<span class="sc">::</span><span class="fu">list_rbind</span>())</span>
<span id="cb63-1060"><a href="#cb63-1060" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1061"><a href="#cb63-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1064"><a href="#cb63-1064" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1065"><a href="#cb63-1065" aria-hidden="true" tabindex="-1"></a><span class="co"># set colour palette </span></span>
<span id="cb63-1066"><a href="#cb63-1066" aria-hidden="true" tabindex="-1"></a>pal<span class="ot">=</span><span class="fu">c</span>(wesanderson<span class="sc">::</span><span class="fu">wes_palette</span>(<span class="st">"Royal2"</span>),</span>
<span id="cb63-1067"><a href="#cb63-1067" aria-hidden="true" tabindex="-1"></a>      wesanderson<span class="sc">::</span><span class="fu">wes_palette</span>(<span class="st">"Moonrise3"</span>))</span>
<span id="cb63-1068"><a href="#cb63-1068" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1069"><a href="#cb63-1069" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1072"><a href="#cb63-1072" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1073"><a href="#cb63-1073" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-1074"><a href="#cb63-1074" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-1075"><a href="#cb63-1075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1076"><a href="#cb63-1076" aria-hidden="true" tabindex="-1"></a><span class="co"># this is supposed to use pb_dat I think</span></span>
<span id="cb63-1077"><a href="#cb63-1077" aria-hidden="true" tabindex="-1"></a>lrt_merged<span class="sc">$</span>Endothelial <span class="sc">%&gt;%</span></span>
<span id="cb63-1078"><a href="#cb63-1078" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(contrast) <span class="sc">%&gt;%</span></span>
<span id="cb63-1079"><a href="#cb63-1079" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get top 3 genes by significance</span></span>
<span id="cb63-1080"><a href="#cb63-1080" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice_min</span>(adj_pval, <span class="at">n =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-1081"><a href="#cb63-1081" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb63-1082"><a href="#cb63-1082" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(name <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(deg_direction)) <span class="sc">%&gt;%</span></span>
<span id="cb63-1083"><a href="#cb63-1083" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> contrast, <span class="at">y =</span> logFC,<span class="at">colour=</span>deg_direction)) <span class="sc">+</span></span>
<span id="cb63-1084"><a href="#cb63-1084" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">height=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb63-1085"><a href="#cb63-1085" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun.data =</span> <span class="st">"mean_cl_normal"</span>,</span>
<span id="cb63-1086"><a href="#cb63-1086" aria-hidden="true" tabindex="-1"></a>               <span class="co">#aes(shape="mean"), </span></span>
<span id="cb63-1087"><a href="#cb63-1087" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> <span class="st">"grey"</span>,</span>
<span id="cb63-1088"><a href="#cb63-1088" aria-hidden="true" tabindex="-1"></a>               <span class="at">geom =</span> <span class="st">"crossbar"</span>,</span>
<span id="cb63-1089"><a href="#cb63-1089" aria-hidden="true" tabindex="-1"></a>               <span class="at">show.legend =</span> T)<span class="sc">+</span></span>
<span id="cb63-1090"><a href="#cb63-1090" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="st">""</span>, <span class="at">values=</span><span class="fu">c</span>(<span class="st">"mean"</span><span class="ot">=</span><span class="st">"x"</span>))<span class="sc">+</span></span>
<span id="cb63-1091"><a href="#cb63-1091" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span> <span class="st">"Sum pseudobulk expression counts (unnormalised)"</span>, </span>
<span id="cb63-1092"><a href="#cb63-1092" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">"Phenotype"</span>,</span>
<span id="cb63-1093"><a href="#cb63-1093" aria-hidden="true" tabindex="-1"></a>       <span class="at">colour=</span><span class="st">"DEG direction"</span>) <span class="sc">+</span></span>
<span id="cb63-1094"><a href="#cb63-1094" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> name, <span class="at">scales =</span> <span class="st">"free_y"</span>)<span class="sc">+</span></span>
<span id="cb63-1095"><a href="#cb63-1095" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb63-1096"><a href="#cb63-1096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb63-1097"><a href="#cb63-1097" aria-hidden="true" tabindex="-1"></a>       <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb63-1098"><a href="#cb63-1098" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_colour_viridis(discrete = T)</span></span>
<span id="cb63-1099"><a href="#cb63-1099" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>pal)</span>
<span id="cb63-1100"><a href="#cb63-1100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1101"><a href="#cb63-1101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1104"><a href="#cb63-1104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1105"><a href="#cb63-1105" aria-hidden="true" tabindex="-1"></a>lrt_merged <span class="ot">&lt;-</span> <span class="fu">map2</span>(lrt_merged, <span class="fu">names</span>(lrt_merged), <span class="sc">~</span> .x <span class="sc">%&gt;%</span></span>
<span id="cb63-1106"><a href="#cb63-1106" aria-hidden="true" tabindex="-1"></a>                     dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> .y)) <span class="sc">%&gt;%</span></span>
<span id="cb63-1107"><a href="#cb63-1107" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">list_rbind</span>()</span>
<span id="cb63-1108"><a href="#cb63-1108" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1109"><a href="#cb63-1109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1110"><a href="#cb63-1110" aria-hidden="true" tabindex="-1"></a><span class="fu">### Counts of DEGs</span></span>
<span id="cb63-1111"><a href="#cb63-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1114"><a href="#cb63-1114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1115"><a href="#cb63-1115" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb63-1116"><a href="#cb63-1116" aria-hidden="true" tabindex="-1"></a>lrt_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1117"><a href="#cb63-1117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(deg_direction), <span class="at">fill =</span> <span class="fu">factor</span>(celltype))) <span class="sc">+</span></span>
<span id="cb63-1118"><a href="#cb63-1118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">stat =</span> <span class="st">"count"</span>) <span class="sc">+</span></span>
<span id="cb63-1119"><a href="#cb63-1119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span><span class="fu">factor</span>(celltype)) <span class="sc">+</span></span>
<span id="cb63-1120"><a href="#cb63-1120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Cell type counts of DEGs"</span>)<span class="sc">+</span></span>
<span id="cb63-1121"><a href="#cb63-1121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span> <span class="st">"Log2 Fold Change"</span>, <span class="at">x =</span> <span class="st">"DEG direction"</span>, <span class="at">fill=</span><span class="st">"Cell Type"</span>) <span class="sc">+</span></span>
<span id="cb63-1122"><a href="#cb63-1122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb63-1123"><a href="#cb63-1123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb63-1124"><a href="#cb63-1124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T)</span>
<span id="cb63-1125"><a href="#cb63-1125" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_manual(values=pal)</span></span>
<span id="cb63-1126"><a href="#cb63-1126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1127"><a href="#cb63-1127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1130"><a href="#cb63-1130" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1131"><a href="#cb63-1131" aria-hidden="true" tabindex="-1"></a>lrt_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1132"><a href="#cb63-1132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> contrast, <span class="at">y =</span> logFC, <span class="at">fill =</span> <span class="fu">factor</span>(contrast))) <span class="sc">+</span></span>
<span id="cb63-1133"><a href="#cb63-1133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb63-1134"><a href="#cb63-1134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span><span class="fu">factor</span>(celltype)) <span class="sc">+</span></span>
<span id="cb63-1135"><a href="#cb63-1135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Cell type Log2 Fold Change of DEGs"</span>) <span class="sc">+</span><span class="co">#,</span></span>
<span id="cb63-1136"><a href="#cb63-1136" aria-hidden="true" tabindex="-1"></a>          <span class="co"># subtitle = paste0("Median LFC of Down and Up regulated DEGs: ",</span></span>
<span id="cb63-1137"><a href="#cb63-1137" aria-hidden="true" tabindex="-1"></a>          <span class="co">#                   round(degs_median_lfc$V1[[2]]*-1,2),", ",</span></span>
<span id="cb63-1138"><a href="#cb63-1138" aria-hidden="true" tabindex="-1"></a>          <span class="co">#                   round(degs_median_lfc$V1[[1]],2)))+</span></span>
<span id="cb63-1139"><a href="#cb63-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span> <span class="st">"Log2 Fold Change"</span>, <span class="at">x =</span> <span class="st">"DEG direction"</span>, <span class="at">fill=</span><span class="st">"Cell Type"</span>) <span class="sc">+</span></span>
<span id="cb63-1140"><a href="#cb63-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb63-1141"><a href="#cb63-1141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">6</span>),</span>
<span id="cb63-1142"><a href="#cb63-1142" aria-hidden="true" tabindex="-1"></a>       <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">6</span>)) <span class="sc">+</span></span>
<span id="cb63-1143"><a href="#cb63-1143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"AD P vs others"</span>, <span class="st">"AD V vs others"</span>, <span class="st">"Control P vs others"</span>, <span class="st">"Control V vs others"</span>))</span>
<span id="cb63-1144"><a href="#cb63-1144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1145"><a href="#cb63-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1146"><a href="#cb63-1146" aria-hidden="true" tabindex="-1"></a><span class="fu">### Volcano plots</span></span>
<span id="cb63-1147"><a href="#cb63-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1150"><a href="#cb63-1150" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1151"><a href="#cb63-1151" aria-hidden="true" tabindex="-1"></a><span class="co"># add colour identifier</span></span>
<span id="cb63-1152"><a href="#cb63-1152" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Up"</span> <span class="ot">=</span> <span class="st">"#FDE725FF"</span>, <span class="st">"Down"</span> <span class="ot">=</span> <span class="st">"#440154FF"</span>)</span>
<span id="cb63-1153"><a href="#cb63-1153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1154"><a href="#cb63-1154" aria-hidden="true" tabindex="-1"></a>lrt_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1155"><a href="#cb63-1155" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale FDR by log10</span></span>
<span id="cb63-1156"><a href="#cb63-1156" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(adj_pval))) <span class="sc">+</span></span>
<span id="cb63-1157"><a href="#cb63-1157" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> deg_direction), <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">size =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb63-1158"><a href="#cb63-1158" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> <span class="fu">factor</span>(celltype), <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb63-1159"><a href="#cb63-1159" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(</span>
<span id="cb63-1160"><a href="#cb63-1160" aria-hidden="true" tabindex="-1"></a>        <span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">colour =</span> <span class="st">"#990000"</span>,</span>
<span id="cb63-1161"><a href="#cb63-1161" aria-hidden="true" tabindex="-1"></a>        <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">linewidth =</span> .<span class="dv">3</span></span>
<span id="cb63-1162"><a href="#cb63-1162" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb63-1163"><a href="#cb63-1163" aria-hidden="true" tabindex="-1"></a>    <span class="co"># geom_vline(xintercept = 0, colour="black", size=.3) +</span></span>
<span id="cb63-1164"><a href="#cb63-1164" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"-log10(FDR)"</span>, <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>, <span class="at">colour =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb63-1165"><a href="#cb63-1165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb63-1166"><a href="#cb63-1166" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb63-1167"><a href="#cb63-1167" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb63-1168"><a href="#cb63-1168" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb63-1169"><a href="#cb63-1169" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb63-1170"><a href="#cb63-1170" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> cols) <span class="sc">+</span></span>
<span id="cb63-1171"><a href="#cb63-1171" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>( <span class="co"># give top three genes per cell type by adjusted p value</span></span>
<span id="cb63-1172"><a href="#cb63-1172" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> lrt_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1173"><a href="#cb63-1173" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb63-1174"><a href="#cb63-1174" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">slice_min</span>(adj_pval, <span class="at">n =</span> <span class="dv">3</span>),</span>
<span id="cb63-1175"><a href="#cb63-1175" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">label =</span> name),</span>
<span id="cb63-1176"><a href="#cb63-1176" aria-hidden="true" tabindex="-1"></a>        <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb63-1177"><a href="#cb63-1177" aria-hidden="true" tabindex="-1"></a>        <span class="at">box.padding =</span> <span class="fu">unit</span>(<span class="fl">0.35</span>, <span class="st">"lines"</span>),</span>
<span id="cb63-1178"><a href="#cb63-1178" aria-hidden="true" tabindex="-1"></a>        <span class="at">point.padding =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"lines"</span>)</span>
<span id="cb63-1179"><a href="#cb63-1179" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb63-1180"><a href="#cb63-1180" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1181"><a href="#cb63-1181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1182"><a href="#cb63-1182" aria-hidden="true" tabindex="-1"></a><span class="fu"># DESeq 2</span></span>
<span id="cb63-1183"><a href="#cb63-1183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1186"><a href="#cb63-1186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1187"><a href="#cb63-1187" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb63-1188"><a href="#cb63-1188" aria-hidden="true" tabindex="-1"></a><span class="co"># read data</span></span>
<span id="cb63-1189"><a href="#cb63-1189" aria-hidden="true" tabindex="-1"></a><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span id="cb63-1190"><a href="#cb63-1190" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span>))</span>
<span id="cb63-1191"><a href="#cb63-1191" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1192"><a href="#cb63-1192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1195"><a href="#cb63-1195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1196"><a href="#cb63-1196" aria-hidden="true" tabindex="-1"></a><span class="co"># make count rownames match data rownames</span></span>
<span id="cb63-1197"><a href="#cb63-1197" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data)</span>
<span id="cb63-1198"><a href="#cb63-1198" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>doner <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"doner-"</span>, sce<span class="sc">$</span>individual)</span>
<span id="cb63-1199"><a href="#cb63-1199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1200"><a href="#cb63-1200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1201"><a href="#cb63-1201" aria-hidden="true" tabindex="-1"></a>Let's first try doing pseudobulk by disease status</span>
<span id="cb63-1202"><a href="#cb63-1202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1205"><a href="#cb63-1205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1206"><a href="#cb63-1206" aria-hidden="true" tabindex="-1"></a><span class="co"># setup columns to aggregate by</span></span>
<span id="cb63-1207"><a href="#cb63-1207" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">Idents</span>(sce)</span>
<span id="cb63-1208"><a href="#cb63-1208" aria-hidden="true" tabindex="-1"></a><span class="co"># remove the "_" in the celltype names and replace with "-"</span></span>
<span id="cb63-1209"><a href="#cb63-1209" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">"-"</span>, sce<span class="sc">$</span>celltype)</span>
<span id="cb63-1210"><a href="#cb63-1210" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>donor <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"donor-"</span>, sce<span class="sc">$</span>individual)</span>
<span id="cb63-1211"><a href="#cb63-1211" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(sce<span class="sc">$</span>diagnosis) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"AD"</span>, <span class="st">"Control"</span>)</span>
<span id="cb63-1212"><a href="#cb63-1212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1213"><a href="#cb63-1213" aria-hidden="true" tabindex="-1"></a><span class="co"># counts matrix individual level</span></span>
<span id="cb63-1214"><a href="#cb63-1214" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> <span class="fu">AggregateExpression</span>(sce, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"celltype"</span>, <span class="st">"donor"</span>, <span class="st">"diagnosis"</span>),</span>
<span id="cb63-1215"><a href="#cb63-1215" aria-hidden="true" tabindex="-1"></a>                    <span class="at">assays =</span> <span class="st">"RNA"</span>,</span>
<span id="cb63-1216"><a href="#cb63-1216" aria-hidden="true" tabindex="-1"></a>                    <span class="at">slot =</span> <span class="st">"counts"</span>,</span>
<span id="cb63-1217"><a href="#cb63-1217" aria-hidden="true" tabindex="-1"></a>                    <span class="at">return.seurat =</span> <span class="cn">FALSE</span>)</span>
<span id="cb63-1218"><a href="#cb63-1218" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">&lt;-</span> cts<span class="sc">$</span>RNA</span>
<span id="cb63-1219"><a href="#cb63-1219" aria-hidden="true" tabindex="-1"></a>cts[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb63-1220"><a href="#cb63-1220" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1221"><a href="#cb63-1221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1222"><a href="#cb63-1222" aria-hidden="true" tabindex="-1"></a>Now we need to do some data manipulation to get the count and column data in the format DESeq2 requires</span>
<span id="cb63-1223"><a href="#cb63-1223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1226"><a href="#cb63-1226" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1227"><a href="#cb63-1227" aria-hidden="true" tabindex="-1"></a><span class="co"># transpose</span></span>
<span id="cb63-1228"><a href="#cb63-1228" aria-hidden="true" tabindex="-1"></a>cts_t <span class="ot">&lt;-</span> <span class="fu">t</span>(cts) <span class="sc">%&gt;%</span></span>
<span id="cb63-1229"><a href="#cb63-1229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb63-1230"><a href="#cb63-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1231"><a href="#cb63-1231" aria-hidden="true" tabindex="-1"></a>split_rows <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'_.*'</span>, <span class="st">''</span>, <span class="fu">rownames</span>(cts_t))</span>
<span id="cb63-1232"><a href="#cb63-1232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1233"><a href="#cb63-1233" aria-hidden="true" tabindex="-1"></a>cts_split <span class="ot">&lt;-</span> <span class="fu">split.data.frame</span>(cts_t,</span>
<span id="cb63-1234"><a href="#cb63-1234" aria-hidden="true" tabindex="-1"></a>                              <span class="at">f =</span> <span class="fu">factor</span>(split_rows))</span>
<span id="cb63-1235"><a href="#cb63-1235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1236"><a href="#cb63-1236" aria-hidden="true" tabindex="-1"></a>cts_split_m <span class="ot">&lt;-</span> <span class="fu">lapply</span>(cts_split, <span class="cf">function</span>(x) {</span>
<span id="cb63-1237"><a href="#cb63-1237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(x)</span>
<span id="cb63-1238"><a href="#cb63-1238" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb63-1239"><a href="#cb63-1239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1240"><a href="#cb63-1240" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to modify column names</span></span>
<span id="cb63-1241"><a href="#cb63-1241" aria-hidden="true" tabindex="-1"></a>modify_column_names <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">number_of_groups =</span> <span class="dv">2</span>) {</span>
<span id="cb63-1242"><a href="#cb63-1242" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb63-1243"><a href="#cb63-1243" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get the number of "_" seperations in the names</span></span>
<span id="cb63-1244"><a href="#cb63-1244" aria-hidden="true" tabindex="-1"></a>  number_of_splits <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">strsplit</span>(<span class="fu">as.character</span>(<span class="fu">names</span>(df)), <span class="st">"_"</span>)[[<span class="dv">1</span>]])</span>
<span id="cb63-1245"><a href="#cb63-1245" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get the last n - 1 groups</span></span>
<span id="cb63-1246"><a href="#cb63-1246" aria-hidden="true" tabindex="-1"></a>  cols <span class="ot">&lt;-</span></span>
<span id="cb63-1247"><a href="#cb63-1247" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map</span>((number_of_splits <span class="sc">-</span> (number_of_groups <span class="sc">-</span> <span class="dv">1</span>))<span class="sc">:</span>number_of_splits,</span>
<span id="cb63-1248"><a href="#cb63-1248" aria-hidden="true" tabindex="-1"></a>        <span class="sc">~</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">as.character</span>(<span class="fu">names</span>(</span>
<span id="cb63-1249"><a href="#cb63-1249" aria-hidden="true" tabindex="-1"></a>          df</span>
<span id="cb63-1250"><a href="#cb63-1250" aria-hidden="true" tabindex="-1"></a>        )), <span class="st">"_"</span>), <span class="st">"["</span>, .x))</span>
<span id="cb63-1251"><a href="#cb63-1251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(df) <span class="ot">&lt;-</span> <span class="fu">do.call</span>(paste, <span class="fu">c</span>(cols, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb63-1252"><a href="#cb63-1252" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb63-1253"><a href="#cb63-1253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb63-1254"><a href="#cb63-1254" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb63-1255"><a href="#cb63-1255" aria-hidden="true" tabindex="-1"></a>cts_split_m <span class="ot">&lt;-</span> <span class="fu">map</span>(cts_split_m, modify_column_names)</span>
<span id="cb63-1256"><a href="#cb63-1256" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1257"><a href="#cb63-1257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1260"><a href="#cb63-1260" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1261"><a href="#cb63-1261" aria-hidden="true" tabindex="-1"></a>cts_split_m<span class="sc">$</span><span class="st">`</span><span class="at">Neuron-ex</span><span class="st">`</span>[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb63-1262"><a href="#cb63-1262" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1263"><a href="#cb63-1263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1266"><a href="#cb63-1266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1267"><a href="#cb63-1267" aria-hidden="true" tabindex="-1"></a><span class="co"># get the col data</span></span>
<span id="cb63-1268"><a href="#cb63-1268" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">&lt;-</span> <span class="fu">map</span>(</span>
<span id="cb63-1269"><a href="#cb63-1269" aria-hidden="true" tabindex="-1"></a>  cts_split_m,</span>
<span id="cb63-1270"><a href="#cb63-1270" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> <span class="fu">data.frame</span>(</span>
<span id="cb63-1271"><a href="#cb63-1271" aria-hidden="true" tabindex="-1"></a>    <span class="at">samples =</span> <span class="fu">colnames</span>(.x),</span>
<span id="cb63-1272"><a href="#cb63-1272" aria-hidden="true" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"Control"</span>, <span class="fu">colnames</span>(.x)), <span class="st">"Control"</span>, <span class="st">"AD"</span>)</span>
<span id="cb63-1273"><a href="#cb63-1273" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb63-1274"><a href="#cb63-1274" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"samples"</span>)</span>
<span id="cb63-1275"><a href="#cb63-1275" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb63-1276"><a href="#cb63-1276" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1277"><a href="#cb63-1277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1280"><a href="#cb63-1280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1281"><a href="#cb63-1281" aria-hidden="true" tabindex="-1"></a><span class="co"># check that count and col data cols/rows match</span></span>
<span id="cb63-1282"><a href="#cb63-1282" aria-hidden="true" tabindex="-1"></a><span class="fu">map2_lgl</span>(cts_split_m, colData, <span class="sc">~</span> <span class="fu">ncol</span>(.x) <span class="sc">==</span> <span class="fu">nrow</span>(.y))</span>
<span id="cb63-1283"><a href="#cb63-1283" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1284"><a href="#cb63-1284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1287"><a href="#cb63-1287" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1288"><a href="#cb63-1288" aria-hidden="true" tabindex="-1"></a><span class="co"># get DESeq2 object</span></span>
<span id="cb63-1289"><a href="#cb63-1289" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">map2</span>(cts_split_m, colData, <span class="sc">~</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> .x,</span>
<span id="cb63-1290"><a href="#cb63-1290" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">colData =</span> .y,</span>
<span id="cb63-1291"><a href="#cb63-1291" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">design =</span> <span class="sc">~</span> condition))</span>
<span id="cb63-1292"><a href="#cb63-1292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1293"><a href="#cb63-1293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1296"><a href="#cb63-1296" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1297"><a href="#cb63-1297" aria-hidden="true" tabindex="-1"></a><span class="co"># run DESeq2</span></span>
<span id="cb63-1298"><a href="#cb63-1298" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">map</span>(dds, DESeq)</span>
<span id="cb63-1299"><a href="#cb63-1299" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1300"><a href="#cb63-1300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1301"><a href="#cb63-1301" aria-hidden="true" tabindex="-1"></a>We can see the result names</span>
<span id="cb63-1302"><a href="#cb63-1302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1305"><a href="#cb63-1305" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1306"><a href="#cb63-1306" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(dds<span class="sc">$</span>Astro)</span>
<span id="cb63-1307"><a href="#cb63-1307" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1308"><a href="#cb63-1308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1309"><a href="#cb63-1309" aria-hidden="true" tabindex="-1"></a>And then extract these results</span>
<span id="cb63-1310"><a href="#cb63-1310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1313"><a href="#cb63-1313" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1314"><a href="#cb63-1314" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">map</span>(dds, <span class="sc">~</span><span class="fu">results</span>(.x, <span class="at">name =</span> <span class="st">"condition_Control_vs_AD"</span>))</span>
<span id="cb63-1315"><a href="#cb63-1315" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res<span class="sc">$</span>Astro)</span>
<span id="cb63-1316"><a href="#cb63-1316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1317"><a href="#cb63-1317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1320"><a href="#cb63-1320" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1321"><a href="#cb63-1321" aria-hidden="true" tabindex="-1"></a><span class="co"># merge celltype dataframes</span></span>
<span id="cb63-1322"><a href="#cb63-1322" aria-hidden="true" tabindex="-1"></a>res_merged <span class="ot">&lt;-</span> <span class="fu">map2</span>(res, <span class="fu">names</span>(res), <span class="sc">~</span> <span class="fu">as.data.frame</span>(.x) <span class="sc">%&gt;%</span></span>
<span id="cb63-1323"><a href="#cb63-1323" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">rownames_to_column</span>(<span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-1324"><a href="#cb63-1324" aria-hidden="true" tabindex="-1"></a>                     dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> .y, <span class="at">deg_direction =</span> <span class="fu">ifelse</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb63-1325"><a href="#cb63-1325" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">list_rbind</span>()</span>
<span id="cb63-1326"><a href="#cb63-1326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1327"><a href="#cb63-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1330"><a href="#cb63-1330" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1331"><a href="#cb63-1331" aria-hidden="true" tabindex="-1"></a><span class="co"># save results</span></span>
<span id="cb63-1332"><a href="#cb63-1332" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(res_merged, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2.rds"</span>))</span>
<span id="cb63-1333"><a href="#cb63-1333" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1334"><a href="#cb63-1334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1335"><a href="#cb63-1335" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plots</span></span>
<span id="cb63-1336"><a href="#cb63-1336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1339"><a href="#cb63-1339" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1340"><a href="#cb63-1340" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb63-1341"><a href="#cb63-1341" aria-hidden="true" tabindex="-1"></a>res_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1342"><a href="#cb63-1342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(deg_direction), <span class="at">fill =</span> <span class="fu">factor</span>(celltype))) <span class="sc">+</span></span>
<span id="cb63-1343"><a href="#cb63-1343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">stat =</span> <span class="st">"count"</span>) <span class="sc">+</span></span>
<span id="cb63-1344"><a href="#cb63-1344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span><span class="fu">factor</span>(celltype)) <span class="sc">+</span></span>
<span id="cb63-1345"><a href="#cb63-1345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Cell type counts of DEGs - Control VS AD"</span>)<span class="sc">+</span></span>
<span id="cb63-1346"><a href="#cb63-1346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span> <span class="st">"Log2 Fold Change"</span>, <span class="at">x =</span> <span class="st">"DEG direction"</span>, <span class="at">fill=</span><span class="st">"Cell Type"</span>) <span class="sc">+</span></span>
<span id="cb63-1347"><a href="#cb63-1347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>()<span class="sc">+</span></span>
<span id="cb63-1348"><a href="#cb63-1348" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb63-1349"><a href="#cb63-1349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis</span>(<span class="at">discrete =</span> T)</span>
<span id="cb63-1350"><a href="#cb63-1350" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_fill_manual(values=pal)</span></span>
<span id="cb63-1351"><a href="#cb63-1351" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1352"><a href="#cb63-1352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1353"><a href="#cb63-1353" aria-hidden="true" tabindex="-1"></a><span class="fu">### Volcano plots</span></span>
<span id="cb63-1354"><a href="#cb63-1354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1357"><a href="#cb63-1357" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1358"><a href="#cb63-1358" aria-hidden="true" tabindex="-1"></a><span class="co"># add colour identifier</span></span>
<span id="cb63-1359"><a href="#cb63-1359" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Up"</span> <span class="ot">=</span> <span class="st">"#FDE725FF"</span>, <span class="st">"Down"</span> <span class="ot">=</span> <span class="st">"#440154FF"</span>)</span>
<span id="cb63-1360"><a href="#cb63-1360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1361"><a href="#cb63-1361" aria-hidden="true" tabindex="-1"></a>res_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1362"><a href="#cb63-1362" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter to significant hits</span></span>
<span id="cb63-1363"><a href="#cb63-1363" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb63-1364"><a href="#cb63-1364" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale FDR by log10</span></span>
<span id="cb63-1365"><a href="#cb63-1365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb63-1366"><a href="#cb63-1366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> deg_direction), <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">size =</span> .<span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb63-1367"><a href="#cb63-1367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> <span class="fu">factor</span>(celltype), <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb63-1368"><a href="#cb63-1368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(</span>
<span id="cb63-1369"><a href="#cb63-1369" aria-hidden="true" tabindex="-1"></a>    <span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>),</span>
<span id="cb63-1370"><a href="#cb63-1370" aria-hidden="true" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">"#990000"</span>,</span>
<span id="cb63-1371"><a href="#cb63-1371" aria-hidden="true" tabindex="-1"></a>    <span class="at">linetype =</span> <span class="st">"dashed"</span>,</span>
<span id="cb63-1372"><a href="#cb63-1372" aria-hidden="true" tabindex="-1"></a>    <span class="at">linewidth =</span> .<span class="dv">3</span></span>
<span id="cb63-1373"><a href="#cb63-1373" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb63-1374"><a href="#cb63-1374" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_vline(xintercept = 0, colour="black", size=.3) +</span></span>
<span id="cb63-1375"><a href="#cb63-1375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"-log10(FDR)"</span>, <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>, <span class="at">colour =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb63-1376"><a href="#cb63-1376" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb63-1377"><a href="#cb63-1377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb63-1378"><a href="#cb63-1378" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb63-1379"><a href="#cb63-1379" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb63-1380"><a href="#cb63-1380" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)</span>
<span id="cb63-1381"><a href="#cb63-1381" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb63-1382"><a href="#cb63-1382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> cols) <span class="sc">+</span></span>
<span id="cb63-1383"><a href="#cb63-1383" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(</span>
<span id="cb63-1384"><a href="#cb63-1384" aria-hidden="true" tabindex="-1"></a>    <span class="co"># give top three genes per cell type by adjusted p value</span></span>
<span id="cb63-1385"><a href="#cb63-1385" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> res_merged <span class="sc">%&gt;%</span></span>
<span id="cb63-1386"><a href="#cb63-1386" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb63-1387"><a href="#cb63-1387" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">slice_min</span>(padj, <span class="at">n =</span> <span class="dv">3</span>),</span>
<span id="cb63-1388"><a href="#cb63-1388" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">label =</span> gene),</span>
<span id="cb63-1389"><a href="#cb63-1389" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb63-1390"><a href="#cb63-1390" aria-hidden="true" tabindex="-1"></a>    <span class="at">box.padding =</span> <span class="fu">unit</span>(<span class="fl">0.35</span>, <span class="st">"lines"</span>),</span>
<span id="cb63-1391"><a href="#cb63-1391" aria-hidden="true" tabindex="-1"></a>    <span class="at">point.padding =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"lines"</span>)</span>
<span id="cb63-1392"><a href="#cb63-1392" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb63-1393"><a href="#cb63-1393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb63-1394"><a href="#cb63-1394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1395"><a href="#cb63-1395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1398"><a href="#cb63-1398" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb63-1399"><a href="#cb63-1399" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb63-1400"><a href="#cb63-1400" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb63-1401"><a href="#cb63-1401" aria-hidden="true" tabindex="-1"></a>        <span class="co">#slight diff approach for contin or cat</span></span>
<span id="cb63-1402"><a href="#cb63-1402" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(y_contin){<span class="co">#need to specify coefficient target as not done in design</span></span>
<span id="cb63-1403"><a href="#cb63-1403" aria-hidden="true" tabindex="-1"></a>            test <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmLRT</span>(fit,<span class="at">coef=</span>y_name)</span>
<span id="cb63-1404"><a href="#cb63-1404" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb63-1405"><a href="#cb63-1405" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span>{<span class="co">#categorical i.e. case/control, no need</span></span>
<span id="cb63-1406"><a href="#cb63-1406" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(coef)){<span class="co">#if user inputted value for case samples</span></span>
<span id="cb63-1407"><a href="#cb63-1407" aria-hidden="true" tabindex="-1"></a>                test <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmLRT</span>(fit,<span class="at">coef=</span><span class="fu">length</span>(new_levels))</span>
<span id="cb63-1408"><a href="#cb63-1408" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb63-1409"><a href="#cb63-1409" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span>{<span class="co">#they didn't direction may not be as they expect</span></span>
<span id="cb63-1410"><a href="#cb63-1410" aria-hidden="true" tabindex="-1"></a>                <span class="fu">message</span>(<span class="fu">paste0</span>(<span class="st">"WARNING: No coefficient passed so direction ma"</span>,</span>
<span id="cb63-1411"><a href="#cb63-1411" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"y not relate to desired variable"</span>))</span>
<span id="cb63-1412"><a href="#cb63-1412" aria-hidden="true" tabindex="-1"></a>                test <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmLRT</span>(fit)</span>
<span id="cb63-1413"><a href="#cb63-1413" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb63-1414"><a href="#cb63-1414" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb63-1415"><a href="#cb63-1415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1416"><a href="#cb63-1416" aria-hidden="true" tabindex="-1"></a>        pvals <span class="ot">&lt;-</span> test<span class="sc">$</span>table</span>
<span id="cb63-1417"><a href="#cb63-1417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1418"><a href="#cb63-1418" aria-hidden="true" tabindex="-1"></a>        <span class="co">#add adj p-values</span></span>
<span id="cb63-1419"><a href="#cb63-1419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1420"><a href="#cb63-1420" aria-hidden="true" tabindex="-1"></a>        pvals<span class="sc">$</span>adj_pval <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">p.adjust</span>(pvals<span class="sc">$</span>PValue,</span>
<span id="cb63-1421"><a href="#cb63-1421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1422"><a href="#cb63-1422" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">method =</span> pval_adjust_method)</span>
<span id="cb63-1423"><a href="#cb63-1423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1424"><a href="#cb63-1424" aria-hidden="true" tabindex="-1"></a>        pvals<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pvals)</span>
<span id="cb63-1425"><a href="#cb63-1425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1426"><a href="#cb63-1426" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(pvals) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb63-1427"><a href="#cb63-1427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1428"><a href="#cb63-1428" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(verbose){</span>
<span id="cb63-1429"><a href="#cb63-1429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1430"><a href="#cb63-1430" aria-hidden="true" tabindex="-1"></a>            numDEGs <span class="ot">&lt;-</span> <span class="fu">nrow</span>(pvals[pvals<span class="sc">$</span>adj_pval<span class="sc">&lt;</span>adj_pval,])</span>
<span id="cb63-1431"><a href="#cb63-1431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1432"><a href="#cb63-1432" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(numDEGs,<span class="st">" DEGs found"</span>)</span>
<span id="cb63-1433"><a href="#cb63-1433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1434"><a href="#cb63-1434" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb63-1435"><a href="#cb63-1435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1436"><a href="#cb63-1436" aria-hidden="true" tabindex="-1"></a>        DE_ct_rtrn[[ct_i]] <span class="ot">&lt;-</span> pvals</span>
<span id="cb63-1437"><a href="#cb63-1437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-1438"><a href="#cb63-1438" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>