<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Blood brain barrier in AD - Perform pathway analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="styles.css">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Blood brain barrier in AD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../06_dissemination/03_website/01_data-analysis.html" rel="" target="" aria-current="page">
 <span class="menu-text">Data analysis</span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools ms-auto">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Perform pathway analysis</li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../06_dissemination/03_website/01_data-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis Pipeline</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/01_fastqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run FastQC on sample</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/02_multiqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run MultiQC</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/03_generate_cellranger_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/04_cellranger_count_set1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/05_cellranger_count_set2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 2</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/06_cellranger_count_set3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 3</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/07_cellranger_aggr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Aggregate cellranger count reports</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/08_scflow_prep.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scflow prep</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/09_scflow_run.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run SCflow</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/10_scflow_results.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Analysis scFlow results</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">04_data_analysis/001_snrnaseq_analysis/11_qseudobulk.qmd</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">04_data_analysis/001_snrnaseq_analysis/12_pathway-analysis.qmd</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages"><span class="header-section-number">1</span> Load packages</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data"><span class="header-section-number">2</span> Read in data</a></li>
  <li><a href="#data-prep" id="toc-data-prep" class="nav-link" data-scroll-target="#data-prep"><span class="header-section-number">3</span> Data prep</a></li>
  <li><a href="#hypergeometric-enrichment" id="toc-hypergeometric-enrichment" class="nav-link" data-scroll-target="#hypergeometric-enrichment"><span class="header-section-number">4</span> Hypergeometric enrichment</a></li>
  <li>
<a href="#gene-set-enrichment-analysis-gsea" id="toc-gene-set-enrichment-analysis-gsea" class="nav-link" data-scroll-target="#gene-set-enrichment-analysis-gsea"><span class="header-section-number">5</span> Gene set enrichment analysis (GSEA)</a>
  <ul class="collapse">
<li>
<a href="#gene-ontology" id="toc-gene-ontology" class="nav-link" data-scroll-target="#gene-ontology"><span class="header-section-number">5.1</span> Gene Ontology</a>
  <ul class="collapse">
<li><a href="#plots" id="toc-plots" class="nav-link" data-scroll-target="#plots"><span class="header-section-number">5.1.1</span> Plots</a></li>
  </ul>
</li>
  <li>
<a href="#kegg" id="toc-kegg" class="nav-link" data-scroll-target="#kegg"><span class="header-section-number">5.2</span> KEGG</a>
  <ul class="collapse">
<li><a href="#plots-1" id="toc-plots-1" class="nav-link" data-scroll-target="#plots-1"><span class="header-section-number">5.2.1</span> Plots</a></li>
  </ul>
</li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Perform pathway analysis</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/i_am.html">i_am</a></span><span class="op">(</span><span class="st">"04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.qmd"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-packages" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Load packages</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="read-in-data" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Read in data</h1>
<p>First we’ll read in the DESeq2 results. Note that all the DESeq2 fold changes were done per celltype and across controls and cases (AD).</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2.rds"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="data-prep" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Data prep</h1>
<p>I’ll start by grabbing the to significant genes per celltype</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># filter to sig genes</span></span>
<span><span class="va">res_sig</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get gene names per celltype</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">res_sig</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>list_of_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">list_of_columns</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">res_sig</span><span class="op">$</span><span class="va">celltype</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="hypergeometric-enrichment" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Hypergeometric enrichment</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">go_results</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">genes</span>, <span class="op">~</span> <span class="fu">enrichGO</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">.x</span>, OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>, </span>
<span>                                    keyType <span class="op">=</span> <span class="st">"SYMBOL"</span>, ont <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># exclude cases where no pathways were found</span></span>
<span><span class="va">go_results_filtered</span> <span class="op">&lt;-</span> <span class="va">go_results</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/keep.html">keep</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># barplot</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">go_results_filtered</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html">barplot</a></span><span class="op">(</span><span class="va">.x</span>, showCategory <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-4.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fit</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$Astro</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-ex`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$OPC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$SMC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-6-8.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="gene-set-enrichment-analysis-gsea" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Gene set enrichment analysis (GSEA)</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># get log2 FC as a vector with the respective gene names, sorted in decreasing </span></span>
<span><span class="co"># order</span></span>
<span><span class="va">named_vectors</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html">nest</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>gene_list <span class="op">=</span> <span class="fu">map</span><span class="op">(</span><span class="va">data</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">log2FoldChange</span>, <span class="va">.x</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="gene-ontology" class="level2" data-number="5.1"><h2 data-number="5.1" class="anchored" data-anchor-id="gene-ontology">
<span class="header-section-number">5.1</span> Gene Ontology</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Define the file path to save the resulting object</span></span>
<span><span class="va">result_file</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>    <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/gene-set-enrichment_go.rds"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if the file exists</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Code for your time-consuming analysis</span></span>
<span>  <span class="va">gse</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">map</span><span class="op">(</span></span>
<span>      <span class="va">named_vectors</span><span class="op">$</span><span class="va">gene_list</span>,</span>
<span>      <span class="op">~</span> <span class="fu">gseGO</span><span class="op">(</span></span>
<span>        <span class="va">.x</span>,</span>
<span>        OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span>        ont <span class="op">=</span> <span class="st">"ALL"</span>,</span>
<span>        keyType <span class="op">=</span> <span class="st">"SYMBOL"</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Save the resulting object</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">gse</span>, <span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="co"># Load the existing resulting object</span></span>
<span>  <span class="va">gse</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">named_vectors</span><span class="op">$</span><span class="va">celltype</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># remove any with no pathways found</span></span>
<span><span class="va">gse</span> <span class="op">&lt;-</span> <span class="va">gse</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/keep.html">keep</a></span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="plots" class="level3" data-number="5.1.1"><h3 data-number="5.1.1" class="anchored" data-anchor-id="plots">
<span class="header-section-number">5.1.1</span> Plots</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dotplots</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">map2</span><span class="op">(</span></span>
<span>    <span class="va">gse</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span>,</span>
<span>    <span class="op">~</span> <span class="fu">dotplot</span><span class="op">(</span><span class="va">.x</span>, showCategory <span class="op">=</span> <span class="fl">10</span>, split <span class="op">=</span> <span class="st">".sign"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span> <span class="op">~</span> <span class="va">.sign</span><span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu">ggtitle</span><span class="op">(</span><span class="va">.y</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">dotplots</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$ambiguous</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Astro</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Endothelial</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Microglia</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-ex`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-in`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Oligo</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$OPC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Pericyte</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$SMC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-10-11.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">map2</span><span class="op">(</span><span class="va">gse</span>,</span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span>,</span>
<span>     <span class="op">~</span> <span class="fu">ridgeplot</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"enrichment distribution"</span><span class="op">)</span> <span class="op">+</span></span>
<span>                  <span class="fu">ggtitle</span><span class="op">(</span><span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$ambiguous</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Astro</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Endothelial</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Microglia</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-ex`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-in`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Oligo</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$OPC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Pericyte</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$SMC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-12-11.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Note, one can change which pathway is plotted here by setting the title and <code>geneSetID</code> accordingly</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">map2</span><span class="op">(</span><span class="va">gse</span>,</span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gse</span><span class="op">)</span>,</span>
<span>     <span class="op">~</span> <span class="fu">gseaplot</span><span class="op">(</span><span class="va">.x</span>, by <span class="op">=</span> <span class="st">"all"</span>, title <span class="op">=</span> <span class="va">.x</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, geneSetID <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$ambiguous</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Astro</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Endothelial</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Microglia</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-ex`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-in`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Oligo</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$OPC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Pericyte</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$SMC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-13-11.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="pubmed-trend-of-enriched-terms" class="level4" data-number="5.1.1.1"><h4 data-number="5.1.1.1" class="anchored" data-anchor-id="pubmed-trend-of-enriched-terms">
<span class="header-section-number">5.1.1.1</span> PubMed trend of enriched terms</h4>
<div class="cell" data-hash="12_pathway_analysis_cache/html/unnamed-chunk-14_af3a3059213d6e34d7ad92d8225249ec">
<details><summary>Code</summary><div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">terms</span> <span class="op">&lt;-</span> <span class="va">gse</span><span class="op">$</span><span class="va">`Neuron-ex`</span><span class="op">$</span><span class="va">Description</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="fu">pmcplot</span><span class="op">(</span><span class="va">terms</span>, <span class="fl">2010</span><span class="op">:</span><span class="fl">2022</span>, proportion <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section></section><section id="kegg" class="level2" data-number="5.2"><h2 data-number="5.2" class="anchored" data-anchor-id="kegg">
<span class="header-section-number">5.2</span> KEGG</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb52"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Convert gene IDs for gseKEGG function</span></span>
<span><span class="co"># We will lose some genes here because not all IDs will be converted</span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">named_vectors</span><span class="op">$</span><span class="va">gene_list</span>,</span>
<span>    <span class="op">~</span> <span class="fu">bitr</span><span class="op">(</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span>,</span>
<span>      fromType <span class="op">=</span> <span class="st">"SYMBOL"</span>,</span>
<span>      toType <span class="op">=</span> <span class="st">"ENTREZID"</span>,</span>
<span>      OrgDb <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/list_c.html">list_rbind</a></span><span class="op">(</span><span class="op">)</span></span>
<span> <span class="co"># remove duplicate IDS </span></span>
<span><span class="va">dedup_ids</span> <span class="op">=</span> <span class="va">ids</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">ids</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># join entrezids to dataframe</span></span>
<span><span class="va">res_entrez</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">dedup_ids</span>, by <span class="op">=</span> <span class="fu">join_by</span><span class="op">(</span><span class="va">gene</span> <span class="op">==</span> <span class="va">SYMBOL</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># remove cases that couldn't be matched</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">ENTREZID</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get log2 FC as a vector with the respective gene names, sorted in decreasing </span></span>
<span><span class="co"># order</span></span>
<span><span class="va">named_vectors</span> <span class="op">&lt;-</span> <span class="va">res_entrez</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">celltype</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/nest.html">nest</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>gene_list <span class="op">=</span> <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">data</span>,</span>
<span>    <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">log2FoldChange</span>, <span class="va">.x</span><span class="op">$</span><span class="va">ENTREZID</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">kegg_organism</span> <span class="op">=</span> <span class="st">"hsa"</span></span>
<span></span>
<span><span class="co"># Define the file path to save the resulting object</span></span>
<span><span class="va">result_file</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>    <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/gene-set-enrichment_kegg.rds"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if the file exists</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Code for your time-consuming analysis</span></span>
<span><span class="va">kk2</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">map</span><span class="op">(</span></span>
<span>    <span class="va">named_vectors</span><span class="op">$</span><span class="va">gene_list</span>,</span>
<span>    <span class="op">~</span> <span class="fu">gseKEGG</span><span class="op">(</span><span class="va">.x</span>, organism <span class="op">=</span> <span class="va">kegg_organism</span>, keyType <span class="op">=</span> <span class="st">"ncbi-geneid"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Save the resulting object</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">kk2</span>, <span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="co"># Load the existing resulting object</span></span>
<span>  <span class="va">kk2</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kk2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">named_vectors</span><span class="op">$</span><span class="va">celltype</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="plots-1" class="level3" data-number="5.2.1"><h3 data-number="5.2.1" class="anchored" data-anchor-id="plots-1">
<span class="header-section-number">5.2.1</span> Plots</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">map2</span><span class="op">(</span></span>
<span>  <span class="va">kk2</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kk2</span><span class="op">)</span>,</span>
<span>  <span class="op">~</span> <span class="fu">dotplot</span><span class="op">(</span></span>
<span>    <span class="va">.x</span>,</span>
<span>    showCategory <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    title <span class="op">=</span> <span class="va">.y</span>,</span>
<span>    split <span class="op">=</span> <span class="st">".sign"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">facet_grid</span><span class="op">(</span><span class="va">.</span> <span class="op">~</span> <span class="va">.sign</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$ambiguous</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Astro</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Endothelial</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Fibroblast</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Microglia</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-ex`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-in`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Oligo</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$OPC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Pericyte</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$SMC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-17-12.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb67"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">map2</span><span class="op">(</span><span class="va">kk2</span>,</span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kk2</span><span class="op">)</span>,</span>
<span>     <span class="op">~</span> <span class="fu">ridgeplot</span><span class="op">(</span><span class="va">.x</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"enrichment distribution"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="va">.y</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$ambiguous</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Astro</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Endothelial</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Fibroblast</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Microglia</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-ex`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Neuron-in`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Oligo</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$OPC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$Pericyte</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$SMC</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-cell`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="12_pathway_analysis_files/figure-html/unnamed-chunk-19-12.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="pathview" class="level4" data-number="5.2.1.1"><h4 data-number="5.2.1.1" class="anchored" data-anchor-id="pathview">
<span class="header-section-number">5.2.1.1</span> Pathview</h4>


<!-- -->

</section></section></section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb80" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Perform pathway analysis</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: true</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="co">#self-contained: true</span></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> true</span></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.qmd"</span>)</span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a><span class="fu"># Load packages</span></span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-24"><a href="#cb80-24" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read in data</span></span>
<span id="cb80-25"><a href="#cb80-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-26"><a href="#cb80-26" aria-hidden="true" tabindex="-1"></a>First we'll read in the DESeq2 results.</span>
<span id="cb80-27"><a href="#cb80-27" aria-hidden="true" tabindex="-1"></a>Note that all the DESeq2 fold changes were done per celltype and across controls and cases (AD).</span>
<span id="cb80-28"><a href="#cb80-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-31"><a href="#cb80-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-32"><a href="#cb80-32" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2.rds"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb80-33"><a href="#cb80-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb80-34"><a href="#cb80-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-35"><a href="#cb80-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-36"><a href="#cb80-36" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data prep</span></span>
<span id="cb80-37"><a href="#cb80-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-38"><a href="#cb80-38" aria-hidden="true" tabindex="-1"></a>I'll start by grabbing the to significant genes per celltype</span>
<span id="cb80-39"><a href="#cb80-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-42"><a href="#cb80-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-43"><a href="#cb80-43" aria-hidden="true" tabindex="-1"></a><span class="co"># filter to sig genes</span></span>
<span id="cb80-44"><a href="#cb80-44" aria-hidden="true" tabindex="-1"></a>res_sig <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb80-45"><a href="#cb80-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-46"><a href="#cb80-46" aria-hidden="true" tabindex="-1"></a><span class="co"># get gene names per celltype</span></span>
<span id="cb80-47"><a href="#cb80-47" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> res_sig <span class="sc">%&gt;%</span></span>
<span id="cb80-48"><a href="#cb80-48" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb80-49"><a href="#cb80-49" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">list_of_columns =</span> <span class="fu">list</span>(gene)) <span class="sc">%&gt;%</span></span>
<span id="cb80-50"><a href="#cb80-50" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(list_of_columns)</span>
<span id="cb80-51"><a href="#cb80-51" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genes) <span class="ot">&lt;-</span> <span class="fu">unique</span>(res_sig<span class="sc">$</span>celltype)</span>
<span id="cb80-52"><a href="#cb80-52" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-53"><a href="#cb80-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-54"><a href="#cb80-54" aria-hidden="true" tabindex="-1"></a><span class="fu"># Hypergeometric enrichment</span></span>
<span id="cb80-55"><a href="#cb80-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-58"><a href="#cb80-58" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-59"><a href="#cb80-59" aria-hidden="true" tabindex="-1"></a>go_results <span class="ot">&lt;-</span> <span class="fu">map</span>(genes, <span class="sc">~</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> .x, <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>, </span>
<span id="cb80-60"><a href="#cb80-60" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>, <span class="at">ont =</span> <span class="st">"BP"</span>))</span>
<span id="cb80-61"><a href="#cb80-61" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-62"><a href="#cb80-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-65"><a href="#cb80-65" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-66"><a href="#cb80-66" aria-hidden="true" tabindex="-1"></a><span class="co"># exclude cases where no pathways were found</span></span>
<span id="cb80-67"><a href="#cb80-67" aria-hidden="true" tabindex="-1"></a>go_results_filtered <span class="ot">&lt;-</span> go_results <span class="sc">%&gt;%</span></span>
<span id="cb80-68"><a href="#cb80-68" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">keep</span>(<span class="sc">~</span> <span class="fu">nrow</span>(.) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb80-69"><a href="#cb80-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-70"><a href="#cb80-70" aria-hidden="true" tabindex="-1"></a><span class="co"># barplot</span></span>
<span id="cb80-71"><a href="#cb80-71" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">map</span>(go_results_filtered, <span class="sc">~</span> <span class="fu">plot</span>(<span class="fu">barplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">20</span>)))</span>
<span id="cb80-72"><a href="#cb80-72" aria-hidden="true" tabindex="-1"></a>fit</span>
<span id="cb80-73"><a href="#cb80-73" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-74"><a href="#cb80-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-75"><a href="#cb80-75" aria-hidden="true" tabindex="-1"></a><span class="fu"># Gene set enrichment analysis (GSEA)</span></span>
<span id="cb80-76"><a href="#cb80-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-79"><a href="#cb80-79" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-80"><a href="#cb80-80" aria-hidden="true" tabindex="-1"></a><span class="co"># get log2 FC as a vector with the respective gene names, sorted in decreasing </span></span>
<span id="cb80-81"><a href="#cb80-81" aria-hidden="true" tabindex="-1"></a><span class="co"># order</span></span>
<span id="cb80-82"><a href="#cb80-82" aria-hidden="true" tabindex="-1"></a>named_vectors <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb80-83"><a href="#cb80-83" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb80-84"><a href="#cb80-84" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-85"><a href="#cb80-85" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gene_list =</span> <span class="fu">map</span>(data, <span class="sc">~</span><span class="fu">setNames</span>(.x<span class="sc">$</span>log2FoldChange, .x<span class="sc">$</span>gene) <span class="sc">%&gt;%</span></span>
<span id="cb80-86"><a href="#cb80-86" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">sort</span>(<span class="at">decreasing =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb80-87"><a href="#cb80-87" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-88"><a href="#cb80-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-89"><a href="#cb80-89" aria-hidden="true" tabindex="-1"></a><span class="fu">## Gene Ontology</span></span>
<span id="cb80-90"><a href="#cb80-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-93"><a href="#cb80-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-94"><a href="#cb80-94" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the file path to save the resulting object</span></span>
<span id="cb80-95"><a href="#cb80-95" aria-hidden="true" tabindex="-1"></a>result_file <span class="ot">&lt;-</span></span>
<span id="cb80-96"><a href="#cb80-96" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb80-97"><a href="#cb80-97" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/gene-set-enrichment_go.rds"</span></span>
<span id="cb80-98"><a href="#cb80-98" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb80-99"><a href="#cb80-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-100"><a href="#cb80-100" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the file exists</span></span>
<span id="cb80-101"><a href="#cb80-101" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(result_file)) {</span>
<span id="cb80-102"><a href="#cb80-102" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Code for your time-consuming analysis</span></span>
<span id="cb80-103"><a href="#cb80-103" aria-hidden="true" tabindex="-1"></a>  gse <span class="ot">&lt;-</span></span>
<span id="cb80-104"><a href="#cb80-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map</span>(</span>
<span id="cb80-105"><a href="#cb80-105" aria-hidden="true" tabindex="-1"></a>      named_vectors<span class="sc">$</span>gene_list,</span>
<span id="cb80-106"><a href="#cb80-106" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">gseGO</span>(</span>
<span id="cb80-107"><a href="#cb80-107" aria-hidden="true" tabindex="-1"></a>        .x,</span>
<span id="cb80-108"><a href="#cb80-108" aria-hidden="true" tabindex="-1"></a>        <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span>,</span>
<span id="cb80-109"><a href="#cb80-109" aria-hidden="true" tabindex="-1"></a>        <span class="at">ont =</span> <span class="st">"ALL"</span>,</span>
<span id="cb80-110"><a href="#cb80-110" aria-hidden="true" tabindex="-1"></a>        <span class="at">keyType =</span> <span class="st">"SYMBOL"</span></span>
<span id="cb80-111"><a href="#cb80-111" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb80-112"><a href="#cb80-112" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb80-113"><a href="#cb80-113" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb80-114"><a href="#cb80-114" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the resulting object</span></span>
<span id="cb80-115"><a href="#cb80-115" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(gse, result_file)</span>
<span id="cb80-116"><a href="#cb80-116" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb80-117"><a href="#cb80-117" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the existing resulting object</span></span>
<span id="cb80-118"><a href="#cb80-118" aria-hidden="true" tabindex="-1"></a>  gse <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(result_file)</span>
<span id="cb80-119"><a href="#cb80-119" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb80-120"><a href="#cb80-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-121"><a href="#cb80-121" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gse) <span class="ot">&lt;-</span> named_vectors<span class="sc">$</span>celltype</span>
<span id="cb80-122"><a href="#cb80-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-123"><a href="#cb80-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-126"><a href="#cb80-126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-127"><a href="#cb80-127" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any with no pathways found</span></span>
<span id="cb80-128"><a href="#cb80-128" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> gse <span class="sc">%&gt;%</span></span>
<span id="cb80-129"><a href="#cb80-129" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">keep</span>(<span class="sc">~</span> <span class="fu">nrow</span>(.) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb80-130"><a href="#cb80-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-131"><a href="#cb80-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-132"><a href="#cb80-132" aria-hidden="true" tabindex="-1"></a><span class="fu">### Plots</span></span>
<span id="cb80-133"><a href="#cb80-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-136"><a href="#cb80-136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-137"><a href="#cb80-137" aria-hidden="true" tabindex="-1"></a>dotplots <span class="ot">&lt;-</span></span>
<span id="cb80-138"><a href="#cb80-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb80-139"><a href="#cb80-139" aria-hidden="true" tabindex="-1"></a>    gse,</span>
<span id="cb80-140"><a href="#cb80-140" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gse),</span>
<span id="cb80-141"><a href="#cb80-141" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">dotplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">split =</span> <span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(. <span class="sc">~</span> .sign) <span class="sc">+</span></span>
<span id="cb80-142"><a href="#cb80-142" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggtitle</span>(.y)</span>
<span id="cb80-143"><a href="#cb80-143" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb80-144"><a href="#cb80-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-145"><a href="#cb80-145" aria-hidden="true" tabindex="-1"></a>dotplots</span>
<span id="cb80-146"><a href="#cb80-146" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-147"><a href="#cb80-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-150"><a href="#cb80-150" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-151"><a href="#cb80-151" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb80-152"><a href="#cb80-152" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb80-153"><a href="#cb80-153" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gse,</span>
<span id="cb80-154"><a href="#cb80-154" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(gse),</span>
<span id="cb80-155"><a href="#cb80-155" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">emapplot</span>(.x, <span class="at">showCategory =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb80-156"><a href="#cb80-156" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">ggtitle</span>(.y))</span>
<span id="cb80-157"><a href="#cb80-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-158"><a href="#cb80-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-161"><a href="#cb80-161" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-162"><a href="#cb80-162" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gse,</span>
<span id="cb80-163"><a href="#cb80-163" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(gse),</span>
<span id="cb80-164"><a href="#cb80-164" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">ridgeplot</span>(.x) <span class="sc">+</span></span>
<span id="cb80-165"><a href="#cb80-165" aria-hidden="true" tabindex="-1"></a>       <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"enrichment distribution"</span>) <span class="sc">+</span></span>
<span id="cb80-166"><a href="#cb80-166" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">ggtitle</span>(.y))</span>
<span id="cb80-167"><a href="#cb80-167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-168"><a href="#cb80-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-169"><a href="#cb80-169" aria-hidden="true" tabindex="-1"></a>Note, one can change which pathway is plotted here by setting the title and <span class="in">`geneSetID`</span> accordingly</span>
<span id="cb80-170"><a href="#cb80-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-173"><a href="#cb80-173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-174"><a href="#cb80-174" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gse,</span>
<span id="cb80-175"><a href="#cb80-175" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(gse),</span>
<span id="cb80-176"><a href="#cb80-176" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">gseaplot</span>(.x, <span class="at">by =</span> <span class="st">"all"</span>, <span class="at">title =</span> .x<span class="sc">$</span>Description[<span class="dv">1</span>], <span class="at">geneSetID =</span> <span class="dv">1</span>))</span>
<span id="cb80-177"><a href="#cb80-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-178"><a href="#cb80-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-179"><a href="#cb80-179" aria-hidden="true" tabindex="-1"></a><span class="fu">#### PubMed trend of enriched terms</span></span>
<span id="cb80-180"><a href="#cb80-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-183"><a href="#cb80-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-184"><a href="#cb80-184" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb80-185"><a href="#cb80-185" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb80-186"><a href="#cb80-186" aria-hidden="true" tabindex="-1"></a>terms <span class="ot">&lt;-</span> gse<span class="sc">$</span><span class="st">`</span><span class="at">Neuron-ex</span><span class="st">`</span><span class="sc">$</span>Description[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb80-187"><a href="#cb80-187" aria-hidden="true" tabindex="-1"></a><span class="fu">pmcplot</span>(terms, <span class="dv">2010</span><span class="sc">:</span><span class="dv">2022</span>, <span class="at">proportion =</span> <span class="cn">FALSE</span>)</span>
<span id="cb80-188"><a href="#cb80-188" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-189"><a href="#cb80-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-190"><a href="#cb80-190" aria-hidden="true" tabindex="-1"></a><span class="fu">## KEGG</span></span>
<span id="cb80-191"><a href="#cb80-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-194"><a href="#cb80-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-195"><a href="#cb80-195" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert gene IDs for gseKEGG function</span></span>
<span id="cb80-196"><a href="#cb80-196" aria-hidden="true" tabindex="-1"></a><span class="co"># We will lose some genes here because not all IDs will be converted</span></span>
<span id="cb80-197"><a href="#cb80-197" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">&lt;-</span></span>
<span id="cb80-198"><a href="#cb80-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(</span>
<span id="cb80-199"><a href="#cb80-199" aria-hidden="true" tabindex="-1"></a>    named_vectors<span class="sc">$</span>gene_list,</span>
<span id="cb80-200"><a href="#cb80-200" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">bitr</span>(</span>
<span id="cb80-201"><a href="#cb80-201" aria-hidden="true" tabindex="-1"></a>      <span class="fu">names</span>(.x),</span>
<span id="cb80-202"><a href="#cb80-202" aria-hidden="true" tabindex="-1"></a>      <span class="at">fromType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb80-203"><a href="#cb80-203" aria-hidden="true" tabindex="-1"></a>      <span class="at">toType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb80-204"><a href="#cb80-204" aria-hidden="true" tabindex="-1"></a>      <span class="at">OrgDb =</span> <span class="st">"org.Hs.eg.db"</span></span>
<span id="cb80-205"><a href="#cb80-205" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb80-206"><a href="#cb80-206" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb80-207"><a href="#cb80-207" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">list_rbind</span>()</span>
<span id="cb80-208"><a href="#cb80-208" aria-hidden="true" tabindex="-1"></a> <span class="co"># remove duplicate IDS </span></span>
<span id="cb80-209"><a href="#cb80-209" aria-hidden="true" tabindex="-1"></a>dedup_ids <span class="ot">=</span> ids[<span class="sc">!</span><span class="fu">duplicated</span>(ids[<span class="fu">c</span>(<span class="st">"SYMBOL"</span>)]),]</span>
<span id="cb80-210"><a href="#cb80-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-211"><a href="#cb80-211" aria-hidden="true" tabindex="-1"></a><span class="co"># join entrezids to dataframe</span></span>
<span id="cb80-212"><a href="#cb80-212" aria-hidden="true" tabindex="-1"></a>res_entrez <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb80-213"><a href="#cb80-213" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(dedup_ids, <span class="at">by =</span> <span class="fu">join_by</span>(gene <span class="sc">==</span> SYMBOL)) <span class="sc">%&gt;%</span></span>
<span id="cb80-214"><a href="#cb80-214" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove cases that couldn't be matched</span></span>
<span id="cb80-215"><a href="#cb80-215" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID))</span>
<span id="cb80-216"><a href="#cb80-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-217"><a href="#cb80-217" aria-hidden="true" tabindex="-1"></a><span class="co"># get log2 FC as a vector with the respective gene names, sorted in decreasing </span></span>
<span id="cb80-218"><a href="#cb80-218" aria-hidden="true" tabindex="-1"></a><span class="co"># order</span></span>
<span id="cb80-219"><a href="#cb80-219" aria-hidden="true" tabindex="-1"></a>named_vectors <span class="ot">&lt;-</span> res_entrez <span class="sc">%&gt;%</span></span>
<span id="cb80-220"><a href="#cb80-220" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype) <span class="sc">%&gt;%</span></span>
<span id="cb80-221"><a href="#cb80-221" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() <span class="sc">%&gt;%</span></span>
<span id="cb80-222"><a href="#cb80-222" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gene_list =</span> <span class="fu">map</span>(</span>
<span id="cb80-223"><a href="#cb80-223" aria-hidden="true" tabindex="-1"></a>    data,</span>
<span id="cb80-224"><a href="#cb80-224" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">setNames</span>(.x<span class="sc">$</span>log2FoldChange, .x<span class="sc">$</span>ENTREZID) <span class="sc">%&gt;%</span></span>
<span id="cb80-225"><a href="#cb80-225" aria-hidden="true" tabindex="-1"></a>      <span class="fu">sort</span>(<span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb80-226"><a href="#cb80-226" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb80-227"><a href="#cb80-227" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-228"><a href="#cb80-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-231"><a href="#cb80-231" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-232"><a href="#cb80-232" aria-hidden="true" tabindex="-1"></a>kegg_organism <span class="ot">=</span> <span class="st">"hsa"</span></span>
<span id="cb80-233"><a href="#cb80-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-234"><a href="#cb80-234" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the file path to save the resulting object</span></span>
<span id="cb80-235"><a href="#cb80-235" aria-hidden="true" tabindex="-1"></a>result_file <span class="ot">&lt;-</span></span>
<span id="cb80-236"><a href="#cb80-236" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb80-237"><a href="#cb80-237" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/09_pathway_analysis/gene-set-enrichment_kegg.rds"</span></span>
<span id="cb80-238"><a href="#cb80-238" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb80-239"><a href="#cb80-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-240"><a href="#cb80-240" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the file exists</span></span>
<span id="cb80-241"><a href="#cb80-241" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(result_file)) {</span>
<span id="cb80-242"><a href="#cb80-242" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Code for your time-consuming analysis</span></span>
<span id="cb80-243"><a href="#cb80-243" aria-hidden="true" tabindex="-1"></a>kk2 <span class="ot">&lt;-</span></span>
<span id="cb80-244"><a href="#cb80-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(</span>
<span id="cb80-245"><a href="#cb80-245" aria-hidden="true" tabindex="-1"></a>    named_vectors<span class="sc">$</span>gene_list,</span>
<span id="cb80-246"><a href="#cb80-246" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">gseKEGG</span>(.x, <span class="at">organism =</span> kegg_organism, <span class="at">keyType =</span> <span class="st">"ncbi-geneid"</span>)</span>
<span id="cb80-247"><a href="#cb80-247" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb80-248"><a href="#cb80-248" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb80-249"><a href="#cb80-249" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the resulting object</span></span>
<span id="cb80-250"><a href="#cb80-250" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(kk2, result_file)</span>
<span id="cb80-251"><a href="#cb80-251" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb80-252"><a href="#cb80-252" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the existing resulting object</span></span>
<span id="cb80-253"><a href="#cb80-253" aria-hidden="true" tabindex="-1"></a>  kk2 <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(result_file)</span>
<span id="cb80-254"><a href="#cb80-254" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb80-255"><a href="#cb80-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-256"><a href="#cb80-256" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(kk2) <span class="ot">&lt;-</span> named_vectors<span class="sc">$</span>celltype</span>
<span id="cb80-257"><a href="#cb80-257" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-258"><a href="#cb80-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-259"><a href="#cb80-259" aria-hidden="true" tabindex="-1"></a><span class="fu">### Plots</span></span>
<span id="cb80-260"><a href="#cb80-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-263"><a href="#cb80-263" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-264"><a href="#cb80-264" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(</span>
<span id="cb80-265"><a href="#cb80-265" aria-hidden="true" tabindex="-1"></a>  kk2,</span>
<span id="cb80-266"><a href="#cb80-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(kk2),</span>
<span id="cb80-267"><a href="#cb80-267" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> <span class="fu">dotplot</span>(</span>
<span id="cb80-268"><a href="#cb80-268" aria-hidden="true" tabindex="-1"></a>    .x,</span>
<span id="cb80-269"><a href="#cb80-269" aria-hidden="true" tabindex="-1"></a>    <span class="at">showCategory =</span> <span class="dv">10</span>,</span>
<span id="cb80-270"><a href="#cb80-270" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> .y,</span>
<span id="cb80-271"><a href="#cb80-271" aria-hidden="true" tabindex="-1"></a>    <span class="at">split =</span> <span class="st">".sign"</span></span>
<span id="cb80-272"><a href="#cb80-272" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb80-273"><a href="#cb80-273" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(. <span class="sc">~</span> .sign)</span>
<span id="cb80-274"><a href="#cb80-274" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb80-275"><a href="#cb80-275" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-276"><a href="#cb80-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-279"><a href="#cb80-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-280"><a href="#cb80-280" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb80-281"><a href="#cb80-281" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb80-282"><a href="#cb80-282" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(kk2, <span class="fu">names</span>(kk2),  <span class="sc">~</span> <span class="fu">emapplot</span>(.x) <span class="sc">+</span> <span class="fu">ggtitle</span>(.y))</span>
<span id="cb80-283"><a href="#cb80-283" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-284"><a href="#cb80-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-287"><a href="#cb80-287" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb80-288"><a href="#cb80-288" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(kk2,</span>
<span id="cb80-289"><a href="#cb80-289" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(kk2),</span>
<span id="cb80-290"><a href="#cb80-290" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> <span class="fu">ridgeplot</span>(.x) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"enrichment distribution"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(.y))</span>
<span id="cb80-291"><a href="#cb80-291" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb80-292"><a href="#cb80-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-293"><a href="#cb80-293" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Pathview</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>