<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Blood brain barrier in AD - Annotate cell subtypes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="styles.css">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Blood brain barrier in AD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../06_dissemination/03_website/01_data-analysis.html" rel="" target="" aria-current="page">
 <span class="menu-text">Data analysis</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../06_dissemination/01_reports_conferences/2023-09-12_cardiff-connectome-talk/2023-09-12_cardiff-connectome-talk.html" rel="" target="">
 <span class="menu-text">Talks</span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools ms-auto">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html">Secondary analysis</a></li><li class="breadcrumb-item"><a href="../../04_data_analysis/001_snrnaseq_analysis/13_cell_subtypes.html">Annotate cell subtypes</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../06_dissemination/03_website/01_data-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis Pipeline</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Post-seq QC and Prep</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/01_fastqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run FastQC on sample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/02_multiqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run MultiQC</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/03_generate_cellranger_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/04_cellranger_count_set1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/05_cellranger_count_set2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/06_cellranger_count_set3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/07_cellranger_aggr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Aggregate cellranger count reports</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">scFlow</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/08_scflow_prep.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scflow prep</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/09_scflow_run.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run SCflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/10_scflow_results.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Analysis scFlow results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Initial analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform Pseudobulk</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform pathway analysis</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Secondary analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform Pseudobulk</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform pathway analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/13_cell_subtypes.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Annotate cell subtypes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/14_cell_populations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Cell populations</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages"><span class="header-section-number">1</span> Load packages</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data"><span class="header-section-number">2</span> Read in data</a></li>
  <li>
<a href="#cluster-celltypes" id="toc-cluster-celltypes" class="nav-link" data-scroll-target="#cluster-celltypes"><span class="header-section-number">3</span> Cluster celltypes</a>
  <ul class="collapse">
<li>
<a href="#check-umap-clusters-and-top-marker-genes" id="toc-check-umap-clusters-and-top-marker-genes" class="nav-link" data-scroll-target="#check-umap-clusters-and-top-marker-genes"><span class="header-section-number">3.1</span> Check UMAP clusters and top marker genes</a>
  <ul class="collapse">
<li><a href="#endothelial-cells" id="toc-endothelial-cells" class="nav-link" data-scroll-target="#endothelial-cells"><span class="header-section-number">3.1.1</span> Endothelial cells</a></li>
  <li><a href="#pericytes" id="toc-pericytes" class="nav-link" data-scroll-target="#pericytes"><span class="header-section-number">3.1.2</span> Pericytes</a></li>
  <li><a href="#oligodendrocytes" id="toc-oligodendrocytes" class="nav-link" data-scroll-target="#oligodendrocytes"><span class="header-section-number">3.1.3</span> Oligodendrocytes</a></li>
  <li><a href="#astrocytes" id="toc-astrocytes" class="nav-link" data-scroll-target="#astrocytes"><span class="header-section-number">3.1.4</span> Astrocytes</a></li>
  <li><a href="#vascular-and-leptomeningeal" id="toc-vascular-and-leptomeningeal" class="nav-link" data-scroll-target="#vascular-and-leptomeningeal"><span class="header-section-number">3.1.5</span> Vascular-and-leptomeningeal</a></li>
  <li><a href="#neuron" id="toc-neuron" class="nav-link" data-scroll-target="#neuron"><span class="header-section-number">3.1.6</span> Neuron</a></li>
  <li><a href="#microglia" id="toc-microglia" class="nav-link" data-scroll-target="#microglia"><span class="header-section-number">3.1.7</span> Microglia</a></li>
  <li><a href="#opcs" id="toc-opcs" class="nav-link" data-scroll-target="#opcs"><span class="header-section-number">3.1.8</span> OPCs</a></li>
  </ul>
</li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Annotate cell subtypes</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/i_am.html">i_am</a></span><span class="op">(</span><span class="st">"04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.qmd"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-packages" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Load packages</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here I’ll try clustering the higher level cell types separately to identify sub-clusters. These may be a bit tricky to annotate as I’ve already used the Yang 2022 markers<span class="citation" data-cites="yang_human_2022">(<a href="#ref-yang_human_2022" role="doc-biblioref">Yang et al. 2022</a>)</span>, but I guess we’ll see.</p>
</section><section id="read-in-data" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Read in data</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># make count rownames match data rownames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">donor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"donor-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Check the scFlow annotations</span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">sce</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"cluster_celltype"</span>, raster <span class="op">=</span> <span class="cn">FALSE</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Since I already have some subtypes, and the scFlow auto-annotations are pretty highlevel, I’ll just tweak those and use that as a bases for the initial subclustering.</p>
<p>Not sure if I should merge the oligodendrocytes and their precursors into one label?</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># add yang based annotations as a column</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">yang_annotations</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co"># Add higher level annotations </span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">highlevel_celltypes</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span></span>
<span>    <span class="va">sce</span><span class="op">$</span><span class="va">cluster_celltype</span> <span class="op">==</span> <span class="st">"IN-LAMP5"</span> <span class="op">|</span></span>
<span>      <span class="va">sce</span><span class="op">$</span><span class="va">cluster_celltype</span> <span class="op">==</span> <span class="st">"EN-L3-4"</span>,</span>
<span>    <span class="st">"Neuron"</span>,</span>
<span>    <span class="va">sce</span><span class="op">$</span><span class="va">cluster_celltype</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Check UMAP</span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"highlevel_celltypes"</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="cluster-celltypes" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Cluster celltypes</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Function to generate subclusters based on an annotation</span></span>
<span><span class="va">subcluster</span> <span class="op">&lt;-</span></span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">seurat_object</span>,</span>
<span>           <span class="va">celltypes</span>,</span>
<span>           <span class="va">result_file</span>,</span>
<span>           <span class="va">dims</span> <span class="op">=</span> <span class="fl">20</span>,</span>
<span>           <span class="va">resolution</span> <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>           <span class="va">rerun</span> <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># In order to rerun the results file must be deleted</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">rerun</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    </span>
<span>    <span class="co"># The computation takes a while to run so just load results if they exist</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">map</span><span class="op">(</span><span class="va">celltypes</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">seurat_object</span>, idents <span class="op">=</span> <span class="va">.x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html">set_names</a></span><span class="op">(</span><span class="va">celltypes</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Step 2: Cluster to the subsets</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">.</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">dims</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu">FindClusters</span><span class="op">(</span><span class="va">.</span>, resolution <span class="op">=</span> <span class="va">resolution</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Step 4: Run UMAP</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu">RunUMAP</span><span class="op">(</span><span class="va">.</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Reduce size but keep reductions and graphs</span></span>
<span>      <span class="va">reductions</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">reductions</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">graphs</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">graphs</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="va">args</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"object"</span> <span class="op">=</span> <span class="va">subsets</span>,</span>
<span>             <span class="st">"dimreducs"</span> <span class="op">=</span> <span class="va">reductions</span>,</span>
<span>             <span class="st">"graphs"</span> <span class="op">=</span> <span class="va">graphs</span><span class="op">)</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/pmap.html">pmap</a></span><span class="op">(</span><span class="va">args</span>, <span class="va">DietSeurat</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Find markers</span></span>
<span>      <span class="va">cluster_markers</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>,</span>
<span>            <span class="op">~</span> <span class="fu">FindAllMarkers</span><span class="op">(</span></span>
<span>              <span class="va">.x</span>,</span>
<span>              only.pos <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              min.pct <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>              logfc.threshold <span class="op">=</span> <span class="fl">0.5</span></span>
<span>            <span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Save subsets and cluster markers to a tibble</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">tibble</span><span class="op">(</span></span>
<span>          <span class="st">"celltype"</span> <span class="op">=</span> <span class="va">celltypes</span>,</span>
<span>          <span class="st">"seurat_obj"</span> <span class="op">=</span> <span class="va">subsets</span>,</span>
<span>          <span class="st">"cluster_markers"</span> <span class="op">=</span> <span class="va">cluster_markers</span></span>
<span>        <span class="op">)</span></span>
<span>      <span class="co"># save object</span></span>
<span>      <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">subsets</span>, <span class="va">result_file</span><span class="op">)</span></span>
<span>      </span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>    </span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">subsets</span><span class="op">)</span></span>
<span>    </span>
<span>  <span class="op">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span>      <span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">map</span><span class="op">(</span><span class="va">celltypes</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">sce</span>, idents <span class="op">=</span> <span class="va">.x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html">set_names</a></span><span class="op">(</span><span class="va">celltypes</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Step 2: Cluster to the subsets</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu">FindNeighbors</span><span class="op">(</span><span class="va">.</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="va">dims</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu">FindClusters</span><span class="op">(</span><span class="va">.</span>, resolution <span class="op">=</span> <span class="va">resolution</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Step 4: Run UMAP</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu">RunUMAP</span><span class="op">(</span><span class="va">.</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Reduce size but keep reductions and graphs</span></span>
<span>      <span class="va">reductions</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">reductions</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="va">graphs</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">.x</span><span class="op">@</span><span class="va">graphs</span><span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="va">args</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"object"</span> <span class="op">=</span> <span class="va">subsets</span>,</span>
<span>             <span class="st">"dimreducs"</span> <span class="op">=</span> <span class="va">reductions</span>,</span>
<span>             <span class="st">"graphs"</span> <span class="op">=</span> <span class="va">graphs</span><span class="op">)</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/pmap.html">pmap</a></span><span class="op">(</span><span class="va">args</span>, <span class="va">DietSeurat</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Find markers</span></span>
<span>      <span class="va">cluster_markers</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span>,</span>
<span>            <span class="op">~</span> <span class="fu">FindAllMarkers</span><span class="op">(</span></span>
<span>              <span class="va">.x</span>,</span>
<span>              only.pos <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>              min.pct <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>              logfc.threshold <span class="op">=</span> <span class="fl">0.5</span></span>
<span>            <span class="op">)</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="co"># Save subsets and cluster markers to a tibble</span></span>
<span>      <span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>        <span class="fu">tibble</span><span class="op">(</span></span>
<span>          <span class="st">"celltype"</span> <span class="op">=</span> <span class="va">celltypes</span>,</span>
<span>          <span class="st">"seurat_obj"</span> <span class="op">=</span> <span class="va">subsets</span>,</span>
<span>          <span class="st">"cluster_markers"</span> <span class="op">=</span> <span class="va">cluster_markers</span></span>
<span>        <span class="op">)</span></span>
<span>      <span class="co"># save object</span></span>
<span>      <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">subsets</span>, <span class="va">result_file</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span><span class="va">celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">highlevel_celltypes</span><span class="op">)</span></span>
<span><span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">$</span><span class="va">highlevel_celltypes</span></span>
<span></span>
<span><span class="co"># There is an error if a metadata column has the same name as a gene</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">apoe_status</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">$</span><span class="va">APOE</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">APOE</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="va">subsets</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">subcluster</span><span class="op">(</span></span>
<span>    <span class="va">sce</span>,</span>
<span>    <span class="va">celltypes</span>,</span>
<span>    <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>      <span class="st">"03_data/990_processed_data/001_snrnaseq/10_cell_subtypes/subclusters_highlevel-celltypes.rds"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 70185
Number of edges: 2200772

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8817
Number of communities: 11
Elapsed time: 20 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 55163
Number of edges: 1609707

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8755
Number of communities: 9
Elapsed time: 11 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 159999
Number of edges: 3914416

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9130
Number of communities: 18
Elapsed time: 46 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 81510
Number of edges: 2640391

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9207
Number of communities: 14
Elapsed time: 20 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 6391
Number of edges: 243592

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8685
Number of communities: 8
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 67244
Number of edges: 2322787

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9422
Number of communities: 20
Elapsed time: 14 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15591
Number of edges: 549194

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8824
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 18274
Number of edges: 550483

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8730
Number of communities: 9
Elapsed time: 2 seconds</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Clean up environment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">sce</span>, <span class="va">celltypes</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="check-umap-clusters-and-top-marker-genes" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="check-umap-clusters-and-top-marker-genes">
<span class="header-section-number">3.1</span> Check UMAP clusters and top marker genes</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Step 3: Visualize the UMAP for each cell subtype</span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">map2</span><span class="op">(</span></span>
<span>    <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span>,</span>
<span>    <span class="va">subsets</span><span class="op">$</span><span class="va">celltype</span>,</span>
<span>    <span class="op">~</span> <span class="fu">DimPlot</span><span class="op">(</span></span>
<span>      <span class="va">.</span>,</span>
<span>      reduction <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>      label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>      <span class="fu">ggtitle</span><span class="op">(</span><span class="va">.y</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Check how many genes aren't significant and which have blank gene names</span></span>
<span><span class="fu">map</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">cluster_markers</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">p_val_adj</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$Endothelial
[1] 184505

$Pericytes
[1] 15366

$Oligodendrocytes
[1] 776559

$Astrocytes
[1] 1231493

$`Vascular-and-leptomeningeal`
[1] 30148

$Neuron
[1] 2363406

$Microglia
[1] 27410

$OPCs
[1] 709011</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">map</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">cluster_markers</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">gene</span> <span class="op">==</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$Endothelial
[1] 0

$Pericytes
[1] 0

$Oligodendrocytes
[1] 0

$Astrocytes
[1] 0

$`Vascular-and-leptomeningeal`
[1] 0

$Neuron
[1] 0

$Microglia
[1] 0

$OPCs
[1] 0</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># get top 10 genes per cluster</span></span>
<span><span class="va">top10</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">cluster_markers</span>, <span class="op">~</span> <span class="va">.x</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># filter to sig genes</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="co"># get top genes by avg log2FC</span></span>
<span>  <span class="fu">top_n</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">to_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">top10</span><span class="op">$</span><span class="va">Endothelial</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span></span>
<span><span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu">DotPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Endothelial</span>, features <span class="op">=</span> <span class="va">to_plot</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">coord_flip</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="1920"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># nest top 10 markers by cluster</span></span>
<span><span class="va">top10</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">top10</span>, <span class="va">nest</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># function to rename apoe column</span></span>
<span><span class="co"># modify_sce &lt;- function(sce) {</span></span>
<span><span class="co">#   sce$apoe_status &lt;- sce$APOE</span></span>
<span><span class="co">#   sce$APOE &lt;- NULL</span></span>
<span><span class="co">#   return(sce)</span></span>
<span><span class="co"># }</span></span>
<span><span class="co"># to get points to appear with raster = T, increase point size</span></span>
<span><span class="co">#FeaturePlot(seurat, features = markers, pt.size = 1.5)</span></span>
<span><span class="co"># This takes a while to run, so save the output and only rerun if needed</span></span>
<span><span class="va">marker_plot_results</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>    <span class="st">"03_data/990_processed_data/001_snrnaseq/10_cell_subtypes/subclusters_highlevel-cluster-markers.rds"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">marker_plot_results</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># subsets$seurat_obj &lt;- map(subsets$seurat_obj, modify_sce)</span></span>
<span>  </span>
<span>  <span class="va">sce_markers_plot</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu">map2</span><span class="op">(</span><span class="va">top10</span>, <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span>, <span class="kw">function</span><span class="op">(</span><span class="va">top</span>, <span class="va">seurat</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu">map2</span><span class="op">(</span><span class="va">top</span><span class="op">$</span><span class="va">data</span>, <span class="va">top</span><span class="op">$</span><span class="va">cluster</span>, <span class="kw">function</span><span class="op">(</span><span class="va">data</span>, <span class="va">cluster</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu">FeaturePlot</span><span class="op">(</span><span class="va">seurat</span>,</span>
<span>                    features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>,</span>
<span>                    raster <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>          <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html">plot_annotation</a></span><span class="op">(</span>title <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"Seurat cluster: "</span>, <span class="va">cluster</span><span class="op">)</span><span class="op">)</span></span>
<span>      <span class="op">}</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">set_names</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"cluster_"</span>, <span class="va">top</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">sce_markers_plot</span>, <span class="va">marker_plot_results</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="va">sce_markers_plot</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="va">marker_plot_results</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="endothelial-cells" class="level3" data-number="3.1.1"><h3 data-number="3.1.1" class="anchored" data-anchor-id="endothelial-cells">
<span class="header-section-number">3.1.1</span> Endothelial cells</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">Endothelial</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">Endothelial</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-10.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_9</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-11.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_10</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-13-12.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Cluster 1 is somewhat between Arterial and an astrocyte subtype</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Endothelial subtypes</span></span>
<span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"Arterial"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"Arterial"</span>, <span class="st">"2"</span> <span class="op">=</span> <span class="st">"Capillary"</span>, </span>
<span>                   <span class="st">"3"</span> <span class="op">=</span> <span class="st">"Venous"</span>, <span class="st">"4"</span> <span class="op">=</span> <span class="st">"Capillary/Venous"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Arterial"</span>, </span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Capillary"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"T-Pericyte"</span>, </span>
<span>                   <span class="st">"8"</span> <span class="op">=</span> <span class="st">"Oligodendrocyte/OPC"</span>, <span class="st">"9"</span> <span class="op">=</span> <span class="st">"T-Cell"</span>, <span class="st">"10"</span> <span class="op">=</span> <span class="st">"Astrocyte"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Endothelial</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Endothelial</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Endothelial</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Endothelial</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Endothelial</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="pericytes" class="level3" data-number="3.1.2"><h3 data-number="3.1.2" class="anchored" data-anchor-id="pericytes">
<span class="header-section-number">3.1.2</span> Pericytes</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">Pericytes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">Pericytes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-15-10.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Note that M-Pericyte is ECM-regulating Pericyte</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Pericyte subtypes</span></span>
<span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"Vascular-SMC"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"T-Pericyte"</span>, <span class="st">"2"</span> <span class="op">=</span> <span class="st">"M-Pericyte"</span>, </span>
<span>                   <span class="st">"3"</span> <span class="op">=</span> <span class="st">"T-Pericyte"</span>, <span class="st">"4"</span> <span class="op">=</span> <span class="st">"Vascular-SMC/OPC"</span>, </span>
<span>                   <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Vascular-SMC"</span>, <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Capillary"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"OPC"</span>, </span>
<span>                   <span class="st">"8"</span> <span class="op">=</span> <span class="st">"Perivascular Macrophage"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Pericytes</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Pericytes</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Pericytes</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Pericytes</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Pericytes</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section><section id="oligodendrocytes" class="level3" data-number="3.1.3"><h3 data-number="3.1.3" class="anchored" data-anchor-id="oligodendrocytes">
<span class="header-section-number">3.1.3</span> Oligodendrocytes</h3>
<p>Anna Williams seems to know a lot about oligodendrocytes, so I might look at their stuff for these annotations. Viloa used her markers and found they worked well apparently.</p>
<p>The cluster analysis revealed six oligodendrocyte states (positive for myelin genes such as PLP1, MBP, MAG), two OPC states (positive for PDGFRA, CSPG4 and BCAN) and three committed oligodendrocyte precursor (COP) cell states (positive for GPR17, GPC5 or GAP43), confirming that oligodendroglia are heterogeneous in the healthy adult human CNS (Figure 2a-c).</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">Oligodendrocytes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">Oligodendrocytes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-10.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_9</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-11.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_10</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-12.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_11</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-13.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_12</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-14.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_13</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-15.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_14</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-16.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_15</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-17-17.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>She reports RBFOX1 as a marker for all but two of her oligodendrocyte clusters (A and F), so lets check that marker</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb63"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"RBFOX1"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Seems like it’s only really heavily expressed in clusters 3, 4, 7 and 9 curiously…</p>
<p>Anna Williams paper annotates “Oligo_B” as AFF3+ LGALS1- FMN1-</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb64"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># anna williams oligo B markers</span></span>
<span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AFF3"</span>, <span class="st">"LGALS1"</span>, <span class="st">"FMN1"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Seems to reflect cluster 4, and maybe 5</p>
<p>Oligo_C: FMN1</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb65"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FMN1"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Not super strong expression accross a single cluster, but seems to be in 12, 9 and maybe 9?</p>
<p>Oligo_D: LGALS1+ FMN1- SPARC-</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb66"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"LGALS1"</span>, <span class="st">"FMN1"</span>, <span class="st">"SPARC"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This doesn’t really fit at all…</p>
<p>Oligo_E: HHIP+ OPALIN- AFF3-</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb67"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HHIP"</span>, <span class="st">"OPALIN"</span>, <span class="st">"AFF3"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Well the left clusters express HHIP, but the same clusters express OPALIN, so this doesn’t fit either…</p>
<p>Oligo_A expresses OPALIN, PLXDC2, LAMA2 and PALM2, and Oligo_F expresses SPARC, DHCR24, TUBA1A, TUBB2B, PMP2</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb68"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"OPALIN"</span>, <span class="st">"PLXDC2"</span>, <span class="st">"LAMA2"</span>, <span class="st">"PALM2"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb69"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SPARC"</span>, <span class="st">"DHCR24"</span>, <span class="st">"TUBA1A"</span>, <span class="st">"TUBB2B"</span>, <span class="st">"PMP2"</span><span class="op">)</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-23-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>I guess for now I’ll just label them as RBFOX1 pos/neg oligodendrocytes since her group reports this as a key distinguishing feature.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb70"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Astrocyte subtypes</span></span>
<span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, </span>
<span>                   <span class="st">"2"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"3"</span> <span class="op">=</span> <span class="st">"RBFOX1-Oligo"</span>,</span>
<span>                   <span class="st">"4"</span> <span class="op">=</span> <span class="st">"RBFOX1-Oligo"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>,</span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"RBFOX1-Oligo"</span>,</span>
<span>                   <span class="st">"8"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"9"</span> <span class="op">=</span> <span class="st">"RBFOX1-Oligo"</span>, </span>
<span>                   <span class="st">"10"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"11"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, </span>
<span>                   <span class="st">"12"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"13"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>,</span>
<span>                   <span class="st">"14"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"15"</span> <span class="op">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span></span>
<span>  <span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Oligodendrocytes</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">"umap"</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  raster <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section><section id="astrocytes" class="level3" data-number="3.1.4"><h3 data-number="3.1.4" class="anchored" data-anchor-id="astrocytes">
<span class="header-section-number">3.1.4</span> Astrocytes</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb71"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">Astrocytes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb72"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">Astrocytes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-10.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_9</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-11.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_10</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-12.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_11</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-13.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_12</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-14.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_13</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-25-15.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Cluster 8 seems to be a bit mixed between the pericyte and capillary types in the Yang data. Cluster 13 doesn’t map super well to the Yang data</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb87"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Astrocyte subtypes</span></span>
<span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"Astrocyte-Cortex"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"Capillary"</span>, </span>
<span>                   <span class="st">"2"</span> <span class="op">=</span> <span class="st">"Astrocyte-Cortex"</span>, <span class="st">"3"</span> <span class="op">=</span> <span class="st">"Astrocyte-Hpc"</span>,</span>
<span>                   <span class="st">"4"</span> <span class="op">=</span> <span class="st">"T-Pericyte"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Oligodendrocyte"</span>,</span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Astrocyte-Hpc"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"Astrocyte-Hpc"</span>,</span>
<span>                   <span class="st">"8"</span> <span class="op">=</span> <span class="st">"M-Pericyte"</span>, <span class="st">"9"</span> <span class="op">=</span> <span class="st">"Neuron"</span>, <span class="st">"10"</span> <span class="op">=</span> <span class="st">"Capillary"</span>, </span>
<span>                   <span class="st">"11"</span> <span class="op">=</span> <span class="st">"Microglia"</span>, <span class="st">"12"</span> <span class="op">=</span> <span class="st">"T-Cell"</span>, <span class="st">"13"</span> <span class="op">=</span> <span class="st">"Capillary"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Astrocytes</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Astrocytes</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Astrocytes</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Astrocytes</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Astrocytes</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section><section id="vascular-and-leptomeningeal" class="level3" data-number="3.1.5"><h3 data-number="3.1.5" class="anchored" data-anchor-id="vascular-and-leptomeningeal">
<span class="header-section-number">3.1.5</span> Vascular-and-leptomeningeal</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb88"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb89"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-27-9.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>FB = fibroblast. Cluster 2 doesn’t map very well to Yang. Not sure if cluster 5 maps better to one of the SMC subtypes or pericyte subtypes in the Yang data</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb98"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"Perivascular-FB"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"Perivascular-FB"</span>, </span>
<span>                   <span class="st">"2"</span> <span class="op">=</span> <span class="st">"Meningeal-FB"</span>, <span class="st">"3"</span> <span class="op">=</span> <span class="st">"T-Pericyte"</span>, </span>
<span>                   <span class="st">"4"</span> <span class="op">=</span> <span class="st">"Meningeal-FB"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Vascular-SMC/M-Pericyte"</span>,</span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Capillary/Venous"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"Oligodendrocyte"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">`Vascular-and-leptomeningeal`</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section><section id="neuron" class="level3" data-number="3.1.6"><h3 data-number="3.1.6" class="anchored" data-anchor-id="neuron">
<span class="header-section-number">3.1.6</span> Neuron</h3>
<p>The Yang dataset doesn’t have neuronal subtypes, so I’ll skip this for now. Anna Williams paper and maybe the Paul Matthews paper have some I think.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb99"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">Neuron</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">Neuron</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-10.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_9</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-11.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_10</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-12.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_11</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-13.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_12</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-14.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_13</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-15.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_14</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-16.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_15</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-17.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_16</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-18.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_17</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-19.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_18</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-20.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_19</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-29-21.png" class="img-fluid" width="1152"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb121"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">frank_annotation</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">readxl</span><span class="fu">::</span><span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html">read_excel</a></span><span class="op">(</span></span>
<span>    <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>      <span class="st">"03_data/990_processed_data/001_snrnaseq/91_data_from_frank/Cell Markers - Frank - First 20.xlsx"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">frank_annotation</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
  `List No` `Gene Rank by Cell` gene   avg_log2FC pct.1 pct.2 p_val_adj cluster
      &lt;dbl&gt;               &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;  
1         1                   1 FLT1         4.75 0.97  0.132         0 E      
2         2                   2 ATP10A       4.34 0.953 0.107         0 E      
3         3                   3 ABCB1        4.30 0.948 0.104         0 E      
4         4                   4 MECOM        4.17 0.936 0.077         0 E      
5         5                   5 ANO2         4.01 0.926 0.106         0 E      
6         6                   6 VWF          3.92 0.932 0.083         0 E      
# ℹ 4 more variables: N &lt;dbl&gt;, in_cluster &lt;chr&gt;, gene_type &lt;chr&gt;,
#   description &lt;chr&gt;</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb123"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">frank_annotation</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code> [1] "E"  "T"  "P"  "S"  "O"  "A"  "NE" "F"  "NI" "M"  "OP"</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb125"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">neuro_i</span> <span class="op">&lt;-</span> <span class="va">frank_annotation</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># filter to inhibitory neuron makrers that aren't in excitatory group</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">cluster</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NI"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"NE"</span>, <span class="va">in_cluster</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># get top 10 markers by fold change</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span></span>
<span></span>
<span><span class="va">neuron_i_plot</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">FeaturePlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>,</span>
<span>              features <span class="op">=</span> <span class="va">neuro_i</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>              raster <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html">plot_annotation</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Inhibitory neuronal markers"</span><span class="op">)</span></span>
<span><span class="va">neuron_i_plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Clusters 2, 7, 8, 10 and 13 look look to express these inhibitory markers well.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb126"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">neuro_e</span> <span class="op">&lt;-</span> <span class="va">frank_annotation</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># filter to inhibitory neuron makrers that aren't in excitatory group</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">cluster</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NE"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"NI"</span>, <span class="va">in_cluster</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># get top 10 markers by fold change</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span></span>
<span></span>
<span><span class="va">neuron_e_plot</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">FeaturePlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>,</span>
<span>              features <span class="op">=</span> <span class="va">neuro_e</span><span class="op">$</span><span class="va">gene</span>,</span>
<span>              raster <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html">plot_annotation</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Excitatory neuronal markers"</span><span class="op">)</span></span>
<span><span class="va">neuron_e_plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>The large group of clusters in the bottom right seem look like excitatory neurons.</p>
<p>Anna Williams annotated reelin-positive neurons, so let’s check reelin in these clusters</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb127"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"RELN"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_annotation.html">plot_annotation</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Reelin neurons"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<p>Seems like a subset of cells that span clusters 2 and 7 are expressing RELN, so I suppose those could be regarded as reelin-positive. The fact that it isn’t expressed ubiquitoously throughout the clusters gives me pause though… I’ll leave them as inhibitory neurons for now.</p>
<p>Clusters 12 and 4 (and 18 to a lesser extent) don’t seem to express the inhibitory or excitatory markers I’m using here. Not sure what to label them as currently.</p>
<p>The markers the Anna Williams paper reports seem to line up too (excitatory neurons - SNAP25, SLC17A7; inhibitory neurons - SNAP25, GAD1).</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb128"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SLC17A7"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb129"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">FeaturePlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GAD1"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-33-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb130"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"2"</span> <span class="op">=</span> <span class="st">"In-Neuron"</span>, </span>
<span>                   <span class="st">"3"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"4"</span> <span class="op">=</span> <span class="st">"ambiguous"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>,</span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"In-Neuron"</span>,  <span class="st">"8"</span> <span class="op">=</span> <span class="st">"In-Neuron"</span>, </span>
<span>                   <span class="st">"9"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"10"</span> <span class="op">=</span> <span class="st">"In-Neuron"</span>, <span class="st">"11"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, </span>
<span>                   <span class="st">"12"</span> <span class="op">=</span> <span class="st">"ambiguous"</span>, <span class="st">"13"</span> <span class="op">=</span> <span class="st">"In-Neuron"</span>, <span class="st">"14"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>,</span>
<span>                   <span class="st">"15"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"16"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"17"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>,</span>
<span>                   <span class="st">"18"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"19"</span> <span class="op">=</span> <span class="st">"Ex-Neuron"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Neuron</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section><section id="microglia" class="level3" data-number="3.1.7"><h3 data-number="3.1.7" class="anchored" data-anchor-id="microglia">
<span class="header-section-number">3.1.7</span> Microglia</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb131"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">Microglia</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb132"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">Microglia</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-10.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_9</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-11.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_10</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-35-12.png" class="img-fluid" width="1152"></p>
</div>
</div>
<p>Cluster 0 maps to both perivascular macrophages and microglia - not clear which label to apply. The mapping in this subcluster to the Yang data is somewhat rough in several clusters. Would be good to get different marker lists to try.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb144"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"Microglia"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"Microglia"</span>, </span>
<span>                   <span class="st">"2"</span> <span class="op">=</span> <span class="st">"Perivascular Macrophage"</span>, <span class="st">"3"</span> <span class="op">=</span> <span class="st">"Microglia"</span>, </span>
<span>                   <span class="st">"4"</span> <span class="op">=</span> <span class="st">"Oligodendrocyte"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Perivascular Macrophage"</span>,</span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Neuron"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"Meningeal-FB/Ependymal"</span>, </span>
<span>                   <span class="st">"8"</span> <span class="op">=</span> <span class="st">"Capillary"</span>, <span class="st">"9"</span> <span class="op">=</span> <span class="st">"Astrocyte-Cortex"</span>,</span>
<span>                   <span class="st">"10"</span> <span class="op">=</span> <span class="st">"OPC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Microglia</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Microglia</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Microglia</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Microglia</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">Microglia</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="960"></p>
</div>
</div>
</section><section id="opcs" class="level3" data-number="3.1.8"><h3 data-number="3.1.8" class="anchored" data-anchor-id="opcs">
<span class="header-section-number">3.1.8</span> OPCs</h3>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb145"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plots</span><span class="op">$</span><span class="va">OPCs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="1152"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb146"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sce_markers_plot</span><span class="op">$</span><span class="va">OPCs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$cluster_0</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-2.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_1</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-3.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_2</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-4.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_3</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-5.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_4</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-6.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_5</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-7.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_6</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-8.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_7</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-9.png" class="img-fluid" width="1152"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$cluster_8</code></pre>
</div>
<div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-37-10.png" class="img-fluid" width="1152"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb156"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sub_celltypes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span> <span class="op">=</span> <span class="st">"OPC"</span>, <span class="st">"1"</span> <span class="op">=</span> <span class="st">"Microglia"</span>, </span>
<span>                   <span class="st">"2"</span> <span class="op">=</span> <span class="st">"ambiguous"</span>, <span class="st">"3"</span> <span class="op">=</span> <span class="st">"Oligodendrocyte"</span>, </span>
<span>                   <span class="st">"4"</span> <span class="op">=</span> <span class="st">"Neuron"</span>, <span class="st">"5"</span> <span class="op">=</span> <span class="st">"Meningeal-FB"</span>,</span>
<span>                   <span class="st">"6"</span> <span class="op">=</span> <span class="st">"Astrocyte-Cortex"</span>, <span class="st">"7"</span> <span class="op">=</span> <span class="st">"Neuron"</span>, </span>
<span>                   <span class="st">"8"</span> <span class="op">=</span> <span class="st">"Capillary/T-Perictye"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add the new celltypes</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">OPCs</span><span class="op">$</span><span class="va">subclusters</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Idents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">OPCs</span><span class="op">)</span></span>
<span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">OPCs</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">RenameIdents</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">OPCs</span>, <span class="va">sub_celltypes</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">DimPlot</span><span class="op">(</span><span class="va">subsets</span><span class="op">$</span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">OPCs</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="13_cell_subtypes_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="960"></p>
</div>
</div>
<!-- # Use Yang 2022 data to query cell types -->
<!-- The Seurat package has functions for querying celltype data from one dataset to another -->
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb157"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">gds</span> <span class="op">&lt;-</span> <span class="fu">getGEO</span><span class="op">(</span><span class="st">"GSE163577"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="fu">pData</span><span class="op">(</span><span class="fu">phenoData</span><span class="op">(</span><span class="va">gds</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">6</span>,<span class="fl">8</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expr_matrix</span> <span class="op">&lt;-</span> <span class="fu">exprs</span><span class="op">(</span><span class="va">gds</span><span class="op">$</span><span class="va">GSE163577_series_matrix.txt.gz</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu">CreateSeuratObject</span><span class="op">(</span>counts <span class="op">=</span> <span class="va">expr_matrix</span><span class="op">)</span></span>
<span><span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">seurat_obj</span>, metadata <span class="op">=</span> <span class="va">gds</span><span class="op">$</span><span class="va">GSE163577_series_matrix.txt.gz</span><span class="op">$</span><span class="va">`brain region:ch1`</span>, col.name <span class="op">=</span> <span class="st">"brain_region"</span><span class="op">)</span></span>
<span><span class="fu">AddMetaData</span><span class="op">(</span>object <span class="op">=</span> <span class="va">seurat_obj</span>, metadata <span class="op">=</span> <span class="va">gds</span><span class="op">$</span><span class="va">GSE163577_series_matrix.txt.gz</span><span class="op">$</span><span class="va">`disease:ch1`</span>, col.name <span class="op">=</span> <span class="st">"disease_status"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->


</section></section></section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-yang_human_2022" class="csl-entry" role="listitem">
Yang, Andrew C., Ryan T. Vest, Fabian Kern, Davis P. Lee, Maayan Agam, Christina A. Maat, Patricia M. Losada, et al. 2022. <span>“A Human Brain Vascular Atlas Reveals Diverse Mediators of <span>Alzheimer</span>’s Risk.”</span> <em>Nature</em> 603 (7903): 885–92. <a href="https://doi.org/10.1038/s41586-021-04369-3">https://doi.org/10.1038/s41586-021-04369-3</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb158" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Annotate cell subtypes</span></span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: true</span></span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a><span class="co">#self-contained: true</span></span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> true</span></span>
<span id="cb158-7"><a href="#cb158-7" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb158-8"><a href="#cb158-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb158-9"><a href="#cb158-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-12"><a href="#cb158-12" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-13"><a href="#cb158-13" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.qmd"</span>)</span>
<span id="cb158-14"><a href="#cb158-14" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-15"><a href="#cb158-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-16"><a href="#cb158-16" aria-hidden="true" tabindex="-1"></a><span class="fu"># Load packages</span></span>
<span id="cb158-17"><a href="#cb158-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-20"><a href="#cb158-20" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-21"><a href="#cb158-21" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span>
<span id="cb158-22"><a href="#cb158-22" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-23"><a href="#cb158-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-24"><a href="#cb158-24" aria-hidden="true" tabindex="-1"></a>Here I'll try clustering the higher level cell types separately to identify sub-clusters.</span>
<span id="cb158-25"><a href="#cb158-25" aria-hidden="true" tabindex="-1"></a>These may be a bit tricky to annotate as I've already used the Yang 2022 markers<span class="co">[</span><span class="ot">@yang_human_2022</span><span class="co">]</span>, but I guess we'll see.</span>
<span id="cb158-26"><a href="#cb158-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-27"><a href="#cb158-27" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read in data</span></span>
<span id="cb158-28"><a href="#cb158-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-31"><a href="#cb158-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-32"><a href="#cb158-32" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb158-33"><a href="#cb158-33" aria-hidden="true" tabindex="-1"></a><span class="co"># read data</span></span>
<span id="cb158-34"><a href="#cb158-34" aria-hidden="true" tabindex="-1"></a><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span id="cb158-35"><a href="#cb158-35" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span>))</span>
<span id="cb158-36"><a href="#cb158-36" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-37"><a href="#cb158-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-40"><a href="#cb158-40" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-41"><a href="#cb158-41" aria-hidden="true" tabindex="-1"></a><span class="co"># make count rownames match data rownames</span></span>
<span id="cb158-42"><a href="#cb158-42" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data)</span>
<span id="cb158-43"><a href="#cb158-43" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>donor <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"donor-"</span>, sce<span class="sc">$</span>individual)</span>
<span id="cb158-44"><a href="#cb158-44" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-45"><a href="#cb158-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-48"><a href="#cb158-48" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-49"><a href="#cb158-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the scFlow annotations</span></span>
<span id="cb158-50"><a href="#cb158-50" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(sce, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">group.by =</span> <span class="st">"cluster_celltype"</span>, <span class="at">raster =</span> <span class="cn">FALSE</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-51"><a href="#cb158-51" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-52"><a href="#cb158-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-53"><a href="#cb158-53" aria-hidden="true" tabindex="-1"></a>Since I already have some subtypes, and the scFlow auto-annotations are pretty highlevel, I'll just tweak those and use that as a bases for the initial subclustering.</span>
<span id="cb158-54"><a href="#cb158-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-55"><a href="#cb158-55" aria-hidden="true" tabindex="-1"></a>Not sure if I should merge the oligodendrocytes and their precursors into one label?</span>
<span id="cb158-56"><a href="#cb158-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-59"><a href="#cb158-59" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-60"><a href="#cb158-60" aria-hidden="true" tabindex="-1"></a><span class="co"># add yang based annotations as a column</span></span>
<span id="cb158-61"><a href="#cb158-61" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>yang_annotations <span class="ot">&lt;-</span> <span class="fu">Idents</span>(sce)</span>
<span id="cb158-62"><a href="#cb158-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Add higher level annotations </span></span>
<span id="cb158-63"><a href="#cb158-63" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>highlevel_celltypes <span class="ot">&lt;-</span></span>
<span id="cb158-64"><a href="#cb158-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(</span>
<span id="cb158-65"><a href="#cb158-65" aria-hidden="true" tabindex="-1"></a>    sce<span class="sc">$</span>cluster_celltype <span class="sc">==</span> <span class="st">"IN-LAMP5"</span> <span class="sc">|</span></span>
<span id="cb158-66"><a href="#cb158-66" aria-hidden="true" tabindex="-1"></a>      sce<span class="sc">$</span>cluster_celltype <span class="sc">==</span> <span class="st">"EN-L3-4"</span>,</span>
<span id="cb158-67"><a href="#cb158-67" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Neuron"</span>,</span>
<span id="cb158-68"><a href="#cb158-68" aria-hidden="true" tabindex="-1"></a>    sce<span class="sc">$</span>cluster_celltype</span>
<span id="cb158-69"><a href="#cb158-69" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb158-70"><a href="#cb158-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-71"><a href="#cb158-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Check UMAP</span></span>
<span id="cb158-72"><a href="#cb158-72" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb158-73"><a href="#cb158-73" aria-hidden="true" tabindex="-1"></a>  sce,</span>
<span id="cb158-74"><a href="#cb158-74" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb158-75"><a href="#cb158-75" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"highlevel_celltypes"</span>,</span>
<span id="cb158-76"><a href="#cb158-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span>,</span>
<span id="cb158-77"><a href="#cb158-77" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="cn">TRUE</span></span>
<span id="cb158-78"><a href="#cb158-78" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-79"><a href="#cb158-79" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-80"><a href="#cb158-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-81"><a href="#cb158-81" aria-hidden="true" tabindex="-1"></a><span class="fu"># Cluster celltypes</span></span>
<span id="cb158-82"><a href="#cb158-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-85"><a href="#cb158-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-86"><a href="#cb158-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate subclusters based on an annotation</span></span>
<span id="cb158-87"><a href="#cb158-87" aria-hidden="true" tabindex="-1"></a>subcluster <span class="ot">&lt;-</span></span>
<span id="cb158-88"><a href="#cb158-88" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(seurat_object,</span>
<span id="cb158-89"><a href="#cb158-89" aria-hidden="true" tabindex="-1"></a>           celltypes,</span>
<span id="cb158-90"><a href="#cb158-90" aria-hidden="true" tabindex="-1"></a>           result_file,</span>
<span id="cb158-91"><a href="#cb158-91" aria-hidden="true" tabindex="-1"></a>           <span class="at">dims =</span> <span class="dv">20</span>,</span>
<span id="cb158-92"><a href="#cb158-92" aria-hidden="true" tabindex="-1"></a>           <span class="at">resolution =</span> <span class="fl">0.4</span>,</span>
<span id="cb158-93"><a href="#cb158-93" aria-hidden="true" tabindex="-1"></a>           <span class="at">rerun =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb158-94"><a href="#cb158-94" aria-hidden="true" tabindex="-1"></a>    <span class="co"># In order to rerun the results file must be deleted</span></span>
<span id="cb158-95"><a href="#cb158-95" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (rerun) {</span>
<span id="cb158-96"><a href="#cb158-96" aria-hidden="true" tabindex="-1"></a>      <span class="fu">file.remove</span>(result_file)</span>
<span id="cb158-97"><a href="#cb158-97" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb158-98"><a href="#cb158-98" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-99"><a href="#cb158-99" aria-hidden="true" tabindex="-1"></a>    <span class="co"># The computation takes a while to run so just load results if they exist</span></span>
<span id="cb158-100"><a href="#cb158-100" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(result_file)) {</span>
<span id="cb158-101"><a href="#cb158-101" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span id="cb158-102"><a href="#cb158-102" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span></span>
<span id="cb158-103"><a href="#cb158-103" aria-hidden="true" tabindex="-1"></a>        <span class="fu">map</span>(celltypes, <span class="sc">~</span> <span class="fu">subset</span>(seurat_object, <span class="at">idents =</span> .x)) <span class="sc">%&gt;%</span></span>
<span id="cb158-104"><a href="#cb158-104" aria-hidden="true" tabindex="-1"></a>        purrr<span class="sc">::</span><span class="fu">set_names</span>(celltypes)</span>
<span id="cb158-105"><a href="#cb158-105" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-106"><a href="#cb158-106" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Step 2: Cluster to the subsets</span></span>
<span id="cb158-107"><a href="#cb158-107" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">FindNeighbors</span>(., <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>dims))</span>
<span id="cb158-108"><a href="#cb158-108" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span></span>
<span id="cb158-109"><a href="#cb158-109" aria-hidden="true" tabindex="-1"></a>        <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">FindClusters</span>(., <span class="at">resolution =</span> resolution))</span>
<span id="cb158-110"><a href="#cb158-110" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-111"><a href="#cb158-111" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Step 4: Run UMAP</span></span>
<span id="cb158-112"><a href="#cb158-112" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">RunUMAP</span>(., <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>))</span>
<span id="cb158-113"><a href="#cb158-113" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-114"><a href="#cb158-114" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Reduce size but keep reductions and graphs</span></span>
<span id="cb158-115"><a href="#cb158-115" aria-hidden="true" tabindex="-1"></a>      reductions <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">names</span>(.x<span class="sc">@</span>reductions))</span>
<span id="cb158-116"><a href="#cb158-116" aria-hidden="true" tabindex="-1"></a>      graphs <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">names</span>(.x<span class="sc">@</span>graphs))</span>
<span id="cb158-117"><a href="#cb158-117" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-118"><a href="#cb158-118" aria-hidden="true" tabindex="-1"></a>      args <span class="ot">&lt;-</span></span>
<span id="cb158-119"><a href="#cb158-119" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="st">"object"</span> <span class="ot">=</span> subsets,</span>
<span id="cb158-120"><a href="#cb158-120" aria-hidden="true" tabindex="-1"></a>             <span class="st">"dimreducs"</span> <span class="ot">=</span> reductions,</span>
<span id="cb158-121"><a href="#cb158-121" aria-hidden="true" tabindex="-1"></a>             <span class="st">"graphs"</span> <span class="ot">=</span> graphs)</span>
<span id="cb158-122"><a href="#cb158-122" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">pmap</span>(args, DietSeurat)</span>
<span id="cb158-123"><a href="#cb158-123" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-124"><a href="#cb158-124" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Find markers</span></span>
<span id="cb158-125"><a href="#cb158-125" aria-hidden="true" tabindex="-1"></a>      cluster_markers <span class="ot">&lt;-</span></span>
<span id="cb158-126"><a href="#cb158-126" aria-hidden="true" tabindex="-1"></a>        <span class="fu">map</span>(subsets,</span>
<span id="cb158-127"><a href="#cb158-127" aria-hidden="true" tabindex="-1"></a>            <span class="sc">~</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb158-128"><a href="#cb158-128" aria-hidden="true" tabindex="-1"></a>              .x,</span>
<span id="cb158-129"><a href="#cb158-129" aria-hidden="true" tabindex="-1"></a>              <span class="at">only.pos =</span> <span class="cn">TRUE</span>,</span>
<span id="cb158-130"><a href="#cb158-130" aria-hidden="true" tabindex="-1"></a>              <span class="at">min.pct =</span> <span class="fl">0.2</span>,</span>
<span id="cb158-131"><a href="#cb158-131" aria-hidden="true" tabindex="-1"></a>              <span class="at">logfc.threshold =</span> <span class="fl">0.5</span></span>
<span id="cb158-132"><a href="#cb158-132" aria-hidden="true" tabindex="-1"></a>            ))</span>
<span id="cb158-133"><a href="#cb158-133" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-134"><a href="#cb158-134" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Save subsets and cluster markers to a tibble</span></span>
<span id="cb158-135"><a href="#cb158-135" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span></span>
<span id="cb158-136"><a href="#cb158-136" aria-hidden="true" tabindex="-1"></a>        <span class="fu">tibble</span>(</span>
<span id="cb158-137"><a href="#cb158-137" aria-hidden="true" tabindex="-1"></a>          <span class="st">"celltype"</span> <span class="ot">=</span> celltypes,</span>
<span id="cb158-138"><a href="#cb158-138" aria-hidden="true" tabindex="-1"></a>          <span class="st">"seurat_obj"</span> <span class="ot">=</span> subsets,</span>
<span id="cb158-139"><a href="#cb158-139" aria-hidden="true" tabindex="-1"></a>          <span class="st">"cluster_markers"</span> <span class="ot">=</span> cluster_markers</span>
<span id="cb158-140"><a href="#cb158-140" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb158-141"><a href="#cb158-141" aria-hidden="true" tabindex="-1"></a>      <span class="co"># save object</span></span>
<span id="cb158-142"><a href="#cb158-142" aria-hidden="true" tabindex="-1"></a>      readr<span class="sc">::</span><span class="fu">write_rds</span>(subsets, result_file)</span>
<span id="cb158-143"><a href="#cb158-143" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-144"><a href="#cb158-144" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb158-145"><a href="#cb158-145" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(result_file)</span>
<span id="cb158-146"><a href="#cb158-146" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb158-147"><a href="#cb158-147" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-148"><a href="#cb158-148" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(subsets)</span>
<span id="cb158-149"><a href="#cb158-149" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-150"><a href="#cb158-150" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb158-151"><a href="#cb158-151" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-152"><a href="#cb158-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-155"><a href="#cb158-155" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-156"><a href="#cb158-156" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb158-157"><a href="#cb158-157" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span id="cb158-158"><a href="#cb158-158" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span></span>
<span id="cb158-159"><a href="#cb158-159" aria-hidden="true" tabindex="-1"></a>        <span class="fu">map</span>(celltypes, <span class="sc">~</span> <span class="fu">subset</span>(sce, <span class="at">idents =</span> .x)) <span class="sc">%&gt;%</span></span>
<span id="cb158-160"><a href="#cb158-160" aria-hidden="true" tabindex="-1"></a>        purrr<span class="sc">::</span><span class="fu">set_names</span>(celltypes)</span>
<span id="cb158-161"><a href="#cb158-161" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-162"><a href="#cb158-162" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Step 2: Cluster to the subsets</span></span>
<span id="cb158-163"><a href="#cb158-163" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">FindNeighbors</span>(., <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>dims))</span>
<span id="cb158-164"><a href="#cb158-164" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span></span>
<span id="cb158-165"><a href="#cb158-165" aria-hidden="true" tabindex="-1"></a>        <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">FindClusters</span>(., <span class="at">resolution =</span> resolution))</span>
<span id="cb158-166"><a href="#cb158-166" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-167"><a href="#cb158-167" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Step 4: Run UMAP</span></span>
<span id="cb158-168"><a href="#cb158-168" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">RunUMAP</span>(., <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>))</span>
<span id="cb158-169"><a href="#cb158-169" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-170"><a href="#cb158-170" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Reduce size but keep reductions and graphs</span></span>
<span id="cb158-171"><a href="#cb158-171" aria-hidden="true" tabindex="-1"></a>      reductions <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">names</span>(.x<span class="sc">@</span>reductions))</span>
<span id="cb158-172"><a href="#cb158-172" aria-hidden="true" tabindex="-1"></a>      graphs <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets, <span class="sc">~</span> <span class="fu">names</span>(.x<span class="sc">@</span>graphs))</span>
<span id="cb158-173"><a href="#cb158-173" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-174"><a href="#cb158-174" aria-hidden="true" tabindex="-1"></a>      args <span class="ot">&lt;-</span></span>
<span id="cb158-175"><a href="#cb158-175" aria-hidden="true" tabindex="-1"></a>        <span class="fu">list</span>(<span class="st">"object"</span> <span class="ot">=</span> subsets,</span>
<span id="cb158-176"><a href="#cb158-176" aria-hidden="true" tabindex="-1"></a>             <span class="st">"dimreducs"</span> <span class="ot">=</span> reductions,</span>
<span id="cb158-177"><a href="#cb158-177" aria-hidden="true" tabindex="-1"></a>             <span class="st">"graphs"</span> <span class="ot">=</span> graphs)</span>
<span id="cb158-178"><a href="#cb158-178" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">pmap</span>(args, DietSeurat)</span>
<span id="cb158-179"><a href="#cb158-179" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-180"><a href="#cb158-180" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Find markers</span></span>
<span id="cb158-181"><a href="#cb158-181" aria-hidden="true" tabindex="-1"></a>      cluster_markers <span class="ot">&lt;-</span></span>
<span id="cb158-182"><a href="#cb158-182" aria-hidden="true" tabindex="-1"></a>        <span class="fu">map</span>(subsets,</span>
<span id="cb158-183"><a href="#cb158-183" aria-hidden="true" tabindex="-1"></a>            <span class="sc">~</span> <span class="fu">FindAllMarkers</span>(</span>
<span id="cb158-184"><a href="#cb158-184" aria-hidden="true" tabindex="-1"></a>              .x,</span>
<span id="cb158-185"><a href="#cb158-185" aria-hidden="true" tabindex="-1"></a>              <span class="at">only.pos =</span> <span class="cn">TRUE</span>,</span>
<span id="cb158-186"><a href="#cb158-186" aria-hidden="true" tabindex="-1"></a>              <span class="at">min.pct =</span> <span class="fl">0.2</span>,</span>
<span id="cb158-187"><a href="#cb158-187" aria-hidden="true" tabindex="-1"></a>              <span class="at">logfc.threshold =</span> <span class="fl">0.5</span></span>
<span id="cb158-188"><a href="#cb158-188" aria-hidden="true" tabindex="-1"></a>            ))</span>
<span id="cb158-189"><a href="#cb158-189" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-190"><a href="#cb158-190" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Save subsets and cluster markers to a tibble</span></span>
<span id="cb158-191"><a href="#cb158-191" aria-hidden="true" tabindex="-1"></a>      subsets <span class="ot">&lt;-</span></span>
<span id="cb158-192"><a href="#cb158-192" aria-hidden="true" tabindex="-1"></a>        <span class="fu">tibble</span>(</span>
<span id="cb158-193"><a href="#cb158-193" aria-hidden="true" tabindex="-1"></a>          <span class="st">"celltype"</span> <span class="ot">=</span> celltypes,</span>
<span id="cb158-194"><a href="#cb158-194" aria-hidden="true" tabindex="-1"></a>          <span class="st">"seurat_obj"</span> <span class="ot">=</span> subsets,</span>
<span id="cb158-195"><a href="#cb158-195" aria-hidden="true" tabindex="-1"></a>          <span class="st">"cluster_markers"</span> <span class="ot">=</span> cluster_markers</span>
<span id="cb158-196"><a href="#cb158-196" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb158-197"><a href="#cb158-197" aria-hidden="true" tabindex="-1"></a>      <span class="co"># save object</span></span>
<span id="cb158-198"><a href="#cb158-198" aria-hidden="true" tabindex="-1"></a>      readr<span class="sc">::</span><span class="fu">write_rds</span>(subsets, result_file)</span>
<span id="cb158-199"><a href="#cb158-199" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-200"><a href="#cb158-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-203"><a href="#cb158-203" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-204"><a href="#cb158-204" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Subset the Seurat object for each cell type</span></span>
<span id="cb158-205"><a href="#cb158-205" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(sce<span class="sc">$</span>highlevel_celltypes)</span>
<span id="cb158-206"><a href="#cb158-206" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(sce) <span class="ot">&lt;-</span> sce<span class="sc">$</span>highlevel_celltypes</span>
<span id="cb158-207"><a href="#cb158-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-208"><a href="#cb158-208" aria-hidden="true" tabindex="-1"></a><span class="co"># There is an error if a metadata column has the same name as a gene</span></span>
<span id="cb158-209"><a href="#cb158-209" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>apoe_status <span class="ot">&lt;-</span> sce<span class="sc">$</span>APOE</span>
<span id="cb158-210"><a href="#cb158-210" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>APOE <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb158-211"><a href="#cb158-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-212"><a href="#cb158-212" aria-hidden="true" tabindex="-1"></a>subsets <span class="ot">&lt;-</span></span>
<span id="cb158-213"><a href="#cb158-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subcluster</span>(</span>
<span id="cb158-214"><a href="#cb158-214" aria-hidden="true" tabindex="-1"></a>    sce,</span>
<span id="cb158-215"><a href="#cb158-215" aria-hidden="true" tabindex="-1"></a>    celltypes,</span>
<span id="cb158-216"><a href="#cb158-216" aria-hidden="true" tabindex="-1"></a>    here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb158-217"><a href="#cb158-217" aria-hidden="true" tabindex="-1"></a>      <span class="st">"03_data/990_processed_data/001_snrnaseq/10_cell_subtypes/subclusters_highlevel-celltypes.rds"</span></span>
<span id="cb158-218"><a href="#cb158-218" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb158-219"><a href="#cb158-219" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb158-220"><a href="#cb158-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-221"><a href="#cb158-221" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up environment</span></span>
<span id="cb158-222"><a href="#cb158-222" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(sce, celltypes)</span>
<span id="cb158-223"><a href="#cb158-223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-224"><a href="#cb158-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-225"><a href="#cb158-225" aria-hidden="true" tabindex="-1"></a><span class="fu">## Check UMAP clusters and top marker genes</span></span>
<span id="cb158-226"><a href="#cb158-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-229"><a href="#cb158-229" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-230"><a href="#cb158-230" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 3: Visualize the UMAP for each cell subtype</span></span>
<span id="cb158-231"><a href="#cb158-231" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span></span>
<span id="cb158-232"><a href="#cb158-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb158-233"><a href="#cb158-233" aria-hidden="true" tabindex="-1"></a>    subsets<span class="sc">$</span>seurat_obj,</span>
<span id="cb158-234"><a href="#cb158-234" aria-hidden="true" tabindex="-1"></a>    subsets<span class="sc">$</span>celltype,</span>
<span id="cb158-235"><a href="#cb158-235" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">DimPlot</span>(</span>
<span id="cb158-236"><a href="#cb158-236" aria-hidden="true" tabindex="-1"></a>      .,</span>
<span id="cb158-237"><a href="#cb158-237" aria-hidden="true" tabindex="-1"></a>      <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb158-238"><a href="#cb158-238" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb158-239"><a href="#cb158-239" aria-hidden="true" tabindex="-1"></a>      <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-240"><a href="#cb158-240" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb158-241"><a href="#cb158-241" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggtitle</span>(.y)</span>
<span id="cb158-242"><a href="#cb158-242" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb158-243"><a href="#cb158-243" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-244"><a href="#cb158-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-247"><a href="#cb158-247" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-248"><a href="#cb158-248" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 20</span></span>
<span id="cb158-249"><a href="#cb158-249" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 20</span></span>
<span id="cb158-250"><a href="#cb158-250" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how many genes aren't significant and which have blank gene names</span></span>
<span id="cb158-251"><a href="#cb158-251" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(subsets<span class="sc">$</span>cluster_markers, <span class="sc">~</span> <span class="fu">sum</span>(<span class="fu">which</span>(.x<span class="sc">$</span>p_val_adj <span class="sc">&gt;</span> <span class="fl">0.05</span>)))</span>
<span id="cb158-252"><a href="#cb158-252" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(subsets<span class="sc">$</span>cluster_markers, <span class="sc">~</span> <span class="fu">sum</span>(<span class="fu">which</span>(.x<span class="sc">$</span>gene <span class="sc">==</span> <span class="st">""</span>)))</span>
<span id="cb158-253"><a href="#cb158-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-254"><a href="#cb158-254" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 10 genes per cluster</span></span>
<span id="cb158-255"><a href="#cb158-255" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">map</span>(subsets<span class="sc">$</span>cluster_markers, <span class="sc">~</span> .x <span class="sc">%&gt;%</span></span>
<span id="cb158-256"><a href="#cb158-256" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter to sig genes</span></span>
<span id="cb158-257"><a href="#cb158-257" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p_val_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb158-258"><a href="#cb158-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> </span>
<span id="cb158-259"><a href="#cb158-259" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get top genes by avg log2FC</span></span>
<span id="cb158-260"><a href="#cb158-260" aria-hidden="true" tabindex="-1"></a>  <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="at">wt =</span> avg_log2FC))</span>
<span id="cb158-261"><a href="#cb158-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-262"><a href="#cb158-262" aria-hidden="true" tabindex="-1"></a>to_plot <span class="ot">&lt;-</span> <span class="fu">unique</span>(top10<span class="sc">$</span>Endothelial<span class="sc">$</span>gene)</span>
<span id="cb158-263"><a href="#cb158-263" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">DotPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Endothelial, <span class="at">features =</span> to_plot) <span class="sc">+</span> </span>
<span id="cb158-264"><a href="#cb158-264" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb158-265"><a href="#cb158-265" aria-hidden="true" tabindex="-1"></a>plot</span>
<span id="cb158-266"><a href="#cb158-266" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-267"><a href="#cb158-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-270"><a href="#cb158-270" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-271"><a href="#cb158-271" aria-hidden="true" tabindex="-1"></a><span class="co"># nest top 10 markers by cluster</span></span>
<span id="cb158-272"><a href="#cb158-272" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">map</span>(top10, nest)</span>
<span id="cb158-273"><a href="#cb158-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-274"><a href="#cb158-274" aria-hidden="true" tabindex="-1"></a><span class="co"># function to rename apoe column</span></span>
<span id="cb158-275"><a href="#cb158-275" aria-hidden="true" tabindex="-1"></a><span class="co"># modify_sce &lt;- function(sce) {</span></span>
<span id="cb158-276"><a href="#cb158-276" aria-hidden="true" tabindex="-1"></a><span class="co">#   sce$apoe_status &lt;- sce$APOE</span></span>
<span id="cb158-277"><a href="#cb158-277" aria-hidden="true" tabindex="-1"></a><span class="co">#   sce$APOE &lt;- NULL</span></span>
<span id="cb158-278"><a href="#cb158-278" aria-hidden="true" tabindex="-1"></a><span class="co">#   return(sce)</span></span>
<span id="cb158-279"><a href="#cb158-279" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb158-280"><a href="#cb158-280" aria-hidden="true" tabindex="-1"></a><span class="co"># to get points to appear with raster = T, increase point size</span></span>
<span id="cb158-281"><a href="#cb158-281" aria-hidden="true" tabindex="-1"></a><span class="co">#FeaturePlot(seurat, features = markers, pt.size = 1.5)</span></span>
<span id="cb158-282"><a href="#cb158-282" aria-hidden="true" tabindex="-1"></a><span class="co"># This takes a while to run, so save the output and only rerun if needed</span></span>
<span id="cb158-283"><a href="#cb158-283" aria-hidden="true" tabindex="-1"></a>marker_plot_results <span class="ot">&lt;-</span></span>
<span id="cb158-284"><a href="#cb158-284" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb158-285"><a href="#cb158-285" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/10_cell_subtypes/subclusters_highlevel-cluster-markers.rds"</span></span>
<span id="cb158-286"><a href="#cb158-286" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb158-287"><a href="#cb158-287" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(marker_plot_results)) {</span>
<span id="cb158-288"><a href="#cb158-288" aria-hidden="true" tabindex="-1"></a>  <span class="co"># subsets$seurat_obj &lt;- map(subsets$seurat_obj, modify_sce)</span></span>
<span id="cb158-289"><a href="#cb158-289" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-290"><a href="#cb158-290" aria-hidden="true" tabindex="-1"></a>  sce_markers_plot <span class="ot">&lt;-</span></span>
<span id="cb158-291"><a href="#cb158-291" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map2</span>(top10, subsets<span class="sc">$</span>seurat_obj, <span class="cf">function</span>(top, seurat) {</span>
<span id="cb158-292"><a href="#cb158-292" aria-hidden="true" tabindex="-1"></a>      <span class="fu">map2</span>(top<span class="sc">$</span>data, top<span class="sc">$</span>cluster, <span class="cf">function</span>(data, cluster) {</span>
<span id="cb158-293"><a href="#cb158-293" aria-hidden="true" tabindex="-1"></a>        <span class="fu">FeaturePlot</span>(seurat,</span>
<span id="cb158-294"><a href="#cb158-294" aria-hidden="true" tabindex="-1"></a>                    <span class="at">features =</span> <span class="fu">as.character</span>(data<span class="sc">$</span>gene),</span>
<span id="cb158-295"><a href="#cb158-295" aria-hidden="true" tabindex="-1"></a>                    <span class="at">raster =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb158-296"><a href="#cb158-296" aria-hidden="true" tabindex="-1"></a>          patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">"Seurat cluster: "</span>, cluster))</span>
<span id="cb158-297"><a href="#cb158-297" aria-hidden="true" tabindex="-1"></a>      }) <span class="sc">%&gt;%</span> <span class="fu">set_names</span>(<span class="fu">paste0</span>(<span class="st">"cluster_"</span>, top<span class="sc">$</span>cluster))</span>
<span id="cb158-298"><a href="#cb158-298" aria-hidden="true" tabindex="-1"></a>    })</span>
<span id="cb158-299"><a href="#cb158-299" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(sce_markers_plot, marker_plot_results)</span>
<span id="cb158-300"><a href="#cb158-300" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb158-301"><a href="#cb158-301" aria-hidden="true" tabindex="-1"></a>  sce_markers_plot <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(marker_plot_results)</span>
<span id="cb158-302"><a href="#cb158-302" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb158-303"><a href="#cb158-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-304"><a href="#cb158-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-305"><a href="#cb158-305" aria-hidden="true" tabindex="-1"></a><span class="fu">### Endothelial cells</span></span>
<span id="cb158-306"><a href="#cb158-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-309"><a href="#cb158-309" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-310"><a href="#cb158-310" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-311"><a href="#cb158-311" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-312"><a href="#cb158-312" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>Endothelial</span>
<span id="cb158-313"><a href="#cb158-313" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>Endothelial</span>
<span id="cb158-314"><a href="#cb158-314" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-315"><a href="#cb158-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-316"><a href="#cb158-316" aria-hidden="true" tabindex="-1"></a>Cluster 1 is somewhat between Arterial and an astrocyte subtype</span>
<span id="cb158-317"><a href="#cb158-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-320"><a href="#cb158-320" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-321"><a href="#cb158-321" aria-hidden="true" tabindex="-1"></a><span class="co"># Endothelial subtypes</span></span>
<span id="cb158-322"><a href="#cb158-322" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Arterial"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Arterial"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>, </span>
<span id="cb158-323"><a href="#cb158-323" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Venous"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Capillary/Venous"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Arterial"</span>, </span>
<span id="cb158-324"><a href="#cb158-324" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"T-Pericyte"</span>, </span>
<span id="cb158-325"><a href="#cb158-325" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"Oligodendrocyte/OPC"</span>, <span class="st">"9"</span> <span class="ot">=</span> <span class="st">"T-Cell"</span>, <span class="st">"10"</span> <span class="ot">=</span> <span class="st">"Astrocyte"</span>)</span>
<span id="cb158-326"><a href="#cb158-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-327"><a href="#cb158-327" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-328"><a href="#cb158-328" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Endothelial<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-329"><a href="#cb158-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Endothelial)</span>
<span id="cb158-330"><a href="#cb158-330" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Endothelial <span class="ot">&lt;-</span></span>
<span id="cb158-331"><a href="#cb158-331" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Endothelial, sub_celltypes)</span>
<span id="cb158-332"><a href="#cb158-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-333"><a href="#cb158-333" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Endothelial, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-334"><a href="#cb158-334" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-335"><a href="#cb158-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-336"><a href="#cb158-336" aria-hidden="true" tabindex="-1"></a><span class="fu">### Pericytes</span></span>
<span id="cb158-337"><a href="#cb158-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-340"><a href="#cb158-340" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-341"><a href="#cb158-341" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-342"><a href="#cb158-342" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-343"><a href="#cb158-343" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>Pericytes</span>
<span id="cb158-344"><a href="#cb158-344" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>Pericytes</span>
<span id="cb158-345"><a href="#cb158-345" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-346"><a href="#cb158-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-347"><a href="#cb158-347" aria-hidden="true" tabindex="-1"></a>Note that M-Pericyte is ECM-regulating Pericyte</span>
<span id="cb158-348"><a href="#cb158-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-351"><a href="#cb158-351" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-352"><a href="#cb158-352" aria-hidden="true" tabindex="-1"></a><span class="co"># Pericyte subtypes</span></span>
<span id="cb158-353"><a href="#cb158-353" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Vascular-SMC"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"T-Pericyte"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"M-Pericyte"</span>, </span>
<span id="cb158-354"><a href="#cb158-354" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"T-Pericyte"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Vascular-SMC/OPC"</span>, </span>
<span id="cb158-355"><a href="#cb158-355" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Vascular-SMC"</span>, <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"OPC"</span>, </span>
<span id="cb158-356"><a href="#cb158-356" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"Perivascular Macrophage"</span>)</span>
<span id="cb158-357"><a href="#cb158-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-358"><a href="#cb158-358" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-359"><a href="#cb158-359" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Pericytes<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-360"><a href="#cb158-360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Pericytes)</span>
<span id="cb158-361"><a href="#cb158-361" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Pericytes <span class="ot">&lt;-</span></span>
<span id="cb158-362"><a href="#cb158-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Pericytes, sub_celltypes)</span>
<span id="cb158-363"><a href="#cb158-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-364"><a href="#cb158-364" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Pericytes, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-365"><a href="#cb158-365" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-366"><a href="#cb158-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-367"><a href="#cb158-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-368"><a href="#cb158-368" aria-hidden="true" tabindex="-1"></a><span class="fu">### Oligodendrocytes</span></span>
<span id="cb158-369"><a href="#cb158-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-370"><a href="#cb158-370" aria-hidden="true" tabindex="-1"></a>Anna Williams seems to know a lot about oligodendrocytes, so I might look at their stuff for these annotations.</span>
<span id="cb158-371"><a href="#cb158-371" aria-hidden="true" tabindex="-1"></a>Viloa used her markers and found they worked well apparently.</span>
<span id="cb158-372"><a href="#cb158-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-373"><a href="#cb158-373" aria-hidden="true" tabindex="-1"></a>The cluster analysis revealed six oligodendrocyte states (positive for myelin genes such as PLP1, MBP, MAG), two OPC states (positive for PDGFRA, CSPG4 and BCAN) and three committed oligodendrocyte precursor (COP) cell states (positive for GPR17, GPC5 or GAP43), confirming that oligodendroglia are heterogeneous in the healthy adult human CNS (Figure 2a-c).</span>
<span id="cb158-374"><a href="#cb158-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-375"><a href="#cb158-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-378"><a href="#cb158-378" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-379"><a href="#cb158-379" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-380"><a href="#cb158-380" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-381"><a href="#cb158-381" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>Oligodendrocytes</span>
<span id="cb158-382"><a href="#cb158-382" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>Oligodendrocytes</span>
<span id="cb158-383"><a href="#cb158-383" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-384"><a href="#cb158-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-385"><a href="#cb158-385" aria-hidden="true" tabindex="-1"></a>She reports RBFOX1 as a marker for all but two of her oligodendrocyte clusters (A and F), so lets check that marker</span>
<span id="cb158-386"><a href="#cb158-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-389"><a href="#cb158-389" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-390"><a href="#cb158-390" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-391"><a href="#cb158-391" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-392"><a href="#cb158-392" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"RBFOX1"</span>),</span>
<span id="cb158-393"><a href="#cb158-393" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-394"><a href="#cb158-394" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-395"><a href="#cb158-395" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-396"><a href="#cb158-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-397"><a href="#cb158-397" aria-hidden="true" tabindex="-1"></a>Seems like it's only really heavily expressed in clusters 3, 4, 7 and 9 curiously...</span>
<span id="cb158-398"><a href="#cb158-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-399"><a href="#cb158-399" aria-hidden="true" tabindex="-1"></a>Anna Williams paper annotates "Oligo_B" as AFF3+ LGALS1- FMN1-</span>
<span id="cb158-400"><a href="#cb158-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-403"><a href="#cb158-403" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-404"><a href="#cb158-404" aria-hidden="true" tabindex="-1"></a><span class="co"># anna williams oligo B markers</span></span>
<span id="cb158-405"><a href="#cb158-405" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-406"><a href="#cb158-406" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-407"><a href="#cb158-407" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"AFF3"</span>, <span class="st">"LGALS1"</span>, <span class="st">"FMN1"</span>),</span>
<span id="cb158-408"><a href="#cb158-408" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-409"><a href="#cb158-409" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-410"><a href="#cb158-410" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-411"><a href="#cb158-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-412"><a href="#cb158-412" aria-hidden="true" tabindex="-1"></a>Seems to reflect cluster 4, and maybe 5</span>
<span id="cb158-413"><a href="#cb158-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-414"><a href="#cb158-414" aria-hidden="true" tabindex="-1"></a>Oligo_C: FMN1</span>
<span id="cb158-415"><a href="#cb158-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-418"><a href="#cb158-418" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-419"><a href="#cb158-419" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-420"><a href="#cb158-420" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-421"><a href="#cb158-421" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"FMN1"</span>),</span>
<span id="cb158-422"><a href="#cb158-422" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-423"><a href="#cb158-423" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-424"><a href="#cb158-424" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-425"><a href="#cb158-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-426"><a href="#cb158-426" aria-hidden="true" tabindex="-1"></a>Not super strong expression accross a single cluster, but seems to be in 12, 9 and maybe 9?</span>
<span id="cb158-427"><a href="#cb158-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-428"><a href="#cb158-428" aria-hidden="true" tabindex="-1"></a>Oligo_D: LGALS1+ FMN1- SPARC-</span>
<span id="cb158-429"><a href="#cb158-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-432"><a href="#cb158-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-433"><a href="#cb158-433" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-434"><a href="#cb158-434" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-435"><a href="#cb158-435" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"LGALS1"</span>, <span class="st">"FMN1"</span>, <span class="st">"SPARC"</span>),</span>
<span id="cb158-436"><a href="#cb158-436" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-437"><a href="#cb158-437" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-438"><a href="#cb158-438" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-439"><a href="#cb158-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-440"><a href="#cb158-440" aria-hidden="true" tabindex="-1"></a>This doesn't really fit at all...</span>
<span id="cb158-441"><a href="#cb158-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-442"><a href="#cb158-442" aria-hidden="true" tabindex="-1"></a>Oligo_E: HHIP+ OPALIN- AFF3-</span>
<span id="cb158-443"><a href="#cb158-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-446"><a href="#cb158-446" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-447"><a href="#cb158-447" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-448"><a href="#cb158-448" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-449"><a href="#cb158-449" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"HHIP"</span>, <span class="st">"OPALIN"</span>, <span class="st">"AFF3"</span>),</span>
<span id="cb158-450"><a href="#cb158-450" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-451"><a href="#cb158-451" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-452"><a href="#cb158-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-453"><a href="#cb158-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-454"><a href="#cb158-454" aria-hidden="true" tabindex="-1"></a>Well the left clusters express HHIP, but the same clusters express OPALIN, so this doesn't fit either...</span>
<span id="cb158-455"><a href="#cb158-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-456"><a href="#cb158-456" aria-hidden="true" tabindex="-1"></a>Oligo_A expresses OPALIN, PLXDC2, LAMA2 and PALM2, and Oligo_F expresses SPARC, DHCR24, TUBA1A, TUBB2B, PMP2</span>
<span id="cb158-457"><a href="#cb158-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-460"><a href="#cb158-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-461"><a href="#cb158-461" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-462"><a href="#cb158-462" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-463"><a href="#cb158-463" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"OPALIN"</span>, <span class="st">"PLXDC2"</span>, <span class="st">"LAMA2"</span>, <span class="st">"PALM2"</span>),</span>
<span id="cb158-464"><a href="#cb158-464" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-465"><a href="#cb158-465" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-466"><a href="#cb158-466" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(</span>
<span id="cb158-467"><a href="#cb158-467" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-468"><a href="#cb158-468" aria-hidden="true" tabindex="-1"></a>  <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"SPARC"</span>, <span class="st">"DHCR24"</span>, <span class="st">"TUBA1A"</span>, <span class="st">"TUBB2B"</span>, <span class="st">"PMP2"</span>),</span>
<span id="cb158-469"><a href="#cb158-469" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-470"><a href="#cb158-470" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-471"><a href="#cb158-471" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-472"><a href="#cb158-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-473"><a href="#cb158-473" aria-hidden="true" tabindex="-1"></a>I guess for now I'll just label them as RBFOX1 pos/neg oligodendrocytes since her group reports this as a key distinguishing feature.</span>
<span id="cb158-474"><a href="#cb158-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-477"><a href="#cb158-477" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-478"><a href="#cb158-478" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-479"><a href="#cb158-479" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-480"><a href="#cb158-480" aria-hidden="true" tabindex="-1"></a><span class="co"># Astrocyte subtypes</span></span>
<span id="cb158-481"><a href="#cb158-481" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, </span>
<span id="cb158-482"><a href="#cb158-482" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Oligo"</span>,</span>
<span id="cb158-483"><a href="#cb158-483" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Oligo"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>,</span>
<span id="cb158-484"><a href="#cb158-484" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Oligo"</span>,</span>
<span id="cb158-485"><a href="#cb158-485" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"9"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Oligo"</span>, </span>
<span id="cb158-486"><a href="#cb158-486" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"10"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"11"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, </span>
<span id="cb158-487"><a href="#cb158-487" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"12"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"13"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>,</span>
<span id="cb158-488"><a href="#cb158-488" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"14"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>, <span class="st">"15"</span> <span class="ot">=</span> <span class="st">"RBFOX1-Neg-Oligo"</span>)</span>
<span id="cb158-489"><a href="#cb158-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-490"><a href="#cb158-490" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-491"><a href="#cb158-491" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-492"><a href="#cb158-492" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes)</span>
<span id="cb158-493"><a href="#cb158-493" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes <span class="ot">&lt;-</span></span>
<span id="cb158-494"><a href="#cb158-494" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes, sub_celltypes)</span>
<span id="cb158-495"><a href="#cb158-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-496"><a href="#cb158-496" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb158-497"><a href="#cb158-497" aria-hidden="true" tabindex="-1"></a>  subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Oligodendrocytes,</span>
<span id="cb158-498"><a href="#cb158-498" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">"umap"</span>,</span>
<span id="cb158-499"><a href="#cb158-499" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="cn">TRUE</span>,</span>
<span id="cb158-500"><a href="#cb158-500" aria-hidden="true" tabindex="-1"></a>  <span class="at">raster =</span> <span class="cn">FALSE</span></span>
<span id="cb158-501"><a href="#cb158-501" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-502"><a href="#cb158-502" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-503"><a href="#cb158-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-504"><a href="#cb158-504" aria-hidden="true" tabindex="-1"></a><span class="fu">### Astrocytes</span></span>
<span id="cb158-505"><a href="#cb158-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-508"><a href="#cb158-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-509"><a href="#cb158-509" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-510"><a href="#cb158-510" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-511"><a href="#cb158-511" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>Astrocytes</span>
<span id="cb158-512"><a href="#cb158-512" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>Astrocytes</span>
<span id="cb158-513"><a href="#cb158-513" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-514"><a href="#cb158-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-515"><a href="#cb158-515" aria-hidden="true" tabindex="-1"></a>Cluster 8 seems to be a bit mixed between the pericyte and capillary types in the Yang data.</span>
<span id="cb158-516"><a href="#cb158-516" aria-hidden="true" tabindex="-1"></a>Cluster 13 doesn't map super well to the Yang data</span>
<span id="cb158-517"><a href="#cb158-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-520"><a href="#cb158-520" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-521"><a href="#cb158-521" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-522"><a href="#cb158-522" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-523"><a href="#cb158-523" aria-hidden="true" tabindex="-1"></a><span class="co"># Astrocyte subtypes</span></span>
<span id="cb158-524"><a href="#cb158-524" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Cortex"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>, </span>
<span id="cb158-525"><a href="#cb158-525" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Cortex"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Hpc"</span>,</span>
<span id="cb158-526"><a href="#cb158-526" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"T-Pericyte"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Oligodendrocyte"</span>,</span>
<span id="cb158-527"><a href="#cb158-527" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Hpc"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Hpc"</span>,</span>
<span id="cb158-528"><a href="#cb158-528" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"M-Pericyte"</span>, <span class="st">"9"</span> <span class="ot">=</span> <span class="st">"Neuron"</span>, <span class="st">"10"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>, </span>
<span id="cb158-529"><a href="#cb158-529" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"11"</span> <span class="ot">=</span> <span class="st">"Microglia"</span>, <span class="st">"12"</span> <span class="ot">=</span> <span class="st">"T-Cell"</span>, <span class="st">"13"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>)</span>
<span id="cb158-530"><a href="#cb158-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-531"><a href="#cb158-531" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-532"><a href="#cb158-532" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Astrocytes<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-533"><a href="#cb158-533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Astrocytes)</span>
<span id="cb158-534"><a href="#cb158-534" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Astrocytes <span class="ot">&lt;-</span></span>
<span id="cb158-535"><a href="#cb158-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Astrocytes, sub_celltypes)</span>
<span id="cb158-536"><a href="#cb158-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-537"><a href="#cb158-537" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Astrocytes, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-538"><a href="#cb158-538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-539"><a href="#cb158-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-540"><a href="#cb158-540" aria-hidden="true" tabindex="-1"></a><span class="fu">### Vascular-and-leptomeningeal</span></span>
<span id="cb158-541"><a href="#cb158-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-544"><a href="#cb158-544" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-545"><a href="#cb158-545" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-546"><a href="#cb158-546" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-547"><a href="#cb158-547" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span></span>
<span id="cb158-548"><a href="#cb158-548" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span></span>
<span id="cb158-549"><a href="#cb158-549" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-550"><a href="#cb158-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-551"><a href="#cb158-551" aria-hidden="true" tabindex="-1"></a>FB = fibroblast.</span>
<span id="cb158-552"><a href="#cb158-552" aria-hidden="true" tabindex="-1"></a>Cluster 2 doesn't map very well to Yang.</span>
<span id="cb158-553"><a href="#cb158-553" aria-hidden="true" tabindex="-1"></a>Not sure if cluster 5 maps better to one of the SMC subtypes or pericyte subtypes in the Yang data</span>
<span id="cb158-554"><a href="#cb158-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-557"><a href="#cb158-557" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-558"><a href="#cb158-558" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-559"><a href="#cb158-559" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-560"><a href="#cb158-560" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Perivascular-FB"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Perivascular-FB"</span>, </span>
<span id="cb158-561"><a href="#cb158-561" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Meningeal-FB"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"T-Pericyte"</span>, </span>
<span id="cb158-562"><a href="#cb158-562" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Meningeal-FB"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Vascular-SMC/M-Pericyte"</span>,</span>
<span id="cb158-563"><a href="#cb158-563" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Capillary/Venous"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"Oligodendrocyte"</span>)</span>
<span id="cb158-564"><a href="#cb158-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-565"><a href="#cb158-565" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-566"><a href="#cb158-566" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span><span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-567"><a href="#cb158-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span>)</span>
<span id="cb158-568"><a href="#cb158-568" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span> <span class="ot">&lt;-</span></span>
<span id="cb158-569"><a href="#cb158-569" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span>, sub_celltypes)</span>
<span id="cb158-570"><a href="#cb158-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-571"><a href="#cb158-571" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span><span class="st">`</span><span class="at">Vascular-and-leptomeningeal</span><span class="st">`</span>, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-572"><a href="#cb158-572" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-573"><a href="#cb158-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-574"><a href="#cb158-574" aria-hidden="true" tabindex="-1"></a><span class="fu">### Neuron</span></span>
<span id="cb158-575"><a href="#cb158-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-576"><a href="#cb158-576" aria-hidden="true" tabindex="-1"></a>The Yang dataset doesn't have neuronal subtypes, so I'll skip this for now.</span>
<span id="cb158-577"><a href="#cb158-577" aria-hidden="true" tabindex="-1"></a>Anna Williams paper and maybe the Paul Matthews paper have some I think.</span>
<span id="cb158-578"><a href="#cb158-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-581"><a href="#cb158-581" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-582"><a href="#cb158-582" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-583"><a href="#cb158-583" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-584"><a href="#cb158-584" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>Neuron</span>
<span id="cb158-585"><a href="#cb158-585" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>Neuron</span>
<span id="cb158-586"><a href="#cb158-586" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-587"><a href="#cb158-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-590"><a href="#cb158-590" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-591"><a href="#cb158-591" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-592"><a href="#cb158-592" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-593"><a href="#cb158-593" aria-hidden="true" tabindex="-1"></a>frank_annotation <span class="ot">&lt;-</span></span>
<span id="cb158-594"><a href="#cb158-594" aria-hidden="true" tabindex="-1"></a>  readxl<span class="sc">::</span><span class="fu">read_excel</span>(</span>
<span id="cb158-595"><a href="#cb158-595" aria-hidden="true" tabindex="-1"></a>    here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb158-596"><a href="#cb158-596" aria-hidden="true" tabindex="-1"></a>      <span class="st">"03_data/990_processed_data/001_snrnaseq/91_data_from_frank/Cell Markers - Frank - First 20.xlsx"</span></span>
<span id="cb158-597"><a href="#cb158-597" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb158-598"><a href="#cb158-598" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb158-599"><a href="#cb158-599" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(frank_annotation)</span>
<span id="cb158-600"><a href="#cb158-600" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(frank_annotation<span class="sc">$</span>cluster)</span>
<span id="cb158-601"><a href="#cb158-601" aria-hidden="true" tabindex="-1"></a>neuro_i <span class="ot">&lt;-</span> frank_annotation <span class="sc">%&gt;%</span></span>
<span id="cb158-602"><a href="#cb158-602" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter to inhibitory neuron makrers that aren't in excitatory group</span></span>
<span id="cb158-603"><a href="#cb158-603" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(cluster <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"NI"</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NE"</span>, in_cluster)) <span class="sc">%&gt;%</span></span>
<span id="cb158-604"><a href="#cb158-604" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get top 10 markers by fold change</span></span>
<span id="cb158-605"><a href="#cb158-605" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="at">wt =</span> avg_log2FC)</span>
<span id="cb158-606"><a href="#cb158-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-607"><a href="#cb158-607" aria-hidden="true" tabindex="-1"></a>neuron_i_plot <span class="ot">&lt;-</span></span>
<span id="cb158-608"><a href="#cb158-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron,</span>
<span id="cb158-609"><a href="#cb158-609" aria-hidden="true" tabindex="-1"></a>              <span class="at">features =</span> neuro_i<span class="sc">$</span>gene,</span>
<span id="cb158-610"><a href="#cb158-610" aria-hidden="true" tabindex="-1"></a>              <span class="at">raster =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb158-611"><a href="#cb158-611" aria-hidden="true" tabindex="-1"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">"Inhibitory neuronal markers"</span>)</span>
<span id="cb158-612"><a href="#cb158-612" aria-hidden="true" tabindex="-1"></a>neuron_i_plot</span>
<span id="cb158-613"><a href="#cb158-613" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-614"><a href="#cb158-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-615"><a href="#cb158-615" aria-hidden="true" tabindex="-1"></a>Clusters 2, 7, 8, 10 and 13 look look to express these inhibitory markers well.</span>
<span id="cb158-616"><a href="#cb158-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-619"><a href="#cb158-619" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-620"><a href="#cb158-620" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-621"><a href="#cb158-621" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-622"><a href="#cb158-622" aria-hidden="true" tabindex="-1"></a>neuro_e <span class="ot">&lt;-</span> frank_annotation <span class="sc">%&gt;%</span></span>
<span id="cb158-623"><a href="#cb158-623" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter to inhibitory neuron makrers that aren't in excitatory group</span></span>
<span id="cb158-624"><a href="#cb158-624" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(cluster <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"NE"</span>) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NI"</span>, in_cluster)) <span class="sc">%&gt;%</span></span>
<span id="cb158-625"><a href="#cb158-625" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get top 10 markers by fold change</span></span>
<span id="cb158-626"><a href="#cb158-626" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="at">wt =</span> avg_log2FC)</span>
<span id="cb158-627"><a href="#cb158-627" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-628"><a href="#cb158-628" aria-hidden="true" tabindex="-1"></a>neuron_e_plot <span class="ot">&lt;-</span></span>
<span id="cb158-629"><a href="#cb158-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron,</span>
<span id="cb158-630"><a href="#cb158-630" aria-hidden="true" tabindex="-1"></a>              <span class="at">features =</span> neuro_e<span class="sc">$</span>gene,</span>
<span id="cb158-631"><a href="#cb158-631" aria-hidden="true" tabindex="-1"></a>              <span class="at">raster =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb158-632"><a href="#cb158-632" aria-hidden="true" tabindex="-1"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">"Excitatory neuronal markers"</span>)</span>
<span id="cb158-633"><a href="#cb158-633" aria-hidden="true" tabindex="-1"></a>neuron_e_plot</span>
<span id="cb158-634"><a href="#cb158-634" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-635"><a href="#cb158-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-636"><a href="#cb158-636" aria-hidden="true" tabindex="-1"></a>The large group of clusters in the bottom right seem look like excitatory neurons.</span>
<span id="cb158-637"><a href="#cb158-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-638"><a href="#cb158-638" aria-hidden="true" tabindex="-1"></a>Anna Williams annotated reelin-positive neurons, so let's check reelin in these clusters</span>
<span id="cb158-639"><a href="#cb158-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-642"><a href="#cb158-642" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-643"><a href="#cb158-643" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-644"><a href="#cb158-644" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-645"><a href="#cb158-645" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"RELN"</span>)) <span class="sc">+</span></span>
<span id="cb158-646"><a href="#cb158-646" aria-hidden="true" tabindex="-1"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">"Reelin neurons"</span>)</span>
<span id="cb158-647"><a href="#cb158-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-648"><a href="#cb158-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-649"><a href="#cb158-649" aria-hidden="true" tabindex="-1"></a>Seems like a subset of cells that span clusters 2 and 7 are expressing RELN, so I suppose those could be regarded as reelin-positive.</span>
<span id="cb158-650"><a href="#cb158-650" aria-hidden="true" tabindex="-1"></a>The fact that it isn't expressed ubiquitoously throughout the clusters gives me pause though...</span>
<span id="cb158-651"><a href="#cb158-651" aria-hidden="true" tabindex="-1"></a>I'll leave them as inhibitory neurons for now.</span>
<span id="cb158-652"><a href="#cb158-652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-653"><a href="#cb158-653" aria-hidden="true" tabindex="-1"></a>Clusters 12 and 4 (and 18 to a lesser extent) don't seem to express the inhibitory or excitatory markers I'm using here.</span>
<span id="cb158-654"><a href="#cb158-654" aria-hidden="true" tabindex="-1"></a>Not sure what to label them as currently.</span>
<span id="cb158-655"><a href="#cb158-655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-656"><a href="#cb158-656" aria-hidden="true" tabindex="-1"></a>The markers the Anna Williams paper reports seem to line up too (excitatory neurons - SNAP25, SLC17A7; inhibitory neurons - SNAP25, GAD1).</span>
<span id="cb158-657"><a href="#cb158-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-660"><a href="#cb158-660" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-661"><a href="#cb158-661" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"SLC17A7"</span>))</span>
<span id="cb158-662"><a href="#cb158-662" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"GAD1"</span>))</span>
<span id="cb158-663"><a href="#cb158-663" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-664"><a href="#cb158-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-665"><a href="#cb158-665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-668"><a href="#cb158-668" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-669"><a href="#cb158-669" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-670"><a href="#cb158-670" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-671"><a href="#cb158-671" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"In-Neuron"</span>, </span>
<span id="cb158-672"><a href="#cb158-672" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"ambiguous"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>,</span>
<span id="cb158-673"><a href="#cb158-673" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"In-Neuron"</span>,  <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"In-Neuron"</span>, </span>
<span id="cb158-674"><a href="#cb158-674" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"9"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"10"</span> <span class="ot">=</span> <span class="st">"In-Neuron"</span>, <span class="st">"11"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, </span>
<span id="cb158-675"><a href="#cb158-675" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"12"</span> <span class="ot">=</span> <span class="st">"ambiguous"</span>, <span class="st">"13"</span> <span class="ot">=</span> <span class="st">"In-Neuron"</span>, <span class="st">"14"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>,</span>
<span id="cb158-676"><a href="#cb158-676" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"15"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"16"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"17"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>,</span>
<span id="cb158-677"><a href="#cb158-677" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"18"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>, <span class="st">"19"</span> <span class="ot">=</span> <span class="st">"Ex-Neuron"</span>)</span>
<span id="cb158-678"><a href="#cb158-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-679"><a href="#cb158-679" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-680"><a href="#cb158-680" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-681"><a href="#cb158-681" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron)</span>
<span id="cb158-682"><a href="#cb158-682" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron <span class="ot">&lt;-</span></span>
<span id="cb158-683"><a href="#cb158-683" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron, sub_celltypes)</span>
<span id="cb158-684"><a href="#cb158-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-685"><a href="#cb158-685" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Neuron, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-686"><a href="#cb158-686" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-687"><a href="#cb158-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-688"><a href="#cb158-688" aria-hidden="true" tabindex="-1"></a><span class="fu">### Microglia</span></span>
<span id="cb158-689"><a href="#cb158-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-692"><a href="#cb158-692" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-693"><a href="#cb158-693" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-694"><a href="#cb158-694" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-695"><a href="#cb158-695" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>Microglia</span>
<span id="cb158-696"><a href="#cb158-696" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>Microglia</span>
<span id="cb158-697"><a href="#cb158-697" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-698"><a href="#cb158-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-699"><a href="#cb158-699" aria-hidden="true" tabindex="-1"></a>Cluster 0 maps to both perivascular macrophages and microglia - not clear which label to apply.</span>
<span id="cb158-700"><a href="#cb158-700" aria-hidden="true" tabindex="-1"></a>The mapping in this subcluster to the Yang data is somewhat rough in several clusters.</span>
<span id="cb158-701"><a href="#cb158-701" aria-hidden="true" tabindex="-1"></a>Would be good to get different marker lists to try.</span>
<span id="cb158-702"><a href="#cb158-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-705"><a href="#cb158-705" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-706"><a href="#cb158-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-707"><a href="#cb158-707" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-708"><a href="#cb158-708" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"Microglia"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Microglia"</span>, </span>
<span id="cb158-709"><a href="#cb158-709" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Perivascular Macrophage"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Microglia"</span>, </span>
<span id="cb158-710"><a href="#cb158-710" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Oligodendrocyte"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Perivascular Macrophage"</span>,</span>
<span id="cb158-711"><a href="#cb158-711" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Neuron"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"Meningeal-FB/Ependymal"</span>, </span>
<span id="cb158-712"><a href="#cb158-712" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"Capillary"</span>, <span class="st">"9"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Cortex"</span>,</span>
<span id="cb158-713"><a href="#cb158-713" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"10"</span> <span class="ot">=</span> <span class="st">"OPC"</span>)</span>
<span id="cb158-714"><a href="#cb158-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-715"><a href="#cb158-715" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-716"><a href="#cb158-716" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Microglia<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-717"><a href="#cb158-717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Microglia)</span>
<span id="cb158-718"><a href="#cb158-718" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Microglia <span class="ot">&lt;-</span></span>
<span id="cb158-719"><a href="#cb158-719" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Microglia, sub_celltypes)</span>
<span id="cb158-720"><a href="#cb158-720" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-721"><a href="#cb158-721" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>Microglia, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-722"><a href="#cb158-722" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-723"><a href="#cb158-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-724"><a href="#cb158-724" aria-hidden="true" tabindex="-1"></a><span class="fu">### OPCs</span></span>
<span id="cb158-725"><a href="#cb158-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-728"><a href="#cb158-728" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-729"><a href="#cb158-729" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb158-730"><a href="#cb158-730" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb158-731"><a href="#cb158-731" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span>OPCs</span>
<span id="cb158-732"><a href="#cb158-732" aria-hidden="true" tabindex="-1"></a>sce_markers_plot<span class="sc">$</span>OPCs</span>
<span id="cb158-733"><a href="#cb158-733" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-734"><a href="#cb158-734" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-737"><a href="#cb158-737" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-738"><a href="#cb158-738" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb158-739"><a href="#cb158-739" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb158-740"><a href="#cb158-740" aria-hidden="true" tabindex="-1"></a>sub_celltypes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="st">"OPC"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Microglia"</span>, </span>
<span id="cb158-741"><a href="#cb158-741" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"ambiguous"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Oligodendrocyte"</span>, </span>
<span id="cb158-742"><a href="#cb158-742" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Neuron"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"Meningeal-FB"</span>,</span>
<span id="cb158-743"><a href="#cb158-743" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"Astrocyte-Cortex"</span>, <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"Neuron"</span>, </span>
<span id="cb158-744"><a href="#cb158-744" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"Capillary/T-Perictye"</span>)</span>
<span id="cb158-745"><a href="#cb158-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-746"><a href="#cb158-746" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the new celltypes</span></span>
<span id="cb158-747"><a href="#cb158-747" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>OPCs<span class="sc">$</span>subclusters <span class="ot">&lt;-</span></span>
<span id="cb158-748"><a href="#cb158-748" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>OPCs)</span>
<span id="cb158-749"><a href="#cb158-749" aria-hidden="true" tabindex="-1"></a>subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>OPCs <span class="ot">&lt;-</span></span>
<span id="cb158-750"><a href="#cb158-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">RenameIdents</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>OPCs, sub_celltypes)</span>
<span id="cb158-751"><a href="#cb158-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-752"><a href="#cb158-752" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(subsets<span class="sc">$</span>seurat_obj<span class="sc">$</span>OPCs, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-753"><a href="#cb158-753" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb158-754"><a href="#cb158-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-755"><a href="#cb158-755" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- # Use Yang 2022 data to query cell types --&gt;</span></span>
<span id="cb158-756"><a href="#cb158-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-757"><a href="#cb158-757" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The Seurat package has functions for querying celltype data from one dataset to another --&gt;</span></span>
<span id="cb158-758"><a href="#cb158-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-761"><a href="#cb158-761" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb158-762"><a href="#cb158-762" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb158-763"><a href="#cb158-763" aria-hidden="true" tabindex="-1"></a>gds <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE163577"</span>)</span>
<span id="cb158-764"><a href="#cb158-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-765"><a href="#cb158-765" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(<span class="fu">pData</span>(<span class="fu">phenoData</span>(gds[[<span class="dv">1</span>]]))[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">6</span>,<span class="dv">8</span>)])</span>
<span id="cb158-766"><a href="#cb158-766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-767"><a href="#cb158-767" aria-hidden="true" tabindex="-1"></a>expr_matrix <span class="ot">&lt;-</span> <span class="fu">exprs</span>(gds<span class="sc">$</span>GSE163577_series_matrix.txt.gz)</span>
<span id="cb158-768"><a href="#cb158-768" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> expr_matrix)</span>
<span id="cb158-769"><a href="#cb158-769" aria-hidden="true" tabindex="-1"></a><span class="fu">AddMetaData</span>(<span class="at">object =</span> seurat_obj, <span class="at">metadata =</span> gds<span class="sc">$</span>GSE163577_series_matrix.txt.gz<span class="sc">$</span><span class="st">`</span><span class="at">brain region:ch1</span><span class="st">`</span>, <span class="at">col.name =</span> <span class="st">"brain_region"</span>)</span>
<span id="cb158-770"><a href="#cb158-770" aria-hidden="true" tabindex="-1"></a><span class="fu">AddMetaData</span>(<span class="at">object =</span> seurat_obj, <span class="at">metadata =</span> gds<span class="sc">$</span>GSE163577_series_matrix.txt.gz<span class="sc">$</span><span class="st">`</span><span class="at">disease:ch1</span><span class="st">`</span>, <span class="at">col.name =</span> <span class="st">"disease_status"</span>)</span>
<span id="cb158-771"><a href="#cb158-771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>