<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Blood brain barrier in AD - Cell populations</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="../../site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.29/datatables.js"></script><script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><link href="../../site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="../../site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><link rel="stylesheet" href="styles.css">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Blood brain barrier in AD</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../06_dissemination/03_website/01_data-analysis.html" rel="" target="" aria-current="page">
 <span class="menu-text">Data analysis</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../06_dissemination/01_reports_conferences/2023-09-12_cardiff-connectome-talk/2023-09-12_cardiff-connectome-talk.html" rel="" target="">
 <span class="menu-text">Talks</span></a>
  </li>  
</ul>
<div class="quarto-navbar-tools ms-auto">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html">Secondary analysis</a></li><li class="breadcrumb-item"><a href="../../04_data_analysis/001_snrnaseq_analysis/14_cell_populations.html">Cell populations</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../06_dissemination/03_website/01_data-analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis Pipeline</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Post-seq QC and Prep</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/01_fastqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run FastQC on sample</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/02_multiqc.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run MultiQC</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/03_generate_cellranger_reference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/04_cellranger_count_set1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/05_cellranger_count_set2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/06_cellranger_count_set3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Generate cellranger reference for set 3</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/07_cellranger_aggr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Aggregate cellranger count reports</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">scFlow</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/08_scflow_prep.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scflow prep</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/09_scflow_run.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Run SCflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/10_scflow_results.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Analysis scFlow results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Initial analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform Pseudobulk</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform pathway analysis</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Secondary analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform Pseudobulk</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/12_pathway_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Perform pathway analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/13_cell_subtypes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Annotate cell subtypes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../04_data_analysis/001_snrnaseq_analysis/14_cell_populations.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Cell populations</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages"><span class="header-section-number">1</span> Load packages</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data"><span class="header-section-number">2</span> Read in data</a></li>
  <li>
<a href="#cell-counts" id="toc-cell-counts" class="nav-link" data-scroll-target="#cell-counts"><span class="header-section-number">3</span> Cell counts</a>
  <ul class="collapse">
<li><a href="#check-for-significant-differences-in-celltypes-between-groups" id="toc-check-for-significant-differences-in-celltypes-between-groups" class="nav-link" data-scroll-target="#check-for-significant-differences-in-celltypes-between-groups"><span class="header-section-number">3.1</span> Check for significant differences in celltypes between groups</a></li>
  </ul>
</li>
  <li><a href="#average-expression" id="toc-average-expression" class="nav-link" data-scroll-target="#average-expression"><span class="header-section-number">4</span> Average expression</a></li>
  <li>
<a href="#cellchat" id="toc-cellchat" class="nav-link" data-scroll-target="#cellchat"><span class="header-section-number">5</span> CellChat</a>
  <ul class="collapse">
<li><a href="#create-cellchat-object" id="toc-create-cellchat-object" class="nav-link" data-scroll-target="#create-cellchat-object"><span class="header-section-number">5.1</span> Create CellChat object</a></li>
  <li><a href="#set-ligand-database" id="toc-set-ligand-database" class="nav-link" data-scroll-target="#set-ligand-database"><span class="header-section-number">5.2</span> Set ligand database</a></li>
  <li><a href="#preprocessing-the-expression-data-for-cell-cell-communication-analysis" id="toc-preprocessing-the-expression-data-for-cell-cell-communication-analysis" class="nav-link" data-scroll-target="#preprocessing-the-expression-data-for-cell-cell-communication-analysis"><span class="header-section-number">5.3</span> Preprocessing the expression data for cell-cell communication analysis</a></li>
  <li><a href="#compute-the-communication-probability-and-infer-cellular-communication-network" id="toc-compute-the-communication-probability-and-infer-cellular-communication-network" class="nav-link" data-scroll-target="#compute-the-communication-probability-and-infer-cellular-communication-network"><span class="header-section-number">5.4</span> Compute the communication probability and infer cellular communication network</a></li>
  <li><a href="#infer-the-cell-cell-communication-at-a-signaling-pathway-level" id="toc-infer-the-cell-cell-communication-at-a-signaling-pathway-level" class="nav-link" data-scroll-target="#infer-the-cell-cell-communication-at-a-signaling-pathway-level"><span class="header-section-number">5.5</span> Infer the cell-cell communication at a signaling pathway level</a></li>
  <li><a href="#calculate-the-aggregated-cell-cell-communication-network" id="toc-calculate-the-aggregated-cell-cell-communication-network" class="nav-link" data-scroll-target="#calculate-the-aggregated-cell-cell-communication-network"><span class="header-section-number">5.6</span> Calculate the aggregated cell-cell communication network</a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Cell populations</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header><p>We’re interested in understanding how cell types differ between the vascular and parenchymal fractions in AD and controls. I’ll start by looking at the simple number of cells across each</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/i_am.html">i_am</a></span><span class="op">(</span><span class="st">"04_data_analysis/001_snrnaseq_analysis/14_cell_populations.qmd"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="load-packages" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Load packages</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="read-in-data" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Read in data</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># read data</span></span>
<span><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># make count rownames match data rownames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sce</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">RNA</span><span class="op">@</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">donor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"donor-"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">individual</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="cell-counts" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Cell counts</h1>
<p>Now I’ll get a table of the cell numbers per celltype per prep per diagnosis</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get the counts per celltype per prep per diagnosis</span></span>
<span><span class="va">counts_per_group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, <span class="va">sce</span><span class="op">$</span><span class="va">prep</span>, <span class="va">sce</span><span class="op">$</span><span class="va">diagnosis</span><span class="op">)</span></span>
<span><span class="co"># Check table</span></span>
<span><span class="va">counts_per_group</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>, ,  = Case

             
                  P     V
  Oligo       57502  1050
  Endothelial  1299 46885
  Pericyte      481 21515
  Astro       24225   424
  SMC            30 10966
  Neuron_ex   14691   457
  OPC          6638   207
  Microglia    7552  1104
  Neuron_in    7282   132
  Fibroblast    904  3539
  ambiguous    2310   117
  T-cell        348   697

, ,  = Control

             
                  P     V
  Oligo       63836   258
  Endothelial  2136 61084
  Pericyte      668 25259
  Astro       32564   109
  SMC            31  8563
  Neuron_ex   30756   200
  OPC          9859    86
  Microglia    7428   523
  Neuron_in   14240    66
  Fibroblast    784  2382
  ambiguous    1951    34
  T-cell        404   781</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Convert to df for plots</span></span>
<span><span class="va">counts_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">counts_per_group</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>celltype <span class="op">=</span> <span class="va">Var1</span>, fraction <span class="op">=</span> <span class="va">Var2</span>, diagnosis <span class="op">=</span> <span class="va">Var3</span><span class="op">)</span></span>
<span><span class="co"># Show as datatable</span></span>
<span><span class="va">counts_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">celltype</span>, <span class="va">fraction</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-f0f02263dae0f5bc176b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f0f02263dae0f5bc176b">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48"],["Oligo","Oligo","Oligo","Oligo","Endothelial","Endothelial","Endothelial","Endothelial","Pericyte","Pericyte","Pericyte","Pericyte","Astro","Astro","Astro","Astro","SMC","SMC","SMC","SMC","Neuron_ex","Neuron_ex","Neuron_ex","Neuron_ex","OPC","OPC","OPC","OPC","Microglia","Microglia","Microglia","Microglia","Neuron_in","Neuron_in","Neuron_in","Neuron_in","Fibroblast","Fibroblast","Fibroblast","Fibroblast","ambiguous","ambiguous","ambiguous","ambiguous","T-cell","T-cell","T-cell","T-cell"],["P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V","P","P","V","V"],["Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control","Case","Control"],[57502,63836,1050,258,1299,2136,46885,61084,481,668,21515,25259,24225,32564,424,109,30,31,10966,8563,14691,30756,457,200,6638,9859,207,86,7552,7428,1104,523,7282,14240,132,66,904,784,3539,2382,2310,1951,117,34,348,404,697,781]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>fraction<\/th>\n      <th>diagnosis<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Might be easier to parse as a plot</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Stacked bar plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">counts_df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">fraction</span>, y <span class="op">=</span> <span class="va">Freq</span>, fill <span class="op">=</span> <span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">diagnosis</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Fraction"</span>, y <span class="op">=</span> <span class="st">"Counts"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_viridis_d</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Grouped bar plot</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">counts_df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">fraction</span>, y <span class="op">=</span> <span class="va">Freq</span>, fill <span class="op">=</span> <span class="va">celltype</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_bar</span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span> <span class="va">diagnosis</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Fraction"</span>, y <span class="op">=</span> <span class="st">"Counts"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_viridis_d</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-6-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">counts_df</span>, <span class="va">counts_per_donor</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="check-for-significant-differences-in-celltypes-between-groups" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="check-for-significant-differences-in-celltypes-between-groups">
<span class="header-section-number">3.1</span> Check for significant differences in celltypes between groups</h2>
<p>Add a simple pairwise t-test/wilcox for the cell counts per celltype per fraction in AD vs controls</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get cell counts per celltype and doner</span></span>
<span><span class="va">sce</span><span class="op">$</span><span class="va">fraction_disesase_doner</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sce</span><span class="op">$</span><span class="va">prep</span>, <span class="st">"_"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">diagnosis</span>, <span class="st">"_"</span>, <span class="va">sce</span><span class="op">$</span><span class="va">donor</span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span>, <span class="va">sce</span><span class="op">$</span><span class="va">fraction_disesase_doner</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># convert to dataframe and rename columns</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span></span>
<span>    celltype <span class="op">=</span> <span class="va">Var1</span>,</span>
<span>    fraction_disesase_doner <span class="op">=</span> <span class="va">Var2</span>,</span>
<span>    count <span class="op">=</span> <span class="va">Freq</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Split column into three</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span></span>
<span>    <span class="va">fraction_disesase_doner</span>,</span>
<span>    into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"fraction"</span>, <span class="st">"diagnosis"</span>, <span class="st">"doner"</span><span class="op">)</span>,</span>
<span>    sep <span class="op">=</span> <span class="st">"_"</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="co"># Group by celltype and fraction to perform pairwise t-tests/wilcox </span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">celltype</span>, <span class="va">fraction</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span></span>
<span>    ttest_adj_pval <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/pairwise.t.test.html">pairwise.t.test</a></span><span class="op">(</span><span class="va">count</span>, <span class="va">diagnosis</span>, p.adjust.method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span>, <span class="fl">4</span><span class="op">)</span>,</span>
<span>    wilcox_adj_pval <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/pairwise.wilcox.test.html">pairwise.wilcox.test</a></span><span class="op">(</span><span class="va">count</span>, <span class="va">diagnosis</span>, p.adjust.method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span><span class="op">$</span><span class="va">p.value</span>, <span class="fl">4</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">cell_counts</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-ee619b4a55ab685251e3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ee619b4a55ab685251e3">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24"],["Oligo","Oligo","Endothelial","Endothelial","Pericyte","Pericyte","Astro","Astro","SMC","SMC","Neuron_ex","Neuron_ex","OPC","OPC","Microglia","Microglia","Neuron_in","Neuron_in","Fibroblast","Fibroblast","ambiguous","ambiguous","T-cell","T-cell"],["P","V","P","V","P","V","P","V","P","V","P","V","P","V","P","V","P","V","P","V","P","V","P","V"],[[0.7889],[0.0355],[0.098],[0.3201],[0.2457],[0.607],[0.241],[0.0254],[0.9524],[0.5162],[0.0958],[0.1979],[0.0945],[0.1044],[0.7641],[0.0517],[0.0371],[0.1429],[0.6228],[0.3159],[0.8539],[0.1087],[0.7847],[0.7805]],[[0.9668],[0.0541],[0.1642],[0.7994],[0.3682],[0.6205000000000001],[0.3803],[0.2416],[0.8728],[0.525],[0.0407],[0.2956],[0.144],[0.3391],[0.9889],[0.0809],[0.0617],[0.1645],[0.9888],[0.4407],[0.5145999999999999],[0.6564],[0.6125],[0.8075]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>fraction<\/th>\n      <th>ttest_adj_pval<\/th>\n      <th>wilcox_adj_pval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">ttest_adj_pval</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-de414ecb1ef9a0e054ff" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-de414ecb1ef9a0e054ff">{"x":{"filter":"none","vertical":false,"data":[["1","2","3"],["Oligo","Astro","Neuron_in"],["V","V","P"],[[0.0355],[0.0254],[0.0371]],[[0.0541],[0.2416],[0.0617]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>fraction<\/th>\n      <th>ttest_adj_pval<\/th>\n      <th>wilcox_adj_pval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">wilcox_adj_pval</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-2153cb77a61b3dcd1273" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2153cb77a61b3dcd1273">{"x":{"filter":"none","vertical":false,"data":[["1"],["Neuron_ex"],["P"],[[0.0958]],[[0.0407]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>celltype<\/th>\n      <th>fraction<\/th>\n      <th>ttest_adj_pval<\/th>\n      <th>wilcox_adj_pval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Seems like Astrocytes and Oligodendrocytes are significant in the vascular fraction, and the inhibitory neurons in the parenchymal fraction for the t-tests</p>
<ul>
<li>NOTE: need to add checks for normalcy</li>
<li>NOTE: not working on render - need to debug</li>
</ul>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get unique cell types</span></span>
<span><span class="va">unique_cell_types</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a function to perform the tests and adjust p-values</span></span>
<span><span class="va">perform_tests</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">cell_type</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Subset cell counts for the current cell type in AD and control groups</span></span>
<span>  <span class="va">cell_counts_AD</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">$</span><span class="va">cell_counts</span><span class="op">[</span><span class="va">sce</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"AD"</span> <span class="op">&amp;</span> <span class="va">sce</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">==</span> <span class="va">cell_type</span><span class="op">]</span></span>
<span>  <span class="va">cell_counts_control</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">$</span><span class="va">cell_counts</span><span class="op">[</span><span class="va">sce</span><span class="op">$</span><span class="va">group</span> <span class="op">==</span> <span class="st">"control"</span> <span class="op">&amp;</span> <span class="va">sce</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">==</span> <span class="va">cell_type</span><span class="op">]</span></span>
<span>  </span>
<span>  <span class="co"># Perform t-test</span></span>
<span>  <span class="va">ttest_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/t.test.html">t.test</a></span><span class="op">(</span><span class="va">cell_counts_AD</span>, <span class="va">cell_counts_control</span><span class="op">)</span></span>
<span>  <span class="va">ttest_pvalue</span> <span class="op">&lt;-</span> <span class="va">ttest_result</span><span class="op">$</span><span class="va">p.value</span></span>
<span>  </span>
<span>  <span class="co"># Perform Wilcoxon rank-sum test</span></span>
<span>  <span class="va">wilcox_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test</a></span><span class="op">(</span><span class="va">cell_counts_AD</span>, <span class="va">cell_counts_control</span><span class="op">)</span></span>
<span>  <span class="va">wilcox_pvalue</span> <span class="op">&lt;-</span> <span class="va">wilcox_result</span><span class="op">$</span><span class="va">p.value</span></span>
<span>  </span>
<span>  <span class="co"># Return adjusted p-values</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>CellType <span class="op">=</span> <span class="va">cell_type</span>, TTest_PValue <span class="op">=</span> <span class="va">ttest_pvalue</span>, Wilcox_PValue <span class="op">=</span> <span class="va">wilcox_pvalue</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Apply the function to each unique cell type using map</span></span>
<span><span class="va">test_results</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">unique_cell_types</span>, <span class="va">perform_tests</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract p-values</span></span>
<span><span class="va">ttest_pvalues</span> <span class="op">&lt;-</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">test_results</span>, <span class="st">"TTest_PValue"</span><span class="op">)</span></span>
<span><span class="va">wilcox_pvalues</span> <span class="op">&lt;-</span> <span class="fu">map_dbl</span><span class="op">(</span><span class="va">test_results</span>, <span class="st">"Wilcox_PValue"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Adjust p-values using the Benjamini-Hochberg procedure</span></span>
<span><span class="va">ttest_pvalues_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">ttest_pvalues</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span><span class="va">wilcox_pvalues_adjusted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html">p.adjust</a></span><span class="op">(</span><span class="va">wilcox_pvalues</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a dataframe with the results</span></span>
<span><span class="va">result_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>CellType <span class="op">=</span> <span class="va">unique_cell_types</span>, TTest_PValue <span class="op">=</span> <span class="va">ttest_pvalues</span>, TTest_PValue_Adjusted <span class="op">=</span> <span class="va">ttest_pvalues_adjusted</span>, Wilcox_PValue <span class="op">=</span> <span class="va">wilcox_pvalues</span>, Wilcox_PValue_Adjusted <span class="op">=</span> <span class="va">wilcox_pvalues_adjusted</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">result_df</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section></section><section id="average-expression" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Average expression</h1>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="fu">AverageExpression</span><span class="op">(</span><span class="va">sce</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ident"</span>, <span class="st">"prep"</span>, <span class="st">"diagnosis"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="va">avg_exp</span><span class="op">$</span><span class="va">RNA</span></span>
<span> </span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="op">(</span><span class="va">avg_exp</span>, keep.rownames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">setnames</span><span class="op">(</span><span class="va">avg_exp</span>, <span class="st">"rn"</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># pivot longer such that celltypes are separated out</span></span>
<span><span class="va">avg_exp</span> <span class="op">&lt;-</span> <span class="va">avg_exp</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">!</span><span class="va">gene</span>,</span>
<span>    names_to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"celltype"</span>, <span class="st">".value"</span><span class="op">)</span>,</span>
<span>    names_pattern <span class="op">=</span> <span class="st">"(.*)_(.*)"</span></span>
<span>  <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="cellchat" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> CellChat</h1>
<p><code>CellChat</code> is a package for looking at cell-cell interaction networks</p>
<p>It requires a normalised matrix of gene expression data (genes as rows, cells as columns). I assume from <a href="https://htmlpreview.github.io/?https://github.com/sqjin/CellChat/blob/master/tutorial/CellChat-vignette.html">the vignette</a> that one should look at one group at a time, in this case each fraction per disease state, so 4 groups total.</p>
<p>I’ll start with the vascular fraction of the cases whilst experimenting</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># get the normalised expression data</span></span>
<span><span class="va">expression_data</span> <span class="op">&lt;-</span> <span class="fu">GetAssayData</span><span class="op">(</span><span class="va">sce</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># a dataframe with rownames containing cell mata data</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">@</span><span class="va">meta.data</span></span>
<span><span class="co"># add manual celltype labels</span></span>
<span><span class="va">meta</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="fu">Idents</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="co"># extract the cell names from disease data</span></span>
<span><span class="va">cell_use</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">[</span><span class="va">meta</span><span class="op">$</span><span class="va">diagnosis</span> <span class="op">==</span> <span class="st">"Case"</span> <span class="op">&amp;</span> <span class="va">meta</span><span class="op">$</span><span class="va">prep</span> <span class="op">==</span> <span class="st">"V"</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Here I use <code>expand.grid</code> to get all unique combinations of diagnosis and prep to get cell networks from.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Obtain unique combinations of meta$diagnosis and meta$prep</span></span>
<span><span class="va">combinations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/expand.grid.html">expand.grid</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">diagnosis</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">prep</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert the combinations to a list</span></span>
<span><span class="va">combinations_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">combinations</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the list of unique combinations</span></span>
<span><span class="va">combinations_list</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$Var1
[1] Case    Control Case    Control
Levels: Case Control

$Var2
[1] P P V V
Levels: P V

attr(,"out.attrs")
attr(,"out.attrs")$dim
[1] 2 2

attr(,"out.attrs")$dimnames
attr(,"out.attrs")$dimnames$Var1
[1] "Var1=Case"    "Var1=Control"

attr(,"out.attrs")$dimnames$Var2
[1] "Var2=P" "Var2=V"</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Return all subsets</span></span>
<span><span class="co"># Create a function to generate subsets based on the combinations</span></span>
<span><span class="va">get_subsets</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">diagnosis</span>, <span class="va">prep</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">[</span><span class="va">meta</span><span class="op">$</span><span class="va">diagnosis</span> <span class="op">==</span> <span class="va">diagnosis</span> <span class="op">&amp;</span> <span class="va">meta</span><span class="op">$</span><span class="va">prep</span> <span class="op">==</span> <span class="va">prep</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Use map2 to generate subsets for each combination</span></span>
<span><span class="va">all_subsets</span> <span class="op">&lt;-</span> <span class="fu">map2</span><span class="op">(</span><span class="va">combinations</span><span class="op">$</span><span class="va">Var1</span>, <span class="va">combinations</span><span class="op">$</span><span class="va">Var2</span>, <span class="op">~</span> <span class="fu">get_subsets</span><span class="op">(</span><span class="va">.x</span>, <span class="va">.y</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html">set_names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">combinations_list</span><span class="op">$</span><span class="va">Var1</span>, <span class="st">"+"</span>, <span class="va">combinations_list</span><span class="op">$</span><span class="va">Var2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Print the list of subsets</span></span>
<span><span class="fu">map</span><span class="op">(</span><span class="va">all_subsets</span>, <span class="va">head</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$`Case+P`
[1] "P20_AAACCCAAGAAGCCTG-1" "P20_AAACCCAAGAGGCGGA-1" "P20_AAACCCAAGTAAGCAT-1"
[4] "P20_AAACCCACACTAGGTT-1" "P20_AAACCCAGTTAGGGTG-1" "P20_AAACCCAGTTTCCAAG-1"

$`Control+P`
[1] "P01_AAACCCAAGAGTCCGA-1" "P01_AAACCCAAGATGCCGA-1" "P01_AAACCCAAGGTAAGAG-1"
[4] "P01_AAACCCAAGTAGGAAG-1" "P01_AAACCCACAGTTCACA-1" "P01_AAACCCACATGGGATG-1"

$`Case+V`
[1] "V20_AAACCCATCTGCGAGC-1" "V20_AAACCCATCTTCGTAT-1" "V20_AAACGCTAGGCTAAAT-1"
[4] "V20_AAACGCTGTCGTCTCT-1" "V20_AAAGAACCAGCAGTGA-1" "V20_AAAGAACTCATGACAC-1"

$`Control+V`
[1] "V19_AAACCCAGTCAGGCAA-1" "V19_AAACCCAGTCCTACAA-1" "V19_AAACCCATCCGTGACG-1"
[4] "V19_AAACGAACAGGCATGA-1" "V19_AAACGCTTCAACGAGG-1" "V19_AAACGCTTCTCTGGTC-1"</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare input data for CelChat analysis</span></span>
<span><span class="va">data_input</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">all_subsets</span>, <span class="op">~</span> <span class="va">expression_data</span><span class="op">[</span>, <span class="va">.x</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">all_subsets</span>, <span class="op">~</span> <span class="va">meta</span><span class="op">[</span><span class="va">.x</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="co"># check the cell labels</span></span>
<span><span class="fu">map</span><span class="op">(</span><span class="va">meta</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">.x</span><span class="op">$</span><span class="va">label</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$`Case+P`
 [1] Oligo       ambiguous   Astro       Microglia   OPC         Neuron_in  
 [7] Neuron_ex   Pericyte    Fibroblast  Endothelial T-cell      SMC        
12 Levels: Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC ... T-cell

$`Control+P`
 [1] Neuron_ex   Neuron_in   Oligo       Astro       OPC         ambiguous  
 [7] Microglia   Endothelial Pericyte    T-cell      Fibroblast  SMC        
12 Levels: Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC ... T-cell

$`Case+V`
 [1] Endothelial Pericyte    SMC         Oligo       Fibroblast  Neuron_ex  
 [7] Microglia   T-cell      ambiguous   OPC         Astro       Neuron_in  
12 Levels: Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC ... T-cell

$`Control+V`
 [1] Pericyte    SMC         Endothelial Oligo       Fibroblast  Microglia  
 [7] ambiguous   Astro       OPC         T-cell      Neuron_ex   Neuron_in  
12 Levels: Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC ... T-cell</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">map</span><span class="op">(</span><span class="va">meta</span>, <span class="va">dim</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>$`Case+P`
[1] 123262     37

$`Control+P`
[1] 164657     37

$`Case+V`
[1] 87093    37

$`Control+V`
[1] 99345    37</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">expression_data</span>, <span class="va">combinations</span>, <span class="va">combinations_list</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul>
<li>NOTE: need to remove this once switching to map call, otherwise code will break!</li>
</ul>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Prepare input data for CelChat analysis</span></span>
<span><span class="co">#data_input &lt;- expression_data[, cell_use]</span></span>
<span><span class="co">#meta &lt;- meta[cell_use, ]</span></span>
<span><span class="co"># check the cell labels</span></span>
<span><span class="co">#unique(meta$label)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="create-cellchat-object" class="level2" data-number="5.1"><h2 data-number="5.1" class="anchored" data-anchor-id="create-cellchat-object">
<span class="header-section-number">5.1</span> Create CellChat object</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cellchat</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">map2</span><span class="op">(</span><span class="va">data_input</span>,</span>
<span>       <span class="va">meta</span>,</span>
<span>       <span class="op">~</span> <span class="fu">createCellChat</span><span class="op">(</span></span>
<span>         object <span class="op">=</span> <span class="va">.x</span>,</span>
<span>         meta <span class="op">=</span> <span class="va">.y</span>,</span>
<span>         group.by <span class="op">=</span> <span class="st">"label"</span></span>
<span>       <span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] "Create a CellChat object from a data matrix"
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC Microglia Neuron_in Fibroblast ambiguous T-cell 
[1] "Create a CellChat object from a data matrix"
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC Microglia Neuron_in Fibroblast ambiguous T-cell 
[1] "Create a CellChat object from a data matrix"
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC Microglia Neuron_in Fibroblast ambiguous T-cell 
[1] "Create a CellChat object from a data matrix"
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  Oligo Endothelial Pericyte Astro SMC Neuron_ex OPC Microglia Neuron_in Fibroblast ambiguous T-cell </code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">data_input</span>, <span class="va">meta</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="set-ligand-database" class="level2" data-number="5.2"><h2 data-number="5.2" class="anchored" data-anchor-id="set-ligand-database">
<span class="header-section-number">5.2</span> Set ligand database</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">CellChatDB</span> <span class="op">&lt;-</span> <span class="va">CellChatDB.human</span> <span class="co"># use CellChatDB.mouse if running on mouse data</span></span>
<span><span class="fu">showDatabaseCategory</span><span class="op">(</span><span class="va">CellChatDB</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>One can subset of the DB for analysis, but I’ll use all for now</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Show the structure of the database</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html">glimpse</a></span><span class="op">(</span><span class="va">CellChatDB</span><span class="op">$</span><span class="va">interaction</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>Rows: 1,939
Columns: 11
$ interaction_name   &lt;chr&gt; "TGFB1_TGFBR1_TGFBR2", "TGFB2_TGFBR1_TGFBR2", "TGFB…
$ pathway_name       &lt;chr&gt; "TGFb", "TGFb", "TGFb", "TGFb", "TGFb", "TGFb", "TG…
$ ligand             &lt;chr&gt; "TGFB1", "TGFB2", "TGFB3", "TGFB1", "TGFB1", "TGFB2…
$ receptor           &lt;chr&gt; "TGFbR1_R2", "TGFbR1_R2", "TGFbR1_R2", "ACVR1B_TGFb…
$ agonist            &lt;chr&gt; "TGFb agonist", "TGFb agonist", "TGFb agonist", "TG…
$ antagonist         &lt;chr&gt; "TGFb antagonist", "TGFb antagonist", "TGFb antagon…
$ co_A_receptor      &lt;chr&gt; "", "", "", "", "", "", "", "", "", "", "", "", "",…
$ co_I_receptor      &lt;chr&gt; "TGFb inhibition receptor", "TGFb inhibition recept…
$ evidence           &lt;chr&gt; "KEGG: hsa04350", "KEGG: hsa04350", "KEGG: hsa04350…
$ annotation         &lt;chr&gt; "Secreted Signaling", "Secreted Signaling", "Secret…
$ interaction_name_2 &lt;chr&gt; "TGFB1 - (TGFBR1+TGFBR2)", "TGFB2 - (TGFBR1+TGFBR2)…</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># use a subset of CellChatDB for cell-cell communication analysis</span></span>
<span><span class="co">#CellChatDB_use &lt;- subsetDB(CellChatDB, search = "Secreted Signaling")</span></span>
<span><span class="co"># use all CellChatDB for cell-cell communication analysis</span></span>
<span><span class="va">CellChatDB_use</span> <span class="op">&lt;-</span> <span class="va">CellChatDB</span> <span class="co"># simply use the default CellChatDB</span></span>
<span></span>
<span><span class="co"># set the used database in the object</span></span>
<span><span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>  <span class="va">.x</span><span class="op">@</span><span class="va">DB</span> <span class="op">&lt;-</span> <span class="va">CellChatDB_use</span></span>
<span>  <span class="va">.x</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">CellChatDB_use</span>, <span class="va">CellChatDB</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="preprocessing-the-expression-data-for-cell-cell-communication-analysis" class="level2" data-number="5.3"><h2 data-number="5.3" class="anchored" data-anchor-id="preprocessing-the-expression-data-for-cell-cell-communication-analysis">
<span class="header-section-number">5.3</span> Preprocessing the expression data for cell-cell communication analysis</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># subset the expression data of signaling genes for saving computation cost</span></span>
<span><span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">subsetData</span><span class="op">)</span> <span class="co"># This step is necessary even if using the whole database</span></span>
<span><span class="co">#future::plan("multiprocess", workers = 4) # do parallel</span></span>
<span><span class="fu">future</span><span class="fu">::</span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html">plan</a></span><span class="op">(</span><span class="st">"multicore"</span>, workers <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="co"># do parallel</span></span>
<span><span class="co"># Ran into error with this option being too small</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>future.globals.maxSize <span class="op">=</span> <span class="fl">5000000000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># This takes ages to run, so save the output</span></span>
<span><span class="co"># Define the file path to save the resulting object</span></span>
<span><span class="va">result_file</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>    <span class="st">"03_data/990_processed_data/001_snrnaseq/11_cell_network_interactions/cellchat_preprocessed.rds"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if the file exists</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Code for your time-consuming analysis</span></span>
<span>  <span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">identifyOverExpressedGenes</span><span class="op">)</span></span>
<span>  <span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">identifyOverExpressedInteractions</span><span class="op">)</span></span>
<span></span>
<span>  </span>
<span>  <span class="co"># Save the resulting object</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="co"># Load the existing resulting object</span></span>
<span>  <span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># project gene expression data onto PPI (Optional: when running it, USER should</span></span>
<span><span class="co"># set `raw.use = FALSE` in the function `computeCommunProb()` in order to use</span></span>
<span><span class="co"># the projected data)</span></span>
<span><span class="co"># cellchat &lt;- projectData(cellchat, PPI.human)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="compute-the-communication-probability-and-infer-cellular-communication-network" class="level2" data-number="5.4"><h2 data-number="5.4" class="anchored" data-anchor-id="compute-the-communication-probability-and-infer-cellular-communication-network">
<span class="header-section-number">5.4</span> Compute the communication probability and infer cellular communication network</h2>
<p>If well-known signaling pathways in the studied biological process are not predicted, USER can try <code>truncatedMean</code> to change the method for calculating the average gene expression per cell group.</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Again, this takes quite a while to run</span></span>
<span><span class="co"># Define the file path to save the resulting object</span></span>
<span><span class="va">result_file</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span></span>
<span>    <span class="st">"03_data/990_processed_data/001_snrnaseq/11_cell_network_interactions/cellchat_communication-prob.rds"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if the file exists</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Code for your time-consuming analysis</span></span>
<span>  <span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">computeCommunProb</span><span class="op">)</span></span>
<span>  <span class="co"># Filter out the cell-cell communication if there are only few number of cells in certain cell groups</span></span>
<span>  <span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="op">~</span> <span class="fu">filterCommunication</span><span class="op">(</span><span class="va">.x</span>, min.cells <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="co"># Save the resulting object</span></span>
<span>  <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">write_rds</a></span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="co"># Load the existing resulting object</span></span>
<span>  <span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_rds.html">read_rds</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html">rm</a></span><span class="op">(</span><span class="va">result_file</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We can get the cell interaction networks as a dataframe</p>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">df_net</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">subsetCommunication</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">df_net</span><span class="op">$</span><span class="va">`Case+P`</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>     source    target ligand  receptor        prob pval    interaction_name
1 Microglia Microglia  TGFB1 TGFbR1_R2 0.007058124    0 TGFB1_TGFBR1_TGFBR2
2    T-cell Microglia  TGFB1 TGFbR1_R2 0.007220612    0 TGFB1_TGFBR1_TGFBR2
3 Microglia    T-cell  TGFB1 TGFbR1_R2 0.003806521    0 TGFB1_TGFBR1_TGFBR2
4    T-cell    T-cell  TGFB1 TGFbR1_R2 0.003894446    0 TGFB1_TGFBR1_TGFBR2
5     Astro Microglia  TGFB2 TGFbR1_R2 0.013077737    0 TGFB2_TGFBR1_TGFBR2
6     Astro    T-cell  TGFB2 TGFbR1_R2 0.007075986    0 TGFB2_TGFBR1_TGFBR2
       interaction_name_2 pathway_name         annotation       evidence
1 TGFB1 - (TGFBR1+TGFBR2)         TGFb Secreted Signaling KEGG: hsa04350
2 TGFB1 - (TGFBR1+TGFBR2)         TGFb Secreted Signaling KEGG: hsa04350
3 TGFB1 - (TGFBR1+TGFBR2)         TGFb Secreted Signaling KEGG: hsa04350
4 TGFB1 - (TGFBR1+TGFBR2)         TGFb Secreted Signaling KEGG: hsa04350
5 TGFB2 - (TGFBR1+TGFBR2)         TGFb Secreted Signaling KEGG: hsa04350
6 TGFB2 - (TGFBR1+TGFBR2)         TGFb Secreted Signaling KEGG: hsa04350</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># check how many aren't significant</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">df_net</span><span class="op">$</span><span class="va">`Case+P`</span><span class="op">$</span><span class="va">pval</span> <span class="op">&gt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
</section><section id="infer-the-cell-cell-communication-at-a-signaling-pathway-level" class="level2" data-number="5.5"><h2 data-number="5.5" class="anchored" data-anchor-id="infer-the-cell-cell-communication-at-a-signaling-pathway-level">
<span class="header-section-number">5.5</span> Infer the cell-cell communication at a signaling pathway level</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">computeCommunProbPathway</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="calculate-the-aggregated-cell-cell-communication-network" class="level2" data-number="5.6"><h2 data-number="5.6" class="anchored" data-anchor-id="calculate-the-aggregated-cell-cell-communication-network">
<span class="header-section-number">5.6</span> Calculate the aggregated cell-cell communication network</h2>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">aggregateNet</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">initial_plots</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">comparison</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">groupSize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">idents</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, xpd <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="fu">netVisual_circle</span><span class="op">(</span></span>
<span>    <span class="va">cellchat</span><span class="op">@</span><span class="va">net</span><span class="op">$</span><span class="va">count</span>,</span>
<span>    vertex.weight <span class="op">=</span> <span class="va">groupSize</span>,</span>
<span>    weight.scale <span class="op">=</span> <span class="cn">T</span>,</span>
<span>    label.edge <span class="op">=</span> <span class="cn">F</span>,</span>
<span>    title.name <span class="op">=</span> <span class="st">"Number of interactions"</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="fu">netVisual_circle</span><span class="op">(</span></span>
<span>    <span class="va">cellchat</span><span class="op">@</span><span class="va">net</span><span class="op">$</span><span class="va">weight</span>,</span>
<span>    vertex.weight <span class="op">=</span> <span class="va">groupSize</span>,</span>
<span>    weight.scale <span class="op">=</span> <span class="cn">T</span>,</span>
<span>    label.edge <span class="op">=</span> <span class="cn">F</span>,</span>
<span>    title.name <span class="op">=</span> <span class="st">"Interaction weights/strength"</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="va">comparison</span>, outer <span class="op">=</span> <span class="cn">TRUE</span>, line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>, cex <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">map2</span><span class="op">(</span><span class="va">cellchat</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>, <span class="va">initial_plots</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-24-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-24-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-24-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Case+P`
NULL

$`Control+P`
NULL

$`Case+V`
NULL

$`Control+V`
NULL</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">celltype_plots</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">comparison</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">groupSize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">idents</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">cellchat</span><span class="op">@</span><span class="va">net</span><span class="op">$</span><span class="va">weight</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span>, xpd <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu">map</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">mat2</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span></span>
<span>        <span class="fl">0</span>,</span>
<span>        nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>,</span>
<span>        ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>,</span>
<span>        dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="va">mat2</span><span class="op">[</span><span class="va">.x</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mat</span><span class="op">[</span><span class="va">.x</span>,<span class="op">]</span></span>
<span>    <span class="fu">netVisual_circle</span><span class="op">(</span></span>
<span>      <span class="va">mat2</span>,</span>
<span>      vertex.weight <span class="op">=</span> <span class="va">groupSize</span>,</span>
<span>      weight.scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      edge.weight.max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>,</span>
<span>      title.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span><span class="op">[</span><span class="va">.x</span><span class="op">]</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="va">comparison</span>,</span>
<span>        outer <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>,</span>
<span>        cex <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">map2</span><span class="op">(</span><span class="va">cellchat</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>, <span class="va">celltype_plots</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-25-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-25-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-25-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Case+P`
NULL

$`Control+P`
NULL

$`Case+V`
NULL

$`Control+V`
NULL</code></pre>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">celltype_plots</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">comparison</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">groupSize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">idents</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">cellchat</span><span class="op">@</span><span class="va">net</span><span class="op">$</span><span class="va">weight</span></span>
<span>  </span>
<span>  <span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">map</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>, <span class="op">~</span> <span class="op">{</span></span>
<span>    <span class="va">mat2</span> <span class="op">&lt;-</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span></span>
<span>        <span class="fl">0</span>,</span>
<span>        nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>,</span>
<span>        ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>,</span>
<span>        dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="va">mat2</span><span class="op">[</span><span class="va">.x</span>,<span class="op">]</span> <span class="op">&lt;-</span> <span class="va">mat</span><span class="op">[</span><span class="va">.x</span>,<span class="op">]</span></span>
<span>    <span class="va">plot</span> <span class="op">&lt;-</span> <span class="fu">netVisual_circle</span><span class="op">(</span></span>
<span>      <span class="va">mat2</span>,</span>
<span>      vertex.weight <span class="op">=</span> <span class="va">groupSize</span>,</span>
<span>      weight.scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>      edge.weight.max <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span>,</span>
<span>      title.name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span><span class="op">[</span><span class="va">.x</span><span class="op">]</span></span>
<span>    <span class="op">)</span></span>
<span>    <span class="va">plot</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu">map2</span><span class="op">(</span><span class="va">cellchat</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>, <span class="va">celltype_plots</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-12.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-13.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-14.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-15.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-16.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-17.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-18.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-19.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-20.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-21.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-22.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-23.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-24.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-25.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-26.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-27.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-28.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-29.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-30.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-31.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-32.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-33.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-34.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-35.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-36.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-37.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-38.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-39.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-40.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-41.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-42.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-43.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-44.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-45.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-46.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-47.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-26-48.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<details><summary>Code</summary><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span>, xpd <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">plots</span><span class="op">$</span><span class="va">`Case+P`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output cell-output-stdout">
<pre><code>[[1]]

[[2]]

[[3]]

[[4]]

[[5]]

[[6]]

[[7]]

[[8]]

[[9]]

[[10]]

[[11]]

[[12]]</code></pre>
</div>
<details><summary>Code</summary><div class="sourceCode" id="cb49"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>        outer <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        line <span class="op">=</span> <span class="op">-</span><span class="fl">1.5</span>,</span>
<span>        cex <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<p><img src="14_cell_populations_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>


<!-- -->

</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb50" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> Cell populations</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: true</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co">#self-contained: true</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="an">code-fold:</span><span class="co"> true</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>We're interested in understanding how cell types differ between the vascular and parenchymal fractions in AD and controls.</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>I'll start by looking at the simple number of cells across each</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/001_snrnaseq_analysis/14_cell_populations.qmd"</span>)</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="fu"># Load packages</span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read in data</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a><span class="co"># read data</span></span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a><span class="co">#sce &lt;- readr::read_rds(here::here("03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated.rds"))</span></span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"</span>))</span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a><span class="co"># make count rownames match data rownames</span></span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sce<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>data)</span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>donor <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"donor-"</span>, sce<span class="sc">$</span>individual)</span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-45"><a href="#cb50-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-46"><a href="#cb50-46" aria-hidden="true" tabindex="-1"></a><span class="fu"># Cell counts</span></span>
<span id="cb50-47"><a href="#cb50-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-48"><a href="#cb50-48" aria-hidden="true" tabindex="-1"></a>Now I'll get a table of the cell numbers per celltype per prep per diagnosis</span>
<span id="cb50-49"><a href="#cb50-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-52"><a href="#cb50-52" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-53"><a href="#cb50-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the counts per celltype per prep per diagnosis</span></span>
<span id="cb50-54"><a href="#cb50-54" aria-hidden="true" tabindex="-1"></a>counts_per_group <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">Idents</span>(sce), sce<span class="sc">$</span>prep, sce<span class="sc">$</span>diagnosis)</span>
<span id="cb50-55"><a href="#cb50-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Check table</span></span>
<span id="cb50-56"><a href="#cb50-56" aria-hidden="true" tabindex="-1"></a>counts_per_group</span>
<span id="cb50-57"><a href="#cb50-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-58"><a href="#cb50-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to df for plots</span></span>
<span id="cb50-59"><a href="#cb50-59" aria-hidden="true" tabindex="-1"></a>counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts_per_group) <span class="sc">%&gt;%</span></span>
<span id="cb50-60"><a href="#cb50-60" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltype =</span> Var1, <span class="at">fraction =</span> Var2, <span class="at">diagnosis =</span> Var3)</span>
<span id="cb50-61"><a href="#cb50-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Show as datatable</span></span>
<span id="cb50-62"><a href="#cb50-62" aria-hidden="true" tabindex="-1"></a>counts_df <span class="sc">%&gt;%</span></span>
<span id="cb50-63"><a href="#cb50-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(celltype, fraction) <span class="sc">%&gt;%</span></span>
<span id="cb50-64"><a href="#cb50-64" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb50-65"><a href="#cb50-65" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-66"><a href="#cb50-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-67"><a href="#cb50-67" aria-hidden="true" tabindex="-1"></a>Might be easier to parse as a plot</span>
<span id="cb50-68"><a href="#cb50-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-71"><a href="#cb50-71" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-72"><a href="#cb50-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Stacked bar plot</span></span>
<span id="cb50-73"><a href="#cb50-73" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(counts_df, <span class="fu">aes</span>(<span class="at">x =</span> fraction, <span class="at">y =</span> Freq, <span class="at">fill =</span> celltype)) <span class="sc">+</span></span>
<span id="cb50-74"><a href="#cb50-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb50-75"><a href="#cb50-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> diagnosis) <span class="sc">+</span></span>
<span id="cb50-76"><a href="#cb50-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Fraction"</span>, <span class="at">y =</span> <span class="st">"Counts"</span>) <span class="sc">+</span></span>
<span id="cb50-77"><a href="#cb50-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb50-78"><a href="#cb50-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb50-79"><a href="#cb50-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-80"><a href="#cb50-80" aria-hidden="true" tabindex="-1"></a><span class="co"># Grouped bar plot</span></span>
<span id="cb50-81"><a href="#cb50-81" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(counts_df, <span class="fu">aes</span>(<span class="at">x =</span> fraction, <span class="at">y =</span> Freq, <span class="at">fill =</span> celltype)) <span class="sc">+</span></span>
<span id="cb50-82"><a href="#cb50-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb50-83"><a href="#cb50-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> diagnosis) <span class="sc">+</span></span>
<span id="cb50-84"><a href="#cb50-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Fraction"</span>, <span class="at">y =</span> <span class="st">"Counts"</span>) <span class="sc">+</span></span>
<span id="cb50-85"><a href="#cb50-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb50-86"><a href="#cb50-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb50-87"><a href="#cb50-87" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-88"><a href="#cb50-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-91"><a href="#cb50-91" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-92"><a href="#cb50-92" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(counts_df, counts_per_donor)</span>
<span id="cb50-93"><a href="#cb50-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-94"><a href="#cb50-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-97"><a href="#cb50-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-98"><a href="#cb50-98" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb50-99"><a href="#cb50-99" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb50-100"><a href="#cb50-100" aria-hidden="true" tabindex="-1"></a><span class="co"># Try an ANOVA on the counts per donor for curiosity</span></span>
<span id="cb50-101"><a href="#cb50-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-102"><a href="#cb50-102" aria-hidden="true" tabindex="-1"></a><span class="co"># Get diagnosis</span></span>
<span id="cb50-103"><a href="#cb50-103" aria-hidden="true" tabindex="-1"></a>diagnosis_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb50-104"><a href="#cb50-104" aria-hidden="true" tabindex="-1"></a>  <span class="at">donor =</span> sce<span class="sc">$</span>donor,</span>
<span id="cb50-105"><a href="#cb50-105" aria-hidden="true" tabindex="-1"></a>  <span class="at">diagnosis =</span> sce<span class="sc">$</span>diagnosis) <span class="sc">%&gt;%</span></span>
<span id="cb50-106"><a href="#cb50-106" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb50-107"><a href="#cb50-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-108"><a href="#cb50-108" aria-hidden="true" tabindex="-1"></a>counts_per_donor <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">Idents</span>(sce), sce<span class="sc">$</span>prep, sce<span class="sc">$</span>donor) <span class="sc">%&gt;%</span></span>
<span id="cb50-109"><a href="#cb50-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-110"><a href="#cb50-110" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltype =</span> Var1, <span class="at">fraction =</span> Var2, <span class="at">donor =</span> Var3) <span class="sc">%&gt;%</span></span>
<span id="cb50-111"><a href="#cb50-111" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(diagnosis_df)</span>
<span id="cb50-112"><a href="#cb50-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-113"><a href="#cb50-113" aria-hidden="true" tabindex="-1"></a>anova_donor <span class="ot">&lt;-</span> <span class="fu">aov</span>(Freq <span class="sc">~</span> celltype <span class="sc">+</span>  fraction <span class="sc">+</span> diagnosis, <span class="at">data =</span> counts_per_donor)</span>
<span id="cb50-114"><a href="#cb50-114" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(anova_donor)</span>
<span id="cb50-115"><a href="#cb50-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-116"><a href="#cb50-116" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb50-117"><a href="#cb50-117" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(anova_donor)</span>
<span id="cb50-118"><a href="#cb50-118" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb50-119"><a href="#cb50-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-120"><a href="#cb50-120" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(anova_donor)</span>
<span id="cb50-121"><a href="#cb50-121" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-122"><a href="#cb50-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-123"><a href="#cb50-123" aria-hidden="true" tabindex="-1"></a><span class="fu">## Check for significant differences in celltypes between groups</span></span>
<span id="cb50-124"><a href="#cb50-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-125"><a href="#cb50-125" aria-hidden="true" tabindex="-1"></a>Add a simple pairwise t-test/wilcox for the cell counts per celltype per fraction in AD vs controls</span>
<span id="cb50-126"><a href="#cb50-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-129"><a href="#cb50-129" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-130"><a href="#cb50-130" aria-hidden="true" tabindex="-1"></a><span class="co"># Get cell counts per celltype and doner</span></span>
<span id="cb50-131"><a href="#cb50-131" aria-hidden="true" tabindex="-1"></a>sce<span class="sc">$</span>fraction_disesase_doner <span class="ot">&lt;-</span></span>
<span id="cb50-132"><a href="#cb50-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste0</span>(sce<span class="sc">$</span>prep, <span class="st">"_"</span>, sce<span class="sc">$</span>diagnosis, <span class="st">"_"</span>, sce<span class="sc">$</span>donor)</span>
<span id="cb50-133"><a href="#cb50-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-134"><a href="#cb50-134" aria-hidden="true" tabindex="-1"></a>cell_counts <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">Idents</span>(sce), sce<span class="sc">$</span>fraction_disesase_doner) <span class="sc">%&gt;%</span></span>
<span id="cb50-135"><a href="#cb50-135" aria-hidden="true" tabindex="-1"></a>  <span class="co"># convert to dataframe and rename columns</span></span>
<span id="cb50-136"><a href="#cb50-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-137"><a href="#cb50-137" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb50-138"><a href="#cb50-138" aria-hidden="true" tabindex="-1"></a>    <span class="at">celltype =</span> Var1,</span>
<span id="cb50-139"><a href="#cb50-139" aria-hidden="true" tabindex="-1"></a>    <span class="at">fraction_disesase_doner =</span> Var2,</span>
<span id="cb50-140"><a href="#cb50-140" aria-hidden="true" tabindex="-1"></a>    <span class="at">count =</span> Freq</span>
<span id="cb50-141"><a href="#cb50-141" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb50-142"><a href="#cb50-142" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Split column into three</span></span>
<span id="cb50-143"><a href="#cb50-143" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">separate</span>(</span>
<span id="cb50-144"><a href="#cb50-144" aria-hidden="true" tabindex="-1"></a>    fraction_disesase_doner,</span>
<span id="cb50-145"><a href="#cb50-145" aria-hidden="true" tabindex="-1"></a>    <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"fraction"</span>, <span class="st">"diagnosis"</span>, <span class="st">"doner"</span>),</span>
<span id="cb50-146"><a href="#cb50-146" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">"_"</span></span>
<span id="cb50-147"><a href="#cb50-147" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb50-148"><a href="#cb50-148" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Group by celltype and fraction to perform pairwise t-tests/wilcox </span></span>
<span id="cb50-149"><a href="#cb50-149" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype, fraction) <span class="sc">%&gt;%</span></span>
<span id="cb50-150"><a href="#cb50-150" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(</span>
<span id="cb50-151"><a href="#cb50-151" aria-hidden="true" tabindex="-1"></a>    <span class="at">ttest_adj_pval =</span> <span class="fu">round</span>(<span class="fu">pairwise.t.test</span>(count, diagnosis, <span class="at">p.adjust.method =</span> <span class="st">"BH"</span>)<span class="sc">$</span>p.value, <span class="dv">4</span>),</span>
<span id="cb50-152"><a href="#cb50-152" aria-hidden="true" tabindex="-1"></a>    <span class="at">wilcox_adj_pval =</span> <span class="fu">round</span>(<span class="fu">pairwise.wilcox.test</span>(count, diagnosis, <span class="at">p.adjust.method =</span> <span class="st">"BH"</span>)<span class="sc">$</span>p.value, <span class="dv">4</span>)</span>
<span id="cb50-153"><a href="#cb50-153" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-154"><a href="#cb50-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-155"><a href="#cb50-155" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(cell_counts)</span>
<span id="cb50-156"><a href="#cb50-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-157"><a href="#cb50-157" aria-hidden="true" tabindex="-1"></a>cell_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-158"><a href="#cb50-158" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(ttest_adj_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-159"><a href="#cb50-159" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb50-160"><a href="#cb50-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-161"><a href="#cb50-161" aria-hidden="true" tabindex="-1"></a>cell_counts <span class="sc">%&gt;%</span></span>
<span id="cb50-162"><a href="#cb50-162" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(wilcox_adj_pval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-163"><a href="#cb50-163" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span>
<span id="cb50-164"><a href="#cb50-164" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-165"><a href="#cb50-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-166"><a href="#cb50-166" aria-hidden="true" tabindex="-1"></a>Seems like Astrocytes and Oligodendrocytes are significant in the vascular fraction, and the inhibitory neurons in the parenchymal fraction for the t-tests</span>
<span id="cb50-167"><a href="#cb50-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-168"><a href="#cb50-168" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NOTE: need to add checks for normalcy</span>
<span id="cb50-169"><a href="#cb50-169" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NOTE: not working on render - need to debug</span>
<span id="cb50-170"><a href="#cb50-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-173"><a href="#cb50-173" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-174"><a href="#cb50-174" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb50-175"><a href="#cb50-175" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique cell types</span></span>
<span id="cb50-176"><a href="#cb50-176" aria-hidden="true" tabindex="-1"></a>unique_cell_types <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">Idents</span>(sce))</span>
<span id="cb50-177"><a href="#cb50-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-178"><a href="#cb50-178" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to perform the tests and adjust p-values</span></span>
<span id="cb50-179"><a href="#cb50-179" aria-hidden="true" tabindex="-1"></a>perform_tests <span class="ot">&lt;-</span> <span class="cf">function</span>(cell_type) {</span>
<span id="cb50-180"><a href="#cb50-180" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset cell counts for the current cell type in AD and control groups</span></span>
<span id="cb50-181"><a href="#cb50-181" aria-hidden="true" tabindex="-1"></a>  cell_counts_AD <span class="ot">&lt;-</span> sce<span class="sc">$</span>cell_counts[sce<span class="sc">$</span>group <span class="sc">==</span> <span class="st">"AD"</span> <span class="sc">&amp;</span> sce<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type]</span>
<span id="cb50-182"><a href="#cb50-182" aria-hidden="true" tabindex="-1"></a>  cell_counts_control <span class="ot">&lt;-</span> sce<span class="sc">$</span>cell_counts[sce<span class="sc">$</span>group <span class="sc">==</span> <span class="st">"control"</span> <span class="sc">&amp;</span> sce<span class="sc">$</span>cell_type <span class="sc">==</span> cell_type]</span>
<span id="cb50-183"><a href="#cb50-183" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-184"><a href="#cb50-184" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform t-test</span></span>
<span id="cb50-185"><a href="#cb50-185" aria-hidden="true" tabindex="-1"></a>  ttest_result <span class="ot">&lt;-</span> <span class="fu">t.test</span>(cell_counts_AD, cell_counts_control)</span>
<span id="cb50-186"><a href="#cb50-186" aria-hidden="true" tabindex="-1"></a>  ttest_pvalue <span class="ot">&lt;-</span> ttest_result<span class="sc">$</span>p.value</span>
<span id="cb50-187"><a href="#cb50-187" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-188"><a href="#cb50-188" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform Wilcoxon rank-sum test</span></span>
<span id="cb50-189"><a href="#cb50-189" aria-hidden="true" tabindex="-1"></a>  wilcox_result <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(cell_counts_AD, cell_counts_control)</span>
<span id="cb50-190"><a href="#cb50-190" aria-hidden="true" tabindex="-1"></a>  wilcox_pvalue <span class="ot">&lt;-</span> wilcox_result<span class="sc">$</span>p.value</span>
<span id="cb50-191"><a href="#cb50-191" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-192"><a href="#cb50-192" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return adjusted p-values</span></span>
<span id="cb50-193"><a href="#cb50-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">CellType =</span> cell_type, <span class="at">TTest_PValue =</span> ttest_pvalue, <span class="at">Wilcox_PValue =</span> wilcox_pvalue))</span>
<span id="cb50-194"><a href="#cb50-194" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-195"><a href="#cb50-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-196"><a href="#cb50-196" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each unique cell type using map</span></span>
<span id="cb50-197"><a href="#cb50-197" aria-hidden="true" tabindex="-1"></a>test_results <span class="ot">&lt;-</span> <span class="fu">map</span>(unique_cell_types, perform_tests)</span>
<span id="cb50-198"><a href="#cb50-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-199"><a href="#cb50-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract p-values</span></span>
<span id="cb50-200"><a href="#cb50-200" aria-hidden="true" tabindex="-1"></a>ttest_pvalues <span class="ot">&lt;-</span> <span class="fu">map_dbl</span>(test_results, <span class="st">"TTest_PValue"</span>)</span>
<span id="cb50-201"><a href="#cb50-201" aria-hidden="true" tabindex="-1"></a>wilcox_pvalues <span class="ot">&lt;-</span> <span class="fu">map_dbl</span>(test_results, <span class="st">"Wilcox_PValue"</span>)</span>
<span id="cb50-202"><a href="#cb50-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-203"><a href="#cb50-203" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust p-values using the Benjamini-Hochberg procedure</span></span>
<span id="cb50-204"><a href="#cb50-204" aria-hidden="true" tabindex="-1"></a>ttest_pvalues_adjusted <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(ttest_pvalues, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb50-205"><a href="#cb50-205" aria-hidden="true" tabindex="-1"></a>wilcox_pvalues_adjusted <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(wilcox_pvalues, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb50-206"><a href="#cb50-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-207"><a href="#cb50-207" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dataframe with the results</span></span>
<span id="cb50-208"><a href="#cb50-208" aria-hidden="true" tabindex="-1"></a>result_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">CellType =</span> unique_cell_types, <span class="at">TTest_PValue =</span> ttest_pvalues, <span class="at">TTest_PValue_Adjusted =</span> ttest_pvalues_adjusted, <span class="at">Wilcox_PValue =</span> wilcox_pvalues, <span class="at">Wilcox_PValue_Adjusted =</span> wilcox_pvalues_adjusted)</span>
<span id="cb50-209"><a href="#cb50-209" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(result_df)</span>
<span id="cb50-210"><a href="#cb50-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-211"><a href="#cb50-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-212"><a href="#cb50-212" aria-hidden="true" tabindex="-1"></a><span class="fu"># Average expression</span></span>
<span id="cb50-213"><a href="#cb50-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-216"><a href="#cb50-216" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-217"><a href="#cb50-217" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> <span class="fu">AverageExpression</span>(sce, <span class="at">group.by =</span> <span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"prep"</span>, <span class="st">"diagnosis"</span>))</span>
<span id="cb50-218"><a href="#cb50-218" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> avg_exp<span class="sc">$</span>RNA</span>
<span id="cb50-219"><a href="#cb50-219" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb50-220"><a href="#cb50-220" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> <span class="fu">data.table</span>(avg_exp, <span class="at">keep.rownames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-221"><a href="#cb50-221" aria-hidden="true" tabindex="-1"></a><span class="fu">setnames</span>(avg_exp, <span class="st">"rn"</span>, <span class="st">"gene"</span>)</span>
<span id="cb50-222"><a href="#cb50-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-223"><a href="#cb50-223" aria-hidden="true" tabindex="-1"></a><span class="co"># pivot longer such that celltypes are separated out</span></span>
<span id="cb50-224"><a href="#cb50-224" aria-hidden="true" tabindex="-1"></a>avg_exp <span class="ot">&lt;-</span> avg_exp <span class="sc">%&gt;%</span></span>
<span id="cb50-225"><a href="#cb50-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(</span>
<span id="cb50-226"><a href="#cb50-226" aria-hidden="true" tabindex="-1"></a>    <span class="at">cols =</span> <span class="sc">!</span>gene,</span>
<span id="cb50-227"><a href="#cb50-227" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_to =</span> <span class="fu">c</span>(<span class="st">"celltype"</span>, <span class="st">".value"</span>),</span>
<span id="cb50-228"><a href="#cb50-228" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_pattern =</span> <span class="st">"(.*)_(.*)"</span></span>
<span id="cb50-229"><a href="#cb50-229" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-230"><a href="#cb50-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-231"><a href="#cb50-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-232"><a href="#cb50-232" aria-hidden="true" tabindex="-1"></a><span class="fu"># CellChat</span></span>
<span id="cb50-233"><a href="#cb50-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-234"><a href="#cb50-234" aria-hidden="true" tabindex="-1"></a><span class="in">`CellChat`</span> is a package for looking at cell-cell interaction networks</span>
<span id="cb50-235"><a href="#cb50-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-236"><a href="#cb50-236" aria-hidden="true" tabindex="-1"></a>It requires a normalised matrix of gene expression data (genes as rows, cells as columns).</span>
<span id="cb50-237"><a href="#cb50-237" aria-hidden="true" tabindex="-1"></a>I assume from <span class="co">[</span><span class="ot">the vignette</span><span class="co">](https://htmlpreview.github.io/?https://github.com/sqjin/CellChat/blob/master/tutorial/CellChat-vignette.html)</span> that one should look at one group at a time, in this case each fraction per disease state, so 4 groups total.</span>
<span id="cb50-238"><a href="#cb50-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-239"><a href="#cb50-239" aria-hidden="true" tabindex="-1"></a>I'll start with the vascular fraction of the cases whilst experimenting</span>
<span id="cb50-240"><a href="#cb50-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-243"><a href="#cb50-243" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-244"><a href="#cb50-244" aria-hidden="true" tabindex="-1"></a><span class="co"># get the normalised expression data</span></span>
<span id="cb50-245"><a href="#cb50-245" aria-hidden="true" tabindex="-1"></a>expression_data <span class="ot">&lt;-</span> <span class="fu">GetAssayData</span>(sce, <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">slot =</span> <span class="st">"data"</span>)</span>
<span id="cb50-246"><a href="#cb50-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-247"><a href="#cb50-247" aria-hidden="true" tabindex="-1"></a><span class="co"># a dataframe with rownames containing cell mata data</span></span>
<span id="cb50-248"><a href="#cb50-248" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> sce<span class="sc">@</span>meta.data</span>
<span id="cb50-249"><a href="#cb50-249" aria-hidden="true" tabindex="-1"></a><span class="co"># add manual celltype labels</span></span>
<span id="cb50-250"><a href="#cb50-250" aria-hidden="true" tabindex="-1"></a>meta<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">Idents</span>(sce)</span>
<span id="cb50-251"><a href="#cb50-251" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the cell names from disease data</span></span>
<span id="cb50-252"><a href="#cb50-252" aria-hidden="true" tabindex="-1"></a>cell_use <span class="ot">&lt;-</span> <span class="fu">rownames</span>(meta)[meta<span class="sc">$</span>diagnosis <span class="sc">==</span> <span class="st">"Case"</span> <span class="sc">&amp;</span> meta<span class="sc">$</span>prep <span class="sc">==</span> <span class="st">"V"</span>]</span>
<span id="cb50-253"><a href="#cb50-253" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-254"><a href="#cb50-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-255"><a href="#cb50-255" aria-hidden="true" tabindex="-1"></a>Here I use <span class="in">`expand.grid`</span> to get all unique combinations of diagnosis and prep to get cell networks from.</span>
<span id="cb50-256"><a href="#cb50-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-259"><a href="#cb50-259" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-260"><a href="#cb50-260" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtain unique combinations of meta$diagnosis and meta$prep</span></span>
<span id="cb50-261"><a href="#cb50-261" aria-hidden="true" tabindex="-1"></a>combinations <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="fu">levels</span>(meta<span class="sc">$</span>diagnosis), <span class="fu">levels</span>(meta<span class="sc">$</span>prep))</span>
<span id="cb50-262"><a href="#cb50-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-263"><a href="#cb50-263" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the combinations to a list</span></span>
<span id="cb50-264"><a href="#cb50-264" aria-hidden="true" tabindex="-1"></a>combinations_list <span class="ot">&lt;-</span> <span class="fu">as.list</span>(combinations)</span>
<span id="cb50-265"><a href="#cb50-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-266"><a href="#cb50-266" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the list of unique combinations</span></span>
<span id="cb50-267"><a href="#cb50-267" aria-hidden="true" tabindex="-1"></a>combinations_list</span>
<span id="cb50-268"><a href="#cb50-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-269"><a href="#cb50-269" aria-hidden="true" tabindex="-1"></a><span class="co"># Return all subsets</span></span>
<span id="cb50-270"><a href="#cb50-270" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to generate subsets based on the combinations</span></span>
<span id="cb50-271"><a href="#cb50-271" aria-hidden="true" tabindex="-1"></a>get_subsets <span class="ot">&lt;-</span> <span class="cf">function</span>(diagnosis, prep) {</span>
<span id="cb50-272"><a href="#cb50-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(meta)[meta<span class="sc">$</span>diagnosis <span class="sc">==</span> diagnosis <span class="sc">&amp;</span> meta<span class="sc">$</span>prep <span class="sc">==</span> prep]</span>
<span id="cb50-273"><a href="#cb50-273" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-274"><a href="#cb50-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-275"><a href="#cb50-275" aria-hidden="true" tabindex="-1"></a><span class="co"># Use map2 to generate subsets for each combination</span></span>
<span id="cb50-276"><a href="#cb50-276" aria-hidden="true" tabindex="-1"></a>all_subsets <span class="ot">&lt;-</span> <span class="fu">map2</span>(combinations<span class="sc">$</span>Var1, combinations<span class="sc">$</span>Var2, <span class="sc">~</span> <span class="fu">get_subsets</span>(.x, .y)) <span class="sc">%&gt;%</span></span>
<span id="cb50-277"><a href="#cb50-277" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(<span class="fu">paste0</span>(combinations_list<span class="sc">$</span>Var1, <span class="st">"+"</span>, combinations_list<span class="sc">$</span>Var2))</span>
<span id="cb50-278"><a href="#cb50-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-279"><a href="#cb50-279" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the list of subsets</span></span>
<span id="cb50-280"><a href="#cb50-280" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(all_subsets, head)</span>
<span id="cb50-281"><a href="#cb50-281" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-282"><a href="#cb50-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-285"><a href="#cb50-285" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-286"><a href="#cb50-286" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare input data for CelChat analysis</span></span>
<span id="cb50-287"><a href="#cb50-287" aria-hidden="true" tabindex="-1"></a>data_input <span class="ot">&lt;-</span> <span class="fu">map</span>(all_subsets, <span class="sc">~</span> expression_data[, .x])</span>
<span id="cb50-288"><a href="#cb50-288" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">map</span>(all_subsets, <span class="sc">~</span> meta[.x, ])</span>
<span id="cb50-289"><a href="#cb50-289" aria-hidden="true" tabindex="-1"></a><span class="co"># check the cell labels</span></span>
<span id="cb50-290"><a href="#cb50-290" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(meta, <span class="sc">~</span> <span class="fu">unique</span>(.x<span class="sc">$</span>label))</span>
<span id="cb50-291"><a href="#cb50-291" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(meta, dim)</span>
<span id="cb50-292"><a href="#cb50-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-293"><a href="#cb50-293" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(expression_data, combinations, combinations_list)</span>
<span id="cb50-294"><a href="#cb50-294" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-295"><a href="#cb50-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-296"><a href="#cb50-296" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NOTE: need to remove this once switching to map call, otherwise code will break!</span>
<span id="cb50-297"><a href="#cb50-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-300"><a href="#cb50-300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-301"><a href="#cb50-301" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare input data for CelChat analysis</span></span>
<span id="cb50-302"><a href="#cb50-302" aria-hidden="true" tabindex="-1"></a><span class="co">#data_input &lt;- expression_data[, cell_use]</span></span>
<span id="cb50-303"><a href="#cb50-303" aria-hidden="true" tabindex="-1"></a><span class="co">#meta &lt;- meta[cell_use, ]</span></span>
<span id="cb50-304"><a href="#cb50-304" aria-hidden="true" tabindex="-1"></a><span class="co"># check the cell labels</span></span>
<span id="cb50-305"><a href="#cb50-305" aria-hidden="true" tabindex="-1"></a><span class="co">#unique(meta$label)</span></span>
<span id="cb50-306"><a href="#cb50-306" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-307"><a href="#cb50-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-308"><a href="#cb50-308" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create CellChat object</span></span>
<span id="cb50-309"><a href="#cb50-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-312"><a href="#cb50-312" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-313"><a href="#cb50-313" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span></span>
<span id="cb50-314"><a href="#cb50-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(data_input,</span>
<span id="cb50-315"><a href="#cb50-315" aria-hidden="true" tabindex="-1"></a>       meta,</span>
<span id="cb50-316"><a href="#cb50-316" aria-hidden="true" tabindex="-1"></a>       <span class="sc">~</span> <span class="fu">createCellChat</span>(</span>
<span id="cb50-317"><a href="#cb50-317" aria-hidden="true" tabindex="-1"></a>         <span class="at">object =</span> .x,</span>
<span id="cb50-318"><a href="#cb50-318" aria-hidden="true" tabindex="-1"></a>         <span class="at">meta =</span> .y,</span>
<span id="cb50-319"><a href="#cb50-319" aria-hidden="true" tabindex="-1"></a>         <span class="at">group.by =</span> <span class="st">"label"</span></span>
<span id="cb50-320"><a href="#cb50-320" aria-hidden="true" tabindex="-1"></a>       ))</span>
<span id="cb50-321"><a href="#cb50-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-322"><a href="#cb50-322" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(data_input, meta)</span>
<span id="cb50-323"><a href="#cb50-323" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-324"><a href="#cb50-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-325"><a href="#cb50-325" aria-hidden="true" tabindex="-1"></a><span class="fu">## Set ligand database</span></span>
<span id="cb50-326"><a href="#cb50-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-329"><a href="#cb50-329" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-330"><a href="#cb50-330" aria-hidden="true" tabindex="-1"></a>CellChatDB <span class="ot">&lt;-</span> CellChatDB.human <span class="co"># use CellChatDB.mouse if running on mouse data</span></span>
<span id="cb50-331"><a href="#cb50-331" aria-hidden="true" tabindex="-1"></a><span class="fu">showDatabaseCategory</span>(CellChatDB)</span>
<span id="cb50-332"><a href="#cb50-332" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-333"><a href="#cb50-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-334"><a href="#cb50-334" aria-hidden="true" tabindex="-1"></a>One can subset of the DB for analysis, but I'll use all for now</span>
<span id="cb50-335"><a href="#cb50-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-338"><a href="#cb50-338" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-339"><a href="#cb50-339" aria-hidden="true" tabindex="-1"></a><span class="co"># Show the structure of the database</span></span>
<span id="cb50-340"><a href="#cb50-340" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">glimpse</span>(CellChatDB<span class="sc">$</span>interaction)</span>
<span id="cb50-341"><a href="#cb50-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-342"><a href="#cb50-342" aria-hidden="true" tabindex="-1"></a><span class="co"># use a subset of CellChatDB for cell-cell communication analysis</span></span>
<span id="cb50-343"><a href="#cb50-343" aria-hidden="true" tabindex="-1"></a><span class="co">#CellChatDB_use &lt;- subsetDB(CellChatDB, search = "Secreted Signaling")</span></span>
<span id="cb50-344"><a href="#cb50-344" aria-hidden="true" tabindex="-1"></a><span class="co"># use all CellChatDB for cell-cell communication analysis</span></span>
<span id="cb50-345"><a href="#cb50-345" aria-hidden="true" tabindex="-1"></a>CellChatDB_use <span class="ot">&lt;-</span> CellChatDB <span class="co"># simply use the default CellChatDB</span></span>
<span id="cb50-346"><a href="#cb50-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-347"><a href="#cb50-347" aria-hidden="true" tabindex="-1"></a><span class="co"># set the used database in the object</span></span>
<span id="cb50-348"><a href="#cb50-348" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, <span class="sc">~</span> {</span>
<span id="cb50-349"><a href="#cb50-349" aria-hidden="true" tabindex="-1"></a>  .x<span class="sc">@</span>DB <span class="ot">&lt;-</span> CellChatDB_use</span>
<span id="cb50-350"><a href="#cb50-350" aria-hidden="true" tabindex="-1"></a>  .x</span>
<span id="cb50-351"><a href="#cb50-351" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb50-352"><a href="#cb50-352" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(CellChatDB_use, CellChatDB)</span>
<span id="cb50-353"><a href="#cb50-353" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-354"><a href="#cb50-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-355"><a href="#cb50-355" aria-hidden="true" tabindex="-1"></a><span class="fu">## Preprocessing the expression data for cell-cell communication analysis</span></span>
<span id="cb50-356"><a href="#cb50-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-359"><a href="#cb50-359" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-360"><a href="#cb50-360" aria-hidden="true" tabindex="-1"></a><span class="co"># subset the expression data of signaling genes for saving computation cost</span></span>
<span id="cb50-361"><a href="#cb50-361" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, subsetData) <span class="co"># This step is necessary even if using the whole database</span></span>
<span id="cb50-362"><a href="#cb50-362" aria-hidden="true" tabindex="-1"></a><span class="co">#future::plan("multiprocess", workers = 4) # do parallel</span></span>
<span id="cb50-363"><a href="#cb50-363" aria-hidden="true" tabindex="-1"></a>future<span class="sc">::</span><span class="fu">plan</span>(<span class="st">"multicore"</span>, <span class="at">workers =</span> <span class="dv">4</span>) <span class="co"># do parallel</span></span>
<span id="cb50-364"><a href="#cb50-364" aria-hidden="true" tabindex="-1"></a><span class="co"># Ran into error with this option being too small</span></span>
<span id="cb50-365"><a href="#cb50-365" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">5000000000</span>)</span>
<span id="cb50-366"><a href="#cb50-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-367"><a href="#cb50-367" aria-hidden="true" tabindex="-1"></a><span class="co"># This takes ages to run, so save the output</span></span>
<span id="cb50-368"><a href="#cb50-368" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the file path to save the resulting object</span></span>
<span id="cb50-369"><a href="#cb50-369" aria-hidden="true" tabindex="-1"></a>result_file <span class="ot">&lt;-</span></span>
<span id="cb50-370"><a href="#cb50-370" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb50-371"><a href="#cb50-371" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/11_cell_network_interactions/cellchat_preprocessed.rds"</span></span>
<span id="cb50-372"><a href="#cb50-372" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-373"><a href="#cb50-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-374"><a href="#cb50-374" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the file exists</span></span>
<span id="cb50-375"><a href="#cb50-375" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(result_file)) {</span>
<span id="cb50-376"><a href="#cb50-376" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Code for your time-consuming analysis</span></span>
<span id="cb50-377"><a href="#cb50-377" aria-hidden="true" tabindex="-1"></a>  cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, identifyOverExpressedGenes)</span>
<span id="cb50-378"><a href="#cb50-378" aria-hidden="true" tabindex="-1"></a>  cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, identifyOverExpressedInteractions)</span>
<span id="cb50-379"><a href="#cb50-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-380"><a href="#cb50-380" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-381"><a href="#cb50-381" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the resulting object</span></span>
<span id="cb50-382"><a href="#cb50-382" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(cellchat, result_file)</span>
<span id="cb50-383"><a href="#cb50-383" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb50-384"><a href="#cb50-384" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the existing resulting object</span></span>
<span id="cb50-385"><a href="#cb50-385" aria-hidden="true" tabindex="-1"></a>  cellchat <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(result_file)</span>
<span id="cb50-386"><a href="#cb50-386" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-387"><a href="#cb50-387" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(result_file)</span>
<span id="cb50-388"><a href="#cb50-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-389"><a href="#cb50-389" aria-hidden="true" tabindex="-1"></a><span class="co"># project gene expression data onto PPI (Optional: when running it, USER should</span></span>
<span id="cb50-390"><a href="#cb50-390" aria-hidden="true" tabindex="-1"></a><span class="co"># set `raw.use = FALSE` in the function `computeCommunProb()` in order to use</span></span>
<span id="cb50-391"><a href="#cb50-391" aria-hidden="true" tabindex="-1"></a><span class="co"># the projected data)</span></span>
<span id="cb50-392"><a href="#cb50-392" aria-hidden="true" tabindex="-1"></a><span class="co"># cellchat &lt;- projectData(cellchat, PPI.human)</span></span>
<span id="cb50-393"><a href="#cb50-393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-394"><a href="#cb50-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-395"><a href="#cb50-395" aria-hidden="true" tabindex="-1"></a><span class="fu">## Compute the communication probability and infer cellular communication network</span></span>
<span id="cb50-396"><a href="#cb50-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-397"><a href="#cb50-397" aria-hidden="true" tabindex="-1"></a>If well-known signaling pathways in the studied biological process are not predicted, USER can try <span class="in">`truncatedMean`</span> to change the method for calculating the average gene expression per cell group.</span>
<span id="cb50-398"><a href="#cb50-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-401"><a href="#cb50-401" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-402"><a href="#cb50-402" aria-hidden="true" tabindex="-1"></a><span class="co"># Again, this takes quite a while to run</span></span>
<span id="cb50-403"><a href="#cb50-403" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the file path to save the resulting object</span></span>
<span id="cb50-404"><a href="#cb50-404" aria-hidden="true" tabindex="-1"></a>result_file <span class="ot">&lt;-</span></span>
<span id="cb50-405"><a href="#cb50-405" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb50-406"><a href="#cb50-406" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/11_cell_network_interactions/cellchat_communication-prob.rds"</span></span>
<span id="cb50-407"><a href="#cb50-407" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-408"><a href="#cb50-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-409"><a href="#cb50-409" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the file exists</span></span>
<span id="cb50-410"><a href="#cb50-410" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(result_file)) {</span>
<span id="cb50-411"><a href="#cb50-411" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Code for your time-consuming analysis</span></span>
<span id="cb50-412"><a href="#cb50-412" aria-hidden="true" tabindex="-1"></a>  cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, computeCommunProb)</span>
<span id="cb50-413"><a href="#cb50-413" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter out the cell-cell communication if there are only few number of cells in certain cell groups</span></span>
<span id="cb50-414"><a href="#cb50-414" aria-hidden="true" tabindex="-1"></a>  cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, <span class="sc">~</span> <span class="fu">filterCommunication</span>(.x, <span class="at">min.cells =</span> <span class="dv">10</span>))</span>
<span id="cb50-415"><a href="#cb50-415" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-416"><a href="#cb50-416" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the resulting object</span></span>
<span id="cb50-417"><a href="#cb50-417" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(cellchat, result_file)</span>
<span id="cb50-418"><a href="#cb50-418" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb50-419"><a href="#cb50-419" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Load the existing resulting object</span></span>
<span id="cb50-420"><a href="#cb50-420" aria-hidden="true" tabindex="-1"></a>  cellchat <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(result_file)</span>
<span id="cb50-421"><a href="#cb50-421" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-422"><a href="#cb50-422" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(result_file)</span>
<span id="cb50-423"><a href="#cb50-423" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-424"><a href="#cb50-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-425"><a href="#cb50-425" aria-hidden="true" tabindex="-1"></a>We can get the cell interaction networks as a dataframe</span>
<span id="cb50-426"><a href="#cb50-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-429"><a href="#cb50-429" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-430"><a href="#cb50-430" aria-hidden="true" tabindex="-1"></a>df_net <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, subsetCommunication)</span>
<span id="cb50-431"><a href="#cb50-431" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df_net<span class="sc">$</span><span class="st">`</span><span class="at">Case+P</span><span class="st">`</span>)</span>
<span id="cb50-432"><a href="#cb50-432" aria-hidden="true" tabindex="-1"></a><span class="co"># check how many aren't significant</span></span>
<span id="cb50-433"><a href="#cb50-433" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(df_net<span class="sc">$</span><span class="st">`</span><span class="at">Case+P</span><span class="st">`</span><span class="sc">$</span>pval <span class="sc">&gt;</span> <span class="fl">0.05</span>)</span>
<span id="cb50-434"><a href="#cb50-434" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-435"><a href="#cb50-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-436"><a href="#cb50-436" aria-hidden="true" tabindex="-1"></a><span class="fu">## Infer the cell-cell communication at a signaling pathway level</span></span>
<span id="cb50-437"><a href="#cb50-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-440"><a href="#cb50-440" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-441"><a href="#cb50-441" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, computeCommunProbPathway)</span>
<span id="cb50-442"><a href="#cb50-442" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-443"><a href="#cb50-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-444"><a href="#cb50-444" aria-hidden="true" tabindex="-1"></a><span class="fu">## Calculate the aggregated cell-cell communication network</span></span>
<span id="cb50-445"><a href="#cb50-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-448"><a href="#cb50-448" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-449"><a href="#cb50-449" aria-hidden="true" tabindex="-1"></a>cellchat <span class="ot">&lt;-</span> <span class="fu">map</span>(cellchat, aggregateNet)</span>
<span id="cb50-450"><a href="#cb50-450" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-451"><a href="#cb50-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-454"><a href="#cb50-454" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-455"><a href="#cb50-455" aria-hidden="true" tabindex="-1"></a>initial_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(cellchat, comparison) {</span>
<span id="cb50-456"><a href="#cb50-456" aria-hidden="true" tabindex="-1"></a>  groupSize <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(cellchat<span class="sc">@</span>idents))</span>
<span id="cb50-457"><a href="#cb50-457" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">xpd =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-458"><a href="#cb50-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">netVisual_circle</span>(</span>
<span id="cb50-459"><a href="#cb50-459" aria-hidden="true" tabindex="-1"></a>    cellchat<span class="sc">@</span>net<span class="sc">$</span>count,</span>
<span id="cb50-460"><a href="#cb50-460" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.weight =</span> groupSize,</span>
<span id="cb50-461"><a href="#cb50-461" aria-hidden="true" tabindex="-1"></a>    <span class="at">weight.scale =</span> T,</span>
<span id="cb50-462"><a href="#cb50-462" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.edge =</span> F,</span>
<span id="cb50-463"><a href="#cb50-463" aria-hidden="true" tabindex="-1"></a>    <span class="at">title.name =</span> <span class="st">"Number of interactions"</span></span>
<span id="cb50-464"><a href="#cb50-464" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-465"><a href="#cb50-465" aria-hidden="true" tabindex="-1"></a>  <span class="fu">netVisual_circle</span>(</span>
<span id="cb50-466"><a href="#cb50-466" aria-hidden="true" tabindex="-1"></a>    cellchat<span class="sc">@</span>net<span class="sc">$</span>weight,</span>
<span id="cb50-467"><a href="#cb50-467" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.weight =</span> groupSize,</span>
<span id="cb50-468"><a href="#cb50-468" aria-hidden="true" tabindex="-1"></a>    <span class="at">weight.scale =</span> T,</span>
<span id="cb50-469"><a href="#cb50-469" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.edge =</span> F,</span>
<span id="cb50-470"><a href="#cb50-470" aria-hidden="true" tabindex="-1"></a>    <span class="at">title.name =</span> <span class="st">"Interaction weights/strength"</span></span>
<span id="cb50-471"><a href="#cb50-471" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-472"><a href="#cb50-472" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mtext</span>(comparison, <span class="at">outer =</span> <span class="cn">TRUE</span>, <span class="at">line =</span> <span class="sc">-</span><span class="fl">1.5</span>, <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb50-473"><a href="#cb50-473" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-474"><a href="#cb50-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-475"><a href="#cb50-475" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(cellchat, <span class="fu">names</span>(cellchat), initial_plots)</span>
<span id="cb50-476"><a href="#cb50-476" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-477"><a href="#cb50-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-480"><a href="#cb50-480" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-481"><a href="#cb50-481" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb50-482"><a href="#cb50-482" aria-hidden="true" tabindex="-1"></a>celltype_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(cellchat, comparison) {</span>
<span id="cb50-483"><a href="#cb50-483" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-484"><a href="#cb50-484" aria-hidden="true" tabindex="-1"></a>  groupSize <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(cellchat<span class="sc">@</span>idents))</span>
<span id="cb50-485"><a href="#cb50-485" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-486"><a href="#cb50-486" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">&lt;-</span> cellchat<span class="sc">@</span>net<span class="sc">$</span>weight</span>
<span id="cb50-487"><a href="#cb50-487" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>), <span class="at">xpd =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-488"><a href="#cb50-488" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-489"><a href="#cb50-489" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(mat), <span class="sc">~</span> {</span>
<span id="cb50-490"><a href="#cb50-490" aria-hidden="true" tabindex="-1"></a>    mat2 <span class="ot">&lt;-</span></span>
<span id="cb50-491"><a href="#cb50-491" aria-hidden="true" tabindex="-1"></a>      <span class="fu">matrix</span>(</span>
<span id="cb50-492"><a href="#cb50-492" aria-hidden="true" tabindex="-1"></a>        <span class="dv">0</span>,</span>
<span id="cb50-493"><a href="#cb50-493" aria-hidden="true" tabindex="-1"></a>        <span class="at">nrow =</span> <span class="fu">nrow</span>(mat),</span>
<span id="cb50-494"><a href="#cb50-494" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">ncol</span>(mat),</span>
<span id="cb50-495"><a href="#cb50-495" aria-hidden="true" tabindex="-1"></a>        <span class="at">dimnames =</span> <span class="fu">dimnames</span>(mat)</span>
<span id="cb50-496"><a href="#cb50-496" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb50-497"><a href="#cb50-497" aria-hidden="true" tabindex="-1"></a>    mat2[.x,] <span class="ot">&lt;-</span> mat[.x,]</span>
<span id="cb50-498"><a href="#cb50-498" aria-hidden="true" tabindex="-1"></a>    <span class="fu">netVisual_circle</span>(</span>
<span id="cb50-499"><a href="#cb50-499" aria-hidden="true" tabindex="-1"></a>      mat2,</span>
<span id="cb50-500"><a href="#cb50-500" aria-hidden="true" tabindex="-1"></a>      <span class="at">vertex.weight =</span> groupSize,</span>
<span id="cb50-501"><a href="#cb50-501" aria-hidden="true" tabindex="-1"></a>      <span class="at">weight.scale =</span> <span class="cn">TRUE</span>,</span>
<span id="cb50-502"><a href="#cb50-502" aria-hidden="true" tabindex="-1"></a>      <span class="at">edge.weight.max =</span> <span class="fu">max</span>(mat),</span>
<span id="cb50-503"><a href="#cb50-503" aria-hidden="true" tabindex="-1"></a>      <span class="at">title.name =</span> <span class="fu">rownames</span>(mat)[.x]</span>
<span id="cb50-504"><a href="#cb50-504" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb50-505"><a href="#cb50-505" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb50-506"><a href="#cb50-506" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mtext</span>(comparison,</span>
<span id="cb50-507"><a href="#cb50-507" aria-hidden="true" tabindex="-1"></a>        <span class="at">outer =</span> <span class="cn">TRUE</span>,</span>
<span id="cb50-508"><a href="#cb50-508" aria-hidden="true" tabindex="-1"></a>        <span class="at">line =</span> <span class="sc">-</span><span class="fl">1.5</span>,</span>
<span id="cb50-509"><a href="#cb50-509" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb50-510"><a href="#cb50-510" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-511"><a href="#cb50-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-512"><a href="#cb50-512" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(cellchat, <span class="fu">names</span>(cellchat), celltype_plots)</span>
<span id="cb50-513"><a href="#cb50-513" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-514"><a href="#cb50-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-517"><a href="#cb50-517" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-518"><a href="#cb50-518" aria-hidden="true" tabindex="-1"></a>celltype_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(cellchat, comparison) {</span>
<span id="cb50-519"><a href="#cb50-519" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-520"><a href="#cb50-520" aria-hidden="true" tabindex="-1"></a>  groupSize <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">table</span>(cellchat<span class="sc">@</span>idents))</span>
<span id="cb50-521"><a href="#cb50-521" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-522"><a href="#cb50-522" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">&lt;-</span> cellchat<span class="sc">@</span>net<span class="sc">$</span>weight</span>
<span id="cb50-523"><a href="#cb50-523" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-524"><a href="#cb50-524" aria-hidden="true" tabindex="-1"></a>  plots <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(mat), <span class="sc">~</span> {</span>
<span id="cb50-525"><a href="#cb50-525" aria-hidden="true" tabindex="-1"></a>    mat2 <span class="ot">&lt;-</span></span>
<span id="cb50-526"><a href="#cb50-526" aria-hidden="true" tabindex="-1"></a>      <span class="fu">matrix</span>(</span>
<span id="cb50-527"><a href="#cb50-527" aria-hidden="true" tabindex="-1"></a>        <span class="dv">0</span>,</span>
<span id="cb50-528"><a href="#cb50-528" aria-hidden="true" tabindex="-1"></a>        <span class="at">nrow =</span> <span class="fu">nrow</span>(mat),</span>
<span id="cb50-529"><a href="#cb50-529" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol =</span> <span class="fu">ncol</span>(mat),</span>
<span id="cb50-530"><a href="#cb50-530" aria-hidden="true" tabindex="-1"></a>        <span class="at">dimnames =</span> <span class="fu">dimnames</span>(mat)</span>
<span id="cb50-531"><a href="#cb50-531" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb50-532"><a href="#cb50-532" aria-hidden="true" tabindex="-1"></a>    mat2[.x,] <span class="ot">&lt;-</span> mat[.x,]</span>
<span id="cb50-533"><a href="#cb50-533" aria-hidden="true" tabindex="-1"></a>    plot <span class="ot">&lt;-</span> <span class="fu">netVisual_circle</span>(</span>
<span id="cb50-534"><a href="#cb50-534" aria-hidden="true" tabindex="-1"></a>      mat2,</span>
<span id="cb50-535"><a href="#cb50-535" aria-hidden="true" tabindex="-1"></a>      <span class="at">vertex.weight =</span> groupSize,</span>
<span id="cb50-536"><a href="#cb50-536" aria-hidden="true" tabindex="-1"></a>      <span class="at">weight.scale =</span> <span class="cn">TRUE</span>,</span>
<span id="cb50-537"><a href="#cb50-537" aria-hidden="true" tabindex="-1"></a>      <span class="at">edge.weight.max =</span> <span class="fu">max</span>(mat),</span>
<span id="cb50-538"><a href="#cb50-538" aria-hidden="true" tabindex="-1"></a>      <span class="at">title.name =</span> <span class="fu">rownames</span>(mat)[.x]</span>
<span id="cb50-539"><a href="#cb50-539" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb50-540"><a href="#cb50-540" aria-hidden="true" tabindex="-1"></a>    plot</span>
<span id="cb50-541"><a href="#cb50-541" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb50-542"><a href="#cb50-542" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(plots)</span>
<span id="cb50-543"><a href="#cb50-543" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-544"><a href="#cb50-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-545"><a href="#cb50-545" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">map2</span>(cellchat, <span class="fu">names</span>(cellchat), celltype_plots)</span>
<span id="cb50-546"><a href="#cb50-546" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-547"><a href="#cb50-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-550"><a href="#cb50-550" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-551"><a href="#cb50-551" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb50-552"><a href="#cb50-552" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>), <span class="at">xpd =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-553"><a href="#cb50-553" aria-hidden="true" tabindex="-1"></a>plots<span class="sc">$</span><span class="st">`</span><span class="at">Case+P</span><span class="st">`</span></span>
<span id="cb50-554"><a href="#cb50-554" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="fu">names</span>(plots)[<span class="dv">1</span>],</span>
<span id="cb50-555"><a href="#cb50-555" aria-hidden="true" tabindex="-1"></a>        <span class="at">outer =</span> <span class="cn">TRUE</span>,</span>
<span id="cb50-556"><a href="#cb50-556" aria-hidden="true" tabindex="-1"></a>        <span class="at">line =</span> <span class="sc">-</span><span class="fl">1.5</span>,</span>
<span id="cb50-557"><a href="#cb50-557" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb50-558"><a href="#cb50-558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>