<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.54">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>MAGMA - risk results analysis</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  <link href="../../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="../../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>  
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> MAGMA - risk results analysis</h6>

            <a href="../../04_data_analysis/007_magma/20_magma_risk_results_analysis.qmd" class="btn btn-primary quarto-download-embed" download="20_magma_risk_results_analysis.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">MAGMA - risk results analysis</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data">Load data</a></li>
  <li><a href="#heatmaps" id="toc-heatmaps" class="nav-link" data-scroll-target="#heatmaps">Heatmaps</a>
  <ul class="collapse">
  <li><a href="#subset-to-risk-of-interest" id="toc-subset-to-risk-of-interest" class="nav-link" data-scroll-target="#subset-to-risk-of-interest">Subset to risk of interest</a></li>
  <li><a href="#vertical-heatmap-for-paper" id="toc-vertical-heatmap-for-paper" class="nav-link" data-scroll-target="#vertical-heatmap-for-paper">Vertical heatmap for paper</a></li>
  </ul></li>
  <li><a href="#ppi-modules" id="toc-ppi-modules" class="nav-link" data-scroll-target="#ppi-modules">PPI modules</a>
  <ul class="collapse">
  <li><a href="#magma-gene-pvals" id="toc-magma-gene-pvals" class="nav-link" data-scroll-target="#magma-gene-pvals">MAGMA gene pvals</a></li>
  </ul></li>
  <li><a href="#magma-genes" id="toc-magma-genes" class="nav-link" data-scroll-target="#magma-genes">MAGMA genes</a></li>
  <li><a href="#conditional-anlaysis" id="toc-conditional-anlaysis" class="nav-link" data-scroll-target="#conditional-anlaysis">Conditional anlaysis</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/007_magma/20_magma_risk_results_analysis.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad</code></pre>
</div>
</div></div>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, smiths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'kableExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    group_rows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'janitor'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    chisq.test, fisher.test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    melt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rngtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  To enable shared memory capabilities, try: install.extras('
NMF
')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'future'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    run</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Some features are not enabled in this build of Arrow. Run `arrow_info()` for more information.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'arrow'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    timestamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'viridis'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    viridis_pal</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 methods overwritten by 'proxy':
  method               from    
  print.registry_field registry
  print.registry_entry registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    nrun</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:webshot2':

    resize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built under R 4.4.0 but the current version is
4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI
for R may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'DESeq2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    fpkm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: limma</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DESeq2':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    plotMDS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'edgeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SingleCellExperiment':

    cpm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocParallel'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    register</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation("pathview") within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationFilter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationFilter'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:arrow':

    field</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:future':

    value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ensembldb'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggrepel</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'PCAtools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scran':

    parallelPCA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    biplot, screeplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Seurat':

    components</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    normalize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:future':

    %-&gt;%, %&lt;-%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:NMF':

    algorithm, compare</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'biomaRt'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scRNAseq':

    listDatasets, searchDatasets</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>clusterProfiler v4.12.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use clusterProfiler in published research, please cite:
T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'clusterProfiler'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>DOSE v3.30.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'enrichplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    color_palette</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pals':

    alphabet</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocIO</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rtracklayer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocIO':

    FileForFormat</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    blocks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationHub':

    hubUrl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh37'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'qusage'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    read.gmt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:SNPlocs.Hsapiens.dbSNP155.GRCh37':

    .loadAlleles, .loadLoc, COMPATIBLE_BSGENOMES</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'GenomicFiles'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
simona version 1.2.0
Bioconductor page: http://bioconductor.org/packages/simona/
Github page: https://github.com/jokergoo/simona
Documentation: https://jokergoo.github.io/simona/

If you use it in published research, please cite:
Gu, Z. simona: a Comprehensive R package for Semantic Similarity 
  Analysis on Bio-Ontologies. bioRxiv 2023.

This message can be suppressed by:
  suppressPackageStartupMessages(library(simona))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'ggnetwork':
  method         from  
  fortify.igraph ggtree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'CellChat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scater':

    runPCA, runUMAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
</div></div>
</section>
<section id="load-data" class="level1">
<h1>Load data</h1>
<p>Some of the files have different numbers of headers, so Iâ€™ll write a function to figure out how many lines to skip.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="co"># List files</span></span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/results"</span>), </span>
<span id="cb198-4"><a href="#cb198-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pattern =</span> <span class="st">"*.gsa.out"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb198-5"><a href="#cb198-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-6"><a href="#cb198-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Read data</span></span>
<span id="cb198-7"><a href="#cb198-7" aria-hidden="true" tabindex="-1"></a>parse_magma_set_data <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb198-8"><a href="#cb198-8" aria-hidden="true" tabindex="-1"></a>  file_lines <span class="ot">&lt;-</span> <span class="fu">readLines</span>(file)</span>
<span id="cb198-9"><a href="#cb198-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the index of the line judt before the data starts</span></span>
<span id="cb198-10"><a href="#cb198-10" aria-hidden="true" tabindex="-1"></a>  data_starts_index <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"^# CONDITIONED_INTERNAL"</span>, file_lines)</span>
<span id="cb198-11"><a href="#cb198-11" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read_table</span>(file, <span class="at">skip =</span> data_starts_index, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb198-12"><a href="#cb198-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(data) </span>
<span id="cb198-13"><a href="#cb198-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb198-14"><a href="#cb198-14" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">map</span>(files, parse_magma_set_data)</span>
<span id="cb198-15"><a href="#cb198-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-16"><a href="#cb198-16" aria-hidden="true" tabindex="-1"></a>files_sub <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(<span class="fu">basename</span>(files), <span class="st">"_scz."</span>, <span class="st">"_scz_"</span>) <span class="sc">|&gt;</span></span>
<span id="cb198-17"><a href="#cb198-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"_gene_analysis"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb198-18"><a href="#cb198-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use sub() to remove all characters after the first dot</span></span>
<span id="cb198-19"><a href="#cb198-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*$"</span>, <span class="st">""</span>, .) <span class="sc">|&gt;</span></span>
<span id="cb198-20"><a href="#cb198-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"magma_celltypes_"</span>, <span class="st">""</span>)</span>
<span id="cb198-21"><a href="#cb198-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb198-22"><a href="#cb198-22" aria-hidden="true" tabindex="-1"></a>process_magma_data <span class="ot">&lt;-</span></span>
<span id="cb198-23"><a href="#cb198-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(df,</span>
<span id="cb198-24"><a href="#cb198-24" aria-hidden="true" tabindex="-1"></a>           filename,</span>
<span id="cb198-25"><a href="#cb198-25" aria-hidden="true" tabindex="-1"></a>           <span class="at">celltypes_to_exclude =</span> <span class="fu">c</span>(<span class="st">"ambiguous"</span>, <span class="st">"low-feature-cells"</span>)) {</span>
<span id="cb198-26"><a href="#cb198-26" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>VARIABLE <span class="sc">%in%</span> celltypes_to_exclude) <span class="sc">|&gt;</span></span>
<span id="cb198-27"><a href="#cb198-27" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Add groups and adjusted pvals to data</span></span>
<span id="cb198-28"><a href="#cb198-28" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> filename, <span class="at">p_adj =</span> <span class="fu">p.adjust</span>(P, <span class="at">method =</span> <span class="st">"bonferroni"</span>))</span>
<span id="cb198-29"><a href="#cb198-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb198-30"><a href="#cb198-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb198-31"><a href="#cb198-31" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">map2</span>(data, files_sub, <span class="sc">~</span> <span class="fu">process_magma_data</span>(.x, .y)) <span class="sc">|&gt;</span></span>
<span id="cb198-32"><a href="#cb198-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb198-33"><a href="#cb198-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb198-34"><a href="#cb198-34" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>FULL_NAME <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(df<span class="sc">$</span>FULL_NAME), df<span class="sc">$</span>VARIABLE, df<span class="sc">$</span>FULL_NAME)</span>
<span id="cb198-35"><a href="#cb198-35" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>VARIABLE <span class="ot">&lt;-</span> df<span class="sc">$</span>FULL_NAME</span>
<span id="cb198-36"><a href="#cb198-36" aria-hidden="true" tabindex="-1"></a>magma_heatmap_data <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/magma_celltype_heatmap_data.qs"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a>df <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(VARIABLE <span class="sc">==</span> <span class="st">"M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_module1"</span>) <span class="sc">|&gt;</span> <span class="fu">View</span>()</span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a>celltype <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"([^_]*)_.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, df<span class="sc">$</span>VARIABLE)</span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> df[<span class="fu">grepl</span>(celltype, df<span class="sc">$</span>group),]</span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">apply</span>(df, <span class="dv">1</span>, <span class="cf">function</span>(row) {</span>
<span id="cb199-8"><a href="#cb199-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">any</span>(<span class="fu">sapply</span>(celltype, <span class="cf">function</span>(str) <span class="fu">grepl</span>(str, row[<span class="st">"group"</span>])))</span>
<span id="cb199-9"><a href="#cb199-9" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Remove PPI data</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get ppi data</span></span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a>ppi_modules <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(df, <span class="fu">grepl</span>(<span class="st">"ppi_"</span>, group))</span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(df, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"ppi_"</span>, group))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the regular expression pattern to match everything between the first and third underscore</span></span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The pattern is:</span></span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a><span class="co"># - [^_]+: Match one or more characters that are not an underscore</span></span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a><span class="co"># - _: Match the underscore</span></span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a><span class="co"># - ([^_]+_[^_]+): Capture the content after the first underscore and before the third underscore</span></span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a>pattern <span class="ot">&lt;-</span> <span class="st">"^[^_]+_([^_]+_[^_]+)_.*$"</span></span>
<span id="cb201-7"><a href="#cb201-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the regular expression pattern to match everything after the third underscore</span></span>
<span id="cb201-8"><a href="#cb201-8" aria-hidden="true" tabindex="-1"></a><span class="co"># The pattern is:</span></span>
<span id="cb201-9"><a href="#cb201-9" aria-hidden="true" tabindex="-1"></a><span class="co"># - (?:[^_]*_){3}: Non-capturing group to skip the first three underscores and their preceding text</span></span>
<span id="cb201-10"><a href="#cb201-10" aria-hidden="true" tabindex="-1"></a><span class="co"># - (.*)$: Capture everything after the third underscore until the end of the string</span></span>
<span id="cb201-11"><a href="#cb201-11" aria-hidden="true" tabindex="-1"></a>pattern_2 <span class="ot">&lt;-</span> <span class="st">"^(?:[^_]*_){4}(.*)$"</span></span>
<span id="cb201-12"><a href="#cb201-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Regular expression pattern to match the percent of genes used</span></span>
<span id="cb201-13"><a href="#cb201-13" aria-hidden="true" tabindex="-1"></a>pattern_3 <span class="ot">&lt;-</span> <span class="st">"top.*?percent"</span></span>
<span id="cb201-14"><a href="#cb201-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-15"><a href="#cb201-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the logfc part of the string</span></span>
<span id="cb201-16"><a href="#cb201-16" aria-hidden="true" tabindex="-1"></a>string <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(df<span class="sc">$</span>group, <span class="st">"logfc_threshold_"</span>, <span class="st">""</span>) <span class="sc">|&gt;</span></span>
<span id="cb201-17"><a href="#cb201-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"EUROPEUKBB"</span>, <span class="st">"Bellenguez_2022"</span>) <span class="sc">|&gt;</span></span>
<span id="cb201-18"><a href="#cb201-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"AD_"</span>, <span class="st">"Kunkle_2019_"</span>)</span>
<span id="cb201-19"><a href="#cb201-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Use sub to replace the entire string with the captured group</span></span>
<span id="cb201-20"><a href="#cb201-20" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>input_gene_list <span class="ot">&lt;-</span> <span class="fu">sub</span>(pattern, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb201-21"><a href="#cb201-21" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>gwas_background <span class="ot">&lt;-</span> <span class="fu">sub</span>(pattern_2, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb201-22"><a href="#cb201-22" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>input_filter <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"logfc_threshold_"</span>, df<span class="sc">$</span>group), <span class="st">"logfc_filter"</span>, <span class="st">"no_filter"</span>)</span>
<span id="cb201-23"><a href="#cb201-23" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>celltype_level <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"level1"</span>, df<span class="sc">$</span>group), <span class="st">"level1"</span>, <span class="st">"level2"</span>)</span>
<span id="cb201-24"><a href="#cb201-24" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>percent <span class="ot">&lt;-</span> <span class="fu">regmatches</span>(string, <span class="fu">regexpr</span>(pattern_3, string))</span>
<span id="cb201-25"><a href="#cb201-25" aria-hidden="true" tabindex="-1"></a><span class="co">#df$gwas_disease &lt;- ifelse(grepl("scz_", df$gwas_background), "SCZ", "AD")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Remove runs with no APOE and no window</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"noAPOE"</span>, group)) <span class="sc">|&gt;</span></span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"nowindow"</span>, group)) <span class="sc">|&gt;</span></span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># some duplicates got in - should probably check where from, likely old runs </span></span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># with different names</span></span>
<span id="cb202-6"><a href="#cb202-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"celltypes_"</span>, group))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a><span class="co"># So adjust pval per input list instead</span></span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb203-5"><a href="#cb203-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(input_gene_list, celltype_level, input_filter) <span class="sc">|&gt;</span></span>
<span id="cb203-6"><a href="#cb203-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">p_adj =</span> <span class="fu">p.adjust</span>(P, <span class="at">method =</span> <span class="st">"bonferroni"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb203-7"><a href="#cb203-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb203-8"><a href="#cb203-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Doing this seems to lose pretty much all significance...</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Get a table of significant associations</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a>df <span class="sc">|&gt;</span></span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p_adj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(TYPE, FULL_NAME, group)) <span class="sc">|&gt;</span></span>
<span id="cb204-4"><a href="#cb204-4" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-a21556742c4b205bc118" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a21556742c4b205bc118">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495"],["Microglia","T-cell","Trans-Endo-to-mural","Microglia","T-cell","Microglia","Endothelial","Fibroblast","Neuron_ex","Neuron_in","OPC","Oligo","Pericyte","SMC","T-cell","Endothelial","Neuron_ex","Neuron_in","OPC","Oligo","Pericyte","SMC","T-cell","Trans-Endo-to-mural","Oligo","Microglia","Neuron_ex","Neuron_in","Oligo","Pericyte","SMC","Neuron_ex","Neuron_in","Oligo","SMC","Astro","Endothelial","Fibroblast","Microglia","Pericyte","SMC","T-cell","Trans-Endo-to-mural","Fibroblast","Microglia","Pericyte","T-cell","Astro","Oligo","T-cell","Astro","Endothelial","Fibroblast","Microglia","Neuron_ex","Neuron_in","OPC","Oligo","Pericyte","SMC","T-cell","Trans-Endo-to-mural","Astro","Endothelial","Fibroblast","Microglia","Neuron_ex","Neuron_in","OPC","Oligo","Pericyte","SMC","T-cell","Trans-Endo-to-mural","Astro","Microglia","Microglia","Oligo","Astro","Microglia","Neuron_ex","Neuron_in","OPC","Oligo","Astro","Microglia","Neuron_ex","Neuron_in","OPC","Oligo","OPC","Oligo","Microglia","Neuron_ex","Oligo","Astro","Neuron_ex","Oligo","Oligo","Astro","Microglia","Astro","Microglia","Neuron_in","Oligo","Astro","Microglia","Neuron_ex","Neuron_in","OPC","Oligo","Astro","Microglia","Neuron_ex","Neuron_in","OPC","Oligo","Oligo","SMC","T-cell","SMC","T-cell","T-cell","Endothelial","Fibroblast","Microglia","Pericyte","SMC","Fibroblast","Microglia","Endothelial","Pericyte","SMC","T-cell","Fibroblast","SMC","T-cell","Endothelial","Fibroblast","Pericyte","SMC","T-cell","Endothelial","Fibroblast","Microglia","Pericyte","SMC","T-cell","Endothelial","Microglia","Pericyte","Microglia-activated","Microglia-activated","Perivascular-FB-KAZN2","Astrocyte-activated","Arterial","Arteriolar-SMC","Capillary","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","M-Pericyte","Meningeal-FB","OPC-A","OPC-B","Oligo-A","Oligo-B","Pericyte","Pericyte-2","Perivascular-FB-KAZN2","T-Pericyte","T-cell-mixed","T-cell-parenchymal-1","T-cell-vascular","Vascular-SMC","Vascular-SMC-LINC00486","Arterial","Arteriolar-SMC","Capillary","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","M-Pericyte","Meningeal-FB","OPC-A","OPC-B","Oligo-A","Oligo-B","Pericyte","Perivascular-FB-KAZN2","T-Pericyte","Vascular-SMC","Vascular-SMC-LINC00486","Perivascular","Microglia-quiescent","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","Oligo-A","Oligo-B","Pericyte","T-Pericyte","Vascular-SMC","Vascular-SMC-LINC00486","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","Vascular-SMC","Microglia-activated","Microglia-activated","Pericyte-2","Perivascular-FB-KAZN2","Astrocyte-activated","Arterial","Arteriolar-SMC","Capillary","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","M-Pericyte","Meningeal-FB","Microglia-vascular","OPC-A","OPC-B","Oligo-A","Oligo-B","Pericyte","Pericyte-2","Perivascular-FB-KAZN2","T-Pericyte","T-cell-mixed","T-cell-parenchymal-1","T-cell-vascular","Vascular-SMC","Vascular-SMC-LINC00486","Arterial","Arteriolar-SMC","Capillary","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","M-Pericyte","Meningeal-FB","OPC-A","OPC-B","Oligo-A","Oligo-B","Pericyte","Perivascular-FB-KAZN2","T-Pericyte","Vascular-SMC","Vascular-SMC-LINC00486","Microglia-activated","Microglia-activated","In-Neuron-DPP10-MEF2C","Oligo-A","Astrocyte-activated","Astrocyte-quiescent","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","Microglia-quiescent","Microglia-vascular","OPC-A","OPC-B","Oligo-A","Oligo-B","Perivascular","T-cell-mixed","T-cell-parenchymal-1","Astrocyte-quiescent","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","Microglia-quiescent","Microglia-vascular","OPC-A","OPC-B","Oligo-A","Oligo-B","Perivascular","T-cell-mixed","T-cell-parenchymal-1","Oligo-A","Microglia-activated","Perivascular","T-cell-mixed","T-cell-parenchymal-1","Astrocyte-quiescent","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","In-Neuron-ADARB2-IL1RAPL2-CSCL14","Oligo-A","Oligo-B","Astrocyte-quiescent","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-PRKG1-TBC1D4","Oligo-A","Oligo-B","Oligo-A","Microglia-activated","Microglia-activated","Microglia-quiescent","Ex-Neuron-ADARB2-ERBB4","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","Oligo-A","Astrocyte-activated","Astrocyte-quiescent","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","Microglia-quiescent","Microglia-vascular","OPC-A","OPC-B","Oligo-A","Oligo-B","Perivascular","T-cell-mixed","T-cell-parenchymal-1","Astrocyte-activated","Astrocyte-quiescent","Ex-Neuron-ADARB2-ERBB4","Ex-Neuron-CBLN2-NRGN","Ex-Neuron-DLC1-HS3ST4","Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2","Ex-Neuron-L3-5-RORB-PLCH1","Ex-Neuron-L4-5-RORB-IL1RAPL2","Ex-Neuron-L6-THEMIS-NFIA","In-Neuron-ADARB2-GRM7","In-Neuron-ADARB2-IL1RAPL2-CSCL14","In-Neuron-ADARB2-LAMP5-NRG1","In-Neuron-ADARB2-NRG1-RELN","In-Neuron-DPP10-MEF2C","In-Neuron-PRKG1-TBC1D4","In-Neuron-TRHDE-RALYL","Microglia-quiescent","Microglia-vascular","OPC-A","OPC-B","Oligo-A","Oligo-B","Perivascular","T-cell-mixed","T-cell-parenchymal-1","Oligo-A","Meningeal-FB","T-cell-vascular","Vascular-SMC-LINC00486","Perivascular-FB-KAZN2","Vascular-SMC-LINC00486","Perivascular-FB-KAZN2","M-Pericyte","Pericyte-2","Arteriolar-SMC","Arterial","T-cell-mixed","T-cell-mixed","Perivascular-FB","Meningeal-FB","T-Pericyte","M-Pericyte","Pericyte","Pericyte-2","Vascular-SMC","Arteriolar-SMC","Vascular-SMC-LINC00486","Perivascular-FB-KAZN2","Perivascular-FB","Meningeal-FB","Pericyte-FB","Capillary","Arterial","T-cell-mixed","Microglia-vascular","T-Pericyte","M-Pericyte","Pericyte","Vascular-SMC","Arteriolar-SMC","Vascular-SMC-LINC00486","Perivascular-FB-KAZN2","Perivascular-FB","Meningeal-FB","Pericyte-FB","Capillary","Arterial","T-cell-mixed","T-cell-vascular","T-Pericyte","M-Pericyte","Pericyte","Pericyte-2","Vascular-SMC-LINC00486","Meningeal-FB","Capillary","Arterial","T-cell-mixed"],[1289,1287,1292,1289,1287,1289,1286,1301,1288,1294,1279,1305,1306,1299,1308,1287,1288,1294,1279,1305,1306,1299,1308,1291,1287,249,259,264,258,265,263,259,264,258,263,2493,2580,2590,2578,2588,2597,2586,2592,2590,2578,2588,2587,2495,2580,2586,2557,2616,2633,2622,2571,2615,2588,2618,2635,2640,2632,2634,2558,2618,2634,2623,2572,2615,2589,2619,2637,2641,2632,2635,2493,1263,1263,1268,1266,1288,1292,1290,1302,1298,1266,1288,1292,1290,1302,1298,1262,1268,247,258,256,255,258,256,251,2493,2536,2496,2535,2526,2546,2560,2583,2575,2602,2593,2598,2560,2584,2576,2602,2593,2598,2545,1310,1310,1327,1333,1333,1308,1296,1319,1304,1309,1298,264,1936,1937,1942,1928,1920,1943,1929,1966,1949,1965,1970,1969,1967,1950,1979,1966,1971,1970,1934,1947,1935,1277,1277,1287,1237,1292,1300,1291,1268,1242,1263,1285,1253,1272,1266,1290,1293,1288,1305,1291,1291,1307,1305,1310,1261,1289,1311,1291,1306,1297,1313,1299,1302,1296,1308,1300,1305,1292,1300,1292,1268,1242,1263,1285,1253,1272,1266,1290,1293,1288,1305,1291,1291,1307,1305,1311,1261,1289,1311,1291,1307,1314,1299,1300,1305,260,256,256,258,253,256,257,263,261,258,263,261,260,249,267,267,264,260,256,258,253,256,257,263,261,258,263,261,264,1378,1378,1364,1387,1341,1395,1404,1393,1368,1339,1361,1388,1350,1372,1368,1391,1394,1385,1405,1397,1392,1410,1410,1416,1414,1364,1390,1412,1389,1411,1399,1416,1399,1404,1396,1412,1404,1411,1395,1404,1394,1368,1339,1361,1388,1351,1372,1368,1391,1394,1385,1406,1397,1392,1410,1410,1417,1364,1390,1412,1389,1412,1417,1400,1404,1411,1272,1272,1253,1274,1270,1261,1287,1251,1272,1288,1260,1274,1277,1297,1297,1305,1292,1297,1295,1296,1282,1295,1271,1305,1305,1294,1291,1295,1306,1261,1287,1251,1272,1288,1261,1274,1277,1297,1297,1305,1292,1297,1295,1296,1282,1296,1271,1305,1305,1294,1291,1295,1306,1274,251,256,256,254,259,241,258,256,263,255,250,259,260,241,256,262,263,265,255,250,249,1435,1435,1418,1388,1415,1407,1437,1442,1427,1459,1415,1438,1458,1429,1442,1440,1463,1464,1470,1467,1465,1459,1468,1449,1461,1439,1474,1470,1463,1459,1458,1478,1442,1427,1459,1415,1438,1458,1430,1442,1440,1463,1464,1470,1467,1465,1459,1468,1449,1462,1439,1474,1470,1463,1459,1458,1478,1437,1304,1306,1337,1336,1338,1337,1305,1271,1308,1309,1310,264,2042,2076,2099,2098,2105,2075,2109,2100,2112,2109,2078,2104,2065,2092,2095,2100,2115,2100,2099,2106,2110,2101,2113,2110,2079,2105,2067,2093,2096,2101,2099,2070,2059,2069,2033,2081,2075,2061,2064,2066],[0.08360099999999999,0.096333,0.078349,0.08258500000000001,0.094898,0.10037,0.12887,0.10967,0.19494,0.17871,0.13748,0.13192,0.15652,0.149,0.14693,0.10842,0.1606,0.15199,0.10492,0.10637,0.12561,0.12859,0.11175,0.07301100000000001,0.081523,0.1896,0.34227,0.30635,0.24444,0.21174,0.19859,0.32936,0.30439,0.19851,0.19551,0.071188,0.067635,0.0669,0.05984,0.06293600000000001,0.06358999999999999,0.082173,0.059759,0.073935,0.057659,0.07320500000000001,0.065219,0.074047,0.07204099999999999,0.058084,0.082708,0.11631,0.13011,0.082728,0.13581,0.14314,0.1344,0.12398,0.1222,0.14039,0.13032,0.11976,0.063794,0.095278,0.10263,0.059302,0.10056,0.11555,0.1019,0.10468,0.10163,0.11779,0.11576,0.094166,0.053281,0.079892,0.07385899999999999,0.090628,0.12433,0.10437,0.14764,0.15882,0.161,0.13993,0.10331,0.091541,0.14438,0.1452,0.14955,0.14173,0.07663300000000001,0.089251,0.19169,0.22959,0.30179,0.16067,0.2192,0.28232,0.17624,0.052526,0.07030599999999999,0.053243,0.057131,0.070018,0.073736,0.11726,0.12353,0.13489,0.1707,0.17765,0.11761,0.09574299999999999,0.1154,0.11322,0.14896,0.14417,0.10984,0.07181899999999999,0.070033,0.082799,0.077583,0.081845,0.081386,0.088325,0.073242,0.076918,0.097053,0.065275,0.075422,0.21996,0.070587,0.063401,0.058015,0.061784,0.067703,0.060418,0.077441,0.08883199999999999,0.08365400000000001,0.082925,0.10254,0.093014,0.076214,0.08888,0.073393,0.075588,0.083022,0.087947,0.066702,0.071851,0.07044499999999999,0.08828999999999999,0.096872,0.087356,0.092499,0.12243,0.16135,0.11871,0.13113,0.16721,0.16134,0.19525,0.1622,0.2016,0.16153,0.15691,0.17087,0.22136,0.1991,0.15776,0.18342,0.18633,0.14719,0.14895,0.14649,0.14974,0.12261,0.1405,0.14887,0.11547,0.16028,0.14145,0.12443,0.10979,0.1043,0.14089,0.1617,0.11406,0.13665,0.10158,0.11672,0.17642,0.13417,0.16314,0.15774,0.181,0.14176,0.12735,0.15594,0.20261,0.17162,0.14897,0.17876,0.15804,0.11403,0.16357,0.10714,0.12006,0.099758,0.10565,0.10866,0.13252,0.095919,0.12081,0.13734,0.20359,0.23113,0.3224,0.29471,0.26771,0.32818,0.32872,0.31702,0.25585,0.23985,0.29778,0.29869,0.22591,0.2178,0.24339,0.22668,0.25804,0.23601,0.29179,0.3179,0.2568,0.30609,0.28422,0.33047,0.24127,0.21535,0.29612,0.24258,0.21737,0.084508,0.09617299999999999,0.09876600000000001,0.090965,0.09676999999999999,0.11754,0.14935,0.11201,0.14749,0.16819,0.16133,0.18238,0.17439,0.18439,0.14991,0.16957,0.17883,0.2104,0.19336,0.14444,0.17699,0.18024,0.13602,0.14637,0.093537,0.14107,0.16054,0.1055,0.14776,0.14978,0.1156,0.16583,0.13286,0.12557,0.10469,0.11209,0.13335,0.15457,0.11444,0.12608,0.09926400000000001,0.12182,0.17936,0.12951,0.15611,0.16159,0.16638,0.13025,0.14197,0.15712,0.19751,0.16178,0.13257,0.17135,0.15226,0.10661,0.15675,0.10633,0.11956,0.093011,0.11167,0.11039,0.13395,0.096035,0.11036,0.12887,0.09354,0.10077,0.09834900000000001,0.092171,0.10071,0.1249,0.15661,0.16441,0.18024,0.14659,0.14308,0.15706,0.17288,0.17283,0.17774,0.19442,0.16645,0.19645,0.17911,0.21791,0.1417,0.16771,0.16136,0.16658,0.12852,0.18092,0.12897,0.14146,0.18763,0.112,0.13491,0.15387,0.15376,0.13573,0.12117,0.14494,0.14739,0.14103,0.15296,0.16219,0.14492,0.15589,0.1534,0.17378,0.12152,0.14774,0.14222,0.13853,0.13397,0.17176,0.12894,0.11413,0.14929,0.10012,0.18648,0.22302,0.20956,0.17925,0.20462,0.2075,0.26436,0.21731,0.2572,0.35636,0.25184,0.21992,0.19549,0.21623,0.19815,0.21894,0.268,0.19713,0.34127,0.20123,0.192,0.077322,0.08275,0.080946,0.083472,0.096013,0.09084100000000001,0.09854499999999999,0.1372,0.11155,0.12975,0.15709,0.17795,0.13538,0.13958,0.12717,0.15618,0.17388,0.16253,0.16631,0.16977,0.17202,0.17849,0.21769,0.1496,0.18867,0.17638,0.16879,0.12481,0.20398,0.12793,0.16186,0.1662,0.10734,0.098175,0.11624,0.14223,0.14994,0.12881,0.12243,0.12928,0.1332,0.13994,0.13992,0.14426,0.14847,0.13418,0.15741,0.177,0.12275,0.16508,0.15101,0.14242,0.13013,0.1812,0.12796,0.12137,0.14016,0.09761300000000001,0.073947,0.08756,0.10235,0.08852400000000001,0.10787,0.10328,0.092379,0.079842,0.077127,0.09518699999999999,0.07589799999999999,0.17265,0.072736,0.06778099999999999,0.081416,0.096841,0.10689,0.087628,0.096668,0.10734,0.10547,0.10386,0.082221,0.13439,0.07795100000000001,0.10186,0.072785,0.083034,0.06940399999999999,0.07736899999999999,0.084787,0.10604,0.085453,0.094737,0.10221,0.11258,0.07811700000000001,0.12435,0.075839,0.07610500000000001,0.067673,0.078247,0.074821,0.074087,0.06697599999999999,0.059238,0.075056,0.064397,0.067854,0.068956,0.070728,0.07288600000000001],[0.021333,0.024564,0.020014,0.021067,0.024191,0.025573,0.032471,0.027781,0.049155,0.045159,0.034553,0.033465,0.03972,0.037718,0.037313,0.027291,0.040438,0.038352,0.026332,0.026946,0.031832,0.032506,0.02834,0.018664,0.02081,0.021894,0.039817,0.035976,0.028382,0.024912,0.023278,0.038258,0.035692,0.023014,0.022882,0.024358,0.023479,0.023261,0.020766,0.021876,0.022135,0.028553,0.020785,0.0257,0.020003,0.025438,0.022659,0.025309,0.024973,0.020154,0.028304,0.040188,0.045079,0.028611,0.046585,0.049448,0.046227,0.04285,0.042352,0.048694,0.045142,0.041498,0.021807,0.032889,0.035519,0.020487,0.034454,0.039869,0.035012,0.036142,0.035187,0.04081,0.04005,0.032594,0.018249,0.020195,0.018665,0.022915,0.031101,0.026317,0.03728,0.040075,0.040799,0.03541,0.025807,0.02305,0.036407,0.036587,0.037846,0.035815,0.019385,0.022626,0.022047,0.026657,0.034907,0.01852,0.025413,0.032606,0.020461,0.017973,0.02423,0.018201,0.019658,0.024057,0.025419,0.04015,0.042454,0.0463,0.058848,0.061156,0.04052,0.03274,0.039618,0.038819,0.051289,0.049567,0.037795,0.024812,0.017977,0.021254,0.019834,0.020968,0.020821,0.022716,0.018757,0.019859,0.024925,0.016794,0.019328,0.026142,0.021653,0.019453,0.017821,0.018918,0.020662,0.018536,0.023683,0.027135,0.025455,0.025325,0.03135,0.028431,0.023254,0.027015,0.022454,0.023058,0.025355,0.026853,0.020473,0.022119,0.021627,0.022432,0.024605,0.022268,0.023121,0.030914,0.040859,0.029964,0.032824,0.041455,0.040313,0.049179,0.04038,0.050539,0.040406,0.039594,0.043162,0.055815,0.05051,0.039822,0.0463,0.047302,0.03734,0.037854,0.036575,0.03777,0.03117,0.035465,0.037779,0.02921,0.040776,0.035808,0.031532,0.027763,0.026488,0.035678,0.041022,0.028761,0.034556,0.025614,0.029177,0.043677,0.033477,0.041033,0.039214,0.045311,0.03541,0.03209,0.039335,0.051017,0.043476,0.037552,0.045061,0.040065,0.028887,0.041527,0.026715,0.030241,0.025326,0.026631,0.027545,0.033678,0.024247,0.030549,0.034792,0.024015,0.02702,0.037291,0.034219,0.030785,0.037959,0.038094,0.037159,0.029876,0.027849,0.034904,0.034878,0.026331,0.02485,0.028741,0.026768,0.030303,0.027507,0.0337,0.036856,0.029487,0.035351,0.032888,0.038678,0.028132,0.024967,0.034657,0.028285,0.025488,0.022238,0.0253,0.025861,0.024002,0.025109,0.030751,0.039187,0.029284,0.038239,0.043178,0.041729,0.047603,0.044939,0.047871,0.038867,0.044303,0.046768,0.054863,0.050753,0.037812,0.046258,0.047385,0.035759,0.038557,0.024623,0.036525,0.041931,0.027755,0.038579,0.039389,0.030282,0.043683,0.034803,0.032948,0.027398,0.029487,0.03499,0.040651,0.029898,0.033036,0.025924,0.03154,0.045981,0.033452,0.040689,0.041596,0.043137,0.033722,0.037042,0.041034,0.05143,0.042418,0.034657,0.044722,0.039974,0.027989,0.041247,0.027492,0.031183,0.024435,0.029115,0.029002,0.035247,0.025131,0.028918,0.033844,0.023723,0.025549,0.024731,0.023356,0.025229,0.031185,0.039475,0.040899,0.045185,0.036963,0.035711,0.039403,0.04342,0.04372,0.044962,0.049321,0.04203,0.049694,0.045275,0.055104,0.035652,0.042395,0.040438,0.04226,0.032603,0.045717,0.032556,0.035759,0.047617,0.027925,0.033958,0.038224,0.038493,0.034175,0.030211,0.036312,0.036964,0.035625,0.038639,0.041087,0.036542,0.03938,0.038722,0.043882,0.030533,0.037307,0.035589,0.035094,0.033937,0.043341,0.032503,0.02881,0.037833,0.025438,0.021619,0.026107,0.024532,0.020902,0.023803,0.023296,0.030695,0.025135,0.030147,0.041139,0.02879,0.025545,0.022751,0.02424,0.022885,0.025576,0.031366,0.023158,0.039338,0.02297,0.022203,0.020729,0.022178,0.021576,0.022005,0.025536,0.024097,0.026395,0.036444,0.029489,0.03465,0.041368,0.047209,0.036143,0.036924,0.03378,0.041459,0.046495,0.043473,0.044568,0.045452,0.046025,0.047666,0.058299,0.039827,0.050418,0.046806,0.045288,0.033447,0.054544,0.034164,0.043211,0.044649,0.028474,0.025917,0.031001,0.037402,0.039722,0.034341,0.032353,0.034292,0.035311,0.037366,0.037375,0.038605,0.039695,0.035852,0.041981,0.047338,0.032634,0.044067,0.040018,0.038161,0.034824,0.048384,0.034125,0.032359,0.037601,0.026213,0.01912,0.022459,0.026258,0.022702,0.027644,0.026458,0.023733,0.020263,0.019836,0.024489,0.019533,0.02052,0.022808,0.021408,0.025596,0.030438,0.033645,0.02741,0.030454,0.033753,0.033247,0.032718,0.025735,0.042293,0.024332,0.031975,0.022863,0.02611,0.021892,0.024296,0.02662,0.033342,0.026891,0.029757,0.032183,0.035426,0.024423,0.039089,0.023651,0.023864,0.021234,0.024577,0.023491,0.023429,0.02113,0.018729,0.023548,0.020411,0.02148,0.021764,0.022338,0.023029],[0.02754,0.027635,0.027354,0.027903,0.027953,0.028939,0.031913,0.031668,0.03171,0.032034,0.0323,0.031681,0.031364,0.031416,0.031952,0.030061,0.030365,0.030331,0.030435,0.029807,0.030113,0.029838,0.029828,0.026501,0.026054,0.060591,0.0723,0.072065,0.067452,0.069386,0.069703,0.068083,0.067367,0.06730700000000001,0.067986,0.02059,0.020405,0.020374,0.020064,0.020428,0.020294,0.020573,0.020147,0.020702,0.020446,0.020709,0.02091,0.021532,0.021303,0.021389,0.02336,0.023228,0.023252,0.023201,0.023014,0.023298,0.02373,0.023185,0.023152,0.022923,0.023107,0.022863,0.022098,0.021987,0.021819,0.021808,0.022045,0.022224,0.02231,0.021933,0.021978,0.021943,0.022084,0.02182,0.019176,0.027766,0.028203,0.029692,0.032055,0.032076,0.031572,0.031982,0.032162,0.031984,0.030671,0.029978,0.03036,0.030673,0.030373,0.029983,0.0272,0.026667,0.062251,0.071784,0.069081,0.06593,0.067761,0.06703199999999999,0.059837,0.021143,0.020445,0.021954,0.021452,0.021603,0.021544,0.023393,0.023342,0.022774,0.023073,0.023599,0.023188,0.022333,0.021961,0.022188,0.022453,0.022126,0.022124,0.018924,0.028291,0.028243,0.030853,0.030976,0.029075,0.026092,0.026208,0.025498,0.026193,0.025663,0.025674,0.057045,0.022912,0.022934,0.022741,0.022948,0.023861,0.023485,0.023621,0.025665,0.025886,0.025653,0.025513,0.025803,0.024753,0.024379,0.024384,0.024798,0.024681,0.024684,0.021678,0.021317,0.021628,0.027517,0.027774,0.02812,0.029648,0.031837,0.031516,0.031653,0.031647,0.031682,0.031925,0.031922,0.032014,0.03166,0.031459,0.032148,0.032264,0.031794,0.032211,0.031729,0.031684,0.031993,0.031604,0.031444,0.032299,0.032219,0.031575,0.031398,0.031393,0.031872,0.031581,0.031812,0.032035,0.031411,0.031021,0.031344,0.030838,0.030085,0.029803,0.029985,0.03015,0.030841,0.030374,0.030198,0.030782,0.030642,0.030553,0.030079,0.030167,0.030525,0.029946,0.030359,0.030275,0.029932,0.030088,0.029948,0.030667,0.030314,0.029736,0.029999,0.030132,0.029595,0.030208,0.029927,0.029706,0.057808,0.06376900000000001,0.072727,0.07253800000000001,0.073629,0.07315099999999999,0.07261099999999999,0.071964,0.072075,0.072159,0.071475,0.07302,0.067505,0.069091,0.069149,0.06918100000000001,0.06988,0.068386,0.06877,0.067693,0.06946099999999999,0.06869400000000001,0.06768,0.067978,0.067957,0.06871099999999999,0.067827,0.069173,0.066993,0.026339,0.02669,0.027902,0.027187,0.028527,0.030651,0.030404,0.030713,0.030502,0.030696,0.030791,0.030722,0.030861,0.03049,0.030251,0.030881,0.031122,0.03089,0.03109,0.030588,0.030679,0.030852,0.030412,0.030276,0.029946,0.031102,0.03104,0.030496,0.030313,0.030285,0.030932,0.030456,0.030718,0.030995,0.030365,0.029954,0.030189,0.029729,0.028824,0.028683,0.028916,0.02912,0.029824,0.029402,0.029099,0.029429,0.029555,0.029399,0.029067,0.029059,0.029523,0.028884,0.029225,0.029236,0.028949,0.028958,0.028879,0.029538,0.029274,0.028683,0.029089,0.029063,0.028697,0.029264,0.028902,0.028703,0.027256,0.027709,0.029412,0.029473,0.032145,0.032181,0.031906,0.031741,0.031737,0.031885,0.032229,0.031627,0.03173,0.03198,0.032138,0.03154,0.032274,0.032017,0.031467,0.031961,0.032168,0.031275,0.032201,0.031981,0.031911,0.031868,0.031693,0.032025,0.031555,0.030626,0.030612,0.03065,0.030496,0.030101,0.030644,0.030452,0.030673,0.030693,0.030538,0.030275,0.030431,0.030589,0.03051,0.030513,0.030055,0.030074,0.030723,0.030355,0.030099,0.030198,0.02973,0.030159,0.029942,0.026472,0.060803,0.060363,0.058871,0.059549,0.06805700000000001,0.06830600000000001,0.071737,0.07197099999999999,0.071406,0.068969,0.069481,0.065025,0.067242,0.066537,0.067092,0.067496,0.067855,0.067646,0.06816899999999999,0.066256,0.060006,0.025845,0.02628,0.026728,0.027775,0.027718,0.027816,0.027855,0.030271,0.030247,0.029902,0.03014,0.029903,0.029913,0.030212,0.029942,0.029919,0.03005,0.030345,0.029872,0.030224,0.030103,0.029824,0.030075,0.030177,0.029764,0.030506,0.030096,0.030205,0.030174,0.030159,0.030432,0.029628,0.028714,0.028809,0.028689,0.029047,0.028755,0.028479,0.028758,0.028916,0.028866,0.028934,0.028724,0.028743,0.028672,0.028915,0.02873,0.028745,0.028418,0.028359,0.029034,0.028548,0.028464,0.028634,0.028178,0.028518,0.028521,0.024854,0.025551,0.026131,0.03023,0.030571,0.029211,0.02917,0.026289,0.026723,0.025872,0.026033,0.025571,0.057973,0.023336,0.023196,0.025075,0.02498,0.024845,0.025211,0.024729,0.024865,0.024763,0.02468,0.025357,0.02509,0.025277,0.02511,0.02502,0.025303,0.02434,0.024074,0.024095,0.02398,0.023933,0.023755,0.023763,0.023813,0.024155,0.02401,0.023963,0.024056,0.023848,0.023766,0.023672,0.02091,0.020966,0.020678,0.021201,0.020532,0.020912,0.020777,0.020955,0.020738],[0.0012025,0.00024591,0.0020928,0.0015421,0.00034418,0.00026239,2.7068e-05,0.00026791,4.0289e-10,1.2321e-08,1.0452e-05,1.5731e-05,3.0466e-07,1.0638e-06,2.1461e-06,0.00015556,6.255e-08,2.7448e-07,0.00028408,0.0001801,1.5244e-05,8.2395e-06,9.0022e-05,0.0029384,0.00087866,0.00087804,1.1105e-06,1.0703e-05,0.00014557,0.0011398,0.0021947,6.6463e-07,3.1426e-06,0.0015954,0.002019,0.00027353,0.00045989,0.00051359,0.0014317,0.0010336,0.00086535,3.2599e-05,0.00151,0.00017805,0.0024039,0.00020459,0.00090891,0.00029271,0.00036091,0.0033116,0.00020027,2.7903e-07,1.1163e-08,0.00018199,1.8413e-09,4.1344e-10,7.5564e-09,4.5283e-08,6.615099999999999e-08,4.6583e-10,8.6747e-09,8.226e-08,0.0019487,7.395e-06,1.2892e-06,0.0032759,2.562e-06,1.014e-07,2.4883e-06,9.1762e-07,1.8997e-06,4.0482e-08,8.0712e-08,8.0148e-06,0.002734,0.0020081,0.0044158,0.0011376,5.2728e-05,0.0005705,1.4733e-06,3.4574e-07,2.8129e-07,6.1094e-06,0.00037915,0.0011328,9.992699999999999e-07,1.1134e-06,4.2931e-07,1.1512e-06,0.0024239,0.00040972,0.0010391,0.00069254,6.2932e-06,0.0074123,0.00060999,1.2759e-05,0.0016157,0.0064954,0.00029298,0.007655,0.003874,0.00059662,0.00031089,2.7125e-07,6.1355e-08,1.6144e-09,7.2639e-14,2.7198e-14,1.9889e-07,9.1164e-06,7.516099999999999e-08,1.6964e-07,1.6937e-11,3.7419e-11,3.4807e-07,7.4119e-05,0.0066589,0.001688,0.0059639,0.0041225,0.0025654,0.00035671,0.0026015,0.0012803,0.00010599,0.0054921,0.0016562,5.7887e-05,0.0010339,0.0028537,0.0053739,0.0035517,0.002277,0.0050503,0.00052303,0.0002696,0.0006166500000000001,0.00061473,2.9366e-05,0.00015669,0.0010407,0.00013382,0.001309,0.0011537,0.00038556,0.00018397,0.0010478,0.00037607,0.00056417,0.00066845,0.00024416,0.00094839,0.00090629,6.0424e-05,1.5498e-07,8.8643e-05,1.7201e-05,6.633600000000001e-08,2.1925e-07,4.9049e-10,2.0483e-07,9.880700000000001e-11,1.431e-07,5.3355e-07,6.0025e-08,1.744e-12,3.2634e-10,3.3485e-07,3.6095e-09,2.9312e-09,1.6154e-06,1.094e-06,2.8977e-06,1.6949e-06,5.1805e-05,3.8541e-06,1.0671e-06,0.00014614,1.9575e-07,4.3944e-06,5.1563e-05,0.00023743,0.00038742,3.5061e-06,7.9808e-08,7.5312e-05,2.2906e-06,0.00035348,5.4408e-05,5.438e-09,5.04e-06,3.3451e-08,1.513e-07,1.786e-09,1.7614e-06,1.1559e-05,1.1925e-07,1.6563e-11,5.1041e-09,4.6788e-07,1.8093e-09,6.562599999999999e-08,7.572e-05,2.3965e-08,0.00023884,3.7605e-05,0.00039828,0.00021508,0.00015605,3.8037e-06,0.0007502,2.726e-05,1.9069e-06,0.00021494,0.00014522,4.6779e-06,2.4364e-05,0.00013895,3.6496e-06,3.0136e-06,5.3199e-06,0.00019339,0.00044483,1.5568e-05,2.164e-05,0.00040998,0.00081129,0.00021663,0.00052656,0.00011136,0.00027993,1.1103e-05,1.3383e-06,0.00010948,4.2118e-06,1.3468e-05,5.8932e-07,0.00019298,0.00086368,6.3814e-06,0.00022743,0.00058941,0.00066862,0.00015755,0.00020085,0.00041109,0.00034746,6.3055e-05,4.5555e-07,0.00013313,6.7146e-07,2.1696e-08,8.164e-08,1.489e-09,8.129e-09,7.530199999999999e-10,3.6455e-07,2.0299e-08,4.6576e-09,5.0167e-12,2.5594e-10,1.178e-06,4.0628e-09,2.6319e-09,3.8949e-06,6.7346e-07,0.0008952,2.8943e-06,1.1724e-07,0.00027126,5.5135e-07,3.8388e-07,9.3432e-05,2.631e-08,7.6756e-06,2.5597e-05,0.0002834,9.163800000000001e-05,5.0351e-06,1.0129e-07,3.6074e-05,5.5682e-06,0.00029964,1.445e-05,9.2901e-10,5.3373e-06,4.1214e-08,2.0375e-08,9.2158e-09,4.7423e-06,5.2492e-07,3.2633e-08,1.1607e-11,1.087e-08,2.8885e-06,2.3547e-09,7.3336e-08,0.0001164,2.9018e-08,0.00015991,2.2262e-05,0.00059339,6.2123e-05,7.313999999999999e-05,1.5389e-06,0.00051725,6.7449e-05,3.5965e-06,0.0003005,0.00013852,0.00041419,0.00088358,0.00086649,5.2211e-05,4.6371e-07,1.1251e-07,6.8959e-09,2.1552e-06,4.5426e-06,3.4534e-07,2.578e-08,3.3021e-08,1.6228e-08,3.6328e-10,1.2684e-07,4.3446e-10,6.3966e-09,4.7881e-12,5.3279e-06,4.1651e-08,2.7358e-07,9.6308e-08,2.8338e-05,6.9791e-09,2.3675e-05,5.0325e-06,1.4027e-09,0.00012808,5.2812e-06,2.6138e-07,2.3342e-07,3.2855e-06,3.862e-05,9.7962e-07,7.8215e-07,2.1855e-06,2.7749e-07,4.2998e-08,9.681300000000001e-07,1.7557e-07,2.515e-07,6.2972e-09,2.6498e-05,4.5494e-07,1.8553e-06,2.5357e-06,4.3123e-06,6.5739e-09,7.2715e-06,7.7387e-05,3.1232e-07,7.805e-05,0.0010829,0.00011046,0.0001862,0.0013079,0.0013235,0.0011936,0.00011467,0.0012686,0.00015848,1.2048e-07,0.00014519,0.00036076,0.0018264,0.0005789,0.0015742,0.00059132,3.9354e-05,0.0017864,2.8118e-07,0.0011965,0.00068953,0.0013892,0.00082128,0.0012309,0.0013286,0.00026689,0.00054689,0.00020232,2.9381e-06,0.00011342,7.2036e-06,9.458200000000001e-08,1.3621e-09,3.0307e-06,1.9326e-06,1.0885e-05,9.0604e-08,3.6683e-09,4.3154e-08,1.3141e-08,9.892e-09,5.6135e-09,1.1088e-09,2.383e-13,3.6087e-07,1.1902e-10,3.7704e-09,1.0401e-08,1.8069e-05,7.1418e-12,1.1155e-05,5.3023e-08,1.0324e-08,9.3018e-05,0.0003285,2.5554e-05,4.9307e-07,9.3611e-08,3.0765e-06,1.0413e-05,3.9308e-06,1.9887e-06,6.6863e-07,5.6063e-07,2.6323e-07,1.1361e-07,1.7535e-06,2.1764e-08,3.7993e-10,7.874600000000001e-06,2.9899e-09,1.0056e-07,3.0745e-07,2.4397e-06,1.2805e-10,2.8228e-06,1.0472e-05,4.509e-07,4.3142e-05,0.0019043,0.00040397,0.00035579,0.0018943,0.00011137,0.00020031,0.00022138,0.0014077,0.0014388,0.00012839,0.0015006,0.0014521,0.00091531,0.0017406,0.00058453,5.3143e-05,8.5116e-06,0.00025534,4.654e-05,7.9592e-06,1.0329e-05,1.2952e-05,0.00059376,4.3128e-08,0.0010235,2.5045e-05,0.0018154,0.00051721,0.0021793,0.0006565,0.00021744,4.9259e-06,0.0001788,3.3493e-05,8.5602e-06,1.1476e-06,0.00061177,1.1326e-07,0.00077755,0.00078074,0.0022754,0.00049802,0.00078841,0.00019836,0.0007018,0.0020893,0.00020059,0.00085695,0.00058924,0.00045309,0.00036983,0.00022084],[0.01443,0.00295092,0.0251136,0.0185052,0.00413016,0.00314868,0.000324816,0.00321492,4.83468e-09,1.47852e-07,0.000125424,0.000188772,3.65592e-06,1.27656e-05,2.57532e-05,0.00186672,7.506000000000001e-07,3.29376e-06,0.00340896,0.0021612,0.000182928,9.887399999999999e-05,0.001080264,0.03526079999999999,0.01054392,0.01053648,1.3326e-05,0.000128436,0.00174684,0.0136776,0.0263364,7.975560000000001e-06,3.771120000000001e-05,0.0191448,0.024228,0.00328236,0.00551868,0.00616308,0.0171804,0.0124032,0.0103842,0.000391188,0.01812,0.0021366,0.0288468,0.00245508,0.01090692,0.00351252,0.00433092,0.0397392,0.00240324,3.34836e-06,1.33956e-07,0.00218388,2.20956e-08,4.96128e-09,9.06768e-08,5.433960000000001e-07,7.938119999999999e-07,5.58996e-09,1.040964e-07,9.8712e-07,0.0233844,8.874000000000001e-05,1.54704e-05,0.0393108,3.0744e-05,1.2168e-06,2.98596e-05,1.101144e-05,2.27964e-05,4.85784e-07,9.685440000000001e-07,9.61776e-05,0.032808,0.0120486,0.0264948,0.006825599999999999,0.000316368,0.003423,8.8398e-06,2.07444e-06,1.68774e-06,3.66564e-05,0.0022749,0.0067968,5.99562e-06,6.6804e-06,2.57586e-06,6.907200000000001e-06,0.0145434,0.00245832,0.0062346,0.004155239999999999,3.77592e-05,0.0444738,0.00365994,7.6554e-05,0.0096942,0.0389724,0.00175788,0.04593,0.023244,0.00357972,0.00186534,1.6275e-06,3.6813e-07,9.6864e-09,4.35834e-13,1.63188e-13,1.19334e-06,5.46984e-05,4.50966e-07,1.01784e-06,1.01622e-10,2.24514e-10,2.08842e-06,0.000444714,0.0466123,0.011816,0.0417473,0.0288575,0.0179578,0.00249697,0.0182105,0.008962100000000001,0.00074193,0.0384447,0.0115934,0.000405209,0.007237299999999999,0.0199759,0.0376173,0.0248619,0.015939,0.0353521,0.00366121,0.0018872,0.004316550000000001,0.00430311,0.000205562,0.00109683,0.0072849,0.0009367399999999999,0.009163000000000001,0.008075899999999999,0.00269892,0.00128779,0.0073346,0.00263249,0.00394919,0.0294118,0.01074304,0.04172916,0.03987676,0.002658656,6.81912e-06,0.003900292,0.000756844,2.918784e-06,9.647000000000001e-06,2.158156e-08,9.01252e-06,4.347508000000001e-09,6.2964e-06,2.34762e-05,2.6411e-06,7.6736e-11,1.435896e-08,1.47334e-05,1.58818e-07,1.289728e-07,7.10776e-05,4.8136e-05,0.0001274988,7.45756e-05,0.00227942,0.0001695804,4.695239999999999e-05,0.006430160000000001,8.613e-06,0.0001933536,0.002268772,0.01044692,0.01704648,0.0001542684,3.511552e-06,0.003313728,0.0001007864,0.01555312,0.002393952,2.39272e-07,0.00022176,1.471844e-06,6.6572e-06,7.8584e-08,7.75016e-05,0.000508596,5.247000000000001e-06,7.28772e-10,2.245804e-07,2.058672e-05,7.960919999999999e-08,2.887544e-06,0.00333168,1.05446e-06,0.01050896,0.00165462,0.01752432,0.00946352,0.0068662,0.0001673628,0.0330088,0.00119944,8.39036e-05,0.00945736,0.00638968,0.0002058276,0.001072016,0.006113799999999999,0.0001605824,0.0001325984,0.0002340756,0.00850916,0.01957252,0.000684992,0.00095216,0.01803912,0.03569676,0.00953172,0.02316864,0.00489984,0.01231692,0.000488532,5.88852e-05,0.00481712,0.0001853192,0.0005925920000000001,2.593008e-05,0.008491120000000001,0.03800192,0.0002807816,0.01000692,0.02593404,0.02941928,0.0069322,0.0088374,0.01808796,0.01528824,0.00277442,2.00442e-05,0.005857719999999999,2.954424e-05,9.54624e-07,3.59216e-06,6.5516e-08,3.57676e-07,3.313288e-08,1.60402e-05,8.931559999999999e-07,2.049344e-07,2.207348e-10,1.126136e-08,5.1832e-05,1.787632e-07,1.158036e-07,0.0001713756,2.963224e-05,0.0393888,0.0001273492,5.15856e-06,0.01193544,2.42594e-05,1.689072e-05,0.004111008,1.15764e-06,0.0003377264,0.001126268,0.0124696,0.004032072,0.0002215444,4.45676e-06,0.001587256,0.0002450008,0.01318416,0.0006358000000000001,4.087644e-08,0.0002348412,1.813416e-06,8.964999999999999e-07,4.054952e-07,0.0002086612,2.309648e-05,1.435852e-06,5.10708e-10,4.7828e-07,0.000127094,1.036068e-07,3.226784e-06,0.0051216,1.276792e-06,0.00703604,0.0009795280000000001,0.02610916,0.002733412,0.00321816,6.77116e-05,0.022759,0.002967756,0.000158246,0.008113499999999999,0.00374004,0.01118313,0.02385666,0.02339523,0.001409697,1.252017e-05,3.03777e-06,1.861893e-07,5.81904e-05,0.0001226502,9.32418e-06,6.960599999999999e-07,8.91567e-07,4.38156e-07,9.808560000000001e-09,3.42468e-06,1.173042e-08,1.727082e-07,1.292787e-10,0.0001438533,1.124577e-06,7.38666e-06,2.600316e-06,0.000765126,1.884357e-07,0.000639225,0.0001358775,3.78729e-08,0.00345816,0.0001425924,7.057259999999999e-06,6.30234e-06,8.87085e-05,0.00104274,2.644974e-05,2.111805e-05,5.90085e-05,7.49223e-06,1.160946e-06,2.613951e-05,4.74039e-06,6.7905e-06,1.700244e-07,0.000715446,1.228338e-05,5.00931e-05,6.846390000000001e-05,0.0001164321,1.774953e-07,0.0001963305,0.002089449,8.432639999999999e-06,0.00210735,0.0292383,0.00298242,0.0050274,0.0353133,0.0357345,0.0322272,0.00309609,0.0342522,0.00427896,3.25296e-06,0.00392013,0.009740520000000001,0.0493128,0.0156303,0.0425034,0.01596564,0.001062558,0.0482328,7.59186e-06,0.0323055,0.01861731,0.0375084,0.02217456,0.0332343,0.0358722,0.00720603,0.01476603,0.00546264,7.932869999999999e-05,0.00306234,0.0001944972,2.553714e-06,3.67767e-08,8.18289e-05,5.21802e-05,0.000293895,2.446308e-06,9.904409999999999e-08,1.165158e-06,3.54807e-07,2.67084e-07,1.515645e-07,2.99376e-08,6.4341e-12,9.74349e-06,3.21354e-09,1.018008e-07,2.80827e-07,0.000487863,1.928286e-10,0.000301185,1.431621e-06,2.78748e-07,0.002511486,0.008869500000000001,0.000689958,1.331289e-05,2.527497e-06,8.30655e-05,0.000281151,0.0001061316,5.36949e-05,1.805301e-05,1.513701e-05,7.10721e-06,3.06747e-06,4.73445e-05,5.876279999999999e-07,1.025811e-08,0.0002126142,8.072730000000001e-08,2.71512e-06,8.301150000000001e-06,6.58719e-05,3.45735e-09,7.62156e-05,0.000282744,1.21743e-05,0.001164834,0.0361817,0.00767543,0.006760009999999999,0.0359917,0.00211603,0.00380589,0.00420622,0.0267463,0.0273372,0.00243941,0.0285114,0.0275899,0.01739089,0.0330714,0.01110607,0.001009717,0.0001617204,0.00485146,0.00088426,0.0001512248,0.000196251,0.000246088,0.01128144,8.19432e-07,0.0194465,0.000475855,0.0344926,0.009826990000000001,0.0414067,0.0124735,0.00413136,9.359209999999999e-05,0.0033972,0.000636367,0.0001626438,2.18044e-05,0.01162363,2.15194e-06,0.01477345,0.01483406,0.0432326,0.009462379999999999,0.01497979,0.00376884,0.0133342,0.0396967,0.00381121,0.01628205,0.01119556,0.00860871,0.00702677,0.00419596],["all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","all_controls","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_parenchymal","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular","controls_vascular"],["Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","stroke_35k10k","wmh_35k10k","Bellenguez_2022_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","pd_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","wmh_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","wmh_35k10k","wmh_35k10k","Bellenguez_2022_35k10k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","wmh_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","pd_35k10k","pd_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","wmh_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_50k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","wmh_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_noproxy_35k10k","pd_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","stroke_35k10k","stroke_35k10k","stroke_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","Bellenguez_2022_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","wmh_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","wmh_35k10k","Bellenguez_2022_noproxy_35k10k","Bellenguez_2022_35k10k","Bellenguez_2022_35k10k","pd_35k10k","pd_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","wmh_35k10k","Kunkle_2019_50k","Kunkle_2019_35k10k","scz_25k","scz_25k","scz_50k","scz_50k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","wmh_35k10k","Bellenguez_2022_35k10k","pd_35k10k","pd_35k10k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_25k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","scz_50k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k","stroke_35k10k"],["no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter","no_filter"],["level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level1","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2","level2"],["top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-one-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-one-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-five-percent","top-one-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent","top-ten-percent"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>VARIABLE<\/th>\n      <th>NGENES<\/th>\n      <th>BETA<\/th>\n      <th>BETA_STD<\/th>\n      <th>SE<\/th>\n      <th>P<\/th>\n      <th>p_adj<\/th>\n      <th>input_gene_list<\/th>\n      <th>gwas_background<\/th>\n      <th>input_filter<\/th>\n      <th>celltype_level<\/th>\n      <th>percent<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"VARIABLE","targets":1},{"name":"NGENES","targets":2},{"name":"BETA","targets":3},{"name":"BETA_STD","targets":4},{"name":"SE","targets":5},{"name":"P","targets":6},{"name":"p_adj","targets":7},{"name":"input_gene_list","targets":8},{"name":"gwas_background","targets":9},{"name":"input_filter","targets":10},{"name":"celltype_level","targets":11},{"name":"percent","targets":12}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div></div>
</section>
<section id="heatmaps" class="level1">
<h1>Heatmaps</h1>
<p>I have level 1 and level 2 celltypes, and I have a â€œstandardâ€ and more strict filtering on the input celltype markers, so Iâ€™ll probably make a heatmap for each.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform the p-values to -log10(p-value) to enhance the visualization of small p-values</span></span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>log_p <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(df<span class="sc">$</span>p_adj)</span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>log_p_nominal <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(df<span class="sc">$</span>P)</span>
<span id="cb205-4"><a href="#cb205-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-5"><a href="#cb205-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Cut unneeded parts of group string</span></span>
<span id="cb205-6"><a href="#cb205-6" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(df<span class="sc">$</span>group, <span class="st">"level1_"</span>, <span class="st">""</span>)</span>
<span id="cb205-7"><a href="#cb205-7" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(df<span class="sc">$</span>group, <span class="st">"level2_"</span>, <span class="st">""</span>)</span>
<span id="cb205-8"><a href="#cb205-8" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(df<span class="sc">$</span>group, <span class="st">"logfc_threshold_"</span>, <span class="st">""</span>)</span>
<span id="cb205-9"><a href="#cb205-9" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(df<span class="sc">$</span>group, <span class="st">"top.*?percent_"</span>, <span class="st">""</span>) <span class="sc">|&gt;</span></span>
<span id="cb205-10"><a href="#cb205-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"EUROPEUKBB"</span>, <span class="st">"Bellenguez_2022"</span>) <span class="sc">|&gt;</span></span>
<span id="cb205-11"><a href="#cb205-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"AD_"</span>, <span class="st">"Kunkle_2019_"</span>)</span>
<span id="cb205-12"><a href="#cb205-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-13"><a href="#cb205-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Nest data on broad groups</span></span>
<span id="cb205-14"><a href="#cb205-14" aria-hidden="true" tabindex="-1"></a>df_nest <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span> </span>
<span id="cb205-15"><a href="#cb205-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(gwas_background) <span class="sc">|&gt;</span></span>
<span id="cb205-16"><a href="#cb205-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype_level, percent) <span class="sc">|&gt;</span></span>
<span id="cb205-17"><a href="#cb205-17" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span>
<span id="cb205-18"><a href="#cb205-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-19"><a href="#cb205-19" aria-hidden="true" tabindex="-1"></a>get_heatmap_data <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">nominal_p =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb205-20"><a href="#cb205-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Prepare the data for the heatmap</span></span>
<span id="cb205-21"><a href="#cb205-21" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(nominal_p) {</span>
<span id="cb205-22"><a href="#cb205-22" aria-hidden="true" tabindex="-1"></a>  heatmap_data <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb205-23"><a href="#cb205-23" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(VARIABLE, log_p_nominal, group) <span class="sc">|&gt;</span></span>
<span id="cb205-24"><a href="#cb205-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> VARIABLE, <span class="at">values_from =</span> log_p_nominal) <span class="sc">|&gt;</span></span>
<span id="cb205-25"><a href="#cb205-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb205-26"><a href="#cb205-26" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb205-27"><a href="#cb205-27" aria-hidden="true" tabindex="-1"></a>  heatmap_data <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb205-28"><a href="#cb205-28" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(VARIABLE, log_p, group) <span class="sc">|&gt;</span></span>
<span id="cb205-29"><a href="#cb205-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> VARIABLE, <span class="at">values_from =</span> log_p) <span class="sc">|&gt;</span></span>
<span id="cb205-30"><a href="#cb205-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb205-31"><a href="#cb205-31" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb205-32"><a href="#cb205-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb205-33"><a href="#cb205-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(heatmap_data) <span class="ot">&lt;-</span> heatmap_data<span class="sc">$</span>group</span>
<span id="cb205-34"><a href="#cb205-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove the first column if it contains row names (cell types)</span></span>
<span id="cb205-35"><a href="#cb205-35" aria-hidden="true" tabindex="-1"></a>  heatmap_data <span class="ot">&lt;-</span> heatmap_data[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb205-36"><a href="#cb205-36" aria-hidden="true" tabindex="-1"></a>  heatmap <span class="ot">&lt;-</span> <span class="fu">t</span>(heatmap_data)</span>
<span id="cb205-37"><a href="#cb205-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Replace any NAs with 0</span></span>
<span id="cb205-38"><a href="#cb205-38" aria-hidden="true" tabindex="-1"></a>  <span class="co">#heatmap[is.na(heatmap)] &lt;- 0</span></span>
<span id="cb205-39"><a href="#cb205-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(heatmap)</span>
<span id="cb205-40"><a href="#cb205-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb205-41"><a href="#cb205-41" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> <span class="fu">map</span>(df_nest<span class="sc">$</span>data, get_heatmap_data) <span class="sc">|&gt;</span></span>
<span id="cb205-42"><a href="#cb205-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste0</span>(df_nest<span class="sc">$</span>celltype_level, <span class="st">"_"</span>, df_nest<span class="sc">$</span>percent))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(heatmap_data<span class="sc">$</span><span class="st">`</span><span class="at">level1_EUROPEUKBB_35k10k_top-five-percent</span><span class="st">`</span>, <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>, </span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"Reds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>The T-cell results are interesting. Thereâ€™re 903 T-cells in the controls (level 1) and 496 <code>T-cell-mixed</code>, 326 <code>T-cell-vascular</code>, 63 <code>T-cellparenchymal-1</code> and 18 <code>T-cell-parenchymal-2</code>.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a>make_heatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(heatmap_data, plot_title) {</span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define significance thresholds</span></span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a>  sig_threshold1 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>) <span class="co"># Corresponds to p &lt; 0.05</span></span>
<span id="cb207-4"><a href="#cb207-4" aria-hidden="true" tabindex="-1"></a>  sig_threshold2 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>) <span class="co"># Corresponds to p &lt; 0.01</span></span>
<span id="cb207-5"><a href="#cb207-5" aria-hidden="true" tabindex="-1"></a>  sig_threshold3 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.001</span>) <span class="co"># Corresponds to p &lt; 0.001</span></span>
<span id="cb207-6"><a href="#cb207-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb207-7"><a href="#cb207-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create an annotation matrix with the same dimensions as 'heatmap'</span></span>
<span id="cb207-8"><a href="#cb207-8" aria-hidden="true" tabindex="-1"></a>  annotation_matrix <span class="ot">&lt;-</span></span>
<span id="cb207-9"><a href="#cb207-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">matrix</span>(</span>
<span id="cb207-10"><a href="#cb207-10" aria-hidden="true" tabindex="-1"></a>      <span class="st">""</span>,</span>
<span id="cb207-11"><a href="#cb207-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">nrow =</span> <span class="fu">nrow</span>(heatmap_data),</span>
<span id="cb207-12"><a href="#cb207-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">ncol =</span> <span class="fu">ncol</span>(heatmap_data)</span>
<span id="cb207-13"><a href="#cb207-13" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb207-14"><a href="#cb207-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb207-15"><a href="#cb207-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Populate the annotation matrix with '*', '**', or '***' based on significance levels</span></span>
<span id="cb207-16"><a href="#cb207-16" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold3] <span class="ot">&lt;-</span></span>
<span id="cb207-17"><a href="#cb207-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"***"</span></span>
<span id="cb207-18"><a href="#cb207-18" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold2 <span class="sc">&amp;</span></span>
<span id="cb207-19"><a href="#cb207-19" aria-hidden="true" tabindex="-1"></a>                      heatmap_data <span class="sc">&lt;=</span> sig_threshold3] <span class="ot">&lt;-</span> <span class="st">"**"</span></span>
<span id="cb207-20"><a href="#cb207-20" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold1 <span class="sc">&amp;</span></span>
<span id="cb207-21"><a href="#cb207-21" aria-hidden="true" tabindex="-1"></a>                      heatmap_data <span class="sc">&lt;=</span> sig_threshold2] <span class="ot">&lt;-</span> <span class="st">"*"</span></span>
<span id="cb207-22"><a href="#cb207-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb207-23"><a href="#cb207-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define a color palette for the heatmap</span></span>
<span id="cb207-24"><a href="#cb207-24" aria-hidden="true" tabindex="-1"></a>  color_palette <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>)</span>
<span id="cb207-25"><a href="#cb207-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb207-26"><a href="#cb207-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the heatmap with customizations</span></span>
<span id="cb207-27"><a href="#cb207-27" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb207-28"><a href="#cb207-28" aria-hidden="true" tabindex="-1"></a>    heatmap_data,</span>
<span id="cb207-29"><a href="#cb207-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"-log10(p-value)"</span>,</span>
<span id="cb207-30"><a href="#cb207-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> color_palette,</span>
<span id="cb207-31"><a href="#cb207-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use the defined color palette</span></span>
<span id="cb207-32"><a href="#cb207-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb207-33"><a href="#cb207-33" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (annotation_matrix[i, j] <span class="sc">!=</span> <span class="st">""</span>) {</span>
<span id="cb207-34"><a href="#cb207-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(annotation_matrix[i, j], x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb207-35"><a href="#cb207-35" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb207-36"><a href="#cb207-36" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb207-37"><a href="#cb207-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the clustering method if necessary</span></span>
<span id="cb207-38"><a href="#cb207-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb207-39"><a href="#cb207-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb207-40"><a href="#cb207-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Show row and column names</span></span>
<span id="cb207-41"><a href="#cb207-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_row_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb207-42"><a href="#cb207-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb207-43"><a href="#cb207-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the size of row and column names</span></span>
<span id="cb207-44"><a href="#cb207-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>),</span>
<span id="cb207-45"><a href="#cb207-45" aria-hidden="true" tabindex="-1"></a>    <span class="co">#column_names_rot = 45,</span></span>
<span id="cb207-46"><a href="#cb207-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb207-47"><a href="#cb207-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_max_height =</span> <span class="fu">max_text_height</span>(<span class="fu">colnames</span>(heatmap_data),</span>
<span id="cb207-48"><a href="#cb207-48" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">500</span>)),</span>
<span id="cb207-49"><a href="#cb207-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_title =</span> plot_title</span>
<span id="cb207-50"><a href="#cb207-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a border around the cells</span></span>
<span id="cb207-51"><a href="#cb207-51" aria-hidden="true" tabindex="-1"></a>    <span class="co">#border_color = "grey50",</span></span>
<span id="cb207-52"><a href="#cb207-52" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Additional options like margins</span></span>
<span id="cb207-53"><a href="#cb207-53" aria-hidden="true" tabindex="-1"></a>    <span class="co">#bottom_annotation_height = unit(2, "cm"),</span></span>
<span id="cb207-54"><a href="#cb207-54" aria-hidden="true" tabindex="-1"></a>    <span class="co">#left_annotation_width = unit(2, "cm")</span></span>
<span id="cb207-55"><a href="#cb207-55" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb207-56"><a href="#cb207-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb207-57"><a href="#cb207-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(plot)</span>
<span id="cb207-58"><a href="#cb207-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb207-59"><a href="#cb207-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb207-60"><a href="#cb207-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-61"><a href="#cb207-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Get heatmaps</span></span>
<span id="cb207-62"><a href="#cb207-62" aria-hidden="true" tabindex="-1"></a>heatmaps <span class="ot">&lt;-</span></span>
<span id="cb207-63"><a href="#cb207-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(heatmap_data,</span>
<span id="cb207-64"><a href="#cb207-64" aria-hidden="true" tabindex="-1"></a>       <span class="fu">substr</span>(</span>
<span id="cb207-65"><a href="#cb207-65" aria-hidden="true" tabindex="-1"></a>         <span class="fu">names</span>(heatmap_data),</span>
<span id="cb207-66"><a href="#cb207-66" aria-hidden="true" tabindex="-1"></a>         <span class="at">start =</span> <span class="dv">8</span>,</span>
<span id="cb207-67"><a href="#cb207-67" aria-hidden="true" tabindex="-1"></a>         <span class="at">stop =</span> <span class="fu">nchar</span>(<span class="fu">names</span>(heatmap_data))</span>
<span id="cb207-68"><a href="#cb207-68" aria-hidden="true" tabindex="-1"></a>       ),</span>
<span id="cb207-69"><a href="#cb207-69" aria-hidden="true" tabindex="-1"></a>       make_heatmap)</span>
<span id="cb207-70"><a href="#cb207-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Print them</span></span>
<span id="cb207-71"><a href="#cb207-71" aria-hidden="true" tabindex="-1"></a>level1_hmap_list <span class="ot">&lt;-</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-one-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-five-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-ten-percent</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value),
-log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a>level2_hmap_list <span class="ot">&lt;-</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-one-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-five-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-ten-percent</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)
Warning: Heatmap/annotation names are duplicated: -log10(p-value),
-log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a>level1_hmap_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-12-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a>level2_hmap_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-12-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save them to files</span></span>
<span id="cb214-2"><a href="#cb214-2" aria-hidden="true" tabindex="-1"></a>save_heatmaps <span class="ot">&lt;-</span> <span class="cf">function</span>(heatmap, file_name) {</span>
<span id="cb214-3"><a href="#cb214-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct the full file path</span></span>
<span id="cb214-4"><a href="#cb214-4" aria-hidden="true" tabindex="-1"></a>  file_path <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb214-5"><a href="#cb214-5" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">paste0</span>(<span class="st">"magma_heatmap_"</span>, file_name, <span class="st">".png"</span>))</span>
<span id="cb214-6"><a href="#cb214-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb214-7"><a href="#cb214-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save the heatmap to a file</span></span>
<span id="cb214-8"><a href="#cb214-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(file_path, <span class="at">height =</span> <span class="dv">1200</span>, <span class="at">width =</span> <span class="dv">1000</span>)</span>
<span id="cb214-9"><a href="#cb214-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#svg(file_path, height = 15)</span></span>
<span id="cb214-10"><a href="#cb214-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(heatmap)</span>
<span id="cb214-11"><a href="#cb214-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb214-12"><a href="#cb214-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb214-13"><a href="#cb214-13" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(heatmaps, <span class="fu">names</span>(heatmaps), <span class="sc">~</span> <span class="fu">save_heatmaps</span>(.x, .y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`level1_top-five-percent`
png 
  2 

$`level1_top-one-percent`
png 
  2 

$`level1_top-ten-percent`
png 
  2 

$`level2_top-five-percent`
png 
  2 

$`level2_top-one-percent`
png 
  2 

$`level2_top-ten-percent`
png 
  2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level1.png"</span>), <span class="at">width =</span> <span class="dv">22</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb216-4"><a href="#cb216-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb216-5"><a href="#cb216-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb216-6"><a href="#cb216-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(level1_hmap_list)</span>
<span id="cb216-7"><a href="#cb216-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb216-8"><a href="#cb216-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb218"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb218-2"><a href="#cb218-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb218-3"><a href="#cb218-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level2.png"</span>), <span class="at">width =</span> <span class="dv">24</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb218-4"><a href="#cb218-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb218-5"><a href="#cb218-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb218-6"><a href="#cb218-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(level2_hmap_list)</span>
<span id="cb218-7"><a href="#cb218-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb218-8"><a href="#cb218-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb220"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb220-1"><a href="#cb220-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove scz data to make plot simpler</span></span>
<span id="cb220-2"><a href="#cb220-2" aria-hidden="true" tabindex="-1"></a>heatmap_data_no_scz <span class="ot">&lt;-</span> <span class="fu">map</span>(heatmap_data, <span class="sc">~</span> .x[, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_scz_"</span>, <span class="fu">colnames</span>(.x))])</span>
<span id="cb220-3"><a href="#cb220-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove top 1 percents</span></span>
<span id="cb220-4"><a href="#cb220-4" aria-hidden="true" tabindex="-1"></a>heatmap_data_no_scz <span class="ot">&lt;-</span> heatmap_data_no_scz[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"top-one-percent"</span>, <span class="fu">names</span>(heatmap_data_no_scz))]</span>
<span id="cb220-5"><a href="#cb220-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb220-6"><a href="#cb220-6" aria-hidden="true" tabindex="-1"></a>heatmaps <span class="ot">&lt;-</span></span>
<span id="cb220-7"><a href="#cb220-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(heatmap_data_no_scz,</span>
<span id="cb220-8"><a href="#cb220-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">substr</span>(</span>
<span id="cb220-9"><a href="#cb220-9" aria-hidden="true" tabindex="-1"></a>         <span class="fu">names</span>(heatmap_data_no_scz),</span>
<span id="cb220-10"><a href="#cb220-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">start =</span> <span class="dv">8</span>,</span>
<span id="cb220-11"><a href="#cb220-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">stop =</span> <span class="fu">nchar</span>(<span class="fu">names</span>(heatmap_data_no_scz))</span>
<span id="cb220-12"><a href="#cb220-12" aria-hidden="true" tabindex="-1"></a>       ),</span>
<span id="cb220-13"><a href="#cb220-13" aria-hidden="true" tabindex="-1"></a>       make_heatmap)</span>
<span id="cb220-14"><a href="#cb220-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Print them</span></span>
<span id="cb220-15"><a href="#cb220-15" aria-hidden="true" tabindex="-1"></a>level1_hmap_list <span class="ot">&lt;-</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-five-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-ten-percent</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a>level2_hmap_list <span class="ot">&lt;-</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-five-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-ten-percent</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb224"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1" aria-hidden="true" tabindex="-1"></a>level1_hmap_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-13-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a>level2_hmap_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-13-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-13-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb226-2"><a href="#cb226-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb226-3"><a href="#cb226-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level1_no_scz.png"</span>), <span class="at">width =</span> <span class="dv">18</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb226-4"><a href="#cb226-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb226-5"><a href="#cb226-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb226-6"><a href="#cb226-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(level1_hmap_list)</span>
<span id="cb226-7"><a href="#cb226-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb226-8"><a href="#cb226-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb228-2"><a href="#cb228-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb228-3"><a href="#cb228-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level2_no_scz.png"</span>), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb228-4"><a href="#cb228-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb228-5"><a href="#cb228-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb228-6"><a href="#cb228-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(level2_hmap_list)</span>
<span id="cb228-7"><a href="#cb228-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb228-8"><a href="#cb228-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<section id="subset-to-risk-of-interest" class="level2">
<h2 class="anchored" data-anchor-id="subset-to-risk-of-interest">Subset to risk of interest</h2>
<p>The Bellenguez all controls are the risk hits of interest for the paper, subset to just that</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a>subset_column <span class="ot">&lt;-</span> heatmap_data_no_scz<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-ten-percent</span><span class="st">`</span>[, <span class="st">"all_controls_Bellenguez_2022_35k10k"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(subset_column) <span class="ot">&lt;-</span> <span class="st">"Bellenguez_2022"</span></span>
<span id="cb230-3"><a href="#cb230-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(subset_column)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    Bellenguez_2022
Arterial                0.000000000
Arteriolar-SMC          0.003631949
Artirial-FB             0.000000000
Astrocyte-activated     0.000000000
Astrocyte-quiescent     0.000000000
Capillary               0.000000000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a>heatmap_for_paper <span class="ot">&lt;-</span> <span class="fu">make_heatmap</span>(subset_column, <span class="st">""</span>)</span>
<span id="cb232-2"><a href="#cb232-2" aria-hidden="true" tabindex="-1"></a>heatmap_for_paper</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="">
<figure class="figure">
<p class=""><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-14-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(heatmap_for_paper, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/magma_celltype_heatmap.qs"</span>))</span>
<span id="cb233-2"><a href="#cb233-2" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(subset_column, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/magma_celltype_heatmap_data.qs"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a>heatmap_data_nom <span class="ot">&lt;-</span> <span class="fu">map</span>(df_nest<span class="sc">$</span>data, get_heatmap_data, <span class="at">nominal_p =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste0</span>(df_nest<span class="sc">$</span>celltype_level, <span class="st">"_"</span>, df_nest<span class="sc">$</span>percent))</span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove scz data to make plot simpler</span></span>
<span id="cb234-4"><a href="#cb234-4" aria-hidden="true" tabindex="-1"></a>heatmap_data_no_scz <span class="ot">&lt;-</span> <span class="fu">map</span>(heatmap_data_nom, <span class="sc">~</span> .x[, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_scz_"</span>, <span class="fu">colnames</span>(.x))])</span>
<span id="cb234-5"><a href="#cb234-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove top 1 percents</span></span>
<span id="cb234-6"><a href="#cb234-6" aria-hidden="true" tabindex="-1"></a>heatmap_data_no_scz <span class="ot">&lt;-</span> heatmap_data_no_scz[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"top-one-percent"</span>, <span class="fu">names</span>(heatmap_data_no_scz))]</span>
<span id="cb234-7"><a href="#cb234-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb234-8"><a href="#cb234-8" aria-hidden="true" tabindex="-1"></a>heatmaps <span class="ot">&lt;-</span></span>
<span id="cb234-9"><a href="#cb234-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(heatmap_data_no_scz,</span>
<span id="cb234-10"><a href="#cb234-10" aria-hidden="true" tabindex="-1"></a>       <span class="fu">substr</span>(</span>
<span id="cb234-11"><a href="#cb234-11" aria-hidden="true" tabindex="-1"></a>         <span class="fu">names</span>(heatmap_data_no_scz),</span>
<span id="cb234-12"><a href="#cb234-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">start =</span> <span class="dv">8</span>,</span>
<span id="cb234-13"><a href="#cb234-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">stop =</span> <span class="fu">nchar</span>(<span class="fu">names</span>(heatmap_data_no_scz))</span>
<span id="cb234-14"><a href="#cb234-14" aria-hidden="true" tabindex="-1"></a>       ),</span>
<span id="cb234-15"><a href="#cb234-15" aria-hidden="true" tabindex="-1"></a>       make_heatmap)</span>
<span id="cb234-16"><a href="#cb234-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Print them</span></span>
<span id="cb234-17"><a href="#cb234-17" aria-hidden="true" tabindex="-1"></a>level1_hmap_list <span class="ot">&lt;-</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-five-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-ten-percent</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a>level2_hmap_list <span class="ot">&lt;-</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-five-percent</span><span class="st">`</span> <span class="sc">+</span> heatmaps<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-ten-percent</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a>level1_hmap_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-15-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a>level2_hmap_list</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-15-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-15-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level1_no_scz_nominal.png"</span>), <span class="at">width =</span> <span class="dv">18</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb240-4"><a href="#cb240-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb240-5"><a href="#cb240-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb240-6"><a href="#cb240-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(level1_hmap_list)</span>
<span id="cb240-7"><a href="#cb240-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb240-8"><a href="#cb240-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb242-3"><a href="#cb242-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level2_no_scz_nominal.png"</span>), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb242-4"><a href="#cb242-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb242-5"><a href="#cb242-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb242-6"><a href="#cb242-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(level2_hmap_list)</span>
<span id="cb242-7"><a href="#cb242-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb242-8"><a href="#cb242-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
</section>
<section id="vertical-heatmap-for-paper" class="level2">
<h2 class="anchored" data-anchor-id="vertical-heatmap-for-paper">Vertical heatmap for paper</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a>make_heatmap_v <span class="ot">&lt;-</span> <span class="cf">function</span>(heatmap_data, level) {</span>
<span id="cb244-2"><a href="#cb244-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define significance thresholds</span></span>
<span id="cb244-3"><a href="#cb244-3" aria-hidden="true" tabindex="-1"></a>  sig_threshold1 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>) <span class="co"># Corresponds to p &lt; 0.05</span></span>
<span id="cb244-4"><a href="#cb244-4" aria-hidden="true" tabindex="-1"></a>  sig_threshold2 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>) <span class="co"># Corresponds to p &lt; 0.01</span></span>
<span id="cb244-5"><a href="#cb244-5" aria-hidden="true" tabindex="-1"></a>  sig_threshold3 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.001</span>) <span class="co"># Corresponds to p &lt; 0.001</span></span>
<span id="cb244-6"><a href="#cb244-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb244-7"><a href="#cb244-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create an annotation matrix with the same dimensions as 'heatmap'</span></span>
<span id="cb244-8"><a href="#cb244-8" aria-hidden="true" tabindex="-1"></a>  annotation_matrix <span class="ot">&lt;-</span></span>
<span id="cb244-9"><a href="#cb244-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">matrix</span>(</span>
<span id="cb244-10"><a href="#cb244-10" aria-hidden="true" tabindex="-1"></a>      <span class="st">""</span>,</span>
<span id="cb244-11"><a href="#cb244-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">nrow =</span> <span class="fu">nrow</span>(heatmap_data),</span>
<span id="cb244-12"><a href="#cb244-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">ncol =</span> <span class="fu">ncol</span>(heatmap_data)</span>
<span id="cb244-13"><a href="#cb244-13" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb244-14"><a href="#cb244-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb244-15"><a href="#cb244-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Populate the annotation matrix with '*', '**', or '***' based on significance levels</span></span>
<span id="cb244-16"><a href="#cb244-16" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold3] <span class="ot">&lt;-</span></span>
<span id="cb244-17"><a href="#cb244-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"***"</span></span>
<span id="cb244-18"><a href="#cb244-18" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold2 <span class="sc">&amp;</span></span>
<span id="cb244-19"><a href="#cb244-19" aria-hidden="true" tabindex="-1"></a>                      heatmap_data <span class="sc">&lt;=</span> sig_threshold3] <span class="ot">&lt;-</span> <span class="st">"**"</span></span>
<span id="cb244-20"><a href="#cb244-20" aria-hidden="true" tabindex="-1"></a>  annotation_matrix[heatmap_data <span class="sc">&gt;</span> sig_threshold1 <span class="sc">&amp;</span></span>
<span id="cb244-21"><a href="#cb244-21" aria-hidden="true" tabindex="-1"></a>                      heatmap_data <span class="sc">&lt;=</span> sig_threshold2] <span class="ot">&lt;-</span> <span class="st">"*"</span></span>
<span id="cb244-22"><a href="#cb244-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb244-23"><a href="#cb244-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define a color palette for the heatmap</span></span>
<span id="cb244-24"><a href="#cb244-24" aria-hidden="true" tabindex="-1"></a>  <span class="co">#color_palette &lt;- colorRampPalette(c("blue", "white", "red"))(100)</span></span>
<span id="cb244-25"><a href="#cb244-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb244-26"><a href="#cb244-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the heatmap with customizations</span></span>
<span id="cb244-27"><a href="#cb244-27" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb244-28"><a href="#cb244-28" aria-hidden="true" tabindex="-1"></a>    heatmap_data,</span>
<span id="cb244-29"><a href="#cb244-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"-log10(p-value)"</span>,</span>
<span id="cb244-30"><a href="#cb244-30" aria-hidden="true" tabindex="-1"></a>    <span class="co">#col = color_palette,</span></span>
<span id="cb244-31"><a href="#cb244-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Use the defined color palette</span></span>
<span id="cb244-32"><a href="#cb244-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, width, height, fill) {</span>
<span id="cb244-33"><a href="#cb244-33" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (annotation_matrix[i, j] <span class="sc">!=</span> <span class="st">""</span>) {</span>
<span id="cb244-34"><a href="#cb244-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(annotation_matrix[i, j], x, y, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">fontsize =</span> <span class="dv">10</span>))</span>
<span id="cb244-35"><a href="#cb244-35" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb244-36"><a href="#cb244-36" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb244-37"><a href="#cb244-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the clustering method if necessary</span></span>
<span id="cb244-38"><a href="#cb244-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb244-39"><a href="#cb244-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_columns =</span> <span class="cn">TRUE</span>,</span>
<span id="cb244-40"><a href="#cb244-40" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Show row and column names</span></span>
<span id="cb244-41"><a href="#cb244-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_row_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb244-42"><a href="#cb244-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb244-43"><a href="#cb244-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Adjust the size of row and column names</span></span>
<span id="cb244-44"><a href="#cb244-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>),</span>
<span id="cb244-45"><a href="#cb244-45" aria-hidden="true" tabindex="-1"></a>    <span class="co">#column_names_rot = 45,</span></span>
<span id="cb244-46"><a href="#cb244-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>),</span>
<span id="cb244-47"><a href="#cb244-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">column_names_max_height =</span> <span class="fu">max_text_height</span>(<span class="fu">colnames</span>(heatmap_data),</span>
<span id="cb244-48"><a href="#cb244-48" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">500</span>)),</span>
<span id="cb244-49"><a href="#cb244-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">row_title =</span> level</span>
<span id="cb244-50"><a href="#cb244-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add a border around the cells</span></span>
<span id="cb244-51"><a href="#cb244-51" aria-hidden="true" tabindex="-1"></a>    <span class="co">#border_color = "grey50",</span></span>
<span id="cb244-52"><a href="#cb244-52" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Additional options like margins</span></span>
<span id="cb244-53"><a href="#cb244-53" aria-hidden="true" tabindex="-1"></a>    <span class="co">#bottom_annotation_height = unit(2, "cm"),</span></span>
<span id="cb244-54"><a href="#cb244-54" aria-hidden="true" tabindex="-1"></a>    <span class="co">#left_annotation_width = unit(2, "cm")</span></span>
<span id="cb244-55"><a href="#cb244-55" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb244-56"><a href="#cb244-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb244-57"><a href="#cb244-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(plot)</span>
<span id="cb244-58"><a href="#cb244-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb244-59"><a href="#cb244-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb244-60"><a href="#cb244-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb244-61"><a href="#cb244-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Get heatmaps</span></span>
<span id="cb244-62"><a href="#cb244-62" aria-hidden="true" tabindex="-1"></a>heatmap_data_no_filter <span class="ot">&lt;-</span> heatmap_data[<span class="fu">c</span>(<span class="st">"level1_top-ten-percent"</span>, <span class="st">"level2_top-ten-percent"</span>)]</span>
<span id="cb244-63"><a href="#cb244-63" aria-hidden="true" tabindex="-1"></a>heatmaps_to_join <span class="ot">&lt;-</span> <span class="fu">map2</span>(heatmap_data_no_filter, <span class="fu">c</span>(<span class="st">"Level 1"</span>, <span class="st">"Level 2"</span>), make_heatmap_v)</span>
<span id="cb244-64"><a href="#cb244-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Print them</span></span>
<span id="cb244-65"><a href="#cb244-65" aria-hidden="true" tabindex="-1"></a>heatmaps_to_join</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`level1_top-ten-percent`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-16-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`level2_top-ten-percent`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-16-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb247"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb247-1"><a href="#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb247-2"><a href="#cb247-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb247-3"><a href="#cb247-3" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb247-4"><a href="#cb247-4" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_level1_and_2.png"</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">10</span>, </span>
<span id="cb247-5"><a href="#cb247-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb247-6"><a href="#cb247-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-7"><a href="#cb247-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine heatmaps side-by-side and draw them</span></span>
<span id="cb247-8"><a href="#cb247-8" aria-hidden="true" tabindex="-1"></a>hlist <span class="ot">&lt;-</span> heatmaps_to_join<span class="sc">$</span><span class="st">`</span><span class="at">level1_top-ten-percent</span><span class="st">`</span> <span class="sc">%v%</span> heatmaps_to_join<span class="sc">$</span><span class="st">`</span><span class="at">level2_top-ten-percent</span><span class="st">`</span></span>
<span id="cb247-9"><a href="#cb247-9" aria-hidden="true" tabindex="-1"></a>heatmaps_to_join<span class="sc">$</span></span>
<span id="cb247-10"><a href="#cb247-10" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(hlist)</span>
<span id="cb247-11"><a href="#cb247-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-12"><a href="#cb247-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb247-13"><a href="#cb247-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
<section id="ppi-modules" class="level1">
<h1>PPI modules</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>ppi_modules<span class="sc">$</span>log_p <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(ppi_modules<span class="sc">$</span>p_adj)</span>
<span id="cb248-2"><a href="#cb248-2" aria-hidden="true" tabindex="-1"></a>ppi_modules <span class="ot">&lt;-</span> ppi_modules <span class="sc">|&gt;</span> </span>
<span id="cb248-3"><a href="#cb248-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb248-4"><a href="#cb248-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"pseudobulk"</span>, FULL_NAME) <span class="sc">~</span> <span class="st">"pseudobulk"</span>,</span>
<span id="cb248-5"><a href="#cb248-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grepl</span>(<span class="st">"sig_magma_sig_genes"</span>, group) <span class="sc">~</span> <span class="st">"sig_magma_sig_genes"</span>,</span>
<span id="cb248-6"><a href="#cb248-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">.default =</span> <span class="cn">NA</span></span>
<span id="cb248-7"><a href="#cb248-7" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb248-8"><a href="#cb248-8" aria-hidden="true" tabindex="-1"></a><span class="co"># remove noapoe and nowindow</span></span>
<span id="cb248-9"><a href="#cb248-9" aria-hidden="true" tabindex="-1"></a>ppi_modules <span class="ot">&lt;-</span> ppi_modules <span class="sc">|&gt;</span></span>
<span id="cb248-10"><a href="#cb248-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"noAPOE"</span>, group)) <span class="sc">|&gt;</span></span>
<span id="cb248-11"><a href="#cb248-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"nowindow"</span>, group))</span>
<span id="cb248-12"><a href="#cb248-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-13"><a href="#cb248-13" aria-hidden="true" tabindex="-1"></a>ppi_pseudobulk <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ppi_modules, type <span class="sc">==</span> <span class="st">"pseudobulk"</span>)</span>
<span id="cb248-14"><a href="#cb248-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove first part of string</span></span>
<span id="cb248-15"><a href="#cb248-15" aria-hidden="true" tabindex="-1"></a>string <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(ppi_pseudobulk<span class="sc">$</span>group, <span class="st">"ppi_sig_modules_"</span>, <span class="st">""</span>)</span>
<span id="cb248-16"><a href="#cb248-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the gwas background</span></span>
<span id="cb248-17"><a href="#cb248-17" aria-hidden="true" tabindex="-1"></a>ppi_pseudobulk<span class="sc">$</span>gwas_background <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*pseudobulk_(.*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb248-18"><a href="#cb248-18" aria-hidden="true" tabindex="-1"></a>ppi_pseudobulk<span class="sc">$</span>group <span class="ot">&lt;-</span> ppi_pseudobulk<span class="sc">$</span>gwas_background</span>
<span id="cb248-19"><a href="#cb248-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-20"><a href="#cb248-20" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> <span class="fu">get_heatmap_data</span>(ppi_pseudobulk)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Not much is significant in the adjusted pvaluesâ€¦ Iâ€™ll try the nominal ones as well.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get heatmaps</span></span>
<span id="cb249-2"><a href="#cb249-2" aria-hidden="true" tabindex="-1"></a>ppi_heatmap <span class="ot">&lt;-</span> <span class="fu">make_heatmap</span>(heatmap_data, <span class="st">"PPI modules - p_adj"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: You defined `cell_fun` for a heatmap with more than 100 rows or
columns, which might be very slow to draw. Consider to use the
vectorized version `layer_fun`.</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a>ppi_pseudobulk<span class="sc">$</span>log_p <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(ppi_pseudobulk<span class="sc">$</span>P)</span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> <span class="fu">get_heatmap_data</span>(ppi_pseudobulk)</span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a>ppi_heatmap_noadj <span class="ot">&lt;-</span> <span class="fu">make_heatmap</span>(heatmap_data, <span class="st">"PPI modules - p_nominal"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: You defined `cell_fun` for a heatmap with more than 100 rows or
columns, which might be very slow to draw. Consider to use the
vectorized version `layer_fun`.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a>ppi_heatmap_noadj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-20-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a>joined_plot <span class="ot">&lt;-</span> ppi_heatmap <span class="sc">+</span> ppi_heatmap_noadj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Heatmap/annotation names are duplicated: -log10(p-value)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a>joined_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-20-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11"><img src="20_magma_risk_results_analysis_files/figure-html/unnamed-chunk-20-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb257-2"><a href="#cb257-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb257-3"><a href="#cb257-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">"ppi_magma_heatmap.png"</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">'in'</span>, </span>
<span id="cb257-4"><a href="#cb257-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb257-5"><a href="#cb257-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb257-6"><a href="#cb257-6" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(joined_plot)</span>
<span id="cb257-7"><a href="#cb257-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb257-8"><a href="#cb257-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div></div>
<section id="magma-gene-pvals" class="level2">
<h2 class="anchored" data-anchor-id="magma-gene-pvals">MAGMA gene pvals</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>ppi_magma <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ppi_modules, type <span class="sc">==</span> <span class="st">"sig_magma_sig_genes"</span>)</span>
<span id="cb259-2"><a href="#cb259-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-3"><a href="#cb259-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove first part of string</span></span>
<span id="cb259-4"><a href="#cb259-4" aria-hidden="true" tabindex="-1"></a>string <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(ppi_magma<span class="sc">$</span>group, <span class="st">"ppi_sig_modules_"</span>, <span class="st">""</span>) <span class="sc">|&gt;</span></span>
<span id="cb259-5"><a href="#cb259-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"EUROPEUKBB"</span>, <span class="st">"AD"</span>)</span>
<span id="cb259-6"><a href="#cb259-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the gwas background</span></span>
<span id="cb259-7"><a href="#cb259-7" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>gwas_background <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*sig_magma_sig_genes_(.*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb259-8"><a href="#cb259-8" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>percent <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*(top-.*-percent).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, ppi_magma<span class="sc">$</span>FULL_NAME)</span>
<span id="cb259-9"><a href="#cb259-9" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>input_gene_list <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^([^_]*_){4}(.*)_top-.*$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">2"</span>, ppi_magma<span class="sc">$</span>FULL_NAME)</span>
<span id="cb259-10"><a href="#cb259-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-11"><a href="#cb259-11" aria-hidden="true" tabindex="-1"></a><span class="co"># get unique rows</span></span>
<span id="cb259-12"><a href="#cb259-12" aria-hidden="true" tabindex="-1"></a>ppi_magma <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(ppi_magma)</span>
<span id="cb259-13"><a href="#cb259-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-14"><a href="#cb259-14" aria-hidden="true" tabindex="-1"></a><span class="co"># remove rows where the variable doesn't match the group, not sure where these rows are coming from...</span></span>
<span id="cb259-15"><a href="#cb259-15" aria-hidden="true" tabindex="-1"></a>celltype <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"([^_]*)_.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, ppi_magma<span class="sc">$</span>VARIABLE)</span>
<span id="cb259-16"><a href="#cb259-16" aria-hidden="true" tabindex="-1"></a>ppi_magma <span class="ot">&lt;-</span> ppi_magma[<span class="fu">mapply</span>(grepl, celltype, ppi_magma<span class="sc">$</span>group),]</span>
<span id="cb259-17"><a href="#cb259-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb259-18"><a href="#cb259-18" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>group_no_module <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^(.*)_[^_]*$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, ppi_magma<span class="sc">$</span>VARIABLE)</span>
<span id="cb259-19"><a href="#cb259-19" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">paste0</span>(ppi_magma<span class="sc">$</span>gwas_background)</span>
<span id="cb259-20"><a href="#cb259-20" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>VARIABLE <span class="ot">&lt;-</span> <span class="fu">str_remove</span>(ppi_magma<span class="sc">$</span>VARIABLE, <span class="st">"sig-magma-sig-genes_"</span>) <span class="sc">|&gt;</span></span>
<span id="cb259-21"><a href="#cb259-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove everything between the 3 and 6th _</span></span>
<span id="cb259-22"><a href="#cb259-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"(^[^_]*_[^_]*_[^_]*)_[^_]*_[^_]*_[^_]*(_)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1</span><span class="sc">\\</span><span class="st">2"</span>) <span class="sc">|&gt;</span></span>
<span id="cb259-23"><a href="#cb259-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"EUROPEUKBB"</span>, <span class="st">"AD"</span>)</span>
<span id="cb259-24"><a href="#cb259-24" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(ppi_magma<span class="sc">$</span>VARIABLE, <span class="st">"^[^_]*"</span>)</span>
<span id="cb259-25"><a href="#cb259-25" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>module <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*_"</span>, <span class="st">""</span>, ppi_magma<span class="sc">$</span>VARIABLE)</span>
<span id="cb259-26"><a href="#cb259-26" aria-hidden="true" tabindex="-1"></a>ppi_magma<span class="sc">$</span>celltype_module <span class="ot">&lt;-</span> <span class="fu">paste0</span>(ppi_magma<span class="sc">$</span>celltype, <span class="st">"_"</span>, ppi_magma<span class="sc">$</span>module)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a>module_labels <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(</span>
<span id="cb260-2"><a href="#cb260-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb260-3"><a href="#cb260-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions"</span>,</span>
<span id="cb260-4"><a href="#cb260-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ppi_magma_sig_genes_module_labels.csv"</span></span>
<span id="cb260-5"><a href="#cb260-5" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb260-6"><a href="#cb260-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 81 Columns: 5
â”€â”€ Column specification â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
Delimiter: ","
chr (5): module_label, celltype, gwas, fraction, ontology

â„¹ Use `spec()` to retrieve the full column specification for this data.
â„¹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a>module_labels<span class="sc">$</span>module_function <span class="ot">&lt;-</span></span>
<span id="cb262-2"><a href="#cb262-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_extract</span>(module_labels<span class="sc">$</span>module_label, <span class="st">"^[^-]*"</span>)</span>
<span id="cb262-3"><a href="#cb262-3" aria-hidden="true" tabindex="-1"></a>module_labels<span class="sc">$</span>module <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".*-"</span>, <span class="st">""</span>, module_labels<span class="sc">$</span>module_label)</span>
<span id="cb262-4"><a href="#cb262-4" aria-hidden="true" tabindex="-1"></a>module_labels<span class="sc">$</span>celltype_module <span class="ot">&lt;-</span></span>
<span id="cb262-5"><a href="#cb262-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste0</span>(module_labels<span class="sc">$</span>celltype, <span class="st">"_"</span>, module_labels<span class="sc">$</span>module)</span>
<span id="cb262-6"><a href="#cb262-6" aria-hidden="true" tabindex="-1"></a>module_labels<span class="sc">$</span>percent <span class="ot">&lt;-</span> <span class="st">"top-ten-percent"</span></span>
<span id="cb262-7"><a href="#cb262-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb262-8"><a href="#cb262-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the module labels</span></span>
<span id="cb262-9"><a href="#cb262-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ppi_magma &lt;-</span></span>
<span id="cb262-10"><a href="#cb262-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::left_join(ppi_magma,</span></span>
<span id="cb262-11"><a href="#cb262-11" aria-hidden="true" tabindex="-1"></a><span class="co">#             module_labels,</span></span>
<span id="cb262-12"><a href="#cb262-12" aria-hidden="true" tabindex="-1"></a><span class="co">#             by = join_by(celltype, module, celltype_module, percent, input_gene_list == fraction))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Nest data on broad groups</span></span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a>df_nest <span class="ot">&lt;-</span> ppi_magma <span class="sc">|&gt;</span> </span>
<span id="cb263-3"><a href="#cb263-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(percent <span class="sc">==</span> <span class="st">"top-ten-percent"</span>) <span class="sc">|&gt;</span></span>
<span id="cb263-4"><a href="#cb263-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(gwas_background) <span class="sc">|&gt;</span></span>
<span id="cb263-5"><a href="#cb263-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype, input_gene_list) <span class="sc">|&gt;</span></span>
<span id="cb263-6"><a href="#cb263-6" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb264"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb264-1"><a href="#cb264-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb264-2"><a href="#cb264-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: true</span></span>
<span id="cb264-3"><a href="#cb264-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Without the filter above, there's these odd mismatches between the </span></span>
<span id="cb264-4"><a href="#cb264-4" aria-hidden="true" tabindex="-1"></a><span class="co"># VARIABLE and group columns for the magma gene data...</span></span>
<span id="cb264-5"><a href="#cb264-5" aria-hidden="true" tabindex="-1"></a><span class="fu">grepl</span>(<span class="st">"Perivascular-FB-KAZN2"</span>, ppi_magma<span class="sc">$</span>group[<span class="fu">which</span>(ppi_magma<span class="sc">$</span>VARIABLE <span class="sc">==</span> <span class="st">"Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_module1"</span>)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb265"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb265-1"><a href="#cb265-1" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> <span class="fu">map</span>(df_nest<span class="sc">$</span>data, get_heatmap_data) <span class="sc">|&gt;</span></span>
<span id="cb265-2"><a href="#cb265-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste0</span>(df_nest<span class="sc">$</span>celltype, <span class="st">"_"</span>, df_nest<span class="sc">$</span>input_gene_list))</span>
<span id="cb265-3"><a href="#cb265-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-4"><a href="#cb265-4" aria-hidden="true" tabindex="-1"></a>ppi_heatmap <span class="ot">&lt;-</span></span>
<span id="cb265-5"><a href="#cb265-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(heatmap_data,</span>
<span id="cb265-6"><a href="#cb265-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">str_remove</span>(</span>
<span id="cb265-7"><a href="#cb265-7" aria-hidden="true" tabindex="-1"></a>         <span class="fu">names</span>(heatmap_data),</span>
<span id="cb265-8"><a href="#cb265-8" aria-hidden="true" tabindex="-1"></a>         <span class="st">"_top-.*"</span>),</span>
<span id="cb265-9"><a href="#cb265-9" aria-hidden="true" tabindex="-1"></a>       make_heatmap_v)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb266"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb266-1"><a href="#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb266-2"><a href="#cb266-2" aria-hidden="true" tabindex="-1"></a>joined_plot_top10 <span class="ot">&lt;-</span> ppi_heatmap<span class="sc">$</span><span class="st">`</span><span class="at">all_controls_top-ten-percent</span><span class="st">`</span> <span class="sc">%v%</span> ppi_heatmap<span class="sc">$</span><span class="st">`</span><span class="at">controls_vascular_top-ten-percent</span><span class="st">`</span> <span class="sc">%v%</span> ppi_heatmap<span class="sc">$</span><span class="st">`</span><span class="at">controls_parenchymal_top-ten-percent</span><span class="st">`</span></span>
<span id="cb266-3"><a href="#cb266-3" aria-hidden="true" tabindex="-1"></a>joined_plot_top10</span>
<span id="cb266-4"><a href="#cb266-4" aria-hidden="true" tabindex="-1"></a>joined_plot_top5 <span class="ot">&lt;-</span> ppi_heatmap<span class="sc">$</span><span class="st">`</span><span class="at">all_controls_top-five-percent</span><span class="st">`</span> <span class="sc">%v%</span> ppi_heatmap<span class="sc">$</span><span class="st">`</span><span class="at">controls_vascular_top-five-percent</span><span class="st">`</span> <span class="sc">%v%</span> ppi_heatmap<span class="sc">$</span><span class="st">`</span><span class="at">controls_parenchymal_top-five-percent</span><span class="st">`</span></span>
<span id="cb266-5"><a href="#cb266-5" aria-hidden="true" tabindex="-1"></a>joined_plot_top5</span>
<span id="cb266-6"><a href="#cb266-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb266-7"><a href="#cb266-7" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb266-8"><a href="#cb266-8" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_ppi_magma_sig_genes_top5.png"</span>), <span class="at">width =</span> <span class="dv">18</span>, </span>
<span id="cb266-9"><a href="#cb266-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">10</span>, <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb266-10"><a href="#cb266-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb266-11"><a href="#cb266-11" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(joined_plot_top5)</span>
<span id="cb266-12"><a href="#cb266-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb266-13"><a href="#cb266-13" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb266-14"><a href="#cb266-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Open a graphics device to save the image (e.g., PNG format)</span></span>
<span id="cb266-15"><a href="#cb266-15" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures"</span>, <span class="st">"990_shared_figures"</span>, <span class="st">"001_magma"</span>, </span>
<span id="cb266-16"><a href="#cb266-16" aria-hidden="true" tabindex="-1"></a>               <span class="st">"magma_heatmap_ppi_magma_sig_genes_top10.png"</span>), <span class="at">width =</span> <span class="dv">18</span>, </span>
<span id="cb266-17"><a href="#cb266-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">height =</span> <span class="dv">10</span>, <span class="at">units =</span> <span class="st">'in'</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb266-18"><a href="#cb266-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Draw concatenated heatmaps</span></span>
<span id="cb266-19"><a href="#cb266-19" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(joined_plot_top10)</span>
<span id="cb266-20"><a href="#cb266-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Close the graphics device to save the image</span></span>
<span id="cb266-21"><a href="#cb266-21" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
<section id="magma-genes" class="level1">
<h1>MAGMA genes</h1>
<p>MAGMA also gives gene level data - letâ€™s look at the genes driving the significant celltypes. Iâ€™ll focus on the level 2s for now.</p>
<p>These files have a funny format, theyâ€™re broken up into gene set sections, so thereâ€™s a need to parse the first three header lines of each section and read in the subsequent data.</p>
<p>Note that the files Iâ€™m using as input here are already subset to significant gene sets after multiple testing by MAGMA itself.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb267"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List files</span></span>
<span id="cb267-2"><a href="#cb267-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(</span>
<span id="cb267-3"><a href="#cb267-3" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/results"</span>),</span>
<span id="cb267-4"><a href="#cb267-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">"*.sets.genes.out"</span>,</span>
<span id="cb267-5"><a href="#cb267-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">full.names =</span> <span class="cn">TRUE</span></span>
<span id="cb267-6"><a href="#cb267-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb267-7"><a href="#cb267-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-8"><a href="#cb267-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove PPI files</span></span>
<span id="cb267-9"><a href="#cb267-9" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> files[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"ppi_"</span>, files)]</span>
<span id="cb267-10"><a href="#cb267-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-11"><a href="#cb267-11" aria-hidden="true" tabindex="-1"></a>read_magma_gene_data <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb267-12"><a href="#cb267-12" aria-hidden="true" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">readLines</span>(file)</span>
<span id="cb267-13"><a href="#cb267-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb267-14"><a href="#cb267-14" aria-hidden="true" tabindex="-1"></a>  set_start_indices <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"^# _SET.*VARIABLE"</span>, lines)</span>
<span id="cb267-15"><a href="#cb267-15" aria-hidden="true" tabindex="-1"></a>  set_end_indices <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">which</span>(lines <span class="sc">==</span> <span class="st">""</span>)[<span class="sc">-</span><span class="dv">1</span>], <span class="fu">length</span>(lines))</span>
<span id="cb267-16"><a href="#cb267-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb267-17"><a href="#cb267-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define a function to parse a single section</span></span>
<span id="cb267-18"><a href="#cb267-18" aria-hidden="true" tabindex="-1"></a>  parse_section <span class="ot">&lt;-</span> <span class="cf">function</span>(start_idx, end_idx) {</span>
<span id="cb267-19"><a href="#cb267-19" aria-hidden="true" tabindex="-1"></a>    header_lines <span class="ot">&lt;-</span> lines[start_idx<span class="sc">:</span>(start_idx <span class="sc">+</span> <span class="dv">2</span>)]</span>
<span id="cb267-20"><a href="#cb267-20" aria-hidden="true" tabindex="-1"></a>    set_name <span class="ot">&lt;-</span></span>
<span id="cb267-21"><a href="#cb267-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">sub</span>(<span class="st">".*=</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*</span><span class="sc">\\</span><span class="st">(set</span><span class="sc">\\</span><span class="st">).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, header_lines[<span class="dv">1</span>])</span>
<span id="cb267-22"><a href="#cb267-22" aria-hidden="true" tabindex="-1"></a>    num_genes <span class="ot">&lt;-</span></span>
<span id="cb267-23"><a href="#cb267-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">".*=</span><span class="sc">\\</span><span class="st">s*(</span><span class="sc">\\</span><span class="st">d+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, header_lines[<span class="dv">2</span>]))</span>
<span id="cb267-24"><a href="#cb267-24" aria-hidden="true" tabindex="-1"></a>    p_value <span class="ot">&lt;-</span></span>
<span id="cb267-25"><a href="#cb267-25" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">".*P-VALUE =</span><span class="sc">\\</span><span class="st">s*([0-9.]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, header_lines[<span class="dv">3</span>]))</span>
<span id="cb267-26"><a href="#cb267-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb267-27"><a href="#cb267-27" aria-hidden="true" tabindex="-1"></a>    gene_data <span class="ot">&lt;-</span> lines[(start_idx <span class="sc">+</span> <span class="dv">3</span>)<span class="sc">:</span>end_idx]</span>
<span id="cb267-28"><a href="#cb267-28" aria-hidden="true" tabindex="-1"></a>    set_df <span class="ot">&lt;-</span></span>
<span id="cb267-29"><a href="#cb267-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">read.table</span>(</span>
<span id="cb267-30"><a href="#cb267-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">text =</span> gene_data,</span>
<span id="cb267-31"><a href="#cb267-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb267-32"><a href="#cb267-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">sep =</span> <span class="st">""</span>,</span>
<span id="cb267-33"><a href="#cb267-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">quote =</span> <span class="st">""</span>,</span>
<span id="cb267-34"><a href="#cb267-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">comment.char =</span> <span class="st">""</span></span>
<span id="cb267-35"><a href="#cb267-35" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">|&gt;</span></span>
<span id="cb267-36"><a href="#cb267-36" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">padj =</span> <span class="fu">p.adjust</span>(P, <span class="at">method =</span> <span class="st">"bonferroni"</span>))</span>
<span id="cb267-37"><a href="#cb267-37" aria-hidden="true" tabindex="-1"></a>    <span class="co">#set_df &lt;- read_table(file = gene_data)</span></span>
<span id="cb267-38"><a href="#cb267-38" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb267-39"><a href="#cb267-39" aria-hidden="true" tabindex="-1"></a>    set_df<span class="sc">$</span>SetName <span class="ot">&lt;-</span> set_name</span>
<span id="cb267-40"><a href="#cb267-40" aria-hidden="true" tabindex="-1"></a>    set_df<span class="sc">$</span>NumGenes <span class="ot">&lt;-</span> num_genes</span>
<span id="cb267-41"><a href="#cb267-41" aria-hidden="true" tabindex="-1"></a>    set_df<span class="sc">$</span>SetPValue <span class="ot">&lt;-</span> p_value</span>
<span id="cb267-42"><a href="#cb267-42" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb267-43"><a href="#cb267-43" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Make column names match</span></span>
<span id="cb267-44"><a href="#cb267-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(set_df)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"gene_set"</span></span>
<span id="cb267-45"><a href="#cb267-45" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Make sure chr is a character to account for "X"/"Y"</span></span>
<span id="cb267-46"><a href="#cb267-46" aria-hidden="true" tabindex="-1"></a>    set_df<span class="sc">$</span>CHR <span class="ot">&lt;-</span> <span class="fu">as.character</span>(set_df<span class="sc">$</span>CHR)</span>
<span id="cb267-47"><a href="#cb267-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb267-48"><a href="#cb267-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(set_df)</span>
<span id="cb267-49"><a href="#cb267-49" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb267-50"><a href="#cb267-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb267-51"><a href="#cb267-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use mapply to apply the function to each set of start and end indices</span></span>
<span id="cb267-52"><a href="#cb267-52" aria-hidden="true" tabindex="-1"></a>  sets_list <span class="ot">&lt;-</span></span>
<span id="cb267-53"><a href="#cb267-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mapply</span>(</span>
<span id="cb267-54"><a href="#cb267-54" aria-hidden="true" tabindex="-1"></a>      parse_section,</span>
<span id="cb267-55"><a href="#cb267-55" aria-hidden="true" tabindex="-1"></a>      <span class="at">start_idx =</span> set_start_indices,</span>
<span id="cb267-56"><a href="#cb267-56" aria-hidden="true" tabindex="-1"></a>      <span class="at">end_idx =</span> set_end_indices,</span>
<span id="cb267-57"><a href="#cb267-57" aria-hidden="true" tabindex="-1"></a>      <span class="at">SIMPLIFY =</span> <span class="cn">FALSE</span></span>
<span id="cb267-58"><a href="#cb267-58" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb267-59"><a href="#cb267-59" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list_rbind</span>()</span>
<span id="cb267-60"><a href="#cb267-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sets_list)</span>
<span id="cb267-61"><a href="#cb267-61" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb267-62"><a href="#cb267-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-63"><a href="#cb267-63" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">map</span>(files, read_magma_gene_data)</span>
<span id="cb267-64"><a href="#cb267-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-65"><a href="#cb267-65" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">are_equal</span>(<span class="fu">sum</span>(<span class="fu">map_dbl</span>(df, nrow) <span class="sc">==</span> <span class="dv">0</span>), <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb269"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb269-1"><a href="#cb269-1" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(<span class="fu">basename</span>(files), <span class="st">"_scz."</span>, <span class="st">"_scz_"</span>) <span class="sc">|&gt;</span></span>
<span id="cb269-2"><a href="#cb269-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"_gene_analysis"</span>, <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb269-3"><a href="#cb269-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*$"</span>, <span class="st">""</span>, .) <span class="sc">|&gt;</span></span>
<span id="cb269-4"><a href="#cb269-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"magma_celltypes_"</span>, <span class="st">""</span>)</span>
<span id="cb269-5"><a href="#cb269-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-6"><a href="#cb269-6" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">map2</span>(df, groups, <span class="sc">~</span> .x <span class="sc">|&gt;</span></span>
<span id="cb269-7"><a href="#cb269-7" aria-hidden="true" tabindex="-1"></a>       dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> .y)) <span class="sc">|&gt;</span></span>
<span id="cb269-8"><a href="#cb269-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span>
<span id="cb269-9"><a href="#cb269-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-10"><a href="#cb269-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the logfc part of the string</span></span>
<span id="cb269-11"><a href="#cb269-11" aria-hidden="true" tabindex="-1"></a>string <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(df<span class="sc">$</span>group, <span class="st">"logfc_threshold_"</span>, <span class="st">""</span>) <span class="sc">|&gt;</span></span>
<span id="cb269-12"><a href="#cb269-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"level2_celltypes_"</span>, <span class="st">""</span>)</span>
<span id="cb269-13"><a href="#cb269-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Use sub to replace the entire string with the captured group</span></span>
<span id="cb269-14"><a href="#cb269-14" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>input_gene_list <span class="ot">&lt;-</span> <span class="fu">sub</span>(pattern, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb269-15"><a href="#cb269-15" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>gwas_background <span class="ot">&lt;-</span> <span class="fu">sub</span>(pattern_2, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb269-16"><a href="#cb269-16" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>input_filter <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"logfc_threshold_"</span>, df<span class="sc">$</span>group), <span class="st">"logfc_filter"</span>, <span class="st">"no_filter"</span>)</span>
<span id="cb269-17"><a href="#cb269-17" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>celltype_level <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"level1"</span>, df<span class="sc">$</span>group), <span class="st">"level1"</span>, <span class="st">"level2"</span>)</span>
<span id="cb269-18"><a href="#cb269-18" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>percent <span class="ot">&lt;-</span> <span class="fu">regmatches</span>(string, <span class="fu">regexpr</span>(pattern_3, string))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [28]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb270"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter out scz no window and no APOE</span></span>
<span id="cb270-2"><a href="#cb270-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb270-3"><a href="#cb270-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"scz_|nowindow|noAPOE"</span>, gwas_background))</span>
<span id="cb270-4"><a href="#cb270-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-5"><a href="#cb270-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Get ID translator</span></span>
<span id="cb270-6"><a href="#cb270-6" aria-hidden="true" tabindex="-1"></a>id_translate <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/996_mouse_phenotype_inputs/mouse_ortholog_ids.tsv"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb270-7"><a href="#cb270-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(entrezgene_id, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 18170 Columns: 5
â”€â”€ Column specification â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
Delimiter: "\t"
chr (4): mouse_ensembl, mgi, ensembl_gene_id, gene
dbl (1): entrezgene_id

â„¹ Use `spec()` to retrieve the full column specification for this data.
â„¹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb272"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb272-1"><a href="#cb272-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add Gene symbols and entrezgene_ids</span></span>
<span id="cb272-2"><a href="#cb272-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb272-3"><a href="#cb272-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(id_translate, <span class="at">by =</span> <span class="fu">join_by</span>(GENE <span class="sc">==</span> entrezgene_id))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [29]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb273"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb273-1"><a href="#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get unmatched ID symbols</span></span>
<span id="cb273-2"><a href="#cb273-2" aria-hidden="true" tabindex="-1"></a>missing_ids <span class="ot">&lt;-</span> df<span class="sc">$</span>GENE[<span class="fu">is.na</span>(df<span class="sc">$</span>gene)] <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb273-3"><a href="#cb273-3" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb273-4"><a href="#cb273-4" aria-hidden="true" tabindex="-1"></a>mapped_genes <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'entrezgene_id'</span>, <span class="st">'ensembl_gene_id'</span>, <span class="st">'hgnc_symbol'</span>), </span>
<span id="cb273-5"><a href="#cb273-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">filters =</span> <span class="st">'entrezgene_id'</span>, </span>
<span id="cb273-6"><a href="#cb273-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">values =</span> missing_ids, </span>
<span id="cb273-7"><a href="#cb273-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">mart =</span> ensembl) <span class="sc">|&gt;</span></span>
<span id="cb273-8"><a href="#cb273-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>(entrezgene_id, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb273-9"><a href="#cb273-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-10"><a href="#cb273-10" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb273-11"><a href="#cb273-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(mapped_genes, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"GENE"</span> <span class="ot">=</span> <span class="st">"entrezgene_id"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb273-12"><a href="#cb273-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">ensembl_gene_id =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(ensembl_gene_id.x), ensembl_gene_id.y, ensembl_gene_id.x), <span class="at">hgnc_symbol =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(gene), hgnc_symbol, gene)) <span class="sc">|&gt;</span></span>
<span id="cb273-13"><a href="#cb273-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(gene, ensembl_gene_id.x, ensembl_gene_id.y, group, gene_set))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [30]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb274"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb274-1"><a href="#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save list for PPI and mouse phenotype analysis</span></span>
<span id="cb274-2"><a href="#cb274-2" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(df, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/results/sig_celltype_sig_genes.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Want to check that sig hits for AD are independent of the Microglia activated gene sets in particular.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [31]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb275"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1" aria-hidden="true" tabindex="-1"></a>ad_level2 <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb275-2"><a href="#cb275-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype_level <span class="sc">==</span> <span class="st">"level2"</span> <span class="sc">&amp;</span> gwas_background <span class="sc">==</span> <span class="st">"EUROPEUKBB_35k10k"</span> <span class="sc">&amp;</span> percent <span class="sc">==</span> <span class="st">"top-ten-percent"</span> <span class="sc">&amp;</span> input_gene_list <span class="sc">==</span> <span class="st">"all_controls"</span>)</span>
<span id="cb275-3"><a href="#cb275-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ad_level2<span class="sc">$</span>SetName)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
  Microglia-activated            Pericyte-2 Perivascular-FB-KAZN2 
                 1378                  1364                  1387 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb277"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the gene and Zscores columns per gene set</span></span>
<span id="cb277-2"><a href="#cb277-2" aria-hidden="true" tabindex="-1"></a>gene_zscores <span class="ot">&lt;-</span> ad_level2 <span class="sc">|&gt;</span></span>
<span id="cb277-3"><a href="#cb277-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb277-4"><a href="#cb277-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(GENE, ZSTAT, SetName) <span class="sc">|&gt;</span></span>
<span id="cb277-5"><a href="#cb277-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(SetName) <span class="sc">|&gt;</span></span>
<span id="cb277-6"><a href="#cb277-6" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span>
<span id="cb277-7"><a href="#cb277-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-8"><a href="#cb277-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Check number of genes in common quickly</span></span>
<span id="cb277-9"><a href="#cb277-9" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(gene_zscores<span class="sc">$</span>data[[<span class="dv">1</span>]]<span class="sc">$</span>GENE <span class="sc">%in%</span> gene_zscores<span class="sc">$</span>data[[<span class="dv">2</span>]]<span class="sc">$</span>GENE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 9</code></pre>
</div>
<div class="sourceCode cell-code" id="cb279"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb279-1"><a href="#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="co"># So there is some overlap</span></span>
<span id="cb279-2"><a href="#cb279-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Save data for conditional</span></span>
<span id="cb279-3"><a href="#cb279-3" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(gene_zscores<span class="sc">$</span>SetName, gene_zscores<span class="sc">$</span>data, <span class="sc">~</span> readr<span class="sc">::</span><span class="fu">write_tsv</span>(.y, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/gene_covars"</span>, <span class="fu">paste0</span>(.x, <span class="st">"_gene-covar.txt"</span>))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]
# A tibble: 35 Ã— 2
     GENE ZSTAT
    &lt;dbl&gt; &lt;dbl&gt;
 1   8672  3.63
 2   8440  3.56
 3    274  7.78
 4   9881  3.72
 5   2580  3.53
 6   3133  3.64
 7   3122  5.53
 8   3123  5.96
 9 222537  3.70
10 221895  3.61
# â„¹ 25 more rows

[[2]]
# A tibble: 23 Ã— 2
     GENE ZSTAT
    &lt;dbl&gt; &lt;dbl&gt;
 1   9270  5.06
 2   6868  4.51
 3   8440  3.56
 4    274  7.78
 5  64844  3.9 
 6 130026  4.84
 7  65065  4.19
 8 222537  3.70
 9 221895  3.61
10   1956  3.57
# â„¹ 13 more rows

[[3]]
# A tibble: 24 Ã— 2
     GENE ZSTAT
    &lt;dbl&gt; &lt;dbl&gt;
 1  10772  3.83
 2   9270  5.06
 3   8440  3.56
 4    274  7.78
 5 222537  3.70
 6  54664  4.61
 7 221895  3.61
 8   7586  5.30
 9     19  3.66
10  81619  6.95
# â„¹ 14 more rows</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [32]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb281"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb281-1"><a href="#cb281-1" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(ad_level2, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/sig_genelist"</span>, <span class="st">"all_sig_magma_set_genes.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [33]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb282"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb282-1"><a href="#cb282-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the number of times each gene appears in a group</span></span>
<span id="cb282-2"><a href="#cb282-2" aria-hidden="true" tabindex="-1"></a>gene_counts <span class="ot">&lt;-</span> ad_level2 <span class="sc">%&gt;%</span></span>
<span id="cb282-3"><a href="#cb282-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENE) <span class="sc">%&gt;%</span></span>
<span id="cb282-4"><a href="#cb282-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n_distinct</span>(SetName))</span>
<span id="cb282-5"><a href="#cb282-5" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span> gene_counts <span class="sc">|&gt;</span></span>
<span id="cb282-6"><a href="#cb282-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(count <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb282-7"><a href="#cb282-7" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span> ad_level2 <span class="sc">|&gt;</span></span>
<span id="cb282-8"><a href="#cb282-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">semi_join</span>(unique_genes, <span class="at">by =</span> <span class="st">"GENE"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<ul>
<li>fix code block</li>
</ul>
<div class="cell-container"><div class="cell-decorator"><pre>In [34]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb283"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb283-1"><a href="#cb283-1" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span> unique_genes <span class="sc">|&gt;</span></span>
<span id="cb283-2"><a href="#cb283-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(mapped_genes, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"GENE"</span> <span class="ot">=</span> <span class="st">"entrezgene_id"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb283-3"><a href="#cb283-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">ensembl_gene_id =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(ensembl_gene_id.x), </span>
<span id="cb283-4"><a href="#cb283-4" aria-hidden="true" tabindex="-1"></a>                                         ensembl_gene_id.y, ensembl_gene_id.x), </span>
<span id="cb283-5"><a href="#cb283-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">hgnc_symbol =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(hgnc_symbol.x), hgnc_symbol.y, </span>
<span id="cb283-6"><a href="#cb283-6" aria-hidden="true" tabindex="-1"></a>                                     hgnc_symbol.x)) <span class="sc">|&gt;</span></span>
<span id="cb283-7"><a href="#cb283-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(hgnc_symbol.y, hgnc_symbol.x, ensembl_gene_id.x, </span>
<span id="cb283-8"><a href="#cb283-8" aria-hidden="true" tabindex="-1"></a>                   ensembl_gene_id.y))</span>
<span id="cb283-9"><a href="#cb283-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-10"><a href="#cb283-10" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(unique_genes, </span>
<span id="cb283-11"><a href="#cb283-11" aria-hidden="true" tabindex="-1"></a>                 here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/sig_genelist"</span>, <span class="st">"unique_sig_magma_set_genes.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="conditional-anlaysis" class="level1">
<h1>Conditional anlaysis</h1>
<p>We want to be sure that the sig hits we get are not being driven by the same genes</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [35]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb284"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(</span>
<span id="cb284-2"><a href="#cb284-2" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/sig_genelist/results/"</span>),</span>
<span id="cb284-3"><a href="#cb284-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">"*.gsa.out"</span>,</span>
<span id="cb284-4"><a href="#cb284-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">full.names =</span> <span class="cn">TRUE</span></span>
<span id="cb284-5"><a href="#cb284-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb284-6"><a href="#cb284-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-7"><a href="#cb284-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Read data</span></span>
<span id="cb284-8"><a href="#cb284-8" aria-hidden="true" tabindex="-1"></a>parse_magma_set_data <span class="ot">&lt;-</span> <span class="cf">function</span>(file) {</span>
<span id="cb284-9"><a href="#cb284-9" aria-hidden="true" tabindex="-1"></a>  file_lines <span class="ot">&lt;-</span> <span class="fu">readLines</span>(file)</span>
<span id="cb284-10"><a href="#cb284-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the index of the line judt before the data starts</span></span>
<span id="cb284-11"><a href="#cb284-11" aria-hidden="true" tabindex="-1"></a>  data_starts_index <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"^# CONDITIONED_VARIABLES"</span>, file_lines)</span>
<span id="cb284-12"><a href="#cb284-12" aria-hidden="true" tabindex="-1"></a>  condition <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(file_lines[data_starts_index], <span class="st">"(?&lt;=# CONDITIONED_VARIABLES = ).*?(?= )"</span>)</span>
<span id="cb284-13"><a href="#cb284-13" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read_table</span>(file, <span class="at">skip =</span> data_starts_index, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) <span class="sc">|&gt;</span></span>
<span id="cb284-14"><a href="#cb284-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">condition =</span> condition)</span>
<span id="cb284-15"><a href="#cb284-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(data) </span>
<span id="cb284-16"><a href="#cb284-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb284-17"><a href="#cb284-17" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">map</span>(files, parse_magma_set_data) <span class="sc">|&gt;</span></span>
<span id="cb284-18"><a href="#cb284-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span>
<span id="cb284-19"><a href="#cb284-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-20"><a href="#cb284-20" aria-hidden="true" tabindex="-1"></a><span class="co"># filter to conditioned celltypes</span></span>
<span id="cb284-21"><a href="#cb284-21" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> data <span class="sc">|&gt;</span></span>
<span id="cb284-22"><a href="#cb284-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(VARIABLE <span class="sc">%in%</span> <span class="fu">unique</span>(condition) <span class="sc">&amp;</span> VARIABLE <span class="sc">!=</span> condition) <span class="sc">|&gt;</span></span>
<span id="cb284-23"><a href="#cb284-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(condition) <span class="sc">|&gt;</span></span>
<span id="cb284-24"><a href="#cb284-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">padj =</span> <span class="fu">p.adjust</span>(P,<span class="at">method =</span> <span class="st">"bonferroni"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb284-25"><a href="#cb284-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(TYPE, FULL_NAME)) <span class="sc">|&gt;</span></span>
<span id="cb284-26"><a href="#cb284-26" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb284-27"><a href="#cb284-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-28"><a href="#cb284-28" aria-hidden="true" tabindex="-1"></a>gt<span class="sc">::</span><span class="fu">gt</span>(df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="cvwumlabye" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#cvwumlabye table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#cvwumlabye thead, #cvwumlabye tbody, #cvwumlabye tfoot, #cvwumlabye tr, #cvwumlabye td, #cvwumlabye th {
  border-style: none;
}

#cvwumlabye p {
  margin: 0;
  padding: 0;
}

#cvwumlabye .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#cvwumlabye .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#cvwumlabye .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#cvwumlabye .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#cvwumlabye .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cvwumlabye .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cvwumlabye .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cvwumlabye .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#cvwumlabye .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#cvwumlabye .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#cvwumlabye .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#cvwumlabye .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#cvwumlabye .gt_spanner_row {
  border-bottom-style: hidden;
}

#cvwumlabye .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#cvwumlabye .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#cvwumlabye .gt_from_md > :first-child {
  margin-top: 0;
}

#cvwumlabye .gt_from_md > :last-child {
  margin-bottom: 0;
}

#cvwumlabye .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#cvwumlabye .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#cvwumlabye .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#cvwumlabye .gt_row_group_first td {
  border-top-width: 2px;
}

#cvwumlabye .gt_row_group_first th {
  border-top-width: 2px;
}

#cvwumlabye .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cvwumlabye .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#cvwumlabye .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#cvwumlabye .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cvwumlabye .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cvwumlabye .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#cvwumlabye .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#cvwumlabye .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#cvwumlabye .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cvwumlabye .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cvwumlabye .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#cvwumlabye .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cvwumlabye .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#cvwumlabye .gt_left {
  text-align: left;
}

#cvwumlabye .gt_center {
  text-align: center;
}

#cvwumlabye .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#cvwumlabye .gt_font_normal {
  font-weight: normal;
}

#cvwumlabye .gt_font_bold {
  font-weight: bold;
}

#cvwumlabye .gt_font_italic {
  font-style: italic;
}

#cvwumlabye .gt_super {
  font-size: 65%;
}

#cvwumlabye .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#cvwumlabye .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#cvwumlabye .gt_indent_1 {
  text-indent: 5px;
}

#cvwumlabye .gt_indent_2 {
  text-indent: 10px;
}

#cvwumlabye .gt_indent_3 {
  text-indent: 15px;
}

#cvwumlabye .gt_indent_4 {
  text-indent: 20px;
}

#cvwumlabye .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_col_headings">
<th id="VARIABLE" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">VARIABLE</th>
<th id="MODEL" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">MODEL</th>
<th id="NGENES" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">NGENES</th>
<th id="BETA" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">BETA</th>
<th id="BETA_STD" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">BETA_STD</th>
<th id="SE" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">SE</th>
<th id="P" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">P</th>
<th id="padj" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">padj</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd gt_group_heading_row">
<td colspan="8" id="Microglia-activated" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Microglia-activated</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Microglia-activated  VARIABLE">Pericyte-2</td>
<td class="gt_row gt_right" headers="Microglia-activated  MODEL">30</td>
<td class="gt_row gt_right" headers="Microglia-activated  NGENES">1364</td>
<td class="gt_row gt_right" headers="Microglia-activated  BETA">0.077694</td>
<td class="gt_row gt_right" headers="Microglia-activated  BETA_STD">0.020343</td>
<td class="gt_row gt_right" headers="Microglia-activated  SE">0.028914</td>
<td class="gt_row gt_right" headers="Microglia-activated  P">0.0036082</td>
<td class="gt_row gt_right" headers="Microglia-activated  padj">0.0072164</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Microglia-activated  VARIABLE">Perivascular-FB-KAZN2</td>
<td class="gt_row gt_right" headers="Microglia-activated  MODEL">34</td>
<td class="gt_row gt_right" headers="Microglia-activated  NGENES">1387</td>
<td class="gt_row gt_right" headers="Microglia-activated  BETA">0.071418</td>
<td class="gt_row gt_right" headers="Microglia-activated  BETA_STD">0.018844</td>
<td class="gt_row gt_right" headers="Microglia-activated  SE">0.028016</td>
<td class="gt_row gt_right" headers="Microglia-activated  P">0.0054035</td>
<td class="gt_row gt_right" headers="Microglia-activated  padj">0.0108070</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="8" id="Pericyte-2" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Pericyte-2</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="Pericyte-2  VARIABLE">Microglia-activated</td>
<td class="gt_row gt_right" headers="Pericyte-2  MODEL">23</td>
<td class="gt_row gt_right" headers="Pericyte-2  NGENES">1378</td>
<td class="gt_row gt_right" headers="Pericyte-2  BETA">0.076622</td>
<td class="gt_row gt_right" headers="Pericyte-2  BETA_STD">0.020157</td>
<td class="gt_row gt_right" headers="Pericyte-2  SE">0.027658</td>
<td class="gt_row gt_right" headers="Pericyte-2  P">0.0028036</td>
<td class="gt_row gt_right" headers="Pericyte-2  padj">0.0056072</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Pericyte-2  VARIABLE">Perivascular-FB-KAZN2</td>
<td class="gt_row gt_right" headers="Pericyte-2  MODEL">34</td>
<td class="gt_row gt_right" headers="Pericyte-2  NGENES">1387</td>
<td class="gt_row gt_right" headers="Pericyte-2  BETA">0.062137</td>
<td class="gt_row gt_right" headers="Pericyte-2  BETA_STD">0.016396</td>
<td class="gt_row gt_right" headers="Pericyte-2  SE">0.029745</td>
<td class="gt_row gt_right" headers="Pericyte-2  P">0.0183630</td>
<td class="gt_row gt_right" headers="Pericyte-2  padj">0.0367260</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="8" id="Perivascular-FB-KAZN2" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Perivascular-FB-KAZN2</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Perivascular-FB-KAZN2  VARIABLE">Microglia-activated</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  MODEL">23</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  NGENES">1378</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  BETA">0.079179</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  BETA_STD">0.020830</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  SE">0.027505</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  P">0.0019992</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  padj">0.0039984</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Perivascular-FB-KAZN2  VARIABLE">Pericyte-2</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  MODEL">31</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  NGENES">1364</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  BETA">0.072873</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  BETA_STD">0.019081</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  SE">0.030528</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  P">0.0084978</td>
<td class="gt_row gt_right" headers="Perivascular-FB-KAZN2  padj">0.0169956</td>
</tr>
</tbody>
</table>

</div>
</div>
<div class="sourceCode cell-code" id="cb285"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb285-1"><a href="#cb285-1" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(df, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/994_magma_inputs/sig_genelist/conditional_adjusted_pvals.tsv"</span>))</span>
<span id="cb285-2"><a href="#cb285-2" aria-hidden="true" tabindex="-1"></a><span class="co"># nothing is sig after correction with all celltypes it seems</span></span>
<span id="cb285-3"><a href="#cb285-3" aria-hidden="true" tabindex="-1"></a>data <span class="sc">|&gt;</span></span>
<span id="cb285-4"><a href="#cb285-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(VARIABLE <span class="sc">!=</span> condition) <span class="sc">|&gt;</span></span>
<span id="cb285-5"><a href="#cb285-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(condition) <span class="sc">|&gt;</span></span>
<span id="cb285-6"><a href="#cb285-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">padj =</span> <span class="fu">p.adjust</span>(P,<span class="at">method =</span> <span class="st">"bonferroni"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb285-7"><a href="#cb285-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 0 Ã— 11
# Groups:   condition [0]
# â„¹ 11 variables: VARIABLE &lt;chr&gt;, TYPE &lt;chr&gt;, MODEL &lt;dbl&gt;, NGENES &lt;dbl&gt;,
#   BETA &lt;dbl&gt;, BETA_STD &lt;dbl&gt;, SE &lt;dbl&gt;, P &lt;dbl&gt;, FULL_NAME &lt;chr&gt;,
#   condition &lt;chr&gt;, padj &lt;dbl&gt;</code></pre>
</div>
</div></div>
<p>All the sig hits are still significant after conditioning on each others gene sets and adjusting the pval</p>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","openEffect":"zoom","descPosition":"bottom","selector":".lightbox","loop":false});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>