---
title: MAGMA - risk analysis
execute:
  eval: false
#self-contained: true
code-fold: true
bibliography: references.bib
---

```{r}
here::i_am("04_data_analysis/007_magma/19_magma_risk_analysis.qmd")
```

# Load packages

```{r}
source(here::here("04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"))
```

Ata gave me the Bellenguez GWAS data without UKbiobank proxy cases, but I need to process the data to be inline with the MAGMA format.
The SNP IDs in the annotation file are in a rs<id_num> format, so I also need to map the SNPs to those using the 37 build snpdb that can be found [here](https://ftp.ncbi.nih.gov/snp/latest_release/VCF/)

Note that this file had a annoying 37 lines of header info that `arrow` couldn't deal with, and it's a large file so modifying it is a pain, I used this command for parallel compression:

```{sh}
#| eval: false
zcat GCF_000001405.25.gz | tail -n +38 | pigz > GCF_000001405.25_header_removed.gz
```

```{r}
x <-
  fread(
    here::here(
      "03_data/994_magma_inputs/EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40.tsv"
    )
  )
#x$PVALUE[which(x$PVALUE == 0)] <- 3e-324
# x <-
#   arrow::open_dataset(
#     here::here(
#       "03_data/994_magma_inputs/EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40.tsv"
#     ), format = "tsv"
#   )
x |> dplyr::select(PVALUE) |> collect() |> min()
x <- x |>
  dplyr::mutate(PVALUE = if_else(PVALUE == 0, 3e-324, PVALUE),
                chr_hg38 = gsub("chr", "", chr_hg38))
x |> dplyr::select(PVALUE) |> collect() |> min()
x |> dplyr::select(chr_hg38) |> collect() |> head() 

renames <- c(CHR = "chr_hg38", SNP = "MARKER_ID", BP = "position", A1 = "Allele1", A2 = "Allele2", Nca = "N_cases_meta", Nco = "N_controls_meta", P = "PVALUE")
x2 <-
  x |> dplyr::select(
    c(
      "chr_hg38",
      "MARKER_ID",
      "position",
      "Allele1",
      "Allele2",
      "N_cases_meta",
      "N_controls_meta",
      "PVALUE"
    )
  ) |>
  dplyr::rename(all_of(renames)) 

# Need to match the SNP IDs to the dbsnp rs IDs as that what's in the annotation file
#x2 <- x2 |>
 # separate_wider_delim(SNP, names = c("chromosome", "position", "basepair1", "basepair2"), delim = ":")
rm(x)

head(x2)
#snpdb <- fread(here::here("03_data/994_magma_inputs/GCF_000001405.25.gz"), skip = 38)
#snpdb <- arrow::read_tsv_arrow(here::here("03_data/994_magma_inputs/GCF_000001405.25.gz"), skip = 38, as_data_frame = FALSE)
db_snp_df <-
  arrow::open_dataset(
    here::here(
      "03_data/994_magma_inputs/GCF_000001405.25_header_removed.gz"
    ), format = "tsv"
  )

#db_snp_df <- db_snp_df |> dplyr::rename(CHRO = "#CHROM")

db_snp_df <- db_snp_df |>
  dplyr::select(`#CHROM`, POS, ID, REF, ALT) |>
  dplyr::mutate(CHR = sub("^NC_0{0,5}(\\d+).*", "\\1", `#CHROM`),
                VERSION = as.numeric(sub(".*\\.(\\d+)$", "\\1", `#CHROM`)))
#db_snp_df |> dplyr::select(CHR) |> collect() |> head() 
#db_snp_df |> dplyr::select(CHR) |> collect() |> unique()
#db_snp_df |> dplyr::select(CHRO) |> collect() |> unique()
# Create a mapping table
# chrom_mapping <- data.frame(
#   CHROM = paste0("NC_00000", 1:24, ifelse(1:24 < 10, ".10", ".9")),
#   CHR = c(1:22, "X", "Y")
# )

# Use the mapping table to translate the CHROM column in db_snp_df
#db_snp_df <- dplyr::left_join(db_snp_df, chrom_mapping, by = "CHROM")
#db_snp_df <- merge(db_snp_df, chrom_mapping, by = "CHROM")
#merged_df <- dplyr::left_join(x2, db_snp_df, by = dplyr::join_by(CHR, BP == POS, A1 == REF, A2 == ALT))
#snp_ids <- collect(merged_df)

# Try making a subset of the data
db_snp_df_sub <- db_snp_df |>
  dplyr::filter(POS %in% x2$BP)
db_snp_df_sub <- collect(db_snp_df_sub)

db_snp_df_sub <- db_snp_df_sub |>
  dplyr::group_by(CHR, POS) |>
  dplyr::filter(VERSION == max(VERSION)) |>
  dplyr::ungroup() |>
  dplyr::select(!c(`#CHROM`, VERSION)) |>
  unique()

merged_df <- dplyr::left_join(x2, db_snp_df_sub, by = dplyr::join_by(CHR, BP == POS, A1 == REF, A2 == ALT))
# if the number of rows isn't too much more than 21 million, then try to collect whole file into memory

#merged_df <- inner_join(your_snp_df, snpdb, by = c("chromosome", "position", "basepair"))

# x2 <- x2 |>
#   mutate(SNP = paste0("rs", as.numeric(str_extract(SNP, "(?<=:)[0-9]+(?=:)"))))
x2$N <- x2$Nca + x2$Nco
fwrite(
  x2,
  here::here(
    "03_data/994_magma_inputs/EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40_magma_format.tsv"
  ),
  sep = "\t",
  quote = FALSE
)
# write.table(x2, here::here(
#     "03_data/994_magma_inputs/EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40_magma_format.tsv"
#   ), quote = FALSE, sep = "\t", row.names = FALSE)

```

```{r}
#!/bin/bash
#SBATCH --job-name=MAGMA_step1
#SBATCH -p c_highmem_dri1
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --time=0-01:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=1
#SBATCH --array=1
#SBATCH --account=scw1329
#SBATCH --mail-type=END,FAIL
#SBATCH --mail-user=bernardo-harringtong@cardiff.ac.uk

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "  - SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "  - SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "  - SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "  - SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "Job in the array has:"
echo "    - SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "    - SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"

module purge
module load magma/1.10

WORK_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/994_magma_inputs/"
cd $WORK_DIR

# SNP file from the reference
SNP_Loc_File=${WORK_DIR}"g1000_eur.bim"
# Gene location file - make sure build is correct
Gene_Loc_File=${WORK_DIR}"NCBI37.3.gene.loc"
Output_Prefix="NCBI37_annotated_window_35k10k"

magma \
    --annotate window=35,10 \
    --snp-loc $SNP_Loc_File \
    --gene-loc $Gene_Loc_File \
    --out $Output_Prefix

Output_Prefix="NCBI37_annotated_nowindow"

magma \
    --annotate window=0\
    --snp-loc $SNP_Loc_File \
    --gene-loc $Gene_Loc_File \
    --out $Output_Prefix


echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
```

# Step 2

```{r}
#!/bin/bash
#SBATCH --job-name=MAGMA_step2
#SBATCH -p c_highmem_dri1
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --time=0-10:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=10
#SBATCH --array=1
#SBATCH --mem=120G
#SBATCH --account=scw1329
#SBATCH --mail-type=END,FAIL
#SBATCH --mail-user=bernardo-harringtong@cardiff.ac.uk

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "  - SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "  - SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "  - SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "  - SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "Job in the array has:"
echo "    - SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "    - SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"

module purge
module load magma/1.10
module load parallel/20210322

WORK_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/994_magma_inputs/"
cd $WORK_DIR

#### Without proxies and with APOE - 35k10k window

mkdir temp_annot_35k10k # make a temporary directory to host the intermediate files

# PLINK files from reference (.bed/.bim/.fam)
Data_File=${WORK_DIR}"g1000_eur"
# Output of step 1
Annot_File=${WORK_DIR}"NCBI37_annotated_window_35k10k.genes.annot"
# GWAS summary stats
# Ata has summary stats here: /scratch/scw1751/shared/Bellenguez_2022_sumstats/
# GCST90027158_buildGRCh38_noATGC.tsv is for build 38, but the MAGMA site has reference data for build 37 only, asking sam for link to 38
# Should try with and without APOE in summary stats
# Should also try with and without proxy cases, file: EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40_noATGC_GRCh37.tsv
SNP_Pval_File="/scratch/scw1751/shared/Bellenguez_2022_sumstats/EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40_noATGC_GRCh37.tsv"

Output_Prefix="EUROPE_noproxy_35k10k"

    # run magma in parallel, 10 threads in this case

parallel magma \
   --batch {} 10 \
   --bfile $Data_File \
   --gene-annot $Annot_File \
   --gene-model snp-wise=mean \
   --pval $SNP_Pval_File ncol=N \
   --out temp_annot_35k10k/$Output_Prefix \
::: {1..10}

# merge all intermediate files generated under the temp_annot files
# and send out for one single file set

magma \
   --merge temp_annot_35k10k/$Output_Prefix \
   --out temp_annot_35k10k/$Output_Prefix

# extract merged files for subsequent analysis

cp ./temp_annot_35k10k/$Output_Prefix.genes.* .

# remove the temporary directory

rm -r temp_annot_35k10k

#### Without proxies and with APOE - no window

mkdir temp_annot # make a temporary directory to host the intermediate files

Annot_File=${WORK_DIR}"NCBI37_annotated_nowindow.genes.annot"
SNP_Pval_File="/scratch/scw1751/shared/Bellenguez_2022_sumstats/EADB_release_Feb2020.meta_model_pcs.mac_info_20_GC_OFF.formatted.exc_perc_cases_50.het_5e-8.freq_amp_40_noATGC_GRCh37.tsv"
Output_Prefix="EUROPE_noproxy_nowindow"

# run magma in parallel, 10 threads in this case
parallel magma \
   --batch {} 10 \
   --bfile $Data_File \
   --gene-annot $Annot_File \
   --gene-model snp-wise=mean \
   --pval $SNP_Pval_File ncol=N \
   --out temp_annot/$Output_Prefix \
::: {1..10}

# merge all intermediate files generated under the temp_annot files
# and send out for one single file set

magma \
   --merge temp_annot/$Output_Prefix \
   --out temp_annot/$Output_Prefix

# extract merged files for subsequent analysis

cp ./temp_annot/$Output_Prefix.genes.* .

# remove the temporary directory

rm -r temp_annot


#### With proxies and APOE - 35k10k window

mkdir temp_annot_35k10k_nobb # make a temporary directory to host the intermediate files

Annot_File=${WORK_DIR}"NCBI37_annotated_window_35k10k.genes.annot"
SNP_Pval_File="/scratch/scw1751/shared/Bellenguez_2022_sumstats/GCST90027158_buildGRCh37_noATGC_forMAGMA.tsv"
Output_Prefix="EUROPEUKBB_35k10k"


# run magma in parallel, 10 threads in this case
parallel magma \
   --batch {} 10 \
   --bfile $Data_File \
   --gene-annot $Annot_File \
   --gene-model snp-wise=mean \
   --pval $SNP_Pval_File ncol=N \
   --out temp_annot_35k10k_nobb/$Output_Prefix \
::: {1..10}

# merge all intermediate files generated under the temp_annot files
# and send out for one single file set

magma \
   --merge temp_annot_35k10k_nobb/$Output_Prefix \
   --out temp_annot_35k10k_nobb/$Output_Prefix

# extract merged files for subsequent analysis

cp ./temp_annot_35k10k_nobb/$Output_Prefix.genes.* .

# remove the temporary directory

rm -r temp_annot_35k10k_nobb

#### With proxies and APOE - no window

mkdir temp_annot_nobb # make a temporary directory to host the intermediate files

Annot_File=${WORK_DIR}"NCBI37_annotated_nowindow.genes.annot"
SNP_Pval_File="/scratch/scw1751/shared/Bellenguez_2022_sumstats/GCST90027158_buildGRCh37_noATGC_forMAGMA.tsv"
Output_Prefix="EUROPEUKBB_nowindow"


# run magma in parallel, 10 threads in this case

parallel magma \
   --batch {} 10 \
   --bfile $Data_File \
   --gene-annot $Annot_File \
   --gene-model snp-wise=mean \
   --pval $SNP_Pval_File ncol=N \
   --out temp_annot_nobb/$Output_Prefix \
::: {1..10}

# merge all intermediate files generated under the temp_annot files
# and send out for one single file set

magma \
   --merge temp_annot_nobb/$Output_Prefix \
   --out temp_annot_nobb/$Output_Prefix

# extract merged files for subsequent analysis

cp ./temp_annot_nobb/$Output_Prefix.genes.* .

# remove the temporary directory

rm -r temp_annot_nobb

#### With proxies and without APOE - no window

mkdir temp_annot_nobb # make a temporary directory to host the intermediate files

Annot_File=${WORK_DIR}"NCBI37_annotated_nowindow.genes.annot"
SNP_Pval_File="/scratch/scw1751/shared/Bellenguez_2022_sumstats/GCST90027158_buildGRCh37_noATGC_noAPOE_forMAGMA.tsv"
Output_Prefix="EUROPEUKBB_nowindow_noAPOE"

# run magma in parallel, 10 threads in this case
parallel magma \
   --batch {} 10 \
   --bfile $Data_File \
   --gene-annot $Annot_File \
   --gene-model snp-wise=mean \
   --pval $SNP_Pval_File ncol=N \
   --out temp_annot_nobb/$Output_Prefix \
::: {1..10}

# merge all intermediate files generated under the temp_annot files
# and send out for one single file set

magma \
   --merge temp_annot_nobb/$Output_Prefix \
   --out temp_annot_nobb/$Output_Prefix

# extract merged files for subsequent analysis

cp ./temp_annot_nobb/$Output_Prefix.genes.* .

# remove the temporary directory

rm -r temp_annot_nobb

#### With proxies and without APOE - 35k10k window

mkdir temp_annot_nobb # make a temporary directory to host the intermediate files

Annot_File=${WORK_DIR}"NCBI37_annotated_window_35k10k.genes.annot"
SNP_Pval_File="/scratch/scw1751/shared/Bellenguez_2022_sumstats/GCST90027158_buildGRCh37_noATGC_noAPOE_forMAGMA.tsv"
Output_Prefix="EUROPEUKBB_35k10k_noAPOE"

# run magma in parallel, 10 threads in this case
parallel magma \
   --batch {} 10 \
   --bfile $Data_File \
   --gene-annot $Annot_File \
   --gene-model snp-wise=mean \
   --pval $SNP_Pval_File ncol=N \
   --out temp_annot_nobb/$Output_Prefix \
::: {1..10}

# merge all intermediate files generated under the temp_annot files
# and send out for one single file set

magma \
   --merge temp_annot_nobb/$Output_Prefix \
   --out temp_annot_nobb/$Output_Prefix

# extract merged files for subsequent analysis

cp ./temp_annot_nobb/$Output_Prefix.genes.* .

# remove the temporary directory

rm -r temp_annot_nobb

echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
```

# Step 3

```{r}
#!/bin/bash
#SBATCH --job-name=MAGMA_step3
#SBATCH -p c_highmem_dri1
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --time=0-20:00:00
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=10
#SBATCH --array=1
#SBATCH --account=scw1329
#SBATCH --mail-type=END,FAIL
#SBATCH --mail-user=bernardo-harringtong@cardiff.ac.uk

#####SBATCH --mem=120G

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "  - SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "  - SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "  - SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "  - SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "Job in the array has:"
echo "    - SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "    - SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"

module purge
module load magma/1.10

WORK_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/994_magma_inputs/"
cd $WORK_DIR

## TO BE UPDATED
Set_Annot_File=${WORK_DIR}"magma_celltypes_all_controls.tsv"

# Gene set per celltype analysis
Gene_Results_File=${WORK_DIR}"EUROPEUKBB_35k10k.genes.raw"
Output_Prefix="./results/level2_celltypes_EUROPEUKBB_35k10k"

magma \
 --gene-results $Gene_Results_File \
 --set-annot $Set_Annot_File \
 --out $Output_Prefix

# Assuming WORK_DIR is already set to the directory containing your files
export WORK_DIR

# Populate Set_Annot_Files array with files matching 'magma_celltypes_*'
Set_Annot_Files=(${WORK_DIR}/magma_celltypes_*)

# Populate Gene_Results_Files array with files matching '*.genes.raw'
Gene_Results_Files=(${WORK_DIR}/*.genes.raw)

# Function to run magma with given files and output prefix
run_magma() {
    local Set_Annot_File=$1
    local Gene_Results_File=$2
    local annot_name=$(basename "${Set_Annot_File}" .tsv)
    local gene_name=$(basename "${Gene_Results_File}" .genes.raw)
    local Output_Prefix="./results/level2_celltypes_${annot_name}_${gene_name}"

    # Run magma command
    magma \
        --gene-results $Gene_Results_File \
        --set-annot $Set_Annot_File \
        --out $Output_Prefix
}

# Export the function so it can be used by parallel
export -f run_magma

# Use parallel to run the combinations
parallel run_magma ::: "${Set_Annot_Files[@]}" ::: "${Gene_Results_Files[@]}"

# /workdir/atahualpa.castillomorales/tools/magma_v1.10_static/magma \
#  --gene-results $Gene_Results_File \
#  --set-annot $Set_Annot_File col=2,1 \
#  --out $Output_Prefix

Gene_Results_File=${WORK_DIR}"EUROPEUKBB_35k10k_noAPOE.genes.raw"
Output_Prefix="./results/level2_celltypes_EUROPEUKBB_35k10k_noAPOE"

magma \
 --gene-results $Gene_Results_File \
 --set-annot $Set_Annot_File \
 --out $Output_Prefix



Gene_Results_File=${WORK_DIR}"EUROPEUKBB_nowindow_noAPOE.genes.raw"
Output_Prefix="./results/level2_celltypes_EUROPEUKBB_nowindow_noAPOE"

magma \
 --gene-results $Gene_Results_File \
 --set-annot $Set_Annot_File \
 --out $Output_Prefix


echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
```
