<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.54">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>Protein-protein interactions</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "../../index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="../../site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
     <script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>  <link href="../../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="../../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>  
      </head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Protein-protein interactions</h6>

            <a href="../../04_data_analysis/006_protein-protein-interactions/17_protein-protein_interactions.qmd" class="btn btn-primary quarto-download-embed" download="17_protein-protein_interactions.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Protein-protein interactions</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions">Functions</a></li>
  <li><a href="#read-in-data" id="toc-read-in-data" class="nav-link" data-scroll-target="#read-in-data">Read in data</a></li>
  <li><a href="#data-prep" id="toc-data-prep" class="nav-link" data-scroll-target="#data-prep">Data prep</a>
  <ul class="collapse">
  <li><a href="#magma-genes" id="toc-magma-genes" class="nav-link" data-scroll-target="#magma-genes">MAGMA genes</a></li>
  </ul></li>
  <li><a href="#ppi-test-1.-links-within-gene-set" id="toc-ppi-test-1.-links-within-gene-set" class="nav-link" data-scroll-target="#ppi-test-1.-links-within-gene-set">PPI Test 1. Links within gene set</a>
  <ul class="collapse">
  <li><a href="#background-gene-list" id="toc-background-gene-list" class="nav-link" data-scroll-target="#background-gene-list">Background gene list</a></li>
  <li><a href="#design" id="toc-design" class="nav-link" data-scroll-target="#design">Design</a>
  <ul class="collapse">
  <li><a href="#choose-ppi-network" id="toc-choose-ppi-network" class="nav-link" data-scroll-target="#choose-ppi-network">Choose PPI network</a></li>
  </ul></li>
  <li><a href="#background-list" id="toc-background-list" class="nav-link" data-scroll-target="#background-list">Background list</a></li>
  <li><a href="#gene-lists-for-analysis" id="toc-gene-lists-for-analysis" class="nav-link" data-scroll-target="#gene-lists-for-analysis">Gene lists for analysis</a></li>
  <li><a href="#subnetwork-plot" id="toc-subnetwork-plot" class="nav-link" data-scroll-target="#subnetwork-plot">Subnetwork plot</a></li>
  <li><a href="#test-for-number-of-interactions" id="toc-test-for-number-of-interactions" class="nav-link" data-scroll-target="#test-for-number-of-interactions">Test for number of interactions</a>
  <ul class="collapse">
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a></li>
  </ul></li>
  <li><a href="#louvain-clustering" id="toc-louvain-clustering" class="nav-link" data-scroll-target="#louvain-clustering">Louvain clustering</a></li>
  <li><a href="#for-each-module" id="toc-for-each-module" class="nav-link" data-scroll-target="#for-each-module">For each module</a></li>
  <li><a href="#module-assignment" id="toc-module-assignment" class="nav-link" data-scroll-target="#module-assignment">Module assignment</a>
  <ul class="collapse">
  <li><a href="#go-enrichment" id="toc-go-enrichment" class="nav-link" data-scroll-target="#go-enrichment">GO enrichment</a></li>
  <li><a href="#save-for-magma" id="toc-save-for-magma" class="nav-link" data-scroll-target="#save-for-magma">Save for MAGMA</a></li>
  <li><a href="#treeplots" id="toc-treeplots" class="nav-link" data-scroll-target="#treeplots">Treeplots</a></li>
  <li><a href="#module-labels" id="toc-module-labels" class="nav-link" data-scroll-target="#module-labels">Module labels</a></li>
  </ul></li>
  <li><a href="#plot-labelled-modules" id="toc-plot-labelled-modules" class="nav-link" data-scroll-target="#plot-labelled-modules">Plot labelled modules</a>
  <ul class="collapse">
  <li><a href="#networkd3-graph" id="toc-networkd3-graph" class="nav-link" data-scroll-target="#networkd3-graph">networkD3 graph</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">i_am</span>(<span class="st">"04_data_analysis/006_protein-protein-interactions/17_protein-protein_interactions.qmd"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at /mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad</code></pre>
</div>
</div></div>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, smiths</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'kableExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    group_rows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'janitor'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    chisq.test, fisher.test</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    between, first, last</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    melt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rngtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: cluster</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  To enable shared memory capabilities, try: install.extras('
NMF
')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'future'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    run</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Some features are not enabled in this build of Arrow. Run `arrow_info()` for more information.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'arrow'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    type</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    timestamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'scales'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:readr':

    col_factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    discard</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggpubr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    get_legend</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gridExtra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    combine</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'viridis'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scales':

    viridis_pal</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'pals'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridis':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:viridisLite':

    cividis, inferno, magma, plasma, turbo, viridis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 methods overwritten by 'proxy':
  method               from    
  print.registry_field registry
  print.registry_entry registry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    count</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    nrun</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:data.table':

    first, second</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    expand</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    first, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:reshape':

    expand, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:webshot2':

    resize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:data.table':

    shift</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    reduce</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    collapse, desc, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built under R 4.4.0 but the current version is
4.4.1; it is recomended that you reinstall 'SeuratObject' as the ABI
for R may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DT':

    JS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'DESeq2'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    fpkm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: limma</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:DESeq2':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    plotMDS</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'edgeR'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SingleCellExperiment':

    cpm</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocParallel'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:NMF':

    register</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationDbi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationDbi'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>##############################################################################
Pathview is an open source software package distributed under GNU General
Public License version 3 (GPLv3). Details of GPLv3 is available at
http://www.gnu.org/licenses/gpl-3.0.html. Particullary, users are required to
formally cite the original Pathview paper (not just mention it) in publications
or products. For details, do citation("pathview") within R.

The pathview downloads and uses KEGG data. Non-academic uses may require a KEGG
license agreement (details at http://www.kegg.jp/kegg/legal.html).
##############################################################################</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicFeatures</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: AnnotationFilter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationFilter'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:arrow':

    field</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:future':

    value</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ensembldb'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:dplyr':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocFileCache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: dbplyr</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dbplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    ident, sql</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'AnnotationHub'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggrepel</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'PCAtools'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scran':

    parallelPCA</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    biplot, screeplot</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'igraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Seurat':

    components</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:scater':

    normalize</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:future':

    %-&gt;%, %&lt;-%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:NMF':

    algorithm, compare</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:BiocGenerics':

    normalize, path, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:purrr':

    compose, simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tibble':

    as_data_frame</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    crossing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:dplyr':

    as_data_frame, groups, union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    decompose, spectrum</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    union</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'biomaRt'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scRNAseq':

    listDatasets, searchDatasets</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('download.file.method.GEOquery'='auto')</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting options('GEOquery.inmemory.gpl'=FALSE)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>clusterProfiler v4.12.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use clusterProfiler in published research, please cite:
T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal enrichment tool for interpreting omics data. The Innovation. 2021, 2(3):100141</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'clusterProfiler'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    simplify</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>DOSE v3.30.1  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/

If you use DOSE in published research, please cite:
Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'enrichplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    color_palette</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'XVector'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    compact</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pals':

    alphabet</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grid':

    pattern</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocIO</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: rtracklayer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rtracklayer'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocIO':

    FileForFormat</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    blocks</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationHub':

    hubUrl</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh37'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'qusage'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:clusterProfiler':

    read.gmt</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: replacing previous import 'utils::findMatches' by
'S4Vectors::findMatches' when loading 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SNPlocs.Hsapiens.dbSNP155.GRCh38'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:SNPlocs.Hsapiens.dbSNP155.GRCh37':

    .loadAlleles, .loadLoc, COMPATIBLE_BSGENOMES</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rsamtools</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'GenomicFiles'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:tidyr':

    pack, unpack</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
simona version 1.2.0
Bioconductor page: http://bioconductor.org/packages/simona/
Github page: https://github.com/jokergoo/simona
Documentation: https://jokergoo.github.io/simona/

If you use it in published research, please cite:
Gu, Z. simona: a Comprehensive R package for Semantic Similarity 
  Analysis on Bio-Ontologies. bioRxiv 2023.

This message can be suppressed by:
  suppressPackageStartupMessages(library(simona))
========================================</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================
! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments
   in the original pheatmap() are identically supported in the new function. You 
   can still use the original function by explicitly calling pheatmap::pheatmap().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ComplexHeatmap'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:pheatmap':

    pheatmap</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'ggnetwork':
  method         from  
  fortify.igraph ggtree</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'CellChat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:scater':

    runPCA, runUMAP</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Rcpp</code></pre>
</div>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_config_set</span>(<span class="at">store =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_targets"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="functions" class="level1">
<h1>Functions</h1>
<p>Credit to Jimena for these functions, and indeed the basis for almost all the subsequent code.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Rescale data to range 0 to 1.</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>rescale01 <span class="ot">&lt;-</span> <span class="cf">function</span>(numvect){</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a>  rango <span class="ot">&lt;-</span> <span class="fu">range</span>(numvect)</span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a>  rescalado <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(numvect, <span class="cf">function</span>(x) <span class="cf">if</span>((x <span class="sc">-</span> rango[<span class="dv">1</span>]) <span class="sc">==</span> <span class="dv">0</span>) { <span class="dv">0</span> } <span class="cf">else</span> {(x <span class="sc">-</span> rango[<span class="dv">1</span>]) <span class="sc">/</span> (rango[<span class="dv">2</span>] <span class="sc">-</span> rango[<span class="dv">1</span>])}))</span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(rescalado)</span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-8"><a href="#cb199-8" aria-hidden="true" tabindex="-1"></a>similar_gene_list  <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_list, attr_matrix, <span class="at">rango =</span> <span class="dv">100</span>){</span>
<span id="cb199-9"><a href="#cb199-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># attr_matrix must have at least a column with ensembl_gene_id and the rownames ordered by the attribute</span></span>
<span id="cb199-10"><a href="#cb199-10" aria-hidden="true" tabindex="-1"></a>  d_gl <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">rownames</span>(attr_matrix)[attr_matrix<span class="sc">$</span>ensembl_gene_id <span class="sc">%in%</span> gene_list])</span>
<span id="cb199-11"><a href="#cb199-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Random sample of genes (most with a similar attribute)</span></span>
<span id="cb199-12"><a href="#cb199-12" aria-hidden="true" tabindex="-1"></a>  rango2 <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">seq</span>(<span class="sc">-</span>rango<span class="sc">/</span><span class="dv">2</span>, rango<span class="sc">/</span><span class="dv">2</span>, <span class="dv">1</span>), <span class="dv">0</span>)</span>
<span id="cb199-13"><a href="#cb199-13" aria-hidden="true" tabindex="-1"></a>  rnd_ind <span class="ot">&lt;-</span> <span class="fu">sample</span>(rango2, <span class="at">size =</span> <span class="fu">length</span>(gene_list), <span class="at">replace =</span> <span class="cn">TRUE</span>) </span>
<span id="cb199-14"><a href="#cb199-14" aria-hidden="true" tabindex="-1"></a>  rnd_ind <span class="ot">&lt;-</span> d_gl <span class="sc">+</span> rnd_ind</span>
<span id="cb199-15"><a href="#cb199-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># check the extremes (all should be within the range of D)</span></span>
<span id="cb199-16"><a href="#cb199-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">which</span>(rnd_ind <span class="sc">&lt;</span> <span class="dv">1</span>)) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb199-17"><a href="#cb199-17" aria-hidden="true" tabindex="-1"></a>    rnd_ind[<span class="fu">which</span>(rnd_ind <span class="sc">&lt;</span> <span class="dv">1</span>)] <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span>rango, <span class="dv">1</span>)</span>
<span id="cb199-18"><a href="#cb199-18" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb199-19"><a href="#cb199-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">which</span>(rnd_ind <span class="sc">&gt;</span> <span class="fu">nrow</span>(attr_matrix)))){</span>
<span id="cb199-20"><a href="#cb199-20" aria-hidden="true" tabindex="-1"></a>    rnd_ind[<span class="fu">which</span>(rnd_ind <span class="sc">&gt;</span> <span class="fu">nrow</span>(attr_matrix))] <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(attr_matrix)<span class="sc">:</span>(<span class="fu">nrow</span>(attr_matrix) <span class="sc">-</span> rango), <span class="dv">1</span>)</span>
<span id="cb199-21"><a href="#cb199-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb199-22"><a href="#cb199-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove duplicates and replace with random genes (to keep gene number constant)</span></span>
<span id="cb199-23"><a href="#cb199-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(rnd_ind) <span class="sc">!=</span> <span class="fu">length</span>(<span class="fu">unique</span>(rnd_ind))){</span>
<span id="cb199-24"><a href="#cb199-24" aria-hidden="true" tabindex="-1"></a>    rnd_ind <span class="ot">&lt;-</span> <span class="fu">unique</span>(rnd_ind)</span>
<span id="cb199-25"><a href="#cb199-25" aria-hidden="true" tabindex="-1"></a>    rnd_ind <span class="ot">&lt;-</span> <span class="fu">c</span>(rnd_ind, <span class="fu">sample</span>(<span class="fu">setdiff</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(attr_matrix), rnd_ind), <span class="at">size =</span> (<span class="fu">length</span>(gene_list) <span class="sc">-</span> <span class="fu">length</span>(rnd_ind))))</span>
<span id="cb199-26"><a href="#cb199-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb199-27"><a href="#cb199-27" aria-hidden="true" tabindex="-1"></a>  rnd_smpl <span class="ot">&lt;-</span> <span class="fu">as.character</span>(attr_matrix<span class="sc">$</span>ensembl_gene_id[rnd_ind])</span>
<span id="cb199-28"><a href="#cb199-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(rnd_smpl)</span>
<span id="cb199-29"><a href="#cb199-29" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb199-30"><a href="#cb199-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-31"><a href="#cb199-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Divide genes according to an attribute #</span></span>
<span id="cb199-32"><a href="#cb199-32" aria-hidden="true" tabindex="-1"></a>attribute_2_list20bins <span class="ot">&lt;-</span> <span class="cf">function</span>(attr_matrix, <span class="at">gene_col =</span> <span class="dv">1</span>, <span class="at">attr_col =</span> <span class="dv">2</span>, <span class="at">do_log =</span> <span class="cn">TRUE</span>){</span>
<span id="cb199-33"><a href="#cb199-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(do_log <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb199-34"><a href="#cb199-34" aria-hidden="true" tabindex="-1"></a>    at <span class="ot">&lt;-</span> <span class="fu">rescale01</span>(<span class="fu">log</span>(attr_matrix[, attr_col] <span class="sc">+</span> <span class="dv">1</span>))  </span>
<span id="cb199-35"><a href="#cb199-35" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span>{</span>
<span id="cb199-36"><a href="#cb199-36" aria-hidden="true" tabindex="-1"></a>    at <span class="ot">&lt;-</span> <span class="fu">rescale01</span>(attr_matrix[, attr_col])</span>
<span id="cb199-37"><a href="#cb199-37" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb199-38"><a href="#cb199-38" aria-hidden="true" tabindex="-1"></a>  bines <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, .<span class="dv">05</span>)</span>
<span id="cb199-39"><a href="#cb199-39" aria-hidden="true" tabindex="-1"></a>  gps <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb199-40"><a href="#cb199-40" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(k <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(bines) <span class="sc">-</span> <span class="dv">1</span>))){</span>
<span id="cb199-41"><a href="#cb199-41" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(k <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb199-42"><a href="#cb199-42" aria-hidden="true" tabindex="-1"></a>      gps[[k]] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">union</span>(attr_matrix[<span class="fu">which</span>(at <span class="sc">&lt;=</span> bines[k]), gene_col],</span>
<span id="cb199-43"><a href="#cb199-43" aria-hidden="true" tabindex="-1"></a>                                     attr_matrix[<span class="fu">which</span>(at <span class="sc">&lt;</span> bines[k <span class="sc">+</span> <span class="dv">1</span>]), gene_col]))</span>
<span id="cb199-44"><a href="#cb199-44" aria-hidden="true" tabindex="-1"></a>     } <span class="cf">else</span> {</span>
<span id="cb199-45"><a href="#cb199-45" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span>(k <span class="sc">==</span> <span class="dv">20</span>){</span>
<span id="cb199-46"><a href="#cb199-46" aria-hidden="true" tabindex="-1"></a>         gps[[k]] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">intersect</span>(attr_matrix[<span class="fu">which</span>(at <span class="sc">&gt;=</span> bines[k]), gene_col], </span>
<span id="cb199-47"><a href="#cb199-47" aria-hidden="true" tabindex="-1"></a>                                            attr_matrix[<span class="fu">which</span>(at <span class="sc">&lt;=</span> bines[k <span class="sc">+</span>  <span class="dv">1</span>]), gene_col]))  </span>
<span id="cb199-48"><a href="#cb199-48" aria-hidden="true" tabindex="-1"></a>         } <span class="cf">else</span> {</span>
<span id="cb199-49"><a href="#cb199-49" aria-hidden="true" tabindex="-1"></a>           gps[[k]] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">intersect</span>(attr_matrix[<span class="fu">which</span>(at <span class="sc">&gt;=</span> bines[k]), gene_col],</span>
<span id="cb199-50"><a href="#cb199-50" aria-hidden="true" tabindex="-1"></a>                                              attr_matrix[<span class="fu">which</span>(at <span class="sc">&lt;</span> bines[k <span class="sc">+</span> <span class="dv">1</span>]), gene_col]))</span>
<span id="cb199-51"><a href="#cb199-51" aria-hidden="true" tabindex="-1"></a>           }</span>
<span id="cb199-52"><a href="#cb199-52" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb199-53"><a href="#cb199-53" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb199-54"><a href="#cb199-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(gps)</span>
<span id="cb199-55"><a href="#cb199-55" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="read-in-data" class="level1">
<h1>Read in data</h1>
<div class="cell-container"><div class="cell-decorator"><pre>In [4]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a><span class="co"># read data</span></span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get DESeq2 results</span></span>
<span id="cb200-4"><a href="#cb200-4" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_no-apoe.csv"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb200-5"><a href="#cb200-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">|&gt;</span></span>
<span id="cb200-6"><a href="#cb200-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">apoe_included =</span> <span class="st">"no_apoe"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 996222 Columns: 9
 Column specification 
Delimiter: ","
chr (3): gene, celltype, deg_direction
dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_apoe.csv"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">|&gt;</span></span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">apoe_included =</span> <span class="st">"apoe"</span>) <span class="sc">|&gt;</span></span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 996222 Columns: 9
 Column specification 
Delimiter: ","
chr (3): gene, celltype, deg_direction
dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb204-2"><a href="#cb204-2" aria-hidden="true" tabindex="-1"></a><span class="co"># if(include_apoe) {</span></span>
<span id="cb204-3"><a href="#cb204-3" aria-hidden="true" tabindex="-1"></a><span class="co"># res &lt;- readr::read_csv(here::here("03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_apoe.csv")) %&gt;%</span></span>
<span id="cb204-4"><a href="#cb204-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   na.omit()</span></span>
<span id="cb204-5"><a href="#cb204-5" aria-hidden="true" tabindex="-1"></a><span class="co"># } else {</span></span>
<span id="cb204-6"><a href="#cb204-6" aria-hidden="true" tabindex="-1"></a><span class="co"># res &lt;- readr::read_csv(here::here("03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_no-apoe.csv")) %&gt;%</span></span>
<span id="cb204-7"><a href="#cb204-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   na.omit()</span></span>
<span id="cb204-8"><a href="#cb204-8" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> res <span class="sc">|&gt;</span></span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(x<span class="sc">$</span>celltype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
                          ambiguous                            Arterial 
                                  7                                  85 
                     Arteriolar-SMC                         Artirial-FB 
                                 30                                   7 
                Astrocyte-activated                 Astrocyte-quiescent 
                                986                                1614 
                          Capillary              Ex-Neuron-ADARB2-ERBB4 
                                539                                  11 
               Ex-Neuron-CBLN2-NRGN               Ex-Neuron-DLC1-HS3ST4 
                                  1                                  12 
Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2           Ex-Neuron-L3-5-RORB-PLCH1 
                                108                                  10 
       Ex-Neuron-L4-5-RORB-IL1RAPL2            Ex-Neuron-L6-THEMIS-NFIA 
                                418                                  20 
              In-Neuron-ADARB2-GRM7    In-Neuron-ADARB2-IL1RAPL2-CSCL14 
                                108                                   7 
        In-Neuron-ADARB2-LAMP5-NRG1          In-Neuron-ADARB2-NRG1-RELN 
                                 37                                   1 
              In-Neuron-DPP10-MEF2C              In-Neuron-PRKG1-TBC1D4 
                                 16                                  51 
              In-Neuron-TRHDE-RALYL                   low-feature-cells 
                                  8                                  28 
                         M-Pericyte                 Microglia-activated 
                                218                                  91 
                Microglia-quiescent                             Oligo-A 
                                 37                                1192 
                            Oligo-B                               OPC-A 
                               1175                                   4 
                              OPC-B                            Pericyte 
                                 92                                 133 
                         Pericyte-2                         Pericyte-FB 
                                 70                                  13 
                    Perivascular-FB               Perivascular-FB-KAZN2 
                                 18                                   2 
               T-cell-parenchymal-1                T-cell-parenchymal-2 
                                  1                                   1 
                         T-Pericyte                 Trans-Endo-to-mural 
                                622                                 359 
                       Vascular-SMC              Vascular-SMC-LINC00486 
                                  8                                   2 
                             Venous 
                                539 </code></pre>
</div>
</div></div>
</section>
<section id="data-prep" class="level1">
<h1>Data prep</h1>
<p>We need ensembl IDs for the PPI, so lets get them</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene ensembl IDs</span></span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar_load</span>(gene_ids)</span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a>gene_ensembl_ids <span class="ot">&lt;-</span> gene_ids</span>
<span id="cb207-4"><a href="#cb207-4" aria-hidden="true" tabindex="-1"></a><span class="co"># remove duplicated gene IDs </span></span>
<span id="cb207-5"><a href="#cb207-5" aria-hidden="true" tabindex="-1"></a>unique_ids <span class="ot">=</span> gene_ensembl_ids[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">duplicated</span>(gene_ensembl_ids<span class="sc">$</span>gene)),]</span>
<span id="cb207-6"><a href="#cb207-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb207-7"><a href="#cb207-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Join dataframes</span></span>
<span id="cb207-8"><a href="#cb207-8" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">left_join</span>(res, unique_ids, <span class="at">by =</span> <span class="st">"gene"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Filter to significant genes</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [7]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb208"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to sig genes</span></span>
<span id="cb208-2"><a href="#cb208-2" aria-hidden="true" tabindex="-1"></a>res_sig <span class="ot">&lt;-</span> res <span class="sc">|&gt;</span></span>
<span id="cb208-3"><a href="#cb208-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb208-4"><a href="#cb208-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb208-5"><a href="#cb208-5" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> res_sig <span class="sc">|&gt;</span></span>
<span id="cb208-6"><a href="#cb208-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(apoe_included <span class="sc">==</span>  <span class="st">"no_apoe"</span>)</span>
<span id="cb208-7"><a href="#cb208-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(x<span class="sc">$</span>celltype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
                          ambiguous                            Arterial 
                                  6                                  56 
                     Arteriolar-SMC                         Artirial-FB 
                                 29                                   5 
                Astrocyte-activated                 Astrocyte-quiescent 
                                709                                1089 
                          Capillary                Ex-Neuron-CBLN2-NRGN 
                                215                                   1 
              Ex-Neuron-DLC1-HS3ST4 Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2 
                                 10                                  92 
          Ex-Neuron-L3-5-RORB-PLCH1        Ex-Neuron-L4-5-RORB-IL1RAPL2 
                                 10                                 281 
           Ex-Neuron-L6-THEMIS-NFIA               In-Neuron-ADARB2-GRM7 
                                 19                                  73 
        In-Neuron-ADARB2-LAMP5-NRG1          In-Neuron-ADARB2-NRG1-RELN 
                                 30                                   1 
              In-Neuron-DPP10-MEF2C              In-Neuron-PRKG1-TBC1D4 
                                  9                                  30 
              In-Neuron-TRHDE-RALYL                   low-feature-cells 
                                  3                                   3 
                         M-Pericyte                 Microglia-activated 
                                 86                                  59 
                Microglia-quiescent                             Oligo-A 
                                  9                                 832 
                            Oligo-B                               OPC-B 
                                728                                  63 
                           Pericyte                          Pericyte-2 
                                 91                                  39 
                        Pericyte-FB                T-cell-parenchymal-1 
                                  4                                   1 
               T-cell-parenchymal-2                          T-Pericyte 
                                  1                                 403 
                Trans-Endo-to-mural                        Vascular-SMC 
                                113                                   7 
             Vascular-SMC-LINC00486                              Venous 
                                  1                                 184 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a><span class="co">#write_csv(res_sig, here::here("initial_pseudobulk_sig_genes.csv"))</span></span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb210-3"><a href="#cb210-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Assert that there are no NAs in ensembl IDs</span></span>
<span id="cb210-4"><a href="#cb210-4" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">are_equal</span>(<span class="fu">nrow</span>(dplyr<span class="sc">::</span><span class="fu">filter</span>(res_sig, <span class="fu">is.na</span>(ensembl))), <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] FALSE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(res_sig, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"../../../gmbh_results_summaries/2024-04-52_ad_bbb/pseudobulk_sig_degs.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="magma-genes" class="level2">
<h2 class="anchored" data-anchor-id="magma-genes">MAGMA genes</h2>
<p>Add in the sig MAGMA celltype sig gene list</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get entrez IDs</span></span>
<span id="cb213-3"><a href="#cb213-3" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb213-4"><a href="#cb213-4" aria-hidden="true" tabindex="-1"></a><span class="fu">listAttributes</span>(ensembl)[<span class="fu">grepl</span>(<span class="st">"entrez"</span>, <span class="fu">listAttributes</span>(ensembl)<span class="sc">$</span>name, <span class="at">ignore.case =</span> T),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     name                                 description
57  entrezgene_trans_name               EntrezGene transcript name ID
76 entrezgene_description NCBI gene (formerly Entrezgene) description
77   entrezgene_accession   NCBI gene (formerly Entrezgene) accession
78          entrezgene_id          NCBI gene (formerly Entrezgene) ID
           page
57 feature_page
76 feature_page
77 feature_page
78 feature_page</code></pre>
</div>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| cache: true</span></span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>entrez_ids <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'ensembl_gene_id'</span>, <span class="st">'entrezgene_id'</span>),</span>
<span id="cb215-3"><a href="#cb215-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">filters =</span> <span class="st">'ensembl_gene_id'</span>,</span>
<span id="cb215-4"><a href="#cb215-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">values =</span> gene_ensembl_ids<span class="sc">$</span>ensembl,</span>
<span id="cb215-5"><a href="#cb215-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">mart =</span> ensembl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a>magma_sig <span class="ot">&lt;-</span></span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/994_magma_inputs/results/sig_celltype_sig_genes.tsv"</span></span>
<span id="cb216-4"><a href="#cb216-4" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">|&gt;</span></span>
<span id="cb216-5"><a href="#cb216-5" aria-hidden="true" tabindex="-1"></a>  janitor<span class="sc">::</span><span class="fu">clean_names</span>() <span class="sc">|&gt;</span></span>
<span id="cb216-6"><a href="#cb216-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">entrezgene_id =</span> gene) <span class="sc">|&gt;</span></span>
<span id="cb216-7"><a href="#cb216-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Exclude the top one percents</span></span>
<span id="cb216-8"><a href="#cb216-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>percent <span class="sc">==</span> <span class="st">"top-one-percent"</span>) <span class="sc">|&gt;</span></span>
<span id="cb216-9"><a href="#cb216-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter to level 2</span></span>
<span id="cb216-10"><a href="#cb216-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dplyr::filter(celltype_level == "level2") |&gt;</span></span>
<span id="cb216-11"><a href="#cb216-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat &lt;- vroom(...)
  problems(dat)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 169727 Columns: 24
 Column specification 
Delimiter: "\t"
chr (10): SetName, input_gene_list, gwas_background, input_filter, celltype_...
dbl (14): GENE, CHR, START, STOP, NSNPS, NPARAM, N, ZSTAT, P, ZFITTED_BASE, ...

 Use `spec()` to retrieve the full column specification for this data.
 Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get gene lists per celltype, gwas and percent</span></span>
<span id="cb219-2"><a href="#cb219-2" aria-hidden="true" tabindex="-1"></a>magma_list <span class="ot">&lt;-</span> magma_sig <span class="sc">|&gt;</span></span>
<span id="cb219-3"><a href="#cb219-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ensembl =</span> ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb219-4"><a href="#cb219-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(gwas_background, input_gene_list, percent, set_name, </span>
<span id="cb219-5"><a href="#cb219-5" aria-hidden="true" tabindex="-1"></a>                  celltype_level) <span class="sc">|&gt;</span></span>
<span id="cb219-6"><a href="#cb219-6" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() <span class="sc">|&gt;</span></span>
<span id="cb219-7"><a href="#cb219-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">celltype =</span> set_name) <span class="sc">|&gt;</span></span>
<span id="cb219-8"><a href="#cb219-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> <span class="fu">paste0</span>(celltype, <span class="st">"_sig-magma-sig-genes"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Id be nice to have a PPI for all the gene across all three celltypes of interest as well</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb220"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb220-1"><a href="#cb220-1" aria-hidden="true" tabindex="-1"></a>sig_celltypes_magma <span class="ot">&lt;-</span> magma_sig <span class="sc">|&gt;</span></span>
<span id="cb220-2"><a href="#cb220-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ensembl =</span> ensembl_gene_id) <span class="sc">|&gt;</span></span>
<span id="cb220-3"><a href="#cb220-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(set_name <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Microglia-activated"</span>, <span class="st">"Pericyte-2"</span>, <span class="st">"Perivascular-FB-KAZN2"</span>) <span class="sc">&amp;</span> input_gene_list <span class="sc">==</span> <span class="st">"all_controls"</span> <span class="sc">&amp;</span> percent <span class="sc">==</span> <span class="st">"top-ten-percent"</span> <span class="sc">&amp;</span> gwas_background <span class="sc">==</span> <span class="st">"EUROPEUKBB_35k10k"</span>)</span>
<span id="cb220-4"><a href="#cb220-4" aria-hidden="true" tabindex="-1"></a>sig_celltypes_magma <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">celltype =</span> <span class="st">"sig_magma_celltypes"</span>, <span class="at">data =</span> <span class="fu">list</span>(sig_celltypes_magma))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Therere a few genes that have overlapping expression in the celltypes, itd be good to add those together and see what the interactions look like.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [11]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb221"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb221-1"><a href="#cb221-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> sig_celltypes_magma<span class="sc">$</span>data[[<span class="dv">1</span>]] <span class="sc">|&gt;</span></span>
<span id="cb221-2"><a href="#cb221-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(unique_ids, <span class="at">by =</span> <span class="fu">join_by</span>(ensembl)) <span class="sc">|&gt;</span></span>
<span id="cb221-3"><a href="#cb221-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(set_name) <span class="sc">|&gt;</span></span>
<span id="cb221-4"><a href="#cb221-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">genes =</span> gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Returning more (or less) than 1 row per `summarise()` group was deprecated in
dplyr 1.1.0.
 Please use `reframe()` instead.
 When switching from `summarise()` to `reframe()`, remember that `reframe()`
  always returns an ungrouped data frame and adjust accordingly.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'set_name'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb224"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1" aria-hidden="true" tabindex="-1"></a>extra_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">set_name =</span> <span class="fu">c</span>(<span class="st">"Microglia-activated"</span>), </span>
<span id="cb224-2"><a href="#cb224-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">"ADAM17"</span>, <span class="st">"MARCHF7"</span>, <span class="st">"PLEKHA1"</span>, <span class="st">"MINK1"</span>))</span>
<span id="cb224-3"><a href="#cb224-3" aria-hidden="true" tabindex="-1"></a>mg <span class="ot">&lt;-</span> x <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(set_name <span class="sc">==</span> <span class="st">"Microglia-activated"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb224-4"><a href="#cb224-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(extra_genes)</span>
<span id="cb224-5"><a href="#cb224-5" aria-hidden="true" tabindex="-1"></a>extra_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">set_name =</span> <span class="fu">c</span>(<span class="st">"Pericyte-2"</span>), <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">"IQCK"</span>))</span>
<span id="cb224-6"><a href="#cb224-6" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> x <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(set_name <span class="sc">==</span> <span class="st">"Pericyte-2"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb224-7"><a href="#cb224-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(extra_genes)</span>
<span id="cb224-8"><a href="#cb224-8" aria-hidden="true" tabindex="-1"></a>extra_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">set_name =</span> <span class="fu">c</span>(<span class="st">"Perivascular-FB-KAZN2"</span>), </span>
<span id="cb224-9"><a href="#cb224-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">"TRANK1"</span>, <span class="st">"EGFR"</span>))</span>
<span id="cb224-10"><a href="#cb224-10" aria-hidden="true" tabindex="-1"></a>all <span class="ot">&lt;-</span> x <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(set_name <span class="sc">==</span> <span class="st">"Perivascular-FB-KAZN2"</span>) <span class="sc">|&gt;</span></span>
<span id="cb224-11"><a href="#cb224-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(extra_genes) <span class="sc">|&gt;</span></span>
<span id="cb224-12"><a href="#cb224-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(mg) <span class="sc">|&gt;</span></span>
<span id="cb224-13"><a href="#cb224-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(p2) <span class="sc">|&gt;</span></span>
<span id="cb224-14"><a href="#cb224-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(unique_ids, <span class="at">by =</span> <span class="fu">join_by</span>(genes <span class="sc">==</span> gene)) <span class="sc">|&gt;</span></span>
<span id="cb224-15"><a href="#cb224-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(set_name) <span class="sc">|&gt;</span></span>
<span id="cb224-16"><a href="#cb224-16" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() <span class="sc">|&gt;</span></span>
<span id="cb224-17"><a href="#cb224-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> <span class="fu">case_when</span>(</span>
<span id="cb224-18"><a href="#cb224-18" aria-hidden="true" tabindex="-1"></a>    set_name <span class="sc">==</span> <span class="st">"Microglia-activated"</span> <span class="sc">~</span> <span class="st">"Microglia-activated_sig_magma_extended"</span>,</span>
<span id="cb224-19"><a href="#cb224-19" aria-hidden="true" tabindex="-1"></a>    set_name <span class="sc">==</span> <span class="st">"Pericyte-2"</span> <span class="sc">~</span> <span class="st">"Pericyte-2_sig_magma_extended"</span>,</span>
<span id="cb224-20"><a href="#cb224-20" aria-hidden="true" tabindex="-1"></a>    set_name <span class="sc">==</span> <span class="st">"Perivascular-FB-KAZN2"</span> <span class="sc">~</span> <span class="st">"Perivascular-FB-KAZN2_sig_magma_extended"</span>,</span>
<span id="cb224-21"><a href="#cb224-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">.default =</span> set_name</span>
<span id="cb224-22"><a href="#cb224-22" aria-hidden="true" tabindex="-1"></a>  ), <span class="at">input_gene_list =</span> <span class="st">"all_controls"</span>, <span class="at">percent =</span> <span class="st">"top-ten-percent"</span>, </span>
<span id="cb224-23"><a href="#cb224-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">gwas_backgroup =</span> <span class="fu">unique</span>(sig_celltypes_magma<span class="sc">$</span>data[[<span class="dv">1</span>]]<span class="sc">$</span>gwas_background)) <span class="sc">|&gt;</span></span>
<span id="cb224-24"><a href="#cb224-24" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(celltype, input_gene_list, gwas_backgroup, percent, data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `set_name`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">assert_that</span>(<span class="fu">map_lgl</span>(all<span class="sc">$</span>data, <span class="sc">~</span> <span class="fu">nrow</span>(.x) <span class="sc">==</span> <span class="fu">nrow</span>(<span class="fu">unique</span>(.x))) <span class="sc">|&gt;</span> <span class="fu">sum</span>() <span class="sc">!=</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> res_sig <span class="sc">|&gt;</span></span>
<span id="cb228-2"><a href="#cb228-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(celltype, apoe_included) <span class="sc">|&gt;</span></span>
<span id="cb228-3"><a href="#cb228-3" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>() <span class="sc">|&gt;</span></span>
<span id="cb228-4"><a href="#cb228-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> <span class="fu">paste0</span>(celltype, <span class="st">"_pseudobulk"</span>))</span>
<span id="cb228-5"><a href="#cb228-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb228-6"><a href="#cb228-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge pseudobulk and magma data</span></span>
<span id="cb228-7"><a href="#cb228-7" aria-hidden="true" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> <span class="fu">rbind</span>(gene_list, magma_list) <span class="sc">|&gt;</span></span>
<span id="cb228-8"><a href="#cb228-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(sig_celltypes_magma) <span class="sc">|&gt;</span></span>
<span id="cb228-9"><a href="#cb228-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(all)</span>
<span id="cb228-10"><a href="#cb228-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(all, mg, p2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
<section id="ppi-test-1.-links-within-gene-set" class="level1">
<h1>PPI Test 1. Links within gene set</h1>
<section id="background-gene-list" class="level2">
<h2 class="anchored" data-anchor-id="background-gene-list">Background gene list</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [13]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb229"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb229-1"><a href="#cb229-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb229-2"><a href="#cb229-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-3"><a href="#cb229-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that it can take a while to download the gene lists, so I've set eval </span></span>
<span id="cb229-4"><a href="#cb229-4" aria-hidden="true" tabindex="-1"></a><span class="co"># to false for this chunk.</span></span>
<span id="cb229-5"><a href="#cb229-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-6"><a href="#cb229-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ensembl &lt;- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl")</span></span>
<span id="cb229-7"><a href="#cb229-7" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb229-8"><a href="#cb229-8" aria-hidden="true" tabindex="-1"></a><span class="co"># # Background population (protein coding genes)</span></span>
<span id="cb229-9"><a href="#cb229-9" aria-hidden="true" tabindex="-1"></a><span class="co"># bg &lt;- getBM(attributes = c("ensembl_gene_id", "gene_biotype"), mart = ensembl)</span></span>
<span id="cb229-10"><a href="#cb229-10" aria-hidden="true" tabindex="-1"></a><span class="co"># bg &lt;- bg[which(bg$gene_biotype == "protein_coding"), ]</span></span>
<span id="cb229-11"><a href="#cb229-11" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(bg$ensembl_gene_id, here::here("03_data/993_protein_protein_interaction_data/full_background"), </span></span>
<span id="cb229-12"><a href="#cb229-12" aria-hidden="true" tabindex="-1"></a><span class="co">#             sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE)</span></span>
<span id="cb229-13"><a href="#cb229-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-14"><a href="#cb229-14" aria-hidden="true" tabindex="-1"></a><span class="co"># CDS length # </span></span>
<span id="cb229-15"><a href="#cb229-15" aria-hidden="true" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"cds_length"</span>),</span>
<span id="cb229-16"><a href="#cb229-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">mart =</span> ensembl)</span>
<span id="cb229-17"><a href="#cb229-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cds) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"cds_length"</span>)</span>
<span id="cb229-18"><a href="#cb229-18" aria-hidden="true" tabindex="-1"></a>cds <span class="ot">&lt;-</span> cds[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">is.na</span>(cds<span class="sc">$</span>cds_length)<span class="sc">==</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb229-19"><a href="#cb229-19" aria-hidden="true" tabindex="-1"></a>cds <span class="ot">&lt;-</span> cds[<span class="fu">order</span>(cds<span class="sc">$</span>ensembl_gene_id), ]</span>
<span id="cb229-20"><a href="#cb229-20" aria-hidden="true" tabindex="-1"></a>cds_max <span class="ot">&lt;-</span> <span class="fu">aggregate.data.frame</span>(cds, <span class="at">by =</span> <span class="fu">list</span>(cds<span class="sc">$</span>ensembl_gene_id), <span class="at">FUN =</span> max)</span>
<span id="cb229-21"><a href="#cb229-21" aria-hidden="true" tabindex="-1"></a>cds_max <span class="ot">&lt;-</span> cds_max[, <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"cds_length"</span>)]</span>
<span id="cb229-22"><a href="#cb229-22" aria-hidden="true" tabindex="-1"></a>cds_max <span class="ot">&lt;-</span> cds_max[cds_max<span class="sc">$</span>ensembl_gene_id <span class="sc">%in%</span> bg<span class="sc">$</span>ensembl_gene_id, ]</span>
<span id="cb229-23"><a href="#cb229-23" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(cds_max, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/993_protein_protein_interaction_data/EnsemblGeneID_MaxCDSlength.txt"</span>), </span>
<span id="cb229-24"><a href="#cb229-24" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="design" class="level2">
<h2 class="anchored" data-anchor-id="design">Design</h2>
<p>So Im thinking it makes sense to do this with gene lists from the pseudobulk differential expression per celltype.</p>
<section id="choose-ppi-network" class="level3">
<h3 class="anchored" data-anchor-id="choose-ppi-network">Choose PPI network</h3>
<p>We can either use the full PPI network, or just the known interactions</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [14]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a>use_known_interactions_only <span class="ot">=</span> <span class="cn">FALSE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [15]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb231"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb231-1"><a href="#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # # SELECT YOUR NETWORK FILE # # # </span></span>
<span id="cb231-2"><a href="#cb231-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Merged protein protein interactions.</span></span>
<span id="cb231-3"><a href="#cb231-3" aria-hidden="true" tabindex="-1"></a><span class="co"># my_network &lt;- read_graph(file = here::here("03_data/993_protein_protein_interaction_data/2023May_MergedPPI_.txt"), </span></span>
<span id="cb231-4"><a href="#cb231-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                          format = "ncol")</span></span>
<span id="cb231-5"><a href="#cb231-5" aria-hidden="true" tabindex="-1"></a><span class="co"># For some reason updating igraph has resulted in read.graph being deprecated </span></span>
<span id="cb231-6"><a href="#cb231-6" aria-hidden="true" tabindex="-1"></a><span class="co"># in favour for read_graph, but this is throwing an error</span></span>
<span id="cb231-7"><a href="#cb231-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the file into a dataframe</span></span>
<span id="cb231-8"><a href="#cb231-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ppi_data &lt;- read.table(here::here("03_data/993_protein_protein_interaction_data/MergedPPI_2024Feb_PIO_STRING.txt"), </span></span>
<span id="cb231-9"><a href="#cb231-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                        header = FALSE, as.is = TRUE)</span></span>
<span id="cb231-10"><a href="#cb231-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(use_known_interactions_only) {</span>
<span id="cb231-11"><a href="#cb231-11" aria-hidden="true" tabindex="-1"></a>ppi_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/993_protein_protein_interaction_data/MergedPPI_2024Feb.txt"</span>), </span>
<span id="cb231-12"><a href="#cb231-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">as.is =</span> <span class="cn">TRUE</span>)</span>
<span id="cb231-13"><a href="#cb231-13" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb231-14"><a href="#cb231-14" aria-hidden="true" tabindex="-1"></a>ppi_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/993_protein_protein_interaction_data/MergedPPI_2024Feb.txt"</span>), </span>
<span id="cb231-15"><a href="#cb231-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">as.is =</span> <span class="cn">TRUE</span>)</span>
<span id="cb231-16"><a href="#cb231-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb231-17"><a href="#cb231-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb231-18"><a href="#cb231-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an igraph object from the dataframe</span></span>
<span id="cb231-19"><a href="#cb231-19" aria-hidden="true" tabindex="-1"></a>my_network <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(ppi_data, <span class="at">directed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb231-20"><a href="#cb231-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(ppi_data)</span>
<span id="cb231-21"><a href="#cb231-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the graph</span></span>
<span id="cb231-22"><a href="#cb231-22" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(my_network)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>IGRAPH 75987e1 UN-- 23145 7068831 -- 
+ attr: name (v/c)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To include merged PPI datasets with only physical interactions from STRING</span></span>
<span id="cb233-2"><a href="#cb233-2" aria-hidden="true" tabindex="-1"></a><span class="co"># use 2023May_MergedPPI_PIO_STRING.txt instead</span></span>
<span id="cb233-3"><a href="#cb233-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-4"><a href="#cb233-4" aria-hidden="true" tabindex="-1"></a><span class="fu">edge_attr</span>(my_network, <span class="st">"weight"</span>, <span class="at">index =</span> <span class="fu">E</span>(my_network)) <span class="ot">&lt;-</span> <span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
<section id="background-list" class="level2">
<h2 class="anchored" data-anchor-id="background-list">Background list</h2>
<p>Could want to run against multiple backgrounds in future, but Ill just use the full list of gene from the pseudobulk across all celltypes.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For the background I can use the full gene list including non-significant genes</span></span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a>background_population <span class="ot">&lt;-</span> <span class="fu">unique</span>(res<span class="sc">$</span>ensembl)</span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a>background_population <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">"full_background"</span> <span class="ot">=</span> background_population)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb235"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb235-1"><a href="#cb235-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb235-2"><a href="#cb235-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Path to background populations</span></span>
<span id="cb235-3"><a href="#cb235-3" aria-hidden="true" tabindex="-1"></a>bg_pop <span class="ot">&lt;-</span></span>
<span id="cb235-4"><a href="#cb235-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(</span>
<span id="cb235-5"><a href="#cb235-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">path =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/993_protein_protein_interaction_data/"</span>),</span>
<span id="cb235-6"><a href="#cb235-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb235-7"><a href="#cb235-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pattern =</span> <span class="st">"Background"</span></span>
<span id="cb235-8"><a href="#cb235-8" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb235-9"><a href="#cb235-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-10"><a href="#cb235-10" aria-hidden="true" tabindex="-1"></a><span class="co"># # #  INPUT YOUR BACKGROUND population # # # </span></span>
<span id="cb235-11"><a href="#cb235-11" aria-hidden="true" tabindex="-1"></a><span class="co"># All protein coding genes for example</span></span>
<span id="cb235-12"><a href="#cb235-12" aria-hidden="true" tabindex="-1"></a>background_population <span class="ot">&lt;-</span> <span class="fu">map</span>(bg_pop, <span class="sc">~</span> <span class="fu">read.table</span>(<span class="at">file =</span> .x, </span>
<span id="cb235-13"><a href="#cb235-13" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>, </span>
<span id="cb235-14"><a href="#cb235-14" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)) <span class="sc">|&gt;</span></span>
<span id="cb235-15"><a href="#cb235-15" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">set_names</span>(<span class="fu">basename</span>(bg_pop))</span>
<span id="cb235-16"><a href="#cb235-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-17"><a href="#cb235-17" aria-hidden="true" tabindex="-1"></a><span class="co"># if(length(background_population) == 1){</span></span>
<span id="cb235-18"><a href="#cb235-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   background_population &lt;- background_population[[1]]</span></span>
<span id="cb235-19"><a href="#cb235-19" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [18]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # #  INPUT gene attributes  # # # </span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene length / max CDS length</span></span>
<span id="cb236-3"><a href="#cb236-3" aria-hidden="true" tabindex="-1"></a>gene_length <span class="ot">&lt;-</span> <span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/993_protein_protein_interaction_data/EnsemblGeneID_MaxCDSlength.txt"</span>), </span>
<span id="cb236-4"><a href="#cb236-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>, </span>
<span id="cb236-5"><a href="#cb236-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb236-6"><a href="#cb236-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gene_length) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"gene_length"</span>)</span>
<span id="cb236-7"><a href="#cb236-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-8"><a href="#cb236-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-9"><a href="#cb236-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # # SETUP NUMBER OF RANDOM SAMPLES to create your NULL # # #</span></span>
<span id="cb236-10"><a href="#cb236-10" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb236-11"><a href="#cb236-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-12"><a href="#cb236-12" aria-hidden="true" tabindex="-1"></a><span class="co"># # # Set to "D" to take into account degree, </span></span>
<span id="cb236-13"><a href="#cb236-13" aria-hidden="true" tabindex="-1"></a><span class="co"># # # "GL" to taken into account Gene length, </span></span>
<span id="cb236-14"><a href="#cb236-14" aria-hidden="true" tabindex="-1"></a><span class="co"># # # or "B" to take into account both degree and Gene length</span></span>
<span id="cb236-15"><a href="#cb236-15" aria-hidden="true" tabindex="-1"></a>gene_attribute <span class="ot">&lt;-</span> <span class="st">"B"</span></span>
<span id="cb236-16"><a href="#cb236-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-17"><a href="#cb236-17" aria-hidden="true" tabindex="-1"></a><span class="co"># # # CHANGE the name of your OUTPUT FILE that will contain your test against NULL # # #</span></span>
<span id="cb236-18"><a href="#cb236-18" aria-hidden="true" tabindex="-1"></a>output_file <span class="ot">&lt;-</span></span>
<span id="cb236-19"><a href="#cb236-19" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb236-20"><a href="#cb236-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions"</span>,</span>
<span id="cb236-21"><a href="#cb236-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"PPI_Test1"</span></span>
<span id="cb236-22"><a href="#cb236-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb236-23"><a href="#cb236-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-24"><a href="#cb236-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Min number of genes for a cluster</span></span>
<span id="cb236-25"><a href="#cb236-25" aria-hidden="true" tabindex="-1"></a>min_cluster_size <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb236-26"><a href="#cb236-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-27"><a href="#cb236-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Output Louvain clusters</span></span>
<span id="cb236-28"><a href="#cb236-28" aria-hidden="true" tabindex="-1"></a>output_file2 <span class="ot">&lt;-</span></span>
<span id="cb236-29"><a href="#cb236-29" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb236-30"><a href="#cb236-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions"</span>,</span>
<span id="cb236-31"><a href="#cb236-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"PPI_Test1-Louvain_"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb237"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb237-1"><a href="#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add test type to the output file name</span></span>
<span id="cb237-2"><a href="#cb237-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(gene_attribute <span class="sc">==</span> <span class="st">"D"</span>) {</span>
<span id="cb237-3"><a href="#cb237-3" aria-hidden="true" tabindex="-1"></a>  output_file <span class="ot">&lt;-</span> <span class="fu">paste</span>(output_file, <span class="st">"MatchDegree"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)} <span class="cf">else</span> {</span>
<span id="cb237-4"><a href="#cb237-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(gene_attribute <span class="sc">==</span> <span class="st">"GL"</span>) {</span>
<span id="cb237-5"><a href="#cb237-5" aria-hidden="true" tabindex="-1"></a>  output_file <span class="ot">&lt;-</span> <span class="fu">paste</span>(output_file, <span class="st">"MatchGenelength"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)} <span class="cf">else</span> {</span>
<span id="cb237-6"><a href="#cb237-6" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(gene_attribute <span class="sc">==</span> <span class="st">"B"</span>){</span>
<span id="cb237-7"><a href="#cb237-7" aria-hidden="true" tabindex="-1"></a>  output_file <span class="ot">&lt;-</span> <span class="fu">paste</span>(output_file, <span class="st">"MatchDegreeAndGenelength"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)   </span>
<span id="cb237-8"><a href="#cb237-8" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb237-9"><a href="#cb237-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb237-10"><a href="#cb237-10" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb237-11"><a href="#cb237-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb237-12"><a href="#cb237-12" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(use_known_interactions_only) {</span>
<span id="cb237-13"><a href="#cb237-13" aria-hidden="true" tabindex="-1"></a>  output_file <span class="ot">&lt;-</span> <span class="fu">paste</span>(output_file, <span class="st">"known-interactions"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb237-14"><a href="#cb237-14" aria-hidden="true" tabindex="-1"></a>  output_file2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(output_file2, <span class="st">"known-interactions"</span>, <span class="at">sep =</span> <span class="st">"_"</span>)</span>
<span id="cb237-15"><a href="#cb237-15" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb237-16"><a href="#cb237-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb237-17"><a href="#cb237-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Pull out the ensembl list for each celltype</span></span>
<span id="cb237-18"><a href="#cb237-18" aria-hidden="true" tabindex="-1"></a>my_gene_list <span class="ot">&lt;-</span> <span class="fu">map</span>(gene_list<span class="sc">$</span>data, <span class="sc">~</span> dplyr<span class="sc">::</span><span class="fu">pull</span>(.x, ensembl) <span class="sc">|&gt;</span> <span class="fu">unique</span>()) <span class="sc">|&gt;</span></span>
<span id="cb237-19"><a href="#cb237-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste</span>(gene_list<span class="sc">$</span>celltype, gene_list<span class="sc">$</span>apoe_included, gene_list<span class="sc">$</span>gwas_background, gene_list<span class="sc">$</span>input_gene_list, gene_list<span class="sc">$</span>percent, gene_list<span class="sc">$</span>celltype_level, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb237-20"><a href="#cb237-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_gene_list) <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(<span class="fu">names</span>(my_gene_list), <span class="st">"_NA_NA_NA"</span>, <span class="st">""</span>) <span class="sc">|&gt;</span></span>
<span id="cb237-21"><a href="#cb237-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="st">"_NA"</span>, <span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="gene-lists-for-analysis" class="level2">
<h2 class="anchored" data-anchor-id="gene-lists-for-analysis">Gene lists for analysis</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a>process_network <span class="ot">&lt;-</span> <span class="cf">function</span>(background) {</span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reduce my network to background population of genes tested #</span></span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a>  Gbg <span class="ot">&lt;-</span> <span class="fu">vertex_attr</span>(my_network)<span class="sc">$</span>name</span>
<span id="cb238-4"><a href="#cb238-4" aria-hidden="true" tabindex="-1"></a>  BG <span class="ot">&lt;-</span> <span class="fu">intersect</span>(background, <span class="fu">intersect</span>(Gbg, gene_length<span class="sc">$</span>ensembl_gene_id))</span>
<span id="cb238-5"><a href="#cb238-5" aria-hidden="true" tabindex="-1"></a>  my_network <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> my_network, <span class="at">vids =</span> BG)</span>
<span id="cb238-6"><a href="#cb238-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-7"><a href="#cb238-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create tables with the GENE attributes provided for the background population found in my network #</span></span>
<span id="cb238-8"><a href="#cb238-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate and order genes according to their degree in my network #</span></span>
<span id="cb238-9"><a href="#cb238-9" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(my_network, <span class="at">mode =</span> <span class="st">"all"</span>)</span>
<span id="cb238-10"><a href="#cb238-10" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ensembl_gene_id =</span> <span class="fu">names</span>(D), <span class="at">degree =</span> D)</span>
<span id="cb238-11"><a href="#cb238-11" aria-hidden="true" tabindex="-1"></a>  D <span class="ot">&lt;-</span> D[<span class="fu">order</span>(D<span class="sc">$</span>degree, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),]</span>
<span id="cb238-12"><a href="#cb238-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(D) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(D)</span>
<span id="cb238-13"><a href="#cb238-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-14"><a href="#cb238-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Order genes by Gene length #</span></span>
<span id="cb238-15"><a href="#cb238-15" aria-hidden="true" tabindex="-1"></a>  gene_length <span class="ot">&lt;-</span> gene_length[gene_length<span class="sc">$</span>ensembl_gene_id <span class="sc">%in%</span> BG, ]</span>
<span id="cb238-16"><a href="#cb238-16" aria-hidden="true" tabindex="-1"></a>  gene_length <span class="ot">&lt;-</span> gene_length[<span class="fu">order</span>(gene_length<span class="sc">$</span>gene_length, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb238-17"><a href="#cb238-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(gene_length) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(gene_length)</span>
<span id="cb238-18"><a href="#cb238-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-19"><a href="#cb238-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combined score Degree and Gene length</span></span>
<span id="cb238-20"><a href="#cb238-20" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> <span class="fu">merge</span>(D, gene_length, <span class="at">by =</span> <span class="st">"ensembl_gene_id"</span>)</span>
<span id="cb238-21"><a href="#cb238-21" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(B, </span>
<span id="cb238-22"><a href="#cb238-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">scaled_degree =</span> <span class="fu">rescale01</span>(B<span class="sc">$</span>degree), </span>
<span id="cb238-23"><a href="#cb238-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">scaled_gene_length =</span> <span class="fu">rescale01</span>(B<span class="sc">$</span>gene_length), </span>
<span id="cb238-24"><a href="#cb238-24" aria-hidden="true" tabindex="-1"></a>                  <span class="at">score =</span> <span class="fu">rescale01</span>(B<span class="sc">$</span>degree) <span class="sc">+</span> <span class="fu">rescale01</span>(B<span class="sc">$</span>gene_length))</span>
<span id="cb238-25"><a href="#cb238-25" aria-hidden="true" tabindex="-1"></a>  B <span class="ot">&lt;-</span> B[<span class="fu">order</span>(B<span class="sc">$</span>score,  <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb238-26"><a href="#cb238-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(B) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(gene_length)</span>
<span id="cb238-27"><a href="#cb238-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-28"><a href="#cb238-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Report numbers</span></span>
<span id="cb238-29"><a href="#cb238-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Background population: "</span>, <span class="fu">length</span>(background), <span class="st">" genes"</span>))</span>
<span id="cb238-30"><a href="#cb238-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Genes in the network: "</span>, <span class="fu">length</span>(Gbg), <span class="st">" genes"</span>))</span>
<span id="cb238-31"><a href="#cb238-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(</span>
<span id="cb238-32"><a href="#cb238-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb238-33"><a href="#cb238-33" aria-hidden="true" tabindex="-1"></a>      <span class="st">"Background population contained in the network and used for analysis: "</span>,</span>
<span id="cb238-34"><a href="#cb238-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">length</span>(BG),</span>
<span id="cb238-35"><a href="#cb238-35" aria-hidden="true" tabindex="-1"></a>      <span class="st">" genes"</span></span>
<span id="cb238-36"><a href="#cb238-36" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb238-37"><a href="#cb238-37" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb238-38"><a href="#cb238-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-39"><a href="#cb238-39" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">map</span>(my_gene_list, <span class="sc">~</span> <span class="fu">length</span>(<span class="fu">intersect</span>(.x, BG)))</span>
<span id="cb238-40"><a href="#cb238-40" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">map2</span>(x, <span class="fu">names</span>(my_gene_list), <span class="sc">~</span> <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Genes of interest contained in the network and used for analysis: "</span>, .x, <span class="st">" for celltype "</span>, .y)))</span>
<span id="cb238-41"><a href="#cb238-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-42"><a href="#cb238-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="st">"BG"</span> <span class="ot">=</span> BG, <span class="st">"network"</span> <span class="ot">=</span> my_network, <span class="st">"D"</span> <span class="ot">=</span> D, <span class="st">"B"</span> <span class="ot">=</span> B))</span>
<span id="cb238-43"><a href="#cb238-43" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb238-44"><a href="#cb238-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-45"><a href="#cb238-45" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">map</span>(background_population, process_network)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Background population: 21054 genes"
[1] "Genes in the network: 23145 genes"
[1] "Background population contained in the network and used for analysis: 16947 genes"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype ambiguous_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 25 for celltype Arterial_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Arteriolar-SMC_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype Artirial-FB_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 252 for celltype Astrocyte-activated_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 466 for celltype Astrocyte-quiescent_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 293 for celltype Capillary_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 11 for celltype Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype Ex-Neuron-DLC1-HS3ST4_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 15 for celltype Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 121 for celltype Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 29 for celltype In-Neuron-ADARB2-GRM7_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype In-Neuron-ADARB2-IL1RAPL2-CSCL14_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 7 for celltype In-Neuron-DPP10-MEF2C_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 18 for celltype In-Neuron-PRKG1-TBC1D4_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype In-Neuron-TRHDE-RALYL_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 25 for celltype low-feature-cells_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 120 for celltype M-Pericyte_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 28 for celltype Microglia-activated_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 26 for celltype Microglia-quiescent_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 311 for celltype Oligo-A_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 401 for celltype Oligo-B_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype OPC-A_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 28 for celltype OPC-B_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 39 for celltype Pericyte_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 28 for celltype Pericyte-2_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 8 for celltype Pericyte-FB_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 17 for celltype Perivascular-FB_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype Perivascular-FB-KAZN2_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 200 for celltype T-Pericyte_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 223 for celltype Trans-Endo-to-mural_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 0 for celltype Vascular-SMC_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Vascular-SMC-LINC00486_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 326 for celltype Venous_pseudobulk_apoe"
[1] "Genes of interest contained in the network and used for analysis: 5 for celltype ambiguous_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 51 for celltype Arterial_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 28 for celltype Arteriolar-SMC_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 5 for celltype Artirial-FB_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 631 for celltype Astrocyte-activated_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 969 for celltype Astrocyte-quiescent_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 194 for celltype Capillary_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Ex-Neuron-CBLN2-NRGN_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 10 for celltype Ex-Neuron-DLC1-HS3ST4_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 80 for celltype Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 10 for celltype Ex-Neuron-L3-5-RORB-PLCH1_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 253 for celltype Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 19 for celltype Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 67 for celltype In-Neuron-ADARB2-GRM7_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 24 for celltype In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype In-Neuron-ADARB2-NRG1-RELN_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 9 for celltype In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 24 for celltype In-Neuron-PRKG1-TBC1D4_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype In-Neuron-TRHDE-RALYL_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype low-feature-cells_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 77 for celltype M-Pericyte_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 55 for celltype Microglia-activated_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 7 for celltype Microglia-quiescent_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 709 for celltype Oligo-A_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 640 for celltype Oligo-B_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 60 for celltype OPC-B_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 83 for celltype Pericyte_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 31 for celltype Pericyte-2_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Pericyte-FB_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype T-cell-parenchymal-1_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype T-cell-parenchymal-2_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 369 for celltype T-Pericyte_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 100 for celltype Trans-Endo-to-mural_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype Vascular-SMC_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Vascular-SMC-LINC00486_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 162 for celltype Venous_pseudobulk_no_apoe"
[1] "Genes of interest contained in the network and used for analysis: 20 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 17 for celltype T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 12 for celltype Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 29 for celltype Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 29 for celltype T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 10 for celltype Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype Trans-Endo-to-mural_sig-magma-sig-genes_stroke_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 21 for celltype Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 22 for celltype Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 20 for celltype Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 26 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 15 for celltype Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 19 for celltype SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 17 for celltype T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 19 for celltype Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 35 for celltype Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 46 for celltype Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 33 for celltype Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 37 for celltype T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 12 for celltype Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 15 for celltype Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 14 for celltype T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype Astro_sig-magma-sig-genes_wmh_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 17 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 24 for celltype Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype OPC_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 16 for celltype Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 22 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 11 for celltype Astro_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 16 for celltype Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 12 for celltype Neuron_in_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 11 for celltype Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype Fibroblast_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 5 for celltype SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Fibroblast_sig-magma-sig-genes_wmh_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 13 for celltype Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 10 for celltype Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 11 for celltype SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 13 for celltype T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 8 for celltype Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 8 for celltype T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest contained in the network and used for analysis: 20 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 34 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 21 for celltype Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 20 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 34 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 23 for celltype Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 23 for celltype Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 5 for celltype Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 21 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 29 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 5 for celltype Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 21 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 31 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 23 for celltype Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 11 for celltype Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 7 for celltype In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 7 for celltype Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 1 for celltype Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype T-cell-vascular_sig-magma-sig-genes_Kunkle_2019_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 5 for celltype M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Arteriolar-SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 9 for celltype Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 6 for celltype Meningeal-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype T-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 2 for celltype Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype Vascular-SMC-LINC00486_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Meningeal-FB_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 4 for celltype Capillary_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 3 for celltype T-cell-mixed_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest contained in the network and used for analysis: 55 for celltype sig_magma_celltypes_NA"
[1] "Genes of interest contained in the network and used for analysis: 25 for celltype Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"
[1] "Genes of interest contained in the network and used for analysis: 38 for celltype Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA"
[1] "Genes of interest contained in the network and used for analysis: 24 for celltype Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [21]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">map2</span>(<span class="fu">names</span>(my_gene_list), my_gene_list, <span class="sc">~</span> <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Genes of interest: "</span>, <span class="fu">length</span>(.y), <span class="st">" for celltype "</span>, .x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Genes of interest: 1 for celltype ambiguous_pseudobulk_apoe"
[1] "Genes of interest: 29 for celltype Arterial_pseudobulk_apoe"
[1] "Genes of interest: 1 for celltype Arteriolar-SMC_pseudobulk_apoe"
[1] "Genes of interest: 2 for celltype Artirial-FB_pseudobulk_apoe"
[1] "Genes of interest: 275 for celltype Astrocyte-activated_pseudobulk_apoe"
[1] "Genes of interest: 522 for celltype Astrocyte-quiescent_pseudobulk_apoe"
[1] "Genes of interest: 324 for celltype Capillary_pseudobulk_apoe"
[1] "Genes of interest: 11 for celltype Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe"
[1] "Genes of interest: 2 for celltype Ex-Neuron-DLC1-HS3ST4_pseudobulk_apoe"
[1] "Genes of interest: 16 for celltype Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_apoe"
[1] "Genes of interest: 137 for celltype Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe"
[1] "Genes of interest: 1 for celltype Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_apoe"
[1] "Genes of interest: 35 for celltype In-Neuron-ADARB2-GRM7_pseudobulk_apoe"
[1] "Genes of interest: 7 for celltype In-Neuron-ADARB2-IL1RAPL2-CSCL14_pseudobulk_apoe"
[1] "Genes of interest: 7 for celltype In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_apoe"
[1] "Genes of interest: 7 for celltype In-Neuron-DPP10-MEF2C_pseudobulk_apoe"
[1] "Genes of interest: 21 for celltype In-Neuron-PRKG1-TBC1D4_pseudobulk_apoe"
[1] "Genes of interest: 5 for celltype In-Neuron-TRHDE-RALYL_pseudobulk_apoe"
[1] "Genes of interest: 25 for celltype low-feature-cells_pseudobulk_apoe"
[1] "Genes of interest: 132 for celltype M-Pericyte_pseudobulk_apoe"
[1] "Genes of interest: 32 for celltype Microglia-activated_pseudobulk_apoe"
[1] "Genes of interest: 28 for celltype Microglia-quiescent_pseudobulk_apoe"
[1] "Genes of interest: 359 for celltype Oligo-A_pseudobulk_apoe"
[1] "Genes of interest: 447 for celltype Oligo-B_pseudobulk_apoe"
[1] "Genes of interest: 4 for celltype OPC-A_pseudobulk_apoe"
[1] "Genes of interest: 29 for celltype OPC-B_pseudobulk_apoe"
[1] "Genes of interest: 42 for celltype Pericyte_pseudobulk_apoe"
[1] "Genes of interest: 31 for celltype Pericyte-2_pseudobulk_apoe"
[1] "Genes of interest: 9 for celltype Pericyte-FB_pseudobulk_apoe"
[1] "Genes of interest: 18 for celltype Perivascular-FB_pseudobulk_apoe"
[1] "Genes of interest: 2 for celltype Perivascular-FB-KAZN2_pseudobulk_apoe"
[1] "Genes of interest: 219 for celltype T-Pericyte_pseudobulk_apoe"
[1] "Genes of interest: 244 for celltype Trans-Endo-to-mural_pseudobulk_apoe"
[1] "Genes of interest: 1 for celltype Vascular-SMC_pseudobulk_apoe"
[1] "Genes of interest: 1 for celltype Vascular-SMC-LINC00486_pseudobulk_apoe"
[1] "Genes of interest: 355 for celltype Venous_pseudobulk_apoe"
[1] "Genes of interest: 6 for celltype ambiguous_pseudobulk_no_apoe"
[1] "Genes of interest: 56 for celltype Arterial_pseudobulk_no_apoe"
[1] "Genes of interest: 29 for celltype Arteriolar-SMC_pseudobulk_no_apoe"
[1] "Genes of interest: 5 for celltype Artirial-FB_pseudobulk_no_apoe"
[1] "Genes of interest: 703 for celltype Astrocyte-activated_pseudobulk_no_apoe"
[1] "Genes of interest: 1083 for celltype Astrocyte-quiescent_pseudobulk_no_apoe"
[1] "Genes of interest: 214 for celltype Capillary_pseudobulk_no_apoe"
[1] "Genes of interest: 1 for celltype Ex-Neuron-CBLN2-NRGN_pseudobulk_no_apoe"
[1] "Genes of interest: 10 for celltype Ex-Neuron-DLC1-HS3ST4_pseudobulk_no_apoe"
[1] "Genes of interest: 92 for celltype Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_no_apoe"
[1] "Genes of interest: 10 for celltype Ex-Neuron-L3-5-RORB-PLCH1_pseudobulk_no_apoe"
[1] "Genes of interest: 281 for celltype Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe"
[1] "Genes of interest: 19 for celltype Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_no_apoe"
[1] "Genes of interest: 73 for celltype In-Neuron-ADARB2-GRM7_pseudobulk_no_apoe"
[1] "Genes of interest: 30 for celltype In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_no_apoe"
[1] "Genes of interest: 1 for celltype In-Neuron-ADARB2-NRG1-RELN_pseudobulk_no_apoe"
[1] "Genes of interest: 9 for celltype In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe"
[1] "Genes of interest: 30 for celltype In-Neuron-PRKG1-TBC1D4_pseudobulk_no_apoe"
[1] "Genes of interest: 3 for celltype In-Neuron-TRHDE-RALYL_pseudobulk_no_apoe"
[1] "Genes of interest: 3 for celltype low-feature-cells_pseudobulk_no_apoe"
[1] "Genes of interest: 86 for celltype M-Pericyte_pseudobulk_no_apoe"
[1] "Genes of interest: 59 for celltype Microglia-activated_pseudobulk_no_apoe"
[1] "Genes of interest: 9 for celltype Microglia-quiescent_pseudobulk_no_apoe"
[1] "Genes of interest: 827 for celltype Oligo-A_pseudobulk_no_apoe"
[1] "Genes of interest: 723 for celltype Oligo-B_pseudobulk_no_apoe"
[1] "Genes of interest: 63 for celltype OPC-B_pseudobulk_no_apoe"
[1] "Genes of interest: 91 for celltype Pericyte_pseudobulk_no_apoe"
[1] "Genes of interest: 39 for celltype Pericyte-2_pseudobulk_no_apoe"
[1] "Genes of interest: 4 for celltype Pericyte-FB_pseudobulk_no_apoe"
[1] "Genes of interest: 1 for celltype T-cell-parenchymal-1_pseudobulk_no_apoe"
[1] "Genes of interest: 1 for celltype T-cell-parenchymal-2_pseudobulk_no_apoe"
[1] "Genes of interest: 402 for celltype T-Pericyte_pseudobulk_no_apoe"
[1] "Genes of interest: 113 for celltype Trans-Endo-to-mural_pseudobulk_no_apoe"
[1] "Genes of interest: 7 for celltype Vascular-SMC_pseudobulk_no_apoe"
[1] "Genes of interest: 1 for celltype Vascular-SMC-LINC00486_pseudobulk_no_apoe"
[1] "Genes of interest: 184 for celltype Venous_pseudobulk_no_apoe"
[1] "Genes of interest: 21 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 17 for celltype T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 12 for celltype Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 30 for celltype Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 29 for celltype T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 10 for celltype Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 3 for celltype Trans-Endo-to-mural_sig-magma-sig-genes_stroke_35k10k_all_controls_top-five-percent_level1"
[1] "Genes of interest: 21 for celltype Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 22 for celltype Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 20 for celltype Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 27 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 15 for celltype Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 19 for celltype SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 17 for celltype T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 19 for celltype Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 36 for celltype Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 47 for celltype Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 34 for celltype Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 37 for celltype T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 12 for celltype Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 15 for celltype Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 14 for celltype T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 3 for celltype Astro_sig-magma-sig-genes_wmh_35k10k_all_controls_top-ten-percent_level1"
[1] "Genes of interest: 18 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest: 25 for celltype Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest: 4 for celltype Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest: 1 for celltype OPC_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest: 1 for celltype Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Genes of interest: 16 for celltype Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 23 for celltype Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 11 for celltype Astro_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 16 for celltype Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 12 for celltype Neuron_in_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 11 for celltype Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 2 for celltype Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Genes of interest: 2 for celltype SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 6 for celltype T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 6 for celltype Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 3 for celltype Fibroblast_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 4 for celltype Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 6 for celltype Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 5 for celltype SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 1 for celltype Fibroblast_sig-magma-sig-genes_wmh_35k10k_controls_vascular_top-five-percent_level1"
[1] "Genes of interest: 13 for celltype Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 10 for celltype Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 11 for celltype SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 13 for celltype T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 8 for celltype Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 6 for celltype SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 8 for celltype T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 4 for celltype Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 2 for celltype Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 4 for celltype Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Genes of interest: 21 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest: 35 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest: 21 for celltype Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest: 4 for celltype Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level2"
[1] "Genes of interest: 21 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest: 35 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest: 23 for celltype Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest: 24 for celltype Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest: 5 for celltype Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level2"
[1] "Genes of interest: 22 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest: 30 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest: 6 for celltype In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest: 5 for celltype Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest: 1 for celltype Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Genes of interest: 22 for celltype Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 32 for celltype Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 24 for celltype Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 11 for celltype Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 7 for celltype In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 6 for celltype In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 7 for celltype Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 1 for celltype Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Genes of interest: 4 for celltype Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest: 6 for celltype T-cell-vascular_sig-magma-sig-genes_Kunkle_2019_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest: 5 for celltype M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest: 3 for celltype Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest: 4 for celltype Arteriolar-SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest: 4 for celltype Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Genes of interest: 9 for celltype Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 6 for celltype Meningeal-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 4 for celltype T-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 3 for celltype M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 2 for celltype Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 4 for celltype Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 3 for celltype Vascular-SMC-LINC00486_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 4 for celltype Meningeal-FB_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 4 for celltype Capillary_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 3 for celltype Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 3 for celltype T-cell-mixed_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Genes of interest: 57 for celltype sig_magma_celltypes_NA"
[1] "Genes of interest: 26 for celltype Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"
[1] "Genes of interest: 39 for celltype Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA"
[1] "Genes of interest: 24 for celltype Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="subnetwork-plot" class="level2">
<h2 class="anchored" data-anchor-id="subnetwork-plot">Subnetwork plot</h2>
<ul>
<li>Interaction network of the genes of interest.</li>
</ul>
<div class="cell-container"><div class="cell-decorator"><pre>In [22]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(background_population)</span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-3"><a href="#cb243-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to intersect each cell type gene list with a single background list</span></span>
<span id="cb243-4"><a href="#cb243-4" aria-hidden="true" tabindex="-1"></a>intersect_with_background <span class="ot">&lt;-</span> <span class="cf">function</span>(background_genes, cell_type_gene_list) {</span>
<span id="cb243-5"><a href="#cb243-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(cell_type_gene_list, <span class="sc">~</span> <span class="fu">intersect</span>(.x, <span class="at">y =</span> background_genes))</span>
<span id="cb243-6"><a href="#cb243-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb243-7"><a href="#cb243-7" aria-hidden="true" tabindex="-1"></a>my_gene_list <span class="ot">&lt;-</span> <span class="fu">map</span>(res, <span class="sc">~</span> <span class="fu">intersect_with_background</span>(.x<span class="sc">$</span>BG, my_gene_list))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [23]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define subnetwork</span></span>
<span id="cb244-2"><a href="#cb244-2" aria-hidden="true" tabindex="-1"></a>create_subnetwork <span class="ot">&lt;-</span> <span class="cf">function</span>(network, gene_lists) {</span>
<span id="cb244-3"><a href="#cb244-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(gene_lists, <span class="sc">~</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> network, <span class="at">vids =</span> .x))</span>
<span id="cb244-4"><a href="#cb244-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb244-5"><a href="#cb244-5" aria-hidden="true" tabindex="-1"></a>my_subgraph <span class="ot">&lt;-</span> <span class="fu">map2</span>(res, my_gene_list, <span class="sc">~</span> <span class="fu">create_subnetwork</span>(.x<span class="sc">$</span>network, .y))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>Ill disable the code that generates the plots as they also can be plotted with their clusters later.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [24]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb245"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb245-1"><a href="#cb245-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb245-2"><a href="#cb245-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot subnetwork</span></span>
<span id="cb245-3"><a href="#cb245-3" aria-hidden="true" tabindex="-1"></a><span class="fu">map</span>(my_subgraph, <span class="sc">~</span> <span class="fu">map2</span>(</span>
<span id="cb245-4"><a href="#cb245-4" aria-hidden="true" tabindex="-1"></a>  .x, <span class="fu">names</span>(.x),</span>
<span id="cb245-5"><a href="#cb245-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> <span class="fu">plot</span>(</span>
<span id="cb245-6"><a href="#cb245-6" aria-hidden="true" tabindex="-1"></a>    .x,</span>
<span id="cb245-7"><a href="#cb245-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.size =</span> <span class="dv">3</span>,</span>
<span id="cb245-8"><a href="#cb245-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.color =</span> <span class="st">"darkslateblue"</span>,</span>
<span id="cb245-9"><a href="#cb245-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.frame.color =</span> <span class="st">"darkslateblue"</span>,</span>
<span id="cb245-10"><a href="#cb245-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">edge.color =</span> <span class="st">"grey"</span>,</span>
<span id="cb245-11"><a href="#cb245-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">rescale =</span> <span class="cn">TRUE</span>,</span>
<span id="cb245-12"><a href="#cb245-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label =</span> <span class="cn">NA</span>,</span>
<span id="cb245-13"><a href="#cb245-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">"Celltype: "</span>, .y)</span>
<span id="cb245-14"><a href="#cb245-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb245-15"><a href="#cb245-15" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="test-for-number-of-interactions" class="level2">
<h2 class="anchored" data-anchor-id="test-for-number-of-interactions">Test for number of interactions</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a>perform_permutation <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_list, gene_list_name, my_network, BG, gene_attribute, m, <span class="at">num_cores =</span> <span class="fu">detectCores</span>() <span class="sc">-</span> <span class="dv">1</span>) {</span>
<span id="cb246-2"><a href="#cb246-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb246-3"><a href="#cb246-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Processing "</span>, gene_list_name))</span>
<span id="cb246-4"><a href="#cb246-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb246-5"><a href="#cb246-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gene population of interest</span></span>
<span id="cb246-6"><a href="#cb246-6" aria-hidden="true" tabindex="-1"></a>  gene_list_A <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gene_list, BG)</span>
<span id="cb246-7"><a href="#cb246-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-8"><a href="#cb246-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sum of the edges within gene list A</span></span>
<span id="cb246-9"><a href="#cb246-9" aria-hidden="true" tabindex="-1"></a>  mynet_gl <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(my_network, <span class="at">vids =</span> gene_list_A)</span>
<span id="cb246-10"><a href="#cb246-10" aria-hidden="true" tabindex="-1"></a>  w_mynet_gl <span class="ot">&lt;-</span> <span class="fu">edge.attributes</span>(mynet_gl)<span class="sc">$</span>weight</span>
<span id="cb246-11"><a href="#cb246-11" aria-hidden="true" tabindex="-1"></a>  o <span class="ot">&lt;-</span> <span class="fu">sum</span>(w_mynet_gl)</span>
<span id="cb246-12"><a href="#cb246-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb246-13"><a href="#cb246-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sum of the edges from within a random gene list size A</span></span>
<span id="cb246-14"><a href="#cb246-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Use mclapply to parallelize the replicate function</span></span>
<span id="cb246-15"><a href="#cb246-15" aria-hidden="true" tabindex="-1"></a>  e <span class="ot">&lt;-</span> <span class="fu">mclapply</span>(<span class="dv">1</span><span class="sc">:</span>m, <span class="cf">function</span>(i) {</span>
<span id="cb246-16"><a href="#cb246-16" aria-hidden="true" tabindex="-1"></a>    rnd_smpl <span class="ot">&lt;-</span> <span class="fu">similar_gene_list</span>(<span class="at">gene_list =</span> gene_list_A, <span class="at">attr_matrix =</span> gene_attribute, <span class="at">rango =</span> <span class="dv">100</span>)</span>
<span id="cb246-17"><a href="#cb246-17" aria-hidden="true" tabindex="-1"></a>    rnd_mynet_gl <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(my_network, <span class="at">vids =</span> rnd_smpl)</span>
<span id="cb246-18"><a href="#cb246-18" aria-hidden="true" tabindex="-1"></a>    rnd_w_mynet_gl <span class="ot">&lt;-</span> <span class="fu">edge.attributes</span>(rnd_mynet_gl)<span class="sc">$</span>weight</span>
<span id="cb246-19"><a href="#cb246-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sum</span>(rnd_w_mynet_gl)</span>
<span id="cb246-20"><a href="#cb246-20" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">mc.cores =</span> num_cores)</span>
<span id="cb246-21"><a href="#cb246-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert the list to a vector</span></span>
<span id="cb246-22"><a href="#cb246-22" aria-hidden="true" tabindex="-1"></a>  e <span class="ot">&lt;-</span> <span class="fu">unlist</span>(e)</span>
<span id="cb246-23"><a href="#cb246-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-24"><a href="#cb246-24" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">which</span>(e <span class="sc">&gt;=</span> o)) <span class="sc">/</span> m</span>
<span id="cb246-25"><a href="#cb246-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-26"><a href="#cb246-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write to dataframe</span></span>
<span id="cb246-27"><a href="#cb246-27" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"gene_list"</span> <span class="ot">=</span> gene_list_name, <span class="st">"o"</span> <span class="ot">=</span> o, <span class="st">"e"</span> <span class="ot">=</span> <span class="fu">mean</span>(e), <span class="st">"sem"</span> <span class="ot">=</span> <span class="fu">sd</span>(e), <span class="st">"p"</span> <span class="ot">=</span> p)</span>
<span id="cb246-28"><a href="#cb246-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb246-29"><a href="#cb246-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb246-30"><a href="#cb246-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">"Finished processing "</span>, gene_list_name))</span>
<span id="cb246-31"><a href="#cb246-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb246-32"><a href="#cb246-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-33"><a href="#cb246-33" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span></span>
<span id="cb246-34"><a href="#cb246-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb246-35"><a href="#cb246-35" aria-hidden="true" tabindex="-1"></a>    my_gene_list<span class="sc">$</span>full_background,</span>
<span id="cb246-36"><a href="#cb246-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(my_gene_list<span class="sc">$</span>full_background),</span>
<span id="cb246-37"><a href="#cb246-37" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">perform_permutation</span>(</span>
<span id="cb246-38"><a href="#cb246-38" aria-hidden="true" tabindex="-1"></a>      .x,</span>
<span id="cb246-39"><a href="#cb246-39" aria-hidden="true" tabindex="-1"></a>      .y,</span>
<span id="cb246-40"><a href="#cb246-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">my_network =</span> res<span class="sc">$</span>full_background<span class="sc">$</span>network,</span>
<span id="cb246-41"><a href="#cb246-41" aria-hidden="true" tabindex="-1"></a>      <span class="at">BG =</span> res<span class="sc">$</span>full_background<span class="sc">$</span>BG,</span>
<span id="cb246-42"><a href="#cb246-42" aria-hidden="true" tabindex="-1"></a>      <span class="at">gene_attribute =</span> res<span class="sc">$</span>full_background[[gene_attribute]],</span>
<span id="cb246-43"><a href="#cb246-43" aria-hidden="true" tabindex="-1"></a>      <span class="at">m =</span> m</span>
<span id="cb246-44"><a href="#cb246-44" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb246-45"><a href="#cb246-45" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb246-46"><a href="#cb246-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Processing ambiguous_pseudobulk_apoe"
[1] "Processing Arterial_pseudobulk_apoe"
[1] "Processing Arteriolar-SMC_pseudobulk_apoe"
[1] "Processing Artirial-FB_pseudobulk_apoe"
[1] "Processing Astrocyte-activated_pseudobulk_apoe"
[1] "Processing Astrocyte-quiescent_pseudobulk_apoe"
[1] "Processing Capillary_pseudobulk_apoe"
[1] "Processing Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe"
[1] "Processing Ex-Neuron-DLC1-HS3ST4_pseudobulk_apoe"
[1] "Processing Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_apoe"
[1] "Processing Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe"
[1] "Processing Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_apoe"
[1] "Processing In-Neuron-ADARB2-GRM7_pseudobulk_apoe"
[1] "Processing In-Neuron-ADARB2-IL1RAPL2-CSCL14_pseudobulk_apoe"
[1] "Processing In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_apoe"
[1] "Processing In-Neuron-DPP10-MEF2C_pseudobulk_apoe"
[1] "Processing In-Neuron-PRKG1-TBC1D4_pseudobulk_apoe"
[1] "Processing In-Neuron-TRHDE-RALYL_pseudobulk_apoe"
[1] "Processing low-feature-cells_pseudobulk_apoe"
[1] "Processing M-Pericyte_pseudobulk_apoe"
[1] "Processing Microglia-activated_pseudobulk_apoe"
[1] "Processing Microglia-quiescent_pseudobulk_apoe"
[1] "Processing Oligo-A_pseudobulk_apoe"
[1] "Processing Oligo-B_pseudobulk_apoe"
[1] "Processing OPC-A_pseudobulk_apoe"
[1] "Processing OPC-B_pseudobulk_apoe"
[1] "Processing Pericyte_pseudobulk_apoe"
[1] "Processing Pericyte-2_pseudobulk_apoe"
[1] "Processing Pericyte-FB_pseudobulk_apoe"
[1] "Processing Perivascular-FB_pseudobulk_apoe"
[1] "Processing Perivascular-FB-KAZN2_pseudobulk_apoe"
[1] "Processing T-Pericyte_pseudobulk_apoe"
[1] "Processing Trans-Endo-to-mural_pseudobulk_apoe"
[1] "Processing Vascular-SMC_pseudobulk_apoe"
[1] "Processing Vascular-SMC-LINC00486_pseudobulk_apoe"
[1] "Processing Venous_pseudobulk_apoe"
[1] "Processing ambiguous_pseudobulk_no_apoe"
[1] "Processing Arterial_pseudobulk_no_apoe"
[1] "Processing Arteriolar-SMC_pseudobulk_no_apoe"
[1] "Processing Artirial-FB_pseudobulk_no_apoe"
[1] "Processing Astrocyte-activated_pseudobulk_no_apoe"
[1] "Processing Astrocyte-quiescent_pseudobulk_no_apoe"
[1] "Processing Capillary_pseudobulk_no_apoe"
[1] "Processing Ex-Neuron-CBLN2-NRGN_pseudobulk_no_apoe"
[1] "Processing Ex-Neuron-DLC1-HS3ST4_pseudobulk_no_apoe"
[1] "Processing Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_no_apoe"
[1] "Processing Ex-Neuron-L3-5-RORB-PLCH1_pseudobulk_no_apoe"
[1] "Processing Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe"
[1] "Processing Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_no_apoe"
[1] "Processing In-Neuron-ADARB2-GRM7_pseudobulk_no_apoe"
[1] "Processing In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_no_apoe"
[1] "Processing In-Neuron-ADARB2-NRG1-RELN_pseudobulk_no_apoe"
[1] "Processing In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe"
[1] "Processing In-Neuron-PRKG1-TBC1D4_pseudobulk_no_apoe"
[1] "Processing In-Neuron-TRHDE-RALYL_pseudobulk_no_apoe"
[1] "Processing low-feature-cells_pseudobulk_no_apoe"
[1] "Processing M-Pericyte_pseudobulk_no_apoe"
[1] "Processing Microglia-activated_pseudobulk_no_apoe"
[1] "Processing Microglia-quiescent_pseudobulk_no_apoe"
[1] "Processing Oligo-A_pseudobulk_no_apoe"
[1] "Processing Oligo-B_pseudobulk_no_apoe"
[1] "Processing OPC-B_pseudobulk_no_apoe"
[1] "Processing Pericyte_pseudobulk_no_apoe"
[1] "Processing Pericyte-2_pseudobulk_no_apoe"
[1] "Processing Pericyte-FB_pseudobulk_no_apoe"
[1] "Processing T-cell-parenchymal-1_pseudobulk_no_apoe"
[1] "Processing T-cell-parenchymal-2_pseudobulk_no_apoe"
[1] "Processing T-Pericyte_pseudobulk_no_apoe"
[1] "Processing Trans-Endo-to-mural_pseudobulk_no_apoe"
[1] "Processing Vascular-SMC_pseudobulk_no_apoe"
[1] "Processing Vascular-SMC-LINC00486_pseudobulk_no_apoe"
[1] "Processing Venous_pseudobulk_no_apoe"
[1] "Processing Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Processing Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1"
[1] "Processing Trans-Endo-to-mural_sig-magma-sig-genes_stroke_35k10k_all_controls_top-five-percent_level1"
[1] "Processing Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Astro_sig-magma-sig-genes_wmh_35k10k_all_controls_top-ten-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Processing Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Processing OPC_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Processing Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1"
[1] "Processing Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing Astro_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing Neuron_in_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level1"
[1] "Processing SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing Fibroblast_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing Fibroblast_sig-magma-sig-genes_wmh_35k10k_controls_vascular_top-five-percent_level1"
[1] "Processing Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1"
[1] "Processing Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2"
[1] "Processing Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2"
[1] "Processing Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Processing Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Processing Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"
[1] "Processing Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Processing In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Processing Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Processing Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level2"
[1] "Processing Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2"
[1] "Processing T-cell-vascular_sig-magma-sig-genes_Kunkle_2019_35k10k_controls_vascular_top-five-percent_level2"
[1] "Processing M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Processing Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Processing Arteriolar-SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Processing Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2"
[1] "Processing Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Meningeal-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing T-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Vascular-SMC-LINC00486_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Meningeal-FB_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Capillary_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing T-cell-mixed_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2"
[1] "Processing sig_magma_celltypes_NA"
[1] "Processing Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"
[1] "Processing Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA"
[1] "Processing Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"</code></pre>
</div>
</div></div>
<section id="results" class="level3">
<h3 class="anchored" data-anchor-id="results">Results</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make group column for pseudobulk and magma genes</span></span>
<span id="cb248-2"><a href="#cb248-2" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>input_type <span class="ot">&lt;-</span></span>
<span id="cb248-3"><a href="#cb248-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"_pseudobulk"</span>, results<span class="sc">$</span>gene_list),</span>
<span id="cb248-4"><a href="#cb248-4" aria-hidden="true" tabindex="-1"></a>         <span class="st">"pseudobulk"</span>,</span>
<span id="cb248-5"><a href="#cb248-5" aria-hidden="true" tabindex="-1"></a>         <span class="st">"magma"</span>)</span>
<span id="cb248-6"><a href="#cb248-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Add adjusted pvalues</span></span>
<span id="cb248-7"><a href="#cb248-7" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>pval <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(results<span class="sc">$</span>p <span class="sc">==</span> <span class="dv">0</span>, <span class="fl">1e-8</span>, results<span class="sc">$</span>p)</span>
<span id="cb248-8"><a href="#cb248-8" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> results <span class="sc">|&gt;</span></span>
<span id="cb248-9"><a href="#cb248-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(input_type) <span class="sc">|&gt;</span></span>
<span id="cb248-10"><a href="#cb248-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">padj =</span> <span class="fu">p.adjust</span>(pval, <span class="at">method =</span> <span class="st">"bonferroni"</span>))</span>
<span id="cb248-11"><a href="#cb248-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-12"><a href="#cb248-12" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(results, <span class="fu">paste0</span>(output_file, <span class="st">".tsv"</span>))</span>
<span id="cb248-13"><a href="#cb248-13" aria-hidden="true" tabindex="-1"></a><span class="co">#results &lt;- readr::read_tsv(paste0(output_file, ".tsv"))</span></span>
<span id="cb248-14"><a href="#cb248-14" aria-hidden="true" tabindex="-1"></a>sig_celltypes <span class="ot">&lt;-</span> results <span class="sc">|&gt;</span></span>
<span id="cb248-15"><a href="#cb248-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb248-16"><a href="#cb248-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Table with results of test</span></span>
<span id="cb248-17"><a href="#cb248-17" aria-hidden="true" tabindex="-1"></a>results <span class="sc">|&gt;</span></span>
<span id="cb248-18"><a href="#cb248-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb248-19"><a href="#cb248-19" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">caption =</span> <span class="st">"Test if the number of interactions is higher than expected by chance"</span>,</span>
<span id="cb248-20"><a href="#cb248-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">rownames =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="fu">formatRound</span>(<span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"e"</span>, <span class="st">"sem"</span>), <span class="at">digits =</span> <span class="fu">c</span>(<span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-9b1e127c3d2775ea5c4a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9b1e127c3d2775ea5c4a">{"x":{"filter":"none","vertical":false,"caption":"<caption>Test if the number of interactions is higher than expected by chance<\/caption>","data":[["Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe","Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe","low-feature-cells_pseudobulk_apoe","Microglia-quiescent_pseudobulk_apoe","Oligo-A_pseudobulk_apoe","Oligo-B_pseudobulk_apoe","Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe","In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe","T-Pericyte_pseudobulk_no_apoe","Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1","T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1","Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1","Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1","T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1","Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1","Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1","Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1","Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1","Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1","T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1","Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1","Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1","T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1","Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1","Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1","Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1","Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1","Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1","Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1","Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1","Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1","SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1","T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1","Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1","Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2","Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2","Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2","Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2","Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2","Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2","Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2","Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2","Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2","Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2","Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2","Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2","Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2","Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2","Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2","Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2","sig_magma_celltypes_NA","Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA","Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA","Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"],[11,464,55,50,2356,3780,1710,12,4489,82,50,36,182,110,23,76,85,66,156,44,58,61,83,117,304,119,165,13,40,35,53,102,56,120,46,17,34,26,17,20,12,113,244,61,113,244,66,66,113,167,5,113,190,91,22,6,12,406,81,269,72],[2.8862,352.9053,30.4807,26.2601,2103.213,3472.499,1518.936,2.8985,3517.8165,19.5148,11.0847,5.8179,54.3052,31.4221,6.1257,19.2569,16.2561,18.0314,31.9784,11.3774,14.4936,14.3817,17.3165,44.0093,85.9072,45.484,49.3812,4.807,16.2589,15.8502,13.8698,28.5485,14.3129,25.0104,16.2641,6.9251,5.8594,3.7102,3.3909,6.4605,2.3511,20.8534,63.1275,19.747,20.83,63.1702,25.4577,20.673,20.1381,39.6795,0.8774999999999999,20.1427,43.0732,20.4292,4.708,0.499,2.9823,124.6025,28.1131,74.2419,25.8279],[1.690483757252152,22.30244990661752,6.395632722553116,4.96929747366146,70.40625636329139,97.0527934861864,54.39634179060224,1.642518656440065,96.48003144380054,4.15609290931123,3.101755631702775,2.235853191501937,8.536436026469325,5.686647992142529,2.029657966541682,4.169249411352768,3.883969787746966,4.092394019260717,5.485940484028188,3.088773752208024,3.549735074890828,3.576238814509713,3.955488887512379,6.827025291987261,10.01213326126076,7.093347274563381,7.282443955087873,1.985483622325163,3.609012785697598,3.462738658331233,3.383783822778271,5.227100535478613,3.534663063758331,4.78401301399273,3.583132020105783,2.24129255770703,2.308065067634716,1.796479527818056,1.698289023859265,2.408738248958404,1.392559769730272,4.303342923359885,9.567026526024424,4.314864169968798,4.34448932082309,9.367839003780972,5.605600144970968,4.445677384239715,4.15311367835297,6.413290323950564,0.8505093101246223,4.216386426959719,6.684737117571124,4.952256035786558,2.030061111102688,0.6695101376248446,1.586643770304281,14.05438174786182,5.885624218997058,10.51572358110669,5.594873711817374],[0.0001,0,0.0005,0,0.0005,0.0005999999999999999,0.0004,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.0001,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.0003,0,0,0,0,0,0,0,0,0,0],["pseudobulk","pseudobulk","pseudobulk","pseudobulk","pseudobulk","pseudobulk","pseudobulk","pseudobulk","pseudobulk","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma","magma"],[0.0001,1e-08,0.0005,1e-08,0.0005,0.0005999999999999999,0.0004,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,0.0001,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,0.0003,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08,1e-08],[0.007200000000000001,7.2e-07,0.036,7.2e-07,0.036,0.0432,0.0288,7.2e-07,7.2e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,0.009600000000000001,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,0.0288,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07,9.600000000000001e-07]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>gene_list<\/th>\n      <th>o<\/th>\n      <th>e<\/th>\n      <th>sem<\/th>\n      <th>p<\/th>\n      <th>input_type<\/th>\n      <th>pval<\/th>\n      <th>padj<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":2,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"targets":3,"render":"function(data, type, row, meta) {\n    return type !== 'display' ? data : DTWidget.formatRound(data, 2, 3, \",\", \".\", null);\n  }"},{"className":"dt-right","targets":[1,2,3,4,6,7]},{"name":"gene_list","targets":0},{"name":"o","targets":1},{"name":"e","targets":2},{"name":"sem","targets":3},{"name":"p","targets":4},{"name":"input_type","targets":5},{"name":"pval","targets":6},{"name":"padj","targets":7}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render"],"jsHooks":[]}</script>
</div>
</div></div>
<ul>
<li><p><strong>o</strong> observed number of interactions.</p></li>
<li><p><strong>e</strong> expected number of interactions based on 10^{4} randomizations.</p></li>
<li><p><strong>sem</strong> standard error of the mean based on 10^{4} randomizations.</p></li>
<li><p><strong>p</strong> estimated p value based on 10^{4} randomizations.</p></li>
<li><p><strong>Note</strong> If a weighted network is provided then it is a test for the sum of the edges weights.</p></li>
</ul>
</section>
</section>
<section id="louvain-clustering" class="level2">
<h2 class="anchored" data-anchor-id="louvain-clustering">Louvain clustering</h2>
<ul>
<li>run through funciton code and make sure its working</li>
</ul>
<div class="cell-container"><div class="cell-decorator"><pre>In [27]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a>perform_louvain_clustering <span class="ot">&lt;-</span> <span class="cf">function</span>(graph,</span>
<span id="cb249-2"><a href="#cb249-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">min_cluster_size =</span> <span class="dv">1</span>,</span>
<span id="cb249-3"><a href="#cb249-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">color_palette =</span> <span class="st">"Dark2"</span>,</span>
<span id="cb249-4"><a href="#cb249-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">default_color =</span> <span class="st">"grey"</span>,</span>
<span id="cb249-5"><a href="#cb249-5" aria-hidden="true" tabindex="-1"></a>                                       label) {</span>
<span id="cb249-6"><a href="#cb249-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Perform Louvain clustering</span></span>
<span id="cb249-7"><a href="#cb249-7" aria-hidden="true" tabindex="-1"></a>  louvain <span class="ot">&lt;-</span> <span class="fu">cluster_louvain</span>(graph)</span>
<span id="cb249-8"><a href="#cb249-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-9"><a href="#cb249-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get module membership</span></span>
<span id="cb249-10"><a href="#cb249-10" aria-hidden="true" tabindex="-1"></a>  module_membership <span class="ot">&lt;-</span> <span class="fu">membership</span>(louvain)</span>
<span id="cb249-11"><a href="#cb249-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the sizes of each module</span></span>
<span id="cb249-12"><a href="#cb249-12" aria-hidden="true" tabindex="-1"></a>  module_sizes <span class="ot">&lt;-</span> <span class="fu">sizes</span>(louvain)</span>
<span id="cb249-13"><a href="#cb249-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter out small clusters</span></span>
<span id="cb249-14"><a href="#cb249-14" aria-hidden="true" tabindex="-1"></a>  large_modules <span class="ot">&lt;-</span> <span class="fu">names</span>(module_sizes[module_sizes <span class="sc">&gt;=</span> min_cluster_size])</span>
<span id="cb249-15"><a href="#cb249-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-16"><a href="#cb249-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a filtered membership vector</span></span>
<span id="cb249-17"><a href="#cb249-17" aria-hidden="true" tabindex="-1"></a>  filtered_membership <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(module_membership <span class="sc">%in%</span> large_modules, module_membership, <span class="dv">0</span>)</span>
<span id="cb249-18"><a href="#cb249-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the number of remaining modules</span></span>
<span id="cb249-19"><a href="#cb249-19" aria-hidden="true" tabindex="-1"></a>  num_modules <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(filtered_membership[filtered_membership <span class="sc">!=</span> <span class="dv">0</span>]))</span>
<span id="cb249-20"><a href="#cb249-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-21"><a href="#cb249-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate distinct colors for each module</span></span>
<span id="cb249-22"><a href="#cb249-22" aria-hidden="true" tabindex="-1"></a>  colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">n =</span> num_modules, <span class="at">name =</span> color_palette)</span>
<span id="cb249-23"><a href="#cb249-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-24"><a href="#cb249-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a color vector with the default color for small clusters</span></span>
<span id="cb249-25"><a href="#cb249-25" aria-hidden="true" tabindex="-1"></a>  module_colors <span class="ot">&lt;-</span> <span class="fu">rep</span>(default_color, <span class="fu">length</span>(filtered_membership))</span>
<span id="cb249-26"><a href="#cb249-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assign colors to large modules</span></span>
<span id="cb249-27"><a href="#cb249-27" aria-hidden="true" tabindex="-1"></a>  module_colors[filtered_membership <span class="sc">!=</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> colors[<span class="fu">as.integer</span>(<span class="fu">factor</span>(filtered_membership[filtered_membership <span class="sc">!=</span> <span class="dv">0</span>]))]</span>
<span id="cb249-28"><a href="#cb249-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-29"><a href="#cb249-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get gene names</span></span>
<span id="cb249-30"><a href="#cb249-30" aria-hidden="true" tabindex="-1"></a>  gene_names <span class="ot">&lt;-</span> <span class="fu">V</span>(graph)<span class="sc">$</span>name</span>
<span id="cb249-31"><a href="#cb249-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-32"><a href="#cb249-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create dataframe</span></span>
<span id="cb249-33"><a href="#cb249-33" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb249-34"><a href="#cb249-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene =</span> gene_names,</span>
<span id="cb249-35"><a href="#cb249-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">module =</span> filtered_membership,</span>
<span id="cb249-36"><a href="#cb249-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">module_color =</span> module_colors,</span>
<span id="cb249-37"><a href="#cb249-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> label</span>
<span id="cb249-38"><a href="#cb249-38" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb249-39"><a href="#cb249-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb249-40"><a href="#cb249-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(result)</span>
<span id="cb249-41"><a href="#cb249-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb249-42"><a href="#cb249-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-43"><a href="#cb249-43" aria-hidden="true" tabindex="-1"></a><span class="co"># remove any subgraphs that are empty</span></span>
<span id="cb249-44"><a href="#cb249-44" aria-hidden="true" tabindex="-1"></a>subgraphs <span class="ot">&lt;-</span> my_subgraph<span class="sc">$</span>full_background[<span class="fu">map_lgl</span>(my_subgraph<span class="sc">$</span>full_background, <span class="sc">~</span> <span class="fu">length</span>(.x) <span class="sc">&gt;</span> <span class="dv">0</span>)]</span>
<span id="cb249-45"><a href="#cb249-45" aria-hidden="true" tabindex="-1"></a>module_assignment <span class="ot">&lt;-</span></span>
<span id="cb249-46"><a href="#cb249-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(</span>
<span id="cb249-47"><a href="#cb249-47" aria-hidden="true" tabindex="-1"></a>    subgraphs,</span>
<span id="cb249-48"><a href="#cb249-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(subgraphs),</span>
<span id="cb249-49"><a href="#cb249-49" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">perform_louvain_clustering</span>(.x, <span class="at">min_cluster_size =</span> <span class="dv">3</span>, <span class="at">label =</span> .y)</span>
<span id="cb249-50"><a href="#cb249-50" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in brewer.pal(n = num_modules, name = color_palette): n too large, allowed maximum for palette Dark2 is 8
Returning the palette you asked for with that many colors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels
Warning in brewer.pal(n = num_modules, name = color_palette): minimal value for n is 3, returning requested palette with 3 different levels</code></pre>
</div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [28]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a>ddd <span class="ot">&lt;-</span> <span class="fu">load</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/994_scripts_from_emeka/table_gene_full.Rdata"</span>))</span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>info_annot_gene<span class="ot">&lt;-</span><span class="fu">read.table</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"04_data_analysis/994_scripts_from_emeka/NCBI37.3.gene.loc"</span>),<span class="at">h=</span>F,<span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(info_annot_gene) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'entrezid'</span>, <span class="st">'chr'</span>, <span class="st">'start'</span>, <span class="st">'stop'</span>, <span class="st">'strand'</span>, <span class="st">"hgnc_symbol"</span>)</span>
<span id="cb253-4"><a href="#cb253-4" aria-hidden="true" tabindex="-1"></a>table_gene_du_id<span class="ot">&lt;-</span><span class="fu">merge</span>(table_gene_du,info_annot_gene,<span class="at">by=</span><span class="fu">c</span>(<span class="st">"hgnc_symbol"</span>))</span>
<span id="cb253-5"><a href="#cb253-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(table_gene_du_id)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"gene"</span> </span>
<span id="cb253-6"><a href="#cb253-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-7"><a href="#cb253-7" aria-hidden="true" tabindex="-1"></a>module_assignment <span class="ot">&lt;-</span> <span class="fu">map</span>(module_assignment, dplyr<span class="sc">::</span>left_join, table_gene_du_id, <span class="at">by =</span> <span class="st">"gene"</span>)</span>
<span id="cb253-8"><a href="#cb253-8" aria-hidden="true" tabindex="-1"></a>module_assignment <span class="ot">&lt;-</span> <span class="fu">map</span>(module_assignment, <span class="sc">~</span> .x <span class="sc">|&gt;</span></span>
<span id="cb253-9"><a href="#cb253-9" aria-hidden="true" tabindex="-1"></a>           dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_ids, <span class="at">by =</span> <span class="fu">join_by</span>(gene <span class="sc">==</span> ensembl)) <span class="sc">|&gt;</span></span>
<span id="cb253-10"><a href="#cb253-10" aria-hidden="true" tabindex="-1"></a>           dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gene_symbol =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(hgnc_symbol), gene.y, hgnc_symbol)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [29]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a>modules <span class="ot">&lt;-</span> <span class="fu">list_rbind</span>(module_assignment) <span class="sc">|&gt;</span></span>
<span id="cb254-2"><a href="#cb254-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(label <span class="sc">%in%</span> sig_celltypes<span class="sc">$</span>gene_list) <span class="sc">|&gt;</span></span>
<span id="cb254-3"><a href="#cb254-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"low-feature-cell"</span>, label))</span>
<span id="cb254-4"><a href="#cb254-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-5"><a href="#cb254-5" aria-hidden="true" tabindex="-1"></a>ad_all_controls <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb254-6"><a href="#cb254-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Lymphocyte activation"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Dephosphorylation"</span>),</span>
<span id="cb254-7"><a href="#cb254-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Supramolecular fibre organisation"</span>),</span>
<span id="cb254-8"><a href="#cb254-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Viral process"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Axon terminus"</span>),</span>
<span id="cb254-9"><a href="#cb254-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Cell adhesion"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Lymphocyte activation"</span>),</span>
<span id="cb254-10"><a href="#cb254-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Viral process"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Actin filament"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>),</span>
<span id="cb254-11"><a href="#cb254-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Supramolecular fibre organisation"</span>)</span>
<span id="cb254-12"><a href="#cb254-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb254-13"><a href="#cb254-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-14"><a href="#cb254-14" aria-hidden="true" tabindex="-1"></a><span class="co"># function to make df per vector</span></span>
<span id="cb254-15"><a href="#cb254-15" aria-hidden="true" tabindex="-1"></a>make_ppi_df <span class="ot">&lt;-</span> <span class="cf">function</span>(module_labels, celltype, gwas, fraction, <span class="at">ontology =</span> <span class="cn">NA</span>) {</span>
<span id="cb254-16"><a href="#cb254-16" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span></span>
<span id="cb254-17"><a href="#cb254-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tibble</span>(</span>
<span id="cb254-18"><a href="#cb254-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">module_label =</span> module_labels,</span>
<span id="cb254-19"><a href="#cb254-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">module =</span> <span class="fu">as.numeric</span>(<span class="fu">names</span>(module_labels)),</span>
<span id="cb254-20"><a href="#cb254-20" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> celltype,</span>
<span id="cb254-21"><a href="#cb254-21" aria-hidden="true" tabindex="-1"></a>      <span class="at">gwas =</span> gwas,</span>
<span id="cb254-22"><a href="#cb254-22" aria-hidden="true" tabindex="-1"></a>      <span class="at">fraction =</span> fraction,</span>
<span id="cb254-23"><a href="#cb254-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">ontology =</span> ontology</span>
<span id="cb254-24"><a href="#cb254-24" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb254-25"><a href="#cb254-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb254-26"><a href="#cb254-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb254-27"><a href="#cb254-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-28"><a href="#cb254-28" aria-hidden="true" tabindex="-1"></a>module_labels <span class="ot">&lt;-</span></span>
<span id="cb254-29"><a href="#cb254-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(ad_all_controls,</span>
<span id="cb254-30"><a href="#cb254-30" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(ad_all_controls),</span>
<span id="cb254-31"><a href="#cb254-31" aria-hidden="true" tabindex="-1"></a>       make_ppi_df,</span>
<span id="cb254-32"><a href="#cb254-32" aria-hidden="true" tabindex="-1"></a>       <span class="st">"Bellenguez"</span>,</span>
<span id="cb254-33"><a href="#cb254-33" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all_controls"</span>,</span>
<span id="cb254-34"><a href="#cb254-34" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all"</span>) <span class="sc">|&gt;</span></span>
<span id="cb254-35"><a href="#cb254-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span>
<span id="cb254-36"><a href="#cb254-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-37"><a href="#cb254-37" aria-hidden="true" tabindex="-1"></a>modules <span class="ot">&lt;-</span> modules <span class="sc">|&gt;</span></span>
<span id="cb254-38"><a href="#cb254-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(module_labels, <span class="at">by =</span> <span class="fu">join_by</span>(label, module))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [30]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a>plot_network_ggraph <span class="ot">&lt;-</span> <span class="cf">function</span>(group, module_df, <span class="at">overall_network =</span> my_network,</span>
<span id="cb255-2"><a href="#cb255-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">network_layout =</span> <span class="st">"auto"</span>) {</span>
<span id="cb255-3"><a href="#cb255-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(ggraph)</span>
<span id="cb255-4"><a href="#cb255-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(tidygraph)</span>
<span id="cb255-5"><a href="#cb255-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-6"><a href="#cb255-6" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> module_df <span class="sc">|&gt;</span></span>
<span id="cb255-7"><a href="#cb255-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(group, label)) <span class="sc">|&gt;</span></span>
<span id="cb255-8"><a href="#cb255-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(module) <span class="sc">|&gt;</span></span>
<span id="cb255-9"><a href="#cb255-9" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">genes_in_module =</span> <span class="fu">n</span>()) <span class="sc">|&gt;</span></span>
<span id="cb255-10"><a href="#cb255-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">|&gt;</span></span>
<span id="cb255-11"><a href="#cb255-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(genes_in_module <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">|&gt;</span></span>
<span id="cb255-12"><a href="#cb255-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If there's no module labels just replace with module numbers</span></span>
<span id="cb255-13"><a href="#cb255-13" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">module_label =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(module_label), <span class="fu">as.character</span>(module), module_label))</span>
<span id="cb255-14"><a href="#cb255-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-15"><a href="#cb255-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-16"><a href="#cb255-16" aria-hidden="true" tabindex="-1"></a>  subgraph <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> overall_network, <span class="at">vids =</span> df<span class="sc">$</span>gene)</span>
<span id="cb255-17"><a href="#cb255-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a named vector for easy lookup</span></span>
<span id="cb255-18"><a href="#cb255-18" aria-hidden="true" tabindex="-1"></a>ensembl_to_symbol <span class="ot">&lt;-</span> <span class="fu">setNames</span>(df<span class="sc">$</span>hgnc_symbol, df<span class="sc">$</span>gene)</span>
<span id="cb255-19"><a href="#cb255-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-20"><a href="#cb255-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbols to the graph vertices</span></span>
<span id="cb255-21"><a href="#cb255-21" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(subgraph)<span class="sc">$</span>hgnc_symbol <span class="ot">&lt;-</span> ensembl_to_symbol[<span class="fu">V</span>(subgraph)<span class="sc">$</span>name]</span>
<span id="cb255-22"><a href="#cb255-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert igraph to tidygraph</span></span>
<span id="cb255-23"><a href="#cb255-23" aria-hidden="true" tabindex="-1"></a>  tbl_graph <span class="ot">&lt;-</span> <span class="fu">as_tbl_graph</span>(subgraph)</span>
<span id="cb255-24"><a href="#cb255-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-25"><a href="#cb255-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add module information to the graph</span></span>
<span id="cb255-26"><a href="#cb255-26" aria-hidden="true" tabindex="-1"></a>  tbl_graph <span class="ot">&lt;-</span> tbl_graph <span class="sc">%&gt;%</span></span>
<span id="cb255-27"><a href="#cb255-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">activate</span>(nodes) <span class="sc">%&gt;%</span></span>
<span id="cb255-28"><a href="#cb255-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">module_number =</span> df<span class="sc">$</span>module,</span>
<span id="cb255-29"><a href="#cb255-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">module_color =</span> df<span class="sc">$</span>module_color,</span>
<span id="cb255-30"><a href="#cb255-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">module_label =</span> df<span class="sc">$</span>module_label,</span>
<span id="cb255-31"><a href="#cb255-31" aria-hidden="true" tabindex="-1"></a>           <span class="at">group =</span> df<span class="sc">$</span>label,</span>
<span id="cb255-32"><a href="#cb255-32" aria-hidden="true" tabindex="-1"></a>           <span class="at">gene_name =</span> df<span class="sc">$</span>gene_symbol)</span>
<span id="cb255-33"><a href="#cb255-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-34"><a href="#cb255-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot using ggraph</span></span>
<span id="cb255-35"><a href="#cb255-35" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggraph</span>(tbl_graph, <span class="at">layout =</span> network_layout) <span class="sc">+</span>  <span class="co"># 'fr' layout for Fruchterman-Reingold</span></span>
<span id="cb255-36"><a href="#cb255-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_edge_link</span>(<span class="fu">aes</span>(<span class="at">edge_alpha =</span> <span class="fl">0.5</span>), <span class="at">edge_colour =</span> <span class="st">"grey"</span>, </span>
<span id="cb255-37"><a href="#cb255-37" aria-hidden="true" tabindex="-1"></a>                   <span class="at">edge_linetype =</span> <span class="st">"solid"</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb255-38"><a href="#cb255-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_node_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> module_label), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb255-39"><a href="#cb255-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_node_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> gene_name), <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb255-40"><a href="#cb255-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(df<span class="sc">$</span>module_color, df<span class="sc">$</span>module_label)) <span class="sc">+</span></span>
<span id="cb255-41"><a href="#cb255-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb255-42"><a href="#cb255-42" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb255-43"><a href="#cb255-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">5</span>))) <span class="sc">+</span></span>
<span id="cb255-44"><a href="#cb255-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Module Function"</span>, <span class="at">title =</span> <span class="fu">unique</span>(df<span class="sc">$</span>celltype))</span>
<span id="cb255-45"><a href="#cb255-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-46"><a href="#cb255-46" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(df<span class="sc">$</span>label)) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb255-47"><a href="#cb255-47" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span> group)  <span class="co"># Facet by group</span></span>
<span id="cb255-48"><a href="#cb255-48" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb255-49"><a href="#cb255-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-50"><a href="#cb255-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb255-51"><a href="#cb255-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb255-52"><a href="#cb255-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-53"><a href="#cb255-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the function to subset a single string up to the last underscore</span></span>
<span id="cb255-54"><a href="#cb255-54" aria-hidden="true" tabindex="-1"></a>subset_until_last_underscore <span class="ot">&lt;-</span> <span class="cf">function</span>(input_string) {</span>
<span id="cb255-55"><a href="#cb255-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find the position of the last underscore</span></span>
<span id="cb255-56"><a href="#cb255-56" aria-hidden="true" tabindex="-1"></a>  last_underscore_position <span class="ot">&lt;-</span> <span class="fu">str_locate_all</span>(input_string, <span class="st">"_"</span>)[[<span class="dv">1</span>]][, <span class="dv">1</span>] <span class="sc">%&gt;%</span> <span class="fu">max</span>()</span>
<span id="cb255-57"><a href="#cb255-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-58"><a href="#cb255-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the string up to the last underscore</span></span>
<span id="cb255-59"><a href="#cb255-59" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(last_underscore_position)) {</span>
<span id="cb255-60"><a href="#cb255-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">str_sub</span>(input_string, <span class="dv">1</span>, last_underscore_position <span class="sc">-</span> <span class="dv">1</span>))</span>
<span id="cb255-61"><a href="#cb255-61" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb255-62"><a href="#cb255-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(input_string)</span>
<span id="cb255-63"><a href="#cb255-63" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb255-64"><a href="#cb255-64" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb255-65"><a href="#cb255-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-66"><a href="#cb255-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each element of the vector</span></span>
<span id="cb255-67"><a href="#cb255-67" aria-hidden="true" tabindex="-1"></a>modules<span class="sc">$</span>celltype_label_full <span class="ot">&lt;-</span> <span class="fu">sapply</span>(modules<span class="sc">$</span>label, subset_until_last_underscore)</span>
<span id="cb255-68"><a href="#cb255-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract substring up to the first underscore</span></span>
<span id="cb255-69"><a href="#cb255-69" aria-hidden="true" tabindex="-1"></a>modules<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(modules<span class="sc">$</span>label, <span class="st">"^[^_]*"</span>)</span>
<span id="cb255-70"><a href="#cb255-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-71"><a href="#cb255-71" aria-hidden="true" tabindex="-1"></a><span class="co"># I want to make sure the Amyloid groups get the same colour label</span></span>
<span id="cb255-72"><a href="#cb255-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to assign colors</span></span>
<span id="cb255-73"><a href="#cb255-73" aria-hidden="true" tabindex="-1"></a>assign_amyloid_color <span class="ot">&lt;-</span> <span class="cf">function</span>(df, amyloid_color) {</span>
<span id="cb255-74"><a href="#cb255-74" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb255-75"><a href="#cb255-75" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(label) <span class="sc">%&gt;%</span></span>
<span id="cb255-76"><a href="#cb255-76" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb255-77"><a href="#cb255-77" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Check if "Amyloid" is in the group</span></span>
<span id="cb255-78"><a href="#cb255-78" aria-hidden="true" tabindex="-1"></a>      <span class="at">amyloid_present =</span> <span class="fu">any</span>(module_label <span class="sc">==</span> <span class="st">"Amyloid"</span>),</span>
<span id="cb255-79"><a href="#cb255-79" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Get the original color of "Amyloid" if present</span></span>
<span id="cb255-80"><a href="#cb255-80" aria-hidden="true" tabindex="-1"></a>      <span class="at">original_amyloid_color =</span> <span class="fu">ifelse</span>(module_label <span class="sc">==</span> <span class="st">"Amyloid"</span>, module_color, <span class="cn">NA</span>),</span>
<span id="cb255-81"><a href="#cb255-81" aria-hidden="true" tabindex="-1"></a>      <span class="at">original_amyloid_color =</span> dplyr<span class="sc">::</span><span class="fu">first</span>(<span class="fu">na.omit</span>(original_amyloid_color))</span>
<span id="cb255-82"><a href="#cb255-82" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb255-83"><a href="#cb255-83" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb255-84"><a href="#cb255-84" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb255-85"><a href="#cb255-85" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Assign the desired color to "Amyloid"</span></span>
<span id="cb255-86"><a href="#cb255-86" aria-hidden="true" tabindex="-1"></a>      <span class="at">module_color =</span> <span class="fu">case_when</span>(</span>
<span id="cb255-87"><a href="#cb255-87" aria-hidden="true" tabindex="-1"></a>        module_label <span class="sc">==</span> <span class="st">"Amyloid"</span> <span class="sc">~</span> amyloid_color,</span>
<span id="cb255-88"><a href="#cb255-88" aria-hidden="true" tabindex="-1"></a>        <span class="at">.default =</span> module_color)) <span class="sc">|&gt;</span></span>
<span id="cb255-89"><a href="#cb255-89" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Swap the original "Amyloid" color with the new "Amyloid" color in other labels</span></span>
<span id="cb255-90"><a href="#cb255-90" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">module_color =</span> <span class="fu">case_when</span>(</span>
<span id="cb255-91"><a href="#cb255-91" aria-hidden="true" tabindex="-1"></a>      module_color <span class="sc">==</span> amyloid_color <span class="sc">&amp;</span> module_label <span class="sc">!=</span> <span class="st">"Amyloid"</span> <span class="sc">~</span> original_amyloid_color,</span>
<span id="cb255-92"><a href="#cb255-92" aria-hidden="true" tabindex="-1"></a>      <span class="at">.default =</span> module_color)</span>
<span id="cb255-93"><a href="#cb255-93" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">%&gt;%</span></span>
<span id="cb255-94"><a href="#cb255-94" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>amyloid_present, <span class="sc">-</span>original_amyloid_color)</span>
<span id="cb255-95"><a href="#cb255-95" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb255-96"><a href="#cb255-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb255-97"><a href="#cb255-97" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb255-98"><a href="#cb255-98" aria-hidden="true" tabindex="-1"></a>modules <span class="ot">&lt;-</span> <span class="fu">assign_amyloid_color</span>(modules, <span class="st">"#D95F02"</span>)</span>
<span id="cb255-99"><a href="#cb255-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-100"><a href="#cb255-100" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">map</span>(<span class="fu">unique</span>(modules<span class="sc">$</span>label), plot_network_ggraph, modules, <span class="at">network_layout =</span> <span class="st">"fr"</span>) <span class="sc">|&gt;</span></span>
<span id="cb255-101"><a href="#cb255-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">unique</span>(modules<span class="sc">$</span>label))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggraph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ggraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:sp':

    geometry</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: tidygraph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidygraph'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:XVector':

    slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:biomaRt':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:igraph':

    groups</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:ensembldb':

    filter, select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:AnnotationDbi':

    select</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:IRanges':

    active, slice</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:S4Vectors':

    active, rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:reshape':

    rename</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb270"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1" aria-hidden="true" tabindex="-1"></a>qs<span class="sc">::</span><span class="fu">qsave</span>(plots, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/003_final_figures/001_data_for_plots/ppi_network_plots.qs"</span>))</span>
<span id="cb270-2"><a href="#cb270-2" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(plots, <span class="fu">unique</span>(modules<span class="sc">$</span>label), <span class="sc">~</span> <span class="fu">ggsave</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks"</span>, <span class="fu">paste0</span>(.y, <span class="st">".png"</span>)), <span class="at">plot =</span> .x, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">width =</span> <span class="dv">14</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 233 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 342 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 166 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 326 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe.png"

$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe.png"

$`Microglia-quiescent_pseudobulk_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-quiescent_pseudobulk_apoe.png"

$`Oligo-A_pseudobulk_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Oligo-A_pseudobulk_apoe.png"

$`Oligo-B_pseudobulk_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Oligo-B_pseudobulk_apoe.png"

$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe.png"

$`In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe.png"

$`T-Pericyte_pseudobulk_no_apoe`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-Pericyte_pseudobulk_no_apoe.png"

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1.png"

$`T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1.png"

$`Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1.png"

$`Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1.png"

$`T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1.png"

$`Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1.png"

$`Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1.png"

$`Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1.png"

$`Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1.png"

$`Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1.png"

$`T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1.png"

$`Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1.png"

$`Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1.png"

$`T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1.png"

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1.png"

$`Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1.png"

$`Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1.png"

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1.png"

$`Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1.png"

$`Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1.png"

$`Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1.png"

$`Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1.png"

$`SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1.png"

$`T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1.png"

$`Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1.png"

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2.png"

$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2.png"

$`Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2.png"

$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2.png"

$`Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2.png"

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2.png"

$`Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2.png"

$`Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2.png"

$`Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2.png"

$`Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2.png"

$sig_magma_celltypes_NA
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/sig_magma_celltypes_NA.png"

$`Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA.png"

$`Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA.png"

$`Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA`
[1] "/mnt/b57480ca-5ec4-4912-b90b-3982312c5328/UK Dementia Research Institute Dropbox/Gabriel Bernardo Harrington/backup/research/blood-brain-barrier-in-ad/05_figures/990_shared_figures/002_pseudobulk/ppi_sig_networks/Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA.png"</code></pre>
</div>
</div></div>
</section>
<section id="for-each-module" class="level2">
<h2 class="anchored" data-anchor-id="for-each-module">For each module</h2>
<div class="cell-container"><div class="cell-decorator"><pre>In [31]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb277"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb277-2"><a href="#cb277-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot each community separately</span></span>
<span id="cb277-3"><a href="#cb277-3" aria-hidden="true" tabindex="-1"></a><span class="co">#par(mfrow = c(2, 3))</span></span>
<span id="cb277-4"><a href="#cb277-4" aria-hidden="true" tabindex="-1"></a>mcl <span class="ot">&lt;-</span> <span class="fu">membership</span>(cl)</span>
<span id="cb277-5"><a href="#cb277-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">which</span>(<span class="fu">table</span>(cl<span class="sc">$</span>membership) <span class="sc">&gt;=</span> min_cluster_size)) {</span>
<span id="cb277-6"><a href="#cb277-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#i=1</span></span>
<span id="cb277-7"><a href="#cb277-7" aria-hidden="true" tabindex="-1"></a>  community_vertices <span class="ot">&lt;-</span> <span class="fu">which</span>(mcl <span class="sc">==</span> i)</span>
<span id="cb277-8"><a href="#cb277-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb277-9"><a href="#cb277-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a subgraph for the current community</span></span>
<span id="cb277-10"><a href="#cb277-10" aria-hidden="true" tabindex="-1"></a>  subgraph <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(my_subgraph, community_vertices)</span>
<span id="cb277-11"><a href="#cb277-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># xx &lt;- subset(module_assignment, module == i)</span></span>
<span id="cb277-12"><a href="#cb277-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">V</span>(subgraph)<span class="sc">$</span>hgnc_symbol <span class="ot">&lt;-</span> module_assignment[<span class="fu">match</span>(<span class="fu">V</span>(subgraph)<span class="sc">$</span>name, module_assignment<span class="sc">$</span>gene), <span class="st">"hgnc_symbol"</span>]</span>
<span id="cb277-13"><a href="#cb277-13" aria-hidden="true" tabindex="-1"></a> <span class="co"># V(subgraph)$colors &lt;- module_assignment[match(V(subgraph)$name, module_assignment$gene), "known"]</span></span>
<span id="cb277-14"><a href="#cb277-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-15"><a href="#cb277-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot the subgraph with community-specific information</span></span>
<span id="cb277-16"><a href="#cb277-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#png(paste("Known and 90 pain genes M",i,".png",sep = ""))</span></span>
<span id="cb277-17"><a href="#cb277-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(subgraph,<span class="at">vertex.size=</span><span class="dv">7</span>, <span class="at">vertex.label.dist=</span><span class="dv">2</span>, <span class="at">vertex.label.cex=</span><span class="fl">0.8</span>, </span>
<span id="cb277-18"><a href="#cb277-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">vertex.label =</span> <span class="fu">V</span>(subgraph)<span class="sc">$</span>hgnc_symbol, <span class="at">vertex.color =</span> <span class="fu">V</span>(subgraph)<span class="sc">$</span>colors )</span>
<span id="cb277-19"><a href="#cb277-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">box</span>()</span>
<span id="cb277-20"><a href="#cb277-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#dev.off()</span></span>
<span id="cb277-21"><a href="#cb277-21" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [32]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb278"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb278-1"><a href="#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot clustering</span></span>
<span id="cb278-2"><a href="#cb278-2" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(subgraphs, module_assignment, <span class="sc">~</span> <span class="fu">plot</span>(.x, </span>
<span id="cb278-3"><a href="#cb278-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.size =</span> <span class="dv">3</span>, </span>
<span id="cb278-4"><a href="#cb278-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.color =</span> <span class="fu">as.character</span>(.y<span class="sc">$</span>module_color), </span>
<span id="cb278-5"><a href="#cb278-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.frame.color =</span> .y<span class="sc">$</span>module_color, </span>
<span id="cb278-6"><a href="#cb278-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">edge.color =</span> <span class="st">"grey"</span>,  </span>
<span id="cb278-7"><a href="#cb278-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">rescale =</span> <span class="cn">TRUE</span>, </span>
<span id="cb278-8"><a href="#cb278-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label =</span> <span class="cn">NA</span>,</span>
<span id="cb278-9"><a href="#cb278-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">"Celltype: "</span>, .y<span class="sc">$</span>label[<span class="dv">1</span>])))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-3.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-4.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-4.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-5.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-5.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-6.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-6.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-7.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-7.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-8.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-8.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-9.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-9.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-10.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-10.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-11.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-11.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-12.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-12.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-13.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-13.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-14.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-14.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-15.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-15.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-16.png" class="lightbox" data-gallery="quarto-lightbox-gallery-16"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-16.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-17.png" class="lightbox" data-gallery="quarto-lightbox-gallery-17"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-17.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-18.png" class="lightbox" data-gallery="quarto-lightbox-gallery-18"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-18.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-19.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-19.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-20.png" class="lightbox" data-gallery="quarto-lightbox-gallery-20"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-20.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-21.png" class="lightbox" data-gallery="quarto-lightbox-gallery-21"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-21.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-22.png" class="lightbox" data-gallery="quarto-lightbox-gallery-22"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-22.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-23.png" class="lightbox" data-gallery="quarto-lightbox-gallery-23"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-23.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-24.png" class="lightbox" data-gallery="quarto-lightbox-gallery-24"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-24.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-25.png" class="lightbox" data-gallery="quarto-lightbox-gallery-25"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-25.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-26.png" class="lightbox" data-gallery="quarto-lightbox-gallery-26"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-26.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-27.png" class="lightbox" data-gallery="quarto-lightbox-gallery-27"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-27.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-28.png" class="lightbox" data-gallery="quarto-lightbox-gallery-28"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-28.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-29.png" class="lightbox" data-gallery="quarto-lightbox-gallery-29"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-29.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-30.png" class="lightbox" data-gallery="quarto-lightbox-gallery-30"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-30.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-31.png" class="lightbox" data-gallery="quarto-lightbox-gallery-31"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-31.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-32.png" class="lightbox" data-gallery="quarto-lightbox-gallery-32"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-32.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-33.png" class="lightbox" data-gallery="quarto-lightbox-gallery-33"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-33.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-34.png" class="lightbox" data-gallery="quarto-lightbox-gallery-34"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-34.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-35.png" class="lightbox" data-gallery="quarto-lightbox-gallery-35"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-35.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-36.png" class="lightbox" data-gallery="quarto-lightbox-gallery-36"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-36.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-37.png" class="lightbox" data-gallery="quarto-lightbox-gallery-37"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-37.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-38.png" class="lightbox" data-gallery="quarto-lightbox-gallery-38"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-38.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-39.png" class="lightbox" data-gallery="quarto-lightbox-gallery-39"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-39.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-40.png" class="lightbox" data-gallery="quarto-lightbox-gallery-40"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-40.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-41.png" class="lightbox" data-gallery="quarto-lightbox-gallery-41"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-41.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-42.png" class="lightbox" data-gallery="quarto-lightbox-gallery-42"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-42.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-43.png" class="lightbox" data-gallery="quarto-lightbox-gallery-43"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-43.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-44.png" class="lightbox" data-gallery="quarto-lightbox-gallery-44"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-44.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-45.png" class="lightbox" data-gallery="quarto-lightbox-gallery-45"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-45.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-46.png" class="lightbox" data-gallery="quarto-lightbox-gallery-46"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-46.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-47.png" class="lightbox" data-gallery="quarto-lightbox-gallery-47"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-47.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-48.png" class="lightbox" data-gallery="quarto-lightbox-gallery-48"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-48.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-49.png" class="lightbox" data-gallery="quarto-lightbox-gallery-49"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-49.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-50.png" class="lightbox" data-gallery="quarto-lightbox-gallery-50"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-50.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-51.png" class="lightbox" data-gallery="quarto-lightbox-gallery-51"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-51.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-52.png" class="lightbox" data-gallery="quarto-lightbox-gallery-52"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-52.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-53.png" class="lightbox" data-gallery="quarto-lightbox-gallery-53"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-53.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-54.png" class="lightbox" data-gallery="quarto-lightbox-gallery-54"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-54.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-55.png" class="lightbox" data-gallery="quarto-lightbox-gallery-55"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-55.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-56.png" class="lightbox" data-gallery="quarto-lightbox-gallery-56"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-56.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-57.png" class="lightbox" data-gallery="quarto-lightbox-gallery-57"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-57.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-58.png" class="lightbox" data-gallery="quarto-lightbox-gallery-58"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-58.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-59.png" class="lightbox" data-gallery="quarto-lightbox-gallery-59"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-59.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-60.png" class="lightbox" data-gallery="quarto-lightbox-gallery-60"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-60.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-61.png" class="lightbox" data-gallery="quarto-lightbox-gallery-61"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-61.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-62.png" class="lightbox" data-gallery="quarto-lightbox-gallery-62"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-62.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-63.png" class="lightbox" data-gallery="quarto-lightbox-gallery-63"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-63.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-64.png" class="lightbox" data-gallery="quarto-lightbox-gallery-64"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-64.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-65.png" class="lightbox" data-gallery="quarto-lightbox-gallery-65"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-65.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-66.png" class="lightbox" data-gallery="quarto-lightbox-gallery-66"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-66.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-67.png" class="lightbox" data-gallery="quarto-lightbox-gallery-67"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-67.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-68.png" class="lightbox" data-gallery="quarto-lightbox-gallery-68"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-68.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-69.png" class="lightbox" data-gallery="quarto-lightbox-gallery-69"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-69.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-70.png" class="lightbox" data-gallery="quarto-lightbox-gallery-70"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-70.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-71.png" class="lightbox" data-gallery="quarto-lightbox-gallery-71"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-71.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-72.png" class="lightbox" data-gallery="quarto-lightbox-gallery-72"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-72.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-73.png" class="lightbox" data-gallery="quarto-lightbox-gallery-73"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-73.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-74.png" class="lightbox" data-gallery="quarto-lightbox-gallery-74"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-74.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-75.png" class="lightbox" data-gallery="quarto-lightbox-gallery-75"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-75.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-76.png" class="lightbox" data-gallery="quarto-lightbox-gallery-76"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-76.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-77.png" class="lightbox" data-gallery="quarto-lightbox-gallery-77"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-77.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-78.png" class="lightbox" data-gallery="quarto-lightbox-gallery-78"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-78.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-79.png" class="lightbox" data-gallery="quarto-lightbox-gallery-79"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-79.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-80.png" class="lightbox" data-gallery="quarto-lightbox-gallery-80"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-80.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-81.png" class="lightbox" data-gallery="quarto-lightbox-gallery-81"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-81.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-82.png" class="lightbox" data-gallery="quarto-lightbox-gallery-82"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-82.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-83.png" class="lightbox" data-gallery="quarto-lightbox-gallery-83"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-83.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-84.png" class="lightbox" data-gallery="quarto-lightbox-gallery-84"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-84.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-85.png" class="lightbox" data-gallery="quarto-lightbox-gallery-85"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-85.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-86.png" class="lightbox" data-gallery="quarto-lightbox-gallery-86"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-86.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-87.png" class="lightbox" data-gallery="quarto-lightbox-gallery-87"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-87.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-88.png" class="lightbox" data-gallery="quarto-lightbox-gallery-88"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-88.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-89.png" class="lightbox" data-gallery="quarto-lightbox-gallery-89"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-89.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-90.png" class="lightbox" data-gallery="quarto-lightbox-gallery-90"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-90.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-91.png" class="lightbox" data-gallery="quarto-lightbox-gallery-91"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-91.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-92.png" class="lightbox" data-gallery="quarto-lightbox-gallery-92"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-92.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-93.png" class="lightbox" data-gallery="quarto-lightbox-gallery-93"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-93.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-94.png" class="lightbox" data-gallery="quarto-lightbox-gallery-94"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-94.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-95.png" class="lightbox" data-gallery="quarto-lightbox-gallery-95"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-95.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-96.png" class="lightbox" data-gallery="quarto-lightbox-gallery-96"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-96.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-97.png" class="lightbox" data-gallery="quarto-lightbox-gallery-97"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-97.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-98.png" class="lightbox" data-gallery="quarto-lightbox-gallery-98"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-98.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-99.png" class="lightbox" data-gallery="quarto-lightbox-gallery-99"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-99.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-100.png" class="lightbox" data-gallery="quarto-lightbox-gallery-100"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-100.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-101.png" class="lightbox" data-gallery="quarto-lightbox-gallery-101"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-101.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-102.png" class="lightbox" data-gallery="quarto-lightbox-gallery-102"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-102.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-103.png" class="lightbox" data-gallery="quarto-lightbox-gallery-103"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-103.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-104.png" class="lightbox" data-gallery="quarto-lightbox-gallery-104"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-104.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-105.png" class="lightbox" data-gallery="quarto-lightbox-gallery-105"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-105.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-106.png" class="lightbox" data-gallery="quarto-lightbox-gallery-106"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-106.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-107.png" class="lightbox" data-gallery="quarto-lightbox-gallery-107"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-107.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-108.png" class="lightbox" data-gallery="quarto-lightbox-gallery-108"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-108.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-109.png" class="lightbox" data-gallery="quarto-lightbox-gallery-109"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-109.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-110.png" class="lightbox" data-gallery="quarto-lightbox-gallery-110"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-110.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-111.png" class="lightbox" data-gallery="quarto-lightbox-gallery-111"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-111.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-112.png" class="lightbox" data-gallery="quarto-lightbox-gallery-112"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-112.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-113.png" class="lightbox" data-gallery="quarto-lightbox-gallery-113"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-113.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-114.png" class="lightbox" data-gallery="quarto-lightbox-gallery-114"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-114.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-115.png" class="lightbox" data-gallery="quarto-lightbox-gallery-115"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-115.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-116.png" class="lightbox" data-gallery="quarto-lightbox-gallery-116"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-116.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-117.png" class="lightbox" data-gallery="quarto-lightbox-gallery-117"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-117.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-118.png" class="lightbox" data-gallery="quarto-lightbox-gallery-118"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-118.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-119.png" class="lightbox" data-gallery="quarto-lightbox-gallery-119"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-119.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-120.png" class="lightbox" data-gallery="quarto-lightbox-gallery-120"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-120.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-121.png" class="lightbox" data-gallery="quarto-lightbox-gallery-121"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-121.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-122.png" class="lightbox" data-gallery="quarto-lightbox-gallery-122"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-122.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-123.png" class="lightbox" data-gallery="quarto-lightbox-gallery-123"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-123.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-124.png" class="lightbox" data-gallery="quarto-lightbox-gallery-124"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-124.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-125.png" class="lightbox" data-gallery="quarto-lightbox-gallery-125"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-125.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-126.png" class="lightbox" data-gallery="quarto-lightbox-gallery-126"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-126.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-127.png" class="lightbox" data-gallery="quarto-lightbox-gallery-127"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-127.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-128.png" class="lightbox" data-gallery="quarto-lightbox-gallery-128"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-128.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-129.png" class="lightbox" data-gallery="quarto-lightbox-gallery-129"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-129.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-130.png" class="lightbox" data-gallery="quarto-lightbox-gallery-130"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-130.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-131.png" class="lightbox" data-gallery="quarto-lightbox-gallery-131"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-131.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-132.png" class="lightbox" data-gallery="quarto-lightbox-gallery-132"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-132.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-133.png" class="lightbox" data-gallery="quarto-lightbox-gallery-133"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-133.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-134.png" class="lightbox" data-gallery="quarto-lightbox-gallery-134"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-134.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-135.png" class="lightbox" data-gallery="quarto-lightbox-gallery-135"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-135.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-136.png" class="lightbox" data-gallery="quarto-lightbox-gallery-136"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-136.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-137.png" class="lightbox" data-gallery="quarto-lightbox-gallery-137"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-137.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-138.png" class="lightbox" data-gallery="quarto-lightbox-gallery-138"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-138.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-139.png" class="lightbox" data-gallery="quarto-lightbox-gallery-139"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-139.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-140.png" class="lightbox" data-gallery="quarto-lightbox-gallery-140"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-140.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-141.png" class="lightbox" data-gallery="quarto-lightbox-gallery-141"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-141.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-142.png" class="lightbox" data-gallery="quarto-lightbox-gallery-142"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-142.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-143.png" class="lightbox" data-gallery="quarto-lightbox-gallery-143"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-143.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-144.png" class="lightbox" data-gallery="quarto-lightbox-gallery-144"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-144.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-145.png" class="lightbox" data-gallery="quarto-lightbox-gallery-145"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-145.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-146.png" class="lightbox" data-gallery="quarto-lightbox-gallery-146"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-146.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-147.png" class="lightbox" data-gallery="quarto-lightbox-gallery-147"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-147.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-148.png" class="lightbox" data-gallery="quarto-lightbox-gallery-148"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-148.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-149.png" class="lightbox" data-gallery="quarto-lightbox-gallery-149"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-149.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-150.png" class="lightbox" data-gallery="quarto-lightbox-gallery-150"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-150.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-151.png" class="lightbox" data-gallery="quarto-lightbox-gallery-151"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-151.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-152.png" class="lightbox" data-gallery="quarto-lightbox-gallery-152"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-152.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-153.png" class="lightbox" data-gallery="quarto-lightbox-gallery-153"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-153.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-154.png" class="lightbox" data-gallery="quarto-lightbox-gallery-154"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-154.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-155.png" class="lightbox" data-gallery="quarto-lightbox-gallery-155"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-155.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-156.png" class="lightbox" data-gallery="quarto-lightbox-gallery-156"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-156.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-157.png" class="lightbox" data-gallery="quarto-lightbox-gallery-157"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-157.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-158.png" class="lightbox" data-gallery="quarto-lightbox-gallery-158"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-158.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-159.png" class="lightbox" data-gallery="quarto-lightbox-gallery-159"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-159.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-160.png" class="lightbox" data-gallery="quarto-lightbox-gallery-160"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-160.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-161.png" class="lightbox" data-gallery="quarto-lightbox-gallery-161"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-161.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-162.png" class="lightbox" data-gallery="quarto-lightbox-gallery-162"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-162.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-163.png" class="lightbox" data-gallery="quarto-lightbox-gallery-163"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-163.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-164.png" class="lightbox" data-gallery="quarto-lightbox-gallery-164"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-164.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-165.png" class="lightbox" data-gallery="quarto-lightbox-gallery-165"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-165.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-166.png" class="lightbox" data-gallery="quarto-lightbox-gallery-166"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-166.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-167.png" class="lightbox" data-gallery="quarto-lightbox-gallery-167"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-32-167.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>$ambiguous_pseudobulk_apoe
NULL

$Arterial_pseudobulk_apoe
NULL

$`Arteriolar-SMC_pseudobulk_apoe`
NULL

$`Artirial-FB_pseudobulk_apoe`
NULL

$`Astrocyte-activated_pseudobulk_apoe`
NULL

$`Astrocyte-quiescent_pseudobulk_apoe`
NULL

$Capillary_pseudobulk_apoe
NULL

$`Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe`
NULL

$`Ex-Neuron-DLC1-HS3ST4_pseudobulk_apoe`
NULL

$`Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_apoe`
NULL

$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe`
NULL

$`Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_apoe`
NULL

$`In-Neuron-ADARB2-GRM7_pseudobulk_apoe`
NULL

$`In-Neuron-ADARB2-IL1RAPL2-CSCL14_pseudobulk_apoe`
NULL

$`In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_apoe`
NULL

$`In-Neuron-DPP10-MEF2C_pseudobulk_apoe`
NULL

$`In-Neuron-PRKG1-TBC1D4_pseudobulk_apoe`
NULL

$`In-Neuron-TRHDE-RALYL_pseudobulk_apoe`
NULL

$`low-feature-cells_pseudobulk_apoe`
NULL

$`M-Pericyte_pseudobulk_apoe`
NULL

$`Microglia-activated_pseudobulk_apoe`
NULL

$`Microglia-quiescent_pseudobulk_apoe`
NULL

$`Oligo-A_pseudobulk_apoe`
NULL

$`Oligo-B_pseudobulk_apoe`
NULL

$`OPC-A_pseudobulk_apoe`
NULL

$`OPC-B_pseudobulk_apoe`
NULL

$Pericyte_pseudobulk_apoe
NULL

$`Pericyte-2_pseudobulk_apoe`
NULL

$`Pericyte-FB_pseudobulk_apoe`
NULL

$`Perivascular-FB_pseudobulk_apoe`
NULL

$`Perivascular-FB-KAZN2_pseudobulk_apoe`
NULL

$`T-Pericyte_pseudobulk_apoe`
NULL

$`Trans-Endo-to-mural_pseudobulk_apoe`
NULL

$`Vascular-SMC-LINC00486_pseudobulk_apoe`
NULL

$Venous_pseudobulk_apoe
NULL

$ambiguous_pseudobulk_no_apoe
NULL

$Arterial_pseudobulk_no_apoe
NULL

$`Arteriolar-SMC_pseudobulk_no_apoe`
NULL

$`Artirial-FB_pseudobulk_no_apoe`
NULL

$`Astrocyte-activated_pseudobulk_no_apoe`
NULL

$`Astrocyte-quiescent_pseudobulk_no_apoe`
NULL

$Capillary_pseudobulk_no_apoe
NULL

$`Ex-Neuron-CBLN2-NRGN_pseudobulk_no_apoe`
NULL

$`Ex-Neuron-DLC1-HS3ST4_pseudobulk_no_apoe`
NULL

$`Ex-Neuron-L2-3-CBLN2-LINC02306-CUX2_pseudobulk_no_apoe`
NULL

$`Ex-Neuron-L3-5-RORB-PLCH1_pseudobulk_no_apoe`
NULL

$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe`
NULL

$`Ex-Neuron-L6-THEMIS-NFIA_pseudobulk_no_apoe`
NULL

$`In-Neuron-ADARB2-GRM7_pseudobulk_no_apoe`
NULL

$`In-Neuron-ADARB2-LAMP5-NRG1_pseudobulk_no_apoe`
NULL

$`In-Neuron-ADARB2-NRG1-RELN_pseudobulk_no_apoe`
NULL

$`In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe`
NULL

$`In-Neuron-PRKG1-TBC1D4_pseudobulk_no_apoe`
NULL

$`In-Neuron-TRHDE-RALYL_pseudobulk_no_apoe`
NULL

$`low-feature-cells_pseudobulk_no_apoe`
NULL

$`M-Pericyte_pseudobulk_no_apoe`
NULL

$`Microglia-activated_pseudobulk_no_apoe`
NULL

$`Microglia-quiescent_pseudobulk_no_apoe`
NULL

$`Oligo-A_pseudobulk_no_apoe`
NULL

$`Oligo-B_pseudobulk_no_apoe`
NULL

$`OPC-B_pseudobulk_no_apoe`
NULL

$Pericyte_pseudobulk_no_apoe
NULL

$`Pericyte-2_pseudobulk_no_apoe`
NULL

$`Pericyte-FB_pseudobulk_no_apoe`
NULL

$`T-cell-parenchymal-1_pseudobulk_no_apoe`
NULL

$`T-cell-parenchymal-2_pseudobulk_no_apoe`
NULL

$`T-Pericyte_pseudobulk_no_apoe`
NULL

$`Trans-Endo-to-mural_pseudobulk_no_apoe`
NULL

$`Vascular-SMC_pseudobulk_no_apoe`
NULL

$`Vascular-SMC-LINC00486_pseudobulk_no_apoe`
NULL

$Venous_pseudobulk_no_apoe
NULL

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1`
NULL

$`Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1`
NULL

$`Trans-Endo-to-mural_sig-magma-sig-genes_stroke_35k10k_all_controls_top-five-percent_level1`
NULL

$`Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Astro_sig-magma-sig-genes_wmh_35k10k_all_controls_top-ten-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1`
NULL

$`Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level1`
NULL

$`OPC_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1`
NULL

$`Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level1`
NULL

$`Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`Astro_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`Neuron_in_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`Oligo_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level1`
NULL

$`SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`Fibroblast_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`Fibroblast_sig-magma-sig-genes_wmh_35k10k_controls_vascular_top-five-percent_level1`
NULL

$`Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`SMC_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`T-cell_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`Endothelial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`Microglia_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level1`
NULL

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2`
NULL

$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2`
NULL

$`Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2`
NULL

$`Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2`
NULL

$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2`
NULL

$`Astrocyte-activated_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2`
NULL

$`In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2`
NULL

$`Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2`
NULL

$`Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-five-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`Oligo-A_sig-magma-sig-genes_wmh_35k10k_controls_parenchymal_top-ten-percent_level2`
NULL

$`Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2`
NULL

$`T-cell-vascular_sig-magma-sig-genes_Kunkle_2019_35k10k_controls_vascular_top-five-percent_level2`
NULL

$`M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2`
NULL

$`Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2`
NULL

$`Arteriolar-SMC_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2`
NULL

$`Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-five-percent_level2`
NULL

$`Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Meningeal-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`T-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`M-Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Pericyte_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Pericyte-2_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Vascular-SMC-LINC00486_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Meningeal-FB_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Capillary_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`Arterial_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$`T-cell-mixed_sig-magma-sig-genes_stroke_35k10k_controls_vascular_top-ten-percent_level2`
NULL

$sig_magma_celltypes_NA
NULL

$`Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA`
NULL

$`Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA`
NULL

$`Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA`
NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb280"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb280-1"><a href="#cb280-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save clustering as file</span></span>
<span id="cb280-2"><a href="#cb280-2" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_delim</span>(<span class="fu">list_rbind</span>(module_assignment), <span class="at">file =</span> output_file2)</span>
<span id="cb280-3"><a href="#cb280-3" aria-hidden="true" tabindex="-1"></a><span class="co">#modules &lt;- readr::read_delim(file = output_file2)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="module-assignment" class="level2">
<h2 class="anchored" data-anchor-id="module-assignment">Module assignment</h2>
<p>Having assigned the genes to modules the next thing to do is choose which celltypes to take forward to assign functions via GO pathways to.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [33]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb281"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb281-1"><a href="#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get modules assignment</span></span>
<span id="cb281-2"><a href="#cb281-2" aria-hidden="true" tabindex="-1"></a>modules <span class="ot">&lt;-</span> <span class="fu">list_rbind</span>(module_assignment)</span>
<span id="cb281-3"><a href="#cb281-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-4"><a href="#cb281-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get number of genes per celltypes</span></span>
<span id="cb281-5"><a href="#cb281-5" aria-hidden="true" tabindex="-1"></a>gene_num <span class="ot">&lt;-</span> <span class="fu">map_dbl</span>(my_gene_list<span class="sc">$</span>full_background, length)</span>
<span id="cb281-6"><a href="#cb281-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the gene numbers per celltypes of interest</span></span>
<span id="cb281-7"><a href="#cb281-7" aria-hidden="true" tabindex="-1"></a>gene_num <span class="ot">&lt;-</span> gene_num[<span class="fu">names</span>(gene_num) <span class="sc">%in%</span> sig_celltypes<span class="sc">$</span>gene_list]</span>
<span id="cb281-8"><a href="#cb281-8" aria-hidden="true" tabindex="-1"></a>gene_num</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                             Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe 
                                                                                                                 11 
                                                                       Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe 
                                                                                                                121 
                                                                                  low-feature-cells_pseudobulk_apoe 
                                                                                                                 25 
                                                                                Microglia-quiescent_pseudobulk_apoe 
                                                                                                                 26 
                                                                                            Oligo-A_pseudobulk_apoe 
                                                                                                                311 
                                                                                            Oligo-B_pseudobulk_apoe 
                                                                                                                401 
                                                                    Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe 
                                                                                                                253 
                                                                           In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe 
                                                                                                                  9 
                                                                                      T-Pericyte_pseudobulk_no_apoe 
                                                                                                                369 
                  Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1 
                                                                                                                 20 
                     T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1 
                                                                                                                 17 
        Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level1 
                                                                                                                 12 
                               Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1 
                                                                                                                 29 
                                  T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level1 
                                                                                                                 29 
                                       Microglia_sig-magma-sig-genes_pd_35k10k_all_controls_top-five-percent_level1 
                                                                                                                 10 
                       Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 21 
                 Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 22 
                  Fibroblast_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 20 
                   Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 26 
                    Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 15 
                         SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 19 
                      T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 17 
         Trans-Endo-to-mural_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 19 
                               Fibroblast_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 35 
                                Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 46 
                                 Pericyte_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 33 
                                   T-cell_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 37 
                                            Astro_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 12 
                                            Oligo_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 15 
                                           T-cell_sig-magma-sig-genes_pd_35k10k_all_controls_top-ten-percent_level1 
                                                                                                                 14 
          Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level1 
                                                                                                                 17 
                       Microglia_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level1 
                                                                                                                 24 
               Astro_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1 
                                                                                                                 16 
           Microglia_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level1 
                                                                                                                 22 
                                Microglia_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1 
                                                                                                                 16 
                                    Oligo_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level1 
                                                                                                                 11 
            Endothelial_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1 
                                                                                                                 13 
               Pericyte_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1 
                                                                                                                 10 
                    SMC_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1 
                                                                                                                 11 
                 T-cell_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_vascular_top-ten-percent_level1 
                                                                                                                 13 
                                  Fibroblast_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level1 
                                                                                                                  8 
        Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-five-percent_level2 
                                                                                                                 20 
                     Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2 
                                                                                                                 34 
                   Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_level2 
                                                                                                                 21 
         Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_all_controls_top-ten-percent_level2 
                                                                                                                 20 
                      Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2 
                                                                                                                 34 
                               Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2 
                                                                                                                 23 
                    Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_level2 
                                                                                                                 23 
Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-five-percent_level2 
                                                                                                                 21 
             Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_level2 
                                                                                                                 29 
                                 Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_level2 
                                                                                                                  5 
 Microglia-activated_sig-magma-sig-genes_Bellenguez_2022_noproxy_35k10k_controls_parenchymal_top-ten-percent_level2 
                                                                                                                 21 
              Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2 
                                                                                                                 31 
              Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_level2 
                                                                                                                 23 
                   Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_level2 
                                                                                                                 11 
                                  Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_level2 
                                                                                                                  4 
                             Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_level2 
                                                                                                                  9 
                                                                                             sig_magma_celltypes_NA 
                                                                                                                 55 
                                        Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_NA 
                                                                                                                 25 
                                          Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_NA 
                                                                                                                 38 
                                                   Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_NA 
                                                                                                                 24 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb283"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb283-1"><a href="#cb283-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the celltypes with fewer than 30 genes</span></span>
<span id="cb283-2"><a href="#cb283-2" aria-hidden="true" tabindex="-1"></a>celltypes_with_few_genes <span class="ot">&lt;-</span> <span class="fu">names</span>(gene_num)[gene_num <span class="sc">&lt;</span> <span class="dv">30</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<p>For cases where the number of genes is small, there isnt a need to use the clusters, though there isnt a rule of thumb for a cutoff for this. Given most of the celltypes in this case have more than 30 genes, Ill run those that arent through GO as they are, but the others Ill do per Louvain cluster.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [34]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb284"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1" aria-hidden="true" tabindex="-1"></a>modules_sig <span class="ot">&lt;-</span> modules <span class="sc">|&gt;</span></span>
<span id="cb284-2"><a href="#cb284-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filiter to sig celltypes</span></span>
<span id="cb284-3"><a href="#cb284-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(label <span class="sc">%in%</span> sig_celltypes<span class="sc">$</span>gene_list) <span class="sc">|&gt;</span></span>
<span id="cb284-4"><a href="#cb284-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Update modules for celltypes with low number of genes to all be the same</span></span>
<span id="cb284-5"><a href="#cb284-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">module_updated =</span> <span class="fu">case_when</span>(</span>
<span id="cb284-6"><a href="#cb284-6" aria-hidden="true" tabindex="-1"></a>    label <span class="sc">%in%</span> celltypes_with_few_genes <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb284-7"><a href="#cb284-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">.default =</span> module</span>
<span id="cb284-8"><a href="#cb284-8" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [35]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb285"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb285-1"><a href="#cb285-1" aria-hidden="true" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> modules_sig <span class="sc">%&gt;%</span> </span>
<span id="cb285-2"><a href="#cb285-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(label, module) <span class="sc">%&gt;%</span> </span>
<span id="cb285-3"><a href="#cb285-3" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="go-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="go-enrichment">GO enrichment</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [36]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb286"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb286-1"><a href="#cb286-1" aria-hidden="true" tabindex="-1"></a>convert_to_entrez <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_list) {</span>
<span id="cb286-2"><a href="#cb286-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb286-3"><a href="#cb286-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bitr</span>(gene_list, <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>, <span class="at">toType =</span> <span class="st">"ENTREZID"</span>, <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb286-4"><a href="#cb286-4" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb286-5"><a href="#cb286-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"An error occurred: "</span>, e<span class="sc">$</span>message)</span>
<span id="cb286-6"><a href="#cb286-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># Return NULL if the conversion fails</span></span>
<span id="cb286-7"><a href="#cb286-7" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb286-8"><a href="#cb286-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb286-9"><a href="#cb286-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Ensembl IDs to Entrez IDs</span></span>
<span id="cb286-10"><a href="#cb286-10" aria-hidden="true" tabindex="-1"></a>gene_list_entrez <span class="ot">&lt;-</span> <span class="fu">map</span>(gene_list<span class="sc">$</span>data, <span class="sc">~</span> {</span>
<span id="cb286-11"><a href="#cb286-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb286-12"><a href="#cb286-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">bitr</span>(.x<span class="sc">$</span>gene, <span class="at">fromType =</span> <span class="st">"ENSEMBL"</span>, <span class="at">toType =</span> <span class="st">"ENTREZID"</span>, <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb286-13"><a href="#cb286-13" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb286-14"><a href="#cb286-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"An error occurred: "</span>, e<span class="sc">$</span>message)</span>
<span id="cb286-15"><a href="#cb286-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># Return NULL if the conversion fails</span></span>
<span id="cb286-16"><a href="#cb286-16" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb286-17"><a href="#cb286-17" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns
'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb308"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb308-1"><a href="#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out any NULL entries if there were conversion errors</span></span>
<span id="cb308-2"><a href="#cb308-2" aria-hidden="true" tabindex="-1"></a>gene_list_entrez <span class="ot">&lt;-</span> <span class="fu">map</span>(gene_list_entrez, <span class="sc">~</span> .x <span class="sc">|&gt;</span></span>
<span id="cb308-3"><a href="#cb308-3" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># remove any missing data</span></span>
<span id="cb308-4"><a href="#cb308-4" aria-hidden="true" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID)) <span class="sc">|&gt;</span></span>
<span id="cb308-5"><a href="#cb308-5" aria-hidden="true" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">pull</span>(ENTREZID))</span>
<span id="cb308-6"><a href="#cb308-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb308-7"><a href="#cb308-7" aria-hidden="true" tabindex="-1"></a>entrez_background <span class="ot">&lt;-</span> <span class="fu">convert_to_entrez</span>(gene_ensembl_ids<span class="sc">$</span>ensembl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in bitr(gene_list, fromType = "ENSEMBL", toType = "ENTREZID", OrgDb =
org.Hs.eg.db): 23.25% of input gene IDs are fail to map...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb311"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb311-1"><a href="#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gene_list_entrez) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(gene_list<span class="sc">$</span>label, <span class="st">"_module"</span>, gene_list<span class="sc">$</span>module)</span>
<span id="cb311-2"><a href="#cb311-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-3"><a href="#cb311-3" aria-hidden="true" tabindex="-1"></a>gene_list_entrez_sub <span class="ot">&lt;-</span></span>
<span id="cb311-4"><a href="#cb311-4" aria-hidden="true" tabindex="-1"></a>  gene_list_entrez[<span class="fu">sapply</span>(gene_list_entrez, <span class="cf">function</span>(x)</span>
<span id="cb311-5"><a href="#cb311-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>(x) <span class="sc">&gt;</span> <span class="dv">1</span>)]</span>
<span id="cb311-6"><a href="#cb311-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-7"><a href="#cb311-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save file - it takes a while to compute</span></span>
<span id="cb311-8"><a href="#cb311-8" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span></span>
<span id="cb311-9"><a href="#cb311-9" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb311-10"><a href="#cb311-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions/module_go_terms.rds"</span></span>
<span id="cb311-11"><a href="#cb311-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb311-12"><a href="#cb311-12" aria-hidden="true" tabindex="-1"></a>file <span class="ot">&lt;-</span></span>
<span id="cb311-13"><a href="#cb311-13" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb311-14"><a href="#cb311-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions/module_go_terms_multicore.qs"</span></span>
<span id="cb311-15"><a href="#cb311-15" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb311-16"><a href="#cb311-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-17"><a href="#cb311-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb311-18"><a href="#cb311-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(file)) {</span>
<span id="cb311-19"><a href="#cb311-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plan</span>(<span class="st">"multisession"</span>, <span class="at">workers =</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>() <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb311-20"><a href="#cb311-20" aria-hidden="true" tabindex="-1"></a>  result_list_bp <span class="ot">&lt;-</span> <span class="fu">future_map</span>(</span>
<span id="cb311-21"><a href="#cb311-21" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>data,</span>
<span id="cb311-22"><a href="#cb311-22" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">enrichGO</span>(</span>
<span id="cb311-23"><a href="#cb311-23" aria-hidden="true" tabindex="-1"></a>      <span class="at">gene          =</span> .x<span class="sc">$</span>gene,</span>
<span id="cb311-24"><a href="#cb311-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb311-25"><a href="#cb311-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">keyType       =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb311-26"><a href="#cb311-26" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "BP" for biological process, "MF" for molecular function, "CC" for cellular component</span></span>
<span id="cb311-27"><a href="#cb311-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">ont           =</span> <span class="st">"BP"</span>,</span>
<span id="cb311-28"><a href="#cb311-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">universe      =</span> gene_ensembl_ids<span class="sc">$</span>ensembl,</span>
<span id="cb311-29"><a href="#cb311-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb311-30"><a href="#cb311-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb311-31"><a href="#cb311-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">readable      =</span> <span class="cn">TRUE</span></span>
<span id="cb311-32"><a href="#cb311-32" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">.options =</span> <span class="fu">furrr_options</span>(<span class="at">seed =</span> <span class="dv">123</span>)</span>
<span id="cb311-33"><a href="#cb311-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb311-34"><a href="#cb311-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set_names</span>(<span class="fu">paste0</span>(gene_list<span class="sc">$</span>label, <span class="st">"_module"</span>, gene_list<span class="sc">$</span>module, <span class="st">"_BP"</span>))</span>
<span id="cb311-35"><a href="#cb311-35" aria-hidden="true" tabindex="-1"></a>  result_list_cc <span class="ot">&lt;-</span> <span class="fu">future_map</span>(</span>
<span id="cb311-36"><a href="#cb311-36" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>data,</span>
<span id="cb311-37"><a href="#cb311-37" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">enrichGO</span>(</span>
<span id="cb311-38"><a href="#cb311-38" aria-hidden="true" tabindex="-1"></a>      <span class="at">gene          =</span> .x<span class="sc">$</span>gene,</span>
<span id="cb311-39"><a href="#cb311-39" aria-hidden="true" tabindex="-1"></a>      <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb311-40"><a href="#cb311-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">keyType       =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb311-41"><a href="#cb311-41" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "BP" for biological process, "MF" for molecular function, "CC" for cellular component</span></span>
<span id="cb311-42"><a href="#cb311-42" aria-hidden="true" tabindex="-1"></a>      <span class="at">ont           =</span> <span class="st">"CC"</span>,</span>
<span id="cb311-43"><a href="#cb311-43" aria-hidden="true" tabindex="-1"></a>      <span class="at">universe      =</span> gene_ensembl_ids<span class="sc">$</span>ensembl,</span>
<span id="cb311-44"><a href="#cb311-44" aria-hidden="true" tabindex="-1"></a>      <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb311-45"><a href="#cb311-45" aria-hidden="true" tabindex="-1"></a>      <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb311-46"><a href="#cb311-46" aria-hidden="true" tabindex="-1"></a>      <span class="at">readable      =</span> <span class="cn">TRUE</span></span>
<span id="cb311-47"><a href="#cb311-47" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">.options =</span> <span class="fu">furrr_options</span>(<span class="at">seed =</span> <span class="dv">123</span>)</span>
<span id="cb311-48"><a href="#cb311-48" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb311-49"><a href="#cb311-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set_names</span>(<span class="fu">paste0</span>(gene_list<span class="sc">$</span>label, <span class="st">"_module"</span>, gene_list<span class="sc">$</span>module, <span class="st">"_CC"</span>))</span>
<span id="cb311-50"><a href="#cb311-50" aria-hidden="true" tabindex="-1"></a>  result_list_mf <span class="ot">&lt;-</span> <span class="fu">future_map</span>(</span>
<span id="cb311-51"><a href="#cb311-51" aria-hidden="true" tabindex="-1"></a>    gene_list<span class="sc">$</span>data,</span>
<span id="cb311-52"><a href="#cb311-52" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> <span class="fu">enrichGO</span>(</span>
<span id="cb311-53"><a href="#cb311-53" aria-hidden="true" tabindex="-1"></a>      <span class="at">gene          =</span> .x<span class="sc">$</span>gene,</span>
<span id="cb311-54"><a href="#cb311-54" aria-hidden="true" tabindex="-1"></a>      <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb311-55"><a href="#cb311-55" aria-hidden="true" tabindex="-1"></a>      <span class="at">keyType       =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb311-56"><a href="#cb311-56" aria-hidden="true" tabindex="-1"></a>      <span class="co"># "BP" for biological process, "MF" for molecular function, "CC" for cellular component</span></span>
<span id="cb311-57"><a href="#cb311-57" aria-hidden="true" tabindex="-1"></a>      <span class="at">ont           =</span> <span class="st">"MF"</span>,</span>
<span id="cb311-58"><a href="#cb311-58" aria-hidden="true" tabindex="-1"></a>      <span class="at">universe      =</span> gene_ensembl_ids<span class="sc">$</span>ensembl,</span>
<span id="cb311-59"><a href="#cb311-59" aria-hidden="true" tabindex="-1"></a>      <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb311-60"><a href="#cb311-60" aria-hidden="true" tabindex="-1"></a>      <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb311-61"><a href="#cb311-61" aria-hidden="true" tabindex="-1"></a>      <span class="at">readable      =</span> <span class="cn">TRUE</span></span>
<span id="cb311-62"><a href="#cb311-62" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">.options =</span> <span class="fu">furrr_options</span>(<span class="at">seed =</span> <span class="dv">123</span>)</span>
<span id="cb311-63"><a href="#cb311-63" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb311-64"><a href="#cb311-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set_names</span>(<span class="fu">paste0</span>(gene_list<span class="sc">$</span>label, <span class="st">"_module"</span>, gene_list<span class="sc">$</span>module, <span class="st">"_MF"</span>))</span>
<span id="cb311-65"><a href="#cb311-65" aria-hidden="true" tabindex="-1"></a>  result_list <span class="ot">&lt;-</span> <span class="fu">c</span>(result_list_bp, result_list_mf, result_list_cc)</span>
<span id="cb311-66"><a href="#cb311-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(result_list_bp, result_list_cc, result_list_mf)</span>
<span id="cb311-67"><a href="#cb311-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb311-68"><a href="#cb311-68" aria-hidden="true" tabindex="-1"></a>  qs<span class="sc">::</span><span class="fu">qsave</span>(result_list, file)</span>
<span id="cb311-69"><a href="#cb311-69" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb311-70"><a href="#cb311-70" aria-hidden="true" tabindex="-1"></a>  result_list <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(file)</span>
<span id="cb311-71"><a href="#cb311-71" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb311-72"><a href="#cb311-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-73"><a href="#cb311-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-74"><a href="#cb311-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any NULL entries</span></span>
<span id="cb311-75"><a href="#cb311-75" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> result_list[<span class="sc">!</span><span class="fu">sapply</span>(result_list, is.null)]</span>
<span id="cb311-76"><a href="#cb311-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Assert that there are no nulls in list</span></span>
<span id="cb311-77"><a href="#cb311-77" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">see_if</span>(<span class="fu">sum</span>(<span class="fu">map_lgl</span>(result_list, is.null)) <span class="sc">==</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb313"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb313-1"><a href="#cb313-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the list to keep only elements where the 'result' slot has rows</span></span>
<span id="cb313-2"><a href="#cb313-2" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> result_list[<span class="fu">sapply</span>(result_list, <span class="cf">function</span>(x) <span class="fu">nrow</span>(x<span class="sc">@</span>result) <span class="sc">&gt;</span> <span class="dv">0</span>)]</span>
<span id="cb313-3"><a href="#cb313-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to only those wtih sig enrichments</span></span>
<span id="cb313-4"><a href="#cb313-4" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> result_list[<span class="fu">sapply</span>(result_list, <span class="cf">function</span>(x) <span class="fu">sum</span>(x<span class="sc">@</span>result<span class="sc">$</span>p.adjust <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">&gt;</span> <span class="dv">0</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="save-for-magma" class="level3">
<h3 class="anchored" data-anchor-id="save-for-magma">Save for MAGMA</h3>
<p>I want to run these modules through MAGMA later so Ill save the gene lists whilst I have them in entrez. Note that Im only using the modules that had more than one gene.</p>
<p>Im going to save each celltypes module set to one file.</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [37]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb314"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb314-1"><a href="#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save to MAGMA</span></span>
<span id="cb314-2"><a href="#cb314-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gene_list_entrez_sub) <span class="sc">|&gt;</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_module1"      
[2] "Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_module2"      
[3] "Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_module0"      
[4] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module1"
[5] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module2"
[6] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module3"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb316"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb316-1"><a href="#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a character vector where each element is a single string</span></span>
<span id="cb316-2"><a href="#cb316-2" aria-hidden="true" tabindex="-1"></a>lines_to_write <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">names</span>(gene_list_entrez), <span class="cf">function</span>(name) {</span>
<span id="cb316-3"><a href="#cb316-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(name, <span class="fu">paste</span>(gene_list_entrez[[name]], <span class="at">collapse =</span> <span class="st">" "</span>))</span>
<span id="cb316-4"><a href="#cb316-4" aria-hidden="true" tabindex="-1"></a>}, <span class="at">USE.NAMES =</span> <span class="cn">FALSE</span>)</span>
<span id="cb316-5"><a href="#cb316-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-6"><a href="#cb316-6" aria-hidden="true" tabindex="-1"></a>pseudobulk_lines <span class="ot">&lt;-</span> lines_to_write[<span class="fu">grepl</span>(<span class="st">"_pseudobulk_"</span>, lines_to_write)]</span>
<span id="cb316-7"><a href="#cb316-7" aria-hidden="true" tabindex="-1"></a>magma_gene_list_lines <span class="ot">&lt;-</span> lines_to_write[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"_pseudobulk_"</span>, lines_to_write)]</span>
<span id="cb316-8"><a href="#cb316-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-9"><a href="#cb316-9" aria-hidden="true" tabindex="-1"></a>make_magma_input_per_celltype <span class="ot">&lt;-</span> <span class="cf">function</span>(line_subset, subset_label) {</span>
<span id="cb316-10"><a href="#cb316-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the line subset</span></span>
<span id="cb316-11"><a href="#cb316-11" aria-hidden="true" tabindex="-1"></a>  line_subset <span class="ot">&lt;-</span> line_subset[<span class="fu">grepl</span>(subset_label, line_subset)]</span>
<span id="cb316-12"><a href="#cb316-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get celltypes</span></span>
<span id="cb316-13"><a href="#cb316-13" aria-hidden="true" tabindex="-1"></a>  celltypes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, line_subset) <span class="sc">|&gt;</span> <span class="fu">unique</span>()</span>
<span id="cb316-14"><a href="#cb316-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Match celltypename from begining of line</span></span>
<span id="cb316-15"><a href="#cb316-15" aria-hidden="true" tabindex="-1"></a>  celltypes2 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"^"</span>, celltypes)</span>
<span id="cb316-16"><a href="#cb316-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get per celltype lines</span></span>
<span id="cb316-17"><a href="#cb316-17" aria-hidden="true" tabindex="-1"></a>  per_celltype_lines <span class="ot">&lt;-</span> <span class="fu">map</span>(celltypes2, <span class="sc">~</span> line_subset[<span class="fu">grepl</span>(.x, line_subset)]) <span class="sc">|&gt;</span></span>
<span id="cb316-18"><a href="#cb316-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set_names</span>(<span class="fu">paste0</span>(celltypes, <span class="st">"_"</span>, subset_label))</span>
<span id="cb316-19"><a href="#cb316-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(per_celltype_lines)</span>
<span id="cb316-20"><a href="#cb316-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb316-21"><a href="#cb316-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-22"><a href="#cb316-22" aria-hidden="true" tabindex="-1"></a>pseudobulk_apoe_lines <span class="ot">&lt;-</span></span>
<span id="cb316-23"><a href="#cb316-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_magma_input_per_celltype</span>(pseudobulk_lines, <span class="st">"pseudobulk_apoe"</span>)</span>
<span id="cb316-24"><a href="#cb316-24" aria-hidden="true" tabindex="-1"></a>pseudobulk_no_apoe_lines <span class="ot">&lt;-</span></span>
<span id="cb316-25"><a href="#cb316-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_magma_input_per_celltype</span>(pseudobulk_lines, <span class="st">"pseudobulk_no_apoe"</span>)</span>
<span id="cb316-26"><a href="#cb316-26" aria-hidden="true" tabindex="-1"></a>magma_gene_list_lines <span class="ot">&lt;-</span></span>
<span id="cb316-27"><a href="#cb316-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_magma_input_per_celltype</span>(magma_gene_list_lines, <span class="st">"sig_magma_sig_genes"</span>)</span>
<span id="cb316-28"><a href="#cb316-28" aria-hidden="true" tabindex="-1"></a>magma_list <span class="ot">&lt;-</span> <span class="fu">c</span>(pseudobulk_apoe_lines, pseudobulk_no_apoe_lines, magma_gene_list_lines)</span>
<span id="cb316-29"><a href="#cb316-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Write the character vector to a file</span></span>
<span id="cb316-30"><a href="#cb316-30" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(magma_list,</span>
<span id="cb316-31"><a href="#cb316-31" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(magma_list),</span>
<span id="cb316-32"><a href="#cb316-32" aria-hidden="true" tabindex="-1"></a>     <span class="sc">~</span> readr<span class="sc">::</span><span class="fu">write_lines</span>(.x,</span>
<span id="cb316-33"><a href="#cb316-33" aria-hidden="true" tabindex="-1"></a>                          here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb316-34"><a href="#cb316-34" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"03_data/994_magma_inputs"</span>,</span>
<span id="cb316-35"><a href="#cb316-35" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste0</span>(<span class="st">"ppi_sig_modules_"</span>, .y, <span class="st">".magma.txt"</span>)</span>
<span id="cb316-36"><a href="#cb316-36" aria-hidden="true" tabindex="-1"></a>                          )))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe`
[1] "Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_module1 1956 55103 84959 152404" 
[2] "Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_module2 23768 53942 220164 56884"
[3] "Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_module0 56675 166336 8527"       

$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe`
[1] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module1 87 55698 6782 29953 8736 1303 55812 23105 84140 51134 51571 84542 283234 8515 83723 343990 730094 374864 100885848"                                                                                            
[2] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module2 57630 374291 2670 23542 57595 55885 2902 51560 23154 57569 4685 53358 11255 256691 11076 9796 9715 3765 57528 4062 84687 727800 57026 126129 111064647"                                                        
[3] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module3 3315 572 1871 11180 8682 4691 50809 3300 8568 3727 4783 1051 10009 988 5774 51588 53635 8971 56341 92906 6612 5728 200558 9474 545 8555 94239 4300 84232 83734 91748 1628 1831 2002 1849 7335 400954 1850 7008"
[4] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module4 29929 55596 51447 156 10090 4356 2767 9488 29880 2131 8787 79095 79947 56261 10846 90632 23316 9651 5150"                                                                                                      
[5] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module5 5364 114884 9770 161742 8099 2115 83719 55558 9120 23158 23167 8822 5122 9962 2849 10809"                                                                                                                      
[6] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_module0 80228 271 11067 83638"                                                                                                                                                                                         

$`low-feature-cells_pseudobulk_apoe`
[1] "low-feature-cells_pseudobulk_apoe_module1 3320 122786 5621 1964 124905557 4240 5757 165"
[2] "low-feature-cells_pseudobulk_apoe_module2 1871 5127 4150 10567 7182 83719 127829"       
[3] "low-feature-cells_pseudobulk_apoe_module3 3300 51588 8851 3725 138151"                  
[4] "low-feature-cells_pseudobulk_apoe_module4 23236 105372522 10922 10908 10846 90632 51523"
[5] "low-feature-cells_pseudobulk_apoe_module0 23270"                                        

$`Microglia-quiescent_pseudobulk_apoe`
[1] "Microglia-quiescent_pseudobulk_apoe_module0 87 483 3382 2131 147650 2681"   
[2] "Microglia-quiescent_pseudobulk_apoe_module2 3315 2321 10892 5793 9465"      
[3] "Microglia-quiescent_pseudobulk_apoe_module3 665 5236 2632 9123 23036"       
[4] "Microglia-quiescent_pseudobulk_apoe_module4 22862 55092 6160 5315 6167 7084"
[5] "Microglia-quiescent_pseudobulk_apoe_module5 7852 5025 23042 4360"           

$`Oligo-A_pseudobulk_apoe`
[1] "Oligo-A_pseudobulk_apoe_module1 11116 1186 1820 5310 7016 6604 53615 5129 10691 256364 4752 54776 9891 10278 55527 4298 8099 23135 23241 57178 9640 10908 64423 317762 138639 57710 30818 57140 26012 83931 252884 22843 54868 100131017 116983 59269 9542 84632 118663 138151 51523 2972 140467 5613 340371 65251"                                                                                                                                                                                                                              
[2] "Oligo-A_pseudobulk_apoe_module2 115098 8260 224 6894 10963 10994 8565 5214 9521 55037 10730 8568 4724 2339 4141 57222 5184 9013 81889 54058 4835 22868 5238 2941 1718 79956 25825 9200 29968 3150 56261 7298 4907 63826 7296 84706 6120 51761 145447 64968 222236 8225 1719 283848 374887 347736"                                                                                                                                                                                                                                                
[3] "Oligo-A_pseudobulk_apoe_module3 95681 374618 4238 5119 9338 55007 90693 29099 64792 55036 112752 93343 5151 55020 79583 55084 143458"                                                                                                                                                                                                                                                                                                                                                                                                            
[4] "Oligo-A_pseudobulk_apoe_module4 26586 10427 221061 23353 23047 3480 50809 1080 400506 9765 25957 55086 29894 2176 105376156 7507 10001 5871 7343 23595 26471 165918 4437 79648 23338 271 93183 23312 201973 9859 645455 56910 10438 129530 2822 85407 414 55766 404672 55920 51750 200150 163404"                                                                                                                                                                                                                                                
[5] "Oligo-A_pseudobulk_apoe_module5 79709 4804 9770 8728 23129 22795 80781 4885 26232 55214 84895 22846 10609 337876 113201 285343 1290 85301 201229"                                                                                                                                                                                                                                                                                                                                                                                                
[6] "Oligo-A_pseudobulk_apoe_module6 9817 3643 5836 7408 3949 1509 3099 6794 4150 5970 9252 7852 6576 64135 466 7297 3656 10014 5465 54206 1843 2696 54814 156 7301 80339 6721 6478 10999 257019 6197 7942 4047 58526 4245 11133 6309 254531 3669 8694 200734 49861 1728 84335 8660 81490 81 3303 6195 55733 10603 415116 2119 126129 151525"                                                                                                                                                                                                         
[7] "Oligo-A_pseudobulk_apoe_module7 6731 54407 6507 6533 2036 22979 79187 5364 6238 483 6455 11033 50807 23125 2346 6522 1952 55114 9001 84669 91107 285175 22859 54521 9369 3707 150709 94134 79734 56262 220965 79870 22844 51555 26115 79768 80019 55314 26022 148808 5375 64168 160140 114822 161502 221692 57520 130026 9903 201266 1836 80727 10391 56834 2788 25829 54985 10124 59284 124402 51201 23362 27106 2869 257044 126917 6536 93377 200407 6526 57578 80217 221178 148641 57597 56892 339983 283991 157807 64106 126231 342035 23302"
[8] "Oligo-A_pseudobulk_apoe_module0 4023 79780 27094 22876"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

$`Oligo-B_pseudobulk_apoe`
[1] "Oligo-B_pseudobulk_apoe_module1 55914 5062 2885 665 7798 3643 7466 10818 6386 3480 338692 840 7099 2353 9765 6642 54890 10490 64135 408 5590 7187 6645 5786 3656 22822 23048 9463 51588 6383 23308 102723996 9770 7164 85363 1608 221895 64115 394 10256 9743 54206 53635 10253 604 9099 165918 9891 8408 51322 2035 2696 55114 8030 26207 4034 23710 5778 91107 51123 23643 84166 64762 79735 197259 6478 5789 1454 131566 257019 2909 50488 55083 6197 54885 10867 153222 8073 26259 23365 27296 154743 100101467 200734 4155 8660 9542 6892 7726 282890 221178 51192 10603 415116 10809 65249 147650 2119 338382 2262 53342 79937 100289279"                                                                                                                        
[2] "Oligo-B_pseudobulk_apoe_module2 8260 55860 79709 6894 285282 10963 9125 7112 54499 10611 23353 55421 1487 53371 23047 26227 989 50809 6908 4150 1080 26190 10730 6238 55795 25957 1820 5431 10940 192669 571 100379661 4163 2176 105376156 23126 7329 51631 51065 53615 80349 51184 23595 4752 7257 26471 2960 7023 4437 9793 9455 64769 57091 9913 84669 11097 22868 112869 81617 55778 55915 10069 30827 22850 84515 51241 23493 9465 54619 23160 201973 1316 91133 112950 11321 55280 9255 79780 64418 24139 83473 51754 27352 2029 26173 57455 57577 23033 10438 9753 129530 55316 79979 153642 3669 84187 152002 2218 170384 23398 84450 387521 118663 100 55520 2972 9069 64968 29902 285343 55766 404672 57597 85301 5178 6666 151525 374969 729092 65251 92335"
[3] "Oligo-B_pseudobulk_apoe_module3 54431 57544 224 56994 550 286410 5447 3156 23657 6541 9524 3949 6307 3029 4704 4141 2194 7086 51741 8544 51706 2346 6529 2224 2222 3422 81790 39 5860 10380 6721 10999 563 5229 2941 81855 1717 10908 6652 11343 4047 148808 58526 9200 29968 3157 81544 55349 2277 23052 6309 254013 254531 8694 171586 134429 283212 51380 3638 2180 7296 84706 128989 3988 2592 126129 347736"                                                                                                                                                                                                                                                                                                                                                      
[4] "Oligo-B_pseudobulk_apoe_module4 54407 55117 6533 8884 815 817 576 334 2670 2775 23125 1952 160851 57732 114783 57198 79632 286262 3707 10690 4336 6324 26022 89970 29767 2530 30818 221692 55020 2596 2788 6567 64805 56147 84102 162681 59284 8534 51201 2869 26503 138151 6536 6526 23467 1394 104909134 3705 550643 26074 148641 64881 283991 653082 64106 375612 342035"                                                                                                                                                                                                                                                                                                                                                                                           
[5] "Oligo-B_pseudobulk_apoe_module5 221061 1186 23268 122786 10811 344148 83786 23338 57475 23149 55216 54461 56262 23241 116988 1185 79956 10444 57719 6335 84542 1657 26012 130026 80727 50626 124402 116983 100316904 84288 57578 6450 7581 100271927 348751"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
[6] "Oligo-B_pseudobulk_apoe_module6 200894 57542 283377 3797 27241 55527 51626 79734 29099 6911 79768 9742 55329 161502 92211 9903 57546 22843 348938 151827"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
[7] "Oligo-B_pseudobulk_apoe_module0 828 91120 643699"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      

$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe`
[1] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module1 87 3315 3927 5589 5836 9252 5609 5595 7077 51776 79987 161742 122786 1843 3164 85021 4703 9509 112609 7392 3696 10090 59350 8863 2115 1628 1831 8736 1303 10060 56311 7007 116804918 6775 3488 1212 8321 80305 51571 55617 5122 5627 81848 84870 57188 8515 80319 3026 3304 1850 11072 64288 5549 7008 126129"
[2] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module2 57630 121227 10154 1951 54843 149371 55763 3801 10210 22898 84830 8193 55812 285753 84140 51134 9742 83696 160857 26146 80013 9903 283234 23543 27098 339451 401207 7626"                                                                                                                                     
[3] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module3 1871 5127 23047 50809 9969 23586 546 6868 51588 8295 5424 53615 7343 6949 55596 4739 7328 10274 53635 8971 84148 4438 1954 6612 5027 51447 545 8555 94239 8493 23469 4300 84232 271 23253 8099 83719 112939 79695 1896 140890 23167 400954 901 10522 59274 730094"                                            
[4] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module4 1445 550 57488 8682 51144 11047 1627 5358 4646 5763 84790 10263 57569 23243 2002 253558 64359 3988 84687"                                                                                                                                                                                                     
[5] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module5 41 4988 5577 2781 1759 7223 23542 57595 55885 2902 5332 51560 6855 23551 114794 156 80228 5662 23154 26232 5596 4356 4685 4886 11255 2767 6543 222008 8787 11076 79095 9796 3354 10846 90632 26050 758 59284 84281 56172 3765 257407 343990 27094 57348 442319 9651 4062 5150 2913 56660 374864"              
[6] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module6 29929 3382 119504 29953 419 28966 23105 2762 9488 78996 29880 10559 83723 338811 84436"                                                                                                                                                                                                                       
[7] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module7 6894 54407 11180 9512 8683 5214 374291 60676 56341 26995 9120 23158 81796 9962 56261 63917 22901 7274 145773 353497 7378 91181 64968 6526 6999 10809 57604 440957 200373 56985"                                                                                                                               
[8] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module8 4130 83734 27244 80317 169841 285989"                                                                                                                                                                                                                                                                         
[9] "Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_module0 23096 138241 645369"                                                                                                                                                                                                                                                                                          

$`In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe`
[1] "In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe_module0 8502 22866 445815"    
[2] "In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe_module2 6857 2902 23504"      
[3] "In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe_module3 4208 3745 10846 90632"

$`T-Pericyte_pseudobulk_no_apoe`
[1] "T-Pericyte_pseudobulk_no_apoe_module1 1501 57554 114088 288 29106 9892 57502 57526 9472 4897 9378 1608 30820 1496 5521 50649 1795 114784 9699 344148 9901 257194 5803 160851 2891 83992 1804 10752 285175 6506 56934 84465 57689 4915 2259 2257 4684 4685 138046 10690 64067 10178 79870 1008 152330 8537 460 57476 26230 22854 53353 114805 27092 120114 2272 84457 9515 29091 339479 80333 116135 92737 1946 1600 114792 10497 3751 22986 1826 2898 221938 107 28316 2917 30010 2897 80059 266722 29951 23705 80852 200150 10718 64478 338645 6263 10242 3776 114788 375612 10777 161357 728215"                                                                                                                                                                              
[2] "T-Pericyte_pseudobulk_no_apoe_module2 6539 1739 9351 6558 57498 91869 23657 1355 326625 115416 56995 57038 8613 22868 7915 119559 4731 79770 206358 128387 4358 51754 26152 6563 6505 83959 8644 10391 197258 51084 1646 51380 9942 93210 51523 6536 4482 6584 9054 6526 128077 219902 148641"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
[3] "T-Pericyte_pseudobulk_no_apoe_module3 87 5352 5064 5099 4052 9475 1809 5156 3675 1605 5608 23418 7040 285 6095 10220 3696 3707 10144 6444 57572 384 653583 8840 23768 57157 64093 10219 7035 140685 23764 140876 10699 6695 654 54212 5228 361 27286 442319 953 85301 116535 81615"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
[4] "T-Pericyte_pseudobulk_no_apoe_module4 93323 8458 121457 6747 27316 55223 10717 23589 85302 55255 57410 4436 9782 5530 5879 7324 1153 23195 9129 10521 5315 672 5577 1820 6608 408 11043 2176 105376156 5430 1161 836 51631 8473 85363 5501 64753 3720 1488 9541 10054 55339 80854 105377622 80205 23424 23090 23383 4211 84661 55615 1894 51762 84936 84991 8030 3801 4212 196403 57561 23132 55819 114804 55250 7024 89849 84100 6744 23604 5991 8099 57103 57593 9663 151613 253461 23160 26013 10933 112885 11022 22887 25923 6197 143872 9728 23432 29968 53826 3151 10438 84059 54704 4121 57645 9849 10724 127253 151742 285386 150962 126282 23379 23051 677 55016 56917 9735 25950 9697 286183 54700 10594 170954 107987457 81890 10628 7840 202374 340371 256643 57758"
[5] "T-Pericyte_pseudobulk_no_apoe_module5 80817 54477 9922 83658 374618 56243 84310 115548 205 9322 11228 29911 7597 50807 8601 8853 29886 23143 4034 8382 84239 3423 64131 29775 152485 6752 23400 9729 57623 79839 23093 26137 84527 199223 54622 56171 57577 79925 201456 55790 80727 387921 9394 83891 54558 202151 284391 63920 486 64968 9223 54886 90196 7620 445815 140688 79144 10809 27300 390927 8622 114786 348751 442213"                                                                                                                                                                                                                                                                                                                                              
[6] "T-Pericyte_pseudobulk_no_apoe_module0 203259 23305 7100 8706 121793 4661"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       

$`_sig_magma_sig_genes`
character(0)</code></pre>
</div>
</div></div>
</section>
<section id="treeplots" class="level3">
<h3 class="anchored" data-anchor-id="treeplots">Treeplots</h3>
<div class="cell-container"><div class="cell-decorator"><pre>In [38]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb318"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb318-1"><a href="#cb318-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the first module's GO enrichment result as an example</span></span>
<span id="cb318-2"><a href="#cb318-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(result_list[[<span class="dv">1</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb318-3"><a href="#cb318-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dotplot</span>(result_list[[<span class="dv">1</span>]])</span>
<span id="cb318-4"><a href="#cb318-4" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-38-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-168"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb319"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb319-1"><a href="#cb319-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get semantic similarity of terms</span></span>
<span id="cb319-2"><a href="#cb319-2" aria-hidden="true" tabindex="-1"></a>result_list <span class="ot">&lt;-</span> <span class="fu">map</span>(result_list, pairwise_termsim)</span>
<span id="cb319-3"><a href="#cb319-3" aria-hidden="true" tabindex="-1"></a><span class="fu">treeplot</span>(result_list<span class="sc">$</span><span class="st">`</span><span class="at">Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_module3_BP</span><span class="st">`</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-38-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-169"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-38-2.png" class="img-fluid figure-img" width="672"></a></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb320"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb320-1"><a href="#cb320-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a safe version of your treeplot function that returns NULL on error</span></span>
<span id="cb320-2"><a href="#cb320-2" aria-hidden="true" tabindex="-1"></a>safe_treeplot <span class="ot">&lt;-</span> <span class="fu">possibly</span>(treeplot, <span class="at">otherwise =</span> <span class="cn">NULL</span>)</span>
<span id="cb320-3"><a href="#cb320-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Use map with the safe version of the function</span></span>
<span id="cb320-4"><a href="#cb320-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plan</span>(<span class="st">"multisession"</span>, <span class="at">workers =</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>() <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb320-5"><a href="#cb320-5" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> <span class="fu">future_map</span>(result_list, safe_treeplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.
! # Invaild edge matrix for &lt;phylo&gt;. A &lt;tbl_df&gt; is returned.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: UNRELIABLE VALUE: Future ('&lt;none&gt;') unexpectedly generated random
numbers without specifying argument 'seed'. There is a risk that those random
numbers are not statistically sound and the overall results might be invalid.
To fix this, specify 'seed=TRUE'. This ensures that proper, parallel-safe
random numbers are produced via the L'Ecuyer-CMRG method. To disable this
check, use 'seed=NULL', or set option 'future.rng.onMisuse' to "ignore".</code></pre>
</div>
<div class="sourceCode cell-code" id="cb351"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb351-1"><a href="#cb351-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any NULL entries</span></span>
<span id="cb351-2"><a href="#cb351-2" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> treeplots[<span class="sc">!</span><span class="fu">sapply</span>(treeplots, is.null)]</span>
<span id="cb351-3"><a href="#cb351-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Assert that there are no nulls in list</span></span>
<span id="cb351-4"><a href="#cb351-4" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">see_if</span>(<span class="fu">sum</span>(<span class="fu">map_lgl</span>(treeplots, is.null)) <span class="sc">==</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb353"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb353-1"><a href="#cb353-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace those with no treeplot with dotplots</span></span>
<span id="cb353-2"><a href="#cb353-2" aria-hidden="true" tabindex="-1"></a>no_treeplot <span class="ot">&lt;-</span> <span class="fu">names</span>(result_list)[<span class="sc">!</span><span class="fu">names</span>(result_list) <span class="sc">%in%</span> <span class="fu">names</span>(treeplots)]</span>
<span id="cb353-3"><a href="#cb353-3" aria-hidden="true" tabindex="-1"></a>dotplots <span class="ot">&lt;-</span> <span class="fu">map</span>(result_list[no_treeplot], dotplot)</span>
<span id="cb353-4"><a href="#cb353-4" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">see_if</span>(<span class="fu">sum</span>(<span class="fu">map_lgl</span>(dotplots, is.null)) <span class="sc">==</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb355"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb355-1"><a href="#cb355-1" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> <span class="fu">c</span>(treeplots, dotplots)</span>
<span id="cb355-2"><a href="#cb355-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-3"><a href="#cb355-3" aria-hidden="true" tabindex="-1"></a>treeplots <span class="ot">&lt;-</span> <span class="fu">map2</span>(treeplots, <span class="fu">names</span>(treeplots), <span class="sc">~</span> .x <span class="sc">+</span> <span class="fu">ggtitle</span>(.y))</span>
<span id="cb355-4"><a href="#cb355-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-5"><a href="#cb355-5" aria-hidden="true" tabindex="-1"></a><span class="co"># I want to make patchwork of plots per celltype</span></span>
<span id="cb355-6"><a href="#cb355-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract unique cell types from the plot names</span></span>
<span id="cb355-7"><a href="#cb355-7" aria-hidden="true" tabindex="-1"></a>celltypes <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">sub</span>(<span class="st">"_module.*"</span>, <span class="st">""</span>, <span class="fu">names</span>(treeplots)))</span>
<span id="cb355-8"><a href="#cb355-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-9"><a href="#cb355-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Split the list of plots into a nested list by cell type</span></span>
<span id="cb355-10"><a href="#cb355-10" aria-hidden="true" tabindex="-1"></a>get_celltype_ontology_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(celltype, ontology, plot_list) {</span>
<span id="cb355-11"><a href="#cb355-11" aria-hidden="true" tabindex="-1"></a>  plots_for_type <span class="ot">&lt;-</span> plot_list[<span class="fu">grepl</span>(<span class="fu">paste0</span>(<span class="st">"^"</span>, celltype, <span class="st">"_module"</span>), <span class="fu">names</span>(plot_list))]</span>
<span id="cb355-12"><a href="#cb355-12" aria-hidden="true" tabindex="-1"></a>  plots_ont <span class="ot">&lt;-</span> plots_for_type[<span class="fu">grepl</span>(ontology, <span class="fu">names</span>(plots_for_type))]</span>
<span id="cb355-13"><a href="#cb355-13" aria-hidden="true" tabindex="-1"></a>  combined_plot <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(plots_ont, <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb355-14"><a href="#cb355-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(plots_ont) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb355-15"><a href="#cb355-15" aria-hidden="true" tabindex="-1"></a>  combined_plot <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb355-16"><a href="#cb355-16" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb355-17"><a href="#cb355-17" aria-hidden="true" tabindex="-1"></a>  combined_plot</span>
<span id="cb355-18"><a href="#cb355-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb355-19"><a href="#cb355-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-20"><a href="#cb355-20" aria-hidden="true" tabindex="-1"></a>plots_by_celltype_bp <span class="ot">&lt;-</span></span>
<span id="cb355-21"><a href="#cb355-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(celltypes, get_celltype_ontology_plots, <span class="st">"_BP"</span>, treeplots) <span class="sc">|&gt;</span></span>
<span id="cb355-22"><a href="#cb355-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste0</span>(celltypes, <span class="st">"_BP"</span>))</span>
<span id="cb355-23"><a href="#cb355-23" aria-hidden="true" tabindex="-1"></a>plots_by_celltype_cc <span class="ot">&lt;-</span></span>
<span id="cb355-24"><a href="#cb355-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(celltypes, get_celltype_ontology_plots, <span class="st">"_CC"</span>, treeplots) <span class="sc">|&gt;</span></span>
<span id="cb355-25"><a href="#cb355-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste0</span>(celltypes, <span class="st">"_CC"</span>))</span>
<span id="cb355-26"><a href="#cb355-26" aria-hidden="true" tabindex="-1"></a>plots_by_celltype_mf <span class="ot">&lt;-</span></span>
<span id="cb355-27"><a href="#cb355-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(celltypes, get_celltype_ontology_plots, <span class="st">"_MF"</span>, treeplots) <span class="sc">|&gt;</span></span>
<span id="cb355-28"><a href="#cb355-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">paste0</span>(celltypes, <span class="st">"_MF"</span>))</span>
<span id="cb355-29"><a href="#cb355-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-30"><a href="#cb355-30" aria-hidden="true" tabindex="-1"></a>plots_by_celltype <span class="ot">&lt;-</span></span>
<span id="cb355-31"><a href="#cb355-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(plots_by_celltype_bp,</span>
<span id="cb355-32"><a href="#cb355-32" aria-hidden="true" tabindex="-1"></a>    plots_by_celltype_cc,</span>
<span id="cb355-33"><a href="#cb355-33" aria-hidden="true" tabindex="-1"></a>    plots_by_celltype_mf)</span>
<span id="cb355-34"><a href="#cb355-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove nulls</span></span>
<span id="cb355-35"><a href="#cb355-35" aria-hidden="true" tabindex="-1"></a>plots_by_celltype <span class="ot">&lt;-</span> plots_by_celltype[<span class="sc">!</span><span class="fu">sapply</span>(plots_by_celltype, is.null)]</span>
<span id="cb355-36"><a href="#cb355-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb355-37"><a href="#cb355-37" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(plots_by_celltype_bp, plots_by_celltype_cc, plots_by_celltype_mf)</span>
<span id="cb355-38"><a href="#cb355-38" aria-hidden="true" tabindex="-1"></a><span class="co"># save the plots nice and tall</span></span>
<span id="cb355-39"><a href="#cb355-39" aria-hidden="true" tabindex="-1"></a><span class="fu">walk2</span>(plots_by_celltype, <span class="fu">names</span>(plots_by_celltype), <span class="sc">~</span> <span class="fu">ggsave</span>(</span>
<span id="cb355-40"><a href="#cb355-40" aria-hidden="true" tabindex="-1"></a>  <span class="at">filename =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/ppi_modules"</span>, <span class="fu">paste0</span>(.y, <span class="st">"_combined_plot.png"</span>)),</span>
<span id="cb355-41"><a href="#cb355-41" aria-hidden="true" tabindex="-1"></a>  <span class="at">plot =</span> .x,</span>
<span id="cb355-42"><a href="#cb355-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">width =</span> <span class="dv">14</span>,</span>
<span id="cb355-43"><a href="#cb355-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">height =</span> <span class="dv">34</span></span>
<span id="cb355-44"><a href="#cb355-44" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [39]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb356"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb356-1"><a href="#cb356-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 25</span></span>
<span id="cb356-2"><a href="#cb356-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 14</span></span>
<span id="cb356-3"><a href="#cb356-3" aria-hidden="true" tabindex="-1"></a>plots_by_celltype</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-170"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-171"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-2.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`low-feature-cells_pseudobulk_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-172"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-3.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-quiescent_pseudobulk_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-4.png" class="lightbox" data-gallery="quarto-lightbox-gallery-173"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-4.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-A_pseudobulk_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-5.png" class="lightbox" data-gallery="quarto-lightbox-gallery-174"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-5.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-6.png" class="lightbox" data-gallery="quarto-lightbox-gallery-175"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-6.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-7.png" class="lightbox" data-gallery="quarto-lightbox-gallery-176"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-7.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-Pericyte_pseudobulk_no_apoe_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-8.png" class="lightbox" data-gallery="quarto-lightbox-gallery-177"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-8.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-9.png" class="lightbox" data-gallery="quarto-lightbox-gallery-178"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-9.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-10.png" class="lightbox" data-gallery="quarto-lightbox-gallery-179"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-10.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-11.png" class="lightbox" data-gallery="quarto-lightbox-gallery-180"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-11.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-12.png" class="lightbox" data-gallery="quarto-lightbox-gallery-181"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-12.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-13.png" class="lightbox" data-gallery="quarto-lightbox-gallery-182"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-13.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-14.png" class="lightbox" data-gallery="quarto-lightbox-gallery-183"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-14.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-15.png" class="lightbox" data-gallery="quarto-lightbox-gallery-184"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-15.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-16.png" class="lightbox" data-gallery="quarto-lightbox-gallery-185"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-16.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-17.png" class="lightbox" data-gallery="quarto-lightbox-gallery-186"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-17.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-18.png" class="lightbox" data-gallery="quarto-lightbox-gallery-187"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-18.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-19.png" class="lightbox" data-gallery="quarto-lightbox-gallery-188"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-19.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-20.png" class="lightbox" data-gallery="quarto-lightbox-gallery-189"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-20.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-21.png" class="lightbox" data-gallery="quarto-lightbox-gallery-190"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-21.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-22.png" class="lightbox" data-gallery="quarto-lightbox-gallery-191"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-22.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$sig_magma_celltypes_BP</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-23.png" class="lightbox" data-gallery="quarto-lightbox-gallery-192"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-23.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-24.png" class="lightbox" data-gallery="quarto-lightbox-gallery-193"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-24.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-25.png" class="lightbox" data-gallery="quarto-lightbox-gallery-194"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-25.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_BP`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-26.png" class="lightbox" data-gallery="quarto-lightbox-gallery-195"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-26.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-27.png" class="lightbox" data-gallery="quarto-lightbox-gallery-196"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-27.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-28.png" class="lightbox" data-gallery="quarto-lightbox-gallery-197"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-28.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`low-feature-cells_pseudobulk_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-29.png" class="lightbox" data-gallery="quarto-lightbox-gallery-198"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-29.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-quiescent_pseudobulk_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-30.png" class="lightbox" data-gallery="quarto-lightbox-gallery-199"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-30.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-A_pseudobulk_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-31.png" class="lightbox" data-gallery="quarto-lightbox-gallery-200"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-31.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-32.png" class="lightbox" data-gallery="quarto-lightbox-gallery-201"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-32.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-33.png" class="lightbox" data-gallery="quarto-lightbox-gallery-202"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-33.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-Pericyte_pseudobulk_no_apoe_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-34.png" class="lightbox" data-gallery="quarto-lightbox-gallery-203"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-34.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-35.png" class="lightbox" data-gallery="quarto-lightbox-gallery-204"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-35.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-36.png" class="lightbox" data-gallery="quarto-lightbox-gallery-205"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-36.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-37.png" class="lightbox" data-gallery="quarto-lightbox-gallery-206"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-37.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-38.png" class="lightbox" data-gallery="quarto-lightbox-gallery-207"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-38.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-39.png" class="lightbox" data-gallery="quarto-lightbox-gallery-208"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-39.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-40.png" class="lightbox" data-gallery="quarto-lightbox-gallery-209"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-40.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-41.png" class="lightbox" data-gallery="quarto-lightbox-gallery-210"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-41.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-42.png" class="lightbox" data-gallery="quarto-lightbox-gallery-211"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-42.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-43.png" class="lightbox" data-gallery="quarto-lightbox-gallery-212"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-43.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-44.png" class="lightbox" data-gallery="quarto-lightbox-gallery-213"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-44.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-45.png" class="lightbox" data-gallery="quarto-lightbox-gallery-214"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-45.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-46.png" class="lightbox" data-gallery="quarto-lightbox-gallery-215"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-46.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-47.png" class="lightbox" data-gallery="quarto-lightbox-gallery-216"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-47.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-48.png" class="lightbox" data-gallery="quarto-lightbox-gallery-217"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-48.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$sig_magma_celltypes_CC</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-49.png" class="lightbox" data-gallery="quarto-lightbox-gallery-218"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-49.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-50.png" class="lightbox" data-gallery="quarto-lightbox-gallery-219"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-50.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-51.png" class="lightbox" data-gallery="quarto-lightbox-gallery-220"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-51.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_CC`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-52.png" class="lightbox" data-gallery="quarto-lightbox-gallery-221"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-52.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-ADARB2-ERBB4_pseudobulk_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-53.png" class="lightbox" data-gallery="quarto-lightbox-gallery-222"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-53.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-54.png" class="lightbox" data-gallery="quarto-lightbox-gallery-223"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-54.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`low-feature-cells_pseudobulk_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-55.png" class="lightbox" data-gallery="quarto-lightbox-gallery-224"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-55.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-quiescent_pseudobulk_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-56.png" class="lightbox" data-gallery="quarto-lightbox-gallery-225"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-56.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-A_pseudobulk_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-57.png" class="lightbox" data-gallery="quarto-lightbox-gallery-226"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-57.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-L4-5-RORB-IL1RAPL2_pseudobulk_no_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-58.png" class="lightbox" data-gallery="quarto-lightbox-gallery-227"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-58.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-DPP10-MEF2C_pseudobulk_no_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-59.png" class="lightbox" data-gallery="quarto-lightbox-gallery-228"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-59.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`T-Pericyte_pseudobulk_no_apoe_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-60.png" class="lightbox" data-gallery="quarto-lightbox-gallery-229"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-60.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-61.png" class="lightbox" data-gallery="quarto-lightbox-gallery-230"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-61.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-five-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-62.png" class="lightbox" data-gallery="quarto-lightbox-gallery-231"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-62.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-63.png" class="lightbox" data-gallery="quarto-lightbox-gallery-232"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-63.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-64.png" class="lightbox" data-gallery="quarto-lightbox-gallery-233"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-64.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig-magma-sig-genes_EUROPEUKBB_35k10k_all_controls_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-65.png" class="lightbox" data-gallery="quarto-lightbox-gallery-234"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-65.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-five-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-66.png" class="lightbox" data-gallery="quarto-lightbox-gallery-235"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-66.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Oligo-A_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-five-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-67.png" class="lightbox" data-gallery="quarto-lightbox-gallery-236"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-67.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-68.png" class="lightbox" data-gallery="quarto-lightbox-gallery-237"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-68.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-quiescent_sig-magma-sig-genes_EUROPEUKBB_35k10k_controls_parenchymal_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-69.png" class="lightbox" data-gallery="quarto-lightbox-gallery-238"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-69.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Ex-Neuron-ADARB2-ERBB4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-70.png" class="lightbox" data-gallery="quarto-lightbox-gallery-239"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-70.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-DPP10-MEF2C_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-71.png" class="lightbox" data-gallery="quarto-lightbox-gallery-240"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-71.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`In-Neuron-PRKG1-TBC1D4_sig-magma-sig-genes_pd_35k10k_controls_parenchymal_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-72.png" class="lightbox" data-gallery="quarto-lightbox-gallery-241"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-72.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Meningeal-FB_sig-magma-sig-genes_AD_50k_controls_vascular_top-five-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-73.png" class="lightbox" data-gallery="quarto-lightbox-gallery-242"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-73.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB_sig-magma-sig-genes_pd_35k10k_controls_vascular_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-74.png" class="lightbox" data-gallery="quarto-lightbox-gallery-243"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-74.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$sig_magma_celltypes_MF</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-75.png" class="lightbox" data-gallery="quarto-lightbox-gallery-244"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-75.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Perivascular-FB-KAZN2_sig_magma_extended_NA_all_controls_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-76.png" class="lightbox" data-gallery="quarto-lightbox-gallery-245"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-76.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Microglia-activated_sig_magma_extended_NA_all_controls_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-77.png" class="lightbox" data-gallery="quarto-lightbox-gallery-246"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-77.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
$`Pericyte-2_sig_magma_extended_NA_all_controls_top-ten-percent_MF`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><a href="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-78.png" class="lightbox" data-gallery="quarto-lightbox-gallery-247"><img src="17_protein-protein_interactions_files/figure-html/unnamed-chunk-39-78.png" class="img-fluid figure-img" width="1344"></a></p>
</figure>
</div>
</div>
</div></div>
</section>
<section id="module-labels" class="level3">
<h3 class="anchored" data-anchor-id="module-labels">Module labels</h3>
<p>So there isnt really a mathematical solution to this other than eyeballing the plots and assigning module functions</p>
<section id="pseudobulk-labels" class="level4">
<h4 class="anchored" data-anchor-id="pseudobulk-labels">Pseudobulk labels</h4>
<div class="cell-container"><div class="cell-decorator"><pre>In [40]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb435"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb435-1"><a href="#cb435-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb435-2"><a href="#cb435-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Astro module 1 has memory stuff</span></span>
<span id="cb435-3"><a href="#cb435-3" aria-hidden="true" tabindex="-1"></a>module_assignments <span class="ot">&lt;-</span></span>
<span id="cb435-4"><a href="#cb435-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="co">#astro-activ</span></span>
<span id="cb435-5"><a href="#cb435-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"synapse_regulation-memory"</span>,</span>
<span id="cb435-6"><a href="#cb435-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"plasma_membrane"</span>,</span>
<span id="cb435-7"><a href="#cb435-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anion_transport"</span>,</span>
<span id="cb435-8"><a href="#cb435-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cell_adhesion"</span>,</span>
<span id="cb435-9"><a href="#cb435-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"basal_membrane-exosome"</span>,</span>
<span id="cb435-10"><a href="#cb435-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># astro-quie</span></span>
<span id="cb435-11"><a href="#cb435-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"synaptic_transmission-memory"</span>,</span>
<span id="cb435-12"><a href="#cb435-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ubiquitin-complexes"</span>,</span>
<span id="cb435-13"><a href="#cb435-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cell_junction-actin"</span>,</span>
<span id="cb435-14"><a href="#cb435-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"vascular_transport"</span>,</span>
<span id="cb435-15"><a href="#cb435-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"glycosylation"</span>,</span>
<span id="cb435-16"><a href="#cb435-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"extracellular_matrix"</span>,</span>
<span id="cb435-17"><a href="#cb435-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"chemotaxis-pos_reg_cascades"</span>,</span>
<span id="cb435-18"><a href="#cb435-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ex-neuron l2-3</span></span>
<span id="cb435-19"><a href="#cb435-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"endothelial_migration-preptide_hormone"</span>,</span>
<span id="cb435-20"><a href="#cb435-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"synapse_protein_localisation"</span>,</span>
<span id="cb435-21"><a href="#cb435-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ribonuclieotide_catabolic"</span>,</span>
<span id="cb435-22"><a href="#cb435-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"rna_splicing"</span>,</span>
<span id="cb435-23"><a href="#cb435-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"wnt"</span>,</span>
<span id="cb435-24"><a href="#cb435-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ex-neuron-l4-5</span></span>
<span id="cb435-25"><a href="#cb435-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"extracellular_matrix_organisation"</span>,</span>
<span id="cb435-26"><a href="#cb435-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"negative_reg_protein_phos"</span>,</span>
<span id="cb435-27"><a href="#cb435-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">"chromatid-mitotic_spindle"</span>,</span>
<span id="cb435-28"><a href="#cb435-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">"glycosyltransferase"</span>,</span>
<span id="cb435-29"><a href="#cb435-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bbb_transport-anion"</span>,</span>
<span id="cb435-30"><a href="#cb435-30" aria-hidden="true" tabindex="-1"></a>    <span class="st">"salt_reponse-calcium"</span>,</span>
<span id="cb435-31"><a href="#cb435-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># in-neuron</span></span>
<span id="cb435-32"><a href="#cb435-32" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ubiquitin_ligasse"</span>,</span>
<span id="cb435-33"><a href="#cb435-33" aria-hidden="true" tabindex="-1"></a>    <span class="st">"chloride"</span>,</span>
<span id="cb435-34"><a href="#cb435-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">"IGF"</span>,</span>
<span id="cb435-35"><a href="#cb435-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">"vesicle-synapse"</span>,</span>
<span id="cb435-36"><a href="#cb435-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gtpase_activity"</span>,</span>
<span id="cb435-37"><a href="#cb435-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># M-pericyte</span></span>
<span id="cb435-38"><a href="#cb435-38" aria-hidden="true" tabindex="-1"></a>    <span class="st">"adhesion"</span>,</span>
<span id="cb435-39"><a href="#cb435-39" aria-hidden="true" tabindex="-1"></a>    <span class="st">"endo_retic"</span>,</span>
<span id="cb435-40"><a href="#cb435-40" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cellcell_adhesion"</span>,</span>
<span id="cb435-41"><a href="#cb435-41" aria-hidden="true" tabindex="-1"></a>    <span class="st">"basal_plasma_membrane-stress"</span>,</span>
<span id="cb435-42"><a href="#cb435-42" aria-hidden="true" tabindex="-1"></a>    <span class="co"># microglia-activated</span></span>
<span id="cb435-43"><a href="#cb435-43" aria-hidden="true" tabindex="-1"></a>    <span class="st">"amino_acid_transport-sodium_ion"</span>,</span>
<span id="cb435-44"><a href="#cb435-44" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cholesterol"</span>,</span>
<span id="cb435-45"><a href="#cb435-45" aria-hidden="true" tabindex="-1"></a>    <span class="st">"peptide_catabolic"</span>,</span>
<span id="cb435-46"><a href="#cb435-46" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gtpase_reg-binding"</span>,</span>
<span id="cb435-47"><a href="#cb435-47" aria-hidden="true" tabindex="-1"></a>    <span class="co"># t-pericyte</span></span>
<span id="cb435-48"><a href="#cb435-48" aria-hidden="true" tabindex="-1"></a>    <span class="st">"glutamate-synaptic"</span>,</span>
<span id="cb435-49"><a href="#cb435-49" aria-hidden="true" tabindex="-1"></a>    <span class="st">"basal_plasma"</span>,</span>
<span id="cb435-50"><a href="#cb435-50" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cellcell_adhesion-synapse_organisation-epitheilal_migration"</span>,</span>
<span id="cb435-51"><a href="#cb435-51" aria-hidden="true" tabindex="-1"></a>    <span class="st">"neuron_projection_regulation"</span>,</span>
<span id="cb435-52"><a href="#cb435-52" aria-hidden="true" tabindex="-1"></a>    <span class="co"># trans-endo-to-mural</span></span>
<span id="cb435-53"><a href="#cb435-53" aria-hidden="true" tabindex="-1"></a>    <span class="st">"carboxylic_acid_transport"</span>,</span>
<span id="cb435-54"><a href="#cb435-54" aria-hidden="true" tabindex="-1"></a>    <span class="st">"synapce_reg-chemotaxis"</span>,</span>
<span id="cb435-55"><a href="#cb435-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">"cytoskeleton_actin"</span>,</span>
<span id="cb435-56"><a href="#cb435-56" aria-hidden="true" tabindex="-1"></a>    <span class="st">"pyrimidine_metabolic"</span>)</span>
<span id="cb435-57"><a href="#cb435-57" aria-hidden="true" tabindex="-1"></a>pseudobulk_modules <span class="ot">&lt;-</span> <span class="fu">names</span>(treeplots)[<span class="fu">grepl</span>(<span class="st">"_pseudobulk_"</span>, <span class="fu">names</span>(treeplots))]</span>
<span id="cb435-58"><a href="#cb435-58" aria-hidden="true" tabindex="-1"></a>module_assignments <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="st">"module"</span> <span class="ot">=</span> pseudobulk_modules, <span class="st">"label"</span> <span class="ot">=</span> module_assignments)</span>
<span id="cb435-59"><a href="#cb435-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb435-60"><a href="#cb435-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Save module assignment</span></span>
<span id="cb435-61"><a href="#cb435-61" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb435-62"><a href="#cb435-62" aria-hidden="true" tabindex="-1"></a>  module_assignments,</span>
<span id="cb435-63"><a href="#cb435-63" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb435-64"><a href="#cb435-64" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions"</span>,</span>
<span id="cb435-65"><a href="#cb435-65" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ppi_sig_module_labels.csv"</span></span>
<span id="cb435-66"><a href="#cb435-66" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb435-67"><a href="#cb435-67" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="magma-sig-gene-labels" class="level4">
<h4 class="anchored" data-anchor-id="magma-sig-gene-labels">MAGMA sig gene labels</h4>
<section id="stroke" class="level5">
<h5 class="anchored" data-anchor-id="stroke">Stroke</h5>
<div class="cell-container"><div class="cell-decorator"><pre>In [41]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb436"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb436-1"><a href="#cb436-1" aria-hidden="true" tabindex="-1"></a>vascular_stroke <span class="ot">&lt;-</span></span>
<span id="cb436-2"><a href="#cb436-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb436-3"><a href="#cb436-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Arterial"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-4"><a href="#cb436-4" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"epithelial_cell_migration"</span>,</span>
<span id="cb436-5"><a href="#cb436-5" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"stem_cell_prolif-g2_m_transition"</span>,</span>
<span id="cb436-6"><a href="#cb436-6" aria-hidden="true" tabindex="-1"></a>      <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"macroautophagy"</span>,</span>
<span id="cb436-7"><a href="#cb436-7" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"gtpase-axon"</span>,</span>
<span id="cb436-8"><a href="#cb436-8" aria-hidden="true" tabindex="-1"></a>      <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"angiogenesis"</span></span>
<span id="cb436-9"><a href="#cb436-9" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-10"><a href="#cb436-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Arteriolar-SMC"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-11"><a href="#cb436-11" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"actin_org_stress"</span>,</span>
<span id="cb436-12"><a href="#cb436-12" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"autophagy"</span>,</span>
<span id="cb436-13"><a href="#cb436-13" aria-hidden="true" tabindex="-1"></a>      <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"heart_contraction"</span>,</span>
<span id="cb436-14"><a href="#cb436-14" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"extracell_matrix"</span>,</span>
<span id="cb436-15"><a href="#cb436-15" aria-hidden="true" tabindex="-1"></a>      <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"cell_dev"</span>,</span>
<span id="cb436-16"><a href="#cb436-16" aria-hidden="true" tabindex="-1"></a>      <span class="st">"8"</span> <span class="ot">=</span> <span class="st">"serine_kinase"</span></span>
<span id="cb436-17"><a href="#cb436-17" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-18"><a href="#cb436-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Capilary"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-19"><a href="#cb436-19" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"cell_function"</span>,</span>
<span id="cb436-20"><a href="#cb436-20" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"autophagy"</span>,</span>
<span id="cb436-21"><a href="#cb436-21" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"kidney-sex_development"</span>,</span>
<span id="cb436-22"><a href="#cb436-22" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"mrna_processing"</span></span>
<span id="cb436-23"><a href="#cb436-23" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-24"><a href="#cb436-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Meningeal-FB"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-25"><a href="#cb436-25" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"extracellular_matrix"</span>,</span>
<span id="cb436-26"><a href="#cb436-26" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"dephosphorylation"</span>,</span>
<span id="cb436-27"><a href="#cb436-27" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"autophagy"</span>,</span>
<span id="cb436-28"><a href="#cb436-28" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"glycosylation"</span>,</span>
<span id="cb436-29"><a href="#cb436-29" aria-hidden="true" tabindex="-1"></a>      <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"rna_splicing"</span></span>
<span id="cb436-30"><a href="#cb436-30" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-31"><a href="#cb436-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pericyte"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-32"><a href="#cb436-32" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"epithelium_migration"</span>,</span>
<span id="cb436-33"><a href="#cb436-33" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"fatty_acid_synthesis"</span>,</span>
<span id="cb436-34"><a href="#cb436-34" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"rna_splice"</span>,</span>
<span id="cb436-35"><a href="#cb436-35" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"pcg_protein"</span>,</span>
<span id="cb436-36"><a href="#cb436-36" aria-hidden="true" tabindex="-1"></a>      <span class="st">"7"</span> <span class="ot">=</span> <span class="st">"ubiquitination"</span></span>
<span id="cb436-37"><a href="#cb436-37" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-38"><a href="#cb436-38" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Pericyte-2"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-39"><a href="#cb436-39" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"endosomal"</span>,</span>
<span id="cb436-40"><a href="#cb436-40" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"epithelium_migration"</span>,</span>
<span id="cb436-41"><a href="#cb436-41" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"limb_development"</span>,</span>
<span id="cb436-42"><a href="#cb436-42" aria-hidden="true" tabindex="-1"></a>      <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"cerebellum_development"</span></span>
<span id="cb436-43"><a href="#cb436-43" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-44"><a href="#cb436-44" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T-Pericyte"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-45"><a href="#cb436-45" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"cytoskeleton"</span>,</span>
<span id="cb436-46"><a href="#cb436-46" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"autophagy"</span>,</span>
<span id="cb436-47"><a href="#cb436-47" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"serine_kinase"</span>,</span>
<span id="cb436-48"><a href="#cb436-48" aria-hidden="true" tabindex="-1"></a>      <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"axon_guidance"</span>,</span>
<span id="cb436-49"><a href="#cb436-49" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"rna_polymerase"</span></span>
<span id="cb436-50"><a href="#cb436-50" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-51"><a href="#cb436-51" aria-hidden="true" tabindex="-1"></a>    <span class="st">"M-Pericyte"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-52"><a href="#cb436-52" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"pos_reg_cytoskele"</span>,</span>
<span id="cb436-53"><a href="#cb436-53" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"serine_kinase"</span>,</span>
<span id="cb436-54"><a href="#cb436-54" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"autophagy"</span>,</span>
<span id="cb436-55"><a href="#cb436-55" aria-hidden="true" tabindex="-1"></a>      <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"neg_reg_angiogenesis"</span>,</span>
<span id="cb436-56"><a href="#cb436-56" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"cation_transport"</span></span>
<span id="cb436-57"><a href="#cb436-57" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb436-58"><a href="#cb436-58" aria-hidden="true" tabindex="-1"></a>    <span class="st">"T-cell-mixed"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-59"><a href="#cb436-59" aria-hidden="true" tabindex="-1"></a>        <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"cell_differentiation-axon"</span>,</span>
<span id="cb436-60"><a href="#cb436-60" aria-hidden="true" tabindex="-1"></a>        <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"gtpase"</span>,</span>
<span id="cb436-61"><a href="#cb436-61" aria-hidden="true" tabindex="-1"></a>        <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"rna_splicing"</span>,</span>
<span id="cb436-62"><a href="#cb436-62" aria-hidden="true" tabindex="-1"></a>        <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"ubiqutination"</span></span>
<span id="cb436-63"><a href="#cb436-63" aria-hidden="true" tabindex="-1"></a>      ),</span>
<span id="cb436-64"><a href="#cb436-64" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Vascular-SMC-LINC00486"</span> <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb436-65"><a href="#cb436-65" aria-hidden="true" tabindex="-1"></a>      <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"actin"</span>,</span>
<span id="cb436-66"><a href="#cb436-66" aria-hidden="true" tabindex="-1"></a>      <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"mrna"</span>,</span>
<span id="cb436-67"><a href="#cb436-67" aria-hidden="true" tabindex="-1"></a>      <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"gtpase"</span>,</span>
<span id="cb436-68"><a href="#cb436-68" aria-hidden="true" tabindex="-1"></a>      <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"histone"</span>,</span>
<span id="cb436-69"><a href="#cb436-69" aria-hidden="true" tabindex="-1"></a>      <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"angiogenesis"</span></span>
<span id="cb436-70"><a href="#cb436-70" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb436-71"><a href="#cb436-71" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb436-72"><a href="#cb436-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-73"><a href="#cb436-73" aria-hidden="true" tabindex="-1"></a><span class="co"># function to make df per vector</span></span>
<span id="cb436-74"><a href="#cb436-74" aria-hidden="true" tabindex="-1"></a>make_ppi_df <span class="ot">&lt;-</span> <span class="cf">function</span>(module_labels, celltype, gwas, fraction, <span class="at">ontology =</span> <span class="cn">NA</span>) {</span>
<span id="cb436-75"><a href="#cb436-75" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span></span>
<span id="cb436-76"><a href="#cb436-76" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tibble</span>(</span>
<span id="cb436-77"><a href="#cb436-77" aria-hidden="true" tabindex="-1"></a>      <span class="at">module_label =</span> <span class="fu">paste0</span>(module_labels, <span class="st">"-module"</span>, <span class="fu">names</span>(module_labels)),</span>
<span id="cb436-78"><a href="#cb436-78" aria-hidden="true" tabindex="-1"></a>      <span class="at">celltype =</span> celltype,</span>
<span id="cb436-79"><a href="#cb436-79" aria-hidden="true" tabindex="-1"></a>      <span class="at">gwas =</span> gwas,</span>
<span id="cb436-80"><a href="#cb436-80" aria-hidden="true" tabindex="-1"></a>      <span class="at">fraction =</span> fraction,</span>
<span id="cb436-81"><a href="#cb436-81" aria-hidden="true" tabindex="-1"></a>      <span class="at">ontology =</span> ontology</span>
<span id="cb436-82"><a href="#cb436-82" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb436-83"><a href="#cb436-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb436-84"><a href="#cb436-84" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb436-85"><a href="#cb436-85" aria-hidden="true" tabindex="-1"></a>vascular_stroke <span class="ot">&lt;-</span></span>
<span id="cb436-86"><a href="#cb436-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(vascular_stroke,</span>
<span id="cb436-87"><a href="#cb436-87" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(vascular_stroke),</span>
<span id="cb436-88"><a href="#cb436-88" aria-hidden="true" tabindex="-1"></a>       make_ppi_df,</span>
<span id="cb436-89"><a href="#cb436-89" aria-hidden="true" tabindex="-1"></a>       <span class="st">"stroke"</span>,</span>
<span id="cb436-90"><a href="#cb436-90" aria-hidden="true" tabindex="-1"></a>       <span class="st">"controls_vascular"</span>,</span>
<span id="cb436-91"><a href="#cb436-91" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all"</span>) <span class="sc">|&gt;</span></span>
<span id="cb436-92"><a href="#cb436-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
<section id="ad" class="level5">
<h5 class="anchored" data-anchor-id="ad">AD</h5>
<div class="cell-container"><div class="cell-decorator"><pre>In [42]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb437"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb437-1"><a href="#cb437-1" aria-hidden="true" tabindex="-1"></a>ad_all_controls <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb437-2"><a href="#cb437-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Microglia-activated"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Lymphocyte activation"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Dephosphorylation"</span>),</span>
<span id="cb437-3"><a href="#cb437-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pericyte-2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"Supramolecular fibre organisation"</span>),</span>
<span id="cb437-4"><a href="#cb437-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Perivascular-FB-KAZN2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"2"</span> <span class="ot">=</span> <span class="st">"Amyloid"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"Viral process"</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="st">"Axon terminus"</span>)</span>
<span id="cb437-5"><a href="#cb437-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb437-6"><a href="#cb437-6" aria-hidden="true" tabindex="-1"></a>module_labels <span class="ot">&lt;-</span></span>
<span id="cb437-7"><a href="#cb437-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(ad_all_controls,</span>
<span id="cb437-8"><a href="#cb437-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(ad_all_controls),</span>
<span id="cb437-9"><a href="#cb437-9" aria-hidden="true" tabindex="-1"></a>       make_ppi_df,</span>
<span id="cb437-10"><a href="#cb437-10" aria-hidden="true" tabindex="-1"></a>       <span class="st">"ad"</span>,</span>
<span id="cb437-11"><a href="#cb437-11" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all_controls"</span>,</span>
<span id="cb437-12"><a href="#cb437-12" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all"</span>) <span class="sc">|&gt;</span></span>
<span id="cb437-13"><a href="#cb437-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb437-14"><a href="#cb437-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(vascular_stroke)</span>
<span id="cb437-15"><a href="#cb437-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb437-16"><a href="#cb437-16" aria-hidden="true" tabindex="-1"></a>ad_parenchymal <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb437-17"><a href="#cb437-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Microglia-activated"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"synapse"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"gtpase"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"tcell_activation"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"mirna"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"amyloid"</span>),</span>
<span id="cb437-18"><a href="#cb437-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Microglia-quiescent"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"synaptic_transmission"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"immune_response"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"actin_filament"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"golgi"</span>, <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"ion_transport"</span>)</span>
<span id="cb437-19"><a href="#cb437-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb437-20"><a href="#cb437-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb437-21"><a href="#cb437-21" aria-hidden="true" tabindex="-1"></a>module_labels <span class="ot">&lt;-</span></span>
<span id="cb437-22"><a href="#cb437-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(ad_parenchymal,</span>
<span id="cb437-23"><a href="#cb437-23" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(ad_parenchymal),</span>
<span id="cb437-24"><a href="#cb437-24" aria-hidden="true" tabindex="-1"></a>       make_ppi_df,</span>
<span id="cb437-25"><a href="#cb437-25" aria-hidden="true" tabindex="-1"></a>       <span class="st">"ad"</span>,</span>
<span id="cb437-26"><a href="#cb437-26" aria-hidden="true" tabindex="-1"></a>       <span class="st">"controls_parenchymal"</span>,</span>
<span id="cb437-27"><a href="#cb437-27" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all"</span>) <span class="sc">|&gt;</span></span>
<span id="cb437-28"><a href="#cb437-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb437-29"><a href="#cb437-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(module_labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [43]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb438"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb438-1"><a href="#cb438-1" aria-hidden="true" tabindex="-1"></a>ad_all_controls_bp <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb438-2"><a href="#cb438-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Microglia-activated"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"leukocyte_activation"</span>, <span class="st">"2"</span> <span class="ot">=</span> <span class="st">"actin_filament"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"amyloid"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"gtpas"</span>, <span class="st">"6"</span> <span class="ot">=</span> <span class="st">"viral"</span>),</span>
<span id="cb438-3"><a href="#cb438-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Pericyte-2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"amyloid"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"actin_filament"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"circadian"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"viral"</span>),</span>
<span id="cb438-4"><a href="#cb438-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Perivascular-FB-KAZN2"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"notch"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"amyloid"</span>, <span class="st">"5"</span> <span class="ot">=</span> <span class="st">"viral"</span>)</span>
<span id="cb438-5"><a href="#cb438-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb438-6"><a href="#cb438-6" aria-hidden="true" tabindex="-1"></a>module_labels <span class="ot">&lt;-</span></span>
<span id="cb438-7"><a href="#cb438-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(ad_all_controls_bp,</span>
<span id="cb438-8"><a href="#cb438-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(ad_all_controls_bp),</span>
<span id="cb438-9"><a href="#cb438-9" aria-hidden="true" tabindex="-1"></a>       make_ppi_df,</span>
<span id="cb438-10"><a href="#cb438-10" aria-hidden="true" tabindex="-1"></a>       <span class="st">"ad"</span>,</span>
<span id="cb438-11"><a href="#cb438-11" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all_controls"</span>,</span>
<span id="cb438-12"><a href="#cb438-12" aria-hidden="true" tabindex="-1"></a>       <span class="st">"BP"</span>) <span class="sc">|&gt;</span></span>
<span id="cb438-13"><a href="#cb438-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb438-14"><a href="#cb438-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(module_labels)</span>
<span id="cb438-15"><a href="#cb438-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb438-16"><a href="#cb438-16" aria-hidden="true" tabindex="-1"></a>all_magma_sig_celltype_genes_bp <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb438-17"><a href="#cb438-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"All-celltypes"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="st">"leukocyte_activation"</span>, <span class="st">"3"</span> <span class="ot">=</span> <span class="st">"amyloid"</span>, <span class="st">"4"</span> <span class="ot">=</span> <span class="st">"notch_sig-endothelial-migration"</span>)</span>
<span id="cb438-18"><a href="#cb438-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb438-19"><a href="#cb438-19" aria-hidden="true" tabindex="-1"></a>module_labels <span class="ot">&lt;-</span></span>
<span id="cb438-20"><a href="#cb438-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map2</span>(all_magma_sig_celltype_genes_bp,</span>
<span id="cb438-21"><a href="#cb438-21" aria-hidden="true" tabindex="-1"></a>       <span class="fu">names</span>(all_magma_sig_celltype_genes_bp),</span>
<span id="cb438-22"><a href="#cb438-22" aria-hidden="true" tabindex="-1"></a>       make_ppi_df,</span>
<span id="cb438-23"><a href="#cb438-23" aria-hidden="true" tabindex="-1"></a>       <span class="st">"ad"</span>,</span>
<span id="cb438-24"><a href="#cb438-24" aria-hidden="true" tabindex="-1"></a>       <span class="st">"all_controls"</span>,</span>
<span id="cb438-25"><a href="#cb438-25" aria-hidden="true" tabindex="-1"></a>       <span class="st">"BP"</span>) <span class="sc">|&gt;</span></span>
<span id="cb438-26"><a href="#cb438-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list_rbind</span>() <span class="sc">|&gt;</span></span>
<span id="cb438-27"><a href="#cb438-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(module_labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [44]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb439"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb439-1"><a href="#cb439-1" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(</span>
<span id="cb439-2"><a href="#cb439-2" aria-hidden="true" tabindex="-1"></a>  module_labels,</span>
<span id="cb439-3"><a href="#cb439-3" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb439-4"><a href="#cb439-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions"</span>,</span>
<span id="cb439-5"><a href="#cb439-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ppi_magma_sig_genes_module_labels.csv"</span></span>
<span id="cb439-6"><a href="#cb439-6" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb439-7"><a href="#cb439-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [45]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb440"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb440-1"><a href="#cb440-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb440-2"><a href="#cb440-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbols</span></span>
<span id="cb440-3"><a href="#cb440-3" aria-hidden="true" tabindex="-1"></a>modules_sig <span class="ot">&lt;-</span> modules_sig <span class="sc">|&gt;</span></span>
<span id="cb440-4"><a href="#cb440-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ensembl =</span> gene) <span class="sc">|&gt;</span></span>
<span id="cb440-5"><a href="#cb440-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_ensembl_ids, <span class="at">by =</span> <span class="st">"ensembl"</span>)</span>
<span id="cb440-6"><a href="#cb440-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Check there's no NAs</span></span>
<span id="cb440-7"><a href="#cb440-7" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">are_equal</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(modules_sig<span class="sc">$</span>gene)), <span class="dv">0</span>)</span>
<span id="cb440-8"><a href="#cb440-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-9"><a href="#cb440-9" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to desired celltypes</span></span>
<span id="cb440-10"><a href="#cb440-10" aria-hidden="true" tabindex="-1"></a>module_sub <span class="ot">&lt;-</span> modules_sig <span class="sc">|&gt;</span></span>
<span id="cb440-11"><a href="#cb440-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"EUROPEUKBB"</span>, label) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"all_controls_top-ten-percent"</span>, label) <span class="sc">|</span> label <span class="sc">==</span> <span class="st">"sig_magma_celltypes"</span>) <span class="sc">|&gt;</span></span>
<span id="cb440-12"><a href="#cb440-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">gwas =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"EUROPEUKBB|sig_magma_celltypes"</span>, label), <span class="st">"ad"</span>, <span class="cn">NA</span>),</span>
<span id="cb440-13"><a href="#cb440-13" aria-hidden="true" tabindex="-1"></a>                <span class="co"># use str_extract to extract everything until the first "_"</span></span>
<span id="cb440-14"><a href="#cb440-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">celltype =</span> <span class="fu">str_extract</span>(label, <span class="st">"^[^_]*"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb440-15"><a href="#cb440-15" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype =</span> <span class="fu">ifelse</span>(celltype <span class="sc">==</span> <span class="st">"sig"</span>, <span class="st">"All-celltypes"</span>, celltype))</span>
<span id="cb440-16"><a href="#cb440-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-17"><a href="#cb440-17" aria-hidden="true" tabindex="-1"></a><span class="co"># use str_extract to extract characters after "-module" to get module numbers</span></span>
<span id="cb440-18"><a href="#cb440-18" aria-hidden="true" tabindex="-1"></a>module_labels<span class="sc">$</span>module <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(module_labels<span class="sc">$</span>module_label, <span class="st">"(?&lt;=-module).+"</span>) <span class="sc">|&gt;</span></span>
<span id="cb440-19"><a href="#cb440-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>()</span>
<span id="cb440-20"><a href="#cb440-20" aria-hidden="true" tabindex="-1"></a>module_labels_sub <span class="ot">&lt;-</span> module_labels <span class="sc">|&gt;</span></span>
<span id="cb440-21"><a href="#cb440-21" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(ontology <span class="sc">==</span> <span class="st">"BP"</span>)</span>
<span id="cb440-22"><a href="#cb440-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-23"><a href="#cb440-23" aria-hidden="true" tabindex="-1"></a><span class="co"># join module labels</span></span>
<span id="cb440-24"><a href="#cb440-24" aria-hidden="true" tabindex="-1"></a>module_sub <span class="ot">&lt;-</span> module_sub <span class="sc">|&gt;</span></span>
<span id="cb440-25"><a href="#cb440-25" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(module_labels_sub, <span class="at">by =</span> <span class="fu">join_by</span>(module, gwas, celltype))</span>
<span id="cb440-26"><a href="#cb440-26" aria-hidden="true" tabindex="-1"></a><span class="co"># save data</span></span>
<span id="cb440-27"><a href="#cb440-27" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(module_sub, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"03_data/990_processed_data/007_magma/sig_celltype_module_labels_with_genes.csv"</span>))</span>
<span id="cb440-28"><a href="#cb440-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-29"><a href="#cb440-29" aria-hidden="true" tabindex="-1"></a><span class="co"># look at amyloid modules in particular</span></span>
<span id="cb440-30"><a href="#cb440-30" aria-hidden="true" tabindex="-1"></a>amyloid_modules <span class="ot">&lt;-</span> module_sub <span class="sc">|&gt;</span></span>
<span id="cb440-31"><a href="#cb440-31" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"amyloid"</span>, module_label)) <span class="sc">|&gt;</span></span>
<span id="cb440-32"><a href="#cb440-32" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">!=</span> <span class="st">"All-celltypes"</span>)</span>
<span id="cb440-33"><a href="#cb440-33" aria-hidden="true" tabindex="-1"></a>amyloid_modules <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span>label)</span>
<span id="cb440-34"><a href="#cb440-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-35"><a href="#cb440-35" aria-hidden="true" tabindex="-1"></a>df_list <span class="ot">&lt;-</span> <span class="fu">split</span>(amyloid_modules<span class="sc">$</span>gene, amyloid_modules<span class="sc">$</span>celltype)</span>
<span id="cb440-36"><a href="#cb440-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-37"><a href="#cb440-37" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(df_list), <span class="cf">function</span>(x) {</span>
<span id="cb440-38"><a href="#cb440-38" aria-hidden="true" tabindex="-1"></a>  current_genes <span class="ot">&lt;-</span> df_list[[x]]</span>
<span id="cb440-39"><a href="#cb440-39" aria-hidden="true" tabindex="-1"></a>  other_genes <span class="ot">&lt;-</span> <span class="fu">unlist</span>(df_list[<span class="fu">names</span>(df_list) <span class="sc">!=</span> x])</span>
<span id="cb440-40"><a href="#cb440-40" aria-hidden="true" tabindex="-1"></a>  unique_genes <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(current_genes, other_genes)</span>
<span id="cb440-41"><a href="#cb440-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(unique_genes)</span>
<span id="cb440-42"><a href="#cb440-42" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb440-43"><a href="#cb440-43" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(unique_genes) <span class="ot">&lt;-</span> <span class="fu">names</span>(df_list)</span>
<span id="cb440-44"><a href="#cb440-44" aria-hidden="true" tabindex="-1"></a>unique_genes <span class="ot">&lt;-</span> unique_genes <span class="sc">|&gt;</span></span>
<span id="cb440-45"><a href="#cb440-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stack</span>() <span class="sc">|&gt;</span></span>
<span id="cb440-46"><a href="#cb440-46" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene =</span> values, <span class="at">celltype_specificty =</span> ind) <span class="sc">|&gt;</span></span>
<span id="cb440-47"><a href="#cb440-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">shared_or_unique_gene =</span> <span class="st">"unique"</span>)</span>
<span id="cb440-48"><a href="#cb440-48" aria-hidden="true" tabindex="-1"></a>unique_genes</span>
<span id="cb440-49"><a href="#cb440-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-50"><a href="#cb440-50" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nrow</span>(unique_genes) <span class="sc">!=</span> <span class="fu">nrow</span>(amyloid_modules)) {</span>
<span id="cb440-51"><a href="#cb440-51" aria-hidden="true" tabindex="-1"></a>  common_genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb440-52"><a href="#cb440-52" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gene"</span> <span class="ot">=</span> <span class="fu">Reduce</span>(intersect, df_list),</span>
<span id="cb440-53"><a href="#cb440-53" aria-hidden="true" tabindex="-1"></a>    <span class="st">"celltype_specificty"</span> <span class="ot">=</span> <span class="st">"NA"</span>,</span>
<span id="cb440-54"><a href="#cb440-54" aria-hidden="true" tabindex="-1"></a>    <span class="st">"shared_or_unique_gene"</span> <span class="ot">=</span> <span class="st">"shared"</span></span>
<span id="cb440-55"><a href="#cb440-55" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb440-56"><a href="#cb440-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb440-57"><a href="#cb440-57" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb440-58"><a href="#cb440-58" aria-hidden="true" tabindex="-1"></a>  common_genes <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb440-59"><a href="#cb440-59" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb440-60"><a href="#cb440-60" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(unique_genes, common_genes)</span>
<span id="cb440-61"><a href="#cb440-61" aria-hidden="true" tabindex="-1"></a><span class="co">#readr::write_csv(rbind(unique_genes, common_genes), here::here("../../../gmbh_results_summaries/2024-04-52_ad_bbb/sig_celltype_amyloid_genes.csv"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
</section>
</section>
<section id="plot-labelled-modules" class="level2">
<h2 class="anchored" data-anchor-id="plot-labelled-modules">Plot labelled modules</h2>
<p>Itd be nice to have the PPI networks with the module labels attached</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [46]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb441"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb441-1"><a href="#cb441-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb441-2"><a href="#cb441-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot clustering</span></span>
<span id="cb441-3"><a href="#cb441-3" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(subgraphs, module_assignment, <span class="sc">~</span> <span class="fu">plot</span>(.x, </span>
<span id="cb441-4"><a href="#cb441-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.size =</span> <span class="dv">3</span>, </span>
<span id="cb441-5"><a href="#cb441-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.color =</span> <span class="fu">as.character</span>(.y<span class="sc">$</span>module_color), </span>
<span id="cb441-6"><a href="#cb441-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.frame.color =</span> .y<span class="sc">$</span>module_color, </span>
<span id="cb441-7"><a href="#cb441-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">edge.color =</span> <span class="st">"grey"</span>,  </span>
<span id="cb441-8"><a href="#cb441-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">rescale =</span> <span class="cn">TRUE</span>, </span>
<span id="cb441-9"><a href="#cb441-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">vertex.label =</span> <span class="cn">NA</span>,</span>
<span id="cb441-10"><a href="#cb441-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">"Celltype: "</span>, .y<span class="sc">$</span>label[<span class="dv">1</span>])))</span>
<span id="cb441-11"><a href="#cb441-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb441-12"><a href="#cb441-12" aria-hidden="true" tabindex="-1"></a><span class="co"># subset to celltypes of interest</span></span>
<span id="cb441-13"><a href="#cb441-13" aria-hidden="true" tabindex="-1"></a><span class="co">#module_sub &lt;- module_assignment[(grepl("EUROPEUKBB", names(module_assignment)) &amp; grepl("all_controls_top-ten-percent", names(module_assignment)))] </span></span>
<span id="cb441-14"><a href="#cb441-14" aria-hidden="true" tabindex="-1"></a>subgraphs_sub <span class="ot">&lt;-</span> subgraphs[(<span class="fu">grepl</span>(<span class="st">"EUROPEUKBB"</span>, <span class="fu">names</span>(subgraphs)) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"all_controls_top-ten-percent"</span>, <span class="fu">names</span>(subgraphs)))]</span>
<span id="cb441-15"><a href="#cb441-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb441-16"><a href="#cb441-16" aria-hidden="true" tabindex="-1"></a><span class="co"># make module label list for plots</span></span>
<span id="cb441-17"><a href="#cb441-17" aria-hidden="true" tabindex="-1"></a>module_sub_list <span class="ot">&lt;-</span> module_sub <span class="sc">|&gt;</span></span>
<span id="cb441-18"><a href="#cb441-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(module_label)) <span class="sc">|&gt;</span></span>
<span id="cb441-19"><a href="#cb441-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(label) <span class="sc">|&gt;</span></span>
<span id="cb441-20"><a href="#cb441-20" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>()</span>
<span id="cb441-21"><a href="#cb441-21" aria-hidden="true" tabindex="-1"></a>list <span class="ot">&lt;-</span> module_sub_list<span class="sc">$</span>data <span class="sc">|&gt;</span></span>
<span id="cb441-22"><a href="#cb441-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(module_sub_list<span class="sc">$</span>label)</span>
<span id="cb441-23"><a href="#cb441-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb441-24"><a href="#cb441-24" aria-hidden="true" tabindex="-1"></a><span class="fu">map2</span>(subgraphs_sub, list, <span class="sc">~</span> {</span>
<span id="cb441-25"><a href="#cb441-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">png</span>(</span>
<span id="cb441-26"><a href="#cb441-26" aria-hidden="true" tabindex="-1"></a>    here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb441-27"><a href="#cb441-27" aria-hidden="true" tabindex="-1"></a>      <span class="st">"05_figures/990_shared_figures/ppi_modules/network_plots"</span>,</span>
<span id="cb441-28"><a href="#cb441-28" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(.y<span class="sc">$</span>celltype[<span class="dv">1</span>], <span class="st">"_magma_risk_ppi.png"</span>)</span>
<span id="cb441-29"><a href="#cb441-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb441-30"><a href="#cb441-30" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb441-31"><a href="#cb441-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb441-32"><a href="#cb441-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb441-33"><a href="#cb441-33" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">module_label =</span> <span class="fu">unique</span>(.y<span class="sc">$</span>module_label))</span>
<span id="cb441-34"><a href="#cb441-34" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>colours <span class="ot">&lt;-</span> <span class="fu">rainbow</span>(<span class="at">n =</span> <span class="fu">nrow</span>(df))</span>
<span id="cb441-35"><a href="#cb441-35" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(.y, df, <span class="at">by =</span> <span class="st">"module_label"</span>)</span>
<span id="cb441-36"><a href="#cb441-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(</span>
<span id="cb441-37"><a href="#cb441-37" aria-hidden="true" tabindex="-1"></a>    .x,</span>
<span id="cb441-38"><a href="#cb441-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.size =</span> <span class="dv">3</span>,</span>
<span id="cb441-39"><a href="#cb441-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.color =</span> <span class="fu">as.character</span>(df<span class="sc">$</span>colours),</span>
<span id="cb441-40"><a href="#cb441-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.frame.color =</span> df<span class="sc">$</span>colours,</span>
<span id="cb441-41"><a href="#cb441-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">edge.color =</span> <span class="st">'grey'</span>,</span>
<span id="cb441-42"><a href="#cb441-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">rescale =</span> <span class="cn">TRUE</span>,</span>
<span id="cb441-43"><a href="#cb441-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">vertex.label =</span> <span class="cn">NA</span>,</span>
<span id="cb441-44"><a href="#cb441-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">'Celltype: '</span>, df<span class="sc">$</span>celltype[<span class="dv">1</span>])</span>
<span id="cb441-45"><a href="#cb441-45" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb441-46"><a href="#cb441-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assuming you have a vector of unique labels and corresponding colors</span></span>
<span id="cb441-47"><a href="#cb441-47" aria-hidden="true" tabindex="-1"></a>  unique_labels <span class="ot">&lt;-</span> df <span class="sc">|&gt;</span></span>
<span id="cb441-48"><a href="#cb441-48" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(module_label, colours) <span class="sc">|&gt;</span></span>
<span id="cb441-49"><a href="#cb441-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>()</span>
<span id="cb441-50"><a href="#cb441-50" aria-hidden="true" tabindex="-1"></a>  <span class="co">#colors_for_legend &lt;- unique(as.character(.y)) # Replace with actual colors if different</span></span>
<span id="cb441-51"><a href="#cb441-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend</span>(</span>
<span id="cb441-52"><a href="#cb441-52" aria-hidden="true" tabindex="-1"></a>    <span class="st">"topright"</span>,</span>
<span id="cb441-53"><a href="#cb441-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Position of the le</span></span>
<span id="cb441-54"><a href="#cb441-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend =</span> unique_labels<span class="sc">$</span>module_label,</span>
<span id="cb441-55"><a href="#cb441-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Text in the le</span></span>
<span id="cb441-56"><a href="#cb441-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> unique_labels<span class="sc">$</span>colours,</span>
<span id="cb441-57"><a href="#cb441-57" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Colors in the legend</span></span>
<span id="cb441-58"><a href="#cb441-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">19</span>,</span>
<span id="cb441-59"><a href="#cb441-59" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Type of point to use, 19 is a solid circle</span></span>
<span id="cb441-60"><a href="#cb441-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.8</span> <span class="co"># Size of legend text</span></span>
<span id="cb441-61"><a href="#cb441-61" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb441-62"><a href="#cb441-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb441-63"><a href="#cb441-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb441-64"><a href="#cb441-64" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<section id="networkd3-graph" class="level3">
<h3 class="anchored" data-anchor-id="networkd3-graph">networkD3 graph</h3>
<p>Lets try using <code>networkD3</code> to make some nicer visualisations, and be able to see what genes are in the nodes</p>
<div class="cell-container"><div class="cell-decorator"><pre>In [47]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb442"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb442-1"><a href="#cb442-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb442-2"><a href="#cb442-2" aria-hidden="true" tabindex="-1"></a><span class="co"># In case they are not selected from the ppi network check that they are in the ppi network</span></span>
<span id="cb442-3"><a href="#cb442-3" aria-hidden="true" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">intersect</span>(sig_celltypes_magma<span class="sc">$</span>data[[<span class="dv">1</span>]]<span class="sc">$</span>ensembl, <span class="fu">V</span>(my_network)<span class="sc">$</span>name)</span>
<span id="cb442-4"><a href="#cb442-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-5"><a href="#cb442-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-6"><a href="#cb442-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the ppi network to your genes of interest</span></span>
<span id="cb442-7"><a href="#cb442-7" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> my_network, <span class="at">vids =</span> genes_of_interest)</span>
<span id="cb442-8"><a href="#cb442-8" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">as_edgelist</span>(my_network_sub)</span>
<span id="cb442-9"><a href="#cb442-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-10"><a href="#cb442-10" aria-hidden="true" tabindex="-1"></a><span class="co"># You can use Gene IDs instead of Ensembl Gene IDs if you have the corresponding annotations</span></span>
<span id="cb442-11"><a href="#cb442-11" aria-hidden="true" tabindex="-1"></a><span class="co"># for example gene_info containing Gene_ID / Ensembl IDs  to Gene_Name correspondence </span></span>
<span id="cb442-12"><a href="#cb442-12" aria-hidden="true" tabindex="-1"></a><span class="co"># my_network[, 1] &lt;- gene_info$Gene_name[match(x = my_network[, 1], table = gene_info$Gene_ID)]</span></span>
<span id="cb442-13"><a href="#cb442-13" aria-hidden="true" tabindex="-1"></a><span class="co"># my_network[, 2] &lt;- gene_info$Gene_name[match(x = my_network[, 2], table = gene_info$Gene_ID)]</span></span>
<span id="cb442-14"><a href="#cb442-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-15"><a href="#cb442-15" aria-hidden="true" tabindex="-1"></a>unique_genes_df <span class="ot">&lt;-</span> module_sub <span class="sc">|&gt;</span></span>
<span id="cb442-16"><a href="#cb442-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">!=</span> <span class="st">"All-celltypes"</span>) <span class="sc">|&gt;</span></span>
<span id="cb442-17"><a href="#cb442-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ensembl) <span class="sc">|&gt;</span></span>
<span id="cb442-18"><a href="#cb442-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">celltypes_with_gene =</span> <span class="fu">toString</span>(<span class="fu">unique</span>(celltype))) <span class="sc">|&gt;</span></span>
<span id="cb442-19"><a href="#cb442-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb442-20"><a href="#cb442-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-21"><a href="#cb442-21" aria-hidden="true" tabindex="-1"></a>nodes_table <span class="ot">&lt;-</span> module_sub <span class="sc">|&gt;</span></span>
<span id="cb442-22"><a href="#cb442-22" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(celltype <span class="sc">==</span> <span class="st">"All-celltypes"</span>) <span class="sc">|&gt;</span></span>
<span id="cb442-23"><a href="#cb442-23" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">left_join</span>(unique_genes_df, <span class="at">by =</span> <span class="st">"ensembl"</span>) <span class="sc">|&gt;</span></span>
<span id="cb442-24"><a href="#cb442-24" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">module_plus_celltypes =</span> <span class="fu">paste0</span>(module_label, <span class="st">" "</span>, celltypes_with_gene)) <span class="sc">|&gt;</span></span>
<span id="cb442-25"><a href="#cb442-25" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">module_label =</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(module_label), <span class="st">"no_label"</span>, module_label))</span>
<span id="cb442-26"><a href="#cb442-26" aria-hidden="true" tabindex="-1"></a><span class="co">#dplyr::filter(!is.na(module_label))</span></span>
<span id="cb442-27"><a href="#cb442-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-28"><a href="#cb442-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Nodes table </span></span>
<span id="cb442-29"><a href="#cb442-29" aria-hidden="true" tabindex="-1"></a><span class="co">#nodes &lt;- unique(c(my_network_sub[, 1], my_network_sub[, 2]))</span></span>
<span id="cb442-30"><a href="#cb442-30" aria-hidden="true" tabindex="-1"></a><span class="co">#nodes_table &lt;- data.frame(name = nodes,</span></span>
<span id="cb442-31"><a href="#cb442-31" aria-hidden="true" tabindex="-1"></a><span class="co">#                          # Change here for your relevant annotations </span></span>
<span id="cb442-32"><a href="#cb442-32" aria-hidden="true" tabindex="-1"></a><span class="co">#                          group1 = sample(x = c("A", "B", "C"), size = length(nodes), replace = TRUE),</span></span>
<span id="cb442-33"><a href="#cb442-33" aria-hidden="true" tabindex="-1"></a><span class="co">#                          group2 = sample(x = c("typeI", "typeII"), size = length(nodes), replace = TRUE),</span></span>
<span id="cb442-34"><a href="#cb442-34" aria-hidden="true" tabindex="-1"></a><span class="co">#                          size = 1)</span></span>
<span id="cb442-35"><a href="#cb442-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-36"><a href="#cb442-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Links table</span></span>
<span id="cb442-37"><a href="#cb442-37" aria-hidden="true" tabindex="-1"></a>links_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(my_network_sub)</span>
<span id="cb442-38"><a href="#cb442-38" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X1 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X1, nodes_table<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb442-39"><a href="#cb442-39" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X2 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X2, nodes_table<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb442-40"><a href="#cb442-40" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(links_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"source"</span>, <span class="st">"target"</span>)</span>
<span id="cb442-41"><a href="#cb442-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-42"><a href="#cb442-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot by your first set of annotations</span></span>
<span id="cb442-43"><a href="#cb442-43" aria-hidden="true" tabindex="-1"></a>module_plot <span class="ot">&lt;-</span> <span class="fu">forceNetwork</span>(<span class="at">Links =</span> links_table, <span class="at">Nodes =</span> nodes_table, </span>
<span id="cb442-44"><a href="#cb442-44" aria-hidden="true" tabindex="-1"></a>             <span class="co"># In case you want specific colours</span></span>
<span id="cb442-45"><a href="#cb442-45" aria-hidden="true" tabindex="-1"></a>             <span class="at">colourScale =</span> networkD3<span class="sc">::</span><span class="fu">JS</span>(<span class="st">'d3.scaleOrdinal().domain(["leukocyte_activation-module1", "notch_sig-endothelial-migration-module4", "amyloid-module3", "no_label"]).range(["blue", "red", "limegreen", "grey"])'</span>), </span>
<span id="cb442-46"><a href="#cb442-46" aria-hidden="true" tabindex="-1"></a>             <span class="at">linkColour =</span> <span class="fu">alpha</span>(<span class="st">"grey"</span>, .<span class="dv">5</span>), </span>
<span id="cb442-47"><a href="#cb442-47" aria-hidden="true" tabindex="-1"></a>             <span class="at">charge =</span> <span class="sc">-</span><span class="dv">20</span>, </span>
<span id="cb442-48"><a href="#cb442-48" aria-hidden="true" tabindex="-1"></a>             <span class="at">Source =</span> <span class="st">"source"</span>, </span>
<span id="cb442-49"><a href="#cb442-49" aria-hidden="true" tabindex="-1"></a>             <span class="at">Target =</span> <span class="st">"target"</span>,</span>
<span id="cb442-50"><a href="#cb442-50" aria-hidden="true" tabindex="-1"></a>             <span class="at">NodeID =</span> <span class="st">"gene"</span>,</span>
<span id="cb442-51"><a href="#cb442-51" aria-hidden="true" tabindex="-1"></a>             <span class="at">Group =</span> <span class="st">"module_label"</span>, </span>
<span id="cb442-52"><a href="#cb442-52" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacity =</span> <span class="fl">0.8</span>, </span>
<span id="cb442-53"><a href="#cb442-53" aria-hidden="true" tabindex="-1"></a>             <span class="at">bounded =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-54"><a href="#cb442-54" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-55"><a href="#cb442-55" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacityNoHover =</span> .<span class="dv">5</span>)<span class="co">#,</span></span>
<span id="cb442-56"><a href="#cb442-56" aria-hidden="true" tabindex="-1"></a>             <span class="co">#Nodesize = "celltypes_with_gene")</span></span>
<span id="cb442-57"><a href="#cb442-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-58"><a href="#cb442-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot by your second annotation</span></span>
<span id="cb442-59"><a href="#cb442-59" aria-hidden="true" tabindex="-1"></a>celltypes_plots <span class="ot">&lt;-</span> <span class="fu">forceNetwork</span>(<span class="at">Links =</span> links_table, <span class="at">Nodes =</span> nodes_table, </span>
<span id="cb442-60"><a href="#cb442-60" aria-hidden="true" tabindex="-1"></a>             <span class="co"># In case you want specific colours</span></span>
<span id="cb442-61"><a href="#cb442-61" aria-hidden="true" tabindex="-1"></a>             <span class="at">colourScale =</span> networkD3<span class="sc">::</span><span class="fu">JS</span>(<span class="st">'d3.scaleOrdinal().domain(["typeI", "typeII"]).range(["deeppink", "steelblue"])'</span>), </span>
<span id="cb442-62"><a href="#cb442-62" aria-hidden="true" tabindex="-1"></a>             <span class="at">linkColour =</span> <span class="fu">alpha</span>(<span class="st">"grey"</span>, .<span class="dv">5</span>),</span>
<span id="cb442-63"><a href="#cb442-63" aria-hidden="true" tabindex="-1"></a>             <span class="at">charge =</span> <span class="sc">-</span><span class="dv">20</span>,</span>
<span id="cb442-64"><a href="#cb442-64" aria-hidden="true" tabindex="-1"></a>             <span class="at">Source =</span> <span class="st">"source"</span>, </span>
<span id="cb442-65"><a href="#cb442-65" aria-hidden="true" tabindex="-1"></a>             <span class="at">Target =</span> <span class="st">"target"</span>,</span>
<span id="cb442-66"><a href="#cb442-66" aria-hidden="true" tabindex="-1"></a>             <span class="at">NodeID =</span> <span class="st">"gene"</span>, </span>
<span id="cb442-67"><a href="#cb442-67" aria-hidden="true" tabindex="-1"></a>             <span class="at">Group =</span> <span class="st">"celltypes_with_gene"</span>, </span>
<span id="cb442-68"><a href="#cb442-68" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacity =</span> <span class="fl">0.8</span>, </span>
<span id="cb442-69"><a href="#cb442-69" aria-hidden="true" tabindex="-1"></a>             <span class="at">bounded =</span> <span class="cn">TRUE</span>, </span>
<span id="cb442-70"><a href="#cb442-70" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-71"><a href="#cb442-71" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacityNoHover =</span> .<span class="dv">5</span>) </span>
<span id="cb442-72"><a href="#cb442-72" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> manipulateWidget<span class="sc">::</span><span class="fu">combineWidgets</span>(module_plot, celltypes_plots, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb442-73"><a href="#cb442-73" aria-hidden="true" tabindex="-1"></a>combined_plot</span>
<span id="cb442-74"><a href="#cb442-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-75"><a href="#cb442-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Save both plots to an HTML file</span></span>
<span id="cb442-76"><a href="#cb442-76" aria-hidden="true" tabindex="-1"></a>htmlwidgets<span class="sc">::</span><span class="fu">saveWidget</span>(combined_plot, here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/ppi_modules/network_plots/d3_plots.html"</span>))</span>
<span id="cb442-77"><a href="#cb442-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-78"><a href="#cb442-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Use webshot to take a screenshot of the HTML file and save it as an image</span></span>
<span id="cb442-79"><a href="#cb442-79" aria-hidden="true" tabindex="-1"></a>webshot<span class="sc">::</span><span class="fu">webshot</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/ppi_modules/network_plots/d3_plots.html"</span>), <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"05_figures/990_shared_figures/ppi_modules/network_plots/d3_plots.png"</span>))</span>
<span id="cb442-80"><a href="#cb442-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-81"><a href="#cb442-81" aria-hidden="true" tabindex="-1"></a>nodes_table_sub <span class="ot">&lt;-</span> nodes_table <span class="sc">|&gt;</span></span>
<span id="cb442-82"><a href="#cb442-82" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(celltypes_with_gene <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Pericyte-2"</span>, <span class="st">"Microglia-activated"</span>, <span class="st">"Perivascular-FB-KAZN2"</span>))</span>
<span id="cb442-83"><a href="#cb442-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-84"><a href="#cb442-84" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> my_network, <span class="at">vids =</span> <span class="fu">unique</span>(nodes_table_sub<span class="sc">$</span>ensembl))</span>
<span id="cb442-85"><a href="#cb442-85" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">as_edgelist</span>(my_network_sub)</span>
<span id="cb442-86"><a href="#cb442-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-87"><a href="#cb442-87" aria-hidden="true" tabindex="-1"></a><span class="co"># Links table</span></span>
<span id="cb442-88"><a href="#cb442-88" aria-hidden="true" tabindex="-1"></a>links_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(my_network_sub)</span>
<span id="cb442-89"><a href="#cb442-89" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X1 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X1, nodes_table_sub<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb442-90"><a href="#cb442-90" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X2 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X2, nodes_table_sub<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb442-91"><a href="#cb442-91" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(links_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"source"</span>, <span class="st">"target"</span>)</span>
<span id="cb442-92"><a href="#cb442-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-93"><a href="#cb442-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot by your first set of annotations</span></span>
<span id="cb442-94"><a href="#cb442-94" aria-hidden="true" tabindex="-1"></a>module_plot <span class="ot">&lt;-</span> <span class="fu">forceNetwork</span>(<span class="at">Links =</span> links_table, <span class="at">Nodes =</span> nodes_table_sub, </span>
<span id="cb442-95"><a href="#cb442-95" aria-hidden="true" tabindex="-1"></a>             <span class="co"># In case you want specific colours</span></span>
<span id="cb442-96"><a href="#cb442-96" aria-hidden="true" tabindex="-1"></a>             <span class="at">colourScale =</span> networkD3<span class="sc">::</span><span class="fu">JS</span>(<span class="st">'d3.scaleOrdinal().domain(["leukocyte_activation-module1", "notch_sig-endothelial-migration-module4", "amyloid-module3", "no_label"]).range(["blue", "red", "limegreen", "grey"])'</span>), </span>
<span id="cb442-97"><a href="#cb442-97" aria-hidden="true" tabindex="-1"></a>             <span class="at">linkColour =</span> <span class="fu">alpha</span>(<span class="st">"grey"</span>, .<span class="dv">5</span>), </span>
<span id="cb442-98"><a href="#cb442-98" aria-hidden="true" tabindex="-1"></a>             <span class="at">charge =</span> <span class="sc">-</span><span class="dv">20</span>, </span>
<span id="cb442-99"><a href="#cb442-99" aria-hidden="true" tabindex="-1"></a>             <span class="at">Source =</span> <span class="st">"source"</span>, </span>
<span id="cb442-100"><a href="#cb442-100" aria-hidden="true" tabindex="-1"></a>             <span class="at">Target =</span> <span class="st">"target"</span>,</span>
<span id="cb442-101"><a href="#cb442-101" aria-hidden="true" tabindex="-1"></a>             <span class="at">NodeID =</span> <span class="st">"gene"</span>,</span>
<span id="cb442-102"><a href="#cb442-102" aria-hidden="true" tabindex="-1"></a>             <span class="at">Group =</span> <span class="st">"module_label"</span>, </span>
<span id="cb442-103"><a href="#cb442-103" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacity =</span> <span class="fl">0.8</span>, </span>
<span id="cb442-104"><a href="#cb442-104" aria-hidden="true" tabindex="-1"></a>             <span class="at">bounded =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-105"><a href="#cb442-105" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-106"><a href="#cb442-106" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacityNoHover =</span> .<span class="dv">5</span>)<span class="co">#,</span></span>
<span id="cb442-107"><a href="#cb442-107" aria-hidden="true" tabindex="-1"></a>             <span class="co">#Nodesize = "celltypes_with_gene")</span></span>
<span id="cb442-108"><a href="#cb442-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-109"><a href="#cb442-109" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot by your second annotation</span></span>
<span id="cb442-110"><a href="#cb442-110" aria-hidden="true" tabindex="-1"></a>celltypes_plots <span class="ot">&lt;-</span> <span class="fu">forceNetwork</span>(<span class="at">Links =</span> links_table, <span class="at">Nodes =</span> nodes_table_sub, </span>
<span id="cb442-111"><a href="#cb442-111" aria-hidden="true" tabindex="-1"></a>             <span class="co"># In case you want specific colours</span></span>
<span id="cb442-112"><a href="#cb442-112" aria-hidden="true" tabindex="-1"></a>             <span class="at">colourScale =</span> networkD3<span class="sc">::</span><span class="fu">JS</span>(<span class="st">'d3.scaleOrdinal().domain(["Microglia-activated", "Pericyte-2", "Perivascular-FB-KAZN2"]).range(["deeppink", "steelblue", "yellow"])'</span>), </span>
<span id="cb442-113"><a href="#cb442-113" aria-hidden="true" tabindex="-1"></a>             <span class="at">linkColour =</span> <span class="fu">alpha</span>(<span class="st">"grey"</span>, .<span class="dv">5</span>),</span>
<span id="cb442-114"><a href="#cb442-114" aria-hidden="true" tabindex="-1"></a>             <span class="at">charge =</span> <span class="sc">-</span><span class="dv">20</span>,</span>
<span id="cb442-115"><a href="#cb442-115" aria-hidden="true" tabindex="-1"></a>             <span class="at">Source =</span> <span class="st">"source"</span>, </span>
<span id="cb442-116"><a href="#cb442-116" aria-hidden="true" tabindex="-1"></a>             <span class="at">Target =</span> <span class="st">"target"</span>,</span>
<span id="cb442-117"><a href="#cb442-117" aria-hidden="true" tabindex="-1"></a>             <span class="at">NodeID =</span> <span class="st">"gene"</span>, </span>
<span id="cb442-118"><a href="#cb442-118" aria-hidden="true" tabindex="-1"></a>             <span class="at">Group =</span> <span class="st">"celltypes_with_gene"</span>, </span>
<span id="cb442-119"><a href="#cb442-119" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacity =</span> <span class="fl">0.8</span>, </span>
<span id="cb442-120"><a href="#cb442-120" aria-hidden="true" tabindex="-1"></a>             <span class="at">bounded =</span> <span class="cn">TRUE</span>, </span>
<span id="cb442-121"><a href="#cb442-121" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-122"><a href="#cb442-122" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacityNoHover =</span> .<span class="dv">5</span>) </span>
<span id="cb442-123"><a href="#cb442-123" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> manipulateWidget<span class="sc">::</span><span class="fu">combineWidgets</span>(module_plot, celltypes_plots, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb442-124"><a href="#cb442-124" aria-hidden="true" tabindex="-1"></a>combined_plot</span>
<span id="cb442-125"><a href="#cb442-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-126"><a href="#cb442-126" aria-hidden="true" tabindex="-1"></a>nodes_table_sub <span class="ot">&lt;-</span> nodes_table <span class="sc">|&gt;</span></span>
<span id="cb442-127"><a href="#cb442-127" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(celltypes_with_gene <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Pericyte-2"</span>, <span class="st">"Microglia-activated"</span>, <span class="st">"Perivascular-FB-KAZN2"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb442-128"><a href="#cb442-128" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(module_label <span class="sc">==</span> <span class="st">"amyloid-module3"</span>)</span>
<span id="cb442-129"><a href="#cb442-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-130"><a href="#cb442-130" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> my_network, <span class="at">vids =</span> <span class="fu">unique</span>(nodes_table_sub<span class="sc">$</span>ensembl))</span>
<span id="cb442-131"><a href="#cb442-131" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">as_edgelist</span>(my_network_sub)</span>
<span id="cb442-132"><a href="#cb442-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-133"><a href="#cb442-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Links table</span></span>
<span id="cb442-134"><a href="#cb442-134" aria-hidden="true" tabindex="-1"></a>links_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(my_network_sub)</span>
<span id="cb442-135"><a href="#cb442-135" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X1 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X1, nodes_table_sub<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb442-136"><a href="#cb442-136" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X2 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X2, nodes_table_sub<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb442-137"><a href="#cb442-137" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(links_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"source"</span>, <span class="st">"target"</span>)</span>
<span id="cb442-138"><a href="#cb442-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-139"><a href="#cb442-139" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot by your second annotation</span></span>
<span id="cb442-140"><a href="#cb442-140" aria-hidden="true" tabindex="-1"></a>celltypes_plots <span class="ot">&lt;-</span> <span class="fu">forceNetwork</span>(<span class="at">Links =</span> links_table, <span class="at">Nodes =</span> nodes_table_sub, </span>
<span id="cb442-141"><a href="#cb442-141" aria-hidden="true" tabindex="-1"></a>             <span class="co"># In case you want specific colours</span></span>
<span id="cb442-142"><a href="#cb442-142" aria-hidden="true" tabindex="-1"></a>             <span class="at">colourScale =</span> networkD3<span class="sc">::</span><span class="fu">JS</span>(<span class="st">'d3.scaleOrdinal().domain(["Microglia-activated", "Pericyte-2", "Perivascular-FB-KAZN2"]).range(["deeppink", "steelblue", "yellow"])'</span>), </span>
<span id="cb442-143"><a href="#cb442-143" aria-hidden="true" tabindex="-1"></a>             <span class="at">linkColour =</span> <span class="fu">alpha</span>(<span class="st">"grey"</span>, .<span class="dv">5</span>),</span>
<span id="cb442-144"><a href="#cb442-144" aria-hidden="true" tabindex="-1"></a>             <span class="at">charge =</span> <span class="sc">-</span><span class="dv">20</span>,</span>
<span id="cb442-145"><a href="#cb442-145" aria-hidden="true" tabindex="-1"></a>             <span class="at">Source =</span> <span class="st">"source"</span>, </span>
<span id="cb442-146"><a href="#cb442-146" aria-hidden="true" tabindex="-1"></a>             <span class="at">Target =</span> <span class="st">"target"</span>,</span>
<span id="cb442-147"><a href="#cb442-147" aria-hidden="true" tabindex="-1"></a>             <span class="at">NodeID =</span> <span class="st">"gene"</span>, </span>
<span id="cb442-148"><a href="#cb442-148" aria-hidden="true" tabindex="-1"></a>             <span class="at">Group =</span> <span class="st">"celltypes_with_gene"</span>, </span>
<span id="cb442-149"><a href="#cb442-149" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacity =</span> <span class="fl">0.8</span>, </span>
<span id="cb442-150"><a href="#cb442-150" aria-hidden="true" tabindex="-1"></a>             <span class="at">bounded =</span> <span class="cn">TRUE</span>, </span>
<span id="cb442-151"><a href="#cb442-151" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb442-152"><a href="#cb442-152" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacityNoHover =</span> .<span class="dv">5</span>) </span>
<span id="cb442-153"><a href="#cb442-153" aria-hidden="true" tabindex="-1"></a>celltypes_plots</span>
<span id="cb442-154"><a href="#cb442-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb442-155"><a href="#cb442-155" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> manipulateWidget<span class="sc">::</span><span class="fu">combineWidgets</span>(module_plot, celltypes_plots, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb442-156"><a href="#cb442-156" aria-hidden="true" tabindex="-1"></a>combined_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<div class="cell-container"><div class="cell-decorator"><pre>In [48]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb443"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb443-1"><a href="#cb443-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb443-2"><a href="#cb443-2" aria-hidden="true" tabindex="-1"></a>unique_genes_to_plot <span class="ot">&lt;-</span> <span class="fu">rbind</span>(unique_genes, common_genes) <span class="sc">|&gt;</span></span>
<span id="cb443-3"><a href="#cb443-3" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_ensembl_ids, <span class="at">by =</span> <span class="st">"gene"</span>) <span class="sc">|&gt;</span></span>
<span id="cb443-4"><a href="#cb443-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype_specificty =</span> <span class="fu">as.character</span>(celltype_specificty)) <span class="sc">|&gt;</span></span>
<span id="cb443-5"><a href="#cb443-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">celltype_specificty =</span> <span class="fu">ifelse</span>(celltype_specificty <span class="sc">==</span> <span class="st">"NA"</span>, <span class="st">"Shared-genes"</span>, celltype_specificty))</span>
<span id="cb443-6"><a href="#cb443-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-7"><a href="#cb443-7" aria-hidden="true" tabindex="-1"></a><span class="co"># In case they are not selected from the ppi network check that they are in the ppi network</span></span>
<span id="cb443-8"><a href="#cb443-8" aria-hidden="true" tabindex="-1"></a>genes_of_interest <span class="ot">&lt;-</span> <span class="fu">intersect</span>(unique_genes_to_plot<span class="sc">$</span>ensembl, <span class="fu">V</span>(my_network)<span class="sc">$</span>name)</span>
<span id="cb443-9"><a href="#cb443-9" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">are_equal</span>(<span class="fu">length</span>(unique_genes_to_plot<span class="sc">$</span>gene), <span class="fu">length</span>(genes_of_interest))</span>
<span id="cb443-10"><a href="#cb443-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-11"><a href="#cb443-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-12"><a href="#cb443-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-13"><a href="#cb443-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the ppi network to your genes of interest</span></span>
<span id="cb443-14"><a href="#cb443-14" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">induced_subgraph</span>(<span class="at">graph =</span> my_network, <span class="at">vids =</span> genes_of_interest)</span>
<span id="cb443-15"><a href="#cb443-15" aria-hidden="true" tabindex="-1"></a>my_network_sub <span class="ot">&lt;-</span> <span class="fu">as_edgelist</span>(my_network_sub)</span>
<span id="cb443-16"><a href="#cb443-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-17"><a href="#cb443-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Links table</span></span>
<span id="cb443-18"><a href="#cb443-18" aria-hidden="true" tabindex="-1"></a>links_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(my_network_sub)</span>
<span id="cb443-19"><a href="#cb443-19" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X1 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X1, unique_genes_to_plot<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb443-20"><a href="#cb443-20" aria-hidden="true" tabindex="-1"></a>links_table<span class="sc">$</span>X2 <span class="ot">&lt;-</span> <span class="fu">match</span>(links_table<span class="sc">$</span>X2, unique_genes_to_plot<span class="sc">$</span>ensembl) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb443-21"><a href="#cb443-21" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(links_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"source"</span>, <span class="st">"target"</span>)</span>
<span id="cb443-22"><a href="#cb443-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-23"><a href="#cb443-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot by your first set of annotations</span></span>
<span id="cb443-24"><a href="#cb443-24" aria-hidden="true" tabindex="-1"></a>module_plot <span class="ot">&lt;-</span> <span class="fu">forceNetwork</span>(<span class="at">Links =</span> links_table, <span class="at">Nodes =</span> unique_genes_to_plot, </span>
<span id="cb443-25"><a href="#cb443-25" aria-hidden="true" tabindex="-1"></a>             <span class="co"># In case you want specific colours</span></span>
<span id="cb443-26"><a href="#cb443-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">colourScale =</span> networkD3<span class="sc">::</span><span class="fu">JS</span>(<span class="st">'d3.scaleOrdinal().domain(["Microglia-activated", "Pericyte-2", "Perivascular-FB-KAZN2", "Shared-genes"]).range(["deeppink", "steelblue", "red", "limegreen"])'</span>), </span>
<span id="cb443-27"><a href="#cb443-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">linkColour =</span> <span class="fu">alpha</span>(<span class="st">"grey"</span>, .<span class="dv">5</span>), </span>
<span id="cb443-28"><a href="#cb443-28" aria-hidden="true" tabindex="-1"></a>             <span class="at">charge =</span> <span class="sc">-</span><span class="dv">150</span>, </span>
<span id="cb443-29"><a href="#cb443-29" aria-hidden="true" tabindex="-1"></a>             <span class="at">Source =</span> <span class="st">"source"</span>, </span>
<span id="cb443-30"><a href="#cb443-30" aria-hidden="true" tabindex="-1"></a>             <span class="at">Target =</span> <span class="st">"target"</span>,</span>
<span id="cb443-31"><a href="#cb443-31" aria-hidden="true" tabindex="-1"></a>             <span class="at">NodeID =</span> <span class="st">"gene"</span>,</span>
<span id="cb443-32"><a href="#cb443-32" aria-hidden="true" tabindex="-1"></a>             <span class="at">Group =</span> <span class="st">"celltype_specificty"</span>, </span>
<span id="cb443-33"><a href="#cb443-33" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacity =</span> <span class="dv">1</span>, </span>
<span id="cb443-34"><a href="#cb443-34" aria-hidden="true" tabindex="-1"></a>             <span class="at">bounded =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-35"><a href="#cb443-35" aria-hidden="true" tabindex="-1"></a>             <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-36"><a href="#cb443-36" aria-hidden="true" tabindex="-1"></a>             <span class="at">zoom =</span> <span class="cn">TRUE</span>,</span>
<span id="cb443-37"><a href="#cb443-37" aria-hidden="true" tabindex="-1"></a>             <span class="at">height =</span> <span class="dv">400</span>,</span>
<span id="cb443-38"><a href="#cb443-38" aria-hidden="true" tabindex="-1"></a>             <span class="at">width =</span> <span class="dv">500</span>,</span>
<span id="cb443-39"><a href="#cb443-39" aria-hidden="true" tabindex="-1"></a>             <span class="at">opacityNoHover =</span> .<span class="dv">5</span>)<span class="co">#,</span></span>
<span id="cb443-40"><a href="#cb443-40" aria-hidden="true" tabindex="-1"></a>             <span class="co">#Nodesize = "celltypes_with_gene")</span></span>
<span id="cb443-41"><a href="#cb443-41" aria-hidden="true" tabindex="-1"></a>module_plot</span>
<span id="cb443-42"><a href="#cb443-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-43"><a href="#cb443-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-44"><a href="#cb443-44" aria-hidden="true" tabindex="-1"></a><span class="co"># doesn't work due to phanotomJS error, likely due to incompatible version of openssl</span></span>
<span id="cb443-45"><a href="#cb443-45" aria-hidden="true" tabindex="-1"></a><span class="co"># module_plot &lt;- forceNetwork(Links = links_table, Nodes = unique_genes_to_plot, </span></span>
<span id="cb443-46"><a href="#cb443-46" aria-hidden="true" tabindex="-1"></a><span class="co">#              # In case you want specific colours</span></span>
<span id="cb443-47"><a href="#cb443-47" aria-hidden="true" tabindex="-1"></a><span class="co">#              colourScale = networkD3::JS('d3.scaleOrdinal().domain(["Microglia-activated", "Pericyte-2", "Perivascular-FB-KAZN2", "Shared-genes"]).range(["deeppink", "steelblue", "red", "limegreen"])'), </span></span>
<span id="cb443-48"><a href="#cb443-48" aria-hidden="true" tabindex="-1"></a><span class="co">#              linkColour = alpha("grey", .5), </span></span>
<span id="cb443-49"><a href="#cb443-49" aria-hidden="true" tabindex="-1"></a><span class="co">#              charge = -150, </span></span>
<span id="cb443-50"><a href="#cb443-50" aria-hidden="true" tabindex="-1"></a><span class="co">#              Source = "source", </span></span>
<span id="cb443-51"><a href="#cb443-51" aria-hidden="true" tabindex="-1"></a><span class="co">#              Target = "target",</span></span>
<span id="cb443-52"><a href="#cb443-52" aria-hidden="true" tabindex="-1"></a><span class="co">#              NodeID = "gene",</span></span>
<span id="cb443-53"><a href="#cb443-53" aria-hidden="true" tabindex="-1"></a><span class="co">#              Group = "celltype_specificty", </span></span>
<span id="cb443-54"><a href="#cb443-54" aria-hidden="true" tabindex="-1"></a><span class="co">#              opacity = 1, </span></span>
<span id="cb443-55"><a href="#cb443-55" aria-hidden="true" tabindex="-1"></a><span class="co">#              bounded = TRUE,</span></span>
<span id="cb443-56"><a href="#cb443-56" aria-hidden="true" tabindex="-1"></a><span class="co">#              legend = TRUE,</span></span>
<span id="cb443-57"><a href="#cb443-57" aria-hidden="true" tabindex="-1"></a><span class="co">#              zoom = TRUE,</span></span>
<span id="cb443-58"><a href="#cb443-58" aria-hidden="true" tabindex="-1"></a><span class="co">#              opacityNoHover = .5)#,</span></span>
<span id="cb443-59"><a href="#cb443-59" aria-hidden="true" tabindex="-1"></a><span class="co">#              #Nodesize = "celltypes_with_gene")</span></span>
<span id="cb443-60"><a href="#cb443-60" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb443-61"><a href="#cb443-61" aria-hidden="true" tabindex="-1"></a><span class="co"># htmlwidgets::saveWidget(module_plot, here::here("test_network.html"))</span></span>
<span id="cb443-62"><a href="#cb443-62" aria-hidden="true" tabindex="-1"></a><span class="co"># webshot::webshot(here::here("test_network.html"), file = "test_network.png", vwidth = 500, vheight = 400, zoom = 3)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
<!-- Several of the modules have just one gene in them and I don't trust any enrichment for GO terms there, so I'll replace the module label with the gene label -->
<div class="cell-container"><div class="cell-decorator"><pre>In [49]:</pre></div><div class="cell">
<div class="sourceCode cell-code" id="cb444"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb444-1"><a href="#cb444-1" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb444-2"><a href="#cb444-2" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb444-3"><a href="#cb444-3" aria-hidden="true" tabindex="-1"></a><span class="co"># I've changed to just excluding these now</span></span>
<span id="cb444-4"><a href="#cb444-4" aria-hidden="true" tabindex="-1"></a>gene_num <span class="ot">&lt;-</span> <span class="fu">map_dbl</span>(result_list_subset, <span class="sc">~</span> <span class="fu">length</span>(.x<span class="sc">@</span>gene))</span>
<span id="cb444-5"><a href="#cb444-5" aria-hidden="true" tabindex="-1"></a>one_gene <span class="ot">&lt;-</span> <span class="fu">names</span>(gene_num)[gene_num <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb444-6"><a href="#cb444-6" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">map_chr</span>(one_gene, <span class="sc">~</span> result_list[[.x]]<span class="sc">@</span>gene)</span>
<span id="cb444-7"><a href="#cb444-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(genes) <span class="ot">&lt;-</span> one_gene</span>
<span id="cb444-8"><a href="#cb444-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb444-9"><a href="#cb444-9" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb444-10"><a href="#cb444-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve gene symbols</span></span>
<span id="cb444-11"><a href="#cb444-11" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'entrezgene_id'</span>, <span class="st">'hgnc_symbol'</span>),</span>
<span id="cb444-12"><a href="#cb444-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">filters =</span> <span class="st">'entrezgene_id'</span>,</span>
<span id="cb444-13"><a href="#cb444-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">values =</span> genes,</span>
<span id="cb444-14"><a href="#cb444-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mart =</span> ensembl)</span>
<span id="cb444-15"><a href="#cb444-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb444-16"><a href="#cb444-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the results</span></span>
<span id="cb444-17"><a href="#cb444-17" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(gene_info)</span>
<span id="cb444-18"><a href="#cb444-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure there's no NAs in the conversion</span></span>
<span id="cb444-19"><a href="#cb444-19" aria-hidden="true" tabindex="-1"></a>assertthat<span class="sc">::</span><span class="fu">noNA</span>(gene_info)</span>
<span id="cb444-20"><a href="#cb444-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb444-21"><a href="#cb444-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure genes are arranged in the same way</span></span>
<span id="cb444-22"><a href="#cb444-22" aria-hidden="true" tabindex="-1"></a>module_order <span class="ot">&lt;-</span> <span class="fu">names</span>(genes)[<span class="fu">match</span>(gene_info<span class="sc">$</span>entrezgene_id, genes)]</span>
<span id="cb444-23"><a href="#cb444-23" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> gene_info<span class="sc">$</span>hgnc_symbol</span>
<span id="cb444-24"><a href="#cb444-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gene_symbols) <span class="ot">&lt;-</span> module_order</span>
<span id="cb444-25"><a href="#cb444-25" aria-hidden="true" tabindex="-1"></a><span class="co"># There are some modules not in the module assignment as they didn't get trees </span></span>
<span id="cb444-26"><a href="#cb444-26" aria-hidden="true" tabindex="-1"></a><span class="co"># for being to small. I'll exclude these modules and add them to dataframe</span></span>
<span id="cb444-27"><a href="#cb444-27" aria-hidden="true" tabindex="-1"></a>extra_modules <span class="ot">&lt;-</span> gene_symbols[<span class="sc">!</span><span class="fu">names</span>(gene_symbols) <span class="sc">%in%</span> module_assignments<span class="sc">$</span>module]</span>
<span id="cb444-28"><a href="#cb444-28" aria-hidden="true" tabindex="-1"></a>extra_modules <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">module =</span> <span class="fu">names</span>(extra_modules), <span class="at">label =</span> extra_modules)</span>
<span id="cb444-29"><a href="#cb444-29" aria-hidden="true" tabindex="-1"></a>module_assignments <span class="ot">&lt;-</span> <span class="fu">rbind</span>(module_assignments, extra_modules)</span>
<span id="cb444-30"><a href="#cb444-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Update labels in the dataframe</span></span>
<span id="cb444-31"><a href="#cb444-31" aria-hidden="true" tabindex="-1"></a>module_assignments<span class="sc">$</span>label[<span class="fu">match</span>(<span class="fu">names</span>(gene_symbols), module_assignments<span class="sc">$</span>module)] <span class="ot">&lt;-</span> gene_symbols</span>
<span id="cb444-32"><a href="#cb444-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Save module assignment</span></span>
<span id="cb444-33"><a href="#cb444-33" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(</span>
<span id="cb444-34"><a href="#cb444-34" aria-hidden="true" tabindex="-1"></a>  module_assignments,</span>
<span id="cb444-35"><a href="#cb444-35" aria-hidden="true" tabindex="-1"></a>  here<span class="sc">::</span><span class="fu">here</span>(</span>
<span id="cb444-36"><a href="#cb444-36" aria-hidden="true" tabindex="-1"></a>    <span class="st">"03_data/990_processed_data/001_snrnaseq/12_protein_protein_interactions"</span>,</span>
<span id="cb444-37"><a href="#cb444-37" aria-hidden="true" tabindex="-1"></a>    <span class="st">"ppi_sig_module_labels.csv"</span></span>
<span id="cb444-38"><a href="#cb444-38" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb444-39"><a href="#cb444-39" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div>
</section>
</section>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"selector":".lightbox","descPosition":"bottom","openEffect":"zoom","loop":false,"closeEffect":"zoom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>