---
title: Generate cellranger reference for set 1
execute: 
  eval: false
---

Set 1 contains the first 12 donors, 24 samples (12x P - parenchymal and 12x V - vascular fraction), which are independently sequenced of Set_2 (the remaining 16 samples from 8 donors).
The sequencing was performed for Set_1 initially in Oxford, but there were some issues with index hopping and cleaning the data did not seem to work too well.
So Set_1 was re-sequenced in Cardiff and the folder is: 211008_A00748_0157_AHT7TJDSX2_fastq_L2_3_4

For Set_2 the data was sequenced initially on one Illumina lane, but then more lanes were added for better coverage.
So the FASTQ files across the 4 subdirs need to be combined as input for one sample.
I think there were a couple of corner cases for those samples, one sample was of lower quality and not sequenced again, and one extra run of another sample basically had just one read.
I think the script to run CellRanger for Set_2 should mention this and deal with it.

Due to this funk with the samples I'll keep them are seperate scripts for all three sets.

```{r}
#!/bin/bash

#SBATCH -p c_vhighmem_dri1 ## dev, compute, htc, highmem
#SBATCH --job-name=cellranger_count_set1
#SBATCH --ntasks=40
#SBATCH --ntasks-per-node=40
#SBATCH --array=1-24%2
##### #SBATCH --mem-per-cpu=8000 # memory limit per core
#SBATCH --mem=740G # memory limit per compute node for the job
#SBATCH --time=3-00:00 # maximum job time in D-HH:MM
#SBATCH --account=scw1329
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --mail-user Bernardo-HarringtonG@cardiff.ac.uk # email on fail
#SBATCH --mail-type END,FAIL

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "- SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "- SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "- SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "- SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "This job in the array has:"
echo "- SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "- SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Run on host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"

#-----------------------------------------------------------------------

## FASTQ files, set 1, 24 samples, sequenced in Cardiff
## NOTE, R2 length is 150bp, complete length will be used by default
## alternatively, reads could be trimmed  with Cell Ranger option --r2-length=90
INPUT_DIR="/gluster/dri02/rdscw/shared/webber/Endo_10X/FASTQ/Set_1/211008_A00748_0157_AHT7TJDSX2_fastq_L2_3_4/"

## results
OUTPUT_DIR="/scratch/c.mpmgb/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/04_cellranger_count/01_set1"

## original sample IDs, correpond to FASTQ file names
SAMPLE_ID_FILE="/scratch/c.mpmgb/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_set1_IDs.txt"

## corresponding new sample names to be set by --id
## NOTE: some samples will be de-swapped, as they were wrongly labelled during library prep in Oxford
SAMPLE_ID_NEW_FILE="/scratch/c.mpmgb/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_set1_IDs_swap.txt"

## CR executable
CELL_RANGER="/scratch/c.mpmgb/tools/cellranger-7.1.0/bin/cellranger"

## CR reference
## pre-computed and downloaded from CR website as is (refdata-gex-GRCh38-2020-A.tar.gz)
## based on ensembl v98/gencode v32
## CR_REF="/scratch/c.mpmgb/Tools/CellRanger/CellRanger_references/refdata-gex-GRCh38-2020-A"
## updated CR reference based on ensembl v109 (Gencode v43)
## from get_cellranger_reference.sh
CR_REF="/scratch/c.mpmgb/blood-brain-barrier/03_data/990_processed_data/001_snrnaseq/03_cellranger_reference/GRCh38"

#-----------------------------------------------------------------------
# Cell Ranger count

mkdir -p $OUTPUT_DIR

N=${SLURM_ARRAY_TASK_ID}

SAMPLE_ID=$(cat $SAMPLE_ID_FILE | tail -n+${N} | head -1)
SAMPLE_ID_NEW=$(cat $SAMPLE_ID_NEW_FILE | tail -n+${N} | head -1)

echo "Input dir: "$INPUT_DIR
echo "Sample: "$SAMPLE_ID
echo "Sample new: "$SAMPLE_ID_NEW

cd $OUTPUT_DIR

$CELL_RANGER count --id=$SAMPLE_ID_NEW \
--fastqs=$INPUT_DIR \
--sample=$SAMPLE_ID \
--transcriptome=$CR_REF \
--localcores=40 \
--localmem=600 \
--include-introns=true \
--no-bam

## optional:
## R2 length is longer (150bp) than usually recommended/required (90bp)
## --r2-length=90

echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
```


