---
title: Generate cellranger reference for set 3
execute: 
  eval: false
---

This is for set 3

Note that the sample IDs are in the fastq file names.
The following command was used to extract them:

`find 230327_A00748_0368_AH5CTMDSX5_fastq/ -name '*.fastq.gz' | grep -oP '(?<=/)\w{20}' | cut -c4-7 | sort | uniq  > samples_set3.txt`

There are also some `Undeterimed` files that contain the reads that didn't match to any barcodes, but I manually added the missing bit of the string for that one.

## Note 

Sample P_39 gave an error, it seems the fastq files are all empty.
Spoke to Jo and she can't see anything obviously wrong on her end, so it may be that the sample was just not present due to a pipetting error perhaps.
Could also be a index error which resulted in the sample data getting dumped in the "Undetermined" sample.

```{r}
#!/bin/bash

#SBATCH -p c_highmem_dri1 ## dev, compute, htc, highmem
#SBATCH --job-name=cellranger_count_set3
#SBATCH --ntasks=40
#SBATCH --ntasks-per-node=40
#SBATCH --array=1-40%2
##### #SBATCH --mem-per-cpu=8000 # memory limit per core
#SBATCH --mem=340G # memory limit per compute node for the job
#SBATCH --time=2-10:00 # maximum job time in D-HH:MM
#SBATCH --account=scw1329
#SBATCH -o /scratch/c.mpmgb/hawk_output/%x_out_%A_%a_%J.txt
#SBATCH -e /scratch/c.mpmgb/hawk_output/%x_err_%A_%a_%J.txt
#SBATCH --mail-user Bernardo-HarringtonG@cardiff.ac.uk # email on fail
#SBATCH --mail-type END,FAIL

echo "*****************************************************************"
echo "All jobs in this array have:"
echo "- SLURM_ARRAY_JOB_ID: ${SLURM_ARRAY_JOB_ID}"
echo "- SLURM_ARRAY_TASK_COUNT: ${SLURM_ARRAY_TASK_COUNT}"
echo "- SLURM_ARRAY_TASK_MIN: ${SLURM_ARRAY_TASK_MIN}"
echo "- SLURM_ARRAY_TASK_MAX: ${SLURM_ARRAY_TASK_MAX}"
echo "This job in the array has:"
echo "- SLURM_JOB_ID: ${SLURM_JOB_ID}"
echo "- SLURM_ARRAY_TASK_ID: ${SLURM_ARRAY_TASK_ID}"
echo "Run on host: "`hostname`
echo "Number of threads (nproc): "`nproc`
echo "Total memory in GB: "`free -g | grep -oP '\d+' | sed -n 1p`
echo "Used memory in GB: "`free -g | grep -oP '\d+' | sed -n 2p`
echo "Free memory in GB: "`free -g | grep -oP '\d+' | sed -n 3p`
echo "Username: "`whoami`
echo "Started at: "`date`
echo -e "*****************************************************************\n"


## FASTQ files, set 2 with 16 samples, data from 4 runs, sequenced in Oxford
## NOTE, V_19, only sequenced in 1 run (INPUT_DIR_1)
## NOTE, V_20, 1 run (INPUT_DIR_2) only with a single read, skip that run
## handled via if statement below
INPUT_DIR="/gluster/dri02/rdsmbh/shared/rdsmbh/230327_A00748_0368_AH5CTMDSX5_fastq/"

## results
OUTPUT_DIR="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/04_cellranger_count/03_set3"

## original sample IDs, correpond to FASTQ file names
SAMPLE_ID_FILE="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/90_sample_info/samples_set3.txt"

## CR executable
CELL_RANGER="/scratch/c.mpmgb/tools/cellranger-7.1.0/bin/cellranger"

## CR reference
## pre-computed and downloaded from CR website as is (refdata-gex-GRCh38-2020-A.tar.gz)
## based on ensembl v98/gencode v32
## CR_REF="/scratch/c.mpmfw/Tools/CellRanger/CellRanger_references/refdata-gex-GRCh38-2020-A"
## updated CR reference based on ensembl v108 (Gencode v42)
## from get_cellranger_reference.sh
CR_REF="/scratch/scw1329/gmbh/blood-brain-barrier-in-ad/03_data/990_processed_data/001_snrnaseq/03_cellranger_reference/GRCh38"


#-----------------------------------------------------------------------
# Cell Ranger count

mkdir -p $OUTPUT_DIR

N=${SLURM_ARRAY_TASK_ID}

SAMPLE_ID=$(cat $SAMPLE_ID_FILE | tail -n+${N} | head -1)
# Get the sample ID string after the first "_"
SAMPLE_ID_NEW="${SAMPLE_ID#*_}"

echo "Input dir: "$INPUT_DIR
echo "Sample: "$SAMPLE_ID
echo "Sample new: "$SAMPLE_ID_NEW

cd $OUTPUT_DIR

$CELL_RANGER count --id=$SAMPLE_ID_NEW \
--fastqs=$INPUT_DIR \
--sample=$SAMPLE_ID \
--transcriptome=$CR_REF \
--localcores=40 \
--localmem=330 \
--include-introns=true \
--no-bam

## optional:
## R2 length is longer (151bp) than usually recommended/required (90bp)
## --r2-length=90

echo -e "\n*****************************************************************"
echo "Finished at: "`date`
echo "*****************************************************************"
```

One of the many outputs of interest is the `web_summary.html` report files that have include various metrics worth checking.
I'm not keen to add the whole output to the Git repo due to the file size, so I'll just add these files for now.

Note that there's a also a `web_summary.html` in the `SC_RNA_COUNTER_CS` folder, but I don't want that one here so I'll also remove those.

```{r}
#| purl: false
# get the file paths for all the "web_summary.html" files and save them to a file
find 03_data/990_processed_data/001_snrnaseq/04_cellranger_count/ -name "web_summary.html" > web_summary_files.txt

# the line length for the summary in the SC_RNA_COUNTER_CS dir is longer:
awk '{print length}' web_summary_files.txt | sort -n | uniq -c
# so we'll remove lines based on their length
awk 'length <= max_length' max_length="140" web_summary_files.txt > web_summary_files_clean.txt
# prepend "git add " to the file lines
sed 's/^/git add /' web_summary_files_clean.txt > web_summary_files_git.sh

# run the file
sh web_summary_files_git.sh
# clean up
rm web_summary_files.txt web_summary_files_clean.txt web_summary_files_git.sh
```
