---
title: Perform Pseudobulk
execute:
  eval: true
#self-contained: true
code-fold: true
bibliography: references.bib
---

Note that I'll use the higher level annotations to start with.
This only involved aggregating the venous/capillary/arterial subtypes to a larger "Endothelial" category.

```{r}
here::i_am("04_data_analysis/001_snrnaseq_analysis/11_pseudobulk.qmd")
```

# Load packages

```{r}
source(here::here(
  "04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"
))
```

```{r}
path <- here::here("04_data_analysis/990_code_libraries/pseudobulk")
#source necessary functions from other scripts
source(here::here(path, "make_pseudobulk.R"))
source(here::here(path, "de_analysis.R"))
source(here::here(path, "plot_de_analysis.R"))
```

# Read in data

First we'll read in the scFlow output to a sce

```{r}
#| eval: true
# read data
sce <- readr::read_rds(here::here(
  "03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"
))
```

```{r}
# save gene and ensembl IDs to switch between them as needed
df <- data.frame(
  gene = rownames(sce@assays$RNA@data),
  ensembl = rownames(sce@assays$RNA@counts)
)
# make count rownames match data rownames
rownames(sce@assays$RNA@counts) <- rownames(sce@assays$RNA@data)
sce$highlevel_manual_annotations <- Idents(sce)
```

## Check age breakdown

It might be worth binning ages instead of adding them as a covariate but I need to check the distribution of ages.

```{r}

meta <- sce@meta.data |>
  dplyr::distinct(donor, .keep_all = TRUE)
summary(meta$age)
meta |>
  ggplot(aes(diagnosis, age)) +
  geom_point()

meta |>
  summarise(
    mean = mean(age),
    median = median(age),
    iqr = IQR(age),
    sd = sd(age),
    min = min(age),
    max = max(age),
    .by = diagnosis
  )

sce$donor25 <- if_else(sce$donor == "donor-25", TRUE, FALSE)

DimPlot(sce, reduction = "umap", group.by = "donor25")
```

# Average expression

```{r}
avg_exp <- AverageExpression(sce, group.by = c("ident", "donor"))
avg_exp <- avg_exp$RNA

avg_exp <- as.data.frame(avg_exp) |>
  rownames_to_column(var = "gene")

# pivot longer such that celltypes are separated out
avg_exp <- avg_exp |>
  pivot_longer(
    cols = !gene,
    names_to = c("celltype", ".value"),
    names_pattern = "(.*)_(.*)"
  )

file_out <- here::here(
  "03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-celltype-per-donor.csv"
)
readr::write_csv(avg_exp, file_out)
```

```{r}
# Make column with cluster and annotation combined
sce$cluster_annoated <- paste0(
  "cluster_",
  sce$seurat_clusters,
  "_",
  Idents(sce)
)

avg_exp <- AverageExpression(sce, group.by = c("cluster_annoated", "donor"))
avg_exp <- avg_exp$RNA

avg_exp <- as.data.frame(avg_exp) |>
  rownames_to_column(var = "gene")

# pivot longer such that celltypes are separated out
avg_exp <- avg_exp |>
  pivot_longer(
    cols = !gene,
    names_to = c("celltype", ".value"),
    names_pattern = "(.*)_(.*)"
  )

file_out <- here::here(
  "03_data/990_processed_data/001_snrnaseq/08_pseudobulk/avg-expression-per-cluster-per-donor.csv"
)
readr::write_csv(avg_exp, file_out)
```

```{r}
cluster_count_by_sample <- table(Idents(sce), sce$donor) |>
  as.data.frame() |>
  pivot_wider(names_from = Var2, values_from = Freq) |>
  dplyr::rename(cell_type = Var1)

sce@meta.data |>
  group_by(manual_annotations, donor, prep) |>
  summarise(n = n()) |>
  write_csv(
    "~/Dropbox (UK Dementia Research Institute)/bbb-project_2023-06-30/data/counts_per_celltype-donor-prep.csv"
  )

sce@meta.data |>
  group_by(highlevel_manual_annotations, donor, prep) |>
  summarise(n = n()) |>
  write_csv(
    "~/Dropbox (UK Dementia Research Institute)/bbb-project_2023-06-30/data/counts_per_highlevelcelltype-donor-prep.csv"
  )

cluster_count_by_sample |>
  DT::datatable(
    escape = F,
    rownames = F,
    extensions = 'Buttons',
    options = list(
      dom = 'Blfrtip',
      buttons = list(
        'colvis',
        list(
          extend = 'copy',
          exportOptions = list(columns = ':visible')
        ),
        list(
          extend = 'csv',
          exportOptions = list(columns = ':visible')
        ),
        list(
          extend = 'excel',
          exportOptions = list(columns = ':visible')
        )
      )
    )
  )
```

# Pseudobulk

Convert back to single cell experiment

```{r}
sce <- Seurat::as.SingleCellExperiment(sce)
```

# DESeq 2

```{r}
#| eval: true
# read data
sce <- readr::read_rds(here::here(
  "03_data/990_processed_data/001_snrnaseq/07_scflow_analysis/scflow-sce-annotated-highlevel.rds"
))
```

```{r}
# save gene and ensembl IDs to switch between them as needed
df <- data.frame(
  gene = rownames(sce@assays$RNA@data),
  ensembl = rownames(sce@assays$RNA@counts)
)
# make count rownames match data rownames
rownames(sce@assays$RNA@counts) <- rownames(sce@assays$RNA@data)
```

```{r}
# I'll do the manually annotated ones not the reduced highlevel ones for now
sce$highlevel_manual_annotations <- Idents(sce)
```

## Additional metadata

```{r}
add_meta <- readxl::read_excel(here(
  "03_data/993_additional_metadata/additional_metadata.xlsx"
)) |>
  janitor::clean_names() |>
  dplyr::select(!case_7) |>
  dplyr::rename(diagnosis_add_details = diagnosis)
head(sce@meta.data)
names(add_meta)[names(add_meta) %in% names(sce@meta.data)]

# Add metadata
sce@meta.data <- sce@meta.data |>
  rownames_to_column("row_ids") |>
  dplyr::left_join(add_meta, by = join_by(donor_ID == case_1)) |>
  column_to_rownames("row_ids")
```

Let's first try doing pseudobulk by disease status.

```{r}
# setup columns to aggregate by
sce$celltype <- Idents(sce)
# remove the "_" in the celltype names and replace with "-"
sce$celltype <- gsub("_", "-", sce$celltype)
levels(sce$diagnosis) <- c("AD", "Control")

# counts matrix individual level
cts <- AggregateExpression(
  sce,
  group.by = c("celltype", "donor", "diagnosis"),
  assays = "RNA",
  slot = "counts",
  return.seurat = FALSE
)
cts <- cts$RNA
cts[1:5, 1:10]
```

Now we need to do some data manipulation to get the count and column data in the format DESeq2 requires

```{r}
# transpose
cts_t <- t(cts) |>
  as.data.frame()

split_rows <- gsub('_.*', '', rownames(cts_t))

cts_split <- split.data.frame(cts_t, f = factor(split_rows))

cts_split_m <- lapply(cts_split, function(x) {
  t(x)
})

# Function to modify column names
modify_column_names <- function(df, number_of_groups = 2) {
  df <- as.data.frame(df)
  # get the number of "_" seperations in the names
  number_of_splits <- length(strsplit(as.character(names(df)), "_")[[1]])
  # get the last n - 1 groups
  cols <-
    map(
      (number_of_splits - (number_of_groups - 1)):number_of_splits,
      ~ sapply(
        strsplit(
          as.character(names(
            df
          )),
          "_"
        ),
        "[",
        .x
      )
    )
  names(df) <- do.call(paste, c(cols, sep = "_"))

  return(df)
}
cts_split_m <- map(cts_split_m, modify_column_names)
```

```{r}
cts_split_m$`Neuron-ex`[1:10, 1:2]
```

I need to add additional data for the design

```{r}
# get unique meta data per sample
meta <- sce@meta.data |>
  dplyr::distinct(orig.ident, .keep_all = TRUE) |>
  dplyr::select(donor, sex, age, hypertension_status, apoe_status) |>
  dplyr::mutate(over_85 = if_else(age <= 85, "Under_85", "Over_85"))
```

```{r}
# get the col data
colData <- map(
  cts_split_m,
  ~ data.frame(
    samples = colnames(.x),
    condition = factor(ifelse(grepl("Control", colnames(.x)), "Control", "AD")),
    orig.ident = sapply(strsplit(colnames(.x), "_"), "[", 1)
  ) |>
    dplyr::left_join(meta, by = join_by(orig.ident == donor)) |>
    dplyr::distinct() |>
    column_to_rownames(var = "samples") |>
    dplyr::mutate(apoe_status = str_replace(apoe_status, "/", "_")) |>
    dplyr::mutate(apoe_status = as.factor(apoe_status))
)

# make sure controls are the reference level
colData <- map(
  colData,
  ~ .x |>
    dplyr::mutate(condition = relevel(condition, ref = "Control"))
)


str(colData$Astro)
```

```{r}
# check that count and col data cols/rows match
check <- map2_lgl(cts_split_m, colData, ~ ncol(.x) == nrow(.y))
assertthat::assert_that(sum(check) != 0)
```

## Set APOE coarvar inclusion

- Note - given the APOE genotype is so unbalanced between cases and controls we ultimately elected to not include it in the analysis, but I'll leave the code as is.

```{r}
use_apoe <- FALSE
```

```{r}
# get DESeq2 object
if (use_apoe) {
  dds <- map2(
    cts_split_m,
    colData,
    ~ DESeqDataSetFromMatrix(
      countData = .x,
      colData = .y,
      design = ~ condition + age + sex + apoe_status
    )
  )
} else {
  dds <- map2(
    cts_split_m,
    colData,
    ~ DESeqDataSetFromMatrix(
      countData = .x,
      colData = .y,
      design = ~ condition * hypertension_status + over_85 + sex
    )
  )
}
```

```{r}
# run DESeq2
dds <- map(dds, DESeq)
```

We can see the result names

```{r}
resultsNames(dds$Astro)
```

And then extract these results

```{r}
if (use_apoe) {
  res <- map(dds, ~ results(.x, name = "condition_AD_vs_Control"))
} else {
  # 1. AD effect (controlling for everything else)
  res_ad <- map(
    dds,
    ~ results(.x, contrast = c("condition", "AD", "Control")) |>
      as.data.frame() |>
      dplyr::mutate(contrast = "AD")
  )

  # 2. Hypertension effect
  res_hypertension <- map(
    dds,
    ~ results(.x, contrast = c("hypertension_status", "Yes", "No")) |>
      as.data.frame() |>
      dplyr::mutate(contrast = "hypertension_yes-no")
  )

  # 3. Interaction effect
  res_interaction <- map(
    dds,
    ~ results(.x, name = "conditionAD.hypertension_statusYes") |>
      as.data.frame() |>
      dplyr::mutate(contrast = "AD-hypertension")
  )

  # Combine them
  res <- c(res_ad, res_hypertension, res_interaction)
}
head(res$Astro)
```

```{r}
# merge celltype dataframes
res_merged <- map2(
  res,
  names(res),
  ~ as.data.frame(.x) |>
    rownames_to_column("gene") |>
    dplyr::mutate(
      celltype = .y,
      deg_direction = case_when(
        log2FoldChange > 0 & padj < 0.05 ~ "Up",
        log2FoldChange < 0 & padj < 0.05 ~ "Down",
        .default = "Not Sig"
      )
    )
) |>
  purrr::list_rbind()
```

```{r}
# save results
if (use_apoe) {
  readr::write_rds(
    res_merged,
    here::here(
      "03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_apoe.rds"
    )
  )
} else {
  readr::write_rds(
    res_merged,
    here::here(
      "03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_noapoe_hypertension.rds"
    )
  )
}
```

## Plots

```{r}
#| warning: false
res_merged |>
  ggplot(aes(x = factor(deg_direction), fill = factor(celltype))) +
  geom_histogram(stat = "count") +
  facet_wrap(~ factor(celltype)) +
  ggtitle("Cell type counts of DEGs - Control VS AD") +
  labs(y = "Log2 Fold Change", x = "DEG direction", fill = "Cell Type") +
  theme_cowplot() +
  theme(axis.text = element_text(size = 9)) +
  scale_fill_viridis(discrete = T)
#scale_fill_manual(values=pal)
```

### Volcano plots

```{r}
# add colour identifier
cols <- c("Up" = "#FDE725FF", "Down" = "#440154FF", "Not Sig" = "#d3d3d3")

volc_plot <- res_merged |>
  # scale FDR by log10
  ggplot(aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(colour = deg_direction), stat = "identity", size = .2) +
  facet_wrap(~ factor(celltype), scales = "free") +
  geom_hline(
    yintercept = -log10(0.05),
    colour = "#990000",
    linetype = "dashed",
    linewidth = .3
  ) +
  # geom_vline(xintercept = 0, colour="black", size=.3) +
  labs(y = "-log10(FDR)", x = "Log2 Fold Change", colour = "") +
  theme_cowplot() +
  theme(
    axis.text = element_text(size = 9),
    legend.text = element_text(size = 10)
  ) +
  scale_colour_manual(values = cols) +
  geom_text_repel(
    # give top three genes per cell type by adjusted p value
    data = res_merged |>
      dplyr::group_by(celltype) |>
      dplyr::slice_min(padj, n = 3),
    aes(label = gene),
    size = 3,
    box.padding = unit(0.35, "lines"),
    point.padding = unit(0.3, "lines")
  )

volc_plot
# Save
ggsave(
  here::here("05_figures/990_shared_figures/deseq2_volcano-plot.png"),
  volc_plot,
  device = "png"
)
# Remove
rm(volc_plot)
```

