---
title: Pseudobulk - subtypes
execute:
  eval: true
---

I'll redo the pseudobulk with the updated subtype annotations here

```{r}
here::i_am("04_data_analysis/001_snrnaseq_analysis/15_pseudobulk.qmd")
```

# Load packages

```{r}
source(here::here("04_data_analysis/990_code_libraries/02_library-calls-for-renv.R"))
```

# Read in data

```{r}
#| eval: true
# read data
sce <- qs::qread(here::here("03_data/990_processed_data/001_snrnaseq/11_cell_network_interactions/scflow-sce-annotated.qs"))
```

```{r}
# need to fix rownames in order to subset
gene_ensembl_ids <- data.frame(gene = rownames(sce@assays$RNA@data), ensembl = rownames(sce@assays$RNA@counts))
```

## Endothelin-1

EDN1 is of interest

```{r}
DotPlot(sce, features = "EDN1")

data <- FetchData(sce, vars = c("EDN1", "diagnosis", "highlevel_manual_annotations"))
ggplot(data, aes(y = highlevel_manual_annotations, x = EDN1, fill = diagnosis)) +
  geom_violin() +
  theme_minimal() +
  labs(title = "EDN1 Expression in Different Cell Types",
       x = "Cell Type",
       y = "EDN1 Expression")

avg_cases <- AggregateExpression(sce, group.by = c("ident", "diagnosis")) #|>
end1 <- avg_cases$RNA |>
  as.data.frame() |>
  rownames_to_column("gene") |>
  dplyr::filter(gene == "ENSG00000078401") |>
  pivot_longer(!gene, names_to = "celltype", values_to = "avg_expression") |>
  separate_wider_delim(celltype, delim = "_", names = c("celltype", "diagnosis"))

end1 |>
  dplyr::filter(avg_expression > 1000) |>
  ggplot(aes(gene, celltype, colour = log(avg_expression))) +
  geom_point() +
  facet_wrap(~diagnosis)
```

# GLP1R

```{r}
#| eval: false
Idents(sce) <- sce$subcelltype_annotations
DotPlot(sce, features = "GLP1R")
ggsave(here("GLP1R_dotplot.png"), height = 9, width = 4)
FeaturePlot(sce, features = "GLP1R")
ggsave(here("GLP1R_umap.png"), height = 5, width = 5)
sce$level1 <- if_else(is.na(sce$level1_celltypes_with_endomt_subclusters), sce$highlevel_manual_annotations, sce$level1_celltypes_with_endomt_subclusters)
Idents(sce) <- sce$level1
VlnPlot(sce, features = "GLP1R",  pt.size = 0)
ggsave(here("GLP1R_violin.png"), height = 5, width = 5)
```

```{r}
rm(list = ls()[!ls() %in% "sce"])
```

# DESeq 2

```{r}
# get gene and ensembl IDs
gene_ensembl_ids <- data.frame(gene = rownames(sce@assays$RNA@data), ensembl = rownames(sce@assays$RNA@counts))
```

```{r}
# setup columns to aggregate by
sce$celltype <- Idents(sce)
# remove the "_" in the celltype names and replace with "-"
sce$celltype <- gsub("_", "-", sce$celltype)
sce$diagnosis <- fct_recode(sce$diagnosis, AD = "Case")

rownames(sce@assays$RNA@counts) <- gene_ensembl_ids$gene

# counts matrix individual level
cts <-
  AggregateExpression(
    sce,
    group.by = c("celltype", "donor", "diagnosis"),
    assays = "RNA",
    slot = "counts",
    return.seurat = FALSE
  )
cts <- cts$RNA
cts[1:5, 1:10]
```

Now we need to do some data manipulation to get the count and column data in the format DESeq2 requires

```{r}
# transpose
cts_t <- t(cts) |>
  as.data.frame()

split_rows <- gsub('_.*', '', rownames(cts_t))

cts_split <- split.data.frame(cts_t,
                              f = factor(split_rows))

cts_split_m <- lapply(cts_split, function(x) {
  t(x)
})

# Function to modify column names
modify_column_names <- function(df, number_of_groups = 2) {
  df <- as.data.frame(df)
  # get the number of "_" seperations in the names
  number_of_splits <- length(strsplit(as.character(names(df)), "_")[[1]])
  # get the last n - 1 groups
  cols <-
    map((number_of_splits - (number_of_groups - 1)):number_of_splits,
        ~ sapply(strsplit(as.character(names(
          df
        )), "_"), "[", .x))
  names(df) <- do.call(paste, c(cols, sep = "_"))
  
  return(df)
}
cts_split_m <- map(cts_split_m, modify_column_names)
```

```{r}
cts_split_m$`Ex-Neuron-ADARB2-ERBB4`[1:10,1:4]
```

I need to add additional data for the design

```{r}
# get unique meta data per sample
meta <- sce@meta.data |> 
  dplyr::distinct(orig.ident, .keep_all = TRUE) |>
  dplyr::select(donor, sex, apoe_status, age)
```

```{r}
# get the col data
colData <- map(
  cts_split_m,
  ~ data.frame(
    samples = colnames(.x),
    condition = factor(ifelse(grepl("Control", colnames(.x)), "Control", "AD")),
    donor = sapply(strsplit(colnames(.x), "_"), "[", 1)
  ) |>
    dplyr::left_join(meta, by = "donor") |>
    dplyr::distinct() |>
    column_to_rownames(var = "samples") |>
    dplyr::mutate(apoe_status = str_replace(apoe_status, "/", "-")) |>
    dplyr::mutate(apoe_status = as.factor(apoe_status))
)

# make sure controls are the reference level
colData <- map(colData, ~ .x |>
                 dplyr::mutate(condition = relevel(condition, ref = "Control")))


str(colData$Arterial)
```

```{r}
# check that count and col data cols/rows match
check <- map2_lgl(cts_split_m, colData, ~ ncol(.x) == nrow(.y))
assertthat::assert_that(sum(check) != 0)
```

## With and without APOE 

Want to run the differential expression with and without APOE as a covariate

```{r}
use_apoe = FALSE
if (use_apoe) {
  # get DESeq2 object
  dds <-
    map2(
      cts_split_m,
      colData,
      ~ DESeqDataSetFromMatrix(
        countData = .x,
        colData = .y,
        design = ~ condition + age + sex + apoe_status
      )
    )
} else {
  # Without apoe
  dds <-
    map2(
      cts_split_m,
      colData,
      ~ DESeqDataSetFromMatrix(
        countData = .x,
        colData = .y,
        design = ~ condition + age + sex
      )
    )
}
```

```{r}
# run DESeq2
dds <- map(dds, DESeq)
```

We can see the result names

```{r}
resultsNames(dds$Arterial)
```

And then extract these results

```{r}
res <- map(dds, ~results(.x, name = "condition_AD_vs_Control"))
head(res$Arterial)
```


```{r}
# merge celltype dataframes
res_merged <- map2(res, names(res), ~ as.data.frame(.x) |>
                     rownames_to_column("gene") |>
                     dplyr::mutate(celltype = .y, deg_direction = case_when(
                       log2FoldChange > 0 & padj < 0.05 ~ "Up",
                       log2FoldChange < 0 & padj < 0.05 ~ "Down",
                       .default = "Not Sig"))) |>
  purrr::list_rbind()
head(res_merged)
```

```{r}
# save results
if(use_apoe) {
  readr::write_csv(
    res_merged,
    here::here(
      "03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_apoe.csv"
    )
  )
} else {
  readr::write_csv(
    res_merged,
    here::here(
      "03_data/990_processed_data/001_snrnaseq/08_pseudobulk/pseudobulk_ad-vs-control_deseq2_subtype-annotated_no-apoe.csv"
    )
  )
}
```

## Plots

```{r}
#| warning: false
res_merged |>
  ggplot(aes(x = factor(deg_direction), fill = factor(celltype))) +
  geom_histogram(stat = "count") +
  facet_wrap(~factor(celltype)) +
  ggtitle("Cell type counts of DEGs - Control VS AD")+
  labs(y= "Log2 Fold Change", x = "DEG direction", fill="Cell Type") +
  theme_cowplot()+
  theme(axis.text = element_text(size=9)) +
  scale_fill_viridis(discrete = T)
  #scale_fill_manual(values=pal)
```

### Volcano plots

```{r}
# add colour identifier
cols <- c("Up" = "#FDE725FF", "Down" = "#440154FF", "Not Sig" = "#d3d3d3")

volc_plot <- res_merged |>
  # filter to significant hits
  #dplyr::filter(padj < 0.05) |>
  # scale FDR by log10
  ggplot(aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(colour = deg_direction), stat = "identity", size = .2) +
  facet_wrap( ~ factor(celltype), scales = "free") +
  geom_hline(
    yintercept = -log10(0.05),
    colour = "#990000",
    linetype = "dashed",
    linewidth = .3
  ) +
  # geom_vline(xintercept = 0, colour="black", size=.3) +
  labs(y = "-log10(FDR)", x = "Log2 Fold Change", colour = "") +
  theme_cowplot() +
  theme(
    axis.text = element_text(size = 9),
    #axis.text.x = element_blank(),
    legend.text = element_text(size = 10)
  ) +
  scale_colour_manual(values = cols) +
  geom_text_repel(
    # give top three genes per cell type by adjusted p value
    data = res_merged |>
      dplyr::group_by(celltype) |>
      dplyr::slice_min(padj, n = 3),
    aes(label = gene),
    size = 3,
    box.padding = unit(0.35, "lines"),
    point.padding = unit(0.3, "lines")
  )

volc_plot
# Save
ggsave(here::here("05_figures/990_shared_figures/deseq2_volcano-plot_subtypes.png"), volc_plot, device = "png")
# Remove
rm(volc_plot)
```

