<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.54">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Gabriel Mateus Bernardo Harrington">
<meta name="author" content="Jimena Monzon Sandoval">
<meta name="author" content="Michal Rokicki">
<meta name="author" content="Zameel Cader">
<meta name="author" content="Caleb Webber">
<meta name="dcterms.date" content="2024-07-17">
<meta name="keywords" content="Alzheimer’s diesease, Blood brain barrier">

<title>BBB in AD</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>


<meta name="citation_title" content="BBB in AD">
<meta name="citation_abstract" content="The blood brain barrier, it&amp;amp;#039;s important, probably in AD even
">
<meta name="citation_keywords" content="Alzheimer's diesease,Blood brain barrier">
<meta name="citation_author" content="Gabriel Mateus Bernardo Harrington">
<meta name="citation_author" content="Jimena Monzon Sandoval">
<meta name="citation_author" content="Michal Rokicki">
<meta name="citation_author" content="Zameel Cader">
<meta name="citation_author" content="Caleb Webber">
<meta name="citation_publication_date" content="2024-07-17">
<meta name="citation_cover_date" content="2024-07-17">
<meta name="citation_year" content="2024">
<meta name="citation_online_date" content="2024-07-17">
<meta name="citation_language" content="en">
<meta name="citation_journal_title" content="Cell">
<meta name="citation_reference" content="citation_title=New insights into the genetic etiology of Alzheimer’s disease and related dementias;,citation_author=Céline Bellenguez;,citation_author=Fahri Küçükali;,citation_author=Iris E. Jansen;,citation_author=Luca Kleineidam;,citation_author=Sonia Moreno-Grau;,citation_author=Najaf Amin;,citation_author=Adam C. Naj;,citation_author=Rafael Campos-Martin;,citation_author=Benjamin Grenier-Boley;,citation_author=Victor Andrade;,citation_author=Peter A. Holmans;,citation_author=Anne Boland;,citation_author=Vincent Damotte;,citation_author=Sven J. Lee;,citation_author=Marcos R. Costa;,citation_author=Teemu Kuulasmaa;,citation_author=Qiong Yang;,citation_author=Itziar Rojas;,citation_author=Joshua C. Bis;,citation_author=Amber Yaqub;,citation_author=Ivana Prokic;,citation_author=Julien Chapuis;,citation_author=Shahzad Ahmad;,citation_author=Vilmantas Giedraitis;,citation_author=Dag Aarsland;,citation_author=Pablo Garcia-Gonzalez;,citation_author=Carla Abdelnour;,citation_author=Emilio Alarcón-Martín;,citation_author=Daniel Alcolea;,citation_author=Montserrat Alegret;,citation_author=Ignacio Alvarez;,citation_author=Victoria Álvarez;,citation_author=Nicola J. Armstrong;,citation_author=Anthoula Tsolaki;,citation_author=Carmen Antúnez;,citation_author=Ildebrando Appollonio;,citation_author=Marina Arcaro;,citation_author=Silvana Archetti;,citation_author=Alfonso Arias Pastor;,citation_author=Beatrice Arosio;,citation_author=Lavinia Athanasiu;,citation_author=Henri Bailly;,citation_author=Nerisa Banaj;,citation_author=Miquel Baquero;,citation_author=Sandra Barral;,citation_author=Alexa Beiser;,citation_author=Ana Belén Pastor;,citation_author=Jennifer E. Below;,citation_author=Penelope Benchek;,citation_author=Luisa Benussi;,citation_author=Claudine Berr;,citation_author=Céline Besse;,citation_author=Valentina Bessi;,citation_author=Giuliano Binetti;,citation_author=Alessandra Bizarro;,citation_author=Rafael Blesa;,citation_author=Mercè Boada;,citation_author=Eric Boerwinkle;,citation_author=Barbara Borroni;,citation_author=Silvia Boschi;,citation_author=Paola Bossù;,citation_author=Geir Bråthen;,citation_author=Jan Bressler;,citation_author=Catherine Bresner;,citation_author=Henry Brodaty;,citation_author=Keeley J. Brookes;,citation_author=Luis Ignacio Brusco;,citation_author=Dolores Buiza-Rueda;,citation_author=Katharina Bûrger;,citation_author=Vanessa Burholt;,citation_author=William S. Bush;,citation_author=Miguel Calero;,citation_author=Laura B. Cantwell;,citation_author=Geneviève Chene;,citation_author=Jaeyoon Chung;,citation_author=Michael L. Cuccaro;,citation_author=Ángel Carracedo;,citation_author=Roberta Cecchetti;,citation_author=Laura Cervera-Carles;,citation_author=Camille Charbonnier;,citation_author=Hung-Hsin Chen;,citation_author=Caterina Chillotti;,citation_author=Simona Ciccone;,citation_author=Jurgen A. H. R. Claassen;,citation_author=Christopher Clark;,citation_author=Elisa Conti;,citation_author=Anaïs Corma-Gómez;,citation_author=Emanuele Costantini;,citation_author=Carlo Custodero;,citation_author=Delphine Daian;,citation_author=Maria Carolina Dalmasso;,citation_author=Antonio Daniele;,citation_author=Efthimios Dardiotis;,citation_author=Jean-François Dartigues;,citation_author=Peter Paul Deyn;,citation_author=Katia Paiva Lopes;,citation_author=Lot D. Witte;,citation_author=Stéphanie Debette;,citation_author=Jürgen Deckert;,citation_author=Teodoro Ser;,citation_author=Nicola Denning;,citation_author=Anita DeStefano;,citation_author=Martin Dichgans;,citation_author=Janine Diehl-Schmid;,citation_author=Mónica Diez-Fairen;,citation_author=Paolo Dionigi Rossi;,citation_author=Srdjan Djurovic;,citation_author=Emmanuelle Duron;,citation_author=Emrah Düzel;,citation_author=Carole Dufouil;,citation_author=Gudny Eiriksdottir;,citation_author=Sebastiaan Engelborghs;,citation_author=Valentina Escott-Price;,citation_author=Ana Espinosa;,citation_author=Michael Ewers;,citation_author=Kelley M. Faber;,citation_author=Tagliavini Fabrizio;,citation_author=Sune Fallgaard Nielsen;,citation_author=David W. Fardo;,citation_author=Lucia Farotti;,citation_author=Chiara Fenoglio;,citation_author=Marta Fernández-Fuertes;,citation_author=Raffaele Ferrari;,citation_author=Catarina B. Ferreira;,citation_author=Evelyn Ferri;,citation_author=Bertrand Fin;,citation_author=Peter Fischer;,citation_author=Tormod Fladby;,citation_author=Klaus Fließbach;,citation_author=Bernard Fongang;,citation_author=Myriam Fornage;,citation_author=Juan Fortea;,citation_author=Tatiana M. Foroud;,citation_author=Silvia Fostinelli;,citation_author=Nick C. Fox;,citation_author=Emlio Franco-Macías;,citation_author=María J. Bullido;,citation_author=Ana Frank-García;,citation_author=Lutz Froelich;,citation_author=Brian Fulton-Howard;,citation_author=Daniela Galimberti;,citation_author=Jose Maria García-Alberca;,citation_author=Pablo García-González;,citation_author=Sebastian Garcia-Madrona;,citation_author=Guillermo Garcia-Ribas;,citation_author=Roberta Ghidoni;,citation_author=Ina Giegling;,citation_author=Giaccone Giorgio;,citation_author=Alison M. Goate;,citation_author=Oliver Goldhardt;,citation_author=Duber Gomez-Fonseca;,citation_author=Antonio González-Pérez;,citation_author=Caroline Graff;,citation_author=Giulia Grande;,citation_author=Emma Green;,citation_author=Timo Grimmer;,citation_author=Edna Grünblatt;,citation_author=Michelle Grunin;,citation_author=Vilmundur Gudnason;,citation_author=Tamar Guetta-Baranes;,citation_author=Annakaisa Haapasalo;,citation_author=Georgios Hadjigeorgiou;,citation_author=Jonathan L. Haines;,citation_author=Kara L. Hamilton-Nelson;,citation_author=Harald Hampel;,citation_author=Olivier Hanon;,citation_author=John Hardy;,citation_author=Annette M. Hartmann;,citation_author=Lucrezia Hausner;,citation_author=Janet Harwood;,citation_author=Stefanie Heilmann-Heimbach;,citation_author=Seppo Helisalmi;,citation_author=Michael T. Heneka;,citation_author=Isabel Hernández;,citation_author=Martin J. Herrmann;,citation_author=Per Hoffmann;,citation_author=Clive Holmes;,citation_author=Henne Holstege;,citation_author=Raquel Huerto Vilas;,citation_author=Marc Hulsman;,citation_author=Jack Humphrey;,citation_author=Geert Jan Biessels;,citation_author=Xueqiu Jian;,citation_author=Charlotte Johansson;,citation_author=Gyungah R. Jun;,citation_author=Yuriko Kastumata;,citation_author=John Kauwe;,citation_author=Patrick G. Kehoe;,citation_author=Lena Kilander;,citation_author=Anne Kinhult Ståhlbom;,citation_author=Miia Kivipelto;,citation_author=Anne Koivisto;,citation_author=Johannes Kornhuber;,citation_author=Mary H. Kosmidis;,citation_author=Walter A. Kukull;,citation_author=Pavel P. Kuksa;,citation_author=Brian W. Kunkle;,citation_author=Amanda B. Kuzma;,citation_author=Carmen Lage;,citation_author=Erika J. Laukka;,citation_author=Lenore Launer;,citation_author=Alessandra Lauria;,citation_author=Chien-Yueh Lee;,citation_author=Jenni Lehtisalo;,citation_author=Ondrej Lerch;,citation_author=Alberto Lleó;,citation_author=William Longstreth;,citation_author=Oscar Lopez;,citation_author=Adolfo Lopez Munain;,citation_author=Seth Love;,citation_author=Malin Löwemark;,citation_author=Lauren Luckcuck;,citation_author=Kathryn L. Lunetta;,citation_author=Yiyi Ma;,citation_author=Juan Macías;,citation_author=Catherine A. MacLeod;,citation_author=Wolfgang Maier;,citation_author=Francesca Mangialasche;,citation_author=Marco Spallazzi;,citation_author=Marta Marquié;,citation_author=Rachel Marshall;,citation_author=Eden R. Martin;,citation_author=Angel Martín Montes;,citation_author=Carmen Martínez Rodríguez;,citation_author=Carlo Masullo;,citation_author=Richard Mayeux;,citation_author=Simon Mead;,citation_author=Patrizia Mecocci;,citation_author=Miguel Medina;,citation_author=Alun Meggy;,citation_author=Shima Mehrabian;,citation_author=Silvia Mendoza;,citation_author=Manuel Menéndez-González;,citation_author=Pablo Mir;,citation_author=Susanne Moebus;,citation_author=Merel Mol;,citation_author=Laura Molina-Porcel;,citation_author=Laura Montrreal;,citation_author=Laura Morelli;,citation_author=Fermin Moreno;,citation_author=Kevin Morgan;,citation_author=Thomas Mosley;,citation_author=Markus M. Nöthen;,citation_author=Carolina Muchnik;,citation_author=Shubhabrata Mukherjee;,citation_author=Benedetta Nacmias;,citation_author=Tiia Ngandu;,citation_author=Gael Nicolas;,citation_author=Børge G. Nordestgaard;,citation_author=Robert Olaso;,citation_author=Adelina Orellana;,citation_author=Michela Orsini;,citation_author=Gemma Ortega;,citation_author=Alessandro Padovani;,citation_author=Caffarra Paolo;,citation_author=Goran Papenberg;,citation_author=Lucilla Parnetti;,citation_author=Florence Pasquier;,citation_author=Pau Pastor;,citation_author=Gina Peloso;,citation_author=Alba Pérez-Cordón;,citation_author=Jordi Pérez-Tur;,citation_author=Pierre Pericard;,citation_author=Oliver Peters;,citation_author=Yolande A. L. Pijnenburg;,citation_author=Juan A. Pineda;,citation_author=Gerard Piñol-Ripoll;,citation_author=Claudia Pisanu;,citation_author=Thomas Polak;,citation_author=Julius Popp;,citation_author=Danielle Posthuma;,citation_author=Josef Priller;,citation_author=Raquel Puerta;,citation_author=Olivier Quenez;,citation_author=Inés Quintela;,citation_author=Jesper Qvist Thomassen;,citation_author=Alberto Rábano;,citation_author=Innocenzo Rainero;,citation_author=Farid Rajabli;,citation_author=Inez Ramakers;,citation_author=Luis M. Real;,citation_author=Marcel J. T. Reinders;,citation_author=Christiane Reitz;,citation_author=Dolly Reyes-Dumeyer;,citation_author=Perry Ridge;,citation_author=Steffi Riedel-Heller;,citation_author=Peter Riederer;,citation_author=Natalia Roberto;,citation_author=Eloy Rodriguez-Rodriguez;,citation_author=Arvid Rongve;,citation_author=Irene Rosas Allende;,citation_author=Maitée Rosende-Roca;,citation_author=Jose Luis Royo;,citation_author=Elisa Rubino;,citation_author=Dan Rujescu;,citation_author=María Eugenia Sáez;,citation_author=Paraskevi Sakka;,citation_author=Ingvild Saltvedt;,citation_author=Ángela Sanabria;,citation_author=María Bernal Sánchez-Arjona;,citation_author=Florentino Sanchez-Garcia;,citation_author=Pascual Sánchez Juan;,citation_author=Raquel Sánchez-Valle;,citation_author=Sigrid B. Sando;,citation_author=Chloé Sarnowski;,citation_author=Claudia L. Satizabal;,citation_author=Michela Scamosci;,citation_author=Nikolaos Scarmeas;,citation_author=Elio Scarpini;,citation_author=Philip Scheltens;,citation_author=Norbert Scherbaum;,citation_author=Martin Scherer;,citation_author=Matthias Schmid;,citation_author=Anja Schneider;,citation_author=Jonathan M. Schott;,citation_author=Geir Selbæk;,citation_author=Davide Seripa;,citation_author=Manuel Serrano;,citation_author=Jin Sha;,citation_author=Alexey A. Shadrin;,citation_author=Olivia Skrobot;,citation_author=Susan Slifer;,citation_author=Gijsje J. L. Snijders;,citation_author=Hilkka Soininen;,citation_author=Vincenzo Solfrizzi;,citation_author=Alina Solomon;,citation_author=Yeunjoo Song;,citation_author=Sandro Sorbi;,citation_author=Oscar Sotolongo-Grau;,citation_author=Gianfranco Spalletta;,citation_author=Annika Spottke;,citation_author=Alessio Squassina;,citation_author=Eystein Stordal;,citation_author=Juan Pablo Tartan;,citation_author=Lluís Tárraga;,citation_author=Niccolo Tesí;,citation_author=Anbupalam Thalamuthu;,citation_author=Tegos Thomas;,citation_author=Giuseppe Tosto;,citation_author=Latchezar Traykov;,citation_author=Lucio Tremolizzo;,citation_author=Anne Tybjærg-Hansen;,citation_author=Andre Uitterlinden;,citation_author=Abbe Ullgren;,citation_author=Ingun Ulstein;,citation_author=Sergi Valero;,citation_author=Otto Valladares;,citation_author=Christine Van Broeckhoven;,citation_author=Jeffery Vance;,citation_author=Badri N. Vardarajan;,citation_author=Aad Lugt;,citation_author=Jasper Van Dongen;,citation_author=Jeroen Rooij;,citation_author=John Swieten;,citation_author=Rik Vandenberghe;,citation_author=Frans Verhey;,citation_author=Jean-Sébastien Vidal;,citation_author=Jonathan Vogelgsang;,citation_author=Martin Vyhnalek;,citation_author=Michael Wagner;,citation_author=David Wallon;,citation_author=Li-San Wang;,citation_author=Ruiqi Wang;,citation_author=Leonie Weinhold;,citation_author=Jens Wiltfang;,citation_author=Gill Windle;,citation_author=Bob Woods;,citation_author=Mary Yannakoulia;,citation_author=Habil Zare;,citation_author=Yi Zhao;,citation_author=Xiaoling Zhang;,citation_author=Congcong Zhu;,citation_author=Miren Zulaica;,citation_author=Lindsay A. Farrer;,citation_author=Bruce M. Psaty;,citation_author=Mohsen Ghanbari;,citation_author=Towfique Raj;,citation_author=Perminder Sachdev;,citation_author=Karen Mather;,citation_author=Frank Jessen;,citation_author=M. Arfan Ikram;,citation_author=Alexandre Mendonça;,citation_author=Jakub Hort;,citation_author=Magda Tsolaki;,citation_author=Margaret A. Pericak-Vance;,citation_author=Philippe Amouyel;,citation_author=Julie Williams;,citation_author=Ruth Frikke-Schmidt;,citation_author=Jordi Clarimon;,citation_author=Jean-François Deleuze;,citation_author=Giacomina Rossi;,citation_author=Sudha Seshadri;,citation_author=Ole A. Andreassen;,citation_author=Martin Ingelsson;,citation_author=Mikko Hiltunen;,citation_author=Kristel Sleegers;,citation_author=Gerard D. Schellenberg;,citation_author=Cornelia M. Duijn;,citation_author=Rebecca Sims;,citation_author=Wiesje M. Flier;,citation_author=Agustín Ruiz;,citation_author=Alfredo Ramirez;,citation_author=Jean-Charles Lambert;,citation_publication_date=2022-04;,citation_cover_date=2022-04;,citation_year=2022;,citation_fulltext_html_url=https://www.nature.com/articles/s41588-022-01024-z;,citation_issue=4;,citation_doi=10.1038/s41588-022-01024-z;,citation_volume=54;,citation_language=en;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=Genetic meta-analysis of diagnosed Alzheimer’s disease identifies new risk loci and implicates Aβ, tau, immunity and lipid processing;,citation_author=Brian W. Kunkle;,citation_author=Benjamin Grenier-Boley;,citation_author=Rebecca Sims;,citation_author=Joshua C. Bis;,citation_author=Vincent Damotte;,citation_author=Adam C. Naj;,citation_author=Anne Boland;,citation_author=Maria Vronskaya;,citation_author=Sven J. Lee;,citation_author=Alexandre Amlie-Wolf;,citation_author=Céline Bellenguez;,citation_author=Aura Frizatti;,citation_author=Vincent Chouraki;,citation_author=Eden R. Martin;,citation_author=Kristel Sleegers;,citation_author=Nandini Badarinarayan;,citation_author=Johanna Jakobsdottir;,citation_author=Kara L. Hamilton-Nelson;,citation_author=Sonia Moreno-Grau;,citation_author=Robert Olaso;,citation_author=Rachel Raybould;,citation_author=Yuning Chen;,citation_author=Amanda B. Kuzma;,citation_author=Mikko Hiltunen;,citation_author=Taniesha Morgan;,citation_author=Shahzad Ahmad;,citation_author=Badri N. Vardarajan;,citation_author=Jacques Epelbaum;,citation_author=Per Hoffmann;,citation_author=Merce Boada;,citation_author=Gary W. Beecham;,citation_author=Jean-Guillaume Garnier;,citation_author=Denise Harold;,citation_author=Annette L. Fitzpatrick;,citation_author=Otto Valladares;,citation_author=Marie-Laure Moutet;,citation_author=Amy Gerrish;,citation_author=Albert V. Smith;,citation_author=Liming Qu;,citation_author=Delphine Bacq;,citation_author=Nicola Denning;,citation_author=Xueqiu Jian;,citation_author=Yi Zhao;,citation_author=Maria Del Zompo;,citation_author=Nick C. Fox;,citation_author=Seung-Hoan Choi;,citation_author=Ignacio Mateo;,citation_author=Joseph T. Hughes;,citation_author=Hieab H. Adams;,citation_author=John Malamon;,citation_author=Florentino Sanchez-Garcia;,citation_author=Yogen Patel;,citation_author=Jennifer A. Brody;,citation_author=Beth A. Dombroski;,citation_author=Maria Candida Deniz Naranjo;,citation_author=Makrina Daniilidou;,citation_author=Gudny Eiriksdottir;,citation_author=Shubhabrata Mukherjee;,citation_author=David Wallon;,citation_author=James Uphill;,citation_author=Thor Aspelund;,citation_author=Laura B. Cantwell;,citation_author=Fabienne Garzia;,citation_author=Daniela Galimberti;,citation_author=Edith Hofer;,citation_author=Mariusz Butkiewicz;,citation_author=Bertrand Fin;,citation_author=Elio Scarpini;,citation_author=Chloe Sarnowski;,citation_author=Will S. Bush;,citation_author=Stéphane Meslage;,citation_author=Johannes Kornhuber;,citation_author=Charles C. White;,citation_author=Yuenjoo Song;,citation_author=Robert C. Barber;,citation_author=Sebastiaan Engelborghs;,citation_author=Sabrina Sordon;,citation_author=Dina Voijnovic;,citation_author=Perrie M. Adams;,citation_author=Rik Vandenberghe;,citation_author=Manuel Mayhaus;,citation_author=L. Adrienne Cupples;,citation_author=Marilyn S. Albert;,citation_author=Peter P. De Deyn;,citation_author=Wei Gu;,citation_author=Jayanadra J. Himali;,citation_author=Duane Beekly;,citation_author=Alessio Squassina;,citation_author=Annette M. Hartmann;,citation_author=Adelina Orellana;,citation_author=Deborah Blacker;,citation_author=Eloy Rodriguez-Rodriguez;,citation_author=Simon Lovestone;,citation_author=Melissa E. Garcia;,citation_author=Rachelle S. Doody;,citation_author=Carmen Munoz-Fernadez;,citation_author=Rebecca Sussams;,citation_author=Honghuang Lin;,citation_author=Thomas J. Fairchild;,citation_author=Yolanda A. Benito;,citation_author=Clive Holmes;,citation_author=Hata Karamujić-Čomić;,citation_author=Matthew P. Frosch;,citation_author=Hakan Thonberg;,citation_author=Wolfgang Maier;,citation_author=Gennady Roshchupkin;,citation_author=Bernardino Ghetti;,citation_author=Vilmantas Giedraitis;,citation_author=Amit Kawalia;,citation_author=Shuo Li;,citation_author=Ryan M. Huebinger;,citation_author=Lena Kilander;,citation_author=Susanne Moebus;,citation_author=Isabel Hernández;,citation_author=M. Ilyas Kamboh;,citation_author=RoseMarie Brundin;,citation_author=James Turton;,citation_author=Qiong Yang;,citation_author=Mindy J. Katz;,citation_author=Letizia Concari;,citation_author=Jenny Lord;,citation_author=Alexa S. Beiser;,citation_author=C. Dirk Keene;,citation_author=Seppo Helisalmi;,citation_author=Iwona Kloszewska;,citation_author=Walter A. Kukull;,citation_author=Anne Maria Koivisto;,citation_author=Aoibhinn Lynch;,citation_author=Lluís Tarraga;,citation_author=Eric B. Larson;,citation_author=Annakaisa Haapasalo;,citation_author=Brian Lawlor;,citation_author=Thomas H. Mosley;,citation_author=Richard B. Lipton;,citation_author=Vincenzo Solfrizzi;,citation_author=Michael Gill;,citation_author=W. T. Longstreth;,citation_author=Thomas J. Montine;,citation_author=Vincenza Frisardi;,citation_author=Monica Diez-Fairen;,citation_author=Fernando Rivadeneira;,citation_author=Ronald C. Petersen;,citation_author=Vincent Deramecourt;,citation_author=Ignacio Alvarez;,citation_author=Francesca Salani;,citation_author=Antonio Ciaramella;,citation_author=Eric Boerwinkle;,citation_author=Eric M. Reiman;,citation_author=Nathalie Fievet;,citation_author=Jerome I. Rotter;,citation_author=Joan S. Reisch;,citation_author=Olivier Hanon;,citation_author=Chiara Cupidi;,citation_author=A. G. Andre Uitterlinden;,citation_author=Donald R. Royall;,citation_author=Carole Dufouil;,citation_author=Raffaele Giovanni Maletta;,citation_author=Itziar Rojas;,citation_author=Mary Sano;,citation_author=Alexis Brice;,citation_author=Roberta Cecchetti;,citation_author=Peter St George-Hyslop;,citation_author=Karen Ritchie;,citation_author=Magda Tsolaki;,citation_author=Debby W. Tsuang;,citation_author=Bruno Dubois;,citation_author=David Craig;,citation_author=Chuang-Kuo Wu;,citation_author=Hilkka Soininen;,citation_author=Despoina Avramidou;,citation_author=Roger L. Albin;,citation_author=Laura Fratiglioni;,citation_author=Antonia Germanou;,citation_author=Liana G. Apostolova;,citation_author=Lina Keller;,citation_author=Maria Koutroumani;,citation_author=Steven E. Arnold;,citation_author=Francesco Panza;,citation_author=Olymbia Gkatzima;,citation_author=Sanjay Asthana;,citation_author=Didier Hannequin;,citation_author=Patrice Whitehead;,citation_author=Craig S. Atwood;,citation_author=Paolo Caffarra;,citation_author=Harald Hampel;,citation_author=Inés Quintela;,citation_author=Ángel Carracedo;,citation_author=Lars Lannfelt;,citation_author=David C. Rubinsztein;,citation_author=Lisa L. Barnes;,citation_author=Florence Pasquier;,citation_author=Lutz Frölich;,citation_author=Sandra Barral;,citation_author=Bernadette McGuinness;,citation_author=Thomas G. Beach;,citation_author=Janet A. Johnston;,citation_author=James T. Becker;,citation_author=Peter Passmore;,citation_author=Eileen H. Bigio;,citation_author=Jonathan M. Schott;,citation_author=Thomas D. Bird;,citation_author=Jason D. Warren;,citation_author=Bradley F. Boeve;,citation_author=Michelle K. Lupton;,citation_author=James D. Bowen;,citation_author=Petra Proitsi;,citation_author=Adam Boxer;,citation_author=John F. Powell;,citation_author=James R. Burke;,citation_author=John S. K. Kauwe;,citation_author=Jeffrey M. Burns;,citation_author=Michelangelo Mancuso;,citation_author=Joseph D. Buxbaum;,citation_author=Ubaldo Bonuccelli;,citation_author=Nigel J. Cairns;,citation_author=Andrew McQuillin;,citation_author=Chuanhai Cao;,citation_author=Gill Livingston;,citation_author=Chris S. Carlson;,citation_author=Nicholas J. Bass;,citation_author=Cynthia M. Carlsson;,citation_author=John Hardy;,citation_author=Regina M. Carney;,citation_author=Jose Bras;,citation_author=Minerva M. Carrasquillo;,citation_author=Rita Guerreiro;,citation_author=Mariet Allen;,citation_author=Helena C. Chui;,citation_author=Elizabeth Fisher;,citation_author=Carlo Masullo;,citation_author=Elizabeth A. Crocco;,citation_author=Charles DeCarli;,citation_author=Gina Bisceglio;,citation_author=Malcolm Dick;,citation_author=Li Ma;,citation_author=Ranjan Duara;,citation_author=Neill R. Graff-Radford;,citation_author=Denis A. Evans;,citation_author=Angela Hodges;,citation_author=Kelley M. Faber;,citation_author=Martin Scherer;,citation_author=Kenneth B. Fallon;,citation_author=Matthias Riemenschneider;,citation_author=David W. Fardo;,citation_author=Reinhard Heun;,citation_author=Martin R. Farlow;,citation_author=Heike Kölsch;,citation_author=Steven Ferris;,citation_author=Markus Leber;,citation_author=Tatiana M. Foroud;,citation_author=Isabella Heuser;,citation_author=Douglas R. Galasko;,citation_author=Ina Giegling;,citation_author=Marla Gearing;,citation_author=Michael Hüll;,citation_author=Daniel H. Geschwind;,citation_author=John R. Gilbert;,citation_author=John Morris;,citation_author=Robert C. Green;,citation_author=Kevin Mayo;,citation_author=John H. Growdon;,citation_author=Thomas Feulner;,citation_author=Ronald L. Hamilton;,citation_author=Lindy E. Harrell;,citation_author=Dmitriy Drichel;,citation_author=Lawrence S. Honig;,citation_author=Thomas D. Cushion;,citation_author=Matthew J. Huentelman;,citation_author=Paul Hollingworth;,citation_author=Christine M. Hulette;,citation_author=Bradley T. Hyman;,citation_author=Rachel Marshall;,citation_author=Gail P. Jarvik;,citation_author=Alun Meggy;,citation_author=Erin Abner;,citation_author=Georgina E. Menzies;,citation_author=Lee-Way Jin;,citation_author=Ganna Leonenko;,citation_author=Luis M. Real;,citation_author=Gyungah R. Jun;,citation_author=Clinton T. Baldwin;,citation_author=Detelina Grozeva;,citation_author=Anna Karydas;,citation_author=Giancarlo Russo;,citation_author=Jeffrey A. Kaye;,citation_author=Ronald Kim;,citation_author=Frank Jessen;,citation_author=Neil W. Kowall;,citation_author=Bruno Vellas;,citation_author=Joel H. Kramer;,citation_author=Emma Vardy;,citation_author=Frank M. LaFerla;,citation_author=Karl-Heinz Jöckel;,citation_author=James J. Lah;,citation_author=Martin Dichgans;,citation_author=James B. Leverenz;,citation_author=David Mann;,citation_author=Allan I. Levey;,citation_author=Stuart Pickering-Brown;,citation_author=Andrew P. Lieberman;,citation_author=Norman Klopp;,citation_author=Kathryn L. Lunetta;,citation_author=H.-Erich Wichmann;,citation_author=Constantine G. Lyketsos;,citation_author=Kevin Morgan;,citation_author=Daniel C. Marson;,citation_author=Kristelle Brown;,citation_author=Frank Martiniuk;,citation_author=Christopher Medway;,citation_author=Deborah C. Mash;,citation_author=Markus M. Nöthen;,citation_author=Eliezer Masliah;,citation_author=Nigel M. Hooper;,citation_author=Wayne C. McCormick;,citation_author=Antonio Daniele;,citation_author=Susan M. McCurry;,citation_author=Anthony Bayer;,citation_author=Andrew N. McDavid;,citation_author=John Gallacher;,citation_author=Ann C. McKee;,citation_author=Hendrik Bussche;,citation_author=Marsel Mesulam;,citation_author=Carol Brayne;,citation_author=Bruce L. Miller;,citation_author=Steffi Riedel-Heller;,citation_author=Carol A. Miller;,citation_author=Joshua W. Miller;,citation_author=Ammar Al-Chalabi;,citation_author=John C. Morris;,citation_author=Christopher E. Shaw;,citation_author=Amanda J. Myers;,citation_author=Jens Wiltfang;,citation_author=Sid O’Bryant;,citation_author=John M. Olichney;,citation_author=Victoria Alvarez;,citation_author=Joseph E. Parisi;,citation_author=Andrew B. Singleton;,citation_author=Henry L. Paulson;,citation_author=John Collinge;,citation_author=William R. Perry;,citation_author=Simon Mead;,citation_author=Elaine Peskind;,citation_author=David H. Cribbs;,citation_author=Martin Rossor;,citation_author=Aimee Pierce;,citation_author=Natalie S. Ryan;,citation_author=Wayne W. Poon;,citation_author=Benedetta Nacmias;,citation_author=Huntington Potter;,citation_author=Sandro Sorbi;,citation_author=Joseph F. Quinn;,citation_author=Eleonora Sacchinelli;,citation_author=Ashok Raj;,citation_author=Gianfranco Spalletta;,citation_author=Murray Raskind;,citation_author=Carlo Caltagirone;,citation_author=Paola Bossù;,citation_author=Maria Donata Orfei;,citation_author=Barry Reisberg;,citation_author=Robert Clarke;,citation_author=Christiane Reitz;,citation_author=A. David Smith;,citation_author=John M. Ringman;,citation_author=Donald Warden;,citation_author=Erik D. Roberson;,citation_author=Gordon Wilcock;,citation_author=Ekaterina Rogaeva;,citation_author=Amalia Cecilia Bruni;,citation_author=Howard J. Rosen;,citation_author=Maura Gallo;,citation_author=Roger N. Rosenberg;,citation_author=Yoav Ben-Shlomo;,citation_author=Mark A. Sager;,citation_author=Patrizia Mecocci;,citation_author=Andrew J. Saykin;,citation_author=Pau Pastor;,citation_author=Michael L. Cuccaro;,citation_author=Jeffery M. Vance;,citation_author=Julie A. Schneider;,citation_author=Lori S. Schneider;,citation_author=Susan Slifer;,citation_author=William W. Seeley;,citation_author=Amanda G. Smith;,citation_author=Joshua A. Sonnen;,citation_author=Salvatore Spina;,citation_author=Robert A. Stern;,citation_author=Russell H. Swerdlow;,citation_author=Mitchell Tang;,citation_author=Rudolph E. Tanzi;,citation_author=John Q. Trojanowski;,citation_author=Juan C. Troncoso;,citation_author=Vivianna M. Van Deerlin;,citation_author=Linda J. Van Eldik;,citation_author=Harry V. Vinters;,citation_author=Jean Paul Vonsattel;,citation_author=Sandra Weintraub;,citation_author=Kathleen A. Welsh-Bohmer;,citation_author=Kirk C. Wilhelmsen;,citation_author=Jennifer Williamson;,citation_author=Thomas S. Wingo;,citation_author=Randall L. Woltjer;,citation_author=Clinton B. Wright;,citation_author=Chang-En Yu;,citation_author=Lei Yu;,citation_author=Yasaman Saba;,citation_author=Alberto Pilotto;,citation_author=Maria J. Bullido;,citation_author=Oliver Peters;,citation_author=Paul K. Crane;,citation_author=David Bennett;,citation_author=Paola Bosco;,citation_author=Eliecer Coto;,citation_author=Virginia Boccardi;,citation_author=Phil L. De Jager;,citation_author=Alberto Lleo;,citation_author=Nick Warner;,citation_author=Oscar L. Lopez;,citation_author=Martin Ingelsson;,citation_author=Panagiotis Deloukas;,citation_author=Carlos Cruchaga;,citation_author=Caroline Graff;,citation_author=Rhian Gwilliam;,citation_author=Myriam Fornage;,citation_author=Alison M. Goate;,citation_author=Pascual Sanchez-Juan;,citation_author=Patrick G. Kehoe;,citation_author=Najaf Amin;,citation_author=Nilifur Ertekin-Taner;,citation_author=Claudine Berr;,citation_author=Stéphanie Debette;,citation_author=Seth Love;,citation_author=Lenore J. Launer;,citation_author=Steven G. Younkin;,citation_author=Jean-Francois Dartigues;,citation_author=Chris Corcoran;,citation_author=M. Arfan Ikram;,citation_author=Dennis W. Dickson;,citation_author=Gael Nicolas;,citation_author=Dominique Campion;,citation_author=JoAnn Tschanz;,citation_author=Helena Schmidt;,citation_author=Hakon Hakonarson;,citation_author=Jordi Clarimon;,citation_author=Ron Munger;,citation_author=Reinhold Schmidt;,citation_author=Lindsay A. Farrer;,citation_author=Christine Van Broeckhoven;,citation_author=Michael C. O’Donovan;,citation_author=Anita L. DeStefano;,citation_author=Lesley Jones;,citation_author=Jonathan L. Haines;,citation_author=Jean-Francois Deleuze;,citation_author=Michael J. Owen;,citation_author=Vilmundur Gudnason;,citation_author=Richard Mayeux;,citation_author=Valentina Escott-Price;,citation_author=Bruce M. Psaty;,citation_author=Alfredo Ramirez;,citation_author=Li-San Wang;,citation_author=Agustin Ruiz;,citation_author=Cornelia M. Duijn;,citation_author=Peter A. Holmans;,citation_author=Sudha Seshadri;,citation_author=Julie Williams;,citation_author=Phillippe Amouyel;,citation_author=Gerard D. Schellenberg;,citation_author=Jean-Charles Lambert;,citation_author=Margaret A. Pericak-Vance;,citation_publication_date=2019-03;,citation_cover_date=2019-03;,citation_year=2019;,citation_fulltext_html_url=https://www.nature.com/articles/s41588-019-0358-2;,citation_issue=3;,citation_doi=10.1038/s41588-019-0358-2;,citation_volume=51;,citation_language=en;,citation_journal_title=Nature Genetics;">
<meta name="citation_reference" content="citation_title=scFlow: A scalable and reproducible analysis pipeline for single-cell RNA sequencing dat;,citation_author=Combiz Khozoie;,citation_author=Nurun Fancy;,citation_author=Mahdi Marjaneh;,citation_author=Alan Murphy;,citation_author=Paul Matthews;,citation_author=Nathan Skene;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://doi.org/10.22541/au.162912533.38489960/v2;,citation_doi=10.22541/au.162912533.38489960/v2;">
<meta name="citation_reference" content="citation_title=Nextflow enables reproducible computational workflows;,citation_author=Paolo Di Tommaso;,citation_author=Maria Chatzou;,citation_author=Evan W. Floden;,citation_author=Pablo Prieto Barja;,citation_author=Emilio Palumbo;,citation_author=Cedric Notredame;,citation_publication_date=2017-04;,citation_cover_date=2017-04;,citation_year=2017;,citation_fulltext_html_url=https://www.nature.com/articles/nbt.3820;,citation_issue=4;,citation_doi=10.1038/nbt.3820;,citation_volume=35;,citation_language=en;,citation_journal_title=Nature Biotechnology;">
</head>

<body>

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">BBB in AD</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Authors</div>
          <div class="quarto-title-meta-heading">Affiliations</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Gabriel Mateus Bernardo Harrington, PhD <a href="mailto:bernardo-harringtong@cardiff.ac.uk" class="quarto-title-author-email"><i class="bi bi-envelope"></i></a> <a href="https://orcid.org/0000-0001-6075-3619" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                    <p class="affiliation">
                        <a href="https://ukdri.ac.uk/">
                        UK Dementia Research Institue
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Jimena Monzon Sandoval, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                    <p class="affiliation">
                        <a href="https://ukdri.ac.uk/">
                        UK Dementia Research Institue
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Michal Rokicki, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Zameel Cader, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="https://www.ox.ac.uk/">
                        Oxford University
                        </a>
                      </p>
                  </div>
                      <div class="quarto-title-meta-contents">
            <p class="author">Caleb Webber, PhD </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        <a href="www.cardiff.ac.uk">
                        Cardiff University
                        </a>
                      </p>
                    <p class="affiliation">
                        <a href="https://ukdri.ac.uk/">
                        UK Dementia Research Institue
                        </a>
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">July 17, 2024</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.docx"><i class="bi bi-file-word"></i>MS Word</a></p></div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF (nature)</a></p></div><div class="quarto-title-meta-contents"><p><a href="index-meca.zip" data-meca-link="true"><i class="bi bi-archive"></i>MECA Bundle</a></p></div></div></div>
    </div>

    <div>
      <div class="abstract">
        <div class="block-title">Abstract</div>
        <p>The blood brain barrier, it’s important, probably in AD even</p>
      </div>
    </div>

    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Alzheimer’s diesease, Blood brain barrier</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#summary" id="toc-summary" class="nav-link active" data-scroll-target="#summary"><span class="header-section-number">1</span> Summary</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">2</span> Introduction</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">3</span> Results</a>
  <ul class="collapse">
  <li><a href="#isolating-the-neurovascular-unit" id="toc-isolating-the-neurovascular-unit" class="nav-link" data-scroll-target="#isolating-the-neurovascular-unit"><span class="header-section-number">3.1</span> Isolating the neurovascular unit</a></li>
  <li><a href="#risk-section-summary" id="toc-risk-section-summary" class="nav-link" data-scroll-target="#risk-section-summary"><span class="header-section-number">3.2</span> Risk section summary</a></li>
  <li><a href="#identification-of-risk-associations-in-specific-cell-types" id="toc-identification-of-risk-associations-in-specific-cell-types" class="nav-link" data-scroll-target="#identification-of-risk-associations-in-specific-cell-types"><span class="header-section-number">3.3</span> Identification of Risk Associations in Specific Cell Types</a></li>
  <li><a href="#magma-analysis" id="toc-magma-analysis" class="nav-link" data-scroll-target="#magma-analysis"><span class="header-section-number">3.4</span> MAGMA Analysis</a></li>
  <li><a href="#degs" id="toc-degs" class="nav-link" data-scroll-target="#degs"><span class="header-section-number">3.5</span> DEGs</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="header-section-number">4</span> Discussion</a></li>
  <li><a href="#acknowledgments" id="toc-acknowledgments" class="nav-link" data-scroll-target="#acknowledgments"><span class="header-section-number">5</span> Acknowledgments</a></li>
  <li><a href="#author-contributions" id="toc-author-contributions" class="nav-link" data-scroll-target="#author-contributions"><span class="header-section-number">6</span> Author contributions</a></li>
  <li><a href="#declaration-of-interests" id="toc-declaration-of-interests" class="nav-link" data-scroll-target="#declaration-of-interests"><span class="header-section-number">7</span> Declaration of interests</a></li>
  <li><a href="#figure-titles-and-legends" id="toc-figure-titles-and-legends" class="nav-link" data-scroll-target="#figure-titles-and-legends"><span class="header-section-number">8</span> Figure titles and legends</a></li>
  <li><a href="#tables-with-title-and-legends" id="toc-tables-with-title-and-legends" class="nav-link" data-scroll-target="#tables-with-title-and-legends"><span class="header-section-number">9</span> Tables with title and legends</a></li>
  <li><a href="#star-methods" id="toc-star-methods" class="nav-link" data-scroll-target="#star-methods"><span class="header-section-number">10</span> STAR Methods</a>
  <ul class="collapse">
  <li><a href="#key-resoures-table" id="toc-key-resoures-table" class="nav-link" data-scroll-target="#key-resoures-table"><span class="header-section-number">10.1</span> Key resoures table</a></li>
  <li><a href="#resource-availability" id="toc-resource-availability" class="nav-link" data-scroll-target="#resource-availability"><span class="header-section-number">10.2</span> Resource availability</a></li>
  <li><a href="#experimental-model-and-study-participant-details" id="toc-experimental-model-and-study-participant-details" class="nav-link" data-scroll-target="#experimental-model-and-study-participant-details"><span class="header-section-number">10.3</span> Experimental model and study participant details</a></li>
  <li><a href="#method-details" id="toc-method-details" class="nav-link" data-scroll-target="#method-details"><span class="header-section-number">10.4</span> Method details</a></li>
  <li><a href="#quantification-and-statistical-analysis" id="toc-quantification-and-statistical-analysis" class="nav-link" data-scroll-target="#quantification-and-statistical-analysis"><span class="header-section-number">10.5</span> Quantification and statistical analysis</a></li>
  </ul></li>
  <li><a href="#supplemental-information-titles-and-legends" id="toc-supplemental-information-titles-and-legends" class="nav-link" data-scroll-target="#supplemental-information-titles-and-legends"><span class="header-section-number">11</span> Supplemental information titles and legends</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">12</span> References</a></li>
  </ul>
<div class="quarto-alternate-notebooks"><h2>Notebooks</h2><ul><li><a href="index-preview.html"><i class="bi bi-journal-code"></i>Article Notebook</a></li><li><a href="04_data_analysis/999_results_summary/04_figures-preview.html"><i class="bi bi-journal-code"></i>Build figures</a></li><li><a href="04_data_analysis/005_cell-cell_interactions/01_cellchat-preview.html"><i class="bi bi-journal-code"></i>CellChat</a></li><li><a href="04_data_analysis/005_cell-cell_interactions/02_celltype_deg_overlap-preview.html"><i class="bi bi-journal-code"></i>Celltype DEG overlap</a></li><li><a href="04_data_analysis/990_code_libraries/01_r-packages-used-preview.html"><i class="bi bi-journal-code"></i>create vector of packages used</a></li><li><a href="04_data_analysis/007_magma/19_magma_risk_analysis-preview.html"><i class="bi bi-journal-code"></i>MAGMA - risk analysis</a></li><li><a href="04_data_analysis/007_magma/18_magma_risk_analysis_prep-preview.html"><i class="bi bi-journal-code"></i>MAGMA - risk analysis prep</a></li><li><a href="04_data_analysis/007_magma/20_magma_risk_results_analysis-preview.html"><i class="bi bi-journal-code"></i>MAGMA - risk results analysis</a></li><li><a href="04_data_analysis/007_magma/23_risk_gene_expression-preview.html"><i class="bi bi-journal-code"></i>MAGMA risk gene expression</a></li><li><a href="04_data_analysis/004_mouse_phenotypes/01_MP_enrichment-preview.html"><i class="bi bi-journal-code"></i>MP enrichment analysis</a></li><li><a href="04_data_analysis/007_magma/22_pathway_analysis-preview.html"><i class="bi bi-journal-code"></i>Pathway analysis - MAGMA genes</a></li><li><a href="04_data_analysis/006_protein-protein-interactions/17_protein-protein_interactions-preview.html"><i class="bi bi-journal-code"></i>Protein-protein interactions</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<p>Note that I’ll add a skeleton of all the sections Cell outlines</p>
<section id="summary" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="summary"><span class="header-section-number">1</span> Summary</h2>
<p>The neurovascular unit (NVU) is crucial for brain homeostasis and is implicated in Alzheimer’s disease (AD) pathogenesis. We developed a method to isolate the NVU from parenchymal tissue in post-mortem human brains, achieving over 90% vascular cell enrichment. Using single-nuclei sequencing from 40 samples (20 control and 20 AD cases), we identified risk associations with AD in pericyte and perivascular fibroblast subtypes, as well as in activated microglia. These cells have unique risk signatures linked to amyloid, suggesting a collaborative role in AD risk. Additionally, we discovered EndoMT cells, transitioning from endothelial to mural cells, not previously documented in vascular atlases. This study provides a valuable resource for understanding NVU composition and dynamics in AD, highlighting potential cellular targets for therapeutic intervention and offering new insights into the cellular interactions associated with amyloid and AD risk.</p>
</section>
<section id="introduction" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="introduction"><span class="header-section-number">2</span> Introduction</h2>
<ul>
<li>Role of BBB in AD</li>
<li>Prior vascular atlases (Yang), their methods, shortcomings and how our method addresses them</li>
</ul>
</section>
<section id="results" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="results"><span class="header-section-number">3</span> Results</h2>
<section id="isolating-the-neurovascular-unit" class="level3" data-number="3.1">
<h3 data-number="3.1" class="anchored" data-anchor-id="isolating-the-neurovascular-unit"><span class="header-section-number">3.1</span> Isolating the neurovascular unit</h3>
<div id="fig-method" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A multi-panel figure on our method for isolating the neurovascular unit, and a highlevel data summary. Panel A is a cartoon of the isolation method. Panel B and C are UMAP plots showing the cells from the vascular/parenchymal fractions and the celltype annotation respectively.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-method-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: (A) Neurovascular unit isolation method. (B)"><img src="05_figures/990_shared_figures/003_final_figures/figure1.png" class="img-fluid figure-img" alt="A multi-panel figure on our method for isolating the neurovascular unit, and a highlevel data summary. Panel A is a cartoon of the isolation method. Panel B and C are UMAP plots showing the cells from the vascular/parenchymal fractions and the celltype annotation respectively."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-method-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: (A) Neurovascular unit isolation method. (B)
</figcaption>
</figure>
</div>
<ul>
<li>Explain isolation (or refer to pre-print if it’s ready?)</li>
<li>The quality/purity of the enrichment</li>
<li>The cohort details/summary?</li>
<li>Celltype annotation?</li>
</ul>
</section>
<section id="risk-section-summary" class="level3" data-number="3.2">
<h3 data-number="3.2" class="anchored" data-anchor-id="risk-section-summary"><span class="header-section-number">3.2</span> Risk section summary</h3>
<ul>
<li>TODO:</li>
<li>Run the level 1 celltypes through the PPI/GO/MPO analysis</li>
</ul>
<div id="fig-risk" class="quarto-float quarto-figure quarto-figure-center anchored" alt="A multi-panel figure about Alzheimer's disease risk enrichment.">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-risk-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="05_figures/990_shared_figures/003_final_figures/figure2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: (A) figure legend stuff"><img src="05_figures/990_shared_figures/003_final_figures/figure2.png" class="img-fluid figure-img" alt="A multi-panel figure about Alzheimer's disease risk enrichment."></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-risk-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: (A) figure legend stuff
</figcaption>
</figure>
</div>
<ul>
<li>Analysing Alzheimer’s disease celltype-specific risk with the latest Bellenguez AD GWAS shows a significant enrichment of activated microglia, a pericyte subtype and a perivascular fibroblast subtype.<span class="citation" data-cites="bellenguez2022">(<a href="#ref-bellenguez2022" role="doc-biblioref">Bellenguez et al. 2022</a>)</span></li>
<li>The prior Kunkle GWAS and Bellenguez without the UKBioBank proxy cases (note in the case of the Bellenguez I lost over half the SNPs in the translation to dbSNP rs IDs, dunno what effect keeping them would have…) do not show the same enrichment.<span class="citation" data-cites="kunkle2019">(<a href="#ref-kunkle2019" role="doc-biblioref">Kunkle et al. 2019</a>)</span></li>
<li>GO terms: running GO on the sig magma genes shows amyloid pathways in all three celltypes and regulation of t-cells in microglia/perictye2</li>
</ul>
</section>
<section id="identification-of-risk-associations-in-specific-cell-types" class="level3" data-number="3.3">
<h3 data-number="3.3" class="anchored" data-anchor-id="identification-of-risk-associations-in-specific-cell-types"><span class="header-section-number">3.3</span> Identification of Risk Associations in Specific Cell Types</h3>
<p>We sought to understand what celltype-specific AD risk associations would emerge with a fuller representation of celltypes in the brain. Using single-nuclei sequencing data, we investigated the association of gene expression profiles in various cell types with Alzheimer’s disease (AD) risk. Our analysis identified significant risk associations in three specific cell types: microglia-A, pericyte-2, and perivascular-FB-2.</p>
</section>
<section id="magma-analysis" class="level3" data-number="3.4">
<h3 data-number="3.4" class="anchored" data-anchor-id="magma-analysis"><span class="header-section-number">3.4</span> MAGMA Analysis</h3>
<p>To determine these associations, we employed Multi-marker Analysis of GenoMic Annotation (MAGMA) against the latest Bellenguez AD Genome-Wide Association Study (GWAS).<span class="citation" data-cites="bellenguez2022">(<a href="#ref-bellenguez2022" role="doc-biblioref">Bellenguez et al. 2022</a>)</span> The results revealed significant associations for activated microglia (microglia-A), pericyte-2, and perivascular-FB-2. Microglia-A, which play a crucial role in immune response within the brain, exhibited a pronounced association with AD risk. pericyte-2, involved in blood-brain barrier maintenance and cerebral blood flow regulation, also showed significant associations. Similarly, perivascular-FB-2, essential for the structural integrity of the blood-brain barrier and extracellular matrix composition, were identified as significantly associated with AD risk.</p>
<section id="conditional-analysis" class="level4" data-number="3.4.1">
<h4 data-number="3.4.1" class="anchored" data-anchor-id="conditional-analysis"><span class="header-section-number">3.4.1</span> Conditional Analysis</h4>
<p>We conducted conditional analysis to assess the independence of the risk signals. The results showed significant enrichment for each of the three cell types even when conditioned on the other two cell types, indicating that the observed signals are unique and not confounded by the other cell types.</p>
</section>
<section id="gene-ontology-go-enrichment-analysis" class="level4" data-number="3.4.2">
<h4 data-number="3.4.2" class="anchored" data-anchor-id="gene-ontology-go-enrichment-analysis"><span class="header-section-number">3.4.2</span> Gene Ontology (GO) Enrichment Analysis</h4>
<p>To further understand the biological implications of these associations, we performed Gene Ontology (GO) enrichment analysis on the significant genes identified by MAGMA for each of the aforementioned cell types. The GO analysis revealed several enriched pathways, prominently featuring amyloid-related processes. The enrichment of genes involved in amyloid precursor protein (APP) processing, amyloid-beta (Aβ) formation, and amyloid plaque clearance was particularly notable. This finding aligns with the well-established role of amyloid pathology in AD. Additionally, for microglia-A and pericyte-2 cell types, significant GO terms related to T-cell and lymphocyte regulation were identified, suggesting an immune regulatory component in AD risk.</p>
</section>
<section id="protein-protein-interaction-ppi-networks" class="level4" data-number="3.4.3">
<h4 data-number="3.4.3" class="anchored" data-anchor-id="protein-protein-interaction-ppi-networks"><span class="header-section-number">3.4.3</span> Protein-Protein Interaction (PPI) Networks</h4>
<p>We constructed protein-protein interaction networks for the significant genes identified in each cell type. The PPI networks were significant, and subsequent Louvain clustering of these networks revealed modules associated with amyloid-related processes across all three cell types. This suggests a coordinated risk association involving amyloid processing among microglia-A, pericyte-2, and perivascular-FB-2.</p>
</section>
<section id="mouse-ontology-data-analysis" class="level4" data-number="3.4.4">
<h4 data-number="3.4.4" class="anchored" data-anchor-id="mouse-ontology-data-analysis"><span class="header-section-number">3.4.4</span> Mouse Ontology Data Analysis</h4>
<p>To further validate our findings, we used mouse ontology data to investigate the functional implications of the significant genes. For microglia-A, significant associations were found with abnormal cell chemotaxis and hematoma. pericyte-2 showed significant associations with amyloidosis and tau protein deposits. However, no significant associations were identified for perivascular-FB-2.</p>
</section>
<section id="interpretation" class="level4" data-number="3.4.5">
<h4 data-number="3.4.5" class="anchored" data-anchor-id="interpretation"><span class="header-section-number">3.4.5</span> Interpretation</h4>
<p>The identification of these specific cell types and the enrichment of amyloid-related processes underscore the complex interplay between various cellular components in the brain and AD pathology. Microglia-A, pericyte-2, and perivascular-FB-2 may contribute to disease mechanisms through their roles in immune response, vascular integrity, and amyloid processing, respectively. The independence of these signals, as shown by conditional analysis, highlights the unique contributions of each cell type. Additionally, the PPI network analysis suggests a coordinated risk association involving amyloid-related processes across these cell types. Mouse ontology data further supports these findings, particularly highlighting the roles of microglia-A and pericyte-2 cells in AD-related pathologies.</p>
<p>These findings provide new insights into potential cellular targets for therapeutic intervention in Alzheimer’s disease.</p>
<hr>
<p>This version is more in line with the conventions of scientific writing, avoiding bullet points and providing a more continuous narrative. Adjust the specific details and gene lists as needed based on your actual data.</p>
</section>
</section>
<section id="degs" class="level3" data-number="3.5">
<h3 data-number="3.5" class="anchored" data-anchor-id="degs"><span class="header-section-number">3.5</span> DEGs</h3>
<ul>
<li>ex-neuron 5 - interesting DEGs and sig proportion difference</li>
</ul>
</section>
</section>
<section id="discussion" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="discussion"><span class="header-section-number">4</span> Discussion</h2>
</section>
<section id="acknowledgments" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="acknowledgments"><span class="header-section-number">5</span> Acknowledgments</h2>
</section>
<section id="author-contributions" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="author-contributions"><span class="header-section-number">6</span> Author contributions</h2>
</section>
<section id="declaration-of-interests" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="declaration-of-interests"><span class="header-section-number">7</span> Declaration of interests</h2>
</section>
<section id="figure-titles-and-legends" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="figure-titles-and-legends"><span class="header-section-number">8</span> Figure titles and legends</h2>
</section>
<section id="tables-with-title-and-legends" class="level2" data-number="9">
<h2 data-number="9" class="anchored" data-anchor-id="tables-with-title-and-legends"><span class="header-section-number">9</span> Tables with title and legends</h2>
</section>
<section id="star-methods" class="level2" data-number="10">
<h2 data-number="10" class="anchored" data-anchor-id="star-methods"><span class="header-section-number">10</span> STAR Methods</h2>
<p>Link to info on this here: <a href="https://www.cell.com/star-authors-guide" class="uri">https://www.cell.com/star-authors-guide</a></p>
<p>And a pdf guide <a href="https://www.cell.com/pb-assets/journals/research/cell/methods/Methods_Guide_Cell-1678470557763.pdf">here</a></p>
<ul>
<li>Plan to make the code available on GitHub and make a docker image with the R environment for all the downstream processing - The seurat object from scflow could be included in this image - This could go on ADDI as well?</li>
<li>Raw data will go on GEO I guess</li>
</ul>
<section id="key-resoures-table" class="level3" data-number="10.1">
<h3 data-number="10.1" class="anchored" data-anchor-id="key-resoures-table"><span class="header-section-number">10.1</span> Key resoures table</h3>
</section>
<section id="resource-availability" class="level3" data-number="10.2">
<h3 data-number="10.2" class="anchored" data-anchor-id="resource-availability"><span class="header-section-number">10.2</span> Resource availability</h3>
<section id="lead-contact" class="level4" data-number="10.2.1">
<h4 data-number="10.2.1" class="anchored" data-anchor-id="lead-contact"><span class="header-section-number">10.2.1</span> Lead contact</h4>
<p>Further information and requests for resources and reagents should be directed to and will be fulfilled by the lead contact, <code>name here</code> (<code>email here</code>).</p>
</section>
<section id="materials-availability" class="level4" data-number="10.2.2">
<h4 data-number="10.2.2" class="anchored" data-anchor-id="materials-availability"><span class="header-section-number">10.2.2</span> Materials availability</h4>
<p>No unique reagents were generated for this study.</p>
</section>
<section id="data-and-code-availability" class="level4" data-number="10.2.3">
<h4 data-number="10.2.3" class="anchored" data-anchor-id="data-and-code-availability"><span class="header-section-number">10.2.3</span> Data and code availability</h4>
<ul>
<li>All raw data are available in the GEO database under the accession number GEO:</li>
<li>The Seurat object generated from the scFlow pipeline is included in the Docker image with the R environment used for analysis here:</li>
<li>All original code has been deposited at Zenodo and is publicly available as of the date of publication. DOIs are listed in the key resources table.</li>
</ul>
</section>
</section>
<section id="experimental-model-and-study-participant-details" class="level3" data-number="10.3">
<h3 data-number="10.3" class="anchored" data-anchor-id="experimental-model-and-study-participant-details"><span class="header-section-number">10.3</span> Experimental model and study participant details</h3>
<section id="post-mortem-tissue-donors" class="level4" data-number="10.3.1">
<h4 data-number="10.3.1" class="anchored" data-anchor-id="post-mortem-tissue-donors"><span class="header-section-number">10.3.1</span> Post-mortem tissue donors</h4>
<p>Post-mortem pre-frontal cortex from 20 controls and 20 AD brain donors were collected from <code>brain bank here</code> under <code>ethics details here</code>. The clinicopathological parameters were collected and summarized in <a href="#tbl-donor-metadata" class="quarto-xref">Table&nbsp;1</a>, including gender, age, diagnosis, APOE status, ethnicity and Braak stage. 3 donors where excluded in QC and of the remaining 37 donors, 23 were female and 14 were male with a median age of 87 (range 48-101). Sex and age were included as a covariates in pseudobulk differential gene expression analysis.</p>
</section>
</section>
<section id="method-details" class="level3" data-number="10.4">
<h3 data-number="10.4" class="anchored" data-anchor-id="method-details"><span class="header-section-number">10.4</span> Method details</h3>
<section id="sample-prepneurovascular-unit-isolation" class="level4" data-number="10.4.1">
<h4 data-number="10.4.1" class="anchored" data-anchor-id="sample-prepneurovascular-unit-isolation"><span class="header-section-number">10.4.1</span> Sample prep/neurovascular unit isolation</h4>
</section>
<section id="rna-sequencing" class="level4" data-number="10.4.2">
<h4 data-number="10.4.2" class="anchored" data-anchor-id="rna-sequencing"><span class="header-section-number">10.4.2</span> RNA sequencing</h4>
</section>
</section>
<section id="quantification-and-statistical-analysis" class="level3" data-number="10.5">
<h3 data-number="10.5" class="anchored" data-anchor-id="quantification-and-statistical-analysis"><span class="header-section-number">10.5</span> Quantification and statistical analysis</h3>
<section id="data-processing" class="level4" data-number="10.5.1">
<h4 data-number="10.5.1" class="anchored" data-anchor-id="data-processing"><span class="header-section-number">10.5.1</span> Data Processing</h4>
<p>The raw sequencing data was processed using CellRanger (version 7.1.0, 10X Genomics), which performed initial alignment, filtering, barcode counting, and UMI counting. An updated reference genome (Ensembl version 109 and Gencode version 43 annotations for GRCh38) was generated for use with CellRanger. The resulting gene-barcode matrices were then further processed using the scFlow pipeline implemented in Nextflow.<span class="citation" data-cites="khozoie2021 ditommaso2017">(<a href="#ref-khozoie2021" role="doc-biblioref">Khozoie et al. 2021</a>; <a href="#ref-ditommaso2017" role="doc-biblioref">Di Tommaso et al. 2017</a>)</span></p>
</section>
<section id="quality-control" class="level4" data-number="10.5.2">
<h4 data-number="10.5.2" class="anchored" data-anchor-id="quality-control"><span class="header-section-number">10.5.2</span> Quality Control</h4>
<p>The processed data were read into R (version 4.4.0) using the Seurat package (version 5.1.0).<span class="citation" data-cites="base Seurat">(<a href="#ref-base" role="doc-biblioref"><strong>base?</strong></a>; <a href="#ref-Seurat" role="doc-biblioref"><strong>Seurat?</strong></a>)</span> Basic quality control (QC) measures were applied to ensure data integrity and reliability. Specifically, cells with low feature counts were filtered out, and donors with an insufficient number of high-quality cells were excluded from further analysis, resulting in 3 donors being excluded (both fractions). The criteria for low feature counts and donor exclusion were determined based on established thresholds in the literature and empirical distributions observed in the data.</p>
</section>
<section id="cell-type-annotation" class="level4" data-number="10.5.3">
<h4 data-number="10.5.3" class="anchored" data-anchor-id="cell-type-annotation"><span class="header-section-number">10.5.3</span> Cell Type Annotation</h4>
<p>Cell types were annotated based on canonical marker genes identified from the literature, <span class="citation" data-cites="yang2022">(<a href="#ref-yang2022" role="doc-biblioref"><strong>yang2022?</strong></a>)</span> in particular. The expression levels of these marker genes were used to classify cells into distinct cell types. This annotation process was validated by comparing the identified cell types to known cell type distributions in similar datasets.</p>
</section>
</section>
</section>
<section id="supplemental-information-titles-and-legends" class="level2" data-number="11">
<h2 data-number="11" class="anchored" data-anchor-id="supplemental-information-titles-and-legends"><span class="header-section-number">11</span> Supplemental information titles and legends</h2>
<div class="cell">
<div id="tbl-donor-metadata" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-donor-metadata-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Clinical Information. Clinical characteristics of the cohort and samples included in the multi-omics analysis.
</figcaption>
<div aria-describedby="tbl-donor-metadata-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div id="oqnvdexuob" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#oqnvdexuob table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#oqnvdexuob thead, #oqnvdexuob tbody, #oqnvdexuob tfoot, #oqnvdexuob tr, #oqnvdexuob td, #oqnvdexuob th {
  border-style: none;
}

#oqnvdexuob p {
  margin: 0;
  padding: 0;
}

#oqnvdexuob .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#oqnvdexuob .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#oqnvdexuob .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#oqnvdexuob .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#oqnvdexuob .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#oqnvdexuob .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oqnvdexuob .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#oqnvdexuob .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#oqnvdexuob .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#oqnvdexuob .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#oqnvdexuob .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#oqnvdexuob .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#oqnvdexuob .gt_spanner_row {
  border-bottom-style: hidden;
}

#oqnvdexuob .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#oqnvdexuob .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#oqnvdexuob .gt_from_md > :first-child {
  margin-top: 0;
}

#oqnvdexuob .gt_from_md > :last-child {
  margin-bottom: 0;
}

#oqnvdexuob .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#oqnvdexuob .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#oqnvdexuob .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#oqnvdexuob .gt_row_group_first td {
  border-top-width: 2px;
}

#oqnvdexuob .gt_row_group_first th {
  border-top-width: 2px;
}

#oqnvdexuob .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqnvdexuob .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#oqnvdexuob .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#oqnvdexuob .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oqnvdexuob .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqnvdexuob .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#oqnvdexuob .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#oqnvdexuob .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#oqnvdexuob .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#oqnvdexuob .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#oqnvdexuob .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqnvdexuob .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#oqnvdexuob .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#oqnvdexuob .gt_left {
  text-align: left;
}

#oqnvdexuob .gt_center {
  text-align: center;
}

#oqnvdexuob .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#oqnvdexuob .gt_font_normal {
  font-weight: normal;
}

#oqnvdexuob .gt_font_bold {
  font-weight: bold;
}

#oqnvdexuob .gt_font_italic {
  font-style: italic;
}

#oqnvdexuob .gt_super {
  font-size: 65%;
}

#oqnvdexuob .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#oqnvdexuob .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#oqnvdexuob .gt_indent_1 {
  text-indent: 5px;
}

#oqnvdexuob .gt_indent_2 {
  text-indent: 10px;
}

#oqnvdexuob .gt_indent_3 {
  text-indent: 15px;
}

#oqnvdexuob .gt_indent_4 {
  text-indent: 20px;
}

#oqnvdexuob .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_col_headings">
<th id="donor_id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">donor_id</th>
<th id="sex" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">sex</th>
<th id="age" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">age</th>
<th id="diagnosis" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">diagnosis</th>
<th id="braak_tangle" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">braak_tangle</th>
<th id="pH_cerebellum" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">pH_cerebellum</th>
<th id="ethnicity" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ethnicity</th>
<th id="brain_weight" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">brain_weight</th>
<th id="diagnosis_details" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">diagnosis_details</th>
<th id="apoe_status" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">apoe_status</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP011/2019</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">94</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.04</td>
<td class="gt_row gt_left" headers="ethnicity">White (European)</td>
<td class="gt_row gt_right" headers="brain_weight">1386</td>
<td class="gt_row gt_left" headers="diagnosis_details">Low level AD Braak II, Abnormal deposition of pTDP-43 in medial temporal lobe</td>
<td class="gt_row gt_right" headers="apoe_status">2/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP041/2019</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">87</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.57</td>
<td class="gt_row gt_left" headers="ethnicity">Black Caribbean</td>
<td class="gt_row gt_right" headers="brain_weight">1034</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, LATE</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP067/2018</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">81</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.43</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1172</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, LBD, CVD, Hippocampal sclerosis</td>
<td class="gt_row gt_right" headers="apoe_status">4/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP076/2014</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">95</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.94</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">934</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP032/2018</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">86</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">III</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.62</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1330</td>
<td class="gt_row gt_left" headers="diagnosis_details">Abnormal deposition of tau protein</td>
<td class="gt_row gt_right" headers="apoe_status">2/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP150/2014</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">95</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.17</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1317</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged brain</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP077/2018</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">82</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.90</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1400</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged control, low level AD change</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP090/17</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">70</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Aged brain Braak II</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP102/17</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">81</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged Brain Braak II</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP093/13</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">90</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Alzheimer's disease, Braak V, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP018/14</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">85</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak VI</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP406/16</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">81</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak VI</td>
<td class="gt_row gt_right" headers="apoe_status">2/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP096/13</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">92</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged brain Braak II</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP127/14</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">89</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">IV</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak IV</td>
<td class="gt_row gt_right" headers="apoe_status">4/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP080/15</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">71</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">I</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged brain Braak I</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP054/17</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">73</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak VI</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP109/17</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">77</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">I</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged brain Braak I</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP048/19</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">71</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">I</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Aged brain Braak I</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP011/14</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">93</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">III</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged brain . Braak III</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP160/13</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">88</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">IV</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak IV, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP024/17</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">74</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Control with low level AD Braak II</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP092/19</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">98</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">IV</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, LATE, CVD Braak IV</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP103/13</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">48</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">NA</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">Control</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP037/18</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">101</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, CVD Braak V</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP016/2017</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">89</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.67</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1366</td>
<td class="gt_row gt_left" headers="diagnosis_details">Aged brain, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP163/13</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">89</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">IV</td>
<td class="gt_row gt_right" headers="pH_cerebellum">NA</td>
<td class="gt_row gt_left" headers="ethnicity">NA</td>
<td class="gt_row gt_right" headers="brain_weight">NA</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak IV, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP034/2014</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">87</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.95</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1060</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD</td>
<td class="gt_row gt_right" headers="apoe_status">2/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP008/2015</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">87</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.86</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1101</td>
<td class="gt_row gt_left" headers="diagnosis_details">Normal aged brain</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP002/2019</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">88</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.63</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1151</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">4/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP070/2019</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">85</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.03</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1367</td>
<td class="gt_row gt_left" headers="diagnosis_details">Control with low level AD change Braak II</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP128/2017</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">87</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">I</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.10</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1368</td>
<td class="gt_row gt_left" headers="diagnosis_details">Control, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP080/2014</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">89</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.67</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1101</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, LBD, CVD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP018/2019</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">87</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.93</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1154</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak V</td>
<td class="gt_row gt_right" headers="apoe_status">3/4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP142/2014</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">85</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">VI</td>
<td class="gt_row gt_right" headers="pH_cerebellum">6.22</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1372</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD Braak VI</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP058/2019</td>
<td class="gt_row gt_left" headers="sex">M</td>
<td class="gt_row gt_right" headers="age">89</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.58</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1141</td>
<td class="gt_row gt_left" headers="diagnosis_details">Low level AD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="donor_id">NP177/2013</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">90</td>
<td class="gt_row gt_left" headers="diagnosis">Case</td>
<td class="gt_row gt_left" headers="braak_tangle">V</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.58</td>
<td class="gt_row gt_left" headers="ethnicity">unknown</td>
<td class="gt_row gt_right" headers="brain_weight">1153</td>
<td class="gt_row gt_left" headers="diagnosis_details">AD, LBD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="donor_id">NP063/2019</td>
<td class="gt_row gt_left" headers="sex">F</td>
<td class="gt_row gt_right" headers="age">89</td>
<td class="gt_row gt_left" headers="diagnosis">Control</td>
<td class="gt_row gt_left" headers="braak_tangle">II</td>
<td class="gt_row gt_right" headers="pH_cerebellum">5.69</td>
<td class="gt_row gt_left" headers="ethnicity">White (European)</td>
<td class="gt_row gt_right" headers="brain_weight">1280</td>
<td class="gt_row gt_left" headers="diagnosis_details">Aged brainLow level AD</td>
<td class="gt_row gt_right" headers="apoe_status">3/3</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</figure>
</div>
<a class="quarto-notebook-link" id="nblink-1" href="index-preview.html">Source: Article Notebook</a></div>
<div class="cell">
<div class="cell-output-display">
<div id="jrstpfdtax" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jrstpfdtax table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jrstpfdtax thead, #jrstpfdtax tbody, #jrstpfdtax tfoot, #jrstpfdtax tr, #jrstpfdtax td, #jrstpfdtax th {
  border-style: none;
}

#jrstpfdtax p {
  margin: 0;
  padding: 0;
}

#jrstpfdtax .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jrstpfdtax .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jrstpfdtax .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jrstpfdtax .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jrstpfdtax .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jrstpfdtax .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jrstpfdtax .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jrstpfdtax .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jrstpfdtax .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jrstpfdtax .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jrstpfdtax .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jrstpfdtax .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jrstpfdtax .gt_spanner_row {
  border-bottom-style: hidden;
}

#jrstpfdtax .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jrstpfdtax .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jrstpfdtax .gt_from_md > :first-child {
  margin-top: 0;
}

#jrstpfdtax .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jrstpfdtax .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jrstpfdtax .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jrstpfdtax .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jrstpfdtax .gt_row_group_first td {
  border-top-width: 2px;
}

#jrstpfdtax .gt_row_group_first th {
  border-top-width: 2px;
}

#jrstpfdtax .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jrstpfdtax .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jrstpfdtax .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jrstpfdtax .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jrstpfdtax .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jrstpfdtax .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jrstpfdtax .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jrstpfdtax .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jrstpfdtax .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jrstpfdtax .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jrstpfdtax .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jrstpfdtax .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jrstpfdtax .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jrstpfdtax .gt_left {
  text-align: left;
}

#jrstpfdtax .gt_center {
  text-align: center;
}

#jrstpfdtax .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jrstpfdtax .gt_font_normal {
  font-weight: normal;
}

#jrstpfdtax .gt_font_bold {
  font-weight: bold;
}

#jrstpfdtax .gt_font_italic {
  font-style: italic;
}

#jrstpfdtax .gt_super {
  font-size: 65%;
}

#jrstpfdtax .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jrstpfdtax .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jrstpfdtax .gt_indent_1 {
  text-indent: 5px;
}

#jrstpfdtax .gt_indent_2 {
  text-indent: 10px;
}

#jrstpfdtax .gt_indent_3 {
  text-indent: 15px;
}

#jrstpfdtax .gt_indent_4 {
  text-indent: 20px;
}

#jrstpfdtax .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<caption>R packages used</caption>
<thead>
<tr class="header gt_col_headings">
<th id="package" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">package</th>
<th id="loadedversion" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">loadedversion</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Bioconductor 3.19 (R 4.4.0)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Bioconductor 3.19 (R 4.4.0)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">AnnotationDbi</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.66.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">AnnotationFilter</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.28.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">AnnotationHub</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">3.12.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BSgenome</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.72.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">Biobase</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.64.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocFileCache</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.12.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocGenerics</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">0.50.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocIO</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.14.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocNeighbors</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.22.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocParallel</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.38.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocSingular</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.20.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">BiocVersion</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">3.19.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ComplexHeatmap</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.20.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">DESeq2</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.44.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">DelayedArray</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">0.30.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">DelayedMatrixStats</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.26.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">DropletUtils</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.24.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">EWCE</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.12.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ExperimentHub</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.12.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">GEOquery</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.72.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">GOSemSim</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.30.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">GenomicAlignments</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.40.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">GenomicFeatures</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.56.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">GenomicFiles</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.40.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">HDF5Array</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.32.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">KEGGgraph</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.64.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">MAST</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.30.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">MatrixGenerics</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.16.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">MungeSumstats</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.12.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">PCAtools</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.16.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ProtGenerics</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.36.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">Rgraphviz</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.48.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">Rhdf5lib</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.26.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">Rsamtools</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.20.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">SC3</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.32.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ScaledMatrix</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.12.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">SingleCellExperiment</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.26.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">SummarizedExperiment</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.34.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">UCSC.utils</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.0.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">VariantAnnotation</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.50.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">XVector</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">0.44.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">alabaster.sce</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.4.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">alabaster.schemas</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.4.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">batchelor</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.20.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">beachmat</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.20.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">bluster</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.14.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">cellity</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.32.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">clusterProfiler</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">4.12.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">edgeR</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">4.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">enrichplot</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.24.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ensembldb</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.28.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ewceData</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.12.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">fgsea</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.30.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">gage</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.54.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">gageData</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.42.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">ggtree</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">3.12.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">graph</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.82.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">gypsum</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.0.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">metapod</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.12.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">orthogene</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.10.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">pathview</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.44.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">preprocessCore</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.66.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">qusage</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.38.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">qvalue</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.36.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">rhdf5</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.48.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">rhdf5filters</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.16.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">rrvgo</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.16.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">rtracklayer</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.64.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">scRNAseq</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.18.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">scater</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.32.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">scran</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.32.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">scuttle</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.14.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">simona</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">sparseMatrixStats</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.16.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">speckle</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.4.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">topGO</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">2.56.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">treeio</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.28.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.0)  package">zlibbioc</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.0)  loadedversion">1.50.0</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="Bioconductor" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Bioconductor</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="Bioconductor  package">BSgenome.Hsapiens.1000genomes.hs37d5</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">0.99.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor  package">BSgenome.Hsapiens.NCBI.GRCh38</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">1.3.1000</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor  package">EnsDb.Hsapiens.v79</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">2.99.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor  package">GO.db</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">3.19.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor  package">GenomeInfoDbData</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">1.2.12</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor  package">HDO.db</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">0.99.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor  package">SNPlocs.Hsapiens.dbSNP155.GRCh37</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">0.99.24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor  package">SNPlocs.Hsapiens.dbSNP155.GRCh38</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">0.99.24</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor  package">org.Hs.eg.db</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">3.19.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor  package">org.Mm.eg.db</td>
<td class="gt_row gt_right" headers="Bioconductor  loadedversion">3.19.1</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="RSPM (R 4.4.0)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">RSPM (R 4.4.0)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">BiocManager</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.30.23</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">DirichletReg</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.7-1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">Formula</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.2-5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">HGNChelper</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.8.14</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">NMF</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.27</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">RApiSerialize</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.1.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">RNOmni</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.0.1.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">RSQLite</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.3.7</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">Seurat</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">5.1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">SeuratObject</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">5.0.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">UpSetR</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.4.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">WebGestaltR</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.4.6</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">apcluster</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.4.13</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">babelgene</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">22.9</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">backports</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.5.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">base64url</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">broom</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.0.6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">coda</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.19-4.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">dqrng</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.4.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">english</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.2-6</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">enrichR</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">3.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">estimability</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.5.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">fastmap</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">formattable</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">ggdendro</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">ggfun</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.1.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">ggnetwork</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.5.13</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">googleAuthR</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.0.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">gprofiler2</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.2.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">grr</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.9.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">homologene</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.4.68.19.3.27</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">humaniformat</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.6.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">leaflet</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.2.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">maxLik</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.5-2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">minqa</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.2.7</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">miscTools</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.6-28</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">mvtnorm</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.2-5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">network</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.18.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">openssl</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">paletteer</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.6.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">rliger</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.0.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">rmarkdown</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.27</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">sandwich</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">3.1-0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">sessioninfo</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.2.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">sna</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">2.7-2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">splitstackshape</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.4.8</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">statnet.common</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">4.9.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">stringfish</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.16.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">stringi</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.8.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">systemfonts</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">1.1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">threejs</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.3.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">uwot</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.2.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.0)  package">whisker</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.0)  loadedversion">0.4.1</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Bioconductor 3.19 (R 4.4.1)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Bioconductor 3.19 (R 4.4.1)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">Biostrings</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">2.72.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">DOSE</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">3.30.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">GenomeInfoDb</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.40.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">GenomicRanges</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.56.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">IRanges</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">2.38.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">KEGGREST</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.44.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">ResidualMatrix</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.14.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">S4Arrays</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.4.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">S4Vectors</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">0.42.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">SparseArray</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.4.8</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">alabaster.base</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.4.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">alabaster.matrix</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.4.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">alabaster.ranges</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.4.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">alabaster.se</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">1.4.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">biomaRt</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">2.60.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Bioconductor 3.19 (R 4.4.1)  package">limma</td>
<td class="gt_row gt_right" headers="Bioconductor 3.19 (R 4.4.1)  loadedversion">3.60.3</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="Github (jinworks/CellChat@862ab34277942ca26dad41ed2d168fcbdf952950)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (jinworks/CellChat@862ab34277942ca26dad41ed2d168fcbdf952950)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="Github (jinworks/CellChat@862ab34277942ca26dad41ed2d168fcbdf952950)  package">CellChat</td>
<td class="gt_row gt_right" headers="Github (jinworks/CellChat@862ab34277942ca26dad41ed2d168fcbdf952950)  loadedversion">2.1.2</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="RSPM (R 4.4.1)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">RSPM (R 4.4.1)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">DBI</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.2.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">RcppParallel</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">5.1.8</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">SparseM</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.84</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">XML</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">3.99-0.17</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">aplot</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.2.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">cli</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">3.6.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">crayon</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.5.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">digest</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.6.36</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">downlit</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.4.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">emmeans</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.10.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">evaluate</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.24.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">ggsci</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">3.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">knitr</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.48</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">lme4</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.1-35.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">locfit</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.5-9.10</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">nloptr</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">2.1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">ps</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.7.7</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">reticulate</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.38.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">rlang</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.1.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">robustbase</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.99-3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">scatterpie</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.2.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">secretbase</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.0.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">spatstat.data</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">3.1-2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">spatstat.sparse</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">3.1-0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">spatstat.utils</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">3.0-5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">targets</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">1.7.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">xfun</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">0.45</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="RSPM (R 4.4.1)  package">yaml</td>
<td class="gt_row gt_right" headers="RSPM (R 4.4.1)  loadedversion">2.3.9</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="CRAN (R 4.4.0)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">CRAN (R 4.4.0)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">DEoptimR</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1-3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">DT</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.33</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">FNN</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">GetoptLong</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">GlobalOptions</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">KernSmooth</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.23-24</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">NLP</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.2-1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">Polychrome</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">R.methodsS3</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">R.oo</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.26.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">R.utils</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.12.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">R6</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.5.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">RANN</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.6.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">RColorBrewer</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1-3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">RCurl</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.98-1.14</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ROCR</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0-11</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">RSpectra</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.16-1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">RcppAnnoy</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.0.22</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">RcppHNSW</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.6.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">Rtsne</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.17</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">SnowballC</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.7.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">WriteXLS</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">6.6.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">abind</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.4-5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ape</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">5.8</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">askpass</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">assertthat</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">base64enc</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1-3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">beeswarm</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">bit</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">4.0.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">bit64</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">4.0.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">bitops</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0-7</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">blob</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">bslib</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.7.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">cachem</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">callr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.7.6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">car</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.1-2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">carData</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.0-5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">cellranger</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">circlize</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.16</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">clue</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3-65</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">colorspace</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1-0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">corrgram</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.14</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">cowplot</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">crosstalk</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">curl</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">5.2.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">data.table</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.15.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">dbplyr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.5.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">deldir</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0-4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">dichromat</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0-0.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">doParallel</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.17</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">doRNG</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8.6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">dotCall64</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1-1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">dplyr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">e1071</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.7-14</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">europepmc</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">fansi</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.6</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">farver</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">fastDummies</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.7.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">fastmatch</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1-4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">fftw</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0-8</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">filelock</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">fitdistrplus</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1-11</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">forcats</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">foreach</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">fs</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.6.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">furrr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">future</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.33.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">future.apply</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.11.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gargle</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">generics</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggalluvial</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.12.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggbeeswarm</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.7.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggforce</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggplot2</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.5.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggplotify</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggpubr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.6.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggraph</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.2.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggrepel</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.9.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggridges</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.5.6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggsignif</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.6.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ggvenn</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">globals</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.16.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">glue</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.7.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">goftest</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2-3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">graphlayouts</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gridBase</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4-7</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gridExtra</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gridGraphics</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.5-1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gson</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gt</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.10.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">gtable</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">haven</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.5.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">here</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">hms</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.1.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">htmltools</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.5.8.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">htmlwidgets</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.6.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">httpuv</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.6.15</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">httr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.4.7</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">httr2</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">ica</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0-3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">igraph</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">irlba</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.3.5.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">iterators</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.14</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">janeaustenr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">janitor</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">jquerylib</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">jsonlite</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8.8</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">kableExtra</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.4.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">labelled</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.13.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">later</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">lazyeval</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.2.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">leiden</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.3.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">lifecycle</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">listenv</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.9.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">lmtest</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.9-40</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">lubridate</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.9.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">magrittr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">mapproj</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.11</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">maps</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.4.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">matrixStats</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">memoise</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">mime</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.12</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">miniUI</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">munsell</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.5.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">mvoutlier</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">nlme</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.1-165</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">pals</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">parallelly</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.37.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">patchwork</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">pbapply</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.7-2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">pcaPP</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0-4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">pheatmap</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.12</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">pillar</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.9.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">pkgconfig</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">plyr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8.9</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">png</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1-8</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">polyclip</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.10-6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">prettyunits</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">processx</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.8.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">progress</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">progressr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.14.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">promises</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">proxy</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4-27</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">purrr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">qs</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.26.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rappdirs</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">readr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">readxl</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.4.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">registry</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.5-1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rematch2</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">remotes</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.5.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">reshape</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.8.9</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">reshape2</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.4.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">restfulr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.0.15</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rjson</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.2.21</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rngtools</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rprojroot</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rrcov</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.7-5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rstatix</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.7.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rstudioapi</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.16.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">rsvd</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">sass</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.9</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">scales</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">scattermore</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">scatterplot3d</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3-44</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">sctransform</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">sgeostat</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.0-27</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">shadowtext</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">shape</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.4.6.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">shiny</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8.1.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">slam</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1-50</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">snakecase</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.11.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">sp</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1-4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">spam</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.10-0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">spatstat.explore</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.2-7</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">spatstat.geom</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.2-9</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">spatstat.random</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.2-3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">statmod</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">stringr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">survival</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.7-0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">svglite</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.1.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tensor</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tibble</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.2.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tidygraph</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tidyr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tidyselect</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tidytext</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tidytree</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.6</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">timechange</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tm</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.7-13</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tokenizers</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">treemap</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.4-4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">triebeard</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.1</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tweenr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.0.3</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">tzdb</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">umap</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.2.10.0</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">urltools</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.7.3</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">utf8</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.2.4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">vctrs</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.6.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">vipor</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.7</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">viridis</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.6.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">viridisLite</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.4.2</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">vroom</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.6.5</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">wesanderson</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.3.7</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">withr</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">3.0.0</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">wordcloud</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">2.6</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">xml2</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.3.6</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">xtable</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8-4</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">yulab.utils</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">0.1.4</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.4.0)  package">zoo</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.0)  loadedversion">1.8-12</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="Github (chris-mcginnis-ucsf/DoubletFinder@03e9f37f891ef76a23cc55ea69f940c536ae8f9f)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (chris-mcginnis-ucsf/DoubletFinder@03e9f37f891ef76a23cc55ea69f940c536ae8f9f)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="Github (chris-mcginnis-ucsf/DoubletFinder@03e9f37f891ef76a23cc55ea69f940c536ae8f9f)  package">DoubletFinder</td>
<td class="gt_row gt_right" headers="Github (chris-mcginnis-ucsf/DoubletFinder@03e9f37f891ef76a23cc55ea69f940c536ae8f9f)  loadedversion">2.0.4</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="CRAN (R 4.4.1)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">CRAN (R 4.4.1)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="CRAN (R 4.4.1)  package">MASS</td>
<td class="gt_row gt_right" headers="CRAN (R 4.4.1)  loadedversion">7.3-61</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="CRAN (R 4.3.3)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">CRAN (R 4.3.3)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="CRAN (R 4.3.3)  package">Matrix</td>
<td class="gt_row gt_right" headers="CRAN (R 4.3.3)  loadedversion">1.6-5</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.3.3)  package">boot</td>
<td class="gt_row gt_right" headers="CRAN (R 4.3.3)  loadedversion">1.3-30</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="CRAN (R 4.3.2)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">CRAN (R 4.3.2)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="CRAN (R 4.3.2)  package">Rcpp</td>
<td class="gt_row gt_right" headers="CRAN (R 4.3.2)  loadedversion">1.0.12</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="CRAN (R 4.3.2)  package">cluster</td>
<td class="gt_row gt_right" headers="CRAN (R 4.3.2)  loadedversion">2.1.6</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="Github (satijalab/seurat-wrappers@8d46d6c47c089e193fe5c02a8c23970715918aa9)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (satijalab/seurat-wrappers@8d46d6c47c089e193fe5c02a8c23970715918aa9)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="Github (satijalab/seurat-wrappers@8d46d6c47c089e193fe5c02a8c23970715918aa9)  package">SeuratWrappers</td>
<td class="gt_row gt_right" headers="Github (satijalab/seurat-wrappers@8d46d6c47c089e193fe5c02a8c23970715918aa9)  loadedversion">0.3.5</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="Github (ropensci/bib2df@de0838da561544361d9be6ff20192d0d1c794cde)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (ropensci/bib2df@de0838da561544361d9be6ff20192d0d1c794cde)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="Github (ropensci/bib2df@de0838da561544361d9be6ff20192d0d1c794cde)  package">bib2df</td>
<td class="gt_row gt_right" headers="Github (ropensci/bib2df@de0838da561544361d9be6ff20192d0d1c794cde)  loadedversion">1.1.2.0</td>
</tr>
<tr class="even gt_group_heading_row">
<td colspan="2" id="CRAN (R 4.3.1)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">CRAN (R 4.3.1)</td>
</tr>
<tr class="odd gt_row_group_first">
<td class="gt_row gt_left" headers="CRAN (R 4.3.1)  package">class</td>
<td class="gt_row gt_right" headers="CRAN (R 4.3.1)  loadedversion">7.3-22</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="CRAN (R 4.3.1)  package">lattice</td>
<td class="gt_row gt_right" headers="CRAN (R 4.3.1)  loadedversion">0.22-5</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="CRAN (R 4.2.2)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">CRAN (R 4.2.2)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="CRAN (R 4.2.2)  package">codetools</td>
<td class="gt_row gt_right" headers="CRAN (R 4.2.2)  loadedversion">0.2-19</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Github (theislab/kBET@afc5f431bcbefd73267acc066a0f2e4eaa10a355)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (theislab/kBET@afc5f431bcbefd73267acc066a0f2e4eaa10a355)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Github (theislab/kBET@afc5f431bcbefd73267acc066a0f2e4eaa10a355)  package">kBET</td>
<td class="gt_row gt_right" headers="Github (theislab/kBET@afc5f431bcbefd73267acc066a0f2e4eaa10a355)  loadedversion">0.99.6</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Github (cole-trapnell-lab/monocle3@98402ed0c10cac020524bebbb9300614a799f6d1)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (cole-trapnell-lab/monocle3@98402ed0c10cac020524bebbb9300614a799f6d1)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Github (cole-trapnell-lab/monocle3@98402ed0c10cac020524bebbb9300614a799f6d1)  package">monocle3</td>
<td class="gt_row gt_right" headers="Github (cole-trapnell-lab/monocle3@98402ed0c10cac020524bebbb9300614a799f6d1)  loadedversion">1.3.7</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Github (ropensci/plotly@963b9143ba4102eca8edbd4750d4c85541ddd0f5)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (ropensci/plotly@963b9143ba4102eca8edbd4750d4c85541ddd0f5)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Github (ropensci/plotly@963b9143ba4102eca8edbd4750d4c85541ddd0f5)  package">plotly</td>
<td class="gt_row gt_right" headers="Github (ropensci/plotly@963b9143ba4102eca8edbd4750d4c85541ddd0f5)  loadedversion">4.10.4.9000</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Github (immunogenomics/presto@7636b3d0465c468c35853f82f1717d3a64b3c8f6)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (immunogenomics/presto@7636b3d0465c468c35853f82f1717d3a64b3c8f6)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Github (immunogenomics/presto@7636b3d0465c468c35853f82f1717d3a64b3c8f6)  package">presto</td>
<td class="gt_row gt_right" headers="Github (immunogenomics/presto@7636b3d0465c468c35853f82f1717d3a64b3c8f6)  loadedversion">1.0.0</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Github (neurogenomics/scFlow@065bc13dacee707a0437fb65e006647660fc13c7)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (neurogenomics/scFlow@065bc13dacee707a0437fb65e006647660fc13c7)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Github (neurogenomics/scFlow@065bc13dacee707a0437fb65e006647660fc13c7)  package">scFlow</td>
<td class="gt_row gt_right" headers="Github (neurogenomics/scFlow@065bc13dacee707a0437fb65e006647660fc13c7)  loadedversion">0.7.4</td>
</tr>
<tr class="odd gt_group_heading_row">
<td colspan="2" id="Github (rpolicastro/scProportionTest@37a04900be1c991428da15af7f9aa7b0ad84661e)" class="gt_group_heading" data-quarto-table-cell-role="th" scope="colgroup">Github (rpolicastro/scProportionTest@37a04900be1c991428da15af7f9aa7b0ad84661e)</td>
</tr>
<tr class="even gt_row_group_first">
<td class="gt_row gt_left" headers="Github (rpolicastro/scProportionTest@37a04900be1c991428da15af7f9aa7b0ad84661e)  package">scProportionTest</td>
<td class="gt_row gt_right" headers="Github (rpolicastro/scProportionTest@37a04900be1c991428da15af7f9aa7b0ad84661e)  loadedversion">0.0.0.9000</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="references" class="level2 unnumbered" data-number="12">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">12 References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-bellenguez2022" class="csl-entry" role="listitem">
Bellenguez, Céline, Fahri Küçükali, Iris E. Jansen, Luca Kleineidam, Sonia Moreno-Grau, Najaf Amin, Adam C. Naj, et al. 2022. <span>“New Insights into the Genetic Etiology of Alzheimer<span>’</span>s Disease and Related Dementias.”</span> <em>Nature Genetics</em> 54 (4): 412–36. <a href="https://doi.org/10.1038/s41588-022-01024-z">https://doi.org/10.1038/s41588-022-01024-z</a>.
</div>
<div id="ref-ditommaso2017" class="csl-entry" role="listitem">
Di Tommaso, Paolo, Maria Chatzou, Evan W. Floden, Pablo Prieto Barja, Emilio Palumbo, and Cedric Notredame. 2017. <span>“Nextflow Enables Reproducible Computational Workflows.”</span> <em>Nature Biotechnology</em> 35 (4): 316–19. <a href="https://doi.org/10.1038/nbt.3820">https://doi.org/10.1038/nbt.3820</a>.
</div>
<div id="ref-khozoie2021" class="csl-entry" role="listitem">
Khozoie, Combiz, Nurun Fancy, Mahdi Marjaneh, Alan Murphy, Paul Matthews, and Nathan Skene. 2021. <span>“scFlow: A Scalable and Reproducible Analysis Pipeline for Single-Cell RNA Sequencing Dat.”</span> <a href="https://doi.org/10.22541/au.162912533.38489960/v2">https://doi.org/10.22541/au.162912533.38489960/v2</a>.
</div>
<div id="ref-kunkle2019" class="csl-entry" role="listitem">
Kunkle, Brian W., Benjamin Grenier-Boley, Rebecca Sims, Joshua C. Bis, Vincent Damotte, Adam C. Naj, Anne Boland, et al. 2019. <span>“Genetic Meta-Analysis of Diagnosed Alzheimer<span>’</span>s Disease Identifies New Risk Loci and Implicates Aβ, Tau, Immunity and Lipid Processing.”</span> <em>Nature Genetics</em> 51 (3): 414–30. <a href="https://doi.org/10.1038/s41588-019-0358-2">https://doi.org/10.1038/s41588-019-0358-2</a>.
</div>
</div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{mateus_bernardo_harrington2024,
  author = {Mateus Bernardo Harrington, Gabriel and Monzon Sandoval,
    Jimena and Rokicki, Michal and Cader, Zameel and Webber, Caleb},
  title = {BBB in {AD}},
  date = {2024-07-17},
  langid = {en},
  abstract = {The blood brain barrier, it’s important, probably in AD
    even}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-mateus_bernardo_harrington2024" class="csl-entry quarto-appendix-citeas" role="listitem">
Mateus Bernardo Harrington, Gabriel, Jimena Monzon Sandoval, Michal
Rokicki, Zameel Cader, and Caleb Webber. 2024. <span>“BBB in AD.”</span>
Cell. July 17, 2024.
</div></div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"selector":".lightbox","descPosition":"bottom","openEffect":"zoom","loop":false,"closeEffect":"zoom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>